# Novel metabolic drivers of virulence in *Clostridioides difficile* identified through genome-scale metabolic network analysis

### Abstract

The bacterial pathogen *Clostridioides difficile* causes a toxin-mediated diarrheal illness and is now the leading cause of hospital-acquired infection in the US. Due to growing threats of antibiotic resistance and recurrent infection, targeting components of *C. difficile* metabolism presents a novel approach to combat this infection. Bacterial genome-scale metabolic network reconstructions (GENREs) have been used to identify new therapeutic targets and uncover properties that drive cellular behaviors. With this approach we constructed highly-curated *C. difficile* GENREs for a hyper-virulent isolate (str. R20291) as well as a historic strain (str. 630), and validated these models with both *in vitro* and *in vivo* datasets. Growth simulations of carbon source usage revealed significant correlations with experimentally measured values (*p*-values < 0.001, PPV â‰¥ 92.7%), and single-gene deletion analysis showed accuracies of >89.0% compared with transposon mutant libraries. Next, we utilized each GENRE to identify potential metabolic drivers of multiple virulence-related phenotypes including sporulation and biofilm formation. Through contextualization of these models using transcriptomic data we generated from *in vitro* and infection conditions, we discovered both greater reliance on the Pentose Phosphate Pathway as well as significantly increased usage of cytidine and N-acetylneuraminate when virulence expression is reduced. These novel relationships were subsequently validated experimentally. Collectively, our results highlight the ability of GENREs to identify novel metabolite signals in bacterial pathogenesis.


### Overview

	project
	|- README          		# description of content
	|- LICENSE         		# the license for this project
	|
	|- doc/					# additional documents associated with the study
	|
	|- data/          		# raw and primary data
	|
	|- code/				# all programmatic code (python & R)
	|
	|- results/				# all output from workflows and analyses
	| |- figures/			# manuscript figures
	| +- tables/			# supplementary tables
	|
	|- notebooks/			# jupyter notebooks for the analyses performed during this study

