{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "import copy\n",
    "\n",
    "# Quicker way to read in models\n",
    "import pickle\n",
    "def read_model(fileName, obj='none'):\n",
    "    \n",
    "    fileType = fileName.split('.')[-1]\n",
    "    \n",
    "    if fileType == 'sbml' or fileType == 'xml':\n",
    "        model = cobra.io.read_sbml_model(fileName)\n",
    "    elif fileType == 'json':\n",
    "        model = cobra.io.load_json_model(fileName)\n",
    "    elif fileType == 'yaml':\n",
    "        model = cobra.io.load_yaml_model(fileName)\n",
    "    elif fileType == 'mat':\n",
    "        model = cobra.io.load_matlab_model(fileName)\n",
    "    elif fileType == 'pkl':\n",
    "        model = pickle.load(open(fileName, 'rb'))\n",
    "    else:\n",
    "        raise TypeError('Unrecognized file extension')\n",
    "    \n",
    "    if obj != 'none': model.objective = obj\n",
    "    for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "    return model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# curated Clostridium difficile 630 model\n",
    "iCdG719 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG719.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG719</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fa785b35fd0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1216</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1277</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG719 at 0x7fa785b35fd0>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG719"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "719"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(iCdG719.genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "190.16070481163646"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG719.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "patric password: ········\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'mljenior@patricbrc.org'"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create R20291 draft GENRE\n",
    "\n",
    "import mackinac\n",
    "mackinac.modelseed.ms_client.url = 'http://p3.theseed.org/services/ProbModelSEED/'\n",
    "mackinac.workspace.ws_client.url = 'http://p3.theseed.org/services/Workspace'\n",
    "mackinac.genome.patric_url = 'https://www.patricbrc.org/api/'\n",
    "\n",
    "# PATRIC user information\n",
    "mackinac.get_token('mljenior')\n",
    "# password: matrix54"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_id = '645463.3'\n",
    "template_id = '/chenry/public/modelsupport/templates/GramPosModelTemplate'\n",
    "media_id = '/chenry/public/modelsupport/media/Complete'\n",
    "file_id = '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/CdR20291.draft.json'\n",
    "strain_id = 'Clostridioides difficile R20291'\n",
    "\n",
    "mackinac.reconstruct_modelseed_model(genome_id, template_reference=template_id)\n",
    "mackinac.gapfill_modelseed_model(genome_id, media_reference=media_id)\n",
    "mackinac.optimize_modelseed_model(genome_id)\n",
    "patric_draft = mackinac.create_cobra_model_from_modelseed_model(genome_id)\n",
    "patric_draft.id = strain_id\n",
    "\n",
    "cobra.io.save_json_model(patric_draft, file_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get model gene IDs\n",
    "iCdG719_genes = set([x.id for x in iCdG719.genes])\n",
    "patric_draft = cobra.io.load_json_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/CdR20291.draft.json')\n",
    "draft_R20291_genes = set([x.id for x in patric_draft.genes])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove free biomass components\n",
    "patric_draft.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "patric_draft.reactions.rxn13784_c.remove_from_model(remove_orphans=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in protein alignments\n",
    "\n",
    "# 630 vs R20291\n",
    "Cd630_geneDict = {}\n",
    "R20291_missing = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/proteome_comparisons/Cd630_R20291.tsv', 'r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        Cd630_gene = line.split()[0]\n",
    "        R20291_gene = line.split()[1]\n",
    "        \n",
    "        if not Cd630_gene in iCdG719_genes:\n",
    "            continue\n",
    "        elif R20291_gene == 'none':\n",
    "            R20291_missing.append(Cd630_gene)\n",
    "            continue\n",
    "        else:\n",
    "            Cd630_geneDict[Cd630_gene] = R20291_gene   \n",
    "\n",
    "# R20291 vs 630\n",
    "R20291_geneDict = {}\n",
    "Cd630_missing = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/proteome_comparisons/R20291_Cd630.tsv', 'r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        R20291_gene = line.split()[0]\n",
    "        Cd630_gene = line.split()[1]\n",
    "        \n",
    "        if Cd630_gene == 'none':\n",
    "            Cd630_missing.append(R20291_gene)\n",
    "            continue\n",
    "        else:\n",
    "            R20291_geneDict[R20291_gene] = Cd630_gene  \n",
    "            \n",
    "# Peptoclostridium 630 vs R20291\n",
    "Pepto630_geneDict = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/proteome_comparisons/Pepto630_R20291.tsv', 'r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        Pepto630_gene = line.split()[0]\n",
    "        R20291_gene = line.split()[1]\n",
    "        \n",
    "        if not Pepto630_gene in iCdG719_genes:\n",
    "            continue\n",
    "        elif R20291_gene == 'none':\n",
    "            R20291_missing.append(Pepto630_gene)\n",
    "            continue\n",
    "        else:\n",
    "            Pepto630_geneDict[Pepto630_gene] = R20291_gene   \n",
    "\n",
    "# R20291 vs Peptoclostridium 630\n",
    "R20291_geneDict = {}\n",
    "Pepto630_missing = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/proteome_comparisons/R20291_Pepto630.tsv', 'r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        R20291_gene = line.split()[0]\n",
    "        Pepto630_gene = line.split()[1]\n",
    "        \n",
    "        if Cd630_gene == 'none':\n",
    "            Pepto630_missing.append(R20291_gene)\n",
    "            continue\n",
    "        else:\n",
    "            R20291_geneDict[R20291_gene] = Pepto630_gene  \n",
    "\n",
    "# Create complete translation dictionary\n",
    "geneDict = {}\n",
    "geneDict.update(Cd630_geneDict)\n",
    "geneDict.update(Pepto630_geneDict)\n",
    "\n",
    "# Get gene objects to be removed\n",
    "remove_genes = [iCdG719.genes.get_by_id(x) for x in set(R20291_missing)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create duplicate GENRE\n",
    "draft_GENRE = copy.deepcopy(iCdG719)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Delete genes that are not present in R20291\n",
    "remove_genes = [draft_GENRE.genes.get_by_id(x) for x in set(R20291_missing)]\n",
    "cobra.manipulation.delete.remove_genes(draft_GENRE, remove_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename homologous genes\n",
    "cobra.manipulation.modify.rename_genes(draft_GENRE, geneDict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions added: 4\n"
     ]
    }
   ],
   "source": [
    "# Add genes that only appear in R20291\n",
    "included_genes = list(set(draft_R20291_genes).intersection(set(Cd630_missing)))\n",
    "current_rxn_ids = set([x.id for x in draft_GENRE.reactions])\n",
    "rxns_added = 0\n",
    "add_manually = []\n",
    "for gene in patric_draft.genes:\n",
    "    if not gene.id in included_genes: \n",
    "        add_manually.append(gene.id)\n",
    "        continue\n",
    "    \n",
    "    for rxn in gene.reactions:\n",
    "        if rxn.id in current_rxn_ids:\n",
    "            # update existing gene_reaction_rule\n",
    "            current_gpr = draft_GENRE.reactions.get_by_id(rxn.id).gene_reaction_rule\n",
    "            if current_gpr != '':\n",
    "                current_gpr += ' or ' + gene.id\n",
    "            else:\n",
    "                current_gpr = gene.id\n",
    "            draft_GENRE.reactions.get_by_id(rxn.id).gene_reaction_rule = current_gpr\n",
    "            # update gene name\n",
    "            draft_GENRE.genes.get_by_id(gene.id).name = patric_draft.genes.get_by_id(gene.id).name \n",
    "        else:\n",
    "            draft_GENRE.add_reactions([rxn])\n",
    "            rxns_added += 1\n",
    "            current_gpr = draft_GENRE.reactions.get_by_id(rxn.id).gene_reaction_rule\n",
    "            draft_GENRE.reactions.get_by_id(rxn.id).gene_reaction_rule = current_gpr\n",
    "            \n",
    "print('Reactions added: ' + str(rxns_added))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn10344_c' since it already exists.\n",
      "Ignoring reaction 'rxn05516_c' since it already exists.\n",
      "Ignoring reaction 'rxn10343_c' since it already exists.\n",
      "Ignoring reaction 'rxn10481_c' since it already exists.\n",
      "Ignoring reaction 'rxn05528_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn10344_c' since it already exists.\n",
      "Ignoring reaction 'rxn05516_c' since it already exists.\n",
      "Ignoring reaction 'rxn10343_c' since it already exists.\n",
      "Ignoring reaction 'rxn10481_c' since it already exists.\n",
      "Ignoring reaction 'rxn05528_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn10481_c' since it already exists.\n",
      "Ignoring reaction 'rxn05528_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n",
      "Ignoring reaction 'rxn13783_c' since it already exists.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions added: 46\n"
     ]
    }
   ],
   "source": [
    "# Identify genes that must be added from the draft reconstruction\n",
    "draft_genes = set([gene.id for gene in draft_GENRE.genes])\n",
    "add_manually = set(add_manually).difference(draft_genes)\n",
    "current_rxn_ids = set([x.id for x in draft_GENRE.reactions])\n",
    "\n",
    "for gene in add_manually:\n",
    "    for rxn in patric_draft.genes.get_by_id(gene).reactions:\n",
    "        if rxn.id in current_rxn_ids:\n",
    "            # update existing gene_reaction_rule\n",
    "            current_gpr = draft_GENRE.reactions.get_by_id(rxn.id).gene_reaction_rule\n",
    "            if current_gpr != '':\n",
    "                current_gpr += ' or ' + gene\n",
    "            else:\n",
    "                current_gpr = gene\n",
    "            draft_GENRE.reactions.get_by_id(rxn.id).gene_reaction_rule = current_gpr\n",
    "            # update gene name\n",
    "            draft_GENRE.genes.get_by_id(gene).name = patric_draft.genes.get_by_id(gene).name \n",
    "        else:\n",
    "            draft_GENRE.add_reactions([rxn])\n",
    "            draft_GENRE.genes.get_by_id(gene).name = patric_draft.genes.get_by_id(gene).name \n",
    "            rxns_added += 1\n",
    "\n",
    "print('Reactions added: ' + str(rxns_added))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix a few remaining incorrect GPRs \n",
    "draft_GENRE.reactions.rxn02003_c.gene_reaction_rule = '645463.3.peg.237'\n",
    "draft_GENRE.reactions.rxn14293_c.gene_reaction_rule += ' or 645463.3.peg.260'\n",
    "draft_GENRE.reactions.rxn06517_c.gene_reaction_rule += ' or 645463.3.peg.1978'\n",
    "draft_GENRE.reactions.rxn01404_c.gene_reaction_rule += ' or 645463.3.peg.658'\n",
    "draft_GENRE.reactions.rxn01368_c.gene_reaction_rule += ' or 645463.3.peg.769'\n",
    "draft_GENRE.reactions.rxn00119_c.gene_reaction_rule += ' or 645463.3.peg.2191'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace removed Biomass component\n",
    "cpd02967_c = draft_GENRE.metabolites.get_by_id('cpd02967_c') # N-Acetyl-beta-D-mannosaminyl-1,4-N-acetyl-D-glucosaminyldiphosphoundecaprenol\n",
    "cpd00402_c = draft_GENRE.metabolites.get_by_id('cpd00402_c') # CDPglycerol\n",
    "cpd00046_c = draft_GENRE.metabolites.get_by_id('cpd00046_c') # CMP\n",
    "\n",
    "cpd12894_c = cobra.Metabolite(\n",
    "    'cpd12894_c',\n",
    "    formula='',\n",
    "    name='Teichoic acid',\n",
    "    compartment='cytosol')\n",
    "\n",
    "# R20291 homolog unknown - Stabler et al. (2009). Genome Biol.\n",
    "#gene = cobra.Gene('272563.8.peg.262')\n",
    "#gene.name = 'Putative CDP-glycerol:Poly(Glycerophosphate) glycerophosphotransferase'\n",
    "\n",
    "teichoicacid_rxn = cobra.Reaction('teichoicacid_rxn')\n",
    "#teichoicacid_rxn.gene_reaction_rule = '272563.8.peg.262'\n",
    "teichoicacid_rxn.name = 'Teichoic acid biosynthesis'\n",
    "teichoicacid_rxn.lower_bound = 0.\n",
    "teichoicacid_rxn.upper_bound = 1000.\n",
    "teichoicacid_rxn.add_metabolites({\n",
    "    cpd02967_c: -1.0,\n",
    "    cpd00402_c: -1.0,\n",
    "    cpd00046_c: 1.0,\n",
    "    cpd12894_c: 1.0\n",
    "})\n",
    "draft_GENRE.add_reactions([teichoicacid_rxn])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace Stickland pathway reaction\n",
    "cpd00005_c = draft_GENRE.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "cpd00067_c = draft_GENRE.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00498_c = draft_GENRE.metabolites.get_by_id('cpd00498_c') # 2-Aceto-2-hydroxybutanoate\n",
    "cpd00006_c = draft_GENRE.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd02535_c = draft_GENRE.metabolites.get_by_id('cpd02535_c') # 2,3-Dihydroxy-3-methylvalerate\n",
    "\n",
    "gene = cobra.Gene('645463.3.peg.1493')\n",
    "gene.name = 'Ketol-acid reductoisomerase (NADP(+)) (EC 1.1.1.86)'\n",
    "\n",
    "rxn08764_c = cobra.Reaction('rxn08764_c')\n",
    "rxn08764_c.gene_reaction_rule = '645463.3.peg.1493'\n",
    "rxn08764_c.name = 'Teichoic acid biosynthesis'\n",
    "rxn08764_c.lower_bound = -1000.\n",
    "rxn08764_c.upper_bound = 1000.\n",
    "rxn08764_c.add_metabolites({\n",
    "    cpd00005_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00498_c: -1.0,\n",
    "    cpd00006_c: 1.0,\n",
    "    cpd02535_c: 1.0\n",
    "})\n",
    "draft_GENRE.add_reactions([rxn08764_c])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove DNA demand\n",
    "try:\n",
    "    draft_GENRE.reactions.get_by_id('rxn13783_c').remove_from_model(remove_orphans=True)\n",
    "except:\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Metabolites pruned: 2\n",
      "Reactions pruned: 0\n"
     ]
    }
   ],
   "source": [
    "# Prune orphaned nodes\n",
    "cpd_total = 0\n",
    "rxn_total = 0\n",
    "removed = 1\n",
    "while removed == 1:\n",
    "    removed = 0\n",
    "\n",
    "    # Metabolites\n",
    "    for cpd in draft_GENRE.metabolites:\n",
    "        if len(cpd.reactions) == 0:\n",
    "            cpd.remove_from_model()\n",
    "            removed = 1\n",
    "            cpd_total += 1\n",
    "\n",
    "    # Reactions\n",
    "    for rxn in draft_GENRE.reactions:\n",
    "        if len(rxn.metabolites) == 0: \n",
    "            rxn.remove_from_model()\n",
    "            removed = 1\n",
    "            rxn_total += 1\n",
    "    \n",
    "print('Metabolites pruned: ' + str(cpd_total))\n",
    "print('Reactions pruned: ' + str(rxn_total))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix compartments\n",
    "for cpd in draft_GENRE.metabolites:\n",
    "    if cpd.compartment == 'c':\n",
    "        cpd.compartment = 'cytosol'\n",
    "    elif cpd.compartment == 'e':\n",
    "        cpd.compartment = 'extracellular'\n",
    "    elif cpd.compartment == 'p':\n",
    "        cpd.compartment = 'periplasm'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Open exchange bounds\n",
    "for rxn in draft_GENRE.boundary:\n",
    "    rxn.bounds = (-1000.0,1000.0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "190.16070481163598"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Test that new model grows\n",
    "draft_GENRE.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NCMM: 56.14277852157972\n",
      "MDM: 80.96607010298253\n"
     ]
    }
   ],
   "source": [
    "# Test growth in minimal media conditions\n",
    "\n",
    "# Karlsson et al. (1999). Microbiology.\n",
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "# Theriot et al. (2013). Nature Communications.\n",
    "ncmm = ['cpd00001_e', # water\n",
    "        'cpd00104_e', # Biotin\n",
    "        'cpd00644_e', # Pantothenate\n",
    "        'cpd00263_e', # Pyridoxine\n",
    "        'cpd00149_e', # Cobalt\n",
    "        'cpd00099_e', # Chloride\n",
    "        'cpd10515_e', # Iron\n",
    "        'cpd00030_e', # Manganese\n",
    "        'cpd00254_e', # Magnesium\n",
    "        'cpd00063_e', # Calcium\n",
    "        'cpd00205_e', # Potassium\n",
    "        'cpd00009_e', # Phosphate\n",
    "        'cpd00971_e', # Sodium\n",
    "        'cpd00242_e', # Carbonate\n",
    "        'cpd00322_e', # L-Isoleucine\n",
    "        'cpd00129_e', # L-Proline\n",
    "        'cpd00156_e', # L-Valine\n",
    "        'cpd00107_e', # L-Leucine\n",
    "        'cpd00084_e', # L-Cysteine \n",
    "        'cpd00065_e', # L-Tryptophan\n",
    "        'cpd00060_e', # L-Methionine\n",
    "        'cpd00119_e', # L-Histidine\n",
    "        'cpd00033_e', # Glycine\n",
    "        'cpd00051_e', # L-Arginine\n",
    "        'cpd00161_e'] # L-Threonine\n",
    "\n",
    "ncmm_exchanges = ['EX_' + x for x in ncmm]\n",
    "for rxn in draft_GENRE.boundary:\n",
    "    if not rxn.id in ncmm_exchanges:\n",
    "        rxn.bounds = (0.0,1000.0)\n",
    "    else:\n",
    "        rxn.bounds = (-1000.0,1000.0)\n",
    "print('NCMM: ' + str(draft_GENRE.slim_optimize()))\n",
    "\n",
    "mdm_exchanges = ['EX_' + x for x in mdm]\n",
    "for rxn in draft_GENRE.boundary:\n",
    "    if not rxn.id in mdm_exchanges:\n",
    "        rxn.bounds = (0.0,1000.0)\n",
    "    else:\n",
    "        rxn.bounds = (-1000.0,1000.0)\n",
    "print('MDM: ' + str(draft_GENRE.slim_optimize()))\n",
    "\n",
    "# Leave in MDM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "for x in draft_GENRE.genes:\n",
    "    if len(x.reactions) == 0:\n",
    "        print(x.id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'645463.3.peg.1041'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-23-66328bd4a69a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Remove genes with no reactions\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mremove_genes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mdraft_GENRE\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_by_id\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'645463.3.peg.1041'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1043'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1040'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1042'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1045'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1044'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mcobra\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmanipulation\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mremove_genes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdraft_GENRE\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mremove_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-23-66328bd4a69a>\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# Remove genes with no reactions\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mremove_genes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mdraft_GENRE\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgenes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_by_id\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'645463.3.peg.1041'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1043'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1040'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1042'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1045'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'645463.3.peg.1044'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mcobra\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmanipulation\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mremove_genes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdraft_GENRE\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mremove_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/cobra/core/dictlist.py\u001b[0m in \u001b[0;36mget_by_id\u001b[0;34m(self, id)\u001b[0m\n\u001b[1;32m     56\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mget_by_id\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mid\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m         \u001b[0;34m\"\"\"return the element with a matching id\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 58\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__getitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mid\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mlist_attr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattribute\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: '645463.3.peg.1041'"
     ]
    }
   ],
   "source": [
    "# Remove genes with no reactions\n",
    "remove_genes = [draft_GENRE.genes.get_by_id(x) for x in ['645463.3.peg.1041','645463.3.peg.1043','645463.3.peg.1040','645463.3.peg.1042','645463.3.peg.1045','645463.3.peg.1044']]\n",
    "cobra.manipulation.delete.remove_genes(draft_GENRE, remove_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Name new model\n",
    "draft_GENRE.name = 'Clostridium difficile R20291'\n",
    "draft_GENRE.id = 'iCdR' + str(len(draft_GENRE.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdR700</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fa79f4d7a50</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1238</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1292</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>0.0 + 1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdR700 at 0x7fa79f4d7a50>"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "draft_GENRE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotation information\n",
    "\n",
    "# SBO designations\n",
    "# Metabolites\n",
    "for cpd in draft_GENRE.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "# Reactions\n",
    "for rxn in draft_GENRE.reactions:\n",
    "    substrates = list(rxn.metabolites)\n",
    "    compartments = set([x.compartment for x in substrates])\n",
    "    if 'EX_' in rxn.id:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000627' # exchange\n",
    "    elif len(compartments) > 1:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "    else:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "# Biomass\n",
    "draft_GENRE.reactions.dna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.rna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.teichoicacid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.peptidoglycan_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.cellwall_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.lipid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.cofactor_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.biomass.annotation['sbo'] = 'SBO:0000629'  \n",
    "draft_GENRE.reactions.SK_cpd11416_c.annotation['sbo'] = 'SBO:0000632'  \n",
    "# Genes\n",
    "for gene in draft_GENRE.genes:\n",
    "    gene.annotation['sbo'] = 'SBO:0000243'\n",
    "\n",
    "# RefSeq annotations\n",
    "patric_refseq = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/r20291_refseq_genes.tsv') as ref:\n",
    "    for line in ref:\n",
    "        patric_refseq[line.split()[0]] = line.split()[1]\n",
    "for gene in draft_GENRE.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        gene.annotation['refseq'] = 'NC_013316.1'\n",
    "        try:\n",
    "            gene.annotation['refseq.locustag'] = patric_refseq[gene.id]\n",
    "        except:\n",
    "            continue\n",
    "\n",
    "# Metabolites\n",
    "for cpd in draft_GENRE.metabolites:\n",
    "    if 'cpd' in cpd.id:\n",
    "        new_id = str(cpd.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        cpd.annotation['seed.compound'] = new_id\n",
    "draft_GENRE.metabolites.C21400_c.annotation['kegg.compound'] = 'C21400'        \n",
    "draft_GENRE.metabolites.C21399_c.annotation['kegg.compound'] = 'C21399'        \n",
    "draft_GENRE.metabolites.C21090_c.annotation['kegg.compound'] = 'C21090'        \n",
    "draft_GENRE.metabolites.C21399_e.annotation['kegg.compound'] = 'C21399'        \n",
    "\n",
    "# Reactions\n",
    "for rxn in draft_GENRE.reactions:\n",
    "    if 'rxn' in rxn.id or 'EX_' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['seed.reaction'] = new_id\n",
    "    elif 'R' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['kegg.reaction'] = new_id\n",
    "draft_GENRE.reactions.SK_cpd11416_c.annotation['seed.reaction'] = 'SK_cpd11416'\n",
    "draft_GENRE.reactions.ENOG4108HXH_c.annotation['eggnog.reaction'] = 'ENOG4108HXH'\n",
    "draft_GENRE.reactions.COG3601_c.annotation['cog.reaction'] = 'COG3601'\n",
    "draft_GENRE.reactions.K20025_c.annotation['kegg.reaction'] = 'K20025'\n",
    "\n",
    "# Genes\n",
    "for gene in draft_GENRE.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        gene.annotation['patric'] = 'fig|' + gene.id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct residual gene names\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.1176').name = 'Na(+)-dependent branched-chain amino acid transporter'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.735').name = 'ABC transporter, substrate-binding protein'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2463').name = 'PTS system, glucitol/sorbitol-specific IIB component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.929').name = 'N-acetylgalactosamine-6-phosphate deacetylase'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2324').name = 'ribulose-5-phosphate 4-epimerase'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.46').name = 'Ribulose-5-phosphate 4-epimerase'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.3015').name = 'PTS system, IIB component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.3014').name = 'PTS system, IIC component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.3016').name = 'PTS system, IIA component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.865').name = 'ABC transporter, substrate-binding protein'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.862').name = 'ABC transporter, substrate-binding protein'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.664').name = 'sodium-solute symporter, putative'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.1783').name = 'ABC transporter, ATP-binding protein'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.1421').name = 'DNA-3-methyladenine glycosylase II'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2254').name = '2-oxoglutarate/2-oxoacid ferredoxin oxidoreductase, delta subunit'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2253').name = '2-oxoglutarate/2-oxoacid ferredoxin oxidoreductase, alpha subunit'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.168').name = '4-hydroxyphenylacetate decarboxylase, small subunit'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.167').name = '4-hydroxyphenylacetate decarboxylase, large subunit'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2417').name = 'Indolepyruvate oxidoreductase subunit'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2389').name = 'Glycine reductase component B beta subunit'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.392').name = '2-hydroxyglutaryl-CoA dehydratase, A-component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.393').name = '2-hydroxyglutaryl-CoA dehydratase, B-component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.390').name = 'Hydroxyproline dehydratase putative'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.391').name = 'CoA-substrate-specific enzyme activase; 2-hydroxyglutaryl-CoA dehydratase activator, A-component'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.1466').name = 'NADH-dependent reduced ferredoxin:NADP+ oxidoreductase subunit A'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.169').name = '4-hydroxyphenylacetate decarboxylase activating enzyme'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2250').name = 'Acetyl-CoA synthetase (ADP-forming) alpha and beta chains, putative'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2379').name = 'Succinate-semialdehyde dehydrogenase, CoA-dependent'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.976').name = 'Electron bifurcating butyryl-CoA dehydrogenase (NAD+, ferredoxin)'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.2703').name = 'UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase'\n",
    "draft_GENRE.genes.get_by_id('645463.3.peg.1493').name = 'Ketol-acid reductoisomerase (NADP(+)) (EC 1.1.1.86) | ilvC'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add additional annotation information\n",
    "\n",
    "# BiGG\n",
    "BiGG_rxn_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/rxns_BiGG_ids.txt', 'r') as bigg_rxns:\n",
    "    for line in bigg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        bigg = line[1].strip().split(',')\n",
    "        if len(bigg) == 1: bigg = bigg[0]\n",
    "        BiGG_rxn_ids[modelseed] = bigg\n",
    "BiGG_cpd_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/cpds_BiGG_ids.txt', 'r') as bigg_rxns:\n",
    "    for line in bigg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        bigg = line[1].strip().split(',')\n",
    "        if len(bigg) == 1: bigg = bigg[0]\n",
    "        BiGG_cpd_ids[modelseed] = bigg\n",
    "\n",
    "# KEGG\n",
    "KEGG_rxn_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/rxns_KEGG_ids.txt', 'r') as kegg_rxns:\n",
    "    for line in kegg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        kegg = line[1].strip().split(',')\n",
    "        if len(kegg) == 1: kegg = kegg[0]\n",
    "        KEGG_rxn_ids[modelseed] = kegg\n",
    "KEGG_cpd_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/cpds_KEGG_ids.txt', 'r') as kegg_rxns:\n",
    "    for line in kegg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        kegg = line[1].strip().split(',')\n",
    "        if len(kegg) == 1: kegg = kegg[0]\n",
    "        KEGG_cpd_ids[modelseed] = kegg\n",
    "\n",
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in draft_GENRE.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        draft_GENRE.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        draft_GENRE.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in draft_GENRE.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        draft_GENRE.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        draft_GENRE.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add additional gene annotations\n",
    "\n",
    "uniprot_gene_ids = {}\n",
    "ec_gene_ids = {}\n",
    "refseq_gene_ids = {}\n",
    "kegg_gene_ids = {}\n",
    "patric_genes = set()\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/uniprot_CdR20291.tab', 'r') as gene_annotation:\n",
    "    header = gene_annotation.readline()\n",
    "    for line in gene_annotation:\n",
    "        line = line.split()\n",
    "        uniprot_gene_ids[line[4]] = line[0]\n",
    "        ec_gene_ids[line[4]] = line[1]\n",
    "        refseq_gene_ids[line[4]] = line[2]\n",
    "        kegg_gene_ids[line[4]] = line[3]\n",
    "        patric_genes |= set([line[4]])\n",
    "\n",
    "success = 0\n",
    "for gene in draft_GENRE.genes:\n",
    "    if not gene.id in patric_genes: continue\n",
    "    \n",
    "    success += 1\n",
    "    gene.annotation['uniprot'] = uniprot_gene_ids[gene.id]\n",
    "    gene.annotation['kegg.genes'] = kegg_gene_ids[gene.id]\n",
    "    \n",
    "    if refseq_gene_ids[gene.id] != 'none':\n",
    "        gene.annotation['refseq'] = refseq_gene_ids[gene.id]\n",
    "    \n",
    "    if ec_gene_ids[gene.id] != 'none':\n",
    "        gene.annotation['ec_number'] = ec_gene_ids[gene.id]\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring reaction 'rxn00816_c' since it already exists.\n",
      "Ignoring reaction 'rxn00808_c' since it already exists.\n"
     ]
    }
   ],
   "source": [
    "# Unblock Lactose catabolism\n",
    "\n",
    "cpd00208_c = draft_GENRE.metabolites.cpd00208_c # Lactose\n",
    "cpd00001_c = draft_GENRE.metabolites.cpd00001_c # H2O\n",
    "cpd00027_c = draft_GENRE.metabolites.cpd00027_c # Glucose\n",
    "\n",
    "cpd00108_c = cobra.Metabolite(\n",
    "    'cpd00108_c',\n",
    "    formula='C6H12O6',\n",
    "    name='Galactose',\n",
    "    compartment='cytosol')\n",
    "cpd00108_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00108_c.annotation['seed.compound'] = 'cpd00108'\n",
    "cpd00108_c.annotation['bigg.metabolite'] = 'gal'\n",
    "cpd00108_c.annotation['kegg.compound'] = 'C00124'\n",
    "\n",
    "rxn00816_c = cobra.Reaction('rxn00816_c')\n",
    "rxn00816_c.name = 'Beta-galactosidase'\n",
    "rxn00816_c.lower_bound = 0.\n",
    "rxn00816_c.upper_bound = 1000.\n",
    "rxn00816_c.add_metabolites({\n",
    "    cpd00208_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00027_c: 1.0,\n",
    "    cpd00108_c: 1.0\n",
    "})\n",
    "rxn00816_c.annotation['seed.reaction'] = 'rxn00816'\n",
    "rxn00816_c.annotation['kegg.reaction'] = 'R01100'\n",
    "rxn00816_c.annotation['bigg.reaction'] = 'LACZ'\n",
    "draft_GENRE.add_reactions([rxn00816_c])\n",
    "\n",
    "cpd00002_c = draft_GENRE.metabolites.cpd00002_c # ATP\n",
    "cpd00008_c = draft_GENRE.metabolites.cpd00008_c # ADP\n",
    "cpd00067_c = draft_GENRE.metabolites.cpd00067_c # H\n",
    "cpd00348_c = draft_GENRE.metabolites.cpd00348_c # galactose-1-phosphate \n",
    "\n",
    "rxn00808_c = cobra.Reaction('rxn00808_c')\n",
    "rxn00808_c.name = 'ATP:D-galactose 1-phosphotransferase'\n",
    "rxn00808_c.lower_bound = 0.\n",
    "rxn00808_c.upper_bound = 1000.\n",
    "rxn00808_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00108_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00348_c: 1.0\n",
    "})\n",
    "rxn00808_c.annotation['seed.reaction'] = 'rxn00808'\n",
    "rxn00808_c.annotation['kegg.reaction'] = 'R01092'\n",
    "rxn00808_c.annotation['bigg.reaction'] = 'GAL1'\n",
    "draft_GENRE.add_reactions([rxn00808_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring reaction 'rxn05472_c' since it already exists.\n"
     ]
    }
   ],
   "source": [
    "# Unblock Salicin catabolism\n",
    "\n",
    "cpd01553_c = draft_GENRE.metabolites.cpd01553_c # Saligenin\n",
    "\n",
    "cpd01553_e = cobra.Metabolite(\n",
    "    'cpd01553_e',\n",
    "    formula='C7H8O2',\n",
    "    name='Saligenin',\n",
    "    compartment='extracellular')\n",
    "cpd01553_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd01553_e.annotation['seed.compound'] = 'cpd01553'\n",
    "cpd01553_e.annotation['bigg.metabolite'] = '2hymeph'\n",
    "cpd01553_e.annotation['kegg.compound'] = 'C02323'\n",
    "draft_GENRE.add_boundary(cpd01553_e, type='exchange', reaction_id='EX_cpd01553_e', lb=-1000.0, ub=1000.0)\n",
    "draft_GENRE.reactions.EX_cpd01553_e.name = 'Saligenin exchange'\n",
    "draft_GENRE.reactions.EX_cpd01553_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00067_c = draft_GENRE.metabolites.cpd00067_c # H\n",
    "cpd00067_e = draft_GENRE.metabolites.cpd00067_e # H\n",
    "\n",
    "rxn05472_c = cobra.Reaction('rxn05472_c')\n",
    "rxn05472_c.name = '2-(hydroxymethyl)phenol transport in/out via proton symport'\n",
    "rxn05472_c.lower_bound = -1000.\n",
    "rxn05472_c.upper_bound = 1000.\n",
    "rxn05472_c.add_metabolites({\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd01553_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd01553_c: 1.0\n",
    "})\n",
    "rxn05472_c.annotation['sbo'] = 'SBO:0000185'\n",
    "rxn05472_c.annotation['seed.reaction'] = 'rxn05472'\n",
    "rxn05472_c.annotation['bigg.reaction'] = '2HXMPt6'\n",
    "draft_GENRE.add_reactions([rxn05472_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring reaction 'R01103_c' since it already exists.\n"
     ]
    }
   ],
   "source": [
    "# Correct Melezitose catabolism\n",
    "\n",
    "cpd00001_c = draft_GENRE.metabolites.cpd00001_c # H2O\n",
    "cpd05161_c = draft_GENRE.metabolites.cpd05161_c # Melezitose\n",
    "cpd00027_c = draft_GENRE.metabolites.cpd00027_c # Glucose\n",
    "cpd20885_c = draft_GENRE.metabolites.cpd20885_c # Turanose\n",
    "\n",
    "R01103_c = cobra.Reaction('R01103_c')\n",
    "R01103_c.name = 'Melezitose galactohydrolase'\n",
    "R01103_c.lower_bound = 0.\n",
    "R01103_c.upper_bound = 1000.\n",
    "R01103_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd05161_c: -1.0,\n",
    "    cpd00027_c: 1.0,\n",
    "    cpd20885_c: 1.0\n",
    "})\n",
    "R01103_c.annotation['kegg.reaction'] = 'R01103'\n",
    "R01103_c.annotation['sbo'] = 'SBO:0000176'\n",
    "draft_GENRE.add_reactions([R01103_c])\n",
    "\n",
    "draft_GENRE.id = 'iCdR' + str(len(draft_GENRE.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set media condition\n",
    "for rxn in draft_GENRE.boundary:\n",
    "    if not rxn.id in mdm_exchanges:\n",
    "        rxn.bounds = (0.0,1000.0)\n",
    "    else:\n",
    "        rxn.bounds = (-1000.0,1000.0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "draft_GENRE.reactions.EX_cpd00104_e.name = 'Biotin exchange'\n",
    "draft_GENRE.reactions.rxn05872_c.name = 'Secondary alcohol:NAD+ oxidoreductase'\n",
    "draft_GENRE.reactions.EX_cpd00208_e.name = 'Lactose exchange'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Label tRNA reactions as Biomass\n",
    "tRNA = [\"rxn06449_c\",\"rxn06443_c\",\"rxn06444_c\",\"rxn06445_c\",\"rxn06446_c\",\"rxn06447_c\",\"rxn06448_c\",\"rxn06280_c\",\"rxn06300_c\",\"rxn06432_c\",\"rxn06434_c\",\"rxn06936_c\",\"rxn06435_c\",\"rxn06437_c\",\"rxn06436_c\",\"rxn06438_c\",\"rxn06439_c\",\"rxn06440_c\",\"rxn06441_c\",\"rxn06442_c\"]\n",
    "for rxn in tRNA: draft_GENRE.reactions.get_by_id(rxn).annotation['sbo'] = 'SBO:0000629'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdR700</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fa79f4d7a50</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1238</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1292</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>0.0 + 1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdR700 at 0x7fa79f4d7a50>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "draft_GENRE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "from cobra.flux_analysis import parsimonious\n",
    "\n",
    "# Function to calculate doubling time from objective value\n",
    "def doublingTime(model):\n",
    "    with model as m:\n",
    "        if m.slim_optimize(error_value=0.) < 1e-6:\n",
    "            print('GENRE has no objective flux')\n",
    "        else:\n",
    "            growth = (1. / float(m.slim_optimize())) * 3600.\n",
    "            print(str(round(growth, 2)) + ' minutes doubling time')\n",
    "\n",
    "\n",
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "\n",
    "    #print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    fraction = (float(len(blocked)) / float(len(model.reactions))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions are blocked')\n",
    "    \n",
    "    return blocked\n",
    "\n",
    "\n",
    "# Identify potentially gapfilled reactions, checks against pFBA solution\n",
    "def missingGPR(model):\n",
    "    \n",
    "    noGene = []\n",
    "    exclude = []\n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.genes)) == 0:\n",
    "            if rxn.annotation['sbo'] != 'SBO:0000629':\n",
    "                if rxn in model.boundary:\n",
    "                    exclude.append(rxn.id)\n",
    "                    continue\n",
    "                else:\n",
    "                    noGene.append(rxn.id)\n",
    "    \n",
    "    solution = parsimonious.pfba(model)\n",
    "    active_rxns = set([rxn.id for rxn in model.reactions if abs(solution.fluxes[rxn.id]) > 1e-5])\n",
    "    active_rxns = active_rxns.difference(set(exclude))\n",
    "    noGene_active = set(noGene).intersection(active_rxns)\n",
    "\n",
    "    fraction = float(len(model.reactions)) - float(len(exclude))\n",
    "    fraction = (float(len(noGene)) / fraction) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions without GPRs')\n",
    "    \n",
    "    fraction = (float(len(noGene_active)) / float(len(active_rxns))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% of reactions used in pFBA solution have no GPR')\n",
    "    \n",
    "    return noGene_active\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(model, cytosol='cytosol'):\n",
    "\n",
    "    free = []\n",
    "    with model as m:\n",
    "    \n",
    "        # Close all exchanges\n",
    "        for rxn in m.boundary: m.reactions.get_by_id(rxn.id).lower_bound = 0.\n",
    "    \n",
    "        # Create demand for each reaction and optimize individually\n",
    "        reset_rxn = m.reactions[0].id\n",
    "        for cpd in m.metabolites: \n",
    "            if cpd.compartment == cytosol:\n",
    "                demand = cobra.Reaction('demand')\n",
    "                demand.bounds = (0., 1000.)\n",
    "                demand.add_metabolites({cpd: -1.0})\n",
    "                m.add_reactions([demand])\n",
    "                m.objective = demand\n",
    "                obj_val = m.slim_optimize()\n",
    "                if obj_val > 1e-8: free.append(cpd.id)\n",
    "                m.objective = reset_rxn\n",
    "                m.remove_reactions([demand])\n",
    "    \n",
    "    fraction = (float(len(free)) / float(len(model.metabolites))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(model):\n",
    "    \n",
    "    with model as m:\n",
    "\n",
    "        elements = set()\n",
    "        for cpd in m.metabolites:\n",
    "            try:\n",
    "                elements |= set(cpd.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        massImbal = []\n",
    "        failed = 0\n",
    "        if len(elements) == 0:\n",
    "            print('No elemental data associated with metabolites!')\n",
    "            failed = 1\n",
    "        else:\n",
    "            for rxn in m.reactions:\n",
    "                if rxn.annotation['sbo'] == 'SBO:0000629': \n",
    "                    continue\n",
    "                elif rxn in m.boundary:\n",
    "                    continue\n",
    "\n",
    "                try:\n",
    "                    test = rxn.check_mass_balance()\n",
    "                except ValueError:\n",
    "                    continue\n",
    "\n",
    "                if len(list(test)) > 0:\n",
    "                    if len(set(test.keys()).intersection(elements)) > 0: massImbal.append(rxn.id)\n",
    "                        \n",
    "    if failed != 1:\n",
    "        fraction = (float(len(massImbal)) / float(len(model.reactions))) * 100.\n",
    "        fraction = round(fraction, 2)\n",
    "        print(str(fraction) + '% reactions are mass imbalanced')\n",
    "        \n",
    "    return massImbal\n",
    "\n",
    "\n",
    "def basicCheck(model):\n",
    "    \n",
    "    # Determination\n",
    "    if len(model.reactions) < len(model.metabolites): \n",
    "        print('GENRE is overdetermined')\n",
    "    elif len(model.reactions) > len(model.metabolites):\n",
    "        print('GENRE is underdetermined')\n",
    "    else:\n",
    "        pass\n",
    "    \n",
    "    # Compartments\n",
    "    print('GENRE has ' + str(len(model.compartments.keys())) + ' compartment(s)')\n",
    "    \n",
    "    # Genes\n",
    "    if len(model.genes) == 0: \n",
    "        print('GENRE has no gene data')\n",
    "    else:\n",
    "        print('GENRE has ' + str(len(model.genes)) + ' genes')\n",
    "          \n",
    "    # Growth\n",
    "    doublingTime(model)\n",
    "\n",
    "# Quicker way to read in models\n",
    "import pickle\n",
    "def read_model(fileName, obj='none'):\n",
    "    \n",
    "    fileType = fileName.split('.')[-1]\n",
    "    \n",
    "    if fileType == 'sbml' or fileType == 'xml':\n",
    "        model = cobra.io.read_sbml_model(fileName)\n",
    "    elif fileType == 'json':\n",
    "        model = cobra.io.load_json_model(fileName)\n",
    "    elif fileType == 'yaml':\n",
    "        model = cobra.io.load_yaml_model(fileName)\n",
    "    elif fileType == 'mat':\n",
    "        model = cobra.io.load_matlab_model(fileName)\n",
    "    elif fileType == 'pkl':\n",
    "        model = pickle.load(open(fileName, 'rb'))\n",
    "    else:\n",
    "        raise TypeError('Unrecognized file extension')\n",
    "    \n",
    "    if obj != 'none': model.objective = obj\n",
    "    for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "    return model\n",
    "\n",
    "\n",
    "def save_model(model, fileName, media='none'):\n",
    "    \n",
    "    if media != 'none':\n",
    "        #media_exchanges = ['EX_' + x for x in media]\n",
    "        for rxn in model.boundary:\n",
    "            if not rxn.id in media:\n",
    "                rxn.bounds = (0., 1000.)\n",
    "            else:\n",
    "                rxn.bounds = (-1000., 1000.)\n",
    "    else:\n",
    "        for rxn in model.boundary:\n",
    "            rxn.bounds = (-1000., 1000.)\n",
    "    \n",
    "    sbml_fileName = fileName + '.sbml'\n",
    "    cobra.io.write_sbml_model(model, sbml_fileName)\n",
    "    \n",
    "    json_fileName = fileName + '.json'\n",
    "    cobra.io.save_json_model(model, json_fileName)\n",
    "    \n",
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 700 genes\n",
      "51.39 minutes doubling time\n",
      "48.94% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% reactions are mass imbalanced\n",
      "12.66% reactions without GPRs\n",
      "22.09% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "basicCheck(draft_GENRE)\n",
    "draft_noGPRblocked = blockedReactions(draft_GENRE)\n",
    "draft_free = checkFreeMass(draft_GENRE)\n",
    "draft_massImbal = checkBalance(draft_GENRE)\n",
    "draft_nogene = missingGPR(draft_GENRE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to files\n",
    "cobra.io.write_sbml_model(draft_GENRE, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.sbml')\n",
    "cobra.io.save_json_model(draft_GENRE, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring reaction 'rxn02094_c' since it already exists.\n",
      "Ignoring reaction 'rxn46837_c' since it already exists.\n",
      "Ignoring reaction 'rxn17713_c' since it already exists.\n"
     ]
    }
   ],
   "source": [
    "\n",
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')\n",
    "\n",
    "# Neu5Ac --> N-Acetyl-Neuraminic acid\n",
    "iCdR700.metabolites.cpd00232_c.name = 'N-Acetyl-Neuraminic acid'\n",
    "\n",
    "# D-Tagatose\n",
    "cpd00589_e = cobra.Metabolite(\n",
    "    'cpd00589_e',\n",
    "    formula='C6H12O6',\n",
    "    name='D-Tagatose',\n",
    "    compartment='extracellular')\n",
    "cpd00589_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00589_e.annotation['seed.compound'] = 'cpd00589'\n",
    "cpd00589_e.annotation['bigg.metabolite'] = 'tag__D'\n",
    "cpd00589_e.annotation['kegg.compound'] = 'C00795'\n",
    "#iCdR700.add_boundary(cpd00589_e, type='exchange', reaction_id='EX_cpd00589_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd01553_e.name = 'Tagatose exchange'\n",
    "iCdR700.reactions.EX_cpd01553_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00805_c = iCdR700.metabolites.cpd00805_c # D-Tagatose 6-phosphate\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c # ATP\n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c # ADP\n",
    "cpd00067_c = iCdR700.metabolites.cpd00067_c # H\n",
    "\n",
    "rxn02094_c = cobra.Reaction('rxn02094_c')\n",
    "rxn02094_c.name = 'ATP:D-tagatose 6-phosphotransferase'\n",
    "rxn02094_c.gene_reaction_rule = '( 645463.3.peg.3332 or 645463.3.peg.303 or 645463.3.peg.464 or 645463.3.peg.998 ) and ( 645463.3.peg.306 or 645463.3.peg.465 or 645463.3.peg.3333 or 645463.3.peg.1000 or 645463.3.peg.304 ) and ( 645463.3.peg.307 or 645463.3.peg.3331 or 645463.3.peg.466 or 645463.3.peg.1001 ) and ( 645463.3.peg.467 or 645463.3.peg.1002 or 645463.3.peg.308 or 645463.3.peg.3330 )'\n",
    "rxn02094_c.lower_bound = 0.\n",
    "rxn02094_c.upper_bound = 1000.\n",
    "rxn02094_c.add_metabolites({\n",
    "    cpd00589_e: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00805_c: 1.0\n",
    "})\n",
    "rxn02094_c.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "rxn02094_c.annotation['seed.reaction'] = 'rxn02094'\n",
    "rxn02094_c.annotation['kegg.reaction'] = 'R02927'\n",
    "rxn02094_c.annotation['bigg.reaction'] = 'TAGptspp'\n",
    "iCdR700.add_reactions([rxn02094_c])\n",
    "\n",
    "\n",
    "# N-Acetyl-D-Galactosamine\n",
    "cpd27607_e = cobra.Metabolite(\n",
    "    'cpd27607_e',\n",
    "    formula='C8H15NO6',\n",
    "    name='N-Acetyl-D-Galactosamine',\n",
    "    compartment='extracellular')\n",
    "cpd27607_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd27607_e.annotation['seed.compound'] = 'cpd27607'\n",
    "cpd27607_e.annotation['bigg.metabolite'] = 'acgal'\n",
    "cpd27607_e.annotation['kegg.compound'] = 'C01074'\n",
    "\n",
    "#iCdR700.add_boundary(cpd27607_e, type='exchange', reaction_id='EX_cpd27607_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd27607_e.name = 'N-Acetyl-D-Galactosamine exchange'\n",
    "iCdR700.reactions.EX_cpd27607_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd15390_c = cobra.Metabolite(\n",
    "    'cpd15390_c',\n",
    "    formula='C8H14NO9P',\n",
    "    name='N-Acetyl-D-Galactosamine-1-phosphate',\n",
    "    compartment='cytosol')\n",
    "cpd15390_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd15390_c.annotation['seed.compound'] = 'cpd15390'\n",
    "cpd15390_c.annotation['bigg.metabolite'] = 'acgal1p'\n",
    "cpd15390_c.annotation['kegg.compound'] = 'C18060'\n",
    "\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c # ATP\n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c # ADP\n",
    "cpd00067_c = iCdR700.metabolites.cpd00067_c # H\n",
    "\n",
    "rxn46837_c = cobra.Reaction('rxn46837_c')\n",
    "rxn46837_c.name = 'ATP:N-acetyl-D-galactosamine 1-phosphotransferase'\n",
    "rxn46837_c.gene_reaction_rule = '645463.3.peg.2802 and 645463.3.peg.3106 and 645463.3.peg.3106'\n",
    "rxn46837_c.lower_bound = 0.\n",
    "rxn46837_c.upper_bound = 1000.\n",
    "rxn46837_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd27607_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd15390_c: 1.0\n",
    "})\n",
    "rxn46837_c.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "rxn46837_c.annotation['seed.reaction'] = 'rxn46837'\n",
    "rxn46837_c.annotation['kegg.reaction'] = 'R08967'\n",
    "rxn46837_c.annotation['bigg.reaction'] = 'ACGALK'\n",
    "iCdR700.add_reactions([rxn46837_c])\n",
    "\n",
    "cpd00062_c = iCdR700.metabolites.cpd00062_c # UTP\n",
    "cpd00012_c = iCdR700.metabolites.cpd00012_c # Phosphate\n",
    "cpd00175_c = iCdR700.metabolites.cpd00175_c # UDP-N-acetyl-D-galactosamine\n",
    "\n",
    "rxn17713_c = cobra.Reaction('rxn17713_c')\n",
    "rxn17713_c.name = 'UTP-monosaccharide-1-phosphate uridylyltransferase'\n",
    "rxn17713_c.lower_bound = 0.\n",
    "rxn17713_c.upper_bound = 1000.\n",
    "rxn17713_c.add_metabolites({\n",
    "    cpd00062_c: -1.0,\n",
    "    cpd15390_c: -1.0,\n",
    "    cpd00012_c: 1.0,\n",
    "    cpd00175_c: 1.0\n",
    "})\n",
    "rxn17713_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "rxn17713_c.annotation['seed.reaction'] = 'rxn17713'\n",
    "rxn17713_c.annotation['kegg.reaction'] = 'R10183'\n",
    "rxn17713_c.annotation['bigg.reaction'] = 'GALT'\n",
    "iCdR700.add_reactions([rxn17713_c])\n",
    "\n",
    "save_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700', media=mdm)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Matching Biolog Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "import copy\n",
    "\n",
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "universal_GramPos.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13783_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13784_c.remove_from_model(remove_orphans=True)\n",
    "#universal_GramPos.reactions.rxn11920_c.name = 'Poly-beta-hydroxybutyrate (PHB) synthease'\n",
    "universal_GramPos.reactions.rxn12188_c.name = '2-deoxyglucose-6-phosphatase'\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# R20291-specific\n",
    "\n",
    "# Glycogen\n",
    "rxn12627_c = copy.deepcopy(universal_GramPos.reactions.rxn12627_c)\n",
    "iCdR700.add_reactions([rxn12627_c])\n",
    "iCdR700.reactions.rxn12627_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn12627_c.annotation['seed.reaction'] = 'rxn12627'\n",
    "cpd00155_e = iCdR700.metabolites.cpd00155_e\n",
    "iCdR700.add_boundary(cpd00155_e, type='exchange', reaction_id='EX_cpd00155_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00155_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Glutathione\n",
    "rxn12539_c = copy.deepcopy(universal_GramPos.reactions.rxn12539_c)\n",
    "iCdR700.add_reactions([rxn12539_c])\n",
    "iCdR700.reactions.rxn12539_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn12539_c.annotation['seed.reaction'] = 'rxn12539'\n",
    "iCdR700.metabolites.cpd00042_e.name = 'Glutathione'\n",
    "cpd00042_e = iCdR700.metabolites.cpd00042_e\n",
    "iCdR700.add_boundary(cpd00042_e, type='exchange', reaction_id='EX_cpd00042_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00042_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# beta-Alanine\n",
    "rxn05495_c = copy.deepcopy(universal_GramPos.reactions.rxn05495_c)\n",
    "iCdR700.add_reactions([rxn05495_c])\n",
    "iCdR700.reactions.rxn05495_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05495_c.annotation['seed.reaction'] = 'rxn05495'\n",
    "iCdR700.reactions.rxn05495_c.annotation['bigg.reaction'] = ['BALAt2pp','BALAt2r','ala-Bt2']\n",
    "cpd00085_e = iCdR700.metabolites.cpd00085_e\n",
    "iCdR700.add_boundary(cpd00085_e, type='exchange', reaction_id='EX_cpd00085_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00085_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Agmatine\n",
    "rxn12597_c = copy.deepcopy(universal_GramPos.reactions.rxn12597_c)\n",
    "iCdR700.add_reactions([rxn12597_c])\n",
    "iCdR700.reactions.rxn12597_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn12597_c.annotation['seed.reaction'] = 'rxn12597'\n",
    "cpd00152_e = iCdR700.metabolites.cpd00152_e\n",
    "iCdR700.add_boundary(cpd00152_e, type='exchange', reaction_id='EX_cpd00152_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00152_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Glycerol\n",
    "rxn12529_c = copy.deepcopy(universal_GramPos.reactions.rxn12529_c)\n",
    "iCdR700.add_reactions([rxn12529_c])\n",
    "iCdR700.reactions.rxn12529_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn12529_c.annotation['seed.reaction'] = 'rxn12529'\n",
    "iCdR700.reactions.rxn12529_c.annotation['bigg.reaction'] = 'GLYCt2'\n",
    "cpd00100_e = iCdR700.metabolites.cpd00100_e\n",
    "iCdR700.add_boundary(cpd00100_e, type='exchange', reaction_id='EX_cpd00100_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00100_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Choline\n",
    "rxn09692_c = copy.deepcopy(universal_GramPos.reactions.rxn09692_c)\n",
    "iCdR700.add_reactions([rxn09692_c])\n",
    "iCdR700.reactions.rxn09692_c.name = 'Choline symporter'\n",
    "iCdR700.reactions.rxn09692_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn09692_c.annotation['seed.reaction'] = 'rxn09692'\n",
    "iCdR700.reactions.rxn09692_c.annotation['bigg.reaction'] = ['CHLt2','CHLt2pp','CHLt2r','HNM1']\n",
    "cpd00098_e = iCdR700.metabolites.cpd00098_e\n",
    "iCdR700.add_boundary(cpd00098_e, type='exchange', reaction_id='EX_cpd00098_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00098_e.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring reaction 'rxn18658_c' since it already exists.\n",
      "Ignoring reaction 'rxn05554_c' since it already exists.\n",
      "Ignoring reaction 'rxn05562_c' since it already exists.\n",
      "Ignoring reaction 'rxn05563_c' since it already exists.\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Boundary reaction 'EX_cpd00089_e' already exists.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-50-647c61240af1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     67\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m \u001b[0mcpd00089_e\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0miCdR700\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmetabolites\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcpd00089_e\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 69\u001b[0;31m \u001b[0miCdR700\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd_boundary\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcpd00089_e\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'exchange'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreaction_id\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'EX_cpd00089_e'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlb\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1000.0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mub\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1000.0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     70\u001b[0m \u001b[0miCdR700\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreactions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEX_cpd00089_e\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mannotation\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sbo'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'SBO:0000627'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/cobra/core/model.py\u001b[0m in \u001b[0;36madd_boundary\u001b[0;34m(self, metabolite, type, reaction_id, lb, ub, sbo_term)\u001b[0m\n\u001b[1;32m    617\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mreaction_id\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreactions\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    618\u001b[0m             raise ValueError(\n\u001b[0;32m--> 619\u001b[0;31m                 \"Boundary reaction '{}' already exists.\".format(reaction_id))\n\u001b[0m\u001b[1;32m    620\u001b[0m         \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"{} {}\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmetabolite\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    621\u001b[0m         rxn = Reaction(id=reaction_id, name=name, lower_bound=lb,\n",
      "\u001b[0;31mValueError\u001b[0m: Boundary reaction 'EX_cpd00089_e' already exists."
     ]
    }
   ],
   "source": [
    "# Ala-ala\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdR700.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdR700.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdR700.metabolites.cpd00067_c # H+\n",
    "cpd00731_c = iCdR700.metabolites.cpd00731_c # Ala-ala\n",
    "cpd00731_e = cobra.Metabolite(\n",
    "    'cpd00731_e',\n",
    "    formula='C6H12N2O3',\n",
    "    name='Ala-ala',\n",
    "    compartment='extracellular')\n",
    "cpd00731_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00731_e.annotation['seed.compound'] = 'cpd00731'\n",
    "cpd00731_e.annotation['bigg.metabolite'] = ['alaala', 'diala-L']\n",
    "cpd00731_e.annotation['kegg.compound'] = 'C00993'\n",
    "\n",
    "# Transport\n",
    "rxn18658_c = cobra.Reaction('rxn18658_c')\n",
    "rxn18658_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn18658_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn18658_c.bounds = (0., 1000.)\n",
    "rxn18658_c.add_metabolites({\n",
    "    cpd00731_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00731_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn18658_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn18658_c.annotation['seed.reaction'] = 'rxn18658'\n",
    "rxn18658_c.annotation['bigg.reaction'] = 'ALAALAabcpp'\n",
    "iCdR700.add_reactions([rxn18658_c])\n",
    "\n",
    "cpd00731_e = iCdR700.metabolites.cpd00731_e\n",
    "#iCdR700.add_boundary(cpd00731_e, type='exchange', reaction_id='EX_cpd00731_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00731_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Hydrolysis\n",
    "iCdR700.reactions.rxn07466_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "\n",
    "#--------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Phosphorylated hexose sugars\n",
    "rxn05554_c = copy.deepcopy(universal_GramPos.reactions.rxn05554_c) # Fructose-6-phosphate\n",
    "rxn05562_c = copy.deepcopy(universal_GramPos.reactions.rxn05562_c) # Glucose-1-phosphate\n",
    "rxn05563_c = copy.deepcopy(universal_GramPos.reactions.rxn05563_c) # Glucose-6-phosphate\n",
    "iCdR700.add_reactions([rxn05554_c,rxn05562_c,rxn05563_c])\n",
    "\n",
    "iCdR700.reactions.rxn05554_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05554_c.annotation['seed.reaction'] = 'rxn05554'\n",
    "iCdR700.reactions.rxn05554_c.annotation['bigg.reaction'] = 'F6Pt6_i2'\n",
    "\n",
    "iCdR700.reactions.rxn05562_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05562_c.annotation['seed.reaction'] = 'rxn05562'\n",
    "iCdR700.reactions.rxn05562_c.annotation['bigg.reaction'] = 'G1Pt6_2'\n",
    "\n",
    "iCdR700.reactions.rxn05563_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05563_c.annotation['seed.reaction'] = 'rxn05563'\n",
    "iCdR700.reactions.rxn05563_c.annotation['bigg.reaction'] = 'G6Pt6_2'\n",
    "\n",
    "cpd00072_e = iCdR700.metabolites.cpd00072_e\n",
    "#iCdR700.add_boundary(cpd00072_e, type='exchange', reaction_id='EX_cpd00072_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00072_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00089_e = iCdR700.metabolites.cpd00089_e\n",
    "iCdR700.add_boundary(cpd00089_e, type='exchange', reaction_id='EX_cpd00089_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00089_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00079_e = iCdR700.metabolites.cpd00079_e\n",
    "iCdR700.add_boundary(cpd00079_e, type='exchange', reaction_id='EX_cpd00079_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00079_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Galactose\n",
    "rxn05162_c = copy.deepcopy(universal_GramPos.reactions.rxn05162_c)\n",
    "iCdR700.add_reactions([rxn05162_c])\n",
    "iCdR700.reactions.rxn05162_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05162_c.annotation['seed.reaction'] = ''\n",
    "iCdR700.reactions.rxn05162_c.annotation['bigg.reaction'] = ''\n",
    "cpd00108_e = iCdR700.metabolites.cpd00108_e\n",
    "iCdR700.add_boundary(cpd00108_e, type='exchange', reaction_id='EX_cpd00108_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.rxn18658_c.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "#--------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Intracellular organic acids\n",
    "\n",
    "# 4-Hydroxybenzoate\n",
    "rxn12549_c = copy.deepcopy(universal_GramPos.reactions.rxn12549_c)\n",
    "iCdR700.add_reactions([rxn12549_c])\n",
    "iCdR700.reactions.rxn12549_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn12549_c.annotation['seed.reaction'] = 'rxn12549'\n",
    "iCdR700.reactions.rxn12549_c.annotation['bigg.reaction'] = '4HBZt'\n",
    "cpd00136_e = iCdR700.metabolites.cpd00136_e\n",
    "iCdR700.add_boundary(cpd00136_e, type='exchange', reaction_id='EX_cpd00136_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00136_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# 2-Hydroxybutyrate\n",
    "rxn12585_c = copy.deepcopy(universal_GramPos.reactions.rxn12585_c) \n",
    "iCdR700.add_reactions([rxn12585_c])\n",
    "iCdR700.reactions.rxn12585_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn12585_c.annotation['seed.reaction'] = 'rxn12585'\n",
    "iCdR700.reactions.rxn12585_c.annotation['bigg.reaction'] = '2HBt2'\n",
    "cpd03561_e = iCdR700.metabolites.cpd03561_e\n",
    "iCdR700.add_boundary(cpd03561_e, type='exchange', reaction_id='EX_cpd03561_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd03561_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Fumarate\n",
    "rxn05561_c = copy.deepcopy(universal_GramPos.reactions.rxn05561_c) \n",
    "iCdR700.add_reactions([rxn05561_c])\n",
    "iCdR700.reactions.rxn05561_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05561_c.annotation['seed.reaction'] = 'rxn05561'\n",
    "iCdR700.reactions.rxn05561_c.annotation['bigg.reaction'] = ['FUMt2r','FUMt6','U207']\n",
    "cpd00106_e = iCdR700.metabolites.cpd00106_e\n",
    "iCdR700.add_boundary(cpd00106_e, type='exchange', reaction_id='EX_cpd00106_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00106_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Acetoacetate\n",
    "rxn10960_c = copy.deepcopy(universal_GramPos.reactions.rxn10960_c) \n",
    "iCdR700.add_reactions([rxn10960_c])\n",
    "iCdR700.reactions.rxn10960_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn10960_c.annotation['seed.reaction'] = 'rxn10960'\n",
    "iCdR700.reactions.rxn10960_c.annotation['bigg.reaction'] = ['ACACtex','U126']\n",
    "cpd00142_e = iCdR700.metabolites.cpd00142_e\n",
    "iCdR700.add_boundary(cpd00142_e, type='exchange', reaction_id='EX_cpd00142_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00142_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Shikimate\n",
    "rxn10827_c = copy.deepcopy(universal_GramPos.reactions.rxn10827_c) \n",
    "iCdR700.add_reactions([rxn10827_c])\n",
    "iCdR700.reactions.rxn10827_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn10827_c.annotation['seed.reaction'] = 'rxn10827'\n",
    "iCdR700.reactions.rxn10827_c.annotation['bigg.reaction'] = ['SKMt2','SKMt2pp']\n",
    "iCdR700.reactions.rxn10827_c.name = 'Shikimate transport in via proton symport'\n",
    "cpd00383_e = iCdR700.metabolites.cpd00383_e\n",
    "iCdR700.add_boundary(cpd00383_e, type='exchange', reaction_id='EX_cpd00383_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00383_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Formate\n",
    "rxn05559_c = copy.deepcopy(universal_GramPos.reactions.rxn05559_c) \n",
    "iCdR700.add_reactions([rxn05559_c])\n",
    "iCdR700.reactions.rxn05559_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05559_c.annotation['seed.reaction'] = 'rxn05559'\n",
    "iCdR700.reactions.rxn05559_c.annotation['bigg.reaction'] = ['FORt2','FORt2pp']\n",
    "cpd00047_e = iCdR700.metabolites.cpd00047_e\n",
    "iCdR700.add_boundary(cpd00047_e, type='exchange', reaction_id='EX_cpd00047_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00047_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Glycolate\n",
    "rxn05470_c = copy.deepcopy(universal_GramPos.reactions.rxn05470_c) \n",
    "iCdR700.add_reactions([rxn05470_c])\n",
    "iCdR700.reactions.rxn05470_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05470_c.annotation['seed.reaction'] = 'rxn05470'\n",
    "iCdR700.reactions.rxn05470_c.annotation['bigg.reaction'] = ['GLYCLTt2r','GLYCLTt2rpp']\n",
    "cpd00139_e = iCdR700.metabolites.cpd00139_e\n",
    "iCdR700.add_boundary(cpd00139_e, type='exchange', reaction_id='EX_cpd00139_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00139_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# GABA\n",
    "rxn05564_c = copy.deepcopy(universal_GramPos.reactions.rxn05564_c) \n",
    "iCdR700.add_reactions([rxn05564_c])\n",
    "iCdR700.reactions.rxn05564_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05564_c.annotation['seed.reaction'] = 'rxn05564'\n",
    "iCdR700.reactions.rxn05564_c.annotation['bigg.reaction'] = ['ABUTt2','ABUTt2pp','ABUTt2r','GABAt2','PUT4_4','UGA4_1']\n",
    "cpd00281_e = iCdR700.metabolites.cpd00281_e\n",
    "iCdR700.add_boundary(cpd00281_e, type='exchange', reaction_id='EX_cpd00281_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00281_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "#--------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Hydroxylamine\n",
    "rxn05703_c = copy.deepcopy(universal_GramPos.reactions.rxn05703_c) \n",
    "iCdR700.add_reactions([rxn05703_c])\n",
    "iCdR700.reactions.rxn05703_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05703_c.annotation['seed.reaction'] = 'rxn05703'\n",
    "cpd00165_e = iCdR700.metabolites.cpd00165_e\n",
    "iCdR700.add_boundary(cpd00165_e, type='exchange', reaction_id='EX_cpd00165_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00165_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "#--------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Nucleotide analogs\n",
    "\n",
    "# Deoxyinosine\n",
    "rxn09687_c = copy.deepcopy(universal_GramPos.reactions.rxn09687_c) \n",
    "iCdR700.add_reactions([rxn09687_c])\n",
    "iCdR700.reactions.rxn09687_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn09687_c.annotation['seed.reaction'] = 'rxn09687'\n",
    "iCdR700.reactions.rxn09687_c.annotation['bigg.reaction'] = ['DINSt2','DINSt2pp','U180','U196']\n",
    "cpd03279_e = iCdR700.metabolites.cpd03279_e\n",
    "iCdR700.add_boundary(cpd03279_e, type='exchange', reaction_id='EX_cpd03279_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd03279_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Inosine\n",
    "rxn05316_c = copy.deepcopy(universal_GramPos.reactions.rxn05316_c) \n",
    "iCdR700.add_reactions([rxn05316_c])\n",
    "iCdR700.reactions.rxn05316_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05316_c.annotation['seed.reaction'] = 'rxn05316'\n",
    "iCdR700.reactions.rxn05316_c.annotation['bigg.reaction'] = ['INSt2','INSt2pp','INSt2r','INSt2rpp','U177','U194']\n",
    "cpd00246_e = iCdR700.metabolites.cpd00246_e\n",
    "iCdR700.add_boundary(cpd00246_e, type='exchange', reaction_id='EX_cpd00246_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00246_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Guanosine\n",
    "rxn05204_c = copy.deepcopy(universal_GramPos.reactions.rxn05204_c) \n",
    "iCdR700.add_reactions([rxn05204_c])\n",
    "iCdR700.reactions.rxn05204_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05204_c.annotation['seed.reaction'] = 'rxn05204'\n",
    "iCdR700.reactions.rxn05204_c.annotation['bigg.reaction'] = ['GSNt2','GSNt2m','GSNt2pp','GSNt2r','U175','U192']\n",
    "cpd00311_e = iCdR700.metabolites.cpd00311_e\n",
    "iCdR700.add_boundary(cpd00311_e, type='exchange', reaction_id='EX_cpd00311_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00311_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Cytidine\n",
    "rxn05198_c = copy.deepcopy(universal_GramPos.reactions.rxn05198_c) \n",
    "iCdR700.add_reactions([rxn05198_c])\n",
    "iCdR700.reactions.rxn05198_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05198_c.annotation['seed.reaction'] = 'rxn05198'\n",
    "iCdR700.reactions.rxn05198_c.annotation['bigg.reaction'] = ['CYTDt2','CYTDt2pp','CYTDt2r','CYTDt2rpp','U176','U186','U193']\n",
    "cpd00367_e = iCdR700.metabolites.cpd00367_e\n",
    "iCdR700.add_boundary(cpd00367_e, type='exchange', reaction_id='EX_cpd00367_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00367_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxyguanosine\n",
    "rxn09688_c = copy.deepcopy(universal_GramPos.reactions.rxn09688_c) \n",
    "iCdR700.add_reactions([rxn09688_c])\n",
    "iCdR700.reactions.rxn09688_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn09688_c.annotation['seed.reaction'] = 'rxn09688'\n",
    "iCdR700.reactions.rxn09688_c.annotation['bigg.reaction'] = ['DGSNt2','DGSNt2pp','U181','U197']\n",
    "cpd00277_e = iCdR700.metabolites.cpd00277_e\n",
    "iCdR700.add_boundary(cpd00277_e, type='exchange', reaction_id='EX_cpd00277_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00277_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Adenosine\n",
    "rxn05318_c = copy.deepcopy(universal_GramPos.reactions.rxn05318_c) \n",
    "iCdR700.add_reactions([rxn05318_c])\n",
    "iCdR700.reactions.rxn05318_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05318_c.annotation['seed.reaction'] = 'rxn05318'\n",
    "iCdR700.reactions.rxn05318_c.annotation['bigg.reaction'] = ['ADNt2','ADNt2pp','ADNt2r','ADNt2rpp','U174','U185','U191']\n",
    "cpd00182_e = iCdR700.metabolites.cpd00182_e\n",
    "iCdR700.add_boundary(cpd00182_e, type='exchange', reaction_id='EX_cpd00182_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00182_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxycytidine\n",
    "rxn05205_c = copy.deepcopy(universal_GramPos.reactions.rxn05205_c) \n",
    "iCdR700.add_reactions([rxn05205_c])\n",
    "iCdR700.reactions.rxn05205_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05205_c.annotation['seed.reaction'] = 'rxn05205'\n",
    "iCdR700.reactions.rxn05205_c.annotation['bigg.reaction'] = ['DCYTt2','DCYTt2pp','U183','U189','U199']\n",
    "cpd00654_e = iCdR700.metabolites.cpd00654_e\n",
    "iCdR700.add_boundary(cpd00654_e, type='exchange', reaction_id='EX_cpd00654_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00654_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxyuridine\n",
    "rxn09685_c = copy.deepcopy(universal_GramPos.reactions.rxn09685_c) \n",
    "iCdR700.add_reactions([rxn09685_c])\n",
    "iCdR700.reactions.rxn09685_c.name = 'Deoxyuridine symport'\n",
    "iCdR700.reactions.rxn09685_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn09685_c.annotation['seed.reaction'] = 'rxn09685'\n",
    "iCdR700.reactions.rxn09685_c.annotation['bigg.reaction'] = ['DURIt2','DURIt2pp','U184','U190','U200']\n",
    "cpd00412_e = iCdR700.metabolites.cpd00412_e\n",
    "iCdR700.add_boundary(cpd00412_e, type='exchange', reaction_id='EX_cpd00412_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00412_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxyadenosine\n",
    "rxn05317_c = copy.deepcopy(universal_GramPos.reactions.rxn05317_c) \n",
    "iCdR700.add_reactions([rxn05317_c])\n",
    "iCdR700.reactions.rxn05317_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05317_c.annotation['seed.reaction'] = 'rxn05317'\n",
    "iCdR700.reactions.rxn05317_c.annotation['bigg.reaction'] = ['DADNt2','DADNt2pp','U182','U188','U198']\n",
    "cpd00438_e = iCdR700.metabolites.cpd00438_e\n",
    "iCdR700.add_boundary(cpd00438_e, type='exchange', reaction_id='EX_cpd00438_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00438_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "#--------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Additional amino acids\n",
    "\n",
    "# Citrulline\n",
    "rxn05674_c = copy.deepcopy(universal_GramPos.reactions.rxn05674_c) \n",
    "iCdR700.add_reactions([rxn05674_c])\n",
    "iCdR700.reactions.rxn05674_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05674_c.annotation['seed.reaction'] = 'rxn05674'\n",
    "iCdR700.reactions.rxn05674_c.annotation['bigg.reaction'] = ['CITR_Lt2','CITR_Lt2ex','CITR_Lt2pp']\n",
    "cpd00274_e = iCdR700.metabolites.cpd00274_e\n",
    "iCdR700.add_boundary(cpd00274_e, type='exchange', reaction_id='EX_cpd00274_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00274_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# D-Glutamate\n",
    "rxn05672_c = copy.deepcopy(universal_GramPos.reactions.rxn05672_c) \n",
    "iCdR700.add_reactions([rxn05672_c])\n",
    "iCdR700.reactions.rxn05672_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn05672_c.annotation['seed.reaction'] = 'rxn05672'\n",
    "iCdR700.reactions.rxn05672_c.annotation['bigg.reaction'] = 'GLU-Dt2'\n",
    "cpd00186_e = iCdR700.metabolites.cpd00186_e\n",
    "iCdR700.add_boundary(cpd00186_e, type='exchange', reaction_id='EX_cpd00186_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00186_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Cystathionine\n",
    "cpd00424_e = cobra.Metabolite(\n",
    "    'cpd00424_e',\n",
    "    formula='C7H14N2O4S',\n",
    "    name='Cystathionine',\n",
    "    compartment='extracellular')\n",
    "cpd00424_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00424_e.annotation['seed.compound'] = 'cpd00424'\n",
    "cpd00424_e.annotation['bigg.metabolite'] = 'CST_c'\n",
    "cpd00424_e.annotation['kegg.compound'] = 'C00542'\n",
    "iCdR700.add_boundary(cpd00424_e, type='exchange', reaction_id='EX_cpd00424_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00424_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00424_c = iCdR700.metabolites.cpd00424_c\n",
    "rxn10398_c = cobra.Reaction('rxn10398_c')\n",
    "rxn10398_c.name = 'cystathione diffusion'\n",
    "rxn10398_c.bounds = (-1000., 1000.)\n",
    "rxn10398_c.add_metabolites({\n",
    "    cpd00424_e: -1.0,\n",
    "    cpd00424_c: 1.0 \n",
    "})\n",
    "rxn10398_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn10398_c.annotation['seed.reaction'] = 'rxn10398'\n",
    "rxn10398_c.annotation['bigg.reaction'] = 'CYSTtp'\n",
    "iCdR700.add_reactions([rxn10398_c])\n",
    "iCdR700.reactions.rxn10398_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdR700.reactions.rxn10398_c.annotation['seed.reaction'] = 'rxn10398'\n",
    "iCdR700.reactions.rxn10398_c.annotation['bigg.reaction'] = 'CYSTtp'\n",
    "\n",
    "# LL-2,6-Diaminopimelate\n",
    "cpd00504_e = cobra.Metabolite(\n",
    "    'cpd00504_e',\n",
    "    formula='C7H14N2O4',\n",
    "    name='LL-2,6-Diaminopimelate',\n",
    "    compartment='extracellular')\n",
    "cpd00504_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00504_e.annotation['seed.compound'] = 'cpd00504'\n",
    "cpd00504_e.annotation['bigg.metabolite'] = '26dap_LL'\n",
    "cpd00504_e.annotation['kegg.compound'] = 'C00666'\n",
    "iCdR700.add_boundary(cpd00504_e, type='exchange', reaction_id='EX_cpd00504_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00504_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00001_c = iCdR700.metabolites.cpd00001_c\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c\n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c\n",
    "cpd00009_c = iCdR700.metabolites.cpd00009_c\n",
    "cpd00067_c = iCdR700.metabolites.cpd00067_c\n",
    "cpd00504_c = iCdR700.metabolites.cpd00504_c\n",
    "\n",
    "ID011_c = cobra.Reaction('ID011_c')\n",
    "ID011_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "ID011_c.gene_reaction_rule = '272563.120.peg.772'\n",
    "ID011_c.bounds = (0., 1000.)\n",
    "ID011_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00504_e: -1.0,\n",
    "    cpd00504_c: 1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0 \n",
    "})\n",
    "iCdR700.add_reactions([ID011_c])\n",
    "iCdR700.reactions.ID011_c.annotation['sbo'] = 'SBO:0000185'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Guanine\n",
    "rxn05203_c = copy.deepcopy(universal_GramPos.reactions.rxn05203_c) \n",
    "rxn05203_c.name = 'Guanine symport'\n",
    "rxn05203_c.annotation['sbo'] = 'SBO:0000185'\n",
    "rxn05203_c.annotation['seed.reaction'] = 'rxn05203'\n",
    "rxn05203_c.annotation['bigg.reaction'] = ['FCY21_3','FCY22_3','FCY2_3','GUAt2','GUAt2pp','GUAt2r','YGL186C_3']\n",
    "\n",
    "iCdR700.add_reactions([rxn05203_c])\n",
    "cpd00207_e = iCdR700.metabolites.cpd00207_e\n",
    "iCdR700.add_boundary(cpd00207_e, type='exchange', reaction_id='EX_cpd00207_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00207_e.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_geneDict = {'272563.8.peg.2607': '645463.3.peg.2656',\n",
    "                  '272563.8.peg.2464': '645463.3.peg.2536',\n",
    "                  '272563.8.peg.2736': '645463.3.peg.2384',\n",
    "                  '272563.8.peg.1365': '645463.3.peg.1218'}\n",
    "cobra.manipulation.modify.rename_genes(iCdR700, final_geneDict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 700 genes\n",
      "18.93 minutes doubling time\n",
      "34.59% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% reactions are mass imbalanced\n",
      "15.27% reactions without GPRs\n",
      "24.71% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdR700_rmGenes = basicCheck(iCdR700)\n",
    "iCdR700_noGPRblocked = blockedReactions(iCdR700)\n",
    "iCdR700_free = checkFreeMass(iCdR700)\n",
    "iCdR700_massImbal = checkBalance(iCdR700)\n",
    "iCdR700_nogene = missingGPR(iCdR700)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700', media=mdm)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Integrate CCDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "r20291_genes = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/CdR20291.PATRIC.genes.format.fasta','r') as fasta:\n",
    "    for line in fasta:\n",
    "        r20291_genes[line.split('|')[0]] = line.split('|')[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes found: 700\n"
     ]
    }
   ],
   "source": [
    "success = 0\n",
    "for gene in iCdR700.genes:\n",
    "    try:\n",
    "        gene.annotation['ccds'] = r20291_genes[gene.id]\n",
    "        success += 1\n",
    "    except KeyError:\n",
    "            pass\n",
    "print('Genes found: ' + str(success))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700', media=mdm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdR700</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fa78371e9d0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1244</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1304</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdR700 at 0x7fa78371e9d0>"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdR700"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene Essentiality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# In GENRE, experimentally essential\n",
    "dembek_genes = set(['645463.3.peg.1000','645463.3.peg.1001','645463.3.peg.1002','645463.3.peg.1007',\n",
    "                '645463.3.peg.1016','645463.3.peg.1017','645463.3.peg.102','645463.3.peg.103',\n",
    "                '645463.3.peg.1032','645463.3.peg.1037','645463.3.peg.1057','645463.3.peg.1070',\n",
    "                '645463.3.peg.1077','645463.3.peg.1079','645463.3.peg.1084','645463.3.peg.1086',\n",
    "                '645463.3.peg.1089','645463.3.peg.1096','645463.3.peg.1110','645463.3.peg.1112',\n",
    "                '645463.3.peg.1125','645463.3.peg.1128','645463.3.peg.1129','645463.3.peg.1130',\n",
    "                '645463.3.peg.1176','645463.3.peg.118','645463.3.peg.119','645463.3.peg.1196',\n",
    "                '645463.3.peg.1199','645463.3.peg.1218','645463.3.peg.1223','645463.3.peg.1233',\n",
    "                '645463.3.peg.1240','645463.3.peg.881','645463.3.peg.898','645463.3.peg.904',\n",
    "                '645463.3.peg.905','645463.3.peg.906','645463.3.peg.907','645463.3.peg.909',\n",
    "                '645463.3.peg.910','645463.3.peg.916','645463.3.peg.918','645463.3.peg.919',\n",
    "                '645463.3.peg.920','645463.3.peg.929','645463.3.peg.930','645463.3.peg.954',\n",
    "                '645463.3.peg.955','645463.3.peg.956','645463.3.peg.957','645463.3.peg.969',\n",
    "                '645463.3.peg.971','645463.3.peg.976','645463.3.peg.979','645463.3.peg.980',\n",
    "                '645463.3.peg.981','645463.3.peg.994','645463.3.peg.998'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "190.1607048116364"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdR700.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iCdR700 essential genes: 102\n"
     ]
    }
   ],
   "source": [
    "# Computationally essential\n",
    "from cobra.flux_analysis import single_gene_deletion\n",
    "\n",
    "mdm = ['EX_cpd00001_e','EX_cpd00065_e','EX_cpd00060_e','EX_cpd00322_e','EX_cpd00129_e','EX_cpd00156_e',\n",
    "       'EX_cpd00107_e','EX_cpd00084_e','EX_cpd00149_e','EX_cpd00099_e','EX_cpd10515_e','EX_cpd00030_e', \n",
    "       'EX_cpd00254_e','EX_cpd00063_e','EX_cpd00205_e','EX_cpd00009_e','EX_cpd00971_e','EX_cpd00242_e',\n",
    "       'EX_cpd00104_e','EX_cpd00644_e','EX_cpd00263_e','EX_cpd00027_e']\n",
    "for rxn in iCdR700.boundary:\n",
    "    if rxn.id in mdm:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "\n",
    "deletion_results = single_gene_deletion(iCdR700)\n",
    "\n",
    "iCdR700_essential = []\n",
    "for x, row in deletion_results.iterrows():\n",
    "    if float(row.growth) < 1e-6:\n",
    "        iCdR700_essential.append((list(x)[0]))\n",
    "iCdR700_essential = set(iCdR700_essential)\n",
    "print('iCdR700 essential genes: ' + str(len(iCdR700_essential)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Essential genes in agreement: 16.95%\n"
     ]
    }
   ],
   "source": [
    "iCdR700_genes = set([x.id for x in iCdR700.genes]).intersection(dembek_genes)\n",
    "agree_essential = dembek_genes.intersection(iCdR700_essential)\n",
    "disagree_essential = dembek_genes.difference(iCdR700_essential)\n",
    "ratio = round((float(len(agree_essential)) / float(len(iCdR700_genes))) * 100., 2)\n",
    "print('Essential genes in agreement: ' + str(ratio) + '%')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total: 51\n",
      "645463.3.peg.1128\tphosphopentomutase (EC 5.4.2.7)\n",
      "645463.3.peg.1037\tlactoylglutathione lyase (EC 4.4.1.5)\n",
      "645463.3.peg.920\tNADP-dependent malic enzyme (EC 1.1.1.40)\n",
      "645463.3.peg.906\t3-isopropylmalate dehydratase small subunit (EC 4.2.1.33)\n",
      "645463.3.peg.1016\tPyruvate formate-lyase (EC 2.3.1.54)\n",
      "645463.3.peg.910\tD-3-phosphoglycerate dehydrogenase (EC 1.1.1.95)\n",
      "645463.3.peg.103\tMethionine aminopeptidase (EC 3.4.11.18)\n",
      "645463.3.peg.1007\tPTS system, galactitol-specific IIB component (EC 2.7.1.69)\n",
      "645463.3.peg.1070\tHomoserine dehydrogenase (EC 1.1.1.3)\n",
      "645463.3.peg.881\tAgmatinase (EC 3.5.3.11)\n",
      "645463.3.peg.1196\tCysteine desulfurase (EC 2.8.1.7)\n",
      "645463.3.peg.969\tN-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)\n",
      "645463.3.peg.1079\tAcetate kinase (EC 2.7.2.1)\n",
      "645463.3.peg.916\tAlkanesulfonates ABC transporter ATP-binding protein\n",
      "645463.3.peg.998\tPTS system, mannose-specific IIA component (EC 2.7.1.191)\n",
      "645463.3.peg.102\tAdenylate kinase (EC 2.7.4.3)\n",
      "645463.3.peg.1057\tMethylglyoxal synthase (EC 4.2.3.3)\n",
      "645463.3.peg.930\tGlucosamine-6-phosphate deaminase (EC 3.5.99.6)\n",
      "645463.3.peg.1218\tCytosol aminopeptidase PepA (EC 3.4.11.1)\n",
      "645463.3.peg.979\t3-hydroxybutyryl-CoA dehydratase (EC 4.2.1.55)\n",
      "645463.3.peg.919\tFumarate hydratase class I, beta region (EC 4.2.1.2); L(+)-tartrate dehydratase beta subunit (EC 4.2.1.32) | fumB\n",
      "645463.3.peg.1130\tpyrimidine-nucleoside phosphorylase (EC 2.4.2.2)\n",
      "645463.3.peg.956\tN-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)\n",
      "645463.3.peg.898\tphosphoserine phosphatase (EC 3.1.3.3)\n",
      "645463.3.peg.1096\tFructose-1,6-bisphosphatase, Bacillus type (EC 3.1.3.11)\n",
      "645463.3.peg.1223\tN-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)\n",
      "645463.3.peg.929\tN-acetylgalactosamine-6-phosphate deacetylase\n",
      "645463.3.peg.1240\tAspartokinase (EC 2.7.2.4)\n",
      "645463.3.peg.1077\tGlycolate dehydrogenase (EC 1.1.99.14), subunit GlcD\n",
      "645463.3.peg.1084\t3-oxoacyl-[acyl-carrier-protein] synthase, KASIII (EC 2.3.1.41)\n",
      "645463.3.peg.118\tN-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)\n",
      "645463.3.peg.904\t2-isopropylmalate synthase (EC 2.3.3.13)\n",
      "645463.3.peg.905\t3-isopropylmalate dehydratase large subunit (EC 4.2.1.33)\n",
      "645463.3.peg.994\tAspartate aminotransferase (EC 2.6.1.1)\n",
      "645463.3.peg.1129\tPurine nucleoside phosphorylase (EC 2.4.2.1)\n",
      "645463.3.peg.1017\tPyruvate formate-lyase activating enzyme (EC 1.97.1.4)\n",
      "645463.3.peg.955\tN-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)\n",
      "645463.3.peg.980\t3-hydroxybutyryl-CoA dehydrogenase (EC 1.1.1.157)\n",
      "645463.3.peg.907\t3-isopropylmalate dehydrogenase (EC 1.1.1.85)\n",
      "645463.3.peg.909\tSerine--pyruvate aminotransferase (EC 2.6.1.51)\n",
      "645463.3.peg.976\tElectron bifurcating butyryl-CoA dehydrogenase (NAD+, ferredoxin)\n",
      "645463.3.peg.981\tAcetyl-CoA acetyltransferase (EC 2.3.1.9)\n",
      "645463.3.peg.1125\tADP-ribose pyrophosphatase (EC 3.6.1.13)\n",
      "645463.3.peg.1002\tPTS system, mannose-specific IID component\n",
      "645463.3.peg.1086\tMalonyl CoA-acyl carrier protein transacylase (EC 2.3.1.39)\n",
      "645463.3.peg.957\tN-acetylmuramoyl-L-alanine amidase (EC 3.5.1.28)\n",
      "645463.3.peg.1001\tPTS system, mannose-specific IIC component\n",
      "645463.3.peg.119\tAspartate aminotransferase (EC 2.6.1.1)\n",
      "645463.3.peg.1000\tPTS system, mannose-specific IIB component (EC 2.7.1.191)\n",
      "645463.3.peg.1089\t3-oxoacyl-[acyl-carrier-protein] synthase, KASII (EC 2.3.1.41)\n",
      "645463.3.peg.918\tFumarate hydratase class I, alpha region (EC 4.2.1.2); L(+)-tartrate dehydratase alpha subunit (EC 4.2.1.32) | fumA\n"
     ]
    }
   ],
   "source": [
    "print('Total: ' + str(len(disagree_essential)))\n",
    "for gene in disagree_essential: print(gene + '\\t' + iCdR700.genes.get_by_id(gene).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "\n",
    "import copy\n",
    "\n",
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "universal_GramPos.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13783_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13784_c.remove_from_model(remove_orphans=True)\n",
    "#universal_GramPos.reactions.rxn11920_c.name = 'Poly-beta-hydroxybutyrate (PHB) synthease'\n",
    "universal_GramPos.reactions.rxn12188_c.name = '2-deoxyglucose-6-phosphatase'\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix bounds\n",
    "#iCdR700.reactions.rxn00926_c.bounds = (0.,1000.) # tough...\n",
    "iCdR700.reactions.rxn01021_c.bounds = (0.,1000.)\n",
    "iCdR700.reactions.rxn00143_c.bounds = (0.,1000.)\n",
    "\n",
    "# Adenine synthesis\n",
    "rxn00131_c = copy.deepcopy(universal_GramPos.reactions.rxn00131_c) \n",
    "iCdR700.add_reactions([rxn00131_c])\n",
    "iCdR700.reactions.rxn00131_c.bounds = (0., 1000.)\n",
    "iCdR700.reactions.rxn00131_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdR700.reactions.rxn00131_c.annotation['seed.reaction'] = 'rxn00131'\n",
    "iCdR700.reactions.rxn00131_c.annotation['bigg.reaction'] = 'AMP-NUCLEOSID-RXN'\n",
    "iCdR700.reactions.rxn00131_c.annotation['kegg.reaction'] = 'R00182'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix gene names\n",
    "iCdR700.genes.get_by_id('645463.3.peg.998').name = 'PTS system, mannose-specific IIA component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.1000').name = 'PTS system, mannose-specific IIB component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.1001').name = 'PTS system, mannose-specific IIC component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.1002').name = 'PTS system, mannose-specific IID component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.3332').name = 'PTS system, mannose-specific IIA component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.3330').name = 'PTS system, mannose-specific IID component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.3333').name = 'PTS system, mannose-specific IIB component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.467').name = 'PTS system, mannose-specific IID component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.303').name = 'PTS system, mannose-specific IIA component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.304').name = 'PTS system, mannose-specific IIB component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.466').name = 'PTS system, mannose-specific IIC component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.465').name = 'PTS system, mannose-specific IIB component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.306').name = 'PTS system, mannose-specific IIB component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.308').name = 'PTS system, mannose-specific IID component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.464').name = 'PTS system, mannose-specific IIA component (EC 2.7.1.191)'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.307').name = 'PTS system, mannose-specific IIC component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.3331').name = 'PTS system, mannose-specific IIC component'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.2802').name = 'Phosphoenolpyruvate-protein phosphotransferase of PTS system (EC 2.7.3.9) | ptsI'\n",
    "iCdR700.genes.get_by_id('645463.3.peg.3106').name = 'PTS system, N-acetylglucosamine-specific IIA component (EC 2.7.1.193) / PTS system, N-acetylglucosamine-specific IIB component (EC 2.7.1.193) / PTS system, N-acetylglucosamine-specific IIC component'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['EX_cpd00001_e','EX_cpd00065_e','EX_cpd00060_e','EX_cpd00322_e','EX_cpd00129_e','EX_cpd00156_e',\n",
    "       'EX_cpd00107_e','EX_cpd00084_e','EX_cpd00149_e','EX_cpd00099_e','EX_cpd10515_e','EX_cpd00030_e', \n",
    "       'EX_cpd00254_e','EX_cpd00063_e','EX_cpd00205_e','EX_cpd00009_e','EX_cpd00971_e','EX_cpd00242_e',\n",
    "       'EX_cpd00104_e','EX_cpd00644_e','EX_cpd00263_e','EX_cpd00027_e']\n",
    "for rxn in iCdR700.boundary:\n",
    "    if rxn.id in mdm:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (0., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n",
      "yep!\n"
     ]
    }
   ],
   "source": [
    "strR20291_only = ['645463.3.peg.56','645463.3.peg.57','645463.3.peg.58','645463.3.peg.59','645463.3.peg.217',\n",
    "               '645463.3.peg.237','645463.3.peg.239','645463.3.peg.257','645463.3.peg.258','645463.3.peg.259',\n",
    "               '645463.3.peg.261','645463.3.peg.262','645463.3.peg.326','645463.3.peg.327','645463.3.peg.328',\n",
    "               '645463.3.peg.348','645463.3.peg.357','645463.3.peg.358','645463.3.peg.359','645463.3.peg.381',\n",
    "               '645463.3.peg.382','645463.3.peg.400','645463.3.peg.404','645463.3.peg.409','645463.3.peg.410',\n",
    "               '645463.3.peg.432','645463.3.peg.434','645463.3.peg.436','645463.3.peg.457','645463.3.peg.532',\n",
    "               '645463.3.peg.535','645463.3.peg.536','645463.3.peg.538','645463.3.peg.541','645463.3.peg.542',\n",
    "               '645463.3.peg.543','645463.3.peg.545','645463.3.peg.550','645463.3.peg.551','645463.3.peg.552',\n",
    "               '645463.3.peg.555','645463.3.peg.556','645463.3.peg.561','645463.3.peg.582','645463.3.peg.592',\n",
    "               '645463.3.peg.604','645463.3.peg.621','645463.3.peg.632','645463.3.peg.650','645463.3.peg.659',\n",
    "               '645463.3.peg.663','645463.3.peg.724','645463.3.peg.742','645463.3.peg.768','645463.3.peg.927',\n",
    "               '645463.3.peg.936','645463.3.peg.967','645463.3.peg.1027','645463.3.peg.1030',\n",
    "               '645463.3.peg.1139','645463.3.peg.1140','645463.3.peg.1141','645463.3.peg.1145',\n",
    "               '645463.3.peg.1150','645463.3.peg.1157','645463.3.peg.1206','645463.3.peg.1342',\n",
    "               '645463.3.peg.1343','645463.3.peg.1355','645463.3.peg.1402','645463.3.peg.1433',\n",
    "               '645463.3.peg.1496','645463.3.peg.1497','645463.3.peg.1501','645463.3.peg.1503',\n",
    "               '645463.3.peg.1504','645463.3.peg.1506','645463.3.peg.1507','645463.3.peg.1508',\n",
    "               '645463.3.peg.1509','645463.3.peg.1510','645463.3.peg.1511','645463.3.peg.1512',\n",
    "               '645463.3.peg.1514','645463.3.peg.1515','645463.3.peg.1517','645463.3.peg.1518',\n",
    "               '645463.3.peg.1520','645463.3.peg.1521','645463.3.peg.1526','645463.3.peg.1533',\n",
    "               '645463.3.peg.1538','645463.3.peg.1545','645463.3.peg.1546','645463.3.peg.1558',\n",
    "               '645463.3.peg.1559','645463.3.peg.1561','645463.3.peg.1563','645463.3.peg.1564',\n",
    "               '645463.3.peg.1565','645463.3.peg.1570','645463.3.peg.1664','645463.3.peg.1676',\n",
    "               '645463.3.peg.1806','645463.3.peg.1859','645463.3.peg.1867','645463.3.peg.1868',\n",
    "               '645463.3.peg.1869','645463.3.peg.1870','645463.3.peg.1871','645463.3.peg.1872',\n",
    "               '645463.3.peg.1873','645463.3.peg.1874','645463.3.peg.1876','645463.3.peg.1877',\n",
    "               '645463.3.peg.1878','645463.3.peg.1879','645463.3.peg.1880','645463.3.peg.1881',\n",
    "               '645463.3.peg.1882','645463.3.peg.1883','645463.3.peg.1884','645463.3.peg.1885',\n",
    "               '645463.3.peg.1887','645463.3.peg.1892','645463.3.peg.1895','645463.3.peg.1898',\n",
    "               '645463.3.peg.1900','645463.3.peg.1901','645463.3.peg.1902','645463.3.peg.1904',\n",
    "               '645463.3.peg.1905','645463.3.peg.1906','645463.3.peg.1907','645463.3.peg.1908',\n",
    "               '645463.3.peg.2037','645463.3.peg.2056','645463.3.peg.2059','645463.3.peg.2085',\n",
    "               '645463.3.peg.2086','645463.3.peg.2093','645463.3.peg.2241','645463.3.peg.2344',\n",
    "               '645463.3.peg.2345','645463.3.peg.2426','645463.3.peg.2429','645463.3.peg.2430',\n",
    "               '645463.3.peg.2435','645463.3.peg.2453','645463.3.peg.2454','645463.3.peg.2651',\n",
    "               '645463.3.peg.2652','645463.3.peg.2677','645463.3.peg.2716','645463.3.peg.2866',\n",
    "               '645463.3.peg.2868','645463.3.peg.2869','645463.3.peg.2874','645463.3.peg.2877',\n",
    "               '645463.3.peg.2894','645463.3.peg.2923','645463.3.peg.2934','645463.3.peg.2966',\n",
    "               '645463.3.peg.2997','645463.3.peg.3073','645463.3.peg.3074','645463.3.peg.3075',\n",
    "               '645463.3.peg.3076','645463.3.peg.3077','645463.3.peg.3078','645463.3.peg.3116',\n",
    "               '645463.3.peg.3129','645463.3.peg.3131','645463.3.peg.3132','645463.3.peg.3133',\n",
    "               '645463.3.peg.3135','645463.3.peg.3157','645463.3.peg.3160','645463.3.peg.3165',\n",
    "               '645463.3.peg.3166','645463.3.peg.3167','645463.3.peg.3168','645463.3.peg.3169',\n",
    "               '645463.3.peg.3170','645463.3.peg.3171','645463.3.peg.3172','645463.3.peg.3173',\n",
    "               '645463.3.peg.3174','645463.3.peg.3176','645463.3.peg.3177','645463.3.peg.3178',\n",
    "               '645463.3.peg.3180','645463.3.peg.3181','645463.3.peg.3182','645463.3.peg.3183',\n",
    "               '645463.3.peg.3184','645463.3.peg.3185','645463.3.peg.3197','645463.3.peg.3213',\n",
    "               '645463.3.peg.3214','645463.3.peg.3248','645463.3.peg.3356','645463.3.peg.3357',\n",
    "               '645463.3.peg.3378','645463.3.peg.3380','645463.3.peg.3382','645463.3.peg.3475',\n",
    "               '645463.3.peg.3479','645463.3.peg.3480','645463.3.peg.3481','645463.3.peg.3482',\n",
    "               '645463.3.peg.3483','645463.3.peg.3484','645463.3.peg.3485','645463.3.peg.3486',\n",
    "               '645463.3.peg.3487','645463.3.peg.3488','645463.3.peg.3489','645463.3.peg.3490',\n",
    "               '645463.3.peg.3492','645463.3.peg.3493','645463.3.peg.3568','645463.3.peg.3665',\n",
    "               '645463.3.peg.3669','645463.3.peg.3677','645463.3.peg.3683','645463.3.peg.3686',\n",
    "               '645463.3.peg.3691']\n",
    "\n",
    "draft_PATRIC = cobra.io.load_json_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/CdR20291.draft.json')\n",
    "for cpd in draft_PATRIC.metabolites:\n",
    "    if cpd.compartment == 'c':\n",
    "        cpd.compartment = 'cytosol'\n",
    "    elif cpd.compartment == 'e':\n",
    "        cpd.compartment = 'extracellular'\n",
    "\n",
    "test_rxns = []\n",
    "for gene in strR20291_only:\n",
    "    try:\n",
    "        test_rxns += [x.id for x in draft_PATRIC.genes.get_by_id(gene).reactions]\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for x in test_rxns:\n",
    "    try:\n",
    "        rxn = iCdR700.reactions.get_by_id(x)\n",
    "    except:\n",
    "        print(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700.metabolites.cpd12255_c_c.id = 'cpd12255_c'\n",
    "iCdR700.reactions.rxn00139_c.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Missing exchange reactions\n",
    "iCdR700.metabolites.cpd11574_e.name = 'Molybdate'\n",
    "cpd11574_e = iCdR700.metabolites.cpd11574_e\n",
    "iCdR700.add_boundary(cpd11574_e, type='exchange', reaction_id='EX_cpd11574_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd11574_e.name = 'Molybdate exchange'\n",
    "iCdR700.reactions.EX_cpd11574_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "iCdR700.metabolites.cpd04097_e.name = 'Lead'\n",
    "cpd04097_e = iCdR700.metabolites.cpd04097_e\n",
    "iCdR700.add_boundary(cpd04097_e, type='exchange', reaction_id='EX_cpd04097_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd04097_e.name = 'Lead exchange'\n",
    "iCdR700.reactions.EX_cpd04097_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "iCdR700.metabolites.cpd01012_e.name = 'Cadmium 2+'\n",
    "cpd01012_e = iCdR700.metabolites.cpd01012_e\n",
    "iCdR700.add_boundary(cpd01012_e, type='exchange', reaction_id='EX_cpd01012_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd01012_e.name = 'Cadmium exchange'\n",
    "iCdR700.reactions.EX_cpd01012_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "iCdR700.metabolites.cpd00531_e.name = 'Mercury 2+'\n",
    "cpd00531_e = iCdR700.metabolites.cpd00531_e\n",
    "iCdR700.add_boundary(cpd00531_e, type='exchange', reaction_id='EX_cpd00531_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00531_e.name = 'Mercury exchange'\n",
    "iCdR700.reactions.EX_cpd00531_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "iCdR700.metabolites.cpd00058_e.name = 'Copper 2+'\n",
    "cpd00058_e = iCdR700.metabolites.cpd00058_e\n",
    "iCdR700.add_boundary(cpd00058_e, type='exchange', reaction_id='EX_cpd00058_e', lb=-1000.0, ub=1000.0)\n",
    "iCdR700.reactions.EX_cpd00058_e.name = 'Copper exchange'\n",
    "iCdR700.reactions.EX_cpd00058_e.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.sbml')\n",
    "cobra.io.save_json_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Testing for Possible Energy Generation Cycles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dissipation reactions\n",
    "\n",
    "cpd00009_c = iCdR700.metabolites.cpd00009_c # P\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c # ATP \n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c # ADP \n",
    "cpd00052_c = iCdR700.metabolites.cpd00052_c # CTP \n",
    "cpd00096_c = iCdR700.metabolites.cpd00096_c # CDP \n",
    "cpd00038_c = iCdR700.metabolites.cpd00038_c # GTP \n",
    "cpd00031_c = iCdR700.metabolites.cpd00031_c # GDP \n",
    "cpd00062_c = iCdR700.metabolites.cpd00062_c # UTP \n",
    "cpd00014_c = iCdR700.metabolites.cpd00014_c # UDP \n",
    "cpd00068_c = iCdR700.metabolites.cpd00068_c # ITP \n",
    "cpd00090_c = iCdR700.metabolites.cpd00090_c # IDP \n",
    "cpd00067_c = iCdR700.metabolites.cpd00067_c # H \n",
    "cpd00004_c = iCdR700.metabolites.cpd00004_c # NADH \n",
    "cpd00003_c = iCdR700.metabolites.cpd00003_c # NAD \n",
    "cpd00005_c = iCdR700.metabolites.cpd00005_c # NADPH\n",
    "cpd00006_c = iCdR700.metabolites.cpd00006_c # NADP \n",
    "cpd00982_c = iCdR700.metabolites.cpd00982_c # FADH2\n",
    "cpd00015_c = iCdR700.metabolites.cpd00015_c # FAD \n",
    "\n",
    "test_1 = cobra.Reaction('test_1')\n",
    "test_1.bounds = (0., 1000.)\n",
    "test_1.add_metabolites({ cpd00002_c: -1.0, cpd00008_c: 1.0, cpd00009_c: 1.0})\n",
    "\n",
    "test_2 = cobra.Reaction('test_2')\n",
    "test_2.bounds = (0., 1000.)\n",
    "test_2.add_metabolites({cpd00052_c: -1.0, cpd00096_c: 1.0, cpd00009_c: 1.0})\n",
    "\n",
    "test_3 = cobra.Reaction('test_3')\n",
    "test_3.bounds = (0., 1000.)\n",
    "test_3.add_metabolites({cpd00038_c: -1.0, cpd00031_c: 1.0, cpd00009_c: 1.0})\n",
    "\n",
    "test_4 = cobra.Reaction('test_4')\n",
    "test_4.bounds = (0., 1000.)\n",
    "test_4.add_metabolites({cpd00062_c: -1.0, cpd00014_c: 1.0, cpd00009_c: 1.0})\n",
    "\n",
    "test_5 = cobra.Reaction('test_5')\n",
    "test_5.bounds = (0., 1000.)\n",
    "test_5.add_metabolites({cpd00068_c: -1.0, cpd00090_c: 1.0, cpd00009_c: 1.0})\n",
    "\n",
    "test_6 = cobra.Reaction('test_6')\n",
    "test_6.bounds = (0., 1000.)\n",
    "test_6.add_metabolites({cpd00004_c: -1.0, cpd00003_c: 1.0, cpd00067_c: 1.0})\n",
    "\n",
    "test_7 = cobra.Reaction('test_7')\n",
    "test_7.bounds = (0., 1000.)\n",
    "test_7.add_metabolites({cpd00005_c: -1.0, cpd00006_c: 1.0, cpd00067_c: 1.0})\n",
    "\n",
    "test_8 = cobra.Reaction('test_8')\n",
    "test_8.bounds = (0., 1000.)\n",
    "test_8.add_metabolites({cpd00982_c: -1.0, cpd00015_c: 1.0, cpd00067_c: 2.0})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from copy import deepcopy\n",
    "\n",
    "model = deepcopy(iCdR700)\n",
    "model.add_reactions([test_1,test_2,test_3,test_4,test_5,test_6,test_7,test_8])\n",
    "for x in model.boundary: x.lower_bound = 0.\n",
    "test_rxns = ['test_1','test_2','test_3','test_4','test_5','test_6','test_7','test_8']\n",
    "edc = set()\n",
    "for test_rxn in test_rxns:\n",
    "    \n",
    "    model.objective = test_rxn\n",
    "    solution = model.optimize()\n",
    "    active_rxns = set([rxn.id for rxn in model.reactions if abs(solution.fluxes[rxn.id]) > 1e-6])\n",
    "    if len(active_rxns) > 0: edc |= active_rxns\n",
    "\n",
    "edc = edc.difference(set(test_rxns))\n",
    "edc = edc.difference(set([x.id for x in model.boundary]))\n",
    "print('Total reactions in possible EDCs:', len(edc))\n",
    "\n",
    "# Check reversibility \n",
    "reversible = []\n",
    "irreversible = []\n",
    "for rxn in edc:\n",
    "    if iCdR700.reactions.get_by_id(rxn).lower_bound == 0.:\n",
    "        irreversible.append(rxn)\n",
    "    elif iCdR700.reactions.get_by_id(rxn).upper_bound == 0.:\n",
    "        irreversible.append(rxn)\n",
    "    else:\n",
    "        reversible.append(rxn)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print('Total possible reversibility errors:', len(reversible))\n",
    "for x in reversible: print(x, iCdR700.reactions.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700.reactions.rxn00799_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn05655_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn01870_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00293_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00786_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00541_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn13150_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn02005_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn10036_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00606_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00542_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00519_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn10183_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn01211_c.lower_bound = 0.\n",
    "iCdR700.reactions.rxn00182_c.lower_bound = 0.\n",
    "\n",
    "iCdR700.reactions.rxn05209_c.name = 'Sodium hydrogen symporter'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove incorrect aerobic reactions \n",
    "iCdR700.remove_reactions(['rxn02304_c', 'rxn09180_c', 'rxn30075_c'])\n",
    "\n",
    "# Add other detoxification reaction\n",
    "cpd00001_c = iCdR700.metabolites.cpd00001_c # H2O\n",
    "cpd00023_c = iCdR700.metabolites.cpd00023_c # L-Glutamate\n",
    "cpd00003_c = iCdR700.metabolites.cpd00003_c # NAD\n",
    "cpd00004_c = iCdR700.metabolites.cpd00004_c # NADH\n",
    "cpd00024_c = iCdR700.metabolites.cpd00024_c # 2-Oxoglutarate\n",
    "cpd00067_c = iCdR700.metabolites.cpd00067_c # H+\n",
    "cpd00013_c = iCdR700.metabolites.cpd00013_c # Ammonia\n",
    "\n",
    "rxn30075_c = cobra.Reaction('rxn30075_c')\n",
    "rxn30075_c.name = 'Glutamate dehydrogenase (NAD)'\n",
    "rxn30075_c.gene_reaction_rule = '645463.3.peg.194'\n",
    "rxn30075_c.lower_bound = 0.\n",
    "rxn30075_c.upper_bound = 1000.\n",
    "rxn30075_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00023_c: -1.0,\n",
    "    cpd00024_c: 1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00013_c: 1.0\n",
    "})\n",
    "iCdR700.add_reactions([rxn30075_c])\n",
    "\n",
    "iCdR700.reactions.rxn30075_c.annotation['sbo'] = 'SBO:0000176' \n",
    "iCdR700.reactions.rxn30075_c.annotation['seed.reaction'] = 'rxn30075'\n",
    "iCdR700.reactions.rxn30075_c.annotation['bigg.reaction'] = 'GLUDx'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Incorrect auxotrophies\n",
    "\n",
    "# Correct proline auxotrophy\n",
    "iCdR700.remove_reactions(['rxn00471_c'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA transcription\n",
    "iCdR700.remove_reactions(['rna_rxn'])\n",
    "\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c # ATP\n",
    "cpd00052_c = iCdR700.metabolites.cpd00052_c # CTP\n",
    "cpd00062_c = iCdR700.metabolites.cpd00062_c # UTP\n",
    "cpd00038_c = iCdR700.metabolites.cpd00038_c # GTP\n",
    "cpd00001_c = iCdR700.metabolites.cpd00001_c # H2O\n",
    "\n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdR700.metabolites.cpd00009_c # Phosphate\n",
    "cpd00012_c = iCdR700.metabolites.cpd00012_c # PPi\n",
    "\n",
    "cpd17043_c = iCdR700.metabolites.cpd17043_c # RNA polymer\n",
    "\n",
    "rna_rxn = cobra.Reaction('rna_rxn')\n",
    "rna_rxn.name = 'RNA transcription'\n",
    "rna_rxn.lower_bound = 0.\n",
    "rna_rxn.upper_bound = 1000.\n",
    "rna_rxn.add_metabolites({\n",
    "    cpd00002_c: -1.389,\n",
    "    cpd00052_c: -0.237,\n",
    "    cpd00062_c: -0.5,\n",
    "    cpd00038_c: -0.141,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17043_c: 1.0,\n",
    "    cpd00008_c: 2.0,\n",
    "    cpd00009_c: 2.0,\n",
    "    cpd00012_c: 1.0\n",
    "})\n",
    "iCdR700.add_reactions([rna_rxn])\n",
    "iCdR700.reactions.rna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "\n",
    "# DNA replication\n",
    "iCdR700.remove_reactions(['dna_rxn'])\n",
    "\n",
    "cpd00115_c = iCdR700.metabolites.cpd00115_c # dATP\n",
    "cpd00356_c = iCdR700.metabolites.cpd00356_c # dCTP\n",
    "cpd00357_c = iCdR700.metabolites.cpd00357_c # TTP \n",
    "cpd00241_c = iCdR700.metabolites.cpd00241_c # dGTP\n",
    "cpd00002_c = iCdR700.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdR700.metabolites.cpd00001_c # H2O\n",
    "\n",
    "cpd00008_c = iCdR700.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdR700.metabolites.cpd00009_c # Phosphate\n",
    "cpd00012_c = iCdR700.metabolites.cpd00012_c # PPi\n",
    "\n",
    "cpd17042_c = iCdR700.metabolites.cpd17042_c # DNA polymer\n",
    "\n",
    "dna_rxn = cobra.Reaction('dna_rxn')\n",
    "dna_rxn.name = 'DNA replication'\n",
    "dna_rxn.lower_bound = 0.\n",
    "dna_rxn.upper_bound = 1000.\n",
    "dna_rxn.add_metabolites({\n",
    "    cpd00115_c: -0.5,\n",
    "    cpd00356_c: -0.176,\n",
    "    cpd00357_c: -0.454,\n",
    "    cpd00241_c: -0.308,\n",
    "    cpd00002_c: -2.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17042_c: 1.0,\n",
    "    cpd00008_c: 2.0,\n",
    "    cpd00009_c: 2.0,\n",
    "    cpd00012_c: 1.0\n",
    "})\n",
    "iCdR700.add_reactions([dna_rxn])\n",
    "iCdR700.reactions.dna_rxn.annotation['sbo'] = 'SBO:0000629'  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'iCdR700' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-5b0f0358fa0a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0miCdR700\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'iCdR700' is not defined"
     ]
    }
   ],
   "source": [
    "iCdR700"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.sbml')\n",
    "cobra.io.save_json_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.manipulation.delete import prune_unused_metabolites\n",
    "\n",
    "iCdG693, unused = prune_unused_metabolites(iCdG693)\n",
    "iCdR700, unused = prune_unused_metabolites(iCdR700)\n",
    "\n",
    "iCdG693.reactions.rxn05527_c.name = 'Cytosine symporter'\n",
    "iCdG693.reactions.rxn05204_c.name = 'Guanosine symporter'\n",
    "iCdG693.reactions.rxn05318_c.name = 'Adenosine symporter'\n",
    "iCdG693.reactions.rxn09687_c.name = 'Deoxyinosine symporter'\n",
    "iCdG693.reactions.rxn05871_c.name = 'Primary_alcohol:NAD+ oxidoreductase'\n",
    "\n",
    "iCdR700.reactions.rxn05527_c.name = 'Cytosine symporter'\n",
    "iCdR700.reactions.rxn05204_c.name = 'Guanosine symporter'\n",
    "iCdR700.reactions.rxn05318_c.name = 'Adenosine symporter'\n",
    "iCdR700.reactions.rxn09687_c.name = 'Deoxyinosine symporter'\n",
    "iCdR700.reactions.rxn05871_c.name = 'Primary_alcohol:NAD+ oxidoreductase'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG693 = cobra.io.load_json_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.json')\n",
    "iCdR700 = cobra.io.load_json_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')\n",
    "\n",
    "iCdG693.metabolites.cpd00419_c.name = 'Pyridoxamine'\n",
    "iCdR700.metabolites.cpd00419_c.name = 'Pyridoxamine'\n",
    "\n",
    "iCdG693.genes.get_by_id('272563.8.peg.262').name = 'Putative CDP-glycerol:Poly(Glycerophosphate) glycerophosphotransferase'\n",
    "\n",
    "# Fix residual R20291 gene\n",
    "import copy\n",
    "\n",
    "rxn30075_c = copy.deepcopy(iCdG693.reactions.rxn30075_c)\n",
    "cobra.manipulation.delete.remove_genes(iCdG693, [iCdG693.genes.get_by_id('645463.3.peg.194')])\n",
    "\n",
    "gene = cobra.Gene('272563.120.peg.195')\n",
    "rxn30075_c.gene_reaction_rule = '272563.120.peg.195'\n",
    "iCdG693.add_reactions([rxn30075_c])\n",
    "\n",
    "\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').name = 'NAD-specific glutamate dehydrogenase'\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').annotation['refseq'] = 'AJP09876.1'\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').annotation['refseq.locustag'] = 'CDIF630_00301'\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').annotation['ccds'] = 'ATGTCAGGAAAAGATGTAAATGTCTTCGAGATGGCGCAATCTCAAGTAAAAAATGCATGTGATAAATTAGGTATGGAACCAGCAGTTTATGAATTATTAAAAGAACCTATGAGAGTTATAGAGGTTTCAATTCCAGTAAAAATGGATGATGGTTCTATAAAAACTTTTAAAGGATTTAGATCACAACATAATGATGCAGTAGGGCCAACAAAAGGTGGAATAAGATTCCATCAAAATGTTTCAAGAGACGAAGTAAAAGCTTTATCTATATGGATGACTTTCAAGTGTTCTGTAACAGGTATACCATATGGTGGAGGTAAAGGTGGAATAATAGTAGATCCATCAACTTTATCTCAAGGTGAATTAGAAAGATTAAGTAGAGGATACATAGATGGAATATATAAATTAATAGGTGAGAAAGTTGACGTTCCTGCACCAGACGTAAATACTAATGGTCAAATAATGTCTTGGATGGTTGATGAGTACAATAAATTAACTGGACAAAGTTCTATAGGTGTTATAACTGGTAAGCCAGTTGAATTTGGTGGTTCTTTAGGAAGAACAGCTGCTACTGGATTTGGTGTTGCTGTTACTGCAAGAGAAGCTGCTGCTAAATTAGGAATAGATATGAAAAAAGCTAAAATAGCTGTTCAAGGTATAGGAAACGTTGGTTCTTATACAGTTCTTAACTGTGAAAAACTTGGTGGTACTGTTGTAGCTATGGCTGAATGGTGTAAATCAGAAGGTTCTTATGCTATATACAATGAAAATGGTTTAGATGGTCAAGCTATGTTAGATTATATGAAAGAACATGGTAACTTATTAAACTTCCCAGGAGCTAAGAGAATATCTTTAGAAGAGTTCTGGGCTTCAGATGTTGATATAGTAATACCAGCTGCATTAGAAAACTCTATAACTAAAGAAGTTGCTGAATCTATAAAAGCTAAATTAGTTTGTGAGGCTGCTAATGGACCAACTACTCCAGAGGCTGATGAAGTATTTGCTGAAAGAGGAATAGTTCTTACTCCAGATATATTAACTAATGCTGGTGGAGTTACAGTTTCTTACTTCGAGTGGGTACAAAACTTATATGGATACTACTGGTCAGAAGAAGAAGTAGAGCAAAAAGAAGAAATAGCTATGGTTAAAGCATTTGAATCTATTTGGAAAATCAAAGAAGAGTACAATGTTACAATGAGAGAAGCTGCTTACATGCATTCAATTAAGAAAGTTGCTGAAGCTATGAAATTAAGAGGATGGTACTAA'\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').annotation['uniprot'] = 'DHE2_CLODI'\n",
    "iCdG693.genes.get_by_id('272563.120.peg.195').annotation['ec_number'] = '1.4.1.2'\n",
    "\n",
    "\n",
    "cobra.io.write_sbml_model(iCdG693, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.sbml')\n",
    "cobra.io.save_json_model(iCdG693, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.json')\n",
    "cobra.io.write_sbml_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.sbml')\n",
    "cobra.io.save_json_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
