{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### PATRIC auto-generated model\n",
    "\n",
    "Annotations converted via mackinac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception reporting mode: Verbose\n"
     ]
    }
   ],
   "source": [
    "# Import libraries - REQUIRES pip version 9.0.3\n",
    "import pandas\n",
    "import os\n",
    "from os.path import join\n",
    "import sys\n",
    "import scipy.stats\n",
    "import numpy\n",
    "import math\n",
    "import pickle\n",
    "import copy\n",
    "import time\n",
    "import warnings\n",
    "import gc\n",
    "\n",
    "# Using Cobrapy 0.13.0\n",
    "import cobra\n",
    "import cobra.test\n",
    "import cobra.flux_analysis.gapfilling\n",
    "from cobra.io import write_sbml_model\n",
    "from cobra.flux_analysis import  flux_variability_analysis\n",
    "from cobra.flux_analysis.reaction import assess_component\n",
    "from cobra.manipulation.delete import *\n",
    "from cobra.flux_analysis.parsimonious import add_pfba\n",
    "from cobra.medium import find_boundary_types\n",
    "from cobra.util import solver as sutil\n",
    "from cobra.flux_analysis.sampling import OptGPSampler\n",
    "\n",
    "# Confidence levels in each annotation\n",
    "#import probanno\n",
    "\n",
    "# Import scripted python functions\n",
    "sys.path.append('code/python/')\n",
    "from read_excel import read_excel\n",
    "\n",
    "# Estabish handler for logger\n",
    "import logging\n",
    "logging.basicConfig()\n",
    "logger = logging.getLogger('logger')\n",
    "\n",
    "# Verbose exception printing\n",
    "%xmode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define functions\n",
    "\n",
    "# Identify potentially gapfilled reactions\n",
    "def findGapfilledRxn(model, exclude=[]):\n",
    "    gapfilled = []\n",
    "    transport = findTransports(model)\n",
    "    if not type(exclude) is list:\n",
    "        exclude = [exclude]\n",
    "        \n",
    "    for index in model.reactions:\n",
    "        if len(list(index.genes)) == 0:\n",
    "            if not index in model.boundary:\n",
    "                if not index.id in exclude or not index.id in transport:\n",
    "                    gapfilled.append(index.id)\n",
    "    \n",
    "    if len(gapfilled) > 0:\n",
    "        print(str(len(gapfilled)) + ' metabolic reactions not associated with genes')\n",
    "    \n",
    "    return gapfilled\n",
    "\n",
    "# Check for missing transport and exchange reactions\n",
    "def missingRxns(model, extracellular='e'):\n",
    "\n",
    "    transporters = set(findTransports(model))\n",
    "    exchanges = set([x.id for x in model.exchanges])\n",
    "    \n",
    "    missing_exchanges = []\n",
    "    missing_transports = []\n",
    "    \n",
    "    for metabolite in model.metabolites:\n",
    "        if not metabolite.compartment == extracellular:\n",
    "            continue\n",
    "\n",
    "        curr_rxns = set([x.id for x in list(metabolite.reactions)])\n",
    "        \n",
    "        if not bool(curr_rxns & transporters):\n",
    "            missing_transports.append(metabolite.id)\n",
    "        if not bool(curr_rxns & exchanges):\n",
    "            missing_exchanges.append(metabolite.id)\n",
    "    \n",
    "    if len(missing_transports) != 0:\n",
    "        print(str(len(missing_transports)) + ' extracellular metabolites are missing transport reactions')\n",
    "    if len(missing_exchanges) != 0:\n",
    "        print(str(len(missing_exchanges)) + ' extracellular metabolites are missing exchange reactions')\n",
    "        \n",
    "    return missing_transports, missing_exchanges\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(raw_model, cytosol='cytosol'):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        \n",
    "        # Close all exchanges\n",
    "        for index in model.boundary:\n",
    "            model.reactions.get_by_id(index.id).lower_bound = 0.\n",
    "        \n",
    "        # Identify all metabolites that are produced within the network\n",
    "        demand_metabolites = [x.reactants[0].id for x in model.demands if len(x.reactants) > 0]\n",
    "        demand_metabolites += [x.products[0].id for x in model.demands if len(x.products) > 0]\n",
    "\n",
    "        free = []\n",
    "        for index in model.metabolites: \n",
    "            if index.id in demand_metabolites:\n",
    "                continue\n",
    "            elif not index.compartment in cytosol:\n",
    "                continue\n",
    "            else:\n",
    "                demand = model.add_boundary(index, type='demand')\n",
    "                model.objective = demand\n",
    "                obj_val = model.slim_optimize(error_value=0.)\n",
    "                if obj_val > 1e-8:\n",
    "                    free.append(index.id)\n",
    "                model.remove_reactions([demand])\n",
    "    \n",
    "    if len(free) > 0:\n",
    "        print(str(len(free)) + ' metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(raw_model, exclude=[]):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        imbalanced = []\n",
    "        mass_imbal = 0\n",
    "        charge_imbal = 0\n",
    "        elem_set = set()\n",
    "        for metabolite in model.metabolites:\n",
    "            try:\n",
    "                elem_set |= set(metabolite.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        if len(elem_set) == 0:\n",
    "            imbalanced = model.reactions\n",
    "            mass_imbal = len(model.reactions)\n",
    "            charge_imbal = len(model.reactions)\n",
    "            print('No elemental data associated with metabolites!')\n",
    "        \n",
    "        else:\n",
    "            if not type(exclude) is list: \n",
    "                exclude = [exclude]\n",
    "            for index in model.reactions:\n",
    "                if index in model.boundary or index.id in exclude:\n",
    "                    continue\n",
    "\n",
    "                else:\n",
    "                    try:\n",
    "                        test = index.check_mass_balance()\n",
    "                    except ValueError:\n",
    "                        continue\n",
    "\n",
    "                    if len(list(test)) > 0:\n",
    "                        imbalanced.append(index.id)\n",
    "\n",
    "                        if 'charge' in test.keys():\n",
    "                            charge_imbal += 1\n",
    "                        if len(set(test.keys()).intersection(elem_set)) > 0:\n",
    "                            mass_imbal += 1\n",
    "\n",
    "    if mass_imbal != 0:\n",
    "        print(str(mass_imbal) + ' reactions are mass imbalanced')\n",
    "    if charge_imbal != 0:\n",
    "        print(str(charge_imbal) + ' reactions are charge imbalanced')\n",
    "    \n",
    "    return(imbalanced)\n",
    "\n",
    "\n",
    "# Checks the quality of models by a couple metrics and returns problems\n",
    "def checkQuality(model, exclude=[], cytosol='c'):\n",
    "    \n",
    "    start_time = time.time()\n",
    "    \n",
    "    if model.name != None:\n",
    "        model_name = model.name\n",
    "    else:\n",
    "        model_name = 'model'\n",
    "    \n",
    "    gaps = findGapfilledRxn(model, exclude)\n",
    "    freemass = checkFreeMass(model, cytosol)\n",
    "    balance = checkBalance(model, exclude)\n",
    "    trans, exch = missingRxns(model)\n",
    "    \n",
    "    test = gaps + freemass + balance\n",
    "    if len(test) == 0:\n",
    "        print('No inconsistencies detected')\n",
    "    \n",
    "    duration = int(round(time.time() - start_time))\n",
    "    print('Took ' + str(duration) + ' seconds to analyze ' + model_name) \n",
    "\n",
    "    return gaps, freemass, balance, trans, exch\n",
    "\n",
    "\n",
    "# Trace back through reactions immediately adjacent to a given reaction to identify blocked precursor synthesis\n",
    "def checkPrecursors(model, reaction):\n",
    "    \n",
    "    if isinstance(reaction, str) == True:\n",
    "        reaction = model.reactions.get_by_id(reaction)\n",
    "    \n",
    "    model.objective = reaction\n",
    "    obj_val = max(model.optimize(objective_sense='maximize').objective_value, abs(model.optimize(objective_sense='minimize').objective_value))\n",
    "\n",
    "    if obj_val > 0.001:\n",
    "        print('Able to produce all precursors for this reaction.')\n",
    "        return None\n",
    "    \n",
    "    else:\n",
    "        reactants = reaction.reactants\n",
    "        check = 0\n",
    "        \n",
    "        for reactant in reactants:\n",
    "            sub_reactions = list(reactant.reactions)\n",
    "            \n",
    "            for sub_reaction in sub_reactions:\n",
    "                model.objective = sub_reaction\n",
    "                obj_val = max(model.optimize(objective_sense='maximize').objective_value, abs(model.optimize(objective_sense='minimize').objective_value))\n",
    "\n",
    "                if obj_val < 0.001 and reactant in sub_reaction.products:                    \n",
    "                    print('Cannot acquire ' + str(reactant.id) + ' via ' + str(sub_reaction.id))\n",
    "                elif obj_val < 0.001 and check < 1 and reactant in sub_reaction.reactants:\n",
    "                    print(str(reactant.id) + ' not produced in any reactions.')\n",
    "                    check += 1\n",
    "\n",
    "#------------------------------------------------------------------------------------#\n",
    "\n",
    "# Function to calculate doubling time from objective value\n",
    "def doubling(model):\n",
    "    \n",
    "    with model as m:\n",
    "        ov = m.slim_optimize()\n",
    "        if ov < 1e-6:\n",
    "            growth = 'No growth'\n",
    "        else:\n",
    "            growth = (1 / float(m.slim_optimize())) * 3600\n",
    "        \n",
    "            if growth < 60.0:\n",
    "                growth = str(round(growth, 3)) + ' minutes'\n",
    "            else:\n",
    "                growth = growth / 60\n",
    "                growth = str(round(growth, 3)) + ' hours'\n",
    "            \n",
    "    return growth\n",
    "\n",
    "\n",
    "# Rough transcriptomic integration\n",
    "def roughContextualize(model, transcript_profile, condition):\n",
    "    \n",
    "    orig_OV = model.optimize().objective_value\n",
    "\n",
    "    model_context = copy.deepcopy(model)\n",
    "    abundances = []\n",
    "    \n",
    "    with open(transcript_profile, 'r') as transcription:\n",
    "        transcript_dict = {}\n",
    "        abundances = []\n",
    "        header = transcription.readline().strip().split(',')\n",
    "        column = header.index(condition)\n",
    "        for line in transcription:\n",
    "            line = line.split(',')\n",
    "            transcript_dict[line[0]] = float(line[column])\n",
    "            abundances.append(float(line[column]))\n",
    "    \n",
    "    min_transcription = numpy.percentile(abundances, 50)\n",
    "    penalty_bound = 10\n",
    "    \n",
    "    hits = 0\n",
    "    fails = 0\n",
    "    for gene in list(model_context.genes):\n",
    "    \n",
    "        gene = gene.name\n",
    "    \n",
    "        try:\n",
    "            curr_rxns = list(model_context.genes.get_by_id(gene).reactions)\n",
    "            hits += 1\n",
    "        except KeyError:\n",
    "            fails += 1\n",
    "            continue\n",
    "    \n",
    "        try:\n",
    "            curr_transcription = transcript_dict[gene]\n",
    "        except KeyError:\n",
    "            continue\n",
    "\n",
    "        for reaction in curr_rxns:\n",
    "            curr_id = reaction.id\n",
    "        \n",
    "            if curr_transcription >= min_transcription:            \n",
    "                model_context.reactions.get_by_id(curr_id).lower_bound = -1000\n",
    "                model_context.reactions.get_by_id(curr_id).upper_bound = 1000\n",
    "            elif curr_transcription < min_transcription:\n",
    "                if model_context.reactions.get_by_id(curr_id).lower_bound != 0:\n",
    "                    model_context.reactions.get_by_id(curr_id).lower_bound = -penalty_bound\n",
    "                if model_context.reactions.get_by_id(curr_id).upper_bound != 0:\n",
    "                    model_context.reactions.get_by_id(curr_id).upper_bound = penalty_bound\n",
    "                  \n",
    "\n",
    "    #print('Gene hits across data types: ' + str(hits))\n",
    "    #print('KeyErrors across data types: ' + str(fails) + '\\n')\n",
    "    new_OV = model_context.optimize().objective_value\n",
    "    print('New objective value: ' + str(new_OV))\n",
    "    print('Contextualized doubling time: ' + doubling(new_OV))\n",
    "    \n",
    "    return(model_context)\n",
    "\n",
    "\n",
    "# Checks for availability of reactants of a given reaction\n",
    "def availability(model, target_rxn):\n",
    "\n",
    "    precursors = model.reactions.get_by_id(target_rxn).reactants\n",
    "    \n",
    "    total = 0\n",
    "    unsuccessful = set()\n",
    "    limited = set()\n",
    "    for precursor in precursors:\n",
    "        precursor_rxn = list(model.metabolites.get_by_id(precursor.id).reactions)\n",
    "        \n",
    "        for rxn in precursor_rxn:\n",
    "            if rxn.id == target_rxn:\n",
    "                continue\n",
    "            elif precursor in model.reactions.get_by_id(rxn.id).reactants:\n",
    "                \n",
    "                model.objective = rxn\n",
    "                obj_val = model.slim_optimize()\n",
    "                if obj_val < 1e-8:\n",
    "                    unsuccessful |= set([rxn.id])\n",
    "                    limited |= set([precursor.id])\n",
    "    \n",
    "    print('Failed reactions: ' + str(len(unsuccessful)))\n",
    "    print('Limiting reactants: ' + str(len(limited)))\n",
    "    \n",
    "    return unsuccessful, limited\n",
    "\n",
    "# Removes all metabolites in a list of metabolite ids and all reactions associated with them\n",
    "def removeAll(model, metabolite_list):\n",
    "    \n",
    "    new_model = copy.deepcopy(model)\n",
    "    \n",
    "    for metabolite in metabolite_list:\n",
    "        try:\n",
    "            metabolite = new_model.metabolites.get_by_id(metabolite)\n",
    "            new_model.remove_reactions(metabolite.reactions)\n",
    "            new_model.remove_metabolites([metabolite])\n",
    "        except KeyError:\n",
    "            print(metabolite + ' not found')\n",
    "            continue\n",
    "        \n",
    "    return new_model\n",
    "\n",
    "\n",
    "# Identify transport reactions (for any number compartments)\n",
    "def findTransports(model):\n",
    "    transporters = []\n",
    "    compartments = set(list(model.compartments))\n",
    "    if len(compartments) == 1:\n",
    "        raise Exception('Model only has one compartment!')\n",
    "    \n",
    "    for reaction in model.reactions:\n",
    "        \n",
    "        reactant_compartments = set([x.compartment for x in reaction.reactants])\n",
    "        product_compartments = set([x.compartment for x in reaction.products])\n",
    "        reactant_baseID = set([x.id.split('_')[0] for x in reaction.reactants])\n",
    "        product_baseID = set([x.id.split('_')[0] for x in reaction.products])\n",
    "        \n",
    "        if reactant_compartments == product_compartments and reactant_baseID != product_baseID:\n",
    "            continue\n",
    "        elif bool(compartments & reactant_compartments) == True and bool(compartments & product_compartments) == True:\n",
    "            transporters.append(reaction.id)\n",
    "        \n",
    "    return(transporters)\n",
    "\n",
    "# Removes a given percentage of reactions from a model, ignoring objective\n",
    "def generate_gaps(model, percentage=0.2, prune=False, ignore=[]):\n",
    "    number_to_remove = int(round(len(model.reactions) * percentage))\n",
    "    rxn_ids = [x.id for x in model.reactions]\n",
    "    random.shuffle(rxn_ids)\n",
    "    rxns_to_remove = rxn_ids[-number_to_remove:]\n",
    "        \n",
    "    for rxn in ignore:\n",
    "        try:\n",
    "            rxns_to_remove.remove(rxn)\n",
    "        except ValueError:\n",
    "            continue\n",
    "    \n",
    "    truncated_model = copy.deepcopy(model)\n",
    "    truncated_model.remove_reactions(rxns_to_remove)\n",
    "    \n",
    "    if prune == True:\n",
    "        unused_cpds = prune_unused_metabolites(truncated_model)\n",
    "    \n",
    "    print('Reactions removed: ' + str(len(rxns_to_remove)))\n",
    "    print('New objective value: ' + str(truncated_model.slim_optimize()))\n",
    "    return truncated_model, rxns_to_remove\n",
    "\n",
    "\n",
    "# Calculates the sum of fluxes for a given model\n",
    "def sum_of_fluxes(model):\n",
    "    \n",
    "    with model as m:\n",
    "        solution = m.optimize()\n",
    "        flux_sum = sum(list(solution.fluxes))\n",
    "        \n",
    "    return flux_sum\n",
    "\n",
    "\n",
    "# reports the reaction ids that are in 1 model but not the other\n",
    "def rxn_diff(model_1, model_2):\n",
    "    diff = set([str(x.id) for x in model_1.reactions]).difference([str(y.id) for y in model_2.reactions])\n",
    "    return list(diff)\n",
    "\n",
    "# Check for missing transport and exchange reactions\n",
    "def missingRxn(model, extracellular='extracellular'):\n",
    "\n",
    "    transporters = set(findTransports(model))\n",
    "    exchanges = set([x.id for x in model.exchanges])\n",
    "    \n",
    "    missing_exchanges = []\n",
    "    missing_transports = []\n",
    "    \n",
    "    for metabolite in model.metabolites:\n",
    "        if not metabolite.compartment == extracellular:\n",
    "            continue\n",
    "\n",
    "        curr_rxns = set([x.id for x in list(metabolite.reactions)])\n",
    "        \n",
    "        if not bool(curr_rxns & transporters):\n",
    "            missing_transports.append(metabolite.id)\n",
    "        if not bool(curr_rxns & exchanges):\n",
    "            missing_exchanges.append(metabolite.id)\n",
    "    \n",
    "    if len(missing_transports) != 0:\n",
    "        print(str(len(missing_transports)) + ' extracellular metabolites are missing transport reactions')\n",
    "    if len(missing_exchanges) != 0:\n",
    "        print(str(len(missing_exchanges)) + ' extracellular metabolites are missing exchange reactions')\n",
    "        \n",
    "    return missing_transports, missing_exchanges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in ModelSEED generated model\n",
    "cd630_PATRIC = cobra.io.read_sbml_model('data/PATRIC/272563.8.xml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up counts for tracking changes\n",
    "init_rxn = [str(x.id) for x in cd630_PATRIC.reactions]\n",
    "\n",
    "new_genes = 0\n",
    "new_rxns = 0\n",
    "new_cpds = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "135 metabolic reactions not associated with genes\n",
      "1 reactions are mass imbalanced\n",
      "26 reactions are charge imbalanced\n",
      "Took 27 seconds to analyze Clostridium difficile 630\n"
     ]
    }
   ],
   "source": [
    "cd630_PATRIC_gaps, cd630_PATRIC_free, cd630_PATRIC_imbal, cd630_PATRIC_trans, cd630_PATRIC_exch = checkQuality(cd630_PATRIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cobra/core/model.py:613 \u001b[1;31mUserWarning\u001b[0m: bio1 not in 272563.8\n"
     ]
    }
   ],
   "source": [
    "# General curation\n",
    "\n",
    "# Ensure oxygen exchange and transport exists, but leave anaerobic\n",
    "cpd00532_e = cobra.Metabolite(\n",
    "    'cpd00532_e',\n",
    "    formula='O2',\n",
    "    name='O2-', \n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00532_e, type='exchange', reaction_id='EX_cpd00532_e', lb=0., ub=0.)\n",
    "cpd00532_c = cd630_PATRIC.metabolites.get_by_id('cpd00532_c')\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "\n",
    "rxn09030_c = cobra.Reaction('rxn09030_c')\n",
    "rxn09030_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn09030_c.name = 'O2 diffusion'\n",
    "rxn09030_c.lower_bound = 0.\n",
    "rxn09030_c.upper_bound = 1000.\n",
    "rxn09030_c.add_metabolites({\n",
    "    cpd00532_e: -1.0,\n",
    "    cpd00532_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn09030_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Ensure free water diffusion across compartments\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05319_c').bounds = (-1000., 1000.)\n",
    "\n",
    "# Rework ammonia correctly\n",
    "cd630_PATRIC.metabolites.get_by_id('cpd00013_c').name = 'Ammonium'\n",
    "#cd630_PATRIC.metabolites.get_by_id('cpd00013_c').formula = 'H3N'\n",
    "\n",
    "# Remove old bio1 (generic Gram-positive Biomass function) and macromolecule demand reactions\n",
    "cd630_PATRIC.remove_reactions(['rxn13783_c', 'rxn13784_c', 'rxn13782_c', 'bio1', 'SK_cpd11416_c'])\n",
    "\n",
    "# Fix compartment labels to be more specific\n",
    "for index in cd630_PATRIC.metabolites:\n",
    "    index.notes = {'family': '', 'annotation': 'ModelSEED'}\n",
    "    \n",
    "    if index.compartment == 'c':\n",
    "        index.compartment = 'cytosol'\n",
    "    elif index.compartment == 'e':\n",
    "        index.compartment = 'extracellular'\n",
    "\n",
    "# Add structured note section to each gene\n",
    "for index in cd630_PATRIC.genes:\n",
    "    index.notes = {'annotation': 'PATRIC'}\n",
    "\n",
    "# Parse and label reaction types (non-metabolic)\n",
    "#likelihoods = pickle.load(open('data/PATRIC/272563.8.probs'))\n",
    "for reaction in cd630_PATRIC.reactions:\n",
    "    index.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "    \n",
    "    if reaction in cd630_PATRIC.boundary:\n",
    "        reaction.notes['type'] = 'exchange'\n",
    "        continue\n",
    "    substrates = [x.id.split('_')[-1] for x in reaction.products] + [x.id.split('_')[-1] for x in reaction.reactants]\n",
    "    if 'e' in substrates and 'c' in substrates:\n",
    "        reaction.notes['type'] = 'transport'\n",
    "    \n",
    "    #try:\n",
    "    #    reaction.notes['confidence'] = likelihoods[reaction.id]\n",
    "    #except:\n",
    "    #    continue\n",
    "\n",
    "# Save 2 transport reactions that should have genes associated \n",
    "cd630_PATRIC.reactions.get_by_id('rxn05669_c').gene_reaction_rule = 'CD630_12590' # Valine\n",
    "rxn05669_c = cd630_PATRIC.reactions.get_by_id('rxn05669_c')\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05244_c').gene_reaction_rule = 'CD630_12590' # Isoleucine\n",
    "rxn05244_c = cd630_PATRIC.reactions.get_by_id('rxn05244_c')\n",
    "# Remove previously gapfilled metabolic reactions\n",
    "cd630_PATRIC.remove_reactions(cd630_PATRIC_gaps)\n",
    "cd630_PATRIC.add_reactions([rxn05669_c, rxn05244_c]) # Re-add saved reactions\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Deal with totally orphaned reactions that possess GPRs\n",
    "\n",
    "#  Protein serine <=> [Protein]-D-serine\n",
    "cd630_PATRIC.remove_reactions(['rxn06870_c'])\n",
    "\n",
    "#  Ecgonine <=> Pseudoecgonine\n",
    "cd630_PATRIC.remove_reactions(['rxn04561_c'])\n",
    "\n",
    "#  Isopenicillin N <=> Penicillin N\n",
    "cd630_PATRIC.remove_reactions(['rxn02896_c'])\n",
    "\n",
    "#  Violaxanthin <=> Neoxanthin and 9-cis-Neoxanthin <=> 9-cis-Violaxanthin\n",
    "cd630_PATRIC.remove_reactions(['rxn04757_c', 'rxn04760_c'])\n",
    "\n",
    "# alpha-D-Hexose 1-phosphate <=> alpha-D-Hexose 6-phosphate (very generic)\n",
    "cd630_PATRIC.remove_reactions(['rxn02379_c'])\n",
    "\n",
    "# 3-Lysolecithin <=> 2-Lysolecithin\n",
    "cd630_PATRIC.remove_reactions(['rxn02391_c'])\n",
    "\n",
    "#-----------------------------------------------#\n",
    "\n",
    "# Leaving 3 orphaned reactions with possible ties to known C. difficile metbolism\n",
    "\n",
    "# Lantibiotic resistance\n",
    "# rxn06517_c;  S-Alkyl-L-cysteine S-oxide <=> Dehydroalanine + Alkyl sulfenate\n",
    "\n",
    "# Peptidoglycan synthesis\n",
    "# rxn02975_c;  CDPparatose <=> CDPtyvelose\n",
    "#rxn02737_c;  UDP-glucosamine <=> UDP-galactosamine\n",
    "\n",
    "# Actually part of arabinose utilization\n",
    "# araD; L-ribulose-5-phosphate 4-epimerase; CD630_22780\n",
    "# rxn01292_c;  L-Arabinose <=> L-Ribulose\n",
    "\n",
    "#-----------------------------------------------#\n",
    "\n",
    "# Remove reactions with GPRs that are not actually present or detract from metabolic tasks\n",
    "\n",
    "# L-galactose utilization\n",
    "#cd630_PATRIC.remove_reactions(['rxn07493_c'])\n",
    "\n",
    "# Selenium utiliztion\n",
    "cd630_PATRIC.remove_reactions(['rxn03378_c','rxn02569_c'])\n",
    "\n",
    "# Molybdenum import\n",
    "cd630_PATRIC.remove_reactions(['rxn05619_c'])\n",
    "#cpd11574_e = cd630_PATRIC.metabolites.get_by_id('cpd11574_e')\n",
    "#cpd11574_c = cd630_PATRIC.metabolites.get_by_id('cpd11574_c')\n",
    "#cd630_PATRIC.remove_metabolites([cpd11574_e, cpd11574_c]) # Molybdate\n",
    "\n",
    "# Viomycin\n",
    "cd630_PATRIC.remove_reactions(['rxn02576_c'])\n",
    "\n",
    "#-----------------------------------------------#\n",
    "\n",
    "# Useless macromolecules\n",
    "cd630_PATRIC.remove_reactions(['rxn06197_c'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Components from minimal defined minimal media\n",
    "# All need to feed directly into biomass\n",
    "\n",
    "# Karasawa T, Ikoma S, Yamakawa K, Nakamura S. (1995).\n",
    "# A defined growth medium for Clostridium difficile.\n",
    "# Microbiology 141(2):371-375. \n",
    "# doi:10.1099/13500872-141-2-371\n",
    "\n",
    "# D-Glucose\n",
    "# cpd00027_e via rxn05226_c\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "# Minerals (already in place)\n",
    "\n",
    "# cobalt; cpd00149_e (Co2+)\n",
    "# manganese; cpd00030_e (Mn2+)\n",
    "# magnesium, cpd00254_e (Mg)\n",
    "# sodium; cpd00971_e (Na+)\n",
    "# calcium; cpd00063_e (Ca2+)\n",
    "# iron; cpd10515_e (Fe2+)\n",
    "# potassium; cpd00205_e (K+)\n",
    "# chloride, cpd00099_e (Cl-)\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "# Amino acids\n",
    "\n",
    "# Handled in Stickland fermentation section\n",
    "# isoleucine\n",
    "# proline\n",
    "# valine\n",
    "# leucine\n",
    "\n",
    "# Metabolites used in active tranport reactions\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "# Cysteine\n",
    "cpd00084_e = cobra.Metabolite(\n",
    "    'cpd00084_e',\n",
    "    formula='C3H7NO2S',\n",
    "    name='L-Cysteine', \n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00084_e, type='exchange', reaction_id='EX_cpd00084_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00084_c = cd630_PATRIC.metabolites.get_by_id('cpd00084_c') # L-Cysteine (cytosolic)\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "\n",
    "CD630_07500 = cobra.Gene('CD630_07500')\n",
    "CD630_07500.name = 'ABC-type transport system, aminoacid-family extracellular solute-binding protein'\n",
    "CD630_07500.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "\n",
    "rxn05237_c = cobra.Reaction('rxn05237_c')\n",
    "rxn05237_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05237_c.gene_reaction_rule = 'CD630_07500'\n",
    "rxn05237_c.notes['citations'].append('Karasawa T, Ikoma S, Yamakawa K, Nakamura S. Microbiology 141(2):371-375 doi:10.1099/13500872-141-2-371')\n",
    "rxn05237_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05237_c.lower_bound = 0.\n",
    "rxn05237_c.upper_bound = 1000.\n",
    "rxn05237_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00084_e: -1.0,\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0, # H+\n",
    "    cpd00084_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05237_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Trypotophan\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05663_c').lower_bound = 0.\n",
    "\n",
    "# Methionine\n",
    "# Correct in model already\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "# Vitamins\n",
    "\n",
    "# Biotin (BIOT) - exchange and transport already in model\n",
    "# cpd00104_e -> rxn05223_c -> cpd00104_c\n",
    "cd630_PATRIC.remove_reactions([cd630_PATRIC.reactions.get_by_id('rxn08180_c')]) # remove sythesis step\n",
    "\n",
    "# Pantothenate\n",
    "cpd00644_e = cobra.Metabolite(\n",
    "    'cpd00644_e',\n",
    "    formula='C9H16NO5',\n",
    "    name='PAN', # pantothenate\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00644_e, type='exchange', reaction_id='EX_cpd00644_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00644_c = cd630_PATRIC.metabolites.get_by_id('cpd00644_c') # pantothenate (cytosolic)\n",
    "\n",
    "# transport via panT - http://jb.asm.org/content/191/1/42.full\n",
    "R03018_c = cobra.Reaction('R03018_c')\n",
    "R03018_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'KEGG', 'subsystem': 'ABC transporter'}\n",
    "R03018_c.gene_reaction_rule = '272563.8.peg.164'\n",
    "R03018_c.notes['citations'].append('Karasawa T, Ikoma S, Yamakawa K, Nakamura S. Microbiology 141(2):371-375 doi:10.1099/13500872-141-2-371')\n",
    "R03018_c.name = 'ATP:pantothenate 4-phosphotransferase'\n",
    "R03018_c.lower_bound = 0.\n",
    "R03018_c.upper_bound = 1000.\n",
    "R03018_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00644_e: -1.0,\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0, # H+\n",
    "    cpd00644_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R03018_c])\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "\n",
    "# Pyridoxine\n",
    "cpd00263_e = cobra.Metabolite(\n",
    "    'cpd00263_e',\n",
    "    formula='C8H11NO3',\n",
    "    name='Pyridoxol', # Pyridoxine\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00263_e, type='exchange', reaction_id='EX_cpd00263_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00263_c = cd630_PATRIC.metabolites.get_by_id('cpd00263_c') # \n",
    "\n",
    "COG3601_c = cobra.Reaction('COG3601_c')\n",
    "COG3601_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'COG', 'subsystem': 'ABC transporter'}\n",
    "COG3601_c.gene_reaction_rule = '272563.8.peg.164'\n",
    "COG3601_c.notes['citations'].append('Karasawa T, Ikoma S, Yamakawa K, Nakamura S. Microbiology 141(2):371-375 doi:10.1099/13500872-141-2-371')\n",
    "COG3601_c.name = 'Pyridoxine active transporter'\n",
    "COG3601_c.lower_bound = 0.\n",
    "COG3601_c.upper_bound = 1000.\n",
    "COG3601_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00263_e: -1.0,\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0, # H+\n",
    "    cpd00263_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([COG3601_c])\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "# cpd00478_c\tPyridoxine phosphate (blocked) intracellular\n",
    "\n",
    "# Folate\n",
    "cpd00393_e = cd630_PATRIC.metabolites.get_by_id('cpd00393_e')\n",
    "cpd00393_c = cd630_PATRIC.metabolites.get_by_id('cpd00393_c')\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "\n",
    "rxn05255_c = cobra.Reaction('rxn05255_c')\n",
    "rxn05255_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn05255_c.name = 'Folate transport via proton simport'\n",
    "rxn05255_c.gene_reaction_rule = '272563.8.peg.164'\n",
    "rxn05255_c.lower_bound = 0.\n",
    "rxn05255_c.upper_bound = 1000.\n",
    "rxn05255_c.add_metabolites({\n",
    "    cpd00393_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00393_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05255_c])\n",
    "new_rxns += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TCA cycle - fragmented in strain 630\n",
    "cpd00106_c = cd630_PATRIC.metabolites.get_by_id('cpd00106_c') # Fumarate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00130_c = cd630_PATRIC.metabolites.get_by_id('cpd00130_c') # L-Malate\n",
    "\n",
    "CD630_10030 = cobra.Gene('CD630_10030') # Define gene\n",
    "CD630_10030.name = 'fumA; fumarate hydratase subunit A [KO:K01677] [EC:4.2.1.2]'\n",
    "CD630_10030.notes = {'annotation': 'UniProt'}\n",
    "CD630_10040 = cobra.Gene('CD630_10040') # Define gene\n",
    "CD630_10040.name = 'fumA; fumarate hydratase subunit B [KO:K01678] [EC:4.2.1.2]'\n",
    "CD630_10040.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "new_genes += 1\n",
    "\n",
    "rxn00799_c = cobra.Reaction('rxn00799_c') \n",
    "rxn00799_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'TCA cycle'}\n",
    "rxn00799_c.gene_reaction_rule = 'CD630_10030 and CD630_10040'\n",
    "rxn00799_c.notes['citations'] += ['Dannheim, H., Will, S. E., Schomburg, D., & Neumann‐Schaal, M. (2017). Clostridioides difficile 630Δerm in silico and in vivo – quantitative growth and extensive polysaccharide secretion. FEBS Open Bio, 7(4), 602–615.', 'Dannheim, H., Riedel, T., Neumann-Schaal, M., Bunk, B., Schober, I., Spröer, C., & Schomburg, D. (2017). Manual curation and reannotation of the genomes of Clostridium difficile 630Δerm and C. difficile 630. Journal of Medical Microbiology.']\n",
    "rxn00799_c.name = 'Fumarate hydratase'\n",
    "rxn00799_c.lower_bound = -1000.\n",
    "rxn00799_c.upper_bound = 1000.\n",
    "rxn00799_c.add_metabolites({\n",
    "    cpd00106_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00130_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00799_c])\n",
    "new_rxns += 1\n",
    "\n",
    "cpd00041_c = cd630_PATRIC.metabolites.get_by_id('cpd00041_c') # L-aspartate\n",
    "cpd00036_c = cd630_PATRIC.metabolites.get_by_id('cpd00036_c') # Succinate\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd03470_c = cd630_PATRIC.metabolites.get_by_id('cpd03470_c') # Iminoaspartate\n",
    "\n",
    "CD630_23710 = cobra.Gene('CD630_23710') # Define gene\n",
    "CD630_23710.name = 'L-aspartate oxidase (Quinolinate synthetase B)'\n",
    "CD630_23710.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "\n",
    "rxn08168_c = cobra.Reaction('rxn08168_c')\n",
    "rxn08168_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'TCA cycle'}\n",
    "rxn08168_c.gene_reaction_rule = 'CD630_23710'\n",
    "rxn08168_c.notes['citations'] += ['Dannheim, H., Will, S. E., Schomburg, D., & Neumann‐Schaal, M. (2017). Clostridioides difficile 630Δerm in silico and in vivo – quantitative growth and extensive polysaccharide secretion. FEBS Open Bio, 7(4), 602–615.', 'Dannheim, H., Riedel, T., Neumann-Schaal, M., Bunk, B., Schober, I., Spröer, C., & Schomburg, D. (2017). Manual curation and reannotation of the genomes of Clostridium difficile 630Δerm and C. difficile 630. Journal of Medical Microbiology.']\n",
    "rxn08168_c.name = 'L-aspartate:fumarate oxidoreductase (deaminating)'\n",
    "rxn08168_c.lower_bound = -1000.\n",
    "rxn08168_c.upper_bound = 1000.\n",
    "rxn08168_c.add_metabolites({\n",
    "    cpd00041_c: -1.0,\n",
    "    cpd00106_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd03470_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08168_c])\n",
    "new_rxns += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Oxygen detoxification\n",
    "\n",
    "# Turn off O2 exchange (anaerobc environment)\n",
    "cd630_PATRIC.reactions.get_by_id('EX_cpd00007_e').lower_bound = 0. \n",
    "\n",
    "# Diffusion reaction: rxn05468_c\n",
    "\n",
    "# Remove extraneous aerobic reactions \n",
    "cd630_PATRIC.remove_reactions(['rxn00333_c', 'rxn00262_c', 'rxn00338_c', 'rxn02303_c'])\n",
    "# Keep 'rxn00206_c', 'rxn01600_c', and 'rxn00333_c' due to possible presence for detoxification\n",
    "\n",
    "# Add other detoxification reactions\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00023_c = cd630_PATRIC.metabolites.get_by_id('cpd00023_c') # L-Glutamate\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00024_c = cd630_PATRIC.metabolites.get_by_id('cpd00024_c') # 2-Oxoglutarate\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c') # Ammonia\n",
    "\n",
    "CD630_01790 = cobra.Gene('CD630_01790')\n",
    "CD630_01790.name = 'gluD; Glutamate dehydrogenase (NAD)'\n",
    "CD630_01790.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "\n",
    "rxn30075_c = cobra.Reaction('rxn30075_c')\n",
    "rxn30075_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'Oxygen detoxification'}\n",
    "rxn30075_c.name = 'Glutamate dehydrogenase (NAD)'\n",
    "rxn30075_c.gene_reaction_rule = 'CD630_01790'\n",
    "rxn30075_c.lower_bound = -1000.\n",
    "rxn30075_c.upper_bound = 1000.\n",
    "rxn30075_c.notes['citations'] += 'Dannheim, H., Riedel, T., Neumann-Schaal, M., Bunk, B., Schober, I., Spröer, C., & Schomburg, D. (2017). Manual curation and reannotation of the genomes of Clostridium difficile 630Δerm and C. difficile 630. Journal of Medical Microbiology.'\n",
    "rxn30075_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00023_c: -1.0,\n",
    "    cpd00024_c: 1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00013_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn30075_c])\n",
    "new_rxns += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Nucleic acids\n",
    "\n",
    "# Thymine biosynthesis\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00184_c = cd630_PATRIC.metabolites.get_by_id('cpd00184_c') # Thymidine\n",
    "cpd00509_c = cd630_PATRIC.metabolites.get_by_id('cpd00509_c') # deoxyribose-1-phosphate\n",
    "cpd00151_c = cobra.Metabolite(\n",
    "    'cpd00151_c',\n",
    "    formula='C5H6N2O2',\n",
    "    name='Thymine',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_12250 = cobra.Gene('CD630_12250') \n",
    "CD630_12250.name = 'pdp; Pyrimidine-nucleoside phosphorylase'\n",
    "CD630_12250.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn27328_c = cobra.Reaction('rxn27328_c')\n",
    "rxn27328_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'Thymine biosynthesis'}\n",
    "rxn27328_c.gene_reaction_rule = 'CD630_12250'\n",
    "rxn27328_c.name = 'Thymidine-orthophosphate deoxyribosyltransferase'\n",
    "rxn27328_c.lower_bound = -1000.\n",
    "rxn27328_c.upper_bound = 1000.\n",
    "rxn27328_c.add_metabolites({\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00184_c: -1.0,\n",
    "    cpd00509_c: 1.0,\n",
    "    cpd00151_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn27328_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Future addition:\n",
    "# Linkage between toxin production and purine biosynthesis in Clostridium difficile\n",
    "# http://jmm.microbiologyresearch.org/content/journal/jmm/10.1099/0022-1317-51-1-34#tab2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Host-derived glycans\n",
    "\n",
    "# Stiemsma, L. T., Turvey, S. E., & Finlay, B. B. (2014). An antibiotic-altered microbiota \n",
    "# provides fuel for the enteric foe. Cell Research, 24(1), 5–6. http://doi.org/10.1038/cr.2013.142\n",
    "\n",
    "# N-Acetyl neuraminic acid import\n",
    "cpd00232_e = cobra.Metabolite(\n",
    "    'cpd00232_e',\n",
    "    formula='C11H18NO9',\n",
    "    name='Neu5Ac',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00232_e, type='exchange', reaction_id='EX_cpd00232_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "cpd00232_c = cd630_PATRIC.metabolites.get_by_id('cpd00232_c') # Neu5Ac (intracellular)\n",
    "\n",
    "CD630_24170 = cobra.Gene('CD630_24170') # Define gene\n",
    "CD630_24170.name = 'PTS system, Sorbitol-like IIB component'\n",
    "CD630_24170.notes = {'annotation': 'UniProt'}\n",
    "CD630_24180 = cobra.Gene('CD630_24180') # Define gene\n",
    "CD630_24180.name = 'PTS system, Sorbitol-like IIC component'\n",
    "CD630_24180.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn29571_c = cobra.Reaction('rxn29571_c')\n",
    "rxn29571_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'PTS system'}\n",
    "rxn29571_c.gene_reaction_rule = 'CD630_24170 and CD630_24180'\n",
    "rxn29571_c.name = 'Neu5Ac H+ symporter'\n",
    "rxn29571_c.lower_bound = -1000.\n",
    "rxn29571_c.upper_bound = 1000.\n",
    "rxn29571_c.add_metabolites({\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00232_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00232_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn29571_c])\n",
    "new_rxns += 1\n",
    "# North RA, Kessans SA, Griffin MDW, et al. Cloning, expression, purification, crystallization \n",
    "# and preliminary X-ray diffraction analysis of N-acetylmannosamine-6-phosphate 2-epimerase \n",
    "# from methicillin-resistant Staphylococcus aureus. Acta Crystallogr F Struct Biol Commun. \n",
    "# 2014;70:650–655. doi: 10.1107/S2053230X14007171.\n",
    "\n",
    "# Sialic Acid Catabolism in Staphylococcus aureus\n",
    "# Michael E. Olson, Jessica M. King, Timothy L. Yahr and Alexander R. Horswill \n",
    "# http://jb.asm.org/content/195/8/1779.full\n",
    "cpd00492_e = cobra.Metabolite(\n",
    "    'cpd00492_e',\n",
    "    formula='C8H15NO6',\n",
    "    name='N-Acetyl-D-mannosamine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00492_e, type='exchange', reaction_id='EX_cpd00492_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00492_c = cd630_PATRIC.metabolites.get_by_id('cpd00492_c') # N-Acetylmannosamine\n",
    "\n",
    "rxn05486_c = cobra.Reaction('rxn05486_c')\n",
    "rxn05486_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'PTS system'}\n",
    "rxn05486_c.gene_reaction_rule = 'CD630_24170 and CD630_24180'\n",
    "rxn05486_c.name = 'N-Acetylmannosamine H+ symporter'\n",
    "rxn05486_c.lower_bound = 0.\n",
    "rxn05486_c.upper_bound = 1000.\n",
    "rxn05486_c.add_metabolites({\n",
    "    cpd00492_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00492_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05486_c])\n",
    "new_rxns += 1\n",
    "# # N-Acetyl-D-glucosamine -> GlcN conversion (nagA)\n",
    "cpd00293_c = cd630_PATRIC.metabolites.get_by_id('cpd00293_c') # N-Acetyl-D-glucosamine 6-phosphate\n",
    "cpd00288_c = cd630_PATRIC.metabolites.get_by_id('cpd00288_c') # D-Glucosamine 6-phosphate\n",
    "cpd00029_c = cd630_PATRIC.metabolites.get_by_id('cpd00029_c') # Acetate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # AH2O\n",
    "\n",
    "CD630_10100 = cobra.Gene('CD630_10100') # Define gene\n",
    "CD630_10100.name = 'nagA, N-Acetyl-D-glucosamine-6-phosphate amidohydrolase'\n",
    "CD630_10100.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn01484_c = cobra.Reaction('rxn01484_c') # GlcNAc 6-P deacetylase\n",
    "rxn01484_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01484_c.gene_reaction_rule = 'CD630_10100'\n",
    "rxn01484_c.name = 'N-Acetyl-D-glucosamine-6-phosphate amidohydrolase'\n",
    "rxn01484_c.lower_bound = 0.\n",
    "rxn01484_c.upper_bound = 1000.\n",
    "rxn01484_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00293_c: -1.0,\n",
    "    cpd00288_c: 1.0,\n",
    "    cpd00029_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01484_c])\n",
    "new_rxns += 1\n",
    "# Engevik MA, Yacyshyn MB, Engevik KA, et al. Human Clostridium difficile infection: \n",
    "# altered mucus production and composition. American Journal of Physiology - \n",
    "# Gastrointestinal and Liver Physiology. 2015;308(6):G510-G524. doi:10.1152/ajpgi.00091.2014.\n",
    "\n",
    "\n",
    "CD630_30130 = cobra.Gene('CD630_30130')\n",
    "CD630_30130.name = 'PTS system, IIC component'\n",
    "CD630_30130.notes = {'annotation': 'UniProt'}\n",
    "CD630_30140 = cobra.Gene('CD630_30134')\n",
    "CD630_30140.name = 'PTS system, IIB component'\n",
    "CD630_30140.notes = {'annotation': 'UniProt'}\n",
    "CD630_30150 = cobra.Gene('CD630_30130')\n",
    "CD630_30150.name = 'PTS system, IIA component'\n",
    "CD630_30150.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "cpd00751_e = cobra.Metabolite(\n",
    "    'cpd00751_e',\n",
    "    formula='C6H12O5',\n",
    "    name='L-Fucose',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00751_e, type='exchange', reaction_id='EX_cpd00751_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02262_c').remove_from_model() # remove incorrect transport reaction\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd01912_c = cd630_PATRIC.metabolites.get_by_id('cpd01912_c') # L-Fucose 1-phosphate\n",
    "\n",
    "rxn02262_c = cobra.Reaction('rxn02262_c')\n",
    "rxn02262_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn02262_c.gene_reaction_rule = '272563.8.peg.2384'\n",
    "rxn02262_c.name = 'ATP:6-deoxy-L-galactose 1-phosphotransferase'\n",
    "rxn02262_c.lower_bound = 0.\n",
    "rxn02262_c.upper_bound = 1000.\n",
    "rxn02262_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00751_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd01912_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn02262_c])\n",
    "new_rxns += 1\n",
    "# Fucose utilization\n",
    "cpd00334_c = cobra.Metabolite(\n",
    "    'cpd00334_c',\n",
    "    formula='C3H6O2',\n",
    "    name='L-Lactaldehyde',\n",
    "    compartment='cytosol')\n",
    "cpd00095_c = cd630_PATRIC.metabolites.get_by_id('cpd00095_c') # Glycerone-phosphate\n",
    "\n",
    "CD630_00450 = cobra.Gene('CD630_00450')\n",
    "CD630_00450.name = 'L-fuculose-phosphate aldolase [EC:4.1.2.17]'\n",
    "CD630_00450.notes = {'annotation': 'UniProt'}\n",
    "CD630_22770 = cobra.Gene('CD630_22770')\n",
    "CD630_22770.name = 'L-fuculose-phosphate aldolase [EC:4.1.2.17]'\n",
    "CD630_22770.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn01620_c = cobra.Reaction('rxn01620_c')\n",
    "rxn01620_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01620_c.gene_reaction_rule = 'CD630_00450 or CD630_22770'\n",
    "rxn01620_c.name = 'L-Fuculose-1-phosphate lactaldehyde-lyase'\n",
    "rxn01620_c.lower_bound = -1000.\n",
    "rxn01620_c.upper_bound = 1000.\n",
    "rxn01620_c.add_metabolites({\n",
    "    cpd01912_c: -1.0,\n",
    "    cpd00095_c: 1.0,\n",
    "    cpd00334_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01620_c])\n",
    "new_rxns += 1\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00428_c = cd630_PATRIC.metabolites.get_by_id('cpd00428_c') # 2-Oxopropanal \n",
    "\n",
    "CD630_35150 = cobra.Gene('CD630_35150')\n",
    "CD630_35150.name = 'bifunctional UDP-N-acetylglucosamine pyrophosphorylase / Glucosamine-1-phosphate N-acetyltransferase [EC:2.7.7.23 2.3.1.157]'\n",
    "CD630_35150.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn04937_c = cobra.Reaction('rxn04937_c')\n",
    "rxn04937_c.notes = {'citations': ['North, R. A., et al.. (2018). “Just a spoonful of sugar...”: import of sialic acid across bacterial cell membranes. Biophysical Reviews, 10(2), 219–227.'], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn04937_c.gene_reaction_rule = 'CD630_35150'\n",
    "rxn04937_c.name = 'Lactaldehyde:NAD+ oxidoreductase'\n",
    "rxn04937_c.lower_bound = -1000.\n",
    "rxn04937_c.upper_bound = 1000.\n",
    "rxn04937_c.add_metabolites({\n",
    "    cpd00334_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00428_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn04937_c])\n",
    "new_rxns += 1\n",
    "# Galactose omitted because of Nakamura, et al. (1982).\n",
    "\n",
    "# Mannose import and utilization already intact...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Carbohydrate import (fermentation steps taken care of in SCFA steps)\n",
    "# Nakamura, et al. (1982). Carbohydrate Fermentation by Clostridium difficile. Microbial. Immunol. 26 (2); 107-111.\n",
    "\n",
    "# Already included:\n",
    "# D-Fructose \tC6H12O6 \tcpd00082\n",
    "# D-Glucose \tC6H12O6 \tcpd00027\n",
    "# D-Mannitol \tC6H14O6 \tcpd00314\n",
    "# Sorbitol \tC6H14O6 \tcpd00588\n",
    "# D-Ribose \tC5H10O5 \tcpd00105 \n",
    "# Salicin \tC13H18O7 \tcpd01030\n",
    "# Trehalose (TRHL) \tC12H22O11 \tcpd00794_e \n",
    "\n",
    "cpd05161_e = cobra.Metabolite(\n",
    "    'cpd05161_e',\n",
    "    formula='C18H32O16',\n",
    "    name='Melezitose',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd05161_e, type='exchange', reaction_id='EX_cpd05161_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd05161_c = cobra.Metabolite(\n",
    "    'cpd05161_c',\n",
    "    formula='C18H32O16',\n",
    "    name='Melezitose',\n",
    "    compartment='cytosol')\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "\n",
    "rxn12619_c = cobra.Reaction('rxn12619_c')\n",
    "rxn12619_c.gene_reaction_rule = 'CD630_24170 and CD630_24180'\n",
    "rxn12619_c.name = 'Melezitose transport via proton symport'\n",
    "rxn12619_c.lower_bound = -1000.\n",
    "rxn12619_c.upper_bound = 1000.\n",
    "rxn12619_c.add_metabolites({\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd05161_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd05161_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn12619_c])\n",
    "new_rxns += 1\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "rxn18524_c = cobra.Reaction('rxn18524_c')\n",
    "rxn01620_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01620_c.notes['citations'].append('Nakamura, et al. (1982). Carbohydrate Fermentation by Clostridium difficile. Microbial. Immunol. 26 (2); 107-111.')\n",
    "rxn18524_c.name = 'ATP phosphohydrolase (disaccharide-importing)'\n",
    "rxn18524_c.gene_reaction_rule = 'CD630_30130 and CD630_30140 and CD630_30150'\n",
    "rxn18524_c.lower_bound = 0.\n",
    "rxn18524_c.upper_bound = 1000.\n",
    "rxn18524_c.add_metabolites({\n",
    "    cpd05161_e: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd05161_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn18524_c])\n",
    "new_rxns += 1\n",
    "T4BV06_CLODI = cobra.Gene('T4BV06_CLODI')\n",
    "T4BV06_CLODI.name = 'melA; alpha-galactosidase [EC 3.2.1.22]'\n",
    "T4BV06_CLODI.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "# Cleavage to glucose and turanose the latter of which is an isomer of sucrose. \n",
    "cpd20885_c = cobra.Metabolite(\n",
    "    'cpd20885_c',\n",
    "    formula='C12H22O11',\n",
    "    name='Turanose',\n",
    "    compartment='cytosol')\n",
    "cpd00027_c = cd630_PATRIC.metabolites.get_by_id('cpd00027_c') # D-Glucose\n",
    "\n",
    "R01103_c = cobra.Reaction('R01103_c')\n",
    "R01103_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01103_c.notes['citations'].append('Nakamura, et al. (1982). Carbohydrate Fermentation by Clostridium difficile. Microbial. Immunol. 26 (2); 107-111.')\n",
    "R01103_c.name = 'Raffinose galactohydrolase'\n",
    "R01103_c.gene_reaction_rule = 'T4BV06_CLODI'\n",
    "R01103_c.lower_bound = 0.\n",
    "R01103_c.upper_bound = 1000.\n",
    "R01103_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd05161_c: -1.0,\n",
    "    cpd00027_c: 1.0,\n",
    "    cpd20885_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01103_c])\n",
    "new_rxns += 1\n",
    "# Turanose isomerization to sucrose - spontaneous\n",
    "cpd00076_c = cd630_PATRIC.metabolites.get_by_id('cpd00076_c') # Sucrose\n",
    "ID010_c = cobra.Reaction('ID010_c')\n",
    "ID010_c.name = 'Turanose isomerization'\n",
    "ID010_c.lower_bound = -1000.\n",
    "ID010_c.upper_bound = 1000.\n",
    "ID010_c.add_metabolites({\n",
    "    cpd20885_c: -1.0,\n",
    "    cpd00076_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID010_c])\n",
    "new_rxns += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Amino acid exchange and transport\n",
    "\n",
    "# Branched chain - facilitated diffusion\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+ (extracellular)\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+ (cytosolic)\n",
    "\n",
    "# Leucine\n",
    "cpd00107_e = cobra.Metabolite(\n",
    "    'cpd00107_e',\n",
    "    formula='C6H13NO2',\n",
    "    name='L-Leucine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00107_e, type='exchange', reaction_id='EX_cpd00107_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00107_c = cd630_PATRIC.metabolites.get_by_id('cpd00107_c') # Leucine (cytosol)\n",
    "\n",
    "CD630_12590 = cobra.Gene('CD630_12590')\n",
    "CD630_12590.name = 'brnQ-1; Branched-chain amino acid transport system carrier protein'\n",
    "CD630_12590.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn05243_c = cobra.Reaction('rxn05243_c')\n",
    "rxn05243_c.gene_reaction_rule = 'CD630_12590'\n",
    "rxn05243_c.name = 'L-Leucine hydrogen symport'\n",
    "rxn05243_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn05243_c.notes['citations'] += 'Driessen, A. J., Ubbink-Kok, T., & Konings, W. N. (1988). Amino acid transport by membrane vesicles of an obligate anaerobic bacterium, Clostridium acetobutylicum. Journal of Bacteriology, 170(2), 817–820.'\n",
    "rxn05243_c.lower_bound = 0.\n",
    "rxn05243_c.upper_bound = 1000.\n",
    "rxn05243_c.add_metabolites({\n",
    "    cpd00107_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00107_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05243_c])\n",
    "new_rxns += 1\n",
    "# Valine\n",
    "# Extracellular exchange and transport in place, needs correct GPR\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05669_c').gene_reaction_rule = 'CD630_12590'\n",
    "\n",
    "# Isoleucine\n",
    "# Extracellular exchange and tranport in place, needs correct GPR\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05244_c').gene_reaction_rule = 'CD630_12590'\n",
    "\n",
    "#----------------#\n",
    "\n",
    "# Aromatic - active transport\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "CD630_08730 = cobra.Gene('CD630_08730')\n",
    "CD630_08730.name = 'putative ABC transport system substrate-binding protein'\n",
    "CD630_08730.notes = {'annotation': 'KEGG'}\n",
    "CD630_08760 = cobra.Gene('CD630_08760')\n",
    "CD630_08760.name = 'putative ABC transport system substrate-binding protein'\n",
    "CD630_08760.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "\n",
    "# Tyrosine\n",
    "cpd00069_e = cobra.Metabolite(\n",
    "    'cpd00069_e',\n",
    "    formula='C9H11NO3',\n",
    "    name='L-Tyrosine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00069_e, type='exchange', reaction_id='EX_cpd00069_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00069_c = cd630_PATRIC.metabolites.get_by_id('cpd00069_c') # Tyrosine (cytosolic)\n",
    "\n",
    "rxn18579_c = cobra.Reaction('rxn18579_c') \n",
    "rxn18579_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn18579_c.notes['citations'] += 'Steglich M., et al. (2018). Convergent Loss of ABC Transporter Genes From Clostridioides difficile Genomes Is Associated With Impaired Tyrosine Uptake and p-Cresol Production. Frontiers in Microbiology. doi 10.3389/fmicb.2018.00901.'\n",
    "rxn05243_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn18579_c.name = 'L-Tyrosine ABC transport'\n",
    "rxn18579_c.lower_bound = 0.\n",
    "rxn18579_c.upper_bound = 1000.\n",
    "rxn18579_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00069_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00069_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn18579_c])\n",
    "new_rxns += 1\n",
    "# Phenylalanine\n",
    "cpd00066_e = cobra.Metabolite(\n",
    "    'cpd00066_e',\n",
    "    formula='C9H11NO2',\n",
    "    name='L-Phenylalanine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00066_e, type='exchange', reaction_id='EX_cpd00066_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00066_c = cd630_PATRIC.metabolites.get_by_id('cpd00066_c')\n",
    "\n",
    "rxn18584_c = cobra.Reaction('rxn18584_c') \n",
    "rxn18584_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn18584_c.notes['citations'] += 'Steglich M., et al. (2018). Convergent Loss of ABC Transporter Genes From Clostridioides difficile Genomes Is Associated With Impaired Tyrosine Uptake and p-Cresol Production. Frontiers in Microbiology. doi 10.3389/fmicb.2018.00901.'\n",
    "rxn18584_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn18584_c.name = 'nonpolar-amino-acid-transporting ATPase'\n",
    "rxn18584_c.lower_bound = 0.\n",
    "rxn18584_c.upper_bound = 1000.\n",
    "rxn18584_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00066_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00066_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn18584_c])\n",
    "new_rxns += 1\n",
    "#----------------#\n",
    "\n",
    "# Hydrophobic - active transport\n",
    "\n",
    "# Proline (D and L)\n",
    "cpd00129_e = cobra.Metabolite(\n",
    "    'cpd00129_e',\n",
    "    formula='C5H9NO2',\n",
    "    name='L-Proline',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00129_e, type='exchange', reaction_id='EX_cpd00129_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00129_c = cd630_PATRIC.metabolites.get_by_id('cpd00129_c') # L-Proline\n",
    "# http://www.jbc.org/content/249/21/6965.full.pdf\n",
    "\n",
    "rxn05165_c = cobra.Reaction('rxn05165_c') \n",
    "rxn05165_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05165_c.notes['citations'] += ''\n",
    "rxn05165_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn05165_c.name = 'nonpolar-amino-acid-transporting ATPase'\n",
    "rxn05165_c.lower_bound = 0.\n",
    "rxn05165_c.upper_bound = 1000.\n",
    "rxn05165_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00129_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00129_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05165_c])\n",
    "new_rxns += 1\n",
    "cpd00567_e = cobra.Metabolite(\n",
    "    'cpd00567_e',\n",
    "    formula='C5H9NO2',\n",
    "    name='D-Proline',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00567_e, type='exchange', reaction_id='EX_cpd00567_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00567_c = cd630_PATRIC.metabolites.get_by_id('cpd00567_c') # D-Proline\n",
    "cpd00971_e = cd630_PATRIC.metabolites.get_by_id('cpd00971_e') # Na+ (extracellular)\n",
    "cpd00971_c = cd630_PATRIC.metabolites.get_by_id('cpd00971_c') # Na+ (cytosolic)\n",
    "\n",
    "# Facilitated diffusion for D isomer\n",
    "CD630_06820 = cobra.Gene('CD630_06820')\n",
    "CD630_06820.name = 'Putative sodium:solute symporter'\n",
    "CD630_06820.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "ENOG4108HXH_c = cobra.Reaction('ENOG4108HXH_c') \n",
    "ENOG4108HXH_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'EggNOG'}\n",
    "ENOG4108HXH_c.gene_reaction_rule = 'CD630_06820'\n",
    "ENOG4108HXH_c.name = 'D-proline sodium symporter'\n",
    "ENOG4108HXH_c.lower_bound = 0.\n",
    "ENOG4108HXH_c.upper_bound = 1000.\n",
    "ENOG4108HXH_c.add_metabolites({\n",
    "    cpd00567_e: -1.0,\n",
    "    cpd00971_e: -1.0,\n",
    "    cpd00567_c: 1.0,\n",
    "    cpd00971_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ENOG4108HXH_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Hydroxyproline\n",
    "cpd29317_e = cobra.Metabolite(\n",
    "    'cpd29317_e',\n",
    "    formula='C5H9NO3',\n",
    "    name='4-hydroxyproline',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd29317_e, type='exchange', reaction_id='EX_cpd29317_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd29317_c = cobra.Metabolite(\n",
    "    'cpd29317_c',\n",
    "    formula='C5H9NO3',\n",
    "    name='4-hydroxyproline',\n",
    "    compartment='cytosol')\n",
    "\n",
    "rxn05165_2_c = cobra.Reaction('rxn05165_2_c')\n",
    "rxn05165_2_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'EggNOG', 'subsystem': 'ABC transporter'}\n",
    "rxn05165_2_c.notes['citations'] += ''\n",
    "rxn05165_2_c.name = '4-hydroxyproline ABC transporter'\n",
    "rxn05165_2_c.lower_bound = 0.\n",
    "rxn05165_2_c.upper_bound = 1000.\n",
    "rxn05165_2_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd29317_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd29317_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05165_2_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Glycine\n",
    "cpd00033_e = cobra.Metabolite(\n",
    "    'cpd00033_e',\n",
    "    formula='C2H5NO2',\n",
    "    name='Glycine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00033_e, type='exchange', reaction_id='EX_cpd00033_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00033_c = cd630_PATRIC.metabolites.get_by_id('cpd00033_c') # Glycine\n",
    "\n",
    "rxn10862_c = cobra.Reaction('rxn10862_c') \n",
    "rxn10862_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn10862_c.notes['citations'] += ''\n",
    "rxn10862_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn10862_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn10862_c.lower_bound = 0.\n",
    "rxn10862_c.upper_bound = 1000.\n",
    "rxn10862_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00033_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00033_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10862_c])\n",
    "new_rxns += 1\n",
    "# Hydroxyproline -> D-proline conversion\n",
    "cpd02431_c = cd630_PATRIC.metabolites.get_by_id('cpd02431_c') # 1-Pyrroline-5-carboxylate\n",
    "cpd00006_c = cd630_PATRIC.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd00005_c = cd630_PATRIC.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "\n",
    "CD630_25690 = cobra.Gene('CD630_25690')\n",
    "CD630_25690.name = '4-hydroxyproline epimerase'\n",
    "CD630_25690.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn32054_c = cobra.Reaction('rxn32054_c')\n",
    "rxn32054_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn32054_c.notes['citations'] += 'Watanabe, S., Tanimoto, Y., Nishiwaki, H., & Watanabe, Y. (2015). Identification and Characterization of Bifunctional Proline Racemase/Hydroxyproline Epimerase from Archaea: Discrimination of Substrates and Molecular Evolution. PLoS ONE, 10(3), e0120349.'\n",
    "rxn32054_c.name = '4-Hydroxy-L-proline:NADP+ 5-oxidoreductase'\n",
    "rxn10862_c.gene_reaction_rule = 'CD630_25690'\n",
    "rxn32054_c.lower_bound = 0.\n",
    "rxn32054_c.upper_bound = 1000.\n",
    "rxn32054_c.add_metabolites({\n",
    "    cpd00006_c: -1.0,\n",
    "    cpd29317_c: -1.0,\n",
    "    cpd00005_c: 1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd02431_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn32054_c])\n",
    "new_rxns += 1\n",
    "# rxn00931_c an rxn00929_c in place\n",
    "# NADP + L-Proline <=> NADPH + 2.0 H+ + 1-Pyrroline-5-carboxylate\n",
    "\n",
    "#------------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Non-Stickland substrates\n",
    "# Transport and exchange of non-auxotrophic amino acids\n",
    "\n",
    "# Tryptophan - facilitated diffusion already included\n",
    "\n",
    "#----------------#\n",
    "\n",
    "# Histidine\n",
    "cpd00119_e = cobra.Metabolite(\n",
    "    'cpd00119_e',\n",
    "    formula='C6H9N3O2',\n",
    "    name='L-Histidine',\n",
    "    compartment='extracellular')\n",
    "cpd00119_c = cd630_PATRIC.metabolites.get_by_id('cpd00119_c')\n",
    "\n",
    "CD630_17760 = cobra.Gene('CD630_17760')\n",
    "CD630_17760.name = ' amino acid family ABC transporter ATP-binding protein'\n",
    "CD630_17760.notes = {'annotation': 'NCBI'}\n",
    "\n",
    "rxn05164_c = cobra.Reaction('rxn05164_c') \n",
    "rxn05164_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05164_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05164_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05164_c.lower_bound = 0.\n",
    "rxn05164_c.upper_bound = 1000.\n",
    "rxn05164_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00119_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00119_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05164_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00119_e, type='exchange', reaction_id='EX_cpd00119_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Asparagine\n",
    "cpd00132_e = cobra.Metabolite(\n",
    "    'cpd00132_e',\n",
    "    formula='C4H8N2O3',\n",
    "    name='L-Asparagine',\n",
    "    compartment='extracellular')\n",
    "cpd00132_c = cd630_PATRIC.metabolites.get_by_id('cpd00132_c')\n",
    "\n",
    "rxn10144_c = cobra.Reaction('rxn10144_c') \n",
    "rxn10144_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn10144_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn10144_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn10144_c.lower_bound = 0.\n",
    "rxn10144_c.upper_bound = 1000.\n",
    "rxn10144_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00132_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00132_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10144_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00132_e, type='exchange', reaction_id='EX_cpd00132_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Serine\n",
    "cpd00054_e = cobra.Metabolite(\n",
    "    'cpd00054_e',\n",
    "    formula='C3H7NO3',\n",
    "    name='L-Serine',\n",
    "    compartment='extracellular')\n",
    "cpd00054_c = cd630_PATRIC.metabolites.get_by_id('cpd00054_c')\n",
    "\n",
    "rxn10883_c = cobra.Reaction('rxn10883_c') \n",
    "rxn10883_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn10883_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn10883_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn10883_c.lower_bound = 0.\n",
    "rxn10883_c.upper_bound = 1000.\n",
    "rxn10883_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00054_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00054_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10883_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00054_e, type='exchange', reaction_id='EX_cpd00054_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Threonine\n",
    "cpd00161_e = cobra.Metabolite(\n",
    "    'cpd00161_e',\n",
    "    formula='C4H9NO3',\n",
    "    name='L-Threonine',\n",
    "    compartment='extracellular')\n",
    "cpd00161_c = cd630_PATRIC.metabolites.get_by_id('cpd00161_c')\n",
    "\n",
    "rxn05169_c = cobra.Reaction('rxn05169_c') \n",
    "rxn05169_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05169_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05169_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05169_c.lower_bound = 0.\n",
    "rxn05169_c.upper_bound = 1000.\n",
    "rxn05169_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00161_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00161_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05169_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00161_e, type='exchange', reaction_id='EX_cpd00161_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Glutamine\n",
    "cpd00053_e = cobra.Metabolite(\n",
    "    'cpd00053_e',\n",
    "    formula='C5H10N2O3',\n",
    "    name='L-Glutamine',\n",
    "    compartment='extracellular')\n",
    "cpd00053_c = cd630_PATRIC.metabolites.get_by_id('cpd00053_c')\n",
    "\n",
    "rxn05155_c = cobra.Reaction('rxn05155_c') \n",
    "rxn05155_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05155_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05155_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05155_c.lower_bound = 0.\n",
    "rxn05155_c.upper_bound = 1000.\n",
    "rxn05155_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00053_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00053_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05155_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00053_e, type='exchange', reaction_id='EX_cpd00053_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "\n",
    "# Lysine transport\n",
    "cpd00039_c = cd630_PATRIC.metabolites.get_by_id('cpd00039_c')\n",
    "cpd00039_e = cd630_PATRIC.metabolites.get_by_id('cpd00039_e')\n",
    "\n",
    "rxn05151_c = cobra.Reaction('rxn05151_c') \n",
    "rxn05151_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05151_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05151_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05151_c.lower_bound = 0.\n",
    "rxn05151_c.upper_bound = 1000.\n",
    "rxn05151_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00039_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00039_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05151_c])\n",
    "new_rxns += 1\n",
    "\n",
    "#----------------#\n",
    "\n",
    "# Aspartate (same as Glutamate)\n",
    "cpd00041_e = cobra.Metabolite(\n",
    "    'cpd00041_e',\n",
    "    formula='C4H6NO4',\n",
    "    name='L-Aspartate',\n",
    "    compartment='extracellular')\n",
    "cpd00041_c = cd630_PATRIC.metabolites.get_by_id('cpd00041_c')\n",
    "cpd00971_e = cd630_PATRIC.metabolites.get_by_id('cpd00971_e') # NA+ (extracellular)\n",
    "cpd00971_c = cd630_PATRIC.metabolites.get_by_id('cpd00971_c') # NA+ (extracellular)\n",
    "\n",
    "rxn34357_c = cobra.Reaction('rxn34357_c') \n",
    "rxn34357_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'Cation symporter'}\n",
    "rxn34357_c.gene_reaction_rule = '272563.8.peg.2286 or ( 272563.8.peg.1830 or 272563.8.peg.2818 )'\n",
    "rxn34357_c.name = 'Na+/aspartate symport'\n",
    "rxn34357_c.lower_bound = -1000.\n",
    "rxn34357_c.upper_bound = 1000.\n",
    "rxn34357_c.add_metabolites({\n",
    "    cpd00041_e: -1.0,\n",
    "    cpd00971_e: -1.0,\n",
    "    cpd00041_c: 1.0,\n",
    "    cpd00971_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn34357_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00041_e, type='exchange', reaction_id='EX_cpd00041_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "\n",
    "#35 so far...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd00430_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Phenylacetate exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f8473911310</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd00430_e <=> </p>\n",
       "                    <p style='text-align:right'>Phenylacetate <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd00430_e at 0x7f8473911310>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Stickland fermentation - Oxidative steps\n",
    "\n",
    "# Leucine\n",
    "# L-Leucine:2-oxoglutarate aminotransferase \trxn00806 in model already\n",
    "cpd00107_c = cd630_PATRIC.metabolites.get_by_id('cpd00107_c') # L-Leucine\n",
    "cpd00024_c = cd630_PATRIC.metabolites.get_by_id('cpd00024_c') # 2-Oxoglutarate\n",
    "cpd00023_c = cd630_PATRIC.metabolites.get_by_id('cpd00023_c') # L-Glutamate\n",
    "cpd00200_c = cd630_PATRIC.metabolites.get_by_id('cpd00200_c') # 4MOP (2-Oxoisocaproate)\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "cpd11621_c = cd630_PATRIC.metabolites.get_by_id('cpd11621_c') # Oxidizedferredoxin\n",
    "cpd00011_c = cd630_PATRIC.metabolites.get_by_id('cpd00011_c') # CO2\n",
    "cpd01882_c = cd630_PATRIC.metabolites.get_by_id('cpd01882_c') # Isovaleryl-CoA\n",
    "cpd11620_c = cd630_PATRIC.metabolites.get_by_id('cpd11620_c') # Reducedferredoxin\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "\n",
    "CD630_14930 = cobra.Gene('CD630_14930')\n",
    "CD630_14930.name = 'ilvE; branched-chain-amino-acid aminotransferase [EC 1.2.1.25 ]'\n",
    "CD630_14930.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn14152_c = cobra.Reaction('rxn14152_c') \n",
    "rxn14152_c.name = '4-methyl-2-oxopentanoate oxidoreductase'\n",
    "rxn14152_c.gene_reaction_rule = 'CD630_14930'\n",
    "rxn14152_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn14152_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn14152_c.lower_bound = 0.\n",
    "rxn14152_c.upper_bound = 1000.\n",
    "rxn14152_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00200_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd01882_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn14152_c])\n",
    "\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c') # NH3\n",
    "\n",
    "rxn00804_c = cobra.Reaction('rxn00804_c') \n",
    "rxn00804_c.name = 'Leucine aminotransferase'\n",
    "rxn00804_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00804_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn00804_c.lower_bound = 0.\n",
    "rxn00804_c.upper_bound = 1000.\n",
    "rxn00804_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00107_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00013_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00200_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00804_c])\n",
    "\n",
    "CD630_01120 = cobra.Gene('CD630_01120')\n",
    "CD630_01120.name = 'ptb; Phosphate butyryltransferase'\n",
    "CD630_01120.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "cpd05178_c = cobra.Metabolite(\n",
    "    'cpd05178_c',\n",
    "    formula='C5H10O2',\n",
    "    name='Isovalerate',\n",
    "    compartment='cytosol')\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "R01174_1_c = cobra.Reaction('R01174_1_c') \n",
    "R01174_1_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_1_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_1_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_1_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_1_c.lower_bound = 0.\n",
    "R01174_1_c.upper_bound = 1000.\n",
    "R01174_1_c.add_metabolites({\n",
    "    cpd01882_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd05178_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_1_c])\n",
    "\n",
    "cpd05178_e = cobra.Metabolite(\n",
    "    'cpd05178_e',\n",
    "    formula='C5H10O2',\n",
    "    name='Isovalerate',\n",
    "    compartment='extracellular')\n",
    "ID001_c = cobra.Reaction('ID001_c') \n",
    "ID001_c.name = 'Isovalerate diffusion'\n",
    "ID001_c.gene_reaction_rule = ''\n",
    "ID001_c.lower_bound = -1000.\n",
    "ID001_c.upper_bound = 1000.\n",
    "ID001_c.add_metabolites({\n",
    "    cpd05178_c: -1.0,\n",
    "    cpd05178_e: 1.0,\n",
    "    \n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID001_c])\n",
    "cd630_PATRIC.add_boundary(cpd05178_e, type='exchange', reaction_id='EX_cpd05178_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Valine\n",
    "cpd00156_c = cd630_PATRIC.metabolites.get_by_id('cpd00156_c') # L-Valine\n",
    "cpd00123_c = cd630_PATRIC.metabolites.get_by_id('cpd00123_c') # 3-Methyl-2-oxobutanoate (3mob)\n",
    "\n",
    "R01214_c = cobra.Reaction('R01214_c') \n",
    "R01214_c.name = 'L-valine aminotransferase'\n",
    "R01214_c.gene_reaction_rule = 'CD630_14930'\n",
    "R01214_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01214_c.lower_bound = -1000.\n",
    "R01214_c.upper_bound = 1000.\n",
    "R01214_c.add_metabolites({\n",
    "    cpd00156_c: -1.0,\n",
    "    cpd00024_c: -1.0,\n",
    "    cpd00023_c: 1.0,\n",
    "    cpd00123_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01214_c])\n",
    "\n",
    "cpd00481_c = cd630_PATRIC.metabolites.get_by_id('cpd00481_c') # Isobutyryl-CoA\n",
    "cpd11620_c = cd630_PATRIC.metabolites.get_by_id('cpd11620_c') # Reducedferredoxin\n",
    "cpd11621_c = cd630_PATRIC.metabolites.get_by_id('cpd11621_c') # Oxidizedferredoxin\n",
    "\n",
    "CD630_21990 = cobra.Gene('CD630_21990')\n",
    "CD630_21990.name = 'Putative ferredoxin/flavodoxin oxidoreductase,alpha subunit'\n",
    "CD630_21990.notes = {'annotation': 'UniProt'}\n",
    "CD630_21991 = cobra.Gene('CD630_21991')\n",
    "CD630_21991.name = 'Putative 4Fe-4S ferredoxin iron-sulfur binding domain-containing protein'\n",
    "CD630_21991.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn10618_c = cobra.Reaction('rxn10618_c') \n",
    "rxn10618_c.name = '2-oxoisovalerate ferredoxin reductase'\n",
    "rxn10618_c.gene_reaction_rule = 'CD630_21990 and CD630_21991'\n",
    "rxn10618_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10618_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn10618_c.lower_bound = 0.\n",
    "rxn10618_c.upper_bound = 1000.\n",
    "rxn10618_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00123_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00481_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10618_c])\n",
    "\n",
    "cd630_PATRIC.metabolites.get_by_id('cpd01711_c').formula = 'C4H8O2' # Isobutyrate (correct formula)\n",
    "cpd01711_c = cd630_PATRIC.metabolites.get_by_id('cpd01711_c')\n",
    "\n",
    "R01174_2_c = cobra.Reaction('R01174_2_c') \n",
    "R01174_2_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_2_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_2_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_2_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_2_c.lower_bound = 0.\n",
    "R01174_2_c.upper_bound = 1000.\n",
    "R01174_2_c.add_metabolites({\n",
    "    cpd00481_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd01711_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_2_c])\n",
    "\n",
    "cpd01711_e = cobra.Metabolite(\n",
    "    'cpd01711_e',\n",
    "    formula='C4H8O2',\n",
    "    name='Isobutyrate',\n",
    "    compartment='extracellular')\n",
    "ID002_c = cobra.Reaction('ID002_c') \n",
    "ID002_c.name = 'Isobutyrate diffusion'\n",
    "ID002_c.lower_bound = -1000.\n",
    "ID002_c.upper_bound = 1000.\n",
    "ID002_c.add_metabolites({\n",
    "    cpd01711_c: -1.0,\n",
    "    cpd01711_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID002_c])\n",
    "cd630_PATRIC.add_boundary(cpd01711_e, type='exchange', reaction_id='EX_cpd01711_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Isoleucine\n",
    "cpd00322_c = cd630_PATRIC.metabolites.get_by_id('cpd00322_c') # L-Isoleucine\n",
    "cpd00508_c = cd630_PATRIC.metabolites.get_by_id('cpd00508_c') # 3MOP (3-Methyl-2-oxopentanoate)\n",
    "\n",
    "rxn01575_c = cobra.Reaction('rxn01575_c') \n",
    "rxn01575_c.name = 'L-isoleucine aminotransferase'\n",
    "rxn01575_c.gene_reaction_rule = 'CD630_14930'\n",
    "rxn01575_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01575_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn01575_c.lower_bound = -1000.\n",
    "rxn01575_c.upper_bound = 1000.\n",
    "rxn01575_c.add_metabolites({\n",
    "    cpd00322_c: -1.0,\n",
    "    cpd00024_c: -1.0,\n",
    "    cpd00023_c: 1.0,\n",
    "    cpd00508_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01575_c])\n",
    "\n",
    "# Next reaction already in place:  rxn02269_c;  NAD + CoA + 3MOP --> NADH + CO2 + 2-Methylbutyryl-CoA\n",
    "\n",
    "cpd00760_c = cd630_PATRIC.metabolites.get_by_id('cpd00760_c') # 2-Methylbutyryl-CoA\n",
    "cpd19585_c = cobra.Metabolite(\n",
    "    'cpd19585_c',\n",
    "    formula='C5H10O2',\n",
    "    name='2-Methylbutyrate',\n",
    "    compartment='cytosol')\n",
    "R01174_3_c = cobra.Reaction('R01174_3_c') \n",
    "R01174_3_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_3_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_3_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_3_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_3_c.lower_bound = 0.\n",
    "R01174_3_c.upper_bound = 1000.\n",
    "R01174_3_c.add_metabolites({\n",
    "    cpd00760_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd19585_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_3_c])\n",
    "\n",
    "cpd19585_e = cobra.Metabolite(\n",
    "    'cpd19585_e',\n",
    "    formula='C5H10O2',\n",
    "    name='2-Methylbutyrate',\n",
    "    compartment='extracellular')\n",
    "ID003_c = cobra.Reaction('ID003_c') \n",
    "ID003_c.name = '2-Methylbutyrate diffusion'\n",
    "ID003_c.lower_bound = -1000.\n",
    "ID003_c.upper_bound = 1000.\n",
    "ID003_c.add_metabolites({\n",
    "    cpd19585_c: -1.0,\n",
    "    cpd19585_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID003_c])\n",
    "cd630_PATRIC.add_boundary(cpd19585_e, type='exchange', reaction_id='EX_cpd19585_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Tyrosine\n",
    "cpd00069_c = cd630_PATRIC.metabolites.get_by_id('cpd00069_c') # L-Tyrosine\n",
    "cpd00868_c = cd630_PATRIC.metabolites.get_by_id('cpd00868_c') # p-hydroxyphenylpyruvate\n",
    "#rxn00527_c = cobra.Reaction('rxn00527_c') \n",
    "#rxn00527_c.name = 'L-Tyrosine aminotransferase'\n",
    "#rxn00527_c.gene_reaction_rule = 'CD630_14930'\n",
    "#rxn00527_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "#rxn00527_c.notes['citations'] += 'Neumann-Schaal M., et al. (2015). Time-resolved amino acid uptake of Clostridium difficile 630Δerm and concomitant fermentation product and toxin formation. BMC Microbiology. 15(1); 1471-2180.'\n",
    "#rxn00527_c.lower_bound = -1000.\n",
    "#rxn00527_c.upper_bound = 1000.\n",
    "#rxn00527_c.add_metabolites({\n",
    "#    cpd00069_c: -1.0,\n",
    "#    cpd00024_c: -1.0,\n",
    "#    cpd00023_c: 1.0,\n",
    "#    cpd00868_c: 1.0\n",
    "#})\n",
    "#cd630_PATRIC.add_reactions([rxn00527_c])\n",
    "\n",
    "cpd03165_c = cobra.Metabolite(\n",
    "    'cpd03165_c',\n",
    "    formula='C29H39N7O18P3S',\n",
    "    name='4-Hydroxyphenylacetyl-CoA',\n",
    "    compartment='cytosol')\n",
    "rxn10563_c = cobra.Reaction('rxn10563_c') \n",
    "rxn10563_c.name = 'Indolepyruvate ferredoxin oxidoreductase (hydroxyphenylpyvurate)'\n",
    "rxn10563_c.gene_reaction_rule = 'CD630_21990 and CD630_21991'\n",
    "rxn10563_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10563_c.notes['citations'] += 'Neumann-Schaal M., et al. (2015). Time-resolved amino acid uptake of Clostridium difficile 630Δerm and concomitant fermentation product and toxin formation. BMC Microbiology. 15(1); 1471-2180.'\n",
    "rxn10563_c.lower_bound = 0.\n",
    "rxn10563_c.upper_bound = 1000.\n",
    "rxn10563_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00868_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd03165_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10563_c])\n",
    "\n",
    "cpd00489_c = cobra.Metabolite(\n",
    "    'cpd00489_c',\n",
    "    formula='C8H8O3',\n",
    "    name='4-Hydroxyphenylacetate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "R01174_4_c = cobra.Reaction('R01174_4_c') \n",
    "R01174_4_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_4_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_4_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_4_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_4_c.lower_bound = 0.\n",
    "R01174_4_c.upper_bound = 1000.\n",
    "R01174_4_c.add_metabolites({\n",
    "    cpd03165_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd00489_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_4_c])\n",
    "\n",
    "cpd00489_e = cobra.Metabolite(\n",
    "    'cpd00489_e',\n",
    "    formula='C8H8O3',\n",
    "    name='4-Hydroxyphenylacetate',\n",
    "    compartment='extracellular')\n",
    "ID004_c = cobra.Reaction('ID004_c') \n",
    "ID004_c.name = '4-Hydroxyphenylacetate diffusion'\n",
    "ID004_c.lower_bound = -1000.\n",
    "ID004_c.upper_bound = 1000.\n",
    "ID004_c.add_metabolites({\n",
    "    cpd00489_c: -1.0,\n",
    "    cpd00489_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID004_c])\n",
    "cd630_PATRIC.add_boundary(cpd00489_e, type='exchange', reaction_id='EX_cpd00489_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# p-Cresol formation and secretion\n",
    "cpd01042_c = cobra.Metabolite(\n",
    "    'cpd01042_c',\n",
    "    formula='C7H8O',\n",
    "    name='p-Cresol',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_01530 = cobra.Gene('CD630_01530')\n",
    "CD630_01530.name = '4-hydroxyphenylacetate decarboxylase large subunit [EC:4.1.1.83]'\n",
    "CD630_01530.notes = {'annotation': 'KEGG'}\n",
    "CD630_01540 = cobra.Gene('CD630_01540')\n",
    "CD630_01540.name = '4-hydroxyphenylacetate decarboxylase small subunit [EC:4.1.1.83]'\n",
    "CD630_01540.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn05061_c = cobra.Reaction('rxn05061_c') \n",
    "rxn05061_c.name = '4-hydroxyphenylacetate carboxy-lyase'\n",
    "rxn05061_c.gene_reaction_rule = 'CD630_01530 and CD630_01540'\n",
    "rxn05061_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05061_c.notes['citations'] += 'Selmer, T., & Andrei, P. I. (2001). p-Hydroxyphenylacetate decarboxylase from Clostridium difficile. A novel glycyl radical enzyme catalysing the formation of p-cresol. European Journal of Biochemistry, 268(5), 1363–1372.'\n",
    "rxn05061_c.lower_bound = -1000.\n",
    "rxn05061_c.upper_bound = 1000.\n",
    "rxn05061_c.add_metabolites({\n",
    "    cpd00489_c: -1.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd01042_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05061_c])\n",
    "\n",
    "cpd01042_e = cobra.Metabolite(\n",
    "    'cpd01042_e',\n",
    "    formula='C7H8O',\n",
    "    name='p-Cresol',\n",
    "    compartment='extracellular')\n",
    "ID005_c = cobra.Reaction('ID005_c') \n",
    "ID005_c.name = 'p-Cresol diffusion'\n",
    "ID005_c.lower_bound = -1000.\n",
    "ID005_c.upper_bound = 1000.\n",
    "ID005_c.add_metabolites({\n",
    "    cpd01042_c: -1.0,\n",
    "    cpd01042_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID005_c])\n",
    "cd630_PATRIC.add_boundary(cpd01042_e, type='exchange', reaction_id='EX_cpd01042_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Phenylalanine\n",
    "cpd00066_c = cd630_PATRIC.metabolites.get_by_id('cpd00066_c') # L-Phenylalanine\n",
    "cpd00143_c = cd630_PATRIC.metabolites.get_by_id('cpd00143_c') # Phenylpyruvate\n",
    "rxn21134_c = cobra.Reaction('rxn21134_c') \n",
    "rxn21134_c.name = 'L-phenylalanine aminotransferase'\n",
    "rxn21134_c.gene_reaction_rule = 'CD630_14930'\n",
    "rxn21134_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn21134_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn21134_c.lower_bound = -1000.\n",
    "rxn21134_c.upper_bound = 1000.\n",
    "rxn21134_c.add_metabolites({\n",
    "    cpd00066_c: -1.0,\n",
    "    cpd00024_c: -1.0,\n",
    "    cpd00023_c: 1.0,\n",
    "    cpd00143_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn21134_c])\n",
    "\n",
    "cpd00452_c = cobra.Metabolite(\n",
    "    'cpd00452_c',\n",
    "    formula='C29H38N7O17P3S',\n",
    "    name='Phenylacetyl-CoA',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_23800 = cobra.Gene('CD630_23800')\n",
    "CD630_23800.name = 'Indole pyruvate ferredoxin/flavodoxin oxidoreductase'\n",
    "CD630_23800.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn10562_c = cobra.Reaction('rxn10562_c') \n",
    "rxn10562_c.name = 'indolepyruvate ferredoxin oxidoreductase'\n",
    "rxn10562_c.gene_reaction_rule = 'CD630_23800'\n",
    "rxn10562_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10562_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn10562_c.lower_bound = 0.\n",
    "rxn10562_c.upper_bound = 1000.\n",
    "rxn10562_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00143_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00452_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10562_c])\n",
    "\n",
    "cpd00430_c = cobra.Metabolite(\n",
    "    'cpd00430_c',\n",
    "    formula='C8H8O2',\n",
    "    name='Phenylacetate', # PACT in ModelSEED\n",
    "    compartment='cytosol')\n",
    "\n",
    "R01174_5_c = cobra.Reaction('R01174_5_c') \n",
    "R01174_5_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_5_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_5_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_5_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_5_c.lower_bound = 0.\n",
    "R01174_5_c.upper_bound = 1000.\n",
    "R01174_5_c.add_metabolites({\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00452_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd00430_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_5_c])\n",
    "\n",
    "cpd00430_e = cobra.Metabolite(\n",
    "    'cpd00430_e',\n",
    "    formula='C8H8O2',\n",
    "    name='Phenylacetate', # PACT\n",
    "    compartment='extracellular')\n",
    "\n",
    "rxn11336_c = cobra.Reaction('rxn11336_c') \n",
    "rxn11336_c.name = 'Phenylacetate diffusion'\n",
    "rxn11336_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn11336_c.lower_bound = -1000.\n",
    "rxn11336_c.upper_bound = 1000.\n",
    "rxn11336_c.add_metabolites({\n",
    "    cpd00430_c: -1.0,\n",
    "    cpd00430_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn11336_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00430_e, type='exchange', reaction_id='EX_cpd00430_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd03170_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>4-Hydroxymandelate exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f84738b8dd0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd03170_e <=> </p>\n",
       "                    <p style='text-align:right'>4-Hydroxymandelate <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd03170_e at 0x7f84738b8dd0>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Stickland Reactions - Reductive steps\n",
    "\n",
    "# Frequently needed metabolites\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c') # NH3\n",
    "cpd11421_c = cd630_PATRIC.metabolites.get_by_id('cpd11421_c') # trdrd\n",
    "cpd11420_c = cd630_PATRIC.metabolites.get_by_id('cpd11420_c') # trdox\n",
    "cpd00196_c = cd630_PATRIC.metabolites.get_by_id('cpd00196_c') # Acetylphosphate\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00011_c = cd630_PATRIC.metabolites.get_by_id('cpd00011_c') # CO2\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Proline\n",
    "cpd00129_c = cd630_PATRIC.metabolites.get_by_id('cpd00129_c') # L-Proline\n",
    "cd630_PATRIC.remove_reactions(['rxn02029_c']) # This is incorrect for C. difficile metabolism\n",
    "cpd00339_e = cobra.Metabolite(\n",
    "    'cpd00339_e',\n",
    "    formula='C5H11NO2',\n",
    "    name='5-Aminopentanoate',\n",
    "    compartment='extracellular')\n",
    "# D-Proline reductase (C. difficile)\n",
    "cpd00339_c = cd630_PATRIC.metabolites.get_by_id('cpd00339_c') # 5-Aminopentanoate\n",
    "rxn20606_c = cobra.Reaction('rxn20606_c') \n",
    "rxn20606_c.name = 'D-proline reductase'\n",
    "rxn20606_c.gene_reaction_rule = 'CD630_32440'\n",
    "rxn20606_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn20606_c.notes['citations'] += 'Bouillaut, L., Self, W. T., & Sonenshein, A. L. (2013). Proline-Dependent Regulation of Clostridium difficile Stickland Metabolism. Journal of Bacteriology, 195(4), 844–854.'\n",
    "rxn20606_c.notes['citations'] += 'Jackson, S., Calos, M., Myers, A., & Self, W. T. (2006). Analysis of proline reduction in the nosocomial pathogen Clostridium difficile. Journal of Bacteriology.'\n",
    "rxn20606_c.lower_bound = 0.\n",
    "rxn20606_c.upper_bound = 1000.\n",
    "rxn20606_c.add_metabolites({\n",
    "    cpd00567_c: -1.0,\n",
    "    cpd00004_c: -1.0,\n",
    "    cpd00067_c: -1.0, # H+\n",
    "    cpd00003_c: 1.0,\n",
    "    cpd00339_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn20606_c])\n",
    "# 5-Aminopentanoate (aka 5-aminovalerate) export\n",
    "rxn12566_c = cobra.Reaction('rxn12566_c') \n",
    "rxn12566_c.name = '5-Aminopentanoate transport via proton symport'\n",
    "rxn12566_c.lower_bound = 0.\n",
    "rxn12566_c.upper_bound = 1000.\n",
    "rxn12566_c.add_metabolites({\n",
    "    cpd00339_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00067_e: 1.0,\n",
    "    cpd00339_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn12566_c])\n",
    "cd630_PATRIC.add_boundary(cpd00339_e, type='exchange', reaction_id='EX_cpd00339_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Glycine \n",
    "cpd00033_c = cd630_PATRIC.metabolites.get_by_id('cpd00033_c') # Glycine\n",
    "rxn11676_c = cobra.Reaction('rxn11676_c') \n",
    "rxn11676_c.name = 'Glycine reductase'\n",
    "rxn11676_c.gene_reaction_rule = 'CD630_23540'\n",
    "rxn11676_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn11676_c.notes['citations'] += 'Bouillaut, L., Self, W. T., & Sonenshein, A. L. (2013). Proline-Dependent Regulation of Clostridium difficile Stickland Metabolism. Journal of Bacteriology, 195(4), 844–854.'\n",
    "rxn11676_c.lower_bound = -1000.\n",
    "rxn11676_c.upper_bound = 1000.\n",
    "rxn11676_c.add_metabolites({\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00033_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd11421_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd00013_c: 1.0,\n",
    "    cpd00196_c: 1.0,\n",
    "    cpd11420_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn11676_c])\n",
    "\n",
    "# Acetylphosphate --> Acetate conversion intact\n",
    "# ATP:acetate phosphotransferase; rxn00225; 1 ADP [0] + 1 Acetylphosphate [0] -> 1 ATP [0] + 1 Acetate [0] \n",
    "cd630_PATRIC.reactions.get_by_id('rxn00225_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00225_c').notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00225_c').notes['citations'] += 'Bouillaut, L., Self, W. T., & Sonenshein, A. L. (2013). Proline-Dependent Regulation of Clostridium difficile Stickland Metabolism. Journal of Bacteriology, 195(4), 844–854.'\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Leucine\n",
    "cpd00200_c = cd630_PATRIC.metabolites.get_by_id('cpd00200_c') # 4-Methyl-2-oxopentanoate - 4MOP (2-Oxoisocaproate)\n",
    "cpd01646_c = cd630_PATRIC.metabolites.get_by_id('cpd01646_c') # 2-Isopropylmalate == 2-Hydroxyisocaproate\n",
    "\n",
    "CD630_03940 = cobra.Gene('CD630_03940')\n",
    "CD630_03940.name = 'ldhA; D-lactate dehydrogenase [EC:1.1.1.28]'\n",
    "CD630_03940.notes = {'annotation': 'PATRIC'}\n",
    "\n",
    "rxn13022_c = cobra.Reaction('rxn13022_c') \n",
    "rxn13022_c.name = '3-isopropylmalate dehydrogenase'\n",
    "rxn13022_c.gene_reaction_rule = 'CD630_03940'\n",
    "rxn13022_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn13022_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2006). Characterization of (R)-2-Hydroxyisocaproate Dehydrogenase and a Family III Coenzyme A Transferase Involved in Reduction of l-Leucine to Isocaproate by Clostridium difficile. Applied and Environmental Microbiology, 72(9), 6062–6069.'\n",
    "rxn13022_c.lower_bound = -1000.\n",
    "rxn13022_c.upper_bound = 1000.\n",
    "rxn13022_c.add_metabolites({\n",
    "    cpd00004_c: -1.0,\n",
    "    cpd00011_c: -1.0,\n",
    "    cpd00200_c: -1.0,\n",
    "    cpd00003_c: 1.0,\n",
    "    cpd01646_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn13022_c])\n",
    "\n",
    "CD630_03950 = cobra.Gene('CD630_03950')\n",
    "CD630_03950.name = 'hadA; Isocaprenoyl-CoA:2-hydroxyisocaproate CoA-transferase'\n",
    "CD630_03950.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "cpd23040_c = cobra.Metabolite(\n",
    "    'cpd23040_c',\n",
    "    formula='C27H42N7O18P3S',\n",
    "    name='2-hydroxyisocaproyl-CoA',\n",
    "    compartment='cytosol')\n",
    "# C27H46N7O18P3S would be an uncharged formula, otherwise its -4 charge\n",
    "# C27H42N7O18P3S (original)\n",
    "\n",
    "C21400_c = cobra.Metabolite(\n",
    "    'C21400_c',\n",
    "    formula='C27H46N7O17P3S',\n",
    "    name='Isocaproyl-CoA',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_03980 = cobra.Gene('CD630_03980')\n",
    "CD630_03980.name = 'hadC; (R)-2-hydroxyisocaproyl-CoA dehydratase beta subunit [EC:4.2.1.157]'\n",
    "CD630_03980.notes = {'annotation': 'UniProt'}\n",
    "CD630_03970 = cobra.Gene('CD630_03970')\n",
    "CD630_03970.name = 'hadB; (R)-2-hydroxyisocaproyl-CoA dehydratase alpha subunit [EC:4.2.1.157]'\n",
    "CD630_03970.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "K20025_c = cobra.Reaction('K20025_c') \n",
    "K20025_c.name = '(R)-2-hydroxyisocaproyl-CoA dehydratase'\n",
    "K20025_c.gene_reaction_rule = 'CD630_03980 and CD630_03970'\n",
    "K20025_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "K20025_c.notes['citations'] += 'Britz ML & Wilkinson RG (1982). \"Leucine dissimilation to isovaleric and isocaproic acids by cell suspensions of amino acid fermenting anaerobes: the Stickland reaction revisited.\" Can J Microbiol. 28(3);291-300.'\n",
    "K20025_c.lower_bound = 0.\n",
    "K20025_c.upper_bound = 1000.\n",
    "K20025_c.add_metabolites({\n",
    "    cpd00067_c: -6.0,\n",
    "    cpd23040_c: -1.0,\n",
    "    C21400_c: 1.0,\n",
    "    cpd00001_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([K20025_c])   \n",
    "\n",
    "C21399_c = cobra.Metabolite(\n",
    "    'C21399_c',\n",
    "    formula='C6H12O2',\n",
    "    name='Isocaproate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "cpd00047_c = cd630_PATRIC.metabolites.get_by_id('cpd00047_c') # Formate\n",
    "\n",
    "R11462_c = cobra.Reaction('R11462_c') \n",
    "R11462_c.name = '4-methylpentanoyl-CoA:(R)-2-hydroxy-4-methylpentanoate CoA-transferase [EC:2.8.3.24]'\n",
    "R11462_c.gene_reaction_rule = 'CD630_03950'\n",
    "R11462_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R11462_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2006). Characterization of (R)-2-Hydroxyisocaproate Dehydrogenase and a Family III Coenzyme A Transferase Involved in Reduction of l-Leucine to Isocaproate by Clostridium difficile. Applied and Environmental Microbiology, 72(9), 6062–6069.'\n",
    "R11462_c.lower_bound = -1000.\n",
    "R11462_c.upper_bound = 1000.\n",
    "R11462_c.add_metabolites({\n",
    "    cpd00067_c: -1.0,\n",
    "    C21400_c: -1.0,\n",
    "    cpd01646_c: -1.0,\n",
    "    C21399_c: 1.0,\n",
    "    cpd23040_c: 1.0,\n",
    "    cpd00047_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R11462_c])                                                                           \n",
    "\n",
    "CD630_03960 = cobra.Gene('CD630_03960')\n",
    "CD630_03960.name = 'hadI; 2-hydroxyisocaproyl-CoA dehydratase activator'\n",
    "CD630_03960.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "C21090_c = cobra.Metabolite(\n",
    "    'C21090_c',\n",
    "    formula='C27H44N7O17P3S',\n",
    "    name='Isocaprenoyl-CoA',\n",
    "    compartment='cytosol')\n",
    "\n",
    "R11076_c = cobra.Reaction('R11076_c') \n",
    "R11076_c.name = '(R)-2-hydroxyisocaproyl-CoA dehydratase'\n",
    "R11076_c.gene_reaction_rule = 'CD630_03960 and CD630_03970 and CD630_03980'\n",
    "R11076_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R11076_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2006). Characterization of (R)-2-Hydroxyisocaproate Dehydrogenase and a Family III Coenzyme A Transferase Involved in Reduction of l-Leucine to Isocaproate by Clostridium difficile. Applied and Environmental Microbiology, 72(9), 6062–6069.'\n",
    "R11076_c.lower_bound = -1000.\n",
    "R11076_c.upper_bound = 1000.\n",
    "R11076_c.add_metabolites({\n",
    "    cpd00067_c: -4.0,\n",
    "    cpd23040_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    C21090_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R11076_c])                                                                           \n",
    "\n",
    "C21399_e = cobra.Metabolite(\n",
    "    'C21399_e',\n",
    "    formula='C6H12O2',\n",
    "    name='Isocaproate',\n",
    "    compartment='extracellular')\n",
    "\n",
    "ID006_c = cobra.Reaction('ID006_c') \n",
    "ID006_c.name = 'Isocaproate diffusion' \n",
    "ID006_c.lower_bound = -1000.\n",
    "ID006_c.upper_bound = 1000.\n",
    "ID006_c.add_metabolites({\n",
    "    C21399_e: -1.0,\n",
    "    C21399_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID006_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(C21399_e, type='exchange', reaction_id='EX_C21399_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Phenylalanine\n",
    "cpd00143_c = cd630_PATRIC.metabolites.get_by_id('cpd00143_c') # Phenylpyruvate\n",
    "cpd07326_c = cobra.Metabolite(\n",
    "    'cpd07326_c',\n",
    "    formula='C9H7O2',\n",
    "    name='Cinnamic acid',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_32370_1 = cobra.Gene('CD630_32370_1')\n",
    "CD630_32370_1.name = '3-phenylpropionate/trans-cinnamate dioxygenase subunit alpha [EC:1.14.12.19]'\n",
    "CD630_32370_1.notes = {'annotation': 'KEGG'}\n",
    "CD630_32370_2 = cobra.Gene('CD630_32370_2')\n",
    "CD630_32370_2.name = '3-phenylpropionate/trans-cinnamate dioxygenase subunit beta [EC:1.14.12.19]'\n",
    "CD630_32370_2.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "R06782_c = cobra.Reaction('R06782_c') \n",
    "R06782_c.name = '3-phenylpropanoate dioxygenase'\n",
    "R06782_c.gene_reaction_rule = 'CD630_32370_1 and CD630_32370_2'\n",
    "R06782_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R06782_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2004). Dehydration of (R)-2-hydroxyacyl-CoA to enoyl-CoA in the fermentation of α-amino acids by anaerobic bacteria. FEMS Microbiology Reviews. 28 (4). 455–468.'\n",
    "R06782_c.lower_bound = -1000.\n",
    "R06782_c.upper_bound = 1000.\n",
    "R06782_c.add_metabolites({\n",
    "    cpd00067_c: -2.0,\n",
    "    cpd00143_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd07326_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R06782_c])                                                                           \n",
    "\n",
    "cpd03343_c = cobra.Metabolite(\n",
    "    'cpd03343_c',\n",
    "    formula='C9H9O2',\n",
    "    name='Phenylpropanoate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "\n",
    "K00529 = cobra.Gene('K00529')\n",
    "K00529.name = '3-phenylpropanoate,NADH:oxygen oxidoreductase'\n",
    "K00529.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn01610_c = cobra.Reaction('rxn01610_c') \n",
    "rxn01610_c.name = 'phenylpropanoate:NAD+ delta2-oxidoreductase'\n",
    "rxn01610_c.gene_reaction_rule = 'K00529'\n",
    "rxn01610_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01610_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2004). Dehydration of (R)-2-hydroxyacyl-CoA to enoyl-CoA in the fermentation of α-amino acids by anaerobic bacteria. FEMS Microbiology Reviews. 28 (4). 455–468.'\n",
    "rxn01610_c.lower_bound = 0.\n",
    "rxn01610_c.upper_bound = 1000.\n",
    "rxn01610_c.add_metabolites({\n",
    "    cpd00004_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd07326_c: -1.0,\n",
    "    cpd00003_c: 1.0,\n",
    "    cpd03343_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01610_c])                                                                           \n",
    "\n",
    "cpd03343_e = cobra.Metabolite(\n",
    "    'cpd03343_e',\n",
    "    formula='C9H9O2',\n",
    "    name='Phenylpropanoate',\n",
    "    compartment='extracellular')\n",
    "ID007_c = cobra.Reaction('ID007_c') \n",
    "ID007_c.name = 'Phenylpropanoate diffusion'\n",
    "ID007_c.lower_bound = -1000.\n",
    "ID007_c.upper_bound = 1000.\n",
    "ID007_c.add_metabolites({\n",
    "    cpd03343_e: -1.0,\n",
    "    cpd03343_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID007_c])\n",
    "cd630_PATRIC.add_boundary(cpd03343_e, type='exchange', reaction_id='EX_cpd03343_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Tyrosine\n",
    "cpd00868_c = cd630_PATRIC.metabolites.get_by_id('cpd00868_c') # p-hydroxyphenylpyruvate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') \n",
    "cpd03170_c = cobra.Metabolite(\n",
    "    'cpd03170_c',\n",
    "    formula='C8H7O4',\n",
    "    name='4-Hydroxymandelate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_01550 = cobra.Gene('CD630_01550')\n",
    "CD630_01550.name = 'hpdA; 4-hydroxyphenylacetate decarboxylase activating enzyme'\n",
    "CD630_01550.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn07124_c = cobra.Reaction('rxn07124_c') \n",
    "rxn07124_c.name = 'p-Hydroxyphenylacetate decarboxylase'\n",
    "rxn07124_c.gene_reaction_rule = 'CD630_01550'\n",
    "rxn07124_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn07124_c.notes['citations'] += 'Yu L, Blaser M, Andrei PI, Pierik AJ, & Selmer T. (2006). 4-Hydroxyphenylacetate Decarboxylases:  Properties of a Novel Subclass of Glycyl Radical Enzyme Systems. Biochemistry. 45 (31), 9584-9592.'\n",
    "rxn07124_c.lower_bound = -1000.\n",
    "rxn07124_c.upper_bound = 1000.\n",
    "rxn07124_c.add_metabolites({\n",
    "    cpd00001_c: -2.0,\n",
    "    cpd00868_c: -1.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd03170_c: 1.0,\n",
    "    cpd00067_c: 4.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn07124_c])\n",
    "\n",
    "cpd03170_e = cobra.Metabolite(\n",
    "    'cpd03170_e',\n",
    "    formula='C8H7O4',\n",
    "    name='4-Hydroxymandelate',\n",
    "    compartment='extracellular')\n",
    "ID008_c = cobra.Reaction('ID008_c')\n",
    "ID008_c.name = '2,3-Dihydroxyphenylpropanoate diffusion'\n",
    "ID008_c.lower_bound = -1000.\n",
    "ID008_c.upper_bound = 1000.\n",
    "ID008_c.add_metabolites({\n",
    "    cpd03170_e: -1.0,\n",
    "    cpd03170_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID008_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd03170_e, type='exchange', reaction_id='EX_cpd02501_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#------------------#\n",
    "\n",
    "# Possible other reductive substrates\n",
    "\n",
    "# from Valine\n",
    "#cpd00123_c = cd630_PATRIC.metabolites.get_by_id('cpd00123_c') # 3-Methyl-2-oxobutanoate - 3MOB\n",
    "\n",
    "# from Isoleucine\n",
    "#cpd00508_c = cd630_PATRIC.metabolites.get_by_id('cpd00508_c') # 3MOP (3-Methyl-2-oxopentanoate)\n",
    "\n",
    "# 67 so far..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd00141_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Propionate exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f847361c210</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd00141_e <=> </p>\n",
       "                    <p style='text-align:right'>Propionate <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd00141_e at 0x7f847361c210>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Fatty acid metabolism\n",
    "\n",
    "# Succinate\n",
    "\n",
    "# Exchange and import\n",
    "cpd00036_e = cobra.Metabolite(\n",
    "    'cpd00036_e',\n",
    "    formula='C4H6O4',\n",
    "    name='Succinate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.metabolites.get_by_id('cpd00036_c').formula = 'C4H6O4' # fixes charge\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00036_e, type='exchange', reaction_id='EX_cpd00036_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "cpd00036_c = cd630_PATRIC.metabolites.get_by_id('cpd00036_c') # Succinate (cytosolic)\n",
    "rxn09271_c = cobra.Reaction('rxn09271_c') # Diffusion\n",
    "rxn09271_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn09271_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn09271_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn09271_c.name = 'Succinate transport'\n",
    "rxn09271_c.lower_bound = -1000.\n",
    "rxn09271_c.upper_bound = 1000.\n",
    "rxn09271_c.add_metabolites({\n",
    "    cpd00036_e: -1.0,\n",
    "    cpd00036_c: 1.0\n",
    "})\n",
    "\n",
    "# Succinate --> Succinyl-CoA\n",
    "cpd00078_c = cd630_PATRIC.metabolites.get_by_id('cpd00078_c') # Succinyl-CoA\n",
    "cpd00029_c = cd630_PATRIC.metabolites.get_by_id('cpd00029_c') # Acetate\n",
    "cpd00022_c = cd630_PATRIC.metabolites.get_by_id('cpd00022_c') # Acetyl-CoA\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H\n",
    "\n",
    "CD630_21960 = cobra.Gene('CD630_21960')\n",
    "CD630_21960.name = 'abfD; Putative succinyl-CoA synthetase-like ATP-binding protein'\n",
    "CD630_21960.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn25164_c = cobra.Reaction('rxn25164_c') \n",
    "rxn25164_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn25164_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn25164_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn25164_c.name = 'Succinyl-CoA synthase'\n",
    "rxn25164_c.gene_reaction_rule = 'CD630_21960'\n",
    "rxn25164_c.lower_bound = -1000.\n",
    "rxn25164_c.upper_bound = 1000.\n",
    "rxn25164_c.add_metabolites({\n",
    "    cpd00036_c: -1.0,\n",
    "    cpd00022_c: -1.0,\n",
    "    cpd00078_c: 1.0,\n",
    "    cpd00029_c: 1.0,\n",
    "    cpd00067_c: 2.0\n",
    "})\n",
    "\n",
    "# Succinyl-CoA --> 4-Oxobutanoate\n",
    "cpd00199_c = cd630_PATRIC.metabolites.get_by_id('cpd00199_c') # 4-Oxobutanoate (succinate semialdehyde)\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00005_c = cd630_PATRIC.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "cpd00006_c = cd630_PATRIC.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "\n",
    "CD630_23420 = cobra.Gene('CD630_23420')\n",
    "CD630_23420.name = 'sucD; Succinate-semialdehyde dehydrogenase'\n",
    "CD630_23420.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn16140_c = cobra.Reaction('rxn16140_c') \n",
    "rxn16140_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn16140_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn16140_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn16140_c.name = 'Succinate semialdehyde dehydrogenase'\n",
    "rxn16140_c.gene_reaction_rule = 'CD630_23420'\n",
    "rxn16140_c.lower_bound = -1000.\n",
    "rxn16140_c.upper_bound = 1000.\n",
    "rxn16140_c.add_metabolites({\n",
    "    cpd00078_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00005_c: -1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd00006_c: 1.0,\n",
    "    cpd00199_c: 1.0\n",
    "})\n",
    "\n",
    "# 4-hydroxybutyrate dehydrogenase: rxn01201_c\n",
    "# 4-Oxobutanoate --> 4-Hydroxybutanoate\n",
    "\n",
    "# 4-hydroxybutyrate CoA transferase: rxn03641_c\n",
    "# 4-Hydroxybutanoate --> 4-Hydroxybutyryl-CoA\n",
    "\n",
    "# 4-Hydroxybutyryl-CoA --> 3-Butenoyl-CoA\n",
    "# 3-Butenoyl-CoA --> Crotonyl-CoA \n",
    "# or directly...\n",
    "cpd07946_c = cd630_PATRIC.metabolites.get_by_id('cpd07946_c') # 4-Hydroxybutyryl-CoA\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00650_c = cd630_PATRIC.metabolites.get_by_id('cpd00650_c') # Crotonyl-CoA\n",
    "\n",
    "CD630_23410 = cobra.Gene('CD630_23410')\n",
    "CD630_23410.name = 'abfD; Gamma-aminobutyrate metabolism dehydratase/isomerase'\n",
    "CD630_23410.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn02167_c = cobra.Reaction('rxn02167_c') \n",
    "rxn02167_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn02167_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn02167_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn02167_c.name = '4-hydroxybutyryl-CoA dehydratase'\n",
    "rxn02167_c.gene_reaction_rule = 'CD630_23410'\n",
    "rxn02167_c.lower_bound = -1000.\n",
    "rxn02167_c.upper_bound = 1000.\n",
    "rxn02167_c.add_metabolites({\n",
    "    cpd07946_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd00650_c: 1.0\n",
    "})\n",
    "\n",
    "CD630_10540 = cobra.Gene('CD630_10540')\n",
    "CD630_10540.name = 'bcd2; Butyryl-CoA dehydrogenase'\n",
    "CD630_10540.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "# Crotonyl-CoA --> Butyryl-CoA\n",
    "cpd00120_c = cd630_PATRIC.metabolites.get_by_id('cpd00120_c') # Butyryl-CoA\n",
    "cpd00728_c = cd630_PATRIC.metabolites.get_by_id('cpd00728_c') # 4-Hydroxybutanoate\n",
    "\n",
    "rxn16586_c = cobra.Reaction('rxn16586_c') \n",
    "rxn16586_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn16586_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn16586_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn16586_c.name = 'Butyryl-CoA dehydrgenase'\n",
    "rxn16586_c.gene_reaction_rule = 'CD630_10540'\n",
    "rxn16586_c.lower_bound = 0.\n",
    "rxn16586_c.upper_bound = 1000.\n",
    "rxn16586_c.add_metabolites({\n",
    "    cpd00650_c: -1.0,\n",
    "    cpd00005_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00006_c: 1.0,\n",
    "    cpd00120_c: 1.0\n",
    "})\n",
    "\n",
    "CD630_23390 = cobra.Gene('CD630_23390')\n",
    "CD630_23390.name = 'cat2; 4-hydroxybutyrate CoA-transferase'\n",
    "CD630_23390.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "# Butyryl-CoA --> Butyrate\n",
    "cpd00211_c = cd630_PATRIC.metabolites.get_by_id('cpd00211_c') # Butyrate\n",
    "rxn13713_c = cobra.Reaction('rxn13713_c') \n",
    "rxn13713_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn13713_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn13713_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn13713_c.name = '4-hydroxybutyrate CoA-transferase'\n",
    "rxn13713_c.gene_reaction_rule = 'CD630_23390'\n",
    "rxn13713_c.lower_bound = -1000.\n",
    "rxn13713_c.upper_bound = 1000.\n",
    "rxn13713_c.add_metabolites({\n",
    "    cpd00120_c: -1.0,\n",
    "    cpd00728_c: -1.0,\n",
    "    cpd07946_c: 1.0,\n",
    "    cpd00211_c: 1.0\n",
    "})\n",
    "\n",
    "# Butyryl-CoA --> Butanoylphosphate\n",
    "cpd01662_c = cd630_PATRIC.metabolites.get_by_id('cpd01662_c') # Butanoylphosphate\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "\n",
    "rxn00871_c = cobra.Reaction('rxn00871_c') \n",
    "rxn00871_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00871_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn00871_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn00871_c.name = 'Phosphate butyrltransferase'\n",
    "rxn00871_c.gene_reaction_rule = 'CD630_01120'\n",
    "rxn00871_c.lower_bound = -1000.\n",
    "rxn00871_c.upper_bound = 1000.\n",
    "rxn00871_c.add_metabolites({\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00120_c: -1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd01662_c: 1.0\n",
    "})\n",
    "\n",
    "# Butanoylphosphate --> Butyrate: rxn01236_c (butyrate kinase)\n",
    "\n",
    "# Final export\n",
    "cpd00211_e = cobra.Metabolite(\n",
    "    'cpd00211_e',\n",
    "    formula='C4H7O2',\n",
    "    name='Butyrate',\n",
    "    compartment='extracellular')\n",
    "\n",
    "ID009_c = cobra.Reaction('ID009_c') # Diffusion\n",
    "ID009_c.name = 'Butyrate transport'\n",
    "ID009_c.lower_bound = -1000.\n",
    "ID009_c.upper_bound = 1000.\n",
    "ID009_c.add_metabolites({\n",
    "    cpd00211_e: -1.0,\n",
    "    cpd00211_c: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00211_e, type='exchange', reaction_id='EX_cpd00211_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn09271_c, rxn25164_c, rxn16140_c, rxn02167_c, rxn16586_c, rxn13713_c, rxn00871_c, ID009_c])\n",
    "\n",
    "#----------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Acetate\n",
    "cpd00029_e = cobra.Metabolite(\n",
    "    'cpd00029_e',\n",
    "    formula='C2H3O2',\n",
    "    name='Acetate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00029_e, type='exchange', reaction_id='EX_cpd00029_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00029_c = cd630_PATRIC.metabolites.get_by_id('cpd00029_c') # Acetate\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "rxn08061_c = cobra.Reaction('rxn08061_c') \n",
    "rxn08061_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn08061_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn08061_c.name = 'Acetate transport via proton symport'\n",
    "rxn08061_c.lower_bound = -1000.\n",
    "rxn08061_c.upper_bound = 1000.\n",
    "rxn08061_c.add_metabolites({\n",
    "    cpd00029_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00029_c: 1.0\n",
    "})\n",
    "\n",
    "# Acetate --> Acetyl-CoA in place already rxn03641_c\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "cpd00018_c = cd630_PATRIC.metabolites.get_by_id('cpd00018_c') # AMP\n",
    "rxn30338_c = cobra.Reaction('rxn30338_c') \n",
    "rxn30338_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn30338_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn30338_c.name = 'Acetyl-CoA synthetase'\n",
    "rxn30338_c.gene_reaction_rule = 'CD630_21960'\n",
    "rxn30338_c.lower_bound = 0.\n",
    "rxn30338_c.upper_bound = 1000.\n",
    "rxn30338_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00029_c: -1.0,\n",
    "    cpd00012_c: 1.0,\n",
    "    cpd00018_c: 1.0,\n",
    "    cpd00022_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn08061_c, rxn30338_c])\n",
    "\n",
    "#----------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Lactate \n",
    "cpd00159_e = cobra.Metabolite(\n",
    "    'cpd00159_e',\n",
    "    formula='C3H5O3',\n",
    "    name='L-Lactate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00159_e, type='exchange', reaction_id='EX_cpd00159_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00159_c = cd630_PATRIC.metabolites.get_by_id('cpd00159_c') # L-Lactate\n",
    "rxn05602_c = cobra.Reaction('rxn05602_c') \n",
    "rxn05602_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05602_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn05602_c.name = 'L-lactate transport via proton symport'\n",
    "rxn05602_c.lower_bound = -1000.\n",
    "rxn05602_c.upper_bound = 1000.\n",
    "rxn05602_c.add_metabolites({\n",
    "    cpd00159_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00159_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cpd00221_e = cobra.Metabolite(\n",
    "    'cpd00221_e',\n",
    "    formula='C3H5O3',\n",
    "    name='D-Lactate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00221_e, type='exchange', reaction_id='EX_cpd00221_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00221_c = cd630_PATRIC.metabolites.get_by_id('cpd00221_c') # D-Lactate\n",
    "rxn10171_c = cobra.Reaction('rxn10171_c') \n",
    "rxn10171_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10171_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn10171_c.name = 'D-lactate transport via proton symport'\n",
    "rxn10171_c.lower_bound = -1000.\n",
    "rxn10171_c.upper_bound = 1000.\n",
    "rxn10171_c.add_metabolites({\n",
    "    cpd00221_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00221_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn05602_c, rxn10171_c])\n",
    "\n",
    "#----------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Propionate formation\n",
    "# https://onlinelibrary.wiley.com/doi/pdf/10.1111/1462-2920.13589\n",
    "\n",
    "# Succinyl-CoA -> D-methylmalonyl-CoA\n",
    "cpd00519_c = cd630_PATRIC.metabolites.get_by_id('cpd00519_c') # D-methylmalonyl-CoA\n",
    "rxn00602_c = cobra.Reaction('rxn00602_c') \n",
    "rxn00602_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00602_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn00602_c.name = 'Methylmalonyl-CoA isomerase'\n",
    "rxn00602_c.gene_reaction_rule = 'CD630_10540'\n",
    "rxn00602_c.lower_bound = -1000.\n",
    "rxn00602_c.upper_bound = 1000.\n",
    "rxn00602_c.add_metabolites({\n",
    "    cpd00078_c: -1.0,\n",
    "    cpd00519_c: 1.0\n",
    "})\n",
    "\n",
    "# rxn00678_c: D-methylmalonyl-CoA -> Propionyl-CoA\n",
    "\n",
    "# Propionyl-CoA -> Propionate\n",
    "cpd00141_c = cd630_PATRIC.metabolites.get_by_id('cpd00141_c') # Propionate\n",
    "cpd00086_c = cd630_PATRIC.metabolites.get_by_id('cpd00086_c') # Propionyl-CoA\n",
    "rxn00669_c = cobra.Reaction('rxn00669_c') \n",
    "rxn00669_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00669_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn00669_c.name = 'Propanoate-CoA ligase'\n",
    "rxn00669_c.gene_reaction_rule = 'CD630_01120'\n",
    "rxn00669_c.lower_bound = -1000.\n",
    "rxn00669_c.upper_bound = 1000.\n",
    "rxn00669_c.add_metabolites({\n",
    "    cpd00086_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00141_c: 1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0\n",
    "})\n",
    "\n",
    "cpd00141_e = cobra.Metabolite(\n",
    "    'cpd00141_e',\n",
    "    formula='C3H5O2',\n",
    "    name='Propionate',\n",
    "    compartment='extracellular')\n",
    "rxn09172_c = cobra.Reaction('rxn09172_c') \n",
    "rxn09172_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn09172_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn09172_c.name = 'Propionate diffusion'\n",
    "rxn09172_c.lower_bound = -1000.\n",
    "rxn09172_c.upper_bound = 1000.\n",
    "rxn09172_c.add_metabolites({\n",
    "    cpd00141_c: -1.0,\n",
    "    cpd00141_e: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn00602_c, rxn00669_c, rxn09172_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00141_e, type='exchange', reaction_id='EX_cpd00141_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cpd15793_c not found\n",
      "cpd15794_c not found\n",
      "cpd15795_c not found\n"
     ]
    }
   ],
   "source": [
    "# Membrane components\n",
    "\n",
    "# Peptidoglycan subunits\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02009_c').lower_bound = 0.\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00731_c = cd630_PATRIC.metabolites.get_by_id('cpd00731_c') # Ala-Ala\n",
    "cpd02861_c = cd630_PATRIC.metabolites.get_by_id('cpd02861_c') # UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-L-lysine\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd02862_c = cd630_PATRIC.metabolites.get_by_id('cpd02862_c') # UDPMurAc(oyl-L-Ala-D-gamma-Glu-L-Lys-D-Ala-D-Ala)\n",
    "    \n",
    "CD630_26550 = cobra.Gene('CD630_26550')\n",
    "CD630_26550.name = 'murF; UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase'\n",
    "CD630_26550.notes = {'annotation': 'KEGG'}\n",
    "    \n",
    "rxn03141_c = cobra.Reaction('rxn03141_c') \n",
    "rxn03141_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn03141_c.notes['citations'] += ''\n",
    "rxn03141_c.name = 'UDP-N-acetylmuramoyl-tripeptide---D-alanyl-D-alanine ligase'\n",
    "rxn03141_c.gene_reaction_rule = 'CD630_26550'\n",
    "rxn03141_c.lower_bound = 0.\n",
    "rxn03141_c.upper_bound = 1000.\n",
    "rxn03141_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00731_c: -1.0,\n",
    "    cpd02861_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd02862_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn03141_c])\n",
    "\n",
    "\n",
    "# Ultimately synthesizes both:\n",
    "# cpd03495_c\tUndecaprenyl-diphospho-N-acetylmuramoyl--N-acetylglucosamine-L-ala-D-glu-meso-2-6-diaminopimeloyl-D-ala-D-ala\n",
    "# cpd03491_c\tUndecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-D-alanyl-D-alanine\n",
    "# to be included in peptidoglycan polymer\n",
    "\n",
    "#----------------------------------------------------------------------------#\n",
    "\n",
    "# Remove extraneous metabolites and associated reactions\n",
    "\n",
    "# Macromolecular components of ModelSEED Gram+ cell wall\n",
    "# Peptidoglycan polymer (n-1 subunits)\n",
    "# Peptidoglycan polymer (n subunits)\n",
    "# Stearoylcardiolipin (B. subtilis)\n",
    "# Isoheptadecanoylcardiolipin (B. subtilis)\n",
    "# Anteisoheptadecanoylcardiolipin (B. subtilis)\n",
    "# Stearoyllipoteichoic acid (n=24), linked, unsubstituted\n",
    "# Stearoyllipoteichoic acid (n=24), linked, D-alanine substituted\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, unsubstituted\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, N-acetyl-D-glucosamine\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, D-alanine substituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, D-alanine substituted\n",
    "# Stearoyllipoteichoic acid (n=24), linked, N-acetyl-D-glucosamine\n",
    "# Stearoyllipoteichoic acid (n=24), linked, glucose substituted\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, glucose substituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, glucose substituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, unsubstituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, N-acetyl-D-glucosamine\n",
    "# glycerol teichoic acid (n=45), linked, unsubstituted\n",
    "# glycerol teichoic acid (n=45), linked, D-ala substituted\n",
    "# glycerol teichoic acid (n=45), linked, glucose substituted\n",
    "\n",
    "macro = ['cpd15666_c','cpd15665_c','cpd15793_c','cpd15794_c','cpd15795_c','cpd15748_c','cpd15775_c', 'cpd15749_c','cpd15767_c','cpd15776_c','cpd15777_c','cpd15766_c','cpd15757_c','cpd15758_c', 'cpd15759_c','cpd15750_c','cpd15768_c','cpd15667_c','cpd15668_c','cpd15669_c']\n",
    "cd630_PATRIC = removeAll(cd630_PATRIC, macro)\n",
    "\n",
    "\n",
    "# B. subtilis-annotated cardiolipin molecules\n",
    "# Stearoylcardiolipin (B. subtilis)\n",
    "# Isoheptadecanoylcardiolipin (B. subtilis)\n",
    "# Anteisoheptadecanoylcardiolipin (B. subtilis)\n",
    "\n",
    "cardiolipin = ['cpd15793_c','cpd15794_c','cpd15795_c']\n",
    "cd630_PATRIC = removeAll(cd630_PATRIC, cardiolipin)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add missing intracellular metabolites\n",
    "\n",
    "cpd00314_c = cobra.Metabolite(\n",
    "    'cpd00314_c',\n",
    "    formula='C6H14O6',\n",
    "    name='D-Mannitol',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_23310 = cobra.Gene('CD630_23310')\n",
    "CD630_23310.name = 'mtlD; Mannitol-1-phosphate 5-dehydrogenase'\n",
    "CD630_23310.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00082_c = cd630_PATRIC.metabolites.get_by_id('cpd00082_c') # D-Fructose\n",
    "\n",
    "rxn00629_c = cobra.Reaction('rxn00629_c')\n",
    "rxn00629_c.gene_reaction_rule = 'CD630_23310'\n",
    "rxn00629_c.name = 'D-Mannitol:NAD+ 2-oxidoreductase'\n",
    "rxn00629_c.lower_bound = -1000.\n",
    "rxn00629_c.upper_bound = 1000.\n",
    "rxn00629_c.add_metabolites({\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00314_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00082_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00629_c])\n",
    "\n",
    "#-------------------------------------------------------------------------------------#\n",
    "\n",
    "cpd00588_c = cobra.Metabolite(\n",
    "    'cpd00588_c',\n",
    "    formula='C6H14O6',\n",
    "    name='Sorbitol',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_07680 = cobra.Gene('CD630_07680')\n",
    "CD630_07680.name = 'srlD; Sorbitol 6-phosphate 2-dehydrogenase (Glucitol)'\n",
    "CD630_07680.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn00634_c = cobra.Reaction('rxn00634_c')\n",
    "rxn00634_c.gene_reaction_rule = 'CD630_07680'\n",
    "rxn00634_c.name = 'D-Glucitol:NAD+ 2-oxidoreductase'\n",
    "rxn00634_c.lower_bound = -1000.\n",
    "rxn00634_c.upper_bound = 1000.\n",
    "rxn00634_c.add_metabolites({\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00588_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00082_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00634_c])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Biomass composition\n",
    "\n",
    "Adapted from..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DNA replication\n",
    "\n",
    "cpd00115_c = cd630_PATRIC.metabolites.get_by_id('cpd00115_c') # dATP\n",
    "cpd00356_c = cd630_PATRIC.metabolites.get_by_id('cpd00356_c') # dCTP\n",
    "cpd00357_c = cd630_PATRIC.metabolites.get_by_id('cpd00357_c') # TTP\n",
    "cpd00241_c = cd630_PATRIC.metabolites.get_by_id('cpd00241_c') # dGTP\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "\n",
    "cpd17042_c = cobra.Metabolite(\n",
    "    'cpd17042_c',\n",
    "    formula='',\n",
    "    name='DNA polymer',\n",
    "    compartment='cytosol')\n",
    "cpd17042_c.notes = {'Macromolecule'}\n",
    "\n",
    "dna_rxn = cobra.Reaction('dna_rxn')\n",
    "dna_rxn.name = 'DNA replication'\n",
    "dna_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "dna_rxn.lower_bound = 0.\n",
    "dna_rxn.upper_bound = 1000.\n",
    "dna_rxn.add_metabolites({\n",
    "    cpd00115_c: -1.0,\n",
    "    cpd00356_c: -0.5,\n",
    "    cpd00357_c: -1.0,\n",
    "    cpd00241_c: -0.5,\n",
    "    cpd00002_c: -4.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17042_c: 1.0,\n",
    "    cpd00008_c: 4.0,\n",
    "    cpd00009_c: 4.0,\n",
    "    cpd00012_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([dna_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA transcription\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00052_c = cd630_PATRIC.metabolites.get_by_id('cpd00052_c') # CTP\n",
    "cpd00062_c = cd630_PATRIC.metabolites.get_by_id('cpd00062_c') # UTP\n",
    "cpd00038_c = cd630_PATRIC.metabolites.get_by_id('cpd00038_c') # GTP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "\n",
    "cpd17043_c = cobra.Metabolite(\n",
    "    'cpd17043_c',\n",
    "    formula='',\n",
    "    name='RNA polymer',\n",
    "    compartment='cytosol')\n",
    "cpd17043_c.notes = {'Macromolecule'}\n",
    "\n",
    "rna_rxn = cobra.Reaction('rna_rxn')\n",
    "rna_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "rna_rxn.notes['citations'] += ''\n",
    "rna_rxn.name = 'RNA transcription'\n",
    "rna_rxn.lower_bound = 0.\n",
    "rna_rxn.upper_bound = 1000.\n",
    "rna_rxn.add_metabolites({\n",
    "    cpd00002_c: -2.0,\n",
    "    cpd00052_c: -0.5,\n",
    "    cpd00062_c: -0.5,\n",
    "    cpd00038_c: -0.5,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17043_c: 1.0,\n",
    "    cpd00008_c: 2.0,\n",
    "    cpd00009_c: 2.0,\n",
    "    cpd00012_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rna_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Protein biosynthesis\n",
    "\n",
    "# tRNA synthesis\n",
    "cpd00035_c = cd630_PATRIC.metabolites.get_by_id('cpd00035_c') # L-Alanine\n",
    "\n",
    "\n",
    "cpd00051_c = cd630_PATRIC.metabolites.get_by_id('cpd00051_c') # L-Arginine\n",
    "cpd00132_c = cd630_PATRIC.metabolites.get_by_id('cpd00132_c') # L-Asparagine\n",
    "cpd00041_c = cd630_PATRIC.metabolites.get_by_id('cpd00041_c') # L-Aspartate\n",
    "cpd00084_c = cd630_PATRIC.metabolites.get_by_id('cpd00084_c') # L-Cysteine\n",
    "cpd00053_c = cd630_PATRIC.metabolites.get_by_id('cpd00053_c') # L-Glutamine\n",
    "cpd00023_c = cd630_PATRIC.metabolites.get_by_id('cpd00023_c') # L-Glutamate\n",
    "cpd00033_c = cd630_PATRIC.metabolites.get_by_id('cpd00033_c') # Glycine\n",
    "cpd00119_c = cd630_PATRIC.metabolites.get_by_id('cpd00119_c') # L-Histidine\n",
    "cpd00322_c = cd630_PATRIC.metabolites.get_by_id('cpd00322_c') # L-Isoleucine\n",
    "cpd00107_c = cd630_PATRIC.metabolites.get_by_id('cpd00107_c') # L-Leucine\n",
    "cpd00039_c = cd630_PATRIC.metabolites.get_by_id('cpd00039_c') # L-Lysine\n",
    "cpd00060_c = cd630_PATRIC.metabolites.get_by_id('cpd00060_c') # L-Methionine\n",
    "cpd00066_c = cd630_PATRIC.metabolites.get_by_id('cpd00066_c') # L-Phenylalanine\n",
    "cpd00129_c = cd630_PATRIC.metabolites.get_by_id('cpd00129_c') # L-Proline\n",
    "cpd00054_c = cd630_PATRIC.metabolites.get_by_id('cpd00054_c') # L-Serine\n",
    "cpd00161_c = cd630_PATRIC.metabolites.get_by_id('cpd00161_c') # L-Threonine\n",
    "cpd00065_c = cd630_PATRIC.metabolites.get_by_id('cpd00065_c') # L-Tryptophan\n",
    "cpd00069_c = cd630_PATRIC.metabolites.get_by_id('cpd00069_c') # L-Tyrosine\n",
    "cpd00156_c = cd630_PATRIC.metabolites.get_by_id('cpd00156_c') # L-Valine\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "cpd17041_c = cobra.Metabolite(\n",
    "    'cpd17041_c',\n",
    "    formula='',\n",
    "    name='Protein polymer',\n",
    "    compartment='cytosol')\n",
    "cpd17041_c.notes = {'Macromolecule'}\n",
    "\n",
    "protein_rxn = cobra.Reaction('protein_rxn')\n",
    "protein_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "protein_rxn.notes['citations'] += ''\n",
    "protein_rxn.name = 'Protein biosynthesis'\n",
    "protein_rxn.lower_bound = 0.\n",
    "protein_rxn.upper_bound = 1000.\n",
    "protein_rxn.add_metabolites({\n",
    "    cpd00035_c: -0.5,\n",
    "    cpd00051_c: -0.25,\n",
    "    cpd00132_c: -0.5,\n",
    "    cpd00041_c: -0.5,\n",
    "    cpd00084_c: -0.05,\n",
    "    cpd00053_c: -0.25,\n",
    "    cpd00023_c: -0.5,\n",
    "    cpd00033_c: -0.5,\n",
    "    cpd00119_c: -0.05,\n",
    "    cpd00322_c: -0.5,\n",
    "    cpd00107_c: -0.5,\n",
    "    cpd00039_c: -0.5,\n",
    "    cpd00060_c: -0.25,\n",
    "    cpd00066_c: -0.5,\n",
    "    cpd00129_c: -0.25,\n",
    "    cpd00054_c: -0.5,\n",
    "    cpd00161_c: -0.5,\n",
    "    cpd00065_c: -0.05,\n",
    "    cpd00069_c: -0.25,\n",
    "    cpd00156_c: -0.5,\n",
    "    cpd00002_c: -20.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17041_c: 1.0,\n",
    "    cpd00008_c: 20.0,\n",
    "    cpd00009_c: 20.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([protein_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cell wall synthesis\n",
    "\n",
    "cpd02967_c = cd630_PATRIC.metabolites.get_by_id('cpd02967_c') # N-Acetyl-beta-D-mannosaminyl-1,4-N-acetyl-D-glucosaminyldiphosphoundecaprenol\n",
    "cpd00402_c = cd630_PATRIC.metabolites.get_by_id('cpd00402_c') # CDPglycerol\n",
    "cpd00046_c = cd630_PATRIC.metabolites.get_by_id('cpd00046_c') # CMP\n",
    "\n",
    "cpd12894_c = cobra.Metabolite(\n",
    "    'cpd12894_c',\n",
    "    formula='',\n",
    "    name='Teichoic acid',\n",
    "    compartment='cytosol')\n",
    "cpd12894_c.notes = {'type':'Macromolecule'}\n",
    "\n",
    "CD630_02440 = cobra.Gene('CD630_02440')\n",
    "CD630_02440.name = 'Putative CDP-glycerol:Poly(Glycerophosphate) glycerophosphotransferase'\n",
    "CD630_02440.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "teichoicacid_rxn = cobra.Reaction('teichoicacid_rxn')\n",
    "teichoicacid_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "teichoicacid_rxn.notes['citations'] += ''\n",
    "teichoicacid_rxn.gene_reaction_rule = 'CD630_02440'\n",
    "teichoicacid_rxn.name = 'Teichoic acid biosynthesis'\n",
    "teichoicacid_rxn.lower_bound = 0.\n",
    "teichoicacid_rxn.upper_bound = 1000.\n",
    "teichoicacid_rxn.add_metabolites({\n",
    "    cpd02967_c: -1.0,\n",
    "    cpd00402_c: -1.0,\n",
    "    cpd00046_c: 1.0,\n",
    "    cpd12894_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([teichoicacid_rxn])\n",
    "\n",
    "#---------------------------------------#\n",
    "\n",
    "# Peptidoglycan subunits\n",
    " # Undecaprenyl-diphospho-N-acetylmuramoyl--N-acetylglucosamine-L-ala-D-glu-meso-2-6-diaminopimeloyl-D-ala-D-ala (right)\n",
    "cpd03495_c = cd630_PATRIC.metabolites.get_by_id('cpd03495_c')\n",
    "# Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-D-alanyl-D-alanine (left)\n",
    "cpd03491_c = cd630_PATRIC.metabolites.get_by_id('cpd03491_c') \n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd02229_c = cd630_PATRIC.metabolites.get_by_id('cpd02229_c') # Bactoprenyl diphosphate\n",
    "cpd00117_c = cd630_PATRIC.metabolites.get_by_id('cpd00117_c') # D-Alanine\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "cpd16661_c = cobra.Metabolite(\n",
    "    'cpd16661_c',\n",
    "    formula='',\n",
    "    name='Peptidoglycan polymer',\n",
    "    compartment='cytosol')\n",
    "cpd16661_c.notes = {'type':'Macromolecule'}\n",
    "                            \n",
    "CD630_26510 = cobra.Gene('CD630_26510')\n",
    "CD630_26510.name = 'murG; UDP-N-acetylglucosamine--N-acetylmuramyl-(pentapeptide) pyrophosphoryl-undecaprenol N-acetylglucosamine transferase'\n",
    "CD630_26510.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "peptidoglycan_rxn = cobra.Reaction('peptidoglycan_rxn')\n",
    "peptidoglycan_rxn.gene_reaction_rule = 'CD630_26510'\n",
    "peptidoglycan_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "peptidoglycan_rxn.name = 'Peptidoglycan synthesis'\n",
    "peptidoglycan_rxn.lower_bound = 0.\n",
    "peptidoglycan_rxn.upper_bound = 1000.\n",
    "peptidoglycan_rxn.add_metabolites({\n",
    "    cpd03491_c: -1.0,\n",
    "    cpd03495_c: -1.0,\n",
    "    cpd00002_c: -4.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd16661_c: 1.0,\n",
    "    cpd02229_c: 1.0,\n",
    "    cpd00117_c: 0.5, # D-Alanine\n",
    "    cpd00008_c: 4.0, # ADP\n",
    "    cpd00009_c: 4.0 # Phosphate\n",
    "})\n",
    "cd630_PATRIC.add_reactions([peptidoglycan_rxn])\n",
    "\n",
    "#---------------------------------------#\n",
    "\n",
    "cellwall_c = cobra.Metabolite(\n",
    "    'cellwall_c',\n",
    "    formula='',\n",
    "    name='Cell Wall polymer',\n",
    "    compartment='cytosol')\n",
    "cellwall_c.notes = {'type':'Macromolecule'}\n",
    "\n",
    "cellwall_rxn = cobra.Reaction('cellwall_rxn')\n",
    "cellwall_rxn.name = 'Cell wall biosynthesis'\n",
    "cellwall_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "cellwall_rxn.lower_bound = 0.\n",
    "cellwall_rxn.upper_bound = 1000.\n",
    "cellwall_rxn.add_metabolites({\n",
    "    cpd16661_c: -1.5,\n",
    "    cpd12894_c: -0.05,\n",
    "    cellwall_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([cellwall_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Lipid pool\n",
    "\n",
    "cpd15543_c = cd630_PATRIC.metabolites.get_by_id('cpd15543_c') # Phosphatidylglycerophosphate ditetradecanoyl\n",
    "cpd15545_c = cd630_PATRIC.metabolites.get_by_id('cpd15545_c') # Phosphatidylglycerophosphate dihexadecanoyl\n",
    "cpd15540_c = cd630_PATRIC.metabolites.get_by_id('cpd15540_c') # Phosphatidylglycerol dioctadecanoyl\n",
    "cpd15728_c = cd630_PATRIC.metabolites.get_by_id('cpd15728_c') # Diglucosyl-1,2 dipalmitoylglycerol\n",
    "cpd15729_c = cd630_PATRIC.metabolites.get_by_id('cpd15729_c') # Diglucosyl-1,2 dimyristoylglycerol\n",
    "cpd15737_c = cd630_PATRIC.metabolites.get_by_id('cpd15737_c') # Monoglucosyl-1,2 dipalmitoylglycerol\n",
    "cpd15738_c = cd630_PATRIC.metabolites.get_by_id('cpd15738_c') # Monoglucosyl-1,2 dimyristoylglycerol\n",
    "\n",
    "cpd11852_c = cobra.Metabolite(\n",
    "    'cpd11852_c',\n",
    "    formula='',\n",
    "    name='Lipid Pool',\n",
    "    compartment='cytosol')\n",
    "cpd11852_c.notes = {'type':'Metabolite pool'}\n",
    "\n",
    "lipid_rxn = cobra.Reaction('lipid_rxn')\n",
    "lipid_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "lipid_rxn.notes['citations'] += ''\n",
    "lipid_rxn.name = 'Lipid composition'\n",
    "lipid_rxn.lower_bound = 0.\n",
    "lipid_rxn.upper_bound = 1000.\n",
    "lipid_rxn.add_metabolites({\n",
    "    cpd15543_c: -0.005,\n",
    "    cpd15545_c: -0.005,\n",
    "    cpd15540_c: -0.005,\n",
    "    cpd15728_c: -0.005,\n",
    "    cpd15729_c: -0.005,\n",
    "    cpd15737_c: -0.005,\n",
    "    cpd15738_c: -0.005,\n",
    "    cpd11852_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([lipid_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ions, Vitamins, & Cofactors\n",
    "\n",
    "# Vitamins\n",
    "cpd00104_c = cd630_PATRIC.metabolites.get_by_id('cpd00104_c') # Biotin          MDM\n",
    "cpd00644_c = cd630_PATRIC.metabolites.get_by_id('cpd00644_c') # Pantothenate    MDM\n",
    "cpd00263_c = cd630_PATRIC.metabolites.get_by_id('cpd00263_c') # Pyridoxine      MDM\n",
    "cpd00393_c = cd630_PATRIC.metabolites.get_by_id('cpd00393_c') #  folate\n",
    "cpd00133_c = cd630_PATRIC.metabolites.get_by_id('cpd00133_c') #  nicotinamide\n",
    "cpd00443_c = cd630_PATRIC.metabolites.get_by_id('cpd00443_c') #  p-aminobenzoic acid\n",
    "cpd00220_c = cd630_PATRIC.metabolites.get_by_id('cpd00220_c') #  riboflavin  \n",
    "cpd00305_c = cd630_PATRIC.metabolites.get_by_id('cpd00305_c') #  thiamin\n",
    "\n",
    "# Ions\n",
    "cpd00149_c = cd630_PATRIC.metabolites.get_by_id('cpd00149_c') # Cobalt\n",
    "cpd00030_c = cd630_PATRIC.metabolites.get_by_id('cpd00030_c') # Manganese\n",
    "cpd00254_c = cd630_PATRIC.metabolites.get_by_id('cpd00254_c') # Magnesium\n",
    "cpd00971_c = cd630_PATRIC.metabolites.get_by_id('cpd00971_c') # Sodium\n",
    "cpd00063_c = cd630_PATRIC.metabolites.get_by_id('cpd00063_c') # Calcium\n",
    "cpd10515_c = cd630_PATRIC.metabolites.get_by_id('cpd10515_c') # Iron\n",
    "cpd00205_c = cd630_PATRIC.metabolites.get_by_id('cpd00205_c') # Potassium\n",
    "cpd00099_c = cd630_PATRIC.metabolites.get_by_id('cpd00099_c') # Chloride\n",
    "\n",
    "# Cofactors\n",
    "cpd00022_c = cd630_PATRIC.metabolites.get_by_id('cpd00022_c') # Acetyl-CoA\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "cpd00015_c = cd630_PATRIC.metabolites.get_by_id('cpd00015_c') # FAD\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00006_c = cd630_PATRIC.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd00005_c = cd630_PATRIC.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "\n",
    "# Energy molecules\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "cpd00038_c = cd630_PATRIC.metabolites.get_by_id('cpd00038_c') # GTP\n",
    "cpd00031_c = cd630_PATRIC.metabolites.get_by_id('cpd00031_c') # GDP\n",
    "\n",
    "#cpd00274_c = cd630_PATRIC.metabolites.get_by_id('cpd00274_c') # Citrulline\n",
    "#cpd00907_c = cd630_PATRIC.metabolites.get_by_id('cpd00907_c') # alpha-D-glucose-1,6-biphosphate\n",
    "\n",
    "cofactor_c = cobra.Metabolite(\n",
    "    'cofactor_c',\n",
    "    formula='',\n",
    "    name='Cofactor Pool',\n",
    "    compartment='cytosol')\n",
    "cofactor_c.notes = {'type':'Metabolite pool'}\n",
    "\n",
    "cofactor_rxn = cobra.Reaction('cofactor_rxn')\n",
    "cofactor_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "cofactor_rxn.notes['citations'] += 'Karasawa T, Ikoma S, Yamakawa K, & Nakamura S. (1995). A defined growth medium for Clostridium difficile.Microbiology. 141(2); 371-5.'\n",
    "cofactor_rxn.name = 'Cofactor Pool'\n",
    "cofactor_rxn.lower_bound = 0.\n",
    "cofactor_rxn.upper_bound = 1000.\n",
    "cofactor_rxn.add_metabolites({\n",
    "    cpd00104_c: -0.005, \n",
    "    cpd00644_c: -0.005, \n",
    "    cpd00263_c: -0.005, \n",
    "    cpd00393_c: -0.005,\n",
    "    cpd00133_c: -0.005,\n",
    "    cpd00443_c: -0.005,\n",
    "    cpd00220_c: -0.005,\n",
    "    cpd00305_c: -0.005,\n",
    "    cpd00149_c: -0.005, \n",
    "    cpd00030_c: -0.005, \n",
    "    cpd00254_c: -0.005, \n",
    "    cpd00971_c: -0.005, \n",
    "    cpd00063_c: -0.005, \n",
    "    cpd10515_c: -0.005, \n",
    "    cpd00205_c: -0.005,\n",
    "    cpd00099_c: -0.005, \n",
    "    cpd00022_c: -0.005,\n",
    "    cpd00010_c: -0.0005,\n",
    "    cpd00015_c: -0.0005,\n",
    "    cpd00003_c: -0.005,\n",
    "    cpd00004_c: -0.005,\n",
    "    cpd00006_c: -0.005,\n",
    "    cpd00005_c: -0.005,\n",
    "    cpd00002_c: -0.005, \n",
    "    cpd00008_c: -0.005, \n",
    "    cpd00009_c: -0.5,\n",
    "    cpd00012_c: -0.005, \n",
    "    cpd00038_c: -0.005, \n",
    "    cpd00031_c: -0.005, \n",
    "\n",
    "    cofactor_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([cofactor_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final Biomass\n",
    "\n",
    "cpd11416_c = cobra.Metabolite(\n",
    "    'cpd11416_c',\n",
    "    formula='',\n",
    "    name='Biomass',\n",
    "    compartment='cytosol')\n",
    "cpd11416_c.notes = {'type':'Biomass'}\n",
    "\n",
    "biomass_rxn = cobra.Reaction('biomass')\n",
    "biomass_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "biomass_rxn.name = 'Biomass Reaction'\n",
    "biomass_rxn.lower_bound = 0.\n",
    "biomass_rxn.upper_bound = 1000.\n",
    "biomass_rxn.add_metabolites({\n",
    "    cpd17041_c: -0.4,  # Protein\n",
    "    cpd17043_c: -0.15, # RNA\n",
    "    cpd17042_c: -0.05, # DNA\n",
    "    cpd11852_c: -0.05, # Lipid\n",
    "    cellwall_c: -0.2,\n",
    "    cofactor_c: -0.2,\n",
    "    cpd00001_c: -20.0,\n",
    "    cpd00002_c: -20.0,\n",
    "    cpd00008_c: 20.0,\n",
    "    cpd00009_c: 20.0,\n",
    "    cpd11416_c: 1.0 # Biomass\n",
    "})\n",
    "cd630_PATRIC.add_reactions([biomass_rxn])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd11416_c, type='sink', reaction_id='EX_biomass', lb=0.0, ub=1000.0)\n",
    "\n",
    "# Set new objective\n",
    "cd630_PATRIC.objective = biomass_rxn\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Deal with extracellular metabolites without transporters          # sulfate     # CoA\n",
    "for cpd in ['cpd10516_e', 'cpd01188_e', 'cpd00028_e', 'cpd11606_e', 'cpd00048_e', 'cpd00010_e']:\n",
    "    try:\n",
    "        cd630_PATRIC.metabolites.get_by_id(cpd).remove_from_model()\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "#---------------------------------------------------------------------------#\n",
    "\n",
    "# Manually add necessary reactions\n",
    "\n",
    "# Diffusion reactions\n",
    "\n",
    "# H2O\n",
    "cpd00001_e = cd630_PATRIC.metabolites.get_by_id('cpd00001_e')\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c')\n",
    "rxn08686_c = cobra.Reaction('rxn08686_c') \n",
    "rxn08686_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn08686_c.name = 'H2O transport via diffusion'\n",
    "rxn08686_c.lower_bound = -1000.\n",
    "rxn08686_c.upper_bound = 1000.\n",
    "rxn08686_c.add_metabolites({\n",
    "    cpd00001_e: -1.0,\n",
    "    cpd00001_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08686_c])\n",
    "\n",
    "# O2\n",
    "cpd00007_e = cd630_PATRIC.metabolites.get_by_id('cpd00007_e')\n",
    "cpd00007_c = cd630_PATRIC.metabolites.get_by_id('cpd00007_c')\n",
    "rxn09031_c = cobra.Reaction('rxn09031_c') \n",
    "rxn09031_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn09031_c.name = 'Oxygen transport via diffusion'\n",
    "rxn09031_c.lower_bound = -1000.\n",
    "rxn09031_c.upper_bound = 1000.\n",
    "rxn09031_c.add_metabolites({\n",
    "    cpd00007_e: -1.0,\n",
    "    cpd00007_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn09031_c])\n",
    "\n",
    "# CO2\n",
    "cpd00011_e = cd630_PATRIC.metabolites.get_by_id('cpd00011_e')\n",
    "cpd00011_c = cd630_PATRIC.metabolites.get_by_id('cpd00011_c')\n",
    "rxn05467_c = cobra.Reaction('rxn05467_c') \n",
    "rxn05467_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn05467_c.name = 'CO2 transport via diffusion'\n",
    "rxn05467_c.lower_bound = -1000.\n",
    "rxn05467_c.upper_bound = 1000.\n",
    "rxn05467_c.add_metabolites({\n",
    "    cpd00011_e: -1.0,\n",
    "    cpd00011_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05467_c])\n",
    "\n",
    "# Ca2+\n",
    "cpd00063_e = cd630_PATRIC.metabolites.get_by_id('cpd00063_e')\n",
    "cpd00063_c = cd630_PATRIC.metabolites.get_by_id('cpd00063_c')\n",
    "rxn08186_c = cobra.Reaction('rxn08186_c') \n",
    "rxn08186_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn08186_c.name = 'Calcium transport via diffusion'\n",
    "rxn08186_c.lower_bound = -1000.\n",
    "rxn08186_c.upper_bound = 1000.\n",
    "rxn08186_c.add_metabolites({\n",
    "    cpd00063_e: -1.0,\n",
    "    cpd00063_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08186_c])\n",
    "\n",
    "# Mn2+\n",
    "cpd00030_e = cd630_PATRIC.metabolites.get_by_id('cpd00030_e')\n",
    "cpd00030_c = cd630_PATRIC.metabolites.get_by_id('cpd00030_c')\n",
    "rxn08952_c = cobra.Reaction('rxn08952_c') \n",
    "rxn08952_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn08952_c.name = 'Manganese transport via diffusion'\n",
    "rxn08952_c.lower_bound = -1000.\n",
    "rxn08952_c.upper_bound = 1000.\n",
    "rxn08952_c.add_metabolites({\n",
    "    cpd00030_e: -1.0,\n",
    "    cpd00030_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08952_c])\n",
    "\n",
    "#---------------------------------#\n",
    "\n",
    "# Active transport\n",
    "\n",
    "# ocdca\n",
    "cpd01080_e = cd630_PATRIC.metabolites.get_by_id('cpd01080_e')\n",
    "cpd01080_c = cd630_PATRIC.metabolites.get_by_id('cpd01080_c')\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e')\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c')\n",
    "\n",
    "rxn10161_c = cobra.Reaction('rxn10161_c') \n",
    "rxn10161_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn10161_c.name = 'Octadecanoate transport via proton symport '\n",
    "rxn10161_c.lower_bound = -1000.\n",
    "rxn10161_c.upper_bound = 1000.\n",
    "rxn10161_c.add_metabolites({\n",
    "    cpd01080_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd01080_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10161_c])\n",
    "\n",
    "# Spermidine\n",
    "cpd00264_e = cd630_PATRIC.metabolites.get_by_id('cpd00264_e')\n",
    "cpd00264_c = cd630_PATRIC.metabolites.get_by_id('cpd00264_c')\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c')\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c')\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c')\n",
    "\n",
    "rxn05175_c = cobra.Reaction('rxn05175_c') \n",
    "rxn05175_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05175_c.name = 'Spermidine transport via ABC system'\n",
    "rxn05175_c.lower_bound = 0.\n",
    "rxn05175_c.upper_bound = 1000.\n",
    "rxn05175_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00264_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00264_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05175_c])\n",
    "\n",
    "# meso-2,6-Diaminopimelate\n",
    "cpd00516_e = cd630_PATRIC.metabolites.get_by_id('cpd00516_e')\n",
    "cpd00516_c = cd630_PATRIC.metabolites.get_by_id('cpd00516_c')\n",
    "\n",
    "rxn08304_c = cobra.Reaction('rxn08304_c') \n",
    "rxn08304_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn08304_c.name = 'M-diaminopimelic acid ABC transport'\n",
    "rxn08304_c.lower_bound = 0.\n",
    "rxn08304_c.upper_bound = 1000.\n",
    "rxn08304_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00516_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00516_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08304_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reversible reactions containing ATP: 87\n"
     ]
    }
   ],
   "source": [
    "# Check irreversibility of ATP consuming reactions\n",
    "\n",
    "reversible = []\n",
    "for rxn in list(cd630_PATRIC.metabolites.get_by_id('cpd00002_c').reactions):\n",
    "    if rxn.reversibility == True:\n",
    "        reversible.append(rxn.id)\n",
    "print('Reversible reactions containing ATP: ' + str(len(reversible)))\n",
    "\n",
    "# Set correct reversibility\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08298_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01396_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02503_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01517_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02262_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08297_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10257_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10256_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00392_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00134_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00669_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08295_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01100_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02314_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00337_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01434_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01917_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01127_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02175_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01513_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02489_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00547_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01343_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00077_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03536_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01739_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00151_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01807_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01135_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10255_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00772_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08294_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01607_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00690_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03075_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02362_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00303_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10258_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01671_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05054_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn06672_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn09320_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00364_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08647_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn04384_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn07584_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00695_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02937_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10253_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00615_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01114_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn12510_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00124_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00181_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00440_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03108_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01219_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10042_c').lower_bound = 0. # Proton motive force...\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00239_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02484_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00770_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03908_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05814_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00105_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn07841_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08296_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01300_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01951_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08299_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01492_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01492_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00097_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01509_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00780_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01754_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02155_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00545_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01102_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10254_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05145_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00119_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01199_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00410_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08300_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03540_c').lower_bound = 0.\n",
    "\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01236_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01987_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00985_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00148_c').upper_bound = 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reversible reactions containing GTP: 3\n"
     ]
    }
   ],
   "source": [
    "# Check irreversibility of GTP consuming reactions\n",
    "\n",
    "reversible = []\n",
    "for rxn in list(cd630_PATRIC.metabolites.get_by_id('cpd00038_c').reactions):\n",
    "    if rxn.reversibility == True:\n",
    "        reversible.append(rxn.id)\n",
    "print('Reversible reactions containing GTP: ' + str(len(reversible)))\n",
    "\n",
    "# Fix bounds\n",
    "cd630_PATRIC.reactions.get_by_id('rxn04413_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03537_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00641_c').lower_bound = 0.\n",
    "\n",
    "\n",
    "# GTP generation\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c')\n",
    "cpd00031_c = cd630_PATRIC.metabolites.get_by_id('cpd00031_c')\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c')\n",
    "cpd00038_c = cd630_PATRIC.metabolites.get_by_id('cpd00038_c')\n",
    "rxn00237_c = cobra.Reaction('rxn00237_c') \n",
    "rxn00237_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'Nucleotide biosynthesis'}\n",
    "rxn00237_c.name = 'GDP phosphotransferase'\n",
    "rxn00237_c.lower_bound = 0.\n",
    "rxn00237_c.upper_bound = 1000.\n",
    "rxn05175_c.add_metabolites({\n",
    "    cpd00031_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00038_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00237_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remaining minimal media components\n",
    "\n",
    "# Ammonia (extracellular)\n",
    "cpd00013_e = cobra.Metabolite(\n",
    "    'cpd00013_e',\n",
    "    formula='H4N',\n",
    "    name='Ammonia',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00013_e, type='exchange', reaction_id='EX_cpd00013_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c')\n",
    "rxn05466_c = cobra.Reaction('rxn05466_c') \n",
    "rxn05466_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05466_c.name = 'Ammonia diffusion'\n",
    "rxn05466_c.lower_bound = -1000.\n",
    "rxn05466_c.upper_bound = 1000.\n",
    "rxn05466_c.add_metabolites({\n",
    "    cpd00013_e: -1.0,\n",
    "    cpd00013_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05466_c])\n",
    "\n",
    "# Carbonate (extracellular)\n",
    "cpd00242_c = cd630_PATRIC.metabolites.get_by_id('cpd00242_c')\n",
    "cpd00242_e = cobra.Metabolite(\n",
    "    'cpd00242_e',\n",
    "    formula='CHO3',\n",
    "    name='Carbonate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00242_e, type='exchange', reaction_id='EX_cpd00242_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "rxn10826_c = cobra.Reaction('rxn10826_c') \n",
    "rxn10826_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10826_c.name = 'Carbonate diffusion'\n",
    "rxn10826_c.lower_bound = -1000.\n",
    "rxn10826_c.upper_bound = 1000.\n",
    "rxn10826_c.add_metabolites({\n",
    "    cpd00242_e: -1.0,\n",
    "    cpd00242_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10826_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# FAD biosynthesis\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00050_c = cd630_PATRIC.metabolites.get_by_id('cpd00050_c') # FMN\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "cpd00015_c = cd630_PATRIC.metabolites.get_by_id('cpd00015_c') # FAD\n",
    "\n",
    "try:\n",
    "    cd630_PATRIC.reactions.get_by_id('rxn30210_c').remove_from_model()\n",
    "except:\n",
    "    pass\n",
    "\n",
    "rxn30210_c = cobra.Reaction('rxn30210_c') \n",
    "rxn30210_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn30210_c.name = 'FAD synthetase'\n",
    "rxn30210_c.gene_reaction_rule = '272563.8.peg.1382 and 272563.8.peg.1382'\n",
    "rxn30210_c.lower_bound = 0.\n",
    "rxn30210_c.upper_bound = 1000.\n",
    "rxn30210_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00050_c: -1.0,\n",
    "    cpd00012_c: 1.0,\n",
    "    cpd00015_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn30210_c])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import of last missing vitamins\n",
    "\n",
    "# Nicotinamide\n",
    "cpd00133_e = cobra.Metabolite(\n",
    "    'cpd00133_e',\n",
    "    formula='C6H6N2O',\n",
    "    name='Nicotinamide',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00133_e, type='exchange', reaction_id='EX_cpd00133_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00133_c = cd630_PATRIC.metabolites.get_by_id('cpd00133_c')\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c')\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e')\n",
    "rxn11334_c = cobra.Reaction('rxn11334_c') \n",
    "rxn11334_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn11334_c.name = 'Nicotinamide symporter'\n",
    "rxn11334_c.gene_reaction_rule = '272563.8.peg.164 or 272563.8.peg.1796'\n",
    "rxn11334_c.lower_bound = 0.\n",
    "rxn11334_c.upper_bound = 1000.\n",
    "rxn11334_c.add_metabolites({\n",
    "    cpd00133_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00133_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn11334_c])\n",
    "\n",
    "# p-Aminobenzoic acid (ABEE)\n",
    "cpd00443_e = cobra.Metabolite(\n",
    "    'cpd00443_e',\n",
    "    formula='C7H6NO2',\n",
    "    name='Nicotinamide',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00443_e, type='exchange', reaction_id='EX_cpd00443_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00443_c = cd630_PATRIC.metabolites.get_by_id('cpd00443_c') #  p-aminobenzoic acid\n",
    "rxn10421_c = cobra.Reaction('rxn10421_c') \n",
    "rxn10421_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10421_c.name = 'p-Aminobenzoic acid symporter'\n",
    "rxn10421_c.gene_reaction_rule = '272563.8.peg.164 or 272563.8.peg.1796'\n",
    "rxn10421_c.lower_bound = 0.\n",
    "rxn10421_c.upper_bound = 1000.\n",
    "rxn10421_c.add_metabolites({\n",
    "    cpd00443_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00443_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10421_c])\n",
    "\n",
    "# Folate, Thiamin, and Riboflavin in place already"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>272563.8</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f84b432b950</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1306</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1182</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>-1.0*biomass_reverse_01e59 + 1.0*biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model 272563.8 at 0x7f84b432b950>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check progress\n",
    "cd630_PATRIC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(cd630_PATRIC, 'data/cdiff_temp.sbml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gapfilling (complete media)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dependencies\n",
    "import numpy\n",
    "import copy\n",
    "import time\n",
    "import cobra\n",
    "from cobra.util import solver\n",
    "from optlang.symbolics import Zero\n",
    "from itertools import chain\n",
    "from cobra.manipulation.delete import *\n",
    "from cobra.flux_analysis.sampling import ACHRSampler\n",
    "\n",
    "# pFBA gapfiller\n",
    "def pfba_gapfill(model, reaction_bag, sample=True, extracellular='e', tasks=False):\n",
    "    '''\n",
    "    Function that utilizes iterations of pFBA solution with a universal reaction bag \n",
    "    in order to gapfill a model.\n",
    "    \n",
    "    Parameters\n",
    "    ----------\n",
    "    model : cobra.Model\n",
    "        Model to be gapfilled\n",
    "    reaction_bag : cobra.Model\n",
    "        Reaction bag reference to use during gapfilling\n",
    "    sample : bool\n",
    "        Perform flux sampling to identify all possible active reactions from universal\n",
    "    extracellular : string\n",
    "        Label for extracellular compartment of model\n",
    "    tasks : list\n",
    "        List of strings representing reaction IDs that must be able to carry flux\n",
    "    '''\n",
    "    start_time = time.time()\n",
    "    \n",
    "    # Save some basic network info for downstream membership testing\n",
    "    orig_rxn_ids = set([str(w.id) for w in model.reactions])\n",
    "    orig_cpd_ids = set([str(x.id) for x in model.metabolites])\n",
    "    univ_rxn_ids = set([str(y.id) for y in reaction_bag.reactions])\n",
    "    \n",
    "    # Find overlap in model and reaction bag\n",
    "    overlap_rxn_ids = univ_rxn_ids.intersection(orig_rxn_ids)\n",
    "    \n",
    "    # Get model objective reaction ID\n",
    "    obj = get_objective(model)\n",
    "\n",
    "    # Modify universal reaction bag\n",
    "    new_rxn_ids = set()\n",
    "    print('Creating universal model')\n",
    "    with reaction_bag as universal:\n",
    "\n",
    "        # Remove overlapping reactions from universal bag, and reset objective if needed\n",
    "        for rxn in overlap_rxn_ids: universal.reactions.get_by_id(rxn).remove_from_model()\n",
    "\n",
    "        # Add model reactions to universal bag\n",
    "        universal.add_reactions(list(copy.deepcopy(model.reactions)))\n",
    "        \n",
    "        # Set up pFBA objective\n",
    "        print('Formulating pFBA objective')\n",
    "        rxn_vars = ((rxn.forward_variable, rxn.reverse_variable) for rxn in universal.reactions if rxn.id not in orig_rxn_ids)\n",
    "        rxn_vars = chain(*rxn_vars)\n",
    "        orig_rxn_vars = ((rxn.forward_variable, rxn.reverse_variable) for rxn in universal.reactions if rxn.id in orig_rxn_ids)\n",
    "        orig_rxn_vars = chain(*orig_rxn_vars)\n",
    "        pfba_obj = universal.problem.Objective(Zero, direction='min', sloppy=True)\n",
    "        universal.objective = pfba_obj\n",
    "        universal.objective.set_linear_coefficients({x: 1.0 for x in rxn_vars})\n",
    "        universal.objective.set_linear_coefficients({y: 0.0 for y in orig_rxn_vars})\n",
    "        universal.solver.update()\n",
    "        \n",
    "        # Set minimum lower bound for previous objective\n",
    "        universal.reactions.get_by_id(obj).lower_bound = 1.0\n",
    "        \n",
    "        # Set lower bounds for metaboloic tasks\n",
    "        if tasks == True:\n",
    "            for rxn in tasks:\n",
    "                try:\n",
    "                    universal.reactions.get_by_id(rxn).lower_bound = 0.001\n",
    "                except:\n",
    "                    continue\n",
    "                \n",
    "        # Run FBA and save solution\n",
    "        solution = universal.optimize()\n",
    "        \n",
    "        # Identify reactions from universal that are now active\n",
    "        if sample == True:\n",
    "            print('Generating flux sampling object')\n",
    "            optgp_object = ACHRSampler(universal)\n",
    "            \n",
    "            print('Sampling flux distribution for active reactions')\n",
    "            flux_samples = optgp_object.sample(1000)\n",
    "            for distribution in flux_samples.iterrows():\n",
    "                for rxn in list(flux_samples.columns):\n",
    "                    if rxn in orig_rxn_ids:\n",
    "                        continue\n",
    "                    elif abs(distribution[1][rxn]) > 1e-6:\n",
    "                        new_rxn_ids |= set([rxn])\n",
    "            \n",
    "        else:\n",
    "            print('Finding active reactions in single flux distribution')\n",
    "            rxns = list(solution.fluxes.index)\n",
    "            fluxes = list(solution.fluxes)\n",
    "            for flux in range(0, len(fluxes)):\n",
    "                if rxns[flux] in orig_rxn_ids:\n",
    "                    continue\n",
    "                elif abs(fluxes[flux]) > 1e-6:\n",
    "                    new_rxn_ids |= set([rxns[flux]])\n",
    "    \n",
    "    # Screen new reaction IDs\n",
    "    new_rxn_ids = new_rxn_ids.difference(orig_rxn_ids)\n",
    "    \n",
    "    # Get reactions and metabolites to be added to the model\n",
    "    new_rxns = copy.deepcopy([reaction_bag.reactions.get_by_id(rxn) for rxn in new_rxn_ids])\n",
    "    new_cpd_ids = set()\n",
    "    for rxn in new_rxns: new_cpd_ids |= set([str(x.id) for x in list(rxn.metabolites)])\n",
    "    new_cpd_ids = new_cpd_ids.difference(orig_cpd_ids)\n",
    "    new_cpds = copy.deepcopy([reaction_bag.metabolites.get_by_id(cpd) for cpd in new_cpd_ids])\n",
    "    \n",
    "    # Create gapfilled model \n",
    "    new_model = copy.deepcopy(model)\n",
    "    new_model.add_metabolites(new_cpds)\n",
    "    new_model.add_reactions(new_rxns)\n",
    "    \n",
    "    # Identify extracellular metabolites still need exchanges\n",
    "    new_exchanges = extend_exchanges(new_model, new_cpd_ids, extracellular)\n",
    "    \n",
    "    # Report to user\n",
    "    new_rxn_ids = len(new_rxn_ids) + new_exchanges\n",
    "    new_cpd_ids = len(new_cpd_ids)\n",
    "    print('Gapfilled ' + str(new_rxn_ids) + ' reactions and ' + str(new_cpd_ids) + ' metabolites') \n",
    "    if new_model.slim_optimize() > 1e-6: print('Objective now carries flux')\n",
    "\n",
    "    duration = str(round(((time.time() - start_time) / 60.0), 1)).split('.')\n",
    "    mins = duration[0]\n",
    "    secs = str(round(((float(duration[1]) / 10.0) + 60.0)))\n",
    "    print('Finished in ' + mins + ' minutes and ' + secs + ' seconds')\n",
    "    return new_model\n",
    "\n",
    "\n",
    "# Adds missing exchanges for extracellulart metabolites\n",
    "def extend_exchanges(model, cpd_ids, outside):\n",
    "        \n",
    "    model_exchanges = set()\n",
    "    for rxn in model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            if outside in [str(cpd.compartment) for cpd in rxn.metabolites]:\n",
    "                model_exchanges |= set([rxn.id])\n",
    "\n",
    "    new_ex_ids = 0\n",
    "    cpds = [model.metabolites.get_by_id(cpd) for cpd in cpd_ids if model.metabolites.get_by_id(cpd).compartment == outside]\n",
    "    for cpd in cpds:\n",
    "        rxns = set([str(rxn.id) for rxn in cpds.reactions])\n",
    "        if bool(rxns & model_exchanges) == False:\n",
    "            new_id = 'EX_' + cpd.id\n",
    "            model.add_boundary(cpd, type='exchange', reaction_id=new_id, lb=-1000.0, ub=1000.0)\n",
    "            new_ex_ids += 1\n",
    "\n",
    "    return new_ex_ids\n",
    "\n",
    "\n",
    "# Returns the reaction ID of the objective reaction\n",
    "def get_objective(model):\n",
    "    \n",
    "    if len(list(model.objective.variables)) == 0:\n",
    "        raise IndexError('Model has no objective set.')\n",
    "    \n",
    "    expression = str(model.objective.expression).split()\n",
    "    if 'reverse' in expression[0]:\n",
    "        obj_id = expression[2].split('*')[-1]\n",
    "    else:\n",
    "        obj_id = expression[0].split('*')[-1]\n",
    "            \n",
    "    return obj_id\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import cobra\n",
    "import symengine\n",
    "\n",
    "# pFBA gapfiller\n",
    "def pfba_gapfill(model, objective, universal, extracellular='extracellular', ignore=[]):\n",
    "    start_time = time.time()\n",
    "    '''\n",
    "    Parameters\n",
    "    ----------\n",
    "    model_file : str\n",
    "        Model to be gapfilled\n",
    "    objective : \n",
    "    universal_file : str\n",
    "        Reaction bag reference\n",
    "    iters : int\n",
    "        Number of iterations to repeat gapfilling process\n",
    "    extracellular : str\n",
    "        Label for extracellular compartment of model\n",
    "    '''\n",
    "    # Create temporary models to modify for gapfilling\n",
    "    temp_model = copy.deepcopy(model)\n",
    "    temp_universal = copy.deepcopy(universal)\n",
    "\n",
    "    # Save some basic network info for downstream membership testing\n",
    "    target_rxns = set([str(x.id) for x in model.reactions])\n",
    "    target_cpds = set([str(y.id) for y in model.metabolites])\n",
    "    ref_rxns = set([str(z.id) for z in universal.reactions])\n",
    "    shared_rxns = ref_rxns.intersection(target_rxns)\n",
    "    \n",
    "    # Define minimum objective value\n",
    "    obj_val = temp_model.slim_optimize()\n",
    "    obj_constraint = constrained_model.problem.Constraint(temp_model.objective.expression, lb=obj_val*0.01, ub=obj_val)\n",
    "    temp_model.add_cons_vars([prev_obj_constraint])\n",
    "    temp_model.solver.update()\n",
    "                \n",
    "    # Open exchanges are all open\n",
    "    for rxn in temp_model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            rxn.bounds = (min(rxn.lower_bound, -1000.), max(rxn.upper_bound, 1000.))\n",
    "    for rxn in temp_universal.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            rxn.bounds = (min(rxn.lower_bound, -1000.), max(rxn.upper_bound, 1000.))\n",
    "        \n",
    "    # Remove overlapping reactions from universal bag, add model reactions to universal bag\n",
    "    for rxn in shared_rxns: temp_universal.reactions.get_by_id(rxn).remove_from_model()\n",
    "    temp_universal.add_reactions(list(copy.deepcopy(model.reactions)))\n",
    "        \n",
    "    # Set up pFBA objective\n",
    "    pfba_expr = symengine.RealDouble(0)\n",
    "    for rxn in temp_universal.reactions:\n",
    "        if rxn.id in target_rxns:\n",
    "            pfba_expr += 1.0 * rxn.forward_variable\n",
    "            pfba_expr += 1.0 * rxn.reverse_variable\n",
    "        else:\n",
    "            pfba_expr += 0.0 * rxn.forward_variable\n",
    "            pfba_expr += 0.0 * rxn.reverse_variable\n",
    "        \n",
    "    temp_universal.objective = temp_universal.problem.Objective(pfba_expr, direction='min', sloppy=True)\n",
    "    temp_universal.solver.update()\n",
    "\n",
    "    # Run FBA and save solution\n",
    "    solution = temp_universal.optimize()\n",
    "    active_rxns = set([rxn.id for rxn in temp_universal.reactions if abs(solution.fluxes[rxn.id]) > 1e-6])\n",
    "    \n",
    "    # Screen new reaction IDs\n",
    "    new_rxns = active_rxns.difference(target_rxns)\n",
    "    \n",
    "    # Get reactions and metabolites to be added to the model\n",
    "    new_rxns = copy.deepcopy([temp_universal.reactions.get_by_id(rxn) for rxn in new_rxns])\n",
    "    new_cpds = set()\n",
    "    for rxn in new_rxns: new_cpds |= set([str(x.id) for x in list(rxn.metabolites)]).difference(target_cpds)\n",
    "    new_cpds = copy.deepcopy([temp_universal.metabolites.get_by_id(cpd) for cpd in new_cpds])\n",
    "    \n",
    "    # Create gapfilled model \n",
    "    new_model = copy.deepcopy(model)\n",
    "    new_model.add_metabolites(new_cpds)\n",
    "    new_model.add_reactions(new_rxns)\n",
    "    \n",
    "    # Identify extracellular metabolites still need exchanges\n",
    "    new_exchs = 0\n",
    "    model_exchanges = set()\n",
    "    rxns = set([str(rxn.id) for rxn in model.reactions])\n",
    "    for rxn in new_model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            if extracellular in [str(cpd.compartment) for cpd in rxn.metabolites]:\n",
    "                model_exchanges |= set([rxn.id])                \n",
    "    for cpd in new_model.metabolites:\n",
    "        if cpd.compartment != extracellular: continue\n",
    "        current_rxns = set([x.id for x in cpd.reactions])\n",
    "        if bool(current_rxns & model_exchanges) == False:\n",
    "            new_id = 'EX_' + cpd.id\n",
    "            new_model.add_boundary(cpd, type='exchange', reaction_id=new_id, lb=-1000.0, ub=1000.0)\n",
    "            new_exchs += 1\n",
    "    \n",
    "    # Report to user\n",
    "    print('\\nGapfilled ' + str(len(new_rxns) + new_exchs) + ' reactions and ' + str(len(new_cpds)) + ' metabolites') \n",
    "    if new_model.slim_optimize() <= 1e-6: print('WARNING: Objective does not carries flux')\n",
    "    \n",
    "    return new_model\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd630_PATRIC = cobra.io.read_sbml_model('data/cdiff_temp.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in curated universal reacton bag\n",
    "universal = cobra.io.load_json_model('data/universal.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress: 100%   \n",
      "Gapfilled 30 reactions and 0 metabolites\n",
      "Objective now carries flux\n",
      "Finished in 18 minutes and 60.0 seconds\n"
     ]
    }
   ],
   "source": [
    "# Gap filling to biomass on complete media\n",
    "\n",
    "# Set metabolic tasks\n",
    "scfa_rxns = ['R01174_5_c','rxn05061_c','R01174_4_c','R01174_3_c','R01174_2_c','rxn01236_c',\n",
    "             'R01174_1_c','rxn11676_c','rxn01610_c','rxn07124_c','rxn00669_c','rxn13713_c', 'R11462_c']\n",
    "\n",
    "# Flux-sampling pFBA gapfill\n",
    "cd630_PATRIC_gapfilled = pfba_ensemble_gapfill(cd630_PATRIC, universal, extracellular='extracellular', tasks=scfa_rxns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>272563.8</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f1b7cb9e050</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1306</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1212</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model 272563.8 at 0x7f1b7cb9e050>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC_gapfilled"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to be added: 30\n"
     ]
    }
   ],
   "source": [
    "# Rerun gapfill until no new reactions were found, reseting the kernel each time\n",
    "\n",
    "iter1 = ['rxn02269_c', 'rxn09142_c', 'rxn09145_c', 'rxn00620_c', 'rxn01105_c', 'rxn05293_c', 'rxn02286_c', 'rxn09147_c', \n",
    "         'rxn01953_c', 'rxn02011_c', 'rxn07466_c', 'rxn09429_c', 'rxn03852_c', 'rxn13140_c', 'rxn09065_c', 'rxn01520_c', \n",
    "         'rxn13914_c', 'rxn09063_c', 'rxn09348_c', 'rxn05744_c', 'rxn08040_c']\n",
    "iter2 = ['rxn02269_c', 'rxn13139_c', 'rxn00620_c', 'rxn05293_c', 'rxn09341_c', 'rxn02286_c', 'rxn01953_c', 'rxn02011_c',\n",
    "         'rxn07466_c', 'rxn09429_c', 'rxn09126_c', 'rxn09124_c', 'rxn13140_c', 'rxn09065_c', 'rxn01520_c', 'rxn00553_c',\n",
    "         'rxn03852_c', 'rxn13914_c', 'rxn09063_c', 'rxn05744_c', 'rxn08040_c', 'rxn09163_c']\n",
    "iter3 = ['rxn02269_c', 'rxn09145_c', 'rxn13139_c', 'rxn00620_c', 'rxn05293_c', 'rxn02286_c', 'rxn09147_c', 'rxn01953_c', 'rxn00605_c',\n",
    "         'rxn02011_c', 'rxn00579_c', 'rxn07466_c', 'rxn09429_c', 'rxn03852_c', 'rxn13140_c', 'rxn09065_c', 'rxn03164_c', 'rxn01520_c',\n",
    "         'rxn09063_c', 'rxn09348_c', 'rxn05744_c', 'rxn08040_c', 'rxn09163_c']\n",
    "iter4 = ['rxn02269_c', 'rxn13139_c', 'rxn00620_c', 'rxn05293_c', 'rxn02286_c', 'rxn01953_c', 'rxn00605_c', 'rxn02011_c', 'rxn00579_c',\n",
    "         'rxn07466_c', 'rxn09429_c', 'rxn09126_c', 'rxn09124_c', 'rxn13140_c', 'rxn09065_c', 'rxn01520_c', 'rxn03852_c', 'rxn13914_c',\n",
    "         'rxn09063_c', 'rxn09348_c', 'rxn05744_c', 'rxn08040_c', 'rxn09163_c']\n",
    "\n",
    "new_rxn_ids = set(iter1 + iter2 + iter3 + iter4)\n",
    "print('Reactions to be added: ' + str(len(new_rxn_ids)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cobra.io.write_sbml_model(cd630_PATRIC_gapfilled, 'data/cdiff_temp2.sbml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Gapfilling to minimal media"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to change media condition based on a list of exchange reaction IDs\n",
    "def changeMedia(full_model, components):\n",
    "    model = copy.deepcopy(full_model)\n",
    "    \n",
    "    media = copy.copy(components)\n",
    "    media = ['EX_' + x for x in media]\n",
    "    \n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.products)) == 0:\n",
    "            if rxn.id in media:\n",
    "                model.reactions.get_by_id(rxn.id).lower_bound = -1000.0\n",
    "                media.remove(rxn.id)\n",
    "            else:\n",
    "                model.reactions.get_by_id(rxn.id).lower_bound = 0.0\n",
    "    if len(media) != 0:\n",
    "        print(media)\n",
    "    return model\n",
    "\n",
    "# Undoes change media conditions\n",
    "def unblockExch(model):\n",
    "    exchanges = []\n",
    "    for rxn in model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            exchanges.append(rxn.id)\n",
    "    \n",
    "    for rxn in exchanges:\n",
    "        model.reactions.get_by_id(rxn).lower_bound = -1000.0\n",
    "        model.reactions.get_by_id(rxn).upper_bound = 1000.0\n",
    "    \n",
    "    # Keep anaerobic\n",
    "    model.reactions.get_by_id('EX_cpd00532_e').lower_bound = 0.0\n",
    "    model.reactions.get_by_id('EX_cpd00532_e').upper_bound = 0.0\n",
    "            \n",
    "    return model\n",
    "\n",
    "\n",
    "# Actually prune all unused metabolites and reactions (innate function does not work)\n",
    "def removeUnused(model):\n",
    "    removed_cpd = set()\n",
    "    removed_rxn = set()\n",
    "    unused_current_cpd = 1\n",
    "    unused_current_rxn = 1\n",
    "    \n",
    "    while unused_current_cpd != 0 or unused_current_rxn != 0:\n",
    "        unused_cpd = prune_unused_metabolites(model)\n",
    "        removed_cpd |= set(unused_cpd)\n",
    "        unused_rxn = prune_unused_reactions(model)\n",
    "        removed_rxn |= set(unused_rxn)\n",
    "        \n",
    "        unused_current_cpd = len(unused_cpd)\n",
    "        unused_current_rxn = len(unused_rxn)\n",
    "    \n",
    "    print('Pruned ' + str(len(removed_cpd)) + ' metabolites from model')\n",
    "    print('Pruned ' + str(len(removed_rxn)) + ' reactions from model')\n",
    "        \n",
    "    return(list(removed_cpd), list(removed_rxn))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Karlsson et al. (1999). Microbiology.\n",
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "# Theriot et al. (2013). Nature Communications.\n",
    "ncmm = ['cpd00001_e', # water\n",
    "        'cpd00104_e', # Biotin\n",
    "        'cpd00644_e', # Pantothenate\n",
    "        'cpd00263_e', # Pyridoxine\n",
    "        'cpd00149_e', # Cobalt\n",
    "        'cpd00099_e', # Chloride\n",
    "        'cpd10515_e', # Iron\n",
    "        'cpd00030_e', # Manganese\n",
    "        'cpd00254_e', # Magnesium\n",
    "        'cpd00063_e', # Calcium\n",
    "        'cpd00205_e', # Potassium\n",
    "        'cpd00009_e', # Phosphate\n",
    "        'cpd00971_e', # Sodium\n",
    "        'cpd00242_e', # Carbonate\n",
    "        'cpd00322_e', # L-Isoleucine\n",
    "        'cpd00129_e', # L-Proline\n",
    "        'cpd00156_e', # L-Valine\n",
    "        'cpd00107_e', # L-Leucine\n",
    "        'cpd00084_e', # L-Cysteine \n",
    "        'cpd00065_e', # L-Tryptophan\n",
    "        'cpd00060_e', # L-Methionine\n",
    "        'cpd00119_e', # L-Histidine\n",
    "        'cpd00033_e', # Glycine\n",
    "        'cpd00051_e', # L-Arginine\n",
    "        'cpd00161_e'] # L-Threonine\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cd630_PATRIC_gapfilled = cobra.io.read_sbml_model('data/cdiff_temp2.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress: 46.69%Failed to find new reactions twice in a row, ending search\n",
      "Progress: 100%   \n",
      "Gapfilled 38 reactions and 7 metabolites\n",
      "Objective now carries flux\n",
      "Completed in 8.0 minutes\n"
     ]
    }
   ],
   "source": [
    "# change to C. difficle minimal media\n",
    "cd630_PATRIC_mdm = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "cd630_PATRIC_min = pfba_ensemble_gapfill(cd630_PATRIC_mdm, universal, extracellular='extracellular')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to be added: 26\n"
     ]
    }
   ],
   "source": [
    "iter1_mdm = ['rxn01014_c', 'rxn10175_c', 'rxn02476_c', 'rxn00777_c', 'rxn01359_c', 'rxn12676_c', 'rxn02928_c', 'rxn01538_c', \n",
    "             'rxn01182_c', 'rxn12239_c', 'rxn03167_c', 'rxn05039_c', 'rxn10484_c', 'rxn05291_c', 'rxn10030_c', 'rxn01256_c',\n",
    "             'rxn01644_c', 'rxn08131_c', 'rxn00530_c', 'rxn00010_c', 'rxn10053_c', 'rxn01669_c']\n",
    "iter2_mdm = ['rxn01014_c', 'rxn10175_c', 'rxn02476_c', 'rxn00777_c', 'rxn00970_c', 'rxn13251_c', 'rxn02928_c', 'rxn01538_c', \n",
    "             'rxn12239_c', 'rxn03167_c', 'rxn05039_c', 'rxn10484_c', 'rxn05291_c', 'rxn01256_c', 'rxn01644_c', 'rxn08131_c',\n",
    "             'rxn00530_c', 'rxn00958_c', 'rxn00010_c', 'rxn10053_c', 'rxn01669_c','rxn01539_c']\n",
    "iter3_mdm = ['rxn01014_c', 'rxn10175_c', 'rxn02476_c', 'rxn00777_c', 'rxn12676_c', 'rxn13251_c', 'rxn02928_c', 'rxn01538_c',\n",
    "             'rxn01182_c', 'rxn12239_c', 'rxn03167_c', 'rxn05039_c', 'rxn10484_c', 'rxn10030_c', 'rxn01256_c', 'rxn01644_c',\n",
    "             'rxn08131_c', 'rxn00530_c', 'rxn00010_c', 'rxn10053_c', 'rxn01669_c']\n",
    "\n",
    "mdm_rxn_ids = set(iter1_mdm + iter2_mdm + iter3_mdm)\n",
    "print('Reactions to be added: ' + str(len(mdm_rxn_ids)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress: 13.34%Failed to find new reactions twice in a row, ending search\n",
      "Progress: 100%   \n",
      "Gapfilled 0 reactions and 0 metabolites\n",
      "Objective now carries flux\n",
      "Completed in 3.1 minutes\n"
     ]
    }
   ],
   "source": [
    "# change to C. difficle no-carbohydrate minimal media\n",
    "cd630_PATRIC_ncmm = changeMedia(cd630_PATRIC_min, ncmm)\n",
    "cd630_PATRIC_min2 = pfba_ensemble_gapfill(cd630_PATRIC_ncmm, universal, extracellular='extracellular')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to be added: 8\n"
     ]
    }
   ],
   "source": [
    "iter1_ncmm = ['rxn00285_c', 'rxn05227_c', 'rxn10036_c', 'rxn10840_c', 'rxn00341_c', 'rxn13012_c', 'rxn09119_c',\n",
    "              'rxn10663_c']\n",
    "iter2_ncmm = ['rxn00285_c', 'rxn05227_c', 'rxn10036_c', 'rxn10840_c', 'rxn00341_c', 'rxn13012_c', 'rxn09119_c',\n",
    "              'rxn10663_c']\n",
    "\n",
    "ncmm_rxn_ids= set(iter1_ncmm + iter2_ncmm)\n",
    "print('Reactions to be added: ' + str(len(ncmm_rxn_ids)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gapfilled_rxn_ids = set(list(new_rxn_ids) + list(mdm_rxn_ids) + list(ncmm_rxn_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 extracellular metabolites are missing transport reactions\n"
     ]
    }
   ],
   "source": [
    "# Gapfill model with union of pFBA gapfill iterations\n",
    "new_rxns = copy.deepcopy([universal.reactions.get_by_id(rxn) for rxn in gapfilled_rxn_ids])\n",
    "new_cpd_ids = set()\n",
    "for rxn in new_rxns: new_cpd_ids |= set([str(x.id) for x in list(rxn.metabolites)])\n",
    "orig_cpd_ids = set([str(y.id) for y in cd630_PATRIC.metabolites])\n",
    "new_cpd_ids = new_cpd_ids.difference(orig_cpd_ids)\n",
    "new_cpds = copy.deepcopy([universal.metabolites.get_by_id(cpd) for cpd in new_cpd_ids])\n",
    "\n",
    "cd630_PATRIC_gapfilled = copy.deepcopy(cd630_PATRIC)\n",
    "cd630_PATRIC_gapfilled.add_metabolites(new_cpds)\n",
    "cd630_PATRIC_gapfilled.add_reactions(new_rxns)\n",
    "\n",
    "trans, ex = missingRxn(cd630_PATRIC_gapfilled)\n",
    "for x in ex:\n",
    "    try:\n",
    "        new_id = 'EX_' + x\n",
    "        cpd = cd630_PATRIC_gapfilled.metabolites.get_by_id(x)\n",
    "        cd630_PATRIC_gapfilled.add_boundary(cpd, type='exchange', reaction_id=new_id, lb=-1000.0, ub=1000.0)\n",
    "    except ValueError:\n",
    "        pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'15.808 minutes'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doubling(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'48.873 minutes'"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'50.831 minutes'"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, ncmm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Collect and label gapfilled reactions\n",
    "for rxn in gapfilled_rxn_ids:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id(rxn).notes = 'gapfilled'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(cd630_PATRIC_gapfilled, 'data/cdf_temp3.sbml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fixing remaining mass-balancing reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removes reactions from unbalanced list that contain R-groups in their substrate formulas, or are charged imbalanced\n",
    "def trueUnbalance(model, ignore=[]):\n",
    "    imbal = []\n",
    "    rxn_ids = [str(x.id) for x in model.reactions]\n",
    "    \n",
    "    for rxn in rxn_ids:\n",
    "        test = list(model.reactions.get_by_id(rxn).check_mass_balance().keys())\n",
    "        \n",
    "        if rxn in ignore:\n",
    "            continue\n",
    "        elif len(test) == 0:\n",
    "            continue\n",
    "        elif test == ['charge']:\n",
    "            continue\n",
    "        elif 'EX_' in rxn:\n",
    "            continue\n",
    "            \n",
    "        formula = ''.join([x.formula for x in list(model.reactions.get_by_id(rxn).metabolites)])\n",
    "        if 'R' not in formula:\n",
    "            imbal.append(str(rxn))\n",
    "    \n",
    "    formula_dict = {}\n",
    "    for x in imbal:\n",
    "        react = [i.formula for i in model.reactions.get_by_id(x).reactants]\n",
    "        prod = [j.formula for j in model.reactions.get_by_id(x).products]\n",
    "        if react == prod:\n",
    "            imbal.remove(x)\n",
    "            continue\n",
    "        else:\n",
    "            formula_dict[x] = [react, prod]\n",
    "    \n",
    "    print(str(len(formula_dict.keys())) + ' mass-unbalanced reactions')\n",
    "    return(formula_dict)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cd630_PATRIC_gapfilled = cobra.io.read_sbml_model('data/cdf_temp3.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 mass-unbalanced reactions\n"
     ]
    }
   ],
   "source": [
    "true_imbal = trueUnbalance(cd630_PATRIC_gapfilled, ignore=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00106_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00106_c')\n",
    "cpd00282_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00282_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00247_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00247_c')\n",
    "try:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13251_c').remove_from_model()\n",
    "except:\n",
    "    pass\n",
    "rxn13251_c = cobra.Reaction('rxn13251_c') \n",
    "rxn13251_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn13251_c.name = 'Fumarate dependent DHORD'\n",
    "rxn13251_c.lower_bound = -1000.\n",
    "rxn13251_c.upper_bound = 1000.\n",
    "rxn13251_c.add_metabolites({\n",
    "    cpd00106_c: -1.0,\n",
    "    cpd00282_c: -1.0,\n",
    "    cpd00067_c: -2.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd00247_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn13251_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00067_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00067_c')\n",
    "cpd00239_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00239_c')\n",
    "cpd00822_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00822_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00135_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00135_c')\n",
    "try:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05957_c').remove_from_model()\n",
    "except:\n",
    "    pass\n",
    "rxn05957_c = cobra.Reaction('rxn05957_c') \n",
    "rxn05957_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05957_c.name = 'O4-succinyl-L-homoserine:hydrogen sulfide S-(3-amino-3-carboxypropyl)transferase'\n",
    "rxn05957_c.gene_reaction_rule = '272563.8.peg.1916'\n",
    "rxn05957_c.lower_bound = 0.\n",
    "rxn05957_c.upper_bound = 1000.\n",
    "rxn05957_c.add_metabolites({\n",
    "    cpd00067_c: -2.0,\n",
    "    cpd00239_c: -1.0,\n",
    "    cpd00822_c: -1.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd00135_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn05957_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00002_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00002_c')\n",
    "cpd00010_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00010_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00008_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00008_c')\n",
    "cpd00009_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00009_c')\n",
    "cpd00078_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00078_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00285_c').remove_from_model()\n",
    "rxn00285_c = cobra.Reaction('rxn00285_c') \n",
    "rxn00285_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00285_c.name = 'Succinate:CoA ligase (ADP-forming)'\n",
    "rxn00285_c.lower_bound = 0.\n",
    "rxn00285_c.upper_bound = 1000.\n",
    "rxn00285_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00036_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00078_c: 1.0,\n",
    "    cpd00067_c: 2.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn00285_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00001_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00001_c')\n",
    "cpd03989_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03989_c')\n",
    "cpd00461_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00461_c')\n",
    "cpd01711_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01711_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn03665_c').remove_from_model()\n",
    "rxn03665_c = cobra.Reaction('rxn03665_c') \n",
    "rxn03665_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn03665_c.name = '2-Hydroxy-6-oxo-7-methylocta-2,4-dienoate acylhydrolase'\n",
    "rxn03665_c.gene_reaction_rule = '272563.8.peg.3588'\n",
    "rxn03665_c.lower_bound = 0.\n",
    "rxn03665_c.upper_bound = 1000.\n",
    "rxn03665_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd03989_c: -1.0,\n",
    "    cpd00461_c: 1.0,\n",
    "    cpd01711_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn03665_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00001_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00001_c')\n",
    "cpd02731_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd02731_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00461_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00461_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn01895_c').remove_from_model()\n",
    "rxn01895_c = cobra.Reaction('rxn01895_c') \n",
    "rxn01895_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01895_c.name = '2-Hydroxy-6-oxonona-2,4-diene-1,9-dioate succinylhydrolase'\n",
    "rxn01895_c.gene_reaction_rule = '272563.8.peg.3588'\n",
    "rxn01895_c.lower_bound = 0.\n",
    "rxn01895_c.upper_bound = 1000.\n",
    "rxn01895_c.add_metabolites({  \n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd02731_c: -1.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd00461_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn01895_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "C21400_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('C21400_c')\n",
    "cpd01646_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01646_c')\n",
    "C21399_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('C21399_c')\n",
    "cpd00047_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00047_c')\n",
    "cpd23040_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd23040_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('R11462_c').remove_from_model()\n",
    "R11462_c = cobra.Reaction('R11462_c') \n",
    "R11462_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "R11462_c.name = '4-methylpentanoyl-CoA:(R)-2-hydroxy-4-methylpentanoate CoA-transferase [EC:2.8.3.24]'\n",
    "R11462_c.gene_reaction_rule = 'CD630_03950'\n",
    "R11462_c.lower_bound = 0.\n",
    "R11462_c.upper_bound = 1000.\n",
    "R11462_c.add_metabolites({  \n",
    "    C21400_c: -1.0,\n",
    "    cpd01646_c: -1.0,\n",
    "    C21399_c: 1.0,\n",
    "    cpd00047_c: 1.0,\n",
    "    cpd23040_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([R11462_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        nogene_gapfilled_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "                if m.reactions.get_by_id(rxn).notes == 'gapfilled':\n",
    "                    nogene_gapfilled_blocked.append(rxn)\n",
    "    \n",
    "    print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    print(str(len(nogene_blocked)) + ' reactions without GPRs are blocked')\n",
    "    print(str(len(nogene_gapfilled_blocked)) + ' gapfilled reactions without GPRs are blocked')\n",
    "    \n",
    "    return blocked, nogene_blocked, nogene_gapfilled_blocked\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "519 total reactions are blocked\n",
      "33 reactions without GPRs are blocked\n",
      "0 gapfilled reactions without GPRs are blocked\n"
     ]
    }
   ],
   "source": [
    "# Check for blocked reactions\n",
    "blocked, no_gene, gapfilled_blocked = blockedReactions(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gapfilled reaction with ATP of GTP: 8\n"
     ]
    }
   ],
   "source": [
    "check = []\n",
    "for x in gapfilled_rxn_ids:\n",
    "    rxn = cd630_PATRIC_gapfilled.reactions.get_by_id(x)\n",
    "    cpds = [str(y.id) for y in rxn.metabolites]\n",
    "    if 'cpd00002_c' in cpds or 'cpd00038_c' in cpds:\n",
    "        if rxn.lower_bound < -1e-6 and rxn.upper_bound > 1e-6:\n",
    "            check.append(str(rxn.id))\n",
    "print('Gapfilled reaction with ATP of GTP: ' + str(len(check)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct gapfilled reaction reversibility\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn01182_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13012_c').lower_bound = 0.0 # really slows it down, consumes 6 ATP\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10663_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10053_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13139_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13914_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn09562_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01953_c.bounds = (-1000, 0)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn00292_c.bounds = (-1000, 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'5.322 hours'"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, ncmm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2.321 hours'"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'35.606 minutes'"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doubling(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove reactions with no GPR that are gapfilled and don't have strong biological evidence\n",
    "remove_blocked = ['EX_cpd11576_e','EX_cpd04097_e','EX_cpd01012_e','EX_cpd00058_e',\n",
    "                  'EX_cpd11578_e','EX_cpd11579_e','EX_cpd00210_e','EX_cpd00532_e',\n",
    "                  'EX_cpd00176_e','EX_cpd03724_e','EX_cpd03696_e','EX_cpd08023_e',\n",
    "                  'EX_cpd01092_e','EX_cpd01262_e','EX_cpd11596_e','EX_cpd01914_e',\n",
    "                  'EX_cpd00531_e','EX_cpd11575_e','EX_cpd09878_e','EX_cpd01329_e',\n",
    "                  'rxn09030_c']\n",
    "\n",
    "for rxn in remove_blocked:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id(rxn).remove_from_model()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2.321 hours'"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'35.606 minutes'"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doubling(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Edit SBML to ensure compatibility with Cobra read-in function"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pruning model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Identify potentially gapfilled reactions\n",
    "def missingGPR(model, exclude=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass']):\n",
    "    gapfilled = []\n",
    "    transport = findTransports(model)\n",
    "    if not type(exclude) is list:\n",
    "        exclude = [exclude]\n",
    "        \n",
    "    for index in model.reactions:\n",
    "        if len(list(index.genes)) == 0:\n",
    "            if not index in model.boundary:\n",
    "                if not index.id in exclude or not index.id in transport:\n",
    "                    gapfilled.append(index.id)\n",
    "    \n",
    "    if len(gapfilled) > 0:\n",
    "        print(str(len(gapfilled)) + ' metabolic reactions not associated with genes')\n",
    "    \n",
    "    return gapfilled\n",
    "\n",
    "# Check for missing transport and exchange reactions\n",
    "def missingRxn(model, extracellular='extracellular'):\n",
    "\n",
    "    transporters = set(findTransports(model))\n",
    "    exchanges = set([str(rxn.id) for rxn in model.reactions if len(list(rxn.products)) == 0])\n",
    "    \n",
    "    missing_exchanges = []\n",
    "    missing_transports = []\n",
    "    \n",
    "    for metabolite in model.metabolites:\n",
    "        if not metabolite.compartment == extracellular:\n",
    "            continue\n",
    "\n",
    "        curr_rxns = set([x.id for x in list(metabolite.reactions)])\n",
    "        \n",
    "        if not bool(curr_rxns & transporters):\n",
    "            missing_transports.append(metabolite.id)\n",
    "        if not bool(curr_rxns & exchanges):\n",
    "            missing_exchanges.append(metabolite.id)\n",
    "    \n",
    "    if len(missing_transports) != 0:\n",
    "        print(str(len(missing_transports)) + ' extracellular metabolites are missing transport reactions')\n",
    "    if len(missing_exchanges) != 0:\n",
    "        print(str(len(missing_exchanges)) + ' extracellular metabolites are missing exchange reactions')\n",
    "        \n",
    "    return missing_transports, missing_exchanges\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(raw_model, cytosol='cytosol'):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        \n",
    "        # Close all exchanges\n",
    "        for index in model.boundary:\n",
    "            model.reactions.get_by_id(index.id).lower_bound = 0.\n",
    "        \n",
    "        # Identify all metabolites that are produced within the network\n",
    "        demand_metabolites = [x.reactants[0].id for x in model.demands if len(x.reactants) > 0] + [x.products[0].id for x in model.demands if len(x.products) > 0]\n",
    "\n",
    "        free = []\n",
    "        for index in model.metabolites: \n",
    "            if index.id in demand_metabolites:\n",
    "                continue\n",
    "            elif not index.compartment in cytosol:\n",
    "                continue\n",
    "            else:\n",
    "                demand = model.add_boundary(index, type='demand')\n",
    "                model.objective = demand\n",
    "                obj_val = model.slim_optimize(error_value=0.)\n",
    "                if obj_val > 1e-8:\n",
    "                    free.append(index.id)\n",
    "                model.remove_reactions([demand])\n",
    "    \n",
    "    if len(free) > 0:\n",
    "        print(str(len(free)) + ' metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(raw_model, exclude=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass']):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        imbalanced = []\n",
    "        mass_imbal = 0\n",
    "        charge_imbal = 0\n",
    "        elem_set = set()\n",
    "        for metabolite in model.metabolites:\n",
    "            try:\n",
    "                elem_set |= set(metabolite.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        if len(elem_set) == 0:\n",
    "            imbalanced = model.reactions\n",
    "            mass_imbal = len(model.reactions)\n",
    "            charge_imbal = len(model.reactions)\n",
    "            print('No elemental data associated with metabolites!')\n",
    "        \n",
    "        else:\n",
    "            if not type(exclude) is list: \n",
    "                exclude = [exclude]\n",
    "            for index in model.reactions:\n",
    "                if index in model.boundary or index.id in exclude:\n",
    "                    continue\n",
    "\n",
    "                else:\n",
    "                    try:\n",
    "                        test = index.check_mass_balance()\n",
    "                    except ValueError:\n",
    "                        continue\n",
    "\n",
    "                    if len(list(test)) > 0:\n",
    "                        #imbalanced.append(index.id)\n",
    "\n",
    "                        #if 'charge' in test.keys():\n",
    "                        #    charge_imbal += 1\n",
    "                        if len(set(test.keys()).intersection(elem_set)) > 0:\n",
    "                            mass_imbal += 1\n",
    "                            imbalanced.append(index.id)\n",
    "\n",
    "    if mass_imbal != 0:\n",
    "        print(str(mass_imbal) + ' reactions are mass imbalanced')\n",
    "    #if charge_imbal != 0:\n",
    "    #    print(str(charge_imbal) + ' reactions are charge imbalanced')\n",
    "    \n",
    "    return(imbalanced)\n",
    "\n",
    "\n",
    "def checkDetermination(model):\n",
    "    if len(model.reactions) < len(model.metabolites):\n",
    "        print('Model is overdetermined')\n",
    "    if len(model.reactions) > len(model.metabolites):\n",
    "        print('Model is underdetermined')\n",
    "\n",
    "\n",
    "# Checks the quality of models by a couple metrics and returns problems\n",
    "def checkQuality(model, exclude=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass'], cytosol='cytosol'):\n",
    "    \n",
    "    checkDetermination(model)\n",
    "    gaps = missingGPR(model, exclude)\n",
    "    freemass = checkFreeMass(model, cytosol)\n",
    "    balance = checkBalance(model, exclude)\n",
    "    trans, exch = missingRxn(model)\n",
    "    \n",
    "    test = gaps + freemass + balance\n",
    "    if len(test) == 0:\n",
    "        print('No inconsistencies detected')\n",
    "\n",
    "    return gaps, freemass, balance, trans, exch\n",
    "\n",
    "\n",
    "# Prints the names of reactions or metabolites based on IDs\n",
    "def printNames(model, id_list):\n",
    "    for x in id_list:\n",
    "        try:\n",
    "            print(model.metabolites.get_by_id(x).name)\n",
    "            continue\n",
    "        except:\n",
    "            pass\n",
    "        \n",
    "        print(model.reactions.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model is overdetermined\n",
      "108 metabolic reactions not associated with genes\n",
      "3 extracellular metabolites are missing transport reactions\n",
      "20 extracellular metabolites are missing exchange reactions\n"
     ]
    }
   ],
   "source": [
    "no_gpr, freemass, unbalanced, no_trans, no_exch = checkQuality(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix no transporter reactions\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00034_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('EX_cpd00034_e').remove_from_model()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix no exchanges\n",
    "\n",
    "# L-methionine R-oxide\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11576_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11576_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05614_c').remove_from_model()\n",
    "\n",
    "# Pb\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd04097_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd04097_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10344_c').remove_from_model()\n",
    "\n",
    "# Cd+\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01012_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01012_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05516_c').remove_from_model()\n",
    "\n",
    "# Cu2+\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00058_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00058_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10481_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05528_c').remove_from_model()\n",
    "\n",
    "# hexanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11578_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11578_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05585_c').remove_from_model()\n",
    "\n",
    "# ethanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11579_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11579_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05552_c').remove_from_model()\n",
    "\n",
    "# Taurine\n",
    "cpd00210_e = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00210_e')\n",
    "cd630_PATRIC_gapfilled.add_boundary(cpd00210_e, type='exchange', reaction_id='EX_cpd00210_e', lb=-1000., ub=1000.)\n",
    "\n",
    "# 2-keto-3-deoxygluconate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00176_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00176_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05681_c').remove_from_model()\n",
    "\n",
    "# Ferrichrome\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03724_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03724_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05188_c').remove_from_model()\n",
    "\n",
    "# Ursin\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03696_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05501_c').remove_from_model()\n",
    "\n",
    "# Allantoin\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01092_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01092_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05682_c').remove_from_model()\n",
    "\n",
    "# Amylotriose\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01262_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01262_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05608_c').remove_from_model()\n",
    "\n",
    "# butanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11596_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11596_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05512_c').remove_from_model()\n",
    "\n",
    "# L-Methionine S-oxide\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01914_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01914_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05613_c').remove_from_model()\n",
    "\n",
    "# Hg\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00531_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00531_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10343_c').remove_from_model()\n",
    "\n",
    "# MOPS\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11575_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11575_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05620_c').remove_from_model()\n",
    "\n",
    "# Sulfoacetate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd09878_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd09878_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05656_c').remove_from_model()\n",
    "\n",
    "# Maltohexaose\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01329_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01329_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10174_c').remove_from_model()\n",
    "\n",
    "# methanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd08023_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd08023_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05621_c').remove_from_model()\n",
    "\n",
    "# O2-\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00532_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00532_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00206_c').remove_from_model()\n",
    "\n",
    "# Molybdate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11574_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11574_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('EX_cpd11574_e').remove_from_model()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model is overdetermined\n",
      "108 metabolic reactions not associated with genes\n"
     ]
    }
   ],
   "source": [
    "no_gpr, freemass, unbalanced, no_trans, no_exch = checkQuality(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct obvious auxotrophies\n",
    "\n",
    "# TTP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn10060_c.upper_bound = 0.0\n",
    "\n",
    "# dAMP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01507_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01507_c.upper_bound = 0.0\n",
    "\n",
    "# dAMP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01218_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01218_c.upper_bound = 0.0\n",
    "\n",
    "# dGMP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01445_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01445_c.upper_bound = 0.0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix last reaction reversibilities\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01538_c.lower_bound = 0.0 # ATP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01105_c.lower_bound = 0.0 # GTP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01105_c.lower_bound = 0.0 # GTP\n",
    "\n",
    "# GTP biosynthesis\n",
    "cd630_PATRIC_gapfilled.reactions.rxn13139_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn13139_c.upper_bound = 0.0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix incorrect biosynthesis of limiting amino acids\n",
    "\n",
    "# L-Proline - cpd00129_c\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00931_c').lower_bound = 0.\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00929_c').lower_bound = 0.\n",
    "\n",
    "# L-Cysteine - cpd00084_c\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn12844_c').lower_bound = 0.\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00650_c').lower_bound = 0.\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn09177_c').lower_bound = 0.\n",
    "\n",
    "# L-Leucine - cpd00107_c\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn12845_c').lower_bound = 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Close dipeptide exchange reactions\n",
    "dipeptides = ['EX_cpd11585_e','EX_cpd11582_e','EX_cpd11586_e','EX_cpd01017_e','EX_cpd11591_e',\n",
    "              'EX_cpd11593_e','EX_cpd15605_e','EX_cpd15603_e','EX_cpd11580_e','EX_cpd15606_e',\n",
    "              'EX_cpd11583_e','EX_cpd11589_e','EX_cpd11588_e','EX_cpd15604_e','EX_cpd11592_e',\n",
    "              'EX_cpd11587_e','EX_cpd11590_e','EX_cpd11581_e','EX_cpd11584_e']\n",
    "\n",
    "for x in dipeptides:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id(x).lower_bound = 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pruned 90 metabolites from model\n",
      "Pruned 7 reactions from model\n"
     ]
    }
   ],
   "source": [
    "unused_c, unused_r = removeUnused(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found and removed 0 reactions\n"
     ]
    }
   ],
   "source": [
    "# Removed orphaned reactions with no substrates\n",
    "removed = 0\n",
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    if len(list(rxn.reactants)) == 0 and len(list(rxn.products)) == 0:\n",
    "        cd630_PATRIC_gapfilled.reactions.get_by_id(rxn.id).remove_from_model()\n",
    "        removed += 1\n",
    "print('Found and removed ' + str(removed) + ' reactions')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "481 total reactions are blocked\n",
      "11 reactions without GPRs are blocked\n",
      "0 gapfilled reactions without GPRs are blocked\n"
     ]
    }
   ],
   "source": [
    "# Check blocked reactions\n",
    "blocked, nogene_blocked, nogene_gapfilled_blocked = blockedReactions(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove uninformative metabolites/reactions\n",
    "remove_cpd = ['cpd11440_c', #fa6\n",
    "              'cpd11438_c', #fa4\n",
    "              'cpd00666_c', #Tartrate\n",
    "              'cpd00158_c', #CELB\n",
    "              'cpd11433_c', #fa12\n",
    "              'cpd00392_c', #Phycite\n",
    "              'cpd11430_c', #fa1\n",
    "              'cpd11431_c', #fa11\n",
    "              'cpd11436_c'] #fa3\n",
    "for cpd in remove_cpd:\n",
    "    try:\n",
    "        metabolite = cd630_PATRIC_gapfilled.metabolites.get_by_id(cpd)\n",
    "        cd630_PATRIC_gapfilled.remove_reactions(metabolite.reactions)\n",
    "        cd630_PATRIC_gapfilled.remove_metabolites([metabolite])\n",
    "    except KeyError:\n",
    "        print(metabolite + ' not found')\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inactive_metabolites = [m for m in cd630_PATRIC_gapfilled.metabolites if len(m.reactions) == 0]\n",
    "cd630_PATRIC_gapfilled.remove_metabolites(inactive_metabolites)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "792"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(cd630_PATRIC_gapfilled.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct names\n",
    "cd630_PATRIC_gapfilled.metabolites.cpd00443_e.name = 'ABEE'\n",
    "cd630_PATRIC_gapfilled.reactions.rxn03185_c.name = 'alpha-N-acetylneuraminate alpha-2,8-sialyltransferase'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdJ792</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fd2281f0350</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1179</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1194</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdJ792 at 0x7fd2281f0350>"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC_gapfilled"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add RefSeq annotations\n",
    "\n",
    "patric_refseq = {}\n",
    "with open('/home/matthew/Desktop/repos/Jenior_iCdJ794_2019/data/patric_refseq_genes.tsv') as ref:\n",
    "    for line in ref:\n",
    "        line = line.split()\n",
    "        patric_refseq[line[0]] = line[1]\n",
    "        \n",
    "for gene in cd630_PATRIC_gapfilled.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        try:\n",
    "            gene.annotation['refseq'] = patric_refseq[gene.id]\n",
    "        except:\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "\n",
    "# Metabolites\n",
    "for cpd in cd630_PATRIC_gapfilled.metabolites:\n",
    "    if 'cpd' in cpd.id:\n",
    "        new_id = str(cpd.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        cpd.annotation['seed.compound'] = new_id\n",
    "\n",
    "cd630_PATRIC_gapfilled.metabolites.C21400_c.annotation['kegg'] = 'C21400'        \n",
    "cd630_PATRIC_gapfilled.metabolites.C21399_c.annotation['kegg'] = 'C21399'        \n",
    "cd630_PATRIC_gapfilled.metabolites.C21090_c.annotation['kegg'] = 'C21090'        \n",
    "cd630_PATRIC_gapfilled.metabolites.C21399_e.annotation['kegg'] = 'C21399'        \n",
    "\n",
    "\n",
    "# Reactions\n",
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    if 'rxn' in rxn.id or 'EX_' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['seed.reaction'] = new_id\n",
    "    elif 'R' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['kegg.genes'] = new_id\n",
    "\n",
    "cd630_PATRIC_gapfilled.reactions.SK_cpd11416_c.annotation['seed.reaction'] = 'SK_cpd11416'\n",
    "cd630_PATRIC_gapfilled.reactions.ENOG4108HXH_c.annotation['eggnog'] = 'ENOG4108HXH'\n",
    "cd630_PATRIC_gapfilled.reactions.COG3601_c.annotation['cog'] = 'COG3601'\n",
    "cd630_PATRIC_gapfilled.reactions.K20025_c.annotation['kegg.genes'] = 'K20025'\n",
    "\n",
    "\n",
    "# Genes\n",
    "for gene in cd630_PATRIC_gapfilled.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        gene.annotation['patric'] = 'fig|' + gene.id\n",
    "    elif 'CD630_' in gene.id:\n",
    "        gene.annotation['uniprot'] = gene.id\n",
    "\n",
    "cd630_PATRIC_gapfilled.genes.T4BV06_CLODI.annotation['kegg'] = 'T4BV06_CLODI'\n",
    "cd630_PATRIC_gapfilled.genes.K00529.annotation['kegg'] = 'K00529'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add SBOs\n",
    "\n",
    "# Metabolites\n",
    "for cpd in cd630_PATRIC_gapfilled.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "\n",
    "# Reactions\n",
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    substrates = list(rxn.metabolites)\n",
    "    compartments = set([x.compartment for x in substrates])\n",
    "    \n",
    "    if 'EX_' in rxn.id:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000627' # exchange\n",
    "    elif len(compartments) > 1:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "    else:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "\n",
    "# Biomass\n",
    "cd630_PATRIC_gapfilled.reactions.dna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.rna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.teichoicacid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.peptidoglycan_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.cellwall_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.lipid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.cofactor_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.biomass.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.SK_cpd11416_c.annotation['sbo'] = 'SBO:0000632'  \n",
    "\n",
    "\n",
    "# Genes\n",
    "for gene in cd630_PATRIC_gapfilled.genes:\n",
    "    gene.annotation['sbo'] = 'SBO:0000243'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import flux_variability_analysis\n",
    "\n",
    "def test_unbounded(model):\n",
    "    \n",
    "    fva_result = flux_variability_analysis(model, fraction_of_optimum=1.0)\n",
    "\n",
    "    lower_bounds = numpy.asarray([rxn.lower_bound for rxn in model.reactions], dtype=float)\n",
    "    upper_bounds = numpy.asarray([rxn.upper_bound for rxn in model.reactions], dtype=float)\n",
    "    lower_bound = numpy.nanmedian(lower_bounds[lower_bounds != 0.0])\n",
    "    upper_bound = numpy.nanmedian(upper_bounds[upper_bounds != 0.0])\n",
    "\n",
    "    # Find those reactions whose flux is close to or outside of the median\n",
    "    # upper or lower bound, i.e., appears unconstrained.\n",
    "    unlimited_flux = fva_result.loc[\n",
    "            numpy.isclose(fva_result[\"maximum\"], upper_bound, atol=1e-7) |\n",
    "            (fva_result[\"maximum\"] > upper_bound) |\n",
    "            numpy.isclose(fva_result[\"minimum\"], lower_bound, atol=1e-7) |\n",
    "            (fva_result[\"minimum\"] < lower_bound)].index.tolist()\n",
    "    \n",
    "    saved = []\n",
    "    for rxn in unlimited_flux:\n",
    "        \n",
    "        reaction = model.reactions.get_by_id(rxn) \n",
    "        substrates = set([x.id for x in reaction.metabolites])\n",
    "        ignore = set(['cpd00003_c','cpd00004_c','cpd00005_c','cpd00006_c'])\n",
    "        \n",
    "        if 'EX_' in rxn:\n",
    "            continue\n",
    "        elif reaction.bounds != (-1000., 1000.): \n",
    "            continue\n",
    "        elif 'transferase' in reaction.name:\n",
    "            continue\n",
    "        elif reaction.name.find('mutase') != -1:\n",
    "            continue\n",
    "        elif 'isomerase' in reaction.name:\n",
    "            continue\n",
    "        elif 'hydrolase' in reaction.name:\n",
    "            continue \n",
    "        elif len(substrates & ignore) >= 1:\n",
    "            continue\n",
    "        else:\n",
    "            saved.append(rxn)\n",
    "        \n",
    "    print('Potentially unbounded reactions in default medium: ' + str(len(saved)))\n",
    "    return saved\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "unlimited = test_unbounded(cd630_PATRIC_gapfilled)\n",
    "\n",
    "# Correct reversibility to close potentially leaky cycles\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01322_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05617_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05607_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05655_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05496_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05202_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn00606_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn10184_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05494_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05610_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn00575_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05209_c.bounds = (-1000., 0.) # Na+ antiporter\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05226_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn02005_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05518_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn29571_c.bounds = (0., 1000.) # Neu5Ac H+ symporter\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05486_c.bounds = (0., 1000.) # N-Acetylmannosamine H+ symporter\n",
    "cd630_PATRIC_gapfilled.reactions.rxn10175_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05485_c.bounds = (0., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct last few gene IDs\n",
    "corrected_genes = {'CD630_00450': '272563.120.peg.49', \n",
    "                   'CD630_01120': '272563.120.peg.124', \n",
    "                   'CD630_01530': '272563.120.peg.166', \n",
    "                   'CD630_01540': '272563.120.peg.167', \n",
    "                   'CD630_01550': '272563.120.peg.168', \n",
    "                   'CD630_01790': '272563.120.peg.195', \n",
    "                   'CD630_02440': '272563.8.peg.262', \n",
    "                   'CD630_03940': '272563.120.peg.414', \n",
    "                   'CD630_03950': '272563.120.peg.415', \n",
    "                   'CD630_03960': '272563.120.peg.416', \n",
    "                   'CD630_03970': '272563.120.peg.417', \n",
    "                   'CD630_03980': '272563.120.peg.418', \n",
    "                   'CD630_06820': '272563.8.peg.717', \n",
    "                   'CD630_07500': '272563.120.peg.772', \n",
    "                   'CD630_07680': '272563.120.peg.790', \n",
    "                   'CD630_08730': '272563.120.peg.897', \n",
    "                   'CD630_08760': '272563.120.peg.900', \n",
    "                   'CD630_10030': '272563.120.peg.1042', \n",
    "                   'CD630_10040': '272563.120.peg.1043', \n",
    "                   'CD630_10100': '272563.120.peg.1049', \n",
    "                   'CD630_10540': '272563.120.peg.1094', \n",
    "                   'CD630_12250': '272563.8.peg.1284', \n",
    "                   'CD630_12590': '272563.120.peg.1317', \n",
    "                   'CD630_14930': '272563.120.peg.1562', \n",
    "                   'CD630_17760': '272563.120.peg.1869', \n",
    "                   'CD630_21960': '272563.120.peg.2318', \n",
    "                   'CD630_21990': '272563.120.peg.2321', \n",
    "                   'CD630_21991': '272563.120.peg.2322', \n",
    "                   'CD630_22770': '272563.120.peg.2404', \n",
    "                   'CD630_23310': '272563.120.peg.2462', \n",
    "                   'CD630_23390': '272563.120.peg.2470', \n",
    "                   'CD630_23410': '272563.120.peg.2472', \n",
    "                   'CD630_23420': '272563.120.peg.2473', \n",
    "                   'CD630_23540': '272563.120.peg.2485', \n",
    "                   'CD630_23710': '272563.120.peg.2504', \n",
    "                   'CD630_23800': '272563.120.peg.2513', \n",
    "                   'CD630_24170': '272563.120.peg.2551', \n",
    "                   'CD630_24180': '272563.120.peg.2552', \n",
    "                   'CD630_25690': '272563.120.peg.2710', \n",
    "                   'CD630_26510': '272563.8.peg.2775', \n",
    "                   'CD630_26550': '272563.120.peg.2798', \n",
    "                   'CD630_30130': '272563.8.peg.3157', \n",
    "                   'CD630_30140': '272563.120.peg.3181', \n",
    "                   'CD630_30150': '272563.120.peg.3182', \n",
    "                   'CD630_32370_1': '272563.120.peg.3419', \n",
    "                   'CD630_32370_2': '272563.120.peg.3419', \n",
    "                   'CD630_32440': '272563.120.peg.3426', \n",
    "                   'CD630_35150': '272563.120.peg.3715',\n",
    "                   'K00529': '272563.120.peg.1609',\n",
    "                   'T4BV06_CLODI': '1151372.4.peg.1723'}\n",
    "cobra.manipulation.modify.rename_genes(cd630_PATRIC_gapfilled, corrected_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Name new model\n",
    "cd630_PATRIC_gapfilled.name = 'Clostridium difficile 630'\n",
    "cd630_PATRIC_gapfilled.id = 'iCdR' + str(len(cd630_PATRIC_gapfilled.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    fixed_gpr = rxn.gene_reaction_rule.replace('( ','').replace(' )','')\n",
    "    rxn.gene_reaction_rule = fixed_gpr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(cd630_PATRIC_gapfilled, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.sbml')\n",
    "cobra.io.save_json_model(cd630_PATRIC_gapfilled, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "iCdG791 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gene annotations\n",
    "patric_refseq = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/cd630_patric_refseq_genes.tsv') as ref:\n",
    "    for line in ref:\n",
    "        line = line.split()\n",
    "        patric_refseq[line[0]] = line[1]\n",
    "        \n",
    "for gene in iCdG791.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        try:\n",
    "            gene.annotation['refseq'] = patric_refseq[gene.id]\n",
    "            gene.annotation['patric'] = 'fig|' + gene.id\n",
    "        except:\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Metabolite annotations\n",
    "for cpd in iCdG791.metabolites:\n",
    "    if 'cpd' in cpd.id:\n",
    "        new_id = str(cpd.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        cpd.annotation['seed.compound'] = new_id\n",
    "\n",
    "iCdG791.metabolites.C21400_c.annotation['kegg'] = 'C21400'        \n",
    "iCdG791.metabolites.C21399_c.annotation['kegg'] = 'C21399'        \n",
    "iCdG791.metabolites.C21090_c.annotation['kegg'] = 'C21090'        \n",
    "iCdG791.metabolites.C21399_e.annotation['kegg'] = 'C21399' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reaction annotations\n",
    "for rxn in iCdG791.reactions:\n",
    "    if 'rxn' in rxn.id or 'EX_' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['seed.reaction'] = new_id\n",
    "    elif 'R' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['kegg.genes'] = new_id\n",
    "\n",
    "iCdG791.reactions.SK_cpd11416_c.annotation['seed.reaction'] = 'SK_cpd11416'\n",
    "iCdG791.reactions.ENOG4108HXH_c.annotation['eggnog'] = 'ENOG4108HXH'\n",
    "iCdG791.reactions.COG3601_c.annotation['cog'] = 'COG3601'\n",
    "iCdG791.reactions.K20025_c.annotation['kegg.genes'] = 'K20025'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add SBOs\n",
    "\n",
    "# Metabolites\n",
    "for cpd in iCdG791.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "\n",
    "# Reactions\n",
    "for rxn in iCdG791.reactions:\n",
    "    substrates = list(rxn.metabolites)\n",
    "    compartments = set([x.compartment for x in substrates])\n",
    "    \n",
    "    if 'EX_' in rxn.id:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000627' # exchange\n",
    "    elif len(compartments) > 1:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "    else:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "\n",
    "# Biomass\n",
    "iCdG791.reactions.dna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.rna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.teichoicacid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.peptidoglycan_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.cellwall_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.lipid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.cofactor_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.biomass.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.SK_cpd11416_c.annotation['sbo'] = 'SBO:0000632'  \n",
    "\n",
    "# Genes\n",
    "for gene in iCdG791.genes:\n",
    "    gene.annotation['sbo'] = 'SBO:0000243'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct residual gene names\n",
    "gene_name_fixes = {'272563.120.peg.1317':'Na(+)-dependent branched-chain amino acid transporter',\n",
    "'272563.120.peg.772':'ABC transporter, substrate-binding protein (cluster 3, basic aa/glutamine/opines)',\n",
    "'272563.120.peg.1043':'Fumarate hydratase class I, beta region (EC 4.2.1.2); L(+)-tartrate dehydratase beta subunit (EC 4.2.1.32) | fumB',\n",
    "'272563.120.peg.1042':'Fumarate hydratase class I, alpha region (EC 4.2.1.2); L(+)-tartrate dehydratase alpha subunit (EC 4.2.1.32) | fumA',\n",
    "'272563.120.peg.2504':'L-aspartate oxidase (EC 1.4.3.16) | nadB',\n",
    "'272563.120.peg.195':'NAD-specific glutamate dehydrogenase (EC 1.4.1.2) | gluD',\n",
    "'272563.120.peg.2551':'PTS system, glucitol/sorbitol-specific IIB component (EC 2.7.1.198) / PTS system, glucitol/sorbitol-specific IIC component 2 | srlE_2',\n",
    "'272563.120.peg.2552':'PTS system, glucitol/sorbitol-specific IIC component | srlA',\n",
    "'272563.120.peg.1049':'N-acetylgalactosamine-6-phosphate deacetylase | nagA',\n",
    "'272563.120.peg.2404':'ribulose-5-phosphate 4-epimerase and related epimerases and aldolases',\n",
    "'272563.120.peg.49':'Ribulose-5-phosphate 4-epimerase and related epimerases and aldolases',\n",
    "'272563.120.peg.3715':'N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23) / Glucosamine-1-phosphate N-acetyltransferase (EC 2.3.1.157) | glmU',\n",
    "'272563.120.peg.3181':'PTS system, IIB component',\n",
    "'272563.8.peg.3157':'PTS system, IIC component',\n",
    "'272563.120.peg.3182':'PTS system, IIA component',\n",
    "'1151372.4.peg.1723':'Branched-chain amino acid aminotransferase (EC 2.6.1.42) | ilvE',\n",
    "'272563.120.peg.900':'ABC transporter, substrate-binding protein',\n",
    "'272563.120.peg.897':'ABC transporter, substrate-binding protein',\n",
    "'272563.8.peg.717':'sodium-solute symporter, putative',\n",
    "'272563.120.peg.2710':'Mannosylglycerate hydrolase (EC 3.2.1.170)',\n",
    "'272563.120.peg.1869':'ABC transporter, ATP-binding protein (cluster 3, basic aa/glutamine/opines)',\n",
    "'272563.120.peg.1562':'DNA-3-methyladenine glycosylase II (EC 3.2.2.21)',\n",
    "'272563.120.peg.124':'Phosphate butyryltransferase (EC 2.3.1.19) | ptb',\n",
    "'272563.120.peg.2322':'2-oxoglutarate/2-oxoacid ferredoxin oxidoreductase, delta subunit, ferredoxin-like 4Fe-4S binding protein (EC 1.2.7.-)',\n",
    "'272563.120.peg.2321':'2-oxoglutarate/2-oxoacid ferredoxin oxidoreductase, alpha subunit (EC 1.2.7.-)',\n",
    "'272563.120.peg.167':'4-hydroxyphenylacetate decarboxylase, small subunit (EC 4.1.1.83) | hpdC',\n",
    "'272563.120.peg.166':'4-hydroxyphenylacetate decarboxylase, large subunit (EC 4.1.1.83) | hpdB',\n",
    "'272563.120.peg.2513':'Indolepyruvate oxidoreductase subunit IorB (EC 1.2.7.8) | iorB',\n",
    "'272563.120.peg.3426':'D-proline reductase, 45 kDa subunit (EC 1.21.4.1) / D-proline reductase, 23 kDa subunit (EC 1.21.4.1) | prdA',\n",
    "'272563.120.peg.2485':'Glycine reductase component B beta subunit (EC 1.21.4.2) / Glycine reductase component B alpha subunit (EC 1.21.4.2) | grdE',\n",
    "'272563.120.peg.414':'D-lactate dehydrogenase (EC 1.1.1.28) | ldhA',\n",
    "'272563.120.peg.417':'2-hydroxyglutaryl-CoA dehydratase, D-component HgdA | hadB',\n",
    "'272563.120.peg.418':'2-hydroxyglutaryl-CoA dehydratase, D-component HgdB | hadC',\n",
    "'272563.120.peg.415':'Hydroxyproline dehydratase putative | hadA',\n",
    "'272563.120.peg.416':'CoA-substrate-specific enzyme activase; 2-hydroxyglutaryl-CoA dehydratase activator, A-component | hadI',\n",
    "'272563.120.peg.3419':'Proline racemase (EC 5.1.1.4) | prdF',\n",
    "'272563.120.peg.1609':'NADH-dependent reduced ferredoxin:NADP+ oxidoreductase subunit A',\n",
    "'272563.120.peg.168':'4-hydroxyphenylacetate decarboxylase activating enzyme | hpdA',\n",
    "'272563.120.peg.2318':'Acetyl-CoA synthetase (ADP-forming) alpha and beta chains, putative',\n",
    "'272563.120.peg.2473':'Succinate-semialdehyde dehydrogenase, CoA-dependent | sucD',\n",
    "'272563.120.peg.2472':'4-hydroxybutanoyl-CoA dehydratase (EC 4.2.1.120) / Vinylacetyl-CoA Delta-isomerase (EC 5.3.3.3) | abfD',\n",
    "'272563.120.peg.1094':'Electron bifurcating butyryl-CoA dehydrogenase (NAD+, ferredoxin) | bcd2',\n",
    "'272563.120.peg.2470':'4-hydroxybutyrate coenzyme A transferase | abfT',\n",
    "'272563.120.peg.2798':'UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase (EC 6.3.2.10) | murF',\n",
    "'272563.8.peg.262':'hypothetical protein',\n",
    "'272563.120.peg.2462':'Mannitol-1-phosphate 5-dehydrogenase (EC 1.1.1.17) | mtlD',\n",
    "'272563.120.peg.790':'Sorbitol-6-phosphate 2-dehydrogenase (EC 1.1.1.140) | gutD'}\n",
    "\n",
    "for gene in gene_name_fixes.keys():\n",
    "    iCdG791.genes.get_by_id(gene).name = gene_name_fixes[gene]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to files\n",
    "cobra.io.write_sbml_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG791.sbml')\n",
    "cobra.io.save_json_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG791.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Deal with genes that have now reactions\n",
    "import cobra\n",
    "iCdG791 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct names\n",
    "\n",
    "# Reactions\n",
    "iCdG791.reactions.rxn01987_c.name = 'Ribokinase'\n",
    "iCdG791.reactions.rxn05496_c.name = 'Amino acid transporter (ala-L)'\n",
    "iCdG791.reactions.rxn05202_c.name = 'Xanthine ion-coupled transport;'\n",
    "iCdG791.reactions.rxn12191_c.name = 'Choline monooxygenase (NADP)'\n",
    "iCdG791.reactions.rxn05663_c.name = 'L-tryptophan reversible transport via proton symport'\n",
    "iCdG791.reactions.rxn01213_c.name = 'Geranyl-diphosphate synthase'\n",
    "iCdG791.reactions.rxn10770_c.name = 'Choline dehydrogenase'\n",
    "iCdG791.reactions.rxn05616_c.name = 'Magnesium (+2) transport in via diffusion'\n",
    "iCdG791.reactions.rxn05669_c.name = 'Amino acid transporter (val-L)'\n",
    "iCdG791.reactions.rxn05244_c.name = 'Branched chain amino acid:H+ symporter (Isoleucine)'\n",
    "iCdG791.reactions.EX_cpd00276_e.name = 'D-Glucosamine exchange'\n",
    "iCdG791.reactions.rxn08783_c.name = 'D-lactate dehydrogenase'\n",
    "iCdG791.reactions.rxn03463_c.name = 'alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn03185_c.name = 'alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn02687_c.name = '(R)-mandelate dehydrogenase'\n",
    "iCdG791.reactions.rxn04779_c.name = ' N-cyclopropylmelamine deaminase'\n",
    "iCdG791.reactions.rxn07875_c.name = 'geranal dehydrogenase'\n",
    "iCdG791.reactions.rxn05127_c.name = 'gamma-glutamyl-gamma aminobutyric acid dehydrogenase'\n",
    "iCdG791.reactions.rxn11610_c.name = '2-hydroxymuconic semialdehyde hydrolase'\n",
    "iCdG791.reactions.rxn04781_c.name = 'N-cyclopropylammelide alkylamino hydrolase'\n",
    "iCdG791.reactions.rxn03891_c.name = 'trans-hexaprenyltranstransferase'\n",
    "iCdG791.reactions.rxn03762_c.name = 'CEA synthetase'\n",
    "iCdG791.reactions.rxn03536_c.name = 'Adenosyl cobinamide kinase'\n",
    "iCdG791.reactions.rxn03282_c.name = 'alcohol dehydrogenase'\n",
    "iCdG791.reactions.rxn04723_c.name = '3-methylsalicylaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn07261_c.name = 'ethanolaminephosphotransferase'\n",
    "iCdG791.reactions.rxn03540_c.name = 'Ado-cobyric acid synthase [glutamine hydrolyzing]'\n",
    "iCdG791.reactions.rxn04712_c.name = 'naphthyl-2-methylsuccinate synthase'\n",
    "iCdG791.reactions.rxn07622_c.name = 'dihydrophloroglucinol hydrolase'\n",
    "iCdG791.reactions.rxn06905_c.name = 'TCOA hydrolase'\n",
    "iCdG791.reactions.rxn04737_c.name = '2-naphthaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn03921_c.name = 'aldehyde dehydrogenase [EC:1.2.1.-]'\n",
    "iCdG791.reactions.rxn03721_c.name = 'Parathion hydrolase'\n",
    "iCdG791.reactions.rxn04727_c.name = '1-naphthaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn03537_c.name = 'Adenosyl cobinamide phosphate guanyltransferase'\n",
    "iCdG791.reactions.rxn08352_c.name = '(E)-4-hydroxy-3-methylbut-2-en-1-yl diphosphate reductase'\n",
    "iCdG791.reactions.rxn03135_c.name = 'Imidazole glycerol phosphate synthase amidotransferase'\n",
    "iCdG791.reactions.rxn03837_c.name = 'aryldialkylphosphatase'\n",
    "iCdG791.reactions.rxn06752_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn04413_c.name = 'adenosylcobinamide kinase'\n",
    "iCdG791.reactions.rxn04604_c.name = '(2E,4Z)-2-hydroxy-6-oxona-2,4-dienedioate succinylhydrolase'\n",
    "iCdG791.reactions.rxn03005_c.name = 'N2-Formyl-N1-(5-phospho-D-ribosyl)glycinamide formyltransferase'\n",
    "iCdG791.reactions.rxn07876_c.name = 'citronellal dehydrogenase'\n",
    "iCdG791.reactions.rxn06394_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn04780_c.name = 'N-cyclopropylammeline deaminase'\n",
    "iCdG791.reactions.rxn03486_c.name = '2-hydroxymuconic semi-aldehyde (m-xylene)'\n",
    "iCdG791.reactions.rxn07609_c.name = '2-hydroxy-6-oxo-6-(2-carboxyphenyl)-hexa-2,4-dienoate hydrolase'\n",
    "iCdG791.reactions.rxn03538_c.name = 'Adenosylcobalamin 5-phosphate synthase'\n",
    "iCdG791.reactions.rxn04238_c.name = '4-isopropenyl-2-ketocyclohexane-1-carboxyl-CoA hydrolase'\n",
    "iCdG791.reactions.rxn01208_c.name = '2-Oxo-4-methyl-3-carboxypentanoate decarboxylation'\n",
    "iCdG791.reactions.rxn03467_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn06750_c.name = 'alpha-N-acetylneuraminate alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn03892_c.name = 'all-trans-hexaprenyl-diphosphate synthase'\n",
    "iCdG791.reactions.rxn07493_c.name = 'GDP-L-galactose phosphorylase'\n",
    "iCdG791.reactions.rxn02362_c.name = 'cyclic 2,3-diphosphoglycerate synthase'\n",
    "iCdG791.reactions.rxn01905_c.name = 'cysteine sulfinic acid deaminase'\n",
    "iCdG791.reactions.rxn02604_c.name = 'vicianin beta-glucosidase'\n",
    "iCdG791.reactions.rxn05126_c.name = '4-(gamma-L-glutamylamino)butanal:NAD+ oxidoreductase'\n",
    "iCdG791.reactions.rxn04658_c.name = 'LysW-gamma-L-alpha-aminoadipate 6-semialdehyde:NADP+ 6-oxidoreductase (phosphorylating)'\n",
    "iCdG791.reactions.rxn04744_c.name = '4-hydroxymethylsalicyaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn03873_c.name = '6-oxocyclohex-1-ene-1-carbonyl-CoA hydrolase (decyclizing)'\n",
    "iCdG791.reactions.rxn07885_c.name = 'citronellal dehydrogenase'\n",
    "iCdG791.reactions.rxn06760_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn06759_c.name = 'alpha-N-acetylneuraminate alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn11584_c.name = 'aryldialkylphosphatase'\n",
    "iCdG791.reactions.rxn00970_c.name = 'Hydroxybenzoate synthase'\n",
    "iCdG791.reactions.rxn05039_c.name = '5-amino-6-(5-phosphoribitylamino)uracil phosphatase'\n",
    "iCdG791.reactions.rxn03365_c.name = 'R04916'\n",
    "iCdG791.reactions.rxn03366_c.name = 'R04916'\n",
    "iCdG791.reactions.rxn00568_c.name = 'Nitrite reductase (NAD(P)H)'\n",
    "\n",
    "# Metabolites\n",
    "iCdG791.metabolites.cpd00443_c.name = '4-Aminobenzoate'\n",
    "iCdG791.metabolites.cpd00443_e.name = '4-Aminobenzoate'\n",
    "iCdG791.metabolites.cpd00103_c.name = '5-Phospho-alpha-D-ribose 1-diphosphate'\n",
    "iCdG791.metabolites.cpd00226_c.name = 'Hypoxanthine'\n",
    "iCdG791.metabolites.cpd00104_c.name = 'Biotin'\n",
    "iCdG791.metabolites.cpd00104_e.name = 'Biotin'\n",
    "iCdG791.metabolites.cpd00202_c.name = '2-Isopentenyl diphosphate'\n",
    "iCdG791.metabolites.cpd11421_c.name = 'Reduced thioredoxin'\n",
    "iCdG791.metabolites.cpd11420_c.name = 'Oxidized thioredoxin'\n",
    "iCdG791.metabolites.cpd00309_c.name = 'Xanthine'\n",
    "iCdG791.metabolites.cpd00309_e.name = 'Xanthine'\n",
    "iCdG791.metabolites.cpd00276_e.name = 'D-Glucosamine'\n",
    "iCdG791.metabolites.cpd01080_c.name = 'Octadecanoic acid'\n",
    "iCdG791.metabolites.cpd00050_c.name = 'Flavin mononucleotide'\n",
    "iCdG791.metabolites.cpd00508_c.name = '(3S)-3-Methyl-2-oxopentanoate'\n",
    "iCdG791.metabolites.cpd00644_c.name = '(R)-Pantothenate'\n",
    "iCdG791.metabolites.cpd00226_e.name = 'Hypoxanthine'\n",
    "iCdG791.metabolites.cpd01080_e.name = 'Octadecanoic acid'\n",
    "iCdG791.metabolites.cpd02996_c.name = 'N-Acetyl-D-galactosaminyl-(N-acetylneuraminyl)-D-galactosyl-1,4- beta-D-glucosyl-N-acylsphingosine'\n",
    "iCdG791.metabolites.cpd00361_c.name = '(S)-Acetoin'\n",
    "iCdG791.metabolites.cpd00497_c.name = '(9-D-Ribosylxanthine)-5-phosphate'\n",
    "iCdG791.metabolites.cpd11434_c.name = 'Anteiso-C17:0 CoA Anteisoheptadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd02921_c.name = 'S-2-5-Amino-1-(5-phospho-D-ribosyl)imidazole-4-carboxamido-succinate'\n",
    "iCdG791.metabolites.cpd00042_c.name = 'Glutathione'\n",
    "iCdG791.metabolites.cpd11616_c.name = 'Aldehyde'\n",
    "iCdG791.metabolites.cpd02678_c.name = '5-P-ribosyl-N-formylglycineamide'\n",
    "iCdG791.metabolites.cpd02394_c.name = '5-Phosphoribosylglycinamide'\n",
    "iCdG791.metabolites.cpd02591_c.name = '(2E,6E,10E,14E)-geranylfarnesyl diphosphate'\n",
    "iCdG791.metabolites.cpd00200_c.name = '2-Oxoisocaproate'\n",
    "iCdG791.metabolites.cpd00599_c.name = '2-hydroxybenzoate'\n",
    "iCdG791.metabolites.cpd00853_c.name = '2,3-Bisphospho-D-glycerate'\n",
    "iCdG791.metabolites.cpd11432_c.name = 'Iso-C17:0 CoA Isoheptadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd11441_c.name = 'Iso-C16:0 CoA Isohexadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd11435_c.name = 'Iso-C14:0 CoA Isotetradecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd11439_c.name = 'Anteiso-C15:0 CoA Anteisopentadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd02972_c.name = 'alpha-N-Acetylneuraminyl-2,8-alpha-N-acetylneuraminyl-2,3-beta-D-galactosyl-R'\n",
    "iCdG791.metabolites.cpd00794_e.name = 'D-Trehalose'\n",
    "iCdG791.metabolites.cpd00794_c.name = 'D-Trehalose'\n",
    "iCdG791.metabolites.cpd00668_c.name = '2-Acetolactate'\n",
    "iCdG791.metabolites.cpd02631_c.name = '(3R,4S,5R)-3,4,5-Trihydroxy-1,2-cyclohexanedione'\n",
    "iCdG791.metabolites.cpd02968_c.name = 'UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-6-carboxy-L-lysyl-D-alanyl-D-alanine'\n",
    "iCdG791.metabolites.cpd00208_e.name = 'Lactose'\n",
    "iCdG791.metabolites.cpd00208_c.name = 'Lactose'\n",
    "iCdG791.metabolites.cpd00158_e.name = 'Cellobiose'\n",
    "iCdG791.metabolites.cpd00644_e.name = 'Pantothenate'\n",
    "iCdG791.metabolites.cpd02964_c.name = 'UDP-N-acetylmuramoyl-L-alanyl-D-gamma-glutamyl-meso-2-6-diaminopimelate'\n",
    "iCdG791.metabolites.cpd01741_c.name = 'Dodecanoate'\n",
    "iCdG791.metabolites.cpd00354_c.name = 'Nucleoside 5-diphosphate'\n",
    "iCdG791.metabolites.cpd02920_c.name = '2-Amino-4-hydroxy-6-hydroxymethyl-7-8-dihydropteridinediphosphate'\n",
    "iCdG791.metabolites.cpd00327_c.name = 'Stearoyl-CoA'\n",
    "iCdG791.metabolites.cpd11437_c.name = 'Iso-C15:0 CoA Isopentadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd00646_c.name = '4-Hydroxynitrobenzene'\n",
    "iCdG791.metabolites.cpd04011_c.name = 'Diethylthiophosphoric acid'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 278\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89.76884522354938"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove those genes that only act on macromolecules\n",
    "from cobra.manipulation.delete import remove_genes\n",
    "genes_to_be_removed = ['272563.8.peg.14','272563.8.peg.24','272563.8.peg.3559','272563.8.peg.9','272563.8.peg.13','272563.8.peg.2710','272563.8.peg.2183','272563.8.peg.1314','272563.8.peg.2596','272563.8.peg.3838','272563.8.peg.3388','272563.8.peg.3083','272563.8.peg.10','272563.8.peg.348','272563.8.peg.1360','272563.8.peg.1361','272563.8.peg.3664','272563.8.peg.1339','272563.8.peg.3717','272563.8.peg.77','272563.8.peg.1373','272563.8.peg.113','272563.8.peg.3857','272563.8.peg.1312','272563.8.peg.1527','272563.8.peg.1570','272563.8.peg.78','272563.8.peg.3858','272563.8.peg.89','272563.8.peg.94','272563.8.peg.83','272563.8.peg.1235','272563.8.peg.2253','272563.8.peg.69','272563.8.peg.95','272563.8.peg.1383','272563.8.peg.80','272563.8.peg.103','272563.8.peg.1221','272563.8.peg.112','272563.8.peg.3837','272563.8.peg.120','272563.8.peg.119','272563.8.peg.96','272563.8.peg.93','272563.8.peg.1322','272563.8.peg.721','272563.8.peg.97','272563.8.peg.2685','272563.8.peg.88','272563.8.peg.91','272563.8.peg.74','272563.8.peg.92','272563.8.peg.111','272563.8.peg.2567','272563.8.peg.73','272563.8.peg.86','272563.8.peg.101','272563.8.peg.100','272563.8.peg.98','272563.8.peg.1318','272563.8.peg.2595','272563.8.peg.99','272563.8.peg.85','272563.8.peg.114','272563.8.peg.109','272563.8.peg.3833','272563.8.peg.110','272563.8.peg.87','272563.8.peg.1219','272563.8.peg.90','272563.8.peg.722','272563.8.peg.79','272563.8.peg.84','272563.8.peg.3839','272563.8.peg.102','272563.8.peg.3653']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Lead, cadmium, zinc and mercury transporters\n",
    "genes_to_be_removed = ['272563.8.peg.620','272563.8.peg.1714','272563.8.peg.2961']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Pyruvate formate-lyase activating enzyme\n",
    "genes_to_be_removed = ['272563.8.peg.795','272563.8.peg.3436','272563.8.peg.1176']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# L(+)-tartrate dehydratase beta subunit (EC 4.2.1.32)\n",
    "genes_to_be_removed = ['272563.8.peg.1055','272563.8.peg.1054']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Sulfur carrier protein ThiS\n",
    "genes_to_be_removed = ['272563.8.peg.1781','272563.8.peg.1782']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Others\n",
    "remove_genes(iCdG791, ['272563.8.peg.907'])  # Predicted molybdate-responsive regulator YvgK in bacilli\n",
    "remove_genes(iCdG791, ['272563.8.peg.2617']) # Selenide,water dikinase (EC 2.7.9.3)\n",
    "remove_genes(iCdG791, ['272563.8.peg.2586']) # Hypothetical radical SAM family enzyme in heat shock gene cluster\n",
    "remove_genes(iCdG791, ['272563.8.peg.2223']) # Copper-translocating P-type ATPase (EC 3.6.3.4)\n",
    "remove_genes(iCdG791, ['272563.8.peg.3769']) # Methionine synthase activation domain (EC 2.1.1.13)\n",
    "remove_genes(iCdG791, ['272563.8.peg.472'])  # ornithine racemase (EC 5.1.1.12)\n",
    "remove_genes(iCdG791, ['272563.8.peg.3730']) # tRNA(Ile)-lysidine synthetase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 184\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Protein biosynthesis - correction to integrate genes that create respective tRNAs\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd17043_c = iCdG791.metabolites.get_by_id('cpd17043_c') # RNA\n",
    "cpd00067_c = iCdG791.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00018_c = iCdG791.metabolites.get_by_id('cpd00018_c') # AMP\n",
    "cpd00012_c = iCdG791.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "\n",
    "# L-Alanine\n",
    "cpd00035_c = iCdG791.metabolites.get_by_id('cpd00035_c') \n",
    "cpd11770_c = cobra.Metabolite(\n",
    "    'cpd11770_c',\n",
    "    formula='',\n",
    "    name='L-Alanyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd11770_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd11770_c.annotation['seed.compound'] = 'cpd11770'\n",
    "rxn06300_c = cobra.Reaction('rxn06300_c')\n",
    "rxn06300_c.gene_reaction_rule = '272563.8.peg.1347'\n",
    "rxn06300_c.name = 'L-Alanine:tRNA(Ala) ligase (AMP-forming)'\n",
    "rxn06300_c.bounds = (-1000., 1000.)\n",
    "rxn06300_c.add_metabolites({\n",
    "    cpd00035_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd11770_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06300_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06300_c.annotation['seed.reaction'] = 'rxn06300'\n",
    "\n",
    "# L-Arginine\n",
    "cpd00051_c = iCdG791.metabolites.get_by_id('cpd00051_c') \n",
    "cpd12036_c = cobra.Metabolite(\n",
    "    'cpd12036_c',\n",
    "    formula='',\n",
    "    name='L-Arginyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12036_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12036_c.annotation['seed.compound'] = 'cpd12036'\n",
    "rxn06432_c = cobra.Reaction('rxn06432_c')\n",
    "rxn06432_c.gene_reaction_rule = '272563.8.peg.746'\n",
    "rxn06432_c.name = 'L-Arginine:tRNA(Arg) ligase (AMP-forming)'\n",
    "rxn06432_c.bounds = (-1000., 1000.)\n",
    "rxn06432_c.add_metabolites({\n",
    "    cpd00051_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12036_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06432_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06432_c.annotation['seed.reaction'] = 'rxn06432'\n",
    "\n",
    "# L-Asparagine\n",
    "cpd00132_c = iCdG791.metabolites.get_by_id('cpd00132_c') \n",
    "cpd12313_c = cobra.Metabolite(\n",
    "    'cpd12313_c',\n",
    "    formula='',\n",
    "    name='L-Asparagyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12313_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12313_c.annotation['seed.compound'] = 'cpd12313'\n",
    "rxn06434_c = cobra.Reaction('rxn06434_c')\n",
    "rxn06434_c.gene_reaction_rule = '272563.8.peg.2360'\n",
    "rxn06434_c.name = 'L-Asparagine:tRNA(Asn) ligase (AMP-forming)'\n",
    "rxn06434_c.bounds = (-1000., 1000.)\n",
    "rxn06434_c.add_metabolites({\n",
    "    cpd00132_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12313_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06434_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06434_c.annotation['seed.reaction'] = 'rxn06434'\n",
    "\n",
    "# L-Aspartate\n",
    "cpd00041_c = iCdG791.metabolites.get_by_id('cpd00041_c') \n",
    "cpd12226_c = cobra.Metabolite(\n",
    "    'cpd12226_c',\n",
    "    formula='',\n",
    "    name='L-Aspartyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12226_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12226_c.annotation['seed.compound'] = 'cpd12226'\n",
    "rxn06936_c = cobra.Reaction('rxn06936_c')\n",
    "rxn06936_c.gene_reaction_rule = '272563.8.peg.2866'\n",
    "rxn06936_c.name = 'L-Aspartate:tRNA(Asp) ligase (AMP-forming)'\n",
    "rxn06936_c.bounds = (-1000., 1000.)\n",
    "rxn06936_c.add_metabolites({\n",
    "    cpd00041_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12226_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06936_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06936_c.annotation['seed.reaction'] = 'rxn06936'\n",
    "\n",
    "# L-Cysteine\n",
    "cpd00084_c = iCdG791.metabolites.get_by_id('cpd00084_c') \n",
    "cpd12255_c = cobra.Metabolite(\n",
    "    'cpd12255_c_c',\n",
    "    formula='',\n",
    "    name='Cysteinyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12255_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12255_c.annotation['seed.compound'] = 'cpd12255'\n",
    "rxn06435_c = cobra.Reaction('rxn06435_c')\n",
    "rxn06435_c.gene_reaction_rule = '272563.8.peg.62'\n",
    "rxn06435_c.name = 'L-Cysteine:tRNA(Cys) ligase (AMP-forming)'\n",
    "rxn06435_c.bounds = (-1000., 1000.)\n",
    "rxn06435_c.add_metabolites({\n",
    "    cpd00084_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12255_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06435_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06435_c.annotation['seed.reaction'] = 'rxn06435'\n",
    "\n",
    "# L-Glutamine - no gene\n",
    "cpd00053_c = iCdG791.metabolites.get_by_id('cpd00053_c') \n",
    "cpd12060_c = cobra.Metabolite(\n",
    "    'cpd12060_c',\n",
    "    formula='',\n",
    "    name='L-Glutaminyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12060_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12060_c.annotation['seed.compound'] = 'cpd12060'\n",
    "rxn06437_c = cobra.Reaction('rxn06437_c')\n",
    "rxn06437_c.gene_reaction_rule = '272563.120.peg.2173'\n",
    "rxn06437_c.name = 'L-Glutamine:tRNA(Gln) ligase (AMP-forming)'\n",
    "rxn06437_c.bounds = (-1000., 1000.)\n",
    "rxn06437_c.add_metabolites({\n",
    "    cpd00053_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12060_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06437_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06437_c.annotation['seed.reaction'] = 'rxn06437'\n",
    "\n",
    "# L-Glutamate - no gene\n",
    "cpd00023_c = iCdG791.metabolites.get_by_id('cpd00023_c') \n",
    "cpd12828_c = cobra.Metabolite(\n",
    "    'cpd12828_c',\n",
    "    formula='',\n",
    "    name='L-Glutamyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12828_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12828_c.annotation['seed.compound'] = 'cpd12828'\n",
    "rxn06436_c = cobra.Reaction('rxn06436_c')\n",
    "rxn06436_c.gene_reaction_rule = '272563.120.peg.55'\n",
    "rxn06436_c.name = 'L-Glutamate:tRNA(Gln) ligase (AMP-forming)'\n",
    "rxn06436_c.bounds = (-1000., 1000.)\n",
    "rxn06436_c.add_metabolites({\n",
    "    cpd00023_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12828_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06436_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06436_c.annotation['seed.reaction'] = 'rxn06436'\n",
    "\n",
    "# Glycine\n",
    "cpd00033_c = iCdG791.metabolites.get_by_id('cpd00033_c') \n",
    "cpd12100_c = cobra.Metabolite(\n",
    "    'cpd12100_c',\n",
    "    formula='',\n",
    "    name='Glycyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12100_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12100_c.annotation['seed.compound'] = 'cpd12100'\n",
    "rxn06438_c = cobra.Reaction('rxn06438_c')\n",
    "rxn06438_c.gene_reaction_rule = '272563.8.peg.2551 and 272563.8.peg.2552'\n",
    "rxn06438_c.name = 'glycine:tRNA(Gly) ligase (AMP-forming)'\n",
    "rxn06438_c.bounds = (-1000., 1000.)\n",
    "rxn06438_c.add_metabolites({\n",
    "    cpd00033_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12100_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06438_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06438_c.annotation['seed.reaction'] = 'rxn06438'\n",
    "\n",
    "# L-Histidine\n",
    "cpd00119_c = iCdG791.metabolites.get_by_id('cpd00119_c') \n",
    "cpd12228_c = cobra.Metabolite(\n",
    "    'cpd12228_c',\n",
    "    formula='',\n",
    "    name='L-Histidyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12228_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12228_c.annotation['seed.compound'] = 'cpd12228'\n",
    "rxn06439_c = cobra.Reaction('rxn06439_c')\n",
    "rxn06439_c.gene_reaction_rule = '272563.8.peg.2867'\n",
    "rxn06439_c.name = 'L-Histidine:tRNA(His) ligase (AMP-forming)'\n",
    "rxn06439_c.bounds = (-1000., 1000.)\n",
    "rxn06439_c.add_metabolites({\n",
    "    cpd00119_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12228_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06439_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06439_c.annotation['seed.reaction'] = 'rxn06439'\n",
    "\n",
    "# L-Isoleucine\n",
    "cpd00322_c = iCdG791.metabolites.get_by_id('cpd00322_c') \n",
    "cpd12256_c = cobra.Metabolite(\n",
    "    'cpd12256_c',\n",
    "    formula='',\n",
    "    name='L-Isoleucyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12256_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12256_c.annotation['seed.compound'] = 'cpd12256'\n",
    "rxn06440_c = cobra.Reaction('rxn06440_c')\n",
    "rxn06440_c.gene_reaction_rule = '272563.8.peg.2741'\n",
    "rxn06440_c.name = 'L-Isoleucine:tRNA(Ile) ligase (AMP-forming)'\n",
    "rxn06440_c.bounds = (-1000., 1000.)\n",
    "rxn06440_c.add_metabolites({\n",
    "    cpd00322_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12256_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06440_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06440_c.annotation['seed.reaction'] = 'rxn06440'\n",
    "\n",
    "# L-Leucine\n",
    "cpd00107_c = iCdG791.metabolites.get_by_id('cpd00107_c') \n",
    "cpd12003_c = cobra.Metabolite(\n",
    "    'cpd12003_c',\n",
    "    formula='',\n",
    "    name='Leucyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12003_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12003_c.annotation['seed.compound'] = 'cpd12003'\n",
    "rxn06441_c = cobra.Reaction('rxn06441_c')\n",
    "rxn06441_c.gene_reaction_rule = '272563.8.peg.2644'\n",
    "rxn06441_c.name = 'L-Leucine:tRNA(Leu) ligase (AMP-forming)'\n",
    "rxn06441_c.bounds = (-1000., 1000.)\n",
    "rxn06441_c.add_metabolites({\n",
    "    cpd00107_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12003_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06441_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06441_c.annotation['seed.reaction'] = 'rxn06441'\n",
    "\n",
    "# L-Lysine\n",
    "cpd00039_c = iCdG791.metabolites.get_by_id('cpd00039_c') \n",
    "cpd01326_c = cobra.Metabolite(\n",
    "    'cpd01326_c',\n",
    "    formula='',\n",
    "    name='L-Lysyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd01326_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd01326_c.annotation['seed.compound'] = 'cpd01326'\n",
    "rxn06442_c = cobra.Reaction('rxn06442_c')\n",
    "rxn06442_c.gene_reaction_rule = '272563.8.peg.3722'\n",
    "rxn06442_c.name = 'L-Lysine:tRNALys ligase (AMP-forming)'\n",
    "rxn06442_c.bounds = (-1000., 1000.)\n",
    "rxn06442_c.add_metabolites({\n",
    "    cpd00039_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd01326_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06442_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06442_c.annotation['seed.reaction'] = 'rxn06442'\n",
    "\n",
    "# L-Methionine\n",
    "cpd00060_c = iCdG791.metabolites.get_by_id('cpd00060_c') \n",
    "cpd12105_c = cobra.Metabolite(\n",
    "    'cpd12105_c',\n",
    "    formula='',\n",
    "    name='Methionyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12105_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12105_c.annotation['seed.compound'] = 'cpd12105'\n",
    "rxn06443_c = cobra.Reaction('rxn06443_c')\n",
    "rxn06443_c.gene_reaction_rule = '272563.8.peg.3708'\n",
    "rxn06443_c.name = 'L-Methionine:tRNAMet ligase (AMP-forming)'\n",
    "rxn06443_c.bounds = (-1000., 1000.)\n",
    "rxn06443_c.add_metabolites({\n",
    "    cpd00060_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12105_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06443_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06443_c.annotation['seed.reaction'] = 'rxn06443'\n",
    "\n",
    "# L-Phenylalanine\n",
    "cpd00066_c = iCdG791.metabolites.get_by_id('cpd00066_c') \n",
    "cpd12335_c = cobra.Metabolite(\n",
    "    'cpd12335_c',\n",
    "    formula='',\n",
    "    name='Phenylalanyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12335_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12335_c.annotation['seed.compound'] = 'cpd12335'\n",
    "rxn06444_c = cobra.Reaction('rxn06444_c')\n",
    "rxn06444_c.gene_reaction_rule = '272563.8.peg.734 and 272563.8.peg.735'\n",
    "rxn06444_c.name = 'L-Phenylalanine:tRNA(Ala) ligase (AMP-forming)'\n",
    "rxn06444_c.bounds = (-1000., 1000.)\n",
    "rxn06444_c.add_metabolites({\n",
    "    cpd00066_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12335_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06444_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06444_c.annotation['seed.reaction'] = 'rxn06444'\n",
    "\n",
    "# L-Proline\n",
    "cpd00129_c = iCdG791.metabolites.get_by_id('cpd00129_c') \n",
    "cpd12164_c = cobra.Metabolite(\n",
    "    'cpd12164_c',\n",
    "    formula='',\n",
    "    name='L-Prolyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12164_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12164_c.annotation['seed.compound'] = 'cpd12164'\n",
    "rxn06445_c = cobra.Reaction('rxn06445_c')\n",
    "rxn06445_c.gene_reaction_rule = '272563.120.peg.53'\n",
    "rxn06445_c.name = 'L-proline:tRNA(Pro) ligase (AMP-forming)'\n",
    "rxn06445_c.bounds = (-1000., 1000.)\n",
    "rxn06445_c.add_metabolites({\n",
    "    cpd00129_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12164_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06445_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06445_c.annotation['seed.reaction'] = 'rxn06445'\n",
    "\n",
    "# L-Serine\n",
    "cpd00054_c = iCdG791.metabolites.get_by_id('cpd00054_c') \n",
    "cpd12132_c = cobra.Metabolite(\n",
    "    'cpd12132_c',\n",
    "    formula='',\n",
    "    name='L-Seryl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12132_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12132_c.annotation['seed.compound'] = 'cpd12132'\n",
    "rxn06446_c = cobra.Reaction('rxn06446_c')\n",
    "rxn06446_c.gene_reaction_rule = '272563.8.peg.22 or 272563.8.peg.2968'\n",
    "rxn06446_c.name = 'L-Serine:tRNA(Ser) ligase (AMP-forming)'\n",
    "rxn06446_c.bounds = (-1000., 1000.)\n",
    "rxn06446_c.add_metabolites({\n",
    "    cpd00054_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12132_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06446_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06446_c.annotation['seed.reaction'] = 'rxn06446'\n",
    "\n",
    "# L-Threonine\n",
    "cpd00161_c = iCdG791.metabolites.get_by_id('cpd00161_c') \n",
    "cpd12229_c = cobra.Metabolite(\n",
    "    'cpd12229_c',\n",
    "    formula='',\n",
    "    name='Threonyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12229_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12229_c.annotation['seed.compound'] = 'cpd12229'\n",
    "rxn06447_c = cobra.Reaction('rxn06447_c')\n",
    "rxn06447_c.gene_reaction_rule = '272563.8.peg.601'\n",
    "rxn06447_c.name = 'L-Threonine:tRNA(Thr) ligase (AMP-forming)'\n",
    "rxn06447_c.bounds = (-1000., 1000.)\n",
    "rxn06447_c.add_metabolites({\n",
    "    cpd00161_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12229_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06447_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06447_c.annotation['seed.reaction'] = 'rxn06447'\n",
    "\n",
    "# L-Tryptophan\n",
    "cpd00065_c = iCdG791.metabolites.get_by_id('cpd00065_c') \n",
    "cpd12336_c = cobra.Metabolite(\n",
    "    'cpd12336_c',\n",
    "    formula='',\n",
    "    name='L-Tryptophanyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12336_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12336_c.annotation['seed.compound'] = 'cpd12336'\n",
    "rxn06448_c = cobra.Reaction('rxn06448_c')\n",
    "rxn06448_c.gene_reaction_rule = '272563.8.peg.2733'\n",
    "rxn06448_c.name = 'L-Tryptophan -tRNA(Trp) ligase (AMP-forming)'\n",
    "rxn06448_c.bounds = (-1000., 1000.)\n",
    "rxn06448_c.add_metabolites({\n",
    "    cpd00065_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12336_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06448_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06448_c.annotation['seed.reaction'] = 'rxn06448'\n",
    "\n",
    "# L-Tyrosine\n",
    "cpd00069_c = iCdG791.metabolites.get_by_id('cpd00069_c') \n",
    "cpd12194_c = cobra.Metabolite(\n",
    "    'cpd12194_c',\n",
    "    formula='',\n",
    "    name='L-Tyrosyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12194_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12194_c.annotation['seed.compound'] = 'cpd12194'\n",
    "rxn06280_c = cobra.Reaction('rxn06280_c')\n",
    "rxn06280_c.gene_reaction_rule = '272563.8.peg.1597'\n",
    "rxn06280_c.name = 'L-Tyrosine:tRNA(Tyr) ligase (AMP-forming)'\n",
    "rxn06280_c.bounds = (-1000., 1000.)\n",
    "rxn06280_c.add_metabolites({\n",
    "    cpd00069_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12194_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06280_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06280_c.annotation['seed.reaction'] = 'rxn06280'\n",
    "\n",
    "# L-Valine\n",
    "cpd00156_c = iCdG791.metabolites.get_by_id('cpd00156_c') \n",
    "cpd12133_c = cobra.Metabolite(\n",
    "    'cpd12133_c',\n",
    "    formula='',\n",
    "    name='L-Valyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12133_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12133_c.annotation['seed.compound'] = 'cpd12133'\n",
    "rxn06449_c = cobra.Reaction('rxn06449_c')\n",
    "rxn06449_c.gene_reaction_rule = '272563.8.peg.3409'\n",
    "rxn06449_c.name = 'L-Valine:tRNAVal ligase (AMP-forming)'\n",
    "rxn06449_c.bounds = (-1000., 1000.)\n",
    "rxn06449_c.add_metabolites({\n",
    "    cpd00156_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # RNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12133_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06449_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06449_c.annotation['seed.reaction'] = 'rxn06449'\n",
    "\n",
    "iCdG791.add_reactions([rxn06300_c,rxn06432_c,rxn06434_c,rxn06936_c,rxn06435_c,rxn06437_c,rxn06436_c,\n",
    "                       rxn06438_c,rxn06439_c,rxn06440_c,rxn06441_c,rxn06442_c,rxn06443_c,rxn06444_c,\n",
    "                       rxn06445_c,rxn06446_c,rxn06447_c,rxn06448_c,rxn06280_c,rxn06449_c])\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.2173').name = 'Glutaminyl-tRNA synthetase (EC 6.1.1.18)'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.2173').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.2173').annotation['patric'] = 'fig|272563.120.peg.2173'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.55').name = 'Glutamyl-tRNA synthetase (EC 6.1.1.17)'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.55').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.55').annotation['patric'] = 'fig|272563.120.peg.55'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.53').name = 'Prolyl-tRNA synthetase (EC 6.1.1.15)'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.53').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.53').annotation['patric'] = 'fig|272563.120.peg.53'\n",
    "\n",
    "#---------------------------------------------------------------------------#\n",
    "\n",
    "iCdG791.reactions.protein_rxn.remove_from_model(remove_orphans=False)\n",
    "cpd00001_c = iCdG791.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00008_c = iCdG791.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = iCdG791.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd17041_c = iCdG791.metabolites.get_by_id('cpd17041_c') # Protein\n",
    "\n",
    "# Create new reaction, now with tRNAs\n",
    "protein_rxn = cobra.Reaction('protein_rxn')\n",
    "protein_rxn.name = 'Protein biosynthesis'\n",
    "protein_rxn.lower_bound = 0.\n",
    "protein_rxn.upper_bound = 1000.\n",
    "protein_rxn.add_metabolites({\n",
    "    cpd11770_c: -0.264, # Alanine\n",
    "    cpd12036_c: -0.160, # Arginine\n",
    "    cpd12313_c: -0.322, # Asparagine\n",
    "    cpd12226_c: -0.286, # Aspartate\n",
    "    cpd12255_c: -0.062, # Cysteine\n",
    "    cpd12060_c: -0.120, # Glutamine\n",
    "    cpd12828_c: -0.367, # Glutamate\n",
    "    cpd12100_c: -0.314, # Glycine\n",
    "    cpd12228_c: -0.068, # Histidine\n",
    "    cpd12256_c: -0.498, # Isoleucine\n",
    "    cpd12003_c: -0.455, # Leucine\n",
    "    cpd01326_c: -0.457, # Lysine\n",
    "    cpd12105_c: -0.135, # Methionine\n",
    "    cpd12335_c: -0.219, # Phenylalanine\n",
    "    cpd12164_c: -0.137, # Proline\n",
    "    cpd12132_c: -0.331, # Serine\n",
    "    cpd12229_c: -0.248, # Threonine\n",
    "    cpd12336_c: -0.032, # Tryptophan\n",
    "    cpd12194_c: -0.208, # Tyrosine\n",
    "    cpd12133_c: -0.328, # Valine\n",
    "    cpd00002_c: -20.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17041_c: 1.0,\n",
    "    cpd00008_c: 20.0,\n",
    "    cpd00009_c: 20.0\n",
    "})\n",
    "\n",
    "iCdG791.add_reactions([protein_rxn])\n",
    "iCdG791.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 164\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "82.79962839526671"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd11581_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>gly-asn-L exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f3496646dd0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd11581_e <=> </p>\n",
       "                    <p style='text-align:right'>gly-asn-L <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd11581_e at 0x7f3496646dd0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Dipeptides\n",
    "# Methionyl aminopeptidase and associated active transport + exchanges\n",
    "\n",
    "# gly-glu-L \n",
    "rxn12634_c = universal_GramPos.reactions.rxn12634_c\n",
    "rxn12634_c.lower_bound = 0.\n",
    "rxn12634_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12600_c.remove_from_model(remove_orphans=False)\n",
    "rxn05534_c = universal_GramPos.reactions.rxn05534_c\n",
    "rxn05534_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12634_c,rxn05534_c])\n",
    "cpd11592_e = iCdG791.metabolites.cpd11592_e\n",
    "iCdG791.add_boundary(cpd11592_e, type='exchange', reaction_id='EX_cpd11592_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# met-L-ala-L\n",
    "rxn12636_c = universal_GramPos.reactions.rxn12636_c\n",
    "rxn12636_c.lower_bound = 0.\n",
    "rxn12636_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12544_c.remove_from_model(remove_orphans=False)\n",
    "rxn05536_c = universal_GramPos.reactions.rxn05536_c\n",
    "rxn05536_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12636_c,rxn05536_c])\n",
    "cpd11590_e = iCdG791.metabolites.cpd11590_e\n",
    "iCdG791.add_boundary(cpd11590_e, type='exchange', reaction_id='EX_cpd11590_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# gly-pro-L \n",
    "rxn12638_c = universal_GramPos.reactions.rxn12638_c\n",
    "rxn12638_c.lower_bound = 0.\n",
    "rxn12638_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12576_c.remove_from_model(remove_orphans=False)\n",
    "rxn05538_c = universal_GramPos.reactions.rxn05538_c\n",
    "rxn05538_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12638_c,rxn05538_c])\n",
    "cpd11588_e = iCdG791.metabolites.cpd11588_e\n",
    "iCdG791.add_boundary(cpd11588_e, type='exchange', reaction_id='EX_cpd11588_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# ala-L-glu-L\n",
    "rxn12640_c = universal_GramPos.reactions.rxn12640_c\n",
    "rxn12640_c.lower_bound = 0.\n",
    "rxn12640_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12523_c.remove_from_model(remove_orphans=False)\n",
    "rxn05541_c = universal_GramPos.reactions.rxn05541_c\n",
    "rxn05541_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12640_c,rxn05541_c])\n",
    "cpd11586_e = iCdG791.metabolites.cpd11586_e\n",
    "iCdG791.add_boundary(cpd11586_e, type='exchange', reaction_id='EX_cpd11586_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# ala-L-Thr-L\n",
    "rxn12644_c = universal_GramPos.reactions.rxn12644_c\n",
    "rxn12644_c.lower_bound = 0.\n",
    "rxn12644_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12616_c.remove_from_model(remove_orphans=False)\n",
    "rxn05545_c = universal_GramPos.reactions.rxn05545_c\n",
    "rxn05545_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12644_c,rxn05545_c])\n",
    "cpd11582_e = iCdG791.metabolites.cpd11582_e\n",
    "iCdG791.add_boundary(cpd11582_e, type='exchange', reaction_id='EX_cpd11582_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# gly-asn-L\n",
    "rxn12645_c = universal_GramPos.reactions.rxn12645_c\n",
    "rxn12645_c.lower_bound = 0.\n",
    "rxn12645_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12542_c.remove_from_model(remove_orphans=False)\n",
    "rxn05546_c = universal_GramPos.reactions.rxn05546_c\n",
    "rxn05546_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12645_c,rxn05546_c])\n",
    "cpd11581_e = iCdG791.metabolites.cpd11581_e \n",
    "iCdG791.add_boundary(cpd11581_e, type='exchange', reaction_id='EX_cpd11581_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd15606_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Gly-Tyr exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f3496862610</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd15606_e <=> </p>\n",
       "                    <p style='text-align:right'>Gly-Tyr <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd15606_e at 0x7f3496862610>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remaining dipeptide import and Aminopeptidases in universal bag\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H\n",
    "\n",
    "# L-Prolinylglycine\n",
    "rxn08133_c = universal_GramPos.reactions.rxn08133_c\n",
    "rxn08133_c.lower_bound = 0.\n",
    "rxn08133_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "cpd15551_e = cobra.Metabolite(\n",
    "    'cpd15551_e',\n",
    "    formula='C7H11N2O3',\n",
    "    name='L-Prolinylglycine',\n",
    "    compartment='extracellular')\n",
    "cpd15551_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd15551_e.annotation['seed.compound'] = 'cpd15551'\n",
    "cpd15551_c = universal_GramPos.metabolites.cpd15551_c\n",
    "rxn09189_c = cobra.Reaction('rxn09189_c')\n",
    "rxn09189_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn09189_c.name = 'L-Prolinylglycine (Pro-Gly) transport via ABC system'\n",
    "rxn09189_c.bounds = (0., 1000.)\n",
    "rxn09189_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd15551_e: -1.0, # L-Prolinylglycine\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Pi\n",
    "    cpd15551_c: 1.0, # L-Prolinylglycine\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn09189_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn09189_c.annotation['seed.reaction'] = 'rxn09189'\n",
    "iCdG791.add_reactions([rxn08133_c,rxn09189_c])\n",
    "iCdG791.add_boundary(cpd15551_e, type='exchange', reaction_id='EX_cpd15551_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Cys\n",
    "rxn12844_c = universal_GramPos.reactions.rxn12844_c\n",
    "rxn12844_c.lower_bound = 0.\n",
    "rxn12844_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12848_c = universal_GramPos.reactions.rxn12848_c\n",
    "rxn12848_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12844_c,rxn12848_c])\n",
    "cpd15603_e = iCdG791.metabolites.cpd15603_e\n",
    "iCdG791.add_boundary(cpd15603_e, type='exchange', reaction_id='EX_cpd15603_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Leu\n",
    "rxn12845_c = universal_GramPos.reactions.rxn12845_c\n",
    "rxn12845_c.lower_bound = 0.\n",
    "rxn12845_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12849_c = universal_GramPos.reactions.rxn12849_c\n",
    "rxn12849_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12845_c,rxn12849_c])\n",
    "cpd15604_e = iCdG791.metabolites.cpd15604_e\n",
    "iCdG791.add_boundary(cpd15604_e, type='exchange', reaction_id='EX_cpd15604_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Phe\n",
    "rxn12846_c = universal_GramPos.reactions.rxn12846_c\n",
    "rxn12846_c.lower_bound = 0.\n",
    "rxn12846_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12850_c = universal_GramPos.reactions.rxn12850_c\n",
    "rxn12850_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12846_c,rxn12850_c])\n",
    "cpd15605_e = iCdG791.metabolites.cpd15605_e\n",
    "iCdG791.add_boundary(cpd15605_e, type='exchange', reaction_id='EX_cpd15605_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Try\n",
    "rxn12847_c = universal_GramPos.reactions.rxn12847_c\n",
    "rxn12847_c.lower_bound = 0.\n",
    "rxn12847_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12851_c = universal_GramPos.reactions.rxn12851_c\n",
    "rxn12851_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12851_c,rxn12847_c])\n",
    "cpd15606_e = iCdG791.metabolites.cpd15606_e\n",
    "iCdG791.add_boundary(cpd15606_e, type='exchange', reaction_id='EX_cpd15606_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 150\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "82.98461354618698"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dipeptide-associated reactions not in universal bag\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H+\n",
    "\n",
    "# L-ala-L-asp\n",
    "cpd29695_e = cobra.Metabolite(\n",
    "    'cpd29695_e',\n",
    "    formula='C7H11N2O5',\n",
    "    name='L-ala-L-asp',\n",
    "    compartment='extracellular')\n",
    "cpd29695_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29695_e.annotation['seed.compound'] = 'cpd29695'\n",
    "iCdG791.add_boundary(cpd29695_e, type='exchange', reaction_id='EX_cpd29695_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29695_c = cobra.Metabolite(\n",
    "    'cpd29695_c',\n",
    "    formula='C7H11N2O5',\n",
    "    name='L-ala-L-asp',\n",
    "    compartment='cytosol')\n",
    "cpd29695_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29695_c.annotation['seed.compound'] = 'cpd29695'\n",
    "\n",
    "# Transport\n",
    "rxn30259_c = cobra.Reaction('rxn30259_c')\n",
    "rxn30259_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30259_c.name = 'Dipeptide transport via ABC system (ala-asp)'\n",
    "rxn30259_c.bounds = (0., 1000.)\n",
    "rxn30259_c.add_metabolites({\n",
    "    cpd29695_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29695_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30259_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30259_c.annotation['seed.reaction'] = 'rxn30259'\n",
    "iCdG791.add_reactions([rxn30259_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00041_c = iCdG791.metabolites.cpd00041_c # Aspartate\n",
    "rxn30237_c = cobra.Reaction('rxn30237_c')\n",
    "rxn30237_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30237_c.name = 'aminopeptidase (ala-asp)'\n",
    "rxn30237_c.bounds = (0., 1000.)\n",
    "rxn30237_c.add_metabolites({\n",
    "    cpd29695_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00041_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30237_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30237_c.annotation['seed.reaction'] = 'rxn30237'\n",
    "iCdG791.add_reactions([rxn30237_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-gly\n",
    "cpd29694_e = cobra.Metabolite(\n",
    "    'cpd29694_e',\n",
    "    formula='C5H13N2O2',\n",
    "    name='L-ala-gly',\n",
    "    compartment='extracellular')\n",
    "cpd29694_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29694_e.annotation['seed.compound'] = 'cpd29694'\n",
    "iCdG791.add_boundary(cpd29694_e, type='exchange', reaction_id='EX_cpd29694_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29694_c = cobra.Metabolite(\n",
    "    'cpd29694_c',\n",
    "    formula='C5H13N2O2',\n",
    "    name='L-ala-gly',\n",
    "    compartment='cytosol')\n",
    "cpd29694_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29694_c.annotation['seed.compound'] = 'cpd29694'\n",
    "\n",
    "# Transport\n",
    "rxn30262_c = cobra.Reaction('rxn30262_c')\n",
    "rxn30262_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30262_c.name = 'Dipeptide transport via ABC system (ala-gly)'\n",
    "rxn30262_c.bounds = (0., 1000.)\n",
    "rxn30262_c.add_metabolites({\n",
    "    cpd29694_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29694_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30262_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30262_c.annotation['seed.reaction'] = 'rxn30262'\n",
    "iCdG791.add_reactions([rxn30262_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "rxn30238_c = cobra.Reaction('rxn30238_c')\n",
    "rxn30238_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30238_c.name = 'Aminopeptidase (ala-gly)'\n",
    "rxn30238_c.bounds = (0., 1000.)\n",
    "rxn30238_c.add_metabolites({\n",
    "    cpd29694_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00033_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30238_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30238_c.annotation['seed.reaction'] = 'rxn30238'\n",
    "iCdG791.add_reactions([rxn30238_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-his\n",
    "cpd29697_e = cobra.Metabolite(\n",
    "    'cpd29697_e',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-his',\n",
    "    compartment='extracellular')\n",
    "cpd29697_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29697_e.annotation['seed.compound'] = 'cpd29697'\n",
    "iCdG791.add_boundary(cpd29697_e, type='exchange', reaction_id='EX_cpd29697_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29697_c = cobra.Metabolite(\n",
    "    'cpd29697_c',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-his',\n",
    "    compartment='cytosol')\n",
    "cpd29697_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29697_c.annotation['seed.compound'] = 'cpd29697'\n",
    "\n",
    "# Transport\n",
    "rxn30264_c = cobra.Reaction('rxn30264_c')\n",
    "rxn30264_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30264_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30264_c.bounds = (0., 1000.)\n",
    "rxn30264_c.add_metabolites({\n",
    "    cpd29697_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29697_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30264_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30264_c.annotation['seed.reaction'] = 'rxn30264'\n",
    "iCdG791.add_reactions([rxn30264_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00119_c = iCdG791.metabolites.cpd00119_c # Histidine\n",
    "rxn30239_c = cobra.Reaction('rxn30239_c')\n",
    "rxn30239_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30239_c.name = 'Aminopeptidase'\n",
    "rxn30239_c.bounds = (0., 1000.)\n",
    "rxn30239_c.add_metabolites({\n",
    "    cpd29697_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00119_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30239_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30239_c.annotation['seed.reaction'] = 'rxn30239'\n",
    "iCdG791.add_reactions([rxn30239_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-leu\n",
    "cpd29698_e = cobra.Metabolite(\n",
    "    'cpd29698_e',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-leu',\n",
    "    compartment='extracellular')\n",
    "cpd29698_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29698_e.annotation['seed.compound'] = 'cpd29698'\n",
    "iCdG791.add_boundary(cpd29698_e, type='exchange', reaction_id='EX_cpd29698_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29698_c = cobra.Metabolite(\n",
    "    'cpd29698_c',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-leu',\n",
    "    compartment='cytosol')\n",
    "cpd29698_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29698_c.annotation['seed.compound'] = 'cpd29698'\n",
    "\n",
    "# Transport\n",
    "rxn30266_c = cobra.Reaction('rxn30266_c')\n",
    "rxn30266_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30266_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30266_c.bounds = (0., 1000.)\n",
    "rxn30266_c.add_metabolites({\n",
    "    cpd29698_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29698_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30266_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30266_c.annotation['seed.reaction'] = 'rxn30266'\n",
    "iCdG791.add_reactions([rxn30266_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00107_c = iCdG791.metabolites.cpd00107_c # Leucine\n",
    "rxn30240_c = cobra.Reaction('rxn30240_c')\n",
    "rxn30240_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30240_c.name = 'Aminopeptidase'\n",
    "rxn30240_c.bounds = (0., 1000.)\n",
    "rxn30240_c.add_metabolites({\n",
    "    cpd29698_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00107_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30240_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30240_c.annotation['seed.reaction'] = 'rxn30240'\n",
    "iCdG791.add_reactions([rxn30240_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-gln\n",
    "cpd29696_e = cobra.Metabolite(\n",
    "    'cpd29696_e',\n",
    "    formula='C8H15N3O4',\n",
    "    name='L-ala-L-gln',\n",
    "    compartment='extracellular')\n",
    "cpd29696_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29696_e.annotation['seed.compound'] = 'cpd29696'\n",
    "iCdG791.add_boundary(cpd29696_e, type='exchange', reaction_id='EX_cpd29696_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29696_c = cobra.Metabolite(\n",
    "    'cpd29696_c',\n",
    "    formula='C8H15N3O4',\n",
    "    name='L-ala-L-gln',\n",
    "    compartment='cytosol')\n",
    "cpd29696_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29696_c.annotation['seed.compound'] = 'cpd29696'\n",
    "\n",
    "# Transport\n",
    "rxn30268_c = cobra.Reaction('rxn30268_c')\n",
    "rxn30268_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30268_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30268_c.bounds = (0., 1000.)\n",
    "rxn30268_c.add_metabolites({\n",
    "    cpd29696_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29696_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30268_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30268_c.annotation['seed.reaction'] = 'rxn30268'\n",
    "iCdG791.add_reactions([rxn30268_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00053_c = iCdG791.metabolites.cpd00053_c # Glutamine\n",
    "rxn30241_c = cobra.Reaction('rxn30241_c')\n",
    "rxn30241_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30241_c.name = 'Aminopeptidase'\n",
    "rxn30241_c.bounds = (0., 1000.)\n",
    "rxn30241_c.add_metabolites({\n",
    "    cpd29696_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00053_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30241_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30241_c.annotation['seed.reaction'] = 'rxn30241'\n",
    "iCdG791.add_reactions([rxn30241_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-thr\n",
    "cpd29699_e = cobra.Metabolite(\n",
    "    'cpd29699_e',\n",
    "    formula='C7H14N2O4',\n",
    "    name='L-ala-L-thr',\n",
    "    compartment='extracellular')\n",
    "cpd29699_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29699_e.annotation['seed.compound'] = 'cpd29699'\n",
    "iCdG791.add_boundary(cpd29699_e, type='exchange', reaction_id='EX_cpd29699_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29699_c = cobra.Metabolite(\n",
    "    'cpd29699_c',\n",
    "    formula='C7H14N2O4',\n",
    "    name='L-ala-L-thr',\n",
    "    compartment='cytosol')\n",
    "cpd29699_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29699_c.annotation['seed.compound'] = 'cpd29699'\n",
    "\n",
    "# Transport\n",
    "rxn30270_c = cobra.Reaction('rxn30270_c')\n",
    "rxn30270_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30270_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30270_c.bounds = (0., 1000.)\n",
    "rxn30270_c.add_metabolites({\n",
    "    cpd29699_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29699_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30270_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30270_c.annotation['seed.reaction'] = 'rxn30270'\n",
    "iCdG791.add_reactions([rxn30270_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00161_c = iCdG791.metabolites.cpd00161_c # Threonine\n",
    "rxn30242_c = cobra.Reaction('rxn30242_c')\n",
    "rxn30242_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30242_c.name = 'Aminopeptidase'\n",
    "rxn30242_c.bounds = (0., 1000.)\n",
    "rxn30242_c.add_metabolites({\n",
    "    cpd29699_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00161_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30242_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30242_c.annotation['seed.reaction'] = 'rxn30242'\n",
    "iCdG791.add_reactions([rxn30242_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# gly-L-met\n",
    "cpd29693_e = cobra.Metabolite(\n",
    "    'cpd29693_e',\n",
    "    formula='C7H14N2O3S',\n",
    "    name='gly-L-met',\n",
    "    compartment='extracellular')\n",
    "cpd29693_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29693_e.annotation['seed.compound'] = 'cpd29693'\n",
    "iCdG791.add_boundary(cpd29693_e, type='exchange', reaction_id='EX_cpd29693_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29693_c = cobra.Metabolite(\n",
    "    'cpd29693_c',\n",
    "    formula='C7H14N2O3S',\n",
    "    name='gly-L-met',\n",
    "    compartment='cytosol')\n",
    "cpd29693_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29693_c.annotation['seed.compound'] = 'cpd29693'\n",
    "\n",
    "# Transport\n",
    "rxn30274_c = cobra.Reaction('rxn30274_c')\n",
    "rxn30274_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30274_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30274_c.bounds = (0., 1000.)\n",
    "rxn30274_c.add_metabolites({\n",
    "    cpd29693_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29693_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30274_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30274_c.annotation['seed.reaction'] = 'rxn30274'\n",
    "iCdG791.add_reactions([rxn30274_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "cpd00060_c = iCdG791.metabolites.cpd00060_c # Methionine\n",
    "rxn30244_c = cobra.Reaction('rxn30244_c')\n",
    "rxn30244_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30244_c.name = 'Aminopeptidase'\n",
    "rxn30244_c.bounds = (0., 1000.)\n",
    "rxn30244_c.add_metabolites({\n",
    "    cpd29693_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00033_c: 1.0, # peptide 1\n",
    "    cpd00060_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30244_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30244_c.annotation['seed.reaction'] = 'rxn30244'\n",
    "iCdG791.add_reactions([rxn30244_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# gly-L-asn\n",
    "cpd29690_e = cobra.Metabolite(\n",
    "    'cpd29690_e',\n",
    "    formula='C6H11N3O4',\n",
    "    name='gly-L-asn',\n",
    "    compartment='extracellular')\n",
    "cpd29690_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29690_e.annotation['seed.compound'] = 'cpd29690'\n",
    "iCdG791.add_boundary(cpd29690_e, type='exchange', reaction_id='EX_cpd29690_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29690_c = cobra.Metabolite(\n",
    "    'cpd29690_c',\n",
    "    formula='C6H11N3O4',\n",
    "    name='gly-L-asn',\n",
    "    compartment='cytosol')\n",
    "cpd29690_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29690_c.annotation['seed.compound'] = 'cpd29690'\n",
    "\n",
    "# Transport\n",
    "rxn30276_c = cobra.Reaction('rxn30276_c')\n",
    "rxn30276_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30276_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30276_c.bounds = (0., 1000.)\n",
    "rxn30276_c.add_metabolites({\n",
    "    cpd29690_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29690_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30276_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30276_c.annotation['seed.reaction'] = 'rxn30276'\n",
    "iCdG791.add_reactions([rxn30276_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "cpd00132_c = iCdG791.metabolites.cpd00132_c # Asparagine\n",
    "\n",
    "rxn30245_c = cobra.Reaction('rxn30245_c')\n",
    "rxn30245_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30245_c.name = 'Aminopeptidase'\n",
    "rxn30245_c.bounds = (0., 1000.)\n",
    "rxn30245_c.add_metabolites({\n",
    "    cpd29690_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00033_c: 1.0, # peptide 1\n",
    "    cpd00132_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30245_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30245_c.annotation['seed.reaction'] = 'rxn30245'\n",
    "iCdG791.add_reactions([rxn30245_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# gly-L-gln\n",
    "cpd29691_e = cobra.Metabolite(\n",
    "    'cpd29691_e',\n",
    "    formula='C7H13N3O4',\n",
    "    name='gly-L-gln',\n",
    "    compartment='extracellular')\n",
    "cpd29691_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29691_e.annotation['seed.compound'] = 'cpd29691'\n",
    "iCdG791.add_boundary(cpd29691_e, type='exchange', reaction_id='EX_cpd29691_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29691_c = cobra.Metabolite(\n",
    "    'cpd29691_c',\n",
    "    formula='C7H13N3O4',\n",
    "    name='gly-L-gln',\n",
    "    compartment='cytosol')\n",
    "cpd29691_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29691_c.annotation['seed.compound'] = 'cpd29691'\n",
    "\n",
    "# Transport\n",
    "rxn30278_c = cobra.Reaction('rxn30278_c')\n",
    "rxn30278_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30278_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30278_c.bounds = (0., 1000.)\n",
    "rxn30278_c.add_metabolites({\n",
    "    cpd29691_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29691_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30278_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30278_c.annotation['seed.reaction'] = 'rxn30278'\n",
    "iCdG791.add_reactions([rxn30278_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "cpd00053_c = iCdG791.metabolites.cpd00053_c # Glutamine\n",
    "rxn30246_c = cobra.Reaction('rxn30246_c')\n",
    "rxn30246_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30246_c.name = 'Aminopeptidase'\n",
    "rxn30246_c.bounds = (0., 1000.)\n",
    "rxn30246_c.add_metabolites({\n",
    "    cpd29691_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00033_c: 1.0, # peptide 1\n",
    "    cpd00053_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30246_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30246_c.annotation['seed.reaction'] = 'rxn30246'\n",
    "iCdG791.add_reactions([rxn30246_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-met-L-ala\n",
    "cpd29700_e = cobra.Metabolite(\n",
    "    'cpd29700_e',\n",
    "    formula='C8H16N2O3S',\n",
    "    name='L-met-L-ala',\n",
    "    compartment='extracellular')\n",
    "cpd29700_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29700_e.annotation['seed.compound'] = 'cpd29700'\n",
    "iCdG791.add_boundary(cpd29700_e, type='exchange', reaction_id='EX_cpd29700_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29700_c = cobra.Metabolite(\n",
    "    'cpd29700_c',\n",
    "    formula='C8H16N2O3S',\n",
    "    name='L-met-L-ala',\n",
    "    compartment='cytosol')\n",
    "cpd29700_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29700_c.annotation['seed.compound'] = 'cpd29700'\n",
    "\n",
    "# Transport\n",
    "rxn30280_c = cobra.Reaction('rxn30280_c')\n",
    "rxn30280_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30280_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30280_c.bounds = (0., 1000.)\n",
    "rxn30280_c.add_metabolites({\n",
    "    cpd29700_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29700_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30280_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30280_c.annotation['seed.reaction'] = 'rxn30280'\n",
    "iCdG791.add_reactions([rxn30280_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00060_c = iCdG791.metabolites.cpd00060_c # Methionine\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "rxn30247_c = cobra.Reaction('rxn30247_c')\n",
    "rxn30247_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30247_c.name = 'Aminopeptidase'\n",
    "rxn30247_c.bounds = (0., 1000.)\n",
    "rxn30247_c.add_metabolites({\n",
    "    cpd29700_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00060_c: 1.0, # peptide 1\n",
    "    cpd00035_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30247_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30247_c.annotation['seed.reaction'] = 'rxn30247'\n",
    "iCdG791.add_reactions([rxn30247_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ATP synthase\n",
    "iCdG791.reactions.rxn10042_c.remove_from_model(remove_orphans=False)\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H\n",
    "\n",
    "cpd00067_p = cobra.Metabolite(\n",
    "    'cpd00067_p',\n",
    "    formula='H',\n",
    "    name='H+',\n",
    "    compartment='periplasm')\n",
    "cpd00067_p.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00067_p.annotation['seed.compound'] = 'cpd00067'\n",
    "\n",
    "cpd00982_c = iCdG791.metabolites.cpd00982_c # FADH2\n",
    "cpd00015_c = iCdG791.metabolites.cpd00015_c # FAD\n",
    "\n",
    "rxn12212_c = cobra.Reaction('rxn12212_c')\n",
    "rxn12212_c.gene_reaction_rule = '272563.120.peg.1189 and 272563.8.peg.1199 and 272563.120.peg.1185 and 272563.120.peg.1186 and 272563.120.peg.1188 and 272563.120.peg.1187'\n",
    "rxn12212_c.name = 'FAD oxidation (electron transport complex)'\n",
    "rxn12212_c.bounds = (-1000., 1000.)\n",
    "rxn12212_c.add_metabolites({\n",
    "    cpd00982_c: -1.0, # FADH2\n",
    "    cpd00015_c: 1.0, # FAD\n",
    "    cpd00067_p: 2.0 # H+\n",
    "})\n",
    "rxn12212_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn12212_c.annotation['seed.reaction'] = 'rxn12212'\n",
    "iCdG791.add_reactions([rxn12212_c])\n",
    "\n",
    "cpd00003_c = iCdG791.metabolites.cpd00003_c # NAD\n",
    "cpd00004_c = iCdG791.metabolites.cpd00004_c # NADH\n",
    "\n",
    "rxn12213_c = cobra.Reaction('rxn12213_c')\n",
    "rxn12213_c.gene_reaction_rule = '272563.120.peg.1189 and 272563.8.peg.1199 and 272563.120.peg.1185 and 272563.120.peg.1186 and 272563.120.peg.1188 and 272563.120.peg.1187'\n",
    "rxn12213_c.name = 'FAD oxidation (electron transport complex)'\n",
    "rxn12213_c.bounds = (-1000., 1000.)\n",
    "rxn12213_c.add_metabolites({\n",
    "    cpd00004_c: -1.0, # FADH2\n",
    "    cpd00003_c: 1.0, # FAD\n",
    "    cpd00067_p: 2.0 # H+\n",
    "})\n",
    "rxn12213_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn12213_c.annotation['seed.reaction'] = 'rxn12213'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1189').name = 'Ion-translocating oxidoreductase complex subunit A | rnfA'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1189').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1189').annotation['patric'] = 'fig|272563.120.peg.1189'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.8.peg.1199').name = 'Ion-translocating oxidoreductase complex subunit B | rnfB'\n",
    "iCdG791.genes.get_by_id('272563.8.peg.1199').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.8.peg.1199').annotation['patric'] = 'fig|272563.8.peg.1199'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1185').name = 'Ion-translocating oxidoreductase complex subunit C | rnfC'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1185').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1185').annotation['patric'] = 'fig|272563.120.peg.1185'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1186').name = 'Ion-translocating oxidoreductase complex subunit D | rnfD'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1186').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1186').annotation['patric'] = 'fig|272563.120.peg.1186'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1188').name = 'Ion-translocating oxidoreductase complex subunit E | rnfE'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1188').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1188').annotation['patric'] = 'fig|272563.120.peg.1188'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1187').name = 'Ion-translocating oxidoreductase complex subunit G | rnfG'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1187').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1187').annotation['patric'] = 'fig|272563.120.peg.1187'\n",
    "\n",
    "rxn10042_c = cobra.Reaction('rxn10042_c')\n",
    "rxn10042_c.gene_reaction_rule = '272563.8.peg.3633 and 272563.8.peg.3632 and 272563.8.peg.3636 and 272563.8.peg.3635 and 272563.8.peg.3634 and (272563.8.peg.3640 or 272563.8.peg.3641)'\n",
    "rxn10042_c.name = 'ATP synthase'\n",
    "rxn10042_c.bounds = (0., 1000.)\n",
    "rxn10042_c.add_metabolites({\n",
    "    cpd00008_c: -1.0, # ADP\n",
    "    cpd00009_c: -0.05, # Pi\n",
    "    cpd00067_p: -1.0, # H+\n",
    "    cpd00001_c: 1.0, # H20\n",
    "    cpd00002_c: 1.0, # ATP\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn10042_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn10042_c.annotation['seed.reaction'] = 'rxn10042'\n",
    "iCdG791.add_reactions([rxn10042_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 143\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "75.01730567597323"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# PTS systems\n",
    "\n",
    "# Salicin-specific\n",
    "rxn05647_c = iCdG791.reactions.rxn05647_c\n",
    "iCdG791.reactions.rxn05647_c.remove_from_model(remove_orphans=False)\n",
    "rxn05647_c.lower_bound = 0.\n",
    "rxn05647_c.gene_reaction_rule = '272563.8.peg.408 or 272563.8.peg.854 or 272563.8.peg.3282 or 272563.8.peg.3260 or 272563.8.peg.3241 or 272563.8.peg.3270'\n",
    "iCdG791.add_reactions([rxn05647_c])\n",
    "\n",
    "# Fructose-specific\n",
    "rxn05560_c = iCdG791.reactions.rxn05560_c\n",
    "iCdG791.reactions.rxn05560_c.remove_from_model(remove_orphans=False)\n",
    "rxn05560_c.lower_bound = 0.\n",
    "rxn05560_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG791.add_reactions([rxn05560_c])\n",
    "rxn10175_c = iCdG791.reactions.rxn10175_c\n",
    "iCdG791.reactions.rxn10175_c.remove_from_model(remove_orphans=False)\n",
    "rxn10175_c.lower_bound = 0.\n",
    "rxn10175_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG791.add_reactions([rxn10175_c])\n",
    "\n",
    "# Mannitol-specific\n",
    "rxn05617_c = iCdG791.reactions.rxn05617_c\n",
    "iCdG791.reactions.rxn05617_c.remove_from_model(remove_orphans=False)\n",
    "rxn05617_c.lower_bound = 0.\n",
    "rxn05617_c.gene_reaction_rule = '(272563.8.peg.2449 or 272563.8.peg.3804) and 272563.8.peg.2450 and 272563.8.peg.2451'\n",
    "iCdG791.add_reactions([rxn05617_c])\n",
    "\n",
    "# Maltose-specific\n",
    "rxn05607_c = iCdG791.reactions.rxn05607_c\n",
    "iCdG791.reactions.rxn05607_c.remove_from_model(remove_orphans=False)\n",
    "rxn05607_c.lower_bound = 0.\n",
    "rxn05607_c.gene_reaction_rule = '272563.8.peg.3174 or 272563.8.peg.2632 or 272563.8.peg.1403'\n",
    "iCdG791.add_reactions([rxn05607_c])\n",
    "\n",
    "# Galactitol-specific\n",
    "rxn05567_c = iCdG791.reactions.rxn05567_c\n",
    "iCdG791.reactions.rxn05567_c.remove_from_model(remove_orphans=False)\n",
    "rxn05567_c.lower_bound = 0.\n",
    "rxn05567_c.gene_reaction_rule = '(272563.8.peg.50 or 272563.8.peg.2396) and (272563.8.peg.2395 or 272563.8.peg.1137 or 272563.8.peg.2443 or 272563.8.peg.51) and (272563.8.peg.52 or 272563.8.peg.2394 or 272563.8.peg.2442 or 272563.8.peg.2444)'\n",
    "iCdG791.add_reactions([rxn05567_c])\n",
    "\n",
    "# Sorbitol-specific\n",
    "rxn10184_c = iCdG791.reactions.rxn10184_c\n",
    "iCdG791.reactions.rxn10184_c.remove_from_model(remove_orphans=False)\n",
    "rxn10184_c.lower_bound = 0.\n",
    "rxn10184_c.gene_reaction_rule = '(272563.8.peg.804 or 272563.8.peg.2532) and (272563.8.peg.801 or 272563.8.peg.2536)'\n",
    "iCdG791.add_reactions([rxn10184_c])\n",
    "\n",
    "# Mannose-specific\n",
    "rxn05610_c = iCdG791.reactions.rxn05610_c\n",
    "iCdG791.reactions.rxn05610_c.remove_from_model(remove_orphans=False)\n",
    "rxn05610_c.lower_bound = 0.\n",
    "rxn05610_c.gene_reaction_rule = '(272563.8.peg.3431 or 272563.8.peg.304 or 272563.8.peg.515 or 272563.8.peg.1128) and (272563.8.peg.307 or 272563.8.peg.516 or 272563.8.peg.3432 or 272563.8.peg.1130 or 272563.8.peg.305) and (272563.8.peg.308 or 272563.8.peg.3430 or 272563.8.peg.517 or 272563.8.peg.1131) and (272563.8.peg.518 or 272563.8.peg.1132 or 272563.8.peg.309 or 272563.8.peg.3429)'\n",
    "iCdG791.add_reactions([rxn05610_c])\n",
    "\n",
    "# Glucose-specific\n",
    "rxn05226_c = iCdG791.reactions.rxn05226_c\n",
    "iCdG791.reactions.rxn05226_c.remove_from_model(remove_orphans=False)\n",
    "rxn05226_c.lower_bound = 0.\n",
    "rxn05226_c.gene_reaction_rule = '(272563.8.peg.3202 or 272563.8.peg.3247 or 272563.8.peg.3171 or 272563.8.peg.2634) and (272563.8.peg.2791 or 272563.8.peg.2790) and (272563.8.peg.3174 or 272563.8.peg.2632 or 272563.8.peg.1403)'\n",
    "iCdG791.add_reactions([rxn05226_c])\n",
    "\n",
    "# Cellobiose-specific\n",
    "rxn05518_c = iCdG791.reactions.rxn05518_c\n",
    "iCdG791.reactions.rxn05518_c.remove_from_model(remove_orphans=False)\n",
    "rxn05518_c.lower_bound = 0.\n",
    "rxn05518_c.gene_reaction_rule = '(272563.8.peg.3225 or 272563.8.peg.3012 or 272563.8.peg.153 or 272563.8.peg.901 or 272563.8.peg.3820 or 272563.8.peg.3192 or 272563.8.peg.3825) and (272563.8.peg.3224 or 272563.8.peg.3009 or 272563.8.peg.151 or 272563.8.peg.3822 or 272563.8.peg.902) and (272563.8.peg.3823 or 272563.8.peg.3226 or 272563.8.peg.152 or 272563.8.peg.3013 or 272563.8.peg.1679)'\n",
    "iCdG791.add_reactions([rxn05518_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 71\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71.1778745591431"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final reactions needed\n",
    "cpd00320_c = cobra.Metabolite(\n",
    "    'cpd00320_c',\n",
    "    formula='C4H6NO4',\n",
    "    name='D-Aspartate',\n",
    "    compartment='cytosol')\n",
    "cpd00320_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00320_c.annotation['seed.compound'] = 'cpd00320'\n",
    "cpd00320_e = cobra.Metabolite(\n",
    "    'cpd00320_e',\n",
    "    formula='C4H6NO4',\n",
    "    name='D-Aspartate',\n",
    "    compartment='extracellular')\n",
    "cpd00320_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00320_e.annotation['seed.compound'] = 'cpd00320'\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H\n",
    "rxn05178_c = cobra.Reaction('rxn05178_c')\n",
    "rxn05178_c.name = 'D-Aspartate-ABC transport'\n",
    "rxn05178_c.bounds = (-1000., 1000.)\n",
    "rxn05178_c.add_metabolites({\n",
    "    cpd00320_e: -1.0, \n",
    "    cpd00002_c: -1.0, \n",
    "    cpd00001_c: -1.0, \n",
    "    cpd00008_c: 1.0, \n",
    "    cpd00009_c: 1.0, \n",
    "    cpd00067_c: 1.0, \n",
    "    cpd00320_c: 1.0 \n",
    "})\n",
    "rxn05178_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn05178_c.annotation['seed.reaction'] = 'rxn05178'\n",
    "iCdG791.add_reactions([rxn05178_c])\n",
    "cpd00041_c = iCdG791.metabolites.cpd00041_c\n",
    "rxn00348_c = cobra.Reaction('rxn00348_c')\n",
    "rxn00348_c.gene_reaction_rule = '272563.8.peg.2370 or 272563.8.peg.2133'\n",
    "rxn00348_c.name = 'aspartate racemase'\n",
    "rxn00348_c.bounds = (-1000., 1000.)\n",
    "rxn00348_c.add_metabolites({\n",
    "    cpd00041_c: -1.0, # L-aspartate\n",
    "    cpd00320_c: 1.0 # D-aspartate\n",
    "})\n",
    "rxn00348_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn00348_c.annotation['seed.reaction'] = 'rxn00348'\n",
    "iCdG791.add_reactions([rxn00348_c])\n",
    "iCdG791.add_boundary(cpd00320_e, type='exchange', reaction_id='EX_cpd00320_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "\n",
    "cpd02711_e = cobra.Metabolite(\n",
    "    'cpd02711_e',\n",
    "    formula='C6H8O9P',\n",
    "    name='2-keto-3-deoxygluconate',\n",
    "    compartment='extracellular')\n",
    "cpd02711_c = iCdG791.metabolites.cpd02711_c\n",
    "cpd00067_e = iCdG791.metabolites.cpd00067_e\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c\n",
    "rxn05681_c = cobra.Reaction('rxn05681_c')\n",
    "rxn05681_c.gene_reaction_rule = '272563.8.peg.3148 or 272563.8.peg.2999'\n",
    "rxn05681_c.name = '2-dehydro-3-deoxy-D-gluconate transport via proton symport, reversible'\n",
    "rxn05681_c.bounds = (-1000., 1000.)\n",
    "rxn05681_c.add_metabolites({\n",
    "    cpd02711_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd02711_c: 1.0 \n",
    "})\n",
    "rxn05681_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn05681_c.annotation['seed.reaction'] = 'rxn05681'\n",
    "iCdG791.add_reactions([rxn05681_c])\n",
    "iCdG791.add_boundary(cpd02711_e, type='exchange', reaction_id='EX_cpd02711_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "\n",
    "cpd00025_c = iCdG791.metabolites.cpd00025_c\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c\n",
    "rxn00008_c = cobra.Reaction('rxn00008_c')\n",
    "rxn00008_c.gene_reaction_rule = '272563.8.peg.1708'\n",
    "rxn00008_c.name = 'Mn(II):hydrogen-peroxide oxidoreductase'\n",
    "rxn00008_c.bounds = (0., 1000.)\n",
    "rxn00008_c.add_metabolites({\n",
    "    cpd00025_c: -1.0, # H202\n",
    "    cpd00067_c: -2.0, # H+\n",
    "    cpd00001_c: 2.0 # H20\n",
    "})\n",
    "rxn00008_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn00008_c.annotation['seed.reaction'] = 'rxn00008'\n",
    "iCdG791.add_reactions([rxn00008_c])\n",
    "\n",
    "cpd00038_c = iCdG791.metabolites.cpd00038_c # GTP\n",
    "cpd00031_c = iCdG791.metabolites.cpd00031_c # GDP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Pi\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H+\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H20\n",
    "cpd10515_c = iCdG791.metabolites.cpd10515_c # Fe\n",
    "cpd10515_e = iCdG791.metabolites.cpd10515_e # Fe\n",
    "rxn34423_c = cobra.Reaction('rxn34423_c')\n",
    "rxn34423_c.gene_reaction_rule = '272563.8.peg.3427 or 272563.8.peg.1551 or 272563.8.peg.1593'\n",
    "rxn34423_c.name = 'ferrous iron uptake (GTP-dependent), plasma membrane'\n",
    "rxn34423_c.bounds = (0., 1000.)\n",
    "rxn34423_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00038_c: -1.0,\n",
    "    cpd10515_e: -1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00031_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd10515_c: 1.0\n",
    "})\n",
    "rxn34423_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn34423_c.annotation['seed.reaction'] = 'rxn34423'\n",
    "iCdG791.add_reactions([rxn34423_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Other miscellaneous genes\n",
    "iCdG791.reactions.rxn05172_c.gene_reaction_rule = '(272563.8.peg.1554 or 272563.8.peg.3135) and (272563.8.peg.1556 or 272563.8.peg.2482) and (272563.8.peg.3134 or 272563.8.peg.1555 or 272563.8.peg.1052)'\n",
    "iCdG791.reactions.rxn05298_c.gene_reaction_rule = '272563.8.peg.2286 or 272563.8.peg.2818 or 272563.8.peg.1830'\n",
    "iCdG791.reactions.rxn00337_c.gene_reaction_rule = '272563.8.peg.1389 or 272563.8.peg.2161'\n",
    "iCdG791.reactions.rxn05209_c.gene_reaction_rule = '272563.8.peg.2793 or 272563.8.peg.1732 or 272563.8.peg.473 or 272563.8.peg.2861'\n",
    "iCdG791.reactions.rxn01522_c.gene_reaction_rule = '(272563.8.peg.2194 or 272563.8.peg.3328) and (272563.8.peg.2181 or 272563.8.peg.2187) and (272563.8.peg.2188 or 272563.8.peg.2195)'\n",
    "iCdG791.reactions.rxn00414_c.gene_reaction_rule = '(272563.8.peg.3759 or 272563.8.peg.3761) and (272563.8.peg.3760 or 272563.8.peg.3762)'\n",
    "iCdG791.reactions.rxn20606_c.gene_reaction_rule = '272563.120.peg.3426 and (272563.8.peg.3394 or 272563.8.peg.3392) and (272563.8.peg.3396 or 272563.8.peg.3393)'\n",
    "iCdG791.reactions.rxn00785_c.gene_reaction_rule = '(272563.8.peg.3625 or 272563.8.peg.2439) and (272563.8.peg.2438 or 272563.8.peg.3624)'\n",
    "iCdG791.reactions.rxn01200_c.gene_reaction_rule = '(272563.8.peg.3625 or 272563.8.peg.2439) and (272563.8.peg.2438 or 272563.8.peg.3624)'\n",
    "iCdG791.reactions.rxn01301_c.gene_reaction_rule = '272563.8.peg.1224 or 272563.8.peg.2531 or 272563.8.peg.1655'\n",
    "iCdG791.reactions.rxn00908_c.gene_reaction_rule = '272563.8.peg.240 and 272563.8.peg.766'\n",
    "iCdG791.reactions.rxn01225_c.gene_reaction_rule = '272563.8.peg.3332 or 272563.8.peg.1740'\n",
    "iCdG791.reactions.rxn03136_c.gene_reaction_rule = '272563.8.peg.513 or 272563.8.peg.237'\n",
    "iCdG791.reactions.rxn13022_c.gene_reaction_rule = '272563.8.peg.1041 and 272563.8.peg.1042'\n",
    "iCdG791.reactions.rxn03886_c.gene_reaction_rule = '272563.8.peg.805'\n",
    "iCdG791.reactions.rxn00178_c.gene_reaction_rule = '272563.8.peg.1111 or 272563.8.peg.2800'\n",
    "iCdG791.reactions.rxn08817_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08849_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08842_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08819_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08818_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08840_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08848_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08851_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08846_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08820_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08847_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08850_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08821_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08843_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08839_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08844_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08822_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08845_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08841_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08838_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08823_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn00693_c.gene_reaction_rule = '272563.8.peg.3768 or 272563.8.peg.1815'\n",
    "iCdG791.reactions.rxn00143_c.gene_reaction_rule = '272563.8.peg.2697 or 272563.8.peg.2734'\n",
    "iCdG791.reactions.rxn01519_c.gene_reaction_rule = '272563.8.peg.2521'\n",
    "iCdG791.reactions.rxn00493_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn02320_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn00527_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn01270_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn02320_c.gene_reaction_rule = '272563.8.peg.1625'\n",
    "iCdG791.reactions.rxn03167_c.gene_reaction_rule = '272563.8.peg.1521'\n",
    "iCdG791.reactions.rxn00549_c.gene_reaction_rule = '272563.8.peg.1250'\n",
    "iCdG791.reactions.rxn08040_c.gene_reaction_rule = '272563.8.peg.818'\n",
    "iCdG791.reactions.rxn05555_c.gene_reaction_rule = '272563.8.peg.1726 and (272563.8.peg.1725 or 272563.8.peg.1724) and 272563.8.peg.1727'\n",
    "iCdG791.reactions.rxn05183_c.gene_reaction_rule = '(272563.8.peg.1563 or 272563.8.peg.1730) and 272563.8.peg.1562'\n",
    "iCdG791.reactions.rxn00146_c.gene_reaction_rule = '272563.120.peg.414 or 272563.8.peg.414'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Glucose ABC transport\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H+\n",
    "cpd00027_e = iCdG791.metabolites.cpd00027_e # Glucose\n",
    "cpd00079_c = iCdG791.metabolites.cpd00079_c # Glucose 6-p\n",
    "\n",
    "# Transport\n",
    "rxn05147_c = cobra.Reaction('rxn05147_c')\n",
    "rxn05147_c.gene_reaction_rule = '(272563.8.peg.3202 or 272563.8.peg.3247 or 272563.8.peg.3171 or 272563.8.peg.2634) and (272563.8.peg.2791 or 272563.8.peg.2790) and (272563.8.peg.3174 or 272563.8.peg.2632 or 272563.8.peg.1403)'\n",
    "rxn05147_c.name = 'D-Glucose-ABC transport'\n",
    "rxn05147_c.bounds = (0., 1000.)\n",
    "rxn05147_c.add_metabolites({\n",
    "    cpd00027_e: -1.0, # g\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00079_c: 1.0, # g 6-p\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn05147_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn05147_c.annotation['seed.reaction'] = 'rxn05147'\n",
    "iCdG791.add_reactions([rxn05147_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 0\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "97.22639015406979"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename GENRE\n",
    "iCdG791.id = 'iCdG' + str(len(iCdG791.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add additional annotation information\n",
    "\n",
    "# BiGG\n",
    "BiGG_rxn_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/rxns_BiGG_ids.txt', 'r') as bigg_rxns:\n",
    "    for line in bigg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        bigg = line[1].strip().split(',')\n",
    "        if len(bigg) == 1: bigg = bigg[0]\n",
    "        BiGG_rxn_ids[modelseed] = bigg\n",
    "BiGG_cpd_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/cpds_BiGG_ids.txt', 'r') as bigg_rxns:\n",
    "    for line in bigg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        bigg = line[1].strip().split(',')\n",
    "        if len(bigg) == 1: bigg = bigg[0]\n",
    "        BiGG_cpd_ids[modelseed] = bigg\n",
    "\n",
    "# KEGG\n",
    "KEGG_rxn_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/rxns_KEGG_ids.txt', 'r') as kegg_rxns:\n",
    "    for line in kegg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        kegg = line[1].strip().split(',')\n",
    "        if len(kegg) == 1: kegg = kegg[0]\n",
    "        KEGG_rxn_ids[modelseed] = kegg\n",
    "KEGG_cpd_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/cpds_KEGG_ids.txt', 'r') as kegg_rxns:\n",
    "    for line in kegg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        kegg = line[1].strip().split(',')\n",
    "        if len(kegg) == 1: kegg = kegg[0]\n",
    "        KEGG_cpd_ids[modelseed] = kegg\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in iCdG791.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        iCdG791.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        iCdG791.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in iCdG791.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        iCdG791.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        iCdG791.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removed blocked, no-gene reactions\n",
    "# Lysophospholipase L1 (periplasm)\n",
    "lysophospholipase = ['rxn08807_c','rxn08796_c','rxn08802_c','rxn08814_c','rxn08801_c','rxn08798_c','rxn08804_c','rxn08812_c','rxn08803_c','rxn08813_c','rxn08806_c','rxn08810_c','rxn08805_c','rxn08811_c','rxn08800_c','rxn08816_c']\n",
    "for rxn in lysophospholipase: iCdG791.reactions.get_by_id(rxn).remove_from_model(remove_orphans=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG706</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f349aff0850</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1182</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1205</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG706 at 0x7f349aff0850>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save model\n",
    "cobra.io.write_sbml_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG706.sbml')\n",
    "cobra.io.save_json_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG706.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in again for gap-filling\n",
    "import cobra\n",
    "iCdG706 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG706.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Custom gap-filler\n",
    "import copy\n",
    "import cobra\n",
    "import symengine\n",
    "\n",
    "# pFBA gapfiller\n",
    "def pfba_gapfill(model, universal, obj='biomass', demand='none', extracellular='extracellular'):\n",
    "    '''\n",
    "    Parameters\n",
    "    ----------\n",
    "    model_file : model\n",
    "        Model to be gapfilled\n",
    "    universal_file : model\n",
    "        Reaction bag reference\n",
    "    obj : str\n",
    "        Name of objective reaction\n",
    "    demand : str\n",
    "        Demand metabolite ID, circumvents objective\n",
    "    extracellular : str\n",
    "        Label for extracellular compartment of model\n",
    "    '''\n",
    "\n",
    "    # Save some basic network info for downstream membership testing\n",
    "    target_rxns = set([str(x.id) for x in model.reactions])\n",
    "    target_cpds = set([str(y.id) for y in model.metabolites])\n",
    "    ref_rxns = set([str(z.id) for z in universal.reactions])\n",
    "    shared_rxns = ref_rxns.intersection(target_rxns)\n",
    "        \n",
    "    # Remove overlapping reactions from universal bag, add model reactions to universal bag\n",
    "    temp_universal = copy.deepcopy(universal)\n",
    "    for rxn in shared_rxns: temp_universal.reactions.get_by_id(rxn).remove_from_model()\n",
    "    temp_universal.add_reactions(list(copy.deepcopy(model.reactions)))\n",
    "    \n",
    "    # Set objective\n",
    "    if demand != 'none':\n",
    "        cpd = temp_universal.metabolites.get_by_id(demand)\n",
    "        demand = cobra.Reaction('demand')\n",
    "        demand.bounds = (1., 1000.)\n",
    "        demand.add_metabolites({cpd: -1.0})\n",
    "        temp_universal.add_reactions([demand])\n",
    "        temp_universal.objective = demand.id\n",
    "    else:\n",
    "        temp_universal.objective = obj\n",
    "    obj_constraint = temp_universal.problem.Constraint(temp_universal.objective.expression, lb=10., ub=1000.)\n",
    "    temp_universal.add_cons_vars([obj_constraint])\n",
    "    temp_universal.solver.update()\n",
    "    \n",
    "    # Test to make sure objective flux is acheived with addition of reaction bag\n",
    "    solution = temp_universal.optimize()\n",
    "    if solution.objective_value <= 1e-6: \n",
    "        raise ValueError('ERROR: No objective flux possible with current reaction set.')    \n",
    "    \n",
    "    ## Set up pFBA objective\n",
    "    pfba_expr = symengine.RealDouble(0)\n",
    "    for rxn in temp_universal.reactions:\n",
    "        if rxn.id in target_rxns:\n",
    "            pfba_expr += 0. * rxn.forward_variable\n",
    "            pfba_expr += 0. * rxn.reverse_variable\n",
    "        else:\n",
    "            pfba_expr += 1. * rxn.forward_variable\n",
    "            pfba_expr += 1. * rxn.reverse_variable\n",
    "    temp_universal.objective = temp_universal.problem.Objective(pfba_expr, direction='min', sloppy=True)\n",
    "    temp_universal.solver.update()\n",
    "\n",
    "    # Run FBA and save solution\n",
    "    solution = temp_universal.optimize()\n",
    "    active_rxns = set([rxn.id for rxn in temp_universal.reactions if abs(solution.fluxes[rxn.id]) > 1e-6])\n",
    "    \n",
    "    # Screen new reaction IDs\n",
    "    new_rxns = active_rxns.difference(target_rxns)\n",
    "    new_rxns = new_rxns.difference(set(['demand']))\n",
    "    print('New reactions: ' + str(len(new_rxns)))\n",
    "    print('New objective value: ' + str(solution.objective_value))\n",
    "    \n",
    "    return new_rxns\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "universal_GramPos.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13783_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13784_c.remove_from_model(remove_orphans=True)\n",
    "#universal_GramPos.reactions.rxn11920_c.name = 'Poly-beta-hydroxybutyrate (PHB) synthease'\n",
    "universal_GramPos.reactions.rxn12188_c.name = '2-deoxyglucose-6-phosphatase'\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e', # Pyridoxine\n",
    "       'EX_cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "# Theriot et al. (2013). Nature Communications.\n",
    "ncmm = ['EX_cpd00001_e', # water\n",
    "        'EX_cpd00104_e', # Biotin\n",
    "        'EX_cpd00644_e', # Pantothenate\n",
    "        'EX_cpd00263_e', # Pyridoxine\n",
    "        'EX_cpd00149_e', # Cobalt\n",
    "        'EX_cpd00099_e', # Chloride\n",
    "        'EX_cpd10515_e', # Iron\n",
    "        'EX_cpd00030_e', # Manganese\n",
    "        'EX_cpd00254_e', # Magnesium\n",
    "        'EX_cpd00063_e', # Calcium\n",
    "        'EX_cpd00205_e', # Potassium\n",
    "        'EX_cpd00009_e', # Phosphate\n",
    "        'EX_cpd00971_e', # Sodium\n",
    "        'EX_cpd00242_e', # Carbonate\n",
    "        'EX_cpd00322_e', # L-Isoleucine\n",
    "        'EX_cpd00129_e', # L-Proline\n",
    "        'EX_cpd00156_e', # L-Valine\n",
    "        'EX_cpd00107_e', # L-Leucine\n",
    "        'EX_cpd00084_e', # L-Cysteine \n",
    "        'EX_cpd00065_e', # L-Tryptophan\n",
    "        'EX_cpd00060_e', # L-Methionine\n",
    "        'EX_cpd00119_e', # L-Histidine\n",
    "        'EX_cpd00033_e', # Glycine\n",
    "        'EX_cpd00051_e', # L-Arginine\n",
    "        'EX_cpd00161_e'] # L-Threonine\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New reactions: 1\n",
      "New objective value: 0.0025000000001482686\n"
     ]
    }
   ],
   "source": [
    "# Change media and gapfill - MDM\n",
    "for rxn in iCdG706.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Pre-gapfill objective value: ' + str(test))\n",
    "\n",
    "gapfilled_rxns = pfba_gapfill(iCdG706, universal_GramPos) # Biomass alone\n",
    "gapfilled_rxns1 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17043_c') # RNA polymer\n",
    "gapfilled_rxns2 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17041_c') # Protein polymer\n",
    "gapfilled_rxns3 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd12894_c') # Teichoic acid\n",
    "gapfilled_rxns4 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd16661_c') # Peptidoglycan polymer\n",
    "gapfilled_rxns5 = pfba_gapfill(iCdG706, universal_GramPos, demand='cellwall_c') # Cell Wall polymer\n",
    "gapfilled_rxns6 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd11852_c') # Lipid Pool\n",
    "gapfilled_rxns7 = pfba_gapfill(iCdG706, universal_GramPos, demand='cofactor_c') # Cofactor Pool\n",
    "gapfilled_rxns8 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17042_c') # DNA polymer\n",
    "\n",
    "mdm_gapfilled_rxns = gapfilled_rxns1.union(gapfilled_rxns2).union(gapfilled_rxns3).union(gapfilled_rxns4).union(gapfilled_rxns5).union(gapfilled_rxns6).union(gapfilled_rxns7).union(gapfilled_rxns8).union(gapfilled_rxns)\n",
    "print('Total reactions to be added: ' + str(len(mdm_gapfilled_rxns)))\n",
    "\n",
    "rxns_to_be_added = []\n",
    "for rxn in mdm_gapfilled_rxns: \n",
    "    rxns_to_be_added.append(copy.deepcopy(universal_GramPos.reactions.get_by_id(rxn)))\n",
    "iCdG706.add_reactions(rxns_to_be_added)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Post-gapfill objective value: ' + str(test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pre-gapfill objective value: 12.559386622123775\n",
      "New reactions: 0\n",
      "New objective value: 3.9164191226586654e-14\n",
      "New reactions: 0\n",
      "New objective value: 10.000000000000194\n",
      "New reactions: 0\n",
      "New objective value: 9.999999999999883\n",
      "New reactions: 0\n",
      "New objective value: 10.000000000000401\n",
      "New reactions: 2\n",
      "New objective value: 61.93957115009697\n",
      "New reactions: 4\n",
      "New objective value: 385.581871345032\n",
      "New reactions: 0\n",
      "New objective value: 10.0\n",
      "New reactions: 0\n",
      "New objective value: 10.0\n",
      "New reactions: 0\n",
      "New objective value: 10.000000000000016\n",
      "Total reactions to be added: 5\n",
      "Post-gapfill objective value: 44.875570765519235\n"
     ]
    }
   ],
   "source": [
    "# Change media and gapfill - NCMM\n",
    "for rxn in iCdG706.boundary:\n",
    "    if not rxn.id in ncmm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Pre-gapfill objective value: ' + str(test))\n",
    "\n",
    "gapfilled_rxns = pfba_gapfill(iCdG706, universal_GramPos) # Biomass alone\n",
    "gapfilled_rxns1 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17043_c') # RNA polymer\n",
    "gapfilled_rxns2 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17041_c') # Protein polymer\n",
    "gapfilled_rxns3 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd12894_c') # Teichoic acid\n",
    "gapfilled_rxns4 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd16661_c') # Peptidoglycan polymer\n",
    "gapfilled_rxns5 = pfba_gapfill(iCdG706, universal_GramPos, demand='cellwall_c') # Cell Wall polymer\n",
    "gapfilled_rxns6 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd11852_c') # Lipid Pool\n",
    "gapfilled_rxns7 = pfba_gapfill(iCdG706, universal_GramPos, demand='cofactor_c') # Cofactor Pool\n",
    "gapfilled_rxns8 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17042_c') # DNA polymer\n",
    "\n",
    "ncmm_gapfilled_rxns = gapfilled_rxns1.union(gapfilled_rxns2).union(gapfilled_rxns3).union(gapfilled_rxns4).union(gapfilled_rxns5).union(gapfilled_rxns6).union(gapfilled_rxns7).union(gapfilled_rxns8).union(gapfilled_rxns)\n",
    "print('Total reactions to be added: ' + str(len(ncmm_gapfilled_rxns)))\n",
    "\n",
    "rxns_to_be_added = []\n",
    "for rxn in ncmm_gapfilled_rxns: \n",
    "    rxns_to_be_added.append(copy.deepcopy(universal_GramPos.reactions.get_by_id(rxn)))\n",
    "iCdG706.add_reactions(rxns_to_be_added)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Post-gapfill objective value: ' + str(test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct names and possible gene annotations\n",
    "all_gapfilled_rxns = ncmm_gapfilled_rxns.union(mdm_gapfilled_rxns)\n",
    "\n",
    "iCdG706.reactions.rxn09655_c.name = 'Trehalose transport via proton symport'\n",
    "iCdG706.reactions.rxn09658_c.name = 'Sucrose transport via proton symport'\n",
    "iCdG706.reactions.rxn10183_c.name = 'N-acetyl-D-mannosamine transport via PTS'\n",
    "\n",
    "iCdG706.reactions.EX_cpd00208_e.name = 'Lactose exchange'\n",
    "\n",
    "gene = cobra.Gene('272563.120.peg.1564')\n",
    "gene.annotation['patric'] = 'fig|272563.120.peg.1564'\n",
    "gene.annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG706.reactions.rxn00504_c.gene_reaction_rule = '272563.120.peg.1564'\n",
    "iCdG706.genes.get_by_id('272563.120.peg.1564').name = 'Pyrroline-5-carboxylate reductase | proC'\n",
    "\n",
    "iCdG706.reactions.rxn09655_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG706.reactions.rxn09658_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG706.reactions.rxn10183_c.gene_reaction_rule = '(272563.8.peg.3202 or 272563.8.peg.3247 or 272563.8.peg.3171 or 272563.8.peg.2634) and (272563.8.peg.2790 or 272563.8.peg.2791)'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename GENRE\n",
    "iCdG706.id = 'iCdG' + str(len(iCdG706.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "for rxn in iCdG706.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG707</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f40ef778210</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1184</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1214</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG707 at 0x7f40ef778210>"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG706"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in iCdG706.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        iCdG706.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        iCdG706.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in iCdG706.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        iCdG706.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        iCdG706.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Label tRNA reactions as Biomass\n",
    "iCdG706.reactions.rxn06300_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06432_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06434_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06936_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06435_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06437_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06436_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06438_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06439_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06440_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06441_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06442_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06443_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06444_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06445_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06446_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06447_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06448_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06280_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06449_c.annotation['sbo'] = 'SBO:0000629'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save model\n",
    "cobra.io.write_sbml_model(iCdG706, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG707.sbml')\n",
    "cobra.io.save_json_model(iCdG706, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG707.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "iCdG707 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG707.sbml')\n",
    "for rxn in iCdG707.boundary: rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import parsimonious\n",
    "\n",
    "# Function to calculate doubling time from objective value\n",
    "def doublingTime(model):\n",
    "    with model as m:\n",
    "        if m.slim_optimize(error_value=0.) < 1e-6:\n",
    "            print('GENRE has no objective flux')\n",
    "        else:\n",
    "            growth = (1. / float(m.slim_optimize())) * 3600.\n",
    "            print(str(round(growth, 2)) + ' minutes doubling time')\n",
    "\n",
    "\n",
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "\n",
    "    #print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    fraction = (float(len(blocked)) / float(len(model.reactions))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions are blocked')\n",
    "    \n",
    "    return blocked\n",
    "\n",
    "\n",
    "# Identify potentially gapfilled reactions, checks against pFBA solution\n",
    "def missingGPR(model, exclude=['biomass']):\n",
    "    \n",
    "    noGene = []        \n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.genes)) == 0:\n",
    "            if rxn.id not in exclude:\n",
    "                if len(list(rxn.reactants)) == 0 or len(list(rxn.products)) == 0:\n",
    "                    exclude.append(rxn.id)\n",
    "                    continue\n",
    "                else:\n",
    "                    noGene.append(rxn.id)\n",
    "    \n",
    "    solution = parsimonious.pfba(model)\n",
    "    active_rxns = set([rxn.id for rxn in model.reactions if abs(solution.fluxes[rxn.id]) > 1e-5])\n",
    "    active_rxns = active_rxns.difference(set(exclude))\n",
    "    noGene_active = set(noGene).intersection(active_rxns)\n",
    "\n",
    "    fraction = float(len(model.reactions)) - float(len(exclude))\n",
    "    fraction = (float(len(noGene)) / fraction) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions without GPRs')\n",
    "    \n",
    "    fraction = (float(len(noGene_active)) / float(len(active_rxns))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% of reactions used in pFBA solution have no GPR')\n",
    "    \n",
    "    return noGene_active\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(model, cytosol='cytosol'):\n",
    "\n",
    "    free = []\n",
    "    with model as m:\n",
    "    \n",
    "        # Close all exchanges\n",
    "        for rxn in m.boundary: m.reactions.get_by_id(rxn.id).lower_bound = 0.\n",
    "    \n",
    "        # Create demand for each reaction and optimize individually\n",
    "        reset_rxn = m.reactions[0].id\n",
    "        for cpd in m.metabolites: \n",
    "            if cpd.compartment == cytosol:\n",
    "                demand = cobra.Reaction('demand')\n",
    "                demand.bounds = (0., 1000.)\n",
    "                demand.add_metabolites({cpd: -1.0})\n",
    "                m.add_reactions([demand])\n",
    "                m.objective = demand\n",
    "                obj_val = m.slim_optimize()\n",
    "                if obj_val > 1e-8: free.append(cpd.id)\n",
    "                m.objective = reset_rxn\n",
    "                m.remove_reactions([demand])\n",
    "    \n",
    "    fraction = (float(len(free)) / float(len(model.metabolites))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(model, exclude=['biomass']):\n",
    "    \n",
    "    with model as m:\n",
    "\n",
    "        elements = set()\n",
    "        for cpd in m.metabolites:\n",
    "            try:\n",
    "                elements |= set(cpd.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        massImbal = []\n",
    "        failed = 0\n",
    "        if len(elements) == 0:\n",
    "            print('No elemental data associated with metabolites!')\n",
    "            failed = 1\n",
    "        else:\n",
    "            for rxn in m.reactions:\n",
    "                if rxn in m.boundary or rxn.id in exclude: \n",
    "                    continue\n",
    "                elif rxn.annotation['sbo'] == 'SBO:0000629':\n",
    "                    continue\n",
    "\n",
    "                try:\n",
    "                    test = rxn.check_mass_balance()\n",
    "                except ValueError:\n",
    "                    continue\n",
    "\n",
    "                if len(list(test)) > 0:\n",
    "                    if len(set(test.keys()).intersection(elements)) > 0: massImbal.append(rxn.id)\n",
    "                        \n",
    "    if failed != 1:\n",
    "        fraction = (float(len(massImbal)) / float(len(model.reactions))) * 100.\n",
    "        fraction = round(fraction, 2)\n",
    "        print(str(fraction) + '% reactions are mass imbalanced')\n",
    "        \n",
    "    return massImbal\n",
    "\n",
    "\n",
    "def basicCheck(model):\n",
    "    \n",
    "    # Determination\n",
    "    if len(model.reactions) < len(model.metabolites): \n",
    "        print('GENRE is overdetermined')\n",
    "    elif len(model.reactions) > len(model.metabolites):\n",
    "        print('GENRE is underdetermined')\n",
    "    else:\n",
    "        pass\n",
    "    \n",
    "    # Compartments\n",
    "    print('GENRE has ' + str(len(model.compartments.keys())) + ' compartment(s)')\n",
    "    \n",
    "    # Genes\n",
    "    if len(model.genes) == 0: \n",
    "        print('GENRE has no gene data')\n",
    "    else:\n",
    "        print('GENRE has ' + str(len(model.genes)) + ' genes')\n",
    "          \n",
    "    # Growth\n",
    "    doublingTime(model)\n",
    "\n",
    "# Quicker way to read in models\n",
    "import pickle\n",
    "def read_model(fileName, obj='none'):\n",
    "    \n",
    "    fileType = fileName.split('.')[-1]\n",
    "    \n",
    "    if fileType == 'sbml' or fileType == 'xml':\n",
    "        model = cobra.io.read_sbml_model(fileName)\n",
    "    elif fileType == 'json':\n",
    "        model = cobra.io.load_json_model(fileName)\n",
    "    elif fileType == 'yaml':\n",
    "        model = cobra.io.load_yaml_model(fileName)\n",
    "    elif fileType == 'mat':\n",
    "        model = cobra.io.load_matlab_model(fileName)\n",
    "    elif fileType == 'pkl':\n",
    "        model = pickle.load(open(fileName, 'rb'))\n",
    "    else:\n",
    "        raise TypeError('Unrecognized file extension')\n",
    "    \n",
    "    if obj != 'none': model.objective = obj\n",
    "    for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "    return model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 707 genes\n",
      "22.3 minutes doubling time\n",
      "37.31% reactions are blocked\n",
      "0.49% reactions are mass imbalanced\n",
      "13.57% reactions without GPRs\n",
      "24.82% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "basicCheck(iCdG707)\n",
    "iCdG707_noGPRblocked = blockedReactions(iCdG707)\n",
    "iCdG707_free = checkFreeMass(iCdG707)\n",
    "iCdG707_massImbal = checkBalance(iCdG707, exclude=['biomass'])\n",
    "iCdG707_nogene = missingGPR(iCdG707, exclude=['biomass'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct mass imbalances\n",
    "\n",
    "cpd00001_c = iCdG707.metabolites.cpd00001_c # H2O\n",
    "cpd29694_c = iCdG707.metabolites.cpd29694_c # L-ala-gly\n",
    "cpd00033_c = iCdG707.metabolites.cpd00033_c # Glycine\n",
    "cpd00035_c = iCdG707.metabolites.cpd00035_c # L-Alanine\n",
    "cpd00067_c = iCdG707.metabolites.cpd00067_c # H+\n",
    "\n",
    "iCdG707.reactions.rxn30238_c.remove_from_model()\n",
    "\n",
    "rxn30238_c = cobra.Reaction('rxn30238_c') \n",
    "rxn30238_c.name = 'Aminopeptidase (ala-gly)'\n",
    "rxn30238_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30238_c.lower_bound = 0.\n",
    "rxn30238_c.upper_bound = 1000.\n",
    "rxn30238_c.add_metabolites({\n",
    "    cpd00001_c: -2.0,\n",
    "    cpd29694_c: -1.0,\n",
    "    cpd00035_c: 1.0,\n",
    "    cpd00033_c: 1.0,\n",
    "    cpd00067_c: 5.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn30238_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00982_c = iCdG707.metabolites.cpd00982_c # FADH2\n",
    "cpd00015_c = iCdG707.metabolites.cpd00015_c # FAD\n",
    "cpd00067_p = iCdG707.metabolites.cpd00067_p # H+\n",
    "\n",
    "iCdG707.reactions.rxn12212_c.remove_from_model()\n",
    "\n",
    "rxn12212_c = cobra.Reaction('rxn12212_c') \n",
    "rxn12212_c.name = 'FAD oxidation (electron transport complex)'\n",
    "rxn12212_c.gene_reaction_rule = '272563.120.peg.1189 and 272563.8.peg.1199 and 272563.120.peg.1185 and 272563.120.peg.1186 and 272563.120.peg.1188 and 272563.120.peg.1187'\n",
    "rxn12212_c.lower_bound = -1000.\n",
    "rxn12212_c.upper_bound = 1000.\n",
    "rxn12212_c.add_metabolites({\n",
    "    cpd00982_c: -1.0,\n",
    "    cpd00015_c: 1.0,\n",
    "    cpd00067_p: 3.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn12212_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00106_c = iCdG707.metabolites.cpd00106_c # Fumarate\n",
    "cpd00791_c = iCdG707.metabolites.cpd00791_c # ProtoporphyrinogenIX\n",
    "cpd00036_c = iCdG707.metabolites.cpd00036_c # Succinate\n",
    "cpd01476_c = iCdG707.metabolites.cpd01476_c # Protoporphyrin\n",
    "\n",
    "iCdG707.reactions.rxn09180_c.remove_from_model()\n",
    "\n",
    "rxn09180_c = cobra.Reaction('rxn09180_c') \n",
    "rxn09180_c.name = 'protoporphyrinogen oxidase (anaerobic)'\n",
    "rxn09180_c.lower_bound = -1000.\n",
    "rxn09180_c.upper_bound = 1000.\n",
    "rxn09180_c.add_metabolites({\n",
    "    cpd00067_p: -6.0,\n",
    "    cpd00106_c: -3.0,\n",
    "    cpd00791_c: -1.0,\n",
    "    cpd00036_c: 3.0,\n",
    "    cpd01476_c: 1.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn09180_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00008_c = iCdG707.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG707.metabolites.cpd00009_c # Pi\n",
    "cpd00002_c = iCdG707.metabolites.cpd00002_c # ATP\n",
    "\n",
    "iCdG707.reactions.rxn10042_c.remove_from_model()\n",
    "\n",
    "rxn10042_c = cobra.Reaction('rxn10042_c') \n",
    "rxn10042_c.name = 'ATP synthase'\n",
    "rxn10042_c.gene_reaction_rule = '272563.8.peg.3633 and 272563.8.peg.3632 and 272563.8.peg.3636 and 272563.8.peg.3635 and 272563.8.peg.3634 and ( 272563.8.peg.3640 or 272563.8.peg.3641 )'\n",
    "rxn10042_c.lower_bound = 0.\n",
    "rxn10042_c.upper_bound = 1000.\n",
    "rxn10042_c.add_metabolites({\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00067_p: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd00002_c: 1.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn10042_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00027_e = iCdG707.metabolites.cpd00027_e # Glucose\n",
    "cpd00079_c = iCdG707.metabolites.cpd00079_c # D-glucose-6-phosphate\n",
    "\n",
    "iCdG707.reactions.rxn05147_c.remove_from_model()\n",
    "\n",
    "rxn05147_c = cobra.Reaction('rxn05147_c') \n",
    "rxn05147_c.name = 'D-Glucose-ABC transport'\n",
    "rxn12212_c.gene_reaction_rule = '272563.8.peg.3633 and 272563.8.peg.3632 and 272563.8.peg.3636 and 272563.8.peg.3635 and 272563.8.peg.3634 and ( 272563.8.peg.3640 or 272563.8.peg.3641 )'\n",
    "rxn05147_c.lower_bound = 0.\n",
    "rxn05147_c.upper_bound = 1000.\n",
    "rxn05147_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00027_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00079_c: 1.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn05147_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "iCdG707.metabolites.cpd29698_c.formula = 'C9H18N2O3'\n",
    "iCdG707.metabolites.cpd29698_e.formula = 'C9H18N2O3'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Unblock Lactose catabolism\n",
    "\n",
    "cpd00208_c = iCdG707.metabolites.cpd00208_c # Lactose\n",
    "cpd00001_c = iCdG707.metabolites.cpd00001_c # H2O\n",
    "cpd00027_c = iCdG707.metabolites.cpd00027_c # Glucose\n",
    "\n",
    "cpd00108_c = cobra.Metabolite(\n",
    "    'cpd00108_c',\n",
    "    formula='C6H12O6',\n",
    "    name='Galactose',\n",
    "    compartment='cytosol')\n",
    "cpd00108_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00108_c.annotation['seed.compound'] = 'cpd00108'\n",
    "cpd00108_c.annotation['bigg.metabolite'] = 'gal'\n",
    "cpd00108_c.annotation['kegg.compound'] = 'C00124'\n",
    "\n",
    "rxn00816_c = cobra.Reaction('rxn00816_c')\n",
    "rxn00816_c.name = 'Beta-galactosidase'\n",
    "rxn00816_c.lower_bound = 0.\n",
    "rxn00816_c.upper_bound = 1000.\n",
    "rxn00816_c.add_metabolites({\n",
    "    cpd00208_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00027_c: 1.0,\n",
    "    cpd00108_c: 1.0\n",
    "})\n",
    "rxn00816_c.annotation['sbo'] = ''\n",
    "rxn00816_c.annotation['seed.reaction'] = 'rxn00816'\n",
    "rxn00816_c.annotation['kegg.reaction'] = 'R01100'\n",
    "rxn00816_c.annotation['bigg.reaction'] = 'LACZ'\n",
    "iCdG707.add_reactions([rxn00816_c])\n",
    "\n",
    "cpd00002_c = iCdG707.metabolites.cpd00002_c # ATP\n",
    "cpd00008_c = iCdG707.metabolites.cpd00008_c # ADP\n",
    "cpd00067_c = iCdG707.metabolites.cpd00067_c # H\n",
    "cpd00348_c = iCdG707.metabolites.cpd00348_c # galactose-1-phosphate \n",
    "\n",
    "rxn00808_c = cobra.Reaction('rxn00808_c')\n",
    "rxn00808_c.name = 'ATP:D-galactose 1-phosphotransferase'\n",
    "rxn00808_c.lower_bound = 0.\n",
    "rxn00808_c.upper_bound = 1000.\n",
    "rxn00808_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00108_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00348_c: 1.0\n",
    "})\n",
    "rxn00808_c.annotation['sbo'] = ''\n",
    "rxn00808_c.annotation['seed.reaction'] = 'rxn00808'\n",
    "rxn00808_c.annotation['kegg.reaction'] = 'R01092'\n",
    "rxn00808_c.annotation['bigg.reaction'] = 'GAL1'\n",
    "iCdG707.add_reactions([rxn00808_c])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Unblock Salicin catabolism\n",
    "\n",
    "cpd01553_c = iCdG707.metabolites.cpd01553_c # Saligenin\n",
    "\n",
    "cpd01553_e = cobra.Metabolite(\n",
    "    'cpd01553_e',\n",
    "    formula='C7H8O2',\n",
    "    name='Saligenin',\n",
    "    compartment='extracellular')\n",
    "cpd01553_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd01553_e.annotation['seed.compound'] = 'cpd01553'\n",
    "cpd01553_e.annotation['bigg.metabolite'] = '2hymeph'\n",
    "cpd01553_e.annotation['kegg.compound'] = 'C02323'\n",
    "iCdG707.add_boundary(cpd01553_e, type='exchange', reaction_id='EX_cpd01553_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG707.reactions.EX_cpd01553_e.name = 'Saligenin exchange'\n",
    "iCdG707.reactions.EX_cpd01553_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00067_c = iCdG707.metabolites.cpd00067_c # H\n",
    "cpd00067_e = iCdG707.metabolites.cpd00067_e # H\n",
    "\n",
    "rxn05472_c = cobra.Reaction('rxn05472_c')\n",
    "rxn05472_c.name = '2-(hydroxymethyl)phenol transport in/out via proton symport'\n",
    "rxn05472_c.lower_bound = -1000.\n",
    "rxn05472_c.upper_bound = 1000.\n",
    "rxn05472_c.add_metabolites({\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd01553_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd01553_c: 1.0\n",
    "})\n",
    "rxn05472_c.annotation['sbo'] = 'SBO:0000185'\n",
    "rxn05472_c.annotation['seed.reaction'] = 'rxn05472'\n",
    "rxn05472_c.annotation['bigg.reaction'] = '2HXMPt6'\n",
    "iCdG707.add_reactions([rxn05472_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct Melezitose catabolism\n",
    "remove_gene = [iCdG707.genes.get_by_id('1151372.4.peg.1723')]\n",
    "cobra.manipulation.delete.remove_genes(iCdG707, remove_gene)\n",
    "\n",
    "cpd00001_c = iCdG707.metabolites.cpd00001_c # H2O\n",
    "cpd05161_c = iCdG707.metabolites.cpd05161_c # Melezitose\n",
    "cpd00027_c = iCdG707.metabolites.cpd00027_c # Glucose\n",
    "cpd20885_c = iCdG707.metabolites.cpd20885_c # Turanose\n",
    "\n",
    "R01103_c = cobra.Reaction('R01103_c')\n",
    "R01103_c.name = 'Melezitose galactohydrolase'\n",
    "R01103_c.lower_bound = 0.\n",
    "R01103_c.upper_bound = 1000.\n",
    "R01103_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd05161_c: -1.0,\n",
    "    cpd00027_c: 1.0,\n",
    "    cpd20885_c: 1.0\n",
    "})\n",
    "R01103_c.annotation['kegg.reaction'] = 'R01103'\n",
    "R01103_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG707.add_reactions([R01103_c])\n",
    "\n",
    "iCdG707.id = 'iCdG' + str(len(iCdG707.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e', # Pyridoxine\n",
    "       'EX_cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "for rxn in iCdG707.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in iCdG707.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        iCdG707.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        iCdG707.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in iCdG707.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        iCdG707.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        iCdG707.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "70.05819375319561"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG707.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_gene_ids = {}\n",
    "ec_gene_ids = {}\n",
    "refseq_gene_ids = {}\n",
    "kegg_gene_ids = {}\n",
    "patric_genes = set()\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/uniprot_Cd630.tab', 'r') as gene_annotation:\n",
    "    header = gene_annotation.readline()\n",
    "    for line in gene_annotation:\n",
    "        line = line.split()\n",
    "        uniprot_gene_ids[line[4]] = line[0]\n",
    "        ec_gene_ids[line[4]] = line[1]\n",
    "        refseq_gene_ids[line[4]] = line[2]\n",
    "        kegg_gene_ids[line[4]] = line[3]\n",
    "        patric_genes |= set([line[4]])\n",
    "\n",
    "success = 0\n",
    "for gene in iCdG707.genes:\n",
    "    if not gene.id in patric_genes: continue\n",
    "    \n",
    "    success += 1\n",
    "    gene.annotation['uniprot'] = uniprot_gene_ids[gene.id]\n",
    "    gene.annotation['kegg.genes'] = kegg_gene_ids[gene.id]\n",
    "    \n",
    "    if refseq_gene_ids[gene.id] != 'none':\n",
    "        gene.annotation['refseq'] = refseq_gene_ids[gene.id]\n",
    "    \n",
    "    if ec_gene_ids[gene.id] != 'none':\n",
    "        gene.annotation['ec_number'] = ec_gene_ids[gene.id]\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final annotation fixes\n",
    "\n",
    "# no reaction annotation \n",
    "iCdG707.reactions.rxn12212_c.annotation['seed.reaction'] = 'rxn12212'\n",
    "\n",
    "# no seed annotation\n",
    "iCdG707.metabolites.cpd02711_e.annotation['seed.compound'] = 'cpd02711'\n",
    "\n",
    "# No kegg reaction\n",
    "for rxn in [\"R03018_c\",\"R01103_c\",\"R01174_1_c\",\"R01214_c\",\"R01174_2_c\",\"R01174_3_c\",\"R01174_4_c\",\"R01174_5_c\",\"K20025_c\",\"R11076_c\",\"R06782_c\",\"R11462_c\"]:\n",
    "    iCdG707.reactions.get_by_id(rxn).annotation['kegg.reaction'] = rxn.split('_')[0]\n",
    "\n",
    "# no seed annotation\n",
    "for rxn in [\"rxn30238_c\",\"rxn12212_c\",\"rxn09180_c\",\"rxn10042_c\",\"rxn05147_c\"]:\n",
    "    iCdG707.reactions.get_by_id(rxn).annotation['seed.reaction'] = rxn.split('_')[0]\n",
    "\n",
    "# SBO\n",
    "iCdG707.metabolites.cpd02711_e.annotation['sbo'] = 'SBO:0000247'\n",
    "for rxn in [\"rxn30238_c\",\"rxn12212_c\",\"rxn09180_c\",\"rxn10042_c\",\"rxn05147_c\"]:\n",
    "    iCdG707.reactions.get_by_id(rxn).annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG707.genes.get_by_id('272563.120.peg.1564').annotation['sbo'] = 'SBO:0000243'\n",
    "for rxn in [\"rxn30270_c\",\"rxn05546_c\",\"rxn09658_c\",\"rxn09189_c\",\"rxn30274_c\",\"rxn05534_c\",\"rxn10183_c\",\"rxn09655_c\",\"rxn12848_c\",\"rxn05536_c\",\"rxn30276_c\",\"rxn05538_c\",\"rxn05147_c\",\"rxn12849_c\",\"rxn30278_c\",\"rxn05541_c\",\"rxn12850_c\",\"rxn30280_c\",\"rxn05545_c\",\"rxn12851_c\",\"rxn30264_c\",\"rxn05178_c\",\"rxn30268_c\",\"rxn30259_c\",\"rxn05681_c\",\"rxn30266_c\",\"rxn30262_c\",\"rxn34423_c\"]:\n",
    "    iCdG707.reactions.get_by_id(rxn).annotation['sbo'] = 'SBO:0000185'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save model\n",
    "cobra.io.write_sbml_model(iCdG707, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG706.sbml')\n",
    "cobra.io.save_json_model(iCdG707, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG706.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "from cobra.flux_analysis import parsimonious\n",
    "\n",
    "# Function to calculate doubling time from objective value\n",
    "def doublingTime(model):\n",
    "    with model as m:\n",
    "        if m.slim_optimize(error_value=0.) < 1e-6:\n",
    "            print('GENRE has no objective flux')\n",
    "        else:\n",
    "            growth = (1. / float(m.slim_optimize())) * 3600.\n",
    "            print(str(round(growth, 2)) + ' minutes doubling time')\n",
    "\n",
    "\n",
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "\n",
    "    #print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    fraction = (float(len(blocked)) / float(len(model.reactions))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions are blocked')\n",
    "    \n",
    "    return blocked\n",
    "\n",
    "\n",
    "# Identify potentially gapfilled reactions, checks against pFBA solution\n",
    "def missingGPR(model):\n",
    "    \n",
    "    noGene = []\n",
    "    exclude = []\n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.genes)) == 0:\n",
    "            if rxn.annotation['sbo'] != 'SBO:0000629':\n",
    "                if rxn in model.boundary:\n",
    "                    exclude.append(rxn.id)\n",
    "                    continue\n",
    "                else:\n",
    "                    noGene.append(rxn.id)\n",
    "    \n",
    "    solution = parsimonious.pfba(model)\n",
    "    active_rxns = set([rxn.id for rxn in model.reactions if abs(solution.fluxes[rxn.id]) > 1e-5])\n",
    "    active_rxns = active_rxns.difference(set(exclude))\n",
    "    noGene_active = set(noGene).intersection(active_rxns)\n",
    "\n",
    "    fraction = float(len(model.reactions)) - float(len(exclude))\n",
    "    fraction = (float(len(noGene)) / fraction) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions without GPRs')\n",
    "    \n",
    "    fraction = (float(len(noGene_active)) / float(len(active_rxns))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% of reactions used in pFBA solution have no GPR')\n",
    "    \n",
    "    return noGene_active\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(model, cytosol='cytosol'):\n",
    "\n",
    "    free = []\n",
    "    with model as m:\n",
    "    \n",
    "        # Close all exchanges\n",
    "        for rxn in m.boundary: m.reactions.get_by_id(rxn.id).lower_bound = 0.\n",
    "    \n",
    "        # Create demand for each reaction and optimize individually\n",
    "        reset_rxn = m.reactions[0].id\n",
    "        for cpd in m.metabolites: \n",
    "            if cpd.compartment == cytosol:\n",
    "                demand = cobra.Reaction('demand')\n",
    "                demand.bounds = (0., 1000.)\n",
    "                demand.add_metabolites({cpd: -1.0})\n",
    "                m.add_reactions([demand])\n",
    "                m.objective = demand\n",
    "                obj_val = m.slim_optimize()\n",
    "                if obj_val > 1e-8: free.append(cpd.id)\n",
    "                m.objective = reset_rxn\n",
    "                m.remove_reactions([demand])\n",
    "    \n",
    "    fraction = (float(len(free)) / float(len(model.metabolites))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(model):\n",
    "    \n",
    "    with model as m:\n",
    "\n",
    "        elements = set()\n",
    "        for cpd in m.metabolites:\n",
    "            try:\n",
    "                elements |= set(cpd.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        massImbal = []\n",
    "        failed = 0\n",
    "        if len(elements) == 0:\n",
    "            print('No elemental data associated with metabolites!')\n",
    "            failed = 1\n",
    "        else:\n",
    "            for rxn in m.reactions:\n",
    "                if rxn.annotation['sbo'] == 'SBO:0000629': \n",
    "                    continue\n",
    "                elif rxn in m.boundary:\n",
    "                    continue\n",
    "\n",
    "                try:\n",
    "                    test = rxn.check_mass_balance()\n",
    "                except ValueError:\n",
    "                    continue\n",
    "\n",
    "                if len(list(test)) > 0:\n",
    "                    if len(set(test.keys()).intersection(elements)) > 0: massImbal.append(rxn.id)\n",
    "                        \n",
    "    if failed != 1:\n",
    "        fraction = (float(len(massImbal)) / float(len(model.reactions))) * 100.\n",
    "        fraction = round(fraction, 2)\n",
    "        print(str(fraction) + '% reactions are mass imbalanced')\n",
    "        \n",
    "    return massImbal\n",
    "\n",
    "\n",
    "def basicCheck(model):\n",
    "    \n",
    "    # Determination\n",
    "    if len(model.reactions) < len(model.metabolites): \n",
    "        print('GENRE is overdetermined')\n",
    "    elif len(model.reactions) > len(model.metabolites):\n",
    "        print('GENRE is underdetermined')\n",
    "    else:\n",
    "        pass\n",
    "    \n",
    "    # Compartments\n",
    "    print('GENRE has ' + str(len(model.compartments.keys())) + ' compartment(s)')\n",
    "    \n",
    "    # Genes\n",
    "    if len(model.genes) == 0: \n",
    "        print('GENRE has no gene data')\n",
    "    else:\n",
    "        print('GENRE has ' + str(len(model.genes)) + ' genes')\n",
    "    no_rxns = []\n",
    "    for gene in model.genes:\n",
    "        if len(gene.reactions) == 0:\n",
    "            no_rxns.append(gene.id)\n",
    "    if len(no_rxns) > 0:\n",
    "          print('GENRE has ' + str(len(no_rxns)) + ' genes that are not associated with reactions')\n",
    "            \n",
    "    # Growth\n",
    "    doublingTime(model)\n",
    "    \n",
    "    return no_rxns\n",
    "\n",
    "# Quicker way to read in models\n",
    "import pickle\n",
    "def read_model(fileName, obj='none'):\n",
    "    \n",
    "    fileType = fileName.split('.')[-1]\n",
    "    \n",
    "    if fileType == 'sbml' or fileType == 'xml':\n",
    "        model = cobra.io.read_sbml_model(fileName)\n",
    "    elif fileType == 'json':\n",
    "        model = cobra.io.load_json_model(fileName)\n",
    "    elif fileType == 'yaml':\n",
    "        model = cobra.io.load_yaml_model(fileName)\n",
    "    elif fileType == 'mat':\n",
    "        model = cobra.io.load_matlab_model(fileName)\n",
    "    elif fileType == 'pkl':\n",
    "        model = pickle.load(open(fileName, 'rb'))\n",
    "    else:\n",
    "        raise TypeError('Unrecognized file extension')\n",
    "    \n",
    "    if obj != 'none': model.objective = obj\n",
    "    for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "    return model\n",
    "\n",
    "\n",
    "def save_model(model, fileName, media='none'):\n",
    "    \n",
    "    if media != 'none':\n",
    "        media_exchanges = ['EX_' + x for x in media]\n",
    "        for rxn in model.boundary:\n",
    "            if not rxn.id in media_exchanges:\n",
    "                rxn.bounds = (0., 1000.)\n",
    "            else:\n",
    "                rxn.bounds = (-1000., 1000.)\n",
    "    else:\n",
    "        for rxn in model.boundary:\n",
    "            rxn.bounds = (-1000., 1000.)\n",
    "    \n",
    "    sbml_fileName = fileName + '.sbml'\n",
    "    cobra.io.write_sbml_model(model, sbml_fileName)\n",
    "    \n",
    "    json_fileName = fileName + '.json'\n",
    "    cobra.io.save_json_model(model, json_fileName)\n",
    "    \n",
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "iCdG706 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG706.sbml')\n",
    "\n",
    "# Neu5Ac --> N-Acetyl-Neuraminic acid\n",
    "iCdG706.metabolites.cpd00232_c.name = 'N-Acetyl-Neuraminic acid'\n",
    "\n",
    "# D-Tagatose\n",
    "cpd00589_e = cobra.Metabolite(\n",
    "    'cpd00589_e',\n",
    "    formula='C6H12O6',\n",
    "    name='D-Tagatose',\n",
    "    compartment='extracellular')\n",
    "cpd00589_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00589_e.annotation['seed.compound'] = 'cpd00589'\n",
    "cpd00589_e.annotation['bigg.metabolite'] = 'tag__D'\n",
    "cpd00589_e.annotation['kegg.compound'] = 'C00795'\n",
    "iCdG706.add_boundary(cpd00589_e, type='exchange', reaction_id='EX_cpd00589_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG706.reactions.EX_cpd01553_e.name = 'Tagatose exchange'\n",
    "iCdG706.reactions.EX_cpd01553_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00805_c = iCdG706.metabolites.cpd00805_c # D-Tagatose 6-phosphate\n",
    "cpd00002_c = iCdG706.metabolites.cpd00002_c # ATP\n",
    "cpd00008_c = iCdG706.metabolites.cpd00008_c # ADP\n",
    "cpd00067_c = iCdG706.metabolites.cpd00067_c # H\n",
    "\n",
    "rxn02094_c = cobra.Reaction('rxn02094_c')\n",
    "rxn02094_c.name = 'ATP:D-tagatose 6-phosphotransferase'\n",
    "rxn02094_c.gene_reaction_rule = '( 645463.3.peg.3332 or 645463.3.peg.303 or 645463.3.peg.464 or 645463.3.peg.998 ) and ( 645463.3.peg.306 or 645463.3.peg.465 or 645463.3.peg.3333 or 645463.3.peg.1000 or 645463.3.peg.304 ) and ( 645463.3.peg.307 or 645463.3.peg.3331 or 645463.3.peg.466 or 645463.3.peg.1001 ) and ( 645463.3.peg.467 or 645463.3.peg.1002 or 645463.3.peg.308 or 645463.3.peg.3330 )'\n",
    "rxn02094_c.lower_bound = 0.\n",
    "rxn02094_c.upper_bound = 1000.\n",
    "rxn02094_c.add_metabolites({\n",
    "    cpd00589_e: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00805_c: 1.0\n",
    "})\n",
    "rxn02094_c.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "rxn02094_c.annotation['seed.reaction'] = 'rxn02094'\n",
    "rxn02094_c.annotation['kegg.reaction'] = 'R02927'\n",
    "rxn02094_c.annotation['bigg.reaction'] = 'TAGptspp'\n",
    "iCdG706.add_reactions([rxn02094_c])\n",
    "\n",
    "\n",
    "# N-Acetyl-D-Galactosamine\n",
    "cpd27607_e = cobra.Metabolite(\n",
    "    'cpd27607_e',\n",
    "    formula='C8H15NO6',\n",
    "    name='N-Acetyl-D-Galactosamine',\n",
    "    compartment='extracellular')\n",
    "cpd27607_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd27607_e.annotation['seed.compound'] = 'cpd27607'\n",
    "cpd27607_e.annotation['bigg.metabolite'] = 'acgal'\n",
    "cpd27607_e.annotation['kegg.compound'] = 'C01074'\n",
    "\n",
    "iCdG706.add_boundary(cpd27607_e, type='exchange', reaction_id='EX_cpd27607_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG706.reactions.EX_cpd27607_e.name = 'N-Acetyl-D-Galactosamine exchange'\n",
    "iCdG706.reactions.EX_cpd27607_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd15390_c = cobra.Metabolite(\n",
    "    'cpd15390_c',\n",
    "    formula='C8H14NO9P',\n",
    "    name='N-Acetyl-D-Galactosamine-1-phosphate',\n",
    "    compartment='cytosol')\n",
    "cpd15390_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd15390_c.annotation['seed.compound'] = 'cpd15390'\n",
    "cpd15390_c.annotation['bigg.metabolite'] = 'acgal1p'\n",
    "cpd15390_c.annotation['kegg.compound'] = 'C18060'\n",
    "\n",
    "cpd00002_c = iCdG706.metabolites.cpd00002_c # ATP\n",
    "cpd00008_c = iCdG706.metabolites.cpd00008_c # ADP\n",
    "cpd00067_c = iCdG706.metabolites.cpd00067_c # H\n",
    "\n",
    "rxn46837_c = cobra.Reaction('rxn46837_c')\n",
    "rxn46837_c.name = 'ATP:N-acetyl-D-galactosamine 1-phosphotransferase'\n",
    "rxn46837_c.gene_reaction_rule = '645463.3.peg.2802 and 645463.3.peg.3106 and 645463.3.peg.3106'\n",
    "rxn46837_c.lower_bound = 0.\n",
    "rxn46837_c.upper_bound = 1000.\n",
    "rxn46837_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd27607_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd15390_c: 1.0\n",
    "})\n",
    "rxn46837_c.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "rxn46837_c.annotation['seed.reaction'] = 'rxn46837'\n",
    "rxn46837_c.annotation['kegg.reaction'] = 'R08967'\n",
    "rxn46837_c.annotation['bigg.reaction'] = 'ACGALK'\n",
    "iCdG706.add_reactions([rxn46837_c])\n",
    "\n",
    "cpd00062_c = iCdG706.metabolites.cpd00062_c # UTP\n",
    "cpd00012_c = iCdG706.metabolites.cpd00012_c # Phosphate\n",
    "cpd00175_c = iCdG706.metabolites.cpd00175_c # UDP-N-acetyl-D-galactosamine\n",
    "\n",
    "rxn17713_c = cobra.Reaction('rxn17713_c')\n",
    "rxn17713_c.name = 'UTP-monosaccharide-1-phosphate uridylyltransferase'\n",
    "rxn17713_c.lower_bound = 0.\n",
    "rxn17713_c.upper_bound = 1000.\n",
    "rxn17713_c.add_metabolites({\n",
    "    cpd00062_c: -1.0,\n",
    "    cpd15390_c: -1.0,\n",
    "    cpd00012_c: 1.0,\n",
    "    cpd00175_c: 1.0\n",
    "})\n",
    "rxn17713_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "rxn17713_c.annotation['seed.reaction'] = 'rxn17713'\n",
    "rxn17713_c.annotation['kegg.reaction'] = 'R10183'\n",
    "rxn17713_c.annotation['bigg.reaction'] = 'GALT'\n",
    "iCdG706.add_reactions([rxn17713_c])\n",
    "\n",
    "save_model(iCdG706, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG706', media=mdm)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 725 genes\n",
      "GENRE has 6 genes that are not associated with reactions\n",
      "20.68 minutes doubling time\n",
      "36.39% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% reactions are mass imbalanced\n",
      "13.39% reactions without GPRs\n",
      "22.26% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdG706 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG706.sbml')\n",
    "\n",
    "remove_genes = basicCheck(iCdG706)\n",
    "iCdG706_noGPRblocked = blockedReactions(iCdG706)\n",
    "iCdG706_free = checkFreeMass(iCdG706)\n",
    "iCdG706_massImbal = checkBalance(iCdG706)\n",
    "iCdG706_nogene = missingGPR(iCdG706)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "remove_genes = [iCdG706.genes.get_by_id(x) for x in remove_genes]\n",
    "cobra.manipulation.delete.remove_genes(iCdG706, remove_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG706.id = 'iCdG' + str(len(iCdG706.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG719</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f3fb459f810</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1189</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1223</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG719 at 0x7f3fb459f810>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG706"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_model(iCdG706, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG719', media=mdm)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Matching Biolog Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG719 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG719.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "import copy\n",
    "\n",
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "universal_GramPos.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13783_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13784_c.remove_from_model(remove_orphans=True)\n",
    "#universal_GramPos.reactions.rxn11920_c.name = 'Poly-beta-hydroxybutyrate (PHB) synthease'\n",
    "universal_GramPos.reactions.rxn12188_c.name = '2-deoxyglucose-6-phosphatase'\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ala-ala\n",
    "cpd00002_c = iCdG719.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG719.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG719.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG719.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG719.metabolites.cpd00067_c # H+\n",
    "cpd00731_c = iCdG719.metabolites.cpd00731_c # Ala-ala\n",
    "cpd00731_e = cobra.Metabolite(\n",
    "    'cpd00731_e',\n",
    "    formula='C6H12N2O3',\n",
    "    name='Ala-ala',\n",
    "    compartment='extracellular')\n",
    "cpd00731_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00731_e.annotation['seed.compound'] = 'cpd00731'\n",
    "cpd00731_e.annotation['bigg.metabolite'] = ['alaala', 'diala-L']\n",
    "cpd00731_e.annotation['kegg.compound'] = 'C00993'\n",
    "\n",
    "# Transport\n",
    "rxn18658_c = cobra.Reaction('rxn18658_c')\n",
    "rxn18658_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn18658_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn18658_c.bounds = (0., 1000.)\n",
    "rxn18658_c.add_metabolites({\n",
    "    cpd00731_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00731_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn18658_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn18658_c.annotation['seed.reaction'] = 'rxn18658'\n",
    "rxn18658_c.annotation['bigg.reaction'] = 'ALAALAabcpp'\n",
    "iCdG719.add_reactions([rxn18658_c])\n",
    "\n",
    "cpd00731_e = iCdG719.metabolites.cpd00731_e\n",
    "iCdG719.add_boundary(cpd00731_e, type='exchange', reaction_id='EX_cpd00731_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00731_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Hydrolysis\n",
    "iCdG719.reactions.rxn07466_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Phosphorylated hexose sugars\n",
    "rxn05554_c = copy.deepcopy(universal_GramPos.reactions.rxn05554_c) # Fructose-6-phosphate\n",
    "rxn05562_c = copy.deepcopy(universal_GramPos.reactions.rxn05562_c) # Glucose-1-phosphate\n",
    "rxn05563_c = copy.deepcopy(universal_GramPos.reactions.rxn05563_c) # Glucose-6-phosphate\n",
    "iCdG719.add_reactions([rxn05554_c,rxn05562_c,rxn05563_c])\n",
    "\n",
    "iCdG719.reactions.rxn05554_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05554_c.annotation['seed.reaction'] = 'rxn05554'\n",
    "iCdG719.reactions.rxn05554_c.annotation['bigg.reaction'] = 'F6Pt6_i2'\n",
    "\n",
    "iCdG719.reactions.rxn05562_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05562_c.annotation['seed.reaction'] = 'rxn05562'\n",
    "iCdG719.reactions.rxn05562_c.annotation['bigg.reaction'] = 'G1Pt6_2'\n",
    "\n",
    "iCdG719.reactions.rxn05563_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05563_c.annotation['seed.reaction'] = 'rxn05563'\n",
    "iCdG719.reactions.rxn05563_c.annotation['bigg.reaction'] = 'G6Pt6_2'\n",
    "\n",
    "cpd00072_e = iCdG719.metabolites.cpd00072_e\n",
    "iCdG719.add_boundary(cpd00072_e, type='exchange', reaction_id='EX_cpd00072_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00072_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00089_e = iCdG719.metabolites.cpd00089_e\n",
    "iCdG719.add_boundary(cpd00089_e, type='exchange', reaction_id='EX_cpd00089_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00089_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00079_e = iCdG719.metabolites.cpd00079_e\n",
    "iCdG719.add_boundary(cpd00079_e, type='exchange', reaction_id='EX_cpd00079_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00079_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Galactose\n",
    "rxn05162_c = copy.deepcopy(universal_GramPos.reactions.rxn05162_c)\n",
    "iCdG719.add_reactions([rxn05162_c])\n",
    "iCdG719.reactions.rxn05162_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05162_c.annotation['seed.reaction'] = ''\n",
    "iCdG719.reactions.rxn05162_c.annotation['bigg.reaction'] = ''\n",
    "cpd00108_e = iCdG719.metabolites.cpd00108_e\n",
    "iCdG719.add_boundary(cpd00108_e, type='exchange', reaction_id='EX_cpd00108_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.rxn18658_c.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Intracellular organic acids\n",
    "\n",
    "# 4-Hydroxybenzoate\n",
    "rxn12549_c = copy.deepcopy(universal_GramPos.reactions.rxn12549_c)\n",
    "iCdG719.add_reactions([rxn12549_c])\n",
    "iCdG719.reactions.rxn12549_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn12549_c.annotation['seed.reaction'] = 'rxn12549'\n",
    "iCdG719.reactions.rxn12549_c.annotation['bigg.reaction'] = '4HBZt'\n",
    "cpd00136_e = iCdG719.metabolites.cpd00136_e\n",
    "iCdG719.add_boundary(cpd00136_e, type='exchange', reaction_id='EX_cpd00136_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00136_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# 2-Hydroxybutyrate\n",
    "rxn12585_c = copy.deepcopy(universal_GramPos.reactions.rxn12585_c) \n",
    "iCdG719.add_reactions([rxn12585_c])\n",
    "iCdG719.reactions.rxn12585_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn12585_c.annotation['seed.reaction'] = 'rxn12585'\n",
    "iCdG719.reactions.rxn12585_c.annotation['bigg.reaction'] = '2HBt2'\n",
    "cpd03561_e = iCdG719.metabolites.cpd03561_e\n",
    "iCdG719.add_boundary(cpd03561_e, type='exchange', reaction_id='EX_cpd03561_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd03561_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Fumarate\n",
    "rxn05561_c = copy.deepcopy(universal_GramPos.reactions.rxn05561_c) \n",
    "iCdG719.add_reactions([rxn05561_c])\n",
    "iCdG719.reactions.rxn05561_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05561_c.annotation['seed.reaction'] = 'rxn05561'\n",
    "iCdG719.reactions.rxn05561_c.annotation['bigg.reaction'] = ['FUMt2r','FUMt6','U207']\n",
    "cpd00106_e = iCdG719.metabolites.cpd00106_e\n",
    "iCdG719.add_boundary(cpd00106_e, type='exchange', reaction_id='EX_cpd00106_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00106_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Acetoacetate\n",
    "rxn10960_c = copy.deepcopy(universal_GramPos.reactions.rxn10960_c) \n",
    "iCdG719.add_reactions([rxn10960_c])\n",
    "iCdG719.reactions.rxn10960_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn10960_c.annotation['seed.reaction'] = 'rxn10960'\n",
    "iCdG719.reactions.rxn10960_c.annotation['bigg.reaction'] = ['ACACtex','U126']\n",
    "cpd00142_e = iCdG719.metabolites.cpd00142_e\n",
    "iCdG719.add_boundary(cpd00142_e, type='exchange', reaction_id='EX_cpd00142_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00142_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Shikimate\n",
    "rxn10827_c = copy.deepcopy(universal_GramPos.reactions.rxn10827_c) \n",
    "iCdG719.add_reactions([rxn10827_c])\n",
    "iCdG719.reactions.rxn10827_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn10827_c.annotation['seed.reaction'] = 'rxn10827'\n",
    "iCdG719.reactions.rxn10827_c.annotation['bigg.reaction'] = ['SKMt2','SKMt2pp']\n",
    "iCdG719.reactions.rxn10827_c.name = 'Shikimate transport in via proton symport'\n",
    "cpd00383_e = iCdG719.metabolites.cpd00383_e\n",
    "iCdG719.add_boundary(cpd00383_e, type='exchange', reaction_id='EX_cpd00383_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00383_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Formate\n",
    "rxn05559_c = copy.deepcopy(universal_GramPos.reactions.rxn05559_c) \n",
    "iCdG719.add_reactions([rxn05559_c])\n",
    "iCdG719.reactions.rxn05559_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05559_c.annotation['seed.reaction'] = 'rxn05559'\n",
    "iCdG719.reactions.rxn05559_c.annotation['bigg.reaction'] = ['FORt2','FORt2pp']\n",
    "cpd00047_e = iCdG719.metabolites.cpd00047_e\n",
    "iCdG719.add_boundary(cpd00047_e, type='exchange', reaction_id='EX_cpd00047_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00047_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Glycolate\n",
    "rxn05470_c = copy.deepcopy(universal_GramPos.reactions.rxn05470_c) \n",
    "iCdG719.add_reactions([rxn05470_c])\n",
    "iCdG719.reactions.rxn05470_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05470_c.annotation['seed.reaction'] = 'rxn05470'\n",
    "iCdG719.reactions.rxn05470_c.annotation['bigg.reaction'] = ['GLYCLTt2r','GLYCLTt2rpp']\n",
    "cpd00139_e = iCdG719.metabolites.cpd00139_e\n",
    "iCdG719.add_boundary(cpd00139_e, type='exchange', reaction_id='EX_cpd00139_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00139_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# GABA\n",
    "rxn05564_c = copy.deepcopy(universal_GramPos.reactions.rxn05564_c) \n",
    "iCdG719.add_reactions([rxn05564_c])\n",
    "iCdG719.reactions.rxn05564_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05564_c.annotation['seed.reaction'] = 'rxn05564'\n",
    "iCdG719.reactions.rxn05564_c.annotation['bigg.reaction'] = ['ABUTt2','ABUTt2pp','ABUTt2r','GABAt2','PUT4_4','UGA4_1']\n",
    "cpd00281_e = iCdG719.metabolites.cpd00281_e\n",
    "iCdG719.add_boundary(cpd00281_e, type='exchange', reaction_id='EX_cpd00281_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00281_e.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Hydroxylamine\n",
    "rxn05703_c = copy.deepcopy(universal_GramPos.reactions.rxn05703_c) \n",
    "iCdG719.add_reactions([rxn05703_c])\n",
    "iCdG719.reactions.rxn05703_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05703_c.annotation['seed.reaction'] = 'rxn05703'\n",
    "cpd00165_e = iCdG719.metabolites.cpd00165_e\n",
    "iCdG719.add_boundary(cpd00165_e, type='exchange', reaction_id='EX_cpd00165_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00165_e.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Nucleotide analogs\n",
    "\n",
    "# Deoxyinosine\n",
    "rxn09687_c = copy.deepcopy(universal_GramPos.reactions.rxn09687_c) \n",
    "iCdG719.add_reactions([rxn09687_c])\n",
    "iCdG719.reactions.rxn09687_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn09687_c.annotation['seed.reaction'] = 'rxn09687'\n",
    "iCdG719.reactions.rxn09687_c.annotation['bigg.reaction'] = ['DINSt2','DINSt2pp','U180','U196']\n",
    "cpd03279_e = iCdG719.metabolites.cpd03279_e\n",
    "iCdG719.add_boundary(cpd03279_e, type='exchange', reaction_id='EX_cpd03279_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd03279_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Inosine\n",
    "rxn05316_c = copy.deepcopy(universal_GramPos.reactions.rxn05316_c) \n",
    "iCdG719.add_reactions([rxn05316_c])\n",
    "iCdG719.reactions.rxn05316_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05316_c.annotation['seed.reaction'] = 'rxn05316'\n",
    "iCdG719.reactions.rxn05316_c.annotation['bigg.reaction'] = ['INSt2','INSt2pp','INSt2r','INSt2rpp','U177','U194']\n",
    "cpd00246_e = iCdG719.metabolites.cpd00246_e\n",
    "iCdG719.add_boundary(cpd00246_e, type='exchange', reaction_id='EX_cpd00246_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00246_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Guanosine\n",
    "rxn05204_c = copy.deepcopy(universal_GramPos.reactions.rxn05204_c) \n",
    "iCdG719.add_reactions([rxn05204_c])\n",
    "iCdG719.reactions.rxn05204_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05204_c.annotation['seed.reaction'] = 'rxn05204'\n",
    "iCdG719.reactions.rxn05204_c.annotation['bigg.reaction'] = ['GSNt2','GSNt2m','GSNt2pp','GSNt2r','U175','U192']\n",
    "cpd00311_e = iCdG719.metabolites.cpd00311_e\n",
    "iCdG719.add_boundary(cpd00311_e, type='exchange', reaction_id='EX_cpd00311_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00311_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Cytidine\n",
    "rxn05198_c = copy.deepcopy(universal_GramPos.reactions.rxn05198_c) \n",
    "iCdG719.add_reactions([rxn05198_c])\n",
    "iCdG719.reactions.rxn05198_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05198_c.annotation['seed.reaction'] = 'rxn05198'\n",
    "iCdG719.reactions.rxn05198_c.annotation['bigg.reaction'] = ['CYTDt2','CYTDt2pp','CYTDt2r','CYTDt2rpp','U176','U186','U193']\n",
    "cpd00367_e = iCdG719.metabolites.cpd00367_e\n",
    "iCdG719.add_boundary(cpd00367_e, type='exchange', reaction_id='EX_cpd00367_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00367_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxyguanosine\n",
    "rxn09688_c = copy.deepcopy(universal_GramPos.reactions.rxn09688_c) \n",
    "iCdG719.add_reactions([rxn09688_c])\n",
    "iCdG719.reactions.rxn09688_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn09688_c.annotation['seed.reaction'] = 'rxn09688'\n",
    "iCdG719.reactions.rxn09688_c.annotation['bigg.reaction'] = ['DGSNt2','DGSNt2pp','U181','U197']\n",
    "cpd00277_e = iCdG719.metabolites.cpd00277_e\n",
    "iCdG719.add_boundary(cpd00277_e, type='exchange', reaction_id='EX_cpd00277_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00277_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Adenosine\n",
    "rxn05318_c = copy.deepcopy(universal_GramPos.reactions.rxn05318_c) \n",
    "iCdG719.add_reactions([rxn05318_c])\n",
    "iCdG719.reactions.rxn05318_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05318_c.annotation['seed.reaction'] = 'rxn05318'\n",
    "iCdG719.reactions.rxn05318_c.annotation['bigg.reaction'] = ['ADNt2','ADNt2pp','ADNt2r','ADNt2rpp','U174','U185','U191']\n",
    "cpd00182_e = iCdG719.metabolites.cpd00182_e\n",
    "iCdG719.add_boundary(cpd00182_e, type='exchange', reaction_id='EX_cpd00182_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00182_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxycytidine\n",
    "rxn05205_c = copy.deepcopy(universal_GramPos.reactions.rxn05205_c) \n",
    "iCdG719.add_reactions([rxn05205_c])\n",
    "iCdG719.reactions.rxn05205_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05205_c.annotation['seed.reaction'] = 'rxn05205'\n",
    "iCdG719.reactions.rxn05205_c.annotation['bigg.reaction'] = ['DCYTt2','DCYTt2pp','U183','U189','U199']\n",
    "cpd00654_e = iCdG719.metabolites.cpd00654_e\n",
    "iCdG719.add_boundary(cpd00654_e, type='exchange', reaction_id='EX_cpd00654_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00654_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxyuridine\n",
    "rxn09685_c = copy.deepcopy(universal_GramPos.reactions.rxn09685_c) \n",
    "iCdG719.add_reactions([rxn09685_c])\n",
    "iCdG719.reactions.rxn09685_c.name = 'Deoxyuridine symport'\n",
    "iCdG719.reactions.rxn09685_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn09685_c.annotation['seed.reaction'] = 'rxn09685'\n",
    "iCdG719.reactions.rxn09685_c.annotation['bigg.reaction'] = ['DURIt2','DURIt2pp','U184','U190','U200']\n",
    "cpd00412_e = iCdG719.metabolites.cpd00412_e\n",
    "iCdG719.add_boundary(cpd00412_e, type='exchange', reaction_id='EX_cpd00412_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00412_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# Deoxyadenosine\n",
    "rxn05317_c = copy.deepcopy(universal_GramPos.reactions.rxn05317_c) \n",
    "iCdG719.add_reactions([rxn05317_c])\n",
    "iCdG719.reactions.rxn05317_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05317_c.annotation['seed.reaction'] = 'rxn05317'\n",
    "iCdG719.reactions.rxn05317_c.annotation['bigg.reaction'] = ['DADNt2','DADNt2pp','U182','U188','U198']\n",
    "cpd00438_e = iCdG719.metabolites.cpd00438_e\n",
    "iCdG719.add_boundary(cpd00438_e, type='exchange', reaction_id='EX_cpd00438_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00438_e.annotation['sbo'] = 'SBO:0000627'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Additional amino acids\n",
    "\n",
    "# Citrulline\n",
    "rxn05674_c = copy.deepcopy(universal_GramPos.reactions.rxn05674_c) \n",
    "iCdG719.add_reactions([rxn05674_c])\n",
    "iCdG719.reactions.rxn05674_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05674_c.annotation['seed.reaction'] = 'rxn05674'\n",
    "iCdG719.reactions.rxn05674_c.annotation['bigg.reaction'] = ['CITR_Lt2','CITR_Lt2ex','CITR_Lt2pp']\n",
    "cpd00274_e = iCdG719.metabolites.cpd00274_e\n",
    "iCdG719.add_boundary(cpd00274_e, type='exchange', reaction_id='EX_cpd00274_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00274_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "# D-Glutamate\n",
    "rxn05672_c = copy.deepcopy(universal_GramPos.reactions.rxn05672_c) \n",
    "iCdG719.add_reactions([rxn05672_c])\n",
    "iCdG719.reactions.rxn05672_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05672_c.annotation['seed.reaction'] = 'rxn05672'\n",
    "iCdG719.reactions.rxn05672_c.annotation['bigg.reaction'] = 'GLU-Dt2'\n",
    "cpd00186_e = iCdG719.metabolites.cpd00186_e\n",
    "iCdG719.add_boundary(cpd00186_e, type='exchange', reaction_id='EX_cpd00186_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00186_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "#--------------------------------------------------------------------------#\n",
    "\n",
    "# Cystathionine\n",
    "cpd00424_e = cobra.Metabolite(\n",
    "    'cpd00424_e',\n",
    "    formula='C7H14N2O4S',\n",
    "    name='Cystathionine',\n",
    "    compartment='extracellular')\n",
    "cpd00424_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00424_e.annotation['seed.compound'] = 'cpd00424'\n",
    "cpd00424_e.annotation['bigg.metabolite'] = 'CST_c'\n",
    "cpd00424_e.annotation['kegg.compound'] = 'C00542'\n",
    "iCdG719.add_boundary(cpd00424_e, type='exchange', reaction_id='EX_cpd00424_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00424_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00424_c = iCdG719.metabolites.cpd00424_c\n",
    "rxn10398_c = cobra.Reaction('rxn10398_c')\n",
    "rxn10398_c.name = 'cystathione diffusion'\n",
    "rxn10398_c.bounds = (-1000., 1000.)\n",
    "rxn10398_c.add_metabolites({\n",
    "    cpd00424_e: -1.0,\n",
    "    cpd00424_c: 1.0 \n",
    "})\n",
    "rxn10398_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn10398_c.annotation['seed.reaction'] = 'rxn10398'\n",
    "rxn10398_c.annotation['bigg.reaction'] = 'CYSTtp'\n",
    "iCdG719.add_reactions([rxn10398_c])\n",
    "iCdG719.reactions.rxn10398_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn10398_c.annotation['seed.reaction'] = 'rxn10398'\n",
    "iCdG719.reactions.rxn10398_c.annotation['bigg.reaction'] = 'CYSTtp'\n",
    "\n",
    "# LL-2,6-Diaminopimelate\n",
    "cpd00504_e = cobra.Metabolite(\n",
    "    'cpd00504_e',\n",
    "    formula='C7H14N2O4',\n",
    "    name='LL-2,6-Diaminopimelate',\n",
    "    compartment='extracellular')\n",
    "cpd00504_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00504_e.annotation['seed.compound'] = 'cpd00504'\n",
    "cpd00504_e.annotation['bigg.metabolite'] = '26dap_LL'\n",
    "cpd00504_e.annotation['kegg.compound'] = 'C00666'\n",
    "iCdG719.add_boundary(cpd00504_e, type='exchange', reaction_id='EX_cpd00504_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00504_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "cpd00001_c = iCdG719.metabolites.cpd00001_c\n",
    "cpd00002_c = iCdG719.metabolites.cpd00002_c\n",
    "cpd00008_c = iCdG719.metabolites.cpd00008_c\n",
    "cpd00009_c = iCdG719.metabolites.cpd00009_c\n",
    "cpd00067_c = iCdG719.metabolites.cpd00067_c\n",
    "cpd00504_c = iCdG719.metabolites.cpd00504_c\n",
    "\n",
    "ID011_c = cobra.Reaction('ID011_c')\n",
    "ID011_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "ID011_c.gene_reaction_rule = '272563.120.peg.772'\n",
    "ID011_c.bounds = (0., 1000.)\n",
    "ID011_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00504_e: -1.0,\n",
    "    cpd00504_c: 1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0 \n",
    "})\n",
    "iCdG719.add_reactions([ID011_c])\n",
    "iCdG719.reactions.ID011_c.annotation['sbo'] = 'SBO:0000185'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Guanine\n",
    "rxn05203_c = copy.deepcopy(universal_GramPos.reactions.rxn05203_c) \n",
    "rxn05203_c.name = 'Guanine symport'\n",
    "rxn05203_c.annotation['sbo'] = 'SBO:0000185'\n",
    "rxn05203_c.annotation['seed.reaction'] = 'rxn05203'\n",
    "rxn05203_c.annotation['bigg.reaction'] = ['FCY21_3','FCY22_3','FCY2_3','GUAt2','GUAt2pp','GUAt2r','YGL186C_3']\n",
    "\n",
    "iCdG719.add_reactions([rxn05203_c])\n",
    "cpd00207_e = iCdG719.metabolites.cpd00207_e\n",
    "iCdG719.add_boundary(cpd00207_e, type='exchange', reaction_id='EX_cpd00207_e', lb=-1000.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd00207_e.annotation['sbo'] = 'SBO:0000627'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 719 genes\n",
      "18.93 minutes doubling time\n",
      "34.22% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% reactions are mass imbalanced\n",
      "15.2% reactions without GPRs\n",
      "24.23% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdG719_rmGenes = basicCheck(iCdG719)\n",
    "iCdG719_noGPRblocked = blockedReactions(iCdG719)\n",
    "iCdG719_free = checkFreeMass(iCdG719)\n",
    "iCdG719_massImbal = checkBalance(iCdG719)\n",
    "iCdG719_nogene = missingGPR(iCdG719)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_model(iCdG719, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG719', media=mdm)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Integrate CCDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG719 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG719.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd630_genes = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/Cd630.PATRIC.genes.format.fasta','r') as fasta:\n",
    "    for line in fasta:\n",
    "        cd630_genes[line.split('|')[0]] = line.split('|')[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes found: 653\n"
     ]
    }
   ],
   "source": [
    "success = 0\n",
    "for gene in iCdG719.genes:\n",
    "    try:\n",
    "        gene.annotation['ccds'] = cd630_genes[gene.id]\n",
    "        success += 1\n",
    "    except KeyError:\n",
    "            pass\n",
    "print('Genes found: ' + str(success))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "import copy\n",
    "\n",
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "universal_GramPos.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13783_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13784_c.remove_from_model(remove_orphans=True)\n",
    "#universal_GramPos.reactions.rxn11920_c.name = 'Poly-beta-hydroxybutyrate (PHB) synthease'\n",
    "universal_GramPos.reactions.rxn12188_c.name = '2-deoxyglucose-6-phosphatase'\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "    if cpd.compartment == 'c':\n",
    "        cpd.compartment = 'cytosol'\n",
    "    elif cpd.compartment == 'e':\n",
    "        cpd.compartment = 'extracellular'\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix bounds\n",
    "#iCdR700.reactions.rxn00926_c.bounds = (0.,1000.) # tough...\n",
    "iCdG719.reactions.rxn01021_c.bounds = (0.,1000.)\n",
    "iCdG719.reactions.rxn00143_c.bounds = (0.,1000.)\n",
    "\n",
    "# Adenine synthesis\n",
    "rxn00131_c = copy.deepcopy(universal_GramPos.reactions.rxn00131_c) \n",
    "iCdG719.add_reactions([rxn00131_c])\n",
    "iCdG719.reactions.rxn00131_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn00131_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG719.reactions.rxn00131_c.annotation['seed.reaction'] = 'rxn00131'\n",
    "iCdG719.reactions.rxn00131_c.annotation['bigg.reaction'] = 'AMP-NUCLEOSID-RXN'\n",
    "iCdG719.reactions.rxn00131_c.annotation['kegg.reaction'] = 'R00182'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix last genes\n",
    "geneDict = {'645463.3.peg.3332':'272563.8.peg.3431', '645463.3.peg.3330':'272563.8.peg.3429', '645463.3.peg.3333':'272563.8.peg.3432', \n",
    "            '645463.3.peg.467':'272563.8.peg.518', '645463.3.peg.1000':'272563.8.peg.1130', '645463.3.peg.303':'272563.8.peg.304', \n",
    "            '645463.3.peg.304':'272563.8.peg.305', '645463.3.peg.466':'272563.8.peg.517', '645463.3.peg.998':'272563.8.peg.1128', \n",
    "            '645463.3.peg.465':'272563.8.peg.516', '645463.3.peg.1002':'272563.8.peg.1132', '645463.3.peg.306':'272563.8.peg.307', \n",
    "            '645463.3.peg.1001':'272563.8.peg.1131', '645463.3.peg.308':'272563.8.peg.309', '645463.3.peg.464':'272563.8.peg.515', \n",
    "            '645463.3.peg.307':'272563.8.peg.308', '645463.3.peg.3331':'272563.8.peg.3430', '645463.3.peg.2802':'272563.8.peg.2882',\n",
    "            '645463.3.peg.3106':'272563.8.peg.3245'}\n",
    "\n",
    "cobra.manipulation.modify.rename_genes(iCdG719, geneDict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'272563.8' is not a valid SBML 'SId'.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['rxn05547_c', 'rxn05540_c', 'rxn05535_c', 'rxn05543_c', 'rxn05537_c', 'rxn05539_c', 'rxn05542_c', 'rxn05533_c', 'rxn05544_c']\n"
     ]
    }
   ],
   "source": [
    "# Include missing genes\n",
    "\n",
    "str630_only = ['272563.8.peg.1','272563.8.peg.2','272563.8.peg.3','272563.8.peg.4','272563.8.peg.5',\n",
    "               '272563.8.peg.6','272563.8.peg.7','272563.8.peg.8','272563.8.peg.220','272563.8.peg.229','272563.8.peg.230','272563.8.peg.233','272563.8.peg.259','272563.8.peg.260','272563.8.peg.261','272563.8.peg.262','272563.8.peg.352','272563.8.peg.376','272563.8.peg.377','272563.8.peg.392','272563.8.peg.400','272563.8.peg.402','272563.8.peg.407','272563.8.peg.427','272563.8.peg.429','272563.8.peg.449','272563.8.peg.450','272563.8.peg.451','272563.8.peg.456','272563.8.peg.458','272563.8.peg.464','272563.8.peg.476','272563.8.peg.477','272563.8.peg.530','272563.8.peg.536','272563.8.peg.595','272563.8.peg.605','272563.8.peg.611','272563.8.peg.615','272563.8.peg.633','272563.8.peg.638','272563.8.peg.650','272563.8.peg.651','272563.8.peg.662','272563.8.peg.679','272563.8.peg.685','272563.8.peg.713','272563.8.peg.895','272563.8.peg.944','272563.8.peg.945','272563.8.peg.946','272563.8.peg.949','272563.8.peg.950','272563.8.peg.951','272563.8.peg.953','272563.8.peg.954','272563.8.peg.955','272563.8.peg.956','272563.8.peg.957','272563.8.peg.958','272563.8.peg.959','272563.8.peg.960','272563.8.peg.961','272563.8.peg.962','272563.8.peg.963','272563.8.peg.964','272563.8.peg.965','272563.8.peg.966','272563.8.peg.967','272563.8.peg.968','272563.8.peg.969','272563.8.peg.970','272563.8.peg.971','272563.8.peg.973','272563.8.peg.974','272563.8.peg.975','272563.8.peg.976','272563.8.peg.977','272563.8.peg.978','272563.8.peg.980','272563.8.peg.981','272563.8.peg.982','272563.8.peg.984','272563.8.peg.985','272563.8.peg.986','272563.8.peg.987','272563.8.peg.988','272563.8.peg.993','272563.8.peg.1007','272563.8.peg.1008','272563.8.peg.1009','272563.8.peg.1021','272563.8.peg.1025','272563.8.peg.1026','272563.8.peg.1027','272563.8.peg.1057','272563.8.peg.1146','272563.8.peg.1154','272563.8.peg.1157','272563.8.peg.1158','272563.8.peg.1159','272563.8.peg.1161','272563.8.peg.1162','272563.8.peg.1163','272563.8.peg.1164','272563.8.peg.1165','272563.8.peg.1166','272563.8.peg.1168','272563.8.peg.1172','272563.8.peg.1300','272563.8.peg.1380','272563.8.peg.1571','272563.8.peg.1622','272563.8.peg.1733','272563.8.peg.1798','272563.8.peg.1827','272563.8.peg.1828','272563.8.peg.1854','272563.8.peg.1910','272563.8.peg.1937','272563.8.peg.1952','272563.8.peg.1953','272563.8.peg.1954','272563.8.peg.1979','272563.8.peg.1980','272563.8.peg.1983','272563.8.peg.1998','272563.8.peg.2080','272563.8.peg.2100','272563.8.peg.2101','272563.8.peg.2103','272563.8.peg.2110','272563.8.peg.2114','272563.8.peg.2116','272563.8.peg.2141','272563.8.peg.2144','272563.8.peg.2146','272563.8.peg.2297','272563.8.peg.2335','272563.8.peg.2336','272563.8.peg.2337','272563.8.peg.2340','272563.8.peg.2399','272563.8.peg.2402','272563.8.peg.2407','272563.8.peg.2409','272563.8.peg.2414','272563.8.peg.2418','272563.8.peg.2640','272563.8.peg.2650','272563.8.peg.2651','272563.8.peg.2729','272563.8.peg.2730','272563.8.peg.2731','272563.8.peg.2732','272563.8.peg.2752','272563.8.peg.2790','272563.8.peg.2793','272563.8.peg.2849','272563.8.peg.2915','272563.8.peg.3020','272563.8.peg.3021','272563.8.peg.3022','272563.8.peg.3026','272563.8.peg.3038','272563.8.peg.3039','272563.8.peg.3040','272563.8.peg.3054','272563.8.peg.3059','272563.8.peg.3060','272563.8.peg.3061','272563.8.peg.3062','272563.8.peg.3063','272563.8.peg.3066','272563.8.peg.3068','272563.8.peg.3069','272563.8.peg.3070','272563.8.peg.3071','272563.8.peg.3072','272563.8.peg.3073','272563.8.peg.3076','272563.8.peg.3077','272563.8.peg.3078','272563.8.peg.3079','272563.8.peg.3080','272563.8.peg.3081','272563.8.peg.3082','272563.8.peg.3084','272563.8.peg.3086','272563.8.peg.3087','272563.8.peg.3088','272563.8.peg.3089','272563.8.peg.3090','272563.8.peg.3091','272563.8.peg.3092','272563.8.peg.3096','272563.8.peg.3161','272563.8.peg.3200','272563.8.peg.3265','272563.8.peg.3289','272563.8.peg.3291','272563.8.peg.3292','272563.8.peg.3293','272563.8.peg.3294','272563.8.peg.3299','272563.8.peg.3399','272563.8.peg.3455','272563.8.peg.3457','272563.8.peg.3481','272563.8.peg.3485','272563.8.peg.3487','272563.8.peg.3488','272563.8.peg.3494','272563.8.peg.3496','272563.8.peg.3497','272563.8.peg.3500','272563.8.peg.3504','272563.8.peg.3505','272563.8.peg.3531','272563.8.peg.3538','272563.8.peg.3542','272563.8.peg.3548','272563.8.peg.3550','272563.8.peg.3808','272563.8.peg.3851']\n",
    "\n",
    "cd630_PATRIC = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/272563.8.xml')\n",
    "for cpd in cd630_PATRIC.metabolites:\n",
    "    if cpd.compartment == 'c':\n",
    "        cpd.compartment = 'cytosol'\n",
    "    elif cpd.compartment == 'e':\n",
    "        cpd.compartment = 'extracellular'\n",
    "\n",
    "new_rxns = []\n",
    "for gene in str630_only:\n",
    "    try:\n",
    "        rxns = [x.id for x in cd630_PATRIC.genes.get_by_id(gene).reactions]\n",
    "        for x in rxns:\n",
    "            try:\n",
    "                rxn = iCdG719.reactions.get_by_id(x)\n",
    "            except:\n",
    "                new_rxns.append(x)\n",
    "                pass\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "print(new_rxns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add missing reactions\n",
    "\n",
    "rxn05542_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05542_c) \n",
    "iCdG719.add_reactions([rxn05542_c])\n",
    "iCdG719.reactions.rxn05542_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05542_c.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "iCdG719.reactions.rxn05542_c.annotation['seed.reaction'] = 'rxn05542'\n",
    "iCdG719.reactions.rxn05542_c.annotation['bigg.reaction'] = 'DIPEPabc4'\n",
    "\n",
    "cpd11585_e = iCdG719.metabolites.cpd11585_e\n",
    "iCdG719.add_boundary(cpd11585_e, type='exchange', reaction_id='EX_cpd11585_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11585_e.name = 'Ala-Gly exchange'\n",
    "iCdG719.reactions.EX_cpd11585_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12641_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12641_c) \n",
    "iCdG719.add_reactions([rxn12641_c])\n",
    "iCdG719.reactions.rxn12641_c.name = 'Ala-Gly aminopeptidase'\n",
    "iCdG719.reactions.rxn12641_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12641_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn12641_c.annotation['seed.reaction'] = 'rxn12641'\n",
    "iCdG719.reactions.rxn12641_c.annotation['bigg.reaction'] = 'MEAMP1(ala-gly)'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05547_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05547_c) \n",
    "iCdG719.add_reactions([rxn05547_c])\n",
    "iCdG719.reactions.rxn05547_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05547_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05547_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05547_c.annotation['seed.reaction'] = 'rxn05547'\n",
    "iCdG719.reactions.rxn05547_c.annotation['bigg.reaction'] = 'DIPEPabc9'\n",
    "\n",
    "cpd11580_e = iCdG719.metabolites.cpd11580_e\n",
    "iCdG719.add_boundary(cpd11580_e, type='exchange', reaction_id='EX_cpd11580_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11580_e.name = 'Gly-Gln exchange'\n",
    "iCdG719.reactions.EX_cpd11580_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12646_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12646_c) \n",
    "iCdG719.add_reactions([rxn12646_c])\n",
    "iCdG719.reactions.rxn12646_c.name = 'Gly-Gln aminopeptidase'\n",
    "iCdG719.reactions.rxn12646_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12646_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG719.reactions.rxn12646_c.annotation['seed.reaction'] = 'rxn12646'\n",
    "iCdG719.reactions.rxn12646_c.annotation['bigg.reaction'] = 'DIPEPabc2'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05540_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05540_c) \n",
    "iCdG719.add_reactions([rxn05540_c])\n",
    "iCdG719.reactions.rxn05540_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05540_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05540_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05540_c.annotation['seed.reaction'] = 'rxn05540'\n",
    "iCdG719.reactions.rxn05540_c.annotation['bigg.reaction'] = 'DIPEPabc2'\n",
    "\n",
    "cpd11587_e = iCdG719.metabolites.cpd11587_e\n",
    "iCdG719.add_boundary(cpd11587_e, type='exchange', reaction_id='EX_cpd11587_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11587_e.name = 'Ala-Gln exchange'\n",
    "iCdG719.reactions.EX_cpd11587_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12639_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12639_c) \n",
    "iCdG719.add_reactions([rxn12639_c])\n",
    "iCdG719.reactions.rxn12639_c.name = 'Ala-Gln aminopeptidase'\n",
    "iCdG719.reactions.rxn12639_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12639_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG719.reactions.rxn12639_c.annotation['seed.reaction'] = 'rxn12639'\n",
    "iCdG719.reactions.rxn12639_c.annotation['bigg.reaction'] = 'MEAMP1(ala-gln)'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05537_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05537_c) \n",
    "iCdG719.add_reactions([rxn05537_c])\n",
    "iCdG719.reactions.rxn05537_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05537_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05537_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05537_c.annotation['seed.reaction'] = 'rxn05537'\n",
    "iCdG719.reactions.rxn05537_c.annotation['bigg.reaction'] = 'DIPEPabc13'\n",
    "\n",
    "cpd11589_e = iCdG719.metabolites.cpd11589_e\n",
    "iCdG719.add_boundary(cpd11589_e, type='exchange', reaction_id='EX_cpd11589_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11589_e.name = 'Gly-Asp exchange'\n",
    "iCdG719.reactions.EX_cpd11589_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12637_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12637_c) \n",
    "iCdG719.add_reactions([rxn12637_c])\n",
    "iCdG719.reactions.rxn12637_c.name = 'Gly-Asp aminopeptidase'\n",
    "iCdG719.reactions.rxn12637_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12637_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG719.reactions.rxn12637_c.annotation['seed.reaction'] = 'rxn12637'\n",
    "iCdG719.reactions.rxn12637_c.annotation['bigg.reaction'] = ['AMPEP13','MEAMP1(gly-asp)']\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05543_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05543_c) \n",
    "iCdG719.add_reactions([rxn05543_c])\n",
    "iCdG719.reactions.rxn05543_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05543_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05543_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05543_c.annotation['seed.reaction'] = 'rxn05543'\n",
    "iCdG719.reactions.rxn05543_c.annotation['bigg.reaction'] = 'DIPEPabc5'\n",
    "\n",
    "cpd11584_e = iCdG719.metabolites.cpd11584_e\n",
    "iCdG719.add_boundary(cpd11584_e, type='exchange', reaction_id='EX_cpd11584_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11584_e.name = 'Ala-His exchange'\n",
    "iCdG719.reactions.EX_cpd11584_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12642_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12642_c) \n",
    "iCdG719.add_reactions([rxn12642_c])\n",
    "iCdG719.reactions.rxn12642_c.name = 'Ala-His aminopeptidase'\n",
    "iCdG719.reactions.rxn12642_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12642_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn12642_c.annotation['seed.reaction'] = 'rxn12642'\n",
    "iCdG719.reactions.rxn12642_c.annotation['bigg.reaction'] = 'MEAMP1(ala-his)'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05539_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05539_c) \n",
    "iCdG719.add_reactions([rxn05539_c])\n",
    "iCdG719.reactions.rxn05539_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05539_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05539_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05539_c.annotation['seed.reaction'] = 'rxn05539'\n",
    "iCdG719.reactions.rxn05539_c.annotation['bigg.reaction'] = ['CGLYabcpp','DIPEPabc15']\n",
    "\n",
    "cpd01017_e = iCdG719.metabolites.cpd01017_e\n",
    "iCdG719.add_boundary(cpd01017_e, type='exchange', reaction_id='EX_cpd01017_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd01017_e.name = 'Cys-Gly exchange'\n",
    "iCdG719.reactions.EX_cpd01017_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn00650_c = copy.deepcopy(cd630_PATRIC.reactions.rxn00650_c) \n",
    "iCdG719.add_reactions([rxn00650_c])\n",
    "iCdG719.reactions.rxn00650_c.name = 'Cys-Gly aminopeptidase'\n",
    "iCdG719.reactions.rxn00650_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn00650_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn00650_c.annotation['seed.reaction'] = 'rxn00650'\n",
    "iCdG719.reactions.rxn00650_c.annotation['bigg.reaction'] = ['AMPTASECG','TRIA']\n",
    "iCdG719.reactions.rxn00650_c.annotation['kegg.reaction'] = 'R00899'\n",
    "\n",
    "rxn00350_c = copy.deepcopy(cd630_PATRIC.reactions.rxn00350_c) \n",
    "iCdG719.add_reactions([rxn00350_c])\n",
    "iCdG719.reactions.rxn00350_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn00350_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn00350_c.annotation['seed.reaction'] = 'rxn00350'\n",
    "iCdG719.reactions.rxn00350_c.annotation['bigg.reaction'] = ['GGTAe2','GTHRDHpp']\n",
    "iCdG719.reactions.rxn00350_c.annotation['kegg.reaction'] = 'R00494'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05535_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05535_c) \n",
    "iCdG719.add_reactions([rxn05535_c])\n",
    "iCdG719.reactions.rxn05535_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05535_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05535_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05535_c.annotation['seed.reaction'] = 'rxn05535'\n",
    "iCdG719.reactions.rxn05535_c.annotation['bigg.reaction'] = 'DIPEPabc11'\n",
    "\n",
    "cpd11591_e = iCdG719.metabolites.cpd11591_e\n",
    "iCdG719.add_boundary(cpd11591_e, type='exchange', reaction_id='EX_cpd11591_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11591_e.name = 'Gly-Met exchange'\n",
    "iCdG719.reactions.EX_cpd11591_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12635_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12635_c) \n",
    "iCdG719.add_reactions([rxn12635_c])\n",
    "iCdG719.reactions.rxn12635_c.name = 'Gly-Met aminopeptidase'\n",
    "iCdG719.reactions.rxn12635_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12635_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn12635_c.annotation['seed.reaction'] = 'rxn12635'\n",
    "iCdG719.reactions.rxn12635_c.annotation['bigg.reaction'] = 'MEAMP1(gly-met)'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05544_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05544_c) \n",
    "iCdG719.add_reactions([rxn05544_c])\n",
    "iCdG719.reactions.rxn05544_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05544_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05544_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05544_c.annotation['seed.reaction'] = 'rxn05544'\n",
    "iCdG719.reactions.rxn05544_c.annotation['bigg.reaction'] = 'DIPEPabc6'\n",
    "\n",
    "cpd11583_e = iCdG719.metabolites.cpd11583_e\n",
    "iCdG719.add_boundary(cpd11583_e, type='exchange', reaction_id='EX_cpd11583_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11583_e.name = 'Ala-Leu exchange'\n",
    "iCdG719.reactions.EX_cpd11583_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12643_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12643_c) \n",
    "iCdG719.add_reactions([rxn12643_c])\n",
    "iCdG719.reactions.rxn12643_c.name = 'Ala-Leu aminopeptidase'\n",
    "iCdG719.reactions.rxn12643_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12643_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn12643_c.annotation['seed.reaction'] = 'rxn12643'\n",
    "iCdG719.reactions.rxn12643_c.annotation['bigg.reaction'] = 'MEAMP1(ala-leu)'\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "rxn05533_c = copy.deepcopy(cd630_PATRIC.reactions.rxn05533_c) \n",
    "iCdG719.add_reactions([rxn05533_c])\n",
    "iCdG719.reactions.rxn05533_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn05533_c.name = 'oligopeptide-transporting ATPase'\n",
    "iCdG719.reactions.rxn05533_c.annotation['sbo'] = 'SBO:0000185'\n",
    "iCdG719.reactions.rxn05533_c.annotation['seed.reaction'] = 'rxn05533'\n",
    "iCdG719.reactions.rxn05533_c.annotation['bigg.reaction'] = 'DIPEPabc1'\n",
    "\n",
    "cpd11593_e = iCdG719.metabolites.cpd11593_e\n",
    "iCdG719.add_boundary(cpd11593_e, type='exchange', reaction_id='EX_cpd11593_e', lb=0.0, ub=1000.0)\n",
    "iCdG719.reactions.EX_cpd11593_e.name = 'Ala-Asp exchange'\n",
    "iCdG719.reactions.EX_cpd11593_e.annotation['sbo'] = 'SBO:0000627'\n",
    "\n",
    "rxn12633_c = copy.deepcopy(cd630_PATRIC.reactions.rxn12633_c) \n",
    "iCdG719.add_reactions([rxn12633_c])\n",
    "iCdG719.reactions.rxn12633_c.name = 'Ala-Asp aminopeptidase'\n",
    "iCdG719.reactions.rxn12633_c.bounds = (0., 1000.)\n",
    "iCdG719.reactions.rxn12633_c.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "iCdG719.reactions.rxn12633_c.annotation['seed.reaction'] = 'rxn12633'\n",
    "iCdG719.reactions.rxn12633_c.annotation['bigg.reaction'] = 'MEAMP1(ala-asp)'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG719.id = 'iCdG' + str(len(iCdG719.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG700</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f3aff278550</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1235</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1308</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG700 at 0x7f3aff278550>"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG719"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "mdm = ['EX_' + x for x in mdm]\n",
    "for rxn in iCdG719.boundary:\n",
    "    rxn.bounds = (-1000., 1000.)\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "80.96607010298041"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG719.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_model(iCdG719, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG700')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 700 genes\n",
      "18.93 minutes doubling time\n",
      "33.49% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% reactions are mass imbalanced\n",
      "15.1% reactions without GPRs\n",
      "25.66% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdG700_rmGenes = basicCheck(iCdG700)\n",
    "iCdG700_noGPRblocked = blockedReactions(iCdG700)\n",
    "iCdG700_free = checkFreeMass(iCdG700)\n",
    "iCdG700_massImbal = checkBalance(iCdG700)\n",
    "iCdG700_nogene = missingGPR(iCdG700)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "for cpd in iCdG700.metabolites: cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "iCdG700.reactions.rxn18658_c.annotation['sbo'] = 'SBO:0000185'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd630_genes = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/Cd630.PATRIC.genes.format.fasta','r') as fasta:\n",
    "    for line in fasta:\n",
    "        cd630_genes[line.split('|')[0]] = line.split('|')[1]\n",
    "        \n",
    "pepto630_genes = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/Pepto630.PATRIC.genes.format.fasta','r') as fasta:\n",
    "    for line in fasta:\n",
    "        pepto630_genes[line.split('|')[0]] = line.split('|')[1]        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes found: 653\n"
     ]
    }
   ],
   "source": [
    "success = 0\n",
    "failed = []\n",
    "for gene in iCdG700.genes:\n",
    "    try:\n",
    "        gene.annotation['ccds'] = cd630_genes[gene.id]\n",
    "        success += 1\n",
    "    except KeyError:\n",
    "        failed.append(gene.id)\n",
    "        pass\n",
    "print('Genes found: ' + str(success))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes found: 47\n"
     ]
    }
   ],
   "source": [
    "success = 0\n",
    "failed = []\n",
    "for gene in iCdG700.genes:\n",
    "    try:\n",
    "        gene.annotation['ccds'] = pepto630_genes[gene.id]\n",
    "        success += 1\n",
    "    except KeyError:\n",
    "        failed.append(gene.id)\n",
    "        pass\n",
    "print('Genes found: ' + str(success))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG700.metabolites.cpd12255_c_c.id = 'cpd12255_c'\n",
    "iCdG700.reactions.rxn00139_c.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(iCdG700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG700.sbml')\n",
    "cobra.io.save_json_model(iCdG700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "patric password: ········\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'mljenior@patricbrc.org'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create Peptoclostridium draft GENRE\n",
    "import cobra\n",
    "import mackinac\n",
    "mackinac.modelseed.ms_client.url = 'http://p3.theseed.org/services/ProbModelSEED/'\n",
    "mackinac.workspace.ws_client.url = 'http://p3.theseed.org/services/Workspace'\n",
    "mackinac.genome.patric_url = 'https://www.patricbrc.org/api/'\n",
    "\n",
    "# PATRIC user information\n",
    "mackinac.get_token('mljenior')\n",
    "# password: matrix54"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_id = '272563.8'\n",
    "template_id = '/chenry/public/modelsupport/templates/GramPosModelTemplate'\n",
    "media_id = '/chenry/public/modelsupport/media/Complete'\n",
    "file_id = '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/Cd630.draft.json'\n",
    "strain_id = 'Clostridium difficile 630'\n",
    "\n",
    "mackinac.reconstruct_modelseed_model(genome_id, template_reference=template_id)\n",
    "mackinac.gapfill_modelseed_model(genome_id, media_reference=media_id)\n",
    "mackinac.optimize_modelseed_model(genome_id)\n",
    "patric_draft = mackinac.create_cobra_model_from_modelseed_model(genome_id)\n",
    "patric_draft.id = strain_id\n",
    "\n",
    "cobra.io.save_json_model(patric_draft, file_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridium difficile 630</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f651b295350</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1235</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1136</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*bio1 - 1.0*bio1_reverse_b18f7</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>c, e</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridium difficile 630 at 0x7f651b295350>"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_draft"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 genes\n",
      "{'272563.8.peg.1781', '272563.8.peg.3769', '272563.8.peg.795', '272563.8.peg.1176', '272563.8.peg.2251', '272563.8.peg.1782', '272563.8.peg.3436'}\n"
     ]
    }
   ],
   "source": [
    "draft_genes = set([x.id for x in patric_draft.genes])\n",
    "current_genes = set([y.id for y in iCdG700.genes])\n",
    "\n",
    "new_genes = draft_genes.difference(current_genes)\n",
    "\n",
    "current_rxns = set([z.id for z in iCdG700.reactions])\n",
    "save_genes = []\n",
    "for x in new_genes:\n",
    "    rxns = set([i.id for i in patric_draft.genes.get_by_id(x).reactions])\n",
    "    \n",
    "    if len(rxns.intersection(current_rxns)) > 0:\n",
    "        save_genes.append(x)\n",
    "save_genes = set(save_genes)\n",
    "print(str(len(save_genes)) + ' genes')\n",
    "print(save_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Integrate to existing reactions\n",
    "iCdG700.reactions.rxn00157_c.gene_reaction_rule = '( ( 272563.8.peg.3435 or 272563.8.peg.1175 or 272563.8.peg.796 ) or ( 272563.8.peg.795 or 272563.8.peg.1176 or 272563.8.peg.3436 ) )'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.795').name = 'Pyruvate formate-lyase activating enzyme (EC 1.97.1.4)'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.1176').name = 'Pyruvate formate-lyase activating enzyme (EC 1.97.1.4)'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.3436').name = 'Pyruvate formate-lyase activating enzyme (EC 1.97.1.4)'\n",
    "\n",
    "iCdG700.reactions.rxn00119_c.gene_reaction_rule += ' or 272563.8.peg.2251'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.2251').name = 'Uridylate kinase (EC 2.7.4.-)'\n",
    "\n",
    "iCdG700.reactions.rxn09310_c.gene_reaction_rule = '272563.8.peg.1781 and 272563.8.peg.1782'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.1781').name = 'Sulfur carrier protein ThiS subunit'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.1782').name = 'Sulfur carrier protein ThiS subunit'\n",
    "\n",
    "iCdG700.reactions.rxn00693_c.gene_reaction_rule += ' or 272563.8.peg.3769'\n",
    "iCdG700.genes.get_by_id('272563.8.peg.3769').name = 'Methionine synthase activation domain (EC 2.1.1.13)'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprot_gene_ids = {}\n",
    "ec_gene_ids = {}\n",
    "refseq_gene_ids = {}\n",
    "kegg_gene_ids = {}\n",
    "patric_genes = set()\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/uniprot_Cd630.tab', 'r') as gene_annotation:\n",
    "    header = gene_annotation.readline()\n",
    "    for line in gene_annotation:\n",
    "        line = line.split()\n",
    "        uniprot_gene_ids[line[4]] = line[0]\n",
    "        ec_gene_ids[line[4]] = line[1]\n",
    "        refseq_gene_ids[line[4]] = line[2]\n",
    "        kegg_gene_ids[line[4]] = line[3]\n",
    "        patric_genes |= set([line[4]])\n",
    "\n",
    "cd630_genes = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/Cd630.PATRIC.genes.format.fasta','r') as fasta:\n",
    "    for line in fasta:\n",
    "        cd630_genes[line.split('|')[0]] = line.split('|')[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct labels and annotation data\n",
    "for gene in iCdG700.genes:\n",
    "    gene.annotation['sbo'] = 'SBO:0000243'\n",
    "    \n",
    "    gene.annotation['patric'] = 'fig|' + gene.id\n",
    "    try:\n",
    "        gene.annotation['uniprot'] = uniprot_gene_ids[gene.id]\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        gene.annotation['kegg.genes'] = kegg_gene_ids[gene.id]\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        gene.annotation['refseq'] = refseq_gene_ids[gene.id]\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        gene.annotation['ec_number'] = ec_gene_ids[gene.id]\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        gene.annotation['ccds'] = cd630_genes[gene.id]\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "for rxn in patric_draft.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000185'\n",
    "    \n",
    "for cpd in patric_draft.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    \n",
    "    if cpd.compartment == 'c':\n",
    "        cpd.compartment = 'cytosol'\n",
    "    elif cpd.compartment == 'e':\n",
    "        cpd.compartment = 'extracellular'\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "pepto_only = []\n",
    "pepto_dict = {}\n",
    "with open('/home/mjenior/Desktop/pepto630_vs_cd630.patric_id.tsv', 'r') as alignment:\n",
    "    header = alignment.readline()\n",
    "    for line in alignment:\n",
    "        line = line.split()\n",
    "        if line[1] == 'none':\n",
    "            pepto_only.append(line[0])\n",
    "        else:\n",
    "            pepto_dict[line[0]] = line[1]\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG700_peptogenes = []\n",
    "for x in iCdG700.genes:\n",
    "    if '272563.120.' in x.id:\n",
    "        iCdG700_peptogenes.append(x.id)\n",
    "iCdG700_peptogenes = set(iCdG700_peptogenes)\n",
    "\n",
    "replace_genes = {}\n",
    "for x in iCdG700_peptogenes:\n",
    "    replace_genes[x] = pepto_dict[x]\n",
    "    \n",
    "cobra.manipulation.modify.rename_genes(iCdG700, replace_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG700.id = 'iCdG' + str(len(iCdG700.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG692</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f64ce6c61d0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1235</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1308</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG692 at 0x7f64ce6c61d0>"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG700"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 692 genes\n",
      "18.48 minutes doubling time\n",
      "33.49% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% reactions are mass imbalanced\n",
      "14.92% reactions without GPRs\n",
      "24.63% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdG700_rmGenes = basicCheck(iCdG700)\n",
    "iCdG700_noGPRblocked = blockedReactions(iCdG700)\n",
    "iCdG700_free = checkFreeMass(iCdG700)\n",
    "iCdG700_massImbal = checkBalance(iCdG700)\n",
    "iCdG700_nogene = missingGPR(iCdG700)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "mdm = ['EX_' + x for x in mdm]\n",
    "for rxn in iCdG700.boundary:\n",
    "    rxn.bounds = (-1000., 1000.)\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(iCdG700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG692.sbml')\n",
    "cobra.io.save_json_model(iCdG700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG692.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
