{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "from cobra.flux_analysis import parsimonious\n",
    "\n",
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "\n",
    "    #print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    fraction = (float(len(blocked)) / float(len(model.reactions))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions are blocked')\n",
    "    \n",
    "    return blocked\n",
    "\n",
    "\n",
    "# Identify potentially gapfilled reactions, checks against pFBA solution\n",
    "def missingGPR(model):\n",
    "    \n",
    "    noGene = []\n",
    "    exclude = []\n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.genes)) == 0:\n",
    "            if rxn in model.boundary:\n",
    "                exclude.append(rxn.id)\n",
    "                continue\n",
    "            else:\n",
    "                noGene.append(rxn.id)\n",
    "    \n",
    "    solution = parsimonious.pfba(model)\n",
    "    active_rxns = set([rxn.id for rxn in model.reactions if abs(solution.fluxes[rxn.id]) > 1e-5])\n",
    "    active_rxns = active_rxns.difference(set(exclude))\n",
    "    noGene_active = set(noGene).intersection(active_rxns)\n",
    "\n",
    "    fraction = float(len(model.reactions)) - float(len(exclude))\n",
    "    fraction = (float(len(noGene)) / fraction) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions without GPRs')\n",
    "    \n",
    "    fraction = (float(len(noGene_active)) / float(len(active_rxns))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% of reactions used in pFBA solution have no GPR')\n",
    "    \n",
    "    return noGene_active\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(model, cytosol='cytosol'):\n",
    "    \n",
    "    #if len(model.compartments.keys()) == 0:\n",
    "    \n",
    "    free = []\n",
    "    with model as m:\n",
    "    \n",
    "        # Close all exchanges\n",
    "        for rxn in m.boundary: rxn.lower_bound = 0.\n",
    "    \n",
    "        # Create demand for each reaction and optimize individually\n",
    "        reset_rxn = m.reactions[0].id\n",
    "        for cpd in m.metabolites: \n",
    "            if cpd.compartment == cytosol:\n",
    "                demand = cobra.Reaction('demand')\n",
    "                demand.bounds = (0., 1000.)\n",
    "                demand.add_metabolites({cpd: -1.0})\n",
    "                m.add_reactions([demand])\n",
    "                m.objective = demand\n",
    "                obj_val = m.slim_optimize()\n",
    "                if obj_val > 1e-8: free.append(cpd.id)\n",
    "                m.objective = reset_rxn\n",
    "                m.remove_reactions([demand])\n",
    "    \n",
    "    fraction = (float(len(free)) / float(len(model.metabolites))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkLostMass(model):\n",
    "\n",
    "    cycled = []\n",
    "    with model as m:\n",
    "    \n",
    "        # Close all exchanges\n",
    "        for rxn in m.boundary: rxn.upper_bound = 0.\n",
    "    \n",
    "        # Create demand for each reaction and optimize individually\n",
    "        for rxn in m.reactions: \n",
    "            m.objective = rxn\n",
    "            if m.slim_optimize() > 1e-6:\n",
    "                for cpd in rxn.products:\n",
    "                    cycled.append(cpd.id)\n",
    "            \n",
    "    cycled = set(cycled)\n",
    "    fraction = (float(len(cycled)) / float(len(model.metabolites))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% metabolites are involved in potential cycles')\n",
    "\n",
    "    return(cycled)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(model):\n",
    "    \n",
    "    with model as m:\n",
    "\n",
    "        elements = set()\n",
    "        for cpd in m.metabolites:\n",
    "            try:\n",
    "                elements |= set(cpd.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        massImbal = []\n",
    "        failed = 0\n",
    "        if len(elements) == 0:\n",
    "            print('No elemental data associated with metabolites!')\n",
    "            failed = 1\n",
    "        else:\n",
    "            for rxn in m.reactions:\n",
    "                if rxn in m.boundary:\n",
    "                    continue\n",
    "\n",
    "                try:\n",
    "                    test = rxn.check_mass_balance()\n",
    "                except ValueError:\n",
    "                    continue\n",
    "\n",
    "                if len(list(test)) > 0:\n",
    "                    if len(set(test.keys()).intersection(elements)) > 0: massImbal.append(rxn.id)\n",
    "                        \n",
    "    if failed != 1:\n",
    "        fraction = (float(len(massImbal)) / float(len(model.reactions))) * 100.\n",
    "        fraction = round(fraction, 2)\n",
    "        print(str(fraction) + '% reactions are mass imbalanced')\n",
    "        \n",
    "    return massImbal\n",
    "\n",
    "\n",
    "def basicCheck(model):\n",
    "    \n",
    "    # Determination\n",
    "    determination = float(len(model.reactions)) / float(len(model.metabolites))\n",
    "    determination = round(determination, 3)\n",
    "    if len(model.reactions) < len(model.metabolites): \n",
    "        statement = ' (overdetermined)'\n",
    "    elif len(model.reactions) > len(model.metabolites):\n",
    "        statement = ' (underdetermined)'\n",
    "    print('Reactions to metabolites ratio: ' + str(determination) + statement)\n",
    "    \n",
    "    # Compartments\n",
    "    print('GENRE has ' + str(len(model.compartments.keys())) + ' compartment(s)')\n",
    "    \n",
    "    # Genes\n",
    "    if len(model.genes) == 0: \n",
    "        print('GENRE has no gene data')\n",
    "    else:\n",
    "        print('GENRE has ' + str(len(model.genes)) + ' genes')\n",
    "    no_rxns = []\n",
    "    for gene in model.genes:\n",
    "          if len(gene.reactions) == 0:\n",
    "                no_rxns.append(gene.id)\n",
    "    if len(no_rxns) > 0:\n",
    "        print('\\t' + str(len(no_rxns)) + ' are not associated with reactions')\n",
    "        \n",
    "    # Growth\n",
    "    ov = model.slim_optimize(error_value=0.)\n",
    "    if ov < 1e-6:\n",
    "        for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        ov = model.slim_optimize(error_value=0.)\n",
    "        if ov < 1e-6:\n",
    "            print('GENRE cannot acheive objective flux')\n",
    "        else:\n",
    "            ov = round(ov, 3)\n",
    "            print(str(ov) + ' objective flux, only in complete media')\n",
    "    else:\n",
    "        ov = round(ov, 3)\n",
    "        print(str(ov) + ' objective flux in current media')\n",
    "\n",
    "# Quicker way to read in models\n",
    "import pickle\n",
    "def read_model(fileName, obj='none'):\n",
    "    \n",
    "    fileType = fileName.split('.')[-1]\n",
    "    \n",
    "    if fileType == 'sbml' or fileType == 'xml':\n",
    "        model = cobra.io.read_sbml_model(fileName)\n",
    "    elif fileType == 'json':\n",
    "        model = cobra.io.load_json_model(fileName)\n",
    "    elif fileType == 'yaml':\n",
    "        model = cobra.io.load_yaml_model(fileName)\n",
    "    elif fileType == 'mat':\n",
    "        model = cobra.io.load_matlab_model(fileName)\n",
    "    elif fileType == 'pkl':\n",
    "        model = pickle.load(open(fileName, 'rb'))\n",
    "    else:\n",
    "        raise TypeError('Unrecognized file extension')\n",
    "    \n",
    "    if obj != 'none': model.objective = obj\n",
    "    for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "    return model\n",
    "\n",
    "def count_components(model):\n",
    "    \n",
    "    metabolic = 0\n",
    "    transporter = 0\n",
    "    exchs = [x.id for x in model.boundary]\n",
    "    for rxn in model.reactions:\n",
    "        if rxn.id in exchs: \n",
    "            continue\n",
    "        else:\n",
    "            substrate_comp = set([x.compartment for x in rxn.reactants])\n",
    "            product_comp = set([x.compartment for x in rxn.products])\n",
    "            if substrate_comp != product_comp:\n",
    "                transporter += 1\n",
    "            else:\n",
    "                metabolic += 1\n",
    "        \n",
    "    print('Exchange reactions:', len(exchs))\n",
    "    print('Transport reactions:', transporter)\n",
    "    print('Metabolic reactions:', metabolic, '\\n')\n",
    "    \n",
    "    compartments = set(model.compartments.keys())\n",
    "    compartment_counts = {}\n",
    "    for x in compartments: \n",
    "        compartment_counts[x] = 0\n",
    "    for cpd in model.metabolites:\n",
    "        compartment_counts[cpd.compartment] += 1\n",
    "    \n",
    "    for x in compartments:\n",
    "        print(x, 'metabolites:', compartment_counts[x])\n",
    "\n",
    "def intracellular_exchanges(model, cytosol='cytosol'):\n",
    "    \n",
    "    intra_exch = []\n",
    "    for rxn in model.boundary:\n",
    "        compartments = set([x.compartment for x in rxn.metabolites])\n",
    "        if cytosol in compartments:\n",
    "            intra_exch.append(rxn.id)\n",
    "        \n",
    "    print('Intracellular exchange reactions:', len(intra_exch))\n",
    "    return intra_exch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Previous Reconstructions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### iMLTC806cdf\n",
    "Larocque, M., Chénard, T., & Najmanovich, R. (2014). A curated C. difficile strain 630 metabolic network: prediction of essential targets and inhibitors. BMC Systems Biology, 8, 117. http://doi.org/10.1186/s12918-014-0117-z\n",
    "\n",
    "#### icdf834\n",
    "Kashaf, S. S., Angione, C., & Lió, P. (2017). Making life difficult for Clostridium difficile: augmenting the pathogen’s metabolic model with transcriptomic and codon usage data for better therapeutic target characterization. BMC Systems Biology, 11, 25. http://doi.org/10.1186/s12918-017-0395-3\n",
    "\n",
    "#### iHD992\n",
    "Dannheim, H., Will, S. E., Schomburg, D., & Neumann‐Schaal, M. (2017). Clostridioides difficile 630Δerm in silico and in vivo – quantitative growth and extensive polysaccharide secretion. FEBS Open Bio, 7(4), 602–615. http://doi.org/10.1002/2211-5463.12208\n",
    "\n",
    "#### iCN900\n",
    "BiGG model for C. difficile 630\n",
    "\n",
    "#### cd630_PATRIC & cdR20291_PATRIC\n",
    "Draft reconstructions created with automated PATRIC server workflow and converted to ModelSEED namespace using Mackinac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Model does not contain SBML fbc package information.\n",
      "SBML package 'layout' not supported by cobrapy, information is not parsed\n",
      "SBML package 'render' not supported by cobrapy, information is not parsed\n",
      "urn:miriam:Clostridium difficile strain 630 does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pepyr_c \"phosphoenolpyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_idp_c \"IDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyr_c \"pyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_itp_c \"ITP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sprm_c \"spermine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acoa_c \"acetyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_coa_c \"coenzyme A\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gth_c \"glutathione\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2o_c \"H2O\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2o_b \"H2O\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cysgly_c \"cys-gly\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gluL_c \"L-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gluL_b \"L-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ivalcoa_c \"isovaleryl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pi_c \"phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pi_b \"phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isop_c \"isopentanoylphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dtmp_c \"dTMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_atp_c \"ATP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dtdp_c \"dTDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adp_c \"ADP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2alac_c \"2-acetolactate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_da_c \"diacetyl\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_co2_c \"CO2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_co2_b \"CO2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aro_c \"L-arogenate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pheL_c \"L-phenylalanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pheL_b \"L-phenylalanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_utp_c \"UTP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uri5p_c \"uridine-5'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyo_c \"glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyo_b \"glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nad_c \"NAD+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyald_c \"D-glyceraldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nadh_c \"NADH\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h_c \"H+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h_b \"H+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rbl5p_c \"L-ribulose-5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xyl5p_c \"D-xylulose-5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3ipm_c \"(2R,3S)-3-isopropylmalate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2ip3os_c \"(2S)-2-isopropyl-3-oxosuccinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gluD_c \"D-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glcAD_e \"alpha-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glcAD_b \"alpha-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h_e \"H+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glcAD_c \"alpha-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udp_c \"UDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fald_c \"formaldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thf_c \"tetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_510mlethf_c \"5,10-methylenetetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_actid_c \"Acetamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_actid_b \"Acetamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_actid_e \"Acetamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23diap_c \"2,3-diaminopropanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23diap_b \"2,3-diaminopropanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nh3_c \"ammonia\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nh3_b \"ammonia\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2p4c5dp2me_c \"2-phospho-4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2me24cdp_c \"2-C-methyl-D-erythritol-2,4-cyclodiphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cmp_c \"CMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_r3hbat_c \"(R)-3-hydroxybutanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aac_c \"acetoacetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2oa_c \"2-oxoadipate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gcoa_c \"glutaryl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isocapcoa_c \"Isocaproyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isocap_c \"Isocaproate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isocap_b \"Isocaproate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_btal_c \"butanal\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nadph_c \"NADPH\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_buta_c \"n-butanol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nadp_c \"NADP+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hoam_c \"Hydroxylamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hoam_b \"Hydroxylamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hepdp_c \"all-trans-heptaprenyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ipdp_c \"isopentenyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_octdp_c \"all-trans-octaprenyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppi_c \"diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3h3mgcoa_c \"(S)-3-hydroxy-3-methylglutaryl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acgdp_c \"adenosylcobinamide-GDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rz_c \"alpha-ribazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_coB12_c \"coenzyme B12\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gmp_c \"GMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dgtp_c \"dGTP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dgdp_c \"dGDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2o6pamg2d_e \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2o6pamg2d_b \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2o6pamg2d_c \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nacmnc_c \"nicotinate mononucleotide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nacaddn_c \"nicotinate adenine dinucleotide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_leuL_c \"L-leucine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_leuL_b \"L-leucine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2oglut_c \"2-oxoglutarate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4m2op_c \"4-methyl-2-oxopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dump_c \"dUMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_78dhfmg_c \"7,8-dihydrofolate monoglutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cdpg_e \"CDP-glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cdpg_b \"CDP-glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cdpg_c \"CDP-glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_damp_c \"dAMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dad_c \"deoxyadenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dad_b \"deoxyadenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isocaprecoa_c \"2-Isocaprenoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2s6h24ch1c_c \"2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2sucbz_c \"2-succinylbenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_escul_e \"esculin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_escul_b \"esculin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_escul_c \"esculin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pap_c \"adenosine 3',5'-bisphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_laldS_c \"(S)-lactaldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lacS_c \"(S)-lactate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_for_c \"formate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_for_b \"formate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbshclin_c \"Cobalt-sirohydrochlorin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_amet_c \"S-adenosyl-L-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbfIII_c \"Cobalt-factor III\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ahcys_c \"S-adenosyl-L-homocysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pi_e \"phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_na_e \"Na+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_na_b \"Na+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_na_c \"Na+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc1pA_c \"alpha-D-glucose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc_c \"beta-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc_b \"beta-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uri_c \"uridine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uri_b \"uridine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cytd_c \"cytidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abald_c \"4-aminobutyraldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abat_c \"4-Aminobutanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_no2_c \"nitrite\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_no2_b \"nitrite\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lacR_c \"(R)-lactate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uqnn8_c \"ubiquinone-8\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uqnl8_c \"ubiquinol-8\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc6p_c \"beta-D-glucose-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_6pg15l_c \"6-phospho-D-glucono-1,5-lactone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glcn_e \"D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glcn_b \"D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glcn_c \"D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_1h2m2b4dp_c \"1-hydroxy-2-methyl-2-(E)-butenyl 4-diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adcbam_c \"adenosylcobinamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gtp_c \"GTP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adcobamp_c \"adenosylcobinamide-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gdp_c \"GDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_succoa_c \"succinyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sucsald_c \"succinate semialdehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptcys_c \"N-((R)-pantothenoyl)-L-cysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptth_c \"pantetheine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2oiv_c \"2-oxoisovalerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2ipm_c \"(2S)-2-isopropylmalate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_asp4p_c \"L-aspartyl-4-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aspsald_c \"L-aspartate-semialdehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_56dhthy_c \"5,6-dihydrothymine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3uib_c \"3-ureido-isobutyrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc6pA_c \"alpha-D-glucose 6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fru6p_c \"D-fructose-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dhap_c \"dihydroxyacetone phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_g3p_c \"sn-glycerol-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_g3p_b \"sn-glycerol-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3ppyr_c \"3-Phosphonopyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sucr_e \"sucrose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sucr_b \"sucrose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sucr6p_c \"sucrose-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ncamrib_c \"nicotinamide-beta-riboside\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ncam_c \"nicotinamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ncam_b \"nicotinamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ribD_c \"D-ribose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ribD_b \"D-ribose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_abdmagdpu_c \"N-acetyl-beta-D-mannosaminyl-1,4-N-acetyl-D-glucosaminyldiphosphoundecaprenol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lcacoa_c \"long-chain acyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fadh2_c \"FADH2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fad_c \"FAD\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_icit_c \"isocitrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ura_e \"uracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ura_b \"uracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ura_c \"uracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dudp_c \"dUDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ade_c \"adenine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5pr1dp_c \"5-phospho-alpha-D-ribose 1-diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_amp_c \"AMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_man6p_c \"D-mannose 6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_man1p_c \"alpha-D-mannose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_bio_e \"biotin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_bio_b \"biotin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_bio_c \"biotin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dcmp_c \"dCMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dcyt_c \"deoxycytidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dcyt_b \"deoxycytidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_eimzgp_c \"D-erythro-imidazole-glycerol-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_imzap_c \"imidazole acetol-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uppg3_c \"uroporphyrinogen-III\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pc1_c \"precorrin-1\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2dh3dg_c \"2-dehydro-3-deoxy-D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2dh3dg_b \"2-dehydro-3-deoxy-D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2dh3dg6p_c \"2-dehydro-3-deoxy-D-gluconate-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hco3_e \"bicarbonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hco3_b \"bicarbonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hco3_c \"bicarbonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dctp_c \"dCTP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dutp_c \"dUTP\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fardp_c \"(2E,6E)-farnesyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_grgrdp_c \"geranylgeranyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_alaD_c \"D-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_btcoa_c \"butanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ac_c \"acetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ac_b \"acetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2but_c \"2-butanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sh17bp_c \"sedoheptulose 1,7-bisphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_shl7p_c \"D-sedoheptulose-7-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thymd_c \"thymidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thymd_b \"thymidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_78dhpte_c \"7,8-dihydropteroate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hisp_c \"L-histidinol-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_histd_c \"histidinol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_1ag3p_c \"1-acyl-sn-glycerol-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3dhq_c \"3-dehydroquinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3dhsk_c \"3-dehydroshikimate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adn_c \"adenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adn_b \"adenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hicit_c \"homoisocitrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gsn_c \"guanosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gua_c \"guanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptpphygen_c \"protoporphyrinogen IX\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mqn_c \"menaquinone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptphyrin_c \"protoporphyrin IX\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mql_c \"menaquinol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4ppcys_c \"R-4'-phosphopantothenoyl-L-cysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3g12dag_c \"3-D-glucosyl-1,2-diacylglycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpglu_c \"UDP-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dgdag_c \"diglucosyl-diacylglycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cdp_c \"CDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_succ_c \"succinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gly_e \"glycine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gly_b \"glycine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gly_c \"glycine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23dh23dhvz_c \"2,3-dihydro-2,3-dihydroxybenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23dhbz_c \"2,3-dihydroxybenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23dhbz_b \"2,3-dihydroxybenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_asnL_c \"L-asparagine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_asnL_b \"L-asparagine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aspL_c \"L-aspartate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aspL_b \"L-aspartate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sucr_c \"sucrose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fru_c \"D-fructose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fru_b \"D-fructose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cysthL_c \"L-cystathionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hcys_c \"L-homocysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dgsn_c \"deoxyguanosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_drin1p_c \"deoxyribose-1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_skm3p_c \"shikimate-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5epyrsk3p_c \"5-enolpyruvyl-shikimate-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thioredoxred_c \"reduced thioredoxin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dcdp_c \"dCDP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thioredoxoxi_c \"oxidized thioredoxin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_metL_c \"L-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_metL_b \"L-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ch4s_c \"methanethiol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2obut_c \"2-oxobutanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hser_c \"L-homoserine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ahser_c \"O-acetyl-L-homoserine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbprc5a_c \"cobalt-precorrin-5A\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbpcr5b_c \"cobalt-precorrin-5B\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aald_c \"acetaldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_56dhura_c \"5,6-dihydrouracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3uprop_c \"3-ureidopropionate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_bio5a_c \"Biotinyl-5'-AMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abg_e \"4-aminobenzoate-glutamate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abg_b \"4-aminobenzoate-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abz_c \"4-aminobenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abz_b \"4-aminobenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lacgth_c \"S-lactoyl-glutathione\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_agmp_c \"N-acetylglutamyl-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aglu5sald_c \"N-acetyl-L-glutamate 5-semialdehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_r5p_c \"D-ribitol 5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rl5p_c \"D-ribulose-5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_arab_c \"L-arabinose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_arab_b \"L-arabinose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rblL_c \"L-ribulose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_c1ala_e \"cob(I)alamin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_c1ala_b \"cob(I)alamin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_c1ala_c \"cob(I)alamin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malt_e \"maltose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malt_b \"maltose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malt6p_c \"maltose 6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_67dm81rlum_c \"6,7-dimethyl-8-(1-D-ribityl)lumazine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ribflv_c \"riboflavin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ribflv_b \"riboflavin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5a6raura_c \"5-amino-6-(D-ribitylamino)uracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thm_e \"thiamin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thm_b \"thiamin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thm_c \"thiamin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_14dh2no_c \"1,4-dihydroxy-2-naphthoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2dmmqn_c \"2-demethylmenaquinone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3abcoa_c \"L-3-aminobutyryl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ctncoa_c \"crotonyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ap_c \"acetylphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tmam_c \"trimethylamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tmam_b \"trimethylamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyb_c \"glycine betaine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyb_b \"glycine betaine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dap_c \"L,L-diaminopimelate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dapime_c \"meso-diaminopimelate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_skm_c \"shikimate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5a15primz_c \"5-amino-1-(5-phospho-D-ribosyl)imidazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5a15pri4c_c \"5-amino-1-(5-phospho-D-ribosyl)imidazole-4-carboxylate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_Prot_c \"protein\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_RNA_c \"ribonucleic acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_DNA_c \"deoxyribonucleic acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mnl_e \"mannitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mnl_b \"mannitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mnl1p_c \"mannitol-1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4s4h2345th2sdl_c \"(4S)-4-hydroxy-2,3,4,5-tetrahydro-(2S)-dipicolinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2345thd_c \"(S)-2,3,4,5-tetrahydrodipicolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpaga_c \"UDP-alpha-N-acetyl-D-glucosamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpama_c \"UDP-N-acetyl-D-mannosamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_proL_e \"L-proline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_proL_b \"L-proline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_proL_c \"L-proline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_510methf_c \"5,10-methenyltetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyc_c \"D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dad_e \"deoxyadenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4ppt_c \"4'-phosphopantetheine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adman_e \"alpha-D-mannose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adman_b \"alpha-D-mannose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_m5tade_c \"S-methyl-5'-thioadenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_m5trib_c \"S-methyl-5-thio-D-ribose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tartsald_c \"tartronate semialdehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fe2_c \"Fe2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fe2_b \"Fe2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_proheme_c \"protoheme IX\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_betald_c \"betaine aldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dethio_c \"dethiobiotin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dethio_b \"dethiobiotin\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5doadn_c \"5'-deoxyadenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malS_e \"(S)-malate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malS_b \"(S)-malate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malS_c \"(S)-malate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbp_c \"carbamoyl-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dpcoa_c \"dephospho-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ca2_e \"Ca2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ca2_b \"Ca2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ca2_c \"Ca2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_escut_c \"esculetin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_escut_b \"esculetin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_escut_e \"esculetin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_proD_c \"D-proline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5mt23dopp_c \"5-(methylthio)-2,3-dioxopentyl phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2h3k5mt1pp_c \"2-hydroxy-3-keto-5-methylthio-1-phosphopentene\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ctp_c \"CTP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fmnh2_c \"FMNH2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fmn_c \"FMN\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_k_e \"K+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_k_b \"K+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_k_c \"K+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pbng_c \"porphobilinogen\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hmb_c \"hydroxymethylbilane\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_so3_c \"sulfite\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_so3_b \"sulfite\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2s_c \"hydrogen sulfide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_alaL_c \"L-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_alaL_b \"L-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_serL_c \"L-serine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_serL_b \"L-serine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tre_e \"trehalose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tre_b \"trehalose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tre6p_c \"trehalose 6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glnL_c \"L-glutamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glnL_b \"L-glutamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thymd_e \"thymidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_10fthf_c \"10-formyl-tetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2mbutp_c \"2-methylbutanoyl-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2mbut_c \"2-Methylbutyric acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2mbut_b \"2-Methylbutyric acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5prga_c \"5-phospho-ribosyl-glycineamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5prfga_c \"5'-phosphoribosyl-N-formylglycineamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ins_e \"inosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ins_b \"inosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ins_c \"inosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_6hm78dhpte_c \"6-hydroxymethyl-7,8-dihydropterin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udcpdp_c \"di-trans,octa-cis-undecaprenyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_prfiaicarp_c \"phosphoribulosylformimino-AICAR-P\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aimzcrnt_c \"aminoimidazole carboxamide ribonucleotide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5mthf_c \"5-methyl-tetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malcoa_c \"malonyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2maacoa_c \"2-methylacetoacetyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppcoa_c \"propanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ametam_c \"S-adenosyl-L-methioninamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_spmd_c \"spermidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_spmd_b \"spermidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4hcoa_c \"4-hydroxybutanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_vacoa_c \"vinylacetyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xmp_c \"XMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_spmd_e \"spermidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hxan_c \"hypoxanthine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hxan_b \"hypoxanthine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rib1p_c \"alpha-D-ribose-1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rib5p_c \"D-ribose-5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc_e \"beta-D-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_chor_c \"chorismate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_chor_b \"chorismate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pphn_c \"prephenate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4mpo_c \"4-methylphenol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4mpo_b \"4-methylphenol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4mpo_e \"4-methylphenol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_oxac_c \"oxaloacetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_et4p_c \"D-erythrose-4-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3dahs7p_c \"3-deoxy-D-arabino-heptulosonate-7-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_asucc_c \"adenylo-succinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fum_c \"fumarate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aacoa_c \"acetoacetyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_78dhnpte_c \"7,8-dihydro-D-neopterin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gcald_c \"glycolaldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gal1pA_c \"alpha-D-galactose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpgal_c \"UDP-D-galactose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_quln_c \"quinolinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpamr_c \"UDP-N-acetylmuramate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpagaepyr_c \"UDP-N-acetylglucosamine-enolpyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpagaepyr_b \"UDP-N-acetylglucosamine-enolpyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mthgxl_c \"methylglyoxal\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_s3m2op_c \"(S)-3-methyl-2-oxopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2mbcoa_c \"2-methylbutanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2pg_c \"2-phospho-D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3pg_c \"3-phospho-D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_metL_e \"L-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_6hmdhptedp_c \"6-hydroxymethyl-dihydropterin diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hcit_c \"homocitrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_duri_c \"deoxyuridine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xan_c \"xanthine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xan_b \"xanthine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_35dah_c \"(3S,5S)-3,5-diaminohexanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5a3oh_c \"(S)-5-amino-3-oxohexanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptgp_c \"L-1-phosphatidylglycerol-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptg_c \"L-1-phosphatidyl-glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3a2mp_c \"(R)-3-amino-2-methylpropanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2m3op_c \"2-methyl-3-oxopropanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mevR_c \"(R)-mevalonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mevR_b \"(R)-mevalonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_s5ddr5lhcys_c \"S-(5-deoxy-D-ribos-5-yl)-L-homocysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbpc3_c \"Cobalt-precorrin-3\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbpc4_c \"Cobalt-precorrin-4\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_casp_c \"N-carbamoyl-L-aspartate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xtsn_c \"xanthosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_24dap_c \"(2R,4S)-2,4-diaminopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2a4op_c \"2-amino-4-oxopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acnam_c \"N-acetylneuraminate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acnam_b \"N-acetylneuraminate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ama_c \"N-acetyl-D-mannosamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_15pramp_c \"1-(5-phospho-D-ribosyl)-AMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_15pr55pramai4c_c \"1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino]imidazole-4-carboxamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_g3p_e \"sn-glycerol-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fru1p_c \"fructose-1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lysL_c \"L-lysine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lysL_b \"L-lysine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3pt13lg_c \"3-phosphatidyl-1'-(3'-O-L-lysyl)glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_p1p3b5adtp_c \"P1,P3-Bis(5'-adenosyl) triphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pppgpp_c \"pppGpp\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pc2_c \"precorrin-2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_shclin_c \"Sirohydrochlorin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dcyt_e \"deoxycytidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cdpdag_c \"CDP-diacylglycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptd_c \"phosphatidate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tag16bp_c \"tagatose-1,6-bisphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyald3p_c \"D-glyceraldehyde-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acap_c \"adenosyl-cobinamide phosphate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_altd_c \"D-altronate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpamagdapaa_c \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminopimelyl-D-alanyl-D-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ichor_c \"isochorismate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cysL_c \"L-cysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cysL_b \"L-cysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4m5het_c \"4-methyl-5-(beta-hydroxyethyl)thiazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2mccoa_c \"(E)-2-methylcrotonoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2m3hbcoa_c \"2-methyl-3-hydroxybutyryl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glnL_e \"L-glutamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4a2m5dpmp_c \"4-amino-2-methyl-5-diphosphomethylpyrimidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4m52poet_c \"4-methyl-5-(2-phosphonooxyethyl)thiazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thmp_c \"thiamin phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pratp_c \"phosphoribosyl-ATP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_valL_c \"L-valine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_valL_b \"L-valine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fru16bp_c \"fructose-1,6-bisphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyam5p_c \"pyridoxamine 5'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyal5p_c \"pyridoxal 5'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_12dag_c \"1,2-Diacyl-sn-glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gpgl_c \"glycerophosphoglycoglycerolipid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_at_c \"acetone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_at_b \"acetone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_at_e \"acetone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_zn2_e \"Zn2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_zn2_b \"Zn2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_zn2_c \"Zn2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ileL_e \"L-isoleucine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ileL_b \"L-isoleucine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ileL_c \"L-isoleucine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyin_e \"pyridoxine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyin_b \"pyridoxine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyin_c \"pyridoxine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_din_c \"deoxyinosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_no2_e \"nitrite\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ga6p_c \"D-glucosamine-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ga1p_c \"D-glucosamine 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xyld_e \"D-xylose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xyld_b \"D-xylose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xyld_c \"D-xylose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thmdp_c \"thiamin diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dhettp_c \"alpha,beta-dihydroxyethyl-TPP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2iac_c \"2-iminoacetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glx_c \"glyoxylate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cdlp_c \"cardiolipin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_13bpg_c \"1,3-bisphospho-D-glycerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hcy_c \"hydrogen cyanide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3mpyr_c \"3-mercaptopyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tcy_c \"thiocyanate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tmam_e \"trimethylamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5mtr1p_c \"5-methylthioribulose-1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hxan_e \"hypoxanthine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_56dmbimz_c \"5,6-dimethylbenzimidazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_rz5p_c \"alpha-ribazole-5'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nac_c \"nicotinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nac_b \"nicotinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tyrL_e \"L-tyrosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tyrL_b \"L-tyrosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tyrL_c \"L-tyrosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppgpp_c \"ppGpp\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mthx5p_c \"1-methylthio-xylulose 5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_1dxyl5p_c \"1-deoxy-D-xylulose 5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptrc_e \"putrescine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptrc_b \"putrescine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ptrc_c \"putrescine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_no3_e \"nitrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_no3_b \"nitrate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_no3_c \"nitrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_phser_c \"O-phospho-L-homoserine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thrL_c \"L-threonine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thrL_b \"L-threonine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppacal_c \"Phosphonoacetaldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppac_c \"phosphonoacetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ornD_c \"D-ornithine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_c1yda_c \"cob(I)yrinate a,c-diamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acbda_c \"adenosyl-cobyrinate a,c-diamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pppi_c \"PPPi\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isobutcoa_c \"isobutanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppat_c \"phosphonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppat_b \"phosphonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2a3ob_c \"2-amino-3-oxobutanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_25da65prap_c \"2,5-diamino-6-(5-phospho-D-ribosylamino)pyrimidin-4(3H)-one\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5a65prsaura_c \"5-amino-6-(5-phospho-D-ribosylamino)uracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_butp_c \"butanoyl phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_eto_e \"ethanol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_eto_b \"ethanol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_eto_c \"ethanol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gluL_e \"L-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adman_c \"alpha-D-mannose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glc1p_c \"beta-D-glucose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_argL_c \"L-arginine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_argL_b \"L-arginine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_his_c \"L-histidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_his_b \"L-histidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thym_c \"thymine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_citrL_c \"L-citrulline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_1dgtl4p_c \"1-deoxy-L-glycero-tetrulose 4-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adpglu_c \"ADP-glucose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_suchser_c \"O-succinyl-L-homoserine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cppogen3_c \"coproporphyrinogen III\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_datp_c \"dATP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4a2m5pmp_c \"4-amino-2-methyl-5-phosphomethylpyrimidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_12dh5mtp_c \"1,2-dihydroxy-5-(methylthio)pent-1-en-3-one\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dho_c \"(S)-dihydroorotate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_orot_c \"orotate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_palm_c \"palmitate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5prfgad_c \"5-phosphoribosyl-N-formylglycineamidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acnam_e \"N-acetylneuraminate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aorn_c \"N-acetyl-L-ornithine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ornL_c \"L-ornithine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aglu_c \"N-acetyl-L-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5a65praura_c \"5-amino-6-(5-phospho-D-ribitylamino)uracil\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_serL_e \"L-serine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3pser_c \"3-phospho-L-serine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2sucbzcoa_c \"2-succinylbenzoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fru_e \"D-fructose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gccoa_c \"glutaconyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_uri_e \"uridine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sor_e \"D-sorbitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sor_b \"D-sorbitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sor_c \"D-sorbitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_chol_c \"choline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_chol_b \"choline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isuc_c \"alpha-iminosuccinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lysL_e \"L-lysine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2dhp_c \"2-dehydropantoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4s45dhp23do_c \"(4S)-4,5-dihydroxypentan-2,3-dione\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4s45dhp23do_b \"(4S)-4,5-dihydroxypentan-2,3-dione\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4s45dhp23do_e \"(4S)-4,5-dihydroxypentan-2,3-dione\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_34hppyr_c \"3-(4-hydroxyphenyl)pyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gal_c \"beta-D-galactose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gal_b \"beta-D-galactose\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gal6p_c \"D-galactose 6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5pra_c \"5-phospho-beta-D-ribosyl-amine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mg2_e \"Mg2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mg2_b \"Mg2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mg2_c \"Mg2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ppat_e \"phosphonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ncam_e \"nicotinamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dttp_c \"dTTP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_6pg_c \"6-phospho-D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5apn_c \"5-aminopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5apn_b \"5-aminopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pntoR_e \"(R)-pantothenate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pntoR_b \"(R)-pantothenate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pntoR_c \"(R)-pantothenate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malt_c \"maltose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isobutp_c \"isobutanoylphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isobuta_c \"isobutyric acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isobuta_b \"isobutyric acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_lpteichoic_c \"lipoteichoic acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpcpp_c \"ditrans,octacis-undecaprenyl phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpagapp_c \"undecaprenyl-N-acetyl-alpha-D-glucosaminyl-pyrophosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3hbcoa_c \"(S)-3-hydroxybutanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aga_e \"N-acetyl-beta-D-glucosamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aga_b \"N-acetyl-beta-D-glucosamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aga6p_c \"N-acetyl-D-glucosamine-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_argL_e \"L-arginine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_co_c \"carbon monoxide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_leuL_e \"L-leucine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dadp_c \"dADP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2ippm_c \"2-isopropylmaleate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_galA_c \"alpha-D-galactose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aga1p_c \"N-acetyl-alpha-D-glucosamine 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fruf_c \"beta-D-fructofuranose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isocap_e \"Isocaproate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hpyr_c \"hydroxypyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_etoa_c \"ethanolamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_etoa_b \"ethanolamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_78dhnpte3p_c \"7,8-dihydroneopterin 3'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ival_c \"Isovalerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ival_b \"Isovalerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ival_e \"Isovalerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_phpyr_c \"phenylpyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_alaala_c \"D-alanyl-D-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cobalt_e \"Co2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cobalt_b \"Co2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cobalt_c \"Co2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_celb_e \"cellobiose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_celb_b \"cellobiose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_celb6p_c \"cellobiose 6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dmadp_c \"dimethylallyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fe2_e \"Fe2+\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_etoa_e \"ethanolamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fuc1p_c \"L-fuculose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tsul_c \"thiosulfate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_arab_e \"L-arabinose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5apn_e \"5-aminopentanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hista_c \"histidinal\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_chol_e \"choline\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_r3hbatcoa_c \"(R)-3-Hydroxybutanoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ctnycoa_c \"crotonoyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_for_e \"formate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbpcr8x_c \"cobalt-precorrin-8x\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cobat_c \"Cobyrinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_argsuc_c \"L-arginino-succinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_urea1c_c \"urea-1-carboxylate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4u5imzc_c \"4-ureido-5-imidazole carboxylate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_actnR_c \"(R)-acetoin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_actnS_c \"(S)-acetoin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2hisocapcoa_c \"(R)-2-hydroxyisocaproyl-CoA\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xyl_c \"D-xylulose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyb_e \"glycine betaine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2drib5p_c \"2-deoxy-D-ribose-5-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pendp_c \"all-trans-pentaprenyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_xan_e \"xanthine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_drib_c \"deoxyribose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_raf_c \"raffinose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_raf_b \"raffinose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_raf_e \"raffinose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_m5trib1p_c \"S-methyl-5-thio-alpha-D-ribose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4hplald_c \"4-Hydroxyphenylacetaldehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_naspdi_c \"N-acetylspermidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adprib_c \"ADP-ribose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2drib1p_c \"2-deoxy-alpha-D-ribose 1-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2aepat_e \"(2-aminoethyl)phosphonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2aepat_b \"(2-aminoethyl)phosphonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2aepat_c \"(2-aminoethyl)phosphonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_trpL_c \"L-tryptophan\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_trpL_b \"L-tryptophan\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_imp_c \"IMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_alaL_e \"L-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbdhpcr6_c \"Cobalt-dihydro-precorrin 6\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbpcr7_c \"cobalt-precorrin-7\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpamala_c \"UDP-N-acetylmuramyl-L-Ala\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpamag_c \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_grdp_c \"geranyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ac_e \"acetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cit_c \"citrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acon_c \"cis-aconitate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_36dah_c \"(3S)-3,6-diaminohexanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4imzo_c \"4-imidazolone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fgly_c \"N-formimino-glycine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4a4dchor_c \"4-amino-4-deoxychorismate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_c2yda_c \"cob(II)yrinate a,c-diamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_orot5p_c \"orotidine-5'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2mbut_e \"2-Methylbutyric acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4a5imzc_c \"4-amino-5-imidazole carboxylate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4aimz_c \"4-aminoimidazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_3phpyr_c \"3-phospho-hydroxypyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_isobuta_e \"isobutyric acid\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2a2hbut_c \"2-aceto-2-hydroxy-butanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23dh3mv_c \"2,3-dihydroxy-3-methylvalerate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_78dhnptetp_c \"7,8-dihydroneopterin triphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyam_c \"pyridoxamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pyal_c \"pyridoxal\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_s2a6kp_c \"N-succinyl-2-amino-6-ketopimelate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_s26dap_c \"N-succinyl-L,L-2,6-diaminopimelate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4pptnD_c \"D-4'-phosphopantothenate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abg_c \"4-aminobenzoate-glutamate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dgo15lac_c \"D-glucono-1,5-lactone\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aspL_e \"L-aspartate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpamagdap_c \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminopimelate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cysL_e \"L-cysteine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tag6p_c \"tagatose-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5fithf_c \"5-formiminotetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hexdp_c \"all-trans-hexaprenyl diphosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_celb_c \"cellobiose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nacput_c \"N-acetylputrescine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gtol_e \"galactitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gtol_b \"galactitol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gtol1p_c \"galactitol-1-phosphate\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_serD_c \"D-serine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5pr4sc5ai_c \"5'-phosphoribosyl-4-(N-succinocarboxamide)-5-aminoimidazole\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4hbut_c \"4-hydroxybutyrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hoam_e \"Hydroxylamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ibtal_c \"isobutanal\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ibtol_c \"isobutanol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ibtol_b \"isobutanol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_alaB_c \"beta-alanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pheL_e \"L-phenylalanine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thr3p_c \"L-threonine O-3-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_1a2po2p_c \"(R)-1-amino-2-propanol O-2-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ncammnuc_c \"nicotinamide mononucleotide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_co2_e \"CO2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5all_c \"5-aminolevulinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_sor6p_c \"D-sorbitol-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nac_e \"nicotinate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_acb_c \"adenosyl-cobyrate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4hoplac_c \"4-Hydroxyphenylacetate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_metD_c \"D-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_metD_b \"D-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gal_e \"beta-D-galactose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tagd_c \"D-Tagaturonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tagd_b \"D-Tagaturonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_aga_c \"N-acetyl-beta-D-glucosamine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_csn_c \"cytosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_csn_b \"cytosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_csn_e \"cytosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_malsald_c \"malonate semialdehyde\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_asnL_e \"L-asparagine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_hpsamdhllys_c \"H-protein-S-(aminomethyldihydrolipoyl)lysine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_agm_c \"agmatine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2hisocap_c \"(R)-2-hydroxyisocaproate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_urea_c \"urea\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fuc_e \"L-fuculose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_fuc_b \"L-fuculose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2me4p_c \"2-C-methyl-D-erythritol-4-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_so3_e \"sulfite\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4cyt5dp2me_c \"4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dachi6p_c \"diacetylchitobiose-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2_c \"H2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2_b \"H2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_adn_e \"adenosine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_valL_e \"L-valine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_gdpm_c \"GDP-alpha-D-mannose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_prfaca_c \"phosphoribosyl-formamido-carboxamide\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyo_e \"glycerol\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_5fthf_c \"5-formyl-tetrahydrofolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_cbprc6a_c \"cobalt-precorrin-6A\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dgmp_c \"dGMP\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_metD_e \"D-methionine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_trpL_e \"L-tryptophan\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_nh3_e \"ammonia\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_4abz_e \"4-aminobenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dethio_e \"dethiobiotin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_his_e \"L-histidine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_2dh3dg_e \"2-dehydro-3-deoxy-D-gluconate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_mevR_e \"(R)-mevalonate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dachi_e \"diacetylchitobiose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_dachi_b \"diacetylchitobiose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_chor_e \"chorismate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2o_e \"H2O\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_thrL_e \"L-threonine\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_psuri5p_c \"pseudouridine 5'-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23diap_e \"2,3-diaminopropanoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ama6p_c \"N-acetyl-D-mannosamine-6-phosphate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ibtol_e \"isobutanol\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ribD_e \"D-ribose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_udpagaepyr_e \"UDP-N-acetylglucosamine-enolpyruvate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tgt_e \"tagatose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tgt_b \"tagatose\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_ribflv_e \"riboflavin\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_h2_e \"H2\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_glyclt_c \"glycolate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_pantR_c \"(R)-pantoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_23dhbz_e \"2,3-dihydroxybenzoate\">\n",
      "Use of FORMULA in the notes element is discouraged, use fbc:chemicalFormula instead: <Species M_tagd_e \"D-Tagaturonate\">\n",
      "http://identifiers.org/UniProt/ does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id\n",
      "http://identifiers.org/UniProt/ does not conform to 'http(s)://identifiers.org/collection/id' or'http(s)://identifiers.org/COLLECTION:id\n",
      "Adding exchange reaction EX_h2o_b with default bounds for boundary metabolite: h2o_b.\n",
      "Adding exchange reaction EX_gluL_b with default bounds for boundary metabolite: gluL_b.\n",
      "Adding exchange reaction EX_pi_b with default bounds for boundary metabolite: pi_b.\n",
      "Adding exchange reaction EX_co2_b with default bounds for boundary metabolite: co2_b.\n",
      "Adding exchange reaction EX_pheL_b with default bounds for boundary metabolite: pheL_b.\n",
      "Adding exchange reaction EX_glyo_b with default bounds for boundary metabolite: glyo_b.\n",
      "Adding exchange reaction EX_h_b with default bounds for boundary metabolite: h_b.\n",
      "Adding exchange reaction EX_glcAD_b with default bounds for boundary metabolite: glcAD_b.\n",
      "Adding exchange reaction EX_actid_b with default bounds for boundary metabolite: actid_b.\n",
      "Adding exchange reaction EX_23diap_b with default bounds for boundary metabolite: 23diap_b.\n",
      "Adding exchange reaction EX_nh3_b with default bounds for boundary metabolite: nh3_b.\n",
      "Adding exchange reaction EX_isocap_b with default bounds for boundary metabolite: isocap_b.\n",
      "Adding exchange reaction EX_hoam_b with default bounds for boundary metabolite: hoam_b.\n",
      "Adding exchange reaction EX_2o6pamg2d_b with default bounds for boundary metabolite: 2o6pamg2d_b.\n",
      "Adding exchange reaction EX_leuL_b with default bounds for boundary metabolite: leuL_b.\n",
      "Adding exchange reaction EX_cdpg_b with default bounds for boundary metabolite: cdpg_b.\n",
      "Adding exchange reaction EX_dad_b with default bounds for boundary metabolite: dad_b.\n",
      "Adding exchange reaction EX_escul_b with default bounds for boundary metabolite: escul_b.\n",
      "Adding exchange reaction EX_for_b with default bounds for boundary metabolite: for_b.\n",
      "Adding exchange reaction EX_na_b with default bounds for boundary metabolite: na_b.\n",
      "Adding exchange reaction EX_glc_b with default bounds for boundary metabolite: glc_b.\n",
      "Adding exchange reaction EX_uri_b with default bounds for boundary metabolite: uri_b.\n",
      "Adding exchange reaction EX_no2_b with default bounds for boundary metabolite: no2_b.\n",
      "Adding exchange reaction EX_glcn_b with default bounds for boundary metabolite: glcn_b.\n",
      "Adding exchange reaction EX_g3p_b with default bounds for boundary metabolite: g3p_b.\n",
      "Adding exchange reaction EX_sucr_b with default bounds for boundary metabolite: sucr_b.\n",
      "Adding exchange reaction EX_ncam_b with default bounds for boundary metabolite: ncam_b.\n",
      "Adding exchange reaction EX_ribD_b with default bounds for boundary metabolite: ribD_b.\n",
      "Adding exchange reaction EX_ura_b with default bounds for boundary metabolite: ura_b.\n",
      "Adding exchange reaction EX_bio_b with default bounds for boundary metabolite: bio_b.\n",
      "Adding exchange reaction EX_dcyt_b with default bounds for boundary metabolite: dcyt_b.\n",
      "Adding exchange reaction EX_2dh3dg_b with default bounds for boundary metabolite: 2dh3dg_b.\n",
      "Adding exchange reaction EX_hco3_b with default bounds for boundary metabolite: hco3_b.\n",
      "Adding exchange reaction EX_ac_b with default bounds for boundary metabolite: ac_b.\n",
      "Adding exchange reaction EX_thymd_b with default bounds for boundary metabolite: thymd_b.\n",
      "Adding exchange reaction EX_adn_b with default bounds for boundary metabolite: adn_b.\n",
      "Adding exchange reaction EX_gly_b with default bounds for boundary metabolite: gly_b.\n",
      "Adding exchange reaction EX_23dhbz_b with default bounds for boundary metabolite: 23dhbz_b.\n",
      "Adding exchange reaction EX_asnL_b with default bounds for boundary metabolite: asnL_b.\n",
      "Adding exchange reaction EX_aspL_b with default bounds for boundary metabolite: aspL_b.\n",
      "Adding exchange reaction EX_fru_b with default bounds for boundary metabolite: fru_b.\n",
      "Adding exchange reaction EX_metL_b with default bounds for boundary metabolite: metL_b.\n",
      "Adding exchange reaction EX_4abg_b with default bounds for boundary metabolite: 4abg_b.\n",
      "Adding exchange reaction EX_4abz_b with default bounds for boundary metabolite: 4abz_b.\n",
      "Adding exchange reaction EX_arab_b with default bounds for boundary metabolite: arab_b.\n",
      "Adding exchange reaction EX_c1ala_b with default bounds for boundary metabolite: c1ala_b.\n",
      "Adding exchange reaction EX_malt_b with default bounds for boundary metabolite: malt_b.\n",
      "Adding exchange reaction EX_ribflv_b with default bounds for boundary metabolite: ribflv_b.\n",
      "Adding exchange reaction EX_thm_b with default bounds for boundary metabolite: thm_b.\n",
      "Adding exchange reaction EX_tmam_b with default bounds for boundary metabolite: tmam_b.\n",
      "Adding exchange reaction EX_glyb_b with default bounds for boundary metabolite: glyb_b.\n",
      "Adding exchange reaction EX_mnl_b with default bounds for boundary metabolite: mnl_b.\n",
      "Adding exchange reaction EX_proL_b with default bounds for boundary metabolite: proL_b.\n",
      "Adding exchange reaction EX_adman_b with default bounds for boundary metabolite: adman_b.\n",
      "Adding exchange reaction EX_fe2_b with default bounds for boundary metabolite: fe2_b.\n",
      "Adding exchange reaction EX_dethio_b with default bounds for boundary metabolite: dethio_b.\n",
      "Adding exchange reaction EX_malS_b with default bounds for boundary metabolite: malS_b.\n",
      "Adding exchange reaction EX_ca2_b with default bounds for boundary metabolite: ca2_b.\n",
      "Adding exchange reaction EX_escut_b with default bounds for boundary metabolite: escut_b.\n",
      "Adding exchange reaction EX_k_b with default bounds for boundary metabolite: k_b.\n",
      "Adding exchange reaction EX_so3_b with default bounds for boundary metabolite: so3_b.\n",
      "Adding exchange reaction EX_alaL_b with default bounds for boundary metabolite: alaL_b.\n",
      "Adding exchange reaction EX_serL_b with default bounds for boundary metabolite: serL_b.\n",
      "Adding exchange reaction EX_tre_b with default bounds for boundary metabolite: tre_b.\n",
      "Adding exchange reaction EX_glnL_b with default bounds for boundary metabolite: glnL_b.\n",
      "Adding exchange reaction EX_2mbut_b with default bounds for boundary metabolite: 2mbut_b.\n",
      "Adding exchange reaction EX_ins_b with default bounds for boundary metabolite: ins_b.\n",
      "Adding exchange reaction EX_spmd_b with default bounds for boundary metabolite: spmd_b.\n",
      "Adding exchange reaction EX_hxan_b with default bounds for boundary metabolite: hxan_b.\n",
      "Adding exchange reaction EX_chor_b with default bounds for boundary metabolite: chor_b.\n",
      "Adding exchange reaction EX_4mpo_b with default bounds for boundary metabolite: 4mpo_b.\n",
      "Adding exchange reaction EX_udpagaepyr_b with default bounds for boundary metabolite: udpagaepyr_b.\n",
      "Adding exchange reaction EX_xan_b with default bounds for boundary metabolite: xan_b.\n",
      "Adding exchange reaction EX_mevR_b with default bounds for boundary metabolite: mevR_b.\n",
      "Adding exchange reaction EX_acnam_b with default bounds for boundary metabolite: acnam_b.\n",
      "Adding exchange reaction EX_lysL_b with default bounds for boundary metabolite: lysL_b.\n",
      "Adding exchange reaction EX_cysL_b with default bounds for boundary metabolite: cysL_b.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Adding exchange reaction EX_valL_b with default bounds for boundary metabolite: valL_b.\n",
      "Adding exchange reaction EX_at_b with default bounds for boundary metabolite: at_b.\n",
      "Adding exchange reaction EX_zn2_b with default bounds for boundary metabolite: zn2_b.\n",
      "Adding exchange reaction EX_ileL_b with default bounds for boundary metabolite: ileL_b.\n",
      "Adding exchange reaction EX_pyin_b with default bounds for boundary metabolite: pyin_b.\n",
      "Adding exchange reaction EX_xyld_b with default bounds for boundary metabolite: xyld_b.\n",
      "Adding exchange reaction EX_nac_b with default bounds for boundary metabolite: nac_b.\n",
      "Adding exchange reaction EX_tyrL_b with default bounds for boundary metabolite: tyrL_b.\n",
      "Adding exchange reaction EX_ptrc_b with default bounds for boundary metabolite: ptrc_b.\n",
      "Adding exchange reaction EX_no3_b with default bounds for boundary metabolite: no3_b.\n",
      "Adding exchange reaction EX_thrL_b with default bounds for boundary metabolite: thrL_b.\n",
      "Adding exchange reaction EX_ppat_b with default bounds for boundary metabolite: ppat_b.\n",
      "Adding exchange reaction EX_eto_b with default bounds for boundary metabolite: eto_b.\n",
      "Adding exchange reaction EX_argL_b with default bounds for boundary metabolite: argL_b.\n",
      "Adding exchange reaction EX_his_b with default bounds for boundary metabolite: his_b.\n",
      "Adding exchange reaction EX_sor_b with default bounds for boundary metabolite: sor_b.\n",
      "Adding exchange reaction EX_chol_b with default bounds for boundary metabolite: chol_b.\n",
      "Adding exchange reaction EX_4s45dhp23do_b with default bounds for boundary metabolite: 4s45dhp23do_b.\n",
      "Adding exchange reaction EX_gal_b with default bounds for boundary metabolite: gal_b.\n",
      "Adding exchange reaction EX_mg2_b with default bounds for boundary metabolite: mg2_b.\n",
      "Adding exchange reaction EX_5apn_b with default bounds for boundary metabolite: 5apn_b.\n",
      "Adding exchange reaction EX_pntoR_b with default bounds for boundary metabolite: pntoR_b.\n",
      "Adding exchange reaction EX_isobuta_b with default bounds for boundary metabolite: isobuta_b.\n",
      "Adding exchange reaction EX_aga_b with default bounds for boundary metabolite: aga_b.\n",
      "Adding exchange reaction EX_etoa_b with default bounds for boundary metabolite: etoa_b.\n",
      "Adding exchange reaction EX_ival_b with default bounds for boundary metabolite: ival_b.\n",
      "Adding exchange reaction EX_cobalt_b with default bounds for boundary metabolite: cobalt_b.\n",
      "Adding exchange reaction EX_celb_b with default bounds for boundary metabolite: celb_b.\n",
      "Adding exchange reaction EX_raf_b with default bounds for boundary metabolite: raf_b.\n",
      "Adding exchange reaction EX_2aepat_b with default bounds for boundary metabolite: 2aepat_b.\n",
      "Adding exchange reaction EX_trpL_b with default bounds for boundary metabolite: trpL_b.\n",
      "Adding exchange reaction EX_gtol_b with default bounds for boundary metabolite: gtol_b.\n",
      "Adding exchange reaction EX_ibtol_b with default bounds for boundary metabolite: ibtol_b.\n",
      "Adding exchange reaction EX_metD_b with default bounds for boundary metabolite: metD_b.\n",
      "Adding exchange reaction EX_tagd_b with default bounds for boundary metabolite: tagd_b.\n",
      "Adding exchange reaction EX_csn_b with default bounds for boundary metabolite: csn_b.\n",
      "Adding exchange reaction EX_fuc_b with default bounds for boundary metabolite: fuc_b.\n",
      "Adding exchange reaction EX_h2_b with default bounds for boundary metabolite: h2_b.\n",
      "Adding exchange reaction EX_dachi_b with default bounds for boundary metabolite: dachi_b.\n",
      "Adding exchange reaction EX_tgt_b with default bounds for boundary metabolite: tgt_b.\n",
      "Adding exchange reaction EX_biomass_b with default bounds for boundary metabolite: biomass_b.\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_tagd \"D-Tagaturonate transporter \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_tagd \"D-Tagaturonate transporter \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ppat \"phosphonate ABC transporter ATP-binding protein\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ppat \"phosphonate ABC transporter ATP-binding protein\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_gluL \"amino-acid ABC transport (L-glutamate)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_gluL \"amino-acid ABC transport (L-glutamate)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_asnL \"L-asparagine transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_asnL \"L-asparagine transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glyb \"glycine betaine ABC transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glyb \"glycine betaine ABC transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_aspL2 \"L-aspartate transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_aspL2 \"L-aspartate transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ptrc \"putrescine ABC transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ptrc \"putrescine ABC transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_metD \"D-methionine ABC transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_metD \"D-methionine ABC transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_chol \"choline ABC transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_chol \"choline ABC transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_atp \"sodium extrusion ABC transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_atp \"sodium extrusion ABC transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glnL \"amino-acid ABC transport (L-glutamine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glnL \"amino-acid ABC transport (L-glutamine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_pi2 \"phosphate active transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_pi2 \"phosphate active transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_argL \"L-arginine transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_argL \"L-arginine transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_his \"amino-acid ABC transport (L-histidine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_his \"amino-acid ABC transport (L-histidine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_tyrL \"amino-acid ABC transport (L-tyrosine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_tyrL \"amino-acid ABC transport (L-tyrosine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_zn2 \"zinc exporter ABC transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_zn2 \"zinc exporter ABC transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_g3p \"glycerol-3-phosphate-transporting ATPase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_g3p \"glycerol-3-phosphate-transporting ATPase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_fe2 \"Iron atp driven transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_fe2 \"Iron atp driven transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_no2 \"Nitrate ATP transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_no2 \"Nitrate ATP transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_cysL \"amino-acid ABC transport (L-cysteine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_cysL \"amino-acid ABC transport (L-cysteine)\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_c1ala \"Vitamin B12 transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_c1ala \"Vitamin B12 transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_metL \"amino-acid ABC transport (L-methionine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_metL \"amino-acid ABC transport (L-methionine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_trpL \"amino-acid ABC transport (L-tryptophan)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_trpL \"amino-acid ABC transport (L-tryptophan)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_spmd \"spermine ABC transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_spmd \"spermine ABC transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_lysL \"amino-acid ABC transport (L-lysine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_lysL \"amino-acid ABC transport (L-lysine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_leuL \"amino-acid ABC transport (L-leucine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_leuL \"amino-acid ABC transport (L-leucine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_alaL2 \"L-alanine transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_alaL2 \"L-alanine transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_pheL \"amino-acid ABC transport (L-phenylalanine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_pheL \"amino-acid ABC transport (L-phenylalanine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_cobalt \"The ATP-binding Cassette (ABC) Superfamily\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_cobalt \"The ATP-binding Cassette (ABC) Superfamily\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ca \"Calcium ATP dependent transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ca \"Calcium ATP dependent transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_valL \"amino-acid ABC transport (L-valine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_valL \"amino-acid ABC transport (L-valine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_proL \"amino-acid ABC transport (L-proline)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_proL \"amino-acid ABC transport (L-proline)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_mg3 \"P-type magnesium transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_mg3 \"P-type magnesium transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_thrL \"amino-acid ABC transport (L-threonine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_thrL \"amino-acid ABC transport (L-threonine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_k \"potassium transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_k \"potassium transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_gly \"amino-acid ABC transport (glycine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_gly \"amino-acid ABC transport (glycine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_mg2 \"magnesium ABC transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_mg2 \"magnesium ABC transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_serL \"amino-acid ABC transport (L-serine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_serL \"amino-acid ABC transport (L-serine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ileL \"amino-acid ABC transport (L-isoleucine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ileL \"amino-acid ABC transport (L-isoleucine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ribD \"ribose ABC transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ribD \"ribose ABC transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_470 \"asparagine synthase (glutamine-hydrolysing)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_470 \"asparagine synthase (glutamine-hydrolysing)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_222 \"aspartate-ammonia ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_222 \"aspartate-ammonia ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_88 \"succinate semialdehyde:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_88 \"succinate semialdehyde:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_317 \"Acetamide amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_317 \"Acetamide amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_575 \"L-glutamate:NAD+ oxidoreductase (deaminating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_575 \"L-glutamate:NAD+ oxidoreductase (deaminating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_712 \"4-aminobutyrate:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_712 \"4-aminobutyrate:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_411 \"succinate semialdehyde:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_411 \"succinate semialdehyde:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_439 \"aspartate:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_439 \"aspartate:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_445 \"glutamine-fructose-6-phosphate transaminase (isomerizing)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_445 \"glutamine-fructose-6-phosphate transaminase (isomerizing)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_142 \"4-aminobutanoate:pyruvate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_142 \"4-aminobutanoate:pyruvate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_12 \"glutamate racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_12 \"glutamate racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_365 \"argininosuccinate lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_365 \"argininosuccinate lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_92 \"asparaginase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_92 \"asparaginase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_543 \"glutamate synthase (NADPH)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_543 \"glutamate synthase (NADPH)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_675 \"L-glutamine aminohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_675 \"L-glutamine aminohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_688 \"4-aminobutyrate transaminase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_688 \"4-aminobutyrate transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_336 \"L-alanine:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_336 \"L-alanine:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_759 \"4-aminobutanoate:glyoxylate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_759 \"4-aminobutanoate:glyoxylate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_506 \"UDP-N-acetylmuramate-L-alanine ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_506 \"UDP-N-acetylmuramate-L-alanine ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_3 \"glutathione gamma-glutamylaminopeptidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_3 \"glutathione gamma-glutamylaminopeptidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_428 \"UDP-N-acetylmuramoyl-L-alanine-D-glutamate ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_428 \"UDP-N-acetylmuramoyl-L-alanine-D-glutamate ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_526 \"glutamate-ammonia ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_526 \"glutamate-ammonia ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_357 \"alanine racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_357 \"alanine racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_346 \"D-alanine-D-alanine ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_346 \"D-alanine-D-alanine ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_396 \"L-Serine:pyruvate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_396 \"L-Serine:pyruvate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_446 \"argininosuccinate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_446 \"argininosuccinate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_561 \"L-glutamate:NAD+ oxidoreductase (transaminating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_561 \"L-glutamate:NAD+ oxidoreductase (transaminating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_583 \"malonate semialdehide decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_583 \"malonate semialdehide decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_347 \"aspartate 1-decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_347 \"aspartate 1-decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_380 \"glucosamine-6-phosphate deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_380 \"glucosamine-6-phosphate deaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_571 \"ATP:N-acetyl-D-glucosamine 6-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_571 \"ATP:N-acetyl-D-glucosamine 6-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_331 \"UDP-N-acetylglucosamine diphosphorylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_331 \"UDP-N-acetylglucosamine diphosphorylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_227 \"phosphoglucosamine mutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_227 \"phosphoglucosamine mutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_594 \"glucosamine-1-phosphate N-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_594 \"glucosamine-1-phosphate N-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_514 \"N-acetylglucosamine-6-phosphate deacetylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_514 \"N-acetylglucosamine-6-phosphate deacetylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_548 \"UDP-N-acetyl-D-glucosamine 2-epimerase (UDP-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_548 \"UDP-N-acetyl-D-glucosamine 2-epimerase (UDP-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_301 \"UDP-N-acetylglucosamine 1-carboxyvinyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_301 \"UDP-N-acetylglucosamine 1-carboxyvinyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_176 \"UDP-N-acetylmuramate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_176 \"UDP-N-acetylmuramate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_194 \"N-acetylneuraminate pyruvate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_194 \"N-acetylneuraminate pyruvate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_727 \"N-acetyl-D-glucosamine 6-phosphate 2-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_727 \"N-acetyl-D-glucosamine 6-phosphate 2-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_741 \"ATP:N-acetyl-D-mannosamine 6-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_741 \"ATP:N-acetyl-D-mannosamine 6-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_223 \"UDP-N-acetylmuramate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_223 \"UDP-N-acetylmuramate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_121 \"UDP-N-acetyl-D-glucosamine 2-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_121 \"UDP-N-acetyl-D-glucosamine 2-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_602 \"L-arginine carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_602 \"L-arginine carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_367 \"allophanate hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_367 \"allophanate hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_493 \"acetyl-CoA:putrescine N-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_493 \"acetyl-CoA:putrescine N-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_106 \"N-acetyl-gamma-glutamyl-phosphate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_106 \"N-acetyl-gamma-glutamyl-phosphate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_638 \"agmatine amidinohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_638 \"agmatine amidinohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_467 \"ornithine carbamoyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_467 \"ornithine carbamoyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_747 \"acetylornithine amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_747 \"acetylornithine amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_278 \"N2-Acetyl-L-ornithine:L-glutamate N-acetyltransferase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_278 \"N2-Acetyl-L-ornithine:L-glutamate N-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_42 \"4-aminobutyraldehyde:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_42 \"4-aminobutyraldehyde:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_635 \"acetylglutamate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_635 \"acetylglutamate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_311 \"amino-acid N-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_311 \"amino-acid N-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_135 \"proline racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_135 \"proline racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_627 \"urea:bicarbonate ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_627 \"urea:bicarbonate ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_295 \"putrescine aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_295 \"putrescine aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_748 \"acetylornithine transaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_748 \"acetylornithine transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_337 \"L-arginine iminohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_337 \"L-arginine iminohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_CW \"Cell wall synthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_CW \"Cell wall synthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Biomass \"Biomass Reaction\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Biomass \"Biomass Reaction\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_RNA \"RNA synthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_RNA \"RNA synthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_DNA \"DNA synthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_DNA \"DNA synthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_CLP \"Peptidoglycan synthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_CLP \"Peptidoglycan synthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_prot \"Protein synthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_prot \"Protein synthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_lip \"Lipid composition\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_lip \"Lipid composition\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Bio_SPs \"Solute Pool\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Bio_SPs \"Solute Pool\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_525 \"acetyl-CoA:4-hydroxybutanoate CoA-transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_525 \"acetyl-CoA:4-hydroxybutanoate CoA-transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_616 \"vinylacetyl-CoA delta3-delta2-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_616 \"vinylacetyl-CoA delta3-delta2-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_530 \"butanal:NADP+ oxidoreductase (CoA-acylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_530 \"butanal:NADP+ oxidoreductase (CoA-acylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_519 \"(S)-3-hydroxybutanoyl-CoA:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_519 \"(S)-3-hydroxybutanoyl-CoA:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_72 \"butanoyl-CoA:acetate CoA-transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_72 \"butanoyl-CoA:acetate CoA-transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_502 \"(S)-acetoin:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_502 \"(S)-acetoin:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_156 \"4-hydroxybutanoyl-CoA hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_156 \"4-hydroxybutanoyl-CoA hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_615 \"(R)-acetoin:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_615 \"(R)-acetoin:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_504 \"trans-2-enoyl-CoA reductase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_504 \"trans-2-enoyl-CoA reductase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_507 \"succinate semialdehyde hydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_507 \"succinate semialdehyde hydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_18 \"(R)-3-hydroxybutanoate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_18 \"(R)-3-hydroxybutanoate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_463 \"oxidoreductase acting on the CH-OH group of donors with NAD+ as acceptor\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_463 \"oxidoreductase acting on the CH-OH group of donors with NAD+ as acceptor\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_500 \"carbonate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_500 \"carbonate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_379 \"(R)-acetoin racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_379 \"(R)-acetoin racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_175 \"butanal:NAD+ oxidoreductase (CoA-acylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_175 \"butanal:NAD+ oxidoreductase (CoA-acylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_6 \"2-acetolactate decarboxylation\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_6 \"2-acetolactate decarboxylation\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_287 \"Glutaconyl-CoA carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_287 \"Glutaconyl-CoA carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_272 \"phosphate butyryltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_272 \"phosphate butyryltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_21 \"oxidoreductase acting on the CH-OH group of donors with NADP+ as acceptor\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_21 \"oxidoreductase acting on the CH-OH group of donors with NADP+ as acceptor\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_251 \"butyrate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_251 \"butyrate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_523 \"acetolactate decarboxylase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_523 \"acetolactate decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_mg \"magnesium transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_mg \"magnesium transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_k4 \"potassium:H+ antiporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_k4 \"potassium:H+ antiporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_for \"Formate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_for \"Formate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_no \"nitrite transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_no \"nitrite transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_k3 \"potassium ion chanel\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_k3 \"potassium ion chanel\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_zn \"zinc transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_zn \"zinc transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_61 \"isocitrate:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_61 \"isocitrate:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_708 \"2-oxoglutarate dehydrogenase complex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_708 \"2-oxoglutarate dehydrogenase complex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_370 \"pyruvate carboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_370 \"pyruvate carboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_433 \"citrate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_433 \"citrate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_171 \"isocitrate dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_171 \"isocitrate dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_544 \"acetyl-CoA:oxaloacetate C-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_544 \"acetyl-CoA:oxaloacetate C-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_310 \"succinyl-CoA synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_310 \"succinyl-CoA synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_606 \"isocitrate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_606 \"isocitrate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_198 \"fumarate hydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_198 \"fumarate hydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_254 \"malate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_254 \"malate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_557 \"methionine adenosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_557 \"methionine adenosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_99 \"acetyl-CoA:L-homoserine O-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_99 \"acetyl-CoA:L-homoserine O-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_231 \"adenosylhomocysteinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_231 \"adenosylhomocysteinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_2 \"spermine N1-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_2 \"spermine N1-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_562 \"5-(methylthio)-2,3-dioxopentyl-phosphate phosphohydrolase (isomerizing)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_562 \"5-(methylthio)-2,3-dioxopentyl-phosphate phosphohydrolase (isomerizing)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_152 \"methionine synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_152 \"methionine synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_265 \"2-hydroxy-3-keto-5-methylthio-1-phosphopentene phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_265 \"2-hydroxy-3-keto-5-methylthio-1-phosphopentene phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_714 \"2,3-diketo-5-methylthiopentenyl-1-phosphate enolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_714 \"2,3-diketo-5-methylthiopentenyl-1-phosphate enolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_687 \"cystathionine gamma-synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_687 \"cystathionine gamma-synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_711 \"S-adenosylmethioninamine:putrescine 3-aminopropyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_711 \"S-adenosylmethioninamine:putrescine 3-aminopropyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_680 \"S-adenosyl-L-methionine carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_680 \"S-adenosyl-L-methionine carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_469 \"cystathionine gamma-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_469 \"cystathionine gamma-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_242 \"L-cystyeine sulfurtransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_242 \"L-cystyeine sulfurtransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_418 \"spermidine N1-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_418 \"spermidine N1-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_155 \"S-adenosylmethioninamine:spermidine 3-aminopropyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_155 \"S-adenosylmethioninamine:spermidine 3-aminopropyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_560 \"methionine racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_560 \"methionine racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_189 \"S-adenosyl-L-homocysteine homocysteinylribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_189 \"S-adenosyl-L-homocysteine homocysteinylribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_235 \"5-methylthioribulose-1-phosphate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_235 \"5-methylthioribulose-1-phosphate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_136 \"5-(methylthio)-2,3-dioxopentyl phosphate keto-enol-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_136 \"5-(methylthio)-2,3-dioxopentyl phosphate keto-enol-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_98 \"L-methionine methanothiol-lyase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_98 \"L-methionine methanothiol-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_259 \"homoserine O-succinyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_259 \"homoserine O-succinyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_358 \"O-acetyl-L-homoserine:methanethiol 3-amino-3-carboxypropyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_358 \"O-acetyl-L-homoserine:methanethiol 3-amino-3-carboxypropyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_309 \"S-(5-deoxy-D-ribos-5-yl)-L-homocysteine homocysteine-lyase [(4S)-4,5-dihydroxypentan-2,3-dione-forming]indole-3-acetate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_309 \"S-(5-deoxy-D-ribos-5-yl)-L-homocysteine homocysteine-lyase [(4S)-4,5-dihydroxypentan-2,3-dione-forming]indole-3-acetate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_410 \"ATP:S-methyl-5-thio-D-ribose 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_410 \"ATP:S-methyl-5-thio-D-ribose 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_737 \"S-methyl-5-thio-alpha-D-ribose 1-phosphate ketol-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_737 \"S-methyl-5-thio-alpha-D-ribose 1-phosphate ketol-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_127 \"methylthioadenosine methylthioribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_127 \"methylthioadenosine methylthioribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_94 \"cystathionine beta-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_94 \"cystathionine beta-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_167 \"L-homocysteine Methyltransferases\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_167 \"L-homocysteine Methyltransferases\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_664 \"ornithine racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_664 \"ornithine racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_443 \"2-amino-4-ketopentanoate thiolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_443 \"2-amino-4-ketopentanoate thiolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_243 \"D-ornithine 4,5-aminomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_243 \"D-ornithine 4,5-aminomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_319 \"D-threo-2,4-diaminopentanoate:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_319 \"D-threo-2,4-diaminopentanoate:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_193 \"D-threo-2,4-diaminopentanoate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_193 \"D-threo-2,4-diaminopentanoate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ca2 \"Exchange Ca2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ca2 \"Exchange Ca2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_tyrL \"Exchange L-tyrosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_tyrL \"Exchange L-tyrosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_chor \"Exchange chorismate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_chor \"Exchange chorismate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_pheL \"Exchange L-phenylalanine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_pheL \"Exchange L-phenylalanine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_fru \"Exchange D-fructose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_fru \"Exchange D-fructose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_serL \"Exchange L-serine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_serL \"Exchange L-serine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_cysL \"Exchange L-cysteine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_cysL \"Exchange L-cysteine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_adn \"Exchange adenosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_adn \"Exchange adenosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_dcyt \"Exchange deoxycytidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_dcyt \"Exchange deoxycytidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_mnl \"Exchange mannitol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_mnl \"Exchange mannitol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_tagd \"Exchange D-Tagaturonate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_tagd \"Exchange D-Tagaturonate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_k \"Exchange K+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_k \"Exchange K+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_so3 \"Exchange sulfite\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_so3 \"Exchange sulfite\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_chol \"Exchange choline\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_chol \"Exchange choline\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_uri \"Exchange uridine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_uri \"Exchange uridine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ura \"Exchange uracil\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ura \"Exchange uracil\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_glc \"Exchange beta-D-glucose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_glc \"Exchange beta-D-glucose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_nac \"Exchange nicotinate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_nac \"Exchange nicotinate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_argL \"Exchange L-arginine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_argL \"Exchange L-arginine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_udpagaepyr \"Exchange UDP-N-acetylglucosamine-enolpyruvate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_udpagaepyr \"Exchange UDP-N-acetylglucosamine-enolpyruvate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_hxan \"Exchange hypoxanthine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_hxan \"Exchange hypoxanthine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_metL \"Exchange L-methionine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_metL \"Exchange L-methionine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_thm \"Exchange thiamin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_thm \"Exchange thiamin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_proL \"Exchange L-proline\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_proL \"Exchange L-proline\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_aspL \"Exchange L-aspartate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_aspL \"Exchange L-aspartate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ribflv \"Exchange riboflavin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ribflv \"Exchange riboflavin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_hco3 \"Exchange bicarbonate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_hco3 \"Exchange bicarbonate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_h \"Exchange H+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_h \"Exchange H+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_valL \"Exchange L-valine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_valL \"Exchange L-valine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_4abz \"Exchange 4-aminobenzoate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_4abz \"Exchange 4-aminobenzoate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_thymd \"Exchange thymidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_thymd \"Exchange thymidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_pntoR \"Exchange (R)-pantothenate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_pntoR \"Exchange (R)-pantothenate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_dad \"Exchange deoxyadenosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_dad \"Exchange deoxyadenosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_g3p \"Exchange sn-glycerol-3-phosphate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_g3p \"Exchange sn-glycerol-3-phosphate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_na \"Exchange Na+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_na \"Exchange Na+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_gly \"Exchange glycine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_gly \"Exchange glycine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_h2o \"Exchange H2O\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_h2o \"Exchange H2O\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_pyin \"Exchange pyridoxine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_pyin \"Exchange pyridoxine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_leuL \"Exchange L-leucine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_leuL \"Exchange L-leucine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ileL \"Exchange L-isoleucine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ileL \"Exchange L-isoleucine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_cobalt \"Exchange Co2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_cobalt \"Exchange Co2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_zn2 \"Exchange Zn2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_zn2 \"Exchange Zn2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_pi \"Exchange phosphate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_pi \"Exchange phosphate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_gluL \"Exchange L-glutamate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_gluL \"Exchange L-glutamate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_glnL \"Exchange L-glutamine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_glnL \"Exchange L-glutamine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_asnL \"Exchange L-asparagine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_asnL \"Exchange L-asparagine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_thrL \"Exchange L-threonine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_thrL \"Exchange L-threonine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_trpL \"Exchange L-tryptophan\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_trpL \"Exchange L-tryptophan\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_nh3 \"Exchange ammonia\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_nh3 \"Exchange ammonia\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_co2 \"Exchange CO2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_co2 \"Exchange CO2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_alaL \"Exchange L-alanine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_alaL \"Exchange L-alanine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ribD \"Exchange D-ribose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ribD \"Exchange D-ribose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_h2 \"Exchange H2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_h2 \"Exchange H2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_his \"Exchange L-histidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_his \"Exchange L-histidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_bio \"Exchange biotin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_bio \"Exchange biotin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_mg2 \"Exchange Mg2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_mg2 \"Exchange Mg2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_lysL \"Exchange L-lysine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_lysL \"Exchange L-lysine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ins \"Exchange inosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ins \"Exchange inosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_c1ala \"Exchange cob(I)alamin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_c1ala \"Exchange cob(I)alamin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_ncam \"Exchange nicotinamide\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_ncam \"Exchange nicotinamide\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_400 \"butanoyl-CoA dehydrogenase (FAD)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_400 \"butanoyl-CoA dehydrogenase (FAD)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_325 \"enoyl-CoA hydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_325 \"enoyl-CoA hydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_481 \"acetyl-CoA C-acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_481 \"acetyl-CoA C-acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_390 \"3-hydroxyacyl-CoA dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_390 \"3-hydroxyacyl-CoA dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_471 \"acyl-CoA hydrolase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_471 \"acyl-CoA hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_60 \"Beta-Oxidation of Palmitoyl-CoA\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_60 \"Beta-Oxidation of Palmitoyl-CoA\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_409 \"(S)-3-Hydroxybutanoyl-CoA 3-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_409 \"(S)-3-Hydroxybutanoyl-CoA 3-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_341 \"dihydroneopterin monophosphate dephosphorylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_341 \"dihydroneopterin monophosphate dephosphorylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_149 \"6-hydroxymethyl-7,8-dihydropteridine:4-aminobenzoate 2-amino-4-hydroxydihydropteridine-6-methenyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_149 \"6-hydroxymethyl-7,8-dihydropteridine:4-aminobenzoate 2-amino-4-hydroxydihydropteridine-6-methenyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_457 \"dihydroneopterin triphosphate pyrophosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_457 \"dihydroneopterin triphosphate pyrophosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_180 \"2-amino-4-hydroxy-6-hydroxymethyldihydropteridine diphosphokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_180 \"2-amino-4-hydroxy-6-hydroxymethyldihydropteridine diphosphokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_495 \"4-amino-4-deoxychorismate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_495 \"4-amino-4-deoxychorismate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_75 \"dihydrofolate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_75 \"dihydrofolate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_438 \"4-amino-4-deoxychorismate pyruvate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_438 \"4-amino-4-deoxychorismate pyruvate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_104 \"D-aminocylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_104 \"D-aminocylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_344 \"dihydropteroate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_344 \"dihydropteroate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_524 \"GTP 7,8-8,9-dihydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_524 \"GTP 7,8-8,9-dihydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_172 \"dihydroneopterin aldolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_172 \"dihydroneopterin aldolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_64 \"phosphomannomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_64 \"phosphomannomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_219 \"1-phosphofructokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_219 \"1-phosphofructokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_545 \"UTP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_545 \"UTP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_541 \"D-sobitol-6-phosphate:NAD+ 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_541 \"D-sobitol-6-phosphate:NAD+ 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_726 \"mannose-1-phosphate guanylyltransferase (GDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_726 \"mannose-1-phosphate guanylyltransferase (GDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_610 \"ATP:D-hexose 6 phosphotransferase alpha galactose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_610 \"ATP:D-hexose 6 phosphotransferase alpha galactose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_253 \"ATP:D-hexose 6 phosphotransferase mannose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_253 \"ATP:D-hexose 6 phosphotransferase mannose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_276 \"fructose-bisphosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_276 \"fructose-bisphosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_567 \"sedoheptulose-7-phosphate:D-glyceraldehyde-3-phosphate glyceronetransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_567 \"sedoheptulose-7-phosphate:D-glyceraldehyde-3-phosphate glyceronetransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_542 \"ATP:D-fructose 6-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_542 \"ATP:D-fructose 6-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_732 \"D-fructose 1,6-bisphosphate D-glyceraldehyde-3-phosphate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_732 \"D-fructose 1,6-bisphosphate D-glyceraldehyde-3-phosphate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_302 \"ATP:D-hexose 6 phosphotransferase beta galactose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_302 \"ATP:D-hexose 6 phosphotransferase beta galactose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_353 \"L-fuculose-1-phosphate lactaldehyde-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_353 \"L-fuculose-1-phosphate lactaldehyde-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_290 \"ITP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_290 \"ITP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_674 \"sedoheptulose 1,7-bisphosphate D-glyceraldehyde-3-phosphate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_674 \"sedoheptulose 1,7-bisphosphate D-glyceraldehyde-3-phosphate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_613 \"ATP:sedoheptulose-7-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_613 \"ATP:sedoheptulose-7-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_436 \"alpha-D-Glucose aldose-ketose-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_436 \"alpha-D-Glucose aldose-ketose-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_649 \"mannose-6-phosphate isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_649 \"mannose-6-phosphate isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_611 \"mannitol-1-phosphate 5-dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_611 \"mannitol-1-phosphate 5-dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_676 \"sorbitol dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_676 \"sorbitol dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_527 \"CTP:D-fructose-6-phosphate 1-phosphotransferase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_527 \"CTP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_173 \"UTP hexose-1-phosphate uridylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_173 \"UTP hexose-1-phosphate uridylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_199 \"fructose-bisphosphate aldolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_199 \"fructose-bisphosphate aldolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_297 \"ATP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_297 \"ATP:D-fructose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_659 \"mannose-1-phosphate guanylyltransferase (GTP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_659 \"mannose-1-phosphate guanylyltransferase (GTP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_333 \"Fructokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_333 \"Fructokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_491 \"CTP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_491 \"CTP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_329 \"D-galactose 1-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_329 \"D-galactose 1-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_255 \"D-glucose 1-phosphate 1-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_255 \"D-glucose 1-phosphate 1-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_532 \"ATP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_532 \"ATP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_742 \"Galactitol-1-phosphate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_742 \"Galactitol-1-phosphate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_417 \"UDP-D-glucose:alpha-D-galactose 1-phosphate uridylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_417 \"UDP-D-glucose:alpha-D-galactose 1-phosphate uridylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_444 \"UDP-glucose 4-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_444 \"UDP-glucose 4-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_480 \"UTP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_480 \"UTP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_763 \"galactokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_763 \"galactokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_205 \"tagatose-bisphosphate aldolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_205 \"tagatose-bisphosphate aldolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_631 \"ITP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_631 \"ITP:tagatose-6-phosphate 1-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_633 \"galactose-6-phosphate isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_633 \"galactose-6-phosphate isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_9 \"glycerol dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_9 \"glycerol dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_429 \"1,2-diacylglycerol 3-glucosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_429 \"1,2-diacylglycerol 3-glucosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_298 \"glycerol dehydrogenase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_298 \"glycerol dehydrogenase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_549 \"acyl-CoA:1-acyl-sn-glycerol-3-phosphate 2-O-acyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_549 \"acyl-CoA:1-acyl-sn-glycerol-3-phosphate 2-O-acyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_455 \"diacylglycerol kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_455 \"diacylglycerol kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_678 \"aldehyde dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_678 \"aldehyde dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_273 \"palmitate biosynthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_273 \"palmitate biosynthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_477 \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminoheptanedioate:D-alanyl-D-alanine ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_477 \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminoheptanedioate:D-alanyl-D-alanine ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_86 \"hexosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_86 \"hexosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_564 \"glycerol kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_564 \"glycerol kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_224 \"phosphatidylglycerol-membrane-oligosaccharide\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_224 \"phosphatidylglycerol-membrane-oligosaccharide\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_740 \"Acyl-CoA thioesterase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_740 \"Acyl-CoA thioesterase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_109 \"glycerol-3-phosphate dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_109 \"glycerol-3-phosphate dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_204 \"CDP-diacylglycerol phosphatidylhydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_204 \"CDP-diacylglycerol phosphatidylhydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_588 \"CDP-diacylglycerol-glycerol-3-phosphate 3-phosphatidyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_588 \"CDP-diacylglycerol-glycerol-3-phosphate 3-phosphatidyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_56 \"glycerol-3-phosphate dehydrogenase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_56 \"glycerol-3-phosphate dehydrogenase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_546 \"phosphatidate cytidylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_546 \"phosphatidate cytidylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_78 \"glycerol-3-phosphate O-acyltransferase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_78 \"glycerol-3-phosphate O-acyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_185 \"phosphatidylglycerophosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_185 \"phosphatidylglycerophosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_232 \"cardiolipin biosynthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_232 \"cardiolipin biosynthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_267 \"glycerol-3-phosphate cytidylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_267 \"glycerol-3-phosphate cytidylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_340 \"ethanolamine ammonia-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_340 \"ethanolamine ammonia-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_249 \"2-amino-3-oxobutanoate CoA-ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_249 \"2-amino-3-oxobutanoate CoA-ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_751 \"Glycine dehydrogenase (aminomethyl-transferring)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_751 \"Glycine dehydrogenase (aminomethyl-transferring)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_501 \"D-serine ammonia-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_501 \"D-serine ammonia-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_165 \"homoserine dehydrogenase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_165 \"homoserine dehydrogenase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_283 \"phosphoserine phosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_283 \"phosphoserine phosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_494 \"aspartate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_494 \"aspartate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_556 \"threonine aldolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_556 \"threonine aldolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_625 \"L-threonine:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_625 \"L-threonine:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_240 \"threonine synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_240 \"threonine synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_130 \"betaine aldehyde:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_130 \"betaine aldehyde:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_536 \"L-cysteinylglycine dipeptidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_536 \"L-cysteinylglycine dipeptidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_293 \"choline dehydrogenase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_293 \"choline dehydrogenase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_330 \"L-threonine dehydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_330 \"L-threonine dehydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_143 \"glycine hydroxymethyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_143 \"glycine hydroxymethyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_452 \"phosphoglycerate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_452 \"phosphoglycerate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_743 \"serine racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_743 \"serine racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_665 \"3-phospho-L-serinephosphoserine transaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_665 \"3-phospho-L-serinephosphoserine transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_132 \"homoserine dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_132 \"homoserine dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_124 \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate 6-phosphomannohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_124 \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate 6-phosphomannohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_609 \"glycerate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_609 \"glycerate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_385 \"homoserine kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_385 \"homoserine kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_52 \"aspartate-semialdehyde dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_52 \"aspartate-semialdehyde dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_320 \"L-serine ammonia-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_320 \"L-serine ammonia-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_694 \"succinyl-CoA:glycine C-succinyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_694 \"succinyl-CoA:glycine C-succinyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_729 \"choline dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_729 \"choline dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_393 \"glyceraldehyde-3-phosphate dehydrogenase (phosphorylating) NADP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_393 \"glyceraldehyde-3-phosphate dehydrogenase (phosphorylating) NADP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_55 \"alpha-D-glucose 6-phosphate ketol-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_55 \"alpha-D-glucose 6-phosphate ketol-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_496 \"ATP:oxaloacetate carboxylyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_496 \"ATP:oxaloacetate carboxylyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_622 \"D-glyceraldehyde 3-phosphate:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_622 \"D-glyceraldehyde 3-phosphate:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_360 \"acetaldehyde:NAD+ oxidoreductase (acetate forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_360 \"acetaldehyde:NAD+ oxidoreductase (acetate forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_401 \"phosphoglucomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_401 \"phosphoglucomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_221 \"dihydrolipoamide dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_221 \"dihydrolipoamide dehydrogenase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_53 \"pyruvate:ferredoxin oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_53 \"pyruvate:ferredoxin oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_179 \"phosphoglycerate mutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_179 \"phosphoglycerate mutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_414 \"ATP:alpha-D-glucose 6-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_414 \"ATP:alpha-D-glucose 6-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_621 \"beta-D-glucose 6-phosphate ketol-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_621 \"beta-D-glucose 6-phosphate ketol-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_498 \"glucose-6-phosphate 1-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_498 \"glucose-6-phosphate 1-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_285 \"glyceraldehyde-3-phosphate dehydrogenase (phosphorylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_285 \"glyceraldehyde-3-phosphate dehydrogenase (phosphorylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_233 \"phosphoglycerate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_233 \"phosphoglycerate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_660 \"L-lactate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_660 \"L-lactate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_39 \"D-glucose 1-phosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_39 \"D-glucose 1-phosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_657 \"ATP:beta-D-glucose 6-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_657 \"ATP:beta-D-glucose 6-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_147 \"D-glucose 1-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_147 \"D-glucose 1-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_721 \"D-glucose aldose-ketose-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_721 \"D-glucose aldose-ketose-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_596 \"triose-phosphate isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_596 \"triose-phosphate isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_391 \"phosphopyruvate hydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_391 \"phosphopyruvate hydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_140 \"acetate:CoA ligase (AMP-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_140 \"acetate:CoA ligase (AMP-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_383 \"ethanol dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_383 \"ethanol dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_252 \"ATP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_252 \"ATP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_654 \"AMP:fructose-1,6-bisphosphate phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_654 \"AMP:fructose-1,6-bisphosphate phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_473 \"gluconolactonase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_473 \"gluconolactonase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_339 \"hydroxypyruvate isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_339 \"hydroxypyruvate isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_753 \"L-Serine:glyoxylate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_753 \"L-Serine:glyoxylate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_754 \"glycolate fermentation\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_754 \"glycolate fermentation\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_362 \"(3R)-3-Hydroxybutanoyl-CoA hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_362 \"(3R)-3-Hydroxybutanoyl-CoA hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_752 \"Glycolaldehyde dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_752 \"Glycolaldehyde dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_128 \"D-glycerate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_128 \"D-glycerate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_374 \"glycine cleavage\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_374 \"glycine cleavage\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_535 \"2-hydroxy-3-oxopropionate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_535 \"2-hydroxy-3-oxopropionate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_90 \"formate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_90 \"formate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_230 \"glycine oxidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_230 \"glycine oxidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_211 \"10-formyltetrahydrofolate aminohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_211 \"10-formyltetrahydrofolate aminohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_151 \"imidazole glycerol phosphate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_151 \"imidazole glycerol phosphate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_196 \"phosphoribosyl-AMP cyclohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_196 \"phosphoribosyl-AMP cyclohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_361 \"histidinol:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_361 \"histidinol:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_215 \"phosphoribosyl-ATP diphosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_215 \"phosphoribosyl-ATP diphosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_76 \"histidinol-phosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_76 \"histidinol-phosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_191 \"N-carbamoul-L-aspartate hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_191 \"N-carbamoul-L-aspartate hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_268 \"1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino]imidazole-4-carboxamide isomerase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_268 \"1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino]imidazole-4-carboxamide isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_338 \"ATP phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_338 \"ATP phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_713 \"histidinal:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_713 \"histidinal:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_164 \"histidinol-phosphate transaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_164 \"histidinol-phosphate transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_66 \"imidazoleglycerol-phosphate dehydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_66 \"imidazoleglycerol-phosphate dehydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_28 \"L-Leucine:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_28 \"L-Leucine:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_31 \"acyl-CoA dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_31 \"acyl-CoA dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_605 \"2-hydroxyisocaproate CoA transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_605 \"2-hydroxyisocaproate CoA transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_382 \"2-hydroxyisocaproyl-CoA dehydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_382 \"2-hydroxyisocaproyl-CoA dehydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_685 \"2-hydroxyisocaproate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_685 \"2-hydroxyisocaproate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_20 \"Isocaprenoyl-CoA CoA transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_20 \"Isocaprenoyl-CoA CoA transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_550 \"dihydrodipicolinate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_550 \"dihydrodipicolinate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_181 \"(R)-2-hydroxybutane-1,2,4-tricarboxylate hydro-lyase [(1R,2S)-1-hydroxybutane-1,2,4-tricarboxylate-forming]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_181 \"(R)-2-hydroxybutane-1,2,4-tricarboxylate hydro-lyase [(1R,2S)-1-hydroxybutane-1,2,4-tricarboxylate-forming]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_626 \"succinyl-diaminopimelate desuccinylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_626 \"succinyl-diaminopimelate desuccinylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_686 \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamate:meso-2,6-diaminoheptanedioate ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_686 \"UDP-N-acetylmuramoyl-L-alanyl-D-glutamate:meso-2,6-diaminoheptanedioate ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_115 \"diaminopimelate epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_115 \"diaminopimelate epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_82 \"(1R,2S)-1-hydroxybutane-1,2,4-tricarboxylate:NAD+ oxidoreductase (decarboxylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_82 \"(1R,2S)-1-hydroxybutane-1,2,4-tricarboxylate:NAD+ oxidoreductase (decarboxylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_725 \"acetyl-CoA:2-oxoglutarate C-acetyltransferase (thioester-hydrolysing, carboxymethyl forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_725 \"acetyl-CoA:2-oxoglutarate C-acetyltransferase (thioester-hydrolysing, carboxymethyl forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_120 \"dihydrodipicolinate reductase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_120 \"dihydrodipicolinate reductase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_701 \"dihydrodipicolinate reductase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_701 \"dihydrodipicolinate reductase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_462 \"succinyldiaminopimelate transaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_462 \"succinyldiaminopimelate transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_652 \"2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-succinyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_652 \"2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-succinyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_479 \"diaminopimelate decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_479 \"diaminopimelate decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_434 \"L-lysine 2,3-aminomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_434 \"L-lysine 2,3-aminomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_184 \"(3S,5S)-3,5-diaminohexanoate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_184 \"(3S,5S)-3,5-diaminohexanoate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_349 \"3-keto-5-aminohexanoate cleavage\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_349 \"3-keto-5-aminohexanoate cleavage\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_113 \"L-3-aminobutyryl-CoA amminoa-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_113 \"L-3-aminobutyryl-CoA amminoa-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_574 \"(3S)-3,6-diaminohexanoate 5,6-aminomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_574 \"(3S)-3,6-diaminohexanoate 5,6-aminomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_288 \"butyrate-acetoacetate CoA-transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_288 \"butyrate-acetoacetate CoA-transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_531 \"N-Ribosylnicotinamide:orthophosphate ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_531 \"N-Ribosylnicotinamide:orthophosphate ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_174 \"nicotinate-nucleotide diphosphorylase (carboxylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_174 \"nicotinate-nucleotide diphosphorylase (carboxylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_395 \"Nicotinamide amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_395 \"Nicotinamide amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_551 \"nicotinamide-nucleotide adenylyltransferase (NAD+ forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_551 \"nicotinamide-nucleotide adenylyltransferase (NAD+ forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_218 \"deamino-NAD+:ammonia ligase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_218 \"deamino-NAD+:ammonia ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_27 \"nicotinamide-nucleotide adenylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_27 \"nicotinamide-nucleotide adenylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_58 \"N-ribosylnicotinamide ribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_58 \"N-ribosylnicotinamide ribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_294 \"L-aspartate:NADP+ oxidoreductase (deaminating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_294 \"L-aspartate:NADP+ oxidoreductase (deaminating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_313 \"NAD+ synthase (glutamine-hydrolysing)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_313 \"NAD+ synthase (glutamine-hydrolysing)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_304 \"L-aspartate:NAD+ oxidoreductase (deaminating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_304 \"L-aspartate:NAD+ oxidoreductase (deaminating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_262 \"nicotinate mononucleotide:diphosphate phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_262 \"nicotinate mononucleotide:diphosphate phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_533 \"Nicotinamide ribonucleotide phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_533 \"Nicotinamide ribonucleotide phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_580 \"NAD+ phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_580 \"NAD+ phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_41 \"NADP+ 2'-phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_41 \"NADP+ 2'-phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_399 \"alpha-iminosucinate:dihydroxyacetone phosphate alkyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_399 \"alpha-iminosucinate:dihydroxyacetone phosphate alkyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_559 \"NAD+ glycohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_559 \"NAD+ glycohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_415 \"NAD+ kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_415 \"NAD+ kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_651 \"L-aspartate:fumarate oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_651 \"L-aspartate:fumarate oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_286 \"ferredoxin-nitrate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_286 \"ferredoxin-nitrate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_386 \"nitrate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_386 \"nitrate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_521 \"ferredoxin-nitrite reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_521 \"ferredoxin-nitrite reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_662 \"nitrite reductase [NADPH]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_662 \"nitrite reductase [NADPH]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_22 \"ammonia:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_22 \"ammonia:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_43 \"nitrite reductase [NADH]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_43 \"nitrite reductase [NADH]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_697 \"5,10-Methenyltetrahydrofolate:5'-phosphoribosylglycinamide formyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_697 \"5,10-Methenyltetrahydrofolate:5'-phosphoribosylglycinamide formyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_482 \"5-formiminotetrahydrofolate ammonia-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_482 \"5-formiminotetrahydrofolate ammonia-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_591 \"tetrahydrofolate:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_591 \"tetrahydrofolate:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_723 \"methylenetetrahydrofolate reductase (ferredoxin)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_723 \"methylenetetrahydrofolate reductase (ferredoxin)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_326 \"CO-methylating acetyl-CoA synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_326 \"CO-methylating acetyl-CoA synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_14 \"Spontaneous tetrahydrofolate transformation\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_14 \"Spontaneous tetrahydrofolate transformation\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_758 \"5,10-Methenyltetrahydrofolate aminomethyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_758 \"5,10-Methenyltetrahydrofolate aminomethyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_342 \"5-methyltetrahydrofolate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_342 \"5-methyltetrahydrofolate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_145 \"5,10-methenyltetrahydrofolate 5-hydrolase (decyclizing)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_145 \"5,10-methenyltetrahydrofolate 5-hydrolase (decyclizing)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_123 \"methylenetetrahydrofolate dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_123 \"methylenetetrahydrofolate dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_483 \"5-methyltetrahydrofolate:corrinoid Co-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_483 \"5-methyltetrahydrofolate:corrinoid Co-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_669 \"5-formyltetrahydrofolate cyclo-ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_669 \"5-formyltetrahydrofolate cyclo-ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_499 \"methylenetetrahydrofolate dehydrogenase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_499 \"methylenetetrahydrofolate dehydrogenase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_597 \"aminomethyltransferase (H-Gcv-protein)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_597 \"aminomethyltransferase (H-Gcv-protein)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_623 \"formate:tetrahydrofolate ligase (ADP-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_623 \"formate:tetrahydrofolate ligase (ADP-forming)\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_427 \"NADH:ubiquinol-8 oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_427 \"NADH:ubiquinol-8 oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_48 \"succinyl-CoA reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_48 \"succinyl-CoA reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_378 \"carbon dioxide:ferredoxin oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_378 \"carbon dioxide:ferredoxin oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_566 \"ferredoxin-NAD+ reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_566 \"ferredoxin-NAD+ reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_466 \"carbon monoxide:acceptor oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_466 \"carbon monoxide:acceptor oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_244 \"ferredoxin-NADP+ reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_244 \"ferredoxin-NADP+ reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_648 \"hydrogen:ferredoxin oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_648 \"hydrogen:ferredoxin oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_316 \"carbon dioxide reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_316 \"carbon dioxide reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_fru \"D-fructose transport phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_fru \"D-fructose transport phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_udpagaepyr \"UDP-N-acetylglucosamine-enolpyruvate Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_udpagaepyr \"UDP-N-acetylglucosamine-enolpyruvate Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_gtol \"Galactitol pep transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_gtol \"Galactitol pep transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_malt \"maltose transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_malt \"maltose transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_tre \"trehalose transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_tre \"trehalose transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_adman \"PEP dependant alpha-D-mannose Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_adman \"PEP dependant alpha-D-mannose Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glcAD2 \"alpha-D-glucose transport phosphotranferase \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glcAD2 \"alpha-D-glucose transport phosphotranferase \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_2o6pamg2d \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_2o6pamg2d \"2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_celb \"cellobiose transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_celb \"cellobiose transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_sucr \"sucrose transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_sucr \"sucrose transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_aga \"D-glucosamine transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_aga \"D-glucosamine transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_mnl \"mannitol transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_mnl \"mannitol transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_dachi \"diacetylchitobiose transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_dachi \"diacetylchitobiose transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_fru2 \"D-fructose transport phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_fru2 \"D-fructose transport phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_tgt \"Tagatose transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_tgt \"Tagatose transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_sor2 \"D-sorbitol transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_sor2 \"D-sorbitol transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glc \"D-glucose transport phosphotranferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glc \"D-glucose transport phosphotranferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_fuc \"L-fuculose PTS transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_fuc \"L-fuculose PTS transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_169 \"acetate CoA-transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_169 \"acetate CoA-transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_420 \"3-methyl-2-oxobutanoate hydroxymethyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_420 \"3-methyl-2-oxobutanoate hydroxymethyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_296 \"2-dehydropantoate formaldehyde-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_296 \"2-dehydropantoate formaldehyde-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_139 \"phosphopantothenoylcysteine decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_139 \"phosphopantothenoylcysteine decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_405 \"glutaryl-CoA dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_405 \"glutaryl-CoA dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_756 \"pantoate--beta-alanine ligase \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_756 \"pantoate--beta-alanine ligase \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_464 \"D-4'-phosphopantothenate:L-cysteine ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_464 \"D-4'-phosphopantothenate:L-cysteine ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_570 \"pantothenate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_570 \"pantothenate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_197 \"[acyl-carrier-protein] phosphodiesterase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_197 \"[acyl-carrier-protein] phosphodiesterase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_50 \"N-((R)-pantothenoyl)-L-cysteine carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_50 \"N-((R)-pantothenoyl)-L-cysteine carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_34 \"holo-[acyl-carrier-protein] synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_34 \"holo-[acyl-carrier-protein] synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_134 \"dephospho-CoA kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_134 \"dephospho-CoA kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_515 \"phosphopantothenate-cysteine ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_515 \"phosphopantothenate-cysteine ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_125 \"ATP:pantetheine 4'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_125 \"ATP:pantetheine 4'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_449 \"2-acetolactate pyruvate-lyase (carboxylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_449 \"2-acetolactate pyruvate-lyase (carboxylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_85 \"ATP:N-((R)-pantothenoyl)-L-cysteine 4-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_85 \"ATP:N-((R)-pantothenoyl)-L-cysteine 4-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_663 \"malonyl CoA-acyl carrier protein transacylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_663 \"malonyl CoA-acyl carrier protein transacylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_755 \"2-dehydropantoate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_755 \"2-dehydropantoate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_644 \"pantetheine-phosphate adenylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_644 \"pantetheine-phosphate adenylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_398 \"UTP-glucose-1-phosphate uridylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_398 \"UTP-glucose-1-phosphate uridylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_592 \"ribitol-5-phosphate 2-dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_592 \"ribitol-5-phosphate 2-dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_107 \"D-ribitol-5-phosphate:NAD+ 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_107 \"D-ribitol-5-phosphate:NAD+ 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_108 \"L-arabinose isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_108 \"L-arabinose isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_569 \"D-altronate:NAD+ 3-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_569 \"D-altronate:NAD+ 3-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_208 \"D-altronate hydro-lyase (2-dehydro-3-deoxy-D-gluconate-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_208 \"D-altronate hydro-lyase (2-dehydro-3-deoxy-D-gluconate-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_624 \"xylulokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_624 \"xylulokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_332 \"ribulokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_332 \"ribulokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_10 \"L-ribulose-5-phosphate 4-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_10 \"L-ribulose-5-phosphate 4-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_384 \"D-xylose aldose-ketose-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_384 \"D-xylose aldose-ketose-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_368 \"D-erythrose-4-phosphate:D-xylulose-5-phosphate glycolaldehyde transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_368 \"D-erythrose-4-phosphate:D-xylulose-5-phosphate glycolaldehyde transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_629 \"ribulose-phosphate 3-epimerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_629 \"ribulose-phosphate 3-epimerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_419 \"gluconokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_419 \"gluconokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_402 \"D-xylulose 5-phosphate D-glyceraldehyde-3-phosphate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_402 \"D-xylulose 5-phosphate D-glyceraldehyde-3-phosphate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_513 \"ribose-phosphate diphosphokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_513 \"ribose-phosphate diphosphokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_424 \"ribose-5-phosphate isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_424 \"ribose-5-phosphate isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_160 \"D-ribose 1,5-phosphomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_160 \"D-ribose 1,5-phosphomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_656 \"2-dehydro-3-deoxy-phosphogluconate aldolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_656 \"2-dehydro-3-deoxy-phosphogluconate aldolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_658 \"6-phospho-D-gluconate:NADP+ 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_658 \"6-phospho-D-gluconate:NADP+ 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_68 \"2-dehydro-3-deoxygluconokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_68 \"2-dehydro-3-deoxygluconokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_73 \"sedoheptulose 1,7-bisphosphate 1-phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_73 \"sedoheptulose 1,7-bisphosphate 1-phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_307 \"ATP:(D)-glycerate 2-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_307 \"ATP:(D)-glycerate 2-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_620 \"6-phospho-D-gluconate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_620 \"6-phospho-D-gluconate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_404 \"deoxyribokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_404 \"deoxyribokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_312 \"6-phosphogluconolactonase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_312 \"6-phosphogluconolactonase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_229 \"ribokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_229 \"ribokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_577 \"transketolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_577 \"transketolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_490 \"phosphopentomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_490 \"phosphopentomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_634 \"phosphate acetyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_634 \"phosphate acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_650 \"D-glucose:ubiquinone oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_650 \"D-glucose:ubiquinone oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_397 \"deoxyribose-phosphate aldolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_397 \"deoxyribose-phosphate aldolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_116 \"shikimate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_116 \"shikimate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_558 \"4-Hydroxyphenylacetaldehyde:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_558 \"4-Hydroxyphenylacetaldehyde:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_300 \"prephenate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_300 \"prephenate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_345 \"prephenate dehydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_345 \"prephenate dehydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_585 \"L-arogenate:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_585 \"L-arogenate:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_435 \"shikimate dehydrogenase (NADP+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_435 \"shikimate dehydrogenase (NADP+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_335 \"chorismate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_335 \"chorismate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_318 \"3-dehydroquinate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_318 \"3-dehydroquinate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_568 \"shikimate dehydrogenase (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_568 \"shikimate dehydrogenase (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_7 \"L-arogenate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_7 \"L-arogenate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_578 \"L-arogenate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_578 \"L-arogenate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_96 \"3-phosphoshikimate 1-carboxyvinyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_96 \"3-phosphoshikimate 1-carboxyvinyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_161 \"chorismate mutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_161 \"chorismate mutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_79 \"3-dehydroquinate dehydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_79 \"3-dehydroquinate dehydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_416 \"4-hydroxyphenylpyruvate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_416 \"4-hydroxyphenylpyruvate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_476 \"phenylalanine:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_476 \"phenylalanine:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_166 \"3-deoxy-7-phosphoheptulonate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_166 \"3-deoxy-7-phosphoheptulonate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_19 \"2-oxoadipate dehydrogenase complex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_19 \"2-oxoadipate dehydrogenase complex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_576 \"4-hydroxyphenylacetate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_576 \"4-hydroxyphenylacetate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_516 \"3-phosphonopyruvate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_516 \"3-phosphonopyruvate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_699 \"2-aminoethylphosphonate--pyruvate transaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_699 \"2-aminoethylphosphonate--pyruvate transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_598 \"2-Phosphonoacetaldehyde phosphonohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_598 \"2-Phosphonoacetaldehyde phosphonohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_57 \"phosphoenolpyruvate 2,3-phosphonomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_57 \"phosphoenolpyruvate 2,3-phosphonomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_492 \"acetaldehyde:NAD+ oxidoreductase (CoA-acylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_492 \"acetaldehyde:NAD+ oxidoreductase (CoA-acylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_303 \"Phosphonoacetate phosphonohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_303 \"Phosphonoacetate phosphonohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_248 \"phosphonate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_248 \"phosphonate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_376 \"diphosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_376 \"diphosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_241 \"phosphonoacetate hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_241 \"phosphonoacetate hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_693 \"triphosphate phosphohydrlase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_693 \"triphosphate phosphohydrlase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_586 \"4-ureido-5-imidazole carboxylate amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_586 \"4-ureido-5-imidazole carboxylate amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_263 \"ribonucleotide-reductase (ATP)\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_263 \"ribonucleotide-reductase (ATP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_192 \"xanthosine 5'-phosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_192 \"xanthosine 5'-phosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_706 \"xanthosine ribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_706 \"xanthosine ribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_77 \"deoxyadenosine kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_77 \"deoxyadenosine kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_603 \"glycine reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_603 \"glycine reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_226 \"deoxyinosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_226 \"deoxyinosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_126 \"GTP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_126 \"GTP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_26 \"dGTP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_26 \"dGTP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_80 \"adenosine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_80 \"adenosine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_350 \"hypoxanthine:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_350 \"hypoxanthine:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_672 \"ATP:dGDP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_672 \"ATP:dGDP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_421 \"ADP-ribose ribophosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_421 \"ADP-ribose ribophosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_489 \"4-aminoimidazole aminohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_489 \"4-aminoimidazole aminohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_683 \"ATP:dGMP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_683 \"ATP:dGMP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_451 \"4-ureido-5-imidazole carboxylate decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_451 \"4-ureido-5-imidazole carboxylate decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_275 \"XMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_275 \"XMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_111 \"adenylate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_111 \"adenylate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_762 \"ATP:deoxyguanosine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_762 \"ATP:deoxyguanosine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_305 \"amidophosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_305 \"amidophosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_83 \"guanosine ribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_83 \"guanosine ribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_237 \"GDP diphosphokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_237 \"GDP diphosphokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_589 \"5-formiminotetrahydrofolate:glycine N-formiminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_589 \"5-formiminotetrahydrofolate:glycine N-formiminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_733 \"guanosine-3',5'-bis(diphosphate) 3'-diphosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_733 \"guanosine-3',5'-bis(diphosphate) 3'-diphosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_392 \"GMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_392 \"GMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_642 \"phosphoribosylformylglycinamidine cyclo-ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_642 \"phosphoribosylformylglycinamidine cyclo-ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_159 \"Inosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_159 \"Inosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_632 \"dATP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_632 \"dATP:pyruvate 2-O-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_604 \"guanosine 5'-monophosphate phosphahydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_604 \"guanosine 5'-monophosphate phosphahydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_372 \"xanthine amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_372 \"xanthine amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_406 \"inosine ribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_406 \"inosine ribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_100 \"deoxyadenosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_100 \"deoxyadenosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_537 \"deoxyadenylate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_537 \"deoxyadenylate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_343 \"guanosine 3'-diphosphate 5'-triphosphate 5'-phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_343 \"guanosine 3'-diphosphate 5'-triphosphate 5'-phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_475 \"deoxyadenosine deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_475 \"deoxyadenosine deaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_30 \"2'-Deoxyadenosine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_30 \"2'-Deoxyadenosine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_81 \"ATP:GMP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_81 \"ATP:GMP phosphotransferase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_148 \"glycinamide ribonucleotide transformylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_148 \"glycinamide ribonucleotide transformylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_730 \"1-(5'-Phosphoribosyl)-5-amino-4-(N-succinocarboxamide)-imidazole AMP-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_730 \"1-(5'-Phosphoribosyl)-5-amino-4-(N-succinocarboxamide)-imidazole AMP-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_681 \"Ap3A synthetase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_681 \"Ap3A synthetase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_274 \"phosphoribosylformylglycinamidine synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_274 \"phosphoribosylformylglycinamidine synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_745 \"IMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_745 \"IMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_95 \"deoxyguanosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_95 \"deoxyguanosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_724 \"adenosine deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_724 \"adenosine deaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_666 \"IMP cyclohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_666 \"IMP cyclohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_412 \"10-Formyltetrahydrofolate:5'-phosphoribosylglycinamide formyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_412 \"10-Formyltetrahydrofolate:5'-phosphoribosylglycinamide formyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_736 \"ATP:dADP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_736 \"ATP:dADP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_423 \"deoxyribose-1-phosphate isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_423 \"deoxyribose-1-phosphate isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_168 \"adenylosuccinate lyase (fumarate-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_168 \"adenylosuccinate lyase (fumarate-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_157 \"GMP synthase (glutamine-hydrolysing)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_157 \"GMP synthase (glutamine-hydrolysing)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_183 \"guanine deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_183 \"guanine deaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_728 \"ribonucleoside-diphosphate reductase (ADP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_728 \"ribonucleoside-diphosphate reductase (ADP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_682 \"phosphoribosylaminoimidazolecarboxamide formyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_682 \"phosphoribosylaminoimidazolecarboxamide formyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_503 \"phosphoribosylaminoimidazolesuccinocarboxamide synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_503 \"phosphoribosylaminoimidazolesuccinocarboxamide synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_201 \"P1,P3-bis(5'-adenosyl)-triphosphate adenylohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_201 \"P1,P3-bis(5'-adenosyl)-triphosphate adenylohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_595 \"adenylosuccinate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_595 \"adenylosuccinate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_63 \"AMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_63 \"AMP:diphosphate phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_437 \"4-imidazolonase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_437 \"4-imidazolonase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_170 \"ribonucleotide-triphosphate reductase (GTP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_170 \"ribonucleotide-triphosphate reductase (GTP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_381 \"ATP:GDP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_381 \"ATP:GDP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_619 \"xanthosine:phosphate ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_619 \"xanthosine:phosphate ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_282 \"adenosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_282 \"adenosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_387 \"phosphoribosylamine-glycine ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_387 \"phosphoribosylamine-glycine ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_739 \"2'-Deoxyguanosine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_739 \"2'-Deoxyguanosine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_684 \"guanosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_684 \"guanosine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_118 \"phosphoribosylaminoimidazole carboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_118 \"phosphoribosylaminoimidazole carboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_679 \"ATP phosphohydrolase (NGAM)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_679 \"ATP phosphohydrolase (NGAM)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_195 \"xanthosine 5'-phosphate:ammonia ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_195 \"xanthosine 5'-phosphate:ammonia ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_703 \"ribonucleoside-diphosphate reductase (GDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_703 \"ribonucleoside-diphosphate reductase (GDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_512 \"carbamate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_512 \"carbamate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_425 \"IMP dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_425 \"IMP dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_612 \"thioredoxin oxydation\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_612 \"thioredoxin oxydation\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_673 \"ATP:GTP 3'-diphosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_673 \"ATP:GTP 3'-diphosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_202 \"guanosine 3'-diphosphate 5'-triphosphate 3'-diphosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_202 \"guanosine 3'-diphosphate 5'-triphosphate 3'-diphosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_655 \"adenosine ribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_655 \"adenosine ribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_271 \"(S)-dihydroorotate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_271 \"(S)-dihydroorotate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_707 \"(S)-dihydroorotate:fumarate oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_707 \"(S)-dihydroorotate:fumarate oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_696 \"dCTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_696 \"dCTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_62 \"ATP:dUMP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_62 \"ATP:dUMP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_460 \"ribonucleotide-triphosphate reductase (UTP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_460 \"ribonucleotide-triphosphate reductase (UTP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_486 \"3-ureido-isobutyrate amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_486 \"3-ureido-isobutyrate amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_277 \"ribonucleotide-diphosphate reductase (UDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_277 \"ribonucleotide-diphosphate reductase (UDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_1 \"phosphoenolpyruvate kinase (IDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_1 \"phosphoenolpyruvate kinase (IDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_87 \"ATP:CMP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_87 \"ATP:CMP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_715 \"UTP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_715 \"UTP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_327 \"dATP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_327 \"dATP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_593 \"orotate phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_593 \"orotate phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_582 \"phosphoenolpyruvate kinase (CDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_582 \"phosphoenolpyruvate kinase (CDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_97 \"ribonucleotide-diphosphate reductase (CDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_97 \"ribonucleotide-diphosphate reductase (CDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_8 \"UTP diphosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_8 \"UTP diphosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_641 \"dCTP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_641 \"dCTP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_600 \"dGTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_600 \"dGTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_206 \"ATP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_206 \"ATP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_102 \"5,6-Dihydrouracil amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_102 \"5,6-Dihydrouracil amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_69 \"dCTP deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_69 \"dCTP deaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_13 \"nucleoside-phosphate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_13 \"nucleoside-phosphate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_216 \"aspartate carbamoyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_216 \"aspartate carbamoyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_363 \"thymidine phosphorylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_363 \"thymidine phosphorylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_628 \"CTP phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_628 \"CTP phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_614 \"uridine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_614 \"uridine:phosphate alpha-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_511 \"5,6-dihydrouracil:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_511 \"5,6-dihydrouracil:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_579 \"cytosine deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_579 \"cytosine deaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_144 \"CTP synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_144 \"CTP synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_677 \"(S)-dihydroorotate:quinone oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_677 \"(S)-dihydroorotate:quinone oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_505 \"ATP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_505 \"ATP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_289 \"carbamoyl-phosphate synthase (glutamine-hydrolysing) (part 1)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_289 \"carbamoyl-phosphate synthase (glutamine-hydrolysing) (part 1)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_74 \"thymidine kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_74 \"thymidine kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_54 \"hydantoinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_54 \"hydantoinase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_661 \"nucleoside-diphosphate kinase (dTDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_661 \"nucleoside-diphosphate kinase (dTDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_299 \"GTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_299 \"GTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_448 \"nucleoside-diphosphate kinase (dCDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_448 \"nucleoside-diphosphate kinase (dCDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_38 \"5,10-methylenetetrahydrofolate,NADPH:dUMP C-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_38 \"5,10-methylenetetrahydrofolate,NADPH:dUMP C-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_377 \"GTP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_377 \"GTP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_608 \"pyrimidine-nucleoside phosphorylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_608 \"pyrimidine-nucleoside phosphorylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_690 \"nucleoside-diphosphate kinase (CDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_690 \"nucleoside-diphosphate kinase (CDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_213 \"deoxyuridine:phosphate ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_213 \"deoxyuridine:phosphate ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_371 \"dihydroorotase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_371 \"dihydroorotase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_695 \"CTP diphosphohydrolase (diphosphate-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_695 \"CTP diphosphohydrolase (diphosphate-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_547 \"dTTP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_547 \"dTTP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_182 \"deoxycytidine aminohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_182 \"deoxycytidine aminohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_653 \"ITP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_653 \"ITP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_122 \"UDP phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_122 \"UDP phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_334 \"nucleoside-diphosphate kinase (UDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_334 \"nucleoside-diphosphate kinase (UDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_65 \"2'-Deoxycytidine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_65 \"2'-Deoxycytidine 5'-monophosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_590 \"nucleoside-triphosphate diphosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_590 \"nucleoside-triphosphate diphosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_522 \"N-carbamoyl-beta-alanine amidohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_522 \"N-carbamoyl-beta-alanine amidohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_133 \"carbamoyl-phosphate synthase (glutamine-hydrolysing) (part 2)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_133 \"carbamoyl-phosphate synthase (glutamine-hydrolysing) (part 2)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_5 \"dTMP kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_5 \"dTMP kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_517 \"CDP phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_517 \"CDP phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_667 \"ITP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_667 \"ITP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_441 \"UTP:cytidine 5'-phosphotransferase Formate acetyltransferase-glycine dihydroflavodoxin:S-adenosyl-L-methionine oxidoreductase (S-adenosyl-L-methionine cleaving)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_441 \"UTP:cytidine 5'-phosphotransferase Formate acetyltransferase-glycine dihydroflavodoxin:S-adenosyl-L-methionine oxidoreductase (S-adenosyl-L-methionine cleaving)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_407 \"phosphoenolpyruvate kinase (UDP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_407 \"phosphoenolpyruvate kinase (UDP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_239 \"dUTP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_239 \"dUTP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_137 \"ribonucleotide-triphosphate reductase (CTP)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_137 \"ribonucleotide-triphosphate reductase (CTP)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_117 \"uracil phosphoribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_117 \"uracil phosphoribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_29 \"thymidylate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_29 \"thymidylate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_447 \"orotidine-5'-phosphate decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_447 \"orotidine-5'-phosphate decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_734 \"5,6-dihydrothymine:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_734 \"5,6-dihydrothymine:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_348 \"dUTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_348 \"dUTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_413 \"dATP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_413 \"dATP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_269 \"ATP:dCMP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_269 \"ATP:dCMP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_16 \"UTP phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_16 \"UTP phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_261 \"dCMP deaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_261 \"dCMP deaminase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_534 \"thioredoxin-disulfide reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_534 \"thioredoxin-disulfide reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_718 \"pseudouridine-5'-phosphate glycosidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_718 \"pseudouridine-5'-phosphate glycosidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_40 \"cytidine aminohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_40 \"cytidine aminohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_528 \"uridine ribohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_528 \"uridine ribohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_601 \"ATP:dUTP phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_601 \"ATP:dUTP phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_750 \"Thymidylate 5'-phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_750 \"Thymidylate 5'-phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_442 \"dGTP:uridine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_442 \"dGTP:uridine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_432 \"dTTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_432 \"dTTP:cytidine 5'-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_153 \"acetyl-CoA carboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_153 \"acetyl-CoA carboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_474 \"V-type sodium ATP synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_474 \"V-type sodium ATP synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_565 \"oxaloacetate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_565 \"oxaloacetate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_32 \"pyruvate:NAD+ 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_32 \"pyruvate:NAD+ 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_35 \"(S)-lactaldehyde dehydrogenase \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_35 \"(S)-lactaldehyde dehydrogenase \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_49 \"ATP synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_49 \"ATP synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_44 \"lactate:ubiquinone-8 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_44 \"lactate:ubiquinone-8 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_322 \"lactate racemase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_322 \"lactate racemase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_280 \"acetate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_280 \"acetate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_188 \"ATP:pyruvate,phosphate phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_188 \"ATP:pyruvate,phosphate phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_163 \"oxaloacetate:phosphate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_163 \"oxaloacetate:phosphate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_563 \"pyruvate, water dikinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_563 \"pyruvate, water dikinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_186 \"(R)-3-amino-2-methylpropanoate:pyruvate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_186 \"(R)-3-amino-2-methylpropanoate:pyruvate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_458 \"D-lactate dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_458 \"D-lactate dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_394 \"methylglyoxalase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_394 \"methylglyoxalase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_177 \"pyruvaldehyde:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_177 \"pyruvaldehyde:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_89 \"acetyl:succinate CoA-transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_89 \"acetyl:succinate CoA-transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_105 \"S-lactoyl-glutathione hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_105 \"S-lactoyl-glutathione hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_306 \"acetoacetate decarboxylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_306 \"acetoacetate decarboxylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_246 \"fumarate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_246 \"fumarate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_291 \"malate dehydrogenase (oxaloacetate-decarboxylating) (NAD+)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_291 \"malate dehydrogenase (oxaloacetate-decarboxylating) (NAD+)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_716 \"methylglyoxal synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_716 \"methylglyoxal synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_36 \"pyruvate formate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_36 \"pyruvate formate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_710 \"(R)-2-hydroxy-acid:acceptor 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_710 \"(R)-2-hydroxy-acid:acceptor 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_234 \"3-mercaptopyruvate:cyanide sulfurtransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_234 \"3-mercaptopyruvate:cyanide sulfurtransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_643 \"lactate:menaquinone 2-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_643 \"lactate:menaquinone 2-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_702 \"FMN adenylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_702 \"FMN adenylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_459 \"FMNH2:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_459 \"FMNH2:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_692 \"FMNH2:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_692 \"FMNH2:NADP+ oxidoreductase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_162 \"riboflavin kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_162 \"riboflavin kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_257 \"D-ribulose 5-phosphate formate-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_257 \"D-ribulose 5-phosphate formate-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_375 \"5-amino-6-(5-phosphoribitylamino)uracil phosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_375 \"5-amino-6-(5-phosphoribitylamino)uracil phosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_356 \"GTP cyclohydrolase II\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_356 \"GTP cyclohydrolase II\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_599 \"5-amino-6-(D-ribitylamino)uracil butanedionetransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_599 \"5-amino-6-(D-ribitylamino)uracil butanedionetransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_250 \"5-amino-6-(5-phosphoribosylamino)uracil:NADP+ 1'-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_250 \"5-amino-6-(5-phosphoribosylamino)uracil:NADP+ 1'-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_279 \"5-amino-6-(5-phosphoribosylamino)uracil reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_279 \"5-amino-6-(5-phosphoribosylamino)uracil reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_110 \"riboflavin synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_110 \"riboflavin synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_na \"Secretion Na+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_na \"Secretion Na+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_glyo \"Secretion glycerol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_glyo \"Secretion glycerol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_tmam \"Secretion trimethylamine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_tmam \"Secretion trimethylamine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ribflv \"Secretion riboflavin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ribflv \"Secretion riboflavin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_dcyt \"Secretion deoxycytidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_dcyt \"Secretion deoxycytidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_aga \"Secretion N-acetyl-beta-D-glucosamine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_aga \"Secretion N-acetyl-beta-D-glucosamine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_h \"Secretion H+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_h \"Secretion H+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_2mbut \"Secretion 2-Methylbutyric acid\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_2mbut \"Secretion 2-Methylbutyric acid\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_co2 \"Secretion CO2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_co2 \"Secretion CO2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_cdpg \"Secretion CDP-glycerol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_cdpg \"Secretion CDP-glycerol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_udpagaepyr \"Secretion UDP-N-acetylglucosamine-enolpyruvate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_udpagaepyr \"Secretion UDP-N-acetylglucosamine-enolpyruvate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_cobalt \"Secretion Co2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_cobalt \"Secretion Co2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_gtol \"Secretion galactitol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_gtol \"Secretion galactitol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_2dh3dg \"Secretion 2-dehydro-3-deoxy-D-gluconate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_2dh3dg \"Secretion 2-dehydro-3-deoxy-D-gluconate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_no2 \"Secretion nitrite\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_no2 \"Secretion nitrite\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_gly \"Secretion glycine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_gly \"Secretion glycine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_sucr \"Secretion sucrose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_sucr \"Secretion sucrose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_eto \"Secretion ethanol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_eto \"Secretion ethanol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_g3p \"Secretion sn-glycerol-3-phosphate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_g3p \"Secretion sn-glycerol-3-phosphate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_lysL \"Secretion L-lysine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_lysL \"Secretion L-lysine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_proL \"Secretion L-proline\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_proL \"Secretion L-proline\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_csn \"Secretion cytosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_csn \"Secretion cytosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ribD \"Secretion D-ribose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ribD \"Secretion D-ribose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_valL \"Secretion L-valine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_valL \"Secretion L-valine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_dachi \"Secretion diacetylchitobiose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_dachi \"Secretion diacetylchitobiose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_4abz \"Secretion 4-aminobenzoate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_4abz \"Secretion 4-aminobenzoate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ac \"Secretion acetate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ac \"Secretion acetate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ival \"Secretion Isovalerate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ival \"Secretion Isovalerate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ca2 \"Secretion Ca2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ca2 \"Secretion Ca2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_hxan \"Secretion hypoxanthine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_hxan \"Secretion hypoxanthine\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_metL \"Secretion L-methionine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_metL \"Secretion L-methionine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_trpL \"Secretion L-tryptophan\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_trpL \"Secretion L-tryptophan\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_actid \"Secretion Acetamide\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_actid \"Secretion Acetamide\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_arab \"Secretion L-arabinose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_arab \"Secretion L-arabinose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_glcn \"Secretion D-gluconate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_glcn \"Secretion D-gluconate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_alaL \"Secretion L-alanine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_alaL \"Secretion L-alanine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_leuL \"Secretion L-leucine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_leuL \"Secretion L-leucine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_4s45dhp23do \"Secretion (4S)-4,5-dihydroxypentan-2,3-dione\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_4s45dhp23do \"Secretion (4S)-4,5-dihydroxypentan-2,3-dione\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_uri \"Secretion uridine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_uri \"Secretion uridine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_2o6pamg2d \"Secretion 2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_2o6pamg2d \"Secretion 2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_adn \"Secretion adenosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_adn \"Secretion adenosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ileL \"Secretion L-isoleucine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ileL \"Secretion L-isoleucine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_dethio \"Secretion dethiobiotin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_dethio \"Secretion dethiobiotin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_nac \"Secretion nicotinate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_nac \"Secretion nicotinate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_glnL \"Secretion L-glutamine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_glnL \"Secretion L-glutamine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_escul \"Secretion esculin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_escul \"Secretion esculin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_metD \"Secretion D-methionine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_metD \"Secretion D-methionine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_spmd \"Secretion spermidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_spmd \"Secretion spermidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_xan \"Secretion xanthine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_xan \"Secretion xanthine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ptrc \"Secretion putrescine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ptrc \"Secretion putrescine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_thymd \"Secretion thymidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_thymd \"Secretion thymidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_nh3 \"Secretion ammonia\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_nh3 \"Secretion ammonia\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_fru \"Secretion D-fructose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_fru \"Secretion D-fructose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_thm \"Secretion thiamin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_thm \"Secretion thiamin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_c1ala \"Secretion cob(I)alamin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_c1ala \"Secretion cob(I)alamin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_for \"Secretion formate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_for \"Secretion formate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_fe2 \"Secretion Fe2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_fe2 \"Secretion Fe2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_so3 \"Secretion sulfite\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_so3 \"Secretion sulfite\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_4abg \"Secretion 4-aminobenzoate-glutamate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_4abg \"Secretion 4-aminobenzoate-glutamate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_fuc \"Secretion L-fuculose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_fuc \"Secretion L-fuculose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_chol \"Secretion choline\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_chol \"Secretion choline\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_malS \"Secretion (S)-malate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_malS \"Secretion (S)-malate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_chor \"Secretion chorismate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_chor \"Secretion chorismate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_hco3 \"Secretion bicarbonate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_hco3 \"Secretion bicarbonate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_malt \"Secretion maltose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_malt \"Secretion maltose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ppat \"Secretion phosphonate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ppat \"Secretion phosphonate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_h2 \"Secretion H2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_h2 \"Secretion H2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_bio \"Secretion biotin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_bio \"Secretion biotin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_xyld \"Secretion D-xylose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_xyld \"Secretion D-xylose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_pntoR \"Secretion (R)-pantothenate\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_pntoR \"Secretion (R)-pantothenate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_23dhbz \"Secretion of 2,3-dihydroxybenzoate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_23dhbz \"Secretion of 2,3-dihydroxybenzoate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_thrL \"Secretion L-threonine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_thrL \"Secretion L-threonine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_etoa \"Secretion ethanolamine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_etoa \"Secretion ethanolamine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_dad \"Secretion deoxyadenosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_dad \"Secretion deoxyadenosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_isobuta \"Secretion isobutyric acid\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_isobuta \"Secretion isobutyric acid\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_glyb \"Secretion glycine betaine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_glyb \"Secretion glycine betaine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_5apn \"Secretion 5-aminopentanoate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_5apn \"Secretion 5-aminopentanoate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_adman \"Secretion alpha-D-mannose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_adman \"Secretion alpha-D-mannose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_2aepat \"Secretion (2-aminoethyl)phosphonate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_2aepat \"Secretion (2-aminoethyl)phosphonate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_gluL \"Secretion L-glutamate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_gluL \"Secretion L-glutamate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_aspL \"Secretion L-aspartate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_aspL \"Secretion L-aspartate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_cysL \"Secretion L-cysteine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_cysL \"Secretion L-cysteine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_escut \"Secretion esculetin\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_escut \"Secretion esculetin\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_acnam \"Secretion N-acetylneuraminate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_acnam \"Secretion N-acetylneuraminate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ura \"Secretion uracil\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ura \"Secretion uracil\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_tyrL \"Secretion L-tyrosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_tyrL \"Secretion L-tyrosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_hoam \"Secretion Hydroxylamine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_hoam \"Secretion Hydroxylamine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_mevR \"Secretion (R)-mevalonate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_mevR \"Secretion (R)-mevalonate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_pi \"Secretion phosphate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_pi \"Secretion phosphate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_raf \"Secretion raffinose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_raf \"Secretion raffinose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_h2o \"Secretion H2O\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_h2o \"Secretion H2O\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_his \"Secretion L-histidine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_his \"Secretion L-histidine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_zn2 \"Secretion Zn2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_zn2 \"Secretion Zn2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_mg2 \"Secretion Mg2+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_mg2 \"Secretion Mg2+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_4mpo \"Secretion 4-methylphenol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_4mpo \"Secretion 4-methylphenol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_k \"Secretion K+\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_k \"Secretion K+\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_argL \"Secretion L-arginine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_argL \"Secretion L-arginine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_mnl \"Secretion mannitol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_mnl \"Secretion mannitol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_gal \"Secretion beta-D-galactose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_gal \"Secretion beta-D-galactose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_no3 \"Secretion nitrate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_no3 \"Secretion nitrate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_tgt \"Secretion tagatose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_tgt \"Secretion tagatose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_tre \"Secretion trehalose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_tre \"Secretion trehalose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ins \"Secretion inosine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ins \"Secretion inosine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_at \"Secretion acetone\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_at \"Secretion acetone\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_pheL \"Secretion L-phenylalanine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_pheL \"Secretion L-phenylalanine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_23diap \"Secretion 2,3-diaminopropanoate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_23diap \"Secretion 2,3-diaminopropanoate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ncam \"Secretion nicotinamide\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ncam \"Secretion nicotinamide\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_glc \"Secretion beta-D-glucose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_glc \"Secretion beta-D-glucose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_pyin \"Secretion pyridoxine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_pyin \"Secretion pyridoxine\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_celb \"Secretion cellobiose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_celb \"Secretion cellobiose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_ibtol \"Secretion isobutanol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_ibtol \"Secretion isobutanol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_asnL \"Secretion L-asparagine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_asnL \"Secretion L-asparagine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_isocap \"Secretion Isocaproate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_isocap \"Secretion Isocaproate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_sor \"Secretion D-sorbitol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_sor \"Secretion D-sorbitol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_serL \"Secretion L-serine\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_serL \"Secretion L-serine\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Sec_glcAD \"Secretion alpha-D-glucose\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Sec_glcAD \"Secretion alpha-D-glucose\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_744 \"ADP-glucose Glucose-1-phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_744 \"ADP-glucose Glucose-1-phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_422 \"beta-fructofuranosidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_422 \"beta-fructofuranosidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_488 \"cellobiose glucohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_488 \"cellobiose glucohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_258 \"ATP:alpha-D-glucose-1-phosphate adenylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_258 \"ATP:alpha-D-glucose-1-phosphate adenylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_93 \"alpha-glucosidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_93 \"alpha-glucosidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_646 \"diacetylchitobiose-6-phosphate hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_646 \"diacetylchitobiose-6-phosphate hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_raf \"Raffinose transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_raf \"Raffinose transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_450 \"6-phospho-beta-glucosidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_450 \"6-phospho-beta-glucosidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_509 \"Esculin hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_509 \"Esculin hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_308 \"maltose-6'-phosphate glucosidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_308 \"maltose-6'-phosphate glucosidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_45 \"glucose-6-phosphate:NADP+ 1-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_45 \"glucose-6-phosphate:NADP+ 1-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_408 \"Raffinose galactohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_408 \"Raffinose galactohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_373 \"beta-phosphoglucomutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_373 \"beta-phosphoglucomutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_573 \"alpha,alpha-phosphotrehalase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_573 \"alpha,alpha-phosphotrehalase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_315 \"maltose phosphorylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_315 \"maltose phosphorylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_314 \"D-proline reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_314 \"D-proline reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_321 \"Isobutyric acid phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_321 \"Isobutyric acid phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_146 \"2-methylbutanoyl phosphate transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_146 \"2-methylbutanoyl phosphate transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_114 \"betaine reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_114 \"betaine reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_4 \"phosphate isopentyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_4 \"phosphate isopentyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_71 \"D-Alanine:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_71 \"D-Alanine:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_352 \"phosphate isobutanoyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_352 \"phosphate isobutanoyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_178 \"(S)-3-methyl-2-oxopentanoate:NAD+ 2-oxidoreductase (CoA-methylbutanoylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_178 \"(S)-3-methyl-2-oxopentanoate:NAD+ 2-oxidoreductase (CoA-methylbutanoylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_366 \" isopentanoylphosphate phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_366 \" isopentanoylphosphate phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_709 \"isoleucine dehydrogenase \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_709 \"isoleucine dehydrogenase \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_518 \"2-methylbutanoyl CoA transferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_518 \"2-methylbutanoyl CoA transferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_323 \"4-methyl-2-oxopentanoate:NAD+ oxidoreductase (CoA-mehtylpropanoylating)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_323 \"4-methyl-2-oxopentanoate:NAD+ oxidoreductase (CoA-mehtylpropanoylating)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_247 \"3-methyl-2-oxobutanoate dehydrogenase (ferredoxin)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_247 \"3-methyl-2-oxobutanoate dehydrogenase (ferredoxin)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_719 \"hydrogen-sulfide:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_719 \"hydrogen-sulfide:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_158 \"Hydrogen-sulfide:ferredoxin oxidoreductase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_158 \"Hydrogen-sulfide:ferredoxin oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_510 \"3-mercaptopyruvate sulfurtransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_510 \"3-mercaptopyruvate sulfurtransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_584 \"sulfite reduction\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_584 \"sulfite reduction\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_355 \"thiosulfate sulfurtransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_355 \"thiosulfate sulfurtransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_141 \"hydrogen-sulfide:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_141 \"hydrogen-sulfide:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_266 \"Adenosine 3',5'-bisphosphate 3'-phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_266 \"Adenosine 3',5'-bisphosphate 3'-phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_485 \"trans-hexaprenyltranstransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_485 \"trans-hexaprenyltranstransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_630 \"Lipoteichoic acid biosynthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_630 \"Lipoteichoic acid biosynthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_607 \"N-acetylglucosaminyldiphosphoundecaprenol N-acetyl-beta-D-mannosaminyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_607 \"N-acetylglucosaminyldiphosphoundecaprenol N-acetyl-beta-D-mannosaminyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_324 \"UDP-N-acetyl-D-glucosamine:undecaprenyl-phosphate N-acetyl-D-glucosamine phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_324 \"UDP-N-acetyl-D-glucosamine:undecaprenyl-phosphate N-acetyl-D-glucosamine phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_200 \"lysyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_200 \"lysyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_59 \"CDPglycerol:poly(glycerophosphate) glycerophosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_59 \"CDPglycerol:poly(glycerophosphate) glycerophosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_617 \"geranyltranstransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_617 \"geranyltranstransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_292 \"4-hydroxy-3-methylbut-2-enyl diphosphate reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_292 \"4-hydroxy-3-methylbut-2-enyl diphosphate reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_640 \"dimethylallyl diphosphate:NADP+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_640 \"dimethylallyl diphosphate:NADP+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_187 \"(R)-mevalonate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_187 \"(R)-mevalonate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_749 \"4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_749 \"4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_645 \"2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_645 \"2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_705 \"2-C-methyl-D-erythritol-2,4-cyclodiphosphate:reduced ferredoxin oxidase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_705 \"2-C-methyl-D-erythritol-2,4-cyclodiphosphate:reduced ferredoxin oxidase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_637 \"1-deoxy-D-xylulose-5-phosphate reductoisomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_637 \"1-deoxy-D-xylulose-5-phosphate reductoisomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_150 \"(2E,6E)-farnesyl-diphosphate:isopentenyl-diphosphate cistransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_150 \"(2E,6E)-farnesyl-diphosphate:isopentenyl-diphosphate cistransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_70 \"farnesyltranstransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_70 \"farnesyltranstransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_431 \"dimethylallyltranstransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_431 \"dimethylallyltranstransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_639 \"undecaprenyl-diphosphate phosphohydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_639 \"undecaprenyl-diphosphate phosphohydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_403 \"geranylgeranyl-diphosphate:isopentenyl-diphosphate transtransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_403 \"geranylgeranyl-diphosphate:isopentenyl-diphosphate transtransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_24 \"hydroxymethylglutaryl-CoA lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_24 \"hydroxymethylglutaryl-CoA lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_587 \"isopentenyl-diphosphate delta-isomerase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_587 \"isopentenyl-diphosphate delta-isomerase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_46 \"isopentenyl-diphosphate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_46 \"isopentenyl-diphosphate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_17 \"2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_17 \"2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_351 \"dimethylallyl diphosphate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_351 \"dimethylallyl diphosphate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_689 \"trans-pentaprenyltranstransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_689 \"trans-pentaprenyltranstransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_720 \"L-tyrosine 4-methylphenol-lyase (2-iminoacetate-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_720 \"L-tyrosine 4-methylphenol-lyase (2-iminoacetate-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_468 \"xylulose-5-phosphate : thiamin diphosphate glycolaldehydetransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_468 \"xylulose-5-phosphate : thiamin diphosphate glycolaldehydetransferase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_225 \"ATP:4-methyl-5-(beta-hydroxyethyl)thiazole 2-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_225 \"ATP:4-methyl-5-(beta-hydroxyethyl)thiazole 2-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_214 \"thiamine-phosphate diphosphorylase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_214 \"thiamine-phosphate diphosphorylase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_636 \"phosphomethylpyrimidine synthase (2'deoxyadenosine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_636 \"phosphomethylpyrimidine synthase (2'deoxyadenosine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_668 \"Phosphomethylpyrimidine synthase (5'deoxyadenosine)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_668 \"Phosphomethylpyrimidine synthase (5'deoxyadenosine)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_238 \"1-methylthio-xylulose 5-phosphate sulfo-lyase \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_238 \"1-methylthio-xylulose 5-phosphate sulfo-lyase \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_264 \"phosphomethylpyrimidine kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_264 \"phosphomethylpyrimidine kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_228 \"sedoheptulose-7-phosphate : thiamin diphosphate glycolaldehydetransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_228 \"sedoheptulose-7-phosphate : thiamin diphosphate glycolaldehydetransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_15 \"2,3-diaminopropanoate ammonia-lyase (adding water; pyruvate-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_15 \"2,3-diaminopropanoate ammonia-lyase (adding water; pyruvate-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_478 \"1-deoxy-D-xylulose-5-phosphate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_478 \"1-deoxy-D-xylulose-5-phosphate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_581 \"ATP:thiamine phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_581 \"ATP:thiamine phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_210 \"4-methyl-5-(beta-hydroxyethyl)thiazole Biosynthesis\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_210 \"4-methyl-5-(beta-hydroxyethyl)thiazole Biosynthesis\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_388 \"thiamine diphosphokinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_388 \"thiamine diphosphokinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_389 \"thiamine-phosphate kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_389 \"thiamine-phosphate kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_mevR \"(R)-mevalonate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_mevR \"(R)-mevalonate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_xyld \"D-xylose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_xyld \"D-xylose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_isobuta \"Isobutyric acid transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_isobuta \"Isobutyric acid transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_fe \"Ferrous iron transport \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_fe \"Ferrous iron transport \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ival \"Isovalerate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ival \"Isovalerate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_4s45dhp23do \"(4S)-4,5-dihydroxypentan-2,3-dione transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_4s45dhp23do \"(4S)-4,5-dihydroxypentan-2,3-dione transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_h \"H+ Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_h \"H+ Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_23dhbz \"2,3-dihydroxybenzoate Transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_23dhbz \"2,3-dihydroxybenzoate Transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_hoam \"Hydroxylamine transporter \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_hoam \"Hydroxylamine transporter \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_escul \"Esculin transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_escul \"Esculin transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_23diap \"2,3-diaminopropanoate transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_23diap \"2,3-diaminopropanoate transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_pyin \"Pyridoxine Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_pyin \"Pyridoxine Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_escut \"Esculetin transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_escut \"Esculetin transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_2aepat \"(2-aminoethyl)phosphonate transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_2aepat \"(2-aminoethyl)phosphonate transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_bio \"Biotin transport \">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_bio \"Biotin transport \">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_pntoR \"(R)-pantothenate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_pntoR \"(R)-pantothenate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_cdpg \"CDP-glycerol:poly(glycerophosphate) glycerophosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_cdpg \"CDP-glycerol:poly(glycerophosphate) glycerophosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ibtol \"isobutanol transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ibtol \"isobutanol transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_cobalt2 \"cobalt transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_cobalt2 \"cobalt transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glyo \"glycerol transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glyo \"glycerol transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_2mbut \"2-Methylbutyric acid transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_2mbut \"2-Methylbutyric acid transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_actid \"Acetamide transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_actid \"Acetamide transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_nac \"nicotinate transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_nac \"nicotinate transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_etoa \"ethanolamine transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_etoa \"ethanolamine transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_5apn \"5-aminopentanoate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_5apn \"5-aminopentanoate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_dethio \"dethiobiotin transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_dethio \"dethiobiotin transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_thm \"thiamin transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_thm \"thiamin transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_4abz \"p-Aminobenzoic acid transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_4abz \"p-Aminobenzoic acid transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_4mpo \"4-methylphenol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_4mpo \"4-methylphenol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_pi3 \"phosphate passive transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_pi3 \"phosphate passive transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_isocap \"Isocaproate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_isocap \"Isocaproate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_nh \"ammonia transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_nh \"ammonia transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ncam \"nicotinamide transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ncam \"nicotinamide transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_tmam \" trimethylamine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_tmam \" trimethylamine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_284 \"naphthoate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_284 \"naphthoate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_119 \"NADH:menaquinone oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_119 \"NADH:menaquinone oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_472 \"2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_472 \"2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_722 \"tyrosine:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_722 \"tyrosine:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_112 \"all-trans-polyprenyl diphosphate:1,4-dihydroxy-2-naphthoate polyprenyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_112 \"all-trans-polyprenyl diphosphate:1,4-dihydroxy-2-naphthoate polyprenyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_746 \"S-adenosyl-L-methione:demethylmenaquinone methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_746 \"S-adenosyl-L-methione:demethylmenaquinone methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_23 \"all-trans-octaprenyl-diphosphate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_23 \"all-trans-octaprenyl-diphosphate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_33 \"hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_33 \"hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_209 \"isochorismate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_209 \"isochorismate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_555 \"o-succinylbenzoate-CoA ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_555 \"o-succinylbenzoate-CoA ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_51 \"2-isopropylmalate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_51 \"2-isopropylmalate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_212 \"2-methylcrotonoyl-CoA hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_212 \"2-methylcrotonoyl-CoA hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_11 \"3-isopropylmalate dehydrogenase(step1)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_11 \"3-isopropylmalate dehydrogenase(step1)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_354 \"2-methyl-3-hydroxybutyryl-CoA:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_354 \"2-methyl-3-hydroxybutyryl-CoA:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_757 \"(3S)-3-Methyl-2-oxopentanoate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_757 \"(3S)-3-Methyl-2-oxopentanoate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_328 \"2-Isopropylmalate hydro-lyase(step 1)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_328 \"2-Isopropylmalate hydro-lyase(step 1)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_217 \"leucine transaminase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_217 \"leucine transaminase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_454 \"2-Isopropylmalate hydro-lyase(step 2)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_454 \"2-Isopropylmalate hydro-lyase(step 2)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_735 \"2,3-dihydroxy-3-methylvalerate dehydratase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_735 \"2,3-dihydroxy-3-methylvalerate dehydratase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_369 \"L-valine:pyruvate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_369 \"L-valine:pyruvate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_691 \"2-methylbutanoyl-CoA:FAD 2,3-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_691 \"2-methylbutanoyl-CoA:FAD 2,3-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_731 \"pyruvate:2-oxobutanoate acetaldehydetransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_731 \"pyruvate:2-oxobutanoate acetaldehydetransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_453 \"2-aceto-2-hydroxybutanoate oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_453 \"2-aceto-2-hydroxybutanoate oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_154 \"propanoyl-CoA : acetyl-CoA acetyltransferase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_154 \"propanoyl-CoA : acetyl-CoA acetyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_359 \"3-isopropylmalate dehydrogenase(step2)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_359 \"3-isopropylmalate dehydrogenase(step2)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_520 \"isobutanal hydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_520 \"isobutanal hydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_430 \"2-methyl-3-oxopropanoate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_430 \"2-methyl-3-oxopropanoate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_539 \"propanoyl-CoA:formate C-propanoyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_539 \"propanoyl-CoA:formate C-propanoyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_203 \"S-adenosyl-L-methionine:uroporphyrin-III C-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_203 \"S-adenosyl-L-methionine:uroporphyrin-III C-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_497 \"ATP:cob(I0alamin Co-beta-adenosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_497 \"ATP:cob(I0alamin Co-beta-adenosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_256 \"S-adenosyl-L-methionine:precorrin-1 C-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_256 \"S-adenosyl-L-methionine:precorrin-1 C-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_207 \"adenosylcobinamide-phosphate guanylyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_207 \"adenosylcobinamide-phosphate guanylyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_236 \"nicotinate mononucleotide:5,6-dimethylbenzimidazole phospho-D-ribosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_236 \"nicotinate mononucleotide:5,6-dimethylbenzimidazole phospho-D-ribosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_465 \"coproporphyrinogen dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_465 \"coproporphyrinogen dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_131 \"biotin synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_131 \"biotin synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_456 \"hydroxymethylbilane hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_456 \"hydroxymethylbilane hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_529 \"threonine-phosphate carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_529 \"threonine-phosphate carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_538 \"adenosylcobinamide kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_538 \"adenosylcobinamide kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_717 \"S-adenosyl-L-methionine:uroporphyrin-III C-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_717 \"S-adenosyl-L-methionine:uroporphyrin-III C-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_47 \"adenosylcobinamide kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_47 \"adenosylcobinamide kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_37 \"S-adenosyl-L-methionine:cobalt-factor-II C20-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_37 \"S-adenosyl-L-methionine:cobalt-factor-II C20-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_260 \"Uroporphyrinogen-III carboxy-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_260 \"Uroporphyrinogen-III carboxy-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_553 \"ATP:L-threonine O-phosphotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_553 \"ATP:L-threonine O-phosphotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_103 \"ATP:Biotin adenyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_103 \"ATP:Biotin adenyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_671 \"S-adenosyl-L-methionine:cobalt-precorrin-5B (C1)-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_671 \"S-adenosyl-L-methionine:cobalt-precorrin-5B (C1)-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_91 \"2,3-dihydro-2,3-dihydroxybenzoate:NAD+ oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_91 \"2,3-dihydro-2,3-dihydroxybenzoate:NAD+ oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_540 \"5-aminolevulinate hydro-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_540 \"5-aminolevulinate hydro-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_281 \"cobalt-precorrin 3 dehydrogenase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_281 \"cobalt-precorrin 3 dehydrogenase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_25 \"cobalamin synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_25 \"cobalamin synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_704 \"Isochorismate pyruvate-hydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_704 \"Isochorismate pyruvate-hydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_426 \"cobalt-precorrin-7 (C15)-methyltransferase [decarboxylating]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_426 \"cobalt-precorrin-7 (C15)-methyltransferase [decarboxylating]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_245 \"ATP:cob(I)yrinate a,c-diamide cobeta-adenosyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_245 \"ATP:cob(I)yrinate a,c-diamide cobeta-adenosyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_440 \"cob(II)yrinate a,c-diamide:FMN oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_440 \"cob(II)yrinate a,c-diamide:FMN oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_84 \"protoporphyrinogen IX dehydrogenase (menaquinone)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_84 \"protoporphyrinogen IX dehydrogenase (menaquinone)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_552 \"adenosyl-cobyrate:(R)-1-amino-2-propanol O-2-phosphate ligase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_552 \"adenosyl-cobyrate:(R)-1-amino-2-propanol O-2-phosphate ligase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_572 \"S-adenosyl-L-methionine:precorrin-7 C15-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_572 \"S-adenosyl-L-methionine:precorrin-7 C15-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_138 \"porphobilinogen ammonia-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_138 \"porphobilinogen ammonia-lyase\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_618 \"adenosyl-cobyrate synthase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_618 \"adenosyl-cobyrate synthase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_364 \"Precorrin-8X methylmutase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_364 \"Precorrin-8X methylmutase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_487 \"cobalt-precorrin-4 methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_487 \"cobalt-precorrin-4 methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_129 \"ferrochelatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_129 \"ferrochelatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_738 \"precorrin-6A reductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_738 \"precorrin-6A reductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_700 \"adenosylcobalamin phosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_700 \"adenosylcobalamin phosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_67 \"S-adenosyl-L-methionine:uroporphyrinogen-III C-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_67 \"S-adenosyl-L-methionine:uroporphyrinogen-III C-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_190 \"S-adenosyl-L-methionine:precorrin-3B C17-methyltransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_190 \"S-adenosyl-L-methionine:precorrin-3B C17-methyltransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_670 \"cobyrinate:L-glutamine amido-ligase (ADP-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_670 \"cobyrinate:L-glutamine amido-ligase (ADP-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_760 \"adenosylcobinamide-GDP ribazoletransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_760 \"adenosylcobinamide-GDP ribazoletransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_484 \"sirohydrochlorin cobalt-lyase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_484 \"sirohydrochlorin cobalt-lyase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_101 \"cobalt-precorrin 5A acylhydrolase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_101 \"cobalt-precorrin 5A acylhydrolase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_761 \"adenosylcobalamin/alpha-ribazole phosphatase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_761 \"adenosylcobalamin/alpha-ribazole phosphatase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_508 \"pyridoxamine:pyruvate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_508 \"pyridoxamine:pyruvate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_554 \"pyridoxal kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_554 \"pyridoxal kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_220 \"Pyridoxamine-5'-phosphate:2-oxoglutarate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_220 \"Pyridoxamine-5'-phosphate:2-oxoglutarate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_647 \"Pyridoxine:NADP+ 4-oxidoreductase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_647 \"Pyridoxine:NADP+ 4-oxidoreductase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_461 \"pyridoxamine:oxaloacetate aminotransferase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_461 \"pyridoxamine:oxaloacetate aminotransferase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_698 \"pyridoxal kinase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_698 \"pyridoxal kinase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_ID_270 \"Pyridoxamine-5'-phosphate:2-oxoglutarate aminotransferase (D-glutamate-forming)\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_ID_270 \"Pyridoxamine-5'-phosphate:2-oxoglutarate aminotransferase (D-glutamate-forming)\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_co \"carbon dioxide diffusion\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_co \"carbon dioxide diffusion\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_h2o \"water diffusion\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_h2o \"water diffusion\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_eto \"ethanol transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_eto \"ethanol transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_hco \"hco3 transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_hco \"hco3 transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_at \"acetone export\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_at \"acetone export\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_h4 \"hydrogen diffusion\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_h4 \"hydrogen diffusion\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_adman2 \"alpha-D-mannose Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_adman2 \"alpha-D-mannose Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glcAD \"alpha-D-glucose Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glcAD \"alpha-D-glucose Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_uri \"uridine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_uri \"uridine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ura \"uracil transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ura \"uracil transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_h3 \"sodium ion:proton antiporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_h3 \"sodium ion:proton antiporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ca2 \"Calcium:Sodium antiport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ca2 \"Calcium:Sodium antiport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_xyld2 \"D-xylose Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_xyld2 \"D-xylose Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ac \"Acetate Transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ac \"Acetate Transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glcn \"gluconate permease\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glcn \"gluconate permease\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_pi \"sodium-dependent phosphate transporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_pi \"sodium-dependent phosphate transporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_dad \"deoxyadenosine transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_dad \"deoxyadenosine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_dcyt \"deoxycytidine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_dcyt \"deoxycytidine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_celb2 \"cellobiose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_celb2 \"cellobiose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_thymd \"thymidine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_thymd \"thymidine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_malS \"(S)-malate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_malS \"(S)-malate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ribflv \"riboflavin transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ribflv \"riboflavin transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_hxan \"hypoxanthine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_hxan \"hypoxanthine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_2dh3dg \"2-keto-3-deoxygluconate permease\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_2dh3dg \"2-keto-3-deoxygluconate permease\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_h2 \"sodium:H+ symporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_h2 \"sodium:H+ symporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_sor \"D-sorbitol import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_sor \"D-sorbitol import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_malt2 \"maltose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_malt2 \"maltose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_k2 \"potassium ion uptake\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_k2 \"potassium ion uptake\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_csn \"cytosine:H+ cotransport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_csn \"cytosine:H+ cotransport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_sucr2 \"sucrose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_sucr2 \"sucrose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_ins \"inosine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_ins \"inosine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_4abg \"aminobenzoyl-glutamate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_4abg \"aminobenzoyl-glutamate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_arab \"L-arabinose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_arab \"L-arabinose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_acnam \"Transport of N-acetylneuraminate\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_acnam \"Transport of N-acetylneuraminate\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_xan \"xanthine permease\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_xan \"xanthine permease\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_gal \"beta-D-galactose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_gal \"beta-D-galactose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_so \"sodium:sulfite symporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_so \"sodium:sulfite symporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_xan2 \"xanthine:uracil cotransport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_xan2 \"xanthine:uracil cotransport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_alaL \"sodium ion:alanine symporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_alaL \"sodium ion:alanine symporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_adn \"adenosine transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_adn \"adenosine transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_glc2 \"beta-D-glucose import\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_glc2 \"beta-D-glucose import\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_gluL2 \"glutamate:sodium cotransport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_gluL2 \"glutamate:sodium cotransport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_chor \"chorismate transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_chor \"chorismate transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Trans_aspL \"proton/sodium ion:glutamate/aspartate symporter\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Trans_aspL \"proton/sodium ion:glutamate/aspartate symporter\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_Ex_biomass \"Exchange Biomass\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_Ex_biomass \"Exchange Biomass\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp1 \"CD0043, CD0042, CD0041\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp1 \"CD0043, CD0042, CD0041\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp2 \"CD0118, CD2199, CD2198\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp2 \"CD0118, CD2199, CD2198\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp3 \"CD0135, CD0136, CD0137\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp3 \"CD0135, CD0136, CD0137\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp4 \"CD0206, CD0207, CD0208\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp4 \"CD0206, CD0207, CD0208\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp5 \"CD0284, CD0285\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp5 \"CD0284, CD0285\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp6 \"CD0286, CD0287, CD0288, CD0289\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp6 \"CD0286, CD0287, CD0288, CD0289\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp7 \"CD0431, CD0430\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp7 \"CD0431, CD0430\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp8 \"CD0491, CD0492, CD0493, CD0494\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp8 \"CD0491, CD0492, CD0493, CD0494\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp9 \"CD0723, acoA, acoB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp9 \"CD0723, acoA, acoB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp10 \"CD0723, acoL\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp10 \"CD0723, acoL\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp11 \"CD0728, CD0725, CD0727, CD0717, CD0724\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp11 \"CD0728, CD0725, CD0727, CD0717, CD0724\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp12 \"CD0728, CD0726, CD0727, CD0717, CD0724\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp12 \"CD0728, CD0726, CD0727, CD0717, CD0724\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp13 \"CD0863, CD0862, CD0861\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp13 \"CD0863, CD0862, CD0861\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp14 \"CD1000, CD1001, CD0999\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp14 \"CD1000, CD1001, CD0999\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp15 \"CD1074, CD1076, CD1077, CD1078\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp15 \"CD1074, CD1076, CD1077, CD1078\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp16 \"CD1083, CD2280, CD2282\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp16 \"CD1083, CD2280, CD2282\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp17 \"CD1386, CD1387\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp17 \"CD1386, CD1387\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp18 \"CD1587, CD1588, CD1589\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp18 \"CD1587, CD1588, CD1589\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp19 \"CD1904, CD2989, CD2990\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp19 \"CD1904, CD2989, CD2990\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp20 \"CD1979, CD1978, CD1980\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp20 \"CD1979, CD1978, CD1980\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp21 \"CD2077, pyrK\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp21 \"CD2077, pyrK\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp22 \"CD2087, pucC1, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp22 \"CD2087, pucC1, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp23 \"CD2087, pucC1, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp23 \"CD2087, pucC1, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp24 \"CD2087, pucC1, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp24 \"CD2087, pucC1, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp25 \"CD2087, pucC2, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp25 \"CD2087, pucC2, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp26 \"CD2087, pucC2, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp26 \"CD2087, pucC2, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp27 \"CD2087, pucC2, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp27 \"CD2087, pucC2, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp28 \"CD2087, xdhB, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp28 \"CD2087, xdhB, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp29 \"CD2087, xdhB, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp29 \"CD2087, xdhB, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp30 \"CD2087, xdhB, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp30 \"CD2087, xdhB, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp31 \"CD2327, CD2326, CD2325\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp31 \"CD2327, CD2326, CD2325\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp32 \"CD2332, CD2334\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp32 \"CD2332, CD2334\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp33 \"CD2512, CD2510\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp33 \"CD2512, CD2510\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp34 \"CD2566, CD2567, CD2568\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp34 \"CD2566, CD2567, CD2568\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp35 \"CD2875, CD2876, CD2878\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp35 \"CD2875, CD2876, CD2878\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp36 \"CD2975, CD2877, CD2878\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp36 \"CD2975, CD2877, CD2878\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp37 \"CD3015, CD3014, CD3013\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp37 \"CD3015, CD3014, CD3013\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp38 \"CD3027, CD3030\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp38 \"CD3027, CD3030\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp39 \"CD3030, CD3027\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp39 \"CD3030, CD3027\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp40 \"CD3058, CD3061\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp40 \"CD3058, CD3061\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp41 \"CD3080, CD3049, CD3048\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp41 \"CD3080, CD3049, CD3048\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp42 \"CD3080, CD3081, CD3082\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp42 \"CD3080, CD3081, CD3082\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp43 \"CD3101, CD3103\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp43 \"CD3101, CD3103\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp44 \"CD3179, pyrK\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp44 \"CD3179, pyrK\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp45 \"CD3278, CD3279, CD3277, CD3276\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp45 \"CD3278, CD3279, CD3277, CD3276\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp46 \"CD3417, CD3416, CD3415, CD3414\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp46 \"CD3417, CD3416, CD3415, CD3414\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp47 \"CD3445, CD3444, CD3443\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp47 \"CD3445, CD3444, CD3443\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp48 \"CD3629, CD3630\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp48 \"CD3629, CD3630\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp49 \"accA, accD, accB, accC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp49 \"accA, accD, accB, accC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp50 \"asrA, asrB, asrC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp50 \"asrA, asrB, asrC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp51 \"atpA, atpB, atpF, atpH, atpG, atpD, atpC, atpI, atpZ\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp51 \"atpA, atpB, atpF, atpH, atpG, atpD, atpC, atpI, atpZ\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp52 \"atpA, atpB, atpF, atpH, atpG, atpD, atpE, atpI, atpZ\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp52 \"atpA, atpB, atpF, atpH, atpG, atpD, atpE, atpI, atpZ\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp53 \"carB1, carB2, carA\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp53 \"carB1, carB2, carA\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp54 \"carB1, carB2, pyrAA2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp54 \"carB1, carB2, pyrAA2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp55 \"carB, carA\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp55 \"carB, carA\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp56 \"carB, pyrAA2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp56 \"carB, pyrAA2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp57 \"cbiO2, cbiO1, cbiQ, cbiN, cbiM\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp57 \"cbiO2, cbiO1, cbiQ, cbiN, cbiM\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp58 \"cbiO, CD0326, cbiN, cbiM\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp58 \"cbiO, CD0326, cbiN, cbiM\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp59 \"celC, celB, celA\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp59 \"celC, celB, celA\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp60 \"crt1, CD0803, hbd, thlA2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp60 \"crt1, CD0803, hbd, thlA2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp61 \"eutA, eutB, eutC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp61 \"eutA, eutB, eutC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp62 \"eutB, eutC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp62 \"eutB, eutC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp63 \"fabH, fabF, CD2577, fabZ\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp63 \"fabH, fabF, CD2577, fabZ\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp64 \"fabH, fabF, fabG, fabZ\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp64 \"fabH, fabF, fabG, fabZ\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp65 \"fumA, fumB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp65 \"fumA, fumB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp66 \"gcvTPA, gcvH\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp66 \"gcvTPA, gcvH\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp67 \"gcvTPA, gcvPB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp67 \"gcvTPA, gcvPB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp68 \"grdA, grdE, grdD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp68 \"grdA, grdE, grdD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp69 \"grdB, grdC, grdX\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp69 \"grdB, grdC, grdX\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp70 \"grdC, grdB, grdX\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp70 \"grdC, grdB, grdX\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp71 \"grdD, grdA, grdX\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp71 \"grdD, grdA, grdX\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp72 \"gutA, srlEa, srlEb, srlB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp72 \"gutA, srlEa, srlEb, srlB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp73 \"hadI, hadB, hadC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp73 \"hadI, hadB, hadC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp74 \"hisH, hisF\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp74 \"hisH, hisF\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp75 \"hisZ, hisG\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp75 \"hisZ, hisG\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp76 \"hpdC, hpdB, hpdA\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp76 \"hpdC, hpdB, hpdA\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp77 \"hymA, hymB, CD3407\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp77 \"hymA, hymB, CD3407\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp78 \"kdpA, kdpB, kdpC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp78 \"kdpA, kdpB, kdpC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp79 \"leuC, leuD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp79 \"leuC, leuD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp80 \"metN, metI, metQ, CD1653\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp80 \"metN, metI, metQ, CD1653\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp81 \"nrdE, nrdF\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp81 \"nrdE, nrdF\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp82 \"ntpD, ntpB, ntpA, ntpC, ntpE, ntpF, ntpK\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp82 \"ntpD, ntpB, ntpA, ntpC, ntpE, ntpF, ntpK\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp83 \"opuCA, opuCC, CD3215\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp83 \"opuCA, opuCC, CD3215\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp84 \"oraS, oraE\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp84 \"oraS, oraE\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp85 \"ortA, ortB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp85 \"ortA, ortB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp86 \"pabA, pabB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp86 \"pabA, pabB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp87 \"potA, potC, potB, potD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp87 \"potA, potC, potB, potD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp88 \"prdA, prdB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp88 \"prdA, prdB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp89 \"prdF, prdE, prdD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp89 \"prdF, prdE, prdD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp90 \"ptsG-A, ptsG-BC\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp90 \"ptsG-A, ptsG-BC\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp91 \"pyrD, pyrK\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp91 \"pyrD, pyrK\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp92 \"pyrK, CD2077\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp92 \"pyrK, CD2077\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp93 \"pyrK, CD3179\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp93 \"pyrK, CD3179\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp94 \"rbsA, rbsC, rbsB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp94 \"rbsA, rbsC, rbsB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp95 \"ribH, ribE\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp95 \"ribH, ribE\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp96 \"rpiB1, rpiB2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp96 \"rpiB1, rpiB2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp97 \"scoA, scoB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp97 \"scoA, scoB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp98 \"srlA, srlEa, srlEb, srlB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp98 \"srlA, srlEa, srlEb, srlB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp99 \"thiF, thiS, thiG, CD2156, thiI\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp99 \"thiF, thiS, thiG, CD2156, thiI\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp100 \"thiF, thiS, thiG, thiH, thiI\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp100 \"thiF, thiS, thiG, thiH, thiI\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp101 \"xdhA1, pucC1, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp101 \"xdhA1, pucC1, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp102 \"xdhA1, pucC1, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp102 \"xdhA1, pucC1, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp103 \"xdhA1, pucC1, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp103 \"xdhA1, pucC1, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp104 \"xdhA1, pucC2, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp104 \"xdhA1, pucC2, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp105 \"xdhA1, pucC2, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp105 \"xdhA1, pucC2, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp106 \"xdhA1, pucC2, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp106 \"xdhA1, pucC2, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp107 \"xdhA1, xdhB, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp107 \"xdhA1, xdhB, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp108 \"xdhA1, xdhB, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp108 \"xdhA1, xdhB, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp109 \"xdhA1, xdhB, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp109 \"xdhA1, xdhB, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp110 \"xdhA2, pucC1, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp110 \"xdhA2, pucC1, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp111 \"xdhA2, pucC1, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp111 \"xdhA2, pucC1, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp112 \"xdhA2, pucC1, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp112 \"xdhA2, pucC1, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp113 \"xdhA2, pucC2, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp113 \"xdhA2, pucC2, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp114 \"xdhA2, pucC2, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp114 \"xdhA2, pucC2, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp115 \"xdhA2, pucC2, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp115 \"xdhA2, pucC2, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp116 \"xdhA2, xdhB, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp116 \"xdhA2, xdhB, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp117 \"xdhA2, xdhB, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp117 \"xdhA2, xdhB, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp118 \"xdhA2, xdhB, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp118 \"xdhA2, xdhB, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp119 \"xdhA3, pucC1, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp119 \"xdhA3, pucC1, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp120 \"xdhA3, pucC1, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp120 \"xdhA3, pucC1, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp121 \"xdhA3, pucC1, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp121 \"xdhA3, pucC1, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp122 \"xdhA3, pucC2, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp122 \"xdhA3, pucC2, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp123 \"xdhA3, pucC2, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp123 \"xdhA3, pucC2, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp124 \"xdhA3, pucC2, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp124 \"xdhA3, pucC2, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp125 \"xdhA3, xdhB, xdhC1, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp125 \"xdhA3, xdhB, xdhC1, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp126 \"xdhA3, xdhB, xdhC2, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp126 \"xdhA3, xdhB, xdhC2, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp127 \"xdhA3, xdhB, xdhC3, xdhD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp127 \"xdhA3, xdhB, xdhC3, xdhD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction R_comp128 \"xynA, xynB, xynC, xynD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction R_comp128 \"xynA, xynB, xynC, xynD\">'\n",
      "No objective coefficients in model. Unclear what should be optimized\n",
      "Missing flux bounds on reactions set to default bounds.As best practise and to avoid confusion flux bounds should be set explicitly on all reactions.\n",
      "'M_1dgtl4p[c]' is not a valid SBML 'SId'.\n",
      "'M_tre[e]' is not a valid SBML 'SId'.\n",
      "'M_no3[c]' is not a valid SBML 'SId'.\n",
      "'M_rib5p[c]' is not a valid SBML 'SId'.\n",
      "'M_ptpphygen[c]' is not a valid SBML 'SId'.\n",
      "'M_ctnycoa[c]' is not a valid SBML 'SId'.\n",
      "'M_5mtr1p[c]' is not a valid SBML 'SId'.\n",
      "'M_c1ala[e]' is not a valid SBML 'SId'.\n",
      "'M_2hisocap[c]' is not a valid SBML 'SId'.\n",
      "'M_cbpc3[c]' is not a valid SBML 'SId'.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'M_6hm78dhpte[c]' is not a valid SBML 'SId'.\n",
      "'M_adc5p[c]' is not a valid SBML 'SId'.\n",
      "'M_sucr[c]' is not a valid SBML 'SId'.\n",
      "'M_s2a6kp[c]' is not a valid SBML 'SId'.\n",
      "'M_pantR[c]' is not a valid SBML 'SId'.\n",
      "'M_acap[c]' is not a valid SBML 'SId'.\n",
      "'M_cysL[c]' is not a valid SBML 'SId'.\n",
      "'M_pheL[e]' is not a valid SBML 'SId'.\n",
      "'M_pppi[c]' is not a valid SBML 'SId'.\n",
      "'M_nh3[c]' is not a valid SBML 'SId'.\n",
      "'M_hcy[c]' is not a valid SBML 'SId'.\n",
      "'M_3uprop[c]' is not a valid SBML 'SId'.\n",
      "'M_78dhpte[c]' is not a valid SBML 'SId'.\n",
      "'M_butp[c]' is not a valid SBML 'SId'.\n",
      "'M_4pptnD[c]' is not a valid SBML 'SId'.\n",
      "'M_sh17bp[c]' is not a valid SBML 'SId'.\n",
      "'M_tcy[c]' is not a valid SBML 'SId'.\n",
      "'M_datp[c]' is not a valid SBML 'SId'.\n",
      "'M_2a3ob[c]' is not a valid SBML 'SId'.\n",
      "'M_itp[c]' is not a valid SBML 'SId'.\n",
      "'M_drib[c]' is not a valid SBML 'SId'.\n",
      "'M_naspdi[c]' is not a valid SBML 'SId'.\n",
      "'M_argsuc[c]' is not a valid SBML 'SId'.\n",
      "'M_dpcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_ga6p[c]' is not a valid SBML 'SId'.\n",
      "'M_ptphyrin[c]' is not a valid SBML 'SId'.\n",
      "'M_2hisocapcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_fru1p[c]' is not a valid SBML 'SId'.\n",
      "'M_spmd[c]' is not a valid SBML 'SId'.\n",
      "'M_dethio[c]' is not a valid SBML 'SId'.\n",
      "'M_fmn[c]' is not a valid SBML 'SId'.\n",
      "'M_glyclt[c]' is not a valid SBML 'SId'.\n",
      "'M_fes[c]' is not a valid SBML 'SId'.\n",
      "'M_histd[c]' is not a valid SBML 'SId'.\n",
      "'M_oxac[c]' is not a valid SBML 'SId'.\n",
      "'M_thm[e]' is not a valid SBML 'SId'.\n",
      "'M_ac[c]' is not a valid SBML 'SId'.\n",
      "'M_CLP[c]' is not a valid SBML 'SId'.\n",
      "'M_fru16bp[c]' is not a valid SBML 'SId'.\n",
      "'M_actnS[c]' is not a valid SBML 'SId'.\n",
      "'M_5a3oh[c]' is not a valid SBML 'SId'.\n",
      "'M_serL[c]' is not a valid SBML 'SId'.\n",
      "'M_aga[c]' is not a valid SBML 'SId'.\n",
      "'M_2o6pamg2d[c]' is not a valid SBML 'SId'.\n",
      "'M_Lip[c]' is not a valid SBML 'SId'.\n",
      "'M_gccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_na[c]' is not a valid SBML 'SId'.\n",
      "'M_510methf[c]' is not a valid SBML 'SId'.\n",
      "'M_abdmagdpu[c]' is not a valid SBML 'SId'.\n",
      "'M_ga1p[c]' is not a valid SBML 'SId'.\n",
      "'M_urea[c]' is not a valid SBML 'SId'.\n",
      "'M_2oiv[c]' is not a valid SBML 'SId'.\n",
      "'M_pyal[c]' is not a valid SBML 'SId'.\n",
      "'M_leuL[e]' is not a valid SBML 'SId'.\n",
      "'M_dhettp[c]' is not a valid SBML 'SId'.\n",
      "'M_2a4op[c]' is not a valid SBML 'SId'.\n",
      "'M_nacaddn[c]' is not a valid SBML 'SId'.\n",
      "'M_m5trib1p[c]' is not a valid SBML 'SId'.\n",
      "'M_2iac[c]' is not a valid SBML 'SId'.\n",
      "'M_hpsamdhllys[c]' is not a valid SBML 'SId'.\n",
      "'M_thm[c]' is not a valid SBML 'SId'.\n",
      "'M_nacput[c]' is not a valid SBML 'SId'.\n",
      "'M_isocap[c]' is not a valid SBML 'SId'.\n",
      "'M_dad[c]' is not a valid SBML 'SId'.\n",
      "'M_2mbcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_bio5a[c]' is not a valid SBML 'SId'.\n",
      "'M_ribflv[c]' is not a valid SBML 'SId'.\n",
      "'M_fum[c]' is not a valid SBML 'SId'.\n",
      "'M_fum[e]' is not a valid SBML 'SId'.\n",
      "'M_lacgth[c]' is not a valid SBML 'SId'.\n",
      "'M_4a5imzc[c]' is not a valid SBML 'SId'.\n",
      "'M_dachi[e]' is not a valid SBML 'SId'.\n",
      "'M_k[e]' is not a valid SBML 'SId'.\n",
      "'M_ornL[c]' is not a valid SBML 'SId'.\n",
      "'M_dcdp[c]' is not a valid SBML 'SId'.\n",
      "'M_5fithf[c]' is not a valid SBML 'SId'.\n",
      "'M_s26dap[c]' is not a valid SBML 'SId'.\n",
      "'M_acb[c]' is not a valid SBML 'SId'.\n",
      "'M_hco3[c]' is not a valid SBML 'SId'.\n",
      "'M_adman[c]' is not a valid SBML 'SId'.\n",
      "'M_35dah[c]' is not a valid SBML 'SId'.\n",
      "'M_gal6p[c]' is not a valid SBML 'SId'.\n",
      "'M_rz[c]' is not a valid SBML 'SId'.\n",
      "'M_pyin[e]' is not a valid SBML 'SId'.\n",
      "'M_uqnl8[c]' is not a valid SBML 'SId'.\n",
      "'M_ac[e]' is not a valid SBML 'SId'.\n",
      "'M_cdpg[e]' is not a valid SBML 'SId'.\n",
      "'M_eto[e]' is not a valid SBML 'SId'.\n",
      "'M_din[c]' is not a valid SBML 'SId'.\n",
      "'M_78dhnptetp[c]' is not a valid SBML 'SId'.\n",
      "'M_malt[e]' is not a valid SBML 'SId'.\n",
      "'M_adn[c]' is not a valid SBML 'SId'.\n",
      "'M_2obut[c]' is not a valid SBML 'SId'.\n",
      "'M_imzap[c]' is not a valid SBML 'SId'.\n",
      "'M_dachi6p[c]' is not a valid SBML 'SId'.\n",
      "'M_gtol[e]' is not a valid SBML 'SId'.\n",
      "'M_da[c]' is not a valid SBML 'SId'.\n",
      "'M_5a15pri4c[c]' is not a valid SBML 'SId'.\n",
      "'M_raf[c]' is not a valid SBML 'SId'.\n",
      "'M_tartsald[c]' is not a valid SBML 'SId'.\n",
      "'M_ichor[c]' is not a valid SBML 'SId'.\n",
      "'M_malacp[c]' is not a valid SBML 'SId'.\n",
      "'M_lpteichoic[c]' is not a valid SBML 'SId'.\n",
      "'M_4m2op[c]' is not a valid SBML 'SId'.\n",
      "'M_5mthf[c]' is not a valid SBML 'SId'.\n",
      "'M_glcAD[e]' is not a valid SBML 'SId'.\n",
      "'M_12dh5mtp[c]' is not a valid SBML 'SId'.\n",
      "'M_mthx5p[c]' is not a valid SBML 'SId'.\n",
      "'M_nadp[c]' is not a valid SBML 'SId'.\n",
      "'M_glyo[e]' is not a valid SBML 'SId'.\n",
      "'M_2s6h24ch1c[c]' is not a valid SBML 'SId'.\n",
      "'M_thmp[c]' is not a valid SBML 'SId'.\n",
      "'M_ppac[c]' is not a valid SBML 'SId'.\n",
      "'M_leuL[c]' is not a valid SBML 'SId'.\n",
      "'M_mnl[e]' is not a valid SBML 'SId'.\n",
      "'M_citrL[c]' is not a valid SBML 'SId'.\n",
      "'M_tag6p[c]' is not a valid SBML 'SId'.\n",
      "'M_2m3op[c]' is not a valid SBML 'SId'.\n",
      "'M_idp[c]' is not a valid SBML 'SId'.\n",
      "'M_nadh[c]' is not a valid SBML 'SId'.\n",
      "'M_4m52poet[c]' is not a valid SBML 'SId'.\n",
      "'M_23dhbz[e]' is not a valid SBML 'SId'.\n",
      "'M_gly[c]' is not a valid SBML 'SId'.\n",
      "'M_laldS[c]' is not a valid SBML 'SId'.\n",
      "'M_glc6p[c]' is not a valid SBML 'SId'.\n",
      "'M_fad[c]' is not a valid SBML 'SId'.\n",
      "'M_5apn[c]' is not a valid SBML 'SId'.\n",
      "'M_udpglu[c]' is not a valid SBML 'SId'.\n",
      "'M_gly[e]' is not a valid SBML 'SId'.\n",
      "'M_aglu[c]' is not a valid SBML 'SId'.\n",
      "'M_thrL[c]' is not a valid SBML 'SId'.\n",
      "'M_udpagaepyr[c]' is not a valid SBML 'SId'.\n",
      "'M_RNA[c]' is not a valid SBML 'SId'.\n",
      "'M_skm[c]' is not a valid SBML 'SId'.\n",
      "'M_ppat[c]' is not a valid SBML 'SId'.\n",
      "'M_vacoa[c]' is not a valid SBML 'SId'.\n",
      "'M_tre6p[c]' is not a valid SBML 'SId'.\n",
      "'M_imp[c]' is not a valid SBML 'SId'.\n",
      "'M_glnL[e]' is not a valid SBML 'SId'.\n",
      "'M_aorn[c]' is not a valid SBML 'SId'.\n",
      "'M_ppacal[c]' is not a valid SBML 'SId'.\n",
      "'M_et4p[c]' is not a valid SBML 'SId'.\n",
      "'M_gdp[c]' is not a valid SBML 'SId'.\n",
      "'M_isobuta[e]' is not a valid SBML 'SId'.\n",
      "'M_fe2[e]' is not a valid SBML 'SId'.\n",
      "'M_s3m2op[c]' is not a valid SBML 'SId'.\n",
      "'M_4mpo[e]' is not a valid SBML 'SId'.\n",
      "'M_fe2[c]' is not a valid SBML 'SId'.\n",
      "'M_cmp[c]' is not a valid SBML 'SId'.\n",
      "'M_at[c]' is not a valid SBML 'SId'.\n",
      "'M_tmam[e]' is not a valid SBML 'SId'.\n",
      "'M_15pramp[c]' is not a valid SBML 'SId'.\n",
      "'M_lysL[e]' is not a valid SBML 'SId'.\n",
      "'M_c1ala[c]' is not a valid SBML 'SId'.\n",
      "'M_2sucbzcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_s5ddr5lhcys[c]' is not a valid SBML 'SId'.\n",
      "'M_psuri5p[c]' is not a valid SBML 'SId'.\n",
      "'M_3ipm[c]' is not a valid SBML 'SId'.\n",
      "'M_aacoa[c]' is not a valid SBML 'SId'.\n",
      "'M_glcAD[c]' is not a valid SBML 'SId'.\n",
      "'M_alaL[c]' is not a valid SBML 'SId'.\n",
      "'M_actid[e]' is not a valid SBML 'SId'.\n",
      "'M_tagd[e]' is not a valid SBML 'SId'.\n",
      "'M_eimzgp[c]' is not a valid SBML 'SId'.\n",
      "'M_2a2hbut[c]' is not a valid SBML 'SId'.\n",
      "'M_mnl1p[c]' is not a valid SBML 'SId'.\n",
      "'M_5a6raura[c]' is not a valid SBML 'SId'.\n",
      "'M_cobalt[e]' is not a valid SBML 'SId'.\n",
      "'M_14dh2no[c]' is not a valid SBML 'SId'.\n",
      "'M_tyrL[c]' is not a valid SBML 'SId'.\n",
      "'M_pyam5p[c]' is not a valid SBML 'SId'.\n",
      "'M_prfaca[c]' is not a valid SBML 'SId'.\n",
      "'M_23dhbz[c]' is not a valid SBML 'SId'.\n",
      "'M_pyam[c]' is not a valid SBML 'SId'.\n",
      "'M_pepyr[c]' is not a valid SBML 'SId'.\n",
      "'M_chol[e]' is not a valid SBML 'SId'.\n",
      "'M_hpyr[c]' is not a valid SBML 'SId'.\n",
      "'M_alaL[e]' is not a valid SBML 'SId'.\n",
      "'M_ca2[e]' is not a valid SBML 'SId'.\n",
      "'M_isuc[c]' is not a valid SBML 'SId'.\n",
      "'M_cysgly[c]' is not a valid SBML 'SId'.\n",
      "'M_4hcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_dctp[c]' is not a valid SBML 'SId'.\n",
      "'M_at[e]' is not a valid SBML 'SId'.\n",
      "'M_udpamr[c]' is not a valid SBML 'SId'.\n",
      "'M_g3p[c]' is not a valid SBML 'SId'.\n",
      "'M_2mbut[e]' is not a valid SBML 'SId'.\n",
      "'M_thymd[c]' is not a valid SBML 'SId'.\n",
      "'M_uri[e]' is not a valid SBML 'SId'.\n",
      "'M_xan[e]' is not a valid SBML 'SId'.\n",
      "'M_acnam[c]' is not a valid SBML 'SId'.\n",
      "'M_5pra[c]' is not a valid SBML 'SId'.\n",
      "'M_aacp[c]' is not a valid SBML 'SId'.\n",
      "'M_4hbut[c]' is not a valid SBML 'SId'.\n",
      "'M_adp[c]' is not a valid SBML 'SId'.\n",
      "'M_nac[c]' is not a valid SBML 'SId'.\n",
      "'M_ctp[c]' is not a valid SBML 'SId'.\n",
      "'M_uri[c]' is not a valid SBML 'SId'.\n",
      "'M_hacp[c]' is not a valid SBML 'SId'.\n",
      "'M_udpamag[c]' is not a valid SBML 'SId'.\n",
      "'M_4cyt5dp2me[c]' is not a valid SBML 'SId'.\n",
      "'M_adpglu[c]' is not a valid SBML 'SId'.\n",
      "'M_pppgpp[c]' is not a valid SBML 'SId'.\n",
      "'M_5fthf[c]' is not a valid SBML 'SId'.\n",
      "'M_4hoplac[c]' is not a valid SBML 'SId'.\n",
      "'M_5a15primz[c]' is not a valid SBML 'SId'.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'M_2dh3dg[e]' is not a valid SBML 'SId'.\n",
      "'M_dad[e]' is not a valid SBML 'SId'.\n",
      "'M_5all[c]' is not a valid SBML 'SId'.\n",
      "'M_ama6p[c]' is not a valid SBML 'SId'.\n",
      "'M_escut[e]' is not a valid SBML 'SId'.\n",
      "'M_10fthf[c]' is not a valid SBML 'SId'.\n",
      "'M_isobutp[c]' is not a valid SBML 'SId'.\n",
      "'M_acgdp[c]' is not a valid SBML 'SId'.\n",
      "'M_glx[c]' is not a valid SBML 'SId'.\n",
      "'M_thmdp[c]' is not a valid SBML 'SId'.\n",
      "'M_ins[e]' is not a valid SBML 'SId'.\n",
      "'M_ribflv[e]' is not a valid SBML 'SId'.\n",
      "'M_23diap[c]' is not a valid SBML 'SId'.\n",
      "'M_5prfga[c]' is not a valid SBML 'SId'.\n",
      "'M_cbprc6a[c]' is not a valid SBML 'SId'.\n",
      "'M_2mbut[c]' is not a valid SBML 'SId'.\n",
      "'M_escul[e]' is not a valid SBML 'SId'.\n",
      "'M_fru[e]' is not a valid SBML 'SId'.\n",
      "'M_ade[c]' is not a valid SBML 'SId'.\n",
      "'M_nac[e]' is not a valid SBML 'SId'.\n",
      "'M_betald[c]' is not a valid SBML 'SId'.\n",
      "'M_glc[e]' is not a valid SBML 'SId'.\n",
      "'M_argL[e]' is not a valid SBML 'SId'.\n",
      "'M_5pr4sc5ai[c]' is not a valid SBML 'SId'.\n",
      "'M_hisp[c]' is not a valid SBML 'SId'.\n",
      "'M_actnR[c]' is not a valid SBML 'SId'.\n",
      "'M_4abat[c]' is not a valid SBML 'SId'.\n",
      "'M_gth[c]' is not a valid SBML 'SId'.\n",
      "'M_gth[e]' is not a valid SBML 'SId'.\n",
      "'M_6hmdhptedp[c]' is not a valid SBML 'SId'.\n",
      "'M_cbp[c]' is not a valid SBML 'SId'.\n",
      "'M_ileL[e]' is not a valid SBML 'SId'.\n",
      "'M_mg2[e]' is not a valid SBML 'SId'.\n",
      "'M_dcyt[e]' is not a valid SBML 'SId'.\n",
      "'M_dcmp[c]' is not a valid SBML 'SId'.\n",
      "'M_4s45dhp23do[c]' is not a valid SBML 'SId'.\n",
      "'M_ileL[c]' is not a valid SBML 'SId'.\n",
      "'M_glcn[c]' is not a valid SBML 'SId'.\n",
      "'M_pi[c]' is not a valid SBML 'SId'.\n",
      "'M_hexdp[c]' is not a valid SBML 'SId'.\n",
      "'M_k[c]' is not a valid SBML 'SId'.\n",
      "'M_1dxyl5p[c]' is not a valid SBML 'SId'.\n",
      "'M_sucsald[c]' is not a valid SBML 'SId'.\n",
      "'M_2aepat[c]' is not a valid SBML 'SId'.\n",
      "'M_pendp[c]' is not a valid SBML 'SId'.\n",
      "'M_mevR[e]' is not a valid SBML 'SId'.\n",
      "'M_ca2[c]' is not a valid SBML 'SId'.\n",
      "'M_gmp[c]' is not a valid SBML 'SId'.\n",
      "'M_2mccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_dump[c]' is not a valid SBML 'SId'.\n",
      "'M_csn[e]' is not a valid SBML 'SId'.\n",
      "'M_pyal5p[c]' is not a valid SBML 'SId'.\n",
      "'M_cbfIII[c]' is not a valid SBML 'SId'.\n",
      "'M_23dh3mv[c]' is not a valid SBML 'SId'.\n",
      "'M_uqnn8[c]' is not a valid SBML 'SId'.\n",
      "'M_5pr1dp[c]' is not a valid SBML 'SId'.\n",
      "'M_fuc[e]' is not a valid SBML 'SId'.\n",
      "'M_rl5p[c]' is not a valid SBML 'SId'.\n",
      "'M_glc1pA[c]' is not a valid SBML 'SId'.\n",
      "'M_3pg[c]' is not a valid SBML 'SId'.\n",
      "'M_4s4h2345th2sdl[c]' is not a valid SBML 'SId'.\n",
      "'M_67dm81rlum[c]' is not a valid SBML 'SId'.\n",
      "'M_lacR[c]' is not a valid SBML 'SId'.\n",
      "'M_hcys[c]' is not a valid SBML 'SId'.\n",
      "'M_nadph[c]' is not a valid SBML 'SId'.\n",
      "'M_5prga[c]' is not a valid SBML 'SId'.\n",
      "'M_dgtp[c]' is not a valid SBML 'SId'.\n",
      "'M_proL[e]' is not a valid SBML 'SId'.\n",
      "'M_2mbutp[c]' is not a valid SBML 'SId'.\n",
      "'M_xyld[e]' is not a valid SBML 'SId'.\n",
      "'M_prfiaicarp[c]' is not a valid SBML 'SId'.\n",
      "'M_adprib[c]' is not a valid SBML 'SId'.\n",
      "'M_r3hbat[c]' is not a valid SBML 'SId'.\n",
      "'M_gtol1p[c]' is not a valid SBML 'SId'.\n",
      "'M_5doadn[c]' is not a valid SBML 'SId'.\n",
      "'M_alaala[c]' is not a valid SBML 'SId'.\n",
      "'M_pi[e]' is not a valid SBML 'SId'.\n",
      "'M_gluL[c]' is not a valid SBML 'SId'.\n",
      "'M_2pg[c]' is not a valid SBML 'SId'.\n",
      "'M_chol[c]' is not a valid SBML 'SId'.\n",
      "'M_dtdp[c]' is not a valid SBML 'SId'.\n",
      "'M_tyrL[e]' is not a valid SBML 'SId'.\n",
      "'M_ibtal[c]' is not a valid SBML 'SId'.\n",
      "'M_adcobamp[c]' is not a valid SBML 'SId'.\n",
      "'M_3dahs7p[c]' is not a valid SBML 'SId'.\n",
      "'M_4s45dhp23do[e]' is not a valid SBML 'SId'.\n",
      "'M_r3hbatcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_acnam[e]' is not a valid SBML 'SId'.\n",
      "'M_alaD[c]' is not a valid SBML 'SId'.\n",
      "'M_rbl5p[c]' is not a valid SBML 'SId'.\n",
      "'M_trpL[e]' is not a valid SBML 'SId'.\n",
      "'M_ncammnuc[c]' is not a valid SBML 'SId'.\n",
      "'M_cbshclin[c]' is not a valid SBML 'SId'.\n",
      "'M_sor[e]' is not a valid SBML 'SId'.\n",
      "'M_glyald3p[c]' is not a valid SBML 'SId'.\n",
      "'M_hoam[c]' is not a valid SBML 'SId'.\n",
      "'M_raf[e]' is not a valid SBML 'SId'.\n",
      "'M_isobutcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_2drib1p[c]' is not a valid SBML 'SId'.\n",
      "'M_1h2m2b4dp[c]' is not a valid SBML 'SId'.\n",
      "'M_uri5p[c]' is not a valid SBML 'SId'.\n",
      "'M_cbdhpcr6[c]' is not a valid SBML 'SId'.\n",
      "'M_nacmnc[c]' is not a valid SBML 'SId'.\n",
      "'M_aga6p[c]' is not a valid SBML 'SId'.\n",
      "'M_so3[c]' is not a valid SBML 'SId'.\n",
      "'M_so3[e]' is not a valid SBML 'SId'.\n",
      "'M_isobuta[c]' is not a valid SBML 'SId'.\n",
      "'M_15pr55pramai4c[c]' is not a valid SBML 'SId'.\n",
      "'M_dttp[c]' is not a valid SBML 'SId'.\n",
      "'M_arab[e]' is not a valid SBML 'SId'.\n",
      "'M_ibtol[c]' is not a valid SBML 'SId'.\n",
      "'M_celb[e]' is not a valid SBML 'SId'.\n",
      "'M_4aimz[c]' is not a valid SBML 'SId'.\n",
      "'M_thf[c]' is not a valid SBML 'SId'.\n",
      "'M_thioredoxoxi[c]' is not a valid SBML 'SId'.\n",
      "'M_fmnh2[c]' is not a valid SBML 'SId'.\n",
      "'M_glc1p[c]' is not a valid SBML 'SId'.\n",
      "'M_36dah[c]' is not a valid SBML 'SId'.\n",
      "'M_hista[c]' is not a valid SBML 'SId'.\n",
      "'M_malt[c]' is not a valid SBML 'SId'.\n",
      "'M_ncam[e]' is not a valid SBML 'SId'.\n",
      "'M_co[c]' is not a valid SBML 'SId'.\n",
      "'M_udpamagdap[c]' is not a valid SBML 'SId'.\n",
      "'M_xyld[c]' is not a valid SBML 'SId'.\n",
      "'M_fald[c]' is not a valid SBML 'SId'.\n",
      "'M_dap[c]' is not a valid SBML 'SId'.\n",
      "'M_chor[e]' is not a valid SBML 'SId'.\n",
      "'M_dtmp[c]' is not a valid SBML 'SId'.\n",
      "'M_ptrc[e]' is not a valid SBML 'SId'.\n",
      "'M_metL[e]' is not a valid SBML 'SId'.\n",
      "'M_no2[c]' is not a valid SBML 'SId'.\n",
      "'M_aro[c]' is not a valid SBML 'SId'.\n",
      "'M_CW[c]' is not a valid SBML 'SId'.\n",
      "'M_fru[c]' is not a valid SBML 'SId'.\n",
      "'M_4ppcys[c]' is not a valid SBML 'SId'.\n",
      "'M_4abg[e]' is not a valid SBML 'SId'.\n",
      "'M_aspsald[c]' is not a valid SBML 'SId'.\n",
      "'M_lacS[c]' is not a valid SBML 'SId'.\n",
      "'M_lacS[e]' is not a valid SBML 'SId'.\n",
      "'M_udpama[c]' is not a valid SBML 'SId'.\n",
      "'M_man6p[c]' is not a valid SBML 'SId'.\n",
      "'M_cdpg[c]' is not a valid SBML 'SId'.\n",
      "'M_orot[c]' is not a valid SBML 'SId'.\n",
      "'M_78dhnpte3p[c]' is not a valid SBML 'SId'.\n",
      "'M_thr3p[c]' is not a valid SBML 'SId'.\n",
      "'M_sor[c]' is not a valid SBML 'SId'.\n",
      "'M_hgcvpdhllys[c]' is not a valid SBML 'SId'.\n",
      "'M_sprm[c]' is not a valid SBML 'SId'.\n",
      "'M_aspL[e]' is not a valid SBML 'SId'.\n",
      "'M_fesm[c]' is not a valid SBML 'SId'.\n",
      "'M_fadh2[c]' is not a valid SBML 'SId'.\n",
      "'M_2dh3dg6p[c]' is not a valid SBML 'SId'.\n",
      "'M_ahcys[c]' is not a valid SBML 'SId'.\n",
      "'M_ibtol[e]' is not a valid SBML 'SId'.\n",
      "'M_cbpc4[c]' is not a valid SBML 'SId'.\n",
      "'M_aga1p[c]' is not a valid SBML 'SId'.\n",
      "'M_cysL[e]' is not a valid SBML 'SId'.\n",
      "'M_4a2m5dpmp[c]' is not a valid SBML 'SId'.\n",
      "'M_xyl[c]' is not a valid SBML 'SId'.\n",
      "'M_4u5imzc[c]' is not a valid SBML 'SId'.\n",
      "'M_cobalt[c]' is not a valid SBML 'SId'.\n",
      "'M_gal[e]' is not a valid SBML 'SId'.\n",
      "'M_celb6p[c]' is not a valid SBML 'SId'.\n",
      "'M_thrL[e]' is not a valid SBML 'SId'.\n",
      "'M_rblL[c]' is not a valid SBML 'SId'.\n",
      "'M_asp4p[c]' is not a valid SBML 'SId'.\n",
      "'M_dgdp[c]' is not a valid SBML 'SId'.\n",
      "'M_aspL[c]' is not a valid SBML 'SId'.\n",
      "'M_78dhnpte[c]' is not a valid SBML 'SId'.\n",
      "'M_udpaga[c]' is not a valid SBML 'SId'.\n",
      "'M_ptth[c]' is not a valid SBML 'SId'.\n",
      "'M_ppat[e]' is not a valid SBML 'SId'.\n",
      "'M_argL[c]' is not a valid SBML 'SId'.\n",
      "'M_ppcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_udpamagdapaa[c]' is not a valid SBML 'SId'.\n",
      "'M_m5trib[c]' is not a valid SBML 'SId'.\n",
      "'M_4mpo[c]' is not a valid SBML 'SId'.\n",
      "'M_udpagapp[c]' is not a valid SBML 'SId'.\n",
      "'M_valL[c]' is not a valid SBML 'SId'.\n",
      "'M_hcit[c]' is not a valid SBML 'SId'.\n",
      "'M_succ[c]' is not a valid SBML 'SId'.\n",
      "'M_succ[e]' is not a valid SBML 'SId'.\n",
      "'M_dgsn[c]' is not a valid SBML 'SId'.\n",
      "'M_unsc[c]' is not a valid SBML 'SId'.\n",
      "'M_2ipm[c]' is not a valid SBML 'SId'.\n",
      "'M_isocap[e]' is not a valid SBML 'SId'.\n",
      "'M_4m5het[c]' is not a valid SBML 'SId'.\n",
      "'M_56dhura[c]' is not a valid SBML 'SId'.\n",
      "'M_hepdp[c]' is not a valid SBML 'SId'.\n",
      "'M_ap[c]' is not a valid SBML 'SId'.\n",
      "'M_pyr[c]' is not a valid SBML 'SId'.\n",
      "'M_duri[c]' is not a valid SBML 'SId'.\n",
      "'M_pntoR[c]' is not a valid SBML 'SId'.\n",
      "'M_casp[c]' is not a valid SBML 'SId'.\n",
      "'M_2but[c]' is not a valid SBML 'SId'.\n",
      "'M_pc1[c]' is not a valid SBML 'SId'.\n",
      "'M_suchser[c]' is not a valid SBML 'SId'.\n",
      "'M_3a2mp[c]' is not a valid SBML 'SId'.\n",
      "'M_6pg15l[c]' is not a valid SBML 'SId'.\n",
      "'M_for[e]' is not a valid SBML 'SId'.\n",
      "'M_asnL[c]' is not a valid SBML 'SId'.\n",
      "'M_dethio[e]' is not a valid SBML 'SId'.\n",
      "'M_glc[c]' is not a valid SBML 'SId'.\n",
      "'M_cysthL[c]' is not a valid SBML 'SId'.\n",
      "'M_adcbam[c]' is not a valid SBML 'SId'.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'M_2sucbz[c]' is not a valid SBML 'SId'.\n",
      "'M_atp[c]' is not a valid SBML 'SId'.\n",
      "'M_metD[c]' is not a valid SBML 'SId'.\n",
      "'M_glcn[e]' is not a valid SBML 'SId'.\n",
      "'M_teichoic[c]' is not a valid SBML 'SId'.\n",
      "'M_feroxred[c]' is not a valid SBML 'SId'.\n",
      "'M_elecacceptred[c]' is not a valid SBML 'SId'.\n",
      "'M_2maacoa[c]' is not a valid SBML 'SId'.\n",
      "'M_no2[e]' is not a valid SBML 'SId'.\n",
      "'M_altd[c]' is not a valid SBML 'SId'.\n",
      "'M_23diap[e]' is not a valid SBML 'SId'.\n",
      "'M_adman[e]' is not a valid SBML 'SId'.\n",
      "'M_his[e]' is not a valid SBML 'SId'.\n",
      "'M_co2[c]' is not a valid SBML 'SId'.\n",
      "'M_2dmmqn[c]' is not a valid SBML 'SId'.\n",
      "'M_trpL[c]' is not a valid SBML 'SId'.\n",
      "'M_cppogen3[c]' is not a valid SBML 'SId'.\n",
      "'M_isop[c]' is not a valid SBML 'SId'.\n",
      "'M_3uib[c]' is not a valid SBML 'SId'.\n",
      "'M_4abald[c]' is not a valid SBML 'SId'.\n",
      "'M_thioredoxred[c]' is not a valid SBML 'SId'.\n",
      "'M_dmadp[c]' is not a valid SBML 'SId'.\n",
      "'M_tmam[c]' is not a valid SBML 'SId'.\n",
      "'M_hser[c]' is not a valid SBML 'SId'.\n",
      "'M_ura[c]' is not a valid SBML 'SId'.\n",
      "'M_bio[e]' is not a valid SBML 'SId'.\n",
      "'M_xmp[c]' is not a valid SBML 'SId'.\n",
      "'M_glyc[c]' is not a valid SBML 'SId'.\n",
      "'M_5apn[e]' is not a valid SBML 'SId'.\n",
      "'M_fardp[c]' is not a valid SBML 'SId'.\n",
      "'M_coB12[c]' is not a valid SBML 'SId'.\n",
      "'M_ncamrib[c]' is not a valid SBML 'SId'.\n",
      "'M_4ppt[c]' is not a valid SBML 'SId'.\n",
      "'M_cytd[c]' is not a valid SBML 'SId'.\n",
      "'M_malt6p[c]' is not a valid SBML 'SId'.\n",
      "'M_ins[c]' is not a valid SBML 'SId'.\n",
      "'M_thymd[e]' is not a valid SBML 'SId'.\n",
      "'M_gsn[c]' is not a valid SBML 'SId'.\n",
      "'M_serL[e]' is not a valid SBML 'SId'.\n",
      "'M_for[c]' is not a valid SBML 'SId'.\n",
      "'M_amet[c]' is not a valid SBML 'SId'.\n",
      "'M_uppg3[c]' is not a valid SBML 'SId'.\n",
      "'M_isocaprecoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3ppyr[c]' is not a valid SBML 'SId'.\n",
      "'M_g3p[e]' is not a valid SBML 'SId'.\n",
      "'M_agmp[c]' is not a valid SBML 'SId'.\n",
      "'M_cobat[c]' is not a valid SBML 'SId'.\n",
      "'M_aglu5sald[c]' is not a valid SBML 'SId'.\n",
      "'M_glc6pA[c]' is not a valid SBML 'SId'.\n",
      "'M_nad[c]' is not a valid SBML 'SId'.\n",
      "'M_h2[e]' is not a valid SBML 'SId'.\n",
      "'M_drin1p[c]' is not a valid SBML 'SId'.\n",
      "'M_4abg[c]' is not a valid SBML 'SId'.\n",
      "'M_dutp[c]' is not a valid SBML 'SId'.\n",
      "'M_urea1c[c]' is not a valid SBML 'SId'.\n",
      "'M_hxan[c]' is not a valid SBML 'SId'.\n",
      "'M_proL[c]' is not a valid SBML 'SId'.\n",
      "'M_phser[c]' is not a valid SBML 'SId'.\n",
      "'M_ptcys[c]' is not a valid SBML 'SId'.\n",
      "'M_aac[c]' is not a valid SBML 'SId'.\n",
      "'M_etoa[c]' is not a valid SBML 'SId'.\n",
      "'M_aald[c]' is not a valid SBML 'SId'.\n",
      "'M_hxan[e]' is not a valid SBML 'SId'.\n",
      "'M_2aepat[e]' is not a valid SBML 'SId'.\n",
      "'M_ppgpp[c]' is not a valid SBML 'SId'.\n",
      "'M_5a65praura[c]' is not a valid SBML 'SId'.\n",
      "'M_hmb[c]' is not a valid SBML 'SId'.\n",
      "'M_4imzo[c]' is not a valid SBML 'SId'.\n",
      "'M_5epyrsk3p[c]' is not a valid SBML 'SId'.\n",
      "'M_2me4p[c]' is not a valid SBML 'SId'.\n",
      "'M_2o6pamg2d[e]' is not a valid SBML 'SId'.\n",
      "'M_ival[e]' is not a valid SBML 'SId'.\n",
      "'M_m5tade[c]' is not a valid SBML 'SId'.\n",
      "'M_DNA[c]' is not a valid SBML 'SId'.\n",
      "'M_pheL[c]' is not a valid SBML 'SId'.\n",
      "'M_amp[c]' is not a valid SBML 'SId'.\n",
      "'M_pphn[c]' is not a valid SBML 'SId'.\n",
      "'M_aga[e]' is not a valid SBML 'SId'.\n",
      "'M_4abz[c]' is not a valid SBML 'SId'.\n",
      "'M_udpcpp[c]' is not a valid SBML 'SId'.\n",
      "'M_rib1p[c]' is not a valid SBML 'SId'.\n",
      "'M_mql[c]' is not a valid SBML 'SId'.\n",
      "'M_xan[c]' is not a valid SBML 'SId'.\n",
      "'M_proD[c]' is not a valid SBML 'SId'.\n",
      "'M_btal[c]' is not a valid SBML 'SId'.\n",
      "'M_pntoR[e]' is not a valid SBML 'SId'.\n",
      "'M_gal[c]' is not a valid SBML 'SId'.\n",
      "'M_24dap[c]' is not a valid SBML 'SId'.\n",
      "'M_cbpcr5b[c]' is not a valid SBML 'SId'.\n",
      "'M_13bpg[c]' is not a valid SBML 'SId'.\n",
      "'M_na[e]' is not a valid SBML 'SId'.\n",
      "'M_gdpm[c]' is not a valid SBML 'SId'.\n",
      "'M_elecacceptoxi[c]' is not a valid SBML 'SId'.\n",
      "'M_malS[e]' is not a valid SBML 'SId'.\n",
      "'M_hgcvpllys[c]' is not a valid SBML 'SId'.\n",
      "'M_buta[c]' is not a valid SBML 'SId'.\n",
      "'M_4a4dchor[c]' is not a valid SBML 'SId'.\n",
      "'M_tgt[e]' is not a valid SBML 'SId'.\n",
      "'M_acon[c]' is not a valid SBML 'SId'.\n",
      "'M_h2o[c]' is not a valid SBML 'SId'.\n",
      "'M_glyald[c]' is not a valid SBML 'SId'.\n",
      "'M_icit[c]' is not a valid SBML 'SId'.\n",
      "'M_ch4s[c]' is not a valid SBML 'SId'.\n",
      "'M_fru6p[c]' is not a valid SBML 'SId'.\n",
      "'M_his[c]' is not a valid SBML 'SId'.\n",
      "'M_lysL[c]' is not a valid SBML 'SId'.\n",
      "'M_dho[c]' is not a valid SBML 'SId'.\n",
      "'M_gal1pA[c]' is not a valid SBML 'SId'.\n",
      "'M_78dhfmg[c]' is not a valid SBML 'SId'.\n",
      "'M_3h3mgcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_ptrc[c]' is not a valid SBML 'SId'.\n",
      "'M_ahser[c]' is not a valid SBML 'SId'.\n",
      "'M_malS[c]' is not a valid SBML 'SId'.\n",
      "'M_2m3hbcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_tsul[c]' is not a valid SBML 'SId'.\n",
      "'M_h[c]' is not a valid SBML 'SId'.\n",
      "'M_succoa[c]' is not a valid SBML 'SId'.\n",
      "'M_dapime[c]' is not a valid SBML 'SId'.\n",
      "'M_ametam[c]' is not a valid SBML 'SId'.\n",
      "'M_agm[c]' is not a valid SBML 'SId'.\n",
      "'M_zn2[e]' is not a valid SBML 'SId'.\n",
      "'M_3mpyr[c]' is not a valid SBML 'SId'.\n",
      "'M_mevR[c]' is not a valid SBML 'SId'.\n",
      "'M_ura[e]' is not a valid SBML 'SId'.\n",
      "'M_udpamala[c]' is not a valid SBML 'SId'.\n",
      "'M_mg2[c]' is not a valid SBML 'SId'.\n",
      "'M_sor6p[c]' is not a valid SBML 'SId'.\n",
      "'M_alaB[c]' is not a valid SBML 'SId'.\n",
      "'M_actid[c]' is not a valid SBML 'SId'.\n",
      "'M_ctncoa[c]' is not a valid SBML 'SId'.\n",
      "'M_2345thd[c]' is not a valid SBML 'SId'.\n",
      "'M_xtsn[c]' is not a valid SBML 'SId'.\n",
      "'M_dadp[c]' is not a valid SBML 'SId'.\n",
      "'M_bio[c]' is not a valid SBML 'SId'.\n",
      "'M_escul[c]' is not a valid SBML 'SId'.\n",
      "'M_serD[c]' is not a valid SBML 'SId'.\n",
      "'M_rz5p[c]' is not a valid SBML 'SId'.\n",
      "'M_adn[e]' is not a valid SBML 'SId'.\n",
      "'M_1a2po2p[c]' is not a valid SBML 'SId'.\n",
      "'M_dgo15lac[c]' is not a valid SBML 'SId'.\n",
      "'M_glyo[c]' is not a valid SBML 'SId'.\n",
      "'M_fruf[c]' is not a valid SBML 'SId'.\n",
      "'M_hicit[c]' is not a valid SBML 'SId'.\n",
      "'M_shl7p[c]' is not a valid SBML 'SId'.\n",
      "'M_proheme[c]' is not a valid SBML 'SId'.\n",
      "'M_dcyt[c]' is not a valid SBML 'SId'.\n",
      "'M_2ippm[c]' is not a valid SBML 'SId'.\n",
      "'M_3phpyr[c]' is not a valid SBML 'SId'.\n",
      "'M_nh3[e]' is not a valid SBML 'SId'.\n",
      "'M_biomass[c]' is not a valid SBML 'SId'.\n",
      "'M_5prfgad[c]' is not a valid SBML 'SId'.\n",
      "'M_h[e]' is not a valid SBML 'SId'.\n",
      "'M_ribD[e]' is not a valid SBML 'SId'.\n",
      "'M_c1yda[c]' is not a valid SBML 'SId'.\n",
      "'M_2h3k5mt1pp[c]' is not a valid SBML 'SId'.\n",
      "'M_asucc[c]' is not a valid SBML 'SId'.\n",
      "'M_escut[c]' is not a valid SBML 'SId'.\n",
      "'M_56dmbimz[c]' is not a valid SBML 'SId'.\n",
      "'M_2ip3os[c]' is not a valid SBML 'SId'.\n",
      "'M_coa[c]' is not a valid SBML 'SId'.\n",
      "'M_feroxoxi[c]' is not a valid SBML 'SId'.\n",
      "'M_25da65prap[c]' is not a valid SBML 'SId'.\n",
      "'M_acbda[c]' is not a valid SBML 'SId'.\n",
      "'M_asnL[e]' is not a valid SBML 'SId'.\n",
      "'M_3dhq[c]' is not a valid SBML 'SId'.\n",
      "'M_mthgxl[c]' is not a valid SBML 'SId'.\n",
      "'M_3pser[c]' is not a valid SBML 'SId'.\n",
      "'M_man1p[c]' is not a valid SBML 'SId'.\n",
      "'M_sucr[e]' is not a valid SBML 'SId'.\n",
      "'M_zn2[c]' is not a valid SBML 'SId'.\n",
      "'M_ssc[c]' is not a valid SBML 'SId'.\n",
      "'M_h2[c]' is not a valid SBML 'SId'.\n",
      "'M_dgmp[c]' is not a valid SBML 'SId'.\n",
      "'M_2dh3dg[c]' is not a valid SBML 'SId'.\n",
      "'M_acoa[c]' is not a valid SBML 'SId'.\n",
      "'M_co2[e]' is not a valid SBML 'SId'.\n",
      "'M_2alac[c]' is not a valid SBML 'SId'.\n",
      "'M_cit[c]' is not a valid SBML 'SId'.\n",
      "'M_celb[c]' is not a valid SBML 'SId'.\n",
      "'M_glyb[c]' is not a valid SBML 'SId'.\n",
      "'M_mqn[c]' is not a valid SBML 'SId'.\n",
      "'M_ornD[c]' is not a valid SBML 'SId'.\n",
      "'M_udpagaepyr[e]' is not a valid SBML 'SId'.\n",
      "'M_isocapcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_ncam[c]' is not a valid SBML 'SId'.\n",
      "'M_tagd[c]' is not a valid SBML 'SId'.\n",
      "'M_sucr6p[c]' is not a valid SBML 'SId'.\n",
      "'M_6pg[c]' is not a valid SBML 'SId'.\n",
      "'M_r5p[c]' is not a valid SBML 'SId'.\n",
      "'M_udp[c]' is not a valid SBML 'SId'.\n",
      "'M_csn[c]' is not a valid SBML 'SId'.\n",
      "'M_glyb[e]' is not a valid SBML 'SId'.\n",
      "'M_3abcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_gcald[c]' is not a valid SBML 'SId'.\n",
      "'M_fgly[c]' is not a valid SBML 'SId'.\n",
      "'M_spmd[e]' is not a valid SBML 'SId'.\n",
      "'M_quln[c]' is not a valid SBML 'SId'.\n",
      "'M_gpgl[c]' is not a valid SBML 'SId'.\n",
      "'M_ppi[c]' is not a valid SBML 'SId'.\n",
      "'M_4hplald[c]' is not a valid SBML 'SId'.\n",
      "'M_damp[c]' is not a valid SBML 'SId'.\n",
      "'M_hoam[e]' is not a valid SBML 'SId'.\n",
      "'M_2me24cdp[c]' is not a valid SBML 'SId'.\n",
      "'M_naspin[c]' is not a valid SBML 'SId'.\n",
      "'M_gtp[c]' is not a valid SBML 'SId'.\n",
      "'M_grgrdp[c]' is not a valid SBML 'SId'.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'M_valL[e]' is not a valid SBML 'SId'.\n",
      "'M_ama[c]' is not a valid SBML 'SId'.\n",
      "'M_ivalcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_pratp[c]' is not a valid SBML 'SId'.\n",
      "'M_SPs[c]' is not a valid SBML 'SId'.\n",
      "'M_xyl5p[c]' is not a valid SBML 'SId'.\n",
      "'M_metD[e]' is not a valid SBML 'SId'.\n",
      "'M_c2yda[c]' is not a valid SBML 'SId'.\n",
      "'M_ipdp[c]' is not a valid SBML 'SId'.\n",
      "'M_udpgal[c]' is not a valid SBML 'SId'.\n",
      "'M_pap[c]' is not a valid SBML 'SId'.\n",
      "'M_aimzcrnt[c]' is not a valid SBML 'SId'.\n",
      "'M_orot5p[c]' is not a valid SBML 'SId'.\n",
      "'M_2dhp[c]' is not a valid SBML 'SId'.\n",
      "'M_h2o[e]' is not a valid SBML 'SId'.\n",
      "'M_cbpcr7[c]' is not a valid SBML 'SId'.\n",
      "'M_shclin[c]' is not a valid SBML 'SId'.\n",
      "'M_23dh23dhvz[c]' is not a valid SBML 'SId'.\n",
      "'M_gluD[c]' is not a valid SBML 'SId'.\n",
      "'M_gluL[e]' is not a valid SBML 'SId'.\n",
      "'M_thym[c]' is not a valid SBML 'SId'.\n",
      "'M_2drib5p[c]' is not a valid SBML 'SId'.\n",
      "'M_pc2[c]' is not a valid SBML 'SId'.\n",
      "'M_510mlethf[c]' is not a valid SBML 'SId'.\n",
      "'M_skm3p[c]' is not a valid SBML 'SId'.\n",
      "'M_2oa[c]' is not a valid SBML 'SId'.\n",
      "'M_galA[c]' is not a valid SBML 'SId'.\n",
      "'M_2p4c5dp2me[c]' is not a valid SBML 'SId'.\n",
      "'M_utp[c]' is not a valid SBML 'SId'.\n",
      "'M_dudp[c]' is not a valid SBML 'SId'.\n",
      "'M_56dhthy[c]' is not a valid SBML 'SId'.\n",
      "'M_metL[c]' is not a valid SBML 'SId'.\n",
      "'M_h2s[c]' is not a valid SBML 'SId'.\n",
      "'M_hco3[e]' is not a valid SBML 'SId'.\n",
      "'M_ribD[c]' is not a valid SBML 'SId'.\n",
      "'M_udcpdp[c]' is not a valid SBML 'SId'.\n",
      "'M_p1p3b5adtp[c]' is not a valid SBML 'SId'.\n",
      "'M_4abz[e]' is not a valid SBML 'SId'.\n",
      "'M_malsald[c]' is not a valid SBML 'SId'.\n",
      "'M_3dhsk[c]' is not a valid SBML 'SId'.\n",
      "'M_glnL[c]' is not a valid SBML 'SId'.\n",
      "'M_gua[c]' is not a valid SBML 'SId'.\n",
      "'M_cbpcr8x[c]' is not a valid SBML 'SId'.\n",
      "'M_tag16bp[c]' is not a valid SBML 'SId'.\n",
      "'M_cbprc5a[c]' is not a valid SBML 'SId'.\n",
      "'M_34hppyr[c]' is not a valid SBML 'SId'.\n",
      "'M_malcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_pbng[c]' is not a valid SBML 'SId'.\n",
      "'M_ival[c]' is not a valid SBML 'SId'.\n",
      "'M_etoa[e]' is not a valid SBML 'SId'.\n",
      "'M_dhap[c]' is not a valid SBML 'SId'.\n",
      "'M_5a65prsaura[c]' is not a valid SBML 'SId'.\n",
      "'M_no3[e]' is not a valid SBML 'SId'.\n",
      "'M_5mt23dopp[c]' is not a valid SBML 'SId'.\n",
      "'M_gcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_phpyr[c]' is not a valid SBML 'SId'.\n",
      "'M_arab[c]' is not a valid SBML 'SId'.\n",
      "'M_pyin[c]' is not a valid SBML 'SId'.\n",
      "'M_grdp[c]' is not a valid SBML 'SId'.\n",
      "'M_octdp[c]' is not a valid SBML 'SId'.\n",
      "'M_chor[c]' is not a valid SBML 'SId'.\n",
      "'M_eto[c]' is not a valid SBML 'SId'.\n",
      "'M_2oglut[c]' is not a valid SBML 'SId'.\n",
      "'M_2oglut[e]' is not a valid SBML 'SId'.\n",
      "'M_Prot[c]' is not a valid SBML 'SId'.\n",
      "'M_cdp[c]' is not a valid SBML 'SId'.\n",
      "'M_fuc1p[c]' is not a valid SBML 'SId'.\n",
      "'M_4a2m5pmp[c]' is not a valid SBML 'SId'.\n",
      "'M_prpnte[c]' is not a valid SBML 'SId'.\n",
      "'M_cmylphosph[c]' is not a valid SBML 'SId'.\n",
      "'M_ncmylasp[c]' is not a valid SBML 'SId'.\n",
      "'M_glusemald[c]' is not a valid SBML 'SId'.\n",
      "'M_2am2prpnte[c]' is not a valid SBML 'SId'.\n",
      "'M_palmacd[c]' is not a valid SBML 'SId'.\n",
      "'M_butacp[c]' is not a valid SBML 'SId'.\n",
      "'M_hexacp[c]' is not a valid SBML 'SId'.\n",
      "'M_octacp[c]' is not a valid SBML 'SId'.\n",
      "'M_decacp[c]' is not a valid SBML 'SId'.\n",
      "'M_dodecacp[c]' is not a valid SBML 'SId'.\n",
      "'M_myrstacp[c]' is not a valid SBML 'SId'.\n",
      "'M_palmacp[c]' is not a valid SBML 'SId'.\n",
      "'M_sracp[c]' is not a valid SBML 'SId'.\n",
      "'M_btcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_hexcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_octcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_deccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_dodeccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_myrstcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_palmcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_srcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxybutacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxyhexacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxyoctacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxydecacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxydodecacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxymyristacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxypalmacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3hbcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxyhexcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxyoctcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxydeccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxydodeccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxymyristcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3hydroxypalmcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxohexacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxooctacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxodecacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxododecacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxomyristacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxopalmacp[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxohexcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxooctcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxodeccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxododeccoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxomyristcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_3oxopalmcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_but2enacp[c]' is not a valid SBML 'SId'.\n",
      "'M_thex2enacp[c]' is not a valid SBML 'SId'.\n",
      "'M_toct2enacp[c]' is not a valid SBML 'SId'.\n",
      "'M_2dec2enacp[c]' is not a valid SBML 'SId'.\n",
      "'M_2dodec2enacp[c]' is not a valid SBML 'SId'.\n",
      "'M_ttetdec2enoylacp[c]' is not a valid SBML 'SId'.\n",
      "'M_thexdecenoylacp[c]' is not a valid SBML 'SId'.\n",
      "'M_but2encoa[c]' is not a valid SBML 'SId'.\n",
      "'M_thex2encoa[c]' is not a valid SBML 'SId'.\n",
      "'M_toct2encoa[c]' is not a valid SBML 'SId'.\n",
      "'M_2dec2encoa[c]' is not a valid SBML 'SId'.\n",
      "'M_2dodec2encoa[c]' is not a valid SBML 'SId'.\n",
      "'M_ttetdec2enoylcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_thexdecenoylcoa[c]' is not a valid SBML 'SId'.\n",
      "'M_acacp[c]' is not a valid SBML 'SId'.\n",
      "'M_acacacp[c]' is not a valid SBML 'SId'.\n",
      "'M_1hexdecg3p[c]' is not a valid SBML 'SId'.\n",
      "'M_1tetdecg3p[c]' is not a valid SBML 'SId'.\n",
      "'M_1octdecg3p[c]' is not a valid SBML 'SId'.\n",
      "'M_12dhexdec3gp[c]' is not a valid SBML 'SId'.\n",
      "'M_12dtetradecg3p[c]' is not a valid SBML 'SId'.\n",
      "'M_12dioctdecg3p[c]' is not a valid SBML 'SId'.\n",
      "'M_mglucsyldpalmgl[c]' is not a valid SBML 'SId'.\n",
      "'M_mglucsyldmygl[c]' is not a valid SBML 'SId'.\n",
      "'M_mglucsyldstgl[c]' is not a valid SBML 'SId'.\n",
      "'M_dgludpalmgl[c]' is not a valid SBML 'SId'.\n",
      "'M_dgludmygl[c]' is not a valid SBML 'SId'.\n",
      "'M_digludstgl[c]' is not a valid SBML 'SId'.\n",
      "'M_12dgdihexdec[c]' is not a valid SBML 'SId'.\n",
      "'M_12dgditetdec[c]' is not a valid SBML 'SId'.\n",
      "'M_12dgdioctdec[c]' is not a valid SBML 'SId'.\n",
      "'M_cdpdhexdecglc[c]' is not a valid SBML 'SId'.\n",
      "'M_cdpdtetdecglc[c]' is not a valid SBML 'SId'.\n",
      "'M_cdpoctdecglc[c]' is not a valid SBML 'SId'.\n",
      "'M_pgpdihexdec[c]' is not a valid SBML 'SId'.\n",
      "'M_pgpditetdec[c]' is not a valid SBML 'SId'.\n",
      "'M_pgpdioctdec[c]' is not a valid SBML 'SId'.\n",
      "'M_phosglcdihexdec[c]' is not a valid SBML 'SId'.\n",
      "'M_phosglcditetdec[c]' is not a valid SBML 'SId'.\n",
      "'M_phosglcdioctdec[c]' is not a valid SBML 'SId'.\n",
      "'M_palmphgl[c]' is not a valid SBML 'SId'.\n",
      "'M_myrphgl[c]' is not a valid SBML 'SId'.\n",
      "'M_strphglc[c]' is not a valid SBML 'SId'.\n",
      "'M_palmcdlpn[c]' is not a valid SBML 'SId'.\n",
      "'M_mycdlpn[c]' is not a valid SBML 'SId'.\n",
      "'M_strcdlpn[c]' is not a valid SBML 'SId'.\n",
      "'M_pmf[c]' is not a valid SBML 'SId'.\n",
      "'M_dgslnt[c]' is not a valid SBML 'SId'.\n",
      "'M_gslnt[c]' is not a valid SBML 'SId'.\n",
      "'M_sectrna[c]' is not a valid SBML 'SId'.\n",
      "'M_sertrnasec[c]' is not a valid SBML 'SId'.\n",
      "'M_sel[c]' is not a valid SBML 'SId'.\n",
      "'M_sel[e]' is not a valid SBML 'SId'.\n",
      "'M_seln[c]' is not a valid SBML 'SId'.\n",
      "'M_selnp[c]' is not a valid SBML 'SId'.\n",
      "'M_slnt[c]' is not a valid SBML 'SId'.\n",
      "'M_slnt[e]' is not a valid SBML 'SId'.\n",
      "'M_selenlhcys[c]' is not a valid SBML 'SId'.\n",
      "'M_selencysthne[c]' is not a valid SBML 'SId'.\n",
      "'M_trnasecys[c]' is not a valid SBML 'SId'.\n",
      "'M_gthox[c]' is not a valid SBML 'SId'.\n",
      "'M_gthox[e]' is not a valid SBML 'SId'.\n",
      "'M_selprotein[c]' is not a valid SBML 'SId'.\n",
      "'M_pyr5carb[c]' is not a valid SBML 'SId'.\n",
      "'R_Ex_sel(e)' is not a valid SBML 'SId'.\n",
      "'R_Ex_slnt(e)' is not a valid SBML 'SId'.\n",
      "Model does not contain SBML fbc package information.\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Protein_via_tRNA \"Protein_via_tRNA\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Protein_via_tRNA \"Protein_via_tRNA\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction RNA_Biosyn \"RNA-Biosyn\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction RNA_Biosyn \"RNA-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction DNA_Biosyn \"DNA-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction DNA_Biosyn \"DNA-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Lipid_Biosyn \"Lipid-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Lipid_Biosyn \"Lipid-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_CWG_Biosyn \"L-CWG-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_CWG_Biosyn \"L-CWG-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Peptidoglycan_Biosyn \"Peptidoglycan-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Peptidoglycan_Biosyn \"Peptidoglycan-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction P_CWG_Biosyn \"P-CWG-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction P_CWG_Biosyn \"P-CWG-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Glycogen_Biosyn \"Glycogen-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Glycogen_Biosyn \"Glycogen-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Ion_Biosyn \"Ion-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Ion_Biosyn \"Ion-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Metabolite_Biosyn \"Metabolite-Biosyn\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Metabolite_Biosyn \"Metabolite-Biosyn\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Protease \"Protease\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Protease \"Protease\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction RNAase \"RNAase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction RNAase \"RNAase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction DNAase \"DNAase\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction DNAase \"DNAase\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Biomass \"Biomass\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Biomass \"Biomass\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NGAM \"NGAM\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NGAM \"NGAM\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H_exchange \"H+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H_exchange \"H+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H2O_exchange \"H2O_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H2O_exchange \"H2O_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H2_exchange \"H2_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H2_exchange \"H2_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H2S_exchange \"H2S_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H2S_exchange \"H2S_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CO2_exchange \"CO2_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CO2_exchange \"CO2_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CO_exchange \"CO_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CO_exchange \"CO_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ammonium_exchange \"ammonium_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ammonium_exchange \"ammonium_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfite_exchange \"sulfite_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfite_exchange \"sulfite_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphate_exchange \"phosphate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphate_exchange \"phosphate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction selenide_exchange \"selenide_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction selenide_exchange \"selenide_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Co2_exchange \"Co2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Co2_exchange \"Co2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Na_exchange \"Na+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Na_exchange \"Na+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction K_exchange \"K+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction K_exchange \"K+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Mg2_exchange \"Mg2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Mg2_exchange \"Mg2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Fe2_exchange \"Fe2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Fe2_exchange \"Fe2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Fe3_exchange \"Fe3+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Fe3_exchange \"Fe3+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Ca2_exchange \"Ca2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Ca2_exchange \"Ca2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Ni2_exchange \"Ni2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Ni2_exchange \"Ni2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Cd2_exchange \"Cd2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Cd2_exchange \"Cd2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Zn2_exchange \"Zn2+_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Zn2_exchange \"Zn2+_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdate_exchange \"molybdate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdate_exchange \"molybdate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_Glucose_exchange \"D-Glucose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_Glucose_exchange \"D-Glucose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction urea_exchange \"urea_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction urea_exchange \"urea_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methanethiol_exchange \"methanethiol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methanethiol_exchange \"methanethiol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanol_exchange \"ethanol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanol_exchange \"ethanol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanol_exchange \"propanol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanol_exchange \"propanol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction butanol_exchange \"butanol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction butanol_exchange \"butanol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isobutanol_exchange \"isobutanol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isobutanol_exchange \"isobutanol_exchange\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_methylphenol_exchange \"4-methylphenol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_methylphenol_exchange \"4-methylphenol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetone_exchange \"acetone_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetone_exchange \"acetone_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_pantothenate_exchange \"(R)-pantothenate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_pantothenate_exchange \"(R)-pantothenate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_exchange \"biotin_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_exchange \"biotin_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pyridoxine_exchange \"pyridoxine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pyridoxine_exchange \"pyridoxine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminobutanoate_exchange \"2-aminobutanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminobutanoate_exchange \"2-aminobutanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_methyl_5_thio_D_ribose_exchange \"S-methyl-5-thio-D-ribose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_methyl_5_thio_D_ribose_exchange \"S-methyl-5-thio-D-ribose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminopentanoate_exchange \"5-aminopentanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminopentanoate_exchange \"5-aminopentanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction THMF_exchange \"THMF_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction THMF_exchange \"THMF_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_deoxy_D_ribose_exchange \"5-deoxy-D-ribose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_deoxy_D_ribose_exchange \"5-deoxy-D-ribose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction beta_alanine_exchange \"beta-alanine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction beta_alanine_exchange \"beta-alanine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_amino_2_methylpropanoate_exchange \"3-amino-2-methylpropanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_amino_2_methylpropanoate_exchange \"3-amino-2-methylpropanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aminoaceton_exchange \"aminoaceton_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aminoaceton_exchange \"aminoaceton_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolaldehyde_exchange \"glycolaldehyde_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolaldehyde_exchange \"glycolaldehyde_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_methyl_5_thio_alpha_D_ribose_1_phosphate_exchange \"S-methyl-5-thio-alpha-D-ribose_1-phosphate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_methyl_5_thio_alpha_D_ribose_1_phosphate_exchange \"S-methyl-5-thio-alpha-D-ribose_1-phosphate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_exchange \"valine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_exchange \"valine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_exchange \"leucine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_exchange \"leucine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_exchange \"isoleucine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_exchange \"isoleucine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_exchange \"tryptophan_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_exchange \"tryptophan_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cysteine_exchange \"cysteine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cysteine_exchange \"cysteine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methionine_exchange \"methionine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methionine_exchange \"methionine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction proline_exchange \"proline_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction proline_exchange \"proline_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_exchange \"glycine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_exchange \"glycine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction alanine_exchange \"alanine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction alanine_exchange \"alanine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aspartate_exchange \"aspartate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aspartate_exchange \"aspartate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamate_exchange \"glutamate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamate_exchange \"glutamate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_exchange \"serine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_exchange \"serine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_exchange \"threonine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_exchange \"threonine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_exchange \"tyrosine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_exchange \"tyrosine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_exchange \"phenylalanine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_exchange \"phenylalanine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_exchange \"arginine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_exchange \"arginine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_exchange \"histidine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_exchange \"histidine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_exchange \"lysine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_exchange \"lysine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formate_exchange \"formate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formate_exchange \"formate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetate_exchange \"acetate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetate_exchange \"acetate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_exchange \"propanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_exchange \"propanoate_exchange\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction butanoate_exchange \"butanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction butanoate_exchange \"butanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valerate_exchange \"valerate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valerate_exchange \"valerate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_exchange \"hexanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_exchange \"hexanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _S_lactate_exchange \"(S)-lactate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _S_lactate_exchange \"(S)-lactate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_lactate_exchange \"(R)-lactate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_lactate_exchange \"(R)-lactate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isocaproate_exchange \"isocaproate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isocaproate_exchange \"isocaproate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_exchange \"(R)-3-hydroxybutanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_exchange \"(R)-3-hydroxybutanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction indole_3_acetate_exchange \"indole-3-acetate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction indole_3_acetate_exchange \"indole-3-acetate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_hydroxyphenylacetate_exchange \"4-hydroxyphenylacetate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_hydroxyphenylacetate_exchange \"4-hydroxyphenylacetate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylacetate_exchange \"phenylacetate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylacetate_exchange \"phenylacetate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_imidazoleacetate_exchange \"4-imidazoleacetate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_imidazoleacetate_exchange \"4-imidazoleacetate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_phenylpropanoate_exchange \"3-phenylpropanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_phenylpropanoate_exchange \"3-phenylpropanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_4_hydroxyphenyl_propanoate_exchange \"3-(4-hydroxyphenyl)propanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_4_hydroxyphenyl_propanoate_exchange \"3-(4-hydroxyphenyl)propanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isobutyrate_exchange \"isobutyrate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isobutyrate_exchange \"isobutyrate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isovalerate_exchange \"isovalerate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isovalerate_exchange \"isovalerate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_methylbutanoate_exchange \"2-methylbutanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_methylbutanoate_exchange \"2-methylbutanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H2O_transport \"H2O_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H2O_transport \"H2O_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H2_transport \"H2_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H2_transport \"H2_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CO2_transport \"CO2_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CO2_transport \"CO2_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CO_transport \"CO_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CO_transport \"CO_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction H2S_transport \"H2S_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction H2S_transport \"H2S_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methanethiol_transport \"methanethiol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methanethiol_transport \"methanethiol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spontaneous_no_pathway_RXN_11811 \"spontaneous_no-pathway_RXN-11811\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spontaneous_no_pathway_RXN_11811 \"spontaneous_no-pathway_RXN-11811\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ammonia_transport \"ammonia_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ammonia_transport \"ammonia_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spontaneous_no_pathway_RXN_11811_ex_ \"spontaneous_no-pathway_RXN-11811[ex]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spontaneous_no_pathway_RXN_11811_ex_ \"spontaneous_no-pathway_RXN-11811[ex]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spontaneous_sulfite_dissotiation \"spontaneous_sulfite_dissotiation\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spontaneous_sulfite_dissotiation \"spontaneous_sulfite_dissotiation\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfite_transport \"sulfite_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfite_transport \"sulfite_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spontaneous_sulfite_dissotiation_ex_ \"spontaneous_sulfite_dissotiation[ex]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spontaneous_sulfite_dissotiation_ex_ \"spontaneous_sulfite_dissotiation[ex]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spontaneous_selenide_dissotiation \"spontaneous_selenide_dissotiation\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spontaneous_selenide_dissotiation \"spontaneous_selenide_dissotiation\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction selenide_transport \"selenide_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction selenide_transport \"selenide_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spontaneous_selenide_dissotiation_ex_ \"spontaneous_selenide_dissotiation[ex]\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spontaneous_selenide_dissotiation_ex_ \"spontaneous_selenide_dissotiation[ex]\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanol_transport \"ethanol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanol_transport \"ethanol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanol_transport \"propanol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanol_transport \"propanol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction butanol_transport \"butanol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction butanol_transport \"butanol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isobutanol_transport \"isobutanol_transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isobutanol_transport \"isobutanol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_methylphenol_transport \"4-methylphenol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_methylphenol_transport \"4-methylphenol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetone_transport \"acetone_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetone_transport \"acetone_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aminoaceton_transport \"aminoaceton_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aminoaceton_transport \"aminoaceton_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction urea_transport \"urea_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction urea_transport \"urea_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction THMF_transport \"THMF_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction THMF_transport \"THMF_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_methyl_5_thio_D_ribose_transport \"S-methyl-5-thio-D-ribose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_methyl_5_thio_D_ribose_transport \"S-methyl-5-thio-D-ribose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_methyl_5_thio_alpha_D_ribose_1_phosphate_transport \"S-methyl-5-thio-alpha-D-ribose_1-phosphate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_methyl_5_thio_alpha_D_ribose_1_phosphate_transport \"S-methyl-5-thio-alpha-D-ribose_1-phosphate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_deoxy_D_ribose_transport \"5-deoxy-D-ribose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_deoxy_D_ribose_transport \"5-deoxy-D-ribose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolaldehyde_transport \"glycolaldehyde_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolaldehyde_transport \"glycolaldehyde_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formate_transport \"formate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formate_transport \"formate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetate_transport \"acetate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetate_transport \"acetate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_transport \"propanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_transport \"propanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction n_butanoate_transport \"n-butanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction n_butanoate_transport \"n-butanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_transport \"pentanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_transport \"pentanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_transport \"hexanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_transport \"hexanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_transport \"(R)-3-hydroxybutanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_transport \"(R)-3-hydroxybutanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _S_lactate_transport \"(S)-lactate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _S_lactate_transport \"(S)-lactate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_lactate_transport \"(R)-lactate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_lactate_transport \"(R)-lactate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction indole_3_acetate_transport \"indole-3-acetate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction indole_3_acetate_transport \"indole-3-acetate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_hydroxyphenylacetate_transport \"4-hydroxyphenylacetate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_hydroxyphenylacetate_transport \"4-hydroxyphenylacetate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylacetate_transport \"phenylacetate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylacetate_transport \"phenylacetate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_imidazoleacetate_transport \"4-imidazoleacetate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_imidazoleacetate_transport \"4-imidazoleacetate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_phenylpropanoate_transport \"3-phenylpropanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_phenylpropanoate_transport \"3-phenylpropanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_4_hydroxyphenyl_propanoate_transport \"3-(4-hydroxyphenyl)propanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_4_hydroxyphenyl_propanoate_transport \"3-(4-hydroxyphenyl)propanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isobutyrate_transport \"isobutyrate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isobutyrate_transport \"isobutyrate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isovalerate_transport \"isovalerate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isovalerate_transport \"isovalerate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_methylbutanoate_transport \"2-methylbutanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_methylbutanoate_transport \"2-methylbutanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isocaproate_transport \"isocaproate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isocaproate_transport \"isocaproate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_antiport \"threonine_antiport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_antiport \"threonine_antiport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_transport \"serine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_transport \"serine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_transport \"threonine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_transport \"threonine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction alanine_transport \"alanine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction alanine_transport \"alanine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_transport \"glycine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_transport \"glycine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_transport \"valine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_transport \"valine_transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_transport \"leucine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_transport \"leucine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_transport \"isoleucine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_transport \"isoleucine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_transport \"tryptophan_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_transport \"tryptophan_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_transport \"tyrosine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_transport \"tyrosine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_transport \"phenylalanine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_transport \"phenylalanine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamate_transport \"glutamate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamate_transport \"glutamate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction proline_transport \"proline_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction proline_transport \"proline_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cysteine_transport \"cysteine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cysteine_transport \"cysteine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_transport \"lysine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_transport \"lysine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminopentanoate_transport \"5-aminopentanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminopentanoate_transport \"5-aminopentanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminobutanoate_transport \"2-aminobutanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminobutanoate_transport \"2-aminobutanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction beta_alanine_transport \"beta-alanine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction beta_alanine_transport \"beta-alanine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_pantothenate_transport \"(R)-pantothenate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_pantothenate_transport \"(R)-pantothenate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_3_amino_2_methylpropanoate_transport \"(R)-3-amino-2-methylpropanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_3_amino_2_methylpropanoate_transport \"(R)-3-amino-2-methylpropanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_3_6_3_transport \"arginine_3.6.3.-_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_3_6_3_transport \"arginine_3.6.3.-_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methionine_transport \"methionine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methionine_transport \"methionine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aspartate_transport \"aspartate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aspartate_transport \"aspartate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_transport \"histidine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_transport \"histidine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pyridoxine_transport \"pyridoxine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pyridoxine_transport \"pyridoxine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphate_transport \"phosphate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphate_transport \"phosphate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Ni2_transport \"Ni2+_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Ni2_transport \"Ni2+_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Co2_transport_3_6_3_2_3_6_3_5 \"Co2+_transport_3.6.3.2_3.6.3.5\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Co2_transport_3_6_3_2_3_6_3_5 \"Co2+_transport_3.6.3.2_3.6.3.5\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Mg2_transport_3_6_3_2_ \"Mg2+_transport_3.6.3.2_\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Mg2_transport_3_6_3_2_ \"Mg2+_transport_3.6.3.2_\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Fe3_transport_3_6_3_30 \"Fe3+_transport_3.6.3.30\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Fe3_transport_3_6_3_30 \"Fe3+_transport_3.6.3.30\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction K_transport_3_6_3_12 \"K+_transport_3.6.3.12\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction K_transport_3_6_3_12 \"K+_transport_3.6.3.12\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Ca2_transport_3_6_3_8 \"Ca2+_transport_3.6.3.8\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Ca2_transport_3_6_3_8 \"Ca2+_transport_3.6.3.8\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdate_transport_3_6_3_29 \"molybdate_transport_3.6.3.29\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdate_transport_3_6_3_29 \"molybdate_transport_3.6.3.29\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Fe2_transport \"Fe2+_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Fe2_transport \"Fe2+_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Cd2_transport_3_6_3_3 \"Cd2+_transport_3.6.3.3\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Cd2_transport_3_6_3_3 \"Cd2+_transport_3.6.3.3\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Zn2_transport_3_6_3_5 \"Zn2+_transport_3.6.3.5\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Zn2_transport_3_6_3_5 \"Zn2+_transport_3.6.3.5\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_transport \"biotin_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_transport \"biotin_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Mg2_transport \"Mg2+_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Mg2_transport \"Mg2+_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction PO4_Na_transport \"PO4/Na_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction PO4_Na_transport \"PO4/Na_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction potassium_transport \"potassium_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction potassium_transport \"potassium_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction F_type_ATPase_transport_3_6_3_14_mod_ATPSYN_RXN \"F-type-ATPase_transport_3.6.3.14_mod_ATPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction F_type_ATPase_transport_3_6_3_14_mod_ATPSYN_RXN \"F-type-ATPase_transport_3.6.3.14_mod_ATPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction V_type_ATPase_transport_3_6_3_15_mod_3_6_3_15_RXN \"V-type-ATPase_transport_3.6.3.15_mod_3.6.3.15-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction V_type_ATPase_transport_3_6_3_15_mod_3_6_3_15_RXN \"V-type-ATPase_transport_3.6.3.15_mod_3.6.3.15-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction rnfABCDEG_transport \"rnfABCDEG_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction rnfABCDEG_transport \"rnfABCDEG_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sodium_transport \"sodium_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sodium_transport \"sodium_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction PTS_system_2_7_3_9_2_7_3_9_RXN \"PTS-system_2.7.3.9_2.7.3.9-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction PTS_system_2_7_3_9_2_7_3_9_RXN \"PTS-system_2.7.3.9_2.7.3.9-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glucose_transport_2_7_1_199 \"glucose_transport_2.7.1.199\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glucose_transport_2_7_1_199 \"glucose_transport_2.7.1.199\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_1_5_3_1_9_PGLUCISOM_RXN \"glycolysis-I/1_5.3.1.9_PGLUCISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_1_5_3_1_9_PGLUCISOM_RXN \"glycolysis-I/1_5.3.1.9_PGLUCISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_2b_3_1_3_11_F16BDEPHOS_RXN \"glycolysis-I/2b_3.1.3.11_F16BDEPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_2b_3_1_3_11_F16BDEPHOS_RXN \"glycolysis-I/2b_3.1.3.11_F16BDEPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_2a_2_7_1_11_6PFRUCTPHOS_RXN \"glycolysis-I/2a_2.7.1.11_6PFRUCTPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_2a_2_7_1_11_6PFRUCTPHOS_RXN \"glycolysis-I/2a_2.7.1.11_6PFRUCTPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_3_4_1_2_13_F16ALDOLASE_RXN \"glycolysis-I/3_4.1.2.13_F16ALDOLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_3_4_1_2_13_F16ALDOLASE_RXN \"glycolysis-I/3_4.1.2.13_F16ALDOLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_p4_5_3_1_1_TRIOSEPISOMERIZATION_RXN \"glycolysis-I/p4_5.3.1.1_TRIOSEPISOMERIZATION-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_p4_5_3_1_1_TRIOSEPISOMERIZATION_RXN \"glycolysis-I/p4_5.3.1.1_TRIOSEPISOMERIZATION-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_4_1_2_1_12_GAPDHSYNEC_RXN_WOP \"glycolysis-I/4_1.2.1.12_GAPDHSYNEC-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_4_1_2_1_12_GAPDHSYNEC_RXN_WOP \"glycolysis-I/4_1.2.1.12_GAPDHSYNEC-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_5_2_7_2_3_PHOSGLYPHOS_RXN \"glycolysis-I/5_2.7.2.3_PHOSGLYPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_5_2_7_2_3_PHOSGLYPHOS_RXN \"glycolysis-I/5_2.7.2.3_PHOSGLYPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_6_5_4_2_12_3PGAREARR_RXN \"glycolysis-I/6_5.4.2.12_3PGAREARR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_6_5_4_2_12_3PGAREARR_RXN \"glycolysis-I/6_5.4.2.12_3PGAREARR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_7_4_2_1_11_2PGADEHYDRAT_RXN \"glycolysis-I/7_4.2.1.11_2PGADEHYDRAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_7_4_2_1_11_2PGADEHYDRAT_RXN \"glycolysis-I/7_4.2.1.11_2PGADEHYDRAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_I_8_2_7_1_40_PEPDEPHOS_RXN \"glycolysis-I/8_2.7.1.40_PEPDEPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_I_8_2_7_1_40_PEPDEPHOS_RXN \"glycolysis-I/8_2.7.1.40_PEPDEPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_IV_2c_2_7_1_90_2_7_1_90_RXN \"glycolysis-IV/2c_2.7.1.90_2.7.1.90-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_IV_2c_2_7_1_90_2_7_1_90_RXN \"glycolysis-IV/2c_2.7.1.90_2.7.1.90-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_IV_4b_1_2_1_9_1_2_1_9_RXN \"glycolysis-IV/4b_1.2.1.9_1.2.1.9-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_IV_4b_1_2_1_9_1_2_1_9_RXN \"glycolysis-IV/4b_1.2.1.9_1.2.1.9-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycolysis_C_8b_2_7_9_1_PYRUVATEORTHOPHOSPHATE_DIKINASE_RXN \"glycolysis-C/8b_2.7.9.1_PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycolysis_C_8b_2_7_9_1_PYRUVATEORTHOPHOSPHATE_DIKINASE_RXN \"glycolysis-C/8b_2.7.9.1_PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylglyoxal_degradation_I_p1_4_2_3_3_METHGLYSYN_RXN \"methylglyoxal-degradation-I/p1_4.2.3.3_METHGLYSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylglyoxal_degradation_I_p1_4_2_3_3_METHGLYSYN_RXN \"methylglyoxal-degradation-I/p1_4.2.3.3_METHGLYSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylglyoxal_degradation_I_1_4_4_1_5_GLYOXI_RXN \"methylglyoxal-degradation-I/1_4.4.1.5_GLYOXI-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylglyoxal_degradation_I_1_4_4_1_5_GLYOXI_RXN \"methylglyoxal-degradation-I/1_4.4.1.5_GLYOXI-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylglyoxal_degradation_I_2_3_1_2_6_GLYOXII_RXN \"methylglyoxal-degradation-I/2_3.1.2.6_GLYOXII-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylglyoxal_degradation_I_2_3_1_2_6_GLYOXII_RXN \"methylglyoxal-degradation-I/2_3.1.2.6_GLYOXII-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_1_moddir_6_3_4_3_FORMATETHFLIG_RXN \"Wood-Ljungdahl/1_moddir_6.3.4.3_FORMATETHFLIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_1_moddir_6_3_4_3_FORMATETHFLIG_RXN \"Wood-Ljungdahl/1_moddir_6.3.4.3_FORMATETHFLIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_2_moddir_3_5_4_9_METHENYLTHFCYCLOHYDRO_RXN \"Wood-Ljungdahl/2_moddir_3.5.4.9_METHENYLTHFCYCLOHYDRO-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_2_moddir_3_5_4_9_METHENYLTHFCYCLOHYDRO_RXN \"Wood-Ljungdahl/2_moddir_3.5.4.9_METHENYLTHFCYCLOHYDRO-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_3_1_5_1_5_METHYLENETHFDEHYDROG_NADP_RXN \"Wood-Ljungdahl/3_1.5.1.5_METHYLENETHFDEHYDROG-NADP-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_3_1_5_1_5_METHYLENETHFDEHYDROG_NADP_RXN \"Wood-Ljungdahl/3_1.5.1.5_METHYLENETHFDEHYDROG-NADP-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_4_manual_1_5_HD_MetFV \"Wood-Ljungdahl/4_manual_1.5.-_HD-MetFV\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_4_manual_1_5_HD_MetFV \"Wood-Ljungdahl/4_manual_1.5.-_HD-MetFV\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_5_2_1_1_258_METHCOCLTH_RXN \"Wood-Ljungdahl/5_2.1.1.258_METHCOCLTH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_5_2_1_1_258_METHCOCLTH_RXN \"Wood-Ljungdahl/5_2.1.1.258_METHCOCLTH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_p6_moddir_1_2_7_4_1_2_7_4_RXN \"Wood-Ljungdahl/p6_moddir_1.2.7.4_1.2.7.4-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_p6_moddir_1_2_7_4_1_2_7_4_RXN \"Wood-Ljungdahl/p6_moddir_1.2.7.4_1.2.7.4-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Wood_Ljungdahl_6_2_3_1_169_ACETYLSYNCLTH_RXN \"Wood-Ljungdahl/6_2.3.1.169_ACETYLSYNCLTH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Wood_Ljungdahl_6_2_3_1_169_ACETYLSYNCLTH_RXN \"Wood-Ljungdahl/6_2.3.1.169_ACETYLSYNCLTH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CO_drgradation_C_manual_HD_CO_DEHYD \"CO-drgradation-C_manual_HD-CO-DEHYD\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CO_drgradation_C_manual_HD_CO_DEHYD \"CO-drgradation-C_manual_HD-CO-DEHYD\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dihydrogen_production_A_moddir_1_12_7_2_HYDROG_RXN \"dihydrogen-production-A_moddir_1.12.7.2_HYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dihydrogen_production_A_moddir_1_12_7_2_HYDROG_RXN \"dihydrogen-production-A_moddir_1.12.7.2_HYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dihydrogen_production_B_manual_1_12_1_HD_Bifur_Hyd \"dihydrogen-production-B_manual_1.12.1.-_HD-Bifur-Hyd\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dihydrogen_production_B_manual_1_12_1_HD_Bifur_Hyd \"dihydrogen-production-B_manual_1.12.1.-_HD-Bifur-Hyd\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dihydrogen_production_C_manual_1_1_99_33_1_12_1_3_HD_Form_Hyd \"dihydrogen-production-C_manual_1.1.99.33_1.12.1.3_HD-Form-Hyd\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dihydrogen_production_C_manual_1_1_99_33_1_12_1_3_HD_Form_Hyd \"dihydrogen-production-C_manual_1.1.99.33_1.12.1.3_HD-Form-Hyd\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppp_non_ox_1a_5_3_1_6_RIB5PISOM_RXN \"ppp-non-ox/1a_5.3.1.6_RIB5PISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppp_non_ox_1a_5_3_1_6_RIB5PISOM_RXN \"ppp-non-ox/1a_5.3.1.6_RIB5PISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppp_non_ox_1b_5_1_3_1_RIBULP3EPIM_RXN \"ppp-non-ox/1b_5.1.3.1_RIBULP3EPIM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppp_non_ox_1b_5_1_3_1_RIBULP3EPIM_RXN \"ppp-non-ox/1b_5.1.3.1_RIBULP3EPIM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppp_non_ox_2_2_2_1_1_R06863 \"ppp-non-ox/2_2.2.1.1_R06863\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppp_non_ox_2_2_2_1_1_R06863 \"ppp-non-ox/2_2.2.1.1_R06863\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppp_non_ox_3_2_2_1_1_R06861 \"ppp-non-ox/3_2.2.1.1_R06861\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppp_non_ox_3_2_2_1_1_R06861 \"ppp-non-ox/3_2.2.1.1_R06861\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppp_non_ox_4_2_2_1_2_TRANSALDOL_RXN \"ppp-non-ox/4_2.2.1.2_TRANSALDOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppp_non_ox_4_2_2_1_2_TRANSALDOL_RXN \"ppp-non-ox/4_2.2.1.2_TRANSALDOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppp_non_ox_5_2_2_1_1_2TRANSKETO_RXN \"ppp-non-ox/5_2.2.1.1_2TRANSKETO-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppp_non_ox_5_2_2_1_1_2TRANSKETO_RXN \"ppp-non-ox/5_2.2.1.1_2TRANSKETO-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction PRPP_biosynthesis_I_2_7_6_1_PRPPSYN_RXN \"PRPP-biosynthesis-I_2.7.6.1_PRPPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction PRPP_biosynthesis_I_2_7_6_1_PRPPSYN_RXN \"PRPP-biosynthesis-I_2.7.6.1_PRPPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sedoheptulose_bisphosphate_bypass_2_7_1_11_RXN0_6541 \"sedoheptulose-bisphosphate-bypass_2.7.1.11_RXN0-6541\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sedoheptulose_bisphosphate_bypass_2_7_1_11_RXN0_6541 \"sedoheptulose-bisphosphate-bypass_2.7.1.11_RXN0-6541\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sedoheptulose_bisphosphate_bypass_4_1_2_13_SEDOBISALDOL_RXN \"sedoheptulose-bisphosphate-bypass_4.1.2.13_SEDOBISALDOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sedoheptulose_bisphosphate_bypass_4_1_2_13_SEDOBISALDOL_RXN \"sedoheptulose-bisphosphate-bypass_4.1.2.13_SEDOBISALDOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction Calvin_Benson_Bassham_cycle_3_1_3_11_SEDOHEPTULOSE_BISPHOSPHATASE_RXN \"Calvin-Benson-Bassham-cycle_3.1.3.11_SEDOHEPTULOSE-BISPHOSPHATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction Calvin_Benson_Bassham_cycle_3_1_3_11_SEDOHEPTULOSE_BISPHOSPHATASE_RXN \"Calvin-Benson-Bassham-cycle_3.1.3.11_SEDOHEPTULOSE-BISPHOSPHATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_1_4_2_1_1_CARBODEHYDRAT_RXN \"TCA-cycle/1_4.2.1.1_CARBODEHYDRAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_1_4_2_1_1_CARBODEHYDRAT_RXN \"TCA-cycle/1_4.2.1.1_CARBODEHYDRAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_2_6_4_1_1_PYRUVATE_CARBOXYLASE_RXN \"TCA-cycle/2_6.4.1.1_PYRUVATE-CARBOXYLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_2_6_4_1_1_PYRUVATE_CARBOXYLASE_RXN \"TCA-cycle/2_6.4.1.1_PYRUVATE-CARBOXYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_3_2_3_3_3_CITSYN_RXN \"TCA-cycle/3_2.3.3.3_CITSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_3_2_3_3_3_CITSYN_RXN \"TCA-cycle/3_2.3.3.3_CITSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_4_4_2_1_3_ACONITATEDEHYDR_RXN \"TCA-cycle/4_4.2.1.3_ACONITATEDEHYDR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_4_4_2_1_3_ACONITATEDEHYDR_RXN \"TCA-cycle/4_4.2.1.3_ACONITATEDEHYDR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_5_4_2_1_3_ACONITATEHYDR_RXN \"TCA-cycle/5_4.2.1.3_ACONITATEHYDR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_5_4_2_1_3_ACONITATEHYDR_RXN \"TCA-cycle/5_4.2.1.3_ACONITATEHYDR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_6_1_1_1_41_ISOCITRATE_DEHYDROGENASE_NAD_RXN \"TCA-cycle/6_1.1.1.41_ISOCITRATE-DEHYDROGENASE-NAD+-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_6_1_1_1_41_ISOCITRATE_DEHYDROGENASE_NAD_RXN \"TCA-cycle/6_1.1.1.41_ISOCITRATE-DEHYDROGENASE-NAD+-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_7_4_2_1_2_FUMHYDR_RXN \"TCA-cycle/7_4.2.1.2_FUMHYDR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_7_4_2_1_2_FUMHYDR_RXN \"TCA-cycle/7_4.2.1.2_FUMHYDR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_8_1_1_1_38_1_1_1_39_RXN \"TCA-cycle/8_1.1.1.38_1.1.1.39-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_8_1_1_1_38_1_1_1_39_RXN \"TCA-cycle/8_1.1.1.38_1.1.1.39-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction TCA_cycle_9_1_1_1_38_OXALODECARB_RXN \"TCA-cycle/9_1.1.1.38_OXALODECARB-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction TCA_cycle_9_1_1_1_38_OXALODECARB_RXN \"TCA-cycle/9_1.1.1.38_OXALODECARB-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADP_reduction_manual_HD_NfnAB \"NADP-reduction_manual_HD-NfnAB\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADP_reduction_manual_HD_NfnAB \"NADP-reduction_manual_HD-NfnAB\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_1_6_4_1_2_ACETYL_COA_CARBOXYLTRANSFER_RXN \"fatty-acid-I/1_6.4.1.2_ACETYL-COA-CARBOXYLTRANSFER-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_1_6_4_1_2_ACETYL_COA_CARBOXYLTRANSFER_RXN \"fatty-acid-I/1_6.4.1.2_ACETYL-COA-CARBOXYLTRANSFER-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_2_2_3_1_39_MALONYL_COA_ACP_TRANSACYL_RXN \"fatty-acid-I/2_2.3.1.39_MALONYL-COA-ACP-TRANSACYL-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_2_2_3_1_39_MALONYL_COA_ACP_TRANSACYL_RXN \"fatty-acid-I/2_2.3.1.39_MALONYL-COA-ACP-TRANSACYL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3a_manual_HD_acet \"fatty-acid-I/3a_manual_HD-acet\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3a_manual_HD_acet \"fatty-acid-I/3a_manual_HD-acet\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3b_manual_HD_prop \"fatty-acid-I/3b_manual_HD-prop\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3b_manual_HD_prop \"fatty-acid-I/3b_manual_HD-prop\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3c_manual_HD_buta \"fatty-acid-I/3c_manual_HD-buta\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3c_manual_HD_buta \"fatty-acid-I/3c_manual_HD-buta\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3d_manual_HD_ibut \"fatty-acid-I/3d_manual_HD-ibut\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3d_manual_HD_ibut \"fatty-acid-I/3d_manual_HD-ibut\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3e_manual_HD_vale \"fatty-acid-I/3e_manual_HD-vale\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3e_manual_HD_vale \"fatty-acid-I/3e_manual_HD-vale\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3f_manual_HD_ival \"fatty-acid-I/3f_manual_HD-ival\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3f_manual_HD_ival \"fatty-acid-I/3f_manual_HD-ival\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3g_manual_HD_aval \"fatty-acid-I/3g_manual_HD-aval\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3g_manual_HD_aval \"fatty-acid-I/3g_manual_HD-aval\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3h_manual_HD_icap \"fatty-acid-I/3h_manual_HD-icap\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3h_manual_HD_icap \"fatty-acid-I/3h_manual_HD-icap\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_3i_manual_HD_hexa \"fatty-acid-I/3i_manual_HD-hexa\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_3i_manual_HD_hexa \"fatty-acid-I/3i_manual_HD-hexa\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fatty_acid_I_4_manual_2_3_1_n2_HD_2_3_1_n2 \"fatty-acid-I/4_manual_2.3.1.n2_HD-2.3.1.n2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fatty_acid_I_4_manual_2_3_1_n2_HD_2_3_1_n2 \"fatty-acid-I/4_manual_2.3.1.n2_HD-2.3.1.n2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_1_modnames_1_1_1_94_1_1_1_261_RXN \"CDP-diacylglycerol-biosyn-I/1_modnames_1.1.1.94_1.1.1.261-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_1_modnames_1_1_1_94_1_1_1_261_RXN \"CDP-diacylglycerol-biosyn-I/1_modnames_1.1.1.94_1.1.1.261-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_2_manual_2_3_1_n3_HD_2_3_1_n3 \"CDP-diacylglycerol-biosyn-I/2_manual_2.3.1.n3_HD-2.3.1.n3\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_2_manual_2_3_1_n3_HD_2_3_1_n3 \"CDP-diacylglycerol-biosyn-I/2_manual_2.3.1.n3_HD-2.3.1.n3\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_3_2_3_1_51_BS321039 \"CDP-diacylglycerol-biosyn-I/3_2.3.1.51_BS321039\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_3_2_3_1_51_BS321039 \"CDP-diacylglycerol-biosyn-I/3_2.3.1.51_BS321039\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_4_2_7_7_41_CDPDIGLYSYN_RXN \"CDP-diacylglycerol-biosyn-I/4_2.7.7.41_CDPDIGLYSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CDP_diacylglycerol_biosyn_I_4_2_7_7_41_CDPDIGLYSYN_RXN \"CDP-diacylglycerol-biosyn-I/4_2.7.7.41_CDPDIGLYSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cardiolipin_biosyn_I_1_2_7_8_5_PHOSPHAGLYPSYN_RXN \"cardiolipin-biosyn-I/1_2.7.8.5_PHOSPHAGLYPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cardiolipin_biosyn_I_1_2_7_8_5_PHOSPHAGLYPSYN_RXN \"cardiolipin-biosyn-I/1_2.7.8.5_PHOSPHAGLYPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cardiolipin_biosyn_I_2_3_1_3_27_PGPPHOSPHA_RXN \"cardiolipin-biosyn-I/2_3.1.3.27_PGPPHOSPHA-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cardiolipin_biosyn_I_2_3_1_3_27_PGPPHOSPHA_RXN \"cardiolipin-biosyn-I/2_3.1.3.27_PGPPHOSPHA-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cardiolipin_biosyn_I_3_2_7_8_CARDIOLIPSYN_RXN \"cardiolipin-biosyn-I/3_2.7.8.-_CARDIOLIPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cardiolipin_biosyn_I_3_2_7_8_CARDIOLIPSYN_RXN \"cardiolipin-biosyn-I/3_2.7.8.-_CARDIOLIPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycerol_degradation_I_2_7_1_30_GLYCEROL_KIN_RXN \"glycerol-degradation-I_2.7.1.30_GLYCEROL-KIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycerol_degradation_I_2_7_1_30_GLYCEROL_KIN_RXN \"glycerol-degradation-I_2.7.1.30_GLYCEROL-KIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_1_2_6_1_16_L_GLN_FRUCT_6_P_AMINOTRANS_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/1_2.6.1.16_L-GLN-FRUCT-6-P-AMINOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_1_2_6_1_16_L_GLN_FRUCT_6_P_AMINOTRANS_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/1_2.6.1.16_L-GLN-FRUCT-6-P-AMINOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_2_5_4_2_10_5_4_2_10_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/2_5.4.2.10_5.4.2.10-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_2_5_4_2_10_5_4_2_10_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/2_5.4.2.10_5.4.2.10-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_3_2_3_1_157_2_3_1_157_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/3_2.3.1.157_2.3.1.157-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_3_2_3_1_157_2_3_1_157_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/3_2.3.1.157_2.3.1.157-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_4_2_7_7_23_NAG1P_URIDYLTRANS_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/4_2.7.7.23_NAG1P-URIDYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_glucosamine_biosyn_I_4_2_7_7_23_NAG1P_URIDYLTRANS_RXN \"UDP-N-acetyl-D-glucosamine-biosyn-I/4_2.7.7.23_NAG1P-URIDYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_galactosamine_biosyn_I_5_1_3_2_UDP_N_ACETYLGLUCOSAMINE_4_EPIMERASE_RXN \"UDP-N-acetyl-D-galactosamine-biosyn-I_5.1.3.2_UDP-N-ACETYLGLUCOSAMINE-4-EPIMERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_N_acetyl_D_galactosamine_biosyn_I_5_1_3_2_UDP_N_ACETYLGLUCOSAMINE_4_EPIMERASE_RXN \"UDP-N-acetyl-D-galactosamine-biosyn-I_5.1.3.2_UDP-N-ACETYLGLUCOSAMINE-4-EPIMERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GDP_mannose_biosyn_1_5_3_1_8_MANNPISOM_RXN \"GDP-mannose-biosyn/1_5.3.1.8_MANNPISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GDP_mannose_biosyn_1_5_3_1_8_MANNPISOM_RXN \"GDP-mannose-biosyn/1_5.3.1.8_MANNPISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GDP_mannose_biosyn_2_5_4_2_8_PHOSMANMUT_RXN \"GDP-mannose-biosyn/2_5.4.2.8_PHOSMANMUT-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GDP_mannose_biosyn_2_5_4_2_8_PHOSMANMUT_RXN \"GDP-mannose-biosyn/2_5.4.2.8_PHOSMANMUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GDP_mannose_biosyn_3_2_7_7_13_2_7_7_13_RXN \"GDP-mannose-biosyn/3_2.7.7.13_2.7.7.13-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GDP_mannose_biosyn_3_2_7_7_13_2_7_7_13_RXN \"GDP-mannose-biosyn/3_2.7.7.13_2.7.7.13-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_glucose_biosyn_1_5_4_2_2_5_4_2_8_PHOSPHOGLUCMUT_RXN \"UDP-glucose-biosyn/1_5.4.2.2_5.4.2.8_PHOSPHOGLUCMUT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_glucose_biosyn_1_5_4_2_2_5_4_2_8_PHOSPHOGLUCMUT_RXN \"UDP-glucose-biosyn/1_5.4.2.2_5.4.2.8_PHOSPHOGLUCMUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UDP_glucose_biosyn_2_2_7_7_9_GLUC1PURIDYLTRANS_RXN \"UDP-glucose-biosyn/2_2.7.7.9_GLUC1PURIDYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UDP_glucose_biosyn_2_2_7_7_9_GLUC1PURIDYLTRANS_RXN \"UDP-glucose-biosyn/2_2.7.7.9_GLUC1PURIDYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_1_2_5_1_7_UDPNACETYLGLUCOSAMENOLPYRTRANS_RXN \"peptidoglycan-biosyn-I/1_2.5.1.7_UDPNACETYLGLUCOSAMENOLPYRTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_1_2_5_1_7_UDPNACETYLGLUCOSAMENOLPYRTRANS_RXN \"peptidoglycan-biosyn-I/1_2.5.1.7_UDPNACETYLGLUCOSAMENOLPYRTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_2_1_3_1_98_UDPNACETYLMURAMATEDEHYDROG_RXN \"peptidoglycan-biosyn-I/2_1.3.1.98_UDPNACETYLMURAMATEDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_2_1_3_1_98_UDPNACETYLMURAMATEDEHYDROG_RXN \"peptidoglycan-biosyn-I/2_1.3.1.98_UDPNACETYLMURAMATEDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_3_6_3_2_8_UDP_NACMUR_ALA_LIG_RXN \"peptidoglycan-biosyn-I/3_6.3.2.8_UDP-NACMUR-ALA-LIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_3_6_3_2_8_UDP_NACMUR_ALA_LIG_RXN \"peptidoglycan-biosyn-I/3_6.3.2.8_UDP-NACMUR-ALA-LIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_glutamate_5_1_1_3_GLUTRACE_RXN \"D-glutamate_5.1.1.3_GLUTRACE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_glutamate_5_1_1_3_GLUTRACE_RXN \"D-glutamate_5.1.1.3_GLUTRACE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_4_6_3_2_9_UDP_NACMURALA_GLU_LIG_RXN \"peptidoglycan-biosyn-I/4_6.3.2.9_UDP-NACMURALA-GLU-LIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_4_6_3_2_9_UDP_NACMURALA_GLU_LIG_RXN \"peptidoglycan-biosyn-I/4_6.3.2.9_UDP-NACMURALA-GLU-LIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_5_6_3_2_13_UDP_NACMURALGLDAPLIG_RXN \"peptidoglycan-biosyn-I/5_6.3.2.13_UDP-NACMURALGLDAPLIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_5_6_3_2_13_UDP_NACMURALGLDAPLIG_RXN \"peptidoglycan-biosyn-I/5_6.3.2.13_UDP-NACMURALGLDAPLIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_alanyl_D_alanine_1_5_1_1_1_ALARACECAT_RXN \"D-alanyl-D-alanine/1_5.1.1.1_ALARACECAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_alanyl_D_alanine_1_5_1_1_1_ALARACECAT_RXN \"D-alanyl-D-alanine/1_5.1.1.1_ALARACECAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_alanyl_D_alanine_2_6_3_2_4_DALADALALIG_RXN \"D-alanyl-D-alanine/2_6.3.2.4_DALADALALIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_alanyl_D_alanine_2_6_3_2_4_DALADALALIG_RXN \"D-alanyl-D-alanine/2_6.3.2.4_DALADALALIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_6_6_3_2_10_UDP_NACMURALGLDAPAALIG_RXN \"peptidoglycan-biosyn-I/6_6.3.2.10_UDP-NACMURALGLDAPAALIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_6_6_3_2_10_UDP_NACMURALGLDAPAALIG_RXN \"peptidoglycan-biosyn-I/6_6.3.2.10_UDP-NACMURALGLDAPAALIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_p7_3_6_1_27_UNDECAPRENYL_DIPHOSPHATASE_RXN \"peptidoglycan-biosyn-I/p7_3.6.1.27_UNDECAPRENYL-DIPHOSPHATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_p7_3_6_1_27_UNDECAPRENYL_DIPHOSPHATASE_RXN \"peptidoglycan-biosyn-I/p7_3.6.1.27_UNDECAPRENYL-DIPHOSPHATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_7_2_7_8_13_PHOSNACMURPENTATRANS_RXN \"peptidoglycan-biosyn-I/7_2.7.8.13_PHOSNACMURPENTATRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_7_2_7_8_13_PHOSNACMURPENTATRANS_RXN \"peptidoglycan-biosyn-I/7_2.7.8.13_PHOSNACMURPENTATRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_8_2_4_1_227_NACGLCTRANS_RXN \"peptidoglycan-biosyn-I/8_2.4.1.227_NACGLCTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_8_2_4_1_227_NACGLCTRANS_RXN \"peptidoglycan-biosyn-I/8_2.4.1.227_NACGLCTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_10_manual_3_4_16_4_3_5_1_104_HD_murein_mods \"peptidoglycan-biosyn-I/10_manual_3.4.16.4_3.5.1.104_HD-murein-mods\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_biosyn_I_10_manual_3_4_16_4_3_5_1_104_HD_murein_mods \"peptidoglycan-biosyn-I/10_manual_3.4.16.4_3.5.1.104_HD-murein-mods\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_degr_1_manual_3_5_1_28_3_2_1_96_3_5_1_104_HD_appABCF \"peptidoglycan-degr/1_manual_3.5.1.28_3.2.1.96_3.5.1.104_HD-appABCF\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_degr_1_manual_3_5_1_28_3_2_1_96_3_5_1_104_HD_appABCF \"peptidoglycan-degr/1_manual_3.5.1.28_3.2.1.96_3.5.1.104_HD-appABCF\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_degr_2a_2_7_1_170_RXN0_4621 \"peptidoglycan-degr/2a_2.7.1.170_RXN0-4621\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_degr_2a_2_7_1_170_RXN0_4621 \"peptidoglycan-degr/2a_2.7.1.170_RXN0-4621\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_degr_2b_manual_2_7_1_59_HD_2_7_1_59 \"peptidoglycan-degr/2b_manual_2.7.1.59_HD-2.7.1.59\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_degr_2b_manual_2_7_1_59_HD_2_7_1_59 \"peptidoglycan-degr/2b_manual_2.7.1.59_HD-2.7.1.59\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_degr_3_4_2_1_126_RXN0_4641 \"peptidoglycan-degr/3_4.2.1.126_RXN0-4641\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_degr_3_4_2_1_126_RXN0_4641 \"peptidoglycan-degr/3_4.2.1.126_RXN0-4641\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction peptidoglycan_degr_4_2_7_1_59_N_ACETYLGLUCOSAMINE_KINASE_RXN \"peptidoglycan-degr/4_2.7.1.59_N-ACETYLGLUCOSAMINE-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction peptidoglycan_degr_4_2_7_1_59_N_ACETYLGLUCOSAMINE_KINASE_RXN \"peptidoglycan-degr/4_2.7.1.59_N-ACETYLGLUCOSAMINE-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylglucosamine_degr_I_1_3_5_1_25_NAG6PDEACET_RXN \"N-acetylglucosamine-degr-I/1_3.5.1.25_NAG6PDEACET-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylglucosamine_degr_I_1_3_5_1_25_NAG6PDEACET_RXN \"N-acetylglucosamine-degr-I/1_3.5.1.25_NAG6PDEACET-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylglucosamine_degr_I_2_3_5_99_6_BS363214 \"N-acetylglucosamine-degr-I/2_3.5.99.6_BS363214\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylglucosamine_degr_I_2_3_5_99_6_BS363214 \"N-acetylglucosamine-degr-I/2_3.5.99.6_BS363214\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction muropeptide_degr_1_3_4_17_13_RXN0_5227 \"muropeptide-degr/1_3.4.17.13_RXN0-5227\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction muropeptide_degr_1_3_4_17_13_RXN0_5227 \"muropeptide-degr/1_3.4.17.13_RXN0-5227\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction muropeptide_degr_2_RXN0_961 \"muropeptide-degr/2_RXN0-961\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction muropeptide_degr_2_RXN0_961 \"muropeptide-degr/2_RXN0-961\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction muropeptide_degr_3_5_1_1_20_RXN0_5228 \"muropeptide-degr/3_5.1.1.20_RXN0-5228\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction muropeptide_degr_3_5_1_1_20_RXN0_5228 \"muropeptide-degr/3_5.1.1.20_RXN0-5228\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction muropeptide_degr_4_3_4_13_18_RXN0_6981 \"muropeptide-degr/4_3.4.13.18_RXN0-6981\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction muropeptide_degr_4_3_4_13_18_RXN0_6981 \"muropeptide-degr/4_3.4.13.18_RXN0-6981\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_CWG_biosyn_1a_3_1_3_4_PHOSPHATIDATE_PHOSPHATASE_RXN \"L-CWG-biosyn/1a_3.1.3.4_PHOSPHATIDATE-PHOSPHATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_CWG_biosyn_1a_3_1_3_4_PHOSPHATIDATE_PHOSPHATASE_RXN \"L-CWG-biosyn/1a_3.1.3.4_PHOSPHATIDATE-PHOSPHATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_CWG_biosyn_1b_2_7_1_107_DIACYLGLYKIN_RXN \"L-CWG-biosyn/1b_2.7.1.107_DIACYLGLYKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_CWG_biosyn_1b_2_7_1_107_DIACYLGLYKIN_RXN \"L-CWG-biosyn/1b_2.7.1.107_DIACYLGLYKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_CWG_biosyn_2_2_4_1_315_2_4_1_337_2_4_1_157_RXN \"L-CWG-biosyn/2_2.4.1.315_2.4.1.337_2.4.1.157-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_CWG_biosyn_2_2_4_1_315_2_4_1_337_2_4_1_157_RXN \"L-CWG-biosyn/2_2.4.1.315_2.4.1.337_2.4.1.157-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_CWG_biosyn_3_2_4_1_315_RXN_15117 \"L-CWG-biosyn/3_2.4.1.315_RXN-15117\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_CWG_biosyn_3_2_4_1_315_RXN_15117 \"L-CWG-biosyn/3_2.4.1.315_RXN-15117\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_CWG_biosyn_4_2_4_1_315_RXN_15118 \"L-CWG-biosyn/4_2.4.1.315_RXN-15118\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_CWG_biosyn_4_2_4_1_315_RXN_15118 \"L-CWG-biosyn/4_2.4.1.315_RXN-15118\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flagellin_modification_1_manual_2_4_HD_fla1 \"flagellin-modification/1_manual_2.4.-_HD-fla1\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flagellin_modification_1_manual_2_4_HD_fla1 \"flagellin-modification/1_manual_2.4.-_HD-fla1\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flagellin_modification_p1_manual_3_1_3_HD_fla2 \"flagellin-modification/p1_manual_3.1.3.-_HD-fla2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flagellin_modification_p1_manual_3_1_3_HD_fla2 \"flagellin-modification/p1_manual_3.1.3.-_HD-fla2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flagellin_modification_2_manual_3_1_4_HD_fla3 \"flagellin-modification/2_manual_3.1.4.-_HD-fla3\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flagellin_modification_2_manual_3_1_4_HD_fla3 \"flagellin-modification/2_manual_3.1.4.-_HD-fla3\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flagellin_modification_3_manual_2_1_1_HD_fla4 \"flagellin-modification/3_manual_2.1.1.-_HD-fla4\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flagellin_modification_3_manual_2_1_1_HD_fla4 \"flagellin-modification/3_manual_2.1.1.-_HD-fla4\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_biosyn_I_1_2_7_7_27_GLUC1PADENYLTRANS_RXN \"glycogen-biosyn-I/1_2.7.7.27_GLUC1PADENYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_biosyn_I_1_2_7_7_27_GLUC1PADENYLTRANS_RXN \"glycogen-biosyn-I/1_2.7.7.27_GLUC1PADENYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_biosyn_I_2_manual_2_4_1_21_HD_2_4_1_21 \"glycogen-biosyn-I/2_manual_2.4.1.21_HD-2.4.1.21\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_biosyn_I_2_manual_2_4_1_21_HD_2_4_1_21 \"glycogen-biosyn-I/2_manual_2.4.1.21_HD-2.4.1.21\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_biosyn_I_3_2_4_1_18_3_2_1_54_3_2_1_41_3_2_1_1_GLYCOGEN_BRANCH_RXN \"glycogen-biosyn-I/3_2.4.1.18_3.2.1.54_3.2.1.41_3.2.1.1_GLYCOGEN-BRANCH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_biosyn_I_3_2_4_1_18_3_2_1_54_3_2_1_41_3_2_1_1_GLYCOGEN_BRANCH_RXN \"glycogen-biosyn-I/3_2.4.1.18_3.2.1.54_3.2.1.41_3.2.1.1_GLYCOGEN-BRANCH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_degr_manual_3_2_1_54_3_2_1_41_3_2_1_1_HD_Glycogen \"glycogen-degr_manual_3.2.1.54_3.2.1.41_3.2.1.1_HD-Glycogen\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_degr_manual_3_2_1_54_3_2_1_41_3_2_1_1_HD_Glycogen \"glycogen-degr_manual_3.2.1.54_3.2.1.41_3.2.1.1_HD-Glycogen\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_degr_3_2_1_133_3_2_1_1_RXN_12188 \"glycogen-degr_3.2.1.133_3.2.1.1_RXN-12188\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_degr_3_2_1_133_3_2_1_1_RXN_12188 \"glycogen-degr_3.2.1.133_3.2.1.1_RXN-12188\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_degr_2_4_1_64_2_4_1_8_ALPHAALPHA_TREHALOSE_PHOSPHORYLASE_RXN \"glycogen-degr_2.4.1.64_2.4.1.8_ALPHAALPHA-TREHALOSE-PHOSPHORYLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_degr_2_4_1_64_2_4_1_8_ALPHAALPHA_TREHALOSE_PHOSPHORYLASE_RXN \"glycogen-degr_2.4.1.64_2.4.1.8_ALPHAALPHA-TREHALOSE-PHOSPHORYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycogen_degr_3_2_1_10_RXN_15910 \"glycogen-degr_3.2.1.10_RXN-15910\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycogen_degr_3_2_1_10_RXN_15910 \"glycogen-degr_3.2.1.10_RXN-15910\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glucose_degr_1_2_7_1_2_GLUCOKIN_RXN \"glucose-degr/1_2.7.1.2_GLUCOKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glucose_degr_1_2_7_1_2_GLUCOKIN_RXN \"glucose-degr/1_2.7.1.2_GLUCOKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glucose_degr_2_5_4_2_6_BETA_PHOSPHOGLUCOMUTASE_RXN \"glucose-degr/2_5.4.2.6_BETA-PHOSPHOGLUCOMUTASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glucose_degr_2_5_4_2_6_BETA_PHOSPHOGLUCOMUTASE_RXN \"glucose-degr/2_5.4.2.6_BETA-PHOSPHOGLUCOMUTASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glucose_degr_3_5_3_1_9_R03321 \"glucose-degr/3_5.3.1.9_R03321\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glucose_degr_3_5_3_1_9_R03321 \"glucose-degr/3_5.3.1.9_R03321\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_1_manual_2_4_1_12_HD_2_4_1_12 \"exopolysaccharide-biosynthesis/1_manual_2.4.1.12_HD-2.4.1.12\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_1_manual_2_4_1_12_HD_2_4_1_12 \"exopolysaccharide-biosynthesis/1_manual_2.4.1.12_HD-2.4.1.12\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_2_transport \"exopolysaccharide-biosynthesis/2_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_2_transport \"exopolysaccharide-biosynthesis/2_transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_3_manual_2_3_1_HD_PSac \"exopolysaccharide-biosynthesis/3_manual_2.3.1.-_HD-PSac\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_3_manual_2_3_1_HD_PSac \"exopolysaccharide-biosynthesis/3_manual_2.3.1.-_HD-PSac\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_4_manual_3_2_1_4_HD_3_2_1_4 \"exopolysaccharide-biosynthesis/4_manual_3.2.1.4_HD-3.2.1.4\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction exopolysaccharide_biosynthesis_4_manual_3_2_1_4_HD_3_2_1_4 \"exopolysaccharide-biosynthesis/4_manual_3.2.1.4_HD-3.2.1.4\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction exopolysaccharide_exchange \"exopolysaccharide_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction exopolysaccharide_exchange \"exopolysaccharide_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamate_6_3_1_2_GLUTAMINESYN_RXN \"glutamate_6.3.1.2_GLUTAMINESYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamate_6_3_1_2_GLUTAMINESYN_RXN \"glutamate_6.3.1.2_GLUTAMINESYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamate_3_5_1_2_6_3_5_2_6_3_5_5_6_3_5_4_GLUTAMIN_RXN \"glutamate_3.5.1.2_6.3.5.2_6.3.5.5_6.3.5.4_GLUTAMIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamate_3_5_1_2_6_3_5_2_6_3_5_5_6_3_5_4_GLUTAMIN_RXN \"glutamate_3.5.1.2_6.3.5.2_6.3.5.5_6.3.5.4_GLUTAMIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamate_1_4_1_2_GLUTAMATE_DEHYDROGENASE_NADP_RXN_WOP \"glutamate_1.4.1.2_GLUTAMATE-DEHYDROGENASE-NADP+-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamate_1_4_1_2_GLUTAMATE_DEHYDROGENASE_NADP_RXN_WOP \"glutamate_1.4.1.2_GLUTAMATE-DEHYDROGENASE-NADP+-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aspartate_2_6_1_1_ASPAMINOTRANS_RXN \"aspartate_2.6.1.1_ASPAMINOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aspartate_2_6_1_1_ASPAMINOTRANS_RXN \"aspartate_2.6.1.1_ASPAMINOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aspartate_6_3_5_4_ASNSYNB_RXN \"aspartate_6.3.5.4_ASNSYNB-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aspartate_6_3_5_4_ASNSYNB_RXN \"aspartate_6.3.5.4_ASNSYNB-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aspartate_6_3_1_1_6_3_5_4_ASNSYNA_RXN \"aspartate_6.3.1.1_6.3.5.4_ASNSYNA-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aspartate_6_3_1_1_6_3_5_4_ASNSYNA_RXN \"aspartate_6.3.1.1_6.3.5.4_ASNSYNA-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction aspartate_3_5_5_4_3_5_1_1_ASPARAGHYD_RXN \"aspartate_3.5.5.4_3.5.1.1_ASPARAGHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction aspartate_3_5_5_4_3_5_1_1_ASPARAGHYD_RXN \"aspartate_3.5.5.4_3.5.1.1_ASPARAGHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_1_2_7_2_4_ASPARTATEKIN_RXN \"lysine-biosyn-II/1_2.7.2.4_ASPARTATEKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_1_2_7_2_4_ASPARTATEKIN_RXN \"lysine-biosyn-II/1_2.7.2.4_ASPARTATEKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_2_1_2_1_11_ASPARTATE_SEMIALDEHYDE_DEHYDROGENASE_RXN \"lysine-biosyn-II/2_1.2.1.11_ASPARTATE-SEMIALDEHYDE-DEHYDROGENASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_2_1_2_1_11_ASPARTATE_SEMIALDEHYDE_DEHYDROGENASE_RXN \"lysine-biosyn-II/2_1.2.1.11_ASPARTATE-SEMIALDEHYDE-DEHYDROGENASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_3_4_3_3_7_DIHYDRODIPICSYN_RXN \"lysine-biosyn-II/3_4.3.3.7_DIHYDRODIPICSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_3_4_3_3_7_DIHYDRODIPICSYN_RXN \"lysine-biosyn-II/3_4.3.3.7_DIHYDRODIPICSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_4_1_17_1_8_RXN_14014 \"lysine-biosyn-II/4_1.17.1.8_RXN-14014\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_4_1_17_1_8_RXN_14014 \"lysine-biosyn-II/4_1.17.1.8_RXN-14014\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_5_2_3_1_89_2_3_1_89_RXN \"lysine-biosyn-II/5_2.3.1.89_2.3.1.89-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_5_2_3_1_89_2_3_1_89_RXN \"lysine-biosyn-II/5_2.3.1.89_2.3.1.89-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_6_2_6_1_RXN_4822 \"lysine-biosyn-II/6_2.6.1.-_RXN-4822\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_6_2_6_1_RXN_4822 \"lysine-biosyn-II/6_2.6.1.-_RXN-4822\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_7_3_5_1_47_N_ACETYLDIAMINOPIMELATE_DEACETYLASE_RXN \"lysine-biosyn-II/7_3.5.1.47_N-ACETYLDIAMINOPIMELATE-DEACETYLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_7_3_5_1_47_N_ACETYLDIAMINOPIMELATE_DEACETYLASE_RXN \"lysine-biosyn-II/7_3.5.1.47_N-ACETYLDIAMINOPIMELATE-DEACETYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_8_5_1_1_7_DIAMINOPIMEPIM_RXN \"lysine-biosyn-II/8_5.1.1.7_DIAMINOPIMEPIM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_8_5_1_1_7_DIAMINOPIMEPIM_RXN \"lysine-biosyn-II/8_5.1.1.7_DIAMINOPIMEPIM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lysine_biosyn_II_9_4_1_1_20_DIAMINOPIMDECARB_RXN \"lysine-biosyn-II/9_4.1.1.20_DIAMINOPIMDECARB-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lysine_biosyn_II_9_4_1_1_20_DIAMINOPIMDECARB_RXN \"lysine-biosyn-II/9_4.1.1.20_DIAMINOPIMDECARB-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_1_2_3_1_1_N_ACETYLTRANSFER_RXN \"arginine-biosyn-II/1_2.3.1.1_N-ACETYLTRANSFER-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_1_2_3_1_1_N_ACETYLTRANSFER_RXN \"arginine-biosyn-II/1_2.3.1.1_N-ACETYLTRANSFER-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_2_2_7_2_8_ACETYLGLUTKIN_RXN \"arginine-biosyn-II/2_2.7.2.8_ACETYLGLUTKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_2_2_7_2_8_ACETYLGLUTKIN_RXN \"arginine-biosyn-II/2_2.7.2.8_ACETYLGLUTKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_3_1_2_1_38_N_ACETYLGLUTPREDUCT_RXN \"arginine-biosyn-II/3_1.2.1.38_N-ACETYLGLUTPREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_3_1_2_1_38_N_ACETYLGLUTPREDUCT_RXN \"arginine-biosyn-II/3_1.2.1.38_N-ACETYLGLUTPREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_4_2_6_1_11_ACETYLORNTRANSAM_RXN \"arginine-biosyn-II/4_2.6.1.11_ACETYLORNTRANSAM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_4_2_6_1_11_ACETYLORNTRANSAM_RXN \"arginine-biosyn-II/4_2.6.1.11_ACETYLORNTRANSAM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_5_2_3_1_35_GLUTAMATE_N_ACETYLTRANSFERASE_RXN \"arginine-biosyn-II/5_2.3.1.35_GLUTAMATE-N-ACETYLTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_5_2_3_1_35_GLUTAMATE_N_ACETYLTRANSFERASE_RXN \"arginine-biosyn-II/5_2.3.1.35_GLUTAMATE-N-ACETYLTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_p6_6_3_5_5_CARBPSYN_RXN \"arginine-biosyn-II/p6_6.3.5.5_CARBPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_p6_6_3_5_5_CARBPSYN_RXN \"arginine-biosyn-II/p6_6.3.5.5_CARBPSYN-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_6_2_1_3_3_ORNCARBAMTRANSFER_RXN \"arginine-biosyn-II/6_2.1.3.3_ORNCARBAMTRANSFER-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_6_2_1_3_3_ORNCARBAMTRANSFER_RXN \"arginine-biosyn-II/6_2.1.3.3_ORNCARBAMTRANSFER-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_7_6_3_4_5_ARGSUCCINSYN_RXN \"arginine-biosyn-II/7_6.3.4.5_ARGSUCCINSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_7_6_3_4_5_ARGSUCCINSYN_RXN \"arginine-biosyn-II/7_6.3.4.5_ARGSUCCINSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_biosyn_II_8_4_3_2_1_ARGSUCCINLYA_RXN \"arginine-biosyn-II/8_4.3.2.1_ARGSUCCINLYA-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_biosyn_II_8_4_3_2_1_ARGSUCCINLYA_RXN \"arginine-biosyn-II/8_4.3.2.1_ARGSUCCINLYA-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_degradation_1_3_5_3_6_ARGININE_DEIMINASE_RXN \"arginine-degradation/1_3.5.3.6_ARGININE-DEIMINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_degradation_1_3_5_3_6_ARGININE_DEIMINASE_RXN \"arginine-degradation/1_3.5.3.6_ARGININE-DEIMINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arginine_degradation_2_3_5_1_87_CITRULLINASE_RXN \"arginine-degradation/2_3.5.1.87_CITRULLINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arginine_degradation_2_3_5_1_87_CITRULLINASE_RXN \"arginine-degradation/2_3.5.1.87_CITRULLINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction putrescine_biosyn_I_1_4_1_1_19_ARGDECARBOX_RXN \"putrescine-biosyn-I/1_4.1.1.19_ARGDECARBOX-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction putrescine_biosyn_I_1_4_1_1_19_ARGDECARBOX_RXN \"putrescine-biosyn-I/1_4.1.1.19_ARGDECARBOX-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction putrescine_biosyn_I_2_3_5_3_11_AGMATIN_RXN \"putrescine-biosyn-I/2_3.5.3.11_AGMATIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction putrescine_biosyn_I_2_3_5_3_11_AGMATIN_RXN \"putrescine-biosyn-I/2_3.5.3.11_AGMATIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spermidine_biosyn_I_1_4_1_1_50_SAMDECARB_RXN \"spermidine-biosyn-I/1_4.1.1.50_SAMDECARB-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spermidine_biosyn_I_1_4_1_1_50_SAMDECARB_RXN \"spermidine-biosyn-I/1_4.1.1.50_SAMDECARB-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spermidine_biosyn_I_2_2_5_1_16_SPERMIDINESYN_RXN \"spermidine-biosyn-I/2_2.5.1.16_SPERMIDINESYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spermidine_biosyn_I_2_2_5_1_16_SPERMIDINESYN_RXN \"spermidine-biosyn-I/2_2.5.1.16_SPERMIDINESYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spermine_biosynthesis_2_5_1_16_SPERMINE_SYNTHASE_RXN \"spermine-biosynthesis_2.5.1.16_SPERMINE-SYNTHASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spermine_biosynthesis_2_5_1_16_SPERMINE_SYNTHASE_RXN \"spermine-biosynthesis_2.5.1.16_SPERMINE-SYNTHASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylputrescine_biosyn_2_3_1_57_RXN_0 \"N-acetylputrescine-biosyn_2.3.1.57_RXN-0\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylputrescine_biosyn_2_3_1_57_RXN_0 \"N-acetylputrescine-biosyn_2.3.1.57_RXN-0\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylspermidine_biosyn_2_3_1_57_SPERMACTRAN_RXN \"N-acetylspermidine-biosyn_2.3.1.57_SPERMACTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylspermidine_biosyn_2_3_1_57_SPERMACTRAN_RXN \"N-acetylspermidine-biosyn_2.3.1.57_SPERMACTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylspermine_biosyn_2_3_1_57_RXN_9939 \"N-acetylspermine-biosyn_2.3.1.57_RXN-9939\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylspermine_biosyn_2_3_1_57_RXN_9939 \"N-acetylspermine-biosyn_2.3.1.57_RXN-9939\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_diacetylspermine_biosyn_2_3_1_57_BS321087 \"N-diacetylspermine-biosyn_2.3.1.57_BS321087\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_diacetylspermine_biosyn_2_3_1_57_BS321087 \"N-diacetylspermine-biosyn_2.3.1.57_BS321087\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_methyl_5_thioadenosine_degr_3_2_2_16_3_2_2_9_METHYLTHIOADENOSINE_NUCLEOSIDASE_RXN \"S-methyl-5'-thioadenosine-degr_3.2.2.16_3.2.2.9_METHYLTHIOADENOSINE-NUCLEOSIDASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_methyl_5_thioadenosine_degr_3_2_2_16_3_2_2_9_METHYLTHIOADENOSINE_NUCLEOSIDASE_RXN \"S-methyl-5'-thioadenosine-degr_3.2.2.16_3.2.2.9_METHYLTHIOADENOSINE-NUCLEOSIDASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_methyl_5_thioadenosine_degr_3_2_2_9_5_METHYLTHIOADENOSINE_PHOSPHORYLASE_RXN \"S-methyl-5'-thioadenosine-degr_3.2.2.9_5-METHYLTHIOADENOSINE-PHOSPHORYLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_methyl_5_thioadenosine_degr_3_2_2_9_5_METHYLTHIOADENOSINE_PHOSPHORYLASE_RXN \"S-methyl-5'-thioadenosine-degr_3.2.2.9_5-METHYLTHIOADENOSINE-PHOSPHORYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_1_3_2_2_9_ADENOSYLHOMOCYSTEINE_NUCLEOSIDASE_RXN \"S-adenosyl-L-methionine-cycle-I/1_3.2.2.9_ADENOSYLHOMOCYSTEINE-NUCLEOSIDASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_1_3_2_2_9_ADENOSYLHOMOCYSTEINE_NUCLEOSIDASE_RXN \"S-adenosyl-L-methionine-cycle-I/1_3.2.2.9_ADENOSYLHOMOCYSTEINE-NUCLEOSIDASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_2_4_4_1_21_RIBOSYLHOMOCYSTEINASE_RXN \"S-adenosyl-L-methionine-cycle-I/2_4.4.1.21_RIBOSYLHOMOCYSTEINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_2_4_4_1_21_RIBOSYLHOMOCYSTEINASE_RXN \"S-adenosyl-L-methionine-cycle-I/2_4.4.1.21_RIBOSYLHOMOCYSTEINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_3_2_1_1_13_HOMOCYSMETB12_RXN \"S-adenosyl-L-methionine-cycle-I/3_2.1.1.13_HOMOCYSMETB12-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_3_2_1_1_13_HOMOCYSMETB12_RXN \"S-adenosyl-L-methionine-cycle-I/3_2.1.1.13_HOMOCYSMETB12-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_4_2_5_1_6_S_ADENMETSYN_RXN \"S-adenosyl-L-methionine-cycle-I/4_2.5.1.6_S-ADENMETSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction S_adenosyl_L_methionine_cycle_I_4_2_5_1_6_S_ADENMETSYN_RXN \"S-adenosyl-L-methionine-cycle-I/4_2.5.1.6_S-ADENMETSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction autoinducer_AI_2_biosyn_1_spontaneous_RXN_10015 \"autoinducer-AI-2-biosyn/1_spontaneous_RXN-10015\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction autoinducer_AI_2_biosyn_1_spontaneous_RXN_10015 \"autoinducer-AI-2-biosyn/1_spontaneous_RXN-10015\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction autoinducer_AI_2_biosyn_2_spontaneous_RXN_10017 \"autoinducer-AI-2-biosyn/2_spontaneous_RXN-10017\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction autoinducer_AI_2_biosyn_2_spontaneous_RXN_10017 \"autoinducer-AI-2-biosyn/2_spontaneous_RXN-10017\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methionine_biosyn_1_2_3_1_46_HOMSUCTRAN_RXN \"methionine-biosyn/1_2.3.1.46_HOMSUCTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methionine_biosyn_1_2_3_1_46_HOMSUCTRAN_RXN \"methionine-biosyn/1_2.3.1.46_HOMSUCTRAN-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methionine_biosyn_2_2_5_1_49_RXN_9384 \"methionine-biosyn/2_2.5.1.49_RXN-9384\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methionine_biosyn_2_2_5_1_49_RXN_9384 \"methionine-biosyn/2_2.5.1.49_RXN-9384\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methionine_degr_II_4_4_1_11_RXN_14301 \"methionine-degr-II_4.4.1.11_RXN-14301\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methionine_degr_II_4_4_1_11_RXN_14301 \"methionine-degr-II_4.4.1.11_RXN-14301\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminobut_2_enoate_degr_1_spontaneous_4_4_1_1_4_3_1_19_3_5_99_7_4_4_1_2_RXN_15121 \"2-aminobut-2-enoate-degr/1_spontaneous_4.4.1.1_4.3.1.19_3.5.99.7_4.4.1.2_RXN-15121\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminobut_2_enoate_degr_1_spontaneous_4_4_1_1_4_3_1_19_3_5_99_7_4_4_1_2_RXN_15121 \"2-aminobut-2-enoate-degr/1_spontaneous_4.4.1.1_4.3.1.19_3.5.99.7_4.4.1.2_RXN-15121\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminobut_2_enoate_degr_2_spontaneous_3_5_99_7_3_5_99_10_4_4_1_1_4_3_1_19_4_4_1_2_3_5_99_RXN_15123 \"2-aminobut-2-enoate-degr/2_spontaneous_3.5.99.7_3.5.99.10_4.4.1.1_4.3.1.19_4.4.1.2_3.5.99.-_RXN-15123\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminobut_2_enoate_degr_2_spontaneous_3_5_99_7_3_5_99_10_4_4_1_1_4_3_1_19_4_4_1_2_3_5_99_RXN_15123 \"2-aminobut-2-enoate-degr/2_spontaneous_3.5.99.7_3.5.99.10_4.4.1.1_4.3.1.19_4.4.1.2_3.5.99.-_RXN-15123\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminoprop_2_enoate_degr_1_spontaneous_4_4_1_8_4_4_1_13_4_4_1_25_4_3_1_17_4_4_1_28_RXN_15124 \"2-aminoprop-2-enoate-degr/1_spontaneous_4.4.1.8_4.4.1.13_4.4.1.25_4.3.1.17_4.4.1.28_RXN-15124\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminoprop_2_enoate_degr_1_spontaneous_4_4_1_8_4_4_1_13_4_4_1_25_4_3_1_17_4_4_1_28_RXN_15124 \"2-aminoprop-2-enoate-degr/1_spontaneous_4.4.1.8_4.4.1.13_4.4.1.25_4.3.1.17_4.4.1.28_RXN-15124\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminoprop_2_enoate_degr_2_spontaneous_4_4_1_8_4_4_1_13_4_3_1_17_4_4_1_28_3_5_99_10_4_4_1_25_RXN_15127 \"2-aminoprop-2-enoate-degr/2_spontaneous_4.4.1.8_4.4.1.13_4.3.1.17_4.4.1.28_3.5.99.10_4.4.1.25_RXN-15127\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminoprop_2_enoate_degr_2_spontaneous_4_4_1_8_4_4_1_13_4_3_1_17_4_4_1_28_3_5_99_10_4_4_1_25_RXN_15127 \"2-aminoprop-2-enoate-degr/2_spontaneous_4.4.1.8_4.4.1.13_4.3.1.17_4.4.1.28_3.5.99.10_4.4.1.25_RXN-15127\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminobutanoate_2_6_1_42_BS356546 \"2-aminobutanoate_2.6.1.42_BS356546\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminobutanoate_2_6_1_42_BS356546 \"2-aminobutanoate_2.6.1.42_BS356546\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_from_homoserine_0_1_1_1_3_HOMOSERDEHYDROG_RXN \"threonine-from-homoserine/0_1.1.1.3_HOMOSERDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_from_homoserine_0_1_1_1_3_HOMOSERDEHYDROG_RXN \"threonine-from-homoserine/0_1.1.1.3_HOMOSERDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_from_homoserine_1_2_7_1_39_HOMOSERKIN_RXN \"threonine-from-homoserine/1_2.7.1.39_HOMOSERKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_from_homoserine_1_2_7_1_39_HOMOSERKIN_RXN \"threonine-from-homoserine/1_2.7.1.39_HOMOSERKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_from_homoserine_2_4_2_3_1_THRESYN_RXN \"threonine-from-homoserine/2_4.2.3.1_THRESYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_from_homoserine_2_4_2_3_1_THRESYN_RXN \"threonine-from-homoserine/2_4.2.3.1_THRESYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_degr_IV_4_1_2_5_THREONINE_ALDOLASE_RXN \"threonine-degr-IV_4.1.2.5_THREONINE-ALDOLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_degr_IV_4_1_2_5_THREONINE_ALDOLASE_RXN \"threonine-degr-IV_4.1.2.5_THREONINE-ALDOLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_degr_V_4_3_1_19_RXN_15122 \"threonine-degr-V_4.3.1.19_RXN-15122\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_degr_V_4_3_1_19_RXN_15122 \"threonine-degr-V_4.3.1.19_RXN-15122\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_degr_II_1_1_1_1_103_THREODEHYD_RXN \"threonine-degr-II/1_1.1.1.103_THREODEHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_degr_II_1_1_1_1_103_THREODEHYD_RXN \"threonine-degr-II/1_1.1.1.103_THREODEHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction threonine_degr_II_2_spontaneous_THREOSPON_RXN \"threonine-degr-II/2_spontaneous_THREOSPON-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction threonine_degr_II_2_spontaneous_THREOSPON_RXN \"threonine-degr-II/2_spontaneous_THREOSPON-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction homoserine_degr_4_4_1_1_4_3_1_19_4_4_1_11_HOMOSERDEAM_RXN \"homoserine-degr_4.4.1.1_4.3.1.19_4.4.1.11_HOMOSERDEAM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction homoserine_degr_4_4_1_1_4_3_1_19_4_4_1_11_HOMOSERDEAM_RXN \"homoserine-degr_4.4.1.1_4.3.1.19_4.4.1.11_HOMOSERDEAM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_biosyn_2_1_2_1_GLYOHMETRANS_RXN \"serine-biosyn_2.1.2.1_GLYOHMETRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_biosyn_2_1_2_1_GLYOHMETRANS_RXN \"serine-biosyn_2.1.2.1_GLYOHMETRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_degr_A_4_3_1_17_RXN_15125 \"serine-degr-A_4.3.1.17_RXN-15125\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_degr_A_4_3_1_17_RXN_15125 \"serine-degr-A_4.3.1.17_RXN-15125\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_degr_B_1a_2_6_1_51_SERINE_PYRUVATE_AMINOTRANSFERASE_RXN \"serine-degr-B/1a_2.6.1.51_SERINE--PYRUVATE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_degr_B_1a_2_6_1_51_SERINE_PYRUVATE_AMINOTRANSFERASE_RXN \"serine-degr-B/1a_2.6.1.51_SERINE--PYRUVATE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_degr_B_1b_2_6_1_2_RXN_974 \"serine-degr-B/1b_2.6.1.2_RXN-974\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_degr_B_1b_2_6_1_2_RXN_974 \"serine-degr-B/1b_2.6.1.2_RXN-974\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_degr_B_2a_1_1_1_81_1_1_1_60_GLYCERATE_DEHYDROGENASE_RXN \"serine-degr-B/2a_1.1.1.81_1.1.1.60_GLYCERATE-DEHYDROGENASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_degr_B_2a_1_1_1_81_1_1_1_60_GLYCERATE_DEHYDROGENASE_RXN \"serine-degr-B/2a_1.1.1.81_1.1.1.60_GLYCERATE-DEHYDROGENASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_degr_B_2b_1_1_1_81_RXN0_300 \"serine-degr-B/2b_1.1.1.81_RXN0-300\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_degr_B_2b_1_1_1_81_RXN0_300 \"serine-degr-B/2b_1.1.1.81_RXN0-300\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction serine_degr_B_3_2_7_1_31_GLY3KIN_RXN \"serine-degr-B/3_2.7.1.31_GLY3KIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction serine_degr_B_3_2_7_1_31_GLY3KIN_RXN \"serine-degr-B/3_2.7.1.31_GLY3KIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction selenocysteine_biosyn_I_1_2_7_9_3_2_7_9_3_RXN \"selenocysteine-biosyn-I/1_2.7.9.3_2.7.9.3-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction selenocysteine_biosyn_I_1_2_7_9_3_2_7_9_3_RXN \"selenocysteine-biosyn-I/1_2.7.9.3_2.7.9.3-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction selenocysteine_biosyn_I_2_2_9_1_1_2_9_1_1_RXN \"selenocysteine-biosyn-I/2_2.9.1.1_2.9.1.1-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction selenocysteine_biosyn_I_2_2_9_1_1_2_9_1_1_RXN \"selenocysteine-biosyn-I/2_2.9.1.1_2.9.1.1-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction seleno_amino_acid_biosynthesis_2_5_1_47_RXN_12726 \"seleno-amino-acid-biosynthesis_2.5.1.47_RXN-12726\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction seleno_amino_acid_biosynthesis_2_5_1_47_RXN_12726 \"seleno-amino-acid-biosynthesis_2.5.1.47_RXN-12726\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction selenocysteine_degr_mod_4_4_1_16_mod_SELENOCYSTEINE_LYASE_RXN \"selenocysteine-degr_mod_4.4.1.16_mod_SELENOCYSTEINE-LYASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction selenocysteine_degr_mod_4_4_1_16_mod_SELENOCYSTEINE_LYASE_RXN \"selenocysteine-degr_mod_4.4.1.16_mod_SELENOCYSTEINE-LYASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cysteine_biosyn_I_1_2_5_1_47_ACSERLY_RXN \"cysteine-biosyn-I/1_2.5.1.47_ACSERLY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cysteine_biosyn_I_1_2_5_1_47_ACSERLY_RXN \"cysteine-biosyn-I/1_2.5.1.47_ACSERLY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cysteine_biosyn_I_2_2_3_1_30_SERINE_O_ACETTRAN_RXN \"cysteine-biosyn-I/2_2.3.1.30_SERINE-O-ACETTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cysteine_biosyn_I_2_2_3_1_30_SERINE_O_ACETTRAN_RXN \"cysteine-biosyn-I/2_2.3.1.30_SERINE-O-ACETTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cysteine_degr_II_4_4_1_8_4_4_1_1_LCYSDESULF_RXN \"cysteine-degr-II_4.4.1.8_4.4.1.1_LCYSDESULF-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cysteine_degr_II_4_4_1_8_4_4_1_1_LCYSDESULF_RXN \"cysteine-degr-II_4.4.1.8_4.4.1.1_LCYSDESULF-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfite_degr_manual_1_8_1_HD_sulf \"sulfite-degr_manual_1.8.1.-_HD-sulf\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfite_degr_manual_1_8_1_HD_sulf \"sulfite-degr_manual_1.8.1.-_HD-sulf\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_1_2_4_2_17_ATPPHOSPHORIBOSYLTRANS_RXN \"histidine-biosyn/1_2.4.2.17_ATPPHOSPHORIBOSYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_1_2_4_2_17_ATPPHOSPHORIBOSYLTRANS_RXN \"histidine-biosyn/1_2.4.2.17_ATPPHOSPHORIBOSYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_2_3_6_1_31_HISTPRATPHYD_RXN \"histidine-biosyn/2_3.6.1.31_HISTPRATPHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_2_3_6_1_31_HISTPRATPHYD_RXN \"histidine-biosyn/2_3.6.1.31_HISTPRATPHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_3_3_5_4_19_HISTCYCLOHYD_RXN \"histidine-biosyn/3_3.5.4.19_HISTCYCLOHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_3_3_5_4_19_HISTCYCLOHYD_RXN \"histidine-biosyn/3_3.5.4.19_HISTCYCLOHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_4_5_3_1_16_PRIBFAICARPISOM_RXN \"histidine-biosyn/4_5.3.1.16_PRIBFAICARPISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_4_5_3_1_16_PRIBFAICARPISOM_RXN \"histidine-biosyn/4_5.3.1.16_PRIBFAICARPISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_5_4_1_3_2_4_2_GLUTAMIDOTRANS_RXN \"histidine-biosyn/5_4.1.3.-_2.4.2.-_GLUTAMIDOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_5_4_1_3_2_4_2_GLUTAMIDOTRANS_RXN \"histidine-biosyn/5_4.1.3.-_2.4.2.-_GLUTAMIDOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_6_4_2_1_19_IMIDPHOSDEHYD_RXN \"histidine-biosyn/6_4.2.1.19_IMIDPHOSDEHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_6_4_2_1_19_IMIDPHOSDEHYD_RXN \"histidine-biosyn/6_4.2.1.19_IMIDPHOSDEHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_7_2_6_1_9_HISTAMINOTRANS_RXN \"histidine-biosyn/7_2.6.1.9_HISTAMINOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_7_2_6_1_9_HISTAMINOTRANS_RXN \"histidine-biosyn/7_2.6.1.9_HISTAMINOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_8_3_1_3_15_HISTIDPHOS_RXN \"histidine-biosyn/8_3.1.3.15_HISTIDPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_8_3_1_3_15_HISTIDPHOS_RXN \"histidine-biosyn/8_3.1.3.15_HISTIDPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_9_1_1_1_23_HISTOLDEHYD_RXN \"histidine-biosyn/9_1.1.1.23_HISTOLDEHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_9_1_1_1_23_HISTOLDEHYD_RXN \"histidine-biosyn/9_1.1.1.23_HISTOLDEHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_biosyn_10_1_1_1_23_HISTALDEHYD_RXN \"histidine-biosyn/10_1.1.1.23_HISTALDEHYD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_biosyn_10_1_1_1_23_HISTALDEHYD_RXN \"histidine-biosyn/10_1.1.1.23_HISTALDEHYD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction alanine_biosyn_4_1_1_12_ASPARTATE_4_DECARBOXYLASE_RXN \"alanine-biosyn_4.1.1.12_ASPARTATE-4-DECARBOXYLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction alanine_biosyn_4_1_1_12_ASPARTATE_4_DECARBOXYLASE_RXN \"alanine-biosyn_4.1.1.12_ASPARTATE-4-DECARBOXYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction alanine_biosyn_2_6_1_2_ALANINE_AMINOTRANSFERASE_RXN \"alanine-biosyn_2.6.1.2_ALANINE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction alanine_biosyn_2_6_1_2_ALANINE_AMINOTRANSFERASE_RXN \"alanine-biosyn_2.6.1.2_ALANINE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ornithine_degr_II_1_5_1_1_12_ORNITHINE_RACEMASE_RXN \"ornithine-degr-II/1_5.1.1.12_ORNITHINE-RACEMASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ornithine_degr_II_1_5_1_1_12_ORNITHINE_RACEMASE_RXN \"ornithine-degr-II/1_5.1.1.12_ORNITHINE-RACEMASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ornithine_degr_II_2_5_4_3_5_ORNMUTST_RXN \"ornithine-degr-II/2_5.4.3.5_ORNMUTST-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ornithine_degr_II_2_5_4_3_5_ORNMUTST_RXN \"ornithine-degr-II/2_5.4.3.5_ORNMUTST-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ornithine_degr_II_3_1_4_1_12_24_DIAMINOPENTANOATE_DEHYDROGENASE_RXN_WOP \"ornithine-degr-II/3_1.4.1.12_24-DIAMINOPENTANOATE-DEHYDROGENASE-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ornithine_degr_II_3_1_4_1_12_24_DIAMINOPENTANOATE_DEHYDROGENASE_RXN_WOP \"ornithine-degr-II/3_1.4.1.12_24-DIAMINOPENTANOATE-DEHYDROGENASE-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ornithine_degr_II_4_2_3_1_B10_AKPTHIOL_RXN \"ornithine-degr-II/4_2.3.1.B10_AKPTHIOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ornithine_degr_II_4_2_3_1_B10_AKPTHIOL_RXN \"ornithine-degr-II/4_2.3.1.B10_AKPTHIOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction proline_degr_1_5_1_1_4_PROLINE_RACEMASE_RXN \"proline-degr/1_5.1.1.4_PROLINE-RACEMASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction proline_degr_1_5_1_1_4_PROLINE_RACEMASE_RXN \"proline-degr/1_5.1.1.4_PROLINE-RACEMASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction proline_degr_2_1_21_4_1_BS370299 \"proline-degr/2_1.21.4.1_BS370299\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction proline_degr_2_1_21_4_1_BS370299 \"proline-degr/2_1.21.4.1_BS370299\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_cleavage_1_moddir_1_4_4_2_GCVP_RXN \"glycine-cleavage/1_moddir_1.4.4.2_GCVP-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_cleavage_1_moddir_1_4_4_2_GCVP_RXN \"glycine-cleavage/1_moddir_1.4.4.2_GCVP-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_cleavage_2_2_1_2_10_GCVT_RXN \"glycine-cleavage/2_2.1.2.10_GCVT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_cleavage_2_2_1_2_10_GCVT_RXN \"glycine-cleavage/2_2.1.2.10_GCVT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_cleavage_3_1_8_1_4_RXN_7719 \"glycine-cleavage/3_1.8.1.4_RXN-7719\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_cleavage_3_1_8_1_4_RXN_7719 \"glycine-cleavage/3_1.8.1.4_RXN-7719\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_degr_1_1_21_4_2_RXN_7566 \"glycine-degr/1_1.21.4.2_RXN-7566\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_degr_1_1_21_4_2_RXN_7566 \"glycine-degr/1_1.21.4.2_RXN-7566\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glycine_degr_2_1_8_1_9_THIOREDOXIN_REDUCT_NADPH_RXN \"glycine-degr/2_1.8.1.9_THIOREDOXIN-REDUCT-NADPH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glycine_degr_2_1_8_1_9_THIOREDOXIN_REDUCT_NADPH_RXN \"glycine-degr/2_1.8.1.9_THIOREDOXIN-REDUCT-NADPH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_1_2_5_1_54_DAHPSYN_RXN \"chorismate-biosyn-I/1_2.5.1.54_DAHPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_1_2_5_1_54_DAHPSYN_RXN \"chorismate-biosyn-I/1_2.5.1.54_DAHPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_2_4_2_3_4_3_DEHYDROQUINATE_SYNTHASE_RXN \"chorismate-biosyn-I/2_4.2.3.4_3-DEHYDROQUINATE-SYNTHASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_2_4_2_3_4_3_DEHYDROQUINATE_SYNTHASE_RXN \"chorismate-biosyn-I/2_4.2.3.4_3-DEHYDROQUINATE-SYNTHASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_3_4_2_1_10_3_DEHYDROQUINATE_DEHYDRATASE_RXN \"chorismate-biosyn-I/3_4.2.1.10_3-DEHYDROQUINATE-DEHYDRATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_3_4_2_1_10_3_DEHYDROQUINATE_DEHYDRATASE_RXN \"chorismate-biosyn-I/3_4.2.1.10_3-DEHYDROQUINATE-DEHYDRATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_4_1_1_1_25_RXN_7968 \"chorismate-biosyn-I/4_1.1.1.25_RXN-7968\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_4_1_1_1_25_RXN_7968 \"chorismate-biosyn-I/4_1.1.1.25_RXN-7968\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_5_2_7_1_71_SHIKIMATE_KINASE_RXN \"chorismate-biosyn-I/5_2.7.1.71_SHIKIMATE-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_5_2_7_1_71_SHIKIMATE_KINASE_RXN \"chorismate-biosyn-I/5_2.7.1.71_SHIKIMATE-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_6_2_5_1_19_2_5_1_19_RXN \"chorismate-biosyn-I/6_2.5.1.19_2.5.1.19-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_6_2_5_1_19_2_5_1_19_RXN \"chorismate-biosyn-I/6_2.5.1.19_2.5.1.19-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chorismate_biosyn_I_7_4_2_3_5_CHORISMATE_SYNTHASE_RXN \"chorismate-biosyn-I/7_4.2.3.5_CHORISMATE-SYNTHASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chorismate_biosyn_I_7_4_2_3_5_CHORISMATE_SYNTHASE_RXN \"chorismate-biosyn-I/7_4.2.3.5_CHORISMATE-SYNTHASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyr_phe_biosyn_5_4_99_5_CHORISMATEMUT_RXN \"tyr/phe-biosyn_5.4.99.5_CHORISMATEMUT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyr_phe_biosyn_5_4_99_5_CHORISMATEMUT_RXN \"tyr/phe-biosyn_5.4.99.5_CHORISMATEMUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_biosyn_I_1_1_3_1_12_PREPHENATEDEHYDROG_RXN \"tyrosine-biosyn_I/1_1.3.1.12_PREPHENATEDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_biosyn_I_1_1_3_1_12_PREPHENATEDEHYDROG_RXN \"tyrosine-biosyn_I/1_1.3.1.12_PREPHENATEDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_biosyn_I_2a_2_6_1_9_2_6_1_1_2_6_1_57_TYROSINE_AMINOTRANSFERASE_RXN \"tyrosine-biosyn_I/2a_2.6.1.9_2.6.1.1_2.6.1.57_TYROSINE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_biosyn_I_2a_2_6_1_9_2_6_1_1_2_6_1_57_TYROSINE_AMINOTRANSFERASE_RXN \"tyrosine-biosyn_I/2a_2.6.1.9_2.6.1.1_2.6.1.57_TYROSINE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_biosyn_I_2b_2_6_1_51_RXN3O_4157 \"tyrosine-biosyn_I/2b_2.6.1.51_RXN3O-4157\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_biosyn_I_2b_2_6_1_51_RXN3O_4157 \"tyrosine-biosyn_I/2b_2.6.1.51_RXN3O-4157\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_biosyn_I_1_4_2_1_51_PREPHENATEDEHYDRAT_RXN \"phenylalanine-biosyn-I/1_4.2.1.51_PREPHENATEDEHYDRAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_biosyn_I_1_4_2_1_51_PREPHENATEDEHYDRAT_RXN \"phenylalanine-biosyn-I/1_4.2.1.51_PREPHENATEDEHYDRAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_biosyn_I_2a_2_6_1_57_2_6_1_9_2_6_1_1_2_6_1_42_PHEAMINOTRANS_RXN \"phenylalanine-biosyn-I/2a_2.6.1.57_2.6.1.9_2.6.1.1_2.6.1.42_PHEAMINOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_biosyn_I_2a_2_6_1_57_2_6_1_9_2_6_1_1_2_6_1_42_PHEAMINOTRANS_RXN \"phenylalanine-biosyn-I/2a_2.6.1.57_2.6.1.9_2.6.1.1_2.6.1.42_PHEAMINOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_biosyn_I_2b_2_6_1_57_RXN_7708 \"phenylalanine-biosyn-I/2b_2.6.1.57_RXN-7708\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_biosyn_I_2b_2_6_1_57_RXN_7708 \"phenylalanine-biosyn-I/2b_2.6.1.57_RXN-7708\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_biosyn_II_1_2_6_1_57_PREPHENATE_ASP_TRANSAMINE_RXN \"phenylalanine-biosyn-II/1_2.6.1.57_PREPHENATE-ASP-TRANSAMINE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_biosyn_II_1_2_6_1_57_PREPHENATE_ASP_TRANSAMINE_RXN \"phenylalanine-biosyn-II/1_2.6.1.57_PREPHENATE-ASP-TRANSAMINE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_biosyn_II_2_4_2_1_51_CARBOXYCYCLOHEXADIENYL_DEHYDRATASE_RXN \"phenylalanine-biosyn-II/2_4.2.1.51_CARBOXYCYCLOHEXADIENYL-DEHYDRATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_biosyn_II_2_4_2_1_51_CARBOXYCYCLOHEXADIENYL_DEHYDRATASE_RXN \"phenylalanine-biosyn-II/2_4.2.1.51_CARBOXYCYCLOHEXADIENYL-DEHYDRATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methionine_from_2_oxo_4_methylthiobutanoate_2_6_1_57_RXN_14147 \"methionine-from-2-oxo-4-methylthiobutanoate_2.6.1.57_RXN-14147\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methionine_from_2_oxo_4_methylthiobutanoate_2_6_1_57_RXN_14147 \"methionine-from-2-oxo-4-methylthiobutanoate_2.6.1.57_RXN-14147\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_degr_2_6_1_27_TRYPTOPHAN_AMINOTRANSFERASE_RXN \"tryptophan-degr_2.6.1.27_TRYPTOPHAN-AMINOTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_degr_2_6_1_27_TRYPTOPHAN_AMINOTRANSFERASE_RXN \"tryptophan-degr_2.6.1.27_TRYPTOPHAN-AMINOTRANSFERASE-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_degr_2_6_1_57_RXN_14685 \"tryptophan-degr_2.6.1.57_RXN-14685\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_degr_2_6_1_57_RXN_14685 \"tryptophan-degr_2.6.1.57_RXN-14685\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_degr_2_6_1_38_HISTTRANSAM_RXN \"histidine-degr_2.6.1.38_HISTTRANSAM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_degr_2_6_1_38_HISTTRANSAM_RXN \"histidine-degr_2.6.1.38_HISTTRANSAM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_ox_1_1_2_7_8_1_2_7_8_RXN \"tryptophan-ox/1_1.2.7.8_1.2.7.8-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_ox_1_1_2_7_8_1_2_7_8_RXN \"tryptophan-ox/1_1.2.7.8_1.2.7.8-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_ox_1_manual_1_2_7_8_HD_1_2_7_8_tyr \"tyrosine-ox/1_manual_1.2.7.8_HD-1.2.7.8-tyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_ox_1_manual_1_2_7_8_HD_1_2_7_8_tyr \"tyrosine-ox/1_manual_1.2.7.8_HD-1.2.7.8-tyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_ox_1_manual_1_2_7_8_HD_1_2_7_8_phe \"phenylalanine-ox/1_manual_1.2.7.8_HD-1.2.7.8-phe\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_ox_1_manual_1_2_7_8_HD_1_2_7_8_phe \"phenylalanine-ox/1_manual_1.2.7.8_HD-1.2.7.8-phe\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_ox_1_manual_1_2_7_8_HD_1_2_7_8_his \"histidine-ox/1_manual_1.2.7.8_HD-1.2.7.8-his\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_ox_1_manual_1_2_7_8_HD_1_2_7_8_his \"histidine-ox/1_manual_1.2.7.8_HD-1.2.7.8-his\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_ox_2_manual_2_3_1_HD_2_3_1_trp \"tryptophan-ox/2_manual_2.3.1.-_HD-2.3.1.-trp\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_ox_2_manual_2_3_1_HD_2_3_1_trp \"tryptophan-ox/2_manual_2.3.1.-_HD-2.3.1.-trp\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_ox_2_2_3_1_manual_HD_2_3_1_tyr \"tyrosine-ox/2_2.3.1.-_manual_HD-2.3.1.-tyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_ox_2_2_3_1_manual_HD_2_3_1_tyr \"tyrosine-ox/2_2.3.1.-_manual_HD-2.3.1.-tyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_ox_2_2_3_1_manual_HD_2_3_1_phe \"phenylalanine-ox/2_2.3.1.-_manual_HD-2.3.1.-phe\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_ox_2_2_3_1_manual_HD_2_3_1_phe \"phenylalanine-ox/2_2.3.1.-_manual_HD-2.3.1.-phe\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_ox_2_2_3_1_manual_HD_2_3_1_his \"histidine-ox/2_2.3.1.-_manual_HD-2.3.1.-his\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_ox_2_2_3_1_manual_HD_2_3_1_his \"histidine-ox/2_2.3.1.-_manual_HD-2.3.1.-his\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tryptophan_ox_3_manual_2_7_2_HD_2_7_2_trp \"tryptophan-ox/3_manual_2.7.2.-_HD-2.7.2.-trp\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tryptophan_ox_3_manual_2_7_2_HD_2_7_2_trp \"tryptophan-ox/3_manual_2.7.2.-_HD-2.7.2.-trp\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_ox_3_manual_2_7_2_HD_2_7_2_tyr \"tyrosine-ox/3_manual_2.7.2.-_HD-2.7.2.-tyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_ox_3_manual_2_7_2_HD_2_7_2_tyr \"tyrosine-ox/3_manual_2.7.2.-_HD-2.7.2.-tyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phenylalanine_ox_3_manual_2_7_2_HD_2_7_2_phe \"phenylalanine-ox/3_manual_2.7.2.-_HD-2.7.2.-phe\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phenylalanine_ox_3_manual_2_7_2_HD_2_7_2_phe \"phenylalanine-ox/3_manual_2.7.2.-_HD-2.7.2.-phe\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction histidine_ox_3_manual_2_7_2_HD_2_7_2_his \"histidine-ox/3_manual_2.7.2.-_HD-2.7.2.-his\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction histidine_ox_3_manual_2_7_2_HD_2_7_2_his \"histidine-ox/3_manual_2.7.2.-_HD-2.7.2.-his\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyrosine_ox_4_4_1_1_83_4_1_1_83_RXN \"tyrosine-ox/4_4.1.1.83_4.1.1.83-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyrosine_ox_4_4_1_1_83_4_1_1_83_RXN \"tyrosine-ox/4_4.1.1.83_4.1.1.83-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phe_red_1_1_1_1_222_BS186998 \"phe-red/1_1.1.1.222_BS186998\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phe_red_1_1_1_1_222_BS186998 \"phe-red/1_1.1.1.222_BS186998\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phe_red_2_manual_2_8_3_HD_2_8_3_phe \"phe-red/2_manual_2.8.3.-_HD-2.8.3.-phe\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phe_red_2_manual_2_8_3_HD_2_8_3_phe \"phe-red/2_manual_2.8.3.-_HD-2.8.3.-phe\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phe_red_3_manual_4_2_1_HD_4_2_1_phe \"phe-red/3_manual_4.2.1.-_HD-4.2.1.-phe\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phe_red_3_manual_4_2_1_HD_4_2_1_phe \"phe-red/3_manual_4.2.1.-_HD-4.2.1.-phe\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phe_red_4_manual_1_3_98_HD_1_3_98_phe \"phe-red/4_manual_1.3.98.-_HD-1.3.98.-phe\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phe_red_4_manual_1_3_98_HD_1_3_98_phe \"phe-red/4_manual_1.3.98.-_HD-1.3.98.-phe\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyr_red_1_1_1_1_222_1_1_1_222_RXN_WOP \"tyr-red/1_1.1.1.222_1.1.1.222-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyr_red_1_1_1_1_222_1_1_1_222_RXN_WOP \"tyr-red/1_1.1.1.222_1.1.1.222-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyr_red_2_manual_2_8_3_HD_2_8_3_tyr \"tyr-red/2_manual_2.8.3.-_HD-2.8.3.-tyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyr_red_2_manual_2_8_3_HD_2_8_3_tyr \"tyr-red/2_manual_2.8.3.-_HD-2.8.3.-tyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyr_red_3_manual_4_2_1_HD_4_2_1_tyr \"tyr-red/3_manual_4.2.1.-_HD-4.2.1.-tyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyr_red_3_manual_4_2_1_HD_4_2_1_tyr \"tyr-red/3_manual_4.2.1.-_HD-4.2.1.-tyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tyr_red_4_manual_1_3_98_HD_1_3_98_tyr \"tyr-red/4_manual_1.3.98.-_HD-1.3.98.-tyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tyr_red_4_manual_1_3_98_HD_1_3_98_tyr \"tyr-red/4_manual_1.3.98.-_HD-1.3.98.-tyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_biosyn_1_2_2_1_6_RXN_12583 \"valine-biosyn/1_2.2.1.6_RXN-12583\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_biosyn_1_2_2_1_6_RXN_12583 \"valine-biosyn/1_2.2.1.6_RXN-12583\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_biosyn_2_2_2_1_6_RXN_14037 \"valine-biosyn/2_2.2.1.6_RXN-14037\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_biosyn_2_2_2_1_6_RXN_14037 \"valine-biosyn/2_2.2.1.6_RXN-14037\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_biosyn_3_1_1_1_86_RXN_16061 \"valine-biosyn/3_1.1.1.86_RXN-16061\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_biosyn_3_1_1_1_86_RXN_16061 \"valine-biosyn/3_1.1.1.86_RXN-16061\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_biosyn_4_4_2_1_9_DIHYDROXYISOVALDEHYDRAT_RXN \"valine-biosyn/4_4.2.1.9_DIHYDROXYISOVALDEHYDRAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_biosyn_4_4_2_1_9_DIHYDROXYISOVALDEHYDRAT_RXN \"valine-biosyn/4_4.2.1.9_DIHYDROXYISOVALDEHYDRAT-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_biosyn_5_2_6_1_42_BRANCHED_CHAINAMINOTRANSFERVAL_RXN \"valine-biosyn/5_2.6.1.42_BRANCHED-CHAINAMINOTRANSFERVAL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_biosyn_5_2_6_1_42_BRANCHED_CHAINAMINOTRANSFERVAL_RXN \"valine-biosyn/5_2.6.1.42_BRANCHED-CHAINAMINOTRANSFERVAL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_biosyn_1_2_3_3_13_2_ISOPROPYLMALATESYN_RXN \"leucine-biosyn/1_2.3.3.13_2-ISOPROPYLMALATESYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_biosyn_1_2_3_3_13_2_ISOPROPYLMALATESYN_RXN \"leucine-biosyn/1_2.3.3.13_2-ISOPROPYLMALATESYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_biosyn_2_4_2_1_33_RXN_13163 \"leucine-biosyn/2_4.2.1.33_RXN-13163\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_biosyn_2_4_2_1_33_RXN_13163 \"leucine-biosyn/2_4.2.1.33_RXN-13163\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_biosyn_3_1_1_1_85_3_ISOPROPYLMALDEHYDROG_RXN \"leucine-biosyn/3_1.1.1.85_3-ISOPROPYLMALDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_biosyn_3_1_1_1_85_3_ISOPROPYLMALDEHYDROG_RXN \"leucine-biosyn/3_1.1.1.85_3-ISOPROPYLMALDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_biosyn_4_spontaneous_1_1_1_85_RXN_7800 \"leucine-biosyn/4_spontaneous_1.1.1.85_RXN-7800\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_biosyn_4_spontaneous_1_1_1_85_RXN_7800 \"leucine-biosyn/4_spontaneous_1.1.1.85_RXN-7800\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_biosyn_5_2_6_1_42_BRANCHED_CHAINAMINOTRANSFERLEU_RXN \"leucine-biosyn/5_2.6.1.42_BRANCHED-CHAINAMINOTRANSFERLEU-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_biosyn_5_2_6_1_42_BRANCHED_CHAINAMINOTRANSFERLEU_RXN \"leucine-biosyn/5_2.6.1.42_BRANCHED-CHAINAMINOTRANSFERLEU-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_1_2_3_1_182_RXN_7743 \"isoleucine-biosyn-II/1_2.3.1.182_RXN-7743\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_1_2_3_1_182_RXN_7743 \"isoleucine-biosyn-II/1_2.3.1.182_RXN-7743\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_2_4_2_1_35_R_2_METHYLMALATE_DEHYDRATASE_RXN \"isoleucine-biosyn-II/2_4.2.1.35_R-2-METHYLMALATE-DEHYDRATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_2_4_2_1_35_R_2_METHYLMALATE_DEHYDRATASE_RXN \"isoleucine-biosyn-II/2_4.2.1.35_R-2-METHYLMALATE-DEHYDRATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_3_4_2_1_35_RXN_7744 \"isoleucine-biosyn-II/3_4.2.1.35_RXN-7744\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_3_4_2_1_35_RXN_7744 \"isoleucine-biosyn-II/3_4.2.1.35_RXN-7744\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_4_1_1_1_85_RXN_7745 \"isoleucine-biosyn-II/4_1.1.1.85_RXN-7745\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_4_1_1_1_85_RXN_7745 \"isoleucine-biosyn-II/4_1.1.1.85_RXN-7745\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_5_2_2_1_6_R04673 \"isoleucine-biosyn-II/5_2.2.1.6_R04673\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_5_2_2_1_6_R04673 \"isoleucine-biosyn-II/5_2.2.1.6_R04673\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_6_1_1_1_86_ACETOOHBUTREDUCTOISOM_RXN \"isoleucine-biosyn-II/6_1.1.1.86_ACETOOHBUTREDUCTOISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_6_1_1_1_86_ACETOOHBUTREDUCTOISOM_RXN \"isoleucine-biosyn-II/6_1.1.1.86_ACETOOHBUTREDUCTOISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_7_4_2_1_9_DIHYDROXYMETVALDEHYDRAT_RXN \"isoleucine-biosyn-II/7_4.2.1.9_DIHYDROXYMETVALDEHYDRAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_7_4_2_1_9_DIHYDROXYMETVALDEHYDRAT_RXN \"isoleucine-biosyn-II/7_4.2.1.9_DIHYDROXYMETVALDEHYDRAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_8_2_6_1_42_BRANCHED_CHAINAMINOTRANSFERILEU_RXN \"isoleucine-biosyn-II/8_2.6.1.42_BRANCHED-CHAINAMINOTRANSFERILEU-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_biosyn_II_8_2_6_1_42_BRANCHED_CHAINAMINOTRANSFERILEU_RXN \"isoleucine-biosyn-II/8_2.6.1.42_BRANCHED-CHAINAMINOTRANSFERILEU-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_ox_1_1_2_7_7_1_2_7_7_RXN \"valine-ox/1_1.2.7.7_1.2.7.7-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_ox_1_1_2_7_7_1_2_7_7_RXN \"valine-ox/1_1.2.7.7_1.2.7.7-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_ox_1_manual_1_2_7_7_HD_1_2_7_7_leu \"leucine-ox/1_manual_1.2.7.7_HD-1.2.7.7-leu\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_ox_1_manual_1_2_7_7_HD_1_2_7_7_leu \"leucine-ox/1_manual_1.2.7.7_HD-1.2.7.7-leu\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_ox_1_manual_1_2_7_7_HD_1_2_7_7_ile \"isoleucine-ox/1_manual_1.2.7.7_HD-1.2.7.7-ile\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_ox_1_manual_1_2_7_7_HD_1_2_7_7_ile \"isoleucine-ox/1_manual_1.2.7.7_HD-1.2.7.7-ile\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_ox_2_manual_2_3_1_HD_2_3_1_val \"valine-ox/2_manual_2.3.1.-_HD-2.3.1.-val\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_ox_2_manual_2_3_1_HD_2_3_1_val \"valine-ox/2_manual_2.3.1.-_HD-2.3.1.-val\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_ox_2_manual_2_3_1_HD_2_3_1_leu \"leucine-ox/2_manual_2.3.1.-_HD-2.3.1.-leu\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_ox_2_manual_2_3_1_HD_2_3_1_leu \"leucine-ox/2_manual_2.3.1.-_HD-2.3.1.-leu\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_ox_2_manual_2_3_1_HD_2_3_1_ile \"isoleucine-ox/2_manual_2.3.1.-_HD-2.3.1.-ile\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_ox_2_manual_2_3_1_HD_2_3_1_ile \"isoleucine-ox/2_manual_2.3.1.-_HD-2.3.1.-ile\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction valine_ox_3_manual_2_7_2_HD_2_7_2_val \"valine-ox/3_manual_2.7.2.-_HD-2.7.2.-val\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction valine_ox_3_manual_2_7_2_HD_2_7_2_val \"valine-ox/3_manual_2.7.2.-_HD-2.7.2.-val\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leucine_ox_3_manual_2_7_2_HD_7_2_leu \"leucine-ox/3_manual_2.7.2.-_HD-.7.2.-leu\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leucine_ox_3_manual_2_7_2_HD_7_2_leu \"leucine-ox/3_manual_2.7.2.-_HD-.7.2.-leu\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction isoleucine_ox_3_manual_2_7_2_HD_2_7_2_ile \"isoleucine-ox/3_manual_2.7.2.-_HD-2.7.2.-ile\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction isoleucine_ox_3_manual_2_7_2_HD_2_7_2_ile \"isoleucine-ox/3_manual_2.7.2.-_HD-2.7.2.-ile\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leu_red_1_manual_1_1_1_345_HD_1_1_1_345 \"leu-red/1_manual_1.1.1.345_HD-1.1.1.345\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leu_red_1_manual_1_1_1_345_HD_1_1_1_345 \"leu-red/1_manual_1.1.1.345_HD-1.1.1.345\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leu_red_2a_manual_2_8_3_HD_2_8_3_leu_A \"leu-red/2a_manual_2.8.3.-_HD-2.8.3.-leu-A\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leu_red_2a_manual_2_8_3_HD_2_8_3_leu_A \"leu-red/2a_manual_2.8.3.-_HD-2.8.3.-leu-A\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leu_red_3_manual_4_2_1_HD_4_2_1_leu \"leu-red/3_manual_4.2.1.-_HD-4.2.1.-leu\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leu_red_3_manual_4_2_1_HD_4_2_1_leu \"leu-red/3_manual_4.2.1.-_HD-4.2.1.-leu\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leu_red_4_manual_1_3_98_HD_1_3_98_leu \"leu-red/4_manual_1.3.98.-_HD-1.3.98.-leu\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leu_red_4_manual_1_3_98_HD_1_3_98_leu \"leu-red/4_manual_1.3.98.-_HD-1.3.98.-leu\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leu_red_2b_manual_2_8_3_HD_2_8_3_leu_B \"leu-red/2b_manual_2.8.3.-_HD-2.8.3.-leu-B\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leu_red_2b_manual_2_8_3_HD_2_8_3_leu_B \"leu-red/2b_manual_2.8.3.-_HD-2.8.3.-leu-B\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction leu_red_5_manual_2_3_1_HD_2_3_1_leu_red \"leu-red/5_manual_2.3.1.-_HD-2.3.1.-leu-red\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction leu_red_5_manual_2_3_1_HD_2_3_1_leu_red \"leu-red/5_manual_2.3.1.-_HD-2.3.1.-leu-red\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formate_2_3_1_54_PYRUVFORMLY_RXN \"formate_2.3.1.54_PYRUVFORMLY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formate_2_3_1_54_PYRUVFORMLY_RXN \"formate_2.3.1.54_PYRUVFORMLY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formate_2_3_1_54_KETOBUTFORMLY_RXN \"formate_2.3.1.54_KETOBUTFORMLY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formate_2_3_1_54_KETOBUTFORMLY_RXN \"formate_2.3.1.54_KETOBUTFORMLY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_EtOH_1_1_2_1_10_ACETALD_DEHYDROG_RXN \"ferm-EtOH/1_1.2.1.10_ACETALD-DEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_EtOH_1_1_2_1_10_ACETALD_DEHYDROG_RXN \"ferm-EtOH/1_1.2.1.10_ACETALD-DEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_EtOH_2_1_1_1_1_ALCOHOL_DEHYDROGENASE_NADP_RXN_WOP \"ferm-EtOH/2_1.1.1.1_ALCOHOL-DEHYDROGENASE-NADP+-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_EtOH_2_1_1_1_1_ALCOHOL_DEHYDROGENASE_NADP_RXN_WOP \"ferm-EtOH/2_1.1.1.1_ALCOHOL-DEHYDROGENASE-NADP+-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_PropOH_1_1_2_1_87_RXN_12736 \"ferm-PropOH/1_1.2.1.87_RXN-12736\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_PropOH_1_1_2_1_87_RXN_12736 \"ferm-PropOH/1_1.2.1.87_RXN-12736\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_PropOH_2_1_1_1_1_RXN_13198 \"ferm-PropOH/2_1.1.1.1_RXN-13198\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_PropOH_2_1_1_1_1_RXN_13198 \"ferm-PropOH/2_1.1.1.1_RXN-13198\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_ButOH_1_1_2_1_10_1_2_1_87_BUTANAL_DEHYDROGENASE_RXN_WOP \"ferm-ButOH/1_1.2.1.10_1.2.1.87_BUTANAL-DEHYDROGENASE-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_ButOH_1_1_2_1_10_1_2_1_87_BUTANAL_DEHYDROGENASE_RXN_WOP \"ferm-ButOH/1_1.2.1.10_1.2.1.87_BUTANAL-DEHYDROGENASE-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_ButOH_2_1_1_1_RXN_161 \"ferm-ButOH/2_1.1.1.-_RXN-161\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_ButOH_2_1_1_1_RXN_161 \"ferm-ButOH/2_1.1.1.-_RXN-161\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_IbutOH_1_1_2_1_10_RXN_13671 \"ferm-IbutOH/1_1.2.1.10_RXN-13671\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_IbutOH_1_1_2_1_10_RXN_13671 \"ferm-IbutOH/1_1.2.1.10_RXN-13671\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_IbutOH_2_1_1_1_1_RXN_7657 \"ferm-IbutOH/2_1.1.1.1_RXN-7657\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_IbutOH_2_1_1_1_1_RXN_7657 \"ferm-IbutOH/2_1.1.1.1_RXN-7657\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_acetate_1_1_2_7_1_PYRUFLAVREDUCT_RXN \"ferm-acetate/1_1.2.7.1_PYRUFLAVREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_acetate_1_1_2_7_1_PYRUFLAVREDUCT_RXN \"ferm-acetate/1_1.2.7.1_PYRUFLAVREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_acetate_2_2_3_1_8_PHOSACETYLTRANS_RXN \"ferm-acetate/2_2.3.1.8_PHOSACETYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_acetate_2_2_3_1_8_PHOSACETYLTRANS_RXN \"ferm-acetate/2_2.3.1.8_PHOSACETYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_acetate_3_2_7_2_1_moddir_ACETATEKIN_RXN \"ferm-acetate/3_2.7.2.1_moddir_ACETATEKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_acetate_3_2_7_2_1_moddir_ACETATEKIN_RXN \"ferm-acetate/3_2.7.2.1_moddir_ACETATEKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_propanoate_1_1_2_7_1_2_OXOBUTYRATE_SYNTHASE_RXN \"ferm-propanoate/1_1.2.7.1_2-OXOBUTYRATE-SYNTHASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_propanoate_1_1_2_7_1_2_OXOBUTYRATE_SYNTHASE_RXN \"ferm-propanoate/1_1.2.7.1_2-OXOBUTYRATE-SYNTHASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_propanoate_2_2_3_1_222_2_3_1_8_PTAALT_RXN \"ferm-propanoate/2_2.3.1.222_2.3.1.8_PTAALT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_propanoate_2_2_3_1_222_2_3_1_8_PTAALT_RXN \"ferm-propanoate/2_2.3.1.222_2.3.1.8_PTAALT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ferm_propanoate_3_2_7_2_1_moddir_PROPKIN_RXN \"ferm-propanoate/3_2.7.2.1_moddir_PROPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ferm_propanoate_3_2_7_2_1_moddir_PROPKIN_RXN \"ferm-propanoate/3_2.7.2.1_moddir_PROPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetyl_CoA_to_acid_6_2_1_13_ACETATE_COA_LIGASE_ADP_FORMING_RXN \"acetyl-CoA-to-acid_6.2.1.13_ACETATE--COA-LIGASE-ADP-FORMING-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetyl_CoA_to_acid_6_2_1_13_ACETATE_COA_LIGASE_ADP_FORMING_RXN \"acetyl-CoA-to-acid_6.2.1.13_ACETATE--COA-LIGASE-ADP-FORMING-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoyl_CoA_to_acid_6_2_1_13_PROPANOATECOA_LIGASE_RXN \"propanoyl-CoA-to-acid_6.2.1.13_PROPANOATECOA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoyl_CoA_to_acid_6_2_1_13_PROPANOATECOA_LIGASE_RXN \"propanoyl-CoA-to-acid_6.2.1.13_PROPANOATECOA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction butanoyl_CoA_to_acid_manual_6_2_1_13_HD_6_2_1_13 \"butanoyl-CoA-to-acid_manual_6.2.1.13_HD-6.2.1.13\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction butanoyl_CoA_to_acid_manual_6_2_1_13_HD_6_2_1_13 \"butanoyl-CoA-to-acid_manual_6.2.1.13_HD-6.2.1.13\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_ferm_1a_2_8_3_PROPIONLACT_RXN \"propanoate-ferm/1a_2.8.3.-_PROPIONLACT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_ferm_1a_2_8_3_PROPIONLACT_RXN \"propanoate-ferm/1a_2.8.3.-_PROPIONLACT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_ferm_2a_4_2_1_54_RXN_781 \"propanoate-ferm/2a_4.2.1.54_RXN-781\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_ferm_2a_4_2_1_54_RXN_781 \"propanoate-ferm/2a_4.2.1.54_RXN-781\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_ferm_1b_2_8_3_LACCOA_RXN \"propanoate-ferm/1b_2.8.3.-_LACCOA-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_ferm_1b_2_8_3_LACCOA_RXN \"propanoate-ferm/1b_2.8.3.-_LACCOA-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_ferm_2b_4_2_1_54_LACTOYL_COA_DEHYDRATASE_RXN \"propanoate-ferm/2b_4.2.1.54_LACTOYL-COA-DEHYDRATASE-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_ferm_2b_4_2_1_54_LACTOYL_COA_DEHYDRATASE_RXN \"propanoate-ferm/2b_4.2.1.54_LACTOYL-COA-DEHYDRATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction propanoate_ferm_3_manual_1_3_98_HD_1_3_98_prop \"propanoate-ferm/3_manual_1.3.98.-_HD-1.3.98.-prop\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction propanoate_ferm_3_manual_1_3_98_HD_1_3_98_prop \"propanoate-ferm/3_manual_1.3.98.-_HD-1.3.98.-prop\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pyruvate_ferm_to_lactate_moddir_1_1_1_27_L_LACTATE_DEHYDROGENASE_RXN \"pyruvate-ferm-to-lactate_moddir_1.1.1.27_L-LACTATE-DEHYDROGENASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pyruvate_ferm_to_lactate_moddir_1_1_1_27_L_LACTATE_DEHYDROGENASE_RXN \"pyruvate-ferm-to-lactate_moddir_1.1.1.27_L-LACTATE-DEHYDROGENASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lactate_recemation_5_1_2_1_LACTATE_RACEMASE_RXN \"lactate-recemation_5.1.2.1_LACTATE-RACEMASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lactate_recemation_5_1_2_1_LACTATE_RACEMASE_RXN \"lactate-recemation_5.1.2.1_LACTATE-RACEMASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lactate_oxidation_manual_1_1_98_HD_1_1_98_pyr \"lactate-oxidation_manual_1.1.98.-_HD-1.1.98.-pyr\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lactate_oxidation_manual_1_1_98_HD_1_1_98_pyr \"lactate-oxidation_manual_1.1.98.-_HD-1.1.98.-pyr\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_1_2_8_3_18_RXN_8807 \"succinate-to-butyrate/1_2.8.3.18_RXN-8807\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_1_2_8_3_18_RXN_8807 \"succinate-to-butyrate/1_2.8.3.18_RXN-8807\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_2_1_2_1_76_RXN_8891 \"succinate-to-butyrate/2_1.2.1.76_RXN-8891\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_2_1_2_1_76_RXN_8891 \"succinate-to-butyrate/2_1.2.1.76_RXN-8891\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_3_1_1_1_61_RXN_6903_WOP \"succinate-to-butyrate/3_1.1.1.61_RXN-6903_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_3_1_1_1_61_RXN_6903_WOP \"succinate-to-butyrate/3_1.1.1.61_RXN-6903_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_4a_2_8_3_RXN_8889 \"succinate-to-butyrate/4a_2.8.3.-_RXN-8889\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_4a_2_8_3_RXN_8889 \"succinate-to-butyrate/4a_2.8.3.-_RXN-8889\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_4b_manual_2_8_3_HD_4hbut \"succinate-to-butyrate/4b_manual_2.8.3.-_HD-4hbut\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_4b_manual_2_8_3_HD_4hbut \"succinate-to-butyrate/4b_manual_2.8.3.-_HD-4hbut\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_5a_4_2_1_120_RXN_8890 \"succinate-to-butyrate/5a_4.2.1.120_RXN-8890\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_5a_4_2_1_120_RXN_8890 \"succinate-to-butyrate/5a_4.2.1.120_RXN-8890\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_5b_moddir_4_2_1_120_BS403853 \"succinate-to-butyrate/5b_moddir_4.2.1.120_BS403853\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_5b_moddir_4_2_1_120_BS403853 \"succinate-to-butyrate/5b_moddir_4.2.1.120_BS403853\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_to_butyrate_6_5_3_3_3_VINYLACETYL_COA_DELTA_ISOMERASE_RXN \"succinate-to-butyrate/6_5.3.3.3_VINYLACETYL-COA-DELTA-ISOMERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_to_butyrate_6_5_3_3_3_VINYLACETYL_COA_DELTA_ISOMERASE_RXN \"succinate-to-butyrate/6_5.3.3.3_VINYLACETYL-COA-DELTA-ISOMERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetyl_CoA_to_butyrate_II_1_2_3_1_9_ACETYL_COA_ACETYLTRANSFER_RXN \"acetyl-CoA-to-butyrate-II/1_2.3.1.9_ACETYL-COA-ACETYLTRANSFER-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetyl_CoA_to_butyrate_II_1_2_3_1_9_ACETYL_COA_ACETYLTRANSFER_RXN \"acetyl-CoA-to-butyrate-II/1_2.3.1.9_ACETYL-COA-ACETYLTRANSFER-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetyl_CoA_to_butyrate_II_2_1_1_1_35_HD_1_1_1_35 \"acetyl-CoA-to-butyrate-II/2_1.1.1.35_HD-1.1.1.35\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetyl_CoA_to_butyrate_II_2_1_1_1_35_HD_1_1_1_35 \"acetyl-CoA-to-butyrate-II/2_1.1.1.35_HD-1.1.1.35\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetyl_CoA_to_butyrate_II_3_4_2_1_150_RXN_11667 \"acetyl-CoA-to-butyrate-II/3_4.2.1.150_RXN-11667\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetyl_CoA_to_butyrate_II_3_4_2_1_150_RXN_11667 \"acetyl-CoA-to-butyrate-II/3_4.2.1.150_RXN-11667\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction crotonyl_CoA_to_butanoate_1_manual_1_3_98_HD_1_3_98_but \"crotonyl-CoA-to-butanoate/1_manual_1.3.98.-_HD-1.3.98.-but\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction crotonyl_CoA_to_butanoate_1_manual_1_3_98_HD_1_3_98_but \"crotonyl-CoA-to-butanoate/1_manual_1.3.98.-_HD-1.3.98.-but\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction crotonyl_CoA_to_butanoate_2_2_3_1_19_PHOSPHATE_BUTYRYLTRANSFERASE_RXN \"crotonyl-CoA-to-butanoate/2_2.3.1.19_PHOSPHATE-BUTYRYLTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction crotonyl_CoA_to_butanoate_2_2_3_1_19_PHOSPHATE_BUTYRYLTRANSFERASE_RXN \"crotonyl-CoA-to-butanoate/2_2.3.1.19_PHOSPHATE-BUTYRYLTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction crotonyl_CoA_to_butanoate_3_moddir_2_7_2_7_BUTYRATE_KINASE_RXN \"crotonyl-CoA-to-butanoate/3_moddir_2.7.2.7_BUTYRATE-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction crotonyl_CoA_to_butanoate_3_moddir_2_7_2_7_BUTYRATE_KINASE_RXN \"crotonyl-CoA-to-butanoate/3_moddir_2.7.2.7_BUTYRATE-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_ferm_1a_2_8_3_9_2_8_3_9_RXN \"(R)-3-hydroxybutanoate-ferm/1a_2.8.3.9_2.8.3.9-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_ferm_1a_2_8_3_9_2_8_3_9_RXN \"(R)-3-hydroxybutanoate-ferm/1a_2.8.3.9_2.8.3.9-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_ferm_1b_2_8_3_9_ACETOACETYL_COA_TRANSFER_RXN \"(R)-3-hydroxybutanoate-ferm/1b_2.8.3.9_ACETOACETYL-COA-TRANSFER-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_ferm_1b_2_8_3_9_ACETOACETYL_COA_TRANSFER_RXN \"(R)-3-hydroxybutanoate-ferm/1b_2.8.3.9_ACETOACETYL-COA-TRANSFER-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_ferm_2_1_1_1_30_3_HYDROXYBUTYRATE_DEHYDROGENASE_RXN \"(R)-3-hydroxybutanoate-ferm/2_1.1.1.30_3-HYDROXYBUTYRATE-DEHYDROGENASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _R_3_hydroxybutanoate_ferm_2_1_1_1_30_3_HYDROXYBUTYRATE_DEHYDROGENASE_RXN \"(R)-3-hydroxybutanoate-ferm/2_1.1.1.30_3-HYDROXYBUTYRATE-DEHYDROGENASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetone_fermentation_spontaneous_ACETOACETATE_DECARBOXYLASE_RXN \"acetone-fermentation_spontaneous_ACETOACETATE-DECARBOXYLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetone_fermentation_spontaneous_ACETOACETATE_DECARBOXYLASE_RXN \"acetone-fermentation_spontaneous_ACETOACETATE-DECARBOXYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetone_fermentation_ex_spontaneous_ACETOACETATE_DECARBOXYLASE_RXN \"acetone-fermentation-ex_spontaneous_ACETOACETATE-DECARBOXYLASE-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetone_fermentation_ex_spontaneous_ACETOACETATE_DECARBOXYLASE_RXN \"acetone-fermentation-ex_spontaneous_ACETOACETATE-DECARBOXYLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_ferm_1_2_3_1_9_RXN_12561 \"pentanoate-ferm/1_2.3.1.9_RXN-12561\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_ferm_1_2_3_1_9_RXN_12561 \"pentanoate-ferm/1_2.3.1.9_RXN-12561\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_ferm_2_manual_1_1_1_157_HD_1_1_1_157_pent \"pentanoate-ferm/2_manual_1.1.1.157_HD-1.1.1.157-pent\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_ferm_2_manual_1_1_1_157_HD_1_1_1_157_pent \"pentanoate-ferm/2_manual_1.1.1.157_HD-1.1.1.157-pent\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_ferm_3_manual_4_2_1_150_HD_4_2_1_150_pent \"pentanoate-ferm/3_manual_4.2.1.150_HD-4.2.1.150-pent\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_ferm_3_manual_4_2_1_150_HD_4_2_1_150_pent \"pentanoate-ferm/3_manual_4.2.1.150_HD-4.2.1.150-pent\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_ferm_4_manual_1_3_98_HD_1_3_98_pent \"pentanoate-ferm/4_manual_1.3.98.-_HD-1.3.98.-pent\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_ferm_4_manual_1_3_98_HD_1_3_98_pent \"pentanoate-ferm/4_manual_1.3.98.-_HD-1.3.98.-pent\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_ferm_5_manual_2_3_1_19_HD_2_3_1_19_pent \"pentanoate-ferm/5_manual_2.3.1.19_HD-2.3.1.19-pent\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_ferm_5_manual_2_3_1_19_HD_2_3_1_19_pent \"pentanoate-ferm/5_manual_2.3.1.19_HD-2.3.1.19-pent\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pentanoate_ferm_6_manual_2_7_2_7_HD_2_7_2_7_pent \"pentanoate-ferm/6_manual_2.7.2.7_HD-2.7.2.7-pent\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pentanoate_ferm_6_manual_2_7_2_7_HD_2_7_2_7_pent \"pentanoate-ferm/6_manual_2.7.2.7_HD-2.7.2.7-pent\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_ferm_1_2_3_1_9_RXN_12565 \"hexanoate-ferm/1_2.3.1.9_RXN-12565\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_ferm_1_2_3_1_9_RXN_12565 \"hexanoate-ferm/1_2.3.1.9_RXN-12565\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_ferm_2_manual_1_1_1_157_HD_1_1_1_157_hex \"hexanoate-ferm/2_manual_1.1.1.157_HD-1.1.1.157-hex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_ferm_2_manual_1_1_1_157_HD_1_1_1_157_hex \"hexanoate-ferm/2_manual_1.1.1.157_HD-1.1.1.157-hex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_ferm_3_manual_4_2_1_150_HD_4_2_1_150_hex \"hexanoate-ferm/3_manual_4.2.1.150_HD-4.2.1.150-hex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_ferm_3_manual_4_2_1_150_HD_4_2_1_150_hex \"hexanoate-ferm/3_manual_4.2.1.150_HD-4.2.1.150-hex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_ferm_4_manual_1_3_98_HD_1_3_98_hex \"hexanoate-ferm/4_manual_1.3.98.-_HD-1.3.98.-hex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_ferm_4_manual_1_3_98_HD_1_3_98_hex \"hexanoate-ferm/4_manual_1.3.98.-_HD-1.3.98.-hex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_ferm_5_manual_2_3_1_19_HD_2_3_1_19_hex \"hexanoate-ferm/5_manual_2.3.1.19_HD-2.3.1.19-hex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_ferm_5_manual_2_3_1_19_HD_2_3_1_19_hex \"hexanoate-ferm/5_manual_2.3.1.19_HD-2.3.1.19-hex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hexanoate_ferm_6_manual_2_7_2_7_HD_2_7_2_7_hex \"hexanoate-ferm/6_manual_2.7.2.7_HD-2.7.2.7-hex\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hexanoate_ferm_6_manual_2_7_2_7_HD_2_7_2_7_hex \"hexanoate-ferm/6_manual_2.7.2.7_HD-2.7.2.7-hex\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_1_2_4_2_14_PRPPAMIDOTRANS_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/1_2.4.2.14_PRPPAMIDOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_1_2_4_2_14_PRPPAMIDOTRANS_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/1_2.4.2.14_PRPPAMIDOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_2_6_3_4_13_GLYRIBONUCSYN_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/2_6.3.4.13_GLYRIBONUCSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_2_6_3_4_13_GLYRIBONUCSYN_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/2_6.3.4.13_GLYRIBONUCSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_3_2_1_2_2_GART_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/3_2.1.2.2_GART-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_3_2_1_2_2_GART_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/3_2.1.2.2_GART-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_4_6_3_5_3_FGAMSYN_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/4_6.3.5.3_FGAMSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_4_6_3_5_3_FGAMSYN_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/4_6.3.5.3_FGAMSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_5_6_3_3_1_AIRS_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/5_6.3.3.1_AIRS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_aminoimidazole_ribonucleotide_biosyn_II_5_6_3_3_1_AIRS_RXN \"5-aminoimidazole-ribonucleotide-biosyn-II/5_6.3.3.1_AIRS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_1_manual_spontaneous_5_4_99_18_HD_5_4_99_18 \"inosine-5'-phosphate-biosyn/1_manual_spontaneous_5.4.99.18_HD-5.4.99.18\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_1_manual_spontaneous_5_4_99_18_HD_5_4_99_18 \"inosine-5'-phosphate-biosyn/1_manual_spontaneous_5.4.99.18_HD-5.4.99.18\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_2_5_4_99_18_RXN0_743 \"inosine-5'-phosphate-biosyn/2_5.4.99.18_RXN0-743\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_2_5_4_99_18_RXN0_743 \"inosine-5'-phosphate-biosyn/2_5.4.99.18_RXN0-743\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_3_6_3_2_6_SAICARSYN_RXN \"inosine-5'-phosphate-biosyn/3_6.3.2.6_SAICARSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_3_6_3_2_6_SAICARSYN_RXN \"inosine-5'-phosphate-biosyn/3_6.3.2.6_SAICARSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_4_4_3_2_2_AICARSYN_RXN \"inosine-5'-phosphate-biosyn/4_4.3.2.2_AICARSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_4_4_3_2_2_AICARSYN_RXN \"inosine-5'-phosphate-biosyn/4_4.3.2.2_AICARSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_5_2_1_2_3_AICARTRANSFORM_RXN \"inosine-5'-phosphate-biosyn/5_2.1.2.3_AICARTRANSFORM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_5_2_1_2_3_AICARTRANSFORM_RXN \"inosine-5'-phosphate-biosyn/5_2.1.2.3_AICARTRANSFORM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_6_moddir_3_5_4_10_2_1_2_3_IMPCYCLOHYDROLASE_RXN \"inosine-5'-phosphate-biosyn/6_moddir_3.5.4.10_2.1.2.3_IMPCYCLOHYDROLASE-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_5_phosphate_biosyn_6_moddir_3_5_4_10_2_1_2_3_IMPCYCLOHYDROLASE_RXN \"inosine-5'-phosphate-biosyn/6_moddir_3.5.4.10_2.1.2.3_IMPCYCLOHYDROLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_10_METHIONINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.10_METHIONINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_10_METHIONINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.10_METHIONINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_11_SERINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.11_SERINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_11_SERINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.11_SERINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_12_ASPARTATE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.12_ASPARTATE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_12_ASPARTATE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.12_ASPARTATE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_14_GLYCINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.14_GLYCINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_14_GLYCINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.14_GLYCINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_15_PROLINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.15_PROLINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_15_PROLINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.15_PROLINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_16_CYSTEINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.16_CYSTEINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_16_CYSTEINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.16_CYSTEINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_17_GLURS_RXN \"tRNA-charging_6.1.1.17_GLURS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_17_GLURS_RXN \"tRNA-charging_6.1.1.17_GLURS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_18_GLUTAMINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.18_GLUTAMINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_18_GLUTAMINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.18_GLUTAMINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_19_ARGININE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.19_ARGININE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_19_ARGININE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.19_ARGININE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_1_TYROSINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.1_TYROSINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_1_TYROSINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.1_TYROSINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_20_PHENYLALANINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.20_PHENYLALANINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_20_PHENYLALANINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.20_PHENYLALANINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_21_HISTIDINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.21_HISTIDINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_21_HISTIDINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.21_HISTIDINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_22_ASPARAGINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.22_ASPARAGINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_22_ASPARAGINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.22_ASPARAGINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_2_TRYPTOPHAN_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.2_TRYPTOPHAN--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_2_TRYPTOPHAN_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.2_TRYPTOPHAN--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_3_THREONINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.3_THREONINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_3_THREONINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.3_THREONINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_4_LEUCINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.4_LEUCINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_4_LEUCINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.4_LEUCINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_5_ISOLEUCINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.5_ISOLEUCINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_5_ISOLEUCINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.5_ISOLEUCINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_6_LYSINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.6_LYSINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_6_LYSINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.6_LYSINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_7_ALANINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.7_ALANINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_7_ALANINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.7_ALANINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_9_VALINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.9_VALINE--TRNA-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tRNA_charging_6_1_1_9_VALINE_TRNA_LIGASE_RXN \"tRNA-charging_6.1.1.9_VALINE--TRNA-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction selenocysteine_biosynthesis_6_1_1_11_RXN0_2161 \"selenocysteine-biosynthesis_6.1.1.11_RXN0-2161\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction selenocysteine_biosynthesis_6_1_1_11_RXN0_2161 \"selenocysteine-biosynthesis_6.1.1.11_RXN0-2161\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NAD_biosyn_I_1a_1_4_3_16_L_ASPARTATE_OXID_RXN \"NAD-biosyn-I/1a_1.4.3.16_L-ASPARTATE-OXID-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NAD_biosyn_I_1a_1_4_3_16_L_ASPARTATE_OXID_RXN \"NAD-biosyn-I/1a_1.4.3.16_L-ASPARTATE-OXID-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NAD_biosyn_I_1b_1_4_3_16_RXN_9772 \"NAD-biosyn-I/1b_1.4.3.16_RXN-9772\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NAD_biosyn_I_1b_1_4_3_16_RXN_9772 \"NAD-biosyn-I/1b_1.4.3.16_RXN-9772\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NAD_biosyn_I_2_2_5_1_72_QUINOLINATE_SYNTHA_RXN \"NAD-biosyn-I/2_2.5.1.72_QUINOLINATE-SYNTHA-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NAD_biosyn_I_2_2_5_1_72_QUINOLINATE_SYNTHA_RXN \"NAD-biosyn-I/2_2.5.1.72_QUINOLINATE-SYNTHA-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NAD_biosyn_I_3_2_4_2_19_QUINOPRIBOTRANS_RXN \"NAD-biosyn-I/3_2.4.2.19_QUINOPRIBOTRANS-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NAD_biosyn_I_3_2_4_2_19_QUINOPRIBOTRANS_RXN \"NAD-biosyn-I/3_2.4.2.19_QUINOPRIBOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NAD_biosyn_I_4_2_7_7_18_NICONUCADENYLYLTRAN_RXN \"NAD-biosyn-I/4_2.7.7.18_NICONUCADENYLYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NAD_biosyn_I_4_2_7_7_18_NICONUCADENYLYLTRAN_RXN \"NAD-biosyn-I/4_2.7.7.18_NICONUCADENYLYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NAD_biosyn_I_5_6_3_1_5_NAD_SYNTH_NH3_RXN \"NAD-biosyn-I/5_6.3.1.5_NAD-SYNTH-NH3-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NAD_biosyn_I_5_6_3_1_5_NAD_SYNTH_NH3_RXN \"NAD-biosyn-I/5_6.3.1.5_NAD-SYNTH-NH3-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADP_biosyn_2_7_1_23_NAD_KIN_RXN \"NADP-biosyn_2.7.1.23_NAD-KIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADP_biosyn_2_7_1_23_NAD_KIN_RXN \"NADP-biosyn_2.7.1.23_NAD-KIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADH_repair_spontaneous_RXN_12753 \"NADH-repair_spontaneous_RXN-12753\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADH_repair_spontaneous_RXN_12753 \"NADH-repair_spontaneous_RXN-12753\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADH_repair_spontaneous_RXN_12754 \"NADH-repair_spontaneous_RXN-12754\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADH_repair_spontaneous_RXN_12754 \"NADH-repair_spontaneous_RXN-12754\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADH_repair_5_1_99_6_RXN_12752 \"NADH-repair_5.1.99.6_RXN-12752\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADH_repair_5_1_99_6_RXN_12752 \"NADH-repair_5.1.99.6_RXN-12752\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADH_repair_4_2_1_136_RXN0_6727 \"NADH-repair_4.2.1.136_RXN0-6727\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADH_repair_4_2_1_136_RXN0_6727 \"NADH-repair_4.2.1.136_RXN0-6727\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADPH_repair_manual_spontaneous_HD_S_NADPHX \"NADPH-repair_manual_spontaneous_HD-(S)-NADPHX\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADPH_repair_manual_spontaneous_HD_S_NADPHX \"NADPH-repair_manual_spontaneous_HD-(S)-NADPHX\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADPH_repair_manual_spontaneous_HD_R_NADPHX \"NADPH-repair_manual_spontaneous_HD-(R)-NADPHX\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADPH_repair_manual_spontaneous_HD_R_NADPHX \"NADPH-repair_manual_spontaneous_HD-(R)-NADPHX\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADPH_repair_5_1_99_6_RXN_13142 \"NADPH-repair_5.1.99.6_RXN-13142\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADPH_repair_5_1_99_6_RXN_13142 \"NADPH-repair_5.1.99.6_RXN-13142\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NADPH_repair_4_2_1_136_RXN_13141 \"NADPH-repair_4.2.1.136_RXN-13141\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NADPH_repair_4_2_1_136_RXN_13141 \"NADPH-repair_4.2.1.136_RXN-13141\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_A_1_3_1_3_5_RXN_14227 \"nicotine-transitions-A/1_3.1.3.5_RXN-14227\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_A_1_3_1_3_5_RXN_14227 \"nicotine-transitions-A/1_3.1.3.5_RXN-14227\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_A_2_2_4_2_1_R02295 \"nicotine-transitions-A/2_2.4.2.1_R02295\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_A_2_2_4_2_1_R02295 \"nicotine-transitions-A/2_2.4.2.1_R02295\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_B_3_6_1_9_BS165396 \"nicotine-transitions-B_3.6.1.9_BS165396\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_B_3_6_1_9_BS165396 \"nicotine-transitions-B_3.6.1.9_BS165396\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_C_1_3_6_1_9_NADPYROPHOSPHAT_RXN \"nicotine-transitions-C/1_3.6.1.9_NADPYROPHOSPHAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_C_1_3_6_1_9_NADPYROPHOSPHAT_RXN \"nicotine-transitions-C/1_3.6.1.9_NADPYROPHOSPHAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_C_2_3_1_3_5_3_1_3_2_RXN_5841 \"nicotine-transitions-C/2_3.1.3.5_3.1.3.2_RXN-5841\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_C_2_3_1_3_5_3_1_3_2_RXN_5841 \"nicotine-transitions-C/2_3.1.3.5_3.1.3.2_RXN-5841\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_C_3_3_2_2_1_BS94861 \"nicotine-transitions-C/3_3.2.2.1_BS94861\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_C_3_3_2_2_1_BS94861 \"nicotine-transitions-C/3_3.2.2.1_BS94861\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_C_4a_3_5_1_19_NICOTINAMID_RXN \"nicotine-transitions-C/4a_3.5.1.19_NICOTINAMID-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_C_4a_3_5_1_19_NICOTINAMID_RXN \"nicotine-transitions-C/4a_3.5.1.19_NICOTINAMID-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_C_4b_2_4_2_1_RXN0_7092 \"nicotine-transitions-C/4b_2.4.2.1_RXN0-7092\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_C_4b_2_4_2_1_RXN0_7092 \"nicotine-transitions-C/4b_2.4.2.1_RXN0-7092\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction nicotine_transitions_C_5_2_7_7_18_2_7_7_1_RXN \"nicotine-transitions-C/5_2.7.7.18_2.7.7.1-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction nicotine_transitions_C_5_2_7_7_18_2_7_7_1_RXN \"nicotine-transitions-C/5_2.7.7.18_2.7.7.1-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_1_3_5_4_25_GTP_CYCLOHYDRO_II_RXN \"flavin-biosyn-I/1_3.5.4.25_GTP-CYCLOHYDRO-II-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_1_3_5_4_25_GTP_CYCLOHYDRO_II_RXN \"flavin-biosyn-I/1_3.5.4.25_GTP-CYCLOHYDRO-II-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_2_3_5_4_26_RIBOFLAVINSYNDEAM_RXN \"flavin-biosyn-I/2_3.5.4.26_RIBOFLAVINSYNDEAM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_2_3_5_4_26_RIBOFLAVINSYNDEAM_RXN \"flavin-biosyn-I/2_3.5.4.26_RIBOFLAVINSYNDEAM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_3_1_1_1_193_RIBOFLAVINSYNREDUC_RXN \"flavin-biosyn-I/3_1.1.1.193_RIBOFLAVINSYNREDUC-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_3_1_1_1_193_RIBOFLAVINSYNREDUC_RXN \"flavin-biosyn-I/3_1.1.1.193_RIBOFLAVINSYNREDUC-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_4_3_1_3_RIBOPHOSPHAT_RXN \"flavin-biosyn-I/4_3.1.3.-_RIBOPHOSPHAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_4_3_1_3_RIBOPHOSPHAT_RXN \"flavin-biosyn-I/4_3.1.3.-_RIBOPHOSPHAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_5_4_1_99_12_DIOHBUTANONEPSYN_RXN \"flavin-biosyn-I/5_4.1.99.12_DIOHBUTANONEPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_5_4_1_99_12_DIOHBUTANONEPSYN_RXN \"flavin-biosyn-I/5_4.1.99.12_DIOHBUTANONEPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_6_2_5_1_78_LUMAZINESYN_RXN \"flavin-biosyn-I/6_2.5.1.78_LUMAZINESYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_6_2_5_1_78_LUMAZINESYN_RXN \"flavin-biosyn-I/6_2.5.1.78_LUMAZINESYN-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_7_2_5_1_9_RIBOFLAVIN_SYN_RXN \"flavin-biosyn-I/7_2.5.1.9_RIBOFLAVIN-SYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_7_2_5_1_9_RIBOFLAVIN_SYN_RXN \"flavin-biosyn-I/7_2.5.1.9_RIBOFLAVIN-SYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_8_2_7_1_26_RIBOFLAVINKIN_RXN \"flavin-biosyn-I/8_2.7.1.26_RIBOFLAVINKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_8_2_7_1_26_RIBOFLAVINKIN_RXN \"flavin-biosyn-I/8_2.7.1.26_RIBOFLAVINKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction flavin_biosyn_I_9_2_7_7_2_FADSYN_RXN \"flavin-biosyn-I/9_2.7.7.2_FADSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction flavin_biosyn_I_9_2_7_7_2_FADSYN_RXN \"flavin-biosyn-I/9_2.7.7.2_FADSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_1_2_1_2_11_3_CH3_2_OXOBUTANOATE_OH_CH3_XFER_RXN \"phosphopantothenate-biosyn-I/1_2.1.2.11_3-CH3-2-OXOBUTANOATE-OH-CH3-XFER-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_1_2_1_2_11_3_CH3_2_OXOBUTANOATE_OH_CH3_XFER_RXN \"phosphopantothenate-biosyn-I/1_2.1.2.11_3-CH3-2-OXOBUTANOATE-OH-CH3-XFER-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_2_1_1_1_169_2_DEHYDROPANTOATE_REDUCT_RXN \"phosphopantothenate-biosyn-I/2_1.1.1.169_2-DEHYDROPANTOATE-REDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_2_1_1_1_169_2_DEHYDROPANTOATE_REDUCT_RXN \"phosphopantothenate-biosyn-I/2_1.1.1.169_2-DEHYDROPANTOATE-REDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_3_6_3_2_1_PANTOATE_BETA_ALANINE_LIG_RXN \"phosphopantothenate-biosyn-I/3_6.3.2.1_PANTOATE-BETA-ALANINE-LIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_3_6_3_2_1_PANTOATE_BETA_ALANINE_LIG_RXN \"phosphopantothenate-biosyn-I/3_6.3.2.1_PANTOATE-BETA-ALANINE-LIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_4_2_7_1_33_PANTOTHENATE_KIN_RXN \"phosphopantothenate-biosyn-I/4_2.7.1.33_PANTOTHENATE-KIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphopantothenate_biosyn_I_4_2_7_1_33_PANTOTHENATE_KIN_RXN \"phosphopantothenate-biosyn-I/4_2.7.1.33_PANTOTHENATE-KIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_1_6_3_2_5_P_PANTOCYSLIG_RXN \"coenzyme-A-biosyn/1_6.3.2.5_P-PANTOCYSLIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_1_6_3_2_5_P_PANTOCYSLIG_RXN \"coenzyme-A-biosyn/1_6.3.2.5_P-PANTOCYSLIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_2_4_1_1_36_P_PANTOCYSDECARB_RXN \"coenzyme-A-biosyn/2_4.1.1.36_P-PANTOCYSDECARB-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_2_4_1_1_36_P_PANTOCYSDECARB_RXN \"coenzyme-A-biosyn/2_4.1.1.36_P-PANTOCYSDECARB-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_3_2_7_7_3_PANTEPADENYLYLTRAN_RXN \"coenzyme-A-biosyn/3_2.7.7.3_PANTEPADENYLYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_3_2_7_7_3_PANTEPADENYLYLTRAN_RXN \"coenzyme-A-biosyn/3_2.7.7.3_PANTEPADENYLYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_4_2_7_1_24_DEPHOSPHOCOAKIN_RXN \"coenzyme-A-biosyn/4_2.7.1.24_DEPHOSPHOCOAKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction coenzyme_A_biosyn_4_2_7_1_24_DEPHOSPHOCOAKIN_RXN \"coenzyme-A-biosyn/4_2.7.1.24_DEPHOSPHOCOAKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_aminobenzoate_biosyn_1_2_6_1_85_PABASYN_RXN \"4-aminobenzoate-biosyn/1_2.6.1.85_PABASYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_aminobenzoate_biosyn_1_2_6_1_85_PABASYN_RXN \"4-aminobenzoate-biosyn/1_2.6.1.85_PABASYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_aminobenzoate_biosyn_2_4_1_3_38_ADCLY_RXN \"4-aminobenzoate-biosyn/2_4.1.3.38_ADCLY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_aminobenzoate_biosyn_2_4_1_3_38_ADCLY_RXN \"4-aminobenzoate-biosyn/2_4.1.3.38_ADCLY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_1_3_5_4_16_R00428 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/1_3.5.4.16_R00428\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_1_3_5_4_16_R00428 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/1_3.5.4.16_R00428\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_2_3_5_4_16_R05046 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/2_3.5.4.16_R05046\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_2_3_5_4_16_R05046 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/2_3.5.4.16_R05046\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_3_3_5_4_16_R05048 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/3_3.5.4.16_R05048\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_3_3_5_4_16_R05048 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/3_3.5.4.16_R05048\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_4_moddir_3_5_4_16_R04639 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/4_moddir_3.5.4.16_R04639\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_4_moddir_3_5_4_16_R04639 \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/4_moddir_3.5.4.16_R04639\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_5_3_6_1_67_H2NEOPTERINP3PYROPHOSPHOHYDRO_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/5_3.6.1.67_H2NEOPTERINP3PYROPHOSPHOHYDRO-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_5_3_6_1_67_H2NEOPTERINP3PYROPHOSPHOHYDRO_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/5_3.6.1.67_H2NEOPTERINP3PYROPHOSPHOHYDRO-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_6_3_6_1_DIHYDRONEOPTERIN_MONO_P_DEPHOS_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/6_3.6.1.-_DIHYDRONEOPTERIN-MONO-P-DEPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_6_3_6_1_DIHYDRONEOPTERIN_MONO_P_DEPHOS_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/6_3.6.1.-_DIHYDRONEOPTERIN-MONO-P-DEPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_7_4_1_2_25_H2NEOPTERINALDOL_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/7_4.1.2.25_H2NEOPTERINALDOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_7_4_1_2_25_H2NEOPTERINALDOL_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/7_4.1.2.25_H2NEOPTERINALDOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_8_2_7_6_3_H2PTERIDINEPYROPHOSPHOKIN_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/8_2.7.6.3_H2PTERIDINEPYROPHOSPHOKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _6_hydroxymethyl_dihydropterin_PP_biosyn_I_8_2_7_6_3_H2PTERIDINEPYROPHOSPHOKIN_RXN \"6-hydroxymethyl-dihydropterin-PP-biosyn-I/8_2.7.6.3_H2PTERIDINEPYROPHOSPHOKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrahydrofolate_biosynthesis_2_5_1_15_H2PTEROATESYNTH_RXN \"tetrahydrofolate-biosynthesis_2.5.1.15_H2PTEROATESYNTH-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrahydrofolate_biosynthesis_2_5_1_15_H2PTEROATESYNTH_RXN \"tetrahydrofolate-biosynthesis_2.5.1.15_H2PTEROATESYNTH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrahydrofolate_biosynthesis_6_3_2_17_DIHYDROFOLATESYNTH_RXN \"tetrahydrofolate-biosynthesis_6.3.2.17_DIHYDROFOLATESYNTH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrahydrofolate_biosynthesis_6_3_2_17_DIHYDROFOLATESYNTH_RXN \"tetrahydrofolate-biosynthesis_6.3.2.17_DIHYDROFOLATESYNTH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrahydrofolate_biosyn_3_manual_1_5_1_3_HD_1_5_1_3 \"tetrahydrofolate-biosyn/3_manual_1.5.1.3_HD-1.5.1.3\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrahydrofolate_biosyn_3_manual_1_5_1_3_HD_1_5_1_3 \"tetrahydrofolate-biosyn/3_manual_1.5.1.3_HD-1.5.1.3\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrahydrofolate_biosyn_4_manual_6_3_2_17_HD_6_3_2_17 \"tetrahydrofolate-biosyn/4_manual_6.3.2.17_HD-6.3.2.17\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrahydrofolate_biosyn_4_manual_6_3_2_17_HD_6_3_2_17 \"tetrahydrofolate-biosyn/4_manual_6.3.2.17_HD-6.3.2.17\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction folate_transformations_1_3_5_4_9_2_1_2_10_RXN_6321 \"folate-transformations/1_3.5.4.9_2.1.2.10_RXN-6321\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction folate_transformations_1_3_5_4_9_2_1_2_10_RXN_6321 \"folate-transformations/1_3.5.4.9_2.1.2.10_RXN-6321\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction folate_transformations_2_6_3_3_2_5_FORMYL_THF_CYCLO_LIGASE_RXN \"folate-transformations/2_6.3.3.2_5-FORMYL-THF-CYCLO-LIGASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction folate_transformations_2_6_3_3_2_5_FORMYL_THF_CYCLO_LIGASE_RXN \"folate-transformations/2_6.3.3.2_5-FORMYL-THF-CYCLO-LIGASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_1_2_2_1_7_DXS_RXN \"methylerythritol-phosphate-pathway/1_2.2.1.7_DXS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_1_2_2_1_7_DXS_RXN \"methylerythritol-phosphate-pathway/1_2.2.1.7_DXS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_2_1_1_1_267_DXPREDISOM_RXN \"methylerythritol-phosphate-pathway/2_1.1.1.267_DXPREDISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_2_1_1_1_267_DXPREDISOM_RXN \"methylerythritol-phosphate-pathway/2_1.1.1.267_DXPREDISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_3_2_7_7_60_2_7_7_60_RXN \"methylerythritol-phosphate-pathway/3_2.7.7.60_2.7.7.60-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_3_2_7_7_60_2_7_7_60_RXN \"methylerythritol-phosphate-pathway/3_2.7.7.60_2.7.7.60-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_4_2_7_1_148_2_7_1_148_RXN \"methylerythritol-phosphate-pathway/4_2.7.1.148_2.7.1.148-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_4_2_7_1_148_2_7_1_148_RXN \"methylerythritol-phosphate-pathway/4_2.7.1.148_2.7.1.148-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_5_4_6_1_12_RXN0_302 \"methylerythritol-phosphate-pathway/5_4.6.1.12_RXN0-302\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_5_4_6_1_12_RXN0_302 \"methylerythritol-phosphate-pathway/5_4.6.1.12_RXN0-302\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_6_1_17_7_1_RXN0_882 \"methylerythritol-phosphate-pathway/6_1.17.7.1_RXN0-882\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_6_1_17_7_1_RXN0_882 \"methylerythritol-phosphate-pathway/6_1.17.7.1_RXN0-882\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_7a_1_17_7_4_ISPH2_RXN \"methylerythritol-phosphate-pathway/7a_1.17.7.4_ISPH2-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_7a_1_17_7_4_ISPH2_RXN \"methylerythritol-phosphate-pathway/7a_1.17.7.4_ISPH2-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_7b_1_17_7_4_RXN0_884 \"methylerythritol-phosphate-pathway/7b_1.17.7.4_RXN0-884\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylerythritol_phosphate_pathway_7b_1_17_7_4_RXN0_884 \"methylerythritol-phosphate-pathway/7b_1.17.7.4_RXN0-884\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction farnesyl_PP_biosyn_1_2_5_1_1_GPPSYN_RXN \"farnesyl-PP-biosyn/1_2.5.1.1_GPPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction farnesyl_PP_biosyn_1_2_5_1_1_GPPSYN_RXN \"farnesyl-PP-biosyn/1_2.5.1.1_GPPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction farnesyl_PP_biosyn_2a_2_5_1_10_FPPSYN_RXN \"farnesyl-PP-biosyn/2a_2.5.1.10_FPPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction farnesyl_PP_biosyn_2a_2_5_1_10_FPPSYN_RXN \"farnesyl-PP-biosyn/2a_2.5.1.10_FPPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction farnesyl_PP_biosyn_2b_2_5_1_10_BS421953 \"farnesyl-PP-biosyn/2b_2.5.1.10_BS421953\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction farnesyl_PP_biosyn_2b_2_5_1_10_BS421953 \"farnesyl-PP-biosyn/2b_2.5.1.10_BS421953\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction undecaprenyl_PP_biosyn_2_5_1_31_RXN_8999 \"undecaprenyl-PP-biosyn_2.5.1.31_RXN-8999\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction undecaprenyl_PP_biosyn_2_5_1_31_RXN_8999 \"undecaprenyl-PP-biosyn_2.5.1.31_RXN-8999\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_amino_2_methyl_5_diphosphomethylpyrimidine_biosyn_1_4_1_99_17_PYRIMSYN1_RXN \"4-amino-2-methyl-5-diphosphomethylpyrimidine-biosyn/1_4.1.99.17_PYRIMSYN1-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_amino_2_methyl_5_diphosphomethylpyrimidine_biosyn_1_4_1_99_17_PYRIMSYN1_RXN \"4-amino-2-methyl-5-diphosphomethylpyrimidine-biosyn/1_4.1.99.17_PYRIMSYN1-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_amino_2_methyl_5_diphosphomethylpyrimidine_biosyn_2_2_7_4_7_PYRIMSYN3_RXN \"4-amino-2-methyl-5-diphosphomethylpyrimidine-biosyn/2_2.7.4.7_PYRIMSYN3-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_amino_2_methyl_5_diphosphomethylpyrimidine_biosyn_2_2_7_4_7_PYRIMSYN3_RXN \"4-amino-2-methyl-5-diphosphomethylpyrimidine-biosyn/2_2.7.4.7_PYRIMSYN3-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_deoxyadenosine_degr_3_2_2_16_3_2_2_1_3_2_2_9_RXN0_6550 \"5'-deoxyadenosine-degr_3.2.2.16_3.2.2.1_3.2.2.9_RXN0-6550\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_deoxyadenosine_degr_3_2_2_16_3_2_2_1_3_2_2_9_RXN0_6550 \"5'-deoxyadenosine-degr_3.2.2.16_3.2.2.1_3.2.2.9_RXN0-6550\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_1_2_8_1_7_RXN0_308 \"molybdenum-cofactor-biosynthesis/1_2.8.1.7_RXN0-308\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_1_2_8_1_7_RXN0_308 \"molybdenum-cofactor-biosynthesis/1_2.8.1.7_RXN0-308\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_2_2_7_7_80_RXN_11361 \"molybdenum-cofactor-biosynthesis/2_2.7.7.80_RXN-11361\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_2_2_7_7_80_RXN_11361 \"molybdenum-cofactor-biosynthesis/2_2.7.7.80_RXN-11361\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_3_mod_2_8_1_11_RXN_12473 \"molybdenum-cofactor-biosynthesis/3_mod_2.8.1.11_RXN-12473\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_3_mod_2_8_1_11_RXN_12473 \"molybdenum-cofactor-biosynthesis/3_mod_2.8.1.11_RXN-12473\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_4_4_1_99_18_RXN_8340 \"molybdenum-cofactor-biosynthesis/4_4.1.99.18_RXN-8340\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_4_4_1_99_18_RXN_8340 \"molybdenum-cofactor-biosynthesis/4_4.1.99.18_RXN-8340\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_5_2_8_1_12_RXN_8342 \"molybdenum-cofactor-biosynthesis/5_2.8.1.12_RXN-8342\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_5_2_8_1_12_RXN_8342 \"molybdenum-cofactor-biosynthesis/5_2.8.1.12_RXN-8342\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_6_2_7_7_75_RXN_8344 \"molybdenum-cofactor-biosynthesis/6_2.7.7.75_RXN-8344\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_6_2_7_7_75_RXN_8344 \"molybdenum-cofactor-biosynthesis/6_2.7.7.75_RXN-8344\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_7_2_10_1_1_RXN_8348 \"molybdenum-cofactor-biosynthesis/7_2.10.1.1_RXN-8348\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction molybdenum_cofactor_biosynthesis_7_2_10_1_1_RXN_8348 \"molybdenum-cofactor-biosynthesis/7_2.10.1.1_RXN-8348\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiazole_biosynthesis_1_2_8_1_7_RXN_9787 \"thiazole-biosynthesis/1_2.8.1.7_RXN-9787\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiazole_biosynthesis_1_2_8_1_7_RXN_9787 \"thiazole-biosynthesis/1_2.8.1.7_RXN-9787\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiazole_biosynthesis_2_2_7_7_73_RXN_9789 \"thiazole-biosynthesis/2_2.7.7.73_RXN-9789\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiazole_biosynthesis_2_2_7_7_73_RXN_9789 \"thiazole-biosynthesis/2_2.7.7.73_RXN-9789\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiazole_biosynthesis_3_2_8_1_RXN_9788 \"thiazole-biosynthesis/3_2.8.1.-_RXN-9788\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiazole_biosynthesis_3_2_8_1_RXN_9788 \"thiazole-biosynthesis/3_2.8.1.-_RXN-9788\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiazole_biosynthesis_p4_4_1_99_19_RXN_11319 \"thiazole-biosynthesis/p4_4.1.99.19_RXN-11319\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiazole_biosynthesis_p4_4_1_99_19_RXN_11319 \"thiazole-biosynthesis/p4_4.1.99.19_RXN-11319\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiazole_biosynthesis_4_2_8_1_10_THIAZOLSYN2_RXN \"thiazole-biosynthesis/4_2.8.1.10_THIAZOLSYN2-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiazole_biosynthesis_4_2_8_1_10_THIAZOLSYN2_RXN \"thiazole-biosynthesis/4_2.8.1.10_THIAZOLSYN2-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_diphosphate_biosyn_III_1_2_5_1_3_RXN_12611 \"thiamin-diphosphate-biosyn-III/1_2.5.1.3_RXN-12611\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_diphosphate_biosyn_III_1_2_5_1_3_RXN_12611 \"thiamin-diphosphate-biosyn-III/1_2.5.1.3_RXN-12611\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_diphosphate_biosyn_III_2_3_1_3_2_RXNQT_4191 \"thiamin-diphosphate-biosyn-III/2_3.1.3.2_RXNQT-4191\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_diphosphate_biosyn_III_2_3_1_3_2_RXNQT_4191 \"thiamin-diphosphate-biosyn-III/2_3.1.3.2_RXNQT-4191\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_diphosphate_biosyn_III_3_2_7_6_2_THIAMIN_PYROPHOSPHOKINASE_RXN \"thiamin-diphosphate-biosyn-III/3_2.7.6.2_THIAMIN-PYROPHOSPHOKINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_diphosphate_biosyn_III_3_2_7_6_2_THIAMIN_PYROPHOSPHOKINASE_RXN \"thiamin-diphosphate-biosyn-III/3_2.7.6.2_THIAMIN-PYROPHOSPHOKINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_salvage_1_3_5_99_2_THIAMINASE_RXN \"thiamin-salvage/1_3.5.99.2_THIAMINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_salvage_1_3_5_99_2_THIAMINASE_RXN \"thiamin-salvage/1_3.5.99.2_THIAMINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_salvage_2_2_7_1_50_THIAZOLSYN3_RXN \"thiamin-salvage/2_2.7.1.50_THIAZOLSYN3-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_salvage_2_2_7_1_50_THIAZOLSYN3_RXN \"thiamin-salvage/2_2.7.1.50_THIAZOLSYN3-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_salvage_p3_2_7_1_49_OHMETPYRKIN_RXN \"thiamin-salvage/p3_2.7.1.49_OHMETPYRKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_salvage_p3_2_7_1_49_OHMETPYRKIN_RXN \"thiamin-salvage/p3_2.7.1.49_OHMETPYRKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiamin_salvage_4_2_5_1_3_THI_P_SYN_RXN \"thiamin-salvage/4_2.5.1.3_THI-P-SYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiamin_salvage_4_2_5_1_3_THI_P_SYN_RXN \"thiamin-salvage/4_2.5.1.3_THI-P-SYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_1_1_2_1_70_GLUTRNAREDUCT_RXN \"tetrapyrrole-biosynthesis/1_1.2.1.70_GLUTRNAREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_1_1_2_1_70_GLUTRNAREDUCT_RXN \"tetrapyrrole-biosynthesis/1_1.2.1.70_GLUTRNAREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_2_5_4_3_8_GSAAMINOTRANS_RXN \"tetrapyrrole-biosynthesis/2_5.4.3.8_GSAAMINOTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_2_5_4_3_8_GSAAMINOTRANS_RXN \"tetrapyrrole-biosynthesis/2_5.4.3.8_GSAAMINOTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_3_4_2_1_24_PORPHOBILSYNTH_RXN \"tetrapyrrole-biosynthesis/3_4.2.1.24_PORPHOBILSYNTH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_3_4_2_1_24_PORPHOBILSYNTH_RXN \"tetrapyrrole-biosynthesis/3_4.2.1.24_PORPHOBILSYNTH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_4_2_5_1_61_OHMETHYLBILANESYN_RXN \"tetrapyrrole-biosynthesis/4_2.5.1.61_OHMETHYLBILANESYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_4_2_5_1_61_OHMETHYLBILANESYN_RXN \"tetrapyrrole-biosynthesis/4_2.5.1.61_OHMETHYLBILANESYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_5_4_2_1_75_UROGENIIISYN_RXN \"tetrapyrrole-biosynthesis/5_4.2.1.75_UROGENIIISYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tetrapyrrole_biosynthesis_5_4_2_1_75_UROGENIIISYN_RXN \"tetrapyrrole-biosynthesis/5_4.2.1.75_UROGENIIISYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_1_2_1_1_107_UROPORIIIMETHYLTRANSA_RXN \"cobyrinate-biosyn-I/1_2.1.1.107_UROPORIIIMETHYLTRANSA-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_1_2_1_1_107_UROPORIIIMETHYLTRANSA_RXN \"cobyrinate-biosyn-I/1_2.1.1.107_UROPORIIIMETHYLTRANSA-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_2_2_1_1_107_RXN_8675 \"cobyrinate-biosyn-I/2_2.1.1.107_RXN-8675\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_2_2_1_1_107_RXN_8675 \"cobyrinate-biosyn-I/2_2.1.1.107_RXN-8675\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_3_1_3_1_76_DIMETHUROPORDEHYDROG_RXN \"cobyrinate-biosyn-I/3_1.3.1.76_DIMETHUROPORDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_3_1_3_1_76_DIMETHUROPORDEHYDROG_RXN \"cobyrinate-biosyn-I/3_1.3.1.76_DIMETHUROPORDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_4_4_99_1_3_4_99_1_3_RXN \"cobyrinate-biosyn-I/4_4.99.1.3_4.99.1.3-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_4_4_99_1_3_4_99_1_3_RXN \"cobyrinate-biosyn-I/4_4.99.1.3_4.99.1.3-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_5_2_1_1_151_2_1_1_151_RXN \"cobyrinate-biosyn-I/5_2.1.1.151_2.1.1.151-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_5_2_1_1_151_2_1_1_151_RXN \"cobyrinate-biosyn-I/5_2.1.1.151_2.1.1.151-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_6_mod_2_1_1_272_RXN_14314 \"cobyrinate-biosyn-I/6_mod_2.1.1.272_RXN-14314\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_6_mod_2_1_1_272_RXN_14314 \"cobyrinate-biosyn-I/6_mod_2.1.1.272_RXN-14314\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_7_2_1_1_271_RXN_8762 \"cobyrinate-biosyn-I/7_2.1.1.271_RXN-8762\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_7_2_1_1_271_RXN_8762 \"cobyrinate-biosyn-I/7_2.1.1.271_RXN-8762\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_8_3_7_1_12_RXN_8763 \"cobyrinate-biosyn-I/8_3.7.1.12_RXN-8763\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_8_3_7_1_12_RXN_8763 \"cobyrinate-biosyn-I/8_3.7.1.12_RXN-8763\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_9_2_1_1_195_RXN_8764 \"cobyrinate-biosyn-I/9_2.1.1.195_RXN-8764\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_9_2_1_1_195_RXN_8764 \"cobyrinate-biosyn-I/9_2.1.1.195_RXN-8764\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_10_1_3_1_106_RXN_8765 \"cobyrinate-biosyn-I/10_1.3.1.106_RXN-8765\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_10_1_3_1_106_RXN_8765 \"cobyrinate-biosyn-I/10_1.3.1.106_RXN-8765\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_11_2_1_1_196_RXN_8766 \"cobyrinate-biosyn-I/11_2.1.1.196_RXN-8766\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_11_2_1_1_196_RXN_8766 \"cobyrinate-biosyn-I/11_2.1.1.196_RXN-8766\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_12_2_1_1_289_RXN_8767 \"cobyrinate-biosyn-I/12_2.1.1.289_RXN-8767\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_12_2_1_1_289_RXN_8767 \"cobyrinate-biosyn-I/12_2.1.1.289_RXN-8767\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_13_5_4_99_60_RXN_8768 \"cobyrinate-biosyn-I/13_5.4.99.60_RXN-8768\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_13_5_4_99_60_RXN_8768 \"cobyrinate-biosyn-I/13_5.4.99.60_RXN-8768\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_14_6_3_5_11_RXN_14256 \"cobyrinate-biosyn-I/14_6.3.5.11_RXN-14256\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_14_6_3_5_11_RXN_14256 \"cobyrinate-biosyn-I/14_6.3.5.11_RXN-14256\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_15_6_3_5_11_RXN_14257 \"cobyrinate-biosyn-I/15_6.3.5.11_RXN-14257\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobyrinate_biosyn_I_15_6_3_5_11_RXN_14257 \"cobyrinate-biosyn-I/15_6.3.5.11_RXN-14257\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_1_4_1_99_M1_RXN_17122 \"5,6-dimethylbenzimidazole-biosyn-II/1_4.1.99.M1_RXN-17122\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_1_4_1_99_M1_RXN_17122 \"5,6-dimethylbenzimidazole-biosyn-II/1_4.1.99.M1_RXN-17122\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_2_2_1_1_M9_RXN_17123 \"5,6-dimethylbenzimidazole-biosyn-II/2_2.1.1.M9_RXN-17123\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_2_2_1_1_M9_RXN_17123 \"5,6-dimethylbenzimidazole-biosyn-II/2_2.1.1.M9_RXN-17123\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_3_2_1_1_M10_RXN_17124 \"5,6-dimethylbenzimidazole-biosyn-II/3_2.1.1.M10_RXN-17124\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_3_2_1_1_M10_RXN_17124 \"5,6-dimethylbenzimidazole-biosyn-II/3_2.1.1.M10_RXN-17124\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_4_unknown_RXN_17125 \"5,6-dimethylbenzimidazole-biosyn-II/4_unknown_RXN-17125\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _5_6_dimethylbenzimidazole_biosyn_II_4_unknown_RXN_17125 \"5,6-dimethylbenzimidazole-biosyn-II/4_unknown_RXN-17125\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_1_mod_1_16_8_1_R343_RXN \"adenosylcobalamin-biosynthesis-I/1_mod_1.16.8.1_R343-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_1_mod_1_16_8_1_R343_RXN \"adenosylcobalamin-biosynthesis-I/1_mod_1.16.8.1_R343-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_2_2_5_1_17_R344_RXN \"adenosylcobalamin-biosynthesis-I/2_2.5.1.17_R344-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_2_2_5_1_17_R344_RXN \"adenosylcobalamin-biosynthesis-I/2_2.5.1.17_R344-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_3_6_3_5_10_R345_RXN \"adenosylcobalamin-biosynthesis-I/3_6.3.5.10_R345-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_3_6_3_5_10_R345_RXN \"adenosylcobalamin-biosynthesis-I/3_6.3.5.10_R345-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_pp4_2_7_1_177_RXN_8626 \"adenosylcobalamin-biosynthesis-I/pp4_2.7.1.177_RXN-8626\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_pp4_2_7_1_177_RXN_8626 \"adenosylcobalamin-biosynthesis-I/pp4_2.7.1.177_RXN-8626\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_p4_4_1_1_81_4_1_1_81_RXN \"adenosylcobalamin-biosynthesis-I/p4_4.1.1.81_4.1.1.81-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_p4_4_1_1_81_4_1_1_81_RXN \"adenosylcobalamin-biosynthesis-I/p4_4.1.1.81_4.1.1.81-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_4_6_3_1_10_RXN_6261 \"adenosylcobalamin-biosynthesis-I/4_6.3.1.10_RXN-6261\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_4_6_3_1_10_RXN_6261 \"adenosylcobalamin-biosynthesis-I/4_6.3.1.10_RXN-6261\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_5_2_7_7_62_COBINPGUANYLYLTRANS_RXN \"adenosylcobalamin-biosynthesis-I/5_2.7.7.62_COBINPGUANYLYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_5_2_7_7_62_COBINPGUANYLYLTRANS_RXN \"adenosylcobalamin-biosynthesis-I/5_2.7.7.62_COBINPGUANYLYLTRANS-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_pp6_6_3_4_21_NICOTINATEPRIBOSYLTRANS_RXN \"adenosylcobalamin-biosynthesis-I/pp6_6.3.4.21_NICOTINATEPRIBOSYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_pp6_6_3_4_21_NICOTINATEPRIBOSYLTRANS_RXN \"adenosylcobalamin-biosynthesis-I/pp6_6.3.4.21_NICOTINATEPRIBOSYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_p6_2_4_2_21_DMBPPRIBOSYLTRANS_RXN \"adenosylcobalamin-biosynthesis-I/p6_2.4.2.21_DMBPPRIBOSYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_p6_2_4_2_21_DMBPPRIBOSYLTRANS_RXN \"adenosylcobalamin-biosynthesis-I/p6_2.4.2.21_DMBPPRIBOSYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_6a_2_7_8_26_COBALAMIN5PSYN_RXN \"adenosylcobalamin-biosynthesis-I/6a_2.7.8.26_COBALAMIN5PSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_6a_2_7_8_26_COBALAMIN5PSYN_RXN \"adenosylcobalamin-biosynthesis-I/6a_2.7.8.26_COBALAMIN5PSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_7a_3_1_3_73_RXN_8770 \"adenosylcobalamin-biosynthesis-I/7a_3.1.3.73_RXN-8770\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_7a_3_1_3_73_RXN_8770 \"adenosylcobalamin-biosynthesis-I/7a_3.1.3.73_RXN-8770\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_6b_3_1_3_73_RIBAZOLEPHOSPHAT_RXN \"adenosylcobalamin-biosynthesis-I/6b_3.1.3.73_RIBAZOLEPHOSPHAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_6b_3_1_3_73_RIBAZOLEPHOSPHAT_RXN \"adenosylcobalamin-biosynthesis-I/6b_3.1.3.73_RIBAZOLEPHOSPHAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_7b_2_7_8_26_COBALAMINSYN_RXN \"adenosylcobalamin-biosynthesis-I/7b_2.7.8.26_COBALAMINSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_biosynthesis_I_7b_2_7_8_26_COBALAMINSYN_RXN \"adenosylcobalamin-biosynthesis-I/7b_2.7.8.26_COBALAMINSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction protoporphyrin_biosyn_1_4_1_1_37_UROGENDECARBOX_RXN \"protoporphyrin-biosyn/1_4.1.1.37_UROGENDECARBOX-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction protoporphyrin_biosyn_1_4_1_1_37_UROGENDECARBOX_RXN \"protoporphyrin-biosyn/1_4.1.1.37_UROGENDECARBOX-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction protoporphyrin_biosyn_2_modnames_1_3_99_22_HEMN_RXN \"protoporphyrin-biosyn/2_modnames_1.3.99.22_HEMN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction protoporphyrin_biosyn_2_modnames_1_3_99_22_HEMN_RXN \"protoporphyrin-biosyn/2_modnames_1.3.99.22_HEMN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction protoporphyrin_biosyn_3_6_6_1_1_RXN1F_20 \"protoporphyrin-biosyn/3_6.6.1.1_RXN1F-20\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction protoporphyrin_biosyn_3_6_6_1_1_RXN1F_20 \"protoporphyrin-biosyn/3_6.6.1.1_RXN1F-20\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_carboxyl_carrier_protein_assembly_1_6_3_4_15_BIOTINLIG_RXN \"biotin-carboxyl-carrier-protein-assembly/1_6.3.4.15_BIOTINLIG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_carboxyl_carrier_protein_assembly_1_6_3_4_15_BIOTINLIG_RXN \"biotin-carboxyl-carrier-protein-assembly/1_6.3.4.15_BIOTINLIG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_carboxyl_carrier_protein_assembly_2_spontaneous_RXN_7101 \"biotin-carboxyl-carrier-protein-assembly/2_spontaneous_RXN-7101\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_carboxyl_carrier_protein_assembly_2_spontaneous_RXN_7101 \"biotin-carboxyl-carrier-protein-assembly/2_spontaneous_RXN-7101\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_carboxyl_carrier_protein_assembly_3_6_3_4_14_BIOTIN_CARBOXYL_RXN \"biotin-carboxyl-carrier-protein-assembly/3_6.3.4.14_BIOTIN-CARBOXYL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_carboxyl_carrier_protein_assembly_3_6_3_4_14_BIOTIN_CARBOXYL_RXN \"biotin-carboxyl-carrier-protein-assembly/3_6.3.4.14_BIOTIN-CARBOXYL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction PLP_biosyn_2_7_1_49_PNKIN_RXN \"PLP-biosyn_2.7.1.49_PNKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction PLP_biosyn_2_7_1_49_PNKIN_RXN \"PLP-biosyn_2.7.1.49_PNKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_threonylcarbamoyladenylate_2_7_7_87_RXN_14569 \"L-threonylcarbamoyladenylate_2.7.7.87_RXN-14569\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_threonylcarbamoyladenylate_2_7_7_87_RXN_14569 \"L-threonylcarbamoyladenylate_2.7.7.87_RXN-14569\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction diphosphate_3_6_1_1_INORGPYROPHOSPHAT_RXN \"diphosphate_3.6.1.1_INORGPYROPHOSPHAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction diphosphate_3_6_1_1_INORGPYROPHOSPHAT_RXN \"diphosphate_3.6.1.1_INORGPYROPHOSPHAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction triphosphate_3_6_1_1_modb_BS383385 \"triphosphate_3.6.1.1_modb_BS383385\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction triphosphate_3_6_1_1_modb_BS383385 \"triphosphate_3.6.1.1_modb_BS383385\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction diphosphate_2_7_11_32_RXN_13757 \"diphosphate_2.7.11.32_RXN-13757\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction diphosphate_2_7_11_32_RXN_13757 \"diphosphate_2.7.11.32_RXN-13757\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction diphosphate_2_7_4_27_RXN_13756 \"diphosphate_2.7.4.27_RXN-13756\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction diphosphate_2_7_4_27_RXN_13756 \"diphosphate_2.7.4.27_RXN-13756\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_ribonucleotides_de_novo_biosyn_1_6_3_4_4_ADENYLOSUCCINATE_SYNTHASE_RXN \"adenosine-ribonucleotides-de-novo-biosyn/1_6.3.4.4_ADENYLOSUCCINATE-SYNTHASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_ribonucleotides_de_novo_biosyn_1_6_3_4_4_ADENYLOSUCCINATE_SYNTHASE_RXN \"adenosine-ribonucleotides-de-novo-biosyn/1_6.3.4.4_ADENYLOSUCCINATE-SYNTHASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_ribonucleotides_de_novo_biosyn_2_4_3_2_2_AMPSYN_RXN \"adenosine-ribonucleotides-de-novo-biosyn/2_4.3.2.2_AMPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_ribonucleotides_de_novo_biosyn_2_4_3_2_2_AMPSYN_RXN \"adenosine-ribonucleotides-de-novo-biosyn/2_4.3.2.2_AMPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction guanosine_ribonucleotides_de_novo_biosyn_1_1_1_1_205_IMP_DEHYDROG_RXN \"guanosine-ribonucleotides-de-novo-biosyn/1_1.1.1.205_IMP-DEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction guanosine_ribonucleotides_de_novo_biosyn_1_1_1_1_205_IMP_DEHYDROG_RXN \"guanosine-ribonucleotides-de-novo-biosyn/1_1.1.1.205_IMP-DEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction guanosine_ribonucleotides_de_novo_biosyn_2_6_3_5_2_GMP_SYN_GLUT_RXN \"guanosine-ribonucleotides-de-novo-biosyn/2_6.3.5.2_GMP-SYN-GLUT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction guanosine_ribonucleotides_de_novo_biosyn_2_6_3_5_2_GMP_SYN_GLUT_RXN \"guanosine-ribonucleotides-de-novo-biosyn/2_6.3.5.2_GMP-SYN-GLUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_biosyn_1_2_1_3_2_ASPCARBTRANS_RXN \"UMP-biosyn/1_2.1.3.2_ASPCARBTRANS-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_biosyn_1_2_1_3_2_ASPCARBTRANS_RXN \"UMP-biosyn/1_2.1.3.2_ASPCARBTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_biosyn_2_moddir_3_5_2_3_DIHYDROOROT_RXN \"UMP-biosyn/2_moddir_3.5.2.3_DIHYDROOROT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_biosyn_2_moddir_3_5_2_3_DIHYDROOROT_RXN \"UMP-biosyn/2_moddir_3.5.2.3_DIHYDROOROT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_biosyn_3_1_3_1_14_OROTATE_REDUCTASE_NADH_RXN \"UMP-biosyn/3_1.3.1.14_OROTATE-REDUCTASE-NADH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_biosyn_3_1_3_1_14_OROTATE_REDUCTASE_NADH_RXN \"UMP-biosyn/3_1.3.1.14_OROTATE-REDUCTASE-NADH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_biosyn_4_2_4_2_10_OROPRIBTRANS_RXN \"UMP-biosyn/4_2.4.2.10_OROPRIBTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_biosyn_4_2_4_2_10_OROPRIBTRANS_RXN \"UMP-biosyn/4_2.4.2.10_OROPRIBTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_biosyn_5_4_1_1_23_OROTPDECARB_RXN \"UMP-biosyn/5_4.1.1.23_OROTPDECARB-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_biosyn_5_4_1_1_23_OROTPDECARB_RXN \"UMP-biosyn/5_4.1.1.23_OROTPDECARB-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CTP_biosy2_6_3_4_2_CTPSYN_RXN \"CTP-biosy2_6.3.4.2_CTPSYN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CTP_biosy2_6_3_4_2_CTPSYN_RXN \"CTP-biosy2_6.3.4.2_CTPSYN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UTP_biosyn_3_5_4_5_RXN_14118 \"UTP-biosyn_3.5.4.5_RXN-14118\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UTP_biosyn_3_5_4_5_RXN_14118 \"UTP-biosyn_3.5.4.5_RXN-14118\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_ADPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_ADPREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_ADPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_ADPREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_GDPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_GDPREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_GDPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_GDPREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_CDPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_CDPREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_CDPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_CDPREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_UDPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_UDPREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_1_17_4_1_UDPREDUCT_RXN \"deoxyribonucleotides-biosyn_1.17.4.1_UDPREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_723 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-723\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_723 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-723\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_724 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-724\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_724 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-724\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_745 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-745\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_745 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-745\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_746 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-746\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction deoxyribonucleotides_biosyn_mod_1_17_4_2_RXN0_746 \"deoxyribonucleotides-biosyn_mod_1.17.4.2_RXN0-746\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dTTP_biosyn_1_3_6_1_23_DUTP_PYROP_RXN \"dTTP-biosyn/1_3.6.1.23_DUTP-PYROP-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dTTP_biosyn_1_3_6_1_23_DUTP_PYROP_RXN \"dTTP-biosyn/1_3.6.1.23_DUTP-PYROP-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dTTP_biosyn_2_2_1_1_148_RXN_8850 \"dTTP-biosyn/2_2.1.1.148_RXN-8850\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dTTP_biosyn_2_2_1_1_148_RXN_8850 \"dTTP-biosyn/2_2.1.1.148_RXN-8850\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CMP_salvage_1_3_1_3_5_RXN_14026 \"CMP-salvage/1_3.1.3.5_RXN-14026\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CMP_salvage_1_3_1_3_5_RXN_14026 \"CMP-salvage/1_3.1.3.5_RXN-14026\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CMP_salvage_2a_2_7_1_48_CYTIKIN_RXN \"CMP-salvage/2a_2.7.1.48_CYTIKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CMP_salvage_2a_2_7_1_48_CYTIKIN_RXN \"CMP-salvage/2a_2.7.1.48_CYTIKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CMP_salvage_2b_3_5_4_5_CYTIDEAM2_RXN \"CMP-salvage/2b_3.5.4.5_CYTIDEAM2-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CMP_salvage_2b_3_5_4_5_CYTIDEAM2_RXN \"CMP-salvage/2b_3.5.4.5_CYTIDEAM2-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction CMP_salvage_2c_2_4_2_2_BS355365 \"CMP-salvage/2c_2.4.2.2_BS355365\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction CMP_salvage_2c_2_4_2_2_BS355365 \"CMP-salvage/2c_2.4.2.2_BS355365\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_salvage_1_3_1_3_5_3_1_3_2_RXN_14025 \"UMP-salvage/1_3.1.3.5_3.1.3.2_RXN-14025\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_salvage_1_3_1_3_5_3_1_3_2_RXN_14025 \"UMP-salvage/1_3.1.3.5_3.1.3.2_RXN-14025\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_salvage_2a_2_7_1_48_URIDINEKIN_RXN \"UMP-salvage/2a_2.7.1.48_URIDINEKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_salvage_2a_2_7_1_48_URIDINEKIN_RXN \"UMP-salvage/2a_2.7.1.48_URIDINEKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_salvage_2b_2_4_2_2_URPHOS_RXN \"UMP-salvage/2b_2.4.2.2_URPHOS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_salvage_2b_2_4_2_2_URPHOS_RXN \"UMP-salvage/2b_2.4.2.2_URPHOS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction UMP_salvage_3_2_4_2_9_URACIL_PRIBOSYLTRANS_RXN \"UMP-salvage/3_2.4.2.9_URACIL-PRIBOSYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction UMP_salvage_3_2_4_2_9_URACIL_PRIBOSYLTRANS_RXN \"UMP-salvage/3_2.4.2.9_URACIL-PRIBOSYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction AMP_salvage_1_3_1_3_5_3_1_3_2_AMP_DEPHOSPHORYLATION_RXN \"AMP-salvage/1_3.1.3.5_3.1.3.2_AMP-DEPHOSPHORYLATION-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction AMP_salvage_1_3_1_3_5_3_1_3_2_AMP_DEPHOSPHORYLATION_RXN \"AMP-salvage/1_3.1.3.5_3.1.3.2_AMP-DEPHOSPHORYLATION-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction AMP_salvage_2_2_4_2_1_ADENPHOSPHOR_RXN \"AMP-salvage/2_2.4.2.1_ADENPHOSPHOR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction AMP_salvage_2_2_4_2_1_ADENPHOSPHOR_RXN \"AMP-salvage/2_2.4.2.1_ADENPHOSPHOR-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction AMP_salvage_3_2_4_2_8_2_4_2_7_ADENPRIBOSYLTRAN_RXN \"AMP-salvage/3_2.4.2.8_2.4.2.7_ADENPRIBOSYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction AMP_salvage_3_2_4_2_8_2_4_2_7_ADENPRIBOSYLTRAN_RXN \"AMP-salvage/3_2.4.2.8_2.4.2.7_ADENPRIBOSYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GMP_salvage_1_3_1_3_5_3_1_3_2_RXN_7609 \"GMP-salvage/1_3.1.3.5_3.1.3.2_RXN-7609\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GMP_salvage_1_3_1_3_5_3_1_3_2_RXN_7609 \"GMP-salvage/1_3.1.3.5_3.1.3.2_RXN-7609\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GMP_salvage_2_2_4_2_1_RXN0_5199 \"GMP-salvage/2_2.4.2.1_RXN0-5199\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GMP_salvage_2_2_4_2_1_RXN0_5199 \"GMP-salvage/2_2.4.2.1_RXN0-5199\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GMP_salvage_3_3_5_4_3_GUANINE_DEAMINASE_RXN \"GMP-salvage/3_3.5.4.3_GUANINE-DEAMINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GMP_salvage_3_3_5_4_3_GUANINE_DEAMINASE_RXN \"GMP-salvage/3_3.5.4.3_GUANINE-DEAMINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction GMP_salvage_4_2_4_2_8_2_4_2_22_XANPRIBOSYLTRAN_RXN \"GMP-salvage/4_2.4.2.8_2.4.2.22_XANPRIBOSYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction GMP_salvage_4_2_4_2_8_2_4_2_22_XANPRIBOSYLTRAN_RXN \"GMP-salvage/4_2.4.2.8_2.4.2.22_XANPRIBOSYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction XMP_salvage_1_3_1_3_5_XMPXAN_RXN \"XMP-salvage/1_3.1.3.5_XMPXAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction XMP_salvage_1_3_1_3_5_XMPXAN_RXN \"XMP-salvage/1_3.1.3.5_XMPXAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction XMP_salvage_2_2_4_2_1_XANTHOSINEPHOSPHORY_RXN \"XMP-salvage/2_2.4.2.1_XANTHOSINEPHOSPHORY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction XMP_salvage_2_2_4_2_1_XANTHOSINEPHOSPHORY_RXN \"XMP-salvage/2_2.4.2.1_XANTHOSINEPHOSPHORY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction IMP_salvage_3_1_3_5_RXN_7607 \"IMP-salvage_3.1.3.5_RXN-7607\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction IMP_salvage_3_1_3_5_RXN_7607 \"IMP-salvage_3.1.3.5_RXN-7607\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_pa1_3_1_3_5_3_1_3_2_BS73811 \"adenosine-salvage/pa1_3.1.3.5_3.1.3.2_BS73811\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_pa1_3_1_3_5_3_1_3_2_BS73811 \"adenosine-salvage/pa1_3.1.3.5_3.1.3.2_BS73811\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_pb1_3_1_3_5_BS73810 \"adenosine-salvage/pb1_3.1.3.5_BS73810\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_pb1_3_1_3_5_BS73810 \"adenosine-salvage/pb1_3.1.3.5_BS73810\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_1_3_5_4_4_ADENODEAMIN_RXN \"adenosine-salvage/1_3.5.4.4_ADENODEAMIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_1_3_5_4_4_ADENODEAMIN_RXN \"adenosine-salvage/1_3.5.4.4_ADENODEAMIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_2_2_4_2_1_INOPHOSPHOR_RXN \"adenosine-salvage/2_2.4.2.1_INOPHOSPHOR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_2_2_4_2_1_INOPHOSPHOR_RXN \"adenosine-salvage/2_2.4.2.1_INOPHOSPHOR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_3_1_17_1_4_RXN_7682 \"adenosine-salvage/3_1.17.1.4_RXN-7682\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_3_1_17_1_4_RXN_7682 \"adenosine-salvage/3_1.17.1.4_RXN-7682\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dCMP_salvage_1a_3_1_3_5_RXN0_5292 \"dCMP-salvage/1a_3.1.3.5_RXN0-5292\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dCMP_salvage_1a_3_1_3_5_RXN0_5292 \"dCMP-salvage/1a_3.1.3.5_RXN0-5292\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dCMP_salvage_2_3_5_4_5_CYTIDEAM_RXN \"dCMP-salvage/2_3.5.4.5_CYTIDEAM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dCMP_salvage_2_3_5_4_5_CYTIDEAM_RXN \"dCMP-salvage/2_3.5.4.5_CYTIDEAM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dCMP_salvage_11b_3_5_4_12_DCMP_DEAMINASE_RXN \"dCMP-salvage/11b_3.5.4.12_DCMP-DEAMINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dCMP_salvage_11b_3_5_4_12_DCMP_DEAMINASE_RXN \"dCMP-salvage/11b_3.5.4.12_DCMP-DEAMINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dUMP_salvage_1_3_1_3_5_RXN_14143 \"dUMP-salvage/1_3.1.3.5_RXN-14143\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dUMP_salvage_1_3_1_3_5_RXN_14143 \"dUMP-salvage/1_3.1.3.5_RXN-14143\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dUMP_salvage_2_2_4_2_1_2_4_2_2_URA_PHOSPH_RXN \"dUMP-salvage/2_2.4.2.1_2.4.2.2_URA-PHOSPH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dUMP_salvage_2_2_4_2_1_2_4_2_2_URA_PHOSPH_RXN \"dUMP-salvage/2_2.4.2.1_2.4.2.2_URA-PHOSPH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dAMP_salvage_1_3_1_3_5_3_1_3_2_RXN_14161 \"dAMP-salvage/1_3.1.3.5_3.1.3.2_RXN-14161\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dAMP_salvage_1_3_1_3_5_3_1_3_2_RXN_14161 \"dAMP-salvage/1_3.1.3.5_3.1.3.2_RXN-14161\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dAMP_salvage_2a_2_4_2_1_DEOXYADENPHOSPHOR_RXN \"dAMP-salvage/2a_2.4.2.1_DEOXYADENPHOSPHOR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dAMP_salvage_2a_2_4_2_1_DEOXYADENPHOSPHOR_RXN \"dAMP-salvage/2a_2.4.2.1_DEOXYADENPHOSPHOR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dAMP_salvage_2b_3_5_4_4_ADDALT_RXN \"dAMP-salvage/2b_3.5.4.4_ADDALT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dAMP_salvage_2b_3_5_4_4_ADDALT_RXN \"dAMP-salvage/2b_3.5.4.4_ADDALT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dAMP_salvage_3_2_4_2_1_DEOXYINOPHOSPHOR_RXN \"dAMP-salvage/3_2.4.2.1_DEOXYINOPHOSPHOR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dAMP_salvage_3_2_4_2_1_DEOXYINOPHOSPHOR_RXN \"dAMP-salvage/3_2.4.2.1_DEOXYINOPHOSPHOR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dGMP_salvage_1_3_1_3_5_3_1_3_2_RXN_14142 \"dGMP-salvage/1_3.1.3.5_3.1.3.2_RXN-14142\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dGMP_salvage_1_3_1_3_5_3_1_3_2_RXN_14142 \"dGMP-salvage/1_3.1.3.5_3.1.3.2_RXN-14142\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dGMP_salvage_2_2_4_2_1_DEOXYGUANPHOSPHOR_RXN \"dGMP-salvage/2_2.4.2.1_DEOXYGUANPHOSPHOR-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dGMP_salvage_2_2_4_2_1_DEOXYGUANPHOSPHOR_RXN \"dGMP-salvage/2_2.4.2.1_DEOXYGUANPHOSPHOR-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dTMP_salvage_1_3_1_3_5_THYMIDYLATE_5_PHOSPHATASE_RXN \"dTMP-salvage/1_3.1.3.5_THYMIDYLATE-5-PHOSPHATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dTMP_salvage_1_3_1_3_5_THYMIDYLATE_5_PHOSPHATASE_RXN \"dTMP-salvage/1_3.1.3.5_THYMIDYLATE-5-PHOSPHATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dTMP_salvage_2_2_4_2_2_THYM_PHOSPH_RXN \"dTMP-salvage/2_2.4.2.2_THYM-PHOSPH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dTMP_salvage_2_2_4_2_2_THYM_PHOSPH_RXN \"dTMP-salvage/2_2.4.2.2_THYM-PHOSPH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_1_3_2_2_1_BR16044 \"adenosine-salvage/1_3.2.2.1_BR16044\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_1_3_2_2_1_BR16044 \"adenosine-salvage/1_3.2.2.1_BR16044\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosine_salvage_2_3_5_4_2_ADENINE_DEAMINASE_RXN \"adenosine-salvage/2_3.5.4.2_ADENINE-DEAMINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosine_salvage_2_3_5_4_2_ADENINE_DEAMINASE_RXN \"adenosine-salvage/2_3.5.4.2_ADENINE-DEAMINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction guanosine_salvage_1_3_2_2_1_RXN0_366 \"guanosine-salvage/1_3.2.2.1_RXN0-366\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction guanosine_salvage_1_3_2_2_1_RXN0_366 \"guanosine-salvage/1_3.2.2.1_RXN0-366\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction guanosine_salvage_2_2_4_2_8_2_4_2_7_2_4_2_22_GUANPRIBOSYLTRAN_RXN \"guanosine-salvage/2_2.4.2.8_2.4.2.7_2.4.2.22_GUANPRIBOSYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction guanosine_salvage_2_2_4_2_8_2_4_2_7_2_4_2_22_GUANPRIBOSYLTRAN_RXN \"guanosine-salvage/2_2.4.2.8_2.4.2.7_2.4.2.22_GUANPRIBOSYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_salvage_1_3_2_2_1_BR16042 \"inosine-salvage/1_3.2.2.1_BR16042\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_salvage_1_3_2_2_1_BR16042 \"inosine-salvage/1_3.2.2.1_BR16042\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction inosine_salvage_2_2_4_2_8_HYPOXANPRIBOSYLTRAN_RXN \"inosine-salvage/2_2.4.2.8_HYPOXANPRIBOSYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction inosine_salvage_2_2_4_2_8_HYPOXANPRIBOSYLTRAN_RXN \"inosine-salvage/2_2.4.2.8_HYPOXANPRIBOSYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xanthosine_salvage_3_2_2_1_RXN0_363 \"xanthosine-salvage_3.2.2.1_RXN0-363\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xanthosine_salvage_3_2_2_1_RXN0_363 \"xanthosine-salvage_3.2.2.1_RXN0-363\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cytidine_salvage_3_2_2_1_BS214504 \"cytidine-salvage_3.2.2.1_BS214504\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cytidine_salvage_3_2_2_1_BS214504 \"cytidine-salvage_3.2.2.1_BS214504\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_deoxy_alpha_D_ribose_1_phosphate_degr_1_5_4_2_7_D_PPENTOMUT_RXN \"2'-deoxy-alpha-D-ribose-1-phosphate-degr/1_5.4.2.7_D-PPENTOMUT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_deoxy_alpha_D_ribose_1_phosphate_degr_1_5_4_2_7_D_PPENTOMUT_RXN \"2'-deoxy-alpha-D-ribose-1-phosphate-degr/1_5.4.2.7_D-PPENTOMUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_deoxy_alpha_D_ribose_1_phosphate_degr_2_4_1_2_4_DEOXYRIBOSE_P_ALD_RXN \"2'-deoxy-alpha-D-ribose-1-phosphate-degr/2_4.1.2.4_DEOXYRIBOSE-P-ALD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_deoxy_alpha_D_ribose_1_phosphate_degr_2_4_1_2_4_DEOXYRIBOSE_P_ALD_RXN \"2'-deoxy-alpha-D-ribose-1-phosphate-degr/2_4.1.2.4_DEOXYRIBOSE-P-ALD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetaldehyde_degradation_1_2_1_3_1_17_1_4_RXN66_3 \"acetaldehyde-degradation_1.2.1.3_1.17.1.4_RXN66-3\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetaldehyde_degradation_1_2_1_3_1_17_1_4_RXN66_3 \"acetaldehyde-degradation_1.2.1.3_1.17.1.4_RXN66-3\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction PRPP_biosynthesis_II_5_4_2_2_5_4_2_7_PPENTOMUT_RXN \"PRPP-biosynthesis-II_5.4.2.2_5.4.2.7_PPENTOMUT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction PRPP_biosynthesis_II_5_4_2_2_5_4_2_7_PPENTOMUT_RXN \"PRPP-biosynthesis-II_5.4.2.2_5.4.2.7_PPENTOMUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction beta_D_ribopyranose_degradation_modb_2_7_1_15_BR47755 \"beta-D-ribopyranose-degradation_modb_2.7.1.15_BR47755\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction beta_D_ribopyranose_degradation_modb_2_7_1_15_BR47755 \"beta-D-ribopyranose-degradation_modb_2.7.1.15_BR47755\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_RXN_14074 \"NXP-transitions_2.7.4.3_RXN-14074\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_RXN_14074 \"NXP-transitions_2.7.4.3_RXN-14074\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_8_GUANYL_KIN_RXN \"NXP-transitions_2.7.4.8_GUANYL-KIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_8_GUANYL_KIN_RXN \"NXP-transitions_2.7.4.8_GUANYL-KIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_GDPKIN_RXN \"NXP-transitions_2.7.4.6_GDPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_GDPKIN_RXN \"NXP-transitions_2.7.4.6_GDPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_22_RXN_12002 \"NXP-transitions_2.7.4.22_RXN-12002\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_22_RXN_12002 \"NXP-transitions_2.7.4.22_RXN-12002\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_BS358186 \"NXP-transitions_2.7.4.3_BS358186\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_BS358186 \"NXP-transitions_2.7.4.3_BS358186\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_25_2_7_4_3_RXN_11832 \"NXP-transitions_2.7.4.25_2.7.4.3_RXN-11832\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_25_2_7_4_3_RXN_11832 \"NXP-transitions_2.7.4.25_2.7.4.3_RXN-11832\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_CDPKIN_RXN \"NXP-transitions_2.7.4.3_CDPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_CDPKIN_RXN \"NXP-transitions_2.7.4.3_CDPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_ADENYL_KIN_RXN \"NXP-transitions_2.7.4.3_ADENYL-KIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_ADENYL_KIN_RXN \"NXP-transitions_2.7.4.3_ADENYL-KIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_DEOXYADENYLATE_KINASE_RXN \"NXP-transitions_2.7.4.3_DEOXYADENYLATE-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_3_DEOXYADENYLATE_KINASE_RXN \"NXP-transitions_2.7.4.3_DEOXYADENYLATE-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_8_GMKALT_RXN \"NXP-transitions_2.7.4.8_GMKALT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_8_GMKALT_RXN \"NXP-transitions_2.7.4.8_GMKALT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_25_RXN_7913 \"NXP-transitions_2.7.4.25_RXN-7913\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_25_RXN_7913 \"NXP-transitions_2.7.4.25_RXN-7913\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_9_DTMPKI_RXN \"NXP-transitions_2.7.4.9_DTMPKI-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_9_DTMPKI_RXN \"NXP-transitions_2.7.4.9_DTMPKI-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_9_RXN_17141 \"NXP-transitions_2.7.4.9_RXN-17141\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_9_RXN_17141 \"NXP-transitions_2.7.4.9_RXN-17141\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DTDPKIN_RXN \"NXP-transitions_2.7.4.6_DTDPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DTDPKIN_RXN \"NXP-transitions_2.7.4.6_DTDPKIN-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_22_2_7_4_9_2_7_4_25_RXN_14122 \"NXP-transitions_2.7.4.22_2.7.4.9_2.7.4.25_RXN-14122\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_22_2_7_4_9_2_7_4_25_RXN_14122 \"NXP-transitions_2.7.4.22_2.7.4.9_2.7.4.25_RXN-14122\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DADPKIN_RXN \"NXP-transitions_2.7.4.6_DADPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DADPKIN_RXN \"NXP-transitions_2.7.4.6_DADPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DGDPKIN_RXN \"NXP-transitions_2.7.4.6_DGDPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DGDPKIN_RXN \"NXP-transitions_2.7.4.6_DGDPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DUDPKIN_RXN \"NXP-transitions_2.7.4.6_DUDPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DUDPKIN_RXN \"NXP-transitions_2.7.4.6_DUDPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DCDPKIN_RXN \"NXP-transitions_2.7.4.6_DCDPKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction NXP_transitions_2_7_4_6_DCDPKIN_RXN \"NXP-transitions_2.7.4.6_DCDPKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppGpp_biosynthesis_2_7_6_5_GDPPYPHOSKIN_RXN \"ppGpp-biosynthesis_2.7.6.5_GDPPYPHOSKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppGpp_biosynthesis_2_7_6_5_GDPPYPHOSKIN_RXN \"ppGpp-biosynthesis_2.7.6.5_GDPPYPHOSKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppGpp_biosynthesis_2_7_6_5_GTPPYPHOSKIN_RXN \"ppGpp-biosynthesis_2.7.6.5_GTPPYPHOSKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppGpp_biosynthesis_2_7_6_5_GTPPYPHOSKIN_RXN \"ppGpp-biosynthesis_2.7.6.5_GTPPYPHOSKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ppGpp_biosynthesis_3_6_1_40_PPPGPPHYDRO_RXN \"ppGpp-biosynthesis_3.6.1.40_PPPGPPHYDRO-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ppGpp_biosynthesis_3_6_1_40_PPPGPPHYDRO_RXN \"ppGpp-biosynthesis_3.6.1.40_PPPGPPHYDRO-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction uracil_degr_I_1_1_3_1_1_DIHYDROURACIL_DEHYDROGENASE_NAD_RXN \"uracil-degr-I/1_1.3.1.1_DIHYDROURACIL-DEHYDROGENASE-NAD+-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction uracil_degr_I_1_1_3_1_1_DIHYDROURACIL_DEHYDROGENASE_NAD_RXN \"uracil-degr-I/1_1.3.1.1_DIHYDROURACIL-DEHYDROGENASE-NAD+-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction uracil_degr_I_2_3_5_2_2_DIHYDROPYRIMIDINASE_RXN \"uracil-degr-I/2_3.5.2.2_DIHYDROPYRIMIDINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction uracil_degr_I_2_3_5_2_2_DIHYDROPYRIMIDINASE_RXN \"uracil-degr-I/2_3.5.2.2_DIHYDROPYRIMIDINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction uracil_degr_I_3_3_5_1_6_BETA_UREIDOPROPIONASE_RXN \"uracil-degr-I/3_3.5.1.6_BETA-UREIDOPROPIONASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction uracil_degr_I_3_3_5_1_6_BETA_UREIDOPROPIONASE_RXN \"uracil-degr-I/3_3.5.1.6_BETA-UREIDOPROPIONASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thymine_degradation_1_3_1_1_RXN0_6565 \"thymine-degradation_1.3.1.1_RXN0-6565\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thymine_degradation_1_3_1_1_RXN0_6565 \"thymine-degradation_1.3.1.1_RXN0-6565\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thymine_degradation_3_5_2_2_RXN_11211 \"thymine-degradation_3.5.2.2_RXN-11211\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thymine_degradation_3_5_2_2_RXN_11211 \"thymine-degradation_3.5.2.2_RXN-11211\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thymine_degradation_3_5_1_6_RXN_11210 \"thymine-degradation_3.5.1.6_RXN-11210\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thymine_degradation_3_5_1_6_RXN_11210 \"thymine-degradation_3.5.1.6_RXN-11210\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cytosine_exchange \"cytosine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cytosine_exchange \"cytosine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thymine_exchange \"thymine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thymine_exchange \"thymine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cytosine_transport \"cytosine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cytosine_transport \"cytosine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thymine_transport \"thymine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thymine_transport \"thymine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenine_exchange \"adenine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenine_exchange \"adenine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction guanine_exchange \"guanine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction guanine_exchange \"guanine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hypoxanthine_exchange \"hypoxanthine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hypoxanthine_exchange \"hypoxanthine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenine_transport \"adenine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenine_transport \"adenine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction guanine_transport \"guanine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction guanine_transport \"guanine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hypoxanthine_transport \"hypoxanthine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hypoxanthine_transport \"hypoxanthine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction uracil_exchange \"uracil_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction uracil_exchange \"uracil_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction uracil_transport \"uracil_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction uracil_transport \"uracil_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xanthine_exchange \"xanthine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xanthine_exchange \"xanthine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction urate_exchange \"urate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction urate_exchange \"urate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xanthine_transport \"xanthine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xanthine_transport \"xanthine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction urate_transport \"urate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction urate_transport \"urate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction urate_degradation_1_17_1_4_RXN0_901 \"urate-degradation_1.17.1.4_RXN0-901\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction urate_degradation_1_17_1_4_RXN0_901 \"urate-degradation_1.17.1.4_RXN0-901\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction asparagine_exchange \"asparagine_exchange\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction asparagine_exchange \"asparagine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamine_exchange \"glutamine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamine_exchange \"glutamine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glutamine_transport \"glutamine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glutamine_transport \"glutamine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction asparagine_transport \"asparagine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction asparagine_transport \"asparagine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_hydroxyproline_exchange \"4-hydroxyproline_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_hydroxyproline_exchange \"4-hydroxyproline_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_hydroxyproline_transport \"4-hydroxyproline_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_hydroxyproline_transport \"4-hydroxyproline_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_hydroxyproline_degr_1_manual_4_2_1_HD_Hyp \"4-hydroxyproline-degr/1_manual_4.2.1.-_HD-Hyp\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_hydroxyproline_degr_1_manual_4_2_1_HD_Hyp \"4-hydroxyproline-degr/1_manual_4.2.1.-_HD-Hyp\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _4_hydroxyproline_degr_2_1_5_1_2_PYRROLINECARBREDUCT_RXN \"4-hydroxyproline-degr/2_1.5.1.2_PYRROLINECARBREDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _4_hydroxyproline_degr_2_1_5_1_2_PYRROLINECARBREDUCT_RXN \"4-hydroxyproline-degr/2_1.5.1.2_PYRROLINECARBREDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cystathionine_exchange \"cystathionine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cystathionine_exchange \"cystathionine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cystathionine_transport \"cystathionine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cystathionine_transport \"cystathionine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cystathionine_degr_A_4_4_1_8_RXN_15131 \"cystathionine-degr-A_4.4.1.8_RXN-15131\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cystathionine_degr_A_4_4_1_8_RXN_15131 \"cystathionine-degr-A_4.4.1.8_RXN-15131\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cystathionine_degr_B_4_4_1_1_RXN_14048 \"cystathionine-degr-B_4.4.1.1_RXN-14048\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cystathionine_degr_B_4_4_1_1_RXN_14048 \"cystathionine-degr-B_4.4.1.1_RXN-14048\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spermidine_exchange \"spermidine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spermidine_exchange \"spermidine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction putrescine_exchange \"putrescine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction putrescine_exchange \"putrescine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction putrescine_transport_3_6_3_31_modb_BS141431 \"putrescine_transport_3.6.3.31_modb_BS141431\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction putrescine_transport_3_6_3_31_modb_BS141431 \"putrescine_transport_3.6.3.31_modb_BS141431\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction spermidine_transport_3_6_3_31_modb_BS141432 \"spermidine_transport_3.6.3.31_modb_BS141432\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction spermidine_transport_3_6_3_31_modb_BS141432 \"spermidine_transport_3.6.3.31_modb_BS141432\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_3_diaminopropanoate_exchange \"2,3-diaminopropanoate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_3_diaminopropanoate_exchange \"2,3-diaminopropanoate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_3_diaminopropanoate_transport \"2,3-diaminopropanoate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_3_diaminopropanoate_transport \"2,3-diaminopropanoate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_3_diaminopropanoate_degr_4_3_1_15_4_3_1_15_RXN \"2,3-diaminopropanoate-degr_4.3.1.15_4.3.1.15-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_3_diaminopropanoate_degr_4_3_1_15_4_3_1_15_RXN \"2,3-diaminopropanoate-degr_4.3.1.15_4.3.1.15-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydroxylamine_exchange \"hydroxylamine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydroxylamine_exchange \"hydroxylamine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydroxylamine_transport \"hydroxylamine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydroxylamine_transport \"hydroxylamine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydroxylamine_1_7_99_1_HYDROXYLAMINE_REDUCTASE_NADH_RXN \"hydroxylamine_1.7.99.1_HYDROXYLAMINE-REDUCTASE-NADH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydroxylamine_1_7_99_1_HYDROXYLAMINE_REDUCTASE_NADH_RXN \"hydroxylamine_1.7.99.1_HYDROXYLAMINE-REDUCTASE-NADH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_exchange \"ethanolamine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_exchange \"ethanolamine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_transport1 \"ethanolamine_transport1\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_transport1 \"ethanolamine_transport1\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_transport2 \"ethanolamine_transport2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_transport2 \"ethanolamine_transport2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_utilization_1_4_3_1_7_ETHAMLY_RXN \"ethanolamine-utilization/1_4.3.1.7_ETHAMLY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_utilization_1_4_3_1_7_ETHAMLY_RXN \"ethanolamine-utilization/1_4.3.1.7_ETHAMLY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_utilization_2_1_2_1_10_ACETALD_DEHYDROG_RXN \"ethanolamine-utilization/2_1.2.1.10_ACETALD-DEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_utilization_2_1_2_1_10_ACETALD_DEHYDROG_RXN \"ethanolamine-utilization/2_1.2.1.10_ACETALD-DEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_utilization_3_2_3_1_8_PHOSACETYLTRANS_RXN \"ethanolamine-utilization/3_2.3.1.8_PHOSACETYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_utilization_3_2_3_1_8_PHOSACETYLTRANS_RXN \"ethanolamine-utilization/3_2.3.1.8_PHOSACETYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ethanolamine_utilization_4_1_1_1_1_ALCOHOL_DEHYDROGENASE_NADP_RXN_WOP \"ethanolamine-utilization/4_1.1.1.1_ALCOHOL-DEHYDROGENASE-NADP+-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ethanolamine_utilization_4_1_1_1_1_ALCOHOL_DEHYDROGENASE_NADP_RXN_WOP \"ethanolamine-utilization/4_1.1.1.1_ALCOHOL-DEHYDROGENASE-NADP+-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sn_glycerol_3_phosphate_exchange \"sn-glycerol-3-phosphate_exchange\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sn_glycerol_3_phosphate_exchange \"sn-glycerol-3-phosphate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sn_glycerol_3_phosphate_transport \"sn-glycerol-3-phosphate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sn_glycerol_3_phosphate_transport \"sn-glycerol-3-phosphate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_exchange \"methylphosphonate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_exchange \"methylphosphonate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_transport \"methylphosphonate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_transport \"methylphosphonate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methane_transport \"methane_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methane_transport \"methane_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methane_exchange \"methane_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methane_exchange \"methane_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_1_2_7_8_37_RXN0_6732 \"methylphosphonate-degr-II/1_2.7.8.37_RXN0-6732\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_1_2_7_8_37_RXN0_6732 \"methylphosphonate-degr-II/1_2.7.8.37_RXN0-6732\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_2_3_6_1_63_RXN0_6733 \"methylphosphonate-degr-II/2_3.6.1.63_RXN0-6733\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_2_3_6_1_63_RXN0_6733 \"methylphosphonate-degr-II/2_3.6.1.63_RXN0-6733\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_3_4_7_1_1_RXN0_6734 \"methylphosphonate-degr-II/3_4.7.1.1_RXN0-6734\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_3_4_7_1_1_RXN0_6734 \"methylphosphonate-degr-II/3_4.7.1.1_RXN0-6734\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_4_3_1_4_57_RXN_14995 \"methylphosphonate-degr-II/4_3.1.4.57_RXN-14995\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_4_3_1_4_57_RXN_14995 \"methylphosphonate-degr-II/4_3.1.4.57_RXN-14995\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_5_3_1_4_57_RXN_14996 \"methylphosphonate-degr-II/5_3.1.4.57_RXN-14996\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction methylphosphonate_degr_II_5_3_1_4_57_RXN_14996 \"methylphosphonate-degr-II/5_3.1.4.57_RXN-14996\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminoethylphosphonate_exchange \"2-aminoethylphosphonate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminoethylphosphonate_exchange \"2-aminoethylphosphonate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminoethylphosphonate_transport \"2-aminoethylphosphonate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminoethylphosphonate_transport \"2-aminoethylphosphonate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminoethylphosph_degr_I_1_2_6_1_37_2_6_1_37_RXN \"2-aminoethylphosph-degr-I/1_2.6.1.37_2.6.1.37-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminoethylphosph_degr_I_1_2_6_1_37_2_6_1_37_RXN \"2-aminoethylphosph-degr-I/1_2.6.1.37_2.6.1.37-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_aminoethylphosph_degr_I_2_3_11_1_1_PHOSPHONOACETALDEHYDE_HYDROLASE_RXN \"2-aminoethylphosph-degr-I/2_3.11.1.1_PHOSPHONOACETALDEHYDE-HYDROLASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_aminoethylphosph_degr_I_2_3_11_1_1_PHOSPHONOACETALDEHYDE_HYDROLASE_RXN \"2-aminoethylphosph-degr-I/2_3.11.1.1_PHOSPHONOACETALDEHYDE-HYDROLASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphonoacetate_exchange \"phosphonoacetate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphonoacetate_exchange \"phosphonoacetate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphonoacetate_transport \"phosphonoacetate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphonoacetate_transport \"phosphonoacetate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction phosphonoacetate_degradation_3_11_1_2_3_11_1_2_RXN \"phosphonoacetate-degradation_3.11.1.2_3.11.1.2-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction phosphonoacetate_degradation_3_11_1_2_3_11_1_2_RXN \"phosphonoacetate-degradation_3.11.1.2_3.11.1.2-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_exchange \"succinate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_exchange \"succinate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction succinate_transport \"succinate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction succinate_transport \"succinate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tartronate_semialdehyde_exchange \"tartronate-semialdehyde_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tartronate_semialdehyde_exchange \"tartronate-semialdehyde_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tartronate_semialdehyde_transport \"tartronate-semialdehyde_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tartronate_semialdehyde_transport \"tartronate-semialdehyde_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tartronate_semialdehyde_degr_1_1_1_60_TSA_REDUCT_RXN_WOP \"tartronate-semialdehyde-degr_1.1.1.60_TSA-REDUCT-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tartronate_semialdehyde_degr_1_1_1_60_TSA_REDUCT_RXN_WOP \"tartronate-semialdehyde-degr_1.1.1.60_TSA-REDUCT-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction tartronate_semialdehyde_degr_1_1_1_60_TSA_REDUCT_RXN \"tartronate-semialdehyde-degr_1.1.1.60_TSA-REDUCT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction tartronate_semialdehyde_degr_1_1_1_60_TSA_REDUCT_RXN \"tartronate-semialdehyde-degr_1.1.1.60_TSA-REDUCT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetoin_exchange \"acetoin_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetoin_exchange \"acetoin_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetoin_transport \"acetoin_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetoin_transport \"acetoin_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction acetoin_degradation_2_3_1_190_RXN_9718 \"acetoin-degradation_2.3.1.190_RXN-9718\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction acetoin_degradation_2_3_1_190_RXN_9718 \"acetoin-degradation_2.3.1.190_RXN-9718\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_exchange \"3-hydroxy-3-methylglutarate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_exchange \"3-hydroxy-3-methylglutarate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_transport \"3-hydroxy-3-methylglutarate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_transport \"3-hydroxy-3-methylglutarate_transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_degr_1_manual_2_8_3_HD_3MG \"3-hydroxy-3-methylglutarate-degr/1_manual_2.8.3.-_HD-3MG\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_degr_1_manual_2_8_3_HD_3MG \"3-hydroxy-3-methylglutarate-degr/1_manual_2.8.3.-_HD-3MG\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_degr_2_4_1_3_4_HYDROXYMETHYLGLUTARYL_COA_LYASE_RXN \"3-hydroxy-3-methylglutarate-degr/2_4.1.3.4_HYDROXYMETHYLGLUTARYL-COA-LYASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _3_hydroxy_3_methylglutarate_degr_2_4_1_3_4_HYDROXYMETHYLGLUTARYL_COA_LYASE_RXN \"3-hydroxy-3-methylglutarate-degr/2_4.1.3.4_HYDROXYMETHYLGLUTARYL-COA-LYASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydroxyl_radical_exchange \"hydroxyl-radical_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydroxyl_radical_exchange \"hydroxyl-radical_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydroxyl_radical_transport \"hydroxyl-radical_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydroxyl_radical_transport \"hydroxyl-radical_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_spontaneous_RXN_11409 \"oxidized-GTP-and-dGTP-detoxification_spontaneous_RXN-11409\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_spontaneous_RXN_11409 \"oxidized-GTP-and-dGTP-detoxification_spontaneous_RXN-11409\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_spontaneous_RXN_11410 \"oxidized-GTP-and-dGTP-detoxification_spontaneous_RXN-11410\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_spontaneous_RXN_11410 \"oxidized-GTP-and-dGTP-detoxification_spontaneous_RXN-11410\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_3_6_1_55_RXN_11397 \"oxidized-GTP-and-dGTP-detoxification_3.6.1.55_RXN-11397\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_3_6_1_55_RXN_11397 \"oxidized-GTP-and-dGTP-detoxification_3.6.1.55_RXN-11397\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_3_6_1_55_RXN_11396 \"oxidized-GTP-and-dGTP-detoxification_3.6.1.55_RXN-11396\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxidized_GTP_and_dGTP_detoxification_3_6_1_55_RXN_11396 \"oxidized-GTP-and-dGTP-detoxification_3.6.1.55_RXN-11396\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _8_oxo_GMP_transport \"8-oxo-GMP_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _8_oxo_GMP_transport \"8-oxo-GMP_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _8_oxo_dGMP_transport \"8-oxo-dGMP_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _8_oxo_dGMP_transport \"8-oxo-dGMP_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _8_oxo_GMP_exchange \"8-oxo-GMP_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _8_oxo_GMP_exchange \"8-oxo-GMP_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _8_oxo_dGMP_exchange \"8-oxo-dGMP_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _8_oxo_dGMP_exchange \"8-oxo-dGMP_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_exchange \"superoxide_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_exchange \"superoxide_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydrogen_peroxide_exchange \"hydrogen_peroxide_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydrogen_peroxide_exchange \"hydrogen_peroxide_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxygen_exchange \"oxygen_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxygen_exchange \"oxygen_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction mathanol_exchange \"mathanol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction mathanol_exchange \"mathanol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_transport \"superoxide_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_transport \"superoxide_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction hydrogen_peroxide_transport \"hydrogen_peroxide_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction hydrogen_peroxide_transport \"hydrogen_peroxide_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxygen_transport \"oxygen_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxygen_transport \"oxygen_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction mathanol_transport \"mathanol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction mathanol_transport \"mathanol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_1_1_15_1_1_SUPEROX_DISMUT_RXN \"superoxide-radicals-degr-A/1_1.15.1.1_SUPEROX-DISMUT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_1_1_15_1_1_SUPEROX_DISMUT_RXN \"superoxide-radicals-degr-A/1_1.15.1.1_SUPEROX-DISMUT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_2a_1_11_1_6_CATAL_RXN \"superoxide-radicals-degr-A/2a_1.11.1.6_CATAL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_2a_1_11_1_6_CATAL_RXN \"superoxide-radicals-degr-A/2a_1.11.1.6_CATAL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_2b_1_11_1_6_RXN66_1 \"superoxide-radicals-degr-A/2b_1.11.1.6_RXN66-1\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_2b_1_11_1_6_RXN66_1 \"superoxide-radicals-degr-A/2b_1.11.1.6_RXN66-1\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_2c_moddir_1_11_1_6_RXN_14189 \"superoxide-radicals-degr-A/2c_moddir_1.11.1.6_RXN-14189\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_A_2c_moddir_1_11_1_6_RXN_14189 \"superoxide-radicals-degr-A/2c_moddir_1.11.1.6_RXN-14189\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_B_1_1_15_1_2_1_15_1_2_RXN \"superoxide-radicals-degr-B/1_1.15.1.2_1.15.1.2-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_B_1_1_15_1_2_1_15_1_2_RXN \"superoxide-radicals-degr-B/1_1.15.1.2_1.15.1.2-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_B_2_manual_1_11_1_HD_1_11_1_ \"superoxide-radicals-degr-B/2_manual_1.11.1.-_HD-1.11.1.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_B_2_manual_1_11_1_HD_1_11_1_ \"superoxide-radicals-degr-B/2_manual_1.11.1.-_HD-1.11.1.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction superoxide_radicals_degr_B_3_manual_1_4_HD_1_4_ \"superoxide-radicals-degr-B/3_manual_1.4.-_HD-1.4.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction superoxide_radicals_degr_B_3_manual_1_4_HD_1_4_ \"superoxide-radicals-degr-B/3_manual_1.4.-_HD-1.4.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxygen_reduction_A_1_6_3_HD_1_6_3_ \"oxygen-reduction-A_1.6.3.-_HD-1.6.3.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxygen_reduction_A_1_6_3_HD_1_6_3_ \"oxygen-reduction-A_1.6.3.-_HD-1.6.3.-\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction oxygen_reduction_B_1_6_3_4_RXN_14692 \"oxygen-reduction-B_1.6.3.4_RXN-14692\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction oxygen_reduction_B_1_6_3_4_RXN_14692 \"oxygen-reduction-B_1.6.3.4_RXN-14692\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formaldehyde_assimilation_spontaneous_2_1_2_11_RXN_2881 \"formaldehyde-assimilation_spontaneous_2.1.2.11_RXN-2881\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formaldehyde_assimilation_spontaneous_2_1_2_11_RXN_2881 \"formaldehyde-assimilation_spontaneous_2.1.2.11_RXN-2881\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_exchange \"cyanide_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_exchange \"cyanide_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiocyanate_exchange \"thiocyanate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiocyanate_exchange \"thiocyanate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_transport \"cyanide_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_transport \"cyanide_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction thiocyanate_transport \"thiocyanate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction thiocyanate_transport \"thiocyanate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_detoxification_I_1_4_4_1_9_L_3_CYANOALANINE_SYNTHASE_RXN \"cyanide-detoxification-I/1_4.4.1.9_L-3-CYANOALANINE-SYNTHASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_detoxification_I_1_4_4_1_9_L_3_CYANOALANINE_SYNTHASE_RXN \"cyanide-detoxification-I/1_4.4.1.9_L-3-CYANOALANINE-SYNTHASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_detoxification_I_2a_3_5_5_4_3_5_5_4_RXN \"cyanide-detoxification-I/2a_3.5.5.4_3.5.5.4-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_detoxification_I_2a_3_5_5_4_3_5_5_4_RXN \"cyanide-detoxification-I/2a_3.5.5.4_3.5.5.4-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_detoxification_I_2b_4_2_1_65_3_5_5_4_3_CYANOALANINE_HYDRATASE_RXN \"cyanide-detoxification-I/2b_4.2.1.65_3.5.5.4_3-CYANOALANINE-HYDRATASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_detoxification_I_2b_4_2_1_65_3_5_5_4_3_CYANOALANINE_HYDRATASE_RXN \"cyanide-detoxification-I/2b_4.2.1.65_3.5.5.4_3-CYANOALANINE-HYDRATASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_detoxification_III_1_2_8_1_2_R03105 \"cyanide-detoxification-III/1_2.8.1.2_R03105\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_detoxification_III_1_2_8_1_2_R03105 \"cyanide-detoxification-III/1_2.8.1.2_R03105\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyanide_detoxification_III_2_2_8_1_1_THIOSULFATE_SULFURTRANSFERASE_RXN \"cyanide-detoxification-III/2_2.8.1.1_THIOSULFATE-SULFURTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyanide_detoxification_III_2_2_8_1_1_THIOSULFATE_SULFURTRANSFERASE_RXN \"cyanide-detoxification-III/2_2.8.1.1_THIOSULFATE-SULFURTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_cysteine_degradation_III_1_2_6_1_1_CYSTEINE_AMINOTRANSFERASE_RXN \"L-cysteine-degradation-III/1_2.6.1.1_CYSTEINE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_cysteine_degradation_III_1_2_6_1_1_CYSTEINE_AMINOTRANSFERASE_RXN \"L-cysteine-degradation-III/1_2.6.1.1_CYSTEINE-AMINOTRANSFERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction L_cysteine_degradation_III_2_2_8_1_2_MERCAPYSTRANS_RXN \"L-cysteine-degradation-III/2_2.8.1.2_MERCAPYSTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction L_cysteine_degradation_III_2_2_8_1_2_MERCAPYSTRANS_RXN \"L-cysteine-degradation-III/2_2.8.1.2_MERCAPYSTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurochenodeoxycholate_exchange \"taurochenodeoxycholate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurochenodeoxycholate_exchange \"taurochenodeoxycholate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chenodeoxycholate_exchange \"chenodeoxycholate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chenodeoxycholate_exchange \"chenodeoxycholate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurocholate_exchange \"taurocholate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurocholate_exchange \"taurocholate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cholate_exchange \"cholate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cholate_exchange \"cholate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurochenodeoxycholate_transport \"taurochenodeoxycholate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurochenodeoxycholate_transport \"taurochenodeoxycholate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chenodeoxycholate_transport \"chenodeoxycholate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chenodeoxycholate_transport \"chenodeoxycholate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurocholate_transport \"taurocholate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurocholate_transport \"taurocholate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cholate_transport \"cholate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cholate_transport \"cholate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurochenodeoxycholate_degr_1_2_3_1_65_RXN_9852 \"taurochenodeoxycholate-degr/1_2.3.1.65_RXN-9852\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurochenodeoxycholate_degr_1_2_3_1_65_RXN_9852 \"taurochenodeoxycholate-degr/1_2.3.1.65_RXN-9852\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurochenodeoxycholate_degr_2_3_1_2_2_RXN_14289 \"taurochenodeoxycholate-degr/2_3.1.2.2_RXN-14289\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurochenodeoxycholate_degr_2_3_1_2_2_RXN_14289 \"taurochenodeoxycholate-degr/2_3.1.2.2_RXN-14289\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurochenodeoxycholate_degr_3_manual_1_1_1_HD_ChChol \"taurochenodeoxycholate-degr/3_manual_1.1.1.-_HD-ChChol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurochenodeoxycholate_degr_3_manual_1_1_1_HD_ChChol \"taurochenodeoxycholate-degr/3_manual_1.1.1.-_HD-ChChol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurocholate_degr_1_moddir_2_3_1_65_RXN_9800 \"taurocholate-degr/1_moddir_2.3.1.65_RXN-9800\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurocholate_degr_1_moddir_2_3_1_65_RXN_9800 \"taurocholate-degr/1_moddir_2.3.1.65_RXN-9800\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurocholate_degr_2_manual_3_1_2_2_HD_3_1_2_2 \"taurocholate-degr/2_manual_3.1.2.2_HD-3.1.2.2\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurocholate_degr_2_manual_3_1_2_2_HD_3_1_2_2 \"taurocholate-degr/2_manual_3.1.2.2_HD-3.1.2.2\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurocholate_degr_3_manual_1_1_1_HD_Chol \"taurocholate-degr/3_manual_1.1.1.-_HD-Chol\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurocholate_degr_3_manual_1_1_1_HD_Chol \"taurocholate-degr/3_manual_1.1.1.-_HD-Chol\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _7_oxodeoxycholate_transport \"7-oxodeoxycholate_transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _7_oxodeoxycholate_transport \"7-oxodeoxycholate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _7_oxolithocholate_transport \"7-oxolithocholate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _7_oxolithocholate_transport \"7-oxolithocholate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurine_transport \"taurine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurine_transport \"taurine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _7_oxodeoxycholate_exchange \"7-oxodeoxycholate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _7_oxodeoxycholate_exchange \"7-oxodeoxycholate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _7_oxolithocholate_exchange \"7-oxolithocholate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _7_oxolithocholate_exchange \"7-oxolithocholate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction taurine_exchange \"taurine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction taurine_exchange \"taurine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction choline_exchange \"choline_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction choline_exchange \"choline_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction choline_transport \"choline_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction choline_transport \"choline_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction trimethylamine_exchange \"trimethylamine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction trimethylamine_exchange \"trimethylamine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction trimethylamine_transport \"trimethylamine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction trimethylamine_transport \"trimethylamine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction choline_degradation_III_4_3_99_4_RXN_13946 \"choline-degradation-III_4.3.99.4_RXN-13946\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction choline_degradation_III_4_3_99_4_RXN_13946 \"choline-degradation-III_4.3.99.4_RXN-13946\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_arabinose_exchange \"D-arabinose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_arabinose_exchange \"D-arabinose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_arabinose_transport_2_7_1_ \"D-arabinose_transport_2.7.1.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_arabinose_transport_2_7_1_ \"D-arabinose_transport_2.7.1.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_arabinose_degr_5_3_1_13_DARAB5PISOM_RXN \"D-arabinose-degr_5.3.1.13_DARAB5PISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_arabinose_degr_5_3_1_13_DARAB5PISOM_RXN \"D-arabinose-degr_5.3.1.13_DARAB5PISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylan_exchange \"xylan_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylan_exchange \"xylan_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylose_exchange \"xylose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylose_exchange \"xylose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylan_transport \"xylan_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylan_transport \"xylan_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylose_transport_2_7_1_ \"xylose_transport_2.7.1.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylose_transport_2_7_1_ \"xylose_transport_2.7.1.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylan_degradation_3_2_1_177_RXN0_5001 \"xylan-degradation_3.2.1.177_RXN0-5001\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylan_degradation_3_2_1_177_RXN0_5001 \"xylan-degradation_3.2.1.177_RXN0-5001\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylose_degradation_I_1_5_3_1_5_XYLISOM_RXN \"xylose-degradation-I/1_5.3.1.5_XYLISOM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylose_degradation_I_1_5_3_1_5_XYLISOM_RXN \"xylose-degradation-I/1_5.3.1.5_XYLISOM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylose_degradation_I_2_2_7_1_17_XYLULOKIN_RXN \"xylose-degradation-I/2_2.7.1.17_XYLULOKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylose_degradation_I_2_2_7_1_17_XYLULOKIN_RXN \"xylose-degradation-I/2_2.7.1.17_XYLULOKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction galactose_exchange \"galactose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction galactose_exchange \"galactose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction galactose_transport_2_7_1_ \"galactose_transport_2.7.1.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction galactose_transport_2_7_1_ \"galactose_transport_2.7.1.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction galactose_degr_I_1_2_7_7_12_GALACTURIDYLYLTRANS_RXN \"galactose-degr-I/1_2.7.7.12_GALACTURIDYLYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction galactose_degr_I_1_2_7_7_12_GALACTURIDYLYLTRANS_RXN \"galactose-degr-I/1_2.7.7.12_GALACTURIDYLYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction galactose_degr_I_2_5_1_3_2_UDPGLUCEPIM_RXN \"galactose-degr-I/2_5.1.3.2_UDPGLUCEPIM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction galactose_degr_I_2_5_1_3_2_UDPGLUCEPIM_RXN \"galactose-degr-I/2_5.1.3.2_UDPGLUCEPIM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction galactose_degr_I_3_spontaneous_RXN_10639 \"galactose-degr-I/3_spontaneous_RXN-10639\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction galactose_degr_I_3_spontaneous_RXN_10639 \"galactose-degr-I/3_spontaneous_RXN-10639\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylitol_exchange \"xylitol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylitol_exchange \"xylitol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylitol_transport \"xylitol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylitol_transport \"xylitol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylitol_degr_manual_1_1_1_HD_xyl5 \"xylitol-degr_manual_1.1.1.-_HD-xyl5\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylitol_degr_manual_1_1_1_HD_xyl5 \"xylitol-degr_manual_1.1.1.-_HD-xyl5\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylitol_transport_2_7_1_ \"xylitol_transport_2.7.1.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylitol_transport_2_7_1_ \"xylitol_transport_2.7.1.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylitol_degr_manual_1_1_1_HD_xyl1 \"xylitol-degr_manual_1.1.1.-_HD-xyl1\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylitol_degr_manual_1_1_1_HD_xyl1 \"xylitol-degr_manual_1.1.1.-_HD-xyl1\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction xylulose_degradation_1_5_1_3_4_RIBULPEPIM_RXN \"xylulose-degradation/1_5.1.3.4_RIBULPEPIM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction xylulose_degradation_1_5_1_3_4_RIBULPEPIM_RXN \"xylulose-degradation/1_5.1.3.4_RIBULPEPIM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arabitol_exchange \"arabitol_exchange\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arabitol_exchange \"arabitol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arabitol_transport \"arabitol_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arabitol_transport \"arabitol_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction arabitol_degr_1_1_1_301_RXN_11106 \"arabitol-degr_1.1.1.301_RXN-11106\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction arabitol_degr_1_1_1_301_RXN_11106 \"arabitol-degr_1.1.1.301_RXN-11106\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ribitol_exchange \"ribitol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ribitol_exchange \"ribitol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ribitol_transport_2_7_1_ \"ribitol_transport_2.7.1.-\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ribitol_transport_2_7_1_ \"ribitol_transport_2.7.1.-\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction ribitol_degr_1_1_1_137_RIBITOL_5_PHOSPHATE_2_DEHYDROGENASE_RXN_WOP \"ribitol-degr_1.1.1.137_RIBITOL-5-PHOSPHATE-2-DEHYDROGENASE-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction ribitol_degr_1_1_1_137_RIBITOL_5_PHOSPHATE_2_DEHYDROGENASE_RXN_WOP \"ribitol-degr_1.1.1.137_RIBITOL-5-PHOSPHATE-2-DEHYDROGENASE-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_sorbitol_exchange \"D-sorbitol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_sorbitol_exchange \"D-sorbitol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_sorbitol_transport_2_7_1_198 \"D-sorbitol_transport_2.7.1.198\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_sorbitol_transport_2_7_1_198 \"D-sorbitol_transport_2.7.1.198\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_sorbitol_degradation_II_1_1_1_140_SORB6PDEHYDROG_RXN \"D-sorbitol-degradation-II_1.1.1.140_SORB6PDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_sorbitol_degradation_II_1_1_1_140_SORB6PDEHYDROG_RXN \"D-sorbitol-degradation-II_1.1.1.140_SORB6PDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sorbitol_biosyn_1_1_1_17_BS257116 \"sorbitol-biosyn_1.1.1.17_BS257116\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sorbitol_biosyn_1_1_1_17_BS257116 \"sorbitol-biosyn_1.1.1.17_BS257116\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_mannitol_exchange \"D-mannitol_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_mannitol_exchange \"D-mannitol_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction mannitol_transport_2_7_1_197 \"mannitol_transport_2.7.1.197\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction mannitol_transport_2_7_1_197 \"mannitol_transport_2.7.1.197\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction mannitol_degradation_I_1_1_1_17_MANNPDEHYDROG_RXN \"mannitol-degradation-I_1.1.1.17_MANNPDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction mannitol_degradation_I_1_1_1_17_MANNPDEHYDROG_RXN \"mannitol-degradation-I_1.1.1.17_MANNPDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_ribopyranose_exchange \"D-ribopyranose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_ribopyranose_exchange \"D-ribopyranose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_deoxyribose_exchange \"2'-deoxyribose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_deoxyribose_exchange \"2'-deoxyribose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_ribopyranose_transport \"D-ribopyranose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_ribopyranose_transport \"D-ribopyranose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_deoxyribose_transport \"2'-deoxyribose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_deoxyribose_transport \"2'-deoxyribose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_deoxyribose_degradation_2_7_1_15_RXN_14223 \"2'-deoxyribose-degradation_2.7.1.15_RXN-14223\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_deoxyribose_degradation_2_7_1_15_RXN_14223 \"2'-deoxyribose-degradation_2.7.1.15_RXN-14223\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltose_exchange \"maltose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltose_exchange \"maltose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltose_transport_2_7_1_208 \"maltose_transport_2.7.1.208\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltose_transport_2_7_1_208 \"maltose_transport_2.7.1.208\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltose_degr_3_2_1_122_MALTOSE_6_PHOSPHATE_GLUCOSIDASE_RXN \"maltose-degr_3.2.1.122_MALTOSE-6-PHOSPHATE-GLUCOSIDASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltose_degr_3_2_1_122_MALTOSE_6_PHOSPHATE_GLUCOSIDASE_RXN \"maltose-degr_3.2.1.122_MALTOSE-6-PHOSPHATE-GLUCOSIDASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sucrose_exchange \"sucrose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sucrose_exchange \"sucrose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sucrose_transport \"sucrose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sucrose_transport \"sucrose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sucrose_degradation_I_1_3_2_1_B3_BR49793 \"sucrose-degradation-I/1_3.2.1.B3_BR49793\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sucrose_degradation_I_1_3_2_1_B3_BR49793 \"sucrose-degradation-I/1_3.2.1.B3_BR49793\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sucrose_degradation_I_2a_2_7_1_4_FRUCTOKINASE_RXN \"sucrose-degradation-I/2a_2.7.1.4_FRUCTOKINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sucrose_degradation_I_2a_2_7_1_4_FRUCTOKINASE_RXN \"sucrose-degradation-I/2a_2.7.1.4_FRUCTOKINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sucrose_degradation_I_2b_5_3_1_5_BS1055 \"sucrose-degradation-I/2b_5.3.1.5_BS1055\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sucrose_degradation_I_2b_5_3_1_5_BS1055 \"sucrose-degradation-I/2b_5.3.1.5_BS1055\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glucose_iso_5_3_1_9_R02740 \"glucose-iso_5.3.1.9_R02740\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glucose_iso_5_3_1_9_R02740 \"glucose-iso_5.3.1.9_R02740\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction glucose_iso_5_1_3_15_5_3_1_9_spontaneous_GLUCOSE_6_PHOSPHATE_1_EPIMERASE_RXN \"glucose-iso_5.1.3.15_5.3.1.9_spontaneous_GLUCOSE-6-PHOSPHATE-1-EPIMERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction glucose_iso_5_1_3_15_5_3_1_9_spontaneous_GLUCOSE_6_PHOSPHATE_1_EPIMERASE_RXN \"glucose-iso_5.1.3.15_5.3.1.9_spontaneous_GLUCOSE-6-PHOSPHATE-1-EPIMERASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_tagatose_exchange \"D-tagatose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_tagatose_exchange \"D-tagatose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_tagatose_transport \"D-tagatose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_tagatose_transport \"D-tagatose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_tagatose_degr_1_manual_2_7_1_HD_tag \"D-tagatose-degr/1_manual_2.7.1.-_HD-tag\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_tagatose_degr_1_manual_2_7_1_HD_tag \"D-tagatose-degr/1_manual_2.7.1.-_HD-tag\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_tagatose_degr_4_1_2_40_TAGAALDOL_RXN \"D-tagatose-degr/_4.1.2.40_TAGAALDOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_tagatose_degr_4_1_2_40_TAGAALDOL_RXN \"D-tagatose-degr/_4.1.2.40_TAGAALDOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_fructofuranose_exchange \"D-fructofuranose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_fructofuranose_exchange \"D-fructofuranose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fructose_transport_2_7_1_202 \"fructose_transport_2.7.1.202\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fructose_transport_2_7_1_202 \"fructose_transport_2.7.1.202\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction fructose_degradation_2_7_1_56_1PFRUCTPHOSN_RXN \"fructose-degradation_2.7.1.56_1PFRUCTPHOSN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction fructose_degradation_2_7_1_56_1PFRUCTPHOSN_RXN \"fructose-degradation_2.7.1.56_1PFRUCTPHOSN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_1_4_1_2_13_RXN_8631 \"1-deoxy-D-xylulose_5-phosphate-biosyn/1_4.1.2.13_RXN-8631\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_1_4_1_2_13_RXN_8631 \"1-deoxy-D-xylulose_5-phosphate-biosyn/1_4.1.2.13_RXN-8631\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_2a_1_2_1_3_BS426217 \"1-deoxy-D-xylulose_5-phosphate-biosyn/2a_1.2.1.3_BS426217\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_2a_1_2_1_3_BS426217 \"1-deoxy-D-xylulose_5-phosphate-biosyn/2a_1.2.1.3_BS426217\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_2b_2_2_1_7_BS260855 \"1-deoxy-D-xylulose_5-phosphate-biosyn/2b_2.2.1.7_BS260855\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_2b_2_2_1_7_BS260855 \"1-deoxy-D-xylulose_5-phosphate-biosyn/2b_2.2.1.7_BS260855\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_3_2_7_1_17_RXN0_382 \"1-deoxy-D-xylulose_5-phosphate-biosyn/3_2.7.1.17_RXN0-382\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _1_deoxy_D_xylulose_5_phosphate_biosyn_3_2_7_1_17_RXN0_382 \"1-deoxy-D-xylulose_5-phosphate-biosyn/3_2.7.1.17_RXN0-382\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_cellobiose_exchange \"D-cellobiose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_cellobiose_exchange \"D-cellobiose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cellobiose_transport_2_7_1_205 \"cellobiose_transport_2.7.1.205\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cellobiose_transport_2_7_1_205 \"cellobiose_transport_2.7.1.205\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cellobiose_degr_3_2_1_86_6_PHOSPHO_BETA_GLUCOSIDASE_RXN \"cellobiose-degr_3.2.1.86_6-PHOSPHO-BETA-GLUCOSIDASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cellobiose_degr_3_2_1_86_6_PHOSPHO_BETA_GLUCOSIDASE_RXN \"cellobiose-degr_3.2.1.86_6-PHOSPHO-BETA-GLUCOSIDASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chitobiose_exchange \"chitobiose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chitobiose_exchange \"chitobiose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chitobiose_transport_2_7_1_196 \"chitobiose_transport_2.7.1.196\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chitobiose_transport_2_7_1_196 \"chitobiose_transport_2.7.1.196\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chitobiose_degradation_3_5_1_105_RXN0_7001 \"chitobiose-degradation_3.5.1.105_RXN0-7001\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chitobiose_degradation_3_5_1_105_RXN0_7001 \"chitobiose-degradation_3.5.1.105_RXN0-7001\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chitobiose_degradation_3_2_1_86_RXN0_7002 \"chitobiose-degradation_3.2.1.86_RXN0-7002\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chitobiose_degradation_3_2_1_86_RXN0_7002 \"chitobiose-degradation_3.2.1.86_RXN0-7002\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction chitobiose_degradation_2_7_1_59_GLUCOSAMINE_KINASE_RXN \"chitobiose-degradation_2.7.1.59_GLUCOSAMINE-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction chitobiose_degradation_2_7_1_59_GLUCOSAMINE_KINASE_RXN \"chitobiose-degradation_2.7.1.59_GLUCOSAMINE-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction trehalose_exchange \"trehalose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction trehalose_exchange \"trehalose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction trehalose_transport \"trehalose_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction trehalose_transport \"trehalose_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction trehalose_degr_I_3_2_1_122_3_2_1_93_TRE6PHYDRO_RXN \"trehalose-degr-I_3.2.1.122_3.2.1.93_TRE6PHYDRO-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction trehalose_degr_I_3_2_1_122_3_2_1_93_TRE6PHYDRO_RXN \"trehalose-degr-I_3.2.1.122_3.2.1.93_TRE6PHYDRO-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyclodextrin_exchange \"cyclodextrin_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyclodextrin_exchange \"cyclodextrin_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction starch_exchange \"starch_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction starch_exchange \"starch_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltodextrin_exchange \"maltodextrin_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltodextrin_exchange \"maltodextrin_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pullulan_exchange \"pullulan_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pullulan_exchange \"pullulan_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltotriose_exchange \"maltotriose_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltotriose_exchange \"maltotriose_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cyclodextrin_degr_3_2_1_54_CYCLOMALTODEXTRINASE_RXN \"cyclodextrin-degr_3.2.1.54_CYCLOMALTODEXTRINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cyclodextrin_degr_3_2_1_54_CYCLOMALTODEXTRINASE_RXN \"cyclodextrin-degr_3.2.1.54_CYCLOMALTODEXTRINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction starch_degr_manual_2_4_1_18_3_2_1_41_3_2_1_1_HD_starch \"starch-degr_manual_2.4.1.18_3.2.1.41_3.2.1.1_HD-starch\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction starch_degr_manual_2_4_1_18_3_2_1_41_3_2_1_1_HD_starch \"starch-degr_manual_2.4.1.18_3.2.1.41_3.2.1.1_HD-starch\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltodextrin_degr_3_2_1_133_3_2_1_1_RXN_12188 \"maltodextrin-degr_3.2.1.133_3.2.1.1_RXN-12188\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltodextrin_degr_3_2_1_133_3_2_1_1_RXN_12188 \"maltodextrin-degr_3.2.1.133_3.2.1.1_RXN-12188\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction pullulan_degr_3_2_1_54_3_2_1_41_RXN_12302 \"pullulan-degr_3.2.1.54_3.2.1.41_RXN-12302\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction pullulan_degr_3_2_1_54_3_2_1_41_RXN_12302 \"pullulan-degr_3.2.1.54_3.2.1.41_RXN-12302\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction maltotriose_degr_modnames_3_2_1_10_R06199 \"maltotriose-degr_modnames_3.2.1.10_R06199\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction maltotriose_degr_modnames_3_2_1_10_R06199 \"maltotriose-degr_modnames_3.2.1.10_R06199\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylneuraminate_exchange \"N-acetylneuraminate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylneuraminate_exchange \"N-acetylneuraminate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylneuraminate_transport \"N-acetylneuraminate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylneuraminate_transport \"N-acetylneuraminate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylneur_degr_1_4_1_3_3_ACNEULY_RXN \"N-acetylneur-degr/1_4.1.3.3_ACNEULY-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylneur_degr_1_4_1_3_3_ACNEULY_RXN \"N-acetylneur-degr/1_4.1.3.3_ACNEULY-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylneur_degr_2_2_7_1_59_NANK_RXN \"N-acetylneur-degr/2_2.7.1.59_NANK-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylneur_degr_2_2_7_1_59_NANK_RXN \"N-acetylneur-degr/2_2.7.1.59_NANK-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetylneur_degr_3_5_1_3_9_NANE_RXN \"N-acetylneur-degr/3_5.1.3.9_NANE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetylneur_degr_3_5_1_3_9_NANE_RXN \"N-acetylneur-degr/3_5.1.3.9_NANE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetyl_D_galactosamine_exchange \"N-acetyl-D-galactosamine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetyl_D_galactosamine_exchange \"N-acetyl-D-galactosamine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_galactosamine_exchange \"D-galactosamine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_galactosamine_exchange \"D-galactosamine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction N_acetyl_D_galactosamine_transport \"N-acetyl-D-galactosamine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction N_acetyl_D_galactosamine_transport \"N-acetyl-D-galactosamine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_galactosamine_transport \"D-galactosamine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_galactosamine_transport \"D-galactosamine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _N_acetyl_D_galactosamine_degr_1_3_5_1_25_3_5_1_80_RXN \"(N-acetyl-)D-galactosamine-degr/1_3.5.1.25_3.5.1.80-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _N_acetyl_D_galactosamine_degr_1_3_5_1_25_3_5_1_80_RXN \"(N-acetyl-)D-galactosamine-degr/1_3.5.1.25_3.5.1.80-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _N_acetyl_D_galactosamine_degr_2_5_3_1_RXN_13548 \"(N-acetyl-)D-galactosamine-degr/2_5.3.1.-_RXN-13548\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _N_acetyl_D_galactosamine_degr_2_5_3_1_RXN_13548 \"(N-acetyl-)D-galactosamine-degr/2_5.3.1.-_RXN-13548\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _N_acetyl_D_galactosamine_degr_3_2_7_1_11_2_7_1_144_2_7_1_56_TAGAKIN_RXN \"(N-acetyl-)D-galactosamine-degr/3_2.7.1.11_2.7.1.144_2.7.1.56_TAGAKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _N_acetyl_D_galactosamine_degr_3_2_7_1_11_2_7_1_144_2_7_1_56_TAGAKIN_RXN \"(N-acetyl-)D-galactosamine-degr/3_2.7.1.11_2.7.1.144_2.7.1.56_TAGAKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_glucosaminate_exchange \"D-glucosaminate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_glucosaminate_exchange \"D-glucosaminate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_glucosaminate_transport \"D-glucosaminate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_glucosaminate_transport \"D-glucosaminate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_glucosaminate_degr_1_4_3_1_29_RXN_14730 \"D-glucosaminate-degr/1_4.3.1.29_RXN-14730\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_glucosaminate_degr_1_4_3_1_29_RXN_14730 \"D-glucosaminate-degr/1_4.3.1.29_RXN-14730\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_glucosaminate_degr_2_2_7_1_45_DEOXYGLUCONOKIN_RXN \"D-glucosaminate-degr/2_2.7.1.45_DEOXYGLUCONOKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_glucosaminate_degr_2_2_7_1_45_DEOXYGLUCONOKIN_RXN \"D-glucosaminate-degr/2_2.7.1.45_DEOXYGLUCONOKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_glucosaminate_degr_3_4_1_2_14_KDPGALDOL_RXN \"D-glucosaminate-degr/3_4.1.2.14_KDPGALDOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_glucosaminate_degr_3_4_1_2_14_KDPGALDOL_RXN \"D-glucosaminate-degr/3_4.1.2.14_KDPGALDOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_O_alpha_mannosyl_D_glycerate_exchange \"2-O-alpha-mannosyl-D-glycerate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_O_alpha_mannosyl_D_glycerate_exchange \"2-O-alpha-mannosyl-D-glycerate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_O_alpha_mannosyl_D_glycerate_transport_2_7_1_195 \"2-O-alpha-mannosyl-D-glycerate_transport_2.7.1.195\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_O_alpha_mannosyl_D_glycerate_transport_2_7_1_195 \"2-O-alpha-mannosyl-D-glycerate_transport_2.7.1.195\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_O_alpha_mannosyl_D_glycerate_degr_3_2_1_170_RXN0_5216 \"2-O-alpha-mannosyl-D-glycerate-degr_3.2.1.170_RXN0-5216\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_O_alpha_mannosyl_D_glycerate_degr_3_2_1_170_RXN0_5216 \"2-O-alpha-mannosyl-D-glycerate-degr_3.2.1.170_RXN0-5216\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_altronate_exchange \"D-altronate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_altronate_exchange \"D-altronate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_dehydro_3_deoxy_D_gluconate_exchange \"2-dehydro-3-deoxy-D-gluconate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_dehydro_3_deoxy_D_gluconate_exchange \"2-dehydro-3-deoxy-D-gluconate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _2_dehydro_3_deoxy_D_gluconate_transport \"2-dehydro-3-deoxy-D-gluconate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _2_dehydro_3_deoxy_D_gluconate_transport \"2-dehydro-3-deoxy-D-gluconate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_altronate_transport \"D-altronate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_altronate_transport \"D-altronate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction D_altronate_degr_4_2_1_7_ALTRODEHYDRAT_RXN \"D-altronate-degr_4.2.1.7_ALTRODEHYDRAT-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction D_altronate_degr_4_2_1_7_ALTRODEHYDRAT_RXN \"D-altronate-degr_4.2.1.7_ALTRODEHYDRAT-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction riboflavin_exchange \"riboflavin_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction riboflavin_exchange \"riboflavin_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction riboflavin_transport \"riboflavin_transport\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction riboflavin_transport \"riboflavin_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobinamide_exchange \"cobinamide_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobinamide_exchange \"cobinamide_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction coenzyme_B12_exchange \"coenzyme-B12_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction coenzyme_B12_exchange \"coenzyme-B12_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction coenzyme_B12_transport_3_6_3_33 \"coenzyme-B12_transport_3.6.3.33\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction coenzyme_B12_transport_3_6_3_33 \"coenzyme-B12_transport_3.6.3.33\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cobinamide_transport_3_6_3_33 \"cobinamide_transport_3.6.3.33\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cobinamide_transport_3_6_3_33 \"cobinamide_transport_3.6.3.33\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_salvage_from_cobinamide_1_2_5_1_17_BTUR2_RXN \"adenosylcobalamin-salvage-from-cobinamide/1_2.5.1.17_BTUR2-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_salvage_from_cobinamide_1_2_5_1_17_BTUR2_RXN \"adenosylcobalamin-salvage-from-cobinamide/1_2.5.1.17_BTUR2-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction adenosylcobalamin_salvage_from_cobinamide_2_2_7_1_156_COBINAMIDEKIN_RXN \"adenosylcobalamin-salvage-from-cobinamide/2_2.7.1.156_COBINAMIDEKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction adenosylcobalamin_salvage_from_cobinamide_2_2_7_1_156_COBINAMIDEKIN_RXN \"adenosylcobalamin-salvage-from-cobinamide/2_2.7.1.156_COBINAMIDEKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dethiobiotin_exchange \"dethiobiotin_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dethiobiotin_exchange \"dethiobiotin_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction dethiobiotin_transport \"dethiobiotin_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction dethiobiotin_transport \"dethiobiotin_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_biosynthesis_2_8_1_7_RXN_12588 \"biotin-biosynthesis_2.8.1.7_RXN-12588\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_biosynthesis_2_8_1_7_RXN_12588 \"biotin-biosynthesis_2.8.1.7_RXN-12588\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction biotin_biosynthesis_modnames_2_8_1_6_2_8_1_6_RXN \"biotin-biosynthesis_modnames_2.8.1.6_2.8.1.6-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction biotin_biosynthesis_modnames_2_8_1_6_2_8_1_6_RXN \"biotin-biosynthesis_modnames_2.8.1.6_2.8.1.6-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formylaminopyrimidine_exchange \"formylaminopyrimidine_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formylaminopyrimidine_exchange \"formylaminopyrimidine_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction formylaminopyrimidine_transport \"formylaminopyrimidine_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction formylaminopyrimidine_transport \"formylaminopyrimidine_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction base_degraded_thiamine_salvage_3_5_1_RXN_12612 \"base-degraded-thiamine-salvage_3.5.1.-_RXN-12612\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction base_degraded_thiamine_salvage_3_5_1_RXN_12612 \"base-degraded-thiamine-salvage_3.5.1.-_RXN-12612\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction base_degraded_thiamine_salvage_3_5_99_2_RXN_12613 \"base-degraded-thiamine-salvage_3.5.99.2_RXN-12613\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction base_degraded_thiamine_salvage_3_5_99_2_RXN_12613 \"base-degraded-thiamine-salvage_3.5.99.2_RXN-12613\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfolactate_exchange \"sulfolactate_exchange\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfolactate_exchange \"sulfolactate_exchange\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfolactate_transport \"sulfolactate_transport\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfolactate_transport \"sulfolactate_transport\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfolactate_degr_1_4_4_1_24_RXN_11691 \"sulfolactate-degr/1_4.4.1.24_RXN-11691\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfolactate_degr_1_4_4_1_24_RXN_11691 \"sulfolactate-degr/1_4.4.1.24_RXN-11691\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction sulfolactate_degr_2_spontaneous_RXN_8315 \"sulfolactate-degr/2_spontaneous_RXN-8315\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction sulfolactate_degr_2_spontaneous_RXN_8315 \"sulfolactate-degr/2_spontaneous_RXN-8315\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cGDP_2_7_7_65_RXN0_5359 \"cGDP_2.7.7.65_RXN0-5359\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cGDP_2_7_7_65_RXN0_5359 \"cGDP_2.7.7.65_RXN0-5359\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction cGDP_3_1_4_52_RXN0_4181 \"cGDP_3.1.4.52_RXN0-4181\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction cGDP_3_1_4_52_RXN0_4181 \"cGDP_3.1.4.52_RXN0-4181\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_1_RXN_6021 \"~no-pathway_1.1.1.1_RXN-6021\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_1_RXN_6021 \"~no-pathway_1.1.1.1_RXN-6021\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_1_RXN_12582 \"~no-pathway_1.1.1.1_RXN-12582\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_1_RXN_12582 \"~no-pathway_1.1.1.1_RXN-12582\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_22_UGD_RXN \"~no-pathway_1.1.1.22_UGD-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_22_UGD_RXN \"~no-pathway_1.1.1.22_UGD-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_2_1_3_RXN_4142 \"~no-pathway_1.2.1.3_RXN-4142\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_2_1_3_RXN_4142 \"~no-pathway_1.2.1.3_RXN-4142\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_2_1_3_AMINOBUTDEHYDROG_RXN \"~no-pathway_1.2.1.3_AMINOBUTDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_2_1_3_AMINOBUTDEHYDROG_RXN \"~no-pathway_1.2.1.3_AMINOBUTDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_86_R04440 \"~no-pathway_1.1.1.86_R04440\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_1_1_86_R04440 \"~no-pathway_1.1.1.86_R04440\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_5_1_49_PYRROLINE_2_CARBOXYLATE_REDUCTASE_RXN \"~no-pathway_1.5.1.49_PYRROLINE-2-CARBOXYLATE-REDUCTASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_5_1_49_PYRROLINE_2_CARBOXYLATE_REDUCTASE_RXN \"~no-pathway_1.5.1.49_PYRROLINE-2-CARBOXYLATE-REDUCTASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_5_1_49_PYRROLINE_2_CARBOXYLATE_REDUCTASE_RXN_WOP \"~no-pathway_1.5.1.49_PYRROLINE-2-CARBOXYLATE-REDUCTASE-RXN_WOP\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_5_1_49_PYRROLINE_2_CARBOXYLATE_REDUCTASE_RXN_WOP \"~no-pathway_1.5.1.49_PYRROLINE-2-CARBOXYLATE-REDUCTASE-RXN_WOP\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_8_1_14_COA_DISULFIDE_REDUCTASE_NADH_RXN \"~no-pathway_1.8.1.14_COA-DISULFIDE-REDUCTASE-NADH-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_8_1_14_COA_DISULFIDE_REDUCTASE_NADH_RXN \"~no-pathway_1.8.1.14_COA-DISULFIDE-REDUCTASE-NADH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_1_1_79_2_1_1_79_RXN \"~no-pathway_2.1.1.79_2.1.1.79-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_1_1_79_2_1_1_79_RXN \"~no-pathway_2.1.1.79_2.1.1.79-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_3_1_181_RXN0_947 \"~no-pathway_2.3.1.181_RXN0-947\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_3_1_181_RXN0_947 \"~no-pathway_2.3.1.181_RXN0-947\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_3_1_79_MALTACETYLTRAN_RXN \"~no-pathway_2.3.1.79_MALTACETYLTRAN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_3_1_79_MALTACETYLTRAN_RXN \"~no-pathway_2.3.1.79_MALTACETYLTRAN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_5_1_17_COBALADENOSYLTRANS_RXN \"~no-pathway_2.5.1.17_COBALADENOSYLTRANS-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_5_1_17_COBALADENOSYLTRANS_RXN \"~no-pathway_2.5.1.17_COBALADENOSYLTRANS-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_5_1_49_O_ACETYLHOMOSERINE_THIOL_LYASE_RXN \"~no-pathway_2.5.1.49_O-ACETYLHOMOSERINE-THIOL-LYASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_5_1_49_O_ACETYLHOMOSERINE_THIOL_LYASE_RXN \"~no-pathway_2.5.1.49_O-ACETYLHOMOSERINE-THIOL-LYASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_33_PANTETHEINE_KINASE_RXN \"~no-pathway_2.7.1.33_PANTETHEINE-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_33_PANTETHEINE_KINASE_RXN \"~no-pathway_2.7.1.33_PANTETHEINE-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_33_R04391 \"~no-pathway_2.7.1.33_R04391\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_33_R04391 \"~no-pathway_2.7.1.33_R04391\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_49_2_7_1_35_PYRAMKIN_RXN \"~no-pathway_2.7.1.49_2.7.1.35_PYRAMKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_49_2_7_1_35_PYRAMKIN_RXN \"~no-pathway_2.7.1.49_2.7.1.35_PYRAMKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_35_PYRIDOXKIN_RXN \"~no-pathway_2.7.1.35_PYRIDOXKIN-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_35_PYRIDOXKIN_RXN \"~no-pathway_2.7.1.35_PYRIDOXKIN-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_66_UNDECAPRENOL_KINASE_RXN \"~no-pathway_2.7.1.66_UNDECAPRENOL-KINASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_1_66_UNDECAPRENOL_KINASE_RXN \"~no-pathway_2.7.1.66_UNDECAPRENOL-KINASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_7_85_RXN_14338 \"~no-pathway_2.7.7.85_RXN-14338\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_7_85_RXN_14338 \"~no-pathway_2.7.7.85_RXN-14338\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_8_40_RXN_14571 \"~no-pathway_2.7.8.40_RXN-14571\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_8_40_RXN_14571 \"~no-pathway_2.7.8.40_RXN-14571\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_2_7_8_7_HOLO_ACP_SYNTH_RXN \"~no-pathway_2.7.8.7_HOLO-ACP-SYNTH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_2_7_8_7_HOLO_ACP_SYNTH_RXN \"~no-pathway_2.7.8.7_HOLO-ACP-SYNTH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction photorespiration_3_1_3_18_GPH_RXN \"photorespiration_3.1.3.18_GPH-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction photorespiration_3_1_3_18_GPH_RXN \"photorespiration_3.1.3.18_GPH-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_1_4_46_RXN_14073 \"~no-pathway_3.1.4.46_RXN-14073\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_1_4_46_RXN_14073 \"~no-pathway_3.1.4.46_RXN-14073\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_2_1_10_3_2_1_10_RXN \"~no-pathway_3.2.1.10_3.2.1.10-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_2_1_10_3_2_1_10_RXN \"~no-pathway_3.2.1.10_3.2.1.10-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_2_1_133_3_2_1_133_RXN \"~no-pathway_3.2.1.133_3.2.1.133-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_2_1_133_3_2_1_133_RXN \"~no-pathway_3.2.1.133_3.2.1.133-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_2_1_37_3_2_1_37_RXN \"~no-pathway_3.2.1.37_3.2.1.37-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_2_1_37_3_2_1_37_RXN \"~no-pathway_3.2.1.37_3.2.1.37-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_6_1_13_RXN0_1441 \"~no-pathway_3.6.1.13_RXN0-1441\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_6_1_13_RXN0_1441 \"~no-pathway_3.6.1.13_RXN0-1441\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction lactate_biosynthesis_fucose_degradation_4_1_2_17_FUCPALDOL_RXN \"lactate-biosynthesis_fucose-degradation_4.1.2.17_FUCPALDOL-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction lactate_biosynthesis_fucose_degradation_4_1_2_17_FUCPALDOL_RXN \"lactate-biosynthesis_fucose-degradation_4.1.2.17_FUCPALDOL-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_4_1_3_16_RXN_13990 \"~no-pathway_4.1.3.16_RXN-13990\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_4_1_3_16_RXN_13990 \"~no-pathway_4.1.3.16_RXN-13990\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_5_1_1_13_ASPARTATE_RACEMASE_RXN \"~no-pathway_5.1.1.13_ASPARTATE-RACEMASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_5_1_1_13_ASPARTATE_RACEMASE_RXN \"~no-pathway_5.1.1.13_ASPARTATE-RACEMASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_5_1_1_21_RXN_16430 \"~no-pathway_5.1.1.21_RXN-16430\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_5_1_1_21_RXN_16430 \"~no-pathway_5.1.1.21_RXN-16430\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _metabolism_of_biosynthesis_5_1_3_14_UDPGLCNACEPIM_RXN \"~metabolism-of-biosynthesis_5.1.3.14_UDPGLCNACEPIM-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _metabolism_of_biosynthesis_5_1_3_14_UDPGLCNACEPIM_RXN \"~metabolism-of-biosynthesis_5.1.3.14_UDPGLCNACEPIM-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_5_4_3_9_RXN_13569 \"~no-pathway_5.4.3.9_RXN-13569\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_5_4_3_9_RXN_13569 \"~no-pathway_5.4.3.9_RXN-13569\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_6_1_1_13_6_1_1_13_RXN \"~no-pathway_6.1.1.13_6.1.1.13-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_6_1_1_13_6_1_1_13_RXN \"~no-pathway_6.1.1.13_6.1.1.13-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_1_3_15_S_2_HYDROXY_ACID_OXIDASE_RXN \"~no-pathway_1.1.3.15_S-2-HYDROXY-ACID-OXIDASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_1_3_15_S_2_HYDROXY_ACID_OXIDASE_RXN \"~no-pathway_1.1.3.15_S-2-HYDROXY-ACID-OXIDASE-RXN\">'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_11_1_15_RXN0_5468 \"~no-pathway_1.11.1.15_RXN0-5468\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_11_1_15_RXN0_5468 \"~no-pathway_1.11.1.15_RXN0-5468\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_1_2_1_3_ALDHDEHYDROG_RXN \"~no-pathway_1.2.1.3_ALDHDEHYDROG-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_1_2_1_3_ALDHDEHYDROG_RXN \"~no-pathway_1.2.1.3_ALDHDEHYDROG-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_1_1_96_RXN_15041 \"~no-pathway_3.1.1.96_RXN-15041\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_1_1_96_RXN_15041 \"~no-pathway_3.1.1.96_RXN-15041\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_5_1_3_OMEGA_AMIDASE_RXN \"~no-pathway_3.5.1.3_OMEGA-AMIDASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_5_1_3_OMEGA_AMIDASE_RXN \"~no-pathway_3.5.1.3_OMEGA-AMIDASE-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_5_1_81_3_5_1_81_RXN \"~no-pathway_3.5.1.81_3.5.1.81-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_5_1_81_3_5_1_81_RXN \"~no-pathway_3.5.1.81_3.5.1.81-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_3_5_1_87_3_5_1_87_RXN \"~no-pathway_3.5.1.87_3.5.1.87-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_3_5_1_87_3_5_1_87_RXN \"~no-pathway_3.5.1.87_3.5.1.87-RXN\">'\n",
      "Missing lower flux bound set to '-1000.0' for  reaction: '<Reaction _no_pathway_5_5_1_6_CHALCONE_ISOMERASE_RXN \"~no-pathway_5.5.1.6_CHALCONE-ISOMERASE-RXN\">'\n",
      "Missing upper flux bound set to '1000.0' for  reaction: '<Reaction _no_pathway_5_5_1_6_CHALCONE_ISOMERASE_RXN \"~no-pathway_5.5.1.6_CHALCONE-ISOMERASE-RXN\">'\n",
      "No objective coefficients in model. Unclear what should be optimized\n",
      "Missing flux bounds on reactions set to default bounds.As best practise and to avoid confusion flux bounds should be set explicitly on all reactions.\n"
     ]
    }
   ],
   "source": [
    "# Read in models, published\n",
    "iMLTC806cdf = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/iMLTC806cdf/iMLTC806cdf.xml','Biomass')\n",
    "icdf834 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/icdf834/icdf834.sbml','Biomass')\n",
    "iHD992 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/iHD992/iHD992.sbml','Biomass')\n",
    "iCN900 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/iCN900/iCN900.xml','BIOMASS__5')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EX_h2o_b EX_h2o_b h2o_b\n",
      "EX_gluL_b EX_gluL_b gluL_b\n",
      "EX_pi_b EX_pi_b pi_b\n",
      "EX_co2_b EX_co2_b co2_b\n",
      "EX_pheL_b EX_pheL_b pheL_b\n",
      "EX_glyo_b EX_glyo_b glyo_b\n",
      "EX_h_b EX_h_b h_b\n",
      "EX_glcAD_b EX_glcAD_b glcAD_b\n",
      "EX_actid_b EX_actid_b actid_b\n",
      "EX_23diap_b EX_23diap_b 23diap_b\n",
      "EX_nh3_b EX_nh3_b nh3_b\n",
      "EX_isocap_b EX_isocap_b isocap_b\n",
      "EX_hoam_b EX_hoam_b hoam_b\n",
      "EX_2o6pamg2d_b EX_2o6pamg2d_b 2o6pamg2d_b\n",
      "EX_leuL_b EX_leuL_b leuL_b\n",
      "EX_cdpg_b EX_cdpg_b cdpg_b\n",
      "EX_dad_b EX_dad_b dad_b\n",
      "EX_escul_b EX_escul_b escul_b\n",
      "EX_for_b EX_for_b for_b\n",
      "EX_na_b EX_na_b na_b\n",
      "EX_glc_b EX_glc_b glc_b\n",
      "EX_uri_b EX_uri_b uri_b\n",
      "EX_no2_b EX_no2_b no2_b\n",
      "EX_glcn_b EX_glcn_b glcn_b\n",
      "EX_g3p_b EX_g3p_b g3p_b\n",
      "EX_sucr_b EX_sucr_b sucr_b\n",
      "EX_ncam_b EX_ncam_b ncam_b\n",
      "EX_ribD_b EX_ribD_b ribD_b\n",
      "EX_ura_b EX_ura_b ura_b\n",
      "EX_bio_b EX_bio_b bio_b\n",
      "EX_dcyt_b EX_dcyt_b dcyt_b\n",
      "EX_2dh3dg_b EX_2dh3dg_b 2dh3dg_b\n",
      "EX_hco3_b EX_hco3_b hco3_b\n",
      "EX_ac_b EX_ac_b ac_b\n",
      "EX_thymd_b EX_thymd_b thymd_b\n",
      "EX_adn_b EX_adn_b adn_b\n",
      "EX_gly_b EX_gly_b gly_b\n",
      "EX_23dhbz_b EX_23dhbz_b 23dhbz_b\n",
      "EX_asnL_b EX_asnL_b asnL_b\n",
      "EX_aspL_b EX_aspL_b aspL_b\n",
      "EX_fru_b EX_fru_b fru_b\n",
      "EX_metL_b EX_metL_b metL_b\n",
      "EX_4abg_b EX_4abg_b 4abg_b\n",
      "EX_4abz_b EX_4abz_b 4abz_b\n",
      "EX_arab_b EX_arab_b arab_b\n",
      "EX_c1ala_b EX_c1ala_b c1ala_b\n",
      "EX_malt_b EX_malt_b malt_b\n",
      "EX_ribflv_b EX_ribflv_b ribflv_b\n",
      "EX_thm_b EX_thm_b thm_b\n",
      "EX_tmam_b EX_tmam_b tmam_b\n",
      "EX_glyb_b EX_glyb_b glyb_b\n",
      "EX_mnl_b EX_mnl_b mnl_b\n",
      "EX_proL_b EX_proL_b proL_b\n",
      "EX_adman_b EX_adman_b adman_b\n",
      "EX_fe2_b EX_fe2_b fe2_b\n",
      "EX_dethio_b EX_dethio_b dethio_b\n",
      "EX_malS_b EX_malS_b malS_b\n",
      "EX_ca2_b EX_ca2_b ca2_b\n",
      "EX_escut_b EX_escut_b escut_b\n",
      "EX_k_b EX_k_b k_b\n",
      "EX_so3_b EX_so3_b so3_b\n",
      "EX_alaL_b EX_alaL_b alaL_b\n",
      "EX_serL_b EX_serL_b serL_b\n",
      "EX_tre_b EX_tre_b tre_b\n",
      "EX_glnL_b EX_glnL_b glnL_b\n",
      "EX_2mbut_b EX_2mbut_b 2mbut_b\n",
      "EX_ins_b EX_ins_b ins_b\n",
      "EX_spmd_b EX_spmd_b spmd_b\n",
      "EX_hxan_b EX_hxan_b hxan_b\n",
      "EX_chor_b EX_chor_b chor_b\n",
      "EX_4mpo_b EX_4mpo_b 4mpo_b\n",
      "EX_udpagaepyr_b EX_udpagaepyr_b udpagaepyr_b\n",
      "EX_xan_b EX_xan_b xan_b\n",
      "EX_mevR_b EX_mevR_b mevR_b\n",
      "EX_acnam_b EX_acnam_b acnam_b\n",
      "EX_lysL_b EX_lysL_b lysL_b\n",
      "EX_cysL_b EX_cysL_b cysL_b\n",
      "EX_valL_b EX_valL_b valL_b\n",
      "EX_at_b EX_at_b at_b\n",
      "EX_zn2_b EX_zn2_b zn2_b\n",
      "EX_ileL_b EX_ileL_b ileL_b\n",
      "EX_pyin_b EX_pyin_b pyin_b\n",
      "EX_xyld_b EX_xyld_b xyld_b\n",
      "EX_nac_b EX_nac_b nac_b\n",
      "EX_tyrL_b EX_tyrL_b tyrL_b\n",
      "EX_ptrc_b EX_ptrc_b ptrc_b\n",
      "EX_no3_b EX_no3_b no3_b\n",
      "EX_thrL_b EX_thrL_b thrL_b\n",
      "EX_ppat_b EX_ppat_b ppat_b\n",
      "EX_eto_b EX_eto_b eto_b\n",
      "EX_argL_b EX_argL_b argL_b\n",
      "EX_his_b EX_his_b his_b\n",
      "EX_sor_b EX_sor_b sor_b\n",
      "EX_chol_b EX_chol_b chol_b\n",
      "EX_4s45dhp23do_b EX_4s45dhp23do_b 4s45dhp23do_b\n",
      "EX_gal_b EX_gal_b gal_b\n",
      "EX_mg2_b EX_mg2_b mg2_b\n",
      "EX_5apn_b EX_5apn_b 5apn_b\n",
      "EX_pntoR_b EX_pntoR_b pntoR_b\n",
      "EX_isobuta_b EX_isobuta_b isobuta_b\n",
      "EX_aga_b EX_aga_b aga_b\n",
      "EX_etoa_b EX_etoa_b etoa_b\n",
      "EX_ival_b EX_ival_b ival_b\n",
      "EX_cobalt_b EX_cobalt_b cobalt_b\n",
      "EX_celb_b EX_celb_b celb_b\n",
      "EX_raf_b EX_raf_b raf_b\n",
      "EX_2aepat_b EX_2aepat_b 2aepat_b\n",
      "EX_trpL_b EX_trpL_b trpL_b\n",
      "EX_gtol_b EX_gtol_b gtol_b\n",
      "EX_ibtol_b EX_ibtol_b ibtol_b\n",
      "EX_metD_b EX_metD_b metD_b\n",
      "EX_tagd_b EX_tagd_b tagd_b\n",
      "EX_csn_b EX_csn_b csn_b\n",
      "EX_fuc_b EX_fuc_b fuc_b\n",
      "EX_h2_b EX_h2_b h2_b\n",
      "EX_dachi_b EX_dachi_b dachi_b\n",
      "EX_tgt_b EX_tgt_b tgt_b\n",
      "EX_biomass_b EX_biomass_b biomass_b\n",
      "Trans_tagd D-Tagaturonate transporter  h2o_c\n",
      "Trans_ppat phosphonate ABC transporter ATP-binding protein h2o_c\n",
      "Trans_gluL amino-acid ABC transport (L-glutamate) h2o_c\n",
      "Trans_asnL L-asparagine transporter h2o_c\n",
      "Trans_glyb glycine betaine ABC transporter h2o_c\n",
      "Trans_aspL2 L-aspartate transporter h2o_c\n",
      "Trans_ptrc putrescine ABC transport h2o_c\n",
      "Trans_metD D-methionine ABC transporter h2o_c\n",
      "Trans_chol choline ABC transporter h2o_c\n",
      "Trans_atp sodium extrusion ABC transporter h2o_c\n",
      "Trans_glnL amino-acid ABC transport (L-glutamine) h2o_c\n",
      "Trans_pi2 phosphate active transporter h2o_c\n",
      "Trans_argL L-arginine transporter h2o_c\n",
      "Trans_his amino-acid ABC transport (L-histidine) h2o_c\n",
      "Trans_tyrL amino-acid ABC transport (L-tyrosine) h2o_c\n",
      "Trans_zn2 zinc exporter ABC transport h2o_c\n",
      "Trans_g3p glycerol-3-phosphate-transporting ATPase h2o_c\n",
      "Trans_fe2 Iron atp driven transport h2o_c\n",
      "Trans_no2 Nitrate ATP transporter h2o_c\n",
      "Trans_cysL amino-acid ABC transport (L-cysteine) h2o_c\n",
      "Trans_c1ala Vitamin B12 transport h2o_c\n",
      "Trans_metL amino-acid ABC transport (L-methionine) h2o_c\n",
      "Trans_trpL amino-acid ABC transport (L-tryptophan) h2o_c\n",
      "Trans_spmd spermine ABC transport h2o_c\n",
      "Trans_lysL amino-acid ABC transport (L-lysine) h2o_c\n",
      "Trans_leuL amino-acid ABC transport (L-leucine) h2o_c\n",
      "Trans_alaL2 L-alanine transporter h2o_c\n",
      "Trans_pheL amino-acid ABC transport (L-phenylalanine) h2o_c\n",
      "Trans_cobalt The ATP-binding Cassette (ABC) Superfamily h2o_c\n",
      "Trans_ca Calcium ATP dependent transport h2o_c\n",
      "Trans_valL amino-acid ABC transport (L-valine) h2o_c\n",
      "Trans_proL amino-acid ABC transport (L-proline) h2o_c\n",
      "Trans_mg3 P-type magnesium transporter h2o_c\n",
      "Trans_thrL amino-acid ABC transport (L-threonine) h2o_c\n",
      "Trans_k potassium transport h2o_c\n",
      "Trans_gly amino-acid ABC transport (glycine) h2o_c\n",
      "Trans_mg2 magnesium ABC transporter h2o_c\n",
      "Trans_serL amino-acid ABC transport (L-serine) h2o_c\n",
      "Trans_ileL amino-acid ABC transport (L-isoleucine) h2o_c\n",
      "Trans_ribD ribose ABC transporter h2o_c\n",
      "ID_470 asparagine synthase (glutamine-hydrolysing) h2o_c\n",
      "ID_222 aspartate-ammonia ligase atp_c\n",
      "ID_88 succinate semialdehyde:NAD+ oxidoreductase h2o_c\n",
      "ID_317 Acetamide amidohydrolase h2o_c\n",
      "ID_575 L-glutamate:NAD+ oxidoreductase (deaminating) h2o_c\n",
      "ID_712 4-aminobutyrate:2-oxoglutarate aminotransferase 2oglut_c\n",
      "ID_411 succinate semialdehyde:NADP+ oxidoreductase h2o_c\n",
      "ID_439 aspartate:2-oxoglutarate aminotransferase gluL_c\n",
      "ID_445 glutamine-fructose-6-phosphate transaminase (isomerizing) fru6p_c\n",
      "ID_142 4-aminobutanoate:pyruvate aminotransferase pyr_c\n",
      "ID_12 glutamate racemase gluL_c\n",
      "ID_365 argininosuccinate lyase argsuc_c\n",
      "ID_92 asparaginase h2o_c\n",
      "ID_543 glutamate synthase (NADPH) h_c\n",
      "ID_675 L-glutamine aminohydrolase h2o_c\n",
      "ID_688 4-aminobutyrate transaminase 2oglut_c\n",
      "ID_336 L-alanine:2-oxoglutarate aminotransferase 2oglut_c\n",
      "ID_759 4-aminobutanoate:glyoxylate aminotransferase 4abat_c\n",
      "ID_506 UDP-N-acetylmuramate-L-alanine ligase atp_c\n",
      "ID_3 glutathione gamma-glutamylaminopeptidase gth_c\n",
      "ID_428 UDP-N-acetylmuramoyl-L-alanine-D-glutamate ligase atp_c\n",
      "ID_526 glutamate-ammonia ligase gluL_c\n",
      "ID_357 alanine racemase alaL_c\n",
      "ID_346 D-alanine-D-alanine ligase atp_c\n",
      "ID_396 L-Serine:pyruvate aminotransferase pyr_c\n",
      "ID_446 argininosuccinate synthase atp_c\n",
      "ID_561 L-glutamate:NAD+ oxidoreductase (transaminating) nadh_c\n",
      "ID_583 malonate semialdehide decarboxylase malsald_c\n",
      "ID_347 aspartate 1-decarboxylase aspL_c\n",
      "ID_380 glucosamine-6-phosphate deaminase h2o_c\n",
      "ID_571 ATP:N-acetyl-D-glucosamine 6-phosphotransferase atp_c\n",
      "ID_331 UDP-N-acetylglucosamine diphosphorylase utp_c\n",
      "ID_227 phosphoglucosamine mutase ga6p_c\n",
      "ID_594 glucosamine-1-phosphate N-acetyltransferase acoa_c\n",
      "ID_514 N-acetylglucosamine-6-phosphate deacetylase h2o_c\n",
      "ID_548 UDP-N-acetyl-D-glucosamine 2-epimerase (UDP-forming) h2o_c\n",
      "ID_301 UDP-N-acetylglucosamine 1-carboxyvinyltransferase pepyr_c\n",
      "ID_176 UDP-N-acetylmuramate:NAD+ oxidoreductase nad_c\n",
      "ID_194 N-acetylneuraminate pyruvate-lyase acnam_c\n",
      "ID_727 N-acetyl-D-glucosamine 6-phosphate 2-epimerase ama6p_c\n",
      "ID_741 ATP:N-acetyl-D-mannosamine 6-phosphotransferase atp_c\n",
      "ID_223 UDP-N-acetylmuramate dehydrogenase h_c\n",
      "ID_121 UDP-N-acetyl-D-glucosamine 2-epimerase udpaga_c\n",
      "ID_602 L-arginine carboxy-lyase argL_c\n",
      "ID_367 allophanate hydrolase h2o_c\n",
      "ID_493 acetyl-CoA:putrescine N-acetyltransferase acoa_c\n",
      "ID_106 N-acetyl-gamma-glutamyl-phosphate reductase h_c\n",
      "ID_638 agmatine amidinohydrolase h2o_c\n",
      "ID_467 ornithine carbamoyltransferase cbp_c\n",
      "ID_747 acetylornithine amidohydrolase h2o_c\n",
      "ID_278 N2-Acetyl-L-ornithine:L-glutamate N-acetyltransferase gluL_c\n",
      "ID_42 4-aminobutyraldehyde:NAD+ oxidoreductase h2o_c\n",
      "ID_635 acetylglutamate kinase atp_c\n",
      "ID_311 amino-acid N-acetyltransferase acoa_c\n",
      "ID_135 proline racemase proL_c\n",
      "ID_627 urea:bicarbonate ligase atp_c\n",
      "ID_295 putrescine aminotransferase pyr_c\n",
      "ID_748 acetylornithine transaminase gluL_c\n",
      "ID_337 L-arginine iminohydrolase h2o_c\n",
      "Bio_CW Cell wall synthesis teichoic_c\n",
      "Biomass Biomass Reaction h2o_c\n",
      "Bio_RNA RNA synthesis h2o_c\n",
      "Bio_DNA DNA synthesis h2o_c\n",
      "Bio_CLP Peptidoglycan synthesis h2o_c\n",
      "Bio_prot Protein synthesis h2o_c\n",
      "Bio_lip Lipid composition mqn_c\n",
      "Bio_SPs Solute Pool acoa_c\n",
      "ID_525 acetyl-CoA:4-hydroxybutanoate CoA-transferase acoa_c\n",
      "ID_616 vinylacetyl-CoA delta3-delta2-isomerase vacoa_c\n",
      "ID_530 butanal:NADP+ oxidoreductase (CoA-acylating) h_c\n",
      "ID_519 (S)-3-hydroxybutanoyl-CoA:NADP+ oxidoreductase nadp_c\n",
      "ID_72 butanoyl-CoA:acetate CoA-transferase btcoa_c\n",
      "ID_502 (S)-acetoin:NAD+ oxidoreductase da_c\n",
      "ID_156 4-hydroxybutanoyl-CoA hydro-lyase 4hcoa_c\n",
      "ID_615 (R)-acetoin:NAD+ oxidoreductase da_c\n",
      "ID_504 trans-2-enoyl-CoA reductase (NADP+) h_c\n",
      "ID_507 succinate semialdehyde hydrogenase nadh_c\n",
      "ID_18 (R)-3-hydroxybutanoate:NAD+ oxidoreductase nad_c\n",
      "ID_463 oxidoreductase acting on the CH-OH group of donors with NAD+ as acceptor nadh_c\n",
      "ID_500 carbonate hydro-lyase h2o_c\n",
      "ID_379 (R)-acetoin racemase actnR_c\n",
      "ID_175 butanal:NAD+ oxidoreductase (CoA-acylating) nadh_c\n",
      "ID_6 2-acetolactate decarboxylation 2alac_c\n",
      "ID_287 Glutaconyl-CoA carboxy-lyase gccoa_c\n",
      "ID_272 phosphate butyryltransferase pi_c\n",
      "ID_21 oxidoreductase acting on the CH-OH group of donors with NADP+ as acceptor h_c\n",
      "ID_251 butyrate kinase adp_c\n",
      "ID_523 acetolactate decarboxylase 2alac_c\n",
      "Trans_mg magnesium transporter mg2_e\n",
      "Trans_k4 potassium:H+ antiporter h_c\n",
      "Trans_for Formate transport for_e\n",
      "Trans_no nitrite transporter no2_e\n",
      "Trans_k3 potassium ion chanel k_e\n",
      "Trans_zn zinc transporter zn2_e\n",
      "ID_61 isocitrate:NADP+ oxidoreductase nadp_c\n",
      "ID_708 2-oxoglutarate dehydrogenase complex coa_c\n",
      "ID_370 pyruvate carboxylase pyr_c\n",
      "ID_433 citrate hydro-lyase cit_c\n",
      "ID_171 isocitrate dehydrogenase (NAD+) nad_c\n",
      "ID_544 acetyl-CoA:oxaloacetate C-acetyltransferase acoa_c\n",
      "ID_310 succinyl-CoA synthase coa_c\n",
      "ID_606 isocitrate hydro-lyase icit_c\n",
      "ID_198 fumarate hydratase malS_c\n",
      "ID_254 malate:NAD+ oxidoreductase nad_c\n",
      "ID_557 methionine adenosyltransferase h2o_c\n",
      "ID_99 acetyl-CoA:L-homoserine O-acetyltransferase acoa_c\n",
      "ID_231 adenosylhomocysteinase h2o_c\n",
      "ID_2 spermine N1-acetyltransferase sprm_c\n",
      "ID_562 5-(methylthio)-2,3-dioxopentyl-phosphate phosphohydrolase (isomerizing) h2o_c\n",
      "ID_152 methionine synthase hcys_c\n",
      "ID_265 2-hydroxy-3-keto-5-methylthio-1-phosphopentene phosphohydrolase h2o_c\n",
      "ID_714 2,3-diketo-5-methylthiopentenyl-1-phosphate enolase 5mtr1p_c\n",
      "ID_687 cystathionine gamma-synthase cysL_c\n",
      "ID_711 S-adenosylmethioninamine:putrescine 3-aminopropyltransferase ametam_c\n",
      "ID_680 S-adenosyl-L-methionine carboxy-lyase h_c\n",
      "ID_469 cystathionine gamma-lyase h2o_c\n",
      "ID_242 L-cystyeine sulfurtransferase h_c\n",
      "ID_418 spermidine N1-acetyltransferase acoa_c\n",
      "ID_155 S-adenosylmethioninamine:spermidine 3-aminopropyltransferase ametam_c\n",
      "ID_560 methionine racemase metL_c\n",
      "ID_189 S-adenosyl-L-homocysteine homocysteinylribohydrolase h2o_c\n",
      "ID_235 5-methylthioribulose-1-phosphate hydro-lyase 5mtr1p_c\n",
      "ID_136 5-(methylthio)-2,3-dioxopentyl phosphate keto-enol-isomerase 5mt23dopp_c\n",
      "ID_98 L-methionine methanothiol-lyase h2o_c\n",
      "ID_259 homoserine O-succinyltransferase succoa_c\n",
      "ID_358 O-acetyl-L-homoserine:methanethiol 3-amino-3-carboxypropyltransferase ch4s_c\n",
      "ID_309 S-(5-deoxy-D-ribos-5-yl)-L-homocysteine homocysteine-lyase [(4S)-4,5-dihydroxypentan-2,3-dione-forming]indole-3-acetate s5ddr5lhcys_c\n",
      "ID_410 ATP:S-methyl-5-thio-D-ribose 1-phosphotransferase atp_c\n",
      "ID_737 S-methyl-5-thio-alpha-D-ribose 1-phosphate ketol-isomerase m5trib1p_c\n",
      "ID_127 methylthioadenosine methylthioribohydrolase h2o_c\n",
      "ID_94 cystathionine beta-lyase h2o_c\n",
      "ID_167 L-homocysteine Methyltransferases amet_c\n",
      "ID_664 ornithine racemase ornL_c\n",
      "ID_443 2-amino-4-ketopentanoate thiolase coa_c\n",
      "ID_243 D-ornithine 4,5-aminomutase ornD_c\n",
      "ID_319 D-threo-2,4-diaminopentanoate:NADP+ oxidoreductase h2o_c\n",
      "ID_193 D-threo-2,4-diaminopentanoate:NAD+ oxidoreductase h2o_c\n",
      "Ex_ca2 Exchange Ca2+ ca2_b\n",
      "Ex_tyrL Exchange L-tyrosine tyrL_b\n",
      "Ex_chor Exchange chorismate chor_b\n",
      "Ex_pheL Exchange L-phenylalanine pheL_b\n",
      "Ex_fru Exchange D-fructose fru_b\n",
      "Ex_serL Exchange L-serine serL_b\n",
      "Ex_cysL Exchange L-cysteine cysL_b\n",
      "Ex_adn Exchange adenosine adn_b\n",
      "Ex_dcyt Exchange deoxycytidine dcyt_b\n",
      "Ex_mnl Exchange mannitol mnl_b\n",
      "Ex_tagd Exchange D-Tagaturonate tagd_b\n",
      "Ex_k Exchange K+ k_b\n",
      "Ex_so3 Exchange sulfite so3_b\n",
      "Ex_chol Exchange choline chol_b\n",
      "Ex_uri Exchange uridine uri_b\n",
      "Ex_ura Exchange uracil ura_b\n",
      "Ex_glc Exchange beta-D-glucose glc_b\n",
      "Ex_nac Exchange nicotinate nac_b\n",
      "Ex_argL Exchange L-arginine argL_b\n",
      "Ex_udpagaepyr Exchange UDP-N-acetylglucosamine-enolpyruvate udpagaepyr_b\n",
      "Ex_hxan Exchange hypoxanthine hxan_b\n",
      "Ex_metL Exchange L-methionine metL_b\n",
      "Ex_thm Exchange thiamin thm_b\n",
      "Ex_proL Exchange L-proline proL_b\n",
      "Ex_aspL Exchange L-aspartate aspL_b\n",
      "Ex_ribflv Exchange riboflavin ribflv_b\n",
      "Ex_hco3 Exchange bicarbonate hco3_b\n",
      "Ex_h Exchange H+ h_b\n",
      "Ex_valL Exchange L-valine valL_b\n",
      "Ex_4abz Exchange 4-aminobenzoate 4abz_b\n",
      "Ex_thymd Exchange thymidine thymd_b\n",
      "Ex_pntoR Exchange (R)-pantothenate pntoR_b\n",
      "Ex_dad Exchange deoxyadenosine dad_b\n",
      "Ex_g3p Exchange sn-glycerol-3-phosphate g3p_b\n",
      "Ex_na Exchange Na+ na_b\n",
      "Ex_gly Exchange glycine gly_b\n",
      "Ex_h2o Exchange H2O h2o_b\n",
      "Ex_pyin Exchange pyridoxine pyin_b\n",
      "Ex_leuL Exchange L-leucine leuL_b\n",
      "Ex_ileL Exchange L-isoleucine ileL_b\n",
      "Ex_cobalt Exchange Co2+ cobalt_b\n",
      "Ex_zn2 Exchange Zn2+ zn2_b\n",
      "Ex_pi Exchange phosphate pi_b\n",
      "Ex_gluL Exchange L-glutamate gluL_b\n",
      "Ex_glnL Exchange L-glutamine glnL_b\n",
      "Ex_asnL Exchange L-asparagine asnL_b\n",
      "Ex_thrL Exchange L-threonine thrL_b\n",
      "Ex_trpL Exchange L-tryptophan trpL_b\n",
      "Ex_nh3 Exchange ammonia nh3_b\n",
      "Ex_co2 Exchange CO2 co2_b\n",
      "Ex_alaL Exchange L-alanine alaL_b\n",
      "Ex_ribD Exchange D-ribose ribD_b\n",
      "Ex_h2 Exchange H2 h2_b\n",
      "Ex_his Exchange L-histidine his_b\n",
      "Ex_bio Exchange biotin bio_b\n",
      "Ex_mg2 Exchange Mg2+ mg2_b\n",
      "Ex_lysL Exchange L-lysine lysL_b\n",
      "Ex_ins Exchange inosine ins_b\n",
      "Ex_c1ala Exchange cob(I)alamin c1ala_b\n",
      "Ex_ncam Exchange nicotinamide ncam_b\n",
      "ID_400 butanoyl-CoA dehydrogenase (FAD) fad_c\n",
      "ID_325 enoyl-CoA hydratase 3hbcoa_c\n",
      "ID_481 acetyl-CoA C-acetyltransferase coa_c\n",
      "ID_390 3-hydroxyacyl-CoA dehydrogenase nad_c\n",
      "ID_471 acyl-CoA hydrolase h2o_c\n",
      "ID_60 Beta-Oxidation of Palmitoyl-CoA coa_c\n",
      "ID_409 (S)-3-Hydroxybutanoyl-CoA 3-epimerase 3hbcoa_c\n",
      "ID_341 dihydroneopterin monophosphate dephosphorylase h2o_c\n",
      "ID_149 6-hydroxymethyl-7,8-dihydropteridine:4-aminobenzoate 2-amino-4-hydroxydihydropteridine-6-methenyltransferase 4abz_c\n",
      "ID_457 dihydroneopterin triphosphate pyrophosphohydrolase h2o_c\n",
      "ID_180 2-amino-4-hydroxy-6-hydroxymethyldihydropteridine diphosphokinase atp_c\n",
      "ID_495 4-amino-4-deoxychorismate synthase glnL_c\n",
      "ID_75 dihydrofolate synthase gluL_c\n",
      "ID_438 4-amino-4-deoxychorismate pyruvate-lyase 4a4dchor_c\n",
      "ID_104 D-aminocylase h2o_c\n",
      "ID_344 dihydropteroate synthase 4abz_c\n",
      "ID_524 GTP 7,8-8,9-dihydrolase h2o_c\n",
      "ID_172 dihydroneopterin aldolase 78dhnpte_c\n",
      "ID_64 phosphomannomutase man6p_c\n",
      "ID_219 1-phosphofructokinase atp_c\n",
      "ID_545 UTP:D-fructose-6-phosphate 1-phosphotransferase utp_c\n",
      "ID_541 D-sobitol-6-phosphate:NAD+ 2-oxidoreductase nad_c\n",
      "ID_726 mannose-1-phosphate guanylyltransferase (GDP) gdp_c\n",
      "ID_610 ATP:D-hexose 6 phosphotransferase alpha galactose atp_c\n",
      "ID_253 ATP:D-hexose 6 phosphotransferase mannose atp_c\n",
      "ID_276 fructose-bisphosphatase h2o_c\n",
      "ID_567 sedoheptulose-7-phosphate:D-glyceraldehyde-3-phosphate glyceronetransferase shl7p_c\n",
      "ID_542 ATP:D-fructose 6-phosphotransferase atp_c\n",
      "ID_732 D-fructose 1,6-bisphosphate D-glyceraldehyde-3-phosphate-lyase dhap_c\n",
      "ID_302 ATP:D-hexose 6 phosphotransferase beta galactose atp_c\n",
      "ID_353 L-fuculose-1-phosphate lactaldehyde-lyase fuc1p_c\n",
      "ID_290 ITP:D-fructose-6-phosphate 1-phosphotransferase itp_c\n",
      "ID_674 sedoheptulose 1,7-bisphosphate D-glyceraldehyde-3-phosphate-lyase sh17bp_c\n",
      "ID_613 ATP:sedoheptulose-7-phosphate 1-phosphotransferase atp_c\n",
      "ID_436 alpha-D-Glucose aldose-ketose-isomerase glcAD_c\n",
      "ID_649 mannose-6-phosphate isomerase man6p_c\n",
      "ID_611 mannitol-1-phosphate 5-dehydrogenase nad_c\n",
      "ID_676 sorbitol dehydrogenase nad_c\n",
      "ID_527 CTP:D-fructose-6-phosphate 1-phosphotransferase fru6p_c\n",
      "ID_173 UTP hexose-1-phosphate uridylyltransferase utp_c\n",
      "ID_199 fructose-bisphosphate aldolase fru1p_c\n",
      "ID_297 ATP:D-fructose-6-phosphate 1-phosphotransferase atp_c\n",
      "ID_659 mannose-1-phosphate guanylyltransferase (GTP) gtp_c\n",
      "ID_333 Fructokinase atp_c\n",
      "ID_491 CTP:tagatose-6-phosphate 1-phosphotransferase ctp_c\n",
      "ID_329 D-galactose 1-epimerase gal_c\n",
      "ID_255 D-glucose 1-phosphate 1-epimerase glc1pA_c\n",
      "ID_532 ATP:tagatose-6-phosphate 1-phosphotransferase atp_c\n",
      "ID_742 Galactitol-1-phosphate:NAD+ oxidoreductase nad_c\n",
      "ID_417 UDP-D-glucose:alpha-D-galactose 1-phosphate uridylyltransferase udpglu_c\n",
      "ID_444 UDP-glucose 4-epimerase udpgal_c\n",
      "ID_480 UTP:tagatose-6-phosphate 1-phosphotransferase utp_c\n",
      "ID_763 galactokinase atp_c\n",
      "ID_205 tagatose-bisphosphate aldolase tag16bp_c\n",
      "ID_631 ITP:tagatose-6-phosphate 1-phosphotransferase itp_c\n",
      "ID_633 galactose-6-phosphate isomerase gal6p_c\n",
      "ID_9 glycerol dehydrogenase (NAD+) glyo_c\n",
      "ID_429 1,2-diacylglycerol 3-glucosyltransferase udpglu_c\n",
      "ID_298 glycerol dehydrogenase (NADP+) glyo_c\n",
      "ID_549 acyl-CoA:1-acyl-sn-glycerol-3-phosphate 2-O-acyltransferase lcacoa_c\n",
      "ID_455 diacylglycerol kinase adp_c\n",
      "ID_678 aldehyde dehydrogenase (NAD+) h2o_c\n",
      "ID_273 palmitate biosynthesis acoa_c\n",
      "ID_477 UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminoheptanedioate:D-alanyl-D-alanine ligase atp_c\n",
      "ID_86 hexosyltransferase 3g12dag_c\n",
      "ID_564 glycerol kinase atp_c\n",
      "ID_224 phosphatidylglycerol-membrane-oligosaccharide dgdag_c\n",
      "ID_740 Acyl-CoA thioesterase coa_c\n",
      "ID_109 glycerol-3-phosphate dehydrogenase (NAD+) nadh_c\n",
      "ID_204 CDP-diacylglycerol phosphatidylhydrolase h2o_c\n",
      "ID_588 CDP-diacylglycerol-glycerol-3-phosphate 3-phosphatidyltransferase g3p_c\n",
      "ID_56 glycerol-3-phosphate dehydrogenase (NADP+) h_c\n",
      "ID_546 phosphatidate cytidylyltransferase ctp_c\n",
      "ID_78 glycerol-3-phosphate O-acyltransferase g3p_c\n",
      "ID_185 phosphatidylglycerophosphatase h2o_c\n",
      "ID_232 cardiolipin biosynthesis ptg_c\n",
      "ID_267 glycerol-3-phosphate cytidylyltransferase g3p_c\n",
      "ID_340 ethanolamine ammonia-lyase etoa_c\n",
      "ID_249 2-amino-3-oxobutanoate CoA-ligase coa_c\n",
      "ID_751 Glycine dehydrogenase (aminomethyl-transferring) gly_c\n",
      "ID_501 D-serine ammonia-lyase serD_c\n",
      "ID_165 homoserine dehydrogenase (NADP+) h_c\n",
      "ID_283 phosphoserine phosphatase h2o_c\n",
      "ID_494 aspartate kinase atp_c\n",
      "ID_556 threonine aldolase thrL_c\n",
      "ID_625 L-threonine:NAD+ oxidoreductase nad_c\n",
      "ID_240 threonine synthase h2o_c\n",
      "ID_130 betaine aldehyde:NAD+ oxidoreductase h2o_c\n",
      "ID_536 L-cysteinylglycine dipeptidase h2o_c\n",
      "ID_293 choline dehydrogenase (NADP+) nadp_c\n",
      "ID_330 L-threonine dehydratase thrL_c\n",
      "ID_143 glycine hydroxymethyltransferase thf_c\n",
      "ID_452 phosphoglycerate dehydrogenase nad_c\n",
      "ID_743 serine racemase serL_c\n",
      "ID_665 3-phospho-L-serinephosphoserine transaminase gluL_c\n",
      "ID_132 homoserine dehydrogenase (NAD+) nadh_c\n",
      "ID_124 2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate 6-phosphomannohydrolase h2o_c\n",
      "ID_609 glycerate kinase atp_c\n",
      "ID_385 homoserine kinase atp_c\n",
      "ID_52 aspartate-semialdehyde dehydrogenase h_c\n",
      "ID_320 L-serine ammonia-lyase serL_c\n",
      "ID_694 succinyl-CoA:glycine C-succinyltransferase succoa_c\n",
      "ID_729 choline dehydrogenase (NAD+) nad_c\n",
      "ID_393 glyceraldehyde-3-phosphate dehydrogenase (phosphorylating) NADP h_c\n",
      "ID_55 alpha-D-glucose 6-phosphate ketol-isomerase glc6pA_c\n",
      "ID_496 ATP:oxaloacetate carboxylyase atp_c\n",
      "ID_622 D-glyceraldehyde 3-phosphate:NADP+ oxidoreductase h2o_c\n",
      "ID_360 acetaldehyde:NAD+ oxidoreductase (acetate forming) h2o_c\n",
      "ID_401 phosphoglucomutase glc6pA_c\n",
      "ID_221 dihydrolipoamide dehydrogenase nad_c\n",
      "ID_53 pyruvate:ferredoxin oxidoreductase pyr_c\n",
      "ID_179 phosphoglycerate mutase 2pg_c\n",
      "ID_414 ATP:alpha-D-glucose 6-phosphotransferase atp_c\n",
      "ID_621 beta-D-glucose 6-phosphate ketol-isomerase fru6p_c\n",
      "ID_498 glucose-6-phosphate 1-epimerase glc6p_c\n",
      "ID_285 glyceraldehyde-3-phosphate dehydrogenase (phosphorylating) nadh_c\n",
      "ID_233 phosphoglycerate kinase atp_c\n",
      "ID_660 L-lactate dehydrogenase pyr_c\n",
      "ID_39 D-glucose 1-phosphate phosphohydrolase h2o_c\n",
      "ID_657 ATP:beta-D-glucose 6-phosphotransferase atp_c\n",
      "ID_147 D-glucose 1-epimerase glcAD_c\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID_721 D-glucose aldose-ketose-isomerase glc_c\n",
      "ID_596 triose-phosphate isomerase glyald3p_c\n",
      "ID_391 phosphopyruvate hydratase pepyr_c\n",
      "ID_140 acetate:CoA ligase (AMP-forming) coa_c\n",
      "ID_383 ethanol dehydrogenase nad_c\n",
      "ID_252 ATP:pyruvate 2-O-phosphotransferase pepyr_c\n",
      "ID_654 AMP:fructose-1,6-bisphosphate phosphotransferase amp_c\n",
      "ID_473 gluconolactonase h2o_c\n",
      "ID_339 hydroxypyruvate isomerase tartsald_c\n",
      "ID_753 L-Serine:glyoxylate aminotransferase serL_c\n",
      "ID_754 glycolate fermentation h_c\n",
      "ID_362 (3R)-3-Hydroxybutanoyl-CoA hydro-lyase r3hbatcoa_c\n",
      "ID_752 Glycolaldehyde dehydrogenase h2o_c\n",
      "ID_128 D-glycerate:NAD+ oxidoreductase nadh_c\n",
      "ID_374 glycine cleavage nad_c\n",
      "ID_535 2-hydroxy-3-oxopropionate reductase h_c\n",
      "ID_90 formate dehydrogenase nad_c\n",
      "ID_230 glycine oxidase h2o_c\n",
      "ID_211 10-formyltetrahydrofolate aminohydrolase h2o_c\n",
      "ID_151 imidazole glycerol phosphate synthase glnL_c\n",
      "ID_196 phosphoribosyl-AMP cyclohydrolase h2o_c\n",
      "ID_361 histidinol:NAD+ oxidoreductase nad_c\n",
      "ID_215 phosphoribosyl-ATP diphosphatase h2o_c\n",
      "ID_76 histidinol-phosphatase h2o_c\n",
      "ID_191 N-carbamoul-L-aspartate hydrolase co2_c\n",
      "ID_268 1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino]imidazole-4-carboxamide isomerase 15pr55pramai4c_c\n",
      "ID_338 ATP phosphoribosyltransferase atp_c\n",
      "ID_713 histidinal:NAD+ oxidoreductase h2o_c\n",
      "ID_164 histidinol-phosphate transaminase gluL_c\n",
      "ID_66 imidazoleglycerol-phosphate dehydratase eimzgp_c\n",
      "ID_28 L-Leucine:2-oxoglutarate aminotransferase leuL_c\n",
      "ID_31 acyl-CoA dehydrogenase nadh_c\n",
      "ID_605 2-hydroxyisocaproate CoA transferase coa_c\n",
      "ID_382 2-hydroxyisocaproyl-CoA dehydratase h_c\n",
      "ID_685 2-hydroxyisocaproate dehydrogenase nadh_c\n",
      "ID_20 Isocaprenoyl-CoA CoA transferase h2o_c\n",
      "ID_550 dihydrodipicolinate synthase pyr_c\n",
      "ID_181 (R)-2-hydroxybutane-1,2,4-tricarboxylate hydro-lyase [(1R,2S)-1-hydroxybutane-1,2,4-tricarboxylate-forming] hcit_c\n",
      "ID_626 succinyl-diaminopimelate desuccinylase h2o_c\n",
      "ID_686 UDP-N-acetylmuramoyl-L-alanyl-D-glutamate:meso-2,6-diaminoheptanedioate ligase atp_c\n",
      "ID_115 diaminopimelate epimerase dap_c\n",
      "ID_82 (1R,2S)-1-hydroxybutane-1,2,4-tricarboxylate:NAD+ oxidoreductase (decarboxylating) nad_c\n",
      "ID_725 acetyl-CoA:2-oxoglutarate C-acetyltransferase (thioester-hydrolysing, carboxymethyl forming) acoa_c\n",
      "ID_120 dihydrodipicolinate reductase (NADP+) h_c\n",
      "ID_701 dihydrodipicolinate reductase (NAD+) nadh_c\n",
      "ID_462 succinyldiaminopimelate transaminase gluL_c\n",
      "ID_652 2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-succinyltransferase h2o_c\n",
      "ID_479 diaminopimelate decarboxylase dapime_c\n",
      "ID_434 L-lysine 2,3-aminomutase lysL_c\n",
      "ID_184 (3S,5S)-3,5-diaminohexanoate:NAD+ oxidoreductase h2o_c\n",
      "ID_349 3-keto-5-aminohexanoate cleavage acoa_c\n",
      "ID_113 L-3-aminobutyryl-CoA amminoa-lyase 3abcoa_c\n",
      "ID_574 (3S)-3,6-diaminohexanoate 5,6-aminomutase 36dah_c\n",
      "ID_288 butyrate-acetoacetate CoA-transferase aac_c\n",
      "ID_531 N-Ribosylnicotinamide:orthophosphate ribosyltransferase pi_c\n",
      "ID_174 nicotinate-nucleotide diphosphorylase (carboxylating) h_c\n",
      "ID_395 Nicotinamide amidohydrolase h2o_c\n",
      "ID_551 nicotinamide-nucleotide adenylyltransferase (NAD+ forming) atp_c\n",
      "ID_218 deamino-NAD+:ammonia ligase atp_c\n",
      "ID_27 nicotinamide-nucleotide adenylyltransferase atp_c\n",
      "ID_58 N-ribosylnicotinamide ribohydrolase h2o_c\n",
      "ID_294 L-aspartate:NADP+ oxidoreductase (deaminating) nadp_c\n",
      "ID_313 NAD+ synthase (glutamine-hydrolysing) h2o_c\n",
      "ID_304 L-aspartate:NAD+ oxidoreductase (deaminating) nad_c\n",
      "ID_262 nicotinate mononucleotide:diphosphate phosphoribosyltransferase h2o_c\n",
      "ID_533 Nicotinamide ribonucleotide phosphohydrolase h2o_c\n",
      "ID_580 NAD+ phosphohydrolase h2o_c\n",
      "ID_41 NADP+ 2'-phosphohydrolase h2o_c\n",
      "ID_399 alpha-iminosucinate:dihydroxyacetone phosphate alkyltransferase dhap_c\n",
      "ID_559 NAD+ glycohydrolase h2o_c\n",
      "ID_415 NAD+ kinase atp_c\n",
      "ID_651 L-aspartate:fumarate oxidoreductase aspL_c\n",
      "ID_286 ferredoxin-nitrate reductase h2o_c\n",
      "ID_386 nitrate reductase nad_c\n",
      "ID_521 ferredoxin-nitrite reductase h2o_c\n",
      "ID_662 nitrite reductase [NADPH] h_c\n",
      "ID_22 ammonia:NAD+ oxidoreductase h2o_c\n",
      "ID_43 nitrite reductase [NADH] nadh_c\n",
      "ID_697 5,10-Methenyltetrahydrofolate:5'-phosphoribosylglycinamide formyltransferase h2o_c\n",
      "ID_482 5-formiminotetrahydrofolate ammonia-lyase h_c\n",
      "ID_591 tetrahydrofolate:NADP+ oxidoreductase h_c\n",
      "ID_723 methylenetetrahydrofolate reductase (ferredoxin) h_c\n",
      "ID_326 CO-methylating acetyl-CoA synthase coa_c\n",
      "ID_14 Spontaneous tetrahydrofolate transformation fald_c\n",
      "ID_758 5,10-Methenyltetrahydrofolate aminomethyltransferase h2o_c\n",
      "ID_342 5-methyltetrahydrofolate:NAD+ oxidoreductase nadh_c\n",
      "ID_145 5,10-methenyltetrahydrofolate 5-hydrolase (decyclizing) h_c\n",
      "ID_123 methylenetetrahydrofolate dehydrogenase (NAD+) nad_c\n",
      "ID_483 5-methyltetrahydrofolate:corrinoid Co-methyltransferase h_c\n",
      "ID_669 5-formyltetrahydrofolate cyclo-ligase atp_c\n",
      "ID_499 methylenetetrahydrofolate dehydrogenase (NADP+) 510mlethf_c\n",
      "ID_597 aminomethyltransferase (H-Gcv-protein) thf_c\n",
      "ID_623 formate:tetrahydrofolate ligase (ADP-forming) atp_c\n",
      "ID_427 NADH:ubiquinol-8 oxidoreductase nad_c\n",
      "ID_48 succinyl-CoA reductase h_c\n",
      "ID_378 carbon dioxide:ferredoxin oxidoreductase co2_c\n",
      "ID_566 ferredoxin-NAD+ reductase nad_c\n",
      "ID_466 carbon monoxide:acceptor oxidoreductase h2o_c\n",
      "ID_244 ferredoxin-NADP+ reductase h_c\n",
      "ID_648 hydrogen:ferredoxin oxidoreductase h_c\n",
      "ID_316 carbon dioxide reductase co2_c\n",
      "Trans_fru D-fructose transport phosphotransferase pepyr_c\n",
      "Trans_udpagaepyr UDP-N-acetylglucosamine-enolpyruvate Transport pepyr_c\n",
      "Trans_gtol Galactitol pep transporter pepyr_c\n",
      "Trans_malt maltose transport phosphotranferase pepyr_c\n",
      "Trans_tre trehalose transport phosphotranferase pepyr_c\n",
      "Trans_adman PEP dependant alpha-D-mannose Transport pepyr_c\n",
      "Trans_glcAD2 alpha-D-glucose transport phosphotranferase  pepyr_c\n",
      "Trans_2o6pamg2d 2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate Transport pepyr_c\n",
      "Trans_celb cellobiose transport phosphotranferase pepyr_c\n",
      "Trans_sucr sucrose transport phosphotranferase pepyr_c\n",
      "Trans_aga D-glucosamine transporter pepyr_c\n",
      "Trans_mnl mannitol transport phosphotranferase pepyr_c\n",
      "Trans_dachi diacetylchitobiose transport phosphotranferase pepyr_c\n",
      "Trans_fru2 D-fructose transport phosphotransferase pepyr_c\n",
      "Trans_tgt Tagatose transport pepyr_c\n",
      "Trans_sor2 D-sorbitol transport phosphotranferase pepyr_c\n",
      "Trans_glc D-glucose transport phosphotranferase pepyr_c\n",
      "Trans_fuc L-fuculose PTS transporter pepyr_c\n",
      "ID_169 acetate CoA-transferase ac_c\n",
      "ID_420 3-methyl-2-oxobutanoate hydroxymethyltransferase h2o_c\n",
      "ID_296 2-dehydropantoate formaldehyde-lyase 2dhp_c\n",
      "ID_139 phosphopantothenoylcysteine decarboxylase 4ppcys_c\n",
      "ID_405 glutaryl-CoA dehydrogenase gcoa_c\n",
      "ID_756 pantoate--beta-alanine ligase  atp_c\n",
      "ID_464 D-4'-phosphopantothenate:L-cysteine ligase atp_c\n",
      "ID_570 pantothenate kinase atp_c\n",
      "ID_197 [acyl-carrier-protein] phosphodiesterase h2o_c\n",
      "ID_50 N-((R)-pantothenoyl)-L-cysteine carboxy-lyase ptcys_c\n",
      "ID_34 holo-[acyl-carrier-protein] synthase coa_c\n",
      "ID_134 dephospho-CoA kinase atp_c\n",
      "ID_515 phosphopantothenate-cysteine ligase ctp_c\n",
      "ID_125 ATP:pantetheine 4'-phosphotransferase atp_c\n",
      "ID_449 2-acetolactate pyruvate-lyase (carboxylating) 2alac_c\n",
      "ID_85 ATP:N-((R)-pantothenoyl)-L-cysteine 4-phosphotransferase atp_c\n",
      "ID_663 malonyl CoA-acyl carrier protein transacylase h_c\n",
      "ID_755 2-dehydropantoate reductase h_c\n",
      "ID_644 pantetheine-phosphate adenylyltransferase atp_c\n",
      "ID_398 UTP-glucose-1-phosphate uridylyltransferase utp_c\n",
      "ID_592 ribitol-5-phosphate 2-dehydrogenase nadp_c\n",
      "ID_107 D-ribitol-5-phosphate:NAD+ 2-oxidoreductase nad_c\n",
      "ID_108 L-arabinose isomerase arab_c\n",
      "ID_569 D-altronate:NAD+ 3-oxidoreductase nad_c\n",
      "ID_208 D-altronate hydro-lyase (2-dehydro-3-deoxy-D-gluconate-forming) altd_c\n",
      "ID_624 xylulokinase atp_c\n",
      "ID_332 ribulokinase atp_c\n",
      "ID_10 L-ribulose-5-phosphate 4-epimerase rbl5p_c\n",
      "ID_384 D-xylose aldose-ketose-isomerase xyld_c\n",
      "ID_368 D-erythrose-4-phosphate:D-xylulose-5-phosphate glycolaldehyde transferase xyl5p_c\n",
      "ID_629 ribulose-phosphate 3-epimerase rl5p_c\n",
      "ID_419 gluconokinase atp_c\n",
      "ID_402 D-xylulose 5-phosphate D-glyceraldehyde-3-phosphate-lyase pi_c\n",
      "ID_513 ribose-phosphate diphosphokinase atp_c\n",
      "ID_424 ribose-5-phosphate isomerase rl5p_c\n",
      "ID_160 D-ribose 1,5-phosphomutase rib1p_c\n",
      "ID_656 2-dehydro-3-deoxy-phosphogluconate aldolase 2dh3dg6p_c\n",
      "ID_658 6-phospho-D-gluconate:NADP+ 2-oxidoreductase nadp_c\n",
      "ID_68 2-dehydro-3-deoxygluconokinase atp_c\n",
      "ID_73 sedoheptulose 1,7-bisphosphate 1-phosphohydrolase h2o_c\n",
      "ID_307 ATP:(D)-glycerate 2-phosphotransferase atp_c\n",
      "ID_620 6-phospho-D-gluconate hydro-lyase 6pg_c\n",
      "ID_404 deoxyribokinase atp_c\n",
      "ID_312 6-phosphogluconolactonase h2o_c\n",
      "ID_229 ribokinase atp_c\n",
      "ID_577 transketolase xyl5p_c\n",
      "ID_490 phosphopentomutase 2drib1p_c\n",
      "ID_634 phosphate acetyltransferase acoa_c\n",
      "ID_650 D-glucose:ubiquinone oxidoreductase glc_c\n",
      "ID_397 deoxyribose-phosphate aldolase 2drib5p_c\n",
      "ID_116 shikimate kinase atp_c\n",
      "ID_558 4-Hydroxyphenylacetaldehyde:NAD+ oxidoreductase h2o_c\n",
      "ID_300 prephenate dehydrogenase nad_c\n",
      "ID_345 prephenate dehydratase pphn_c\n",
      "ID_585 L-arogenate:2-oxoglutarate aminotransferase gluL_c\n",
      "ID_435 shikimate dehydrogenase (NADP+) h_c\n",
      "ID_335 chorismate synthase 5epyrsk3p_c\n",
      "ID_318 3-dehydroquinate synthase 3dahs7p_c\n",
      "ID_568 shikimate dehydrogenase (NAD+) nadh_c\n",
      "ID_7 L-arogenate hydro-lyase aro_c\n",
      "ID_578 L-arogenate:NAD+ oxidoreductase aro_c\n",
      "ID_96 3-phosphoshikimate 1-carboxyvinyltransferase pepyr_c\n",
      "ID_161 chorismate mutase chor_c\n",
      "ID_79 3-dehydroquinate dehydratase 3dhq_c\n",
      "ID_416 4-hydroxyphenylpyruvate carboxy-lyase 34hppyr_c\n",
      "ID_476 phenylalanine:2-oxoglutarate aminotransferase pheL_c\n",
      "ID_166 3-deoxy-7-phosphoheptulonate synthase pepyr_c\n",
      "ID_19 2-oxoadipate dehydrogenase complex coa_c\n",
      "ID_576 4-hydroxyphenylacetate carboxy-lyase 4hoplac_c\n",
      "ID_516 3-phosphonopyruvate carboxy-lyase 3ppyr_c\n",
      "ID_699 2-aminoethylphosphonate--pyruvate transaminase pyr_c\n",
      "ID_598 2-Phosphonoacetaldehyde phosphonohydrolase h2o_c\n",
      "ID_57 phosphoenolpyruvate 2,3-phosphonomutase pepyr_c\n",
      "ID_492 acetaldehyde:NAD+ oxidoreductase (CoA-acylating) coa_c\n",
      "ID_303 Phosphonoacetate phosphonohydrolase h2o_c\n",
      "ID_248 phosphonate:NAD+ oxidoreductase h2o_c\n",
      "ID_376 diphosphate phosphohydrolase h2o_c\n",
      "ID_241 phosphonoacetate hydrolase h2o_c\n",
      "ID_693 triphosphate phosphohydrlase h2o_c\n",
      "ID_586 4-ureido-5-imidazole carboxylate amidohydrolase h2o_c\n",
      "ID_263 ribonucleotide-reductase (ATP) atp_c\n",
      "ID_192 xanthosine 5'-phosphate phosphohydrolase h2o_c\n",
      "ID_706 xanthosine ribohydrolase h2o_c\n",
      "ID_77 deoxyadenosine kinase atp_c\n",
      "ID_603 glycine reductase pi_c\n",
      "ID_226 deoxyinosine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_126 GTP:pyruvate 2-O-phosphotransferase pepyr_c\n",
      "ID_26 dGTP:pyruvate 2-O-phosphotransferase pyr_c\n",
      "ID_80 adenosine 5'-monophosphate phosphohydrolase h2o_c\n",
      "ID_350 hypoxanthine:NAD+ oxidoreductase h2o_c\n",
      "ID_672 ATP:dGDP phosphotransferase atp_c\n",
      "ID_421 ADP-ribose ribophosphohydrolase amp_c\n",
      "ID_489 4-aminoimidazole aminohydrolase h2o_c\n",
      "ID_683 ATP:dGMP phosphotransferase atp_c\n",
      "ID_451 4-ureido-5-imidazole carboxylate decarboxylase 4a5imzc_c\n",
      "ID_275 XMP:diphosphate phosphoribosyltransferase 5pr1dp_c\n",
      "ID_111 adenylate kinase atp_c\n",
      "ID_762 ATP:deoxyguanosine 5'-phosphotransferase atp_c\n",
      "ID_305 amidophosphoribosyltransferase h2o_c\n",
      "ID_83 guanosine ribohydrolase h2o_c\n",
      "ID_237 GDP diphosphokinase atp_c\n",
      "ID_589 5-formiminotetrahydrofolate:glycine N-formiminotransferase thf_c\n",
      "ID_733 guanosine-3',5'-bis(diphosphate) 3'-diphosphohydrolase h2o_c\n",
      "ID_392 GMP:diphosphate phosphoribosyltransferase 5pr1dp_c\n",
      "ID_642 phosphoribosylformylglycinamidine cyclo-ligase atp_c\n",
      "ID_159 Inosine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_632 dATP:pyruvate 2-O-phosphotransferase pyr_c\n",
      "ID_604 guanosine 5'-monophosphate phosphahydrolase h2o_c\n",
      "ID_372 xanthine amidohydrolase h2o_c\n",
      "ID_406 inosine ribohydrolase h2o_c\n",
      "ID_100 deoxyadenosine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_537 deoxyadenylate kinase atp_c\n",
      "ID_343 guanosine 3'-diphosphate 5'-triphosphate 5'-phosphohydrolase h2o_c\n",
      "ID_475 deoxyadenosine deaminase h2o_c\n",
      "ID_30 2'-Deoxyadenosine 5'-monophosphate phosphohydrolase h2o_c\n",
      "ID_81 ATP:GMP phosphotransferase atp_c\n",
      "ID_148 glycinamide ribonucleotide transformylase atp_c\n",
      "ID_730 1-(5'-Phosphoribosyl)-5-amino-4-(N-succinocarboxamide)-imidazole AMP-lyase 5pr4sc5ai_c\n",
      "ID_681 Ap3A synthetase atp_c\n",
      "ID_274 phosphoribosylformylglycinamidine synthase h2o_c\n",
      "ID_745 IMP:diphosphate phosphoribosyltransferase 5pr1dp_c\n",
      "ID_95 deoxyguanosine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_724 adenosine deaminase h2o_c\n",
      "ID_666 IMP cyclohydrolase prfaca_c\n",
      "ID_412 10-Formyltetrahydrofolate:5'-phosphoribosylglycinamide formyltransferase 10fthf_c\n",
      "ID_736 ATP:dADP phosphotransferase atp_c\n",
      "ID_423 deoxyribose-1-phosphate isomerase drin1p_c\n",
      "ID_168 adenylosuccinate lyase (fumarate-forming) asucc_c\n",
      "ID_157 GMP synthase (glutamine-hydrolysing) h2o_c\n",
      "ID_183 guanine deaminase h2o_c\n",
      "ID_728 ribonucleoside-diphosphate reductase (ADP) adp_c\n",
      "ID_682 phosphoribosylaminoimidazolecarboxamide formyltransferase 10fthf_c\n",
      "ID_503 phosphoribosylaminoimidazolesuccinocarboxamide synthase atp_c\n",
      "ID_201 P1,P3-bis(5'-adenosyl)-triphosphate adenylohydrolase h2o_c\n",
      "ID_595 adenylosuccinate synthase gtp_c\n",
      "ID_63 AMP:diphosphate phosphoribosyltransferase ade_c\n",
      "ID_437 4-imidazolonase h2o_c\n",
      "ID_170 ribonucleotide-triphosphate reductase (GTP) gtp_c\n",
      "ID_381 ATP:GDP phosphotransferase atp_c\n",
      "ID_619 xanthosine:phosphate ribosyltransferase pi_c\n",
      "ID_282 adenosine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_387 phosphoribosylamine-glycine ligase atp_c\n",
      "ID_739 2'-Deoxyguanosine 5'-monophosphate phosphohydrolase h2o_c\n",
      "ID_684 guanosine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_118 phosphoribosylaminoimidazole carboxylase co2_c\n",
      "ID_679 ATP phosphohydrolase (NGAM) h2o_c\n",
      "ID_195 xanthosine 5'-phosphate:ammonia ligase atp_c\n",
      "ID_703 ribonucleoside-diphosphate reductase (GDP) gdp_c\n",
      "ID_512 carbamate kinase adp_c\n",
      "ID_425 IMP dehydrogenase h2o_c\n",
      "ID_612 thioredoxin oxydation thioredoxred_c\n",
      "ID_673 ATP:GTP 3'-diphosphotransferase atp_c\n",
      "ID_202 guanosine 3'-diphosphate 5'-triphosphate 3'-diphosphatase h2o_c\n",
      "ID_655 adenosine ribohydrolase h2o_c\n",
      "ID_271 (S)-dihydroorotate:NAD+ oxidoreductase nad_c\n",
      "ID_707 (S)-dihydroorotate:fumarate oxidoreductase fum_c\n",
      "ID_696 dCTP:cytidine 5'-phosphotransferase cytd_c\n",
      "ID_62 ATP:dUMP phosphotransferase atp_c\n",
      "ID_460 ribonucleotide-triphosphate reductase (UTP) utp_c\n",
      "ID_486 3-ureido-isobutyrate amidohydrolase h2o_c\n",
      "ID_277 ribonucleotide-diphosphate reductase (UDP) udp_c\n",
      "ID_1 phosphoenolpyruvate kinase (IDP) pepyr_c\n",
      "ID_87 ATP:CMP phosphotransferase atp_c\n",
      "ID_715 UTP:uridine 5'-phosphotransferase utp_c\n",
      "ID_327 dATP:cytidine 5'-phosphotransferase cytd_c\n",
      "ID_593 orotate phosphoribosyltransferase 5pr1dp_c\n",
      "ID_582 phosphoenolpyruvate kinase (CDP) pepyr_c\n",
      "ID_97 ribonucleotide-diphosphate reductase (CDP) cdp_c\n",
      "ID_8 UTP diphosphohydrolase h2o_c\n",
      "ID_641 dCTP:uridine 5'-phosphotransferase uri_c\n",
      "ID_600 dGTP:cytidine 5'-phosphotransferase dgtp_c\n",
      "ID_206 ATP:cytidine 5'-phosphotransferase atp_c\n",
      "ID_102 5,6-Dihydrouracil amidohydrolase h2o_c\n",
      "ID_69 dCTP deaminase h2o_c\n",
      "ID_13 nucleoside-phosphate kinase atp_c\n",
      "ID_216 aspartate carbamoyltransferase aspL_c\n",
      "ID_363 thymidine phosphorylase pi_c\n",
      "ID_628 CTP phosphohydrolase h2o_c\n",
      "ID_614 uridine:phosphate alpha-D-ribosyltransferase pi_c\n",
      "ID_511 5,6-dihydrouracil:NADP+ oxidoreductase h_c\n",
      "ID_579 cytosine deaminase h2o_c\n",
      "ID_144 CTP synthase h2o_c\n",
      "ID_677 (S)-dihydroorotate:quinone oxidoreductase uqnn8_c\n",
      "ID_505 ATP:uridine 5'-phosphotransferase atp_c\n",
      "ID_289 carbamoyl-phosphate synthase (glutamine-hydrolysing) (part 1) h2o_c\n",
      "ID_74 thymidine kinase atp_c\n",
      "ID_54 hydantoinase h2o_c\n",
      "ID_661 nucleoside-diphosphate kinase (dTDP) atp_c\n",
      "ID_299 GTP:cytidine 5'-phosphotransferase cytd_c\n",
      "ID_448 nucleoside-diphosphate kinase (dCDP) atp_c\n",
      "ID_38 5,10-methylenetetrahydrofolate,NADPH:dUMP C-methyltransferase h_c\n",
      "ID_377 GTP:uridine 5'-phosphotransferase uri_c\n",
      "ID_608 pyrimidine-nucleoside phosphorylase pi_c\n",
      "ID_690 nucleoside-diphosphate kinase (CDP) atp_c\n",
      "ID_213 deoxyuridine:phosphate ribosyltransferase pi_c\n",
      "ID_371 dihydroorotase casp_c\n",
      "ID_695 CTP diphosphohydrolase (diphosphate-forming) h2o_c\n",
      "ID_547 dTTP:uridine 5'-phosphotransferase uri_c\n",
      "ID_182 deoxycytidine aminohydrolase h2o_c\n",
      "ID_653 ITP:cytidine 5'-phosphotransferase itp_c\n",
      "ID_122 UDP phosphohydrolase h2o_c\n",
      "ID_334 nucleoside-diphosphate kinase (UDP) atp_c\n",
      "ID_65 2'-Deoxycytidine 5'-monophosphate phosphohydrolase h2o_c\n",
      "ID_590 nucleoside-triphosphate diphosphatase h2o_c\n",
      "ID_522 N-carbamoyl-beta-alanine amidohydrolase h2o_c\n",
      "ID_133 carbamoyl-phosphate synthase (glutamine-hydrolysing) (part 2) atp_c\n",
      "ID_5 dTMP kinase dtmp_c\n",
      "ID_517 CDP phosphohydrolase h2o_c\n",
      "ID_667 ITP:uridine 5'-phosphotransferase itp_c\n",
      "ID_441 UTP:cytidine 5'-phosphotransferase Formate acetyltransferase-glycine dihydroflavodoxin:S-adenosyl-L-methionine oxidoreductase (S-adenosyl-L-methionine cleaving) utp_c\n",
      "ID_407 phosphoenolpyruvate kinase (UDP) pepyr_c\n",
      "ID_239 dUTP:uridine 5'-phosphotransferase uri_c\n",
      "ID_137 ribonucleotide-triphosphate reductase (CTP) ctp_c\n",
      "ID_117 uracil phosphoribosyltransferase ura_c\n",
      "ID_29 thymidylate synthase 510mlethf_c\n",
      "ID_447 orotidine-5'-phosphate decarboxylase orot5p_c\n",
      "ID_734 5,6-dihydrothymine:NADP+ oxidoreductase h_c\n",
      "ID_348 dUTP:cytidine 5'-phosphotransferase cytd_c\n",
      "ID_413 dATP:uridine 5'-phosphotransferase uri_c\n",
      "ID_269 ATP:dCMP phosphotransferase atp_c\n",
      "ID_16 UTP phosphohydrolase h2o_c\n",
      "ID_261 dCMP deaminase h2o_c\n",
      "ID_534 thioredoxin-disulfide reductase h_c\n",
      "ID_718 pseudouridine-5'-phosphate glycosidase h2o_c\n",
      "ID_40 cytidine aminohydrolase nh3_c\n",
      "ID_528 uridine ribohydrolase h2o_c\n",
      "ID_601 ATP:dUTP phosphotransferase atp_c\n",
      "ID_750 Thymidylate 5'-phosphohydrolase h2o_c\n",
      "ID_442 dGTP:uridine 5'-phosphotransferase dgtp_c\n",
      "ID_432 dTTP:cytidine 5'-phosphotransferase cytd_c\n",
      "ID_153 acetyl-CoA carboxylase acoa_c\n",
      "ID_474 V-type sodium ATP synthase pi_c\n",
      "ID_565 oxaloacetate carboxy-lyase oxac_c\n",
      "ID_32 pyruvate:NAD+ 2-oxidoreductase pyr_c\n",
      "ID_35 (S)-lactaldehyde dehydrogenase  h2o_c\n",
      "ID_49 ATP synthase pi_c\n",
      "ID_44 lactate:ubiquinone-8 2-oxidoreductase lacR_c\n",
      "ID_322 lactate racemase lacS_c\n",
      "ID_280 acetate kinase adp_c\n",
      "ID_188 ATP:pyruvate,phosphate phosphotransferase pyr_c\n",
      "ID_163 oxaloacetate:phosphate carboxy-lyase pi_c\n",
      "ID_563 pyruvate, water dikinase pyr_c\n",
      "ID_186 (R)-3-amino-2-methylpropanoate:pyruvate aminotransferase pyr_c\n",
      "ID_458 D-lactate dehydrogenase pyr_c\n",
      "ID_394 methylglyoxalase gth_c\n",
      "ID_177 pyruvaldehyde:NAD+ oxidoreductase h2o_c\n",
      "ID_89 acetyl:succinate CoA-transferase acoa_c\n",
      "ID_105 S-lactoyl-glutathione hydrolase h2o_c\n",
      "ID_306 acetoacetate decarboxylase aac_c\n",
      "ID_246 fumarate reductase mql_c\n",
      "ID_291 malate dehydrogenase (oxaloacetate-decarboxylating) (NAD+) nad_c\n",
      "ID_716 methylglyoxal synthase dhap_c\n",
      "ID_36 pyruvate formate-lyase pyr_c\n",
      "ID_710 (R)-2-hydroxy-acid:acceptor 2-oxidoreductase elecacceptoxi_c\n",
      "ID_234 3-mercaptopyruvate:cyanide sulfurtransferase hcy_c\n",
      "ID_643 lactate:menaquinone 2-oxidoreductase lacR_c\n",
      "ID_702 FMN adenylyltransferase atp_c\n",
      "ID_459 FMNH2:NAD+ oxidoreductase nad_c\n",
      "ID_692 FMNH2:NADP+ oxidoreductase nadp_c\n",
      "ID_162 riboflavin kinase atp_c\n",
      "ID_257 D-ribulose 5-phosphate formate-lyase rl5p_c\n",
      "ID_375 5-amino-6-(5-phosphoribitylamino)uracil phosphatase h2o_c\n",
      "ID_356 GTP cyclohydrolase II h2o_c\n",
      "ID_599 5-amino-6-(D-ribitylamino)uracil butanedionetransferase 5a6raura_c\n",
      "ID_250 5-amino-6-(5-phosphoribosylamino)uracil:NADP+ 1'-oxidoreductase h2o_c\n",
      "ID_279 5-amino-6-(5-phosphoribosylamino)uracil reductase h_c\n",
      "ID_110 riboflavin synthase 67dm81rlum_c\n",
      "Sec_na Secretion Na+ na_e\n",
      "Sec_glyo Secretion glycerol glyo_e\n",
      "Sec_tmam Secretion trimethylamine tmam_e\n",
      "Sec_ribflv Secretion riboflavin ribflv_e\n",
      "Sec_dcyt Secretion deoxycytidine dcyt_e\n",
      "Sec_aga Secretion N-acetyl-beta-D-glucosamine aga_e\n",
      "Sec_h Secretion H+ h_e\n",
      "Sec_2mbut Secretion 2-Methylbutyric acid 2mbut_e\n",
      "Sec_co2 Secretion CO2 co2_e\n",
      "Sec_cdpg Secretion CDP-glycerol cdpg_e\n",
      "Sec_udpagaepyr Secretion UDP-N-acetylglucosamine-enolpyruvate udpagaepyr_e\n",
      "Sec_cobalt Secretion Co2+ cobalt_e\n",
      "Sec_gtol Secretion galactitol gtol_e\n",
      "Sec_2dh3dg Secretion 2-dehydro-3-deoxy-D-gluconate 2dh3dg_e\n",
      "Sec_no2 Secretion nitrite no2_e\n",
      "Sec_gly Secretion glycine gly_e\n",
      "Sec_sucr Secretion sucrose sucr_e\n",
      "Sec_eto Secretion ethanol eto_e\n",
      "Sec_g3p Secretion sn-glycerol-3-phosphate g3p_e\n",
      "Sec_lysL Secretion L-lysine lysL_e\n",
      "Sec_proL Secretion L-proline proL_e\n",
      "Sec_csn Secretion cytosine csn_e\n",
      "Sec_ribD Secretion D-ribose ribD_e\n",
      "Sec_valL Secretion L-valine valL_e\n",
      "Sec_dachi Secretion diacetylchitobiose dachi_e\n",
      "Sec_4abz Secretion 4-aminobenzoate 4abz_e\n",
      "Sec_ac Secretion acetate ac_e\n",
      "Sec_ival Secretion Isovalerate ival_e\n",
      "Sec_ca2 Secretion Ca2+ ca2_e\n",
      "Sec_hxan Secretion hypoxanthine hxan_e\n",
      "Sec_metL Secretion L-methionine metL_e\n",
      "Sec_trpL Secretion L-tryptophan trpL_e\n",
      "Sec_actid Secretion Acetamide actid_e\n",
      "Sec_arab Secretion L-arabinose arab_e\n",
      "Sec_glcn Secretion D-gluconate glcn_e\n",
      "Sec_alaL Secretion L-alanine alaL_e\n",
      "Sec_leuL Secretion L-leucine leuL_e\n",
      "Sec_4s45dhp23do Secretion (4S)-4,5-dihydroxypentan-2,3-dione 4s45dhp23do_e\n",
      "Sec_uri Secretion uridine uri_e\n",
      "Sec_2o6pamg2d Secretion 2-O-(6-phospho-alpha-D-mannosyl)-D-glycerate 2o6pamg2d_e\n",
      "Sec_adn Secretion adenosine adn_e\n",
      "Sec_ileL Secretion L-isoleucine ileL_e\n",
      "Sec_dethio Secretion dethiobiotin dethio_e\n",
      "Sec_nac Secretion nicotinate nac_e\n",
      "Sec_glnL Secretion L-glutamine glnL_e\n",
      "Sec_escul Secretion esculin escul_e\n",
      "Sec_metD Secretion D-methionine metD_e\n",
      "Sec_spmd Secretion spermidine spmd_e\n",
      "Sec_xan Secretion xanthine xan_e\n",
      "Sec_ptrc Secretion putrescine ptrc_e\n",
      "Sec_thymd Secretion thymidine thymd_e\n",
      "Sec_nh3 Secretion ammonia nh3_e\n",
      "Sec_fru Secretion D-fructose fru_e\n",
      "Sec_thm Secretion thiamin thm_e\n",
      "Sec_c1ala Secretion cob(I)alamin c1ala_e\n",
      "Sec_for Secretion formate for_e\n",
      "Sec_fe2 Secretion Fe2+ fe2_e\n",
      "Sec_so3 Secretion sulfite so3_e\n",
      "Sec_4abg Secretion 4-aminobenzoate-glutamate 4abg_e\n",
      "Sec_fuc Secretion L-fuculose fuc_e\n",
      "Sec_chol Secretion choline chol_e\n",
      "Sec_malS Secretion (S)-malate malS_e\n",
      "Sec_chor Secretion chorismate chor_e\n",
      "Sec_hco3 Secretion bicarbonate hco3_e\n",
      "Sec_malt Secretion maltose malt_e\n",
      "Sec_ppat Secretion phosphonate ppat_e\n",
      "Sec_h2 Secretion H2 h2_e\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sec_bio Secretion biotin bio_e\n",
      "Sec_xyld Secretion D-xylose xyld_e\n",
      "Sec_pntoR Secretion (R)-pantothenate pntoR_e\n",
      "Ex_23dhbz Secretion of 2,3-dihydroxybenzoate 23dhbz_e\n",
      "Sec_thrL Secretion L-threonine thrL_e\n",
      "Sec_etoa Secretion ethanolamine etoa_e\n",
      "Sec_dad Secretion deoxyadenosine dad_e\n",
      "Sec_isobuta Secretion isobutyric acid isobuta_e\n",
      "Sec_glyb Secretion glycine betaine glyb_e\n",
      "Sec_5apn Secretion 5-aminopentanoate 5apn_e\n",
      "Sec_adman Secretion alpha-D-mannose adman_e\n",
      "Sec_2aepat Secretion (2-aminoethyl)phosphonate 2aepat_e\n",
      "Sec_gluL Secretion L-glutamate gluL_e\n",
      "Sec_aspL Secretion L-aspartate aspL_e\n",
      "Sec_cysL Secretion L-cysteine cysL_e\n",
      "Sec_escut Secretion esculetin escut_e\n",
      "Sec_acnam Secretion N-acetylneuraminate acnam_e\n",
      "Sec_ura Secretion uracil ura_e\n",
      "Sec_tyrL Secretion L-tyrosine tyrL_e\n",
      "Sec_hoam Secretion Hydroxylamine hoam_e\n",
      "Sec_mevR Secretion (R)-mevalonate mevR_e\n",
      "Sec_pi Secretion phosphate pi_e\n",
      "Sec_raf Secretion raffinose raf_e\n",
      "Sec_h2o Secretion H2O h2o_e\n",
      "Sec_his Secretion L-histidine his_e\n",
      "Sec_zn2 Secretion Zn2+ zn2_e\n",
      "Sec_mg2 Secretion Mg2+ mg2_e\n",
      "Sec_4mpo Secretion 4-methylphenol 4mpo_e\n",
      "Sec_k Secretion K+ k_e\n",
      "Sec_argL Secretion L-arginine argL_e\n",
      "Sec_mnl Secretion mannitol mnl_e\n",
      "Sec_gal Secretion beta-D-galactose gal_e\n",
      "Sec_no3 Secretion nitrate no3_e\n",
      "Sec_tgt Secretion tagatose tgt_e\n",
      "Sec_tre Secretion trehalose tre_e\n",
      "Sec_ins Secretion inosine ins_e\n",
      "Sec_at Secretion acetone at_e\n",
      "Sec_pheL Secretion L-phenylalanine pheL_e\n",
      "Sec_23diap Secretion 2,3-diaminopropanoate 23diap_e\n",
      "Sec_ncam Secretion nicotinamide ncam_e\n",
      "Sec_glc Secretion beta-D-glucose glc_e\n",
      "Sec_pyin Secretion pyridoxine pyin_e\n",
      "Sec_celb Secretion cellobiose celb_e\n",
      "Sec_ibtol Secretion isobutanol ibtol_e\n",
      "Sec_asnL Secretion L-asparagine asnL_e\n",
      "Sec_isocap Secretion Isocaproate isocap_e\n",
      "Sec_sor Secretion D-sorbitol sor_e\n",
      "Sec_serL Secretion L-serine serL_e\n",
      "Sec_glcAD Secretion alpha-D-glucose glcAD_e\n",
      "ID_744 ADP-glucose Glucose-1-phosphohydrolase h2o_c\n",
      "ID_422 beta-fructofuranosidase h2o_c\n",
      "ID_488 cellobiose glucohydrolase h2o_c\n",
      "ID_258 ATP:alpha-D-glucose-1-phosphate adenylyltransferase atp_c\n",
      "ID_93 alpha-glucosidase h2o_c\n",
      "ID_646 diacetylchitobiose-6-phosphate hydrolase h2o_c\n",
      "Trans_raf Raffinose transport h2o_c\n",
      "ID_450 6-phospho-beta-glucosidase h2o_c\n",
      "ID_509 Esculin hydrolase h2o_c\n",
      "ID_308 maltose-6'-phosphate glucosidase h2o_c\n",
      "ID_45 glucose-6-phosphate:NADP+ 1-oxidoreductase nadp_c\n",
      "ID_408 Raffinose galactohydrolase h2o_c\n",
      "ID_373 beta-phosphoglucomutase glc1p_c\n",
      "ID_573 alpha,alpha-phosphotrehalase h2o_c\n",
      "ID_315 maltose phosphorylase pi_c\n",
      "ID_314 D-proline reductase nadh_c\n",
      "ID_321 Isobutyric acid phosphotransferase adp_c\n",
      "ID_146 2-methylbutanoyl phosphate transferase adp_c\n",
      "ID_114 betaine reductase h2o_c\n",
      "ID_4 phosphate isopentyltransferase ivalcoa_c\n",
      "ID_71 D-Alanine:2-oxoglutarate aminotransferase 2oglut_c\n",
      "ID_352 phosphate isobutanoyltransferase pi_c\n",
      "ID_178 (S)-3-methyl-2-oxopentanoate:NAD+ 2-oxidoreductase (CoA-methylbutanoylating) coa_c\n",
      "ID_366  isopentanoylphosphate phosphotransferase isop_c\n",
      "ID_709 isoleucine dehydrogenase  2oglut_c\n",
      "ID_518 2-methylbutanoyl CoA transferase pi_c\n",
      "ID_323 4-methyl-2-oxopentanoate:NAD+ oxidoreductase (CoA-mehtylpropanoylating) coa_c\n",
      "ID_247 3-methyl-2-oxobutanoate dehydrogenase (ferredoxin) coa_c\n",
      "ID_719 hydrogen-sulfide:NAD+ oxidoreductase nadh_c\n",
      "ID_158 Hydrogen-sulfide:ferredoxin oxidoreductase h2o_c\n",
      "ID_510 3-mercaptopyruvate sulfurtransferase so3_c\n",
      "ID_584 sulfite reduction h_c\n",
      "ID_355 thiosulfate sulfurtransferase h_c\n",
      "ID_141 hydrogen-sulfide:NADP+ oxidoreductase h_c\n",
      "ID_266 Adenosine 3',5'-bisphosphate 3'-phosphohydrolase h2o_c\n",
      "ID_485 trans-hexaprenyltranstransferase ipdp_c\n",
      "ID_630 Lipoteichoic acid biosynthesis alaD_c\n",
      "ID_607 N-acetylglucosaminyldiphosphoundecaprenol N-acetyl-beta-D-mannosaminyltransferase udpama_c\n",
      "ID_324 UDP-N-acetyl-D-glucosamine:undecaprenyl-phosphate N-acetyl-D-glucosamine phosphotransferase udpaga_c\n",
      "ID_200 lysyltransferase ptg_c\n",
      "ID_59 CDPglycerol:poly(glycerophosphate) glycerophosphotransferase cdpg_c\n",
      "ID_617 geranyltranstransferase ipdp_c\n",
      "ID_292 4-hydroxy-3-methylbut-2-enyl diphosphate reductase h_c\n",
      "ID_640 dimethylallyl diphosphate:NADP+ oxidoreductase h_c\n",
      "ID_187 (R)-mevalonate:NAD+ oxidoreductase coa_c\n",
      "ID_749 4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol kinase atp_c\n",
      "ID_645 2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase ctp_c\n",
      "ID_705 2-C-methyl-D-erythritol-2,4-cyclodiphosphate:reduced ferredoxin oxidase h_c\n",
      "ID_637 1-deoxy-D-xylulose-5-phosphate reductoisomerase h_c\n",
      "ID_150 (2E,6E)-farnesyl-diphosphate:isopentenyl-diphosphate cistransferase ipdp_c\n",
      "ID_70 farnesyltranstransferase ipdp_c\n",
      "ID_431 dimethylallyltranstransferase ipdp_c\n",
      "ID_639 undecaprenyl-diphosphate phosphohydrolase h2o_c\n",
      "ID_403 geranylgeranyl-diphosphate:isopentenyl-diphosphate transtransferase ipdp_c\n",
      "ID_24 hydroxymethylglutaryl-CoA lyase 3h3mgcoa_c\n",
      "ID_587 isopentenyl-diphosphate delta-isomerase dmadp_c\n",
      "ID_46 isopentenyl-diphosphate:NAD+ oxidoreductase nadh_c\n",
      "ID_17 2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase 2p4c5dp2me_c\n",
      "ID_351 dimethylallyl diphosphate:NAD+ oxidoreductase nadh_c\n",
      "ID_689 trans-pentaprenyltranstransferase ipdp_c\n",
      "ID_720 L-tyrosine 4-methylphenol-lyase (2-iminoacetate-forming) elecacceptred_c\n",
      "ID_468 xylulose-5-phosphate : thiamin diphosphate glycolaldehydetransferase xyl5p_c\n",
      "ID_225 ATP:4-methyl-5-(beta-hydroxyethyl)thiazole 2-phosphotransferase atp_c\n",
      "ID_214 thiamine-phosphate diphosphorylase h_c\n",
      "ID_636 phosphomethylpyrimidine synthase (2'deoxyadenosine) amet_c\n",
      "ID_668 Phosphomethylpyrimidine synthase (5'deoxyadenosine) amet_c\n",
      "ID_238 1-methylthio-xylulose 5-phosphate sulfo-lyase  h_c\n",
      "ID_264 phosphomethylpyrimidine kinase atp_c\n",
      "ID_228 sedoheptulose-7-phosphate : thiamin diphosphate glycolaldehydetransferase shl7p_c\n",
      "ID_15 2,3-diaminopropanoate ammonia-lyase (adding water; pyruvate-forming) h2o_c\n",
      "ID_478 1-deoxy-D-xylulose-5-phosphate synthase pyr_c\n",
      "ID_581 ATP:thiamine phosphotransferase atp_c\n",
      "ID_210 4-methyl-5-(beta-hydroxyethyl)thiazole Biosynthesis pyr_c\n",
      "ID_388 thiamine diphosphokinase atp_c\n",
      "ID_389 thiamine-phosphate kinase atp_c\n",
      "Trans_mevR (R)-mevalonate transport mevR_e\n",
      "Trans_xyld D-xylose import xyld_e\n",
      "Trans_isobuta Isobutyric acid transport isobuta_c\n",
      "Trans_fe Ferrous iron transport  fe2_e\n",
      "Trans_ival Isovalerate transport ival_c\n",
      "Trans_4s45dhp23do (4S)-4,5-dihydroxypentan-2,3-dione transporter 4s45dhp23do_c\n",
      "Trans_h H+ Transport h_e\n",
      "Trans_23dhbz 2,3-dihydroxybenzoate Transporter 23dhbz_c\n",
      "Trans_hoam Hydroxylamine transporter  hoam_e\n",
      "Trans_escul Esculin transport escul_e\n",
      "Trans_23diap 2,3-diaminopropanoate transporter 23diap_e\n",
      "Trans_pyin Pyridoxine Transport pyin_e\n",
      "Trans_escut Esculetin transport escut_c\n",
      "Trans_2aepat (2-aminoethyl)phosphonate transporter 2aepat_e\n",
      "Trans_bio Biotin transport  bio_e\n",
      "Trans_pntoR (R)-pantothenate transport pntoR_e\n",
      "Trans_cdpg CDP-glycerol:poly(glycerophosphate) glycerophosphotransferase cdpg_e\n",
      "Trans_ibtol isobutanol transport ibtol_c\n",
      "Trans_cobalt2 cobalt transporter cobalt_e\n",
      "Trans_glyo glycerol transport glyo_e\n",
      "Trans_2mbut 2-Methylbutyric acid transport 2mbut_c\n",
      "Trans_actid Acetamide transport actid_c\n",
      "Trans_nac nicotinate transporter nac_e\n",
      "Trans_etoa ethanolamine transporter etoa_e\n",
      "Trans_5apn 5-aminopentanoate transport 5apn_c\n",
      "Trans_dethio dethiobiotin transport dethio_e\n",
      "Trans_thm thiamin transport thm_e\n",
      "Trans_4abz p-Aminobenzoic acid transport 4abz_e\n",
      "Trans_4mpo 4-methylphenol 4mpo_c\n",
      "Trans_pi3 phosphate passive transporter pi_e\n",
      "Trans_isocap Isocaproate transport isocap_c\n",
      "Trans_nh ammonia transport nh3_e\n",
      "Trans_ncam nicotinamide transport ncam_e\n",
      "Trans_tmam  trimethylamine transport tmam_e\n",
      "ID_284 naphthoate synthase 2sucbzcoa_c\n",
      "ID_119 NADH:menaquinone oxidoreductase nadh_c\n",
      "ID_472 2-succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate synthase 2oglut_c\n",
      "ID_722 tyrosine:2-oxoglutarate aminotransferase 2oglut_c\n",
      "ID_112 all-trans-polyprenyl diphosphate:1,4-dihydroxy-2-naphthoate polyprenyltransferase octdp_c\n",
      "ID_746 S-adenosyl-L-methione:demethylmenaquinone methyltransferase amet_c\n",
      "ID_23 all-trans-octaprenyl-diphosphate synthase hepdp_c\n",
      "ID_33 hydro-lyase 2s6h24ch1c_c\n",
      "ID_209 isochorismate synthase chor_c\n",
      "ID_555 o-succinylbenzoate-CoA ligase coa_c\n",
      "ID_51 2-isopropylmalate synthase acoa_c\n",
      "ID_212 2-methylcrotonoyl-CoA hydro-lyase h2o_c\n",
      "ID_11 3-isopropylmalate dehydrogenase(step1) nad_c\n",
      "ID_354 2-methyl-3-hydroxybutyryl-CoA:NAD+ oxidoreductase nad_c\n",
      "ID_757 (3S)-3-Methyl-2-oxopentanoate carboxy-lyase h_c\n",
      "ID_328 2-Isopropylmalate hydro-lyase(step 1) 2ipm_c\n",
      "ID_217 leucine transaminase 2oglut_c\n",
      "ID_454 2-Isopropylmalate hydro-lyase(step 2) h2o_c\n",
      "ID_735 2,3-dihydroxy-3-methylvalerate dehydratase 23dh3mv_c\n",
      "ID_369 L-valine:pyruvate aminotransferase pyr_c\n",
      "ID_691 2-methylbutanoyl-CoA:FAD 2,3-oxidoreductase fad_c\n",
      "ID_731 pyruvate:2-oxobutanoate acetaldehydetransferase pyr_c\n",
      "ID_453 2-aceto-2-hydroxybutanoate oxidoreductase h_c\n",
      "ID_154 propanoyl-CoA : acetyl-CoA acetyltransferase coa_c\n",
      "ID_359 3-isopropylmalate dehydrogenase(step2) 2ip3os_c\n",
      "ID_520 isobutanal hydrogenase nadh_c\n",
      "ID_430 2-methyl-3-oxopropanoate:NAD+ oxidoreductase coa_c\n",
      "ID_539 propanoyl-CoA:formate C-propanoyltransferase coa_c\n",
      "ID_203 S-adenosyl-L-methionine:uroporphyrin-III C-methyltransferase nad_c\n",
      "ID_497 ATP:cob(I0alamin Co-beta-adenosyltransferase atp_c\n",
      "ID_256 S-adenosyl-L-methionine:precorrin-1 C-methyltransferase amet_c\n",
      "ID_207 adenosylcobinamide-phosphate guanylyltransferase gtp_c\n",
      "ID_236 nicotinate mononucleotide:5,6-dimethylbenzimidazole phospho-D-ribosyltransferase nacmnc_c\n",
      "ID_465 coproporphyrinogen dehydrogenase amet_c\n",
      "ID_131 biotin synthase amet_c\n",
      "ID_456 hydroxymethylbilane hydro-lyase hmb_c\n",
      "ID_529 threonine-phosphate carboxy-lyase thr3p_c\n",
      "ID_538 adenosylcobinamide kinase atp_c\n",
      "ID_717 S-adenosyl-L-methionine:uroporphyrin-III C-methyltransferase amet_c\n",
      "ID_47 adenosylcobinamide kinase adcbam_c\n",
      "ID_37 S-adenosyl-L-methionine:cobalt-factor-II C20-methyltransferase cbshclin_c\n",
      "ID_260 Uroporphyrinogen-III carboxy-lyase uppg3_c\n",
      "ID_553 ATP:L-threonine O-phosphotransferase atp_c\n",
      "ID_103 ATP:Biotin adenyltransferase atp_c\n",
      "ID_671 S-adenosyl-L-methionine:cobalt-precorrin-5B (C1)-methyltransferase h_c\n",
      "ID_91 2,3-dihydro-2,3-dihydroxybenzoate:NAD+ oxidoreductase nad_c\n",
      "ID_540 5-aminolevulinate hydro-lyase 5all_c\n",
      "ID_281 cobalt-precorrin 3 dehydrogenase elecacceptred_c\n",
      "ID_25 cobalamin synthase acgdp_c\n",
      "ID_704 Isochorismate pyruvate-hydrolase h2o_c\n",
      "ID_426 cobalt-precorrin-7 (C15)-methyltransferase [decarboxylating] amet_c\n",
      "ID_245 ATP:cob(I)yrinate a,c-diamide cobeta-adenosyltransferase atp_c\n",
      "ID_440 cob(II)yrinate a,c-diamide:FMN oxidoreductase fmnh2_c\n",
      "ID_84 protoporphyrinogen IX dehydrogenase (menaquinone) ptpphygen_c\n",
      "ID_552 adenosyl-cobyrate:(R)-1-amino-2-propanol O-2-phosphate ligase atp_c\n",
      "ID_572 S-adenosyl-L-methionine:precorrin-7 C15-methyltransferase amet_c\n",
      "ID_138 porphobilinogen ammonia-lyase h2o_c\n",
      "ID_618 adenosyl-cobyrate synthase h2o_c\n",
      "ID_364 Precorrin-8X methylmutase cbpcr8x_c\n",
      "ID_487 cobalt-precorrin-4 methyltransferase amet_c\n",
      "ID_129 ferrochelatase ptphyrin_c\n",
      "ID_738 precorrin-6A reductase h_c\n",
      "ID_700 adenosylcobalamin phosphatase h2o_c\n",
      "ID_67 S-adenosyl-L-methionine:uroporphyrinogen-III C-methyltransferase amet_c\n",
      "ID_190 S-adenosyl-L-methionine:precorrin-3B C17-methyltransferase amet_c\n",
      "ID_670 cobyrinate:L-glutamine amido-ligase (ADP-forming) h2o_c\n",
      "ID_760 adenosylcobinamide-GDP ribazoletransferase acgdp_c\n",
      "ID_484 sirohydrochlorin cobalt-lyase shclin_c\n",
      "ID_101 cobalt-precorrin 5A acylhydrolase h2o_c\n",
      "ID_761 adenosylcobalamin/alpha-ribazole phosphatase h2o_c\n",
      "ID_508 pyridoxamine:pyruvate aminotransferase pyr_c\n",
      "ID_554 pyridoxal kinase atp_c\n",
      "ID_220 Pyridoxamine-5'-phosphate:2-oxoglutarate aminotransferase pyr_c\n",
      "ID_647 Pyridoxine:NADP+ 4-oxidoreductase h_c\n",
      "ID_461 pyridoxamine:oxaloacetate aminotransferase oxac_c\n",
      "ID_698 pyridoxal kinase atp_c\n",
      "ID_270 Pyridoxamine-5'-phosphate:2-oxoglutarate aminotransferase (D-glutamate-forming) 2oglut_c\n",
      "Trans_co carbon dioxide diffusion co2_e\n",
      "Trans_h2o water diffusion h2o_e\n",
      "Trans_eto ethanol transport eto_e\n",
      "Trans_hco hco3 transport hco3_e\n",
      "Trans_at acetone export at_c\n",
      "Trans_h4 hydrogen diffusion h2_e\n",
      "Trans_adman2 alpha-D-mannose Transport h_e\n",
      "Trans_glcAD alpha-D-glucose Transport glcAD_e\n",
      "Trans_uri uridine transport h_e\n",
      "Trans_ura uracil transport h_e\n",
      "Trans_h3 sodium ion:proton antiporter h_c\n",
      "Trans_ca2 Calcium:Sodium antiport na_c\n",
      "Trans_xyld2 D-xylose Transport h_e\n",
      "Trans_ac Acetate Transport h_c\n",
      "Trans_glcn gluconate permease h_e\n",
      "Trans_pi sodium-dependent phosphate transporter pi_e\n",
      "Trans_dad deoxyadenosine transport h_e\n",
      "Trans_dcyt deoxycytidine transport h_e\n",
      "Trans_celb2 cellobiose import h_e\n",
      "Trans_thymd thymidine transport h_e\n",
      "Trans_malS (S)-malate transport h_c\n",
      "Trans_ribflv riboflavin transport h_e\n",
      "Trans_hxan hypoxanthine transport h_e\n",
      "Trans_2dh3dg 2-keto-3-deoxygluconate permease h_e\n",
      "Trans_h2 sodium:H+ symporter h_e\n",
      "Trans_sor D-sorbitol import h_e\n",
      "Trans_malt2 maltose import h_e\n",
      "Trans_k2 potassium ion uptake h_e\n",
      "Trans_csn cytosine:H+ cotransport h_e\n",
      "Trans_sucr2 sucrose import h_e\n",
      "Trans_ins inosine transport h_e\n",
      "Trans_4abg aminobenzoyl-glutamate transport h_e\n",
      "Trans_arab L-arabinose import h_e\n",
      "Trans_acnam Transport of N-acetylneuraminate h_e\n",
      "Trans_xan xanthine permease h_e\n",
      "Trans_gal beta-D-galactose import h_e\n",
      "Trans_so sodium:sulfite symporter na_e\n",
      "Trans_xan2 xanthine:uracil cotransport ura_e\n",
      "Trans_alaL sodium ion:alanine symporter na_e\n",
      "Trans_adn adenosine transport h_e\n",
      "Trans_glc2 beta-D-glucose import h_e\n",
      "Trans_gluL2 glutamate:sodium cotransport na_e\n",
      "Trans_chor chorismate transport h_e\n",
      "Trans_aspL proton/sodium ion:glutamate/aspartate symporter na_e\n",
      "Ex_biomass Exchange Biomass biomass_c\n",
      "comp1 CD0043, CD0042, CD0041 E_398\n",
      "comp2 CD0118, CD2199, CD2198 E_758\n",
      "comp3 CD0135, CD0136, CD0137 E_608\n",
      "comp4 CD0206, CD0207, CD0208 E_521\n",
      "comp5 CD0284, CD0285 E_745\n",
      "comp6 CD0286, CD0287, CD0288, CD0289 E_295\n",
      "comp7 CD0431, CD0430 E_178\n",
      "comp8 CD0491, CD0492, CD0493, CD0494 E_79\n",
      "comp9 CD0723, acoA, acoB E_278\n",
      "comp10 CD0723, acoL E_278\n",
      "comp11 CD0728, CD0725, CD0727, CD0717, CD0724 E_14\n",
      "comp12 CD0728, CD0726, CD0727, CD0717, CD0724 E_14\n",
      "comp13 CD0863, CD0862, CD0861 E_383\n",
      "comp14 CD1000, CD1001, CD0999 E_30\n",
      "comp15 CD1074, CD1076, CD1077, CD1078 E_739\n",
      "comp16 CD1083, CD2280, CD2282 E_453\n",
      "comp17 CD1386, CD1387 E_283\n",
      "comp18 CD1587, CD1588, CD1589 E_588\n",
      "comp19 CD1904, CD2989, CD2990 E_103\n",
      "comp20 CD1979, CD1978, CD1980 E_624\n",
      "comp21 CD2077, pyrK E_4\n",
      "comp22 CD2087, pucC1, xdhC1, xdhD E_92\n",
      "comp23 CD2087, pucC1, xdhC2, xdhD E_92\n",
      "comp24 CD2087, pucC1, xdhC3, xdhD E_92\n",
      "comp25 CD2087, pucC2, xdhC1, xdhD E_92\n",
      "comp26 CD2087, pucC2, xdhC2, xdhD E_92\n",
      "comp27 CD2087, pucC2, xdhC3, xdhD E_92\n",
      "comp28 CD2087, xdhB, xdhC1, xdhD E_92\n",
      "comp29 CD2087, xdhB, xdhC2, xdhD E_92\n",
      "comp30 CD2087, xdhB, xdhC3, xdhD E_92\n",
      "comp31 CD2327, CD2326, CD2325 E_32\n",
      "comp32 CD2332, CD2334 E_569\n",
      "comp33 CD2512, CD2510 E_724\n",
      "comp34 CD2566, CD2567, CD2568 E_219\n",
      "comp35 CD2875, CD2876, CD2878 E_508\n",
      "comp36 CD2975, CD2877, CD2878 E_757\n",
      "comp37 CD3015, CD3014, CD3013 E_265\n",
      "comp38 CD3027, CD3030 E_699\n",
      "comp39 CD3030, CD3027 E_269\n",
      "comp40 CD3058, CD3061 E_666\n",
      "comp41 CD3080, CD3049, CD3048 E_465\n",
      "comp42 CD3080, CD3081, CD3082 E_465\n",
      "comp43 CD3101, CD3103 E_463\n",
      "comp44 CD3179, pyrK E_500\n",
      "comp45 CD3278, CD3279, CD3277, CD3276 E_702\n",
      "comp46 CD3417, CD3416, CD3415, CD3414 E_534\n",
      "comp47 CD3445, CD3444, CD3443 E_130\n",
      "comp48 CD3629, CD3630 E_575\n",
      "comp49 accA, accD, accB, accC E_747\n",
      "comp50 asrA, asrB, asrC E_473\n",
      "comp51 atpA, atpB, atpF, atpH, atpG, atpD, atpC, atpI, atpZ E_291\n",
      "comp52 atpA, atpB, atpF, atpH, atpG, atpD, atpE, atpI, atpZ E_291\n",
      "comp53 carB1, carB2, carA E_612\n",
      "comp54 carB1, carB2, pyrAA2 E_612\n",
      "comp55 carB, carA E_564\n",
      "comp56 carB, pyrAA2 E_564\n",
      "comp57 cbiO2, cbiO1, cbiQ, cbiN, cbiM E_733\n",
      "comp58 cbiO, CD0326, cbiN, cbiM E_420\n",
      "comp59 celC, celB, celA E_209\n",
      "comp60 crt1, CD0803, hbd, thlA2 E_580\n",
      "comp61 eutA, eutB, eutC E_546\n",
      "comp62 eutB, eutC E_213\n",
      "comp63 fabH, fabF, CD2577, fabZ E_93\n",
      "comp64 fabH, fabF, fabG, fabZ E_93\n",
      "comp65 fumA, fumB E_764\n",
      "comp66 gcvTPA, gcvH E_672\n",
      "comp67 gcvTPA, gcvPB E_672\n",
      "comp68 grdA, grdE, grdD E_289\n",
      "comp69 grdB, grdC, grdX E_374\n",
      "comp70 grdC, grdB, grdX E_447\n",
      "comp71 grdD, grdA, grdX E_548\n",
      "comp72 gutA, srlEa, srlEb, srlB E_152\n",
      "comp73 hadI, hadB, hadC E_275\n",
      "comp74 hisH, hisF E_10\n",
      "comp75 hisZ, hisG E_318\n",
      "comp76 hpdC, hpdB, hpdA E_629\n",
      "comp77 hymA, hymB, CD3407 E_479\n",
      "comp78 kdpA, kdpB, kdpC E_71\n",
      "comp79 leuC, leuD E_581\n",
      "comp80 metN, metI, metQ, CD1653 E_376\n",
      "comp81 nrdE, nrdF E_401\n",
      "comp82 ntpD, ntpB, ntpA, ntpC, ntpE, ntpF, ntpK E_165\n",
      "comp83 opuCA, opuCC, CD3215 E_58\n",
      "comp84 oraS, oraE E_651\n",
      "comp85 ortA, ortB E_417\n",
      "comp86 pabA, pabB E_282\n",
      "comp87 potA, potC, potB, potD E_307\n",
      "comp88 prdA, prdB E_623\n",
      "comp89 prdF, prdE, prdD E_156\n",
      "comp90 ptsG-A, ptsG-BC E_226\n",
      "comp91 pyrD, pyrK E_95\n",
      "comp92 pyrK, CD2077 E_776\n",
      "comp93 pyrK, CD3179 E_776\n",
      "comp94 rbsA, rbsC, rbsB E_418\n",
      "comp95 ribH, ribE E_617\n",
      "comp96 rpiB1, rpiB2 E_416\n",
      "comp97 scoA, scoB E_127\n",
      "comp98 srlA, srlEa, srlEb, srlB E_659\n",
      "comp99 thiF, thiS, thiG, CD2156, thiI E_434\n",
      "comp100 thiF, thiS, thiG, thiH, thiI E_434\n",
      "comp101 xdhA1, pucC1, xdhC1, xdhD E_81\n",
      "comp102 xdhA1, pucC1, xdhC2, xdhD E_81\n",
      "comp103 xdhA1, pucC1, xdhC3, xdhD E_81\n",
      "comp104 xdhA1, pucC2, xdhC1, xdhD E_81\n",
      "comp105 xdhA1, pucC2, xdhC2, xdhD E_81\n",
      "comp106 xdhA1, pucC2, xdhC3, xdhD E_81\n",
      "comp107 xdhA1, xdhB, xdhC1, xdhD E_81\n",
      "comp108 xdhA1, xdhB, xdhC2, xdhD E_81\n",
      "comp109 xdhA1, xdhB, xdhC3, xdhD E_81\n",
      "comp110 xdhA2, pucC1, xdhC1, xdhD E_740\n",
      "comp111 xdhA2, pucC1, xdhC2, xdhD E_740\n",
      "comp112 xdhA2, pucC1, xdhC3, xdhD E_740\n",
      "comp113 xdhA2, pucC2, xdhC1, xdhD E_740\n",
      "comp114 xdhA2, pucC2, xdhC2, xdhD E_740\n",
      "comp115 xdhA2, pucC2, xdhC3, xdhD E_740\n",
      "comp116 xdhA2, xdhB, xdhC1, xdhD E_740\n",
      "comp117 xdhA2, xdhB, xdhC2, xdhD E_740\n",
      "comp118 xdhA2, xdhB, xdhC3, xdhD E_740\n",
      "comp119 xdhA3, pucC1, xdhC1, xdhD E_751\n",
      "comp120 xdhA3, pucC1, xdhC2, xdhD E_751\n",
      "comp121 xdhA3, pucC1, xdhC3, xdhD E_751\n",
      "comp122 xdhA3, pucC2, xdhC1, xdhD E_751\n",
      "comp123 xdhA3, pucC2, xdhC2, xdhD E_751\n",
      "comp124 xdhA3, pucC2, xdhC3, xdhD E_751\n",
      "comp125 xdhA3, xdhB, xdhC1, xdhD E_751\n",
      "comp126 xdhA3, xdhB, xdhC2, xdhD E_751\n",
      "comp127 xdhA3, xdhB, xdhC3, xdhD E_751\n",
      "comp128 xynA, xynB, xynC, xynD E_411\n"
     ]
    }
   ],
   "source": [
    "for x in iMLTC806cdf.reactions:\n",
    "    print(x.id, x.name, x.reactants[0].id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>DM_btn</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Demand for biotin</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f9d26d70950</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>btn[c] <=> </p>\n",
       "                    <p style='text-align:right'>Biotin <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction DM_btn at 0x7f9d26d70950>"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#agora_NAP07.metabolites.get_by_id('o2[c]')\n",
    "agora_NAP07.reactions.DM_btn\n",
    "#agora_NAP07.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2ddglcn[e] 2-Dehydro-3-deoxy-D-gluconate\n",
      "2mbut[e] 2-methylbutyrate\n",
      "34dhpha[e] (3,4-dihydroxyphenyl)acetate\n",
      "4abut[e] 4-Aminobutanoate\n",
      "4hphac[e] 4-hydroxyphenylacetate\n",
      "4hpro_LT[e] trans-4-hydroxy-L-proline\n",
      "4mcat[e] 4-Methylcatechol\n",
      "5aptn[e] 5-Aminopentanoate\n",
      "Lcyst[e] L-cysteate(1-)\n",
      "MGlcn100[e] mucin-type O-glycan No 100\n",
      "MGlcn100_rl[e] released mucin-type O-glycan No 100\n",
      "MGlcn101[e] mucin-type O-glycan No 101\n",
      "MGlcn101_rl[e] released mucin-type O-glycan No 101\n",
      "MGlcn102[e] mucin-type O-glycan No 102\n",
      "MGlcn102_rl[e] released mucin-type O-glycan No 102\n",
      "MGlcn103[e] mucin-type O-glycan No 103\n",
      "MGlcn103_rl[e] released mucin-type O-glycan No 103\n",
      "MGlcn108[e] mucin-type O-glycan No 108\n",
      "MGlcn108_rl[e] released mucin-type O-glycan No 108\n",
      "MGlcn10[e] mucin-type O-glycan No 10\n",
      "MGlcn10_rl[e] released mucin-type O-glycan No 10\n",
      "MGlcn116[e] mucin-type O-glycan No 116\n",
      "MGlcn116_rl[e] released mucin-type O-glycan No 116\n",
      "MGlcn117[e] mucin-type O-glycan No 117\n",
      "MGlcn117_rl[e] released mucin-type O-glycan No 117\n",
      "MGlcn118[e] mucin-type O-glycan No 118\n",
      "MGlcn118_rl[e] released mucin-type O-glycan No 118\n",
      "MGlcn119[e] mucin-type O-glycan No 119\n",
      "MGlcn119_rl[e] released mucin-type O-glycan No 119\n",
      "MGlcn120[e] mucin-type O-glycan No 120\n",
      "MGlcn120_rl[e] released mucin-type O-glycan No 120\n",
      "MGlcn121[e] mucin-type O-glycan No 121\n",
      "MGlcn121_rl[e] released mucin-type O-glycan No 121\n",
      "MGlcn122[e] mucin-type O-glycan No 122\n",
      "MGlcn122_rl[e] released mucin-type O-glycan No 122\n",
      "MGlcn123[e] mucin-type O-glycan No 123\n",
      "MGlcn123_rl[e] released mucin-type O-glycan No 123\n",
      "MGlcn124[e] mucin-type O-glycan No 124\n",
      "MGlcn124_rl[e] released mucin-type O-glycan No 124\n",
      "MGlcn126[e] mucin-type O-glycan No 126\n",
      "MGlcn126_rl[e] released mucin-type O-glycan No 126\n",
      "MGlcn127[e] mucin-type O-glycan No 127\n",
      "MGlcn127_rl[e] released mucin-type O-glycan No 127\n",
      "MGlcn128[e] mucin-type O-glycan No 128\n",
      "MGlcn128_rl[e] released mucin-type O-glycan No 128\n",
      "MGlcn129[e] mucin-type O-glycan No 129\n",
      "MGlcn129_rl[e] released mucin-type O-glycan No 129\n",
      "MGlcn12[e] mucin-type O-glycan No 12\n",
      "MGlcn12_rl[e] released mucin-type O-glycan No 12\n",
      "MGlcn130[e] mucin-type O-glycan No 130\n",
      "MGlcn130_rl[e] released mucin-type O-glycan No 130\n",
      "MGlcn131[e] mucin-type O-glycan No 131\n",
      "MGlcn131_rl[e] released mucin-type O-glycan No 131\n",
      "MGlcn132[e] mucin-type O-glycan No 132\n",
      "MGlcn132_rl[e] released mucin-type O-glycan No 132\n",
      "MGlcn133[e] mucin-type O-glycan No 133\n",
      "MGlcn133_rl[e] released mucin-type O-glycan No 133\n",
      "MGlcn134[e] mucin-type O-glycan No 134\n",
      "MGlcn134_rl[e] released mucin-type O-glycan No 134\n",
      "MGlcn135[e] mucin-type O-glycan No 135\n",
      "MGlcn135_rl[e] released mucin-type O-glycan No 135\n",
      "MGlcn136[e] mucin-type O-glycan No 136\n",
      "MGlcn136_rl[e] released mucin-type O-glycan No 136\n",
      "MGlcn13[e] mucin-type O-glycan No 13\n",
      "MGlcn13_rl[e] released mucin-type O-glycan No 13\n",
      "MGlcn147[e] mucin-type O-glycan No 147\n",
      "MGlcn147_rl[e] released mucin-type O-glycan No 147\n",
      "MGlcn14[e] mucin-type O-glycan No 14\n",
      "MGlcn14_rl[e] released mucin-type O-glycan No 14\n",
      "MGlcn150[e] mucin-type O-glycan No 150\n",
      "MGlcn150_rl[e] released mucin-type O-glycan No 150\n",
      "MGlcn151[e] mucin-type O-glycan No 151\n",
      "MGlcn151_rl[e] released mucin-type O-glycan No 151\n",
      "MGlcn152[e] mucin-type O-glycan No 152\n",
      "MGlcn152_rl[e] released mucin-type O-glycan No 152\n",
      "MGlcn15[e] mucin-type O-glycan No 15\n",
      "MGlcn15_rl[e] released mucin-type O-glycan No 15\n",
      "MGlcn162[e] mucin-type O-glycan No 162\n",
      "MGlcn162_rl[e] released mucin-type O-glycan No 162\n",
      "MGlcn164[e] mucin-type O-glycan No 164\n",
      "MGlcn164_rl[e] released mucin-type O-glycan No 164\n",
      "MGlcn165[e] mucin-type O-glycan No 165\n",
      "MGlcn165_rl[e] released mucin-type O-glycan No 165\n",
      "MGlcn166[e] mucin-type O-glycan No 166\n",
      "MGlcn166_rl[e] released mucin-type O-glycan No 166\n",
      "MGlcn167[e] mucin-type O-glycan No 167\n",
      "MGlcn167_rl[e] released mucin-type O-glycan No 167\n",
      "MGlcn168[e] mucin-type O-glycan No 168\n",
      "MGlcn168_rl[e] released mucin-type O-glycan No 168\n",
      "MGlcn169[e] mucin-type O-glycan No 169\n",
      "MGlcn169_rl[e] released mucin-type O-glycan No 169\n",
      "MGlcn16[e] mucin-type O-glycan No 16\n",
      "MGlcn16_rl[e] released mucin-type O-glycan No 16\n",
      "MGlcn171[e] mucin-type O-glycan No 171\n",
      "MGlcn171_rl[e] released mucin-type O-glycan No 171\n",
      "MGlcn174[e] mucin-type O-glycan No 174\n",
      "MGlcn174_rl[e] released mucin-type O-glycan No 174\n",
      "MGlcn175[e] mucin-type O-glycan No 175\n",
      "MGlcn175_rl[e] released mucin-type O-glycan No 175\n",
      "MGlcn176[e] mucin-type O-glycan No 176\n",
      "MGlcn176_rl[e] released mucin-type O-glycan No 176\n",
      "MGlcn177[e] mucin-type O-glycan No 177\n",
      "MGlcn177_rl[e] released mucin-type O-glycan No 177\n",
      "MGlcn178[e] mucin-type O-glycan No 178\n",
      "MGlcn178_rl[e] released mucin-type O-glycan No 178\n",
      "MGlcn179[e] mucin-type O-glycan No 179\n",
      "MGlcn179_rl[e] released mucin-type O-glycan No 179\n",
      "MGlcn17[e] mucin-type O-glycan No 17\n",
      "MGlcn17_rl[e] released mucin-type O-glycan No 17\n",
      "MGlcn180[e] mucin-type O-glycan No 180\n",
      "MGlcn180_rl[e] released mucin-type O-glycan No 180\n",
      "MGlcn181[e] mucin-type O-glycan No 181\n",
      "MGlcn181_rl[e] released mucin-type O-glycan No 181\n",
      "MGlcn183[e] mucin-type O-glycan No 183\n",
      "MGlcn183_rl[e] released mucin-type O-glycan No 183\n",
      "MGlcn185[e] mucin-type O-glycan No 185\n",
      "MGlcn185_rl[e] released mucin-type O-glycan No 185\n",
      "MGlcn186[e] mucin-type O-glycan No 186\n",
      "MGlcn186_rl[e] released mucin-type O-glycan No 186\n",
      "MGlcn187[e] mucin-type O-glycan No 187\n",
      "MGlcn187_rl[e] released mucin-type O-glycan No 187\n",
      "MGlcn188[e] mucin-type O-glycan No 188\n",
      "MGlcn188_rl[e] released mucin-type O-glycan No 188\n",
      "MGlcn189[e] mucin-type O-glycan No 189\n",
      "MGlcn189_rl[e] released mucin-type O-glycan No 189\n",
      "MGlcn18[e] mucin-type O-glycan No 18\n",
      "MGlcn18_rl[e] released mucin-type O-glycan No 18\n",
      "MGlcn190[e] mucin-type O-glycan No 190\n",
      "MGlcn190_rl[e] released mucin-type O-glycan No 190\n",
      "MGlcn191[e] mucin-type O-glycan No 191\n",
      "MGlcn191_rl[e] released mucin-type O-glycan No 191\n",
      "MGlcn192[e] mucin-type O-glycan No 192\n",
      "MGlcn192_rl[e] released mucin-type O-glycan No 192\n",
      "MGlcn193[e] mucin-type O-glycan No 193\n",
      "MGlcn193_rl[e] released mucin-type O-glycan No 193\n",
      "MGlcn19[e] mucin-type O-glycan No 19\n",
      "MGlcn19_rl[e] released mucin-type O-glycan No 19\n",
      "MGlcn20[e] mucin-type O-glycan No 20\n",
      "MGlcn20_rl[e] released mucin-type O-glycan No 20\n",
      "MGlcn22[e] mucin-type O-glycan No 22\n",
      "MGlcn22_rl[e] released mucin-type O-glycan No 22\n",
      "MGlcn2[e] mucin-type O-glycan No 2\n",
      "MGlcn2_rl[e] released mucin-type O-glycan No 2\n",
      "MGlcn3[e] mucin-type O-glycan No 3\n",
      "MGlcn3_rl[e] released mucin-type O-glycan No 3\n",
      "MGlcn46[e] mucin-type O-glycan No 46\n",
      "MGlcn46_rl[e] released mucin-type O-glycan No 46\n",
      "MGlcn48[e] mucin-type O-glycan No 48\n",
      "MGlcn48_rl[e] released mucin-type O-glycan No 48\n",
      "MGlcn4[e] mucin-type O-glycan No 4\n",
      "MGlcn4_rl[e] released mucin-type O-glycan No 4\n",
      "MGlcn50[e] mucin-type O-glycan No 50\n",
      "MGlcn50_rl[e] released mucin-type O-glycan No 50\n",
      "MGlcn51[e] mucin-type O-glycan No 51\n",
      "MGlcn51_rl[e] released mucin-type O-glycan No 51\n",
      "MGlcn53[e] mucin-type O-glycan No 53\n",
      "MGlcn53_rl[e] released mucin-type O-glycan No 53\n",
      "MGlcn54[e] mucin-type O-glycan No 54\n",
      "MGlcn54_rl[e] released mucin-type O-glycan No 54\n",
      "MGlcn57[e] mucin-type O-glycan No 57\n",
      "MGlcn57_rl[e] released mucin-type O-glycan No 57\n",
      "MGlcn58[e] mucin-type O-glycan No 58\n",
      "MGlcn58_rl[e] released mucin-type O-glycan No 58\n",
      "MGlcn59[e] mucin-type O-glycan No 59\n",
      "MGlcn59_rl[e] released mucin-type O-glycan No 59\n",
      "MGlcn5[e] mucin-type O-glycan No 5\n",
      "MGlcn5_rl[e] released mucin-type O-glycan No 5\n",
      "MGlcn60[e] mucin-type O-glycan No 60\n",
      "MGlcn60_rl[e] released mucin-type O-glycan No 60\n",
      "MGlcn61[e] mucin-type O-glycan No 61\n",
      "MGlcn61_rl[e] released mucin-type O-glycan No 61\n",
      "MGlcn62[e] mucin-type O-glycan No 62\n",
      "MGlcn62_rl[e] released mucin-type O-glycan No 62\n",
      "MGlcn63[e] mucin-type O-glycan No 63\n",
      "MGlcn63_rl[e] released mucin-type O-glycan No 63\n",
      "MGlcn64[e] mucin-type O-glycan No 64\n",
      "MGlcn64_rl[e] released mucin-type O-glycan No 64\n",
      "MGlcn65[e] mucin-type O-glycan No 65\n",
      "MGlcn65_rl[e] released mucin-type O-glycan No 65\n",
      "MGlcn66[e] mucin-type O-glycan No 66\n",
      "MGlcn66_rl[e] released mucin-type O-glycan No 66\n",
      "MGlcn67[e] mucin-type O-glycan No 67\n",
      "MGlcn67_rl[e] released mucin-type O-glycan No 67\n",
      "MGlcn68[e] mucin-type O-glycan No 68\n",
      "MGlcn68_rl[e] released mucin-type O-glycan No 68\n",
      "MGlcn6[e] mucin-type O-glycan No 6\n",
      "MGlcn6_rl[e] released mucin-type O-glycan No 6\n",
      "MGlcn78[e] mucin-type O-glycan No 78\n",
      "MGlcn78_rl[e] released mucin-type O-glycan No 78\n",
      "MGlcn7[e] mucin-type O-glycan No 7\n",
      "MGlcn7_rl[e] released mucin-type O-glycan No 7\n",
      "MGlcn81[e] mucin-type O-glycan No 81\n",
      "MGlcn81_rl[e] released mucin-type O-glycan No 81\n",
      "MGlcn88[e] mucin-type O-glycan No 88\n",
      "MGlcn88_rl[e] released mucin-type O-glycan No 88\n",
      "MGlcn8[e] mucin-type O-glycan No 8\n",
      "MGlcn8_rl[e] released mucin-type O-glycan No 8\n",
      "MGlcn9[e] mucin-type O-glycan No 9\n",
      "MGlcn9_rl[e] released mucin-type O-glycan No 9\n",
      "T_antigen[e] T antigen (core 1)\n",
      "Tn_antigen[e] Tn antigen\n",
      "ac[e] acetate\n",
      "acald[e] acetaldehyde\n",
      "acgal[e] N-acetyl-D-galactosamine\n",
      "acgam[e] N-acetyl-D-glucosamine\n",
      "acnam2a6agal[e] Alpha-2,6-N-acetylneuraminate-N-acetylgalactosamine\n",
      "acnam[e] N-acetylneuraminate\n",
      "adocbl[e] Adenosylcobalamin\n",
      "agam1b3agal[e] Beta-1,3-N-acetylglucosamine-N-acetylgalactosamine\n",
      "ala_D[e] D-alanine\n",
      "ala_L[e] L-alanine\n",
      "alaasp[e] L-alanyl-L-aspartate\n",
      "alagln[e] L-alanyl-L-glutamine\n",
      "alaglu[e] L-alanyl-L-glutamate\n",
      "alagly[e] L-alanylglycine\n",
      "alahis[e] L-alanyl-L-histidine\n",
      "alaleu[e] L-alanyl-L-leucine\n",
      "alathr[e] L-alanyl-L-threonine\n",
      "alltn[e] Allantoin\n",
      "arab_L[e] L-arabinose\n",
      "arbt[e] Arbutin\n",
      "arg_L[e] L-argininium(1+)\n",
      "asn_L[e] L-asparagine\n",
      "aso3[e] arsenite\n",
      "aso4[e] arsenate\n",
      "asp_L[e] L-aspartate(1-)\n",
      "bhb[e] (R)-3-hydroxybutyrate\n",
      "btn[e] Biotin\n",
      "but[e] butyrate\n",
      "butso3[e] butanesulfonate\n",
      "ca2[e] calcium(2+)\n",
      "cbl1[e] Cob(I)alamin\n",
      "cbl2[e] Cob(II)alamin\n",
      "cd2[e] Cadmium\n",
      "cellb[e] cellobiose\n",
      "cgly[e] L-cysteinylglycine\n",
      "ch4s[e] Methanethiol\n",
      "cl[e] Chloride\n",
      "co2[e] carbon dioxide\n",
      "cobalt2[e] Co2+\n",
      "core2[e] Core 2\n",
      "core2_rl[e] released Core 2\n",
      "core3[e] Core 3\n",
      "cro4[e] chromate\n",
      "csn[e] cytosine\n",
      "cu2[e] Cu2+\n",
      "cys_L[e] L-cysteine\n",
      "drib[e] 2-deoxy-D-ribose\n",
      "dsT_antigen[e] disialyl-T antigen\n",
      "dsT_antigen_rl[e] released disialyl-T antigen\n",
      "etha[e] etha[c]\n",
      "ethso3[e] ethanesulfonate\n",
      "etoh[e] ethanol\n",
      "f1a[e] F1alpha\n",
      "f1a_rl[e] released F1alpha antigen\n",
      "fe2[e] Fe2+\n",
      "fe3[e] Fe3+\n",
      "fecrm[e] Ferrichrome\n",
      "for[e] Formate\n",
      "fru[e] D-Fructose\n",
      "gal1b3acgal[e] Beta-1,3-Galactose-N-acetylgalactosamine\n",
      "gal[e] D-Galactose\n",
      "galt[e] galactitol\n",
      "gcald[e] glycolaldehyde\n",
      "glc_D[e] D-glucose\n",
      "gln_L[e] L-glutamine\n",
      "glu_L[e] L-glutamate(1-)\n",
      "gly[e] Glycine\n",
      "glyasn[e] Glycyl-L-asparagine\n",
      "glyasp[e] Glycyl-L-aspartate\n",
      "glyb[e] glycine betaine\n",
      "glyc3p[e] Glycerol 3-phosphate\n",
      "glycys[e] Gly-Cys\n",
      "glygln[e] Glycyl-L-glutamine\n",
      "glyglu[e] Glycyl-L-glutamate\n",
      "glyleu[e] Glycylleucine\n",
      "glymet[e] Glycyl-L-methionine\n",
      "glyphe[e] Glycylphenylalanine\n",
      "glypro[e] Glycylproline\n",
      "glytyr[e] Glycyl-L-tyrosine\n",
      "h2[e] Hydrogen\n",
      "h2o[e] Water\n",
      "h2s[e] Hydrogen sulfide\n",
      "h[e] proton\n",
      "hdca[e] Hexadecanoate (n-C16:0)\n",
      "hexs[e] hexanesulfonate\n",
      "hg2[e] Hg2+\n",
      "his_L[e] L-histidine\n",
      "hxan[e] Hypoxanthine\n",
      "ile_L[e] L-isoleucine\n",
      "ind3ac[e] indole-3-acetate\n",
      "ind3ppa[e] Indole-3-propionate\n",
      "isetac[e] Isethionic acid\n",
      "isobut[e] Isobutyrate, 2-Methylpropanoate\n",
      "isocapr[e] Isocaproate\n",
      "isoval[e] Isovalerate, 3-Methylbutanoate\n",
      "k[e] potassium\n",
      "lac_D[e] (R)-lactate\n",
      "lac_L[e] (S)-lactate\n",
      "lcts[e] Lactose\n",
      "leu_L[e] L-leucine\n",
      "lys_L[e] L-lysinium(1+)\n",
      "malt[e] Maltose\n",
      "malthx[e] Maltohexaose\n",
      "malttr[e] maltotriose\n",
      "man[e] D-Mannose\n",
      "met_D[e] D-Methionine\n",
      "met_L[e] L-methionine\n",
      "metala[e] L-methionyl-L-alanine\n",
      "metsox_R_L[e] L-methionine-R-sulfoxide\n",
      "metsox_S_L[e] L-Methionine Sulfoxide\n",
      "mg2[e] magnesium\n",
      "mn2[e] Mn2+\n",
      "mnl[e] D-Mannitol\n",
      "mobd[e] Molybdate\n",
      "mops[e] 3-(n-Morpholino)Propanesulfonic Acid\n",
      "mso3[e] methanesulfonate\n",
      "na1[e] Sodium\n",
      "nac[e] Nicotinate\n",
      "nh4[e] Ammonium\n",
      "no2[e] Nitrite\n",
      "no3[e] Nitrate\n",
      "o2[e] O2\n",
      "ocdca[e] octadecanoate (n-C18:0)\n",
      "orn[e] Ornithine\n",
      "oxa[e] oxalate(2-)\n",
      "pac[e] phenylacetate\n",
      "pb[e] Lead\n",
      "pcresol[e] p-Cresol\n",
      "phe_L[e] L-phenylalanine\n",
      "phpyr[e] keto-phenylpyruvate\n",
      "pi[e] hydrogenphosphate\n",
      "pnto_R[e] (R)-Pantothenate\n",
      "ppa[e] propionate\n",
      "pppn[e] Phenylpropanoate\n",
      "pro_L[e] L-proline\n",
      "ptrc[e] Putrescine\n",
      "pydx[e] Pyridoxal\n",
      "pydxn[e] Pyridoxine\n",
      "pyr[e] pyruvate\n",
      "rib_D[e] D-ribose\n",
      "ribflv[e] Riboflavin\n",
      "sT_antigen[e] sialyl-T antigen\n",
      "sT_antigen_rl[e] released sialyl-Tn antigen\n",
      "sTn_antigen[e] sialyl-Tn antigen\n",
      "salcn[e] Salicin\n",
      "sbt_D[e] D-glucitol\n",
      "ser_L[e] L-serine\n",
      "so4[e] sulfate\n",
      "spmd[e] Spermidine\n",
      "succ[e] Succinate\n",
      "sulfac[e] sulfoacetate\n",
      "taur[e] Taurine\n",
      "thm[e] Thiamin\n",
      "thr_L[e] L-threonine\n",
      "tma[e] trimethylamine\n",
      "tre[e] Trehalose\n",
      "trp_L[e] L-tryptophan\n",
      "ttdca[e] tetradecanoate (n-C14:0)\n",
      "tyr_L[e] L-tyrosine\n",
      "ura[e] uracil\n",
      "urea[e] Urea\n",
      "val_L[e] L-valine\n",
      "xan[e] Xanthine\n",
      "xyl_D[e] aldehydo-D-xylose\n",
      "zn2[e] Zinc\n"
     ]
    }
   ],
   "source": [
    "for cpd in agora_NAP07.metabolites:\n",
    "    if '[e]' in cpd.id:\n",
    "        print(cpd.id, cpd.name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>10m3oundecACP[c]</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>10-methyl-3-oxo-undecanoyl-ACP</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f9d27de90d0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C23H41N2O9PRS</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 2 reaction(s)</strong></td><td>\n",
       "                    3OAS8M9, 3OAR10M11</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite 10m3oundecACP[c] at 0x7f9d27de90d0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_NAP07.metabolites[5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Model does not contain SBML fbc package information.\n",
      "SBML package 'layout' not supported by cobrapy, information is not parsed\n",
      "SBML package 'render' not supported by cobrapy, information is not parsed\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00858_c0 \"L_Glutamate5_semialdehyde_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00001_c0 \"H2O_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00067_c0 \"H_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02431_c0 \"1_Pyrroline_5_carboxylate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00001_e0 \"H2O_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd09027_c0 \"5_Hydroxy_2_oxo_4_ureido_2_5_dihydro_1H_imidazole_5_carboxylate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00011_c0 \"CO2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01567_c0 \"R_Allantoin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02074_c0 \"2_Oxo_6_aminocaproate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00922_c0 \"delta1_Piperideine_2_carboxylate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02414_c0 \"L_2_Amino_6_oxopimelate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02465_c0 \"tetrahydrodipicolinate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00002_c0 \"ATP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00254_e0 \"Mg_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00008_c0 \"ADP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00009_c0 \"Phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00254_c0 \"Mg_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd08625_c0 \"5_Hydroxyisourate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd10516_e0 \"fe3_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd10516_c0 \"fe3_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00007_e0 \"O2_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00007_c0 \"O2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01092_c0 \"Allantoin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00011_e0 \"CO2_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00003_c0 \"NAD_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00035_c0 \"L_Alanine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00357_c0 \"TTP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15793_c0 \"Stearoylcardiolipin_B_subtilis_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15795_c0 \"Anteisoheptadecanoylcardiolipin_B_subtilis_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15757_c0 \"Stearoyllipoteichoic_acid_n24__linked__glucose_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15766_c0 \"Stearoyllipoteichoic_acid_n24__linked__N_acetyl_D_glucosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00166_c0 \"Calomide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00030_c0 \"Mn2_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00322_c0 \"L_Isoleucine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15352_c0 \"2_Demethylmenaquinone_8_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00132_c0 \"L_Asparagine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00053_c0 \"L_Glutamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00119_c0 \"L_Histidine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00063_c0 \"Ca2_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15794_c0 \"Isoheptadecanoylcardiolipin_B_subtilis_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00012_c0 \"PPi_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11493_c0 \"ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00023_c0 \"L_Glutamate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00041_c0 \"L_Aspartate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00161_c0 \"L_Threonine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15696_c0 \"Dianteisoheptadecanoylphosphatidylethanolamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15748_c0 \"Stearoyllipoteichoic_acid_n24__linked__unsubstituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00006_c0 \"NADP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00220_c0 \"Riboflavin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02229_c0 \"Bactoprenyl_diphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd17041_c0 \"Protein_biosynthesis_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11459_c0 \"tcam_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00264_c0 \"Spermidine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15768_c0 \"Anteisoheptadecanoyllipoteichoic_acid_n24__linked__N_acetyl_D_glucosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03422_c0 \"Cobinamide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00156_c0 \"L_Valine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00201_c0 \"10_Formyltetrahydrofolate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00042_c0 \"GSH_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00118_c0 \"Putrescine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00557_c0 \"Siroheme_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15500_c0 \"Menaquinone_8_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00058_c0 \"Cu2_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15758_c0 \"Isoheptadecanoyllipoteichoic_acid_n24__linked__glucose_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00010_c0 \"CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00038_c0 \"GTP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00129_c0 \"L_Proline_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00056_c0 \"TPP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00052_c0 \"CTP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00241_c0 \"dGTP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00149_c0 \"Co2_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00033_c0 \"Glycine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01997_c0 \"Dimethylbenzimidazole_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00060_c0 \"L_Methionine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15775_c0 \"Stearoyllipoteichoic_acid_n24__linked__D_alanine_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15540_c0 \"Phosphatidylglycerol_dioctadecanoyl_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00087_c0 \"Tetrahydrofolate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd17043_c0 \"RNA_transcription_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15533_c0 \"phosphatidylethanolamine_dioctadecanoyl_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00084_c0 \"L_Cysteine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15667_c0 \"glycerol_teichoic_acid_n45__linked__unsubstituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00066_c0 \"L_Phenylalanine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15666_c0 \"Peptidoglycan_polymer_n_1_subunits_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15759_c0 \"Anteisoheptadecanoyllipoteichoic_acid_n24__linked__glucose_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15669_c0 \"glycerol_teichoic_acid_n45__linked__glucose_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00015_c0 \"FAD_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15695_c0 \"Diisoheptadecanoylphosphatidylethanolamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00016_c0 \"Pyridoxal_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00069_c0 \"L_Tyrosine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15776_c0 \"Isoheptadecanoyllipoteichoic_acid_n24__linked__D_alanine_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15665_c0 \"Peptidoglycan_polymer_n_subunits_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00048_c0 \"Sulfate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00062_c0 \"UTP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00065_c0 \"L_Tryptophan_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00107_c0 \"L_Leucine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00205_c0 \"K_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15723_c0 \"Dianteisoheptadecanoylphosphatidylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00028_c0 \"Heme_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00345_c0 \"5_Methyltetrahydrofolate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15722_c0 \"Diisoheptadecanoylphosphatidylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00054_c0 \"L_Serine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15777_c0 \"Anteisoheptadecanoyllipoteichoic_acid_n24__linked__D_alanine_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00356_c0 \"dCTP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15749_c0 \"Isoheptadecanoyllipoteichoic_acid_n24__linked__unsubstituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd12370_c0 \"apo_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00099_c0 \"Cl__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd17042_c0 \"DNA_replication_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd10515_c0 \"Fe2_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00017_c0 \"S_Adenosyl_L_methionine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00051_c0 \"L_Arginine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15668_c0 \"glycerol_teichoic_acid_n45__linked__D_ala_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15560_c0 \"Ubiquinone_8_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15767_c0 \"Isoheptadecanoyllipoteichoic_acid_n24__linked__N_acetyl_D_glucosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00039_c0 \"L_Lysine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00034_c0 \"Zn2_plus__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11416_c0 \"Biomass_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11416_b \"Biomass_c0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00115_c0 \"dATP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15750_c0 \"Anteisoheptadecanoyllipoteichoic_acid_n24__linked__unsubstituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00186_c0 \"D_Glutamate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00890_c0 \"UDP_N_acetylmuramoyl_L_alanine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00525_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00099_e0 \"Cl__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15677_c0 \"1_2_diisoheptadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15683_c0 \"CDP_1_2_diisoheptadecanoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11530_c0 \"9_methyl_3_hydroxy_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11531_c0 \"9_methyl_trans_dec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11509_c0 \"10_methyl_3_hydroxy_dodecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11510_c0 \"10_methyl_trans_dodec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03915_c0 \"CobIyrinate_diamide_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00421_c0 \"Triphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03916_c0 \"Adenosyl_cobyrinate_diamide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00067_e0 \"H_plus__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00655_e0 \"Dephospho_CoA_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00655_c0 \"Dephospho_CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03834_c0 \"Precorrin_4_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00019_c0 \"S_Adenosyl_homocysteine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03839_c0 \"Precorrin_5_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00026_c0 \"UDP_glucose_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15741_c0 \"Monoglucosyl_1_2_dianteisoheptadecanoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00014_c0 \"UDP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15732_c0 \"Diglucosyl_1_2_dianteisoheptadecanoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00149_e0 \"Co2_plus__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00037_c0 \"UDP_N_acetylglucosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00061_c0 \"Phosphoenolpyruvate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02820_c0 \"UDP_N_acetylglucosamine_enolpyruvate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00298_e0 \"dTMP_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00298_c0 \"dTMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15269_c0 \"octadecenoate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00018_c0 \"AMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15274_c0 \"Octadecenoyl_CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01188_e0 \"Lanosterol_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01188_c0 \"Lanosterol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00175_c0 \"UDP_N_acetyl_D_galactosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00091_c0 \"UMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00508_c0 \"3MOP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00004_c0 \"NADH_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00760_c0 \"2_Methylbutyryl_CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00100_c0 \"Glycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15311_c0 \"1_2_Diacyl_sn_glycerol_dioctadecanoyl_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15739_c0 \"Monoglucosyl_1_2_distearoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00186_e0 \"D_Glutamate_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11434_c0 \"fa12coa_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15672_c0 \"1_anteisoheptadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15678_c0 \"1_2_dianteisoheptadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00492_e0 \"N_Acetyl_D_mannosamine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00492_c0 \"N_Acetyl_D_mannosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00005_c0 \"NADPH_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11518_c0 \"14_methyl_trans_hexa_dec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11519_c0 \"14_methyl_hexa_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11517_c0 \"14_methyl_3_hydroxy_hexa_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03760_c0 \"Precorrin_6B_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03761_c0 \"Precorrin_6A_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00080_c0 \"Glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00322_e0 \"L_Isoleucine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03494_c0 \"Undecaprenyl_diphospho_N_acetylmuramoyl_L_alanyl_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03495_c0 \"Undecaprenyl_diphospho_N_acetylmuramoyl__N_acetylglucosamine_L_ala_D_glu_meso_2_6_diaminopimeloyl_D_ala_D_ala_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15702_c0 \"1_2_Dianteisoheptadecanoyl_sn_glycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11513_c0 \"12_methyl_3_hydroxy_tetra_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11514_c0 \"12_methyl_trans_tetra_dec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11515_c0 \"12_methyl_tetra_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00218_e0 \"Niacin_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00218_c0 \"Niacin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00523_c0 \"Trehalose_6_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00027_c0 \"D_Glucose_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00079_c0 \"D_glucose_6_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00861_c0 \"UDP_N_acetyl_D_mannosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00946_c0 \"Undecaprenyl_diphospho_N_acetyl_glucosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02967_c0 \"N_Acetyl_beta_D_mannosaminyl_1_4_N_acetyl_D__glucosaminyldiphosphoundecaprenol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11594_c0 \"Dextrin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11621_c0 \"Oxidizedferredoxin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00024_c0 \"2_Oxoglutarate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11620_c0 \"Reducedferredoxin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02295_c0 \"1_4_Dihydroxy_2_naphthoate_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02557_c0 \"Farnesylfarnesylgeraniol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15353_c0 \"2_Demethylmenaquinol_8_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00776_c0 \"Squalene_2_3_oxide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00125_c0 \"5_10_Methylenetetrahydrofolate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00658_c0 \"Isochorismate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00020_c0 \"Pyruvate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03451_c0 \"SHCHC_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11587_c0 \"Ala_Gln_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00791_c0 \"ProtoporphyrinogenIX_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01476_c0 \"Protoporphyrin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00080_e0 \"Glycerol_3_phosphate_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01620_c0 \"Precorrin_2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03420_c0 \"Precorrin_3A_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00054_e0 \"L_Serine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15684_c0 \"CDP_1_2_dianteisoheptadecanoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11542_c0 \"15_methyl_3_hydroxy_hexa_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11543_c0 \"15_methyl_trans_hexa_dec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00069_e0 \"L_Tyrosine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00013_c0 \"NH3_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00025_c0 \"H2O2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00032_c0 \"Oxaloacetate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00122_c0 \"N_Acetyl_D_glucosamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00731_c0 \"Ala_Ala_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02964_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_gamma_glutamyl_meso_2_6_diaminopimelate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02968_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamyl_6_carboxy_L_lysyl_D_alanyl__D_alanine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11492_c0 \"Malonyl_acyl_carrierprotein__c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11528_c0 \"7_methyl_octanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11529_c0 \"9_methyl_3_oxo_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15358_c0 \"2_octadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01080_c0 \"ocdca_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00113_c0 \"Isopentenyldiphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00202_c0 \"DMAPP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00283_c0 \"Geranyldiphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00046_c0 \"CMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15717_c0 \"Dianteisoheptadecanoylphosphatidylglycerophosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11522_c0 \"5_methyl_3_hydroxy_hexanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11523_c0 \"5_methyl_trans_hex_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00559_c0 \"Supraene_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03833_c0 \"Precorrin_3B_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11496_c0 \"4_methyl_3_oxo_hexanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11497_c0 \"4_methyl_3_hydroxy_hexanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00012_e0 \"PPi_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11537_c0 \"13_methyl_3_oxo_tetra_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11538_c0 \"13_methyl_3_hydroxy_tetra_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00115_e0 \"dATP_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02039_c0 \"1_Aminopropan_2_ol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03917_c0 \"Adenosylcobyric_acid_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03918_c0 \"Adenosyl_cobinamide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00081_c0 \"Sulfite_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00111_c0 \"Oxidized_glutathione_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00193_c0 \"APS_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15323_c0 \"1_Acyl_sn_glycero_3_phosphoglycerol_octadecanoyl_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00350_c0 \"Farnesyldiphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00162_e0 \"Aminoethanol_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00162_c0 \"Aminoethanol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00773_c0 \"UDP_MurNAc_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00286_c0 \"Undecaprenylphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00393_c0 \"Folate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03835_c0 \"Precorrin_8_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03832_c0 \"Hydrogenobyrinate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15662_c0 \"Prenol_45n_teichoic_acid_glucose_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11500_c0 \"6_methyl_3_oxo_octanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11501_c0 \"6_methyl_3_hydroxy_octanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03443_c0 \"3_Octaprenyl_4_hydroxybenzoate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03444_c0 \"2_Octaprenylphenol_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11520_c0 \"isovaleryl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11521_c0 \"5_methyl_3_oxo_hexanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00774_c0 \"UroporphyrinogenIII_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02083_c0 \"CoproporphyrinogenIII_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00327_c0 \"strcoa_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15329_c0 \"1_octadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15526_c0 \"1_2_dioctadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00051_e0 \"L_Arginine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00013_e0 \"NH3_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01298_c0 \"Aminoacetone_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00428_c0 \"2_Oxopropanal_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00873_c0 \"Nicotinate_ribonucleotide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02904_c0 \"alpha_Ribazole_5_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00050_c0 \"FMN_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00031_c0 \"GDP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11587_e0 \"Ala_Gln_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03423_c0 \"alpha_Ribazole_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00215_c0 \"Pyridoxal_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00299_c0 \"dUMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00982_c0 \"FADH2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00081_e0 \"Sulfite_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11544_c0 \"15_methyl_hexa_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11432_c0 \"fa11coa_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11526_c0 \"7_methyl_3_hydroxy_octanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11527_c0 \"7_methyl_trans_oct_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00023_e0 \"L_Glutamate_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00118_e0 \"Putrescine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00696_c0 \"3_5_Cyclic_CMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11524_c0 \"5_methyl_hexanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00065_e0 \"L_Tryptophan_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00402_c0 \"CDPglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15731_c0 \"Diglucosyl_1_2_diisoheptadecanoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03447_c0 \"2_Octaprenyl_6_methoxy_1_4_benzoquinone_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03448_c0 \"2_Octaprenyl_3_methyl_6_methoxy_1_4_benzoquinone_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15690_c0 \"Dianteisoheptadecanoylphosphatidylserine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11533_c0 \"11_methyl_3_oxo_dodecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11534_c0 \"11_methyl_3_hydroxy_dodecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11505_c0 \"8_methyl_3_hydroxy_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11506_c0 \"8_methyl_trans_dec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11503_c0 \"6_methyl_octanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11504_c0 \"8_methyl_3_oxo_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00117_c0 \"D_Alanine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd12547_c0 \"L_2_Lysophosphatidylethanolamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00908_c0 \"Glycerophosphoethanolamine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11525_c0 \"7_methyl_3_oxo_octanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03426_c0 \"Sirohydrochlorin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15661_c0 \"Prenol_45n_teichoic_acid_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00066_e0 \"L_Phenylalanine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11532_c0 \"9_methyl_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00075_c0 \"Nitrite_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11591_e0 \"Gly_Met_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11591_c0 \"Gly_Met_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11511_c0 \"10_methyl_dodecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11512_c0 \"12_methyl_3_oxo_tetra_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00696_e0 \"3_5_Cyclic_CMP_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00028_e0 \"Heme_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11516_c0 \"14_methyl_3_oxo_hexa_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00528_e0 \"N2_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00528_c0 \"N2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00200_c0 \"4MOP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01882_c0 \"Isovaleryl_CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15730_c0 \"Diglucosyl_1_2_distearoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00136_c0 \"4_Hydroxybenzoate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11640_c0 \"H2_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11594_e0 \"Dextrin_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00058_e0 \"Cu2_plus__e0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00156_e0 \"L_Valine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03449_c0 \"2_Octaprenyl_3_methyl_5_hydroxy_6_methoxy_1_4_benzoquinone_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00095_c0 \"Glycerone_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00277_e0 \"Deoxyguanosine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00277_c0 \"Deoxyguanosine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00305_c0 \"Thiamin_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15268_c0 \"Octadecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00205_e0 \"K_plus__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02090_c0 \"Glycerophosphoglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11425_c0 \"2dmmq7_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11606_c0 \"Menaquinone_7_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00030_e0 \"Mn2_plus__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00638_c0 \"Deamido_NAD_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00132_e0 \"L_Asparagine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00264_e0 \"Spermidine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11495_c0 \"2_methylbutyryl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11541_c0 \"15_methyl_3_oxo_hexa_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00034_e0 \"Zn2_plus__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15689_c0 \"Diisoheptadecanoylphosphatidylserine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00010_e0 \"CoA_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00516_e0 \"meso_2_6_Diaminopimelate_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00516_c0 \"meso_2_6_Diaminopimelate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15663_c0 \"Prenol_45n_teichoic_acid_alanine_substituted_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00063_e0 \"Ca2_plus__e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00038_e0 \"GTP_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00357_e0 \"TTP_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11539_c0 \"13_methyl_trans_tetra_dec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15671_c0 \"1_isoheptadecanoyl_sn_glycerol_3_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15740_c0 \"Monoglucosyl_1_2_diisoheptadecanoylglycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00075_e0 \"Nitrite_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11507_c0 \"8_methyl_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11508_c0 \"10_methyl_3_oxo_dodecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00006_e0 \"NADP_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11498_c0 \"4_methyl_trans_hex_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11499_c0 \"4_methyl_hexanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03422_e0 \"Cobinamide_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00216_c0 \"Chorismate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15716_c0 \"Diisoheptadecanoylphosphatidylglycerophosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00017_e0 \"S_Adenosyl_L_methionine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03919_c0 \"Adenosyl_cobinamide_phosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03920_c0 \"Adenosylcobinamide_GDP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03913_c0 \"Hydrogenobyrinate_diamide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03914_c0 \"CobIIyrinate_diamide_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00122_e0 \"N_Acetyl_D_glucosamine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00117_e0 \"D_Alanine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11540_c0 \"13_methyl_tetra_decanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00119_e0 \"L_Histidine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02590_c0 \"all_trans_Heptaprenyl_diphosphate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00126_c0 \"GMP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11535_c0 \"11_methyl_trans_dodec_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11536_c0 \"11_methyl_dodecanoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00129_e0 \"L_Proline_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01772_c0 \"Succinylbenzoate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03445_c0 \"2_Octaprenyl_6_hydroxyphenol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03446_c0 \"2_Octaprenyl_6_methoxyphenol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00107_e0 \"L_Leucine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00305_e0 \"Thiamin_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00220_e0 \"Riboflavin_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd02021_c0 \"Succinylbenzoyl_CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00393_e0 \"Folate_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00033_e0 \"Glycine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11640_e0 \"H2_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00215_e0 \"Pyridoxal_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15701_c0 \"1_2_Diisoheptadecanoyl_sn_glycerol_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11488_c0 \"Acetoacetyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11606_e0 \"Menaquinone_7_e0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00311_c0 \"Guanosine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00161_e0 \"L_Threonine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01017_c0 \"Cys_Gly_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15237_c0 \"hexadecenoate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00022_c0 \"Acetyl_CoA_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00311_e0 \"Guanosine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00111_e0 \"Oxidized_glutathione_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15237_e0 \"hexadecenoate_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01080_e0 \"ocdca_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00834_c0 \"Phosphopantetheine_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11502_c0 \"6_methyl_trans_oct_2_enoyl_ACP_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00029_c0 \"Acetate_c0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00039_e0 \"L_Lysine_e0\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00001_b \"H2O_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00254_b \"Mg_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd10516_b \"fe3_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00007_b \"O2_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00011_b \"CO2_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00099_b \"Cl__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00067_b \"H_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00655_b \"Dephospho_CoA_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00149_b \"Co2_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00298_b \"dTMP_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01188_b \"Lanosterol_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00186_b \"D_Glutamate_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00492_b \"N_Acetyl_D_mannosamine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00322_b \"L_Isoleucine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00218_b \"Niacin_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00080_b \"Glycerol_3_phosphate_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00054_b \"L_Serine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00069_b \"L_Tyrosine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00012_b \"PPi_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00115_b \"dATP_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00162_b \"Aminoethanol_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00051_b \"L_Arginine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00013_b \"NH3_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11587_b \"Ala_Gln_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00081_b \"Sulfite_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00023_b \"L_Glutamate_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00118_b \"Putrescine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00065_b \"L_Tryptophan_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00066_b \"L_Phenylalanine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11591_b \"Gly_Met_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00696_b \"3_5_Cyclic_CMP_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00028_b \"Heme_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00528_b \"N2_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11594_b \"Dextrin_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00058_b \"Cu2_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00156_b \"L_Valine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00277_b \"Deoxyguanosine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00205_b \"K_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00030_b \"Mn2_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00132_b \"L_Asparagine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00264_b \"Spermidine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00034_b \"Zn2_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00010_b \"CoA_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00516_b \"meso_2_6_Diaminopimelate_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00063_b \"Ca2_plus__e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00038_b \"GTP_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00357_b \"TTP_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00075_b \"Nitrite_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00006_b \"NADP_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd03422_b \"Cobinamide_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00017_b \"S_Adenosyl_L_methionine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00122_b \"N_Acetyl_D_glucosamine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00117_b \"D_Alanine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00119_b \"L_Histidine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00129_b \"L_Proline_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00107_b \"L_Leucine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00305_b \"Thiamin_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00220_b \"Riboflavin_e0_b\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00393_b \"Folate_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00033_b \"Glycine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11640_b \"H2_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00215_b \"Pyridoxal_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd11606_b \"Menaquinone_7_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00161_b \"L_Threonine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00311_b \"Guanosine_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00111_b \"Oxidized_glutathione_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd15237_b \"hexadecenoate_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd01080_b \"ocdca_e0_b\">\n",
      "Use of the species charge attribute is discouraged, use fbc:charge instead: <Species M_cpd00039_b \"L_Lysine_e0_b\">\n",
      "Adding exchange reaction EX_cpd11416_b with default bounds for boundary metabolite: cpd11416_b.\n",
      "Adding exchange reaction EX_cpd00001_b with default bounds for boundary metabolite: cpd00001_b.\n",
      "Adding exchange reaction EX_cpd00254_b with default bounds for boundary metabolite: cpd00254_b.\n",
      "Adding exchange reaction EX_cpd10516_b with default bounds for boundary metabolite: cpd10516_b.\n",
      "Adding exchange reaction EX_cpd00007_b with default bounds for boundary metabolite: cpd00007_b.\n",
      "Adding exchange reaction EX_cpd00011_b with default bounds for boundary metabolite: cpd00011_b.\n",
      "Adding exchange reaction EX_cpd00099_b with default bounds for boundary metabolite: cpd00099_b.\n",
      "Adding exchange reaction EX_cpd00067_b with default bounds for boundary metabolite: cpd00067_b.\n",
      "Adding exchange reaction EX_cpd00655_b with default bounds for boundary metabolite: cpd00655_b.\n",
      "Adding exchange reaction EX_cpd00149_b with default bounds for boundary metabolite: cpd00149_b.\n",
      "Adding exchange reaction EX_cpd00298_b with default bounds for boundary metabolite: cpd00298_b.\n",
      "Adding exchange reaction EX_cpd01188_b with default bounds for boundary metabolite: cpd01188_b.\n",
      "Adding exchange reaction EX_cpd00186_b with default bounds for boundary metabolite: cpd00186_b.\n",
      "Adding exchange reaction EX_cpd00492_b with default bounds for boundary metabolite: cpd00492_b.\n",
      "Adding exchange reaction EX_cpd00322_b with default bounds for boundary metabolite: cpd00322_b.\n",
      "Adding exchange reaction EX_cpd00218_b with default bounds for boundary metabolite: cpd00218_b.\n",
      "Adding exchange reaction EX_cpd00080_b with default bounds for boundary metabolite: cpd00080_b.\n",
      "Adding exchange reaction EX_cpd00054_b with default bounds for boundary metabolite: cpd00054_b.\n",
      "Adding exchange reaction EX_cpd00069_b with default bounds for boundary metabolite: cpd00069_b.\n",
      "Adding exchange reaction EX_cpd00012_b with default bounds for boundary metabolite: cpd00012_b.\n",
      "Adding exchange reaction EX_cpd00115_b with default bounds for boundary metabolite: cpd00115_b.\n",
      "Adding exchange reaction EX_cpd00162_b with default bounds for boundary metabolite: cpd00162_b.\n",
      "Adding exchange reaction EX_cpd00051_b with default bounds for boundary metabolite: cpd00051_b.\n",
      "Adding exchange reaction EX_cpd00013_b with default bounds for boundary metabolite: cpd00013_b.\n",
      "Adding exchange reaction EX_cpd11587_b with default bounds for boundary metabolite: cpd11587_b.\n",
      "Adding exchange reaction EX_cpd00081_b with default bounds for boundary metabolite: cpd00081_b.\n",
      "Adding exchange reaction EX_cpd00023_b with default bounds for boundary metabolite: cpd00023_b.\n",
      "Adding exchange reaction EX_cpd00118_b with default bounds for boundary metabolite: cpd00118_b.\n",
      "Adding exchange reaction EX_cpd00065_b with default bounds for boundary metabolite: cpd00065_b.\n",
      "Adding exchange reaction EX_cpd00066_b with default bounds for boundary metabolite: cpd00066_b.\n",
      "Adding exchange reaction EX_cpd11591_b with default bounds for boundary metabolite: cpd11591_b.\n",
      "Adding exchange reaction EX_cpd00696_b with default bounds for boundary metabolite: cpd00696_b.\n",
      "Adding exchange reaction EX_cpd00028_b with default bounds for boundary metabolite: cpd00028_b.\n",
      "Adding exchange reaction EX_cpd00528_b with default bounds for boundary metabolite: cpd00528_b.\n",
      "Adding exchange reaction EX_cpd11594_b with default bounds for boundary metabolite: cpd11594_b.\n",
      "Adding exchange reaction EX_cpd00058_b with default bounds for boundary metabolite: cpd00058_b.\n",
      "Adding exchange reaction EX_cpd00156_b with default bounds for boundary metabolite: cpd00156_b.\n",
      "Adding exchange reaction EX_cpd00277_b with default bounds for boundary metabolite: cpd00277_b.\n",
      "Adding exchange reaction EX_cpd00205_b with default bounds for boundary metabolite: cpd00205_b.\n",
      "Adding exchange reaction EX_cpd00030_b with default bounds for boundary metabolite: cpd00030_b.\n",
      "Adding exchange reaction EX_cpd00132_b with default bounds for boundary metabolite: cpd00132_b.\n",
      "Adding exchange reaction EX_cpd00264_b with default bounds for boundary metabolite: cpd00264_b.\n",
      "Adding exchange reaction EX_cpd00034_b with default bounds for boundary metabolite: cpd00034_b.\n",
      "Adding exchange reaction EX_cpd00010_b with default bounds for boundary metabolite: cpd00010_b.\n",
      "Adding exchange reaction EX_cpd00516_b with default bounds for boundary metabolite: cpd00516_b.\n",
      "Adding exchange reaction EX_cpd00063_b with default bounds for boundary metabolite: cpd00063_b.\n",
      "Adding exchange reaction EX_cpd00038_b with default bounds for boundary metabolite: cpd00038_b.\n",
      "Adding exchange reaction EX_cpd00357_b with default bounds for boundary metabolite: cpd00357_b.\n",
      "Adding exchange reaction EX_cpd00075_b with default bounds for boundary metabolite: cpd00075_b.\n",
      "Adding exchange reaction EX_cpd00006_b with default bounds for boundary metabolite: cpd00006_b.\n",
      "Adding exchange reaction EX_cpd03422_b with default bounds for boundary metabolite: cpd03422_b.\n",
      "Adding exchange reaction EX_cpd00017_b with default bounds for boundary metabolite: cpd00017_b.\n",
      "Adding exchange reaction EX_cpd00122_b with default bounds for boundary metabolite: cpd00122_b.\n",
      "Adding exchange reaction EX_cpd00117_b with default bounds for boundary metabolite: cpd00117_b.\n",
      "Adding exchange reaction EX_cpd00119_b with default bounds for boundary metabolite: cpd00119_b.\n",
      "Adding exchange reaction EX_cpd00129_b with default bounds for boundary metabolite: cpd00129_b.\n",
      "Adding exchange reaction EX_cpd00107_b with default bounds for boundary metabolite: cpd00107_b.\n",
      "Adding exchange reaction EX_cpd00305_b with default bounds for boundary metabolite: cpd00305_b.\n",
      "Adding exchange reaction EX_cpd00220_b with default bounds for boundary metabolite: cpd00220_b.\n",
      "Adding exchange reaction EX_cpd00393_b with default bounds for boundary metabolite: cpd00393_b.\n",
      "Adding exchange reaction EX_cpd00033_b with default bounds for boundary metabolite: cpd00033_b.\n",
      "Adding exchange reaction EX_cpd11640_b with default bounds for boundary metabolite: cpd11640_b.\n",
      "Adding exchange reaction EX_cpd00215_b with default bounds for boundary metabolite: cpd00215_b.\n",
      "Adding exchange reaction EX_cpd11606_b with default bounds for boundary metabolite: cpd11606_b.\n",
      "Adding exchange reaction EX_cpd00161_b with default bounds for boundary metabolite: cpd00161_b.\n",
      "Adding exchange reaction EX_cpd00311_b with default bounds for boundary metabolite: cpd00311_b.\n",
      "Adding exchange reaction EX_cpd00111_b with default bounds for boundary metabolite: cpd00111_b.\n",
      "Adding exchange reaction EX_cpd15237_b with default bounds for boundary metabolite: cpd15237_b.\n",
      "Adding exchange reaction EX_cpd01080_b with default bounds for boundary metabolite: cpd01080_b.\n",
      "Adding exchange reaction EX_cpd00039_b with default bounds for boundary metabolite: cpd00039_b.\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02374_c0 \"R03314_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02374_c0 \"R03314_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05319_c0 \"TRANS_RXNBWI_115401cemaizeexpOH_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05319_c0 \"TRANS_RXNBWI_115401cemaizeexpOH_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04457_c0 \"R06605_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04457_c0 \"R06605_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02916_c0 \"R04175_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02916_c0 \"R04175_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03012_c0 \"R04336_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03012_c0 \"R04336_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10571_c0 \"Mg2_plus__importing_ATPase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10571_c0 \"Mg2_plus__importing_ATPase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04132_c0 \"R06063_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04132_c0 \"R06063_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05195_c0 \"ATP_phosphohydrolase_ferric_ion_transporting_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05195_c0 \"ATP_phosphohydrolase_ferric_ion_transporting_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05468_c0 \"TRANS_RXNAVI_26568ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05468_c0 \"TRANS_RXNAVI_26568ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04133_c0 \"R06064_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04133_c0 \"R06064_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05467_c0 \"CO2_transporter_via_diffusion_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05467_c0 \"CO2_transporter_via_diffusion_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00001_c0 \"diphosphate_phosphohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00001_c0 \"diphosphate_phosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02008_c0 \"UDP_N_acetylmuramoyl_L_alanine_D_glutamate_ligaseADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02008_c0 \"UDP_N_acetylmuramoyl_L_alanine_D_glutamate_ligaseADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10473_c0 \"chloride_Cl_1_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10473_c0 \"chloride_Cl_1_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10220_c0 \"isoheptadecanoyl_phosphatidate_cytidylyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10220_c0 \"isoheptadecanoyl_phosphatidate_cytidylyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05394_c0 \"9_methyl_3_hydroxy_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05394_c0 \"9_methyl_3_hydroxy_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05373_c0 \"10_methyl_3_hydroxy_dodecanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05373_c0 \"10_methyl_3_hydroxy_dodecanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03535_c0 \"ATP_cobIyrinic_acid_a_c_diamide_Cobeta_adenosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03535_c0 \"ATP_cobIyrinic_acid_a_c_diamide_Cobeta_adenosyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13781_c0 \"Dephospho_CoA_transporter_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13781_c0 \"Dephospho_CoA_transporter_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03514_c0 \"S_adenosyl_L_methionine_precorrin_4_C11_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03514_c0 \"S_adenosyl_L_methionine_precorrin_4_C11_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10275_c0 \"anteisoheptadecanoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10275_c0 \"anteisoheptadecanoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10474_c0 \"cobalt_Co_plus_2_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10474_c0 \"cobalt_Co_plus_2_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00461_c0 \"Phosphoenolpyruvate_UDP_N_acetyl_D_glucosamine_1_carboxyvinyl_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00461_c0 \"Phosphoenolpyruvate_UDP_N_acetyl_D_glucosamine_1_carboxyvinyl_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05679_c0 \"TRANS_RXNAVI_27988cebrachyexpTMP_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05679_c0 \"TRANS_RXNAVI_27988cebrachyexpTMP_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09448_c0 \"fatty_acid__CoA_ligase_octadecenoate_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09448_c0 \"fatty_acid__CoA_ligase_octadecenoate_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12860_c0 \"ATP_dependent_permease_AUS1_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12860_c0 \"ATP_dependent_permease_AUS1_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10197_c0 \"TECA4S_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10197_c0 \"TECA4S_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02269_c0 \"S_3_methyl_2_oxopentanoate_NAD_plus__2_oxidoreductase_CoA_methylbutanoylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02269_c0 \"S_3_methyl_2_oxopentanoate_NAD_plus__2_oxidoreductase_CoA_methylbutanoylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10337_c0 \"isoheptadecanoyl_cardiolipin_synthase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10337_c0 \"isoheptadecanoyl_cardiolipin_synthase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10282_c0 \"stearoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10282_c0 \"stearoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05672_c0 \"D_glutamate_transport_in_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05672_c0 \"D_glutamate_transport_in_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10215_c0 \"anteisoheptadecanoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10215_c0 \"anteisoheptadecanoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05486_c0 \"N_acetyl_D_mannosamine_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05486_c0 \"N_acetyl_D_mannosamine_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05382_c0 \"14_methyl_trans_hexa_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05382_c0 \"14_methyl_trans_hexa_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05381_c0 \"14_methyl_3_hydroxy_hexa_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05381_c0 \"14_methyl_3_hydroxy_hexa_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03492_c0 \"precorrin_6Y_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03492_c0 \"precorrin_6Y_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10206_c0 \"anteisoheptadecanoyl_glycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10206_c0 \"anteisoheptadecanoyl_glycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05244_c0 \"TRANS_RXNBWI_115637cemaizeexpILE_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05244_c0 \"TRANS_RXNBWI_115637cemaizeexpILE_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03408_c0 \"UDP_N_acetyl_D_glucosamine_undecaprenyl_diphospho_N_acetylmuramoyl_L_alanyl_gamma_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_4_beta_N_acetylglucosaminlytransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03408_c0 \"UDP_N_acetyl_D_glucosamine_undecaprenyl_diphospho_N_acetylmuramoyl_L_alanyl_gamma_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_4_beta_N_acetylglucosaminlytransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10254_c0 \"anteisoheptadecanoyl_Diacylglycerol_kinase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10254_c0 \"anteisoheptadecanoyl_Diacylglycerol_kinase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05377_c0 \"12_methyl_3_hydroxy_tetra_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05377_c0 \"12_methyl_3_hydroxy_tetra_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05378_c0 \"12_methyl_trans_tetra_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05378_c0 \"12_methyl_trans_tetra_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05310_c0 \"Nicotinate_transport_Plasma_membrane_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05310_c0 \"Nicotinate_transport_Plasma_membrane_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00606_c0 \"alpha_alpha_Trehalose_6_phosphate_phosphoglucohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00606_c0 \"alpha_alpha_Trehalose_6_phosphate_phosphoglucohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03852_c0 \"UDP_N_acetyl_D_mannosamine_N_acetyl_beta_D_glucosaminyldiphosphoundecaprenol_beta_1_4_N_acetylmannosaminyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03852_c0 \"UDP_N_acetyl_D_mannosamine_N_acetyl_beta_D_glucosaminyldiphosphoundecaprenol_beta_1_4_N_acetylmannosaminyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05452_c0 \"14_methyl_hexa_decanoyl_ACP_acyl_carrier_protein_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05452_c0 \"14_methyl_hexa_decanoyl_ACP_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09989_c0 \"Maltodextrin_glucosidase_dextrin_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09989_c0 \"Maltodextrin_glucosidase_dextrin_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12822_c0 \"L_glutamate_ferredoxin_oxidoreductase_transaminating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12822_c0 \"L_glutamate_ferredoxin_oxidoreductase_transaminating_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08333_c0 \"1_4_dihydroxy_2_naphthoate_octaprenyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08333_c0 \"1_4_dihydroxy_2_naphthoate_octaprenyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02290_c0 \"S_2_3_Epoxysqualene_mutase_cyclizing__lanosterol_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02290_c0 \"S_2_3_Epoxysqualene_mutase_cyclizing__lanosterol_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00910_c0 \"5_methyltetrahydrofolate_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00910_c0 \"5_methyltetrahydrofolate_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04675_c0 \"menaquinone_biosynthesis_protein_related_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04675_c0 \"menaquinone_biosynthesis_protein_related_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09926_c0 \"L_amino_acid_ligase_ala_gln_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09926_c0 \"L_amino_acid_ligase_ala_gln_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02304_c0 \"protoporphyrinogen_IX_oxygen_oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02304_c0 \"protoporphyrinogen_IX_oxygen_oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05578_c0 \"TRANS_RXNBWI_115531ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05578_c0 \"TRANS_RXNBWI_115531ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02775_c0 \"S_adenosyl_L_methionine_precorrin_4_C20_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02775_c0 \"S_adenosyl_L_methionine_precorrin_4_C20_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05307_c0 \"TRANS_RXNBWI_115637cemaizeexpSER_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05307_c0 \"TRANS_RXNBWI_115637cemaizeexpSER_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10221_c0 \"anteisoheptadecanoyl_phosphatidate_cytidylyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10221_c0 \"anteisoheptadecanoyl_phosphatidate_cytidylyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05406_c0 \"15_methyl_3_hydroxy_hexa_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05406_c0 \"15_methyl_3_hydroxy_hexa_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05301_c0 \"TRANS_RXNBWI_115637cemaizeexpTYR_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05301_c0 \"TRANS_RXNBWI_115637cemaizeexpTYR_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00262_c0 \"L_Aspartic_acid_oxygen_oxidoreductase_deaminating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00262_c0 \"L_Aspartic_acid_oxygen_oxidoreductase_deaminating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10887_c0 \"Tag_O_reaction_for_initiation_of_teichoic_acid_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10887_c0 \"Tag_O_reaction_for_initiation_of_teichoic_acid_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03164_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamyl_meso_2_6_diaminoheptanedioate_D_alanyl_D_alanine_ligaseADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03164_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamyl_meso_2_6_diaminoheptanedioate_D_alanyl_D_alanine_ligaseADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00340_c0 \"L_aspartate_ammonia_ligase_AMP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00340_c0 \"L_aspartate_ammonia_ligase_AMP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05392_c0 \"7_methyl_octanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05392_c0 \"7_methyl_octanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08822_c0 \"Lysophospholipase_L2_2_acylglycerophosphotidate__n_C18_0_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08822_c0 \"Lysophospholipase_L2_2_acylglycerophosphotidate__n_C18_0_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01213_c0 \"GPPSYN_RXN_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01213_c0 \"GPPSYN_RXN_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10260_c0 \"anteisoheptadecanoyl_CDPdiacylglycerol_sn_glycerol_3_phosphate_3_phosphatidyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10260_c0 \"anteisoheptadecanoyl_CDPdiacylglycerol_sn_glycerol_3_phosphate_3_phosphatidyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05386_c0 \"5_methyl_3_hydroxy_hexanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05386_c0 \"5_methyl_3_hydroxy_hexanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13399_c0 \"Squalene_epoxidase__endoplasmic_reticular_NAD_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13399_c0 \"Squalene_epoxidase__endoplasmic_reticular_NAD_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03513_c0 \"S_adenosyl_L_methionine_precorrin_3B_C17_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03513_c0 \"S_adenosyl_L_methionine_precorrin_3B_C17_methyltransferase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05360_c0 \"4_methyl_3_oxo_hexanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05360_c0 \"4_methyl_3_oxo_hexanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05635_c0 \"TRANS_RXNAVI_12493ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05635_c0 \"TRANS_RXNAVI_12493ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05401_c0 \"13_methyl_3_oxo_tetra_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05401_c0 \"13_methyl_3_oxo_tetra_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00126_c0 \"ATP_L_methionine_S_adenosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00126_c0 \"ATP_L_methionine_S_adenosyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12647_c0 \"TRANS_RXNAVI_26378cebrachyexpDATP_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12647_c0 \"TRANS_RXNAVI_26378cebrachyexpDATP_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05054_c0 \"adenosylcobyric_acid_R_1_aminopropan_2_ol_ligase_ADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05054_c0 \"adenosylcobyric_acid_R_1_aminopropan_2_ol_ligase_ADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03983_c0 \"AMP_sulfite_glutathione_disulfide_oxidoreductase_adenosine_5_phosphosulfate_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03983_c0 \"AMP_sulfite_glutathione_disulfide_oxidoreductase_adenosine_5_phosphosulfate_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03540_c0 \"R05225_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03540_c0 \"R05225_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09163_c0 \"Phospholipase_A2_phosphatidylglycerol__n_C18_0_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09163_c0 \"Phospholipase_A2_phosphatidylglycerol__n_C18_0_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01466_c0 \"Geranyl_diphosphate_isopentenyl_diphosphate_geranyltrans_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01466_c0 \"Geranyl_diphosphate_isopentenyl_diphosphate_geranyltrans_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05551_c0 \"ethanolamine_transport_inout_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05551_c0 \"ethanolamine_transport_inout_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02286_c0 \"UDP_N_acetylmuramate_L_alanine_ligase_ADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02286_c0 \"UDP_N_acetylmuramate_L_alanine_ligase_ADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03904_c0 \"UDP_N_acetylmuramoyl_L_alanyl_gamma_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_undecaprenyl_phosphate_phospho_N_acetylmuramoyl_pentapeptide_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03904_c0 \"UDP_N_acetylmuramoyl_L_alanyl_gamma_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_undecaprenyl_phosphate_phospho_N_acetylmuramoyl_pentapeptide_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn14120_c0 \"5_6_7_8_tetrahydrofolate_NAD_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn14120_c0 \"5_6_7_8_tetrahydrofolate_NAD_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03512_c0 \"Precorrin_8X_11_12_methylmutase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03512_c0 \"Precorrin_8X_11_12_methylmutase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10195_c0 \"Teichoic_acid_glucose_substituted_export_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10195_c0 \"Teichoic_acid_glucose_substituted_export_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05364_c0 \"6_methyl_3_oxo_octanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05364_c0 \"6_methyl_3_oxo_octanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03393_c0 \"3_octaprenyl_4_hydroxybenzoate_carboxy_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03393_c0 \"3_octaprenyl_4_hydroxybenzoate_carboxy_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05384_c0 \"3_methylbutanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05384_c0 \"3_methylbutanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02288_c0 \"Uroporphyrinogen_III_carboxy_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02288_c0 \"Uroporphyrinogen_III_carboxy_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00410_c0 \"UTP_ammonia_ligase_ADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00410_c0 \"UTP_ammonia_ligase_ADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10213_c0 \"stearoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10213_c0 \"stearoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05303_c0 \"TRANS_RXNBWI_115637cemaizeexpARG_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05303_c0 \"TRANS_RXNBWI_115637cemaizeexpARG_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05466_c0 \"TRANS_RXN_173ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05466_c0 \"TRANS_RXN_173ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn11545_c0 \"Precorrin3A_plus_OxygenPrecorrin3B_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn11545_c0 \"Precorrin3A_plus_OxygenPrecorrin3B_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01833_c0 \"aminoacetone_oxygen_oxidoreductasedeaminating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01833_c0 \"aminoacetone_oxygen_oxidoreductasedeaminating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02897_c0 \"Nicotinate_nucleotide_dimethylbenzimidazole_phospho_D_ribosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02897_c0 \"Nicotinate_nucleotide_dimethylbenzimidazole_phospho_D_ribosyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00121_c0 \"FAD_nucleotidohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00121_c0 \"FAD_nucleotidohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09240_c0 \"Sulfate_adenyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09240_c0 \"Sulfate_adenyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00189_c0 \"L_Glutamine_amidohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00189_c0 \"L_Glutamine_amidohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12568_c0 \"Ala_Gln_transport_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12568_c0 \"Ala_Gln_transport_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00252_c0 \"diphosphate_oxaloacetate_carboxy_lyase_transphosphorylatingphosphoenolpyruvate_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00252_c0 \"diphosphate_oxaloacetate_carboxy_lyase_transphosphorylatingphosphoenolpyruvate_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03150_c0 \"R04594_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03150_c0 \"R04594_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00124_c0 \"ATP_pyridoxal_5_phosphotransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00124_c0 \"ATP_pyridoxal_5_phosphotransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04464_c0 \"5_10_methylenetetrahydrofolate_NADPH_dUMP_C_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04464_c0 \"5_10_methylenetetrahydrofolate_NADPH_dUMP_C_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10604_c0 \"sulfate_transport_exchange_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10604_c0 \"sulfate_transport_exchange_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05454_c0 \"15_methyl_hexa_decanoyl_ACP_acyl_carrier_protein_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05454_c0 \"15_methyl_hexa_decanoyl_ACP_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05390_c0 \"7_methyl_3_hydroxy_octanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05390_c0 \"7_methyl_3_hydroxy_octanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05297_c0 \"TRANS_RXNBWI_115637cemaizeexpGLT_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05297_c0 \"TRANS_RXNBWI_115637cemaizeexpGLT_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05687_c0 \"Transport_of_putrescine__extracellular_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05687_c0 \"Transport_of_putrescine__extracellular_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00413_c0 \"CTP_diphosphate_lyase_cyclizing_3_5_cyclic_CMP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00413_c0 \"CTP_diphosphate_lyase_cyclizing_3_5_cyclic_CMP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05387_c0 \"5_methyl_trans_hex_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05387_c0 \"5_methyl_trans_hex_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02285_c0 \"UDP_N_acetylmuramate_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02285_c0 \"UDP_N_acetylmuramate_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05663_c0 \"TRANS_RXNBWI_115637cemaizeexpTRP_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05663_c0 \"TRANS_RXNBWI_115637cemaizeexpTRP_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10292_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10292_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10094_c0 \"S_adenosylmethione_2_demethylmenaquinone_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10094_c0 \"S_adenosylmethione_2_demethylmenaquinone_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03397_c0 \"UDP_L_rhamnose_flavonol_3_O_D_glucoside_L_rhamnosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03397_c0 \"UDP_L_rhamnose_flavonol_3_O_D_glucoside_L_rhamnosyltransferase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00688_c0 \"10_formyltetrahydrofolate_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00688_c0 \"10_formyltetrahydrofolate_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10227_c0 \"anteisoheptadecanoyl_CDPdiacylglycerol_serine_O_phosphatidyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10227_c0 \"anteisoheptadecanoyl_CDPdiacylglycerol_serine_O_phosphatidyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00086_c0 \"glutathione_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00086_c0 \"glutathione_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05397_c0 \"11_methyl_3_oxo_dodecanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05397_c0 \"11_methyl_3_oxo_dodecanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05369_c0 \"8_methyl_3_hydroxy_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05369_c0 \"8_methyl_3_hydroxy_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05367_c0 \"6_methyl_octanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05367_c0 \"6_methyl_octanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00851_c0 \"D_alanine_D_alanine_ligase_ADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00851_c0 \"D_alanine_D_alanine_ligase_ADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08809_c0 \"Lysophospholipase_L1_2_acylglycerophosphoethanolamine__n_C18_1_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08809_c0 \"Lysophospholipase_L1_2_acylglycerophosphoethanolamine__n_C18_1_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05389_c0 \"7_methyl_3_oxo_octanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05389_c0 \"7_methyl_3_oxo_octanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10284_c0 \"anteisoheptadecanoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10284_c0 \"anteisoheptadecanoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02774_c0 \"S_Adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02774_c0 \"S_Adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05440_c0 \"7_methyl_trans_oct_2_enoyl_ACP_NAD_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05440_c0 \"7_methyl_trans_oct_2_enoyl_ACP_NAD_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10193_c0 \"Polyglycerol_phosphate_alpha_glucosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10193_c0 \"Polyglycerol_phosphate_alpha_glucosyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05306_c0 \"TRANS_RXNBWI_115637cemaizeexpPHE_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05306_c0 \"TRANS_RXNBWI_115637cemaizeexpPHE_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05396_c0 \"9_methyl_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05396_c0 \"9_methyl_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10042_c0 \"F1_ATPase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10042_c0 \"F1_ATPase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05407_c0 \"15_methyl_trans_hexa_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05407_c0 \"15_methyl_trans_hexa_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05893_c0 \"Ammonia_ferredoxin_oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05893_c0 \"Ammonia_ferredoxin_oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12552_c0 \"Gly_Met_transport_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12552_c0 \"Gly_Met_transport_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10300_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10300_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05375_c0 \"10_methyl_dodecanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05375_c0 \"10_methyl_dodecanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05719_c0 \"Cytidine__3_5_Cyclic_Monophosphate_transport_in_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05719_c0 \"Cytidine__3_5_Cyclic_Monophosphate_transport_in_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10919_c0 \"protoheme_transport_irreversible_out_via_diffusion_periplasm_to_extracellular_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10919_c0 \"protoheme_transport_irreversible_out_via_diffusion_periplasm_to_extracellular_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05380_c0 \"14_methyl_3_oxo_hexa_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05380_c0 \"14_methyl_3_oxo_hexa_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10577_c0 \"Nitrogen_exchange__diffusion_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10577_c0 \"Nitrogen_exchange__diffusion_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00224_c0 \"protoheme_ferro_lyase_protoporphyrin_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00224_c0 \"protoheme_ferro_lyase_protoporphyrin_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01207_c0 \"4_methyl_2_oxopentanoate_NAD_plus__oxidoreductase_CoA_mehtylpropanoylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01207_c0 \"4_methyl_2_oxopentanoate_NAD_plus__oxidoreductase_CoA_mehtylpropanoylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10273_c0 \"stearoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10273_c0 \"stearoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00804_c0 \"L_leucine_NAD_plus__oxidoreductase_deaminating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00804_c0 \"L_leucine_NAD_plus__oxidoreductase_deaminating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03893_c0 \"all_trans_octaprenyl_diphosphate_4_hydroxybenzoate_3_octaprenyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03893_c0 \"all_trans_octaprenyl_diphosphate_4_hydroxybenzoate_3_octaprenyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10117_c0 \"Hydrogenase_Demethylmenaquinone_8__2_protons_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10117_c0 \"Hydrogenase_Demethylmenaquinone_8__2_protons_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05531_c0 \"Dextrin_transport_in_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05531_c0 \"Dextrin_transport_in_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10310_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10310_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00151_c0 \"ATP_pyruvate_phosphate_phosphotransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00151_c0 \"ATP_pyruvate_phosphate_phosphotransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05374_c0 \"10_methyl_trans_dodec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05374_c0 \"10_methyl_trans_dodec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05212_c0 \"RXN0_14ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05212_c0 \"RXN0_14ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03901_c0 \"undecaprenyl_diphosphate_phosphohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03901_c0 \"undecaprenyl_diphosphate_phosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05391_c0 \"7_methyl_trans_oct_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05391_c0 \"7_methyl_trans_oct_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05669_c0 \"TRANS_RXNBWI_115637cemaizeexpVAL_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05669_c0 \"TRANS_RXNBWI_115637cemaizeexpVAL_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn11946_c0 \"R05614_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn11946_c0 \"R05614_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00748_c0 \"glycerone_phosphate_phosphate_lyase_methylglyoxal_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00748_c0 \"glycerone_phosphate_phosphate_lyase_methylglyoxal_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09688_c0 \"deoxyguanosine_transport_in_via_proton_symport_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09688_c0 \"deoxyguanosine_transport_in_via_proton_symport_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05059_c0 \"dCTP_aminohydrolase_dUMP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05059_c0 \"dCTP_aminohydrolase_dUMP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01352_c0 \"dGTP_triphosphohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01352_c0 \"dGTP_triphosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09632_c0 \"thiamin_diphosphatase__extracellular_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09632_c0 \"thiamin_diphosphatase__extracellular_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05388_c0 \"5_methyl_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05388_c0 \"5_methyl_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09456_c0 \"R08163_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09456_c0 \"R08163_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05206_c0 \"TRANS_RXN_187ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05206_c0 \"TRANS_RXN_187ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08815_c0 \"Lysophospholipase_L1_2_acylglycerophosphoglycerol__n_C18_0_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08815_c0 \"Lysophospholipase_L1_2_acylglycerophosphoglycerol__n_C18_0_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10199_c0 \"Peptidoglycan_subunit_synthesis_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10199_c0 \"Peptidoglycan_subunit_synthesis_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13661_c0 \"S_adenosylmethione_2_demthylmenaquinone_methyltransferase_menaquinone_7_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13661_c0 \"S_adenosylmethione_2_demthylmenaquinone_methyltransferase_menaquinone_7_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09128_c0 \"Phospholipase_A1_phosphatidate__n_C18_0_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09128_c0 \"Phospholipase_A1_phosphatidate__n_C18_0_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10318_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10318_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03491_c0 \"S_Adenosyl_L_methionine_1_precorrin_6Y_C5_15_methyltransferase_C_12_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03491_c0 \"S_Adenosyl_L_methionine_1_precorrin_6Y_C5_15_methyltransferase_C_12_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08669_c0 \"Glycerophosphodiester_phosphodiesterase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08669_c0 \"Glycerophosphodiester_phosphodiesterase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05618_c0 \"manganese_transport_in_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05618_c0 \"manganese_transport_in_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00138_c0 \"deamido_NAD_plus__ammonia_ligase_AMP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00138_c0 \"deamido_NAD_plus__ammonia_ligase_AMP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05508_c0 \"TRANS_RXNBWI_115637cemaizeexpASN_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05508_c0 \"TRANS_RXNBWI_115637cemaizeexpASN_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13782_c0 \"Protein_biosynthesis_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13782_c0 \"Protein_biosynthesis_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00605_c0 \"UDP_glucose_D_glucose_6_phosphate_1_alpha_D_glucosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00605_c0 \"UDP_glucose_D_glucose_6_phosphate_1_alpha_D_glucosyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02287_c0 \"S_adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02287_c0 \"S_adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09660_c0 \"Transport_of_spermidine__extracellular_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09660_c0 \"Transport_of_spermidine__extracellular_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05359_c0 \"4_methyl_3_oxo_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05359_c0 \"4_methyl_3_oxo_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05405_c0 \"15_methyl_3_oxo_hexa_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05405_c0 \"15_methyl_3_oxo_hexa_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13918_c0 \"Zn2_plus__exporting_ATPase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13918_c0 \"Zn2_plus__exporting_ATPase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02056_c0 \"S_Adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02056_c0 \"S_Adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00367_c0 \"CTP_diphosphohydrolase_diphosphate_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00367_c0 \"CTP_diphosphohydrolase_diphosphate_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10226_c0 \"isoheptadecanoyl_CDPdiacylglycerol_serine_O_phosphatidyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10226_c0 \"isoheptadecanoyl_CDPdiacylglycerol_serine_O_phosphatidyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10832_c0 \"CoA_transporter_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10832_c0 \"CoA_transporter_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10204_c0 \"acyl_CoA_sn_glycerol_3_phosphate_1_O_acyltransferase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10204_c0 \"acyl_CoA_sn_glycerol_3_phosphate_1_O_acyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10145_c0 \"M_diaminopimelic_acid_ABC_transport_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10145_c0 \"M_diaminopimelic_acid_ABC_transport_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10196_c0 \"Teichoic_acid_D_alanine_substituted_export_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10196_c0 \"Teichoic_acid_D_alanine_substituted_export_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00830_c0 \"Isopentenyl_diphosphate_delta3_delta2_isomerase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00830_c0 \"Isopentenyl_diphosphate_delta3_delta2_isomerase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05513_c0 \"TRANS_RXN_194ce_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05513_c0 \"TRANS_RXN_194ce_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12663_c0 \"GTP_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12663_c0 \"GTP_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10319_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10319_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01574_c0 \"L_Isoleucine_oxygen_oxidoreductase_deaminating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01574_c0 \"L_Isoleucine_oxygen_oxidoreductase_deaminating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01953_c0 \"N_acetyl_D_mannosamine_N_acetylmannosaminohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01953_c0 \"N_acetyl_D_mannosamine_N_acetylmannosaminohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09686_c0 \"TRANS_RXNAVI_26378cebrachyexpTTP_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09686_c0 \"TRANS_RXNAVI_26378cebrachyexpTTP_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09156_c0 \"Phospholipase_A2_phosphatidylethanolamine__n_C18_0_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09156_c0 \"Phospholipase_A2_phosphatidylethanolamine__n_C18_0_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10191_c0 \"D_alanine_polyphosphoribitol_ligase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10191_c0 \"D_alanine_polyphosphoribitol_ligase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05402_c0 \"13_methyl_3_hydroxy_tetra_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05402_c0 \"13_methyl_3_hydroxy_tetra_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05379_c0 \"12_methyl_tetra_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05379_c0 \"12_methyl_tetra_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10205_c0 \"isoheptadecanoyl_glycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10205_c0 \"isoheptadecanoyl_glycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05887_c0 \"Hydrogen_NAD_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05887_c0 \"Hydrogen_NAD_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10274_c0 \"isoheptadecanoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10274_c0 \"isoheptadecanoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05625_c0 \"nitrite_transport_in_via_proton_symport__reversible_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05625_c0 \"nitrite_transport_in_via_proton_symport__reversible_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05371_c0 \"8_methyl_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05371_c0 \"8_methyl_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09907_c0 \"L_amino_acid_ligase_Gly_Met_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09907_c0 \"L_amino_acid_ligase_Gly_Met_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13360_c0 \"NADP_transporter_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13360_c0 \"NADP_transporter_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05362_c0 \"4_methyl_trans_hex_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05362_c0 \"4_methyl_trans_hex_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10449_c0 \"Cobinamide_transport_via_ABC_system_uptake__periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10449_c0 \"Cobinamide_transport_via_ABC_system_uptake__periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05363_c0 \"4_methyl_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05363_c0 \"4_methyl_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10301_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10301_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00295_c0 \"UDP_N_acetyl_D_glucosamine_4_epimerase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00295_c0 \"UDP_N_acetyl_D_glucosamine_4_epimerase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09345_c0 \"Undecaprenyl_diphosphate_synthase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09345_c0 \"Undecaprenyl_diphosphate_synthase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05385_c0 \"5_methyl_3_oxo_hexanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05385_c0 \"5_methyl_3_oxo_hexanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00966_c0 \"chorismate_pyruvate_lyase_4_hydroxybenzoate_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00966_c0 \"chorismate_pyruvate_lyase_4_hydroxybenzoate_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05376_c0 \"12_methyl_3_oxo_tetra_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05376_c0 \"12_methyl_3_oxo_tetra_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05029_c0 \"ATP_cobinamide_Cobeta_adenosyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05029_c0 \"ATP_cobinamide_Cobeta_adenosyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10259_c0 \"isoheptadecanoyl_CDPdiacylglycerol_sn_glycerol_3_phosphate_3_phosphatidyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10259_c0 \"isoheptadecanoyl_CDPdiacylglycerol_sn_glycerol_3_phosphate_3_phosphatidyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05372_c0 \"10_methyl_3_oxo_dodecanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05372_c0 \"10_methyl_3_oxo_dodecanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09696_c0 \"S_adenosylmethionine_permease_SAM3_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09696_c0 \"S_adenosylmethionine_permease_SAM3_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03537_c0 \"R05222_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03537_c0 \"R05222_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00292_c0 \"UDP_N_acetyl_D_glucosamine_2_epimerase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00292_c0 \"UDP_N_acetyl_D_glucosamine_2_epimerase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10476_c0 \"hydrogenobyrinic_acid_a_c_diamide_cobalt_cobalt_ligase_ADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10476_c0 \"hydrogenobyrinic_acid_a_c_diamide_cobalt_cobalt_ligase_ADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01573_c0 \"L_Isoleucine_NAD_plus__oxidoreductasedeaminating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01573_c0 \"L_Isoleucine_NAD_plus__oxidoreductasedeaminating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12630_c0 \"Transport_of_N_acetylglucosamine__extracellular_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12630_c0 \"Transport_of_N_acetylglucosamine__extracellular_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10482_c0 \"cobIIyrinic_acid_a_c_diamide_reductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10482_c0 \"cobIIyrinic_acid_a_c_diamide_reductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00191_c0 \"L_Alanine_2_oxoglutarate_aminotransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00191_c0 \"L_Alanine_2_oxoglutarate_aminotransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05494_c0 \"D_alanine_transport_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05494_c0 \"D_alanine_transport_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08299_c0 \"diacylglycerol_kinase_n_C18_0_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08299_c0 \"diacylglycerol_kinase_n_C18_0_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10311_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10311_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05403_c0 \"13_methyl_trans_tetra_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05403_c0 \"13_methyl_trans_tetra_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08040_c0 \"UDP_N_acetyl_D_glucosamine_undecaprenyl_phosphate_N_acetyl_D_glucosamine_phosphotransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08040_c0 \"UDP_N_acetyl_D_glucosamine_undecaprenyl_phosphate_N_acetyl_D_glucosamine_phosphotransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05299_c0 \"TRANS_RXNBWI_115637cemaizeexpHIS_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05299_c0 \"TRANS_RXNBWI_115637cemaizeexpHIS_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12008_c0 \"R05611_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12008_c0 \"R05611_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05383_c0 \"3_methylbutanoyl_CoA_acyl_carrier_protein_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05383_c0 \"3_methylbutanoyl_CoA_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03538_c0 \"R05223_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03538_c0 \"R05223_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13784_c0 \"RNA_transcription_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13784_c0 \"RNA_transcription_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05399_c0 \"11_methyl_trans_dodec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05399_c0 \"11_methyl_trans_dodec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05638_c0 \"TRANS_RXNBWI_115637cemaizeexpPRO_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05638_c0 \"TRANS_RXNBWI_115637cemaizeexpPRO_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10291_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10291_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10214_c0 \"isoheptadecanoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10214_c0 \"isoheptadecanoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00620_c0 \"CDPglycerol_phosphoglycerohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00620_c0 \"CDPglycerol_phosphoglycerohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09458_c0 \"stearoyl_CoA_desaturase_n_C18_0CoA___n_C18_1CoA_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09458_c0 \"stearoyl_CoA_desaturase_n_C18_0CoA___n_C18_1CoA_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05358_c0 \"2_methylbutanoyl_CoA_acyl_carrier_protein_transferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05358_c0 \"2_methylbutanoyl_CoA_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05267_c0 \"1_4_dihydroxy_2_naphthoate_octaprenyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05267_c0 \"1_4_dihydroxy_2_naphthoate_octaprenyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02155_c0 \"ATP_nicotinamide_nucleotide_adenylyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02155_c0 \"ATP_nicotinamide_nucleotide_adenylyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05400_c0 \"11_methyl_dodecanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05400_c0 \"11_methyl_dodecanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02832_c0 \"1R_6R_6_hydroxy_2_succinylcyclohexa_2_4_diene_1_carboxylate_hydrolyase_2_succinylbenzoate_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02832_c0 \"1R_6R_6_hydroxy_2_succinylcyclohexa_2_4_diene_1_carboxylate_hydrolyase_2_succinylbenzoate_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10265_c0 \"isoheptadecanoyl_Phosphatidylglycerophosphate_phosphohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10265_c0 \"isoheptadecanoyl_Phosphatidylglycerophosphate_phosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03395_c0 \"S_adenosyl_L_methionine_3_all_trans_octaprenylbenzene_1_2_diol_2_O_methyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03395_c0 \"S_adenosyl_L_methionine_3_all_trans_octaprenylbenzene_1_2_diol_2_O_methyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10192_c0 \"CDP_glycerol_polyglycerophosphate_glycerophosphotransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10192_c0 \"CDP_glycerol_polyglycerophosphate_glycerophosphotransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01258_c0 \"chorismate_hydroxymutase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01258_c0 \"chorismate_hydroxymutase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05243_c0 \"TRANS_RXNBWI_115637cemaizeexpLEU_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05243_c0 \"TRANS_RXNBWI_115637cemaizeexpLEU_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09657_c0 \"Thiamine_transporter_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09657_c0 \"Thiamine_transporter_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05393_c0 \"9_methyl_3_oxo_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05393_c0 \"9_methyl_3_oxo_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10309_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10309_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05645_c0 \"riboflavin_transport_in_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05645_c0 \"riboflavin_transport_in_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02831_c0 \"O_Succinylbenzoate_CoA_ligase_AMP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02831_c0 \"O_Succinylbenzoate_CoA_ligase_AMP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04147_c0 \"squalene_synthase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04147_c0 \"squalene_synthase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn11544_c0 \"HydrogenobyrinateHydrogenobyrinate_a_c_diamide_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn11544_c0 \"HydrogenobyrinateHydrogenobyrinate_a_c_diamide_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00391_c0 \"riboflavin_5_phosphate_phosphohydrolase_acid_optimum_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00391_c0 \"riboflavin_5_phosphate_phosphohydrolase_acid_optimum_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05255_c0 \"folate_transport_via_proton_simport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05255_c0 \"folate_transport_via_proton_simport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05582_c0 \"TRANS_RXNBWI_115637cemaizeexpGLY_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05582_c0 \"TRANS_RXNBWI_115637cemaizeexpGLY_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10542_c0 \"hydrogen_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10542_c0 \"hydrogen_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12666_c0 \"Pyridoxal_transport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12666_c0 \"Pyridoxal_transport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05398_c0 \"11_methyl_3_hydroxy_dodecanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05398_c0 \"11_methyl_3_hydroxy_dodecanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10283_c0 \"isoheptadecanoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10283_c0 \"isoheptadecanoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02667_c0 \"R_1_Aminopropan_2_ol_NAD_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02667_c0 \"R_1_Aminopropan_2_ol_NAD_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05368_c0 \"8_methyl_3_oxo_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05368_c0 \"8_methyl_3_oxo_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10954_c0 \"Fatty_acid_biosynthesis_n_C18_0_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10954_c0 \"Fatty_acid_biosynthesis_n_C18_0_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10338_c0 \"anteisoheptadecanoyl_cardiolipin_synthase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10338_c0 \"anteisoheptadecanoyl_cardiolipin_synthase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10320_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10320_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10232_c0 \"isoheptadecanoyl_phosphatidylserine_decarboxylase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10232_c0 \"isoheptadecanoyl_phosphatidylserine_decarboxylase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12664_c0 \"Menaquinone_transport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12664_c0 \"Menaquinone_transport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn13783_c0 \"DNA_replication_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn13783_c0 \"DNA_replication_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02898_c0 \"O_Succinylbenzoyl_CoA_1_4_dihydroxy_2_naphthoate_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02898_c0 \"O_Succinylbenzoyl_CoA_1_4_dihydroxy_2_naphthoate_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10194_c0 \"Teichoic_acid_unsubstituted_export_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10194_c0 \"Teichoic_acid_unsubstituted_export_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09016_c0 \"Nucleoside_triphosphate_tripolyhydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09016_c0 \"Nucleoside_triphosphate_tripolyhydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05300_c0 \"TRANS_RXNBWI_115637cemaizeexpTHR_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05300_c0 \"TRANS_RXNBWI_115637cemaizeexpTHR_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00650_c0 \"L_cysteinylglycine_dipeptidase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00650_c0 \"L_cysteinylglycine_dipeptidase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10663_c0 \"Fatty_acid_oxidation_n_C16_1_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10663_c0 \"Fatty_acid_oxidation_n_C16_1_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00692_c0 \"5_10_Methylenetetrahydrofolate_glycine_hydroxymethyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00692_c0 \"5_10_Methylenetetrahydrofolate_glycine_hydroxymethyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05204_c0 \"TRANS_RXNBWI_115572cemaizeexpGUANOSINE_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05204_c0 \"TRANS_RXNBWI_115572cemaizeexpGUANOSINE_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn06874_c0 \"reduced_ferredoxin_dinitrogen_oxidoreductase_ATP_hydrolysing_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn06874_c0 \"reduced_ferredoxin_dinitrogen_oxidoreductase_ATP_hydrolysing_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09677_c0 \"glutathione_ox_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09677_c0 \"glutathione_ox_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10336_c0 \"stearoyl_cardiolipin_synthase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10336_c0 \"stearoyl_cardiolipin_synthase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09675_c0 \"Hexadecenoate_transport_via_facilitated_irreversible_diffusion_extracellular_to_periplasm_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09675_c0 \"Hexadecenoate_transport_via_facilitated_irreversible_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10233_c0 \"anteisoheptadecanoyl_phosphatidylserine_decarboxylase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10233_c0 \"anteisoheptadecanoyl_phosphatidylserine_decarboxylase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09467_c0 \"Fatty_acid_oxidation_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09467_c0 \"Fatty_acid_oxidation_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn01073_c0 \"sn_Glycero_3_phosphoethanolamine_glycerophosphohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn01073_c0 \"sn_Glycero_3_phosphoethanolamine_glycerophosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12225_c0 \"2_octaprenylphenol_hydroxylase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12225_c0 \"2_octaprenylphenol_hydroxylase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10161_c0 \"Octadecanoate_transport_via_proton_symport_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10161_c0 \"Octadecanoate_transport_via_proton_symport_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05370_c0 \"8_methyl_trans_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05370_c0 \"8_methyl_trans_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04413_c0 \"R06558_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04413_c0 \"R06558_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn08766_c0 \"FabH_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn08766_c0 \"FabH_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00350_c0 \"glutathione_gamma_glutamylaminopeptidase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00350_c0 \"glutathione_gamma_glutamylaminopeptidase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02175_c0 \"ATP_pantetheine_4_phosphate_adenylyltransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02175_c0 \"ATP_pantetheine_4_phosphate_adenylyltransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02011_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamate_L_meso_2_6_diaminoheptanedioate_gamma_ligase_ADP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02011_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamate_L_meso_2_6_diaminoheptanedioate_gamma_ligase_ADP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00077_c0 \"ATP_NAD_plus__2_phosphotransferase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00077_c0 \"ATP_NAD_plus__2_phosphotransferase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10293_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10293_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05361_c0 \"4_methyl_3_hydroxy_hexanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05361_c0 \"4_methyl_3_hydroxy_hexanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10302_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10302_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn02303_c0 \"Coproporphyrinogen_oxygen_oxidoreductasedecarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn02303_c0 \"Coproporphyrinogen_oxygen_oxidoreductasedecarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn06022_c0 \"Acyl_carrier_protein_4_pantetheine_phosphohydrolase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn06022_c0 \"Acyl_carrier_protein_4_pantetheine_phosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05404_c0 \"13_methyl_tetra_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05404_c0 \"13_methyl_tetra_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10266_c0 \"anteisoheptadecanoyl_Phosphatidylglycerophosphate_phosphohydrolase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10266_c0 \"anteisoheptadecanoyl_Phosphatidylglycerophosphate_phosphohydrolase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn12224_c0 \"2_octaprenyl_6_methoxyphenol_hydroxylase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn12224_c0 \"2_octaprenyl_6_methoxyphenol_hydroxylase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05365_c0 \"6_methyl_3_hydroxy_octanoyl_ACP_hydro_lyase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05365_c0 \"6_methyl_3_hydroxy_octanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn10253_c0 \"isoheptadecanoyl_Diacylglycerol_kinase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn10253_c0 \"isoheptadecanoyl_Diacylglycerol_kinase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05366_c0 \"6_methyl_trans_oct_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05366_c0 \"6_methyl_trans_oct_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn04139_c0 \"2_Octaprenyl_3_methyl_6_methoxy_1_4_benzoquinone__NADPH2_oxygen_oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn04139_c0 \"2_Octaprenyl_3_methyl_6_methoxy_1_4_benzoquinone__NADPH2_oxygen_oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00175_c0 \"Acetate_CoA_ligase_AMP_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00175_c0 \"Acetate_CoA_ligase_AMP_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00301_c0 \"GTP_diphosphohydrolase_diphosphate_forming_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00301_c0 \"GTP_diphosphohydrolase_diphosphate_forming_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00612_c0 \"sn_Glycerol_3_phosphate_NADP_plus__2_oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00612_c0 \"sn_Glycerol_3_phosphate_NADP_plus__2_oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn00056_c0 \"FeII_oxygen_oxidoreductase_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn00056_c0 \"FeII_oxygen_oxidoreductase_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn03534_c0 \"R05219_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn03534_c0 \"R05219_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn05395_c0 \"9_methyl_trans_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn05395_c0 \"9_methyl_trans_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction R_rxn09674_c0 \"TRANS_RXN_58cp_c0\">\n",
      "Use of GENE ASSOCIATION or GENE_ASSOCIATION in the notes element is discouraged, use fbc:gpr instead: <Reaction R_rxn09674_c0 \"TRANS_RXN_58cp_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction bio1 \"GramPositiveBiomass_auto_biomass\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00001_e0 \"EX_H2O_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00254_e0 \"EX_Mg_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd10516_e0 \"EX_fe3_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00007_e0 \"EX_O2_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00011_e0 \"EX_CO2_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd11416_c0 \"EX_Biomass_c0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00099_e0 \"EX_Cl__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00067_e0 \"EX_H_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00655_e0 \"EX_Dephospho_CoA_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00149_e0 \"EX_Co2_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00298_e0 \"EX_dTMP_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd01188_e0 \"EX_Lanosterol_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00186_e0 \"EX_D_Glutamate_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00492_e0 \"EX_N_Acetyl_D_mannosamine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00322_e0 \"EX_L_Isoleucine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00218_e0 \"EX_Niacin_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00080_e0 \"EX_Glycerol_3_phosphate_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00054_e0 \"EX_L_Serine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00069_e0 \"EX_L_Tyrosine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00012_e0 \"EX_PPi_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00115_e0 \"EX_dATP_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00162_e0 \"EX_Aminoethanol_e0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00051_e0 \"EX_L_Arginine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00013_e0 \"EX_NH3_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd11587_e0 \"EX_Ala_Gln_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00081_e0 \"EX_Sulfite_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00023_e0 \"EX_L_Glutamate_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00118_e0 \"EX_Putrescine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00065_e0 \"EX_L_Tryptophan_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00066_e0 \"EX_L_Phenylalanine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd11591_e0 \"EX_Gly_Met_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00696_e0 \"EX_3_5_Cyclic_CMP_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00028_e0 \"EX_Heme_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00528_e0 \"EX_N2_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd11594_e0 \"EX_Dextrin_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00058_e0 \"EX_Cu2_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00156_e0 \"EX_L_Valine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00277_e0 \"EX_Deoxyguanosine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00205_e0 \"EX_K_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00030_e0 \"EX_Mn2_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00132_e0 \"EX_L_Asparagine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00264_e0 \"EX_Spermidine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00034_e0 \"EX_Zn2_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00010_e0 \"EX_CoA_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00516_e0 \"EX_meso_2_6_Diaminopimelate_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00063_e0 \"EX_Ca2_plus__e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00038_e0 \"EX_GTP_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00357_e0 \"EX_TTP_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00075_e0 \"EX_Nitrite_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00006_e0 \"EX_NADP_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd03422_e0 \"EX_Cobinamide_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00017_e0 \"EX_S_Adenosyl_L_methionine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00122_e0 \"EX_N_Acetyl_D_glucosamine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00117_e0 \"EX_D_Alanine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00119_e0 \"EX_L_Histidine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00129_e0 \"EX_L_Proline_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00107_e0 \"EX_L_Leucine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00305_e0 \"EX_Thiamin_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00220_e0 \"EX_Riboflavin_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00393_e0 \"EX_Folate_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00033_e0 \"EX_Glycine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd11640_e0 \"EX_H2_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00215_e0 \"EX_Pyridoxal_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd11606_e0 \"EX_Menaquinone_7_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00161_e0 \"EX_L_Threonine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00311_e0 \"EX_Guanosine_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00111_e0 \"EX_Oxidized_glutathione_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd15237_e0 \"EX_hexadecenoate_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd01080_e0 \"EX_ocdca_e0\">\n",
      "Encoding LOWER_BOUND and UPPER_BOUND in KineticLaw is discouraged, use fbc:fluxBounds instead: <Reaction EX_cpd00039_e0 \"EX_L_Lysine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02374_c0 \"R03314_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05319_c0 \"TRANS_RXNBWI_115401cemaizeexpOH_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04457_c0 \"R06605_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02916_c0 \"R04175_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03012_c0 \"R04336_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10571_c0 \"Mg2_plus__importing_ATPase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04132_c0 \"R06063_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05195_c0 \"ATP_phosphohydrolase_ferric_ion_transporting_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05468_c0 \"TRANS_RXNAVI_26568ce_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04133_c0 \"R06064_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05467_c0 \"CO2_transporter_via_diffusion_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00001_c0 \"diphosphate_phosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02008_c0 \"UDP_N_acetylmuramoyl_L_alanine_D_glutamate_ligaseADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10473_c0 \"chloride_Cl_1_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10220_c0 \"isoheptadecanoyl_phosphatidate_cytidylyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05394_c0 \"9_methyl_3_hydroxy_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05373_c0 \"10_methyl_3_hydroxy_dodecanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03535_c0 \"ATP_cobIyrinic_acid_a_c_diamide_Cobeta_adenosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13781_c0 \"Dephospho_CoA_transporter_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03514_c0 \"S_adenosyl_L_methionine_precorrin_4_C11_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10275_c0 \"anteisoheptadecanoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10474_c0 \"cobalt_Co_plus_2_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00461_c0 \"Phosphoenolpyruvate_UDP_N_acetyl_D_glucosamine_1_carboxyvinyl_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05679_c0 \"TRANS_RXNAVI_27988cebrachyexpTMP_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09448_c0 \"fatty_acid__CoA_ligase_octadecenoate_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12860_c0 \"ATP_dependent_permease_AUS1_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10197_c0 \"TECA4S_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02269_c0 \"S_3_methyl_2_oxopentanoate_NAD_plus__2_oxidoreductase_CoA_methylbutanoylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10337_c0 \"isoheptadecanoyl_cardiolipin_synthase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10282_c0 \"stearoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05672_c0 \"D_glutamate_transport_in_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10215_c0 \"anteisoheptadecanoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05486_c0 \"N_acetyl_D_mannosamine_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05382_c0 \"14_methyl_trans_hexa_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05381_c0 \"14_methyl_3_hydroxy_hexa_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03492_c0 \"precorrin_6Y_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10206_c0 \"anteisoheptadecanoyl_glycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05244_c0 \"TRANS_RXNBWI_115637cemaizeexpILE_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03408_c0 \"UDP_N_acetyl_D_glucosamine_undecaprenyl_diphospho_N_acetylmuramoyl_L_alanyl_gamma_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_4_beta_N_acetylglucosaminlytransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10254_c0 \"anteisoheptadecanoyl_Diacylglycerol_kinase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05377_c0 \"12_methyl_3_hydroxy_tetra_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05378_c0 \"12_methyl_trans_tetra_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05310_c0 \"Nicotinate_transport_Plasma_membrane_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00606_c0 \"alpha_alpha_Trehalose_6_phosphate_phosphoglucohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03852_c0 \"UDP_N_acetyl_D_mannosamine_N_acetyl_beta_D_glucosaminyldiphosphoundecaprenol_beta_1_4_N_acetylmannosaminyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05452_c0 \"14_methyl_hexa_decanoyl_ACP_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09989_c0 \"Maltodextrin_glucosidase_dextrin_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12822_c0 \"L_glutamate_ferredoxin_oxidoreductase_transaminating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08333_c0 \"1_4_dihydroxy_2_naphthoate_octaprenyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02290_c0 \"S_2_3_Epoxysqualene_mutase_cyclizing__lanosterol_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00910_c0 \"5_methyltetrahydrofolate_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04675_c0 \"menaquinone_biosynthesis_protein_related_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09926_c0 \"L_amino_acid_ligase_ala_gln_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02304_c0 \"protoporphyrinogen_IX_oxygen_oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05578_c0 \"TRANS_RXNBWI_115531ce_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02775_c0 \"S_adenosyl_L_methionine_precorrin_4_C20_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05307_c0 \"TRANS_RXNBWI_115637cemaizeexpSER_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10221_c0 \"anteisoheptadecanoyl_phosphatidate_cytidylyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05406_c0 \"15_methyl_3_hydroxy_hexa_decanoyl_ACP_hydro_lyase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05301_c0 \"TRANS_RXNBWI_115637cemaizeexpTYR_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00262_c0 \"L_Aspartic_acid_oxygen_oxidoreductase_deaminating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10887_c0 \"Tag_O_reaction_for_initiation_of_teichoic_acid_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03164_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamyl_meso_2_6_diaminoheptanedioate_D_alanyl_D_alanine_ligaseADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00340_c0 \"L_aspartate_ammonia_ligase_AMP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05392_c0 \"7_methyl_octanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08822_c0 \"Lysophospholipase_L2_2_acylglycerophosphotidate__n_C18_0_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01213_c0 \"GPPSYN_RXN_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10260_c0 \"anteisoheptadecanoyl_CDPdiacylglycerol_sn_glycerol_3_phosphate_3_phosphatidyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05386_c0 \"5_methyl_3_hydroxy_hexanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13399_c0 \"Squalene_epoxidase__endoplasmic_reticular_NAD_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03513_c0 \"S_adenosyl_L_methionine_precorrin_3B_C17_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05360_c0 \"4_methyl_3_oxo_hexanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05635_c0 \"TRANS_RXNAVI_12493ce_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05401_c0 \"13_methyl_3_oxo_tetra_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00126_c0 \"ATP_L_methionine_S_adenosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12647_c0 \"TRANS_RXNAVI_26378cebrachyexpDATP_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05054_c0 \"adenosylcobyric_acid_R_1_aminopropan_2_ol_ligase_ADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03983_c0 \"AMP_sulfite_glutathione_disulfide_oxidoreductase_adenosine_5_phosphosulfate_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03540_c0 \"R05225_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09163_c0 \"Phospholipase_A2_phosphatidylglycerol__n_C18_0_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01466_c0 \"Geranyl_diphosphate_isopentenyl_diphosphate_geranyltrans_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05551_c0 \"ethanolamine_transport_inout_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02286_c0 \"UDP_N_acetylmuramate_L_alanine_ligase_ADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03904_c0 \"UDP_N_acetylmuramoyl_L_alanyl_gamma_D_glutamyl_meso_2_6_diaminopimeloyl_D_alanyl_D_alanine_undecaprenyl_phosphate_phospho_N_acetylmuramoyl_pentapeptide_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn14120_c0 \"5_6_7_8_tetrahydrofolate_NAD_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03512_c0 \"Precorrin_8X_11_12_methylmutase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10195_c0 \"Teichoic_acid_glucose_substituted_export_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05364_c0 \"6_methyl_3_oxo_octanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03393_c0 \"3_octaprenyl_4_hydroxybenzoate_carboxy_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05384_c0 \"3_methylbutanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02288_c0 \"Uroporphyrinogen_III_carboxy_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00410_c0 \"UTP_ammonia_ligase_ADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10213_c0 \"stearoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05303_c0 \"TRANS_RXNBWI_115637cemaizeexpARG_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05466_c0 \"TRANS_RXN_173ce_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn11545_c0 \"Precorrin3A_plus_OxygenPrecorrin3B_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01833_c0 \"aminoacetone_oxygen_oxidoreductasedeaminating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02897_c0 \"Nicotinate_nucleotide_dimethylbenzimidazole_phospho_D_ribosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00121_c0 \"FAD_nucleotidohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09240_c0 \"Sulfate_adenyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00189_c0 \"L_Glutamine_amidohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12568_c0 \"Ala_Gln_transport_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00252_c0 \"diphosphate_oxaloacetate_carboxy_lyase_transphosphorylatingphosphoenolpyruvate_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03150_c0 \"R04594_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00124_c0 \"ATP_pyridoxal_5_phosphotransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04464_c0 \"5_10_methylenetetrahydrofolate_NADPH_dUMP_C_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10604_c0 \"sulfate_transport_exchange_via_proton_symport_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05454_c0 \"15_methyl_hexa_decanoyl_ACP_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05390_c0 \"7_methyl_3_hydroxy_octanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05297_c0 \"TRANS_RXNBWI_115637cemaizeexpGLT_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05687_c0 \"Transport_of_putrescine__extracellular_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00413_c0 \"CTP_diphosphate_lyase_cyclizing_3_5_cyclic_CMP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05387_c0 \"5_methyl_trans_hex_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02285_c0 \"UDP_N_acetylmuramate_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05663_c0 \"TRANS_RXNBWI_115637cemaizeexpTRP_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10292_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10094_c0 \"S_adenosylmethione_2_demethylmenaquinone_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03397_c0 \"UDP_L_rhamnose_flavonol_3_O_D_glucoside_L_rhamnosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00688_c0 \"10_formyltetrahydrofolate_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10227_c0 \"anteisoheptadecanoyl_CDPdiacylglycerol_serine_O_phosphatidyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00086_c0 \"glutathione_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05397_c0 \"11_methyl_3_oxo_dodecanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05369_c0 \"8_methyl_3_hydroxy_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05367_c0 \"6_methyl_octanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00851_c0 \"D_alanine_D_alanine_ligase_ADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08809_c0 \"Lysophospholipase_L1_2_acylglycerophosphoethanolamine__n_C18_1_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05389_c0 \"7_methyl_3_oxo_octanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10284_c0 \"anteisoheptadecanoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02774_c0 \"S_Adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05440_c0 \"7_methyl_trans_oct_2_enoyl_ACP_NAD_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10193_c0 \"Polyglycerol_phosphate_alpha_glucosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05306_c0 \"TRANS_RXNBWI_115637cemaizeexpPHE_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05396_c0 \"9_methyl_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10042_c0 \"F1_ATPase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05407_c0 \"15_methyl_trans_hexa_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05893_c0 \"Ammonia_ferredoxin_oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12552_c0 \"Gly_Met_transport_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10300_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05375_c0 \"10_methyl_dodecanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05719_c0 \"Cytidine__3_5_Cyclic_Monophosphate_transport_in_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10919_c0 \"protoheme_transport_irreversible_out_via_diffusion_periplasm_to_extracellular_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05380_c0 \"14_methyl_3_oxo_hexa_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10577_c0 \"Nitrogen_exchange__diffusion_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00224_c0 \"protoheme_ferro_lyase_protoporphyrin_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01207_c0 \"4_methyl_2_oxopentanoate_NAD_plus__oxidoreductase_CoA_mehtylpropanoylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10273_c0 \"stearoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00804_c0 \"L_leucine_NAD_plus__oxidoreductase_deaminating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03893_c0 \"all_trans_octaprenyl_diphosphate_4_hydroxybenzoate_3_octaprenyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10117_c0 \"Hydrogenase_Demethylmenaquinone_8__2_protons_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05531_c0 \"Dextrin_transport_in_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10310_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00151_c0 \"ATP_pyruvate_phosphate_phosphotransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05374_c0 \"10_methyl_trans_dodec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05212_c0 \"RXN0_14ce_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03901_c0 \"undecaprenyl_diphosphate_phosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05391_c0 \"7_methyl_trans_oct_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05669_c0 \"TRANS_RXNBWI_115637cemaizeexpVAL_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn11946_c0 \"R05614_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00748_c0 \"glycerone_phosphate_phosphate_lyase_methylglyoxal_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09688_c0 \"deoxyguanosine_transport_in_via_proton_symport_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05059_c0 \"dCTP_aminohydrolase_dUMP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01352_c0 \"dGTP_triphosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09632_c0 \"thiamin_diphosphatase__extracellular_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05388_c0 \"5_methyl_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09456_c0 \"R08163_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05206_c0 \"TRANS_RXN_187ce_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08815_c0 \"Lysophospholipase_L1_2_acylglycerophosphoglycerol__n_C18_0_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10199_c0 \"Peptidoglycan_subunit_synthesis_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13661_c0 \"S_adenosylmethione_2_demthylmenaquinone_methyltransferase_menaquinone_7_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09128_c0 \"Phospholipase_A1_phosphatidate__n_C18_0_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10318_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03491_c0 \"S_Adenosyl_L_methionine_1_precorrin_6Y_C5_15_methyltransferase_C_12_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08669_c0 \"Glycerophosphodiester_phosphodiesterase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05618_c0 \"manganese_transport_in_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00138_c0 \"deamido_NAD_plus__ammonia_ligase_AMP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05508_c0 \"TRANS_RXNBWI_115637cemaizeexpASN_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13782_c0 \"Protein_biosynthesis_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00605_c0 \"UDP_glucose_D_glucose_6_phosphate_1_alpha_D_glucosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02287_c0 \"S_adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09660_c0 \"Transport_of_spermidine__extracellular_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05359_c0 \"4_methyl_3_oxo_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05405_c0 \"15_methyl_3_oxo_hexa_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13918_c0 \"Zn2_plus__exporting_ATPase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02056_c0 \"S_Adenosyl_L_methionine_uroporphyrin_III_C_methyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00367_c0 \"CTP_diphosphohydrolase_diphosphate_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10226_c0 \"isoheptadecanoyl_CDPdiacylglycerol_serine_O_phosphatidyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10832_c0 \"CoA_transporter_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10204_c0 \"acyl_CoA_sn_glycerol_3_phosphate_1_O_acyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10145_c0 \"M_diaminopimelic_acid_ABC_transport_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10196_c0 \"Teichoic_acid_D_alanine_substituted_export_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00830_c0 \"Isopentenyl_diphosphate_delta3_delta2_isomerase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05513_c0 \"TRANS_RXN_194ce_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12663_c0 \"GTP_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10319_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01574_c0 \"L_Isoleucine_oxygen_oxidoreductase_deaminating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01953_c0 \"N_acetyl_D_mannosamine_N_acetylmannosaminohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09686_c0 \"TRANS_RXNAVI_26378cebrachyexpTTP_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09156_c0 \"Phospholipase_A2_phosphatidylethanolamine__n_C18_0_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10191_c0 \"D_alanine_polyphosphoribitol_ligase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05402_c0 \"13_methyl_3_hydroxy_tetra_decanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05379_c0 \"12_methyl_tetra_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10205_c0 \"isoheptadecanoyl_glycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05887_c0 \"Hydrogen_NAD_plus__oxidoreductase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10274_c0 \"isoheptadecanoyl_UDP_glucosyltransferase_diglucosyl_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05625_c0 \"nitrite_transport_in_via_proton_symport__reversible_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05371_c0 \"8_methyl_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09907_c0 \"L_amino_acid_ligase_Gly_Met_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13360_c0 \"NADP_transporter_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05362_c0 \"4_methyl_trans_hex_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10449_c0 \"Cobinamide_transport_via_ABC_system_uptake__periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05363_c0 \"4_methyl_hexanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10301_c0 \"isoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00295_c0 \"UDP_N_acetyl_D_glucosamine_4_epimerase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09345_c0 \"Undecaprenyl_diphosphate_synthase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05385_c0 \"5_methyl_3_oxo_hexanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00966_c0 \"chorismate_pyruvate_lyase_4_hydroxybenzoate_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05376_c0 \"12_methyl_3_oxo_tetra_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05029_c0 \"ATP_cobinamide_Cobeta_adenosyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10259_c0 \"isoheptadecanoyl_CDPdiacylglycerol_sn_glycerol_3_phosphate_3_phosphatidyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05372_c0 \"10_methyl_3_oxo_dodecanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09696_c0 \"S_adenosylmethionine_permease_SAM3_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03537_c0 \"R05222_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00292_c0 \"UDP_N_acetyl_D_glucosamine_2_epimerase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10476_c0 \"hydrogenobyrinic_acid_a_c_diamide_cobalt_cobalt_ligase_ADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01573_c0 \"L_Isoleucine_NAD_plus__oxidoreductasedeaminating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12630_c0 \"Transport_of_N_acetylglucosamine__extracellular_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10482_c0 \"cobIIyrinic_acid_a_c_diamide_reductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00191_c0 \"L_Alanine_2_oxoglutarate_aminotransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05494_c0 \"D_alanine_transport_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08299_c0 \"diacylglycerol_kinase_n_C18_0_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10311_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05403_c0 \"13_methyl_trans_tetra_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08040_c0 \"UDP_N_acetyl_D_glucosamine_undecaprenyl_phosphate_N_acetyl_D_glucosamine_phosphotransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05299_c0 \"TRANS_RXNBWI_115637cemaizeexpHIS_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12008_c0 \"R05611_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05383_c0 \"3_methylbutanoyl_CoA_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03538_c0 \"R05223_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13784_c0 \"RNA_transcription_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05399_c0 \"11_methyl_trans_dodec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05638_c0 \"TRANS_RXNBWI_115637cemaizeexpPRO_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10291_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10214_c0 \"isoheptadecanoyl_1_acylglycerol_3_phosphate_O_acyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00620_c0 \"CDPglycerol_phosphoglycerohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09458_c0 \"stearoyl_CoA_desaturase_n_C18_0CoA___n_C18_1CoA_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05358_c0 \"2_methylbutanoyl_CoA_acyl_carrier_protein_transferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05267_c0 \"1_4_dihydroxy_2_naphthoate_octaprenyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02155_c0 \"ATP_nicotinamide_nucleotide_adenylyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05400_c0 \"11_methyl_dodecanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02832_c0 \"1R_6R_6_hydroxy_2_succinylcyclohexa_2_4_diene_1_carboxylate_hydrolyase_2_succinylbenzoate_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10265_c0 \"isoheptadecanoyl_Phosphatidylglycerophosphate_phosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03395_c0 \"S_adenosyl_L_methionine_3_all_trans_octaprenylbenzene_1_2_diol_2_O_methyltransferase_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10192_c0 \"CDP_glycerol_polyglycerophosphate_glycerophosphotransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01258_c0 \"chorismate_hydroxymutase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05243_c0 \"TRANS_RXNBWI_115637cemaizeexpLEU_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09657_c0 \"Thiamine_transporter_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05393_c0 \"9_methyl_3_oxo_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10309_c0 \"stearoyl_lipoteichoic_acid_synthesis_n24__linked__N_acetylglucosamine_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05645_c0 \"riboflavin_transport_in_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02831_c0 \"O_Succinylbenzoate_CoA_ligase_AMP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04147_c0 \"squalene_synthase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn11544_c0 \"HydrogenobyrinateHydrogenobyrinate_a_c_diamide_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00391_c0 \"riboflavin_5_phosphate_phosphohydrolase_acid_optimum_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05255_c0 \"folate_transport_via_proton_simport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05582_c0 \"TRANS_RXNBWI_115637cemaizeexpGLY_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10542_c0 \"hydrogen_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12666_c0 \"Pyridoxal_transport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05398_c0 \"11_methyl_3_hydroxy_dodecanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10283_c0 \"isoheptadecanoyl_UDP_glucosyltransferase_monoglucosyl_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02667_c0 \"R_1_Aminopropan_2_ol_NAD_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05368_c0 \"8_methyl_3_oxo_decanoyl_ACP_NADP_plus__oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10954_c0 \"Fatty_acid_biosynthesis_n_C18_0_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10338_c0 \"anteisoheptadecanoyl_cardiolipin_synthase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10320_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__unlinked__D_alanine_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10232_c0 \"isoheptadecanoyl_phosphatidylserine_decarboxylase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12664_c0 \"Menaquinone_transport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn13783_c0 \"DNA_replication_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02898_c0 \"O_Succinylbenzoyl_CoA_1_4_dihydroxy_2_naphthoate_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10194_c0 \"Teichoic_acid_unsubstituted_export_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09016_c0 \"Nucleoside_triphosphate_tripolyhydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05300_c0 \"TRANS_RXNBWI_115637cemaizeexpTHR_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00650_c0 \"L_cysteinylglycine_dipeptidase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10663_c0 \"Fatty_acid_oxidation_n_C16_1_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00692_c0 \"5_10_Methylenetetrahydrofolate_glycine_hydroxymethyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05204_c0 \"TRANS_RXNBWI_115572cemaizeexpGUANOSINE_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn06874_c0 \"reduced_ferredoxin_dinitrogen_oxidoreductase_ATP_hydrolysing_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09677_c0 \"glutathione_ox_transport_via_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10336_c0 \"stearoyl_cardiolipin_synthase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09675_c0 \"Hexadecenoate_transport_via_facilitated_irreversible_diffusion_extracellular_to_periplasm_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10233_c0 \"anteisoheptadecanoyl_phosphatidylserine_decarboxylase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09467_c0 \"Fatty_acid_oxidation_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn01073_c0 \"sn_Glycero_3_phosphoethanolamine_glycerophosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12225_c0 \"2_octaprenylphenol_hydroxylase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10161_c0 \"Octadecanoate_transport_via_proton_symport_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05370_c0 \"8_methyl_trans_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04413_c0 \"R06558_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn08766_c0 \"FabH_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00350_c0 \"glutathione_gamma_glutamylaminopeptidase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02175_c0 \"ATP_pantetheine_4_phosphate_adenylyltransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02011_c0 \"UDP_N_acetylmuramoyl_L_alanyl_D_glutamate_L_meso_2_6_diaminoheptanedioate_gamma_ligase_ADP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00077_c0 \"ATP_NAD_plus__2_phosphotransferase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10293_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__unsubstituted_c0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05361_c0 \"4_methyl_3_hydroxy_hexanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10302_c0 \"anteisoheptadecanoyl_lipoteichoic_acid_synthesis_n24__linked__glucose_substituted_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn02303_c0 \"Coproporphyrinogen_oxygen_oxidoreductasedecarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn06022_c0 \"Acyl_carrier_protein_4_pantetheine_phosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05404_c0 \"13_methyl_tetra_decanoyl_ACP_malonyl_acyl_carrier_protein_C_acyltransferase_decarboxylating_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10266_c0 \"anteisoheptadecanoyl_Phosphatidylglycerophosphate_phosphohydrolase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn12224_c0 \"2_octaprenyl_6_methoxyphenol_hydroxylase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05365_c0 \"6_methyl_3_hydroxy_octanoyl_ACP_hydro_lyase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn10253_c0 \"isoheptadecanoyl_Diacylglycerol_kinase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05366_c0 \"6_methyl_trans_oct_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn04139_c0 \"2_Octaprenyl_3_methyl_6_methoxy_1_4_benzoquinone__NADPH2_oxygen_oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00175_c0 \"Acetate_CoA_ligase_AMP_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00301_c0 \"GTP_diphosphohydrolase_diphosphate_forming_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00612_c0 \"sn_Glycerol_3_phosphate_NADP_plus__2_oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn00056_c0 \"FeII_oxygen_oxidoreductase_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn03534_c0 \"R05219_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn05395_c0 \"9_methyl_trans_dec_2_enoyl_ACP_NADP_plus__oxidoreductase_A_specific_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction R_rxn09674_c0 \"TRANS_RXN_58cp_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction bio1 \"GramPositiveBiomass_auto_biomass\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00001_e0 \"EX_H2O_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00254_e0 \"EX_Mg_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd10516_e0 \"EX_fe3_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00007_e0 \"EX_O2_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00011_e0 \"EX_CO2_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd11416_c0 \"EX_Biomass_c0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00099_e0 \"EX_Cl__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00067_e0 \"EX_H_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00655_e0 \"EX_Dephospho_CoA_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00149_e0 \"EX_Co2_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00298_e0 \"EX_dTMP_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd01188_e0 \"EX_Lanosterol_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00186_e0 \"EX_D_Glutamate_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00492_e0 \"EX_N_Acetyl_D_mannosamine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00322_e0 \"EX_L_Isoleucine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00218_e0 \"EX_Niacin_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00080_e0 \"EX_Glycerol_3_phosphate_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00054_e0 \"EX_L_Serine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00069_e0 \"EX_L_Tyrosine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00012_e0 \"EX_PPi_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00115_e0 \"EX_dATP_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00162_e0 \"EX_Aminoethanol_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00051_e0 \"EX_L_Arginine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00013_e0 \"EX_NH3_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd11587_e0 \"EX_Ala_Gln_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00081_e0 \"EX_Sulfite_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00023_e0 \"EX_L_Glutamate_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00118_e0 \"EX_Putrescine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00065_e0 \"EX_L_Tryptophan_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00066_e0 \"EX_L_Phenylalanine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd11591_e0 \"EX_Gly_Met_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00696_e0 \"EX_3_5_Cyclic_CMP_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00028_e0 \"EX_Heme_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00528_e0 \"EX_N2_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd11594_e0 \"EX_Dextrin_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00058_e0 \"EX_Cu2_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00156_e0 \"EX_L_Valine_e0\">\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00277_e0 \"EX_Deoxyguanosine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00205_e0 \"EX_K_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00030_e0 \"EX_Mn2_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00132_e0 \"EX_L_Asparagine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00264_e0 \"EX_Spermidine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00034_e0 \"EX_Zn2_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00010_e0 \"EX_CoA_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00516_e0 \"EX_meso_2_6_Diaminopimelate_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00063_e0 \"EX_Ca2_plus__e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00038_e0 \"EX_GTP_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00357_e0 \"EX_TTP_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00075_e0 \"EX_Nitrite_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00006_e0 \"EX_NADP_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd03422_e0 \"EX_Cobinamide_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00017_e0 \"EX_S_Adenosyl_L_methionine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00122_e0 \"EX_N_Acetyl_D_glucosamine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00117_e0 \"EX_D_Alanine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00119_e0 \"EX_L_Histidine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00129_e0 \"EX_L_Proline_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00107_e0 \"EX_L_Leucine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00305_e0 \"EX_Thiamin_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00220_e0 \"EX_Riboflavin_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00393_e0 \"EX_Folate_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00033_e0 \"EX_Glycine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd11640_e0 \"EX_H2_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00215_e0 \"EX_Pyridoxal_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd11606_e0 \"EX_Menaquinone_7_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00161_e0 \"EX_L_Threonine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00311_e0 \"EX_Guanosine_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00111_e0 \"EX_Oxidized_glutathione_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd15237_e0 \"EX_hexadecenoate_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd01080_e0 \"EX_ocdca_e0\">\n",
      "Encoding OBJECTIVE_COEFFICIENT in KineticLaw is discouraged, use fbc:fluxObjective instead: <Reaction EX_cpd00039_e0 \"EX_L_Lysine_e0\">\n"
     ]
    }
   ],
   "source": [
    "# PATRIC drafts\n",
    "cd630_PATRIC = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/cd630_PATRIC.sbml')\n",
    "cdR20291_PATRIC = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/CdR20291.draft.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# AGORA models\n",
    "agora_NAP07 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/AGORA/Clostridium_difficile_NAP07.xml')\n",
    "agora_NAP08 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/AGORA/Clostridium_difficile_NAP08.xml')\n",
    "agora_CD196 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/AGORA/Clostridium_difficile_CD196.xml')\n",
    "agora_R20291 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/old_cdifficile_genres/AGORA/Clostridium_difficile_R20291.xml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>MODEL1409240004</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff140051550</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1756</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1337</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>54</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*Biomass - 1.0*Biomass_reverse_57a34</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytosol, Boundary, Extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model MODEL1409240004 at 0x7ff140051550>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iMLTC806cdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>Biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Biomass Reaction</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f513e331350</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.24 CW_c + 0.03 DNA_c + 0.07 Lip_c + 0.4 Prot_c + 0.12 RNA_c + 0.14 SPs_c + 40.0 atp_c + 40.0 h2o_c <=> 40.0 adp_c + biomass_c + 40.0 pi_c</p>\n",
       "                    <p style='text-align:right'>0.24 Cell Wall + 0.03 deoxyribonucleic acid + 0.07 lipid + 0.4 protein + 0.12 ribonucleic acid + 0.14 Solute Pools + 40.0 ATP + 40.0 H2O <=> 40.0 ADP + Biomass + 40.0 phosphate</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction Biomass at 0x7f513e331350>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iMLTC806cdf.reactions.Biomass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 118\n",
      "Transport reactions: 318\n",
      "Metabolic reactions: 901 \n",
      "\n",
      "C_c metabolites: 1521\n",
      "C_e metabolites: 117\n",
      "C_b metabolites: 118\n"
     ]
    }
   ],
   "source": [
    "count_components(iMLTC806cdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 0.761 (overdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has no gene data\n",
      "1000.0 objective flux in current media\n",
      "14.36% reactions are blocked\n",
      "0.4% metabolites are generated for free\n",
      "27.22% metabolites are lost improperly\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C23H38N7O17P3S(CH2)2n'\n",
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C16H16O2(C5H8)n'\n",
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C16H18O2(C5H8)n'\n",
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C15H14O2(C5H8)n'\n",
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C2H4NO2R(C2H2NOR)n'\n",
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C10H18O13P2R2(C5H8O6PR)n'\n",
      "cobra/core/metabolite.py:89 \u001b[1;31mUserWarning\u001b[0m: invalid formula (has parenthesis) in 'C10H17O8PR2(C5H8O5PR)n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.54% reactions are mass imbalanced\n",
      "100.0% reactions without GPRs\n",
      "100.0% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iMLTC806cdf_rmGenes = basicCheck(iMLTC806cdf)\n",
    "iMLTC806cdf_noGPRblocked = blockedReactions(iMLTC806cdf)\n",
    "iMLTC806cdf_free = checkFreeMass(iMLTC806cdf, cytosol='C_c')\n",
    "iMLTC806cdf_lost = checkLostMass(iMLTC806cdf)\n",
    "iMLTC806cdf_massImbal = checkBalance(iMLTC806cdf)\n",
    "iMLTC806cdf_nogene = missingGPR(iMLTC806cdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>icdf834_ex</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff0edd55cd0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>807</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1168</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*Biomass - 1.0*Biomass_reverse_57a34</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytosol, Extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model icdf834_ex at 0x7ff0edd55cd0>"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "icdf834"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>Biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Biomass Reaction</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f513dc22e50</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.24 CW[c] + 0.03 DNA[c] + 0.07 Lip[c] + 0.4 Prot[c] + 0.12 RNA[c] + 0.14 SPs[c] + 40.0 atp[c] + 40.0 h2o[c] --> 40.0 adp[c] + biomass[c] + 40.0 pi[c]</p>\n",
       "                    <p style='text-align:right'>0.24 Cell Wall + 0.03 deoxyribonucleic acid + 0.07 lipid + 0.4 protein + 0.12 ribonucleic acid + 0.14 Solute Pools + 40.0 ATP + 40.0 H2O --> 40.0 ADP + Biomass + 40.0 phosphate</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction Biomass at 0x7f513dc22e50>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "icdf834.reactions.Biomass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 123\n",
      "Transport reactions: 149\n",
      "Metabolic reactions: 896 \n",
      "\n",
      "Extracellular metabolites: 125\n",
      "Cytosol metabolites: 682\n"
     ]
    }
   ],
   "source": [
    "count_components(icdf834)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.447 (underdetermined)\n",
      "GENRE has 2 compartment(s)\n",
      "GENRE has 832 genes\n",
      "146.523 objective flux in current media\n",
      "15.33% reactions are blocked\n",
      "0.25% metabolites are generated for free\n",
      "42.5% metabolites are lost improperly\n",
      "12.33% reactions are mass imbalanced\n",
      "24.78% reactions without GPRs\n",
      "28.15% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "icdf834_rmGenes = basicCheck(icdf834)\n",
    "icdf834_noGPRblocked = blockedReactions(icdf834)\n",
    "icdf834_free = checkFreeMass(icdf834, cytosol='Cytosol')\n",
    "icdf834_lost = checkLostMass(icdf834)\n",
    "icdf834_massImbal = checkBalance(icdf834)\n",
    "icdf834_nogene = missingGPR(icdf834)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iHD992</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff0edd55410</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>935</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1125</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*Biomass - 1.0*Biomass_reverse_57a34</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytosol</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iHD992 at 0x7ff0edd55410>"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iHD992"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>Biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f513da19350</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>45.0 ATP + 0.043 DNA + 0.005 Glycogen + 45.0 H2O + 0.022 Ion + 0.024 L_CWG + 0.06 Lipid + 0.005 Metabolites + 0.097 P_CWG + 0.145 Peptidoglycan + 0.508 Protein + 0.091 RNA <=> 45.0 ADP + 45.0 H_ + ...</p>\n",
       "                    <p style='text-align:right'>45.0 ATP + 0.043 DNA + 0.005 Glycogen + 45.0 H2O + 0.022 Ion + 0.024 L-CWG + 0.06 Lipid + 0.005 Metabolites + 0.097 P-CWG + 0.145 Peptidoglycan + 0.508 Protein + 0.091 RNA <=> 45.0 ADP + 45.0 H+ + ...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction Biomass at 0x7f513da19350>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iHD992.reactions.Biomass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 161\n",
      "Transport reactions: 0\n",
      "Metabolic reactions: 964 \n",
      "\n",
      "Cytosol metabolites: 935\n"
     ]
    }
   ],
   "source": [
    "count_components(iHD992)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.203 (underdetermined)\n",
      "GENRE has 1 compartment(s)\n",
      "GENRE has no gene data\n",
      "1000.0 objective flux in current media\n",
      "6.31% reactions are blocked\n",
      "74.22% metabolites are generated for free\n",
      "41.82% metabolites are lost improperly\n",
      "No elemental data associated with metabolites!\n",
      "100.0% reactions without GPRs\n",
      "100.0% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iHD992_rmGenes = basicCheck(iHD992)\n",
    "iHD992_noGPRblocked = blockedReactions(iHD992)\n",
    "iHD992_free = checkFreeMass(iHD992, cytosol='Cytosol')\n",
    "iHD992_lost = checkLostMass(iHD992)\n",
    "iHD992_massImbal = checkBalance(iHD992)\n",
    "iHD992_nogene = missingGPR(iHD992)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCN900</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff0ed3adb90</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>885</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1229</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*BIOMASS__5 - 1.0*BIOMASS__5_reverse_2261a</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular space</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCN900 at 0x7ff0ed3adb90>"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCN900"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>BIOMASS__5</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f51239fc7d0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>CELLWALL_c + DNA_c + LIPID_c + PROTEIN_c + RNA_c + SOLPOOL_c + atp_c + h2o_c --> adp_c + biomass_c + pi_c</p>\n",
       "                    <p style='text-align:right'>Cell wall + Deoxyribonucleic acid + Lipid + Protein + Ribonucleic acid + Solute Pools + ATP C10H12N5O13P3 + H2O H2O --> ADP C10H12N5O10P2 + Biomass: total + Phosphate</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction BIOMASS__5 at 0x7f51239fc7d0>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCN900.reactions.BIOMASS__5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 132\n",
      "Transport reactions: 162\n",
      "Metabolic reactions: 935 \n",
      "\n",
      "e metabolites: 136\n",
      "c metabolites: 749\n"
     ]
    }
   ],
   "source": [
    "count_components(iCN900)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.389 (underdetermined)\n",
      "GENRE has 2 compartment(s)\n",
      "GENRE has 900 genes\n",
      "6.25 objective flux in current media\n",
      "21.56% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% metabolites are lost improperly\n",
      "10.74% reactions are mass imbalanced\n",
      "23.79% reactions without GPRs\n",
      "26.67% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCN900_rmGenes = basicCheck(iCN900)\n",
    "iCN900_noGPRblocked = blockedReactions(iCN900)\n",
    "iCN900_free = checkFreeMass(iCN900, cytosol='Cytosol')\n",
    "iCN900_lost = checkLostMass(iCN900)\n",
    "iCN900_massImbal = checkBalance(iCN900)\n",
    "iCN900_nogene = missingGPR(iCN900)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>cd630_PATRIC</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff0ed60ced0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>472</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>460</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*bio1 - 1.0*bio1_reverse_b18f7</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytosol_0, Extracellular_0</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model cd630_PATRIC at 0x7ff0ed60ced0>"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>bio1</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>GramPositiveBiomass_auto_biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f5123852490</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>41.2913947104178 cpd00002_c0 + 0.00793965859468043 cpd00003_c0 + 0.00793965859468043 cpd00006_c0 + 0.00793965859468043 cpd00010_c0 + 0.00793965859468043 cpd00015_c0 + 0.00793965859468043 cpd00016_c...</p>\n",
       "                    <p style='text-align:right'>41.2913947104178 ATP_c0 + 0.00793965859468043 NAD_c0 + 0.00793965859468043 NADP_c0 + 0.00793965859468043 CoA_c0 + 0.00793965859468043 FAD_c0 + 0.00793965859468043 Pyridoxal_phosphate_c0 + 0.0079396...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction bio1 at 0x7f5123852490>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC.reactions.bio1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 73\n",
      "Transport reactions: 70\n",
      "Metabolic reactions: 317 \n",
      "\n",
      "e0 metabolites: 138\n",
      "c0 metabolites: 334\n"
     ]
    }
   ],
   "source": [
    "count_components(cd630_PATRIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 0.975 (overdetermined)\n",
      "GENRE has 2 compartment(s)\n",
      "GENRE has 1 genes\n",
      "10.989 objective flux in current media\n",
      "1.3% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "0.0% metabolites are lost improperly\n",
      "No elemental data associated with metabolites!\n",
      "18.21% reactions without GPRs\n",
      "18.49% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "cd630_PATRIC_rmGenes = basicCheck(cd630_PATRIC)\n",
    "cd630_PATRIC_noGPRblocked = blockedReactions(cd630_PATRIC)\n",
    "cd630_PATRIC_free = checkFreeMass(cd630_PATRIC, cytosol='Cytosol_0')\n",
    "cd630_PATRIC_lost = checkLostMass(cd630_PATRIC)\n",
    "cd630_PATRIC_massImbal = checkBalance(cd630_PATRIC)\n",
    "cd630_PATRIC_nogene = missingGPR(cd630_PATRIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridioides difficile R20291</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff62e3a7350</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1243</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1143</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*bio1 - 1.0*bio1_reverse_b18f7</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>c, e</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridioides difficile R20291 at 0x7ff62e3a7350>"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cdR20291_PATRIC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>bio1</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>GramPositiveBiomass auto biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f5122cb1610</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>41.2913947104178 cpd00002_c + 0.00793965859468043 cpd00003_c + 0.00793965859468043 cpd00006_c + 0.00793965859468043 cpd00010_c + 0.00793965859468043 cpd00015_c + 0.00793965859468043 cpd00016_c + 0....</p>\n",
       "                    <p style='text-align:right'>41.2913947104178 ATP_c + 0.00793965859468043 NAD_c + 0.00793965859468043 NADP_c + 0.00793965859468043 CoA_c + 0.00793965859468043 FAD_c + 0.00793965859468043 Pyridoxal phosphate_c + 0.0079396585946...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction bio1 at 0x7f5122cb1610>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cdR20291_PATRIC.reactions.bio1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 86\n",
      "Transport reactions: 82\n",
      "Metabolic reactions: 975 \n",
      "\n",
      "e metabolites: 82\n",
      "c metabolites: 1161\n"
     ]
    }
   ],
   "source": [
    "count_components(cdR20291_PATRIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 0.92 (overdetermined)\n",
      "GENRE has 2 compartment(s)\n",
      "GENRE has 715 genes\n",
      "26.263 objective flux in current media\n",
      "49.43% reactions are blocked\n",
      "0.24% metabolites are generated for free\n",
      "5.31% metabolites are involved in potential cycles\n",
      "0.09% reactions are mass imbalanced\n",
      "13.21% reactions without GPRs\n",
      "35.09% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "cdR20291_PATRIC_rmGenes = basicCheck(cdR20291_PATRIC)\n",
    "cdR20291_PATRIC_noGPRblocked = blockedReactions(cdR20291_PATRIC)\n",
    "cdR20291_PATRIC_free = checkFreeMass(cdR20291_PATRIC, cytosol='c')\n",
    "cdR20291_PATRIC_lost = checkLostMass(cdR20291_PATRIC)\n",
    "cdR20291_PATRIC_massImbal = checkBalance(cdR20291_PATRIC)\n",
    "cdR20291_PATRIC_nogene = missingGPR(cdR20291_PATRIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridium_difficile_NAP07</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff633aafa50</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1332</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1689</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>75</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass525 - 1.0*biomass525_reverse_5c178</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytoplasm, Extracellular, Mitochondrion</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridium_difficile_NAP07 at 0x7ff633aafa50>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_NAP07"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>biomass525</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>biomass525</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f5122477e50</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.0078094 ACP[c] + 0.092476 PGP[c] + 0.50006 ala_L[c] + 0.0078094 amet[c] + 0.28827 arg_L[c] + 0.23468 asn_L[c] + 0.23468 asp_L[c] + 40.1701 atp[c] + 0.0078094 ca2[c] + 0.0078094 cl[c] + 0.0078094 ...</p>\n",
       "                    <p style='text-align:right'>0.0078094 acyl carrier protein + 0.092476 Peptidoglycan polymer (n subunits) + 0.50006 L-alanine + 0.0078094 S-Adenosyl-L-methionine + 0.28827 L-argininium(1+) + 0.23468 L-asparagine + 0.23468 L-as...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction biomass525 at 0x7f5122477e50>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_NAP07.reactions.biomass525"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 378\n",
      "Transport reactions: 169\n",
      "Metabolic reactions: 1142 \n",
      "\n",
      "m metabolites: 1\n",
      "e metabolites: 366\n",
      "c metabolites: 965\n"
     ]
    }
   ],
   "source": [
    "count_components(agora_NAP07)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intracellular exchange reactions: 0\n"
     ]
    }
   ],
   "source": [
    "agora_NAP07_inexch = intracellular_exchanges(agora_NAP07, cytosol='Cytoplasm')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.268 (underdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 865 genes\n",
      "128.007 objective flux in current media\n",
      "30.25% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "4.28% metabolites are involved in potential cycles\n",
      "0.24% reactions are mass imbalanced\n",
      "11.22% reactions without GPRs\n",
      "27.57% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "agora_NAP07_rmGenes = basicCheck(agora_NAP07)\n",
    "agora_NAP07_noGPRblocked = blockedReactions(agora_NAP07)\n",
    "agora_NAP07_free = checkFreeMass(agora_NAP07, cytosol='Cytosol_0')\n",
    "agora_NAP07_lost = checkLostMass(agora_NAP07)\n",
    "agora_NAP07_massImbal = checkBalance(agora_NAP07)\n",
    "agora_NAP07_nogene = missingGPR(agora_NAP07)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridium_difficile_NAP08</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff64aef95d0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1336</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1693</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>74</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass525 - 1.0*biomass525_reverse_5c178</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytoplasm, Extracellular, Mitochondrion</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridium_difficile_NAP08 at 0x7ff64aef95d0>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_NAP08"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>biomass525</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>biomass525</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f5121bf9510</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.0078094 ACP[c] + 0.092476 PGP[c] + 0.50006 ala_L[c] + 0.0078094 amet[c] + 0.28827 arg_L[c] + 0.23468 asn_L[c] + 0.23468 asp_L[c] + 40.1701 atp[c] + 0.0078094 ca2[c] + 0.0078094 cl[c] + 0.0078094 ...</p>\n",
       "                    <p style='text-align:right'>0.0078094 acyl carrier protein + 0.092476 Peptidoglycan polymer (n subunits) + 0.50006 L-alanine + 0.0078094 S-Adenosyl-L-methionine + 0.28827 L-argininium(1+) + 0.23468 L-asparagine + 0.23468 L-as...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction biomass525 at 0x7f5121bf9510>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_NAP08.reactions.biomass525"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 378\n",
      "Transport reactions: 169\n",
      "Metabolic reactions: 1146 \n",
      "\n",
      "m metabolites: 1\n",
      "e metabolites: 366\n",
      "c metabolites: 969\n"
     ]
    }
   ],
   "source": [
    "count_components(agora_NAP08)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intracellular exchange reactions: 0\n"
     ]
    }
   ],
   "source": [
    "agora_NAP08_inexch = intracellular_exchanges(agora_NAP08, cytosol='Cytoplasm')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.267 (underdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 865 genes\n",
      "127.965 objective flux in current media\n",
      "30.3% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "4.27% metabolites are involved in potential cycles\n",
      "0.24% reactions are mass imbalanced\n",
      "11.19% reactions without GPRs\n",
      "27.67% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "agora_NAP08_rmGenes = basicCheck(agora_NAP08)\n",
    "agora_NAP08_noGPRblocked = blockedReactions(agora_NAP08)\n",
    "agora_NAP08_free = checkFreeMass(agora_NAP08)\n",
    "agora_NAP08_lost = checkLostMass(agora_NAP08)\n",
    "agora_NAP08_massImbal = checkBalance(agora_NAP08)\n",
    "agora_NAP08_nogene = missingGPR(agora_NAP08)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridium_difficile_CD196</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff64a6ba450</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1379</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1809</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>71</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass205 - 1.0*biomass205_reverse_3c9be</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytoplasm, Extracellular, Mitochondrion</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridium_difficile_CD196 at 0x7ff64a6ba450>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_CD196"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>biomass205</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>biomass205</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f5121204d50</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.0079397 10fthf[c] + 0.0079397 2dmmq8[c] + 0.0079397 5mthf[c] + 0.0079397 ACP[c] + 0.0018061 PGP[c] + 0.0079397 adocbl[c] + 0.0018061 ai17tca1[c] + 0.0018061 ai17tcaacgam[c] + 0.0018061 ai17tcaala...</p>\n",
       "                    <p style='text-align:right'>0.0079397 10-Formyltetrahydrofolate + 0.0079397 2-Demethylmenaquinone 8 + 0.0079397 5-Methyltetrahydrofolate + 0.0079397 acyl carrier protein + 0.0018061 Peptidoglycan polymer (n subunits) + 0.0079...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction biomass205 at 0x7f5121204d50>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_CD196.reactions.biomass205"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 394\n",
      "Transport reactions: 175\n",
      "Metabolic reactions: 1240 \n",
      "\n",
      "m metabolites: 1\n",
      "e metabolites: 371\n",
      "c metabolites: 1007\n"
     ]
    }
   ],
   "source": [
    "count_components(agora_CD196)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intracellular exchange reactions: 0\n"
     ]
    }
   ],
   "source": [
    "agora_CD196_inexch = intracellular_exchanges(agora_CD196, cytosol='Cytoplasm')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.312 (underdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 905 genes\n",
      "154.647 objective flux in current media\n",
      "18.19% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "4.93% metabolites are involved in potential cycles\n",
      "0.44% reactions are mass imbalanced\n",
      "16.58% reactions without GPRs\n",
      "34.09% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "agora_CD196_rmGenes = basicCheck(agora_CD196)\n",
    "agora_CD196_noGPRblocked = blockedReactions(agora_CD196)\n",
    "agora_CD196_free = checkFreeMass(agora_CD196)\n",
    "agora_CD196_lost = checkLostMass(agora_CD196)\n",
    "agora_CD196_massImbal = checkBalance(agora_CD196)\n",
    "agora_CD196_nogene = missingGPR(agora_CD196)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>Clostridium_difficile_R20291</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff62f3a7210</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1164</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1402</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>69</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass205 - 1.0*biomass205_reverse_3c9be</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>Cytoplasm, Extracellular, Mitochondrion</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model Clostridium_difficile_R20291 at 0x7ff62f3a7210>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_R20291"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>biomass205</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>biomass205</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f5120b93210</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.0079397 10fthf[c] + 0.0079397 2dmmq8[c] + 0.0079397 5mthf[c] + 0.0079397 ACP[c] + 0.0018061 PGP[c] + 0.0079397 adocbl[c] + 0.0018061 ai17tca1[c] + 0.0018061 ai17tcaacgam[c] + 0.0018061 ai17tcaala...</p>\n",
       "                    <p style='text-align:right'>0.0079397 10-Formyltetrahydrofolate + 0.0079397 2-Demethylmenaquinone 8 + 0.0079397 5-Methyltetrahydrofolate + 0.0079397 acyl carrier protein + 0.0018061 Peptidoglycan polymer (n subunits) + 0.0079...</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction biomass205 at 0x7f5120b93210>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agora_R20291.reactions.biomass205"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exchange reactions: 186\n",
      "Transport reactions: 172\n",
      "Metabolic reactions: 1044 \n",
      "\n",
      "m metabolites: 1\n",
      "e metabolites: 164\n",
      "c metabolites: 999\n"
     ]
    }
   ],
   "source": [
    "count_components(agora_R20291)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intracellular exchange reactions: 0\n"
     ]
    }
   ],
   "source": [
    "agora_CD196_inexch = intracellular_exchanges(agora_CD196, cytosol='Cytoplasm')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.204 (underdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 893 genes\n",
      "154.201 objective flux in current media\n",
      "23.54% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "5.58% metabolites are involved in potential cycles\n",
      "0.57% reactions are mass imbalanced\n",
      "21.74% reactions without GPRs\n",
      "34.25% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "agora_R20291_rmGenes = basicCheck(agora_R20291)\n",
    "agora_R20291_noGPRblocked = blockedReactions(agora_R20291)\n",
    "agora_R20291_free = checkFreeMass(agora_R20291)\n",
    "agora_R20291_lost = checkLostMass(agora_R20291)\n",
    "agora_R20291_massImbal = checkBalance(agora_R20291)\n",
    "agora_R20291_nogene = missingGPR(agora_R20291)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Papin Lab GENREs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG698 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG698.json')\n",
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG693</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f2936a56e90</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1235</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1309</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG693 at 0x7f2936a56e90>"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG698"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Biomass Reaction</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f51203bbad0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.2 cellwall_c + 0.2 cofactor_c + 20.0 cpd00001_c + 20.0 cpd00002_c + 0.05 cpd11852_c + 0.4 cpd17041_c + 0.05 cpd17042_c + 0.15 cpd17043_c --> 20.0 cpd00008_c + 20.0 cpd00009_c + cpd11416_c</p>\n",
       "                    <p style='text-align:right'>0.2 Cell Wall polymer + 0.2 Cofactor Pool + 20.0 H2O + 20.0 ATP + 0.05 Lipid Pool + 0.4 Protein biosynthesis + 0.05 DNA replication + 0.15 RNA transcription --> 20.0 ADP + 20.0 Phosphate + Biomass</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction biomass at 0x7f51203bbad0>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG698.reactions.biomass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.06 (underdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 693 genes\n",
      "143.692 objective flux in current media\n",
      "34.0% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "6.48% metabolites are involved in potential cycles\n",
      "2.14% reactions are mass imbalanced\n",
      "15.78% reactions without GPRs\n",
      "26.34% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdG692_rmGenes = basicCheck(iCdG698)\n",
    "iCdG692_noGPRblocked = blockedReactions(iCdG698)\n",
    "iCdG692_free = checkFreeMass(iCdG698)\n",
    "iCdG692_lost = checkLostMass(iCdG698)\n",
    "iCdG692_massImbal = checkBalance(iCdG698)\n",
    "iCdG692_nogene = missingGPR(iCdG698)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdR700</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07ff3e1df3b90</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1245</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1309</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdR700 at 0x7ff3e1df3b90>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdR700"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Biomass Reaction</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f511fc52910</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>0.2 cellwall_c + 0.2 cofactor_c + 20.0 cpd00001_c + 20.0 cpd00002_c + 0.05 cpd11852_c + 0.4 cpd17041_c + 0.05 cpd17042_c + 0.15 cpd17043_c --> 20.0 cpd00008_c + 20.0 cpd00009_c + cpd11416_c</p>\n",
       "                    <p style='text-align:right'>0.2 Cell Wall polymer + 0.2 Cofactor Pool + 20.0 H2O + 20.0 ATP + 0.05 Lipid Pool + 0.4 Protein biosynthesis + 0.05 DNA replication + 0.15 RNA transcription --> 20.0 ADP + 20.0 Phosphate + Biomass</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction biomass at 0x7f511fc52910>"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdR700.reactions.biomass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to metabolites ratio: 1.053 (underdetermined)\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 700 genes\n",
      "183.666 objective flux in current media\n",
      "34.84% reactions are blocked\n",
      "0.0% metabolites are generated for free\n",
      "12.55% metabolites are involved in potential cycles\n",
      "2.14% reactions are mass imbalanced\n",
      "15.98% reactions without GPRs\n",
      "27.46% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "iCdR700_rmGenes = basicCheck(iCdR700)\n",
    "iCdR700_noGPRblocked = blockedReactions(iCdR700)\n",
    "iCdR700_free = checkFreeMass(iCdR700)\n",
    "iCdR700_lost = checkLostMass(iCdR700)\n",
    "iCdR700_massImbal = checkBalance(iCdR700)\n",
    "iCdR700_nogene = missingGPR(iCdR700)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Comparing GENRE gene content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Determine overlap between genomes and genres\n",
    "genes_630 = set()\n",
    "genes_r20291 = set()\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/gene_alignment.tsv','r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        gene_r20291 = line.split()[0].lstrip('fig|')\n",
    "        if gene_r20291 != 'none': genes_r20291 |= set([gene_r20291])\n",
    "        \n",
    "        gene_630 = line.split()[1].lstrip('fig|')\n",
    "        if gene_630 != 'none': genes_630 |= set([gene_630])\n",
    "\n",
    "genes_iCdR700 = set([y.id for y in iCdR700.genes])\n",
    "included_iCdR700 = genes_iCdR700.intersection(genes_r20291)\n",
    "\n",
    "genes_iCdG719 = set([x.id for x in iCdG719.genes])\n",
    "included_iCdG719 = genes_iCdG719.intersection(genes_630)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Record to table for figure\n",
    "\n",
    "outfile = open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/all_gene_alignment.tsv','w')\n",
    "outfile.write('gene_name\\tcdR20291_gene\\tcd630_gene\\tcdR20291_homolog\\tcd630_homolog\\tiCdR700_included\\tiCdG719_included\\n')\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/gene_alignment.tsv','r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        line = line.strip()\n",
    "        gene_r20291 = line.split()[0].lstrip('fig|')\n",
    "        if gene_r20291 in included_iCdR700:\n",
    "            entry_iCdR700 = '1'\n",
    "        else:\n",
    "            entry_iCdR700 = '0'\n",
    "        \n",
    "        gene_630 = line.split()[1].lstrip('fig|')\n",
    "        if gene_630 in included_iCdG719:\n",
    "            entry_iCdG719 = '1'\n",
    "        else:\n",
    "            entry_iCdG719 = '0'\n",
    "        \n",
    "        gene_name = 'none'\n",
    "        if gene_r20291 != 'none':\n",
    "            try:\n",
    "                gene_name = iCdR700.genes.get_by_id(gene_r20291).name\n",
    "            except:\n",
    "                pass\n",
    "        else:\n",
    "            try:\n",
    "                gene_name = iCdG719.genes.get_by_id(gene_630).name\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        outfile.write(gene_name + '\\t' + line + '\\t' + entry_iCdR700 + '\\t' + entry_iCdG719 + '\\n')\n",
    "\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Growth substrates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG693 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.json')\n",
    "iCdR700 = read_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdG693.reactions.EX_cpd00443_e.name = '4-Aminobenzoate exchange'\n",
    "iCdG693.reactions.rxn05527_c.name = 'Cytosine hydrogen symport'\n",
    "iCdG693.reactions.rxn05318_c.name = 'Adenosine hydrogen symport'\n",
    "iCdG693.reactions.rxn05201_c.name = 'Hypoxanthine hydrogen symport'\n",
    "iCdG693.reactions.rxn05204_c.name = 'Guanosine hydrogen symport'\n",
    "iCdG693.reactions.EX_cpd00226_e.name = 'Hypoxanthine exchange'\n",
    "iCdG693.reactions.EX_cpd00309_e.name = 'Xanthine exchange'\n",
    "iCdG693.reactions.EX_cpd00104_e.name = 'Biotin exchange'\n",
    "iCdG693.reactions.EX_cpd00794_e.name = 'D-Trehalose exchange'\n",
    "iCdG693.reactions.EX_cpd00158_e.name = 'Cellobiose exchange'\n",
    "iCdG693.reactions.EX_cpd00644_e.name = 'Pantothenate exchange'\n",
    "\n",
    "iCdR700.reactions.EX_cpd00443_e.name = '4-Aminobenzoate exchange'\n",
    "iCdR700.reactions.rxn05527_c.name = 'Cytosine hydrogen symport'\n",
    "iCdR700.reactions.rxn05318_c.name = 'Adenosine hydrogen symport'\n",
    "iCdR700.reactions.rxn05201_c.name = 'Hypoxanthine hydrogen symport'\n",
    "iCdR700.reactions.rxn05204_c.name = 'Guanosine hydrogen symport'\n",
    "iCdR700.reactions.EX_cpd00226_e.name = 'Hypoxanthine exchange'\n",
    "iCdR700.reactions.EX_cpd00309_e.name = 'Xanthine exchange'\n",
    "iCdR700.reactions.EX_cpd00104_e.name = 'Biotin exchange'\n",
    "iCdR700.reactions.EX_cpd00794_e.name = 'D-Trehalose exchange'\n",
    "iCdR700.reactions.EX_cpd00158_e.name = 'Cellobiose exchange'\n",
    "iCdR700.reactions.EX_cpd00644_e.name = 'Pantothenate exchange'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "\n",
    "EX_cpd11574_e = copy.deepcopy(iCdR700.reactions.EX_cpd11574_e)\n",
    "rxn05619_c = copy.deepcopy(iCdR700.reactions.rxn05619_c)\n",
    "rxn10344_c = copy.deepcopy(iCdR700.reactions.rxn10344_c)\n",
    "EX_cpd04097_e = copy.deepcopy(iCdR700.reactions.EX_cpd04097_e)\n",
    "rxn05516_c = copy.deepcopy(iCdR700.reactions.rxn05516_c)\n",
    "EX_cpd01012_e = copy.deepcopy(iCdR700.reactions.EX_cpd01012_e)\n",
    "rxn10343_c = copy.deepcopy(iCdR700.reactions.rxn10343_c)\n",
    "EX_cpd00531_e = copy.deepcopy(iCdR700.reactions.EX_cpd00531_e)\n",
    "rxn10481_c = copy.deepcopy(iCdR700.reactions.rxn10481_c)\n",
    "EX_cpd00058_e = copy.deepcopy(iCdR700.reactions.EX_cpd00058_e)\n",
    "rxn05528_c = copy.deepcopy(iCdR700.reactions.rxn05528_c)\n",
    "iCdG693.add_reactions([EX_cpd11574_e,rxn05619_c,rxn10344_c,EX_cpd04097_e,rxn05516_c,EX_cpd01012_e,\n",
    "                      rxn10343_c,EX_cpd00531_e,rxn10481_c,EX_cpd00058_e,rxn05528_c])\n",
    "\n",
    "geneDict = {'645463.3.peg.857':'272563.8.peg.907', '645463.3.peg.1639':'272563.8.peg.1714', \n",
    "            '645463.3.peg.2886':'272563.8.peg.2961', '645463.3.peg.560':'272563.8.peg.620', \n",
    "            '645463.3.peg.2163':'272563.8.peg.2223'}\n",
    "cobra.manipulation.modify.rename_genes(iCdG693, geneDict)\n",
    "\n",
    "iCdG693.genes.get_by_id('272563.8.peg.907').annotation['patric'] = 'fig|272563.8.peg.907'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.907').annotation['refseq'] = 'NC_009089.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.907').annotation['refseq.locustag'] = 'WP_004454197.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.1714').annotation['patric'] = 'fig|272563.8.peg.1714'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.1714').annotation['refseq'] = 'NC_009089.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.1714').annotation['refseq.locustag'] = 'WP_009893200'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.2961').annotation['patric'] = 'fig|272563.8.peg.2961'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.2961').annotation['refseq'] = 'NC_009089.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.2961').annotation['refseq.locustag'] = 'WP_009898004.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.620').annotation['patric'] = 'fig|272563.8.peg.620'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.620').annotation['refseq'] = 'NC_009089.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.620').annotation['refseq.locustag'] = 'WP_011860870.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.2223').annotation['patric'] = 'fig|272563.8.peg.2223'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.2223').annotation['refseq'] = 'NC_009089.1'\n",
    "iCdG693.genes.get_by_id('272563.8.peg.2223').annotation['refseq.locustag'] = 'WP_011861457.1'\n",
    "\n",
    "iCdG693.id = 'iCdG698'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set media\n",
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e', # Pyridoxine\n",
    "       'EX_cpd00027_e'] # Glucose\n",
    "\n",
    "for rxn in iCdG693.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "for rxn in iCdR700.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(iCdG693, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG698.sbml')\n",
    "cobra.io.save_json_model(iCdG693, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG698.json')\n",
    "cobra.io.write_sbml_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.sbml')\n",
    "cobra.io.save_json_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set media\n",
    "mdm_noGluc = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e'] # Pyridoxine\n",
    "\n",
    "for rxn in iCdG693.boundary:\n",
    "    if not rxn.id in mdm_noGluc:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "for rxn in iCdR700.boundary:\n",
    "    if not rxn.id in mdm_noGluc:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "mdm_noGluc_cpds = ['cpd00001_e','cpd00065_e','cpd00060_e','cpd00322_e','cpd00129_e','cpd00156_e',\n",
    "                   'cpd00107_e','cpd00084_e','cpd00149_e','cpd00099_e','cpd10515_e','cpd00030_e',\n",
    "                   'cpd00254_e','cpd00063_e','cpd00205_e','cpd00009_e','cpd00971_e','cpd00242_e',\n",
    "                   'cpd00104_e','cpd00644_e','cpd00263_e']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate biomass shift ratio\n",
    "biolog_cpds = ['cpd00383','cpd00162','cpd00438','cpd00654',\n",
    "               'cpd00277','cpd03279','cpd00412','cpd00136','cpd00339',\n",
    "               'cpd00027','cpd00208','cpd03561','cpd00029','cpd00142','cpd00182','cpd00152',\n",
    "               'cpd29695','cpd29696','cpd11586','cpd29694','cpd29697','cpd29698','cpd11582','cpd03697',\n",
    "               'cpd00085','cpd00211','cpd00098','cpd00137','cpd00424','cpd00367','cpd00307','cpd00731',\n",
    "               'cpd29694','cpd29698','cpd00117','cpd00339','cpd00320','cpd00158','cpd00082','cpd00072',\n",
    "               'cpd00108','cpd00276','cpd00089','cpd00079','cpd00186','cpd00314','cpd00138','cpd05161',\n",
    "               'cpd00637','cpd00644','cpd00105','cpd00588','cpd00161','cpd00794','cpd00154','cpd00080',\n",
    "               'cpd00504','cpd00130','cpd00393','cpd00378','cpd00047','cpd00106','cpd00281','cpd03561',\n",
    "               'cpd00042','cpd11581','cpd15603','cpd29691','cpd11592','cpd15604','cpd29693','cpd11588',\n",
    "               'cpd00100','cpd00033','cpd00155','cpd00139','cpd00207','cpd00311','cpd29317','cpd00165',\n",
    "               'cpd00226','cpd00246','cpd00035','cpd00224','cpd00051','cpd00132','cpd00041','cpd00274',\n",
    "               'cpd00395','cpd00084','cpd00751','cpd00023','cpd00053','cpd00119','cpd00227','cpd00322',\n",
    "               'cpd00159','cpd00107','cpd00039','cpd00130','cpd00060','cpd00064','cpd00066','cpd00129',\n",
    "               'cpd00054','cpd00161','cpd00065','cpd00069','cpd00156','cpd00489','cpd00121','cpd00179',\n",
    "               'cpd29700','cpd27607','cpd00122','cpd00492','cpd00477','cpd00232','cpd00075',\n",
    "               'cpd00738','cpd08928','cpd00247','cpd00443','cpd00489','cpd00009','cpd00457','cpd00141',\n",
    "               'cpd00118','cpd00215','cpd00020','cpd00220','cpd01030','cpd00036','cpd00076','cpd00210',\n",
    "               'cpd03047','cpd00305','cpd00793','cpd00184','cpd00151','cpd20885','cpd00374','cpd00092',\n",
    "               'cpd00300','cpd00249','cpd00309','cpd01217']\n",
    "\n",
    "iCdG693_results = {}\n",
    "objVal_baseline = iCdG693.slim_optimize()\n",
    "for cpd in biolog_cpds:\n",
    "    cpd_e = cpd + '_e'  \n",
    "    exchange_rxn = 'EX_' + cpd_e\n",
    "    try:\n",
    "        iCdG693.reactions.get_by_id(exchange_rxn).bounds = (-1000., 1000.)\n",
    "        objVal = iCdG693.slim_optimize()\n",
    "        if objVal < 0.: objVal = 0.\n",
    "        objRatio = objVal / objVal_baseline\n",
    "        objRatio = round(objRatio, 3)\n",
    "            \n",
    "        if objRatio > 1.: \n",
    "            iCdG693_results[cpd] = [1, objRatio]\n",
    "        else:\n",
    "            iCdG693_results[cpd] = [0, objRatio]\n",
    "            \n",
    "        if not cpd_e in mdm_noGluc_cpds:\n",
    "            iCdG693.reactions.get_by_id(exchange_rxn).bounds = (0., 1000.)\n",
    "    except KeyError:\n",
    "        iCdG693_results[cpd] = ['keyerror', 'keyerror']\n",
    "iCdG693.reactions.EX_cpd00027_e.bounds = (-1000., 1000.) # glucose\n",
    "\n",
    "iCdR700_results = {}\n",
    "objVal_baseline = iCdR700.slim_optimize()\n",
    "for cpd in biolog_cpds:\n",
    "    cpd_e = cpd + '_e'  \n",
    "    exchange_rxn = 'EX_' + cpd_e \n",
    "    try:\n",
    "        iCdR700.reactions.get_by_id(exchange_rxn).bounds = (-1000., 1000.)\n",
    "        objVal = iCdR700.slim_optimize()\n",
    "        if objVal < 0.: objVal = 0.\n",
    "        objRatio = objVal / objVal_baseline\n",
    "        objRatio = round(objRatio, 3)\n",
    "        \n",
    "        if objRatio > 1.: \n",
    "            iCdR700_results[cpd] = [1, objRatio]\n",
    "        else:\n",
    "            iCdR700_results[cpd] = [0, objRatio]\n",
    "        \n",
    "        if not cpd_e in mdm_noGluc_cpds:\n",
    "            iCdR700.reactions.get_by_id(exchange_rxn).bounds = (0., 1000.)\n",
    "    except KeyError:\n",
    "        iCdR700_results[cpd] = ['keyerror', 'keyerror']\n",
    "iCdR700.reactions.EX_cpd00027_e.bounds = (-1000., 1000.)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compare with biolog results\n",
    "oufile1 = open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/biolog_sim.binary.tsv', 'w')\n",
    "oufile1.write('metabolite\\tname\\tgroup\\tiCdG693_sim\\tCd630_biolog\\tiCdR700_sim\\tCdR20291_biolog\\n')\n",
    "\n",
    "oufile2 = open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/biolog_sim.values.tsv', 'w')\n",
    "oufile2.write('metabolite\\tname\\tgroup\\tiCdG693_sim\\tCd630_biolog\\tiCdR700_sim\\tCdR20291_biolog\\n')\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/scaria.tsv', 'r') as scaria:\n",
    "    header = scaria.readline()\n",
    "    for line in scaria:\n",
    "        line = line.split()\n",
    "        cpd = line[2]\n",
    "        name = line[0]\n",
    "        group = line[1]\n",
    "        info = cpd + '\\t' + name + '\\t' + group\n",
    "        \n",
    "        if iCdG693_results[cpd][0] == 'keyerror':\n",
    "            continue\n",
    "        elif iCdR700_results[cpd][0] == 'keyerror':\n",
    "            continue\n",
    "        else:\n",
    "            growth_630 = float(line[3])\n",
    "            if growth_630 > 1.:\n",
    "                binary_630 = 1\n",
    "            else:\n",
    "                binary_630 = 0\n",
    "            sim_630 = iCdG693_results[cpd]\n",
    "        \n",
    "            growth_R2021 = float(line[4])\n",
    "            if growth_R2021 > 1.:\n",
    "                binary_R2021 = 1\n",
    "            else:\n",
    "                binary_R2021 = 0\n",
    "            sim_R2021 = iCdR700_results[cpd]\n",
    "\n",
    "            binary = info + '\\t' + str(sim_630[0]) + '\\t' + str(binary_630) + '\\t' + str(sim_R2021[0]) + '\\t' + str(binary_R2021) + '\\n'\n",
    "            oufile1.write(binary)\n",
    "        \n",
    "            continuous = info + '\\t' + str(sim_630[1]) + '\\t' + str(growth_630) + '\\t' + str(sim_R2021[1]) + '\\t' + str(growth_R2021) + '\\n'\n",
    "            oufile2.write(continuous)\n",
    "\n",
    "oufile1.close()\n",
    "oufile2.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene Essentiality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import single_gene_deletion\n",
    "\n",
    "essential_refseq = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/R20291_essential_Dembek.tsv', 'r') as dembek:\n",
    "    for line in dembek:\n",
    "        essential_refseq.append(line.split()[0])\n",
    "essential_refseq = set(essential_refseq)\n",
    "\n",
    "essential_patric = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/PATRIC/CdR20291.PATRIC_RefSeq.tsv', 'r') as refseq:\n",
    "    for line in refseq:\n",
    "        line = line.split()\n",
    "        if line[1] in essential_refseq:\n",
    "            essential_patric.append(line[0])\n",
    "essential_patric = set(essential_patric)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set media\n",
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e', # Pyridoxine\n",
    "       'EX_cpd00027_e'] # Glucose\n",
    "\n",
    "for rxn in iCdG698.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "for rxn in iCdR700.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iCdG698 essential genes:  109\n"
     ]
    }
   ],
   "source": [
    "iCdG698_essential = []\n",
    "deletion_results = single_gene_deletion(iCdG698)\n",
    "\n",
    "for x, row in deletion_results.iterrows():\n",
    "    if float(row.growth) < 1e-6:\n",
    "        iCdG698_essential.append((list(x)[0]))\n",
    "iCdG698_essential = set(iCdG698_essential)\n",
    "\n",
    "print('iCdG698 essential genes: ', len(iCdG698_essential))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iCdR700 essential genes:  114\n"
     ]
    }
   ],
   "source": [
    "iCdR700_essential = []\n",
    "deletion_results = single_gene_deletion(iCdR700)\n",
    "\n",
    "for x, row in deletion_results.iterrows():\n",
    "    if float(row.growth) < 1e-6:\n",
    "        iCdR700_essential.append((list(x)[0]))\n",
    "iCdR700_essential = set(iCdR700_essential)\n",
    "\n",
    "print('iCdR700 essential genes: ', len(iCdR700_essential))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Metabolite Essentiality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ignore specific peptides\n",
    "peptides = ['EX_cpd11592_e','EX_cpd11590_e','EX_cpd11588_e','EX_cpd11586_e','EX_cpd11582_e','EX_cpd11581_e',\n",
    "              'EX_cpd15551_e','EX_cpd15603_e','EX_cpd15604_e','EX_cpd15605_e','EX_cpd15606_e','EX_cpd29695_e',\n",
    "              'EX_cpd29694_e','EX_cpd29697_e','EX_cpd29698_e','EX_cpd29696_e','EX_cpd29699_e','EX_cpd29693_e',\n",
    "              'EX_cpd29690_e','EX_cpd29691_e','EX_cpd29700_e','EX_cpd00320_e','EX_cpd02711_e','EX_cpd01553_e',\n",
    "              'EX_cpd00589_e','EX_cpd27607_e','EX_cpd00731_e','EX_cpd11580_e','EX_cpd11587_e','EX_cpd11589_e',\n",
    "              'EX_cpd11584_e','EX_cpd01017_e','EX_cpd11591_e','EX_cpd11583_e','EX_cpd11593_e','EX_cpd11585_e',\n",
    "              'EX_cpd00567_e','EX_cpd00064_e','EX_cpd00339_e']\n",
    "\n",
    "for rxn in peptides:\n",
    "    try:\n",
    "        iCdG693.reactions.get_by_id(rxn).bounds = (0.,1000.)\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        iCdR700.reactions.get_by_id(rxn).bounds = (0.,1000.)\n",
    "    except:\n",
    "        pass\n",
    "    \n",
    "#iCdG693.reactions.rxn00693_c.bounds = (0.,1000.)\n",
    "iCdG693.reactions.rxn06443_c.bounds = (0.,1000.)\n",
    "iCdG693.reactions.EX_cpd00027_e.bounds = (0.,1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11.194022322228646"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG693.reactions.EX_cpd00060_e.bounds = (0.,1000.)\n",
    "iCdG693.reactions.rxn00693_c.bounds = (0.,1000.)\n",
    "iCdG693.slim_optimize() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>rxn00693_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>5-Methyltetrahydrofolate:L-homocysteine S-methyltransferase</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f68ca285910</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd00135_c + cpd00345_c --> cpd00060_c + cpd00087_c</p>\n",
       "                    <p style='text-align:right'>Homocysteine + 5-Methyltetrahydrofolate --> L-Methionine + Tetrahydrofolate</p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td>272563.8.peg.3768 or 272563.8.peg.1815 or 272563.8.peg.3769</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>0.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction rxn00693_c at 0x7f68ca285910>"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG693.reactions.rxn00693_c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Metabolite identifier</strong></td><td>cpd00135_c</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Homocysteine</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f68b0787f90</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Formula</strong></td><td>C4H9NO2S</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartment</strong></td><td>cytosol</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>In 7 reaction(s)</strong></td><td>\n",
       "                    rxn00693_c, rxn05957_c, rxn00952_c, rxn06078_c, rxn00950_c, rxn05958_c, rxn00141_c</td>\n",
       "            </tr>\n",
       "        </table>"
      ],
      "text/plain": [
       "<Metabolite cpd00135_c at 0x7f68b0787f90>"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG693.metabolites.cpd00135_c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total: 18\n",
      "L-Tryptophan\n",
      "L-Isoleucine\n",
      "L-Proline\n",
      "L-Valine\n",
      "L-Leucine\n",
      "L-Cysteine\n",
      "Co2+\n",
      "Cl-\n",
      "Fe2+\n",
      "Mn2+\n",
      "Mg\n",
      "Ca2+\n",
      "K+\n",
      "Phosphate\n",
      "Na+\n",
      "Carbonate\n",
      "Biotin\n",
      "Pyridoxol\n"
     ]
    }
   ],
   "source": [
    "iCdG698.reactions.rxn06443_c.bounds = (0.,1000.)\n",
    "iCdG698.reactions.rxn00693_c.bounds = (0.,1000.)\n",
    "\n",
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e'] # Pyridoxine\n",
    "\n",
    "for rxn in iCdG698.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "\n",
    "iCdG698_cpd_essential = []\n",
    "threshold = iCdG698.slim_optimize() * 0.01\n",
    "for rxn in mdm:\n",
    "    iCdG698.reactions.get_by_id(rxn).bounds = (0.,1000.)\n",
    "    if iCdG698.slim_optimize() < threshold: iCdG698_cpd_essential.append(rxn.lstrip('EX_'))\n",
    "    iCdG698.reactions.get_by_id(rxn).bounds = (-1000.,1000.)\n",
    "    \n",
    "print('Total:', len(iCdG698_cpd_essential))\n",
    "for x in iCdG698_cpd_essential: print(iCdG698.metabolites.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total: 16\n",
      "Na+\n",
      "Fe2+\n",
      "Mg\n",
      "K+\n",
      "L-Tryptophan\n",
      "Co2+\n",
      "Cl-\n",
      "Biotin\n",
      "Ca2+\n",
      "L-Valine\n",
      "L-Isoleucine\n",
      "Mn2+\n",
      "L-Cysteine\n",
      "Pyridoxol\n",
      "L-Leucine\n",
      "L-Proline\n"
     ]
    }
   ],
   "source": [
    "iCdG698_cpd_essential = []\n",
    "threshold = iCdG698.slim_optimize() * 0.01\n",
    "for rxn in iCdG698.boundary:\n",
    "    if rxn.id in peptides: \n",
    "        continue\n",
    "    else:\n",
    "        rxn.bounds = (0.,1000.)\n",
    "        if iCdG698.slim_optimize() < threshold: iCdG698_cpd_essential.append(rxn.reactants[0].id)\n",
    "        rxn.bounds = (-1000.,1000.)\n",
    "            \n",
    "print('Total:', len(iCdG698_cpd_essential))\n",
    "for x in iCdG698_cpd_essential: print(iCdG698.metabolites.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total: 17\n",
      "L-Tryptophan\n",
      "L-Proline\n",
      "L-Valine\n",
      "L-Leucine\n",
      "L-Cysteine\n",
      "Co2+\n",
      "Cl-\n",
      "Fe2+\n",
      "Mn2+\n",
      "Mg\n",
      "Ca2+\n",
      "K+\n",
      "Phosphate\n",
      "Na+\n",
      "Carbonate\n",
      "Biotin\n",
      "Pyridoxol\n"
     ]
    }
   ],
   "source": [
    "iCdR700.reactions.rxn06443_c.bounds = (0.,1000.)\n",
    "iCdR700.reactions.rxn00693_c.bounds = (0.,1000.)\n",
    "\n",
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e'] # Pyridoxine\n",
    "\n",
    "for rxn in iCdR700.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "\n",
    "iCdR700_cpd_essential = []\n",
    "threshold = iCdR700.slim_optimize() * 0.01\n",
    "for rxn in mdm:\n",
    "    iCdR700.reactions.get_by_id(rxn).bounds = (0.,1000.)\n",
    "    if iCdR700.slim_optimize() < threshold: iCdR700_cpd_essential.append(rxn.lstrip('EX_'))\n",
    "    iCdR700.reactions.get_by_id(rxn).bounds = (-1000.,1000.)\n",
    "    \n",
    "print('Total:', len(iCdR700_cpd_essential))\n",
    "for x in iCdR700_cpd_essential: print(iCdR700.metabolites.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total: 15\n",
      "Na+\n",
      "Fe2+\n",
      "Mg\n",
      "K+\n",
      "L-Tryptophan\n",
      "Co2+\n",
      "Cl-\n",
      "Biotin\n",
      "Ca2+\n",
      "L-Valine\n",
      "Mn2+\n",
      "L-Cysteine\n",
      "Pyridoxol\n",
      "L-Leucine\n",
      "L-Proline\n"
     ]
    }
   ],
   "source": [
    "iCdR700_cpd_essential = []\n",
    "threshold = iCdR700.slim_optimize() * 0.01\n",
    "for rxn in iCdR700.boundary:\n",
    "    if rxn.id in peptides:\n",
    "        continue\n",
    "    else:\n",
    "        rxn.bounds = (0.,1000.)\n",
    "        if iCdR700.slim_optimize() < threshold: iCdR700_cpd_essential.append(rxn.reactants[0].id)\n",
    "        rxn.bounds = (-1000.,1000.)\n",
    "    \n",
    "print('Total:', len(iCdR700_cpd_essential))\n",
    "for x in iCdR700_cpd_essential: print(iCdR700.metabolites.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Na+\n",
    "Fe2+\n",
    "Mg\n",
    "K+\n",
    "L-Tryptophan\n",
    "Co2+\n",
    "Cl-\n",
    "Biotin\n",
    "Ca2+\n",
    "L-Valine\n",
    "L-Isoleucine\n",
    "Mn2+\n",
    "L-Cysteine\n",
    "Pyridoxol\n",
    "L-Leucine\n",
    "L-Proline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Biomass Composition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "68"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_S1 = open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/results/tables/table_S1.tsv', 'w')\n",
    "table_S1.write('rxn_id\\trxn_name\\tcpd_id\\tcpd_name\\tcoefficient\\tlower_bound\\tupper_bound\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cellwall\n",
    "rxn = iCdR700.reactions.teichoicacid_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)\n",
    "    \n",
    "rxn = iCdR700.reactions.peptidoglycan_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)\n",
    "    \n",
    "rxn = iCdR700.reactions.cellwall_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Lipids\n",
    "rxn = iCdR700.reactions.lipid_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cofactors\n",
    "rxn = iCdR700.reactions.cofactor_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Protein\n",
    "rxn = iCdR700.reactions.protein_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DNA\n",
    "rxn = iCdR700.reactions.dna_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA\n",
    "rxn = iCdR700.reactions.rna_rxn\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# tRNAs\n",
    "trna_rxns = ['rxn06300_c','rxn06432_c','rxn06434_c','rxn06936_c','rxn06435_c','rxn06437_c','rxn06436_c',\n",
    "             'rxn06438_c','rxn06439_c','rxn06440_c','rxn06441_c','rxn06442_c','rxn06443_c','rxn06444_c',\n",
    "             'rxn06445_c','rxn06446_c','rxn06447_c','rxn06448_c','rxn06280_c','rxn06449_c']\n",
    "for x in trna_rxns:\n",
    "    rxn = iCdR700.reactions.get_by_id(x)\n",
    "    for cpd in rxn.metabolites.keys():\n",
    "        entry = [rxn.id, rxn.name]\n",
    "        entry.append(cpd.id)\n",
    "        entry.append(cpd.name)\n",
    "        entry.append(str(rxn.metabolites[cpd]))\n",
    "        entry.append(str(rxn.lower_bound))\n",
    "        entry.append(str(rxn.upper_bound))\n",
    "        entry = '\\t'.join(entry) + '\\n'\n",
    "        entry = entry.replace(' ', '_')\n",
    "        table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final Biomass\n",
    "rxn = iCdR700.reactions.biomass\n",
    "for cpd in rxn.metabolites.keys():\n",
    "    entry = [rxn.id, rxn.name]\n",
    "    entry.append(cpd.id)\n",
    "    entry.append(cpd.name)\n",
    "    entry.append(str(rxn.metabolites[cpd]))\n",
    "    entry.append(str(rxn.lower_bound))\n",
    "    entry.append(str(rxn.upper_bound))\n",
    "    entry = '\\t'.join(entry) + '\\n'\n",
    "    entry = entry.replace(' ', '_')\n",
    "    table_S1.write(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "table_S1.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Aligning Gene Content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 630 vs R20291\n",
    "\n",
    "shared = []\n",
    "R20291_missing = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/proteome_comparisons/Cd630_R20291.tsv', 'r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        Cd630_gene = line.split()[0]\n",
    "        R20291_gene = line.split()[1]\n",
    "\n",
    "        if R20291_gene == 'none':\n",
    "            R20291_missing.append(Cd630_gene)\n",
    "        else:\n",
    "            shared.append(Cd630_gene)\n",
    "\n",
    "# R20291 vs 630\n",
    "Cd630_missing = []\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/proteome_comparisons/R20291_Cd630.tsv', 'r') as genes:\n",
    "    header = genes.readline()\n",
    "    for line in genes:\n",
    "        R20291_gene = line.split()[0]\n",
    "        Cd630_gene = line.split()[1]\n",
    "        \n",
    "        if Cd630_gene == 'none':\n",
    "            Cd630_missing.append(R20291_gene)\n",
    "        else:\n",
    "            shared.append(Cd630_gene)\n",
    "            \n",
    "R20291_missing = set(R20291_missing)\n",
    "Cd630_missing = set(Cd630_missing)\n",
    "shared = set(shared)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shared genes: 684\n"
     ]
    }
   ],
   "source": [
    "total_shared = 0\n",
    "for x in shared:\n",
    "    try:\n",
    "        iCdG693.genes.get_by_id(x)\n",
    "        total_shared += 1\n",
    "    except:\n",
    "        pass\n",
    "print('Shared genes:', total_shared)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "272563.8.peg.1910 rxn00926_c\n",
      "272563.8.peg.2144 rxn01219_c, rxn00364_c, rxn00707_c\n",
      "272563.8.peg.229 rxn01074_c\n",
      "272563.8.peg.895 rxn05536_c, rxn30280_c, rxn05542_c, rxn30264_c, rxn05545_c, rxn30270_c, rxn30274_c, rxn12851_c, rxn05546_c, rxn05540_c, rxn12849_c, rxn30262_c, rxn05534_c, rxn05547_c, rxn30278_c, rxn09189_c, rxn05535_c, rxn05544_c, rxn18658_c, rxn05541_c, rxn30268_c, rxn05533_c, rxn05538_c, rxn30266_c, rxn12848_c, rxn05537_c, rxn30259_c, rxn12850_c, rxn05539_c, rxn30276_c, rxn05543_c\n",
      "272563.8.peg.2793 rxn05209_c\n",
      "272563.8.peg.2790 rxn10183_c, rxn05226_c\n",
      "272563.8.peg.476 rxn00775_c\n",
      "272563.8.peg.262 teichoicacid_rxn\n"
     ]
    }
   ],
   "source": [
    "for gene in R20291_missing:\n",
    "    try:\n",
    "        gene = iCdG693.genes.get_by_id(gene)\n",
    "        reactions = ', '.join([x.id for x in gene.reactions])\n",
    "        print(gene.id, reactions)\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "645463.3.peg.239 rxn02000_c, rxn01116_c\n",
      "645463.3.peg.3485 rxn02796_c, rxn02795_c, rxn04068_c, rxn03094_c, rxn02015_c, rxn03095_c\n",
      "645463.3.peg.2866 rxn00686_c, rxn01602_c, rxn00655_c\n",
      "645463.3.peg.237 rxn02003_c\n",
      "645463.3.peg.57 rxn03063_c, rxn01520_c, rxn07312_c\n",
      "645463.3.peg.2716 rxn05209_c\n",
      "645463.3.peg.58 rxn00686_c, rxn01602_c, rxn00655_c\n"
     ]
    }
   ],
   "source": [
    "for gene in Cd630_missing:\n",
    "    try:\n",
    "        gene = iCdR700.genes.get_by_id(gene)\n",
    "        reactions = ', '.join([x.id for x in gene.reactions])\n",
    "        print(gene.id, reactions)\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "iCdR700 = cobra.io.load_json_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')\n",
    "iCdR700.reactions.rxn05061_c.lower_bound = 0.\n",
    "iCdR700.metabolites.cpd00232_e.name = 'N-Acetyl-Neuraminic acid'\n",
    "iCdR700.reactions.rxn30238_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdR700.reactions.rxn12212_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdR700.reactions.rxn10042_c.annotation['sbo'] = 'SBO:0000176'\n",
    "cobra.io.write_sbml_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.sbml')\n",
    "cobra.io.save_json_model(iCdR700, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdR700.json')\n",
    "\n",
    "iCdG693 = cobra.io.load_json_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.json')\n",
    "iCdG693.reactions.rxn05061_c.lower_bound = 0.\n",
    "iCdG693.reactions.rxn30238_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG693.reactions.rxn12212_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG693.reactions.rxn10042_c.annotation['sbo'] = 'SBO:0000176'\n",
    "iCdG693.metabolites.cpd00232_e.name = 'N-Acetyl-Neuraminic acid'\n",
    "cobra.io.write_sbml_model(iCdG693, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.sbml')\n",
    "cobra.io.save_json_model(iCdG693, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG693.json')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save biomass formulation to table\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/biomass_composition.tsv', 'w') as outfile:\n",
    "    outfile.write('id\\tname\\tbounds\\tflux_expression\\tid_reaction_str\\tname_reaction_str\\n')\n",
    "    for rxn in iCdG693.reactions:\n",
    "        if rxn.annotation['sbo'] == 'SBO:0000629':\n",
    "            entry = [rxn.id, rxn.name, str(rxn.bounds), str(rxn.flux_expression), rxn.build_reaction_string(), rxn.build_reaction_string(use_metabolite_names=True)]\n",
    "            entry = '\\t'.join(entry) +'\\n'\n",
    "            outfile.write(entry)\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
