{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### PATRIC auto-generated model\n",
    "\n",
    "Annotations converted via mackinac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception reporting mode: Verbose\n"
     ]
    }
   ],
   "source": [
    "# Import libraries - REQUIRES pip version 9.0.3\n",
    "import pandas\n",
    "import os\n",
    "from os.path import join\n",
    "import sys\n",
    "import scipy.stats\n",
    "import numpy\n",
    "import math\n",
    "import pickle\n",
    "import copy\n",
    "import time\n",
    "import warnings\n",
    "import gc\n",
    "\n",
    "# Using Cobrapy 0.13.0\n",
    "import cobra\n",
    "import cobra.test\n",
    "import cobra.flux_analysis.gapfilling\n",
    "from cobra.io import write_sbml_model\n",
    "from cobra.flux_analysis import  flux_variability_analysis\n",
    "from cobra.flux_analysis.reaction import assess_component\n",
    "from cobra.manipulation.delete import *\n",
    "from cobra.flux_analysis.parsimonious import add_pfba\n",
    "from cobra.medium import find_boundary_types\n",
    "from cobra.util import solver as sutil\n",
    "from cobra.flux_analysis.sampling import OptGPSampler\n",
    "\n",
    "# Confidence levels in each annotation\n",
    "#import probanno\n",
    "\n",
    "# Import scripted python functions\n",
    "sys.path.append('code/python/')\n",
    "from read_excel import read_excel\n",
    "\n",
    "# Estabish handler for logger\n",
    "import logging\n",
    "logging.basicConfig()\n",
    "logger = logging.getLogger('logger')\n",
    "\n",
    "# Verbose exception printing\n",
    "%xmode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define functions\n",
    "\n",
    "# Identify potentially gapfilled reactions\n",
    "def findGapfilledRxn(model, exclude=[]):\n",
    "    gapfilled = []\n",
    "    transport = findTransports(model)\n",
    "    if not type(exclude) is list:\n",
    "        exclude = [exclude]\n",
    "        \n",
    "    for index in model.reactions:\n",
    "        if len(list(index.genes)) == 0:\n",
    "            if not index in model.boundary:\n",
    "                if not index.id in exclude or not index.id in transport:\n",
    "                    gapfilled.append(index.id)\n",
    "    \n",
    "    if len(gapfilled) > 0:\n",
    "        print(str(len(gapfilled)) + ' metabolic reactions not associated with genes')\n",
    "    \n",
    "    return gapfilled\n",
    "\n",
    "# Check for missing transport and exchange reactions\n",
    "def missingRxns(model, extracellular='e'):\n",
    "\n",
    "    transporters = set(findTransports(model))\n",
    "    exchanges = set([x.id for x in model.exchanges])\n",
    "    \n",
    "    missing_exchanges = []\n",
    "    missing_transports = []\n",
    "    \n",
    "    for metabolite in model.metabolites:\n",
    "        if not metabolite.compartment == extracellular:\n",
    "            continue\n",
    "\n",
    "        curr_rxns = set([x.id for x in list(metabolite.reactions)])\n",
    "        \n",
    "        if not bool(curr_rxns & transporters):\n",
    "            missing_transports.append(metabolite.id)\n",
    "        if not bool(curr_rxns & exchanges):\n",
    "            missing_exchanges.append(metabolite.id)\n",
    "    \n",
    "    if len(missing_transports) != 0:\n",
    "        print(str(len(missing_transports)) + ' extracellular metabolites are missing transport reactions')\n",
    "    if len(missing_exchanges) != 0:\n",
    "        print(str(len(missing_exchanges)) + ' extracellular metabolites are missing exchange reactions')\n",
    "        \n",
    "    return missing_transports, missing_exchanges\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(raw_model, cytosol='cytosol'):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        \n",
    "        # Close all exchanges\n",
    "        for index in model.boundary:\n",
    "            model.reactions.get_by_id(index.id).lower_bound = 0.\n",
    "        \n",
    "        # Identify all metabolites that are produced within the network\n",
    "        demand_metabolites = [x.reactants[0].id for x in model.demands if len(x.reactants) > 0]\n",
    "        demand_metabolites += [x.products[0].id for x in model.demands if len(x.products) > 0]\n",
    "\n",
    "        free = []\n",
    "        for index in model.metabolites: \n",
    "            if index.id in demand_metabolites:\n",
    "                continue\n",
    "            elif not index.compartment in cytosol:\n",
    "                continue\n",
    "            else:\n",
    "                demand = model.add_boundary(index, type='demand')\n",
    "                model.objective = demand\n",
    "                obj_val = model.slim_optimize(error_value=0.)\n",
    "                if obj_val > 1e-8:\n",
    "                    free.append(index.id)\n",
    "                model.remove_reactions([demand])\n",
    "    \n",
    "    if len(free) > 0:\n",
    "        print(str(len(free)) + ' metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(raw_model, exclude=[]):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        imbalanced = []\n",
    "        mass_imbal = 0\n",
    "        charge_imbal = 0\n",
    "        elem_set = set()\n",
    "        for metabolite in model.metabolites:\n",
    "            try:\n",
    "                elem_set |= set(metabolite.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        if len(elem_set) == 0:\n",
    "            imbalanced = model.reactions\n",
    "            mass_imbal = len(model.reactions)\n",
    "            charge_imbal = len(model.reactions)\n",
    "            print('No elemental data associated with metabolites!')\n",
    "        \n",
    "        else:\n",
    "            if not type(exclude) is list: \n",
    "                exclude = [exclude]\n",
    "            for index in model.reactions:\n",
    "                if index in model.boundary or index.id in exclude:\n",
    "                    continue\n",
    "\n",
    "                else:\n",
    "                    try:\n",
    "                        test = index.check_mass_balance()\n",
    "                    except ValueError:\n",
    "                        continue\n",
    "\n",
    "                    if len(list(test)) > 0:\n",
    "                        imbalanced.append(index.id)\n",
    "\n",
    "                        if 'charge' in test.keys():\n",
    "                            charge_imbal += 1\n",
    "                        if len(set(test.keys()).intersection(elem_set)) > 0:\n",
    "                            mass_imbal += 1\n",
    "\n",
    "    if mass_imbal != 0:\n",
    "        print(str(mass_imbal) + ' reactions are mass imbalanced')\n",
    "    if charge_imbal != 0:\n",
    "        print(str(charge_imbal) + ' reactions are charge imbalanced')\n",
    "    \n",
    "    return(imbalanced)\n",
    "\n",
    "\n",
    "# Checks the quality of models by a couple metrics and returns problems\n",
    "def checkQuality(model, exclude=[], cytosol='c'):\n",
    "    \n",
    "    start_time = time.time()\n",
    "    \n",
    "    if model.name != None:\n",
    "        model_name = model.name\n",
    "    else:\n",
    "        model_name = 'model'\n",
    "    \n",
    "    gaps = findGapfilledRxn(model, exclude)\n",
    "    freemass = checkFreeMass(model, cytosol)\n",
    "    balance = checkBalance(model, exclude)\n",
    "    trans, exch = missingRxns(model)\n",
    "    \n",
    "    test = gaps + freemass + balance\n",
    "    if len(test) == 0:\n",
    "        print('No inconsistencies detected')\n",
    "    \n",
    "    duration = int(round(time.time() - start_time))\n",
    "    print('Took ' + str(duration) + ' seconds to analyze ' + model_name) \n",
    "\n",
    "    return gaps, freemass, balance, trans, exch\n",
    "\n",
    "\n",
    "# Trace back through reactions immediately adjacent to a given reaction to identify blocked precursor synthesis\n",
    "def checkPrecursors(model, reaction):\n",
    "    \n",
    "    if isinstance(reaction, str) == True:\n",
    "        reaction = model.reactions.get_by_id(reaction)\n",
    "    \n",
    "    model.objective = reaction\n",
    "    obj_val = max(model.optimize(objective_sense='maximize').objective_value, abs(model.optimize(objective_sense='minimize').objective_value))\n",
    "\n",
    "    if obj_val > 0.001:\n",
    "        print('Able to produce all precursors for this reaction.')\n",
    "        return None\n",
    "    \n",
    "    else:\n",
    "        reactants = reaction.reactants\n",
    "        check = 0\n",
    "        \n",
    "        for reactant in reactants:\n",
    "            sub_reactions = list(reactant.reactions)\n",
    "            \n",
    "            for sub_reaction in sub_reactions:\n",
    "                model.objective = sub_reaction\n",
    "                obj_val = max(model.optimize(objective_sense='maximize').objective_value, abs(model.optimize(objective_sense='minimize').objective_value))\n",
    "\n",
    "                if obj_val < 0.001 and reactant in sub_reaction.products:                    \n",
    "                    print('Cannot acquire ' + str(reactant.id) + ' via ' + str(sub_reaction.id))\n",
    "                elif obj_val < 0.001 and check < 1 and reactant in sub_reaction.reactants:\n",
    "                    print(str(reactant.id) + ' not produced in any reactions.')\n",
    "                    check += 1\n",
    "\n",
    "#------------------------------------------------------------------------------------#\n",
    "\n",
    "# Function to calculate doubling time from objective value\n",
    "def doubling(model):\n",
    "    \n",
    "    with model as m:\n",
    "        ov = m.slim_optimize()\n",
    "        if ov < 1e-6:\n",
    "            growth = 'No growth'\n",
    "        else:\n",
    "            growth = (1 / float(m.slim_optimize())) * 3600\n",
    "        \n",
    "            if growth < 60.0:\n",
    "                growth = str(round(growth, 3)) + ' minutes'\n",
    "            else:\n",
    "                growth = growth / 60\n",
    "                growth = str(round(growth, 3)) + ' hours'\n",
    "            \n",
    "    return growth\n",
    "\n",
    "\n",
    "# Rough transcriptomic integration\n",
    "def roughContextualize(model, transcript_profile, condition):\n",
    "    \n",
    "    orig_OV = model.optimize().objective_value\n",
    "\n",
    "    model_context = copy.deepcopy(model)\n",
    "    abundances = []\n",
    "    \n",
    "    with open(transcript_profile, 'r') as transcription:\n",
    "        transcript_dict = {}\n",
    "        abundances = []\n",
    "        header = transcription.readline().strip().split(',')\n",
    "        column = header.index(condition)\n",
    "        for line in transcription:\n",
    "            line = line.split(',')\n",
    "            transcript_dict[line[0]] = float(line[column])\n",
    "            abundances.append(float(line[column]))\n",
    "    \n",
    "    min_transcription = numpy.percentile(abundances, 50)\n",
    "    penalty_bound = 10\n",
    "    \n",
    "    hits = 0\n",
    "    fails = 0\n",
    "    for gene in list(model_context.genes):\n",
    "    \n",
    "        gene = gene.name\n",
    "    \n",
    "        try:\n",
    "            curr_rxns = list(model_context.genes.get_by_id(gene).reactions)\n",
    "            hits += 1\n",
    "        except KeyError:\n",
    "            fails += 1\n",
    "            continue\n",
    "    \n",
    "        try:\n",
    "            curr_transcription = transcript_dict[gene]\n",
    "        except KeyError:\n",
    "            continue\n",
    "\n",
    "        for reaction in curr_rxns:\n",
    "            curr_id = reaction.id\n",
    "        \n",
    "            if curr_transcription >= min_transcription:            \n",
    "                model_context.reactions.get_by_id(curr_id).lower_bound = -1000\n",
    "                model_context.reactions.get_by_id(curr_id).upper_bound = 1000\n",
    "            elif curr_transcription < min_transcription:\n",
    "                if model_context.reactions.get_by_id(curr_id).lower_bound != 0:\n",
    "                    model_context.reactions.get_by_id(curr_id).lower_bound = -penalty_bound\n",
    "                if model_context.reactions.get_by_id(curr_id).upper_bound != 0:\n",
    "                    model_context.reactions.get_by_id(curr_id).upper_bound = penalty_bound\n",
    "                  \n",
    "\n",
    "    #print('Gene hits across data types: ' + str(hits))\n",
    "    #print('KeyErrors across data types: ' + str(fails) + '\\n')\n",
    "    new_OV = model_context.optimize().objective_value\n",
    "    print('New objective value: ' + str(new_OV))\n",
    "    print('Contextualized doubling time: ' + doubling(new_OV))\n",
    "    \n",
    "    return(model_context)\n",
    "\n",
    "\n",
    "# Checks for availability of reactants of a given reaction\n",
    "def availability(model, target_rxn):\n",
    "\n",
    "    precursors = model.reactions.get_by_id(target_rxn).reactants\n",
    "    \n",
    "    total = 0\n",
    "    unsuccessful = set()\n",
    "    limited = set()\n",
    "    for precursor in precursors:\n",
    "        precursor_rxn = list(model.metabolites.get_by_id(precursor.id).reactions)\n",
    "        \n",
    "        for rxn in precursor_rxn:\n",
    "            if rxn.id == target_rxn:\n",
    "                continue\n",
    "            elif precursor in model.reactions.get_by_id(rxn.id).reactants:\n",
    "                \n",
    "                model.objective = rxn\n",
    "                obj_val = model.slim_optimize()\n",
    "                if obj_val < 1e-8:\n",
    "                    unsuccessful |= set([rxn.id])\n",
    "                    limited |= set([precursor.id])\n",
    "    \n",
    "    print('Failed reactions: ' + str(len(unsuccessful)))\n",
    "    print('Limiting reactants: ' + str(len(limited)))\n",
    "    \n",
    "    return unsuccessful, limited\n",
    "\n",
    "# Removes all metabolites in a list of metabolite ids and all reactions associated with them\n",
    "def removeAll(model, metabolite_list):\n",
    "    \n",
    "    new_model = copy.deepcopy(model)\n",
    "    \n",
    "    for metabolite in metabolite_list:\n",
    "        try:\n",
    "            metabolite = new_model.metabolites.get_by_id(metabolite)\n",
    "            new_model.remove_reactions(metabolite.reactions)\n",
    "            new_model.remove_metabolites([metabolite])\n",
    "        except KeyError:\n",
    "            print(metabolite + ' not found')\n",
    "            continue\n",
    "        \n",
    "    return new_model\n",
    "\n",
    "\n",
    "# Identify transport reactions (for any number compartments)\n",
    "def findTransports(model):\n",
    "    transporters = []\n",
    "    compartments = set(list(model.compartments))\n",
    "    if len(compartments) == 1:\n",
    "        raise Exception('Model only has one compartment!')\n",
    "    \n",
    "    for reaction in model.reactions:\n",
    "        \n",
    "        reactant_compartments = set([x.compartment for x in reaction.reactants])\n",
    "        product_compartments = set([x.compartment for x in reaction.products])\n",
    "        reactant_baseID = set([x.id.split('_')[0] for x in reaction.reactants])\n",
    "        product_baseID = set([x.id.split('_')[0] for x in reaction.products])\n",
    "        \n",
    "        if reactant_compartments == product_compartments and reactant_baseID != product_baseID:\n",
    "            continue\n",
    "        elif bool(compartments & reactant_compartments) == True and bool(compartments & product_compartments) == True:\n",
    "            transporters.append(reaction.id)\n",
    "        \n",
    "    return(transporters)\n",
    "\n",
    "# Removes a given percentage of reactions from a model, ignoring objective\n",
    "def generate_gaps(model, percentage=0.2, prune=False, ignore=[]):\n",
    "    number_to_remove = int(round(len(model.reactions) * percentage))\n",
    "    rxn_ids = [x.id for x in model.reactions]\n",
    "    random.shuffle(rxn_ids)\n",
    "    rxns_to_remove = rxn_ids[-number_to_remove:]\n",
    "        \n",
    "    for rxn in ignore:\n",
    "        try:\n",
    "            rxns_to_remove.remove(rxn)\n",
    "        except ValueError:\n",
    "            continue\n",
    "    \n",
    "    truncated_model = copy.deepcopy(model)\n",
    "    truncated_model.remove_reactions(rxns_to_remove)\n",
    "    \n",
    "    if prune == True:\n",
    "        unused_cpds = prune_unused_metabolites(truncated_model)\n",
    "    \n",
    "    print('Reactions removed: ' + str(len(rxns_to_remove)))\n",
    "    print('New objective value: ' + str(truncated_model.slim_optimize()))\n",
    "    return truncated_model, rxns_to_remove\n",
    "\n",
    "\n",
    "# Calculates the sum of fluxes for a given model\n",
    "def sum_of_fluxes(model):\n",
    "    \n",
    "    with model as m:\n",
    "        solution = m.optimize()\n",
    "        flux_sum = sum(list(solution.fluxes))\n",
    "        \n",
    "    return flux_sum\n",
    "\n",
    "\n",
    "# reports the reaction ids that are in 1 model but not the other\n",
    "def rxn_diff(model_1, model_2):\n",
    "    diff = set([str(x.id) for x in model_1.reactions]).difference([str(y.id) for y in model_2.reactions])\n",
    "    return list(diff)\n",
    "\n",
    "# Check for missing transport and exchange reactions\n",
    "def missingRxn(model, extracellular='extracellular'):\n",
    "\n",
    "    transporters = set(findTransports(model))\n",
    "    exchanges = set([x.id for x in model.exchanges])\n",
    "    \n",
    "    missing_exchanges = []\n",
    "    missing_transports = []\n",
    "    \n",
    "    for metabolite in model.metabolites:\n",
    "        if not metabolite.compartment == extracellular:\n",
    "            continue\n",
    "\n",
    "        curr_rxns = set([x.id for x in list(metabolite.reactions)])\n",
    "        \n",
    "        if not bool(curr_rxns & transporters):\n",
    "            missing_transports.append(metabolite.id)\n",
    "        if not bool(curr_rxns & exchanges):\n",
    "            missing_exchanges.append(metabolite.id)\n",
    "    \n",
    "    if len(missing_transports) != 0:\n",
    "        print(str(len(missing_transports)) + ' extracellular metabolites are missing transport reactions')\n",
    "    if len(missing_exchanges) != 0:\n",
    "        print(str(len(missing_exchanges)) + ' extracellular metabolites are missing exchange reactions')\n",
    "        \n",
    "    return missing_transports, missing_exchanges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in ModelSEED generated model\n",
    "cd630_PATRIC = cobra.io.read_sbml_model('data/PATRIC/272563.8.xml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set up counts for tracking changes\n",
    "init_rxn = [str(x.id) for x in cd630_PATRIC.reactions]\n",
    "\n",
    "new_genes = 0\n",
    "new_rxns = 0\n",
    "new_cpds = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "135 metabolic reactions not associated with genes\n",
      "1 reactions are mass imbalanced\n",
      "26 reactions are charge imbalanced\n",
      "Took 27 seconds to analyze Clostridium difficile 630\n"
     ]
    }
   ],
   "source": [
    "cd630_PATRIC_gaps, cd630_PATRIC_free, cd630_PATRIC_imbal, cd630_PATRIC_trans, cd630_PATRIC_exch = checkQuality(cd630_PATRIC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cobra/core/model.py:613 \u001b[1;31mUserWarning\u001b[0m: bio1 not in 272563.8\n"
     ]
    }
   ],
   "source": [
    "# General curation\n",
    "\n",
    "# Ensure oxygen exchange and transport exists, but leave anaerobic\n",
    "cpd00532_e = cobra.Metabolite(\n",
    "    'cpd00532_e',\n",
    "    formula='O2',\n",
    "    name='O2-', \n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00532_e, type='exchange', reaction_id='EX_cpd00532_e', lb=0., ub=0.)\n",
    "cpd00532_c = cd630_PATRIC.metabolites.get_by_id('cpd00532_c')\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "\n",
    "rxn09030_c = cobra.Reaction('rxn09030_c')\n",
    "rxn09030_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn09030_c.name = 'O2 diffusion'\n",
    "rxn09030_c.lower_bound = 0.\n",
    "rxn09030_c.upper_bound = 1000.\n",
    "rxn09030_c.add_metabolites({\n",
    "    cpd00532_e: -1.0,\n",
    "    cpd00532_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn09030_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Ensure free water diffusion across compartments\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05319_c').bounds = (-1000., 1000.)\n",
    "\n",
    "# Rework ammonia correctly\n",
    "cd630_PATRIC.metabolites.get_by_id('cpd00013_c').name = 'Ammonium'\n",
    "#cd630_PATRIC.metabolites.get_by_id('cpd00013_c').formula = 'H3N'\n",
    "\n",
    "# Remove old bio1 (generic Gram-positive Biomass function) and macromolecule demand reactions\n",
    "cd630_PATRIC.remove_reactions(['rxn13783_c', 'rxn13784_c', 'rxn13782_c', 'bio1', 'SK_cpd11416_c'])\n",
    "\n",
    "# Fix compartment labels to be more specific\n",
    "for index in cd630_PATRIC.metabolites:\n",
    "    index.notes = {'family': '', 'annotation': 'ModelSEED'}\n",
    "    \n",
    "    if index.compartment == 'c':\n",
    "        index.compartment = 'cytosol'\n",
    "    elif index.compartment == 'e':\n",
    "        index.compartment = 'extracellular'\n",
    "\n",
    "# Add structured note section to each gene\n",
    "for index in cd630_PATRIC.genes:\n",
    "    index.notes = {'annotation': 'PATRIC'}\n",
    "\n",
    "# Parse and label reaction types (non-metabolic)\n",
    "#likelihoods = pickle.load(open('data/PATRIC/272563.8.probs'))\n",
    "for reaction in cd630_PATRIC.reactions:\n",
    "    index.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "    \n",
    "    if reaction in cd630_PATRIC.boundary:\n",
    "        reaction.notes['type'] = 'exchange'\n",
    "        continue\n",
    "    substrates = [x.id.split('_')[-1] for x in reaction.products] + [x.id.split('_')[-1] for x in reaction.reactants]\n",
    "    if 'e' in substrates and 'c' in substrates:\n",
    "        reaction.notes['type'] = 'transport'\n",
    "    \n",
    "    #try:\n",
    "    #    reaction.notes['confidence'] = likelihoods[reaction.id]\n",
    "    #except:\n",
    "    #    continue\n",
    "\n",
    "# Save 2 transport reactions that should have genes associated \n",
    "cd630_PATRIC.reactions.get_by_id('rxn05669_c').gene_reaction_rule = 'CD630_12590' # Valine\n",
    "rxn05669_c = cd630_PATRIC.reactions.get_by_id('rxn05669_c')\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05244_c').gene_reaction_rule = 'CD630_12590' # Isoleucine\n",
    "rxn05244_c = cd630_PATRIC.reactions.get_by_id('rxn05244_c')\n",
    "# Remove previously gapfilled metabolic reactions\n",
    "cd630_PATRIC.remove_reactions(cd630_PATRIC_gaps)\n",
    "cd630_PATRIC.add_reactions([rxn05669_c, rxn05244_c]) # Re-add saved reactions\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Deal with totally orphaned reactions that possess GPRs\n",
    "\n",
    "#  Protein serine <=> [Protein]-D-serine\n",
    "cd630_PATRIC.remove_reactions(['rxn06870_c'])\n",
    "\n",
    "#  Ecgonine <=> Pseudoecgonine\n",
    "cd630_PATRIC.remove_reactions(['rxn04561_c'])\n",
    "\n",
    "#  Isopenicillin N <=> Penicillin N\n",
    "cd630_PATRIC.remove_reactions(['rxn02896_c'])\n",
    "\n",
    "#  Violaxanthin <=> Neoxanthin and 9-cis-Neoxanthin <=> 9-cis-Violaxanthin\n",
    "cd630_PATRIC.remove_reactions(['rxn04757_c', 'rxn04760_c'])\n",
    "\n",
    "# alpha-D-Hexose 1-phosphate <=> alpha-D-Hexose 6-phosphate (very generic)\n",
    "cd630_PATRIC.remove_reactions(['rxn02379_c'])\n",
    "\n",
    "# 3-Lysolecithin <=> 2-Lysolecithin\n",
    "cd630_PATRIC.remove_reactions(['rxn02391_c'])\n",
    "\n",
    "#-----------------------------------------------#\n",
    "\n",
    "# Leaving 3 orphaned reactions with possible ties to known C. difficile metbolism\n",
    "\n",
    "# Lantibiotic resistance\n",
    "# rxn06517_c;  S-Alkyl-L-cysteine S-oxide <=> Dehydroalanine + Alkyl sulfenate\n",
    "\n",
    "# Peptidoglycan synthesis\n",
    "# rxn02975_c;  CDPparatose <=> CDPtyvelose\n",
    "#rxn02737_c;  UDP-glucosamine <=> UDP-galactosamine\n",
    "\n",
    "# Actually part of arabinose utilization\n",
    "# araD; L-ribulose-5-phosphate 4-epimerase; CD630_22780\n",
    "# rxn01292_c;  L-Arabinose <=> L-Ribulose\n",
    "\n",
    "#-----------------------------------------------#\n",
    "\n",
    "# Remove reactions with GPRs that are not actually present or detract from metabolic tasks\n",
    "\n",
    "# L-galactose utilization\n",
    "#cd630_PATRIC.remove_reactions(['rxn07493_c'])\n",
    "\n",
    "# Selenium utiliztion\n",
    "cd630_PATRIC.remove_reactions(['rxn03378_c','rxn02569_c'])\n",
    "\n",
    "# Molybdenum import\n",
    "cd630_PATRIC.remove_reactions(['rxn05619_c'])\n",
    "#cpd11574_e = cd630_PATRIC.metabolites.get_by_id('cpd11574_e')\n",
    "#cpd11574_c = cd630_PATRIC.metabolites.get_by_id('cpd11574_c')\n",
    "#cd630_PATRIC.remove_metabolites([cpd11574_e, cpd11574_c]) # Molybdate\n",
    "\n",
    "# Viomycin\n",
    "cd630_PATRIC.remove_reactions(['rxn02576_c'])\n",
    "\n",
    "#-----------------------------------------------#\n",
    "\n",
    "# Useless macromolecules\n",
    "cd630_PATRIC.remove_reactions(['rxn06197_c'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Components from minimal defined minimal media\n",
    "# All need to feed directly into biomass\n",
    "\n",
    "# Karasawa T, Ikoma S, Yamakawa K, Nakamura S. (1995).\n",
    "# A defined growth medium for Clostridium difficile.\n",
    "# Microbiology 141(2):371-375. \n",
    "# doi:10.1099/13500872-141-2-371\n",
    "\n",
    "# D-Glucose\n",
    "# cpd00027_e via rxn05226_c\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "# Minerals (already in place)\n",
    "\n",
    "# cobalt; cpd00149_e (Co2+)\n",
    "# manganese; cpd00030_e (Mn2+)\n",
    "# magnesium, cpd00254_e (Mg)\n",
    "# sodium; cpd00971_e (Na+)\n",
    "# calcium; cpd00063_e (Ca2+)\n",
    "# iron; cpd10515_e (Fe2+)\n",
    "# potassium; cpd00205_e (K+)\n",
    "# chloride, cpd00099_e (Cl-)\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "# Amino acids\n",
    "\n",
    "# Handled in Stickland fermentation section\n",
    "# isoleucine\n",
    "# proline\n",
    "# valine\n",
    "# leucine\n",
    "\n",
    "# Metabolites used in active tranport reactions\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "# Cysteine\n",
    "cpd00084_e = cobra.Metabolite(\n",
    "    'cpd00084_e',\n",
    "    formula='C3H7NO2S',\n",
    "    name='L-Cysteine', \n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00084_e, type='exchange', reaction_id='EX_cpd00084_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00084_c = cd630_PATRIC.metabolites.get_by_id('cpd00084_c') # L-Cysteine (cytosolic)\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "\n",
    "CD630_07500 = cobra.Gene('CD630_07500')\n",
    "CD630_07500.name = 'ABC-type transport system, aminoacid-family extracellular solute-binding protein'\n",
    "CD630_07500.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "\n",
    "rxn05237_c = cobra.Reaction('rxn05237_c')\n",
    "rxn05237_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05237_c.gene_reaction_rule = 'CD630_07500'\n",
    "rxn05237_c.notes['citations'].append('Karasawa T, Ikoma S, Yamakawa K, Nakamura S. Microbiology 141(2):371-375 doi:10.1099/13500872-141-2-371')\n",
    "rxn05237_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05237_c.lower_bound = 0.\n",
    "rxn05237_c.upper_bound = 1000.\n",
    "rxn05237_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00084_e: -1.0,\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0, # H+\n",
    "    cpd00084_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05237_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Trypotophan\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05663_c').lower_bound = 0.\n",
    "\n",
    "# Methionine\n",
    "# Correct in model already\n",
    "\n",
    "#-------------------------------------------------------------------------#\n",
    "\n",
    "# Vitamins\n",
    "\n",
    "# Biotin (BIOT) - exchange and transport already in model\n",
    "# cpd00104_e -> rxn05223_c -> cpd00104_c\n",
    "cd630_PATRIC.remove_reactions([cd630_PATRIC.reactions.get_by_id('rxn08180_c')]) # remove sythesis step\n",
    "\n",
    "# Pantothenate\n",
    "cpd00644_e = cobra.Metabolite(\n",
    "    'cpd00644_e',\n",
    "    formula='C9H16NO5',\n",
    "    name='PAN', # pantothenate\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00644_e, type='exchange', reaction_id='EX_cpd00644_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00644_c = cd630_PATRIC.metabolites.get_by_id('cpd00644_c') # pantothenate (cytosolic)\n",
    "\n",
    "# transport via panT - http://jb.asm.org/content/191/1/42.full\n",
    "R03018_c = cobra.Reaction('R03018_c')\n",
    "R03018_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'KEGG', 'subsystem': 'ABC transporter'}\n",
    "R03018_c.gene_reaction_rule = '272563.8.peg.164'\n",
    "R03018_c.notes['citations'].append('Karasawa T, Ikoma S, Yamakawa K, Nakamura S. Microbiology 141(2):371-375 doi:10.1099/13500872-141-2-371')\n",
    "R03018_c.name = 'ATP:pantothenate 4-phosphotransferase'\n",
    "R03018_c.lower_bound = 0.\n",
    "R03018_c.upper_bound = 1000.\n",
    "R03018_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00644_e: -1.0,\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0, # H+\n",
    "    cpd00644_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R03018_c])\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "\n",
    "# Pyridoxine\n",
    "cpd00263_e = cobra.Metabolite(\n",
    "    'cpd00263_e',\n",
    "    formula='C8H11NO3',\n",
    "    name='Pyridoxol', # Pyridoxine\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00263_e, type='exchange', reaction_id='EX_cpd00263_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00263_c = cd630_PATRIC.metabolites.get_by_id('cpd00263_c') # \n",
    "\n",
    "COG3601_c = cobra.Reaction('COG3601_c')\n",
    "COG3601_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'COG', 'subsystem': 'ABC transporter'}\n",
    "COG3601_c.gene_reaction_rule = '272563.8.peg.164'\n",
    "COG3601_c.notes['citations'].append('Karasawa T, Ikoma S, Yamakawa K, Nakamura S. Microbiology 141(2):371-375 doi:10.1099/13500872-141-2-371')\n",
    "COG3601_c.name = 'Pyridoxine active transporter'\n",
    "COG3601_c.lower_bound = 0.\n",
    "COG3601_c.upper_bound = 1000.\n",
    "COG3601_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00263_e: -1.0,\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0, # H+\n",
    "    cpd00263_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([COG3601_c])\n",
    "new_rxns += 1\n",
    "new_cpds += 1\n",
    "# cpd00478_c\tPyridoxine phosphate (blocked) intracellular\n",
    "\n",
    "# Folate\n",
    "cpd00393_e = cd630_PATRIC.metabolites.get_by_id('cpd00393_e')\n",
    "cpd00393_c = cd630_PATRIC.metabolites.get_by_id('cpd00393_c')\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "\n",
    "rxn05255_c = cobra.Reaction('rxn05255_c')\n",
    "rxn05255_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn05255_c.name = 'Folate transport via proton simport'\n",
    "rxn05255_c.gene_reaction_rule = '272563.8.peg.164'\n",
    "rxn05255_c.lower_bound = 0.\n",
    "rxn05255_c.upper_bound = 1000.\n",
    "rxn05255_c.add_metabolites({\n",
    "    cpd00393_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00393_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05255_c])\n",
    "new_rxns += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# TCA cycle - fragmented in strain 630\n",
    "cpd00106_c = cd630_PATRIC.metabolites.get_by_id('cpd00106_c') # Fumarate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00130_c = cd630_PATRIC.metabolites.get_by_id('cpd00130_c') # L-Malate\n",
    "\n",
    "CD630_10030 = cobra.Gene('CD630_10030') # Define gene\n",
    "CD630_10030.name = 'fumA; fumarate hydratase subunit A [KO:K01677] [EC:4.2.1.2]'\n",
    "CD630_10030.notes = {'annotation': 'UniProt'}\n",
    "CD630_10040 = cobra.Gene('CD630_10040') # Define gene\n",
    "CD630_10040.name = 'fumA; fumarate hydratase subunit B [KO:K01678] [EC:4.2.1.2]'\n",
    "CD630_10040.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "new_genes += 1\n",
    "\n",
    "rxn00799_c = cobra.Reaction('rxn00799_c') \n",
    "rxn00799_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'TCA cycle'}\n",
    "rxn00799_c.gene_reaction_rule = 'CD630_10030 and CD630_10040'\n",
    "rxn00799_c.notes['citations'] += ['Dannheim, H., Will, S. E., Schomburg, D., & Neumann‐Schaal, M. (2017). Clostridioides difficile 630Δerm in silico and in vivo – quantitative growth and extensive polysaccharide secretion. FEBS Open Bio, 7(4), 602–615.', 'Dannheim, H., Riedel, T., Neumann-Schaal, M., Bunk, B., Schober, I., Spröer, C., & Schomburg, D. (2017). Manual curation and reannotation of the genomes of Clostridium difficile 630Δerm and C. difficile 630. Journal of Medical Microbiology.']\n",
    "rxn00799_c.name = 'Fumarate hydratase'\n",
    "rxn00799_c.lower_bound = -1000.\n",
    "rxn00799_c.upper_bound = 1000.\n",
    "rxn00799_c.add_metabolites({\n",
    "    cpd00106_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00130_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00799_c])\n",
    "new_rxns += 1\n",
    "\n",
    "cpd00041_c = cd630_PATRIC.metabolites.get_by_id('cpd00041_c') # L-aspartate\n",
    "cpd00036_c = cd630_PATRIC.metabolites.get_by_id('cpd00036_c') # Succinate\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd03470_c = cd630_PATRIC.metabolites.get_by_id('cpd03470_c') # Iminoaspartate\n",
    "\n",
    "CD630_23710 = cobra.Gene('CD630_23710') # Define gene\n",
    "CD630_23710.name = 'L-aspartate oxidase (Quinolinate synthetase B)'\n",
    "CD630_23710.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "\n",
    "rxn08168_c = cobra.Reaction('rxn08168_c')\n",
    "rxn08168_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'TCA cycle'}\n",
    "rxn08168_c.gene_reaction_rule = 'CD630_23710'\n",
    "rxn08168_c.notes['citations'] += ['Dannheim, H., Will, S. E., Schomburg, D., & Neumann‐Schaal, M. (2017). Clostridioides difficile 630Δerm in silico and in vivo – quantitative growth and extensive polysaccharide secretion. FEBS Open Bio, 7(4), 602–615.', 'Dannheim, H., Riedel, T., Neumann-Schaal, M., Bunk, B., Schober, I., Spröer, C., & Schomburg, D. (2017). Manual curation and reannotation of the genomes of Clostridium difficile 630Δerm and C. difficile 630. Journal of Medical Microbiology.']\n",
    "rxn08168_c.name = 'L-aspartate:fumarate oxidoreductase (deaminating)'\n",
    "rxn08168_c.lower_bound = -1000.\n",
    "rxn08168_c.upper_bound = 1000.\n",
    "rxn08168_c.add_metabolites({\n",
    "    cpd00041_c: -1.0,\n",
    "    cpd00106_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd03470_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08168_c])\n",
    "new_rxns += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Oxygen detoxification\n",
    "\n",
    "# Turn off O2 exchange (anaerobc environment)\n",
    "cd630_PATRIC.reactions.get_by_id('EX_cpd00007_e').lower_bound = 0. \n",
    "\n",
    "# Diffusion reaction: rxn05468_c\n",
    "\n",
    "# Remove extraneous aerobic reactions \n",
    "cd630_PATRIC.remove_reactions(['rxn00333_c', 'rxn00262_c', 'rxn00338_c', 'rxn02303_c'])\n",
    "# Keep 'rxn00206_c', 'rxn01600_c', and 'rxn00333_c' due to possible presence for detoxification\n",
    "\n",
    "# Add other detoxification reactions\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00023_c = cd630_PATRIC.metabolites.get_by_id('cpd00023_c') # L-Glutamate\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00024_c = cd630_PATRIC.metabolites.get_by_id('cpd00024_c') # 2-Oxoglutarate\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c') # Ammonia\n",
    "\n",
    "CD630_01790 = cobra.Gene('CD630_01790')\n",
    "CD630_01790.name = 'gluD; Glutamate dehydrogenase (NAD)'\n",
    "CD630_01790.notes = {'annotation': 'UniProt'}\n",
    "new_genes += 1\n",
    "\n",
    "rxn30075_c = cobra.Reaction('rxn30075_c')\n",
    "rxn30075_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'Oxygen detoxification'}\n",
    "rxn30075_c.name = 'Glutamate dehydrogenase (NAD)'\n",
    "rxn30075_c.gene_reaction_rule = 'CD630_01790'\n",
    "rxn30075_c.lower_bound = -1000.\n",
    "rxn30075_c.upper_bound = 1000.\n",
    "rxn30075_c.notes['citations'] += 'Dannheim, H., Riedel, T., Neumann-Schaal, M., Bunk, B., Schober, I., Spröer, C., & Schomburg, D. (2017). Manual curation and reannotation of the genomes of Clostridium difficile 630Δerm and C. difficile 630. Journal of Medical Microbiology.'\n",
    "rxn30075_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00023_c: -1.0,\n",
    "    cpd00024_c: 1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00013_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn30075_c])\n",
    "new_rxns += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Nucleic acids\n",
    "\n",
    "# Thymine biosynthesis\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00184_c = cd630_PATRIC.metabolites.get_by_id('cpd00184_c') # Thymidine\n",
    "cpd00509_c = cd630_PATRIC.metabolites.get_by_id('cpd00509_c') # deoxyribose-1-phosphate\n",
    "cpd00151_c = cobra.Metabolite(\n",
    "    'cpd00151_c',\n",
    "    formula='C5H6N2O2',\n",
    "    name='Thymine',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_12250 = cobra.Gene('CD630_12250') \n",
    "CD630_12250.name = 'pdp; Pyrimidine-nucleoside phosphorylase'\n",
    "CD630_12250.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn27328_c = cobra.Reaction('rxn27328_c')\n",
    "rxn27328_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'Thymine biosynthesis'}\n",
    "rxn27328_c.gene_reaction_rule = 'CD630_12250'\n",
    "rxn27328_c.name = 'Thymidine-orthophosphate deoxyribosyltransferase'\n",
    "rxn27328_c.lower_bound = -1000.\n",
    "rxn27328_c.upper_bound = 1000.\n",
    "rxn27328_c.add_metabolites({\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00184_c: -1.0,\n",
    "    cpd00509_c: 1.0,\n",
    "    cpd00151_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn27328_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Future addition:\n",
    "# Linkage between toxin production and purine biosynthesis in Clostridium difficile\n",
    "# http://jmm.microbiologyresearch.org/content/journal/jmm/10.1099/0022-1317-51-1-34#tab2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Host-derived glycans\n",
    "\n",
    "# Stiemsma, L. T., Turvey, S. E., & Finlay, B. B. (2014). An antibiotic-altered microbiota \n",
    "# provides fuel for the enteric foe. Cell Research, 24(1), 5–6. http://doi.org/10.1038/cr.2013.142\n",
    "\n",
    "# N-Acetyl neuraminic acid import\n",
    "cpd00232_e = cobra.Metabolite(\n",
    "    'cpd00232_e',\n",
    "    formula='C11H18NO9',\n",
    "    name='Neu5Ac',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00232_e, type='exchange', reaction_id='EX_cpd00232_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "cpd00232_c = cd630_PATRIC.metabolites.get_by_id('cpd00232_c') # Neu5Ac (intracellular)\n",
    "\n",
    "CD630_24170 = cobra.Gene('CD630_24170') # Define gene\n",
    "CD630_24170.name = 'PTS system, Sorbitol-like IIB component'\n",
    "CD630_24170.notes = {'annotation': 'UniProt'}\n",
    "CD630_24180 = cobra.Gene('CD630_24180') # Define gene\n",
    "CD630_24180.name = 'PTS system, Sorbitol-like IIC component'\n",
    "CD630_24180.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn29571_c = cobra.Reaction('rxn29571_c')\n",
    "rxn29571_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'PTS system'}\n",
    "rxn29571_c.gene_reaction_rule = 'CD630_24170 and CD630_24180'\n",
    "rxn29571_c.name = 'Neu5Ac H+ symporter'\n",
    "rxn29571_c.lower_bound = -1000.\n",
    "rxn29571_c.upper_bound = 1000.\n",
    "rxn29571_c.add_metabolites({\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00232_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00232_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn29571_c])\n",
    "new_rxns += 1\n",
    "# North RA, Kessans SA, Griffin MDW, et al. Cloning, expression, purification, crystallization \n",
    "# and preliminary X-ray diffraction analysis of N-acetylmannosamine-6-phosphate 2-epimerase \n",
    "# from methicillin-resistant Staphylococcus aureus. Acta Crystallogr F Struct Biol Commun. \n",
    "# 2014;70:650–655. doi: 10.1107/S2053230X14007171.\n",
    "\n",
    "# Sialic Acid Catabolism in Staphylococcus aureus\n",
    "# Michael E. Olson, Jessica M. King, Timothy L. Yahr and Alexander R. Horswill \n",
    "# http://jb.asm.org/content/195/8/1779.full\n",
    "cpd00492_e = cobra.Metabolite(\n",
    "    'cpd00492_e',\n",
    "    formula='C8H15NO6',\n",
    "    name='N-Acetyl-D-mannosamine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00492_e, type='exchange', reaction_id='EX_cpd00492_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00492_c = cd630_PATRIC.metabolites.get_by_id('cpd00492_c') # N-Acetylmannosamine\n",
    "\n",
    "rxn05486_c = cobra.Reaction('rxn05486_c')\n",
    "rxn05486_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'PTS system'}\n",
    "rxn05486_c.gene_reaction_rule = 'CD630_24170 and CD630_24180'\n",
    "rxn05486_c.name = 'N-Acetylmannosamine H+ symporter'\n",
    "rxn05486_c.lower_bound = 0.\n",
    "rxn05486_c.upper_bound = 1000.\n",
    "rxn05486_c.add_metabolites({\n",
    "    cpd00492_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00492_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05486_c])\n",
    "new_rxns += 1\n",
    "# # N-Acetyl-D-glucosamine -> GlcN conversion (nagA)\n",
    "cpd00293_c = cd630_PATRIC.metabolites.get_by_id('cpd00293_c') # N-Acetyl-D-glucosamine 6-phosphate\n",
    "cpd00288_c = cd630_PATRIC.metabolites.get_by_id('cpd00288_c') # D-Glucosamine 6-phosphate\n",
    "cpd00029_c = cd630_PATRIC.metabolites.get_by_id('cpd00029_c') # Acetate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # AH2O\n",
    "\n",
    "CD630_10100 = cobra.Gene('CD630_10100') # Define gene\n",
    "CD630_10100.name = 'nagA, N-Acetyl-D-glucosamine-6-phosphate amidohydrolase'\n",
    "CD630_10100.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn01484_c = cobra.Reaction('rxn01484_c') # GlcNAc 6-P deacetylase\n",
    "rxn01484_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01484_c.gene_reaction_rule = 'CD630_10100'\n",
    "rxn01484_c.name = 'N-Acetyl-D-glucosamine-6-phosphate amidohydrolase'\n",
    "rxn01484_c.lower_bound = 0.\n",
    "rxn01484_c.upper_bound = 1000.\n",
    "rxn01484_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00293_c: -1.0,\n",
    "    cpd00288_c: 1.0,\n",
    "    cpd00029_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01484_c])\n",
    "new_rxns += 1\n",
    "# Engevik MA, Yacyshyn MB, Engevik KA, et al. Human Clostridium difficile infection: \n",
    "# altered mucus production and composition. American Journal of Physiology - \n",
    "# Gastrointestinal and Liver Physiology. 2015;308(6):G510-G524. doi:10.1152/ajpgi.00091.2014.\n",
    "\n",
    "\n",
    "CD630_30130 = cobra.Gene('CD630_30130')\n",
    "CD630_30130.name = 'PTS system, IIC component'\n",
    "CD630_30130.notes = {'annotation': 'UniProt'}\n",
    "CD630_30140 = cobra.Gene('CD630_30134')\n",
    "CD630_30140.name = 'PTS system, IIB component'\n",
    "CD630_30140.notes = {'annotation': 'UniProt'}\n",
    "CD630_30150 = cobra.Gene('CD630_30130')\n",
    "CD630_30150.name = 'PTS system, IIA component'\n",
    "CD630_30150.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "cpd00751_e = cobra.Metabolite(\n",
    "    'cpd00751_e',\n",
    "    formula='C6H12O5',\n",
    "    name='L-Fucose',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00751_e, type='exchange', reaction_id='EX_cpd00751_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02262_c').remove_from_model() # remove incorrect transport reaction\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd01912_c = cd630_PATRIC.metabolites.get_by_id('cpd01912_c') # L-Fucose 1-phosphate\n",
    "\n",
    "rxn02262_c = cobra.Reaction('rxn02262_c')\n",
    "rxn02262_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn02262_c.gene_reaction_rule = '272563.8.peg.2384'\n",
    "rxn02262_c.name = 'ATP:6-deoxy-L-galactose 1-phosphotransferase'\n",
    "rxn02262_c.lower_bound = 0.\n",
    "rxn02262_c.upper_bound = 1000.\n",
    "rxn02262_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00751_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd01912_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn02262_c])\n",
    "new_rxns += 1\n",
    "# Fucose utilization\n",
    "cpd00334_c = cobra.Metabolite(\n",
    "    'cpd00334_c',\n",
    "    formula='C3H6O2',\n",
    "    name='L-Lactaldehyde',\n",
    "    compartment='cytosol')\n",
    "cpd00095_c = cd630_PATRIC.metabolites.get_by_id('cpd00095_c') # Glycerone-phosphate\n",
    "\n",
    "CD630_00450 = cobra.Gene('CD630_00450')\n",
    "CD630_00450.name = 'L-fuculose-phosphate aldolase [EC:4.1.2.17]'\n",
    "CD630_00450.notes = {'annotation': 'UniProt'}\n",
    "CD630_22770 = cobra.Gene('CD630_22770')\n",
    "CD630_22770.name = 'L-fuculose-phosphate aldolase [EC:4.1.2.17]'\n",
    "CD630_22770.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn01620_c = cobra.Reaction('rxn01620_c')\n",
    "rxn01620_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01620_c.gene_reaction_rule = 'CD630_00450 or CD630_22770'\n",
    "rxn01620_c.name = 'L-Fuculose-1-phosphate lactaldehyde-lyase'\n",
    "rxn01620_c.lower_bound = -1000.\n",
    "rxn01620_c.upper_bound = 1000.\n",
    "rxn01620_c.add_metabolites({\n",
    "    cpd01912_c: -1.0,\n",
    "    cpd00095_c: 1.0,\n",
    "    cpd00334_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01620_c])\n",
    "new_rxns += 1\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00428_c = cd630_PATRIC.metabolites.get_by_id('cpd00428_c') # 2-Oxopropanal \n",
    "\n",
    "CD630_35150 = cobra.Gene('CD630_35150')\n",
    "CD630_35150.name = 'bifunctional UDP-N-acetylglucosamine pyrophosphorylase / Glucosamine-1-phosphate N-acetyltransferase [EC:2.7.7.23 2.3.1.157]'\n",
    "CD630_35150.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn04937_c = cobra.Reaction('rxn04937_c')\n",
    "rxn04937_c.notes = {'citations': ['North, R. A., et al.. (2018). “Just a spoonful of sugar...”: import of sialic acid across bacterial cell membranes. Biophysical Reviews, 10(2), 219–227.'], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn04937_c.gene_reaction_rule = 'CD630_35150'\n",
    "rxn04937_c.name = 'Lactaldehyde:NAD+ oxidoreductase'\n",
    "rxn04937_c.lower_bound = -1000.\n",
    "rxn04937_c.upper_bound = 1000.\n",
    "rxn04937_c.add_metabolites({\n",
    "    cpd00334_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00428_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn04937_c])\n",
    "new_rxns += 1\n",
    "# Galactose omitted because of Nakamura, et al. (1982).\n",
    "\n",
    "# Mannose import and utilization already intact...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Carbohydrate import (fermentation steps taken care of in SCFA steps)\n",
    "# Nakamura, et al. (1982). Carbohydrate Fermentation by Clostridium difficile. Microbial. Immunol. 26 (2); 107-111.\n",
    "\n",
    "# Already included:\n",
    "# D-Fructose \tC6H12O6 \tcpd00082\n",
    "# D-Glucose \tC6H12O6 \tcpd00027\n",
    "# D-Mannitol \tC6H14O6 \tcpd00314\n",
    "# Sorbitol \tC6H14O6 \tcpd00588\n",
    "# D-Ribose \tC5H10O5 \tcpd00105 \n",
    "# Salicin \tC13H18O7 \tcpd01030\n",
    "# Trehalose (TRHL) \tC12H22O11 \tcpd00794_e \n",
    "\n",
    "cpd05161_e = cobra.Metabolite(\n",
    "    'cpd05161_e',\n",
    "    formula='C18H32O16',\n",
    "    name='Melezitose',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd05161_e, type='exchange', reaction_id='EX_cpd05161_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd05161_c = cobra.Metabolite(\n",
    "    'cpd05161_c',\n",
    "    formula='C18H32O16',\n",
    "    name='Melezitose',\n",
    "    compartment='cytosol')\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "\n",
    "rxn12619_c = cobra.Reaction('rxn12619_c')\n",
    "rxn12619_c.gene_reaction_rule = 'CD630_24170 and CD630_24180'\n",
    "rxn12619_c.name = 'Melezitose transport via proton symport'\n",
    "rxn12619_c.lower_bound = -1000.\n",
    "rxn12619_c.upper_bound = 1000.\n",
    "rxn12619_c.add_metabolites({\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd05161_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd05161_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn12619_c])\n",
    "new_rxns += 1\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "rxn18524_c = cobra.Reaction('rxn18524_c')\n",
    "rxn01620_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01620_c.notes['citations'].append('Nakamura, et al. (1982). Carbohydrate Fermentation by Clostridium difficile. Microbial. Immunol. 26 (2); 107-111.')\n",
    "rxn18524_c.name = 'ATP phosphohydrolase (disaccharide-importing)'\n",
    "rxn18524_c.gene_reaction_rule = 'CD630_30130 and CD630_30140 and CD630_30150'\n",
    "rxn18524_c.lower_bound = 0.\n",
    "rxn18524_c.upper_bound = 1000.\n",
    "rxn18524_c.add_metabolites({\n",
    "    cpd05161_e: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd05161_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn18524_c])\n",
    "new_rxns += 1\n",
    "T4BV06_CLODI = cobra.Gene('T4BV06_CLODI')\n",
    "T4BV06_CLODI.name = 'melA; alpha-galactosidase [EC 3.2.1.22]'\n",
    "T4BV06_CLODI.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "# Cleavage to glucose and turanose the latter of which is an isomer of sucrose. \n",
    "cpd20885_c = cobra.Metabolite(\n",
    "    'cpd20885_c',\n",
    "    formula='C12H22O11',\n",
    "    name='Turanose',\n",
    "    compartment='cytosol')\n",
    "cpd00027_c = cd630_PATRIC.metabolites.get_by_id('cpd00027_c') # D-Glucose\n",
    "\n",
    "R01103_c = cobra.Reaction('R01103_c')\n",
    "R01103_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01103_c.notes['citations'].append('Nakamura, et al. (1982). Carbohydrate Fermentation by Clostridium difficile. Microbial. Immunol. 26 (2); 107-111.')\n",
    "R01103_c.name = 'Raffinose galactohydrolase'\n",
    "R01103_c.gene_reaction_rule = 'T4BV06_CLODI'\n",
    "R01103_c.lower_bound = 0.\n",
    "R01103_c.upper_bound = 1000.\n",
    "R01103_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd05161_c: -1.0,\n",
    "    cpd00027_c: 1.0,\n",
    "    cpd20885_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01103_c])\n",
    "new_rxns += 1\n",
    "# Turanose isomerization to sucrose - spontaneous\n",
    "cpd00076_c = cd630_PATRIC.metabolites.get_by_id('cpd00076_c') # Sucrose\n",
    "ID010_c = cobra.Reaction('ID010_c')\n",
    "ID010_c.name = 'Turanose isomerization'\n",
    "ID010_c.lower_bound = -1000.\n",
    "ID010_c.upper_bound = 1000.\n",
    "ID010_c.add_metabolites({\n",
    "    cpd20885_c: -1.0,\n",
    "    cpd00076_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID010_c])\n",
    "new_rxns += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Amino acid exchange and transport\n",
    "\n",
    "# Branched chain - facilitated diffusion\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+ (extracellular)\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+ (cytosolic)\n",
    "\n",
    "# Leucine\n",
    "cpd00107_e = cobra.Metabolite(\n",
    "    'cpd00107_e',\n",
    "    formula='C6H13NO2',\n",
    "    name='L-Leucine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00107_e, type='exchange', reaction_id='EX_cpd00107_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00107_c = cd630_PATRIC.metabolites.get_by_id('cpd00107_c') # Leucine (cytosol)\n",
    "\n",
    "CD630_12590 = cobra.Gene('CD630_12590')\n",
    "CD630_12590.name = 'brnQ-1; Branched-chain amino acid transport system carrier protein'\n",
    "CD630_12590.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn05243_c = cobra.Reaction('rxn05243_c')\n",
    "rxn05243_c.gene_reaction_rule = 'CD630_12590'\n",
    "rxn05243_c.name = 'L-Leucine hydrogen symport'\n",
    "rxn05243_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn05243_c.notes['citations'] += 'Driessen, A. J., Ubbink-Kok, T., & Konings, W. N. (1988). Amino acid transport by membrane vesicles of an obligate anaerobic bacterium, Clostridium acetobutylicum. Journal of Bacteriology, 170(2), 817–820.'\n",
    "rxn05243_c.lower_bound = 0.\n",
    "rxn05243_c.upper_bound = 1000.\n",
    "rxn05243_c.add_metabolites({\n",
    "    cpd00107_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00107_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05243_c])\n",
    "new_rxns += 1\n",
    "# Valine\n",
    "# Extracellular exchange and transport in place, needs correct GPR\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05669_c').gene_reaction_rule = 'CD630_12590'\n",
    "\n",
    "# Isoleucine\n",
    "# Extracellular exchange and tranport in place, needs correct GPR\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05244_c').gene_reaction_rule = 'CD630_12590'\n",
    "\n",
    "#----------------#\n",
    "\n",
    "# Aromatic - active transport\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "CD630_08730 = cobra.Gene('CD630_08730')\n",
    "CD630_08730.name = 'putative ABC transport system substrate-binding protein'\n",
    "CD630_08730.notes = {'annotation': 'KEGG'}\n",
    "CD630_08760 = cobra.Gene('CD630_08760')\n",
    "CD630_08760.name = 'putative ABC transport system substrate-binding protein'\n",
    "CD630_08760.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "\n",
    "# Tyrosine\n",
    "cpd00069_e = cobra.Metabolite(\n",
    "    'cpd00069_e',\n",
    "    formula='C9H11NO3',\n",
    "    name='L-Tyrosine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00069_e, type='exchange', reaction_id='EX_cpd00069_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00069_c = cd630_PATRIC.metabolites.get_by_id('cpd00069_c') # Tyrosine (cytosolic)\n",
    "\n",
    "rxn18579_c = cobra.Reaction('rxn18579_c') \n",
    "rxn18579_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn18579_c.notes['citations'] += 'Steglich M., et al. (2018). Convergent Loss of ABC Transporter Genes From Clostridioides difficile Genomes Is Associated With Impaired Tyrosine Uptake and p-Cresol Production. Frontiers in Microbiology. doi 10.3389/fmicb.2018.00901.'\n",
    "rxn05243_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn18579_c.name = 'L-Tyrosine ABC transport'\n",
    "rxn18579_c.lower_bound = 0.\n",
    "rxn18579_c.upper_bound = 1000.\n",
    "rxn18579_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00069_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00069_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn18579_c])\n",
    "new_rxns += 1\n",
    "# Phenylalanine\n",
    "cpd00066_e = cobra.Metabolite(\n",
    "    'cpd00066_e',\n",
    "    formula='C9H11NO2',\n",
    "    name='L-Phenylalanine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00066_e, type='exchange', reaction_id='EX_cpd00066_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00066_c = cd630_PATRIC.metabolites.get_by_id('cpd00066_c')\n",
    "\n",
    "rxn18584_c = cobra.Reaction('rxn18584_c') \n",
    "rxn18584_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn18584_c.notes['citations'] += 'Steglich M., et al. (2018). Convergent Loss of ABC Transporter Genes From Clostridioides difficile Genomes Is Associated With Impaired Tyrosine Uptake and p-Cresol Production. Frontiers in Microbiology. doi 10.3389/fmicb.2018.00901.'\n",
    "rxn18584_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn18584_c.name = 'nonpolar-amino-acid-transporting ATPase'\n",
    "rxn18584_c.lower_bound = 0.\n",
    "rxn18584_c.upper_bound = 1000.\n",
    "rxn18584_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00066_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00066_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn18584_c])\n",
    "new_rxns += 1\n",
    "#----------------#\n",
    "\n",
    "# Hydrophobic - active transport\n",
    "\n",
    "# Proline (D and L)\n",
    "cpd00129_e = cobra.Metabolite(\n",
    "    'cpd00129_e',\n",
    "    formula='C5H9NO2',\n",
    "    name='L-Proline',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00129_e, type='exchange', reaction_id='EX_cpd00129_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00129_c = cd630_PATRIC.metabolites.get_by_id('cpd00129_c') # L-Proline\n",
    "# http://www.jbc.org/content/249/21/6965.full.pdf\n",
    "\n",
    "rxn05165_c = cobra.Reaction('rxn05165_c') \n",
    "rxn05165_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05165_c.notes['citations'] += ''\n",
    "rxn05165_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn05165_c.name = 'nonpolar-amino-acid-transporting ATPase'\n",
    "rxn05165_c.lower_bound = 0.\n",
    "rxn05165_c.upper_bound = 1000.\n",
    "rxn05165_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00129_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00129_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05165_c])\n",
    "new_rxns += 1\n",
    "cpd00567_e = cobra.Metabolite(\n",
    "    'cpd00567_e',\n",
    "    formula='C5H9NO2',\n",
    "    name='D-Proline',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00567_e, type='exchange', reaction_id='EX_cpd00567_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00567_c = cd630_PATRIC.metabolites.get_by_id('cpd00567_c') # D-Proline\n",
    "cpd00971_e = cd630_PATRIC.metabolites.get_by_id('cpd00971_e') # Na+ (extracellular)\n",
    "cpd00971_c = cd630_PATRIC.metabolites.get_by_id('cpd00971_c') # Na+ (cytosolic)\n",
    "\n",
    "# Facilitated diffusion for D isomer\n",
    "CD630_06820 = cobra.Gene('CD630_06820')\n",
    "CD630_06820.name = 'Putative sodium:solute symporter'\n",
    "CD630_06820.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "ENOG4108HXH_c = cobra.Reaction('ENOG4108HXH_c') \n",
    "ENOG4108HXH_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'EggNOG'}\n",
    "ENOG4108HXH_c.gene_reaction_rule = 'CD630_06820'\n",
    "ENOG4108HXH_c.name = 'D-proline sodium symporter'\n",
    "ENOG4108HXH_c.lower_bound = 0.\n",
    "ENOG4108HXH_c.upper_bound = 1000.\n",
    "ENOG4108HXH_c.add_metabolites({\n",
    "    cpd00567_e: -1.0,\n",
    "    cpd00971_e: -1.0,\n",
    "    cpd00567_c: 1.0,\n",
    "    cpd00971_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ENOG4108HXH_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Hydroxyproline\n",
    "cpd29317_e = cobra.Metabolite(\n",
    "    'cpd29317_e',\n",
    "    formula='C5H9NO3',\n",
    "    name='4-hydroxyproline',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd29317_e, type='exchange', reaction_id='EX_cpd29317_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd29317_c = cobra.Metabolite(\n",
    "    'cpd29317_c',\n",
    "    formula='C5H9NO3',\n",
    "    name='4-hydroxyproline',\n",
    "    compartment='cytosol')\n",
    "\n",
    "rxn05165_2_c = cobra.Reaction('rxn05165_2_c')\n",
    "rxn05165_2_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'EggNOG', 'subsystem': 'ABC transporter'}\n",
    "rxn05165_2_c.notes['citations'] += ''\n",
    "rxn05165_2_c.name = '4-hydroxyproline ABC transporter'\n",
    "rxn05165_2_c.lower_bound = 0.\n",
    "rxn05165_2_c.upper_bound = 1000.\n",
    "rxn05165_2_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd29317_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd29317_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05165_2_c])\n",
    "new_rxns += 1\n",
    "\n",
    "# Glycine\n",
    "cpd00033_e = cobra.Metabolite(\n",
    "    'cpd00033_e',\n",
    "    formula='C2H5NO2',\n",
    "    name='Glycine',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00033_e, type='exchange', reaction_id='EX_cpd00033_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "cpd00033_c = cd630_PATRIC.metabolites.get_by_id('cpd00033_c') # Glycine\n",
    "\n",
    "rxn10862_c = cobra.Reaction('rxn10862_c') \n",
    "rxn10862_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn10862_c.notes['citations'] += ''\n",
    "rxn10862_c.gene_reaction_rule = 'CD630_08730 or CD630_08760'\n",
    "rxn10862_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn10862_c.lower_bound = 0.\n",
    "rxn10862_c.upper_bound = 1000.\n",
    "rxn10862_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00033_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00033_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10862_c])\n",
    "new_rxns += 1\n",
    "# Hydroxyproline -> D-proline conversion\n",
    "cpd02431_c = cd630_PATRIC.metabolites.get_by_id('cpd02431_c') # 1-Pyrroline-5-carboxylate\n",
    "cpd00006_c = cd630_PATRIC.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd00005_c = cd630_PATRIC.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "\n",
    "CD630_25690 = cobra.Gene('CD630_25690')\n",
    "CD630_25690.name = '4-hydroxyproline epimerase'\n",
    "CD630_25690.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn32054_c = cobra.Reaction('rxn32054_c')\n",
    "rxn32054_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn32054_c.notes['citations'] += 'Watanabe, S., Tanimoto, Y., Nishiwaki, H., & Watanabe, Y. (2015). Identification and Characterization of Bifunctional Proline Racemase/Hydroxyproline Epimerase from Archaea: Discrimination of Substrates and Molecular Evolution. PLoS ONE, 10(3), e0120349.'\n",
    "rxn32054_c.name = '4-Hydroxy-L-proline:NADP+ 5-oxidoreductase'\n",
    "rxn10862_c.gene_reaction_rule = 'CD630_25690'\n",
    "rxn32054_c.lower_bound = 0.\n",
    "rxn32054_c.upper_bound = 1000.\n",
    "rxn32054_c.add_metabolites({\n",
    "    cpd00006_c: -1.0,\n",
    "    cpd29317_c: -1.0,\n",
    "    cpd00005_c: 1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd02431_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn32054_c])\n",
    "new_rxns += 1\n",
    "# rxn00931_c an rxn00929_c in place\n",
    "# NADP + L-Proline <=> NADPH + 2.0 H+ + 1-Pyrroline-5-carboxylate\n",
    "\n",
    "#------------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Non-Stickland substrates\n",
    "# Transport and exchange of non-auxotrophic amino acids\n",
    "\n",
    "# Tryptophan - facilitated diffusion already included\n",
    "\n",
    "#----------------#\n",
    "\n",
    "# Histidine\n",
    "cpd00119_e = cobra.Metabolite(\n",
    "    'cpd00119_e',\n",
    "    formula='C6H9N3O2',\n",
    "    name='L-Histidine',\n",
    "    compartment='extracellular')\n",
    "cpd00119_c = cd630_PATRIC.metabolites.get_by_id('cpd00119_c')\n",
    "\n",
    "CD630_17760 = cobra.Gene('CD630_17760')\n",
    "CD630_17760.name = ' amino acid family ABC transporter ATP-binding protein'\n",
    "CD630_17760.notes = {'annotation': 'NCBI'}\n",
    "\n",
    "rxn05164_c = cobra.Reaction('rxn05164_c') \n",
    "rxn05164_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05164_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05164_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05164_c.lower_bound = 0.\n",
    "rxn05164_c.upper_bound = 1000.\n",
    "rxn05164_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00119_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00119_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05164_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00119_e, type='exchange', reaction_id='EX_cpd00119_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Asparagine\n",
    "cpd00132_e = cobra.Metabolite(\n",
    "    'cpd00132_e',\n",
    "    formula='C4H8N2O3',\n",
    "    name='L-Asparagine',\n",
    "    compartment='extracellular')\n",
    "cpd00132_c = cd630_PATRIC.metabolites.get_by_id('cpd00132_c')\n",
    "\n",
    "rxn10144_c = cobra.Reaction('rxn10144_c') \n",
    "rxn10144_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn10144_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn10144_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn10144_c.lower_bound = 0.\n",
    "rxn10144_c.upper_bound = 1000.\n",
    "rxn10144_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00132_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00132_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10144_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00132_e, type='exchange', reaction_id='EX_cpd00132_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Serine\n",
    "cpd00054_e = cobra.Metabolite(\n",
    "    'cpd00054_e',\n",
    "    formula='C3H7NO3',\n",
    "    name='L-Serine',\n",
    "    compartment='extracellular')\n",
    "cpd00054_c = cd630_PATRIC.metabolites.get_by_id('cpd00054_c')\n",
    "\n",
    "rxn10883_c = cobra.Reaction('rxn10883_c') \n",
    "rxn10883_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn10883_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn10883_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn10883_c.lower_bound = 0.\n",
    "rxn10883_c.upper_bound = 1000.\n",
    "rxn10883_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00054_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00054_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10883_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00054_e, type='exchange', reaction_id='EX_cpd00054_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Threonine\n",
    "cpd00161_e = cobra.Metabolite(\n",
    "    'cpd00161_e',\n",
    "    formula='C4H9NO3',\n",
    "    name='L-Threonine',\n",
    "    compartment='extracellular')\n",
    "cpd00161_c = cd630_PATRIC.metabolites.get_by_id('cpd00161_c')\n",
    "\n",
    "rxn05169_c = cobra.Reaction('rxn05169_c') \n",
    "rxn05169_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05169_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05169_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05169_c.lower_bound = 0.\n",
    "rxn05169_c.upper_bound = 1000.\n",
    "rxn05169_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00161_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00161_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05169_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00161_e, type='exchange', reaction_id='EX_cpd00161_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "# Glutamine\n",
    "cpd00053_e = cobra.Metabolite(\n",
    "    'cpd00053_e',\n",
    "    formula='C5H10N2O3',\n",
    "    name='L-Glutamine',\n",
    "    compartment='extracellular')\n",
    "cpd00053_c = cd630_PATRIC.metabolites.get_by_id('cpd00053_c')\n",
    "\n",
    "rxn05155_c = cobra.Reaction('rxn05155_c') \n",
    "rxn05155_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05155_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05155_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05155_c.lower_bound = 0.\n",
    "rxn05155_c.upper_bound = 1000.\n",
    "rxn05155_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00053_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00053_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05155_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00053_e, type='exchange', reaction_id='EX_cpd00053_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "\n",
    "# Lysine transport\n",
    "cpd00039_c = cd630_PATRIC.metabolites.get_by_id('cpd00039_c')\n",
    "cpd00039_e = cd630_PATRIC.metabolites.get_by_id('cpd00039_e')\n",
    "\n",
    "rxn05151_c = cobra.Reaction('rxn05151_c') \n",
    "rxn05151_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05151_c.gene_reaction_rule = 'CD630_17760'\n",
    "rxn05151_c.name = 'ATP phosphohydrolase (polar-amino-acid-importing)'\n",
    "rxn05151_c.lower_bound = 0.\n",
    "rxn05151_c.upper_bound = 1000.\n",
    "rxn05151_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00039_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00039_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05151_c])\n",
    "new_rxns += 1\n",
    "\n",
    "#----------------#\n",
    "\n",
    "# Aspartate (same as Glutamate)\n",
    "cpd00041_e = cobra.Metabolite(\n",
    "    'cpd00041_e',\n",
    "    formula='C4H6NO4',\n",
    "    name='L-Aspartate',\n",
    "    compartment='extracellular')\n",
    "cpd00041_c = cd630_PATRIC.metabolites.get_by_id('cpd00041_c')\n",
    "cpd00971_e = cd630_PATRIC.metabolites.get_by_id('cpd00971_e') # NA+ (extracellular)\n",
    "cpd00971_c = cd630_PATRIC.metabolites.get_by_id('cpd00971_c') # NA+ (extracellular)\n",
    "\n",
    "rxn34357_c = cobra.Reaction('rxn34357_c') \n",
    "rxn34357_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'Cation symporter'}\n",
    "rxn34357_c.gene_reaction_rule = '272563.8.peg.2286 or ( 272563.8.peg.1830 or 272563.8.peg.2818 )'\n",
    "rxn34357_c.name = 'Na+/aspartate symport'\n",
    "rxn34357_c.lower_bound = -1000.\n",
    "rxn34357_c.upper_bound = 1000.\n",
    "rxn34357_c.add_metabolites({\n",
    "    cpd00041_e: -1.0,\n",
    "    cpd00971_e: -1.0,\n",
    "    cpd00041_c: 1.0,\n",
    "    cpd00971_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn34357_c])\n",
    "new_rxns += 1\n",
    "cd630_PATRIC.add_boundary(cpd00041_e, type='exchange', reaction_id='EX_cpd00041_e', lb=-1000.0, ub=1000.0)\n",
    "new_rxns += 1\n",
    "\n",
    "#35 so far...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd00430_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Phenylacetate exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f8473911310</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd00430_e <=> </p>\n",
       "                    <p style='text-align:right'>Phenylacetate <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd00430_e at 0x7f8473911310>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Stickland fermentation - Oxidative steps\n",
    "\n",
    "# Leucine\n",
    "# L-Leucine:2-oxoglutarate aminotransferase \trxn00806 in model already\n",
    "cpd00107_c = cd630_PATRIC.metabolites.get_by_id('cpd00107_c') # L-Leucine\n",
    "cpd00024_c = cd630_PATRIC.metabolites.get_by_id('cpd00024_c') # 2-Oxoglutarate\n",
    "cpd00023_c = cd630_PATRIC.metabolites.get_by_id('cpd00023_c') # L-Glutamate\n",
    "cpd00200_c = cd630_PATRIC.metabolites.get_by_id('cpd00200_c') # 4MOP (2-Oxoisocaproate)\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "cpd11621_c = cd630_PATRIC.metabolites.get_by_id('cpd11621_c') # Oxidizedferredoxin\n",
    "cpd00011_c = cd630_PATRIC.metabolites.get_by_id('cpd00011_c') # CO2\n",
    "cpd01882_c = cd630_PATRIC.metabolites.get_by_id('cpd01882_c') # Isovaleryl-CoA\n",
    "cpd11620_c = cd630_PATRIC.metabolites.get_by_id('cpd11620_c') # Reducedferredoxin\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "\n",
    "CD630_14930 = cobra.Gene('CD630_14930')\n",
    "CD630_14930.name = 'ilvE; branched-chain-amino-acid aminotransferase [EC 1.2.1.25 ]'\n",
    "CD630_14930.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn14152_c = cobra.Reaction('rxn14152_c') \n",
    "rxn14152_c.name = '4-methyl-2-oxopentanoate oxidoreductase'\n",
    "rxn14152_c.gene_reaction_rule = 'CD630_14930'\n",
    "rxn14152_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn14152_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn14152_c.lower_bound = 0.\n",
    "rxn14152_c.upper_bound = 1000.\n",
    "rxn14152_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00200_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd01882_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn14152_c])\n",
    "\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c') # NH3\n",
    "\n",
    "rxn00804_c = cobra.Reaction('rxn00804_c') \n",
    "rxn00804_c.name = 'Leucine aminotransferase'\n",
    "rxn00804_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00804_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn00804_c.lower_bound = 0.\n",
    "rxn00804_c.upper_bound = 1000.\n",
    "rxn00804_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00107_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00013_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00200_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00804_c])\n",
    "\n",
    "CD630_01120 = cobra.Gene('CD630_01120')\n",
    "CD630_01120.name = 'ptb; Phosphate butyryltransferase'\n",
    "CD630_01120.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "cpd05178_c = cobra.Metabolite(\n",
    "    'cpd05178_c',\n",
    "    formula='C5H10O2',\n",
    "    name='Isovalerate',\n",
    "    compartment='cytosol')\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "R01174_1_c = cobra.Reaction('R01174_1_c') \n",
    "R01174_1_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_1_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_1_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_1_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_1_c.lower_bound = 0.\n",
    "R01174_1_c.upper_bound = 1000.\n",
    "R01174_1_c.add_metabolites({\n",
    "    cpd01882_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd05178_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_1_c])\n",
    "\n",
    "cpd05178_e = cobra.Metabolite(\n",
    "    'cpd05178_e',\n",
    "    formula='C5H10O2',\n",
    "    name='Isovalerate',\n",
    "    compartment='extracellular')\n",
    "ID001_c = cobra.Reaction('ID001_c') \n",
    "ID001_c.name = 'Isovalerate diffusion'\n",
    "ID001_c.gene_reaction_rule = ''\n",
    "ID001_c.lower_bound = -1000.\n",
    "ID001_c.upper_bound = 1000.\n",
    "ID001_c.add_metabolites({\n",
    "    cpd05178_c: -1.0,\n",
    "    cpd05178_e: 1.0,\n",
    "    \n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID001_c])\n",
    "cd630_PATRIC.add_boundary(cpd05178_e, type='exchange', reaction_id='EX_cpd05178_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Valine\n",
    "cpd00156_c = cd630_PATRIC.metabolites.get_by_id('cpd00156_c') # L-Valine\n",
    "cpd00123_c = cd630_PATRIC.metabolites.get_by_id('cpd00123_c') # 3-Methyl-2-oxobutanoate (3mob)\n",
    "\n",
    "R01214_c = cobra.Reaction('R01214_c') \n",
    "R01214_c.name = 'L-valine aminotransferase'\n",
    "R01214_c.gene_reaction_rule = 'CD630_14930'\n",
    "R01214_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01214_c.lower_bound = -1000.\n",
    "R01214_c.upper_bound = 1000.\n",
    "R01214_c.add_metabolites({\n",
    "    cpd00156_c: -1.0,\n",
    "    cpd00024_c: -1.0,\n",
    "    cpd00023_c: 1.0,\n",
    "    cpd00123_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01214_c])\n",
    "\n",
    "cpd00481_c = cd630_PATRIC.metabolites.get_by_id('cpd00481_c') # Isobutyryl-CoA\n",
    "cpd11620_c = cd630_PATRIC.metabolites.get_by_id('cpd11620_c') # Reducedferredoxin\n",
    "cpd11621_c = cd630_PATRIC.metabolites.get_by_id('cpd11621_c') # Oxidizedferredoxin\n",
    "\n",
    "CD630_21990 = cobra.Gene('CD630_21990')\n",
    "CD630_21990.name = 'Putative ferredoxin/flavodoxin oxidoreductase,alpha subunit'\n",
    "CD630_21990.notes = {'annotation': 'UniProt'}\n",
    "CD630_21991 = cobra.Gene('CD630_21991')\n",
    "CD630_21991.name = 'Putative 4Fe-4S ferredoxin iron-sulfur binding domain-containing protein'\n",
    "CD630_21991.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn10618_c = cobra.Reaction('rxn10618_c') \n",
    "rxn10618_c.name = '2-oxoisovalerate ferredoxin reductase'\n",
    "rxn10618_c.gene_reaction_rule = 'CD630_21990 and CD630_21991'\n",
    "rxn10618_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10618_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn10618_c.lower_bound = 0.\n",
    "rxn10618_c.upper_bound = 1000.\n",
    "rxn10618_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00123_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00481_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10618_c])\n",
    "\n",
    "cd630_PATRIC.metabolites.get_by_id('cpd01711_c').formula = 'C4H8O2' # Isobutyrate (correct formula)\n",
    "cpd01711_c = cd630_PATRIC.metabolites.get_by_id('cpd01711_c')\n",
    "\n",
    "R01174_2_c = cobra.Reaction('R01174_2_c') \n",
    "R01174_2_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_2_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_2_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_2_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_2_c.lower_bound = 0.\n",
    "R01174_2_c.upper_bound = 1000.\n",
    "R01174_2_c.add_metabolites({\n",
    "    cpd00481_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd01711_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_2_c])\n",
    "\n",
    "cpd01711_e = cobra.Metabolite(\n",
    "    'cpd01711_e',\n",
    "    formula='C4H8O2',\n",
    "    name='Isobutyrate',\n",
    "    compartment='extracellular')\n",
    "ID002_c = cobra.Reaction('ID002_c') \n",
    "ID002_c.name = 'Isobutyrate diffusion'\n",
    "ID002_c.lower_bound = -1000.\n",
    "ID002_c.upper_bound = 1000.\n",
    "ID002_c.add_metabolites({\n",
    "    cpd01711_c: -1.0,\n",
    "    cpd01711_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID002_c])\n",
    "cd630_PATRIC.add_boundary(cpd01711_e, type='exchange', reaction_id='EX_cpd01711_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Isoleucine\n",
    "cpd00322_c = cd630_PATRIC.metabolites.get_by_id('cpd00322_c') # L-Isoleucine\n",
    "cpd00508_c = cd630_PATRIC.metabolites.get_by_id('cpd00508_c') # 3MOP (3-Methyl-2-oxopentanoate)\n",
    "\n",
    "rxn01575_c = cobra.Reaction('rxn01575_c') \n",
    "rxn01575_c.name = 'L-isoleucine aminotransferase'\n",
    "rxn01575_c.gene_reaction_rule = 'CD630_14930'\n",
    "rxn01575_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01575_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn01575_c.lower_bound = -1000.\n",
    "rxn01575_c.upper_bound = 1000.\n",
    "rxn01575_c.add_metabolites({\n",
    "    cpd00322_c: -1.0,\n",
    "    cpd00024_c: -1.0,\n",
    "    cpd00023_c: 1.0,\n",
    "    cpd00508_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01575_c])\n",
    "\n",
    "# Next reaction already in place:  rxn02269_c;  NAD + CoA + 3MOP --> NADH + CO2 + 2-Methylbutyryl-CoA\n",
    "\n",
    "cpd00760_c = cd630_PATRIC.metabolites.get_by_id('cpd00760_c') # 2-Methylbutyryl-CoA\n",
    "cpd19585_c = cobra.Metabolite(\n",
    "    'cpd19585_c',\n",
    "    formula='C5H10O2',\n",
    "    name='2-Methylbutyrate',\n",
    "    compartment='cytosol')\n",
    "R01174_3_c = cobra.Reaction('R01174_3_c') \n",
    "R01174_3_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_3_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_3_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_3_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_3_c.lower_bound = 0.\n",
    "R01174_3_c.upper_bound = 1000.\n",
    "R01174_3_c.add_metabolites({\n",
    "    cpd00760_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd19585_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_3_c])\n",
    "\n",
    "cpd19585_e = cobra.Metabolite(\n",
    "    'cpd19585_e',\n",
    "    formula='C5H10O2',\n",
    "    name='2-Methylbutyrate',\n",
    "    compartment='extracellular')\n",
    "ID003_c = cobra.Reaction('ID003_c') \n",
    "ID003_c.name = '2-Methylbutyrate diffusion'\n",
    "ID003_c.lower_bound = -1000.\n",
    "ID003_c.upper_bound = 1000.\n",
    "ID003_c.add_metabolites({\n",
    "    cpd19585_c: -1.0,\n",
    "    cpd19585_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID003_c])\n",
    "cd630_PATRIC.add_boundary(cpd19585_e, type='exchange', reaction_id='EX_cpd19585_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Tyrosine\n",
    "cpd00069_c = cd630_PATRIC.metabolites.get_by_id('cpd00069_c') # L-Tyrosine\n",
    "cpd00868_c = cd630_PATRIC.metabolites.get_by_id('cpd00868_c') # p-hydroxyphenylpyruvate\n",
    "#rxn00527_c = cobra.Reaction('rxn00527_c') \n",
    "#rxn00527_c.name = 'L-Tyrosine aminotransferase'\n",
    "#rxn00527_c.gene_reaction_rule = 'CD630_14930'\n",
    "#rxn00527_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "#rxn00527_c.notes['citations'] += 'Neumann-Schaal M., et al. (2015). Time-resolved amino acid uptake of Clostridium difficile 630Δerm and concomitant fermentation product and toxin formation. BMC Microbiology. 15(1); 1471-2180.'\n",
    "#rxn00527_c.lower_bound = -1000.\n",
    "#rxn00527_c.upper_bound = 1000.\n",
    "#rxn00527_c.add_metabolites({\n",
    "#    cpd00069_c: -1.0,\n",
    "#    cpd00024_c: -1.0,\n",
    "#    cpd00023_c: 1.0,\n",
    "#    cpd00868_c: 1.0\n",
    "#})\n",
    "#cd630_PATRIC.add_reactions([rxn00527_c])\n",
    "\n",
    "cpd03165_c = cobra.Metabolite(\n",
    "    'cpd03165_c',\n",
    "    formula='C29H39N7O18P3S',\n",
    "    name='4-Hydroxyphenylacetyl-CoA',\n",
    "    compartment='cytosol')\n",
    "rxn10563_c = cobra.Reaction('rxn10563_c') \n",
    "rxn10563_c.name = 'Indolepyruvate ferredoxin oxidoreductase (hydroxyphenylpyvurate)'\n",
    "rxn10563_c.gene_reaction_rule = 'CD630_21990 and CD630_21991'\n",
    "rxn10563_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10563_c.notes['citations'] += 'Neumann-Schaal M., et al. (2015). Time-resolved amino acid uptake of Clostridium difficile 630Δerm and concomitant fermentation product and toxin formation. BMC Microbiology. 15(1); 1471-2180.'\n",
    "rxn10563_c.lower_bound = 0.\n",
    "rxn10563_c.upper_bound = 1000.\n",
    "rxn10563_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00868_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd03165_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10563_c])\n",
    "\n",
    "cpd00489_c = cobra.Metabolite(\n",
    "    'cpd00489_c',\n",
    "    formula='C8H8O3',\n",
    "    name='4-Hydroxyphenylacetate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "R01174_4_c = cobra.Reaction('R01174_4_c') \n",
    "R01174_4_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_4_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_4_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_4_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_4_c.lower_bound = 0.\n",
    "R01174_4_c.upper_bound = 1000.\n",
    "R01174_4_c.add_metabolites({\n",
    "    cpd03165_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd00489_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_4_c])\n",
    "\n",
    "cpd00489_e = cobra.Metabolite(\n",
    "    'cpd00489_e',\n",
    "    formula='C8H8O3',\n",
    "    name='4-Hydroxyphenylacetate',\n",
    "    compartment='extracellular')\n",
    "ID004_c = cobra.Reaction('ID004_c') \n",
    "ID004_c.name = '4-Hydroxyphenylacetate diffusion'\n",
    "ID004_c.lower_bound = -1000.\n",
    "ID004_c.upper_bound = 1000.\n",
    "ID004_c.add_metabolites({\n",
    "    cpd00489_c: -1.0,\n",
    "    cpd00489_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID004_c])\n",
    "cd630_PATRIC.add_boundary(cpd00489_e, type='exchange', reaction_id='EX_cpd00489_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# p-Cresol formation and secretion\n",
    "cpd01042_c = cobra.Metabolite(\n",
    "    'cpd01042_c',\n",
    "    formula='C7H8O',\n",
    "    name='p-Cresol',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_01530 = cobra.Gene('CD630_01530')\n",
    "CD630_01530.name = '4-hydroxyphenylacetate decarboxylase large subunit [EC:4.1.1.83]'\n",
    "CD630_01530.notes = {'annotation': 'KEGG'}\n",
    "CD630_01540 = cobra.Gene('CD630_01540')\n",
    "CD630_01540.name = '4-hydroxyphenylacetate decarboxylase small subunit [EC:4.1.1.83]'\n",
    "CD630_01540.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn05061_c = cobra.Reaction('rxn05061_c') \n",
    "rxn05061_c.name = '4-hydroxyphenylacetate carboxy-lyase'\n",
    "rxn05061_c.gene_reaction_rule = 'CD630_01530 and CD630_01540'\n",
    "rxn05061_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05061_c.notes['citations'] += 'Selmer, T., & Andrei, P. I. (2001). p-Hydroxyphenylacetate decarboxylase from Clostridium difficile. A novel glycyl radical enzyme catalysing the formation of p-cresol. European Journal of Biochemistry, 268(5), 1363–1372.'\n",
    "rxn05061_c.lower_bound = -1000.\n",
    "rxn05061_c.upper_bound = 1000.\n",
    "rxn05061_c.add_metabolites({\n",
    "    cpd00489_c: -1.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd01042_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05061_c])\n",
    "\n",
    "cpd01042_e = cobra.Metabolite(\n",
    "    'cpd01042_e',\n",
    "    formula='C7H8O',\n",
    "    name='p-Cresol',\n",
    "    compartment='extracellular')\n",
    "ID005_c = cobra.Reaction('ID005_c') \n",
    "ID005_c.name = 'p-Cresol diffusion'\n",
    "ID005_c.lower_bound = -1000.\n",
    "ID005_c.upper_bound = 1000.\n",
    "ID005_c.add_metabolites({\n",
    "    cpd01042_c: -1.0,\n",
    "    cpd01042_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID005_c])\n",
    "cd630_PATRIC.add_boundary(cpd01042_e, type='exchange', reaction_id='EX_cpd01042_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Phenylalanine\n",
    "cpd00066_c = cd630_PATRIC.metabolites.get_by_id('cpd00066_c') # L-Phenylalanine\n",
    "cpd00143_c = cd630_PATRIC.metabolites.get_by_id('cpd00143_c') # Phenylpyruvate\n",
    "rxn21134_c = cobra.Reaction('rxn21134_c') \n",
    "rxn21134_c.name = 'L-phenylalanine aminotransferase'\n",
    "rxn21134_c.gene_reaction_rule = 'CD630_14930'\n",
    "rxn21134_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn21134_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn21134_c.lower_bound = -1000.\n",
    "rxn21134_c.upper_bound = 1000.\n",
    "rxn21134_c.add_metabolites({\n",
    "    cpd00066_c: -1.0,\n",
    "    cpd00024_c: -1.0,\n",
    "    cpd00023_c: 1.0,\n",
    "    cpd00143_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn21134_c])\n",
    "\n",
    "cpd00452_c = cobra.Metabolite(\n",
    "    'cpd00452_c',\n",
    "    formula='C29H38N7O17P3S',\n",
    "    name='Phenylacetyl-CoA',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_23800 = cobra.Gene('CD630_23800')\n",
    "CD630_23800.name = 'Indole pyruvate ferredoxin/flavodoxin oxidoreductase'\n",
    "CD630_23800.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn10562_c = cobra.Reaction('rxn10562_c') \n",
    "rxn10562_c.name = 'indolepyruvate ferredoxin oxidoreductase'\n",
    "rxn10562_c.gene_reaction_rule = 'CD630_23800'\n",
    "rxn10562_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10562_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "rxn10562_c.lower_bound = 0.\n",
    "rxn10562_c.upper_bound = 1000.\n",
    "rxn10562_c.add_metabolites({\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00143_c: -1.0,\n",
    "    cpd11621_c: -2.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00452_c: 1.0,\n",
    "    cpd11620_c: 2.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10562_c])\n",
    "\n",
    "cpd00430_c = cobra.Metabolite(\n",
    "    'cpd00430_c',\n",
    "    formula='C8H8O2',\n",
    "    name='Phenylacetate', # PACT in ModelSEED\n",
    "    compartment='cytosol')\n",
    "\n",
    "R01174_5_c = cobra.Reaction('R01174_5_c') \n",
    "R01174_5_c.name = 'Phosphate butyryltransferase'\n",
    "R01174_5_c.gene_reaction_rule = 'CD630_01120'\n",
    "R01174_5_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R01174_5_c.notes['citations'] += 'Kim J, Darley D, & Buckel W. (2005). 2-Hydroxyisocaproyl-CoA dehydratase and its activator from Clostridium difficile. FEBS J. 272; 550-561.'\n",
    "R01174_5_c.lower_bound = 0.\n",
    "R01174_5_c.upper_bound = 1000.\n",
    "R01174_5_c.add_metabolites({\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00452_c: -1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd00430_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R01174_5_c])\n",
    "\n",
    "cpd00430_e = cobra.Metabolite(\n",
    "    'cpd00430_e',\n",
    "    formula='C8H8O2',\n",
    "    name='Phenylacetate', # PACT\n",
    "    compartment='extracellular')\n",
    "\n",
    "rxn11336_c = cobra.Reaction('rxn11336_c') \n",
    "rxn11336_c.name = 'Phenylacetate diffusion'\n",
    "rxn11336_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn11336_c.lower_bound = -1000.\n",
    "rxn11336_c.upper_bound = 1000.\n",
    "rxn11336_c.add_metabolites({\n",
    "    cpd00430_c: -1.0,\n",
    "    cpd00430_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn11336_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00430_e, type='exchange', reaction_id='EX_cpd00430_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd03170_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>4-Hydroxymandelate exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f84738b8dd0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd03170_e <=> </p>\n",
       "                    <p style='text-align:right'>4-Hydroxymandelate <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd03170_e at 0x7f84738b8dd0>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Stickland Reactions - Reductive steps\n",
    "\n",
    "# Frequently needed metabolites\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c') # NH3\n",
    "cpd11421_c = cd630_PATRIC.metabolites.get_by_id('cpd11421_c') # trdrd\n",
    "cpd11420_c = cd630_PATRIC.metabolites.get_by_id('cpd11420_c') # trdox\n",
    "cpd00196_c = cd630_PATRIC.metabolites.get_by_id('cpd00196_c') # Acetylphosphate\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00011_c = cd630_PATRIC.metabolites.get_by_id('cpd00011_c') # CO2\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Proline\n",
    "cpd00129_c = cd630_PATRIC.metabolites.get_by_id('cpd00129_c') # L-Proline\n",
    "cd630_PATRIC.remove_reactions(['rxn02029_c']) # This is incorrect for C. difficile metabolism\n",
    "cpd00339_e = cobra.Metabolite(\n",
    "    'cpd00339_e',\n",
    "    formula='C5H11NO2',\n",
    "    name='5-Aminopentanoate',\n",
    "    compartment='extracellular')\n",
    "# D-Proline reductase (C. difficile)\n",
    "cpd00339_c = cd630_PATRIC.metabolites.get_by_id('cpd00339_c') # 5-Aminopentanoate\n",
    "rxn20606_c = cobra.Reaction('rxn20606_c') \n",
    "rxn20606_c.name = 'D-proline reductase'\n",
    "rxn20606_c.gene_reaction_rule = 'CD630_32440'\n",
    "rxn20606_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn20606_c.notes['citations'] += 'Bouillaut, L., Self, W. T., & Sonenshein, A. L. (2013). Proline-Dependent Regulation of Clostridium difficile Stickland Metabolism. Journal of Bacteriology, 195(4), 844–854.'\n",
    "rxn20606_c.notes['citations'] += 'Jackson, S., Calos, M., Myers, A., & Self, W. T. (2006). Analysis of proline reduction in the nosocomial pathogen Clostridium difficile. Journal of Bacteriology.'\n",
    "rxn20606_c.lower_bound = 0.\n",
    "rxn20606_c.upper_bound = 1000.\n",
    "rxn20606_c.add_metabolites({\n",
    "    cpd00567_c: -1.0,\n",
    "    cpd00004_c: -1.0,\n",
    "    cpd00067_c: -1.0, # H+\n",
    "    cpd00003_c: 1.0,\n",
    "    cpd00339_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn20606_c])\n",
    "# 5-Aminopentanoate (aka 5-aminovalerate) export\n",
    "rxn12566_c = cobra.Reaction('rxn12566_c') \n",
    "rxn12566_c.name = '5-Aminopentanoate transport via proton symport'\n",
    "rxn12566_c.lower_bound = 0.\n",
    "rxn12566_c.upper_bound = 1000.\n",
    "rxn12566_c.add_metabolites({\n",
    "    cpd00339_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00067_e: 1.0,\n",
    "    cpd00339_e: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn12566_c])\n",
    "cd630_PATRIC.add_boundary(cpd00339_e, type='exchange', reaction_id='EX_cpd00339_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Glycine \n",
    "cpd00033_c = cd630_PATRIC.metabolites.get_by_id('cpd00033_c') # Glycine\n",
    "rxn11676_c = cobra.Reaction('rxn11676_c') \n",
    "rxn11676_c.name = 'Glycine reductase'\n",
    "rxn11676_c.gene_reaction_rule = 'CD630_23540'\n",
    "rxn11676_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn11676_c.notes['citations'] += 'Bouillaut, L., Self, W. T., & Sonenshein, A. L. (2013). Proline-Dependent Regulation of Clostridium difficile Stickland Metabolism. Journal of Bacteriology, 195(4), 844–854.'\n",
    "rxn11676_c.lower_bound = -1000.\n",
    "rxn11676_c.upper_bound = 1000.\n",
    "rxn11676_c.add_metabolites({\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00033_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd11421_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd00013_c: 1.0,\n",
    "    cpd00196_c: 1.0,\n",
    "    cpd11420_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn11676_c])\n",
    "\n",
    "# Acetylphosphate --> Acetate conversion intact\n",
    "# ATP:acetate phosphotransferase; rxn00225; 1 ADP [0] + 1 Acetylphosphate [0] -> 1 ATP [0] + 1 Acetate [0] \n",
    "cd630_PATRIC.reactions.get_by_id('rxn00225_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00225_c').notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00225_c').notes['citations'] += 'Bouillaut, L., Self, W. T., & Sonenshein, A. L. (2013). Proline-Dependent Regulation of Clostridium difficile Stickland Metabolism. Journal of Bacteriology, 195(4), 844–854.'\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Leucine\n",
    "cpd00200_c = cd630_PATRIC.metabolites.get_by_id('cpd00200_c') # 4-Methyl-2-oxopentanoate - 4MOP (2-Oxoisocaproate)\n",
    "cpd01646_c = cd630_PATRIC.metabolites.get_by_id('cpd01646_c') # 2-Isopropylmalate == 2-Hydroxyisocaproate\n",
    "\n",
    "CD630_03940 = cobra.Gene('CD630_03940')\n",
    "CD630_03940.name = 'ldhA; D-lactate dehydrogenase [EC:1.1.1.28]'\n",
    "CD630_03940.notes = {'annotation': 'PATRIC'}\n",
    "\n",
    "rxn13022_c = cobra.Reaction('rxn13022_c') \n",
    "rxn13022_c.name = '3-isopropylmalate dehydrogenase'\n",
    "rxn13022_c.gene_reaction_rule = 'CD630_03940'\n",
    "rxn13022_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn13022_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2006). Characterization of (R)-2-Hydroxyisocaproate Dehydrogenase and a Family III Coenzyme A Transferase Involved in Reduction of l-Leucine to Isocaproate by Clostridium difficile. Applied and Environmental Microbiology, 72(9), 6062–6069.'\n",
    "rxn13022_c.lower_bound = -1000.\n",
    "rxn13022_c.upper_bound = 1000.\n",
    "rxn13022_c.add_metabolites({\n",
    "    cpd00004_c: -1.0,\n",
    "    cpd00011_c: -1.0,\n",
    "    cpd00200_c: -1.0,\n",
    "    cpd00003_c: 1.0,\n",
    "    cpd01646_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn13022_c])\n",
    "\n",
    "CD630_03950 = cobra.Gene('CD630_03950')\n",
    "CD630_03950.name = 'hadA; Isocaprenoyl-CoA:2-hydroxyisocaproate CoA-transferase'\n",
    "CD630_03950.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "cpd23040_c = cobra.Metabolite(\n",
    "    'cpd23040_c',\n",
    "    formula='C27H42N7O18P3S',\n",
    "    name='2-hydroxyisocaproyl-CoA',\n",
    "    compartment='cytosol')\n",
    "# C27H46N7O18P3S would be an uncharged formula, otherwise its -4 charge\n",
    "# C27H42N7O18P3S (original)\n",
    "\n",
    "C21400_c = cobra.Metabolite(\n",
    "    'C21400_c',\n",
    "    formula='C27H46N7O17P3S',\n",
    "    name='Isocaproyl-CoA',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_03980 = cobra.Gene('CD630_03980')\n",
    "CD630_03980.name = 'hadC; (R)-2-hydroxyisocaproyl-CoA dehydratase beta subunit [EC:4.2.1.157]'\n",
    "CD630_03980.notes = {'annotation': 'UniProt'}\n",
    "CD630_03970 = cobra.Gene('CD630_03970')\n",
    "CD630_03970.name = 'hadB; (R)-2-hydroxyisocaproyl-CoA dehydratase alpha subunit [EC:4.2.1.157]'\n",
    "CD630_03970.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "K20025_c = cobra.Reaction('K20025_c') \n",
    "K20025_c.name = '(R)-2-hydroxyisocaproyl-CoA dehydratase'\n",
    "K20025_c.gene_reaction_rule = 'CD630_03980 and CD630_03970'\n",
    "K20025_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "K20025_c.notes['citations'] += 'Britz ML & Wilkinson RG (1982). \"Leucine dissimilation to isovaleric and isocaproic acids by cell suspensions of amino acid fermenting anaerobes: the Stickland reaction revisited.\" Can J Microbiol. 28(3);291-300.'\n",
    "K20025_c.lower_bound = 0.\n",
    "K20025_c.upper_bound = 1000.\n",
    "K20025_c.add_metabolites({\n",
    "    cpd00067_c: -6.0,\n",
    "    cpd23040_c: -1.0,\n",
    "    C21400_c: 1.0,\n",
    "    cpd00001_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([K20025_c])   \n",
    "\n",
    "C21399_c = cobra.Metabolite(\n",
    "    'C21399_c',\n",
    "    formula='C6H12O2',\n",
    "    name='Isocaproate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "cpd00047_c = cd630_PATRIC.metabolites.get_by_id('cpd00047_c') # Formate\n",
    "\n",
    "R11462_c = cobra.Reaction('R11462_c') \n",
    "R11462_c.name = '4-methylpentanoyl-CoA:(R)-2-hydroxy-4-methylpentanoate CoA-transferase [EC:2.8.3.24]'\n",
    "R11462_c.gene_reaction_rule = 'CD630_03950'\n",
    "R11462_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R11462_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2006). Characterization of (R)-2-Hydroxyisocaproate Dehydrogenase and a Family III Coenzyme A Transferase Involved in Reduction of l-Leucine to Isocaproate by Clostridium difficile. Applied and Environmental Microbiology, 72(9), 6062–6069.'\n",
    "R11462_c.lower_bound = -1000.\n",
    "R11462_c.upper_bound = 1000.\n",
    "R11462_c.add_metabolites({\n",
    "    cpd00067_c: -1.0,\n",
    "    C21400_c: -1.0,\n",
    "    cpd01646_c: -1.0,\n",
    "    C21399_c: 1.0,\n",
    "    cpd23040_c: 1.0,\n",
    "    cpd00047_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R11462_c])                                                                           \n",
    "\n",
    "CD630_03960 = cobra.Gene('CD630_03960')\n",
    "CD630_03960.name = 'hadI; 2-hydroxyisocaproyl-CoA dehydratase activator'\n",
    "CD630_03960.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "C21090_c = cobra.Metabolite(\n",
    "    'C21090_c',\n",
    "    formula='C27H44N7O17P3S',\n",
    "    name='Isocaprenoyl-CoA',\n",
    "    compartment='cytosol')\n",
    "\n",
    "R11076_c = cobra.Reaction('R11076_c') \n",
    "R11076_c.name = '(R)-2-hydroxyisocaproyl-CoA dehydratase'\n",
    "R11076_c.gene_reaction_rule = 'CD630_03960 and CD630_03970 and CD630_03980'\n",
    "R11076_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R11076_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2006). Characterization of (R)-2-Hydroxyisocaproate Dehydrogenase and a Family III Coenzyme A Transferase Involved in Reduction of l-Leucine to Isocaproate by Clostridium difficile. Applied and Environmental Microbiology, 72(9), 6062–6069.'\n",
    "R11076_c.lower_bound = -1000.\n",
    "R11076_c.upper_bound = 1000.\n",
    "R11076_c.add_metabolites({\n",
    "    cpd00067_c: -4.0,\n",
    "    cpd23040_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    C21090_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R11076_c])                                                                           \n",
    "\n",
    "C21399_e = cobra.Metabolite(\n",
    "    'C21399_e',\n",
    "    formula='C6H12O2',\n",
    "    name='Isocaproate',\n",
    "    compartment='extracellular')\n",
    "\n",
    "ID006_c = cobra.Reaction('ID006_c') \n",
    "ID006_c.name = 'Isocaproate diffusion' \n",
    "ID006_c.lower_bound = -1000.\n",
    "ID006_c.upper_bound = 1000.\n",
    "ID006_c.add_metabolites({\n",
    "    C21399_e: -1.0,\n",
    "    C21399_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID006_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(C21399_e, type='exchange', reaction_id='EX_C21399_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Phenylalanine\n",
    "cpd00143_c = cd630_PATRIC.metabolites.get_by_id('cpd00143_c') # Phenylpyruvate\n",
    "cpd07326_c = cobra.Metabolite(\n",
    "    'cpd07326_c',\n",
    "    formula='C9H7O2',\n",
    "    name='Cinnamic acid',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_32370_1 = cobra.Gene('CD630_32370_1')\n",
    "CD630_32370_1.name = '3-phenylpropionate/trans-cinnamate dioxygenase subunit alpha [EC:1.14.12.19]'\n",
    "CD630_32370_1.notes = {'annotation': 'KEGG'}\n",
    "CD630_32370_2 = cobra.Gene('CD630_32370_2')\n",
    "CD630_32370_2.name = '3-phenylpropionate/trans-cinnamate dioxygenase subunit beta [EC:1.14.12.19]'\n",
    "CD630_32370_2.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "R06782_c = cobra.Reaction('R06782_c') \n",
    "R06782_c.name = '3-phenylpropanoate dioxygenase'\n",
    "R06782_c.gene_reaction_rule = 'CD630_32370_1 and CD630_32370_2'\n",
    "R06782_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'KEGG'}\n",
    "R06782_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2004). Dehydration of (R)-2-hydroxyacyl-CoA to enoyl-CoA in the fermentation of α-amino acids by anaerobic bacteria. FEMS Microbiology Reviews. 28 (4). 455–468.'\n",
    "R06782_c.lower_bound = -1000.\n",
    "R06782_c.upper_bound = 1000.\n",
    "R06782_c.add_metabolites({\n",
    "    cpd00067_c: -2.0,\n",
    "    cpd00143_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd07326_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([R06782_c])                                                                           \n",
    "\n",
    "cpd03343_c = cobra.Metabolite(\n",
    "    'cpd03343_c',\n",
    "    formula='C9H9O2',\n",
    "    name='Phenylpropanoate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "\n",
    "K00529 = cobra.Gene('K00529')\n",
    "K00529.name = '3-phenylpropanoate,NADH:oxygen oxidoreductase'\n",
    "K00529.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "rxn01610_c = cobra.Reaction('rxn01610_c') \n",
    "rxn01610_c.name = 'phenylpropanoate:NAD+ delta2-oxidoreductase'\n",
    "rxn01610_c.gene_reaction_rule = 'K00529'\n",
    "rxn01610_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01610_c.notes['citations'] += 'Kim, J., Darley, D., Selmer, T., & Buckel, W. (2004). Dehydration of (R)-2-hydroxyacyl-CoA to enoyl-CoA in the fermentation of α-amino acids by anaerobic bacteria. FEMS Microbiology Reviews. 28 (4). 455–468.'\n",
    "rxn01610_c.lower_bound = 0.\n",
    "rxn01610_c.upper_bound = 1000.\n",
    "rxn01610_c.add_metabolites({\n",
    "    cpd00004_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd07326_c: -1.0,\n",
    "    cpd00003_c: 1.0,\n",
    "    cpd03343_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn01610_c])                                                                           \n",
    "\n",
    "cpd03343_e = cobra.Metabolite(\n",
    "    'cpd03343_e',\n",
    "    formula='C9H9O2',\n",
    "    name='Phenylpropanoate',\n",
    "    compartment='extracellular')\n",
    "ID007_c = cobra.Reaction('ID007_c') \n",
    "ID007_c.name = 'Phenylpropanoate diffusion'\n",
    "ID007_c.lower_bound = -1000.\n",
    "ID007_c.upper_bound = 1000.\n",
    "ID007_c.add_metabolites({\n",
    "    cpd03343_e: -1.0,\n",
    "    cpd03343_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID007_c])\n",
    "cd630_PATRIC.add_boundary(cpd03343_e, type='exchange', reaction_id='EX_cpd03343_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#---------------#\n",
    "\n",
    "# Tyrosine\n",
    "cpd00868_c = cd630_PATRIC.metabolites.get_by_id('cpd00868_c') # p-hydroxyphenylpyruvate\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') \n",
    "cpd03170_c = cobra.Metabolite(\n",
    "    'cpd03170_c',\n",
    "    formula='C8H7O4',\n",
    "    name='4-Hydroxymandelate',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_01550 = cobra.Gene('CD630_01550')\n",
    "CD630_01550.name = 'hpdA; 4-hydroxyphenylacetate decarboxylase activating enzyme'\n",
    "CD630_01550.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn07124_c = cobra.Reaction('rxn07124_c') \n",
    "rxn07124_c.name = 'p-Hydroxyphenylacetate decarboxylase'\n",
    "rxn07124_c.gene_reaction_rule = 'CD630_01550'\n",
    "rxn07124_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn07124_c.notes['citations'] += 'Yu L, Blaser M, Andrei PI, Pierik AJ, & Selmer T. (2006). 4-Hydroxyphenylacetate Decarboxylases:  Properties of a Novel Subclass of Glycyl Radical Enzyme Systems. Biochemistry. 45 (31), 9584-9592.'\n",
    "rxn07124_c.lower_bound = -1000.\n",
    "rxn07124_c.upper_bound = 1000.\n",
    "rxn07124_c.add_metabolites({\n",
    "    cpd00001_c: -2.0,\n",
    "    cpd00868_c: -1.0,\n",
    "    cpd00011_c: 1.0,\n",
    "    cpd03170_c: 1.0,\n",
    "    cpd00067_c: 4.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn07124_c])\n",
    "\n",
    "cpd03170_e = cobra.Metabolite(\n",
    "    'cpd03170_e',\n",
    "    formula='C8H7O4',\n",
    "    name='4-Hydroxymandelate',\n",
    "    compartment='extracellular')\n",
    "ID008_c = cobra.Reaction('ID008_c')\n",
    "ID008_c.name = '2,3-Dihydroxyphenylpropanoate diffusion'\n",
    "ID008_c.lower_bound = -1000.\n",
    "ID008_c.upper_bound = 1000.\n",
    "ID008_c.add_metabolites({\n",
    "    cpd03170_e: -1.0,\n",
    "    cpd03170_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([ID008_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd03170_e, type='exchange', reaction_id='EX_cpd02501_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "#------------------#\n",
    "\n",
    "# Possible other reductive substrates\n",
    "\n",
    "# from Valine\n",
    "#cpd00123_c = cd630_PATRIC.metabolites.get_by_id('cpd00123_c') # 3-Methyl-2-oxobutanoate - 3MOB\n",
    "\n",
    "# from Isoleucine\n",
    "#cpd00508_c = cd630_PATRIC.metabolites.get_by_id('cpd00508_c') # 3MOP (3-Methyl-2-oxopentanoate)\n",
    "\n",
    "# 67 so far..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd00141_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Propionate exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f847361c210</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd00141_e <=> </p>\n",
       "                    <p style='text-align:right'>Propionate <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd00141_e at 0x7f847361c210>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Fatty acid metabolism\n",
    "\n",
    "# Succinate\n",
    "\n",
    "# Exchange and import\n",
    "cpd00036_e = cobra.Metabolite(\n",
    "    'cpd00036_e',\n",
    "    formula='C4H6O4',\n",
    "    name='Succinate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.metabolites.get_by_id('cpd00036_c').formula = 'C4H6O4' # fixes charge\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00036_e, type='exchange', reaction_id='EX_cpd00036_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "cpd00036_c = cd630_PATRIC.metabolites.get_by_id('cpd00036_c') # Succinate (cytosolic)\n",
    "rxn09271_c = cobra.Reaction('rxn09271_c') # Diffusion\n",
    "rxn09271_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn09271_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn09271_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn09271_c.name = 'Succinate transport'\n",
    "rxn09271_c.lower_bound = -1000.\n",
    "rxn09271_c.upper_bound = 1000.\n",
    "rxn09271_c.add_metabolites({\n",
    "    cpd00036_e: -1.0,\n",
    "    cpd00036_c: 1.0\n",
    "})\n",
    "\n",
    "# Succinate --> Succinyl-CoA\n",
    "cpd00078_c = cd630_PATRIC.metabolites.get_by_id('cpd00078_c') # Succinyl-CoA\n",
    "cpd00029_c = cd630_PATRIC.metabolites.get_by_id('cpd00029_c') # Acetate\n",
    "cpd00022_c = cd630_PATRIC.metabolites.get_by_id('cpd00022_c') # Acetyl-CoA\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H\n",
    "\n",
    "CD630_21960 = cobra.Gene('CD630_21960')\n",
    "CD630_21960.name = 'abfD; Putative succinyl-CoA synthetase-like ATP-binding protein'\n",
    "CD630_21960.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn25164_c = cobra.Reaction('rxn25164_c') \n",
    "rxn25164_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn25164_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn25164_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn25164_c.name = 'Succinyl-CoA synthase'\n",
    "rxn25164_c.gene_reaction_rule = 'CD630_21960'\n",
    "rxn25164_c.lower_bound = -1000.\n",
    "rxn25164_c.upper_bound = 1000.\n",
    "rxn25164_c.add_metabolites({\n",
    "    cpd00036_c: -1.0,\n",
    "    cpd00022_c: -1.0,\n",
    "    cpd00078_c: 1.0,\n",
    "    cpd00029_c: 1.0,\n",
    "    cpd00067_c: 2.0\n",
    "})\n",
    "\n",
    "# Succinyl-CoA --> 4-Oxobutanoate\n",
    "cpd00199_c = cd630_PATRIC.metabolites.get_by_id('cpd00199_c') # 4-Oxobutanoate (succinate semialdehyde)\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00005_c = cd630_PATRIC.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "cpd00006_c = cd630_PATRIC.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "\n",
    "CD630_23420 = cobra.Gene('CD630_23420')\n",
    "CD630_23420.name = 'sucD; Succinate-semialdehyde dehydrogenase'\n",
    "CD630_23420.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn16140_c = cobra.Reaction('rxn16140_c') \n",
    "rxn16140_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn16140_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn16140_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn16140_c.name = 'Succinate semialdehyde dehydrogenase'\n",
    "rxn16140_c.gene_reaction_rule = 'CD630_23420'\n",
    "rxn16140_c.lower_bound = -1000.\n",
    "rxn16140_c.upper_bound = 1000.\n",
    "rxn16140_c.add_metabolites({\n",
    "    cpd00078_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00005_c: -1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd00006_c: 1.0,\n",
    "    cpd00199_c: 1.0\n",
    "})\n",
    "\n",
    "# 4-hydroxybutyrate dehydrogenase: rxn01201_c\n",
    "# 4-Oxobutanoate --> 4-Hydroxybutanoate\n",
    "\n",
    "# 4-hydroxybutyrate CoA transferase: rxn03641_c\n",
    "# 4-Hydroxybutanoate --> 4-Hydroxybutyryl-CoA\n",
    "\n",
    "# 4-Hydroxybutyryl-CoA --> 3-Butenoyl-CoA\n",
    "# 3-Butenoyl-CoA --> Crotonyl-CoA \n",
    "# or directly...\n",
    "cpd07946_c = cd630_PATRIC.metabolites.get_by_id('cpd07946_c') # 4-Hydroxybutyryl-CoA\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00650_c = cd630_PATRIC.metabolites.get_by_id('cpd00650_c') # Crotonyl-CoA\n",
    "\n",
    "CD630_23410 = cobra.Gene('CD630_23410')\n",
    "CD630_23410.name = 'abfD; Gamma-aminobutyrate metabolism dehydratase/isomerase'\n",
    "CD630_23410.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn02167_c = cobra.Reaction('rxn02167_c') \n",
    "rxn02167_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn02167_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn02167_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn02167_c.name = '4-hydroxybutyryl-CoA dehydratase'\n",
    "rxn02167_c.gene_reaction_rule = 'CD630_23410'\n",
    "rxn02167_c.lower_bound = -1000.\n",
    "rxn02167_c.upper_bound = 1000.\n",
    "rxn02167_c.add_metabolites({\n",
    "    cpd07946_c: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd00650_c: 1.0\n",
    "})\n",
    "\n",
    "CD630_10540 = cobra.Gene('CD630_10540')\n",
    "CD630_10540.name = 'bcd2; Butyryl-CoA dehydrogenase'\n",
    "CD630_10540.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "# Crotonyl-CoA --> Butyryl-CoA\n",
    "cpd00120_c = cd630_PATRIC.metabolites.get_by_id('cpd00120_c') # Butyryl-CoA\n",
    "cpd00728_c = cd630_PATRIC.metabolites.get_by_id('cpd00728_c') # 4-Hydroxybutanoate\n",
    "\n",
    "rxn16586_c = cobra.Reaction('rxn16586_c') \n",
    "rxn16586_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn16586_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn16586_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn16586_c.name = 'Butyryl-CoA dehydrgenase'\n",
    "rxn16586_c.gene_reaction_rule = 'CD630_10540'\n",
    "rxn16586_c.lower_bound = 0.\n",
    "rxn16586_c.upper_bound = 1000.\n",
    "rxn16586_c.add_metabolites({\n",
    "    cpd00650_c: -1.0,\n",
    "    cpd00005_c: -1.0,\n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00006_c: 1.0,\n",
    "    cpd00120_c: 1.0\n",
    "})\n",
    "\n",
    "CD630_23390 = cobra.Gene('CD630_23390')\n",
    "CD630_23390.name = 'cat2; 4-hydroxybutyrate CoA-transferase'\n",
    "CD630_23390.notes = {'annotation': 'KEGG'}\n",
    "\n",
    "# Butyryl-CoA --> Butyrate\n",
    "cpd00211_c = cd630_PATRIC.metabolites.get_by_id('cpd00211_c') # Butyrate\n",
    "rxn13713_c = cobra.Reaction('rxn13713_c') \n",
    "rxn13713_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn13713_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn13713_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn13713_c.name = '4-hydroxybutyrate CoA-transferase'\n",
    "rxn13713_c.gene_reaction_rule = 'CD630_23390'\n",
    "rxn13713_c.lower_bound = -1000.\n",
    "rxn13713_c.upper_bound = 1000.\n",
    "rxn13713_c.add_metabolites({\n",
    "    cpd00120_c: -1.0,\n",
    "    cpd00728_c: -1.0,\n",
    "    cpd07946_c: 1.0,\n",
    "    cpd00211_c: 1.0\n",
    "})\n",
    "\n",
    "# Butyryl-CoA --> Butanoylphosphate\n",
    "cpd01662_c = cd630_PATRIC.metabolites.get_by_id('cpd01662_c') # Butanoylphosphate\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "\n",
    "rxn00871_c = cobra.Reaction('rxn00871_c') \n",
    "rxn00871_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00871_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn00871_c.notes['citations'] += 'Ferreyra, J. A., Wu, K. J., Hryckowian, A. J., Bouley, D. M., Weimer, B. C., & Sonnenburg, J. L. (2014). Gut microbiota-produced succinate promotes C. difficile infection after antibiotic treatment or motility disturbance. Cell Host & Microbe, 16(6), 770–777.'\n",
    "rxn00871_c.name = 'Phosphate butyrltransferase'\n",
    "rxn00871_c.gene_reaction_rule = 'CD630_01120'\n",
    "rxn00871_c.lower_bound = -1000.\n",
    "rxn00871_c.upper_bound = 1000.\n",
    "rxn00871_c.add_metabolites({\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00120_c: -1.0,\n",
    "    cpd00010_c: 1.0,\n",
    "    cpd01662_c: 1.0\n",
    "})\n",
    "\n",
    "# Butanoylphosphate --> Butyrate: rxn01236_c (butyrate kinase)\n",
    "\n",
    "# Final export\n",
    "cpd00211_e = cobra.Metabolite(\n",
    "    'cpd00211_e',\n",
    "    formula='C4H7O2',\n",
    "    name='Butyrate',\n",
    "    compartment='extracellular')\n",
    "\n",
    "ID009_c = cobra.Reaction('ID009_c') # Diffusion\n",
    "ID009_c.name = 'Butyrate transport'\n",
    "ID009_c.lower_bound = -1000.\n",
    "ID009_c.upper_bound = 1000.\n",
    "ID009_c.add_metabolites({\n",
    "    cpd00211_e: -1.0,\n",
    "    cpd00211_c: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00211_e, type='exchange', reaction_id='EX_cpd00211_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn09271_c, rxn25164_c, rxn16140_c, rxn02167_c, rxn16586_c, rxn13713_c, rxn00871_c, ID009_c])\n",
    "\n",
    "#----------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Acetate\n",
    "cpd00029_e = cobra.Metabolite(\n",
    "    'cpd00029_e',\n",
    "    formula='C2H3O2',\n",
    "    name='Acetate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00029_e, type='exchange', reaction_id='EX_cpd00029_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00029_c = cd630_PATRIC.metabolites.get_by_id('cpd00029_c') # Acetate\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e') # H+\n",
    "rxn08061_c = cobra.Reaction('rxn08061_c') \n",
    "rxn08061_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn08061_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn08061_c.name = 'Acetate transport via proton symport'\n",
    "rxn08061_c.lower_bound = -1000.\n",
    "rxn08061_c.upper_bound = 1000.\n",
    "rxn08061_c.add_metabolites({\n",
    "    cpd00029_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00029_c: 1.0\n",
    "})\n",
    "\n",
    "# Acetate --> Acetyl-CoA in place already rxn03641_c\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "cpd00018_c = cd630_PATRIC.metabolites.get_by_id('cpd00018_c') # AMP\n",
    "rxn30338_c = cobra.Reaction('rxn30338_c') \n",
    "rxn30338_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn30338_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn30338_c.name = 'Acetyl-CoA synthetase'\n",
    "rxn30338_c.gene_reaction_rule = 'CD630_21960'\n",
    "rxn30338_c.lower_bound = 0.\n",
    "rxn30338_c.upper_bound = 1000.\n",
    "rxn30338_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00029_c: -1.0,\n",
    "    cpd00012_c: 1.0,\n",
    "    cpd00018_c: 1.0,\n",
    "    cpd00022_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn08061_c, rxn30338_c])\n",
    "\n",
    "#----------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Lactate \n",
    "cpd00159_e = cobra.Metabolite(\n",
    "    'cpd00159_e',\n",
    "    formula='C3H5O3',\n",
    "    name='L-Lactate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00159_e, type='exchange', reaction_id='EX_cpd00159_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00159_c = cd630_PATRIC.metabolites.get_by_id('cpd00159_c') # L-Lactate\n",
    "rxn05602_c = cobra.Reaction('rxn05602_c') \n",
    "rxn05602_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05602_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn05602_c.name = 'L-lactate transport via proton symport'\n",
    "rxn05602_c.lower_bound = -1000.\n",
    "rxn05602_c.upper_bound = 1000.\n",
    "rxn05602_c.add_metabolites({\n",
    "    cpd00159_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00159_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cpd00221_e = cobra.Metabolite(\n",
    "    'cpd00221_e',\n",
    "    formula='C3H5O3',\n",
    "    name='D-Lactate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00221_e, type='exchange', reaction_id='EX_cpd00221_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00221_c = cd630_PATRIC.metabolites.get_by_id('cpd00221_c') # D-Lactate\n",
    "rxn10171_c = cobra.Reaction('rxn10171_c') \n",
    "rxn10171_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10171_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn10171_c.name = 'D-lactate transport via proton symport'\n",
    "rxn10171_c.lower_bound = -1000.\n",
    "rxn10171_c.upper_bound = 1000.\n",
    "rxn10171_c.add_metabolites({\n",
    "    cpd00221_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00221_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn05602_c, rxn10171_c])\n",
    "\n",
    "#----------------------------------------------------------------------------------------------#\n",
    "\n",
    "# Propionate formation\n",
    "# https://onlinelibrary.wiley.com/doi/pdf/10.1111/1462-2920.13589\n",
    "\n",
    "# Succinyl-CoA -> D-methylmalonyl-CoA\n",
    "cpd00519_c = cd630_PATRIC.metabolites.get_by_id('cpd00519_c') # D-methylmalonyl-CoA\n",
    "rxn00602_c = cobra.Reaction('rxn00602_c') \n",
    "rxn00602_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00602_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn00602_c.name = 'Methylmalonyl-CoA isomerase'\n",
    "rxn00602_c.gene_reaction_rule = 'CD630_10540'\n",
    "rxn00602_c.lower_bound = -1000.\n",
    "rxn00602_c.upper_bound = 1000.\n",
    "rxn00602_c.add_metabolites({\n",
    "    cpd00078_c: -1.0,\n",
    "    cpd00519_c: 1.0\n",
    "})\n",
    "\n",
    "# rxn00678_c: D-methylmalonyl-CoA -> Propionyl-CoA\n",
    "\n",
    "# Propionyl-CoA -> Propionate\n",
    "cpd00141_c = cd630_PATRIC.metabolites.get_by_id('cpd00141_c') # Propionate\n",
    "cpd00086_c = cd630_PATRIC.metabolites.get_by_id('cpd00086_c') # Propionyl-CoA\n",
    "rxn00669_c = cobra.Reaction('rxn00669_c') \n",
    "rxn00669_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00669_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn00669_c.name = 'Propanoate-CoA ligase'\n",
    "rxn00669_c.gene_reaction_rule = 'CD630_01120'\n",
    "rxn00669_c.lower_bound = -1000.\n",
    "rxn00669_c.upper_bound = 1000.\n",
    "rxn00669_c.add_metabolites({\n",
    "    cpd00086_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00141_c: 1.0,\n",
    "    cpd00002_c: 1.0,\n",
    "    cpd00010_c: 1.0\n",
    "})\n",
    "\n",
    "cpd00141_e = cobra.Metabolite(\n",
    "    'cpd00141_e',\n",
    "    formula='C3H5O2',\n",
    "    name='Propionate',\n",
    "    compartment='extracellular')\n",
    "rxn09172_c = cobra.Reaction('rxn09172_c') \n",
    "rxn09172_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn09172_c.notes['citations'] += 'S. Esquivel-Elizondo, Z. E. Ilhan, E. I. Garcia-Peña, R. Krajmalnik-Brown. (2017). Insights into Butyrate Production in a Controlled Fermentation System via Gene Predictions. mSystems. 2 (4); e00051-17.'\n",
    "rxn09172_c.name = 'Propionate diffusion'\n",
    "rxn09172_c.lower_bound = -1000.\n",
    "rxn09172_c.upper_bound = 1000.\n",
    "rxn09172_c.add_metabolites({\n",
    "    cpd00141_c: -1.0,\n",
    "    cpd00141_e: 1.0\n",
    "})\n",
    "\n",
    "cd630_PATRIC.add_reactions([rxn00602_c, rxn00669_c, rxn09172_c])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd00141_e, type='exchange', reaction_id='EX_cpd00141_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cpd15793_c not found\n",
      "cpd15794_c not found\n",
      "cpd15795_c not found\n"
     ]
    }
   ],
   "source": [
    "# Membrane components\n",
    "\n",
    "# Peptidoglycan subunits\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02009_c').lower_bound = 0.\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00731_c = cd630_PATRIC.metabolites.get_by_id('cpd00731_c') # Ala-Ala\n",
    "cpd02861_c = cd630_PATRIC.metabolites.get_by_id('cpd02861_c') # UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-L-lysine\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd02862_c = cd630_PATRIC.metabolites.get_by_id('cpd02862_c') # UDPMurAc(oyl-L-Ala-D-gamma-Glu-L-Lys-D-Ala-D-Ala)\n",
    "    \n",
    "CD630_26550 = cobra.Gene('CD630_26550')\n",
    "CD630_26550.name = 'murF; UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase'\n",
    "CD630_26550.notes = {'annotation': 'KEGG'}\n",
    "    \n",
    "rxn03141_c = cobra.Reaction('rxn03141_c') \n",
    "rxn03141_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn03141_c.notes['citations'] += ''\n",
    "rxn03141_c.name = 'UDP-N-acetylmuramoyl-tripeptide---D-alanyl-D-alanine ligase'\n",
    "rxn03141_c.gene_reaction_rule = 'CD630_26550'\n",
    "rxn03141_c.lower_bound = 0.\n",
    "rxn03141_c.upper_bound = 1000.\n",
    "rxn03141_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00731_c: -1.0,\n",
    "    cpd02861_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd02862_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn03141_c])\n",
    "\n",
    "\n",
    "# Ultimately synthesizes both:\n",
    "# cpd03495_c\tUndecaprenyl-diphospho-N-acetylmuramoyl--N-acetylglucosamine-L-ala-D-glu-meso-2-6-diaminopimeloyl-D-ala-D-ala\n",
    "# cpd03491_c\tUndecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-D-alanyl-D-alanine\n",
    "# to be included in peptidoglycan polymer\n",
    "\n",
    "#----------------------------------------------------------------------------#\n",
    "\n",
    "# Remove extraneous metabolites and associated reactions\n",
    "\n",
    "# Macromolecular components of ModelSEED Gram+ cell wall\n",
    "# Peptidoglycan polymer (n-1 subunits)\n",
    "# Peptidoglycan polymer (n subunits)\n",
    "# Stearoylcardiolipin (B. subtilis)\n",
    "# Isoheptadecanoylcardiolipin (B. subtilis)\n",
    "# Anteisoheptadecanoylcardiolipin (B. subtilis)\n",
    "# Stearoyllipoteichoic acid (n=24), linked, unsubstituted\n",
    "# Stearoyllipoteichoic acid (n=24), linked, D-alanine substituted\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, unsubstituted\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, N-acetyl-D-glucosamine\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, D-alanine substituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, D-alanine substituted\n",
    "# Stearoyllipoteichoic acid (n=24), linked, N-acetyl-D-glucosamine\n",
    "# Stearoyllipoteichoic acid (n=24), linked, glucose substituted\n",
    "# Isoheptadecanoyllipoteichoic acid (n=24), linked, glucose substituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, glucose substituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, unsubstituted\n",
    "# Anteisoheptadecanoyllipoteichoic acid (n=24), linked, N-acetyl-D-glucosamine\n",
    "# glycerol teichoic acid (n=45), linked, unsubstituted\n",
    "# glycerol teichoic acid (n=45), linked, D-ala substituted\n",
    "# glycerol teichoic acid (n=45), linked, glucose substituted\n",
    "\n",
    "macro = ['cpd15666_c','cpd15665_c','cpd15793_c','cpd15794_c','cpd15795_c','cpd15748_c','cpd15775_c', 'cpd15749_c','cpd15767_c','cpd15776_c','cpd15777_c','cpd15766_c','cpd15757_c','cpd15758_c', 'cpd15759_c','cpd15750_c','cpd15768_c','cpd15667_c','cpd15668_c','cpd15669_c']\n",
    "cd630_PATRIC = removeAll(cd630_PATRIC, macro)\n",
    "\n",
    "\n",
    "# B. subtilis-annotated cardiolipin molecules\n",
    "# Stearoylcardiolipin (B. subtilis)\n",
    "# Isoheptadecanoylcardiolipin (B. subtilis)\n",
    "# Anteisoheptadecanoylcardiolipin (B. subtilis)\n",
    "\n",
    "cardiolipin = ['cpd15793_c','cpd15794_c','cpd15795_c']\n",
    "cd630_PATRIC = removeAll(cd630_PATRIC, cardiolipin)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add missing intracellular metabolites\n",
    "\n",
    "cpd00314_c = cobra.Metabolite(\n",
    "    'cpd00314_c',\n",
    "    formula='C6H14O6',\n",
    "    name='D-Mannitol',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_23310 = cobra.Gene('CD630_23310')\n",
    "CD630_23310.name = 'mtlD; Mannitol-1-phosphate 5-dehydrogenase'\n",
    "CD630_23310.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00082_c = cd630_PATRIC.metabolites.get_by_id('cpd00082_c') # D-Fructose\n",
    "\n",
    "rxn00629_c = cobra.Reaction('rxn00629_c')\n",
    "rxn00629_c.gene_reaction_rule = 'CD630_23310'\n",
    "rxn00629_c.name = 'D-Mannitol:NAD+ 2-oxidoreductase'\n",
    "rxn00629_c.lower_bound = -1000.\n",
    "rxn00629_c.upper_bound = 1000.\n",
    "rxn00629_c.add_metabolites({\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00314_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00082_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00629_c])\n",
    "\n",
    "#-------------------------------------------------------------------------------------#\n",
    "\n",
    "cpd00588_c = cobra.Metabolite(\n",
    "    'cpd00588_c',\n",
    "    formula='C6H14O6',\n",
    "    name='Sorbitol',\n",
    "    compartment='cytosol')\n",
    "\n",
    "CD630_07680 = cobra.Gene('CD630_07680')\n",
    "CD630_07680.name = 'srlD; Sorbitol 6-phosphate 2-dehydrogenase (Glucitol)'\n",
    "CD630_07680.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "rxn00634_c = cobra.Reaction('rxn00634_c')\n",
    "rxn00634_c.gene_reaction_rule = 'CD630_07680'\n",
    "rxn00634_c.name = 'D-Glucitol:NAD+ 2-oxidoreductase'\n",
    "rxn00634_c.lower_bound = -1000.\n",
    "rxn00634_c.upper_bound = 1000.\n",
    "rxn00634_c.add_metabolites({\n",
    "    cpd00003_c: -1.0,\n",
    "    cpd00588_c: -1.0,\n",
    "    cpd00004_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00082_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00634_c])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Biomass composition\n",
    "\n",
    "Adapted from..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DNA replication\n",
    "\n",
    "cpd00115_c = cd630_PATRIC.metabolites.get_by_id('cpd00115_c') # dATP\n",
    "cpd00356_c = cd630_PATRIC.metabolites.get_by_id('cpd00356_c') # dCTP\n",
    "cpd00357_c = cd630_PATRIC.metabolites.get_by_id('cpd00357_c') # TTP\n",
    "cpd00241_c = cd630_PATRIC.metabolites.get_by_id('cpd00241_c') # dGTP\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "\n",
    "cpd17042_c = cobra.Metabolite(\n",
    "    'cpd17042_c',\n",
    "    formula='',\n",
    "    name='DNA polymer',\n",
    "    compartment='cytosol')\n",
    "cpd17042_c.notes = {'Macromolecule'}\n",
    "\n",
    "dna_rxn = cobra.Reaction('dna_rxn')\n",
    "dna_rxn.name = 'DNA replication'\n",
    "dna_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "dna_rxn.lower_bound = 0.\n",
    "dna_rxn.upper_bound = 1000.\n",
    "dna_rxn.add_metabolites({\n",
    "    cpd00115_c: -1.0,\n",
    "    cpd00356_c: -0.5,\n",
    "    cpd00357_c: -1.0,\n",
    "    cpd00241_c: -0.5,\n",
    "    cpd00002_c: -4.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17042_c: 1.0,\n",
    "    cpd00008_c: 4.0,\n",
    "    cpd00009_c: 4.0,\n",
    "    cpd00012_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([dna_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# RNA transcription\n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00052_c = cd630_PATRIC.metabolites.get_by_id('cpd00052_c') # CTP\n",
    "cpd00062_c = cd630_PATRIC.metabolites.get_by_id('cpd00062_c') # UTP\n",
    "cpd00038_c = cd630_PATRIC.metabolites.get_by_id('cpd00038_c') # GTP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "\n",
    "cpd17043_c = cobra.Metabolite(\n",
    "    'cpd17043_c',\n",
    "    formula='',\n",
    "    name='RNA polymer',\n",
    "    compartment='cytosol')\n",
    "cpd17043_c.notes = {'Macromolecule'}\n",
    "\n",
    "rna_rxn = cobra.Reaction('rna_rxn')\n",
    "rna_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "rna_rxn.notes['citations'] += ''\n",
    "rna_rxn.name = 'RNA transcription'\n",
    "rna_rxn.lower_bound = 0.\n",
    "rna_rxn.upper_bound = 1000.\n",
    "rna_rxn.add_metabolites({\n",
    "    cpd00002_c: -2.0,\n",
    "    cpd00052_c: -0.5,\n",
    "    cpd00062_c: -0.5,\n",
    "    cpd00038_c: -0.5,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17043_c: 1.0,\n",
    "    cpd00008_c: 2.0,\n",
    "    cpd00009_c: 2.0,\n",
    "    cpd00012_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rna_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Protein biosynthesis\n",
    "\n",
    "# tRNA synthesis\n",
    "cpd00035_c = cd630_PATRIC.metabolites.get_by_id('cpd00035_c') # L-Alanine\n",
    "\n",
    "\n",
    "cpd00051_c = cd630_PATRIC.metabolites.get_by_id('cpd00051_c') # L-Arginine\n",
    "cpd00132_c = cd630_PATRIC.metabolites.get_by_id('cpd00132_c') # L-Asparagine\n",
    "cpd00041_c = cd630_PATRIC.metabolites.get_by_id('cpd00041_c') # L-Aspartate\n",
    "cpd00084_c = cd630_PATRIC.metabolites.get_by_id('cpd00084_c') # L-Cysteine\n",
    "cpd00053_c = cd630_PATRIC.metabolites.get_by_id('cpd00053_c') # L-Glutamine\n",
    "cpd00023_c = cd630_PATRIC.metabolites.get_by_id('cpd00023_c') # L-Glutamate\n",
    "cpd00033_c = cd630_PATRIC.metabolites.get_by_id('cpd00033_c') # Glycine\n",
    "cpd00119_c = cd630_PATRIC.metabolites.get_by_id('cpd00119_c') # L-Histidine\n",
    "cpd00322_c = cd630_PATRIC.metabolites.get_by_id('cpd00322_c') # L-Isoleucine\n",
    "cpd00107_c = cd630_PATRIC.metabolites.get_by_id('cpd00107_c') # L-Leucine\n",
    "cpd00039_c = cd630_PATRIC.metabolites.get_by_id('cpd00039_c') # L-Lysine\n",
    "cpd00060_c = cd630_PATRIC.metabolites.get_by_id('cpd00060_c') # L-Methionine\n",
    "cpd00066_c = cd630_PATRIC.metabolites.get_by_id('cpd00066_c') # L-Phenylalanine\n",
    "cpd00129_c = cd630_PATRIC.metabolites.get_by_id('cpd00129_c') # L-Proline\n",
    "cpd00054_c = cd630_PATRIC.metabolites.get_by_id('cpd00054_c') # L-Serine\n",
    "cpd00161_c = cd630_PATRIC.metabolites.get_by_id('cpd00161_c') # L-Threonine\n",
    "cpd00065_c = cd630_PATRIC.metabolites.get_by_id('cpd00065_c') # L-Tryptophan\n",
    "cpd00069_c = cd630_PATRIC.metabolites.get_by_id('cpd00069_c') # L-Tyrosine\n",
    "cpd00156_c = cd630_PATRIC.metabolites.get_by_id('cpd00156_c') # L-Valine\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "cpd17041_c = cobra.Metabolite(\n",
    "    'cpd17041_c',\n",
    "    formula='',\n",
    "    name='Protein polymer',\n",
    "    compartment='cytosol')\n",
    "cpd17041_c.notes = {'Macromolecule'}\n",
    "\n",
    "protein_rxn = cobra.Reaction('protein_rxn')\n",
    "protein_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "protein_rxn.notes['citations'] += ''\n",
    "protein_rxn.name = 'Protein biosynthesis'\n",
    "protein_rxn.lower_bound = 0.\n",
    "protein_rxn.upper_bound = 1000.\n",
    "protein_rxn.add_metabolites({\n",
    "    cpd00035_c: -0.5,\n",
    "    cpd00051_c: -0.25,\n",
    "    cpd00132_c: -0.5,\n",
    "    cpd00041_c: -0.5,\n",
    "    cpd00084_c: -0.05,\n",
    "    cpd00053_c: -0.25,\n",
    "    cpd00023_c: -0.5,\n",
    "    cpd00033_c: -0.5,\n",
    "    cpd00119_c: -0.05,\n",
    "    cpd00322_c: -0.5,\n",
    "    cpd00107_c: -0.5,\n",
    "    cpd00039_c: -0.5,\n",
    "    cpd00060_c: -0.25,\n",
    "    cpd00066_c: -0.5,\n",
    "    cpd00129_c: -0.25,\n",
    "    cpd00054_c: -0.5,\n",
    "    cpd00161_c: -0.5,\n",
    "    cpd00065_c: -0.05,\n",
    "    cpd00069_c: -0.25,\n",
    "    cpd00156_c: -0.5,\n",
    "    cpd00002_c: -20.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17041_c: 1.0,\n",
    "    cpd00008_c: 20.0,\n",
    "    cpd00009_c: 20.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([protein_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cell wall synthesis\n",
    "\n",
    "cpd02967_c = cd630_PATRIC.metabolites.get_by_id('cpd02967_c') # N-Acetyl-beta-D-mannosaminyl-1,4-N-acetyl-D-glucosaminyldiphosphoundecaprenol\n",
    "cpd00402_c = cd630_PATRIC.metabolites.get_by_id('cpd00402_c') # CDPglycerol\n",
    "cpd00046_c = cd630_PATRIC.metabolites.get_by_id('cpd00046_c') # CMP\n",
    "\n",
    "cpd12894_c = cobra.Metabolite(\n",
    "    'cpd12894_c',\n",
    "    formula='',\n",
    "    name='Teichoic acid',\n",
    "    compartment='cytosol')\n",
    "cpd12894_c.notes = {'type':'Macromolecule'}\n",
    "\n",
    "CD630_02440 = cobra.Gene('CD630_02440')\n",
    "CD630_02440.name = 'Putative CDP-glycerol:Poly(Glycerophosphate) glycerophosphotransferase'\n",
    "CD630_02440.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "teichoicacid_rxn = cobra.Reaction('teichoicacid_rxn')\n",
    "teichoicacid_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "teichoicacid_rxn.notes['citations'] += ''\n",
    "teichoicacid_rxn.gene_reaction_rule = 'CD630_02440'\n",
    "teichoicacid_rxn.name = 'Teichoic acid biosynthesis'\n",
    "teichoicacid_rxn.lower_bound = 0.\n",
    "teichoicacid_rxn.upper_bound = 1000.\n",
    "teichoicacid_rxn.add_metabolites({\n",
    "    cpd02967_c: -1.0,\n",
    "    cpd00402_c: -1.0,\n",
    "    cpd00046_c: 1.0,\n",
    "    cpd12894_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([teichoicacid_rxn])\n",
    "\n",
    "#---------------------------------------#\n",
    "\n",
    "# Peptidoglycan subunits\n",
    " # Undecaprenyl-diphospho-N-acetylmuramoyl--N-acetylglucosamine-L-ala-D-glu-meso-2-6-diaminopimeloyl-D-ala-D-ala (right)\n",
    "cpd03495_c = cd630_PATRIC.metabolites.get_by_id('cpd03495_c')\n",
    "# Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-D-alanyl-D-alanine (left)\n",
    "cpd03491_c = cd630_PATRIC.metabolites.get_by_id('cpd03491_c') \n",
    "\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "\n",
    "cpd02229_c = cd630_PATRIC.metabolites.get_by_id('cpd02229_c') # Bactoprenyl diphosphate\n",
    "cpd00117_c = cd630_PATRIC.metabolites.get_by_id('cpd00117_c') # D-Alanine\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "\n",
    "cpd16661_c = cobra.Metabolite(\n",
    "    'cpd16661_c',\n",
    "    formula='',\n",
    "    name='Peptidoglycan polymer',\n",
    "    compartment='cytosol')\n",
    "cpd16661_c.notes = {'type':'Macromolecule'}\n",
    "                            \n",
    "CD630_26510 = cobra.Gene('CD630_26510')\n",
    "CD630_26510.name = 'murG; UDP-N-acetylglucosamine--N-acetylmuramyl-(pentapeptide) pyrophosphoryl-undecaprenol N-acetylglucosamine transferase'\n",
    "CD630_26510.notes = {'annotation': 'UniProt'}\n",
    "\n",
    "peptidoglycan_rxn = cobra.Reaction('peptidoglycan_rxn')\n",
    "peptidoglycan_rxn.gene_reaction_rule = 'CD630_26510'\n",
    "peptidoglycan_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "peptidoglycan_rxn.name = 'Peptidoglycan synthesis'\n",
    "peptidoglycan_rxn.lower_bound = 0.\n",
    "peptidoglycan_rxn.upper_bound = 1000.\n",
    "peptidoglycan_rxn.add_metabolites({\n",
    "    cpd03491_c: -1.0,\n",
    "    cpd03495_c: -1.0,\n",
    "    cpd00002_c: -4.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd16661_c: 1.0,\n",
    "    cpd02229_c: 1.0,\n",
    "    cpd00117_c: 0.5, # D-Alanine\n",
    "    cpd00008_c: 4.0, # ADP\n",
    "    cpd00009_c: 4.0 # Phosphate\n",
    "})\n",
    "cd630_PATRIC.add_reactions([peptidoglycan_rxn])\n",
    "\n",
    "#---------------------------------------#\n",
    "\n",
    "cellwall_c = cobra.Metabolite(\n",
    "    'cellwall_c',\n",
    "    formula='',\n",
    "    name='Cell Wall polymer',\n",
    "    compartment='cytosol')\n",
    "cellwall_c.notes = {'type':'Macromolecule'}\n",
    "\n",
    "cellwall_rxn = cobra.Reaction('cellwall_rxn')\n",
    "cellwall_rxn.name = 'Cell wall biosynthesis'\n",
    "cellwall_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "cellwall_rxn.lower_bound = 0.\n",
    "cellwall_rxn.upper_bound = 1000.\n",
    "cellwall_rxn.add_metabolites({\n",
    "    cpd16661_c: -1.5,\n",
    "    cpd12894_c: -0.05,\n",
    "    cellwall_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([cellwall_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Lipid pool\n",
    "\n",
    "cpd15543_c = cd630_PATRIC.metabolites.get_by_id('cpd15543_c') # Phosphatidylglycerophosphate ditetradecanoyl\n",
    "cpd15545_c = cd630_PATRIC.metabolites.get_by_id('cpd15545_c') # Phosphatidylglycerophosphate dihexadecanoyl\n",
    "cpd15540_c = cd630_PATRIC.metabolites.get_by_id('cpd15540_c') # Phosphatidylglycerol dioctadecanoyl\n",
    "cpd15728_c = cd630_PATRIC.metabolites.get_by_id('cpd15728_c') # Diglucosyl-1,2 dipalmitoylglycerol\n",
    "cpd15729_c = cd630_PATRIC.metabolites.get_by_id('cpd15729_c') # Diglucosyl-1,2 dimyristoylglycerol\n",
    "cpd15737_c = cd630_PATRIC.metabolites.get_by_id('cpd15737_c') # Monoglucosyl-1,2 dipalmitoylglycerol\n",
    "cpd15738_c = cd630_PATRIC.metabolites.get_by_id('cpd15738_c') # Monoglucosyl-1,2 dimyristoylglycerol\n",
    "\n",
    "cpd11852_c = cobra.Metabolite(\n",
    "    'cpd11852_c',\n",
    "    formula='',\n",
    "    name='Lipid Pool',\n",
    "    compartment='cytosol')\n",
    "cpd11852_c.notes = {'type':'Metabolite pool'}\n",
    "\n",
    "lipid_rxn = cobra.Reaction('lipid_rxn')\n",
    "lipid_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "lipid_rxn.notes['citations'] += ''\n",
    "lipid_rxn.name = 'Lipid composition'\n",
    "lipid_rxn.lower_bound = 0.\n",
    "lipid_rxn.upper_bound = 1000.\n",
    "lipid_rxn.add_metabolites({\n",
    "    cpd15543_c: -0.005,\n",
    "    cpd15545_c: -0.005,\n",
    "    cpd15540_c: -0.005,\n",
    "    cpd15728_c: -0.005,\n",
    "    cpd15729_c: -0.005,\n",
    "    cpd15737_c: -0.005,\n",
    "    cpd15738_c: -0.005,\n",
    "    cpd11852_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([lipid_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ions, Vitamins, & Cofactors\n",
    "\n",
    "# Vitamins\n",
    "cpd00104_c = cd630_PATRIC.metabolites.get_by_id('cpd00104_c') # Biotin          MDM\n",
    "cpd00644_c = cd630_PATRIC.metabolites.get_by_id('cpd00644_c') # Pantothenate    MDM\n",
    "cpd00263_c = cd630_PATRIC.metabolites.get_by_id('cpd00263_c') # Pyridoxine      MDM\n",
    "cpd00393_c = cd630_PATRIC.metabolites.get_by_id('cpd00393_c') #  folate\n",
    "cpd00133_c = cd630_PATRIC.metabolites.get_by_id('cpd00133_c') #  nicotinamide\n",
    "cpd00443_c = cd630_PATRIC.metabolites.get_by_id('cpd00443_c') #  p-aminobenzoic acid\n",
    "cpd00220_c = cd630_PATRIC.metabolites.get_by_id('cpd00220_c') #  riboflavin  \n",
    "cpd00305_c = cd630_PATRIC.metabolites.get_by_id('cpd00305_c') #  thiamin\n",
    "\n",
    "# Ions\n",
    "cpd00149_c = cd630_PATRIC.metabolites.get_by_id('cpd00149_c') # Cobalt\n",
    "cpd00030_c = cd630_PATRIC.metabolites.get_by_id('cpd00030_c') # Manganese\n",
    "cpd00254_c = cd630_PATRIC.metabolites.get_by_id('cpd00254_c') # Magnesium\n",
    "cpd00971_c = cd630_PATRIC.metabolites.get_by_id('cpd00971_c') # Sodium\n",
    "cpd00063_c = cd630_PATRIC.metabolites.get_by_id('cpd00063_c') # Calcium\n",
    "cpd10515_c = cd630_PATRIC.metabolites.get_by_id('cpd10515_c') # Iron\n",
    "cpd00205_c = cd630_PATRIC.metabolites.get_by_id('cpd00205_c') # Potassium\n",
    "cpd00099_c = cd630_PATRIC.metabolites.get_by_id('cpd00099_c') # Chloride\n",
    "\n",
    "# Cofactors\n",
    "cpd00022_c = cd630_PATRIC.metabolites.get_by_id('cpd00022_c') # Acetyl-CoA\n",
    "cpd00010_c = cd630_PATRIC.metabolites.get_by_id('cpd00010_c') # CoA\n",
    "cpd00015_c = cd630_PATRIC.metabolites.get_by_id('cpd00015_c') # FAD\n",
    "cpd00003_c = cd630_PATRIC.metabolites.get_by_id('cpd00003_c') # NAD\n",
    "cpd00004_c = cd630_PATRIC.metabolites.get_by_id('cpd00004_c') # NADH\n",
    "cpd00006_c = cd630_PATRIC.metabolites.get_by_id('cpd00006_c') # NADP\n",
    "cpd00005_c = cd630_PATRIC.metabolites.get_by_id('cpd00005_c') # NADPH\n",
    "\n",
    "# Energy molecules\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "cpd00038_c = cd630_PATRIC.metabolites.get_by_id('cpd00038_c') # GTP\n",
    "cpd00031_c = cd630_PATRIC.metabolites.get_by_id('cpd00031_c') # GDP\n",
    "\n",
    "#cpd00274_c = cd630_PATRIC.metabolites.get_by_id('cpd00274_c') # Citrulline\n",
    "#cpd00907_c = cd630_PATRIC.metabolites.get_by_id('cpd00907_c') # alpha-D-glucose-1,6-biphosphate\n",
    "\n",
    "cofactor_c = cobra.Metabolite(\n",
    "    'cofactor_c',\n",
    "    formula='',\n",
    "    name='Cofactor Pool',\n",
    "    compartment='cytosol')\n",
    "cofactor_c.notes = {'type':'Metabolite pool'}\n",
    "\n",
    "cofactor_rxn = cobra.Reaction('cofactor_rxn')\n",
    "cofactor_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "cofactor_rxn.notes['citations'] += 'Karasawa T, Ikoma S, Yamakawa K, & Nakamura S. (1995). A defined growth medium for Clostridium difficile.Microbiology. 141(2); 371-5.'\n",
    "cofactor_rxn.name = 'Cofactor Pool'\n",
    "cofactor_rxn.lower_bound = 0.\n",
    "cofactor_rxn.upper_bound = 1000.\n",
    "cofactor_rxn.add_metabolites({\n",
    "    cpd00104_c: -0.005, \n",
    "    cpd00644_c: -0.005, \n",
    "    cpd00263_c: -0.005, \n",
    "    cpd00393_c: -0.005,\n",
    "    cpd00133_c: -0.005,\n",
    "    cpd00443_c: -0.005,\n",
    "    cpd00220_c: -0.005,\n",
    "    cpd00305_c: -0.005,\n",
    "    cpd00149_c: -0.005, \n",
    "    cpd00030_c: -0.005, \n",
    "    cpd00254_c: -0.005, \n",
    "    cpd00971_c: -0.005, \n",
    "    cpd00063_c: -0.005, \n",
    "    cpd10515_c: -0.005, \n",
    "    cpd00205_c: -0.005,\n",
    "    cpd00099_c: -0.005, \n",
    "    cpd00022_c: -0.005,\n",
    "    cpd00010_c: -0.0005,\n",
    "    cpd00015_c: -0.0005,\n",
    "    cpd00003_c: -0.005,\n",
    "    cpd00004_c: -0.005,\n",
    "    cpd00006_c: -0.005,\n",
    "    cpd00005_c: -0.005,\n",
    "    cpd00002_c: -0.005, \n",
    "    cpd00008_c: -0.005, \n",
    "    cpd00009_c: -0.5,\n",
    "    cpd00012_c: -0.005, \n",
    "    cpd00038_c: -0.005, \n",
    "    cpd00031_c: -0.005, \n",
    "\n",
    "    cofactor_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([cofactor_rxn])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final Biomass\n",
    "\n",
    "cpd11416_c = cobra.Metabolite(\n",
    "    'cpd11416_c',\n",
    "    formula='',\n",
    "    name='Biomass',\n",
    "    compartment='cytosol')\n",
    "cpd11416_c.notes = {'type':'Biomass'}\n",
    "\n",
    "biomass_rxn = cobra.Reaction('biomass')\n",
    "biomass_rxn.notes = {'citations': [], 'type': 'biomass', 'annotation': 'none'}\n",
    "biomass_rxn.name = 'Biomass Reaction'\n",
    "biomass_rxn.lower_bound = 0.\n",
    "biomass_rxn.upper_bound = 1000.\n",
    "biomass_rxn.add_metabolites({\n",
    "    cpd17041_c: -0.4,  # Protein\n",
    "    cpd17043_c: -0.15, # RNA\n",
    "    cpd17042_c: -0.05, # DNA\n",
    "    cpd11852_c: -0.05, # Lipid\n",
    "    cellwall_c: -0.2,\n",
    "    cofactor_c: -0.2,\n",
    "    cpd00001_c: -20.0,\n",
    "    cpd00002_c: -20.0,\n",
    "    cpd00008_c: 20.0,\n",
    "    cpd00009_c: 20.0,\n",
    "    cpd11416_c: 1.0 # Biomass\n",
    "})\n",
    "cd630_PATRIC.add_reactions([biomass_rxn])\n",
    "\n",
    "cd630_PATRIC.add_boundary(cpd11416_c, type='sink', reaction_id='EX_biomass', lb=0.0, ub=1000.0)\n",
    "\n",
    "# Set new objective\n",
    "cd630_PATRIC.objective = biomass_rxn\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Deal with extracellular metabolites without transporters          # sulfate     # CoA\n",
    "for cpd in ['cpd10516_e', 'cpd01188_e', 'cpd00028_e', 'cpd11606_e', 'cpd00048_e', 'cpd00010_e']:\n",
    "    try:\n",
    "        cd630_PATRIC.metabolites.get_by_id(cpd).remove_from_model()\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "#---------------------------------------------------------------------------#\n",
    "\n",
    "# Manually add necessary reactions\n",
    "\n",
    "# Diffusion reactions\n",
    "\n",
    "# H2O\n",
    "cpd00001_e = cd630_PATRIC.metabolites.get_by_id('cpd00001_e')\n",
    "cpd00001_c = cd630_PATRIC.metabolites.get_by_id('cpd00001_c')\n",
    "rxn08686_c = cobra.Reaction('rxn08686_c') \n",
    "rxn08686_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn08686_c.name = 'H2O transport via diffusion'\n",
    "rxn08686_c.lower_bound = -1000.\n",
    "rxn08686_c.upper_bound = 1000.\n",
    "rxn08686_c.add_metabolites({\n",
    "    cpd00001_e: -1.0,\n",
    "    cpd00001_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08686_c])\n",
    "\n",
    "# O2\n",
    "cpd00007_e = cd630_PATRIC.metabolites.get_by_id('cpd00007_e')\n",
    "cpd00007_c = cd630_PATRIC.metabolites.get_by_id('cpd00007_c')\n",
    "rxn09031_c = cobra.Reaction('rxn09031_c') \n",
    "rxn09031_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn09031_c.name = 'Oxygen transport via diffusion'\n",
    "rxn09031_c.lower_bound = -1000.\n",
    "rxn09031_c.upper_bound = 1000.\n",
    "rxn09031_c.add_metabolites({\n",
    "    cpd00007_e: -1.0,\n",
    "    cpd00007_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn09031_c])\n",
    "\n",
    "# CO2\n",
    "cpd00011_e = cd630_PATRIC.metabolites.get_by_id('cpd00011_e')\n",
    "cpd00011_c = cd630_PATRIC.metabolites.get_by_id('cpd00011_c')\n",
    "rxn05467_c = cobra.Reaction('rxn05467_c') \n",
    "rxn05467_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn05467_c.name = 'CO2 transport via diffusion'\n",
    "rxn05467_c.lower_bound = -1000.\n",
    "rxn05467_c.upper_bound = 1000.\n",
    "rxn05467_c.add_metabolites({\n",
    "    cpd00011_e: -1.0,\n",
    "    cpd00011_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05467_c])\n",
    "\n",
    "# Ca2+\n",
    "cpd00063_e = cd630_PATRIC.metabolites.get_by_id('cpd00063_e')\n",
    "cpd00063_c = cd630_PATRIC.metabolites.get_by_id('cpd00063_c')\n",
    "rxn08186_c = cobra.Reaction('rxn08186_c') \n",
    "rxn08186_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn08186_c.name = 'Calcium transport via diffusion'\n",
    "rxn08186_c.lower_bound = -1000.\n",
    "rxn08186_c.upper_bound = 1000.\n",
    "rxn08186_c.add_metabolites({\n",
    "    cpd00063_e: -1.0,\n",
    "    cpd00063_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08186_c])\n",
    "\n",
    "# Mn2+\n",
    "cpd00030_e = cd630_PATRIC.metabolites.get_by_id('cpd00030_e')\n",
    "cpd00030_c = cd630_PATRIC.metabolites.get_by_id('cpd00030_c')\n",
    "rxn08952_c = cobra.Reaction('rxn08952_c') \n",
    "rxn08952_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn08952_c.name = 'Manganese transport via diffusion'\n",
    "rxn08952_c.lower_bound = -1000.\n",
    "rxn08952_c.upper_bound = 1000.\n",
    "rxn08952_c.add_metabolites({\n",
    "    cpd00030_e: -1.0,\n",
    "    cpd00030_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08952_c])\n",
    "\n",
    "#---------------------------------#\n",
    "\n",
    "# Active transport\n",
    "\n",
    "# ocdca\n",
    "cpd01080_e = cd630_PATRIC.metabolites.get_by_id('cpd01080_e')\n",
    "cpd01080_c = cd630_PATRIC.metabolites.get_by_id('cpd01080_c')\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e')\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c')\n",
    "\n",
    "rxn10161_c = cobra.Reaction('rxn10161_c') \n",
    "rxn10161_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED'}\n",
    "rxn10161_c.name = 'Octadecanoate transport via proton symport '\n",
    "rxn10161_c.lower_bound = -1000.\n",
    "rxn10161_c.upper_bound = 1000.\n",
    "rxn10161_c.add_metabolites({\n",
    "    cpd01080_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd01080_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10161_c])\n",
    "\n",
    "# Spermidine\n",
    "cpd00264_e = cd630_PATRIC.metabolites.get_by_id('cpd00264_e')\n",
    "cpd00264_c = cd630_PATRIC.metabolites.get_by_id('cpd00264_c')\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c')\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c')\n",
    "cpd00009_c = cd630_PATRIC.metabolites.get_by_id('cpd00009_c')\n",
    "\n",
    "rxn05175_c = cobra.Reaction('rxn05175_c') \n",
    "rxn05175_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn05175_c.name = 'Spermidine transport via ABC system'\n",
    "rxn05175_c.lower_bound = 0.\n",
    "rxn05175_c.upper_bound = 1000.\n",
    "rxn05175_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00264_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00264_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05175_c])\n",
    "\n",
    "# meso-2,6-Diaminopimelate\n",
    "cpd00516_e = cd630_PATRIC.metabolites.get_by_id('cpd00516_e')\n",
    "cpd00516_c = cd630_PATRIC.metabolites.get_by_id('cpd00516_c')\n",
    "\n",
    "rxn08304_c = cobra.Reaction('rxn08304_c') \n",
    "rxn08304_c.notes = {'citations': [], 'type': 'transport', 'annotation': 'ModelSEED', 'subsystem': 'ABC transporter'}\n",
    "rxn08304_c.name = 'M-diaminopimelic acid ABC transport'\n",
    "rxn08304_c.lower_bound = 0.\n",
    "rxn08304_c.upper_bound = 1000.\n",
    "rxn08304_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00516_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00516_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn08304_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reversible reactions containing ATP: 87\n"
     ]
    }
   ],
   "source": [
    "# Check irreversibility of ATP consuming reactions\n",
    "\n",
    "reversible = []\n",
    "for rxn in list(cd630_PATRIC.metabolites.get_by_id('cpd00002_c').reactions):\n",
    "    if rxn.reversibility == True:\n",
    "        reversible.append(rxn.id)\n",
    "print('Reversible reactions containing ATP: ' + str(len(reversible)))\n",
    "\n",
    "# Set correct reversibility\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08298_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01396_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02503_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01517_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02262_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08297_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10257_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10256_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00392_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00134_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00669_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08295_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01100_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02314_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00337_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01434_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01917_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01127_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02175_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01513_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02489_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00547_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01343_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00077_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03536_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01739_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00151_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01807_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01135_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10255_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00772_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08294_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01607_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00690_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03075_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02362_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00303_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10258_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01671_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05054_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn06672_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn09320_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00364_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08647_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn04384_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn07584_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00695_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02937_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10253_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00615_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01114_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn12510_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00124_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00181_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00440_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03108_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01219_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10042_c').lower_bound = 0. # Proton motive force...\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00239_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02484_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00770_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03908_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05814_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00105_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn07841_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08296_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01300_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01951_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08299_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01492_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01492_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00097_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01509_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00780_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01754_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn02155_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00545_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01102_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn10254_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn05145_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00119_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01199_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00410_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn08300_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03540_c').lower_bound = 0.\n",
    "\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01236_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn01987_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00985_c').upper_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00148_c').upper_bound = 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reversible reactions containing GTP: 3\n"
     ]
    }
   ],
   "source": [
    "# Check irreversibility of GTP consuming reactions\n",
    "\n",
    "reversible = []\n",
    "for rxn in list(cd630_PATRIC.metabolites.get_by_id('cpd00038_c').reactions):\n",
    "    if rxn.reversibility == True:\n",
    "        reversible.append(rxn.id)\n",
    "print('Reversible reactions containing GTP: ' + str(len(reversible)))\n",
    "\n",
    "# Fix bounds\n",
    "cd630_PATRIC.reactions.get_by_id('rxn04413_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn03537_c').lower_bound = 0.\n",
    "cd630_PATRIC.reactions.get_by_id('rxn00641_c').lower_bound = 0.\n",
    "\n",
    "\n",
    "# GTP generation\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c')\n",
    "cpd00031_c = cd630_PATRIC.metabolites.get_by_id('cpd00031_c')\n",
    "cpd00008_c = cd630_PATRIC.metabolites.get_by_id('cpd00008_c')\n",
    "cpd00038_c = cd630_PATRIC.metabolites.get_by_id('cpd00038_c')\n",
    "rxn00237_c = cobra.Reaction('rxn00237_c') \n",
    "rxn00237_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED', 'subsystem': 'Nucleotide biosynthesis'}\n",
    "rxn00237_c.name = 'GDP phosphotransferase'\n",
    "rxn00237_c.lower_bound = 0.\n",
    "rxn00237_c.upper_bound = 1000.\n",
    "rxn05175_c.add_metabolites({\n",
    "    cpd00031_c: -1.0,\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00038_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn00237_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remaining minimal media components\n",
    "\n",
    "# Ammonia (extracellular)\n",
    "cpd00013_e = cobra.Metabolite(\n",
    "    'cpd00013_e',\n",
    "    formula='H4N',\n",
    "    name='Ammonia',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00013_e, type='exchange', reaction_id='EX_cpd00013_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00013_c = cd630_PATRIC.metabolites.get_by_id('cpd00013_c')\n",
    "rxn05466_c = cobra.Reaction('rxn05466_c') \n",
    "rxn05466_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05466_c.name = 'Ammonia diffusion'\n",
    "rxn05466_c.lower_bound = -1000.\n",
    "rxn05466_c.upper_bound = 1000.\n",
    "rxn05466_c.add_metabolites({\n",
    "    cpd00013_e: -1.0,\n",
    "    cpd00013_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn05466_c])\n",
    "\n",
    "# Carbonate (extracellular)\n",
    "cpd00242_c = cd630_PATRIC.metabolites.get_by_id('cpd00242_c')\n",
    "cpd00242_e = cobra.Metabolite(\n",
    "    'cpd00242_e',\n",
    "    formula='CHO3',\n",
    "    name='Carbonate',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00242_e, type='exchange', reaction_id='EX_cpd00242_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "rxn10826_c = cobra.Reaction('rxn10826_c') \n",
    "rxn10826_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10826_c.name = 'Carbonate diffusion'\n",
    "rxn10826_c.lower_bound = -1000.\n",
    "rxn10826_c.upper_bound = 1000.\n",
    "rxn10826_c.add_metabolites({\n",
    "    cpd00242_e: -1.0,\n",
    "    cpd00242_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10826_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# FAD biosynthesis\n",
    "cpd00002_c = cd630_PATRIC.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd00050_c = cd630_PATRIC.metabolites.get_by_id('cpd00050_c') # FMN\n",
    "cpd00012_c = cd630_PATRIC.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "cpd00015_c = cd630_PATRIC.metabolites.get_by_id('cpd00015_c') # FAD\n",
    "\n",
    "try:\n",
    "    cd630_PATRIC.reactions.get_by_id('rxn30210_c').remove_from_model()\n",
    "except:\n",
    "    pass\n",
    "\n",
    "rxn30210_c = cobra.Reaction('rxn30210_c') \n",
    "rxn30210_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn30210_c.name = 'FAD synthetase'\n",
    "rxn30210_c.gene_reaction_rule = '272563.8.peg.1382 and 272563.8.peg.1382'\n",
    "rxn30210_c.lower_bound = 0.\n",
    "rxn30210_c.upper_bound = 1000.\n",
    "rxn30210_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00050_c: -1.0,\n",
    "    cpd00012_c: 1.0,\n",
    "    cpd00015_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn30210_c])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import of last missing vitamins\n",
    "\n",
    "# Nicotinamide\n",
    "cpd00133_e = cobra.Metabolite(\n",
    "    'cpd00133_e',\n",
    "    formula='C6H6N2O',\n",
    "    name='Nicotinamide',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00133_e, type='exchange', reaction_id='EX_cpd00133_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00133_c = cd630_PATRIC.metabolites.get_by_id('cpd00133_c')\n",
    "cpd00067_c = cd630_PATRIC.metabolites.get_by_id('cpd00067_c')\n",
    "cpd00067_e = cd630_PATRIC.metabolites.get_by_id('cpd00067_e')\n",
    "rxn11334_c = cobra.Reaction('rxn11334_c') \n",
    "rxn11334_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn11334_c.name = 'Nicotinamide symporter'\n",
    "rxn11334_c.gene_reaction_rule = '272563.8.peg.164 or 272563.8.peg.1796'\n",
    "rxn11334_c.lower_bound = 0.\n",
    "rxn11334_c.upper_bound = 1000.\n",
    "rxn11334_c.add_metabolites({\n",
    "    cpd00133_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00133_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn11334_c])\n",
    "\n",
    "# p-Aminobenzoic acid (ABEE)\n",
    "cpd00443_e = cobra.Metabolite(\n",
    "    'cpd00443_e',\n",
    "    formula='C7H6NO2',\n",
    "    name='Nicotinamide',\n",
    "    compartment='extracellular')\n",
    "cd630_PATRIC.add_boundary(cpd00443_e, type='exchange', reaction_id='EX_cpd00443_e', lb=-1000.0, ub=1000.0)\n",
    "cpd00443_c = cd630_PATRIC.metabolites.get_by_id('cpd00443_c') #  p-aminobenzoic acid\n",
    "rxn10421_c = cobra.Reaction('rxn10421_c') \n",
    "rxn10421_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn10421_c.name = 'p-Aminobenzoic acid symporter'\n",
    "rxn10421_c.gene_reaction_rule = '272563.8.peg.164 or 272563.8.peg.1796'\n",
    "rxn10421_c.lower_bound = 0.\n",
    "rxn10421_c.upper_bound = 1000.\n",
    "rxn10421_c.add_metabolites({\n",
    "    cpd00443_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00443_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC.add_reactions([rxn10421_c])\n",
    "\n",
    "# Folate, Thiamin, and Riboflavin in place already"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>272563.8</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f84b432b950</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1306</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1182</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>-1.0*biomass_reverse_01e59 + 1.0*biomass</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model 272563.8 at 0x7f84b432b950>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check progress\n",
    "cd630_PATRIC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(cd630_PATRIC, 'data/cdiff_temp.sbml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gapfilling (complete media)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dependencies\n",
    "import numpy\n",
    "import copy\n",
    "import time\n",
    "import cobra\n",
    "from cobra.util import solver\n",
    "from optlang.symbolics import Zero\n",
    "from itertools import chain\n",
    "from cobra.manipulation.delete import *\n",
    "from cobra.flux_analysis.sampling import ACHRSampler\n",
    "\n",
    "# pFBA gapfiller\n",
    "def pfba_gapfill(model, reaction_bag, sample=True, extracellular='e', tasks=False):\n",
    "    '''\n",
    "    Function that utilizes iterations of pFBA solution with a universal reaction bag \n",
    "    in order to gapfill a model.\n",
    "    \n",
    "    Parameters\n",
    "    ----------\n",
    "    model : cobra.Model\n",
    "        Model to be gapfilled\n",
    "    reaction_bag : cobra.Model\n",
    "        Reaction bag reference to use during gapfilling\n",
    "    sample : bool\n",
    "        Perform flux sampling to identify all possible active reactions from universal\n",
    "    extracellular : string\n",
    "        Label for extracellular compartment of model\n",
    "    tasks : list\n",
    "        List of strings representing reaction IDs that must be able to carry flux\n",
    "    '''\n",
    "    start_time = time.time()\n",
    "    \n",
    "    # Save some basic network info for downstream membership testing\n",
    "    orig_rxn_ids = set([str(w.id) for w in model.reactions])\n",
    "    orig_cpd_ids = set([str(x.id) for x in model.metabolites])\n",
    "    univ_rxn_ids = set([str(y.id) for y in reaction_bag.reactions])\n",
    "    \n",
    "    # Find overlap in model and reaction bag\n",
    "    overlap_rxn_ids = univ_rxn_ids.intersection(orig_rxn_ids)\n",
    "    \n",
    "    # Get model objective reaction ID\n",
    "    obj = get_objective(model)\n",
    "\n",
    "    # Modify universal reaction bag\n",
    "    new_rxn_ids = set()\n",
    "    print('Creating universal model')\n",
    "    with reaction_bag as universal:\n",
    "\n",
    "        # Remove overlapping reactions from universal bag, and reset objective if needed\n",
    "        for rxn in overlap_rxn_ids: universal.reactions.get_by_id(rxn).remove_from_model()\n",
    "\n",
    "        # Add model reactions to universal bag\n",
    "        universal.add_reactions(list(copy.deepcopy(model.reactions)))\n",
    "        \n",
    "        # Set up pFBA objective\n",
    "        print('Formulating pFBA objective')\n",
    "        rxn_vars = ((rxn.forward_variable, rxn.reverse_variable) for rxn in universal.reactions if rxn.id not in orig_rxn_ids)\n",
    "        rxn_vars = chain(*rxn_vars)\n",
    "        orig_rxn_vars = ((rxn.forward_variable, rxn.reverse_variable) for rxn in universal.reactions if rxn.id in orig_rxn_ids)\n",
    "        orig_rxn_vars = chain(*orig_rxn_vars)\n",
    "        pfba_obj = universal.problem.Objective(Zero, direction='min', sloppy=True)\n",
    "        universal.objective = pfba_obj\n",
    "        universal.objective.set_linear_coefficients({x: 1.0 for x in rxn_vars})\n",
    "        universal.objective.set_linear_coefficients({y: 0.0 for y in orig_rxn_vars})\n",
    "        universal.solver.update()\n",
    "        \n",
    "        # Set minimum lower bound for previous objective\n",
    "        universal.reactions.get_by_id(obj).lower_bound = 1.0\n",
    "        \n",
    "        # Set lower bounds for metaboloic tasks\n",
    "        if tasks == True:\n",
    "            for rxn in tasks:\n",
    "                try:\n",
    "                    universal.reactions.get_by_id(rxn).lower_bound = 0.001\n",
    "                except:\n",
    "                    continue\n",
    "                \n",
    "        # Run FBA and save solution\n",
    "        solution = universal.optimize()\n",
    "        \n",
    "        # Identify reactions from universal that are now active\n",
    "        if sample == True:\n",
    "            print('Generating flux sampling object')\n",
    "            optgp_object = ACHRSampler(universal)\n",
    "            \n",
    "            print('Sampling flux distribution for active reactions')\n",
    "            flux_samples = optgp_object.sample(1000)\n",
    "            for distribution in flux_samples.iterrows():\n",
    "                for rxn in list(flux_samples.columns):\n",
    "                    if rxn in orig_rxn_ids:\n",
    "                        continue\n",
    "                    elif abs(distribution[1][rxn]) > 1e-6:\n",
    "                        new_rxn_ids |= set([rxn])\n",
    "            \n",
    "        else:\n",
    "            print('Finding active reactions in single flux distribution')\n",
    "            rxns = list(solution.fluxes.index)\n",
    "            fluxes = list(solution.fluxes)\n",
    "            for flux in range(0, len(fluxes)):\n",
    "                if rxns[flux] in orig_rxn_ids:\n",
    "                    continue\n",
    "                elif abs(fluxes[flux]) > 1e-6:\n",
    "                    new_rxn_ids |= set([rxns[flux]])\n",
    "    \n",
    "    # Screen new reaction IDs\n",
    "    new_rxn_ids = new_rxn_ids.difference(orig_rxn_ids)\n",
    "    \n",
    "    # Get reactions and metabolites to be added to the model\n",
    "    new_rxns = copy.deepcopy([reaction_bag.reactions.get_by_id(rxn) for rxn in new_rxn_ids])\n",
    "    new_cpd_ids = set()\n",
    "    for rxn in new_rxns: new_cpd_ids |= set([str(x.id) for x in list(rxn.metabolites)])\n",
    "    new_cpd_ids = new_cpd_ids.difference(orig_cpd_ids)\n",
    "    new_cpds = copy.deepcopy([reaction_bag.metabolites.get_by_id(cpd) for cpd in new_cpd_ids])\n",
    "    \n",
    "    # Create gapfilled model \n",
    "    new_model = copy.deepcopy(model)\n",
    "    new_model.add_metabolites(new_cpds)\n",
    "    new_model.add_reactions(new_rxns)\n",
    "    \n",
    "    # Identify extracellular metabolites still need exchanges\n",
    "    new_exchanges = extend_exchanges(new_model, new_cpd_ids, extracellular)\n",
    "    \n",
    "    # Report to user\n",
    "    new_rxn_ids = len(new_rxn_ids) + new_exchanges\n",
    "    new_cpd_ids = len(new_cpd_ids)\n",
    "    print('Gapfilled ' + str(new_rxn_ids) + ' reactions and ' + str(new_cpd_ids) + ' metabolites') \n",
    "    if new_model.slim_optimize() > 1e-6: print('Objective now carries flux')\n",
    "\n",
    "    duration = str(round(((time.time() - start_time) / 60.0), 1)).split('.')\n",
    "    mins = duration[0]\n",
    "    secs = str(round(((float(duration[1]) / 10.0) + 60.0)))\n",
    "    print('Finished in ' + mins + ' minutes and ' + secs + ' seconds')\n",
    "    return new_model\n",
    "\n",
    "\n",
    "# Adds missing exchanges for extracellulart metabolites\n",
    "def extend_exchanges(model, cpd_ids, outside):\n",
    "        \n",
    "    model_exchanges = set()\n",
    "    for rxn in model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            if outside in [str(cpd.compartment) for cpd in rxn.metabolites]:\n",
    "                model_exchanges |= set([rxn.id])\n",
    "\n",
    "    new_ex_ids = 0\n",
    "    cpds = [model.metabolites.get_by_id(cpd) for cpd in cpd_ids if model.metabolites.get_by_id(cpd).compartment == outside]\n",
    "    for cpd in cpds:\n",
    "        rxns = set([str(rxn.id) for rxn in cpds.reactions])\n",
    "        if bool(rxns & model_exchanges) == False:\n",
    "            new_id = 'EX_' + cpd.id\n",
    "            model.add_boundary(cpd, type='exchange', reaction_id=new_id, lb=-1000.0, ub=1000.0)\n",
    "            new_ex_ids += 1\n",
    "\n",
    "    return new_ex_ids\n",
    "\n",
    "\n",
    "# Returns the reaction ID of the objective reaction\n",
    "def get_objective(model):\n",
    "    \n",
    "    if len(list(model.objective.variables)) == 0:\n",
    "        raise IndexError('Model has no objective set.')\n",
    "    \n",
    "    expression = str(model.objective.expression).split()\n",
    "    if 'reverse' in expression[0]:\n",
    "        obj_id = expression[2].split('*')[-1]\n",
    "    else:\n",
    "        obj_id = expression[0].split('*')[-1]\n",
    "            \n",
    "    return obj_id\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import cobra\n",
    "import symengine\n",
    "\n",
    "# pFBA gapfiller\n",
    "def pfba_gapfill(model, objective, universal, extracellular='extracellular', ignore=[]):\n",
    "    start_time = time.time()\n",
    "    '''\n",
    "    Parameters\n",
    "    ----------\n",
    "    model_file : str\n",
    "        Model to be gapfilled\n",
    "    objective : \n",
    "    universal_file : str\n",
    "        Reaction bag reference\n",
    "    iters : int\n",
    "        Number of iterations to repeat gapfilling process\n",
    "    extracellular : str\n",
    "        Label for extracellular compartment of model\n",
    "    '''\n",
    "    # Create temporary models to modify for gapfilling\n",
    "    temp_model = copy.deepcopy(model)\n",
    "    temp_universal = copy.deepcopy(universal)\n",
    "\n",
    "    # Save some basic network info for downstream membership testing\n",
    "    target_rxns = set([str(x.id) for x in model.reactions])\n",
    "    target_cpds = set([str(y.id) for y in model.metabolites])\n",
    "    ref_rxns = set([str(z.id) for z in universal.reactions])\n",
    "    shared_rxns = ref_rxns.intersection(target_rxns)\n",
    "    \n",
    "    # Define minimum objective value\n",
    "    obj_val = temp_model.slim_optimize()\n",
    "    obj_constraint = constrained_model.problem.Constraint(temp_model.objective.expression, lb=obj_val*0.01, ub=obj_val)\n",
    "    temp_model.add_cons_vars([prev_obj_constraint])\n",
    "    temp_model.solver.update()\n",
    "                \n",
    "    # Open exchanges are all open\n",
    "    for rxn in temp_model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            rxn.bounds = (min(rxn.lower_bound, -1000.), max(rxn.upper_bound, 1000.))\n",
    "    for rxn in temp_universal.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            rxn.bounds = (min(rxn.lower_bound, -1000.), max(rxn.upper_bound, 1000.))\n",
    "        \n",
    "    # Remove overlapping reactions from universal bag, add model reactions to universal bag\n",
    "    for rxn in shared_rxns: temp_universal.reactions.get_by_id(rxn).remove_from_model()\n",
    "    temp_universal.add_reactions(list(copy.deepcopy(model.reactions)))\n",
    "        \n",
    "    # Set up pFBA objective\n",
    "    pfba_expr = symengine.RealDouble(0)\n",
    "    for rxn in temp_universal.reactions:\n",
    "        if rxn.id in target_rxns:\n",
    "            pfba_expr += 1.0 * rxn.forward_variable\n",
    "            pfba_expr += 1.0 * rxn.reverse_variable\n",
    "        else:\n",
    "            pfba_expr += 0.0 * rxn.forward_variable\n",
    "            pfba_expr += 0.0 * rxn.reverse_variable\n",
    "        \n",
    "    temp_universal.objective = temp_universal.problem.Objective(pfba_expr, direction='min', sloppy=True)\n",
    "    temp_universal.solver.update()\n",
    "\n",
    "    # Run FBA and save solution\n",
    "    solution = temp_universal.optimize()\n",
    "    active_rxns = set([rxn.id for rxn in temp_universal.reactions if abs(solution.fluxes[rxn.id]) > 1e-6])\n",
    "    \n",
    "    # Screen new reaction IDs\n",
    "    new_rxns = active_rxns.difference(target_rxns)\n",
    "    \n",
    "    # Get reactions and metabolites to be added to the model\n",
    "    new_rxns = copy.deepcopy([temp_universal.reactions.get_by_id(rxn) for rxn in new_rxns])\n",
    "    new_cpds = set()\n",
    "    for rxn in new_rxns: new_cpds |= set([str(x.id) for x in list(rxn.metabolites)]).difference(target_cpds)\n",
    "    new_cpds = copy.deepcopy([temp_universal.metabolites.get_by_id(cpd) for cpd in new_cpds])\n",
    "    \n",
    "    # Create gapfilled model \n",
    "    new_model = copy.deepcopy(model)\n",
    "    new_model.add_metabolites(new_cpds)\n",
    "    new_model.add_reactions(new_rxns)\n",
    "    \n",
    "    # Identify extracellular metabolites still need exchanges\n",
    "    new_exchs = 0\n",
    "    model_exchanges = set()\n",
    "    rxns = set([str(rxn.id) for rxn in model.reactions])\n",
    "    for rxn in new_model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            if extracellular in [str(cpd.compartment) for cpd in rxn.metabolites]:\n",
    "                model_exchanges |= set([rxn.id])                \n",
    "    for cpd in new_model.metabolites:\n",
    "        if cpd.compartment != extracellular: continue\n",
    "        current_rxns = set([x.id for x in cpd.reactions])\n",
    "        if bool(current_rxns & model_exchanges) == False:\n",
    "            new_id = 'EX_' + cpd.id\n",
    "            new_model.add_boundary(cpd, type='exchange', reaction_id=new_id, lb=-1000.0, ub=1000.0)\n",
    "            new_exchs += 1\n",
    "    \n",
    "    # Report to user\n",
    "    print('\\nGapfilled ' + str(len(new_rxns) + new_exchs) + ' reactions and ' + str(len(new_cpds)) + ' metabolites') \n",
    "    if new_model.slim_optimize() <= 1e-6: print('WARNING: Objective does not carries flux')\n",
    "    \n",
    "    return new_model\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd630_PATRIC = cobra.io.read_sbml_model('data/cdiff_temp.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in curated universal reacton bag\n",
    "universal = cobra.io.load_json_model('data/universal.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress: 100%   \n",
      "Gapfilled 30 reactions and 0 metabolites\n",
      "Objective now carries flux\n",
      "Finished in 18 minutes and 60.0 seconds\n"
     ]
    }
   ],
   "source": [
    "# Gap filling to biomass on complete media\n",
    "\n",
    "# Set metabolic tasks\n",
    "scfa_rxns = ['R01174_5_c','rxn05061_c','R01174_4_c','R01174_3_c','R01174_2_c','rxn01236_c',\n",
    "             'R01174_1_c','rxn11676_c','rxn01610_c','rxn07124_c','rxn00669_c','rxn13713_c', 'R11462_c']\n",
    "\n",
    "# Flux-sampling pFBA gapfill\n",
    "cd630_PATRIC_gapfilled = pfba_ensemble_gapfill(cd630_PATRIC, universal, extracellular='extracellular', tasks=scfa_rxns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>272563.8</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f1b7cb9e050</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1306</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1212</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model 272563.8 at 0x7f1b7cb9e050>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC_gapfilled"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to be added: 30\n"
     ]
    }
   ],
   "source": [
    "# Rerun gapfill until no new reactions were found, reseting the kernel each time\n",
    "\n",
    "iter1 = ['rxn02269_c', 'rxn09142_c', 'rxn09145_c', 'rxn00620_c', 'rxn01105_c', 'rxn05293_c', 'rxn02286_c', 'rxn09147_c', \n",
    "         'rxn01953_c', 'rxn02011_c', 'rxn07466_c', 'rxn09429_c', 'rxn03852_c', 'rxn13140_c', 'rxn09065_c', 'rxn01520_c', \n",
    "         'rxn13914_c', 'rxn09063_c', 'rxn09348_c', 'rxn05744_c', 'rxn08040_c']\n",
    "iter2 = ['rxn02269_c', 'rxn13139_c', 'rxn00620_c', 'rxn05293_c', 'rxn09341_c', 'rxn02286_c', 'rxn01953_c', 'rxn02011_c',\n",
    "         'rxn07466_c', 'rxn09429_c', 'rxn09126_c', 'rxn09124_c', 'rxn13140_c', 'rxn09065_c', 'rxn01520_c', 'rxn00553_c',\n",
    "         'rxn03852_c', 'rxn13914_c', 'rxn09063_c', 'rxn05744_c', 'rxn08040_c', 'rxn09163_c']\n",
    "iter3 = ['rxn02269_c', 'rxn09145_c', 'rxn13139_c', 'rxn00620_c', 'rxn05293_c', 'rxn02286_c', 'rxn09147_c', 'rxn01953_c', 'rxn00605_c',\n",
    "         'rxn02011_c', 'rxn00579_c', 'rxn07466_c', 'rxn09429_c', 'rxn03852_c', 'rxn13140_c', 'rxn09065_c', 'rxn03164_c', 'rxn01520_c',\n",
    "         'rxn09063_c', 'rxn09348_c', 'rxn05744_c', 'rxn08040_c', 'rxn09163_c']\n",
    "iter4 = ['rxn02269_c', 'rxn13139_c', 'rxn00620_c', 'rxn05293_c', 'rxn02286_c', 'rxn01953_c', 'rxn00605_c', 'rxn02011_c', 'rxn00579_c',\n",
    "         'rxn07466_c', 'rxn09429_c', 'rxn09126_c', 'rxn09124_c', 'rxn13140_c', 'rxn09065_c', 'rxn01520_c', 'rxn03852_c', 'rxn13914_c',\n",
    "         'rxn09063_c', 'rxn09348_c', 'rxn05744_c', 'rxn08040_c', 'rxn09163_c']\n",
    "\n",
    "new_rxn_ids = set(iter1 + iter2 + iter3 + iter4)\n",
    "print('Reactions to be added: ' + str(len(new_rxn_ids)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cobra.io.write_sbml_model(cd630_PATRIC_gapfilled, 'data/cdiff_temp2.sbml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Gapfilling to minimal media"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to change media condition based on a list of exchange reaction IDs\n",
    "def changeMedia(full_model, components):\n",
    "    model = copy.deepcopy(full_model)\n",
    "    \n",
    "    media = copy.copy(components)\n",
    "    media = ['EX_' + x for x in media]\n",
    "    \n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.products)) == 0:\n",
    "            if rxn.id in media:\n",
    "                model.reactions.get_by_id(rxn.id).lower_bound = -1000.0\n",
    "                media.remove(rxn.id)\n",
    "            else:\n",
    "                model.reactions.get_by_id(rxn.id).lower_bound = 0.0\n",
    "    if len(media) != 0:\n",
    "        print(media)\n",
    "    return model\n",
    "\n",
    "# Undoes change media conditions\n",
    "def unblockExch(model):\n",
    "    exchanges = []\n",
    "    for rxn in model.reactions:\n",
    "        if len(rxn.reactants) == 0 or len(rxn.products) == 0:\n",
    "            exchanges.append(rxn.id)\n",
    "    \n",
    "    for rxn in exchanges:\n",
    "        model.reactions.get_by_id(rxn).lower_bound = -1000.0\n",
    "        model.reactions.get_by_id(rxn).upper_bound = 1000.0\n",
    "    \n",
    "    # Keep anaerobic\n",
    "    model.reactions.get_by_id('EX_cpd00532_e').lower_bound = 0.0\n",
    "    model.reactions.get_by_id('EX_cpd00532_e').upper_bound = 0.0\n",
    "            \n",
    "    return model\n",
    "\n",
    "\n",
    "# Actually prune all unused metabolites and reactions (innate function does not work)\n",
    "def removeUnused(model):\n",
    "    removed_cpd = set()\n",
    "    removed_rxn = set()\n",
    "    unused_current_cpd = 1\n",
    "    unused_current_rxn = 1\n",
    "    \n",
    "    while unused_current_cpd != 0 or unused_current_rxn != 0:\n",
    "        unused_cpd = prune_unused_metabolites(model)\n",
    "        removed_cpd |= set(unused_cpd)\n",
    "        unused_rxn = prune_unused_reactions(model)\n",
    "        removed_rxn |= set(unused_rxn)\n",
    "        \n",
    "        unused_current_cpd = len(unused_cpd)\n",
    "        unused_current_rxn = len(unused_rxn)\n",
    "    \n",
    "    print('Pruned ' + str(len(removed_cpd)) + ' metabolites from model')\n",
    "    print('Pruned ' + str(len(removed_rxn)) + ' reactions from model')\n",
    "        \n",
    "    return(list(removed_cpd), list(removed_rxn))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Karlsson et al. (1999). Microbiology.\n",
    "mdm = ['cpd00001_e', # water\n",
    "       'cpd00065_e', # L-Tryptophan\n",
    "       'cpd00060_e', # L-Methionine\n",
    "       'cpd00322_e', # L-Isoleucine\n",
    "       'cpd00129_e', # L-Proline\n",
    "       'cpd00156_e', # L-Valine\n",
    "       'cpd00107_e', # L-Leucine\n",
    "       'cpd00084_e', # L-Cysteine \n",
    "       'cpd00149_e', # Cobalt\n",
    "       'cpd00099_e', # Chloride\n",
    "       'cpd10515_e', # Iron\n",
    "       'cpd00030_e', # Manganese\n",
    "       'cpd00254_e', # Magnesium\n",
    "       'cpd00063_e', # Calcium\n",
    "       'cpd00205_e', # Potassium\n",
    "       'cpd00009_e', # Phosphate\n",
    "       'cpd00971_e', # Sodium\n",
    "       'cpd00242_e', # Carbonate\n",
    "       'cpd00104_e', # Biotin\n",
    "       'cpd00644_e', # Pantothenate\n",
    "       'cpd00263_e', # Pyridoxine\n",
    "       'cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "# Theriot et al. (2013). Nature Communications.\n",
    "ncmm = ['cpd00001_e', # water\n",
    "        'cpd00104_e', # Biotin\n",
    "        'cpd00644_e', # Pantothenate\n",
    "        'cpd00263_e', # Pyridoxine\n",
    "        'cpd00149_e', # Cobalt\n",
    "        'cpd00099_e', # Chloride\n",
    "        'cpd10515_e', # Iron\n",
    "        'cpd00030_e', # Manganese\n",
    "        'cpd00254_e', # Magnesium\n",
    "        'cpd00063_e', # Calcium\n",
    "        'cpd00205_e', # Potassium\n",
    "        'cpd00009_e', # Phosphate\n",
    "        'cpd00971_e', # Sodium\n",
    "        'cpd00242_e', # Carbonate\n",
    "        'cpd00322_e', # L-Isoleucine\n",
    "        'cpd00129_e', # L-Proline\n",
    "        'cpd00156_e', # L-Valine\n",
    "        'cpd00107_e', # L-Leucine\n",
    "        'cpd00084_e', # L-Cysteine \n",
    "        'cpd00065_e', # L-Tryptophan\n",
    "        'cpd00060_e', # L-Methionine\n",
    "        'cpd00119_e', # L-Histidine\n",
    "        'cpd00033_e', # Glycine\n",
    "        'cpd00051_e', # L-Arginine\n",
    "        'cpd00161_e'] # L-Threonine\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cd630_PATRIC_gapfilled = cobra.io.read_sbml_model('data/cdiff_temp2.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress: 46.69%Failed to find new reactions twice in a row, ending search\n",
      "Progress: 100%   \n",
      "Gapfilled 38 reactions and 7 metabolites\n",
      "Objective now carries flux\n",
      "Completed in 8.0 minutes\n"
     ]
    }
   ],
   "source": [
    "# change to C. difficle minimal media\n",
    "cd630_PATRIC_mdm = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "cd630_PATRIC_min = pfba_ensemble_gapfill(cd630_PATRIC_mdm, universal, extracellular='extracellular')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to be added: 26\n"
     ]
    }
   ],
   "source": [
    "iter1_mdm = ['rxn01014_c', 'rxn10175_c', 'rxn02476_c', 'rxn00777_c', 'rxn01359_c', 'rxn12676_c', 'rxn02928_c', 'rxn01538_c', \n",
    "             'rxn01182_c', 'rxn12239_c', 'rxn03167_c', 'rxn05039_c', 'rxn10484_c', 'rxn05291_c', 'rxn10030_c', 'rxn01256_c',\n",
    "             'rxn01644_c', 'rxn08131_c', 'rxn00530_c', 'rxn00010_c', 'rxn10053_c', 'rxn01669_c']\n",
    "iter2_mdm = ['rxn01014_c', 'rxn10175_c', 'rxn02476_c', 'rxn00777_c', 'rxn00970_c', 'rxn13251_c', 'rxn02928_c', 'rxn01538_c', \n",
    "             'rxn12239_c', 'rxn03167_c', 'rxn05039_c', 'rxn10484_c', 'rxn05291_c', 'rxn01256_c', 'rxn01644_c', 'rxn08131_c',\n",
    "             'rxn00530_c', 'rxn00958_c', 'rxn00010_c', 'rxn10053_c', 'rxn01669_c','rxn01539_c']\n",
    "iter3_mdm = ['rxn01014_c', 'rxn10175_c', 'rxn02476_c', 'rxn00777_c', 'rxn12676_c', 'rxn13251_c', 'rxn02928_c', 'rxn01538_c',\n",
    "             'rxn01182_c', 'rxn12239_c', 'rxn03167_c', 'rxn05039_c', 'rxn10484_c', 'rxn10030_c', 'rxn01256_c', 'rxn01644_c',\n",
    "             'rxn08131_c', 'rxn00530_c', 'rxn00010_c', 'rxn10053_c', 'rxn01669_c']\n",
    "\n",
    "mdm_rxn_ids = set(iter1_mdm + iter2_mdm + iter3_mdm)\n",
    "print('Reactions to be added: ' + str(len(mdm_rxn_ids)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress: 13.34%Failed to find new reactions twice in a row, ending search\n",
      "Progress: 100%   \n",
      "Gapfilled 0 reactions and 0 metabolites\n",
      "Objective now carries flux\n",
      "Completed in 3.1 minutes\n"
     ]
    }
   ],
   "source": [
    "# change to C. difficle no-carbohydrate minimal media\n",
    "cd630_PATRIC_ncmm = changeMedia(cd630_PATRIC_min, ncmm)\n",
    "cd630_PATRIC_min2 = pfba_ensemble_gapfill(cd630_PATRIC_ncmm, universal, extracellular='extracellular')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactions to be added: 8\n"
     ]
    }
   ],
   "source": [
    "iter1_ncmm = ['rxn00285_c', 'rxn05227_c', 'rxn10036_c', 'rxn10840_c', 'rxn00341_c', 'rxn13012_c', 'rxn09119_c',\n",
    "              'rxn10663_c']\n",
    "iter2_ncmm = ['rxn00285_c', 'rxn05227_c', 'rxn10036_c', 'rxn10840_c', 'rxn00341_c', 'rxn13012_c', 'rxn09119_c',\n",
    "              'rxn10663_c']\n",
    "\n",
    "ncmm_rxn_ids= set(iter1_ncmm + iter2_ncmm)\n",
    "print('Reactions to be added: ' + str(len(ncmm_rxn_ids)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gapfilled_rxn_ids = set(list(new_rxn_ids) + list(mdm_rxn_ids) + list(ncmm_rxn_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 extracellular metabolites are missing transport reactions\n"
     ]
    }
   ],
   "source": [
    "# Gapfill model with union of pFBA gapfill iterations\n",
    "new_rxns = copy.deepcopy([universal.reactions.get_by_id(rxn) for rxn in gapfilled_rxn_ids])\n",
    "new_cpd_ids = set()\n",
    "for rxn in new_rxns: new_cpd_ids |= set([str(x.id) for x in list(rxn.metabolites)])\n",
    "orig_cpd_ids = set([str(y.id) for y in cd630_PATRIC.metabolites])\n",
    "new_cpd_ids = new_cpd_ids.difference(orig_cpd_ids)\n",
    "new_cpds = copy.deepcopy([universal.metabolites.get_by_id(cpd) for cpd in new_cpd_ids])\n",
    "\n",
    "cd630_PATRIC_gapfilled = copy.deepcopy(cd630_PATRIC)\n",
    "cd630_PATRIC_gapfilled.add_metabolites(new_cpds)\n",
    "cd630_PATRIC_gapfilled.add_reactions(new_rxns)\n",
    "\n",
    "trans, ex = missingRxn(cd630_PATRIC_gapfilled)\n",
    "for x in ex:\n",
    "    try:\n",
    "        new_id = 'EX_' + x\n",
    "        cpd = cd630_PATRIC_gapfilled.metabolites.get_by_id(x)\n",
    "        cd630_PATRIC_gapfilled.add_boundary(cpd, type='exchange', reaction_id=new_id, lb=-1000.0, ub=1000.0)\n",
    "    except ValueError:\n",
    "        pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'15.808 minutes'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doubling(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'48.873 minutes'"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'50.831 minutes'"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, ncmm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Collect and label gapfilled reactions\n",
    "for rxn in gapfilled_rxn_ids:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id(rxn).notes = 'gapfilled'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(cd630_PATRIC_gapfilled, 'data/cdf_temp3.sbml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fixing remaining mass-balancing reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removes reactions from unbalanced list that contain R-groups in their substrate formulas, or are charged imbalanced\n",
    "def trueUnbalance(model, ignore=[]):\n",
    "    imbal = []\n",
    "    rxn_ids = [str(x.id) for x in model.reactions]\n",
    "    \n",
    "    for rxn in rxn_ids:\n",
    "        test = list(model.reactions.get_by_id(rxn).check_mass_balance().keys())\n",
    "        \n",
    "        if rxn in ignore:\n",
    "            continue\n",
    "        elif len(test) == 0:\n",
    "            continue\n",
    "        elif test == ['charge']:\n",
    "            continue\n",
    "        elif 'EX_' in rxn:\n",
    "            continue\n",
    "            \n",
    "        formula = ''.join([x.formula for x in list(model.reactions.get_by_id(rxn).metabolites)])\n",
    "        if 'R' not in formula:\n",
    "            imbal.append(str(rxn))\n",
    "    \n",
    "    formula_dict = {}\n",
    "    for x in imbal:\n",
    "        react = [i.formula for i in model.reactions.get_by_id(x).reactants]\n",
    "        prod = [j.formula for j in model.reactions.get_by_id(x).products]\n",
    "        if react == prod:\n",
    "            imbal.remove(x)\n",
    "            continue\n",
    "        else:\n",
    "            formula_dict[x] = [react, prod]\n",
    "    \n",
    "    print(str(len(formula_dict.keys())) + ' mass-unbalanced reactions')\n",
    "    return(formula_dict)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cd630_PATRIC_gapfilled = cobra.io.read_sbml_model('data/cdf_temp3.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 mass-unbalanced reactions\n"
     ]
    }
   ],
   "source": [
    "true_imbal = trueUnbalance(cd630_PATRIC_gapfilled, ignore=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00106_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00106_c')\n",
    "cpd00282_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00282_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00247_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00247_c')\n",
    "try:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13251_c').remove_from_model()\n",
    "except:\n",
    "    pass\n",
    "rxn13251_c = cobra.Reaction('rxn13251_c') \n",
    "rxn13251_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn13251_c.name = 'Fumarate dependent DHORD'\n",
    "rxn13251_c.lower_bound = -1000.\n",
    "rxn13251_c.upper_bound = 1000.\n",
    "rxn13251_c.add_metabolites({\n",
    "    cpd00106_c: -1.0,\n",
    "    cpd00282_c: -1.0,\n",
    "    cpd00067_c: -2.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd00247_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn13251_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00067_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00067_c')\n",
    "cpd00239_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00239_c')\n",
    "cpd00822_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00822_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00135_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00135_c')\n",
    "try:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05957_c').remove_from_model()\n",
    "except:\n",
    "    pass\n",
    "rxn05957_c = cobra.Reaction('rxn05957_c') \n",
    "rxn05957_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn05957_c.name = 'O4-succinyl-L-homoserine:hydrogen sulfide S-(3-amino-3-carboxypropyl)transferase'\n",
    "rxn05957_c.gene_reaction_rule = '272563.8.peg.1916'\n",
    "rxn05957_c.lower_bound = 0.\n",
    "rxn05957_c.upper_bound = 1000.\n",
    "rxn05957_c.add_metabolites({\n",
    "    cpd00067_c: -2.0,\n",
    "    cpd00239_c: -1.0,\n",
    "    cpd00822_c: -1.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd00135_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn05957_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00002_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00002_c')\n",
    "cpd00010_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00010_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00008_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00008_c')\n",
    "cpd00009_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00009_c')\n",
    "cpd00078_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00078_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00285_c').remove_from_model()\n",
    "rxn00285_c = cobra.Reaction('rxn00285_c') \n",
    "rxn00285_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn00285_c.name = 'Succinate:CoA ligase (ADP-forming)'\n",
    "rxn00285_c.lower_bound = 0.\n",
    "rxn00285_c.upper_bound = 1000.\n",
    "rxn00285_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00010_c: -1.0,\n",
    "    cpd00036_c: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00078_c: 1.0,\n",
    "    cpd00067_c: 2.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn00285_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00001_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00001_c')\n",
    "cpd03989_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03989_c')\n",
    "cpd00461_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00461_c')\n",
    "cpd01711_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01711_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn03665_c').remove_from_model()\n",
    "rxn03665_c = cobra.Reaction('rxn03665_c') \n",
    "rxn03665_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn03665_c.name = '2-Hydroxy-6-oxo-7-methylocta-2,4-dienoate acylhydrolase'\n",
    "rxn03665_c.gene_reaction_rule = '272563.8.peg.3588'\n",
    "rxn03665_c.lower_bound = 0.\n",
    "rxn03665_c.upper_bound = 1000.\n",
    "rxn03665_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd03989_c: -1.0,\n",
    "    cpd00461_c: 1.0,\n",
    "    cpd01711_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn03665_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cpd00001_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00001_c')\n",
    "cpd02731_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd02731_c')\n",
    "cpd00036_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00036_c')\n",
    "cpd00461_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00461_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn01895_c').remove_from_model()\n",
    "rxn01895_c = cobra.Reaction('rxn01895_c') \n",
    "rxn01895_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "rxn01895_c.name = '2-Hydroxy-6-oxonona-2,4-diene-1,9-dioate succinylhydrolase'\n",
    "rxn01895_c.gene_reaction_rule = '272563.8.peg.3588'\n",
    "rxn01895_c.lower_bound = 0.\n",
    "rxn01895_c.upper_bound = 1000.\n",
    "rxn01895_c.add_metabolites({  \n",
    "    cpd00067_c: -1.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd02731_c: -1.0,\n",
    "    cpd00036_c: 1.0,\n",
    "    cpd00461_c: 1.0 \n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([rxn01895_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "C21400_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('C21400_c')\n",
    "cpd01646_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01646_c')\n",
    "C21399_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('C21399_c')\n",
    "cpd00047_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00047_c')\n",
    "cpd23040_c = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd23040_c')\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('R11462_c').remove_from_model()\n",
    "R11462_c = cobra.Reaction('R11462_c') \n",
    "R11462_c.notes = {'citations': [], 'type': 'metabolic', 'annotation': 'ModelSEED'}\n",
    "R11462_c.name = '4-methylpentanoyl-CoA:(R)-2-hydroxy-4-methylpentanoate CoA-transferase [EC:2.8.3.24]'\n",
    "R11462_c.gene_reaction_rule = 'CD630_03950'\n",
    "R11462_c.lower_bound = 0.\n",
    "R11462_c.upper_bound = 1000.\n",
    "R11462_c.add_metabolites({  \n",
    "    C21400_c: -1.0,\n",
    "    cpd01646_c: -1.0,\n",
    "    C21399_c: 1.0,\n",
    "    cpd00047_c: 1.0,\n",
    "    cpd23040_c: 1.0,\n",
    "    cpd00067_c: 1.0\n",
    "})\n",
    "cd630_PATRIC_gapfilled.add_reactions([R11462_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        nogene_gapfilled_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "                if m.reactions.get_by_id(rxn).notes == 'gapfilled':\n",
    "                    nogene_gapfilled_blocked.append(rxn)\n",
    "    \n",
    "    print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    print(str(len(nogene_blocked)) + ' reactions without GPRs are blocked')\n",
    "    print(str(len(nogene_gapfilled_blocked)) + ' gapfilled reactions without GPRs are blocked')\n",
    "    \n",
    "    return blocked, nogene_blocked, nogene_gapfilled_blocked\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "519 total reactions are blocked\n",
      "33 reactions without GPRs are blocked\n",
      "0 gapfilled reactions without GPRs are blocked\n"
     ]
    }
   ],
   "source": [
    "# Check for blocked reactions\n",
    "blocked, no_gene, gapfilled_blocked = blockedReactions(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gapfilled reaction with ATP of GTP: 8\n"
     ]
    }
   ],
   "source": [
    "check = []\n",
    "for x in gapfilled_rxn_ids:\n",
    "    rxn = cd630_PATRIC_gapfilled.reactions.get_by_id(x)\n",
    "    cpds = [str(y.id) for y in rxn.metabolites]\n",
    "    if 'cpd00002_c' in cpds or 'cpd00038_c' in cpds:\n",
    "        if rxn.lower_bound < -1e-6 and rxn.upper_bound > 1e-6:\n",
    "            check.append(str(rxn.id))\n",
    "print('Gapfilled reaction with ATP of GTP: ' + str(len(check)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct gapfilled reaction reversibility\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn01182_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13012_c').lower_bound = 0.0 # really slows it down, consumes 6 ATP\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10663_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10053_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13139_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn13914_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn09562_c').lower_bound = 0.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01953_c.bounds = (-1000, 0)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn00292_c.bounds = (-1000, 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'5.322 hours'"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, ncmm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2.321 hours'"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'35.606 minutes'"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doubling(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove reactions with no GPR that are gapfilled and don't have strong biological evidence\n",
    "remove_blocked = ['EX_cpd11576_e','EX_cpd04097_e','EX_cpd01012_e','EX_cpd00058_e',\n",
    "                  'EX_cpd11578_e','EX_cpd11579_e','EX_cpd00210_e','EX_cpd00532_e',\n",
    "                  'EX_cpd00176_e','EX_cpd03724_e','EX_cpd03696_e','EX_cpd08023_e',\n",
    "                  'EX_cpd01092_e','EX_cpd01262_e','EX_cpd11596_e','EX_cpd01914_e',\n",
    "                  'EX_cpd00531_e','EX_cpd11575_e','EX_cpd09878_e','EX_cpd01329_e',\n",
    "                  'rxn09030_c']\n",
    "\n",
    "for rxn in remove_blocked:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id(rxn).remove_from_model()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2.321 hours'"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = changeMedia(cd630_PATRIC_gapfilled, mdm)\n",
    "doubling(test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'35.606 minutes'"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doubling(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Edit SBML to ensure compatibility with Cobra read-in function"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pruning model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Identify potentially gapfilled reactions\n",
    "def missingGPR(model, exclude=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass']):\n",
    "    gapfilled = []\n",
    "    transport = findTransports(model)\n",
    "    if not type(exclude) is list:\n",
    "        exclude = [exclude]\n",
    "        \n",
    "    for index in model.reactions:\n",
    "        if len(list(index.genes)) == 0:\n",
    "            if not index in model.boundary:\n",
    "                if not index.id in exclude or not index.id in transport:\n",
    "                    gapfilled.append(index.id)\n",
    "    \n",
    "    if len(gapfilled) > 0:\n",
    "        print(str(len(gapfilled)) + ' metabolic reactions not associated with genes')\n",
    "    \n",
    "    return gapfilled\n",
    "\n",
    "# Check for missing transport and exchange reactions\n",
    "def missingRxn(model, extracellular='extracellular'):\n",
    "\n",
    "    transporters = set(findTransports(model))\n",
    "    exchanges = set([str(rxn.id) for rxn in model.reactions if len(list(rxn.products)) == 0])\n",
    "    \n",
    "    missing_exchanges = []\n",
    "    missing_transports = []\n",
    "    \n",
    "    for metabolite in model.metabolites:\n",
    "        if not metabolite.compartment == extracellular:\n",
    "            continue\n",
    "\n",
    "        curr_rxns = set([x.id for x in list(metabolite.reactions)])\n",
    "        \n",
    "        if not bool(curr_rxns & transporters):\n",
    "            missing_transports.append(metabolite.id)\n",
    "        if not bool(curr_rxns & exchanges):\n",
    "            missing_exchanges.append(metabolite.id)\n",
    "    \n",
    "    if len(missing_transports) != 0:\n",
    "        print(str(len(missing_transports)) + ' extracellular metabolites are missing transport reactions')\n",
    "    if len(missing_exchanges) != 0:\n",
    "        print(str(len(missing_exchanges)) + ' extracellular metabolites are missing exchange reactions')\n",
    "        \n",
    "    return missing_transports, missing_exchanges\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(raw_model, cytosol='cytosol'):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        \n",
    "        # Close all exchanges\n",
    "        for index in model.boundary:\n",
    "            model.reactions.get_by_id(index.id).lower_bound = 0.\n",
    "        \n",
    "        # Identify all metabolites that are produced within the network\n",
    "        demand_metabolites = [x.reactants[0].id for x in model.demands if len(x.reactants) > 0] + [x.products[0].id for x in model.demands if len(x.products) > 0]\n",
    "\n",
    "        free = []\n",
    "        for index in model.metabolites: \n",
    "            if index.id in demand_metabolites:\n",
    "                continue\n",
    "            elif not index.compartment in cytosol:\n",
    "                continue\n",
    "            else:\n",
    "                demand = model.add_boundary(index, type='demand')\n",
    "                model.objective = demand\n",
    "                obj_val = model.slim_optimize(error_value=0.)\n",
    "                if obj_val > 1e-8:\n",
    "                    free.append(index.id)\n",
    "                model.remove_reactions([demand])\n",
    "    \n",
    "    if len(free) > 0:\n",
    "        print(str(len(free)) + ' metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(raw_model, exclude=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass']):\n",
    "    \n",
    "    with raw_model as model:\n",
    "        imbalanced = []\n",
    "        mass_imbal = 0\n",
    "        charge_imbal = 0\n",
    "        elem_set = set()\n",
    "        for metabolite in model.metabolites:\n",
    "            try:\n",
    "                elem_set |= set(metabolite.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        if len(elem_set) == 0:\n",
    "            imbalanced = model.reactions\n",
    "            mass_imbal = len(model.reactions)\n",
    "            charge_imbal = len(model.reactions)\n",
    "            print('No elemental data associated with metabolites!')\n",
    "        \n",
    "        else:\n",
    "            if not type(exclude) is list: \n",
    "                exclude = [exclude]\n",
    "            for index in model.reactions:\n",
    "                if index in model.boundary or index.id in exclude:\n",
    "                    continue\n",
    "\n",
    "                else:\n",
    "                    try:\n",
    "                        test = index.check_mass_balance()\n",
    "                    except ValueError:\n",
    "                        continue\n",
    "\n",
    "                    if len(list(test)) > 0:\n",
    "                        #imbalanced.append(index.id)\n",
    "\n",
    "                        #if 'charge' in test.keys():\n",
    "                        #    charge_imbal += 1\n",
    "                        if len(set(test.keys()).intersection(elem_set)) > 0:\n",
    "                            mass_imbal += 1\n",
    "                            imbalanced.append(index.id)\n",
    "\n",
    "    if mass_imbal != 0:\n",
    "        print(str(mass_imbal) + ' reactions are mass imbalanced')\n",
    "    #if charge_imbal != 0:\n",
    "    #    print(str(charge_imbal) + ' reactions are charge imbalanced')\n",
    "    \n",
    "    return(imbalanced)\n",
    "\n",
    "\n",
    "def checkDetermination(model):\n",
    "    if len(model.reactions) < len(model.metabolites):\n",
    "        print('Model is overdetermined')\n",
    "    if len(model.reactions) > len(model.metabolites):\n",
    "        print('Model is underdetermined')\n",
    "\n",
    "\n",
    "# Checks the quality of models by a couple metrics and returns problems\n",
    "def checkQuality(model, exclude=['dna_rxn','rna_rxn','protein_rxn','teichoicacid_rxn','peptidoglycan_rxn','lipid_rxn','cofactor_rxn','biomass'], cytosol='cytosol'):\n",
    "    \n",
    "    checkDetermination(model)\n",
    "    gaps = missingGPR(model, exclude)\n",
    "    freemass = checkFreeMass(model, cytosol)\n",
    "    balance = checkBalance(model, exclude)\n",
    "    trans, exch = missingRxn(model)\n",
    "    \n",
    "    test = gaps + freemass + balance\n",
    "    if len(test) == 0:\n",
    "        print('No inconsistencies detected')\n",
    "\n",
    "    return gaps, freemass, balance, trans, exch\n",
    "\n",
    "\n",
    "# Prints the names of reactions or metabolites based on IDs\n",
    "def printNames(model, id_list):\n",
    "    for x in id_list:\n",
    "        try:\n",
    "            print(model.metabolites.get_by_id(x).name)\n",
    "            continue\n",
    "        except:\n",
    "            pass\n",
    "        \n",
    "        print(model.reactions.get_by_id(x).name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model is overdetermined\n",
      "108 metabolic reactions not associated with genes\n",
      "3 extracellular metabolites are missing transport reactions\n",
      "20 extracellular metabolites are missing exchange reactions\n"
     ]
    }
   ],
   "source": [
    "no_gpr, freemass, unbalanced, no_trans, no_exch = checkQuality(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix no transporter reactions\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00034_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('EX_cpd00034_e').remove_from_model()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix no exchanges\n",
    "\n",
    "# L-methionine R-oxide\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11576_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11576_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05614_c').remove_from_model()\n",
    "\n",
    "# Pb\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd04097_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd04097_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10344_c').remove_from_model()\n",
    "\n",
    "# Cd+\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01012_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01012_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05516_c').remove_from_model()\n",
    "\n",
    "# Cu2+\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00058_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00058_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10481_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05528_c').remove_from_model()\n",
    "\n",
    "# hexanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11578_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11578_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05585_c').remove_from_model()\n",
    "\n",
    "# ethanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11579_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11579_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05552_c').remove_from_model()\n",
    "\n",
    "# Taurine\n",
    "cpd00210_e = cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00210_e')\n",
    "cd630_PATRIC_gapfilled.add_boundary(cpd00210_e, type='exchange', reaction_id='EX_cpd00210_e', lb=-1000., ub=1000.)\n",
    "\n",
    "# 2-keto-3-deoxygluconate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00176_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00176_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05681_c').remove_from_model()\n",
    "\n",
    "# Ferrichrome\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03724_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03724_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05188_c').remove_from_model()\n",
    "\n",
    "# Ursin\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd03696_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05501_c').remove_from_model()\n",
    "\n",
    "# Allantoin\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01092_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01092_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05682_c').remove_from_model()\n",
    "\n",
    "# Amylotriose\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01262_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01262_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05608_c').remove_from_model()\n",
    "\n",
    "# butanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11596_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11596_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05512_c').remove_from_model()\n",
    "\n",
    "# L-Methionine S-oxide\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01914_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01914_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05613_c').remove_from_model()\n",
    "\n",
    "# Hg\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00531_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00531_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10343_c').remove_from_model()\n",
    "\n",
    "# MOPS\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11575_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11575_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05620_c').remove_from_model()\n",
    "\n",
    "# Sulfoacetate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd09878_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd09878_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05656_c').remove_from_model()\n",
    "\n",
    "# Maltohexaose\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01329_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd01329_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn10174_c').remove_from_model()\n",
    "\n",
    "# methanesulfonate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd08023_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd08023_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn05621_c').remove_from_model()\n",
    "\n",
    "# O2-\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00532_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd00532_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00206_c').remove_from_model()\n",
    "\n",
    "# Molybdate\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11574_e').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.metabolites.get_by_id('cpd11574_c').remove_from_model()\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('EX_cpd11574_e').remove_from_model()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model is overdetermined\n",
      "108 metabolic reactions not associated with genes\n"
     ]
    }
   ],
   "source": [
    "no_gpr, freemass, unbalanced, no_trans, no_exch = checkQuality(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct obvious auxotrophies\n",
    "\n",
    "# TTP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn10060_c.upper_bound = 0.0\n",
    "\n",
    "# dAMP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01507_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01507_c.upper_bound = 0.0\n",
    "\n",
    "# dAMP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01218_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01218_c.upper_bound = 0.0\n",
    "\n",
    "# dGMP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01445_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01445_c.upper_bound = 0.0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix last reaction reversibilities\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01538_c.lower_bound = 0.0 # ATP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01105_c.lower_bound = 0.0 # GTP\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01105_c.lower_bound = 0.0 # GTP\n",
    "\n",
    "# GTP biosynthesis\n",
    "cd630_PATRIC_gapfilled.reactions.rxn13139_c.lower_bound = -1000.0\n",
    "cd630_PATRIC_gapfilled.reactions.rxn13139_c.upper_bound = 0.0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fix incorrect biosynthesis of limiting amino acids\n",
    "\n",
    "# L-Proline - cpd00129_c\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00931_c').lower_bound = 0.\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00929_c').lower_bound = 0.\n",
    "\n",
    "# L-Cysteine - cpd00084_c\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn12844_c').lower_bound = 0.\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn00650_c').lower_bound = 0.\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn09177_c').lower_bound = 0.\n",
    "\n",
    "# L-Leucine - cpd00107_c\n",
    "cd630_PATRIC_gapfilled.reactions.get_by_id('rxn12845_c').lower_bound = 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Close dipeptide exchange reactions\n",
    "dipeptides = ['EX_cpd11585_e','EX_cpd11582_e','EX_cpd11586_e','EX_cpd01017_e','EX_cpd11591_e',\n",
    "              'EX_cpd11593_e','EX_cpd15605_e','EX_cpd15603_e','EX_cpd11580_e','EX_cpd15606_e',\n",
    "              'EX_cpd11583_e','EX_cpd11589_e','EX_cpd11588_e','EX_cpd15604_e','EX_cpd11592_e',\n",
    "              'EX_cpd11587_e','EX_cpd11590_e','EX_cpd11581_e','EX_cpd11584_e']\n",
    "\n",
    "for x in dipeptides:\n",
    "    cd630_PATRIC_gapfilled.reactions.get_by_id(x).lower_bound = 0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pruned 90 metabolites from model\n",
      "Pruned 7 reactions from model\n"
     ]
    }
   ],
   "source": [
    "unused_c, unused_r = removeUnused(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found and removed 0 reactions\n"
     ]
    }
   ],
   "source": [
    "# Removed orphaned reactions with no substrates\n",
    "removed = 0\n",
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    if len(list(rxn.reactants)) == 0 and len(list(rxn.products)) == 0:\n",
    "        cd630_PATRIC_gapfilled.reactions.get_by_id(rxn.id).remove_from_model()\n",
    "        removed += 1\n",
    "print('Found and removed ' + str(removed) + ' reactions')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "481 total reactions are blocked\n",
      "11 reactions without GPRs are blocked\n",
      "0 gapfilled reactions without GPRs are blocked\n"
     ]
    }
   ],
   "source": [
    "# Check blocked reactions\n",
    "blocked, nogene_blocked, nogene_gapfilled_blocked = blockedReactions(cd630_PATRIC_gapfilled)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove uninformative metabolites/reactions\n",
    "remove_cpd = ['cpd11440_c', #fa6\n",
    "              'cpd11438_c', #fa4\n",
    "              'cpd00666_c', #Tartrate\n",
    "              'cpd00158_c', #CELB\n",
    "              'cpd11433_c', #fa12\n",
    "              'cpd00392_c', #Phycite\n",
    "              'cpd11430_c', #fa1\n",
    "              'cpd11431_c', #fa11\n",
    "              'cpd11436_c'] #fa3\n",
    "for cpd in remove_cpd:\n",
    "    try:\n",
    "        metabolite = cd630_PATRIC_gapfilled.metabolites.get_by_id(cpd)\n",
    "        cd630_PATRIC_gapfilled.remove_reactions(metabolite.reactions)\n",
    "        cd630_PATRIC_gapfilled.remove_metabolites([metabolite])\n",
    "    except KeyError:\n",
    "        print(metabolite + ' not found')\n",
    "        continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inactive_metabolites = [m for m in cd630_PATRIC_gapfilled.metabolites if len(m.reactions) == 0]\n",
    "cd630_PATRIC_gapfilled.remove_metabolites(inactive_metabolites)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "792"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(cd630_PATRIC_gapfilled.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct names\n",
    "cd630_PATRIC_gapfilled.metabolites.cpd00443_e.name = 'ABEE'\n",
    "cd630_PATRIC_gapfilled.reactions.rxn03185_c.name = 'alpha-N-acetylneuraminate alpha-2,8-sialyltransferase'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdJ792</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07fd2281f0350</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1179</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1194</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdJ792 at 0x7fd2281f0350>"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd630_PATRIC_gapfilled"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add RefSeq annotations\n",
    "\n",
    "patric_refseq = {}\n",
    "with open('/home/matthew/Desktop/repos/Jenior_iCdJ794_2019/data/patric_refseq_genes.tsv') as ref:\n",
    "    for line in ref:\n",
    "        line = line.split()\n",
    "        patric_refseq[line[0]] = line[1]\n",
    "        \n",
    "for gene in cd630_PATRIC_gapfilled.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        try:\n",
    "            gene.annotation['refseq'] = patric_refseq[gene.id]\n",
    "        except:\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "\n",
    "# Metabolites\n",
    "for cpd in cd630_PATRIC_gapfilled.metabolites:\n",
    "    if 'cpd' in cpd.id:\n",
    "        new_id = str(cpd.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        cpd.annotation['seed.compound'] = new_id\n",
    "\n",
    "cd630_PATRIC_gapfilled.metabolites.C21400_c.annotation['kegg'] = 'C21400'        \n",
    "cd630_PATRIC_gapfilled.metabolites.C21399_c.annotation['kegg'] = 'C21399'        \n",
    "cd630_PATRIC_gapfilled.metabolites.C21090_c.annotation['kegg'] = 'C21090'        \n",
    "cd630_PATRIC_gapfilled.metabolites.C21399_e.annotation['kegg'] = 'C21399'        \n",
    "\n",
    "\n",
    "# Reactions\n",
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    if 'rxn' in rxn.id or 'EX_' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['seed.reaction'] = new_id\n",
    "    elif 'R' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['kegg.genes'] = new_id\n",
    "\n",
    "cd630_PATRIC_gapfilled.reactions.SK_cpd11416_c.annotation['seed.reaction'] = 'SK_cpd11416'\n",
    "cd630_PATRIC_gapfilled.reactions.ENOG4108HXH_c.annotation['eggnog'] = 'ENOG4108HXH'\n",
    "cd630_PATRIC_gapfilled.reactions.COG3601_c.annotation['cog'] = 'COG3601'\n",
    "cd630_PATRIC_gapfilled.reactions.K20025_c.annotation['kegg.genes'] = 'K20025'\n",
    "\n",
    "\n",
    "# Genes\n",
    "for gene in cd630_PATRIC_gapfilled.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        gene.annotation['patric'] = 'fig|' + gene.id\n",
    "    elif 'CD630_' in gene.id:\n",
    "        gene.annotation['uniprot'] = gene.id\n",
    "\n",
    "cd630_PATRIC_gapfilled.genes.T4BV06_CLODI.annotation['kegg'] = 'T4BV06_CLODI'\n",
    "cd630_PATRIC_gapfilled.genes.K00529.annotation['kegg'] = 'K00529'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add SBOs\n",
    "\n",
    "# Metabolites\n",
    "for cpd in cd630_PATRIC_gapfilled.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "\n",
    "# Reactions\n",
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    substrates = list(rxn.metabolites)\n",
    "    compartments = set([x.compartment for x in substrates])\n",
    "    \n",
    "    if 'EX_' in rxn.id:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000627' # exchange\n",
    "    elif len(compartments) > 1:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "    else:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "\n",
    "# Biomass\n",
    "cd630_PATRIC_gapfilled.reactions.dna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.rna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.teichoicacid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.peptidoglycan_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.cellwall_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.lipid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.cofactor_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.biomass.annotation['sbo'] = 'SBO:0000629'  \n",
    "cd630_PATRIC_gapfilled.reactions.SK_cpd11416_c.annotation['sbo'] = 'SBO:0000632'  \n",
    "\n",
    "\n",
    "# Genes\n",
    "for gene in cd630_PATRIC_gapfilled.genes:\n",
    "    gene.annotation['sbo'] = 'SBO:0000243'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import flux_variability_analysis\n",
    "\n",
    "def test_unbounded(model):\n",
    "    \n",
    "    fva_result = flux_variability_analysis(model, fraction_of_optimum=1.0)\n",
    "\n",
    "    lower_bounds = numpy.asarray([rxn.lower_bound for rxn in model.reactions], dtype=float)\n",
    "    upper_bounds = numpy.asarray([rxn.upper_bound for rxn in model.reactions], dtype=float)\n",
    "    lower_bound = numpy.nanmedian(lower_bounds[lower_bounds != 0.0])\n",
    "    upper_bound = numpy.nanmedian(upper_bounds[upper_bounds != 0.0])\n",
    "\n",
    "    # Find those reactions whose flux is close to or outside of the median\n",
    "    # upper or lower bound, i.e., appears unconstrained.\n",
    "    unlimited_flux = fva_result.loc[\n",
    "            numpy.isclose(fva_result[\"maximum\"], upper_bound, atol=1e-7) |\n",
    "            (fva_result[\"maximum\"] > upper_bound) |\n",
    "            numpy.isclose(fva_result[\"minimum\"], lower_bound, atol=1e-7) |\n",
    "            (fva_result[\"minimum\"] < lower_bound)].index.tolist()\n",
    "    \n",
    "    saved = []\n",
    "    for rxn in unlimited_flux:\n",
    "        \n",
    "        reaction = model.reactions.get_by_id(rxn) \n",
    "        substrates = set([x.id for x in reaction.metabolites])\n",
    "        ignore = set(['cpd00003_c','cpd00004_c','cpd00005_c','cpd00006_c'])\n",
    "        \n",
    "        if 'EX_' in rxn:\n",
    "            continue\n",
    "        elif reaction.bounds != (-1000., 1000.): \n",
    "            continue\n",
    "        elif 'transferase' in reaction.name:\n",
    "            continue\n",
    "        elif reaction.name.find('mutase') != -1:\n",
    "            continue\n",
    "        elif 'isomerase' in reaction.name:\n",
    "            continue\n",
    "        elif 'hydrolase' in reaction.name:\n",
    "            continue \n",
    "        elif len(substrates & ignore) >= 1:\n",
    "            continue\n",
    "        else:\n",
    "            saved.append(rxn)\n",
    "        \n",
    "    print('Potentially unbounded reactions in default medium: ' + str(len(saved)))\n",
    "    return saved\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "unlimited = test_unbounded(cd630_PATRIC_gapfilled)\n",
    "\n",
    "# Correct reversibility to close potentially leaky cycles\n",
    "cd630_PATRIC_gapfilled.reactions.rxn01322_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05617_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05607_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05655_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05496_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05202_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn00606_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn10184_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05494_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05610_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn00575_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05209_c.bounds = (-1000., 0.) # Na+ antiporter\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05226_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn02005_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05518_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn29571_c.bounds = (0., 1000.) # Neu5Ac H+ symporter\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05486_c.bounds = (0., 1000.) # N-Acetylmannosamine H+ symporter\n",
    "cd630_PATRIC_gapfilled.reactions.rxn10175_c.bounds = (0., 1000.)\n",
    "cd630_PATRIC_gapfilled.reactions.rxn05485_c.bounds = (0., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct last few gene IDs\n",
    "corrected_genes = {'CD630_00450': '272563.120.peg.49', \n",
    "                   'CD630_01120': '272563.120.peg.124', \n",
    "                   'CD630_01530': '272563.120.peg.166', \n",
    "                   'CD630_01540': '272563.120.peg.167', \n",
    "                   'CD630_01550': '272563.120.peg.168', \n",
    "                   'CD630_01790': '272563.120.peg.195', \n",
    "                   'CD630_02440': '272563.8.peg.262', \n",
    "                   'CD630_03940': '272563.120.peg.414', \n",
    "                   'CD630_03950': '272563.120.peg.415', \n",
    "                   'CD630_03960': '272563.120.peg.416', \n",
    "                   'CD630_03970': '272563.120.peg.417', \n",
    "                   'CD630_03980': '272563.120.peg.418', \n",
    "                   'CD630_06820': '272563.8.peg.717', \n",
    "                   'CD630_07500': '272563.120.peg.772', \n",
    "                   'CD630_07680': '272563.120.peg.790', \n",
    "                   'CD630_08730': '272563.120.peg.897', \n",
    "                   'CD630_08760': '272563.120.peg.900', \n",
    "                   'CD630_10030': '272563.120.peg.1042', \n",
    "                   'CD630_10040': '272563.120.peg.1043', \n",
    "                   'CD630_10100': '272563.120.peg.1049', \n",
    "                   'CD630_10540': '272563.120.peg.1094', \n",
    "                   'CD630_12250': '272563.8.peg.1284', \n",
    "                   'CD630_12590': '272563.120.peg.1317', \n",
    "                   'CD630_14930': '272563.120.peg.1562', \n",
    "                   'CD630_17760': '272563.120.peg.1869', \n",
    "                   'CD630_21960': '272563.120.peg.2318', \n",
    "                   'CD630_21990': '272563.120.peg.2321', \n",
    "                   'CD630_21991': '272563.120.peg.2322', \n",
    "                   'CD630_22770': '272563.120.peg.2404', \n",
    "                   'CD630_23310': '272563.120.peg.2462', \n",
    "                   'CD630_23390': '272563.120.peg.2470', \n",
    "                   'CD630_23410': '272563.120.peg.2472', \n",
    "                   'CD630_23420': '272563.120.peg.2473', \n",
    "                   'CD630_23540': '272563.120.peg.2485', \n",
    "                   'CD630_23710': '272563.120.peg.2504', \n",
    "                   'CD630_23800': '272563.120.peg.2513', \n",
    "                   'CD630_24170': '272563.120.peg.2551', \n",
    "                   'CD630_24180': '272563.120.peg.2552', \n",
    "                   'CD630_25690': '272563.120.peg.2710', \n",
    "                   'CD630_26510': '272563.8.peg.2775', \n",
    "                   'CD630_26550': '272563.120.peg.2798', \n",
    "                   'CD630_30130': '272563.8.peg.3157', \n",
    "                   'CD630_30140': '272563.120.peg.3181', \n",
    "                   'CD630_30150': '272563.120.peg.3182', \n",
    "                   'CD630_32370_1': '272563.120.peg.3419', \n",
    "                   'CD630_32370_2': '272563.120.peg.3419', \n",
    "                   'CD630_32440': '272563.120.peg.3426', \n",
    "                   'CD630_35150': '272563.120.peg.3715',\n",
    "                   'K00529': '272563.120.peg.1609',\n",
    "                   'T4BV06_CLODI': '1151372.4.peg.1723'}\n",
    "cobra.manipulation.modify.rename_genes(cd630_PATRIC_gapfilled, corrected_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Name new model\n",
    "cd630_PATRIC_gapfilled.name = 'Clostridium difficile 630'\n",
    "cd630_PATRIC_gapfilled.id = 'iCdR' + str(len(cd630_PATRIC_gapfilled.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for rxn in cd630_PATRIC_gapfilled.reactions:\n",
    "    fixed_gpr = rxn.gene_reaction_rule.replace('( ','').replace(' )','')\n",
    "    rxn.gene_reaction_rule = fixed_gpr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cobra.io.write_sbml_model(cd630_PATRIC_gapfilled, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.sbml')\n",
    "cobra.io.save_json_model(cd630_PATRIC_gapfilled, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "iCdG791 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gene annotations\n",
    "patric_refseq = {}\n",
    "with open('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/cd630_patric_refseq_genes.tsv') as ref:\n",
    "    for line in ref:\n",
    "        line = line.split()\n",
    "        patric_refseq[line[0]] = line[1]\n",
    "        \n",
    "for gene in iCdG791.genes:\n",
    "    if '.peg.' in gene.id:\n",
    "        try:\n",
    "            gene.annotation['refseq'] = patric_refseq[gene.id]\n",
    "            gene.annotation['patric'] = 'fig|' + gene.id\n",
    "        except:\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Metabolite annotations\n",
    "for cpd in iCdG791.metabolites:\n",
    "    if 'cpd' in cpd.id:\n",
    "        new_id = str(cpd.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        cpd.annotation['seed.compound'] = new_id\n",
    "\n",
    "iCdG791.metabolites.C21400_c.annotation['kegg'] = 'C21400'        \n",
    "iCdG791.metabolites.C21399_c.annotation['kegg'] = 'C21399'        \n",
    "iCdG791.metabolites.C21090_c.annotation['kegg'] = 'C21090'        \n",
    "iCdG791.metabolites.C21399_e.annotation['kegg'] = 'C21399' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reaction annotations\n",
    "for rxn in iCdG791.reactions:\n",
    "    if 'rxn' in rxn.id or 'EX_' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['seed.reaction'] = new_id\n",
    "    elif 'R' in rxn.id:\n",
    "        new_id = str(rxn.id).split('_')\n",
    "        if len(new_id) > 1:\n",
    "            del new_id[-1]\n",
    "        new_id = ''.join(new_id)\n",
    "        rxn.annotation['kegg.genes'] = new_id\n",
    "\n",
    "iCdG791.reactions.SK_cpd11416_c.annotation['seed.reaction'] = 'SK_cpd11416'\n",
    "iCdG791.reactions.ENOG4108HXH_c.annotation['eggnog'] = 'ENOG4108HXH'\n",
    "iCdG791.reactions.COG3601_c.annotation['cog'] = 'COG3601'\n",
    "iCdG791.reactions.K20025_c.annotation['kegg.genes'] = 'K20025'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add SBOs\n",
    "\n",
    "# Metabolites\n",
    "for cpd in iCdG791.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "\n",
    "# Reactions\n",
    "for rxn in iCdG791.reactions:\n",
    "    substrates = list(rxn.metabolites)\n",
    "    compartments = set([x.compartment for x in substrates])\n",
    "    \n",
    "    if 'EX_' in rxn.id:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000627' # exchange\n",
    "    elif len(compartments) > 1:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000185' # transport\n",
    "    else:\n",
    "        rxn.annotation['sbo'] = 'SBO:0000176' # metabolic\n",
    "\n",
    "# Biomass\n",
    "iCdG791.reactions.dna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.rna_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.teichoicacid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.peptidoglycan_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.cellwall_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.lipid_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.cofactor_rxn.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.biomass.annotation['sbo'] = 'SBO:0000629'  \n",
    "iCdG791.reactions.SK_cpd11416_c.annotation['sbo'] = 'SBO:0000632'  \n",
    "\n",
    "# Genes\n",
    "for gene in iCdG791.genes:\n",
    "    gene.annotation['sbo'] = 'SBO:0000243'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct residual gene names\n",
    "gene_name_fixes = {'272563.120.peg.1317':'Na(+)-dependent branched-chain amino acid transporter',\n",
    "'272563.120.peg.772':'ABC transporter, substrate-binding protein (cluster 3, basic aa/glutamine/opines)',\n",
    "'272563.120.peg.1043':'Fumarate hydratase class I, beta region (EC 4.2.1.2); L(+)-tartrate dehydratase beta subunit (EC 4.2.1.32) | fumB',\n",
    "'272563.120.peg.1042':'Fumarate hydratase class I, alpha region (EC 4.2.1.2); L(+)-tartrate dehydratase alpha subunit (EC 4.2.1.32) | fumA',\n",
    "'272563.120.peg.2504':'L-aspartate oxidase (EC 1.4.3.16) | nadB',\n",
    "'272563.120.peg.195':'NAD-specific glutamate dehydrogenase (EC 1.4.1.2) | gluD',\n",
    "'272563.120.peg.2551':'PTS system, glucitol/sorbitol-specific IIB component (EC 2.7.1.198) / PTS system, glucitol/sorbitol-specific IIC component 2 | srlE_2',\n",
    "'272563.120.peg.2552':'PTS system, glucitol/sorbitol-specific IIC component | srlA',\n",
    "'272563.120.peg.1049':'N-acetylgalactosamine-6-phosphate deacetylase | nagA',\n",
    "'272563.120.peg.2404':'ribulose-5-phosphate 4-epimerase and related epimerases and aldolases',\n",
    "'272563.120.peg.49':'Ribulose-5-phosphate 4-epimerase and related epimerases and aldolases',\n",
    "'272563.120.peg.3715':'N-acetylglucosamine-1-phosphate uridyltransferase (EC 2.7.7.23) / Glucosamine-1-phosphate N-acetyltransferase (EC 2.3.1.157) | glmU',\n",
    "'272563.120.peg.3181':'PTS system, IIB component',\n",
    "'272563.8.peg.3157':'PTS system, IIC component',\n",
    "'272563.120.peg.3182':'PTS system, IIA component',\n",
    "'1151372.4.peg.1723':'Branched-chain amino acid aminotransferase (EC 2.6.1.42) | ilvE',\n",
    "'272563.120.peg.900':'ABC transporter, substrate-binding protein',\n",
    "'272563.120.peg.897':'ABC transporter, substrate-binding protein',\n",
    "'272563.8.peg.717':'sodium-solute symporter, putative',\n",
    "'272563.120.peg.2710':'Mannosylglycerate hydrolase (EC 3.2.1.170)',\n",
    "'272563.120.peg.1869':'ABC transporter, ATP-binding protein (cluster 3, basic aa/glutamine/opines)',\n",
    "'272563.120.peg.1562':'DNA-3-methyladenine glycosylase II (EC 3.2.2.21)',\n",
    "'272563.120.peg.124':'Phosphate butyryltransferase (EC 2.3.1.19) | ptb',\n",
    "'272563.120.peg.2322':'2-oxoglutarate/2-oxoacid ferredoxin oxidoreductase, delta subunit, ferredoxin-like 4Fe-4S binding protein (EC 1.2.7.-)',\n",
    "'272563.120.peg.2321':'2-oxoglutarate/2-oxoacid ferredoxin oxidoreductase, alpha subunit (EC 1.2.7.-)',\n",
    "'272563.120.peg.167':'4-hydroxyphenylacetate decarboxylase, small subunit (EC 4.1.1.83) | hpdC',\n",
    "'272563.120.peg.166':'4-hydroxyphenylacetate decarboxylase, large subunit (EC 4.1.1.83) | hpdB',\n",
    "'272563.120.peg.2513':'Indolepyruvate oxidoreductase subunit IorB (EC 1.2.7.8) | iorB',\n",
    "'272563.120.peg.3426':'D-proline reductase, 45 kDa subunit (EC 1.21.4.1) / D-proline reductase, 23 kDa subunit (EC 1.21.4.1) | prdA',\n",
    "'272563.120.peg.2485':'Glycine reductase component B beta subunit (EC 1.21.4.2) / Glycine reductase component B alpha subunit (EC 1.21.4.2) | grdE',\n",
    "'272563.120.peg.414':'D-lactate dehydrogenase (EC 1.1.1.28) | ldhA',\n",
    "'272563.120.peg.417':'2-hydroxyglutaryl-CoA dehydratase, D-component HgdA | hadB',\n",
    "'272563.120.peg.418':'2-hydroxyglutaryl-CoA dehydratase, D-component HgdB | hadC',\n",
    "'272563.120.peg.415':'Hydroxyproline dehydratase putative | hadA',\n",
    "'272563.120.peg.416':'CoA-substrate-specific enzyme activase; 2-hydroxyglutaryl-CoA dehydratase activator, A-component | hadI',\n",
    "'272563.120.peg.3419':'Proline racemase (EC 5.1.1.4) | prdF',\n",
    "'272563.120.peg.1609':'NADH-dependent reduced ferredoxin:NADP+ oxidoreductase subunit A',\n",
    "'272563.120.peg.168':'4-hydroxyphenylacetate decarboxylase activating enzyme | hpdA',\n",
    "'272563.120.peg.2318':'Acetyl-CoA synthetase (ADP-forming) alpha and beta chains, putative',\n",
    "'272563.120.peg.2473':'Succinate-semialdehyde dehydrogenase, CoA-dependent | sucD',\n",
    "'272563.120.peg.2472':'4-hydroxybutanoyl-CoA dehydratase (EC 4.2.1.120) / Vinylacetyl-CoA Delta-isomerase (EC 5.3.3.3) | abfD',\n",
    "'272563.120.peg.1094':'Electron bifurcating butyryl-CoA dehydrogenase (NAD+, ferredoxin) | bcd2',\n",
    "'272563.120.peg.2470':'4-hydroxybutyrate coenzyme A transferase | abfT',\n",
    "'272563.120.peg.2798':'UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase (EC 6.3.2.10) | murF',\n",
    "'272563.8.peg.262':'hypothetical protein',\n",
    "'272563.120.peg.2462':'Mannitol-1-phosphate 5-dehydrogenase (EC 1.1.1.17) | mtlD',\n",
    "'272563.120.peg.790':'Sorbitol-6-phosphate 2-dehydrogenase (EC 1.1.1.140) | gutD'}\n",
    "\n",
    "for gene in gene_name_fixes.keys():\n",
    "    iCdG791.genes.get_by_id(gene).name = gene_name_fixes[gene]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to files\n",
    "cobra.io.write_sbml_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG791.sbml')\n",
    "cobra.io.save_json_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG791.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Deal with genes that have now reactions\n",
    "import cobra\n",
    "iCdG791 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG791.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct names\n",
    "\n",
    "# Reactions\n",
    "iCdG791.reactions.rxn01987_c.name = 'Ribokinase'\n",
    "iCdG791.reactions.rxn05496_c.name = 'Amino acid transporter (ala-L)'\n",
    "iCdG791.reactions.rxn05202_c.name = 'Xanthine ion-coupled transport;'\n",
    "iCdG791.reactions.rxn12191_c.name = 'Choline monooxygenase (NADP)'\n",
    "iCdG791.reactions.rxn05663_c.name = 'L-tryptophan reversible transport via proton symport'\n",
    "iCdG791.reactions.rxn01213_c.name = 'Geranyl-diphosphate synthase'\n",
    "iCdG791.reactions.rxn10770_c.name = 'Choline dehydrogenase'\n",
    "iCdG791.reactions.rxn05616_c.name = 'Magnesium (+2) transport in via diffusion'\n",
    "iCdG791.reactions.rxn05669_c.name = 'Amino acid transporter (val-L)'\n",
    "iCdG791.reactions.rxn05244_c.name = 'Branched chain amino acid:H+ symporter (Isoleucine)'\n",
    "iCdG791.reactions.EX_cpd00276_e.name = 'D-Glucosamine exchange'\n",
    "iCdG791.reactions.rxn08783_c.name = 'D-lactate dehydrogenase'\n",
    "iCdG791.reactions.rxn03463_c.name = 'alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn03185_c.name = 'alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn02687_c.name = '(R)-mandelate dehydrogenase'\n",
    "iCdG791.reactions.rxn04779_c.name = ' N-cyclopropylmelamine deaminase'\n",
    "iCdG791.reactions.rxn07875_c.name = 'geranal dehydrogenase'\n",
    "iCdG791.reactions.rxn05127_c.name = 'gamma-glutamyl-gamma aminobutyric acid dehydrogenase'\n",
    "iCdG791.reactions.rxn11610_c.name = '2-hydroxymuconic semialdehyde hydrolase'\n",
    "iCdG791.reactions.rxn04781_c.name = 'N-cyclopropylammelide alkylamino hydrolase'\n",
    "iCdG791.reactions.rxn03891_c.name = 'trans-hexaprenyltranstransferase'\n",
    "iCdG791.reactions.rxn03762_c.name = 'CEA synthetase'\n",
    "iCdG791.reactions.rxn03536_c.name = 'Adenosyl cobinamide kinase'\n",
    "iCdG791.reactions.rxn03282_c.name = 'alcohol dehydrogenase'\n",
    "iCdG791.reactions.rxn04723_c.name = '3-methylsalicylaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn07261_c.name = 'ethanolaminephosphotransferase'\n",
    "iCdG791.reactions.rxn03540_c.name = 'Ado-cobyric acid synthase [glutamine hydrolyzing]'\n",
    "iCdG791.reactions.rxn04712_c.name = 'naphthyl-2-methylsuccinate synthase'\n",
    "iCdG791.reactions.rxn07622_c.name = 'dihydrophloroglucinol hydrolase'\n",
    "iCdG791.reactions.rxn06905_c.name = 'TCOA hydrolase'\n",
    "iCdG791.reactions.rxn04737_c.name = '2-naphthaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn03921_c.name = 'aldehyde dehydrogenase [EC:1.2.1.-]'\n",
    "iCdG791.reactions.rxn03721_c.name = 'Parathion hydrolase'\n",
    "iCdG791.reactions.rxn04727_c.name = '1-naphthaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn03537_c.name = 'Adenosyl cobinamide phosphate guanyltransferase'\n",
    "iCdG791.reactions.rxn08352_c.name = '(E)-4-hydroxy-3-methylbut-2-en-1-yl diphosphate reductase'\n",
    "iCdG791.reactions.rxn03135_c.name = 'Imidazole glycerol phosphate synthase amidotransferase'\n",
    "iCdG791.reactions.rxn03837_c.name = 'aryldialkylphosphatase'\n",
    "iCdG791.reactions.rxn06752_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn04413_c.name = 'adenosylcobinamide kinase'\n",
    "iCdG791.reactions.rxn04604_c.name = '(2E,4Z)-2-hydroxy-6-oxona-2,4-dienedioate succinylhydrolase'\n",
    "iCdG791.reactions.rxn03005_c.name = 'N2-Formyl-N1-(5-phospho-D-ribosyl)glycinamide formyltransferase'\n",
    "iCdG791.reactions.rxn07876_c.name = 'citronellal dehydrogenase'\n",
    "iCdG791.reactions.rxn06394_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn04780_c.name = 'N-cyclopropylammeline deaminase'\n",
    "iCdG791.reactions.rxn03486_c.name = '2-hydroxymuconic semi-aldehyde (m-xylene)'\n",
    "iCdG791.reactions.rxn07609_c.name = '2-hydroxy-6-oxo-6-(2-carboxyphenyl)-hexa-2,4-dienoate hydrolase'\n",
    "iCdG791.reactions.rxn03538_c.name = 'Adenosylcobalamin 5-phosphate synthase'\n",
    "iCdG791.reactions.rxn04238_c.name = '4-isopropenyl-2-ketocyclohexane-1-carboxyl-CoA hydrolase'\n",
    "iCdG791.reactions.rxn01208_c.name = '2-Oxo-4-methyl-3-carboxypentanoate decarboxylation'\n",
    "iCdG791.reactions.rxn03467_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn06750_c.name = 'alpha-N-acetylneuraminate alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn03892_c.name = 'all-trans-hexaprenyl-diphosphate synthase'\n",
    "iCdG791.reactions.rxn07493_c.name = 'GDP-L-galactose phosphorylase'\n",
    "iCdG791.reactions.rxn02362_c.name = 'cyclic 2,3-diphosphoglycerate synthase'\n",
    "iCdG791.reactions.rxn01905_c.name = 'cysteine sulfinic acid deaminase'\n",
    "iCdG791.reactions.rxn02604_c.name = 'vicianin beta-glucosidase'\n",
    "iCdG791.reactions.rxn05126_c.name = '4-(gamma-L-glutamylamino)butanal:NAD+ oxidoreductase'\n",
    "iCdG791.reactions.rxn04658_c.name = 'LysW-gamma-L-alpha-aminoadipate 6-semialdehyde:NADP+ 6-oxidoreductase (phosphorylating)'\n",
    "iCdG791.reactions.rxn04744_c.name = '4-hydroxymethylsalicyaldehyde dehydrogenase'\n",
    "iCdG791.reactions.rxn03873_c.name = '6-oxocyclohex-1-ene-1-carbonyl-CoA hydrolase (decyclizing)'\n",
    "iCdG791.reactions.rxn07885_c.name = 'citronellal dehydrogenase'\n",
    "iCdG791.reactions.rxn06760_c.name = 'neuraminidase'\n",
    "iCdG791.reactions.rxn06759_c.name = 'alpha-N-acetylneuraminate alpha-2,8-sialyltransferase'\n",
    "iCdG791.reactions.rxn11584_c.name = 'aryldialkylphosphatase'\n",
    "iCdG791.reactions.rxn00970_c.name = 'Hydroxybenzoate synthase'\n",
    "iCdG791.reactions.rxn05039_c.name = '5-amino-6-(5-phosphoribitylamino)uracil phosphatase'\n",
    "iCdG791.reactions.rxn03365_c.name = 'R04916'\n",
    "iCdG791.reactions.rxn03366_c.name = 'R04916'\n",
    "iCdG791.reactions.rxn00568_c.name = 'Nitrite reductase (NAD(P)H)'\n",
    "\n",
    "# Metabolites\n",
    "iCdG791.metabolites.cpd00443_c.name = '4-Aminobenzoate'\n",
    "iCdG791.metabolites.cpd00443_e.name = '4-Aminobenzoate'\n",
    "iCdG791.metabolites.cpd00103_c.name = '5-Phospho-alpha-D-ribose 1-diphosphate'\n",
    "iCdG791.metabolites.cpd00226_c.name = 'Hypoxanthine'\n",
    "iCdG791.metabolites.cpd00104_c.name = 'Biotin'\n",
    "iCdG791.metabolites.cpd00104_e.name = 'Biotin'\n",
    "iCdG791.metabolites.cpd00202_c.name = '2-Isopentenyl diphosphate'\n",
    "iCdG791.metabolites.cpd11421_c.name = 'Reduced thioredoxin'\n",
    "iCdG791.metabolites.cpd11420_c.name = 'Oxidized thioredoxin'\n",
    "iCdG791.metabolites.cpd00309_c.name = 'Xanthine'\n",
    "iCdG791.metabolites.cpd00309_e.name = 'Xanthine'\n",
    "iCdG791.metabolites.cpd00276_e.name = 'D-Glucosamine'\n",
    "iCdG791.metabolites.cpd01080_c.name = 'Octadecanoic acid'\n",
    "iCdG791.metabolites.cpd00050_c.name = 'Flavin mononucleotide'\n",
    "iCdG791.metabolites.cpd00508_c.name = '(3S)-3-Methyl-2-oxopentanoate'\n",
    "iCdG791.metabolites.cpd00644_c.name = '(R)-Pantothenate'\n",
    "iCdG791.metabolites.cpd00226_e.name = 'Hypoxanthine'\n",
    "iCdG791.metabolites.cpd01080_e.name = 'Octadecanoic acid'\n",
    "iCdG791.metabolites.cpd02996_c.name = 'N-Acetyl-D-galactosaminyl-(N-acetylneuraminyl)-D-galactosyl-1,4- beta-D-glucosyl-N-acylsphingosine'\n",
    "iCdG791.metabolites.cpd00361_c.name = '(S)-Acetoin'\n",
    "iCdG791.metabolites.cpd00497_c.name = '(9-D-Ribosylxanthine)-5-phosphate'\n",
    "iCdG791.metabolites.cpd11434_c.name = 'Anteiso-C17:0 CoA Anteisoheptadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd02921_c.name = 'S-2-5-Amino-1-(5-phospho-D-ribosyl)imidazole-4-carboxamido-succinate'\n",
    "iCdG791.metabolites.cpd00042_c.name = 'Glutathione'\n",
    "iCdG791.metabolites.cpd11616_c.name = 'Aldehyde'\n",
    "iCdG791.metabolites.cpd02678_c.name = '5-P-ribosyl-N-formylglycineamide'\n",
    "iCdG791.metabolites.cpd02394_c.name = '5-Phosphoribosylglycinamide'\n",
    "iCdG791.metabolites.cpd02591_c.name = '(2E,6E,10E,14E)-geranylfarnesyl diphosphate'\n",
    "iCdG791.metabolites.cpd00200_c.name = '2-Oxoisocaproate'\n",
    "iCdG791.metabolites.cpd00599_c.name = '2-hydroxybenzoate'\n",
    "iCdG791.metabolites.cpd00853_c.name = '2,3-Bisphospho-D-glycerate'\n",
    "iCdG791.metabolites.cpd11432_c.name = 'Iso-C17:0 CoA Isoheptadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd11441_c.name = 'Iso-C16:0 CoA Isohexadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd11435_c.name = 'Iso-C14:0 CoA Isotetradecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd11439_c.name = 'Anteiso-C15:0 CoA Anteisopentadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd02972_c.name = 'alpha-N-Acetylneuraminyl-2,8-alpha-N-acetylneuraminyl-2,3-beta-D-galactosyl-R'\n",
    "iCdG791.metabolites.cpd00794_e.name = 'D-Trehalose'\n",
    "iCdG791.metabolites.cpd00794_c.name = 'D-Trehalose'\n",
    "iCdG791.metabolites.cpd00668_c.name = '2-Acetolactate'\n",
    "iCdG791.metabolites.cpd02631_c.name = '(3R,4S,5R)-3,4,5-Trihydroxy-1,2-cyclohexanedione'\n",
    "iCdG791.metabolites.cpd02968_c.name = 'UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-6-carboxy-L-lysyl-D-alanyl-D-alanine'\n",
    "iCdG791.metabolites.cpd00208_e.name = 'Lactose'\n",
    "iCdG791.metabolites.cpd00208_c.name = 'Lactose'\n",
    "iCdG791.metabolites.cpd00158_e.name = 'Cellobiose'\n",
    "iCdG791.metabolites.cpd00644_e.name = 'Pantothenate'\n",
    "iCdG791.metabolites.cpd02964_c.name = 'UDP-N-acetylmuramoyl-L-alanyl-D-gamma-glutamyl-meso-2-6-diaminopimelate'\n",
    "iCdG791.metabolites.cpd01741_c.name = 'Dodecanoate'\n",
    "iCdG791.metabolites.cpd00354_c.name = 'Nucleoside 5-diphosphate'\n",
    "iCdG791.metabolites.cpd02920_c.name = '2-Amino-4-hydroxy-6-hydroxymethyl-7-8-dihydropteridinediphosphate'\n",
    "iCdG791.metabolites.cpd00327_c.name = 'Stearoyl-CoA'\n",
    "iCdG791.metabolites.cpd11437_c.name = 'Iso-C15:0 CoA Isopentadecanoyl-CoA'\n",
    "iCdG791.metabolites.cpd00646_c.name = '4-Hydroxynitrobenzene'\n",
    "iCdG791.metabolites.cpd04011_c.name = 'Diethylthiophosphoric acid'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 278\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89.76884522354938"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove those genes that only act on macromolecules\n",
    "from cobra.manipulation.delete import remove_genes\n",
    "genes_to_be_removed = ['272563.8.peg.14','272563.8.peg.24','272563.8.peg.3559','272563.8.peg.9','272563.8.peg.13','272563.8.peg.2710','272563.8.peg.2183','272563.8.peg.1314','272563.8.peg.2596','272563.8.peg.3838','272563.8.peg.3388','272563.8.peg.3083','272563.8.peg.10','272563.8.peg.348','272563.8.peg.1360','272563.8.peg.1361','272563.8.peg.3664','272563.8.peg.1339','272563.8.peg.3717','272563.8.peg.77','272563.8.peg.1373','272563.8.peg.113','272563.8.peg.3857','272563.8.peg.1312','272563.8.peg.1527','272563.8.peg.1570','272563.8.peg.78','272563.8.peg.3858','272563.8.peg.89','272563.8.peg.94','272563.8.peg.83','272563.8.peg.1235','272563.8.peg.2253','272563.8.peg.69','272563.8.peg.95','272563.8.peg.1383','272563.8.peg.80','272563.8.peg.103','272563.8.peg.1221','272563.8.peg.112','272563.8.peg.3837','272563.8.peg.120','272563.8.peg.119','272563.8.peg.96','272563.8.peg.93','272563.8.peg.1322','272563.8.peg.721','272563.8.peg.97','272563.8.peg.2685','272563.8.peg.88','272563.8.peg.91','272563.8.peg.74','272563.8.peg.92','272563.8.peg.111','272563.8.peg.2567','272563.8.peg.73','272563.8.peg.86','272563.8.peg.101','272563.8.peg.100','272563.8.peg.98','272563.8.peg.1318','272563.8.peg.2595','272563.8.peg.99','272563.8.peg.85','272563.8.peg.114','272563.8.peg.109','272563.8.peg.3833','272563.8.peg.110','272563.8.peg.87','272563.8.peg.1219','272563.8.peg.90','272563.8.peg.722','272563.8.peg.79','272563.8.peg.84','272563.8.peg.3839','272563.8.peg.102','272563.8.peg.3653']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Lead, cadmium, zinc and mercury transporters\n",
    "genes_to_be_removed = ['272563.8.peg.620','272563.8.peg.1714','272563.8.peg.2961']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Pyruvate formate-lyase activating enzyme\n",
    "genes_to_be_removed = ['272563.8.peg.795','272563.8.peg.3436','272563.8.peg.1176']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# L(+)-tartrate dehydratase beta subunit (EC 4.2.1.32)\n",
    "genes_to_be_removed = ['272563.8.peg.1055','272563.8.peg.1054']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Sulfur carrier protein ThiS\n",
    "genes_to_be_removed = ['272563.8.peg.1781','272563.8.peg.1782']\n",
    "remove_genes(iCdG791, genes_to_be_removed)\n",
    "\n",
    "# Others\n",
    "remove_genes(iCdG791, ['272563.8.peg.907'])  # Predicted molybdate-responsive regulator YvgK in bacilli\n",
    "remove_genes(iCdG791, ['272563.8.peg.2617']) # Selenide,water dikinase (EC 2.7.9.3)\n",
    "remove_genes(iCdG791, ['272563.8.peg.2586']) # Hypothetical radical SAM family enzyme in heat shock gene cluster\n",
    "remove_genes(iCdG791, ['272563.8.peg.2223']) # Copper-translocating P-type ATPase (EC 3.6.3.4)\n",
    "remove_genes(iCdG791, ['272563.8.peg.3769']) # Methionine synthase activation domain (EC 2.1.1.13)\n",
    "remove_genes(iCdG791, ['272563.8.peg.472'])  # ornithine racemase (EC 5.1.1.12)\n",
    "remove_genes(iCdG791, ['272563.8.peg.3730']) # tRNA(Ile)-lysidine synthetase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 184\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Protein biosynthesis - correction to integrate genes that create respective tRNAs\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.get_by_id('cpd00002_c') # ATP\n",
    "cpd17043_c = iCdG791.metabolites.get_by_id('cpd17043_c') # RNA\n",
    "cpd00067_c = iCdG791.metabolites.get_by_id('cpd00067_c') # H+\n",
    "cpd00018_c = iCdG791.metabolites.get_by_id('cpd00018_c') # AMP\n",
    "cpd00012_c = iCdG791.metabolites.get_by_id('cpd00012_c') # PPi\n",
    "\n",
    "# L-Alanine\n",
    "cpd00035_c = iCdG791.metabolites.get_by_id('cpd00035_c') \n",
    "cpd11770_c = cobra.Metabolite(\n",
    "    'cpd11770_c',\n",
    "    formula='',\n",
    "    name='L-Alanyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd11770_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd11770_c.annotation['seed.compound'] = 'cpd11770'\n",
    "rxn06300_c = cobra.Reaction('rxn06300_c')\n",
    "rxn06300_c.gene_reaction_rule = '272563.8.peg.1347'\n",
    "rxn06300_c.name = 'L-Alanine:tRNA(Ala) ligase (AMP-forming)'\n",
    "rxn06300_c.bounds = (-1000., 1000.)\n",
    "rxn06300_c.add_metabolites({\n",
    "    cpd00035_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd11770_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06300_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06300_c.annotation['seed.reaction'] = 'rxn06300'\n",
    "\n",
    "# L-Arginine\n",
    "cpd00051_c = iCdG791.metabolites.get_by_id('cpd00051_c') \n",
    "cpd12036_c = cobra.Metabolite(\n",
    "    'cpd12036_c',\n",
    "    formula='',\n",
    "    name='L-Arginyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12036_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12036_c.annotation['seed.compound'] = 'cpd12036'\n",
    "rxn06432_c = cobra.Reaction('rxn06432_c')\n",
    "rxn06432_c.gene_reaction_rule = '272563.8.peg.746'\n",
    "rxn06432_c.name = 'L-Arginine:tRNA(Arg) ligase (AMP-forming)'\n",
    "rxn06432_c.bounds = (-1000., 1000.)\n",
    "rxn06432_c.add_metabolites({\n",
    "    cpd00051_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12036_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06432_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06432_c.annotation['seed.reaction'] = 'rxn06432'\n",
    "\n",
    "# L-Asparagine\n",
    "cpd00132_c = iCdG791.metabolites.get_by_id('cpd00132_c') \n",
    "cpd12313_c = cobra.Metabolite(\n",
    "    'cpd12313_c',\n",
    "    formula='',\n",
    "    name='L-Asparagyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12313_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12313_c.annotation['seed.compound'] = 'cpd12313'\n",
    "rxn06434_c = cobra.Reaction('rxn06434_c')\n",
    "rxn06434_c.gene_reaction_rule = '272563.8.peg.2360'\n",
    "rxn06434_c.name = 'L-Asparagine:tRNA(Asn) ligase (AMP-forming)'\n",
    "rxn06434_c.bounds = (-1000., 1000.)\n",
    "rxn06434_c.add_metabolites({\n",
    "    cpd00132_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12313_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06434_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06434_c.annotation['seed.reaction'] = 'rxn06434'\n",
    "\n",
    "# L-Aspartate\n",
    "cpd00041_c = iCdG791.metabolites.get_by_id('cpd00041_c') \n",
    "cpd12226_c = cobra.Metabolite(\n",
    "    'cpd12226_c',\n",
    "    formula='',\n",
    "    name='L-Aspartyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12226_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12226_c.annotation['seed.compound'] = 'cpd12226'\n",
    "rxn06936_c = cobra.Reaction('rxn06936_c')\n",
    "rxn06936_c.gene_reaction_rule = '272563.8.peg.2866'\n",
    "rxn06936_c.name = 'L-Aspartate:tRNA(Asp) ligase (AMP-forming)'\n",
    "rxn06936_c.bounds = (-1000., 1000.)\n",
    "rxn06936_c.add_metabolites({\n",
    "    cpd00041_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12226_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06936_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06936_c.annotation['seed.reaction'] = 'rxn06936'\n",
    "\n",
    "# L-Cysteine\n",
    "cpd00084_c = iCdG791.metabolites.get_by_id('cpd00084_c') \n",
    "cpd12255_c = cobra.Metabolite(\n",
    "    'cpd12255_c_c',\n",
    "    formula='',\n",
    "    name='Cysteinyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12255_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12255_c.annotation['seed.compound'] = 'cpd12255'\n",
    "rxn06435_c = cobra.Reaction('rxn06435_c')\n",
    "rxn06435_c.gene_reaction_rule = '272563.8.peg.62'\n",
    "rxn06435_c.name = 'L-Cysteine:tRNA(Cys) ligase (AMP-forming)'\n",
    "rxn06435_c.bounds = (-1000., 1000.)\n",
    "rxn06435_c.add_metabolites({\n",
    "    cpd00084_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12255_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06435_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06435_c.annotation['seed.reaction'] = 'rxn06435'\n",
    "\n",
    "# L-Glutamine - no gene\n",
    "cpd00053_c = iCdG791.metabolites.get_by_id('cpd00053_c') \n",
    "cpd12060_c = cobra.Metabolite(\n",
    "    'cpd12060_c',\n",
    "    formula='',\n",
    "    name='L-Glutaminyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12060_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12060_c.annotation['seed.compound'] = 'cpd12060'\n",
    "rxn06437_c = cobra.Reaction('rxn06437_c')\n",
    "rxn06437_c.gene_reaction_rule = '272563.120.peg.2173'\n",
    "rxn06437_c.name = 'L-Glutamine:tRNA(Gln) ligase (AMP-forming)'\n",
    "rxn06437_c.bounds = (-1000., 1000.)\n",
    "rxn06437_c.add_metabolites({\n",
    "    cpd00053_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12060_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06437_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06437_c.annotation['seed.reaction'] = 'rxn06437'\n",
    "\n",
    "# L-Glutamate - no gene\n",
    "cpd00023_c = iCdG791.metabolites.get_by_id('cpd00023_c') \n",
    "cpd12828_c = cobra.Metabolite(\n",
    "    'cpd12828_c',\n",
    "    formula='',\n",
    "    name='L-Glutamyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12828_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12828_c.annotation['seed.compound'] = 'cpd12828'\n",
    "rxn06436_c = cobra.Reaction('rxn06436_c')\n",
    "rxn06436_c.gene_reaction_rule = '272563.120.peg.55'\n",
    "rxn06436_c.name = 'L-Glutamate:tRNA(Gln) ligase (AMP-forming)'\n",
    "rxn06436_c.bounds = (-1000., 1000.)\n",
    "rxn06436_c.add_metabolites({\n",
    "    cpd00023_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12828_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06436_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06436_c.annotation['seed.reaction'] = 'rxn06436'\n",
    "\n",
    "# Glycine\n",
    "cpd00033_c = iCdG791.metabolites.get_by_id('cpd00033_c') \n",
    "cpd12100_c = cobra.Metabolite(\n",
    "    'cpd12100_c',\n",
    "    formula='',\n",
    "    name='Glycyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12100_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12100_c.annotation['seed.compound'] = 'cpd12100'\n",
    "rxn06438_c = cobra.Reaction('rxn06438_c')\n",
    "rxn06438_c.gene_reaction_rule = '272563.8.peg.2551 and 272563.8.peg.2552'\n",
    "rxn06438_c.name = 'glycine:tRNA(Gly) ligase (AMP-forming)'\n",
    "rxn06438_c.bounds = (-1000., 1000.)\n",
    "rxn06438_c.add_metabolites({\n",
    "    cpd00033_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12100_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06438_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06438_c.annotation['seed.reaction'] = 'rxn06438'\n",
    "\n",
    "# L-Histidine\n",
    "cpd00119_c = iCdG791.metabolites.get_by_id('cpd00119_c') \n",
    "cpd12228_c = cobra.Metabolite(\n",
    "    'cpd12228_c',\n",
    "    formula='',\n",
    "    name='L-Histidyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12228_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12228_c.annotation['seed.compound'] = 'cpd12228'\n",
    "rxn06439_c = cobra.Reaction('rxn06439_c')\n",
    "rxn06439_c.gene_reaction_rule = '272563.8.peg.2867'\n",
    "rxn06439_c.name = 'L-Histidine:tRNA(His) ligase (AMP-forming)'\n",
    "rxn06439_c.bounds = (-1000., 1000.)\n",
    "rxn06439_c.add_metabolites({\n",
    "    cpd00119_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12228_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06439_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06439_c.annotation['seed.reaction'] = 'rxn06439'\n",
    "\n",
    "# L-Isoleucine\n",
    "cpd00322_c = iCdG791.metabolites.get_by_id('cpd00322_c') \n",
    "cpd12256_c = cobra.Metabolite(\n",
    "    'cpd12256_c',\n",
    "    formula='',\n",
    "    name='L-Isoleucyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12256_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12256_c.annotation['seed.compound'] = 'cpd12256'\n",
    "rxn06440_c = cobra.Reaction('rxn06440_c')\n",
    "rxn06440_c.gene_reaction_rule = '272563.8.peg.2741'\n",
    "rxn06440_c.name = 'L-Isoleucine:tRNA(Ile) ligase (AMP-forming)'\n",
    "rxn06440_c.bounds = (-1000., 1000.)\n",
    "rxn06440_c.add_metabolites({\n",
    "    cpd00322_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12256_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06440_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06440_c.annotation['seed.reaction'] = 'rxn06440'\n",
    "\n",
    "# L-Leucine\n",
    "cpd00107_c = iCdG791.metabolites.get_by_id('cpd00107_c') \n",
    "cpd12003_c = cobra.Metabolite(\n",
    "    'cpd12003_c',\n",
    "    formula='',\n",
    "    name='Leucyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12003_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12003_c.annotation['seed.compound'] = 'cpd12003'\n",
    "rxn06441_c = cobra.Reaction('rxn06441_c')\n",
    "rxn06441_c.gene_reaction_rule = '272563.8.peg.2644'\n",
    "rxn06441_c.name = 'L-Leucine:tRNA(Leu) ligase (AMP-forming)'\n",
    "rxn06441_c.bounds = (-1000., 1000.)\n",
    "rxn06441_c.add_metabolites({\n",
    "    cpd00107_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12003_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06441_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06441_c.annotation['seed.reaction'] = 'rxn06441'\n",
    "\n",
    "# L-Lysine\n",
    "cpd00039_c = iCdG791.metabolites.get_by_id('cpd00039_c') \n",
    "cpd01326_c = cobra.Metabolite(\n",
    "    'cpd01326_c',\n",
    "    formula='',\n",
    "    name='L-Lysyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd01326_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd01326_c.annotation['seed.compound'] = 'cpd01326'\n",
    "rxn06442_c = cobra.Reaction('rxn06442_c')\n",
    "rxn06442_c.gene_reaction_rule = '272563.8.peg.3722'\n",
    "rxn06442_c.name = 'L-Lysine:tRNALys ligase (AMP-forming)'\n",
    "rxn06442_c.bounds = (-1000., 1000.)\n",
    "rxn06442_c.add_metabolites({\n",
    "    cpd00039_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd01326_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06442_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06442_c.annotation['seed.reaction'] = 'rxn06442'\n",
    "\n",
    "# L-Methionine\n",
    "cpd00060_c = iCdG791.metabolites.get_by_id('cpd00060_c') \n",
    "cpd12105_c = cobra.Metabolite(\n",
    "    'cpd12105_c',\n",
    "    formula='',\n",
    "    name='Methionyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12105_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12105_c.annotation['seed.compound'] = 'cpd12105'\n",
    "rxn06443_c = cobra.Reaction('rxn06443_c')\n",
    "rxn06443_c.gene_reaction_rule = '272563.8.peg.3708'\n",
    "rxn06443_c.name = 'L-Methionine:tRNAMet ligase (AMP-forming)'\n",
    "rxn06443_c.bounds = (-1000., 1000.)\n",
    "rxn06443_c.add_metabolites({\n",
    "    cpd00060_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12105_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06443_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06443_c.annotation['seed.reaction'] = 'rxn06443'\n",
    "\n",
    "# L-Phenylalanine\n",
    "cpd00066_c = iCdG791.metabolites.get_by_id('cpd00066_c') \n",
    "cpd12335_c = cobra.Metabolite(\n",
    "    'cpd12335_c',\n",
    "    formula='',\n",
    "    name='Phenylalanyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12335_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12335_c.annotation['seed.compound'] = 'cpd12335'\n",
    "rxn06444_c = cobra.Reaction('rxn06444_c')\n",
    "rxn06444_c.gene_reaction_rule = '272563.8.peg.734 and 272563.8.peg.735'\n",
    "rxn06444_c.name = 'L-Phenylalanine:tRNA(Ala) ligase (AMP-forming)'\n",
    "rxn06444_c.bounds = (-1000., 1000.)\n",
    "rxn06444_c.add_metabolites({\n",
    "    cpd00066_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12335_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06444_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06444_c.annotation['seed.reaction'] = 'rxn06444'\n",
    "\n",
    "# L-Proline\n",
    "cpd00129_c = iCdG791.metabolites.get_by_id('cpd00129_c') \n",
    "cpd12164_c = cobra.Metabolite(\n",
    "    'cpd12164_c',\n",
    "    formula='',\n",
    "    name='L-Prolyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12164_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12164_c.annotation['seed.compound'] = 'cpd12164'\n",
    "rxn06445_c = cobra.Reaction('rxn06445_c')\n",
    "rxn06445_c.gene_reaction_rule = '272563.120.peg.53'\n",
    "rxn06445_c.name = 'L-proline:tRNA(Pro) ligase (AMP-forming)'\n",
    "rxn06445_c.bounds = (-1000., 1000.)\n",
    "rxn06445_c.add_metabolites({\n",
    "    cpd00129_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12164_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06445_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06445_c.annotation['seed.reaction'] = 'rxn06445'\n",
    "\n",
    "# L-Serine\n",
    "cpd00054_c = iCdG791.metabolites.get_by_id('cpd00054_c') \n",
    "cpd12132_c = cobra.Metabolite(\n",
    "    'cpd12132_c',\n",
    "    formula='',\n",
    "    name='L-Seryl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12132_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12132_c.annotation['seed.compound'] = 'cpd12132'\n",
    "rxn06446_c = cobra.Reaction('rxn06446_c')\n",
    "rxn06446_c.gene_reaction_rule = '272563.8.peg.22 or 272563.8.peg.2968'\n",
    "rxn06446_c.name = 'L-Serine:tRNA(Ser) ligase (AMP-forming)'\n",
    "rxn06446_c.bounds = (-1000., 1000.)\n",
    "rxn06446_c.add_metabolites({\n",
    "    cpd00054_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12132_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06446_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06446_c.annotation['seed.reaction'] = 'rxn06446'\n",
    "\n",
    "# L-Threonine\n",
    "cpd00161_c = iCdG791.metabolites.get_by_id('cpd00161_c') \n",
    "cpd12229_c = cobra.Metabolite(\n",
    "    'cpd12229_c',\n",
    "    formula='',\n",
    "    name='Threonyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12229_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12229_c.annotation['seed.compound'] = 'cpd12229'\n",
    "rxn06447_c = cobra.Reaction('rxn06447_c')\n",
    "rxn06447_c.gene_reaction_rule = '272563.8.peg.601'\n",
    "rxn06447_c.name = 'L-Threonine:tRNA(Thr) ligase (AMP-forming)'\n",
    "rxn06447_c.bounds = (-1000., 1000.)\n",
    "rxn06447_c.add_metabolites({\n",
    "    cpd00161_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12229_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06447_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06447_c.annotation['seed.reaction'] = 'rxn06447'\n",
    "\n",
    "# L-Tryptophan\n",
    "cpd00065_c = iCdG791.metabolites.get_by_id('cpd00065_c') \n",
    "cpd12336_c = cobra.Metabolite(\n",
    "    'cpd12336_c',\n",
    "    formula='',\n",
    "    name='L-Tryptophanyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12336_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12336_c.annotation['seed.compound'] = 'cpd12336'\n",
    "rxn06448_c = cobra.Reaction('rxn06448_c')\n",
    "rxn06448_c.gene_reaction_rule = '272563.8.peg.2733'\n",
    "rxn06448_c.name = 'L-Tryptophan -tRNA(Trp) ligase (AMP-forming)'\n",
    "rxn06448_c.bounds = (-1000., 1000.)\n",
    "rxn06448_c.add_metabolites({\n",
    "    cpd00065_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12336_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06448_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06448_c.annotation['seed.reaction'] = 'rxn06448'\n",
    "\n",
    "# L-Tyrosine\n",
    "cpd00069_c = iCdG791.metabolites.get_by_id('cpd00069_c') \n",
    "cpd12194_c = cobra.Metabolite(\n",
    "    'cpd12194_c',\n",
    "    formula='',\n",
    "    name='L-Tyrosyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12194_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12194_c.annotation['seed.compound'] = 'cpd12194'\n",
    "rxn06280_c = cobra.Reaction('rxn06280_c')\n",
    "rxn06280_c.gene_reaction_rule = '272563.8.peg.1597'\n",
    "rxn06280_c.name = 'L-Tyrosine:tRNA(Tyr) ligase (AMP-forming)'\n",
    "rxn06280_c.bounds = (-1000., 1000.)\n",
    "rxn06280_c.add_metabolites({\n",
    "    cpd00069_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # tRNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12194_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06280_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06280_c.annotation['seed.reaction'] = 'rxn06280'\n",
    "\n",
    "# L-Valine\n",
    "cpd00156_c = iCdG791.metabolites.get_by_id('cpd00156_c') \n",
    "cpd12133_c = cobra.Metabolite(\n",
    "    'cpd12133_c',\n",
    "    formula='',\n",
    "    name='L-Valyl-tRNA',\n",
    "    compartment='cytosol')\n",
    "cpd12133_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd12133_c.annotation['seed.compound'] = 'cpd12133'\n",
    "rxn06449_c = cobra.Reaction('rxn06449_c')\n",
    "rxn06449_c.gene_reaction_rule = '272563.8.peg.3409'\n",
    "rxn06449_c.name = 'L-Valine:tRNAVal ligase (AMP-forming)'\n",
    "rxn06449_c.bounds = (-1000., 1000.)\n",
    "rxn06449_c.add_metabolites({\n",
    "    cpd00156_c: -1.0, # amino acid\n",
    "    cpd17043_c: -0.05, # RNA\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd12133_c: 1.0, # AA-tRNA\n",
    "    cpd00018_c: 1.0, # AMP\n",
    "    cpd00012_c: 1.0, # PPi\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn06449_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn06449_c.annotation['seed.reaction'] = 'rxn06449'\n",
    "\n",
    "iCdG791.add_reactions([rxn06300_c,rxn06432_c,rxn06434_c,rxn06936_c,rxn06435_c,rxn06437_c,rxn06436_c,\n",
    "                       rxn06438_c,rxn06439_c,rxn06440_c,rxn06441_c,rxn06442_c,rxn06443_c,rxn06444_c,\n",
    "                       rxn06445_c,rxn06446_c,rxn06447_c,rxn06448_c,rxn06280_c,rxn06449_c])\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.2173').name = 'Glutaminyl-tRNA synthetase (EC 6.1.1.18)'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.2173').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.2173').annotation['patric'] = 'fig|272563.120.peg.2173'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.55').name = 'Glutamyl-tRNA synthetase (EC 6.1.1.17)'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.55').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.55').annotation['patric'] = 'fig|272563.120.peg.55'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.53').name = 'Prolyl-tRNA synthetase (EC 6.1.1.15)'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.53').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.53').annotation['patric'] = 'fig|272563.120.peg.53'\n",
    "\n",
    "#---------------------------------------------------------------------------#\n",
    "\n",
    "iCdG791.reactions.protein_rxn.remove_from_model(remove_orphans=False)\n",
    "cpd00001_c = iCdG791.metabolites.get_by_id('cpd00001_c') # H2O\n",
    "cpd00008_c = iCdG791.metabolites.get_by_id('cpd00008_c') # ADP\n",
    "cpd00009_c = iCdG791.metabolites.get_by_id('cpd00009_c') # Phosphate\n",
    "cpd17041_c = iCdG791.metabolites.get_by_id('cpd17041_c') # Protein\n",
    "\n",
    "# Create new reaction, now with tRNAs\n",
    "protein_rxn = cobra.Reaction('protein_rxn')\n",
    "protein_rxn.name = 'Protein biosynthesis'\n",
    "protein_rxn.lower_bound = 0.\n",
    "protein_rxn.upper_bound = 1000.\n",
    "protein_rxn.add_metabolites({\n",
    "    cpd11770_c: -0.264, # Alanine\n",
    "    cpd12036_c: -0.160, # Arginine\n",
    "    cpd12313_c: -0.322, # Asparagine\n",
    "    cpd12226_c: -0.286, # Aspartate\n",
    "    cpd12255_c: -0.062, # Cysteine\n",
    "    cpd12060_c: -0.120, # Glutamine\n",
    "    cpd12828_c: -0.367, # Glutamate\n",
    "    cpd12100_c: -0.314, # Glycine\n",
    "    cpd12228_c: -0.068, # Histidine\n",
    "    cpd12256_c: -0.498, # Isoleucine\n",
    "    cpd12003_c: -0.455, # Leucine\n",
    "    cpd01326_c: -0.457, # Lysine\n",
    "    cpd12105_c: -0.135, # Methionine\n",
    "    cpd12335_c: -0.219, # Phenylalanine\n",
    "    cpd12164_c: -0.137, # Proline\n",
    "    cpd12132_c: -0.331, # Serine\n",
    "    cpd12229_c: -0.248, # Threonine\n",
    "    cpd12336_c: -0.032, # Tryptophan\n",
    "    cpd12194_c: -0.208, # Tyrosine\n",
    "    cpd12133_c: -0.328, # Valine\n",
    "    cpd00002_c: -20.0,\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd17041_c: 1.0,\n",
    "    cpd00008_c: 20.0,\n",
    "    cpd00009_c: 20.0\n",
    "})\n",
    "\n",
    "iCdG791.add_reactions([protein_rxn])\n",
    "iCdG791.reactions.protein_rxn.annotation['sbo'] = 'SBO:0000629'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 164\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "82.79962839526671"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd11581_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>gly-asn-L exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f3496646dd0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd11581_e <=> </p>\n",
       "                    <p style='text-align:right'>gly-asn-L <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd11581_e at 0x7f3496646dd0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Dipeptides\n",
    "# Methionyl aminopeptidase and associated active transport + exchanges\n",
    "\n",
    "# gly-glu-L \n",
    "rxn12634_c = universal_GramPos.reactions.rxn12634_c\n",
    "rxn12634_c.lower_bound = 0.\n",
    "rxn12634_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12600_c.remove_from_model(remove_orphans=False)\n",
    "rxn05534_c = universal_GramPos.reactions.rxn05534_c\n",
    "rxn05534_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12634_c,rxn05534_c])\n",
    "cpd11592_e = iCdG791.metabolites.cpd11592_e\n",
    "iCdG791.add_boundary(cpd11592_e, type='exchange', reaction_id='EX_cpd11592_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# met-L-ala-L\n",
    "rxn12636_c = universal_GramPos.reactions.rxn12636_c\n",
    "rxn12636_c.lower_bound = 0.\n",
    "rxn12636_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12544_c.remove_from_model(remove_orphans=False)\n",
    "rxn05536_c = universal_GramPos.reactions.rxn05536_c\n",
    "rxn05536_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12636_c,rxn05536_c])\n",
    "cpd11590_e = iCdG791.metabolites.cpd11590_e\n",
    "iCdG791.add_boundary(cpd11590_e, type='exchange', reaction_id='EX_cpd11590_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# gly-pro-L \n",
    "rxn12638_c = universal_GramPos.reactions.rxn12638_c\n",
    "rxn12638_c.lower_bound = 0.\n",
    "rxn12638_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12576_c.remove_from_model(remove_orphans=False)\n",
    "rxn05538_c = universal_GramPos.reactions.rxn05538_c\n",
    "rxn05538_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12638_c,rxn05538_c])\n",
    "cpd11588_e = iCdG791.metabolites.cpd11588_e\n",
    "iCdG791.add_boundary(cpd11588_e, type='exchange', reaction_id='EX_cpd11588_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# ala-L-glu-L\n",
    "rxn12640_c = universal_GramPos.reactions.rxn12640_c\n",
    "rxn12640_c.lower_bound = 0.\n",
    "rxn12640_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12523_c.remove_from_model(remove_orphans=False)\n",
    "rxn05541_c = universal_GramPos.reactions.rxn05541_c\n",
    "rxn05541_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12640_c,rxn05541_c])\n",
    "cpd11586_e = iCdG791.metabolites.cpd11586_e\n",
    "iCdG791.add_boundary(cpd11586_e, type='exchange', reaction_id='EX_cpd11586_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# ala-L-Thr-L\n",
    "rxn12644_c = universal_GramPos.reactions.rxn12644_c\n",
    "rxn12644_c.lower_bound = 0.\n",
    "rxn12644_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12616_c.remove_from_model(remove_orphans=False)\n",
    "rxn05545_c = universal_GramPos.reactions.rxn05545_c\n",
    "rxn05545_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12644_c,rxn05545_c])\n",
    "cpd11582_e = iCdG791.metabolites.cpd11582_e\n",
    "iCdG791.add_boundary(cpd11582_e, type='exchange', reaction_id='EX_cpd11582_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# gly-asn-L\n",
    "rxn12645_c = universal_GramPos.reactions.rxn12645_c\n",
    "rxn12645_c.lower_bound = 0.\n",
    "rxn12645_c.gene_reaction_rule = '272563.8.peg.1673 or 272563.8.peg.106 or 272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "universal_GramPos.reactions.rxn12542_c.remove_from_model(remove_orphans=False)\n",
    "rxn05546_c = universal_GramPos.reactions.rxn05546_c\n",
    "rxn05546_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12645_c,rxn05546_c])\n",
    "cpd11581_e = iCdG791.metabolites.cpd11581_e \n",
    "iCdG791.add_boundary(cpd11581_e, type='exchange', reaction_id='EX_cpd11581_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Reaction identifier</strong></td><td>EX_cpd15606_e</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Name</strong></td><td>Gly-Tyr exchange</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f3496862610</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Stoichiometry</strong></td>\n",
       "                <td>\n",
       "                    <p style='text-align:right'>cpd15606_e <=> </p>\n",
       "                    <p style='text-align:right'>Gly-Tyr <=> </p>\n",
       "                </td>\n",
       "            </tr><tr>\n",
       "                <td><strong>GPR</strong></td><td></td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Lower bound</strong></td><td>-1000.0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Upper bound</strong></td><td>1000.0</td>\n",
       "            </tr>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<Reaction EX_cpd15606_e at 0x7f3496862610>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remaining dipeptide import and Aminopeptidases in universal bag\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H\n",
    "\n",
    "# L-Prolinylglycine\n",
    "rxn08133_c = universal_GramPos.reactions.rxn08133_c\n",
    "rxn08133_c.lower_bound = 0.\n",
    "rxn08133_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "cpd15551_e = cobra.Metabolite(\n",
    "    'cpd15551_e',\n",
    "    formula='C7H11N2O3',\n",
    "    name='L-Prolinylglycine',\n",
    "    compartment='extracellular')\n",
    "cpd15551_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd15551_e.annotation['seed.compound'] = 'cpd15551'\n",
    "cpd15551_c = universal_GramPos.metabolites.cpd15551_c\n",
    "rxn09189_c = cobra.Reaction('rxn09189_c')\n",
    "rxn09189_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn09189_c.name = 'L-Prolinylglycine (Pro-Gly) transport via ABC system'\n",
    "rxn09189_c.bounds = (0., 1000.)\n",
    "rxn09189_c.add_metabolites({\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd15551_e: -1.0, # L-Prolinylglycine\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Pi\n",
    "    cpd15551_c: 1.0, # L-Prolinylglycine\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn09189_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn09189_c.annotation['seed.reaction'] = 'rxn09189'\n",
    "iCdG791.add_reactions([rxn08133_c,rxn09189_c])\n",
    "iCdG791.add_boundary(cpd15551_e, type='exchange', reaction_id='EX_cpd15551_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Cys\n",
    "rxn12844_c = universal_GramPos.reactions.rxn12844_c\n",
    "rxn12844_c.lower_bound = 0.\n",
    "rxn12844_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12848_c = universal_GramPos.reactions.rxn12848_c\n",
    "rxn12848_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12844_c,rxn12848_c])\n",
    "cpd15603_e = iCdG791.metabolites.cpd15603_e\n",
    "iCdG791.add_boundary(cpd15603_e, type='exchange', reaction_id='EX_cpd15603_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Leu\n",
    "rxn12845_c = universal_GramPos.reactions.rxn12845_c\n",
    "rxn12845_c.lower_bound = 0.\n",
    "rxn12845_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12849_c = universal_GramPos.reactions.rxn12849_c\n",
    "rxn12849_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12845_c,rxn12849_c])\n",
    "cpd15604_e = iCdG791.metabolites.cpd15604_e\n",
    "iCdG791.add_boundary(cpd15604_e, type='exchange', reaction_id='EX_cpd15604_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Phe\n",
    "rxn12846_c = universal_GramPos.reactions.rxn12846_c\n",
    "rxn12846_c.lower_bound = 0.\n",
    "rxn12846_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12850_c = universal_GramPos.reactions.rxn12850_c\n",
    "rxn12850_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12846_c,rxn12850_c])\n",
    "cpd15605_e = iCdG791.metabolites.cpd15605_e\n",
    "iCdG791.add_boundary(cpd15605_e, type='exchange', reaction_id='EX_cpd15605_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "# Gly-Try\n",
    "rxn12847_c = universal_GramPos.reactions.rxn12847_c\n",
    "rxn12847_c.lower_bound = 0.\n",
    "rxn12847_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn12851_c = universal_GramPos.reactions.rxn12851_c\n",
    "rxn12851_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "iCdG791.add_reactions([rxn12851_c,rxn12847_c])\n",
    "cpd15606_e = iCdG791.metabolites.cpd15606_e\n",
    "iCdG791.add_boundary(cpd15606_e, type='exchange', reaction_id='EX_cpd15606_e', lb=-1000.0, ub=1000.0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 150\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "82.98461354618698"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dipeptide-associated reactions not in universal bag\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H+\n",
    "\n",
    "# L-ala-L-asp\n",
    "cpd29695_e = cobra.Metabolite(\n",
    "    'cpd29695_e',\n",
    "    formula='C7H11N2O5',\n",
    "    name='L-ala-L-asp',\n",
    "    compartment='extracellular')\n",
    "cpd29695_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29695_e.annotation['seed.compound'] = 'cpd29695'\n",
    "iCdG791.add_boundary(cpd29695_e, type='exchange', reaction_id='EX_cpd29695_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29695_c = cobra.Metabolite(\n",
    "    'cpd29695_c',\n",
    "    formula='C7H11N2O5',\n",
    "    name='L-ala-L-asp',\n",
    "    compartment='cytosol')\n",
    "cpd29695_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29695_c.annotation['seed.compound'] = 'cpd29695'\n",
    "\n",
    "# Transport\n",
    "rxn30259_c = cobra.Reaction('rxn30259_c')\n",
    "rxn30259_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30259_c.name = 'Dipeptide transport via ABC system (ala-asp)'\n",
    "rxn30259_c.bounds = (0., 1000.)\n",
    "rxn30259_c.add_metabolites({\n",
    "    cpd29695_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29695_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30259_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30259_c.annotation['seed.reaction'] = 'rxn30259'\n",
    "iCdG791.add_reactions([rxn30259_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00041_c = iCdG791.metabolites.cpd00041_c # Aspartate\n",
    "rxn30237_c = cobra.Reaction('rxn30237_c')\n",
    "rxn30237_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30237_c.name = 'aminopeptidase (ala-asp)'\n",
    "rxn30237_c.bounds = (0., 1000.)\n",
    "rxn30237_c.add_metabolites({\n",
    "    cpd29695_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00041_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30237_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30237_c.annotation['seed.reaction'] = 'rxn30237'\n",
    "iCdG791.add_reactions([rxn30237_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-gly\n",
    "cpd29694_e = cobra.Metabolite(\n",
    "    'cpd29694_e',\n",
    "    formula='C5H13N2O2',\n",
    "    name='L-ala-gly',\n",
    "    compartment='extracellular')\n",
    "cpd29694_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29694_e.annotation['seed.compound'] = 'cpd29694'\n",
    "iCdG791.add_boundary(cpd29694_e, type='exchange', reaction_id='EX_cpd29694_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29694_c = cobra.Metabolite(\n",
    "    'cpd29694_c',\n",
    "    formula='C5H13N2O2',\n",
    "    name='L-ala-gly',\n",
    "    compartment='cytosol')\n",
    "cpd29694_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29694_c.annotation['seed.compound'] = 'cpd29694'\n",
    "\n",
    "# Transport\n",
    "rxn30262_c = cobra.Reaction('rxn30262_c')\n",
    "rxn30262_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30262_c.name = 'Dipeptide transport via ABC system (ala-gly)'\n",
    "rxn30262_c.bounds = (0., 1000.)\n",
    "rxn30262_c.add_metabolites({\n",
    "    cpd29694_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29694_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30262_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30262_c.annotation['seed.reaction'] = 'rxn30262'\n",
    "iCdG791.add_reactions([rxn30262_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "rxn30238_c = cobra.Reaction('rxn30238_c')\n",
    "rxn30238_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30238_c.name = 'Aminopeptidase (ala-gly)'\n",
    "rxn30238_c.bounds = (0., 1000.)\n",
    "rxn30238_c.add_metabolites({\n",
    "    cpd29694_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00033_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30238_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30238_c.annotation['seed.reaction'] = 'rxn30238'\n",
    "iCdG791.add_reactions([rxn30238_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-his\n",
    "cpd29697_e = cobra.Metabolite(\n",
    "    'cpd29697_e',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-his',\n",
    "    compartment='extracellular')\n",
    "cpd29697_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29697_e.annotation['seed.compound'] = 'cpd29697'\n",
    "iCdG791.add_boundary(cpd29697_e, type='exchange', reaction_id='EX_cpd29697_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29697_c = cobra.Metabolite(\n",
    "    'cpd29697_c',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-his',\n",
    "    compartment='cytosol')\n",
    "cpd29697_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29697_c.annotation['seed.compound'] = 'cpd29697'\n",
    "\n",
    "# Transport\n",
    "rxn30264_c = cobra.Reaction('rxn30264_c')\n",
    "rxn30264_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30264_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30264_c.bounds = (0., 1000.)\n",
    "rxn30264_c.add_metabolites({\n",
    "    cpd29697_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29697_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30264_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30264_c.annotation['seed.reaction'] = 'rxn30264'\n",
    "iCdG791.add_reactions([rxn30264_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00119_c = iCdG791.metabolites.cpd00119_c # Histidine\n",
    "rxn30239_c = cobra.Reaction('rxn30239_c')\n",
    "rxn30239_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30239_c.name = 'Aminopeptidase'\n",
    "rxn30239_c.bounds = (0., 1000.)\n",
    "rxn30239_c.add_metabolites({\n",
    "    cpd29697_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00119_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30239_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30239_c.annotation['seed.reaction'] = 'rxn30239'\n",
    "iCdG791.add_reactions([rxn30239_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-leu\n",
    "cpd29698_e = cobra.Metabolite(\n",
    "    'cpd29698_e',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-leu',\n",
    "    compartment='extracellular')\n",
    "cpd29698_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29698_e.annotation['seed.compound'] = 'cpd29698'\n",
    "iCdG791.add_boundary(cpd29698_e, type='exchange', reaction_id='EX_cpd29698_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29698_c = cobra.Metabolite(\n",
    "    'cpd29698_c',\n",
    "    formula='C9H14N4O3',\n",
    "    name='L-ala-L-leu',\n",
    "    compartment='cytosol')\n",
    "cpd29698_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29698_c.annotation['seed.compound'] = 'cpd29698'\n",
    "\n",
    "# Transport\n",
    "rxn30266_c = cobra.Reaction('rxn30266_c')\n",
    "rxn30266_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30266_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30266_c.bounds = (0., 1000.)\n",
    "rxn30266_c.add_metabolites({\n",
    "    cpd29698_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29698_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30266_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30266_c.annotation['seed.reaction'] = 'rxn30266'\n",
    "iCdG791.add_reactions([rxn30266_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00107_c = iCdG791.metabolites.cpd00107_c # Leucine\n",
    "rxn30240_c = cobra.Reaction('rxn30240_c')\n",
    "rxn30240_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30240_c.name = 'Aminopeptidase'\n",
    "rxn30240_c.bounds = (0., 1000.)\n",
    "rxn30240_c.add_metabolites({\n",
    "    cpd29698_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00107_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30240_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30240_c.annotation['seed.reaction'] = 'rxn30240'\n",
    "iCdG791.add_reactions([rxn30240_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-gln\n",
    "cpd29696_e = cobra.Metabolite(\n",
    "    'cpd29696_e',\n",
    "    formula='C8H15N3O4',\n",
    "    name='L-ala-L-gln',\n",
    "    compartment='extracellular')\n",
    "cpd29696_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29696_e.annotation['seed.compound'] = 'cpd29696'\n",
    "iCdG791.add_boundary(cpd29696_e, type='exchange', reaction_id='EX_cpd29696_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29696_c = cobra.Metabolite(\n",
    "    'cpd29696_c',\n",
    "    formula='C8H15N3O4',\n",
    "    name='L-ala-L-gln',\n",
    "    compartment='cytosol')\n",
    "cpd29696_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29696_c.annotation['seed.compound'] = 'cpd29696'\n",
    "\n",
    "# Transport\n",
    "rxn30268_c = cobra.Reaction('rxn30268_c')\n",
    "rxn30268_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30268_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30268_c.bounds = (0., 1000.)\n",
    "rxn30268_c.add_metabolites({\n",
    "    cpd29696_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29696_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30268_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30268_c.annotation['seed.reaction'] = 'rxn30268'\n",
    "iCdG791.add_reactions([rxn30268_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00053_c = iCdG791.metabolites.cpd00053_c # Glutamine\n",
    "rxn30241_c = cobra.Reaction('rxn30241_c')\n",
    "rxn30241_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30241_c.name = 'Aminopeptidase'\n",
    "rxn30241_c.bounds = (0., 1000.)\n",
    "rxn30241_c.add_metabolites({\n",
    "    cpd29696_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00053_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30241_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30241_c.annotation['seed.reaction'] = 'rxn30241'\n",
    "iCdG791.add_reactions([rxn30241_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-ala-L-thr\n",
    "cpd29699_e = cobra.Metabolite(\n",
    "    'cpd29699_e',\n",
    "    formula='C7H14N2O4',\n",
    "    name='L-ala-L-thr',\n",
    "    compartment='extracellular')\n",
    "cpd29699_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29699_e.annotation['seed.compound'] = 'cpd29699'\n",
    "iCdG791.add_boundary(cpd29699_e, type='exchange', reaction_id='EX_cpd29699_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29699_c = cobra.Metabolite(\n",
    "    'cpd29699_c',\n",
    "    formula='C7H14N2O4',\n",
    "    name='L-ala-L-thr',\n",
    "    compartment='cytosol')\n",
    "cpd29699_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29699_c.annotation['seed.compound'] = 'cpd29699'\n",
    "\n",
    "# Transport\n",
    "rxn30270_c = cobra.Reaction('rxn30270_c')\n",
    "rxn30270_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30270_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30270_c.bounds = (0., 1000.)\n",
    "rxn30270_c.add_metabolites({\n",
    "    cpd29699_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29699_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30270_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30270_c.annotation['seed.reaction'] = 'rxn30270'\n",
    "iCdG791.add_reactions([rxn30270_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "cpd00161_c = iCdG791.metabolites.cpd00161_c # Threonine\n",
    "rxn30242_c = cobra.Reaction('rxn30242_c')\n",
    "rxn30242_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30242_c.name = 'Aminopeptidase'\n",
    "rxn30242_c.bounds = (0., 1000.)\n",
    "rxn30242_c.add_metabolites({\n",
    "    cpd29699_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00035_c: 1.0, # peptide 1\n",
    "    cpd00161_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30242_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30242_c.annotation['seed.reaction'] = 'rxn30242'\n",
    "iCdG791.add_reactions([rxn30242_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# gly-L-met\n",
    "cpd29693_e = cobra.Metabolite(\n",
    "    'cpd29693_e',\n",
    "    formula='C7H14N2O3S',\n",
    "    name='gly-L-met',\n",
    "    compartment='extracellular')\n",
    "cpd29693_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29693_e.annotation['seed.compound'] = 'cpd29693'\n",
    "iCdG791.add_boundary(cpd29693_e, type='exchange', reaction_id='EX_cpd29693_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29693_c = cobra.Metabolite(\n",
    "    'cpd29693_c',\n",
    "    formula='C7H14N2O3S',\n",
    "    name='gly-L-met',\n",
    "    compartment='cytosol')\n",
    "cpd29693_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29693_c.annotation['seed.compound'] = 'cpd29693'\n",
    "\n",
    "# Transport\n",
    "rxn30274_c = cobra.Reaction('rxn30274_c')\n",
    "rxn30274_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30274_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30274_c.bounds = (0., 1000.)\n",
    "rxn30274_c.add_metabolites({\n",
    "    cpd29693_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29693_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30274_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30274_c.annotation['seed.reaction'] = 'rxn30274'\n",
    "iCdG791.add_reactions([rxn30274_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "cpd00060_c = iCdG791.metabolites.cpd00060_c # Methionine\n",
    "rxn30244_c = cobra.Reaction('rxn30244_c')\n",
    "rxn30244_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30244_c.name = 'Aminopeptidase'\n",
    "rxn30244_c.bounds = (0., 1000.)\n",
    "rxn30244_c.add_metabolites({\n",
    "    cpd29693_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00033_c: 1.0, # peptide 1\n",
    "    cpd00060_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30244_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30244_c.annotation['seed.reaction'] = 'rxn30244'\n",
    "iCdG791.add_reactions([rxn30244_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# gly-L-asn\n",
    "cpd29690_e = cobra.Metabolite(\n",
    "    'cpd29690_e',\n",
    "    formula='C6H11N3O4',\n",
    "    name='gly-L-asn',\n",
    "    compartment='extracellular')\n",
    "cpd29690_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29690_e.annotation['seed.compound'] = 'cpd29690'\n",
    "iCdG791.add_boundary(cpd29690_e, type='exchange', reaction_id='EX_cpd29690_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29690_c = cobra.Metabolite(\n",
    "    'cpd29690_c',\n",
    "    formula='C6H11N3O4',\n",
    "    name='gly-L-asn',\n",
    "    compartment='cytosol')\n",
    "cpd29690_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29690_c.annotation['seed.compound'] = 'cpd29690'\n",
    "\n",
    "# Transport\n",
    "rxn30276_c = cobra.Reaction('rxn30276_c')\n",
    "rxn30276_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30276_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30276_c.bounds = (0., 1000.)\n",
    "rxn30276_c.add_metabolites({\n",
    "    cpd29690_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29690_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30276_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30276_c.annotation['seed.reaction'] = 'rxn30276'\n",
    "iCdG791.add_reactions([rxn30276_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "cpd00132_c = iCdG791.metabolites.cpd00132_c # Asparagine\n",
    "\n",
    "rxn30245_c = cobra.Reaction('rxn30245_c')\n",
    "rxn30245_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30245_c.name = 'Aminopeptidase'\n",
    "rxn30245_c.bounds = (0., 1000.)\n",
    "rxn30245_c.add_metabolites({\n",
    "    cpd29690_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00033_c: 1.0, # peptide 1\n",
    "    cpd00132_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30245_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30245_c.annotation['seed.reaction'] = 'rxn30245'\n",
    "iCdG791.add_reactions([rxn30245_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# gly-L-gln\n",
    "cpd29691_e = cobra.Metabolite(\n",
    "    'cpd29691_e',\n",
    "    formula='C7H13N3O4',\n",
    "    name='gly-L-gln',\n",
    "    compartment='extracellular')\n",
    "cpd29691_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29691_e.annotation['seed.compound'] = 'cpd29691'\n",
    "iCdG791.add_boundary(cpd29691_e, type='exchange', reaction_id='EX_cpd29691_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29691_c = cobra.Metabolite(\n",
    "    'cpd29691_c',\n",
    "    formula='C7H13N3O4',\n",
    "    name='gly-L-gln',\n",
    "    compartment='cytosol')\n",
    "cpd29691_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29691_c.annotation['seed.compound'] = 'cpd29691'\n",
    "\n",
    "# Transport\n",
    "rxn30278_c = cobra.Reaction('rxn30278_c')\n",
    "rxn30278_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30278_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30278_c.bounds = (0., 1000.)\n",
    "rxn30278_c.add_metabolites({\n",
    "    cpd29691_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29691_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30278_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30278_c.annotation['seed.reaction'] = 'rxn30278'\n",
    "iCdG791.add_reactions([rxn30278_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00033_c = iCdG791.metabolites.cpd00033_c # Glycine\n",
    "cpd00053_c = iCdG791.metabolites.cpd00053_c # Glutamine\n",
    "rxn30246_c = cobra.Reaction('rxn30246_c')\n",
    "rxn30246_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30246_c.name = 'Aminopeptidase'\n",
    "rxn30246_c.bounds = (0., 1000.)\n",
    "rxn30246_c.add_metabolites({\n",
    "    cpd29691_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00033_c: 1.0, # peptide 1\n",
    "    cpd00053_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30246_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30246_c.annotation['seed.reaction'] = 'rxn30246'\n",
    "iCdG791.add_reactions([rxn30246_c])\n",
    "\n",
    "#------------------------------------------------------------------#\n",
    "\n",
    "# L-met-L-ala\n",
    "cpd29700_e = cobra.Metabolite(\n",
    "    'cpd29700_e',\n",
    "    formula='C8H16N2O3S',\n",
    "    name='L-met-L-ala',\n",
    "    compartment='extracellular')\n",
    "cpd29700_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29700_e.annotation['seed.compound'] = 'cpd29700'\n",
    "iCdG791.add_boundary(cpd29700_e, type='exchange', reaction_id='EX_cpd29700_e', lb=-1000.0, ub=1000.0)\n",
    "cpd29700_c = cobra.Metabolite(\n",
    "    'cpd29700_c',\n",
    "    formula='C8H16N2O3S',\n",
    "    name='L-met-L-ala',\n",
    "    compartment='cytosol')\n",
    "cpd29700_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd29700_c.annotation['seed.compound'] = 'cpd29700'\n",
    "\n",
    "# Transport\n",
    "rxn30280_c = cobra.Reaction('rxn30280_c')\n",
    "rxn30280_c.gene_reaction_rule = '(272563.8.peg.896 or 272563.8.peg.2794 or 272563.8.peg.895) and 272563.8.peg.2798 and (272563.8.peg.894 or 272563.8.peg.2795) and (272563.8.peg.893 or 272563.8.peg.2796)'\n",
    "rxn30280_c.name = 'Dipeptide transport via ABC system'\n",
    "rxn30280_c.bounds = (0., 1000.)\n",
    "rxn30280_c.add_metabolites({\n",
    "    cpd29700_e: -1.0, # dipeptide\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd29700_c: 1.0, # dipeptide\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn30280_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30280_c.annotation['seed.reaction'] = 'rxn30280'\n",
    "iCdG791.add_reactions([rxn30280_c])\n",
    "\n",
    "# Hydrolysis\n",
    "cpd00060_c = iCdG791.metabolites.cpd00060_c # Methionine\n",
    "cpd00035_c = iCdG791.metabolites.cpd00035_c # Alanine\n",
    "rxn30247_c = cobra.Reaction('rxn30247_c')\n",
    "rxn30247_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30247_c.name = 'Aminopeptidase'\n",
    "rxn30247_c.bounds = (0., 1000.)\n",
    "rxn30247_c.add_metabolites({\n",
    "    cpd29700_c: -1.0, # dipeptide\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00060_c: 1.0, # peptide 1\n",
    "    cpd00035_c: 1.0 # peptide 2\n",
    "})\n",
    "rxn30247_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn30247_c.annotation['seed.reaction'] = 'rxn30247'\n",
    "iCdG791.add_reactions([rxn30247_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ATP synthase\n",
    "iCdG791.reactions.rxn10042_c.remove_from_model(remove_orphans=False)\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H\n",
    "\n",
    "cpd00067_p = cobra.Metabolite(\n",
    "    'cpd00067_p',\n",
    "    formula='H',\n",
    "    name='H+',\n",
    "    compartment='periplasm')\n",
    "cpd00067_p.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00067_p.annotation['seed.compound'] = 'cpd00067'\n",
    "\n",
    "cpd00982_c = iCdG791.metabolites.cpd00982_c # FADH2\n",
    "cpd00015_c = iCdG791.metabolites.cpd00015_c # FAD\n",
    "\n",
    "rxn12212_c = cobra.Reaction('rxn12212_c')\n",
    "rxn12212_c.gene_reaction_rule = '272563.120.peg.1189 and 272563.8.peg.1199 and 272563.120.peg.1185 and 272563.120.peg.1186 and 272563.120.peg.1188 and 272563.120.peg.1187'\n",
    "rxn12212_c.name = 'FAD oxidation (electron transport complex)'\n",
    "rxn12212_c.bounds = (-1000., 1000.)\n",
    "rxn12212_c.add_metabolites({\n",
    "    cpd00982_c: -1.0, # FADH2\n",
    "    cpd00015_c: 1.0, # FAD\n",
    "    cpd00067_p: 2.0 # H+\n",
    "})\n",
    "rxn12212_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn12212_c.annotation['seed.reaction'] = 'rxn12212'\n",
    "iCdG791.add_reactions([rxn12212_c])\n",
    "\n",
    "cpd00003_c = iCdG791.metabolites.cpd00003_c # NAD\n",
    "cpd00004_c = iCdG791.metabolites.cpd00004_c # NADH\n",
    "\n",
    "rxn12213_c = cobra.Reaction('rxn12213_c')\n",
    "rxn12213_c.gene_reaction_rule = '272563.120.peg.1189 and 272563.8.peg.1199 and 272563.120.peg.1185 and 272563.120.peg.1186 and 272563.120.peg.1188 and 272563.120.peg.1187'\n",
    "rxn12213_c.name = 'FAD oxidation (electron transport complex)'\n",
    "rxn12213_c.bounds = (-1000., 1000.)\n",
    "rxn12213_c.add_metabolites({\n",
    "    cpd00004_c: -1.0, # FADH2\n",
    "    cpd00003_c: 1.0, # FAD\n",
    "    cpd00067_p: 2.0 # H+\n",
    "})\n",
    "rxn12213_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn12213_c.annotation['seed.reaction'] = 'rxn12213'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1189').name = 'Ion-translocating oxidoreductase complex subunit A | rnfA'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1189').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1189').annotation['patric'] = 'fig|272563.120.peg.1189'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.8.peg.1199').name = 'Ion-translocating oxidoreductase complex subunit B | rnfB'\n",
    "iCdG791.genes.get_by_id('272563.8.peg.1199').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.8.peg.1199').annotation['patric'] = 'fig|272563.8.peg.1199'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1185').name = 'Ion-translocating oxidoreductase complex subunit C | rnfC'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1185').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1185').annotation['patric'] = 'fig|272563.120.peg.1185'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1186').name = 'Ion-translocating oxidoreductase complex subunit D | rnfD'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1186').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1186').annotation['patric'] = 'fig|272563.120.peg.1186'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1188').name = 'Ion-translocating oxidoreductase complex subunit E | rnfE'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1188').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1188').annotation['patric'] = 'fig|272563.120.peg.1188'\n",
    "\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1187').name = 'Ion-translocating oxidoreductase complex subunit G | rnfG'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1187').annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG791.genes.get_by_id('272563.120.peg.1187').annotation['patric'] = 'fig|272563.120.peg.1187'\n",
    "\n",
    "rxn10042_c = cobra.Reaction('rxn10042_c')\n",
    "rxn10042_c.gene_reaction_rule = '272563.8.peg.3633 and 272563.8.peg.3632 and 272563.8.peg.3636 and 272563.8.peg.3635 and 272563.8.peg.3634 and (272563.8.peg.3640 or 272563.8.peg.3641)'\n",
    "rxn10042_c.name = 'ATP synthase'\n",
    "rxn10042_c.bounds = (0., 1000.)\n",
    "rxn10042_c.add_metabolites({\n",
    "    cpd00008_c: -1.0, # ADP\n",
    "    cpd00009_c: -0.05, # Pi\n",
    "    cpd00067_p: -1.0, # H+\n",
    "    cpd00001_c: 1.0, # H20\n",
    "    cpd00002_c: 1.0, # ATP\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn10042_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn10042_c.annotation['seed.reaction'] = 'rxn10042'\n",
    "iCdG791.add_reactions([rxn10042_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 143\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "75.01730567597323"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# PTS systems\n",
    "\n",
    "# Salicin-specific\n",
    "rxn05647_c = iCdG791.reactions.rxn05647_c\n",
    "iCdG791.reactions.rxn05647_c.remove_from_model(remove_orphans=False)\n",
    "rxn05647_c.lower_bound = 0.\n",
    "rxn05647_c.gene_reaction_rule = '272563.8.peg.408 or 272563.8.peg.854 or 272563.8.peg.3282 or 272563.8.peg.3260 or 272563.8.peg.3241 or 272563.8.peg.3270'\n",
    "iCdG791.add_reactions([rxn05647_c])\n",
    "\n",
    "# Fructose-specific\n",
    "rxn05560_c = iCdG791.reactions.rxn05560_c\n",
    "iCdG791.reactions.rxn05560_c.remove_from_model(remove_orphans=False)\n",
    "rxn05560_c.lower_bound = 0.\n",
    "rxn05560_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG791.add_reactions([rxn05560_c])\n",
    "rxn10175_c = iCdG791.reactions.rxn10175_c\n",
    "iCdG791.reactions.rxn10175_c.remove_from_model(remove_orphans=False)\n",
    "rxn10175_c.lower_bound = 0.\n",
    "rxn10175_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG791.add_reactions([rxn10175_c])\n",
    "\n",
    "# Mannitol-specific\n",
    "rxn05617_c = iCdG791.reactions.rxn05617_c\n",
    "iCdG791.reactions.rxn05617_c.remove_from_model(remove_orphans=False)\n",
    "rxn05617_c.lower_bound = 0.\n",
    "rxn05617_c.gene_reaction_rule = '(272563.8.peg.2449 or 272563.8.peg.3804) and 272563.8.peg.2450 and 272563.8.peg.2451'\n",
    "iCdG791.add_reactions([rxn05617_c])\n",
    "\n",
    "# Maltose-specific\n",
    "rxn05607_c = iCdG791.reactions.rxn05607_c\n",
    "iCdG791.reactions.rxn05607_c.remove_from_model(remove_orphans=False)\n",
    "rxn05607_c.lower_bound = 0.\n",
    "rxn05607_c.gene_reaction_rule = '272563.8.peg.3174 or 272563.8.peg.2632 or 272563.8.peg.1403'\n",
    "iCdG791.add_reactions([rxn05607_c])\n",
    "\n",
    "# Galactitol-specific\n",
    "rxn05567_c = iCdG791.reactions.rxn05567_c\n",
    "iCdG791.reactions.rxn05567_c.remove_from_model(remove_orphans=False)\n",
    "rxn05567_c.lower_bound = 0.\n",
    "rxn05567_c.gene_reaction_rule = '(272563.8.peg.50 or 272563.8.peg.2396) and (272563.8.peg.2395 or 272563.8.peg.1137 or 272563.8.peg.2443 or 272563.8.peg.51) and (272563.8.peg.52 or 272563.8.peg.2394 or 272563.8.peg.2442 or 272563.8.peg.2444)'\n",
    "iCdG791.add_reactions([rxn05567_c])\n",
    "\n",
    "# Sorbitol-specific\n",
    "rxn10184_c = iCdG791.reactions.rxn10184_c\n",
    "iCdG791.reactions.rxn10184_c.remove_from_model(remove_orphans=False)\n",
    "rxn10184_c.lower_bound = 0.\n",
    "rxn10184_c.gene_reaction_rule = '(272563.8.peg.804 or 272563.8.peg.2532) and (272563.8.peg.801 or 272563.8.peg.2536)'\n",
    "iCdG791.add_reactions([rxn10184_c])\n",
    "\n",
    "# Mannose-specific\n",
    "rxn05610_c = iCdG791.reactions.rxn05610_c\n",
    "iCdG791.reactions.rxn05610_c.remove_from_model(remove_orphans=False)\n",
    "rxn05610_c.lower_bound = 0.\n",
    "rxn05610_c.gene_reaction_rule = '(272563.8.peg.3431 or 272563.8.peg.304 or 272563.8.peg.515 or 272563.8.peg.1128) and (272563.8.peg.307 or 272563.8.peg.516 or 272563.8.peg.3432 or 272563.8.peg.1130 or 272563.8.peg.305) and (272563.8.peg.308 or 272563.8.peg.3430 or 272563.8.peg.517 or 272563.8.peg.1131) and (272563.8.peg.518 or 272563.8.peg.1132 or 272563.8.peg.309 or 272563.8.peg.3429)'\n",
    "iCdG791.add_reactions([rxn05610_c])\n",
    "\n",
    "# Glucose-specific\n",
    "rxn05226_c = iCdG791.reactions.rxn05226_c\n",
    "iCdG791.reactions.rxn05226_c.remove_from_model(remove_orphans=False)\n",
    "rxn05226_c.lower_bound = 0.\n",
    "rxn05226_c.gene_reaction_rule = '(272563.8.peg.3202 or 272563.8.peg.3247 or 272563.8.peg.3171 or 272563.8.peg.2634) and (272563.8.peg.2791 or 272563.8.peg.2790) and (272563.8.peg.3174 or 272563.8.peg.2632 or 272563.8.peg.1403)'\n",
    "iCdG791.add_reactions([rxn05226_c])\n",
    "\n",
    "# Cellobiose-specific\n",
    "rxn05518_c = iCdG791.reactions.rxn05518_c\n",
    "iCdG791.reactions.rxn05518_c.remove_from_model(remove_orphans=False)\n",
    "rxn05518_c.lower_bound = 0.\n",
    "rxn05518_c.gene_reaction_rule = '(272563.8.peg.3225 or 272563.8.peg.3012 or 272563.8.peg.153 or 272563.8.peg.901 or 272563.8.peg.3820 or 272563.8.peg.3192 or 272563.8.peg.3825) and (272563.8.peg.3224 or 272563.8.peg.3009 or 272563.8.peg.151 or 272563.8.peg.3822 or 272563.8.peg.902) and (272563.8.peg.3823 or 272563.8.peg.3226 or 272563.8.peg.152 or 272563.8.peg.3013 or 272563.8.peg.1679)'\n",
    "iCdG791.add_reactions([rxn05518_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 71\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71.1778745591431"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final reactions needed\n",
    "cpd00320_c = cobra.Metabolite(\n",
    "    'cpd00320_c',\n",
    "    formula='C4H6NO4',\n",
    "    name='D-Aspartate',\n",
    "    compartment='cytosol')\n",
    "cpd00320_c.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00320_c.annotation['seed.compound'] = 'cpd00320'\n",
    "cpd00320_e = cobra.Metabolite(\n",
    "    'cpd00320_e',\n",
    "    formula='C4H6NO4',\n",
    "    name='D-Aspartate',\n",
    "    compartment='extracellular')\n",
    "cpd00320_e.annotation['sbo'] = 'SBO:0000247'\n",
    "cpd00320_e.annotation['seed.compound'] = 'cpd00320'\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H\n",
    "rxn05178_c = cobra.Reaction('rxn05178_c')\n",
    "rxn05178_c.name = 'D-Aspartate-ABC transport'\n",
    "rxn05178_c.bounds = (-1000., 1000.)\n",
    "rxn05178_c.add_metabolites({\n",
    "    cpd00320_e: -1.0, \n",
    "    cpd00002_c: -1.0, \n",
    "    cpd00001_c: -1.0, \n",
    "    cpd00008_c: 1.0, \n",
    "    cpd00009_c: 1.0, \n",
    "    cpd00067_c: 1.0, \n",
    "    cpd00320_c: 1.0 \n",
    "})\n",
    "rxn05178_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn05178_c.annotation['seed.reaction'] = 'rxn05178'\n",
    "iCdG791.add_reactions([rxn05178_c])\n",
    "cpd00041_c = iCdG791.metabolites.cpd00041_c\n",
    "rxn00348_c = cobra.Reaction('rxn00348_c')\n",
    "rxn00348_c.gene_reaction_rule = '272563.8.peg.2370 or 272563.8.peg.2133'\n",
    "rxn00348_c.name = 'aspartate racemase'\n",
    "rxn00348_c.bounds = (-1000., 1000.)\n",
    "rxn00348_c.add_metabolites({\n",
    "    cpd00041_c: -1.0, # L-aspartate\n",
    "    cpd00320_c: 1.0 # D-aspartate\n",
    "})\n",
    "rxn00348_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn00348_c.annotation['seed.reaction'] = 'rxn00348'\n",
    "iCdG791.add_reactions([rxn00348_c])\n",
    "iCdG791.add_boundary(cpd00320_e, type='exchange', reaction_id='EX_cpd00320_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "\n",
    "cpd02711_e = cobra.Metabolite(\n",
    "    'cpd02711_e',\n",
    "    formula='C6H8O9P',\n",
    "    name='2-keto-3-deoxygluconate',\n",
    "    compartment='extracellular')\n",
    "cpd02711_c = iCdG791.metabolites.cpd02711_c\n",
    "cpd00067_e = iCdG791.metabolites.cpd00067_e\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c\n",
    "rxn05681_c = cobra.Reaction('rxn05681_c')\n",
    "rxn05681_c.gene_reaction_rule = '272563.8.peg.3148 or 272563.8.peg.2999'\n",
    "rxn05681_c.name = '2-dehydro-3-deoxy-D-gluconate transport via proton symport, reversible'\n",
    "rxn05681_c.bounds = (-1000., 1000.)\n",
    "rxn05681_c.add_metabolites({\n",
    "    cpd02711_e: -1.0,\n",
    "    cpd00067_e: -1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd02711_c: 1.0 \n",
    "})\n",
    "rxn05681_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn05681_c.annotation['seed.reaction'] = 'rxn05681'\n",
    "iCdG791.add_reactions([rxn05681_c])\n",
    "iCdG791.add_boundary(cpd02711_e, type='exchange', reaction_id='EX_cpd02711_e', lb=-1000.0, ub=1000.0)\n",
    "\n",
    "\n",
    "cpd00025_c = iCdG791.metabolites.cpd00025_c\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c\n",
    "rxn00008_c = cobra.Reaction('rxn00008_c')\n",
    "rxn00008_c.gene_reaction_rule = '272563.8.peg.1708'\n",
    "rxn00008_c.name = 'Mn(II):hydrogen-peroxide oxidoreductase'\n",
    "rxn00008_c.bounds = (0., 1000.)\n",
    "rxn00008_c.add_metabolites({\n",
    "    cpd00025_c: -1.0, # H202\n",
    "    cpd00067_c: -2.0, # H+\n",
    "    cpd00001_c: 2.0 # H20\n",
    "})\n",
    "rxn00008_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn00008_c.annotation['seed.reaction'] = 'rxn00008'\n",
    "iCdG791.add_reactions([rxn00008_c])\n",
    "\n",
    "cpd00038_c = iCdG791.metabolites.cpd00038_c # GTP\n",
    "cpd00031_c = iCdG791.metabolites.cpd00031_c # GDP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Pi\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H+\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H20\n",
    "cpd10515_c = iCdG791.metabolites.cpd10515_c # Fe\n",
    "cpd10515_e = iCdG791.metabolites.cpd10515_e # Fe\n",
    "rxn34423_c = cobra.Reaction('rxn34423_c')\n",
    "rxn34423_c.gene_reaction_rule = '272563.8.peg.3427 or 272563.8.peg.1551 or 272563.8.peg.1593'\n",
    "rxn34423_c.name = 'ferrous iron uptake (GTP-dependent), plasma membrane'\n",
    "rxn34423_c.bounds = (0., 1000.)\n",
    "rxn34423_c.add_metabolites({\n",
    "    cpd00001_c: -1.0,\n",
    "    cpd00038_c: -1.0,\n",
    "    cpd10515_e: -1.0,\n",
    "    cpd00009_c: 1.0,\n",
    "    cpd00031_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd10515_c: 1.0\n",
    "})\n",
    "rxn34423_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn34423_c.annotation['seed.reaction'] = 'rxn34423'\n",
    "iCdG791.add_reactions([rxn34423_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Other miscellaneous genes\n",
    "iCdG791.reactions.rxn05172_c.gene_reaction_rule = '(272563.8.peg.1554 or 272563.8.peg.3135) and (272563.8.peg.1556 or 272563.8.peg.2482) and (272563.8.peg.3134 or 272563.8.peg.1555 or 272563.8.peg.1052)'\n",
    "iCdG791.reactions.rxn05298_c.gene_reaction_rule = '272563.8.peg.2286 or 272563.8.peg.2818 or 272563.8.peg.1830'\n",
    "iCdG791.reactions.rxn00337_c.gene_reaction_rule = '272563.8.peg.1389 or 272563.8.peg.2161'\n",
    "iCdG791.reactions.rxn05209_c.gene_reaction_rule = '272563.8.peg.2793 or 272563.8.peg.1732 or 272563.8.peg.473 or 272563.8.peg.2861'\n",
    "iCdG791.reactions.rxn01522_c.gene_reaction_rule = '(272563.8.peg.2194 or 272563.8.peg.3328) and (272563.8.peg.2181 or 272563.8.peg.2187) and (272563.8.peg.2188 or 272563.8.peg.2195)'\n",
    "iCdG791.reactions.rxn00414_c.gene_reaction_rule = '(272563.8.peg.3759 or 272563.8.peg.3761) and (272563.8.peg.3760 or 272563.8.peg.3762)'\n",
    "iCdG791.reactions.rxn20606_c.gene_reaction_rule = '272563.120.peg.3426 and (272563.8.peg.3394 or 272563.8.peg.3392) and (272563.8.peg.3396 or 272563.8.peg.3393)'\n",
    "iCdG791.reactions.rxn00785_c.gene_reaction_rule = '(272563.8.peg.3625 or 272563.8.peg.2439) and (272563.8.peg.2438 or 272563.8.peg.3624)'\n",
    "iCdG791.reactions.rxn01200_c.gene_reaction_rule = '(272563.8.peg.3625 or 272563.8.peg.2439) and (272563.8.peg.2438 or 272563.8.peg.3624)'\n",
    "iCdG791.reactions.rxn01301_c.gene_reaction_rule = '272563.8.peg.1224 or 272563.8.peg.2531 or 272563.8.peg.1655'\n",
    "iCdG791.reactions.rxn00908_c.gene_reaction_rule = '272563.8.peg.240 and 272563.8.peg.766'\n",
    "iCdG791.reactions.rxn01225_c.gene_reaction_rule = '272563.8.peg.3332 or 272563.8.peg.1740'\n",
    "iCdG791.reactions.rxn03136_c.gene_reaction_rule = '272563.8.peg.513 or 272563.8.peg.237'\n",
    "iCdG791.reactions.rxn13022_c.gene_reaction_rule = '272563.8.peg.1041 and 272563.8.peg.1042'\n",
    "iCdG791.reactions.rxn03886_c.gene_reaction_rule = '272563.8.peg.805'\n",
    "iCdG791.reactions.rxn00178_c.gene_reaction_rule = '272563.8.peg.1111 or 272563.8.peg.2800'\n",
    "iCdG791.reactions.rxn08817_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08849_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08842_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08819_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08818_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08840_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08848_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08851_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08846_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08820_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08847_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08850_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08821_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08843_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08839_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08844_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08822_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08845_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08841_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08838_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn08823_c.gene_reaction_rule = '272563.8.peg.1645'\n",
    "iCdG791.reactions.rxn00693_c.gene_reaction_rule = '272563.8.peg.3768 or 272563.8.peg.1815'\n",
    "iCdG791.reactions.rxn00143_c.gene_reaction_rule = '272563.8.peg.2697 or 272563.8.peg.2734'\n",
    "iCdG791.reactions.rxn01519_c.gene_reaction_rule = '272563.8.peg.2521'\n",
    "iCdG791.reactions.rxn00493_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn02320_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn00527_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn01270_c.gene_reaction_rule = '272563.8.peg.2315 or 272563.8.peg.2624'\n",
    "iCdG791.reactions.rxn02320_c.gene_reaction_rule = '272563.8.peg.1625'\n",
    "iCdG791.reactions.rxn03167_c.gene_reaction_rule = '272563.8.peg.1521'\n",
    "iCdG791.reactions.rxn00549_c.gene_reaction_rule = '272563.8.peg.1250'\n",
    "iCdG791.reactions.rxn08040_c.gene_reaction_rule = '272563.8.peg.818'\n",
    "iCdG791.reactions.rxn05555_c.gene_reaction_rule = '272563.8.peg.1726 and (272563.8.peg.1725 or 272563.8.peg.1724) and 272563.8.peg.1727'\n",
    "iCdG791.reactions.rxn05183_c.gene_reaction_rule = '(272563.8.peg.1563 or 272563.8.peg.1730) and 272563.8.peg.1562'\n",
    "iCdG791.reactions.rxn00146_c.gene_reaction_rule = '272563.120.peg.414 or 272563.8.peg.414'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Glucose ABC transport\n",
    "\n",
    "cpd00002_c = iCdG791.metabolites.cpd00002_c # ATP\n",
    "cpd00001_c = iCdG791.metabolites.cpd00001_c # H2O\n",
    "cpd00008_c = iCdG791.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG791.metabolites.cpd00009_c # Phosphate\n",
    "cpd00067_c = iCdG791.metabolites.cpd00067_c # H+\n",
    "cpd00027_e = iCdG791.metabolites.cpd00027_e # Glucose\n",
    "cpd00079_c = iCdG791.metabolites.cpd00079_c # Glucose 6-p\n",
    "\n",
    "# Transport\n",
    "rxn05147_c = cobra.Reaction('rxn05147_c')\n",
    "rxn05147_c.gene_reaction_rule = '(272563.8.peg.3202 or 272563.8.peg.3247 or 272563.8.peg.3171 or 272563.8.peg.2634) and (272563.8.peg.2791 or 272563.8.peg.2790) and (272563.8.peg.3174 or 272563.8.peg.2632 or 272563.8.peg.1403)'\n",
    "rxn05147_c.name = 'D-Glucose-ABC transport'\n",
    "rxn05147_c.bounds = (0., 1000.)\n",
    "rxn05147_c.add_metabolites({\n",
    "    cpd00027_e: -1.0, # g\n",
    "    cpd00002_c: -1.0, # ATP\n",
    "    cpd00001_c: -1.0, # H2O\n",
    "    cpd00079_c: 1.0, # g 6-p\n",
    "    cpd00008_c: 1.0, # ADP\n",
    "    cpd00009_c: 1.0, # Phosphate\n",
    "    cpd00067_c: 1.0 # H+\n",
    "})\n",
    "rxn05147_c.annotation['sbo'] = 'SBO:0000176'\n",
    "rxn05147_c.annotation['seed.reaction'] = 'rxn05147'\n",
    "iCdG791.add_reactions([rxn05147_c])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reamining genes: 0\n"
     ]
    }
   ],
   "source": [
    "test = 0\n",
    "for x in iCdG791.genes:\n",
    "    if (len(x.reactions)) == 0:\n",
    "        test += 1\n",
    "print('Reamining genes: ' + str(test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "97.22639015406979"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename GENRE\n",
    "iCdG791.id = 'iCdG' + str(len(iCdG791.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add additional annotation information\n",
    "\n",
    "# BiGG\n",
    "BiGG_rxn_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/rxns_BiGG_ids.txt', 'r') as bigg_rxns:\n",
    "    for line in bigg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        bigg = line[1].strip().split(',')\n",
    "        if len(bigg) == 1: bigg = bigg[0]\n",
    "        BiGG_rxn_ids[modelseed] = bigg\n",
    "BiGG_cpd_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/cpds_BiGG_ids.txt', 'r') as bigg_rxns:\n",
    "    for line in bigg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        bigg = line[1].strip().split(',')\n",
    "        if len(bigg) == 1: bigg = bigg[0]\n",
    "        BiGG_cpd_ids[modelseed] = bigg\n",
    "\n",
    "# KEGG\n",
    "KEGG_rxn_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/rxns_KEGG_ids.txt', 'r') as kegg_rxns:\n",
    "    for line in kegg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        kegg = line[1].strip().split(',')\n",
    "        if len(kegg) == 1: kegg = kegg[0]\n",
    "        KEGG_rxn_ids[modelseed] = kegg\n",
    "KEGG_cpd_ids = {}\n",
    "with open('/home/mjenior/Desktop/repos/cpds_KEGG_ids.txt', 'r') as kegg_rxns:\n",
    "    for line in kegg_rxns:\n",
    "        line = line.split('=')\n",
    "        modelseed = line[0]\n",
    "        kegg = line[1].strip().split(',')\n",
    "        if len(kegg) == 1: kegg = kegg[0]\n",
    "        KEGG_cpd_ids[modelseed] = kegg\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in iCdG791.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        iCdG791.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        iCdG791.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in iCdG791.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        iCdG791.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        iCdG791.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removed blocked, no-gene reactions\n",
    "# Lysophospholipase L1 (periplasm)\n",
    "lysophospholipase = ['rxn08807_c','rxn08796_c','rxn08802_c','rxn08814_c','rxn08801_c','rxn08798_c','rxn08804_c','rxn08812_c','rxn08803_c','rxn08813_c','rxn08806_c','rxn08810_c','rxn08805_c','rxn08811_c','rxn08800_c','rxn08816_c']\n",
    "for rxn in lysophospholipase: iCdG791.reactions.get_by_id(rxn).remove_from_model(remove_orphans=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG706</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f349aff0850</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1182</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1205</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG706 at 0x7f349aff0850>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG791"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save model\n",
    "cobra.io.write_sbml_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG706.sbml')\n",
    "cobra.io.save_json_model(iCdG791, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG706.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read in again for gap-filling\n",
    "import cobra\n",
    "iCdG706 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG706.sbml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Custom gap-filler\n",
    "import copy\n",
    "import cobra\n",
    "import symengine\n",
    "\n",
    "# pFBA gapfiller\n",
    "def pfba_gapfill(model, universal, obj='biomass', demand='none', extracellular='extracellular'):\n",
    "    '''\n",
    "    Parameters\n",
    "    ----------\n",
    "    model_file : model\n",
    "        Model to be gapfilled\n",
    "    universal_file : model\n",
    "        Reaction bag reference\n",
    "    obj : str\n",
    "        Name of objective reaction\n",
    "    demand : str\n",
    "        Demand metabolite ID, circumvents objective\n",
    "    extracellular : str\n",
    "        Label for extracellular compartment of model\n",
    "    '''\n",
    "\n",
    "    # Save some basic network info for downstream membership testing\n",
    "    target_rxns = set([str(x.id) for x in model.reactions])\n",
    "    target_cpds = set([str(y.id) for y in model.metabolites])\n",
    "    ref_rxns = set([str(z.id) for z in universal.reactions])\n",
    "    shared_rxns = ref_rxns.intersection(target_rxns)\n",
    "        \n",
    "    # Remove overlapping reactions from universal bag, add model reactions to universal bag\n",
    "    temp_universal = copy.deepcopy(universal)\n",
    "    for rxn in shared_rxns: temp_universal.reactions.get_by_id(rxn).remove_from_model()\n",
    "    temp_universal.add_reactions(list(copy.deepcopy(model.reactions)))\n",
    "    \n",
    "    # Set objective\n",
    "    if demand != 'none':\n",
    "        cpd = temp_universal.metabolites.get_by_id(demand)\n",
    "        demand = cobra.Reaction('demand')\n",
    "        demand.bounds = (1., 1000.)\n",
    "        demand.add_metabolites({cpd: -1.0})\n",
    "        temp_universal.add_reactions([demand])\n",
    "        temp_universal.objective = demand.id\n",
    "    else:\n",
    "        temp_universal.objective = obj\n",
    "    obj_constraint = temp_universal.problem.Constraint(temp_universal.objective.expression, lb=10., ub=1000.)\n",
    "    temp_universal.add_cons_vars([obj_constraint])\n",
    "    temp_universal.solver.update()\n",
    "    \n",
    "    # Test to make sure objective flux is acheived with addition of reaction bag\n",
    "    solution = temp_universal.optimize()\n",
    "    if solution.objective_value <= 1e-6: \n",
    "        raise ValueError('ERROR: No objective flux possible with current reaction set.')    \n",
    "    \n",
    "    ## Set up pFBA objective\n",
    "    pfba_expr = symengine.RealDouble(0)\n",
    "    for rxn in temp_universal.reactions:\n",
    "        if rxn.id in target_rxns:\n",
    "            pfba_expr += 0. * rxn.forward_variable\n",
    "            pfba_expr += 0. * rxn.reverse_variable\n",
    "        else:\n",
    "            pfba_expr += 1. * rxn.forward_variable\n",
    "            pfba_expr += 1. * rxn.reverse_variable\n",
    "    temp_universal.objective = temp_universal.problem.Objective(pfba_expr, direction='min', sloppy=True)\n",
    "    temp_universal.solver.update()\n",
    "\n",
    "    # Run FBA and save solution\n",
    "    solution = temp_universal.optimize()\n",
    "    active_rxns = set([rxn.id for rxn in temp_universal.reactions if abs(solution.fluxes[rxn.id]) > 1e-6])\n",
    "    \n",
    "    # Screen new reaction IDs\n",
    "    new_rxns = active_rxns.difference(target_rxns)\n",
    "    new_rxns = new_rxns.difference(set(['demand']))\n",
    "    print('New reactions: ' + str(len(new_rxns)))\n",
    "    print('New objective value: ' + str(solution.objective_value))\n",
    "    \n",
    "    return new_rxns\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'GramPositive.modeltemplate' is not a valid SBML 'SId'.\n",
      "No objective coefficients in model. Unclear what should be optimized\n"
     ]
    }
   ],
   "source": [
    "# Read in curated universal reacton bag and annotate\n",
    "universal_GramPos = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/GramPos.sbml')\n",
    "universal_GramPos.reactions.rxn13782_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13783_c.remove_from_model(remove_orphans=True)\n",
    "universal_GramPos.reactions.rxn13784_c.remove_from_model(remove_orphans=True)\n",
    "#universal_GramPos.reactions.rxn11920_c.name = 'Poly-beta-hydroxybutyrate (PHB) synthease'\n",
    "universal_GramPos.reactions.rxn12188_c.name = '2-deoxyglucose-6-phosphatase'\n",
    "\n",
    "for cpd in universal_GramPos.metabolites:\n",
    "    cpd.annotation['sbo'] = 'SBO:0000247'\n",
    "    cpd.annotation['seed.compound'] = cpd.id.split('_')[0]    \n",
    "    cpd.compartment = cpd.compartment.lower()\n",
    "\n",
    "for rxn in universal_GramPos.reactions:\n",
    "    rxn.annotation['sbo'] = 'SBO:0000176'\n",
    "    rxn.annotation['seed.reaction'] = rxn.id.split('_')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e', # Pyridoxine\n",
    "       'EX_cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "# Theriot et al. (2013). Nature Communications.\n",
    "ncmm = ['EX_cpd00001_e', # water\n",
    "        'EX_cpd00104_e', # Biotin\n",
    "        'EX_cpd00644_e', # Pantothenate\n",
    "        'EX_cpd00263_e', # Pyridoxine\n",
    "        'EX_cpd00149_e', # Cobalt\n",
    "        'EX_cpd00099_e', # Chloride\n",
    "        'EX_cpd10515_e', # Iron\n",
    "        'EX_cpd00030_e', # Manganese\n",
    "        'EX_cpd00254_e', # Magnesium\n",
    "        'EX_cpd00063_e', # Calcium\n",
    "        'EX_cpd00205_e', # Potassium\n",
    "        'EX_cpd00009_e', # Phosphate\n",
    "        'EX_cpd00971_e', # Sodium\n",
    "        'EX_cpd00242_e', # Carbonate\n",
    "        'EX_cpd00322_e', # L-Isoleucine\n",
    "        'EX_cpd00129_e', # L-Proline\n",
    "        'EX_cpd00156_e', # L-Valine\n",
    "        'EX_cpd00107_e', # L-Leucine\n",
    "        'EX_cpd00084_e', # L-Cysteine \n",
    "        'EX_cpd00065_e', # L-Tryptophan\n",
    "        'EX_cpd00060_e', # L-Methionine\n",
    "        'EX_cpd00119_e', # L-Histidine\n",
    "        'EX_cpd00033_e', # Glycine\n",
    "        'EX_cpd00051_e', # L-Arginine\n",
    "        'EX_cpd00161_e'] # L-Threonine\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New reactions: 1\n",
      "New objective value: 0.0025000000001482686\n"
     ]
    }
   ],
   "source": [
    "# Change media and gapfill - MDM\n",
    "for rxn in iCdG706.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Pre-gapfill objective value: ' + str(test))\n",
    "\n",
    "gapfilled_rxns = pfba_gapfill(iCdG706, universal_GramPos) # Biomass alone\n",
    "gapfilled_rxns1 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17043_c') # RNA polymer\n",
    "gapfilled_rxns2 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17041_c') # Protein polymer\n",
    "gapfilled_rxns3 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd12894_c') # Teichoic acid\n",
    "gapfilled_rxns4 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd16661_c') # Peptidoglycan polymer\n",
    "gapfilled_rxns5 = pfba_gapfill(iCdG706, universal_GramPos, demand='cellwall_c') # Cell Wall polymer\n",
    "gapfilled_rxns6 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd11852_c') # Lipid Pool\n",
    "gapfilled_rxns7 = pfba_gapfill(iCdG706, universal_GramPos, demand='cofactor_c') # Cofactor Pool\n",
    "gapfilled_rxns8 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17042_c') # DNA polymer\n",
    "\n",
    "mdm_gapfilled_rxns = gapfilled_rxns1.union(gapfilled_rxns2).union(gapfilled_rxns3).union(gapfilled_rxns4).union(gapfilled_rxns5).union(gapfilled_rxns6).union(gapfilled_rxns7).union(gapfilled_rxns8).union(gapfilled_rxns)\n",
    "print('Total reactions to be added: ' + str(len(mdm_gapfilled_rxns)))\n",
    "\n",
    "rxns_to_be_added = []\n",
    "for rxn in mdm_gapfilled_rxns: \n",
    "    rxns_to_be_added.append(copy.deepcopy(universal_GramPos.reactions.get_by_id(rxn)))\n",
    "iCdG706.add_reactions(rxns_to_be_added)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Post-gapfill objective value: ' + str(test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pre-gapfill objective value: 12.559386622123775\n",
      "New reactions: 0\n",
      "New objective value: 3.9164191226586654e-14\n",
      "New reactions: 0\n",
      "New objective value: 10.000000000000194\n",
      "New reactions: 0\n",
      "New objective value: 9.999999999999883\n",
      "New reactions: 0\n",
      "New objective value: 10.000000000000401\n",
      "New reactions: 2\n",
      "New objective value: 61.93957115009697\n",
      "New reactions: 4\n",
      "New objective value: 385.581871345032\n",
      "New reactions: 0\n",
      "New objective value: 10.0\n",
      "New reactions: 0\n",
      "New objective value: 10.0\n",
      "New reactions: 0\n",
      "New objective value: 10.000000000000016\n",
      "Total reactions to be added: 5\n",
      "Post-gapfill objective value: 44.875570765519235\n"
     ]
    }
   ],
   "source": [
    "# Change media and gapfill - NCMM\n",
    "for rxn in iCdG706.boundary:\n",
    "    if not rxn.id in ncmm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Pre-gapfill objective value: ' + str(test))\n",
    "\n",
    "gapfilled_rxns = pfba_gapfill(iCdG706, universal_GramPos) # Biomass alone\n",
    "gapfilled_rxns1 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17043_c') # RNA polymer\n",
    "gapfilled_rxns2 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17041_c') # Protein polymer\n",
    "gapfilled_rxns3 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd12894_c') # Teichoic acid\n",
    "gapfilled_rxns4 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd16661_c') # Peptidoglycan polymer\n",
    "gapfilled_rxns5 = pfba_gapfill(iCdG706, universal_GramPos, demand='cellwall_c') # Cell Wall polymer\n",
    "gapfilled_rxns6 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd11852_c') # Lipid Pool\n",
    "gapfilled_rxns7 = pfba_gapfill(iCdG706, universal_GramPos, demand='cofactor_c') # Cofactor Pool\n",
    "gapfilled_rxns8 = pfba_gapfill(iCdG706, universal_GramPos, demand='cpd17042_c') # DNA polymer\n",
    "\n",
    "ncmm_gapfilled_rxns = gapfilled_rxns1.union(gapfilled_rxns2).union(gapfilled_rxns3).union(gapfilled_rxns4).union(gapfilled_rxns5).union(gapfilled_rxns6).union(gapfilled_rxns7).union(gapfilled_rxns8).union(gapfilled_rxns)\n",
    "print('Total reactions to be added: ' + str(len(ncmm_gapfilled_rxns)))\n",
    "\n",
    "rxns_to_be_added = []\n",
    "for rxn in ncmm_gapfilled_rxns: \n",
    "    rxns_to_be_added.append(copy.deepcopy(universal_GramPos.reactions.get_by_id(rxn)))\n",
    "iCdG706.add_reactions(rxns_to_be_added)\n",
    "test = iCdG706.slim_optimize()\n",
    "print('Post-gapfill objective value: ' + str(test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct names and possible gene annotations\n",
    "all_gapfilled_rxns = ncmm_gapfilled_rxns.union(mdm_gapfilled_rxns)\n",
    "\n",
    "iCdG706.reactions.rxn09655_c.name = 'Trehalose transport via proton symport'\n",
    "iCdG706.reactions.rxn09658_c.name = 'Sucrose transport via proton symport'\n",
    "iCdG706.reactions.rxn10183_c.name = 'N-acetyl-D-mannosamine transport via PTS'\n",
    "\n",
    "gene = cobra.Gene('272563.120.peg.1564')\n",
    "gene.annotation['patric'] = 'fig|272563.120.peg.1564'\n",
    "gene.annotation['sbo'] = 'SBO:0000243'\n",
    "iCdG706.reactions.rxn00504_c.gene_reaction_rule = '272563.120.peg.1564'\n",
    "iCdG706.genes.get_by_id('272563.120.peg.1564').name = 'Pyrroline-5-carboxylate reductase | proC'\n",
    "\n",
    "iCdG706.reactions.rxn09655_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG706.reactions.rxn09658_c.gene_reaction_rule = '(272563.8.peg.2383 or 272563.8.peg.224 or 272563.8.peg.3279) and 272563.8.peg.226 and 272563.8.peg.225'\n",
    "iCdG706.reactions.rxn10183_c.gene_reaction_rule = '(272563.8.peg.3202 or 272563.8.peg.3247 or 272563.8.peg.3171 or 272563.8.peg.2634) and (272563.8.peg.2790 or 272563.8.peg.2791)'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename GENRE\n",
    "iCdG706.id = 'iCdG' + str(len(iCdG706.genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "for rxn in iCdG706.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <table>\n",
       "            <tr>\n",
       "                <td><strong>Name</strong></td>\n",
       "                <td>iCdG707</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Memory address</strong></td>\n",
       "                <td>0x07f40ef778210</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of metabolites</strong></td>\n",
       "                <td>1184</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of reactions</strong></td>\n",
       "                <td>1214</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Number of groups</strong></td>\n",
       "                <td>0</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Objective expression</strong></td>\n",
       "                <td>1.0*biomass - 1.0*biomass_reverse_01e59</td>\n",
       "            </tr><tr>\n",
       "                <td><strong>Compartments</strong></td>\n",
       "                <td>cytosol, extracellular, periplasm</td>\n",
       "            </tr>\n",
       "          </table>"
      ],
      "text/plain": [
       "<Model iCdG707 at 0x7f40ef778210>"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG706"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in iCdG706.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        iCdG706.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        iCdG706.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in iCdG706.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        iCdG706.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        iCdG706.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Label tRNA reactions as Biomass\n",
    "iCdG706.reactions.rxn06300_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06432_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06434_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06936_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06435_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06437_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06436_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06438_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06439_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06440_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06441_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06442_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06443_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06444_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06445_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06446_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06447_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06448_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06280_c.annotation['sbo'] = 'SBO:0000629'\n",
    "iCdG706.reactions.rxn06449_c.annotation['sbo'] = 'SBO:0000629'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save model\n",
    "cobra.io.write_sbml_model(iCdG706, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG707.sbml')\n",
    "cobra.io.save_json_model(iCdG706, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/temp/iCdG707.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra\n",
    "iCdG707 = cobra.io.read_sbml_model('/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG707.sbml')\n",
    "for rxn in iCdG707.boundary: rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.flux_analysis import parsimonious\n",
    "\n",
    "# Function to calculate doubling time from objective value\n",
    "def doublingTime(model):\n",
    "    with model as m:\n",
    "        if m.slim_optimize(error_value=0.) < 1e-6:\n",
    "            print('GENRE has no objective flux')\n",
    "        else:\n",
    "            growth = (1. / float(m.slim_optimize())) * 3600.\n",
    "            print(str(round(growth, 2)) + ' minutes doubling time')\n",
    "\n",
    "\n",
    "# Identifies blocked reactions, 1% cutoff for fraction of optimum\n",
    "def blockedReactions(model):\n",
    "    \n",
    "    with model as m:\n",
    "        blocked = cobra.flux_analysis.variability.find_blocked_reactions(m)\n",
    "        nogene_blocked = []\n",
    "        for rxn in blocked:\n",
    "            if m.reactions.get_by_id(rxn).gene_reaction_rule == '':\n",
    "                nogene_blocked.append(rxn)\n",
    "\n",
    "    #print(str(len(blocked)) + ' total reactions are blocked')\n",
    "    fraction = (float(len(blocked)) / float(len(model.reactions))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions are blocked')\n",
    "    \n",
    "    return blocked\n",
    "\n",
    "\n",
    "# Identify potentially gapfilled reactions, checks against pFBA solution\n",
    "def missingGPR(model, exclude=['biomass']):\n",
    "    \n",
    "    noGene = []        \n",
    "    for rxn in model.reactions:\n",
    "        if len(list(rxn.genes)) == 0:\n",
    "            if rxn.id not in exclude:\n",
    "                if len(list(rxn.reactants)) == 0 or len(list(rxn.products)) == 0:\n",
    "                    exclude.append(rxn.id)\n",
    "                    continue\n",
    "                else:\n",
    "                    noGene.append(rxn.id)\n",
    "    \n",
    "    solution = parsimonious.pfba(model)\n",
    "    active_rxns = set([rxn.id for rxn in model.reactions if abs(solution.fluxes[rxn.id]) > 1e-5])\n",
    "    active_rxns = active_rxns.difference(set(exclude))\n",
    "    noGene_active = set(noGene).intersection(active_rxns)\n",
    "\n",
    "    fraction = float(len(model.reactions)) - float(len(exclude))\n",
    "    fraction = (float(len(noGene)) / fraction) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% reactions without GPRs')\n",
    "    \n",
    "    fraction = (float(len(noGene_active)) / float(len(active_rxns))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% of reactions used in pFBA solution have no GPR')\n",
    "    \n",
    "    return noGene_active\n",
    "\n",
    "\n",
    "# Checks which cytosolic metabolites are generated for free (bacteria only)\n",
    "def checkFreeMass(model, cytosol='cytosol'):\n",
    "\n",
    "    free = []\n",
    "    with model as m:\n",
    "    \n",
    "        # Close all exchanges\n",
    "        for rxn in m.boundary: m.reactions.get_by_id(rxn.id).lower_bound = 0.\n",
    "    \n",
    "        # Create demand for each reaction and optimize individually\n",
    "        reset_rxn = m.reactions[0].id\n",
    "        for cpd in m.metabolites: \n",
    "            if cpd.compartment == cytosol:\n",
    "                demand = cobra.Reaction('demand')\n",
    "                demand.bounds = (0., 1000.)\n",
    "                demand.add_metabolites({cpd: -1.0})\n",
    "                m.add_reactions([demand])\n",
    "                m.objective = demand\n",
    "                obj_val = m.slim_optimize()\n",
    "                if obj_val > 1e-8: free.append(cpd.id)\n",
    "                m.objective = reset_rxn\n",
    "                m.remove_reactions([demand])\n",
    "    \n",
    "    fraction = (float(len(free)) / float(len(model.metabolites))) * 100.\n",
    "    fraction = round(fraction, 2)\n",
    "    print(str(fraction) + '% metabolites are generated for free')\n",
    "\n",
    "    return(free)\n",
    "\n",
    "\n",
    "# Check for mass and charge balance in reactions\n",
    "def checkBalance(model, exclude=['biomass']):\n",
    "    \n",
    "    with model as m:\n",
    "\n",
    "        elements = set()\n",
    "        for cpd in m.metabolites:\n",
    "            try:\n",
    "                elements |= set(cpd.elements.keys())\n",
    "            except:\n",
    "                pass\n",
    "        \n",
    "        massImbal = []\n",
    "        failed = 0\n",
    "        if len(elements) == 0:\n",
    "            print('No elemental data associated with metabolites!')\n",
    "            failed = 1\n",
    "        else:\n",
    "            for rxn in m.reactions:\n",
    "                if rxn in m.boundary or rxn.id in exclude: \n",
    "                    continue\n",
    "                elif rxn.annotation['sbo'] == 'SBO:0000629':\n",
    "                    continue\n",
    "\n",
    "                try:\n",
    "                    test = rxn.check_mass_balance()\n",
    "                except ValueError:\n",
    "                    continue\n",
    "\n",
    "                if len(list(test)) > 0:\n",
    "                    if len(set(test.keys()).intersection(elements)) > 0: massImbal.append(rxn.id)\n",
    "                        \n",
    "    if failed != 1:\n",
    "        fraction = (float(len(massImbal)) / float(len(model.reactions))) * 100.\n",
    "        fraction = round(fraction, 2)\n",
    "        print(str(fraction) + '% reactions are mass imbalanced')\n",
    "        \n",
    "    return massImbal\n",
    "\n",
    "\n",
    "def basicCheck(model):\n",
    "    \n",
    "    # Determination\n",
    "    if len(model.reactions) < len(model.metabolites): \n",
    "        print('GENRE is overdetermined')\n",
    "    elif len(model.reactions) > len(model.metabolites):\n",
    "        print('GENRE is underdetermined')\n",
    "    else:\n",
    "        pass\n",
    "    \n",
    "    # Compartments\n",
    "    print('GENRE has ' + str(len(model.compartments.keys())) + ' compartment(s)')\n",
    "    \n",
    "    # Genes\n",
    "    if len(model.genes) == 0: \n",
    "        print('GENRE has no gene data')\n",
    "    else:\n",
    "        print('GENRE has ' + str(len(model.genes)) + ' genes')\n",
    "          \n",
    "    # Growth\n",
    "    doublingTime(model)\n",
    "\n",
    "# Quicker way to read in models\n",
    "import pickle\n",
    "def read_model(fileName, obj='none'):\n",
    "    \n",
    "    fileType = fileName.split('.')[-1]\n",
    "    \n",
    "    if fileType == 'sbml' or fileType == 'xml':\n",
    "        model = cobra.io.read_sbml_model(fileName)\n",
    "    elif fileType == 'json':\n",
    "        model = cobra.io.load_json_model(fileName)\n",
    "    elif fileType == 'yaml':\n",
    "        model = cobra.io.load_yaml_model(fileName)\n",
    "    elif fileType == 'mat':\n",
    "        model = cobra.io.load_matlab_model(fileName)\n",
    "    elif fileType == 'pkl':\n",
    "        model = pickle.load(open(fileName, 'rb'))\n",
    "    else:\n",
    "        raise TypeError('Unrecognized file extension')\n",
    "    \n",
    "    if obj != 'none': model.objective = obj\n",
    "    for rxn in model.boundary: rxn.bounds = (-1000., 1000.)\n",
    "        \n",
    "    return model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GENRE is underdetermined\n",
      "GENRE has 3 compartment(s)\n",
      "GENRE has 707 genes\n",
      "22.3 minutes doubling time\n",
      "37.31% reactions are blocked\n",
      "0.49% reactions are mass imbalanced\n",
      "13.57% reactions without GPRs\n",
      "24.82% of reactions used in pFBA solution have no GPR\n"
     ]
    }
   ],
   "source": [
    "basicCheck(iCdG707)\n",
    "iCdG707_noGPRblocked = blockedReactions(iCdG707)\n",
    "iCdG707_massImbal = checkBalance(iCdG707, exclude=['biomass'])\n",
    "iCdG707_nogene = missingGPR(iCdG707, exclude=['biomass'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Correct mass imbalances\n",
    "\n",
    "cpd00001_c = iCdG707.metabolites.cpd00001_c # H2O\n",
    "cpd29694_c = iCdG707.metabolites.cpd29694_c # L-ala-gly\n",
    "cpd00033_c = iCdG707.metabolites.cpd00033_c # Glycine\n",
    "cpd00035_c = iCdG707.metabolites.cpd00035_c # L-Alanine\n",
    "cpd00067_c = iCdG707.metabolites.cpd00067_c # H+\n",
    "\n",
    "iCdG707.reactions.rxn30238_c.remove_from_model()\n",
    "\n",
    "rxn30238_c = cobra.Reaction('rxn30238_c') \n",
    "rxn30238_c.name = 'Aminopeptidase (ala-gly)'\n",
    "rxn30238_c.gene_reaction_rule = '272563.8.peg.1365 or 272563.8.peg.2607 or 272563.8.peg.2736 or 272563.8.peg.2464'\n",
    "rxn30238_c.lower_bound = 0.\n",
    "rxn30238_c.upper_bound = 1000.\n",
    "rxn30238_c.add_metabolites({\n",
    "    cpd00001_c: -2.0,\n",
    "    cpd29694_c: -1.0,\n",
    "    cpd00035_c: 1.0,\n",
    "    cpd00033_c: 1.0,\n",
    "    cpd00067_c: 5.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn30238_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00982_c = iCdG707.metabolites.cpd00982_c # FADH2\n",
    "cpd00015_c = iCdG707.metabolites.cpd00015_c # FAD\n",
    "cpd00067_p = iCdG707.metabolites.cpd00067_p # H+\n",
    "\n",
    "iCdG707.reactions.rxn12212_c.remove_from_model()\n",
    "\n",
    "rxn12212_c = cobra.Reaction('rxn12212_c') \n",
    "rxn12212_c.name = 'FAD oxidation (electron transport complex)'\n",
    "rxn12212_c.gene_reaction_rule = '272563.120.peg.1189 and 272563.8.peg.1199 and 272563.120.peg.1185 and 272563.120.peg.1186 and 272563.120.peg.1188 and 272563.120.peg.1187'\n",
    "rxn12212_c.lower_bound = -1000.\n",
    "rxn12212_c.upper_bound = 1000.\n",
    "rxn12212_c.add_metabolites({\n",
    "    cpd00982_c: -1.0,\n",
    "    cpd00015_c: 1.0,\n",
    "    cpd00067_p: 3.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn12212_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00106_c = iCdG707.metabolites.cpd00106_c # Fumarate\n",
    "cpd00791_c = iCdG707.metabolites.cpd00791_c # ProtoporphyrinogenIX\n",
    "cpd00036_c = iCdG707.metabolites.cpd00036_c # Succinate\n",
    "cpd01476_c = iCdG707.metabolites.cpd01476_c # Protoporphyrin\n",
    "\n",
    "iCdG707.reactions.rxn09180_c.remove_from_model()\n",
    "\n",
    "rxn09180_c = cobra.Reaction('rxn09180_c') \n",
    "rxn09180_c.name = 'protoporphyrinogen oxidase (anaerobic)'\n",
    "rxn09180_c.lower_bound = -1000.\n",
    "rxn09180_c.upper_bound = 1000.\n",
    "rxn09180_c.add_metabolites({\n",
    "    cpd00067_p: -6.0,\n",
    "    cpd00106_c: -3.0,\n",
    "    cpd00791_c: -1.0,\n",
    "    cpd00036_c: 3.0,\n",
    "    cpd01476_c: 1.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn09180_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00008_c = iCdG707.metabolites.cpd00008_c # ADP\n",
    "cpd00009_c = iCdG707.metabolites.cpd00009_c # Pi\n",
    "cpd00002_c = iCdG707.metabolites.cpd00002_c # ATP\n",
    "\n",
    "iCdG707.reactions.rxn10042_c.remove_from_model()\n",
    "\n",
    "rxn10042_c = cobra.Reaction('rxn10042_c') \n",
    "rxn10042_c.name = 'ATP synthase'\n",
    "rxn10042_c.gene_reaction_rule = '272563.8.peg.3633 and 272563.8.peg.3632 and 272563.8.peg.3636 and 272563.8.peg.3635 and 272563.8.peg.3634 and ( 272563.8.peg.3640 or 272563.8.peg.3641 )'\n",
    "rxn10042_c.lower_bound = 0.\n",
    "rxn10042_c.upper_bound = 1000.\n",
    "rxn10042_c.add_metabolites({\n",
    "    cpd00008_c: -1.0,\n",
    "    cpd00009_c: -1.0,\n",
    "    cpd00067_p: -1.0,\n",
    "    cpd00001_c: 1.0,\n",
    "    cpd00002_c: 1.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn10042_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "cpd00027_e = iCdG707.metabolites.cpd00027_e # Glucose\n",
    "cpd00079_c = iCdG707.metabolites.cpd00079_c # D-glucose-6-phosphate\n",
    "\n",
    "iCdG707.reactions.rxn05147_c.remove_from_model()\n",
    "\n",
    "rxn05147_c = cobra.Reaction('rxn05147_c') \n",
    "rxn05147_c.name = 'D-Glucose-ABC transport'\n",
    "rxn12212_c.gene_reaction_rule = '272563.8.peg.3633 and 272563.8.peg.3632 and 272563.8.peg.3636 and 272563.8.peg.3635 and 272563.8.peg.3634 and ( 272563.8.peg.3640 or 272563.8.peg.3641 )'\n",
    "rxn05147_c.lower_bound = 0.\n",
    "rxn05147_c.upper_bound = 1000.\n",
    "rxn05147_c.add_metabolites({\n",
    "    cpd00002_c: -1.0,\n",
    "    cpd00027_e: -1.0,\n",
    "    cpd00008_c: 1.0,\n",
    "    cpd00067_c: 1.0,\n",
    "    cpd00079_c: 1.0\n",
    "})\n",
    "iCdG707.add_reactions([rxn05147_c])\n",
    "\n",
    "#----------------------------------------------------------------------#\n",
    "\n",
    "iCdG707.metabolites.cpd29698_c.formula = 'C9H18N2O3'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdm = ['EX_cpd00001_e', # water\n",
    "       'EX_cpd00065_e', # L-Tryptophan\n",
    "       'EX_cpd00060_e', # L-Methionine\n",
    "       'EX_cpd00322_e', # L-Isoleucine\n",
    "       'EX_cpd00129_e', # L-Proline\n",
    "       'EX_cpd00156_e', # L-Valine\n",
    "       'EX_cpd00107_e', # L-Leucine\n",
    "       'EX_cpd00084_e', # L-Cysteine \n",
    "       'EX_cpd00149_e', # Cobalt\n",
    "       'EX_cpd00099_e', # Chloride\n",
    "       'EX_cpd10515_e', # Iron\n",
    "       'EX_cpd00030_e', # Manganese\n",
    "       'EX_cpd00254_e', # Magnesium\n",
    "       'EX_cpd00063_e', # Calcium\n",
    "       'EX_cpd00205_e', # Potassium\n",
    "       'EX_cpd00009_e', # Phosphate\n",
    "       'EX_cpd00971_e', # Sodium\n",
    "       'EX_cpd00242_e', # Carbonate\n",
    "       'EX_cpd00104_e', # Biotin\n",
    "       'EX_cpd00644_e', # Pantothenate\n",
    "       'EX_cpd00263_e', # Pyridoxine\n",
    "       'EX_cpd00027_e'] # D-Glucose (Carbohydrate C-source)\n",
    "\n",
    "for rxn in iCdG707.boundary:\n",
    "    if not rxn.id in mdm:\n",
    "        rxn.bounds = (0., 1000.)\n",
    "    else:\n",
    "        rxn.bounds = (-1000., 1000.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add annotations\n",
    "bigg_rxn_success = 0\n",
    "bigg_cpd_success = 0\n",
    "kegg_rxn_success = 0\n",
    "kegg_cpd_success = 0\n",
    "\n",
    "for rxn in iCdG707.reactions:\n",
    "    new_id = rxn.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_rxn_ids[new_id]\n",
    "        iCdG707.reactions.get_by_id(rxn.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_rxn_ids[new_id]\n",
    "        iCdG707.reactions.get_by_id(rxn.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_rxn_success += 1\n",
    "    except:\n",
    "        pass\n",
    "\n",
    "for cpd in iCdG707.metabolites:\n",
    "    new_id = cpd.id.split('_')[0]\n",
    "    try:\n",
    "        bigg_id = BiGG_cpd_ids[new_id]\n",
    "        iCdG707.metabolites.get_by_id(cpd.id).annotation['bigg.metabolite'] = bigg_id\n",
    "        bigg_cpd_success += 1\n",
    "    except:\n",
    "        pass\n",
    "    try:\n",
    "        kegg_id = KEGG_cpd_ids[new_id]\n",
    "        iCdG707.metabolites.get_by_id(cpd.id).annotation['kegg.compound'] = kegg_id\n",
    "        kegg_cpd_success += 1\n",
    "    except:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "70.05819375320453"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG707.slim_optimize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['POTA_CLOD6', '7.6.2.11', 'WP_003437017.1;YP_001087505.1', 'cdf:CD630_10240;pdc:CDIF630_01160', '272563.120.peg.1064']\n",
      "['METN_CLOD6', '7.4.2.11', 'WP_009896423.1;YP_001087990.1', 'cdf:CD630_14890;pdc:CDIF630_01654', '272563.120.peg.1558']\n",
      "['PSTB_CLOD6', '7.3.2.1', 'WP_003432349.1;YP_001089777.1', 'cdf:CD630_32610;pdc:CDIF630_03559', '272563.120.peg.3445']\n",
      "['Q186E6_CLOD6', '7.2.2.6', 'WP_003430007.1;YP_001088093.1', 'cdf:CD630_15920;pdc:CDIF630_01765', '272563.120.peg.1667']\n",
      "['ATPA_CLOD6', '7.1.2.2', 'WP_003421367.1;YP_001089990.1', 'cdf:CD630_34700;pdc:CDIF630_03781', '272563.120.peg.3667']\n",
      "['ATPB_CLOD6', '7.1.2.2', 'WP_003425850.1;YP_001089988.1', 'cdf:CD630_34680;pdc:CDIF630_03779', '272563.120.peg.3665']\n",
      "['VATA_CLOD6', '7.1.2.2', 'WP_003426762.1;YP_001089472.1', 'cdf:CD630_29560;pdc:CDIF630_03239', '272563.120.peg.3122']\n",
      "['Q18B01_CLOD6', '7.-.-.-', 'WP_011861124.1;YP_001087633.1', 'cdf:CD630_11420', '272563.8.peg.1199']\n",
      "['Q18B02_CLOD6', '7.-.-.-', 'WP_003419046.1;YP_001087632.1', 'cdf:CD630_11410;pdc:CDIF630_01288', '272563.120.peg.1189']\n",
      "['Q18AZ8_CLOD6', '7.-.-.-', 'WP_003419045.1;YP_001087631.1', 'cdf:CD630_11400;pdc:CDIF630_01287', '272563.120.peg.1188']\n",
      "['Q18AZ9_CLOD6', '7.-.-.-', 'WP_009902280.1;YP_001087630.1', 'cdf:CD630_11390;pdc:CDIF630_01286', '272563.120.peg.1187']\n",
      "['Q18B00_CLOD6', '7.-.-.-', 'WP_003428487.1;YP_001087629.1', 'cdf:CD630_11380;pdc:CDIF630_01285', '272563.120.peg.1186']\n",
      "['Q18AZ5_CLOD6', '7.-.-.-', 'WP_003428488.1;YP_001087628.1', 'cdf:CD630_11370;pdc:CDIF630_01284', '272563.120.peg.1185']\n",
      "['ECFA1_CLOD6', '7.-.-.-', 'WP_003435660.1;YP_001086568.1', 'cdf:CD630_01000;pdc:CDIF630_00170', '272563.120.peg.110']\n",
      "['Q18BD3_CLOD6', '6.6.1.1', 'WP_009902393.1;YP_001087766.1', 'cdf:CD630_12720;pdc:CDIF630_01425', '272563.120.peg.1330']\n",
      "['DNLJ_CLOD6', '6.5.1.2', 'WP_011861898.1;YP_001089825.2', 'cdf:CD630_33090;pdc:CDIF630_03610', '272563.120.peg.3494']\n",
      "['Q18CA4_CLOD6', '6.4.1.1', 'WP_003425597.1;YP_001086484.1', 'cdf:CD630_00210;pdc:CDIF630_00083', '272563.120.peg.23']\n",
      "['Q181J3_CLOD6', '6.3.5.5', 'WP_011862004.1;YP_001090115.1', 'cdf:CD630_35910;pdc:CDIF630_03913', '272563.120.peg.3796']\n",
      "['Q181J2_CLOD6', '6.3.5.5', 'WP_011862003.1;YP_001090114.1', 'cdf:CD630_35900;pdc:CDIF630_03912', '272563.120.peg.3795']\n",
      "['Q181J1_CLOD6', '6.3.5.5', 'WP_003429883.1;YP_001090113.1', 'cdf:CD630_35890;pdc:CDIF630_03911', '272563.120.peg.3794']\n",
      "['Q181J0_CLOD6', '6.3.5.5', 'WP_009892173.1;YP_001090112.1', 'cdf:CD630_35880;pdc:CDIF630_03910', '272563.120.peg.3793']\n",
      "['Q188I0_CLOD6', '6.3.5.4', 'WP_011860775.1;YP_001086857.1', 'cdf:CD630_03910;pdc:CDIF630_00519', '272563.120.peg.411']\n",
      "['Q186I9_CLOD6', '6.3.5.4', 'WP_003436426.1;YP_001088133.1', 'cdf:CD630_16320;pdc:CDIF630_01809', '272563.120.peg.1710']\n",
      "['Q18CW3_CLOD6', '6.3.5.3', 'WP_011860677.1;YP_001086693.1', 'cdf:CD630_02250;pdc:CDIF630_00347', '272563.120.peg.241']\n",
      "['GUAA_CLOD6', '6.3.5.2', 'WP_003425341.1;YP_001086666.1', 'cdf:CD630_01980;pdc:CDIF630_00320', '272563.120.peg.213']\n",
      "['Q180T2_CLOD6', '6.3.5.11', 'WP_011861968.1;YP_001089954.2', 'cdf:CD630_34340;pdc:CDIF630_03742', '272563.120.peg.3628']\n",
      "['ASSY_CLOD6', '6.3.4.5', 'WP_003439524.1;YP_001088291.1', 'cdf:CD630_17850;pdc:CDIF630_01984', '272563.120.peg.1878']\n",
      "['PURA_CLOD6', '6.3.4.4', 'WP_003434259.1;YP_001090179.1', 'cdf:CD630_36550;pdc:CDIF630_03983', '272563.120.peg.3865']\n",
      "['FTHS_CLOD6', '6.3.4.3', 'WP_009888533.1;YP_001087195.1', 'cdf:CD630_07180;pdc:CDIF630_00834', '272563.120.peg.739']\n",
      "['Q181E1_CLOD6', '6.3.4.21', 'WP_003427657.1;YP_001090064.1', 'cdf:CD630_35430;pdc:CDIF630_03858', '272563.120.peg.3744']\n",
      "['TILS_CLOD6', '6.3.4.19', 'WP_003434952.1;YP_001090083.1', 'cdf:CD630_35600;pdc:CDIF630_03880', '272563.120.peg.3763']\n",
      "['Q181F8_CLOD6', '6.3.4.15', 'WP_009903819.1;YP_001090081.1', 'cdf:CD630_35580;pdc:CDIF630_03878', '272563.120.peg.3761']\n",
      "['Q187P8_CLOD6', '6.3.4.14;_6.4.1.2', 'WP_003435334.1;YP_001088444.1', 'cdf:CD630_19380;pdc:CDIF630_02142', '272563.120.peg.2034']\n",
      "['Q18CW4_CLOD6', '6.3.4.13', 'WP_003436062.1;YP_001086692.1', 'cdf:CD630_02240;pdc:CDIF630_00346', '272563.120.peg.240']\n",
      "['TMCAL_CLOD6', '6.3.4.-', 'WP_011861133.1;YP_001087666.1', 'cdf:CD630_11740;pdc:CDIF630_01322', '272563.120.peg.1224']\n",
      "['Q180V5_CLOD6', '6.3.3.2', 'WP_003437817.1;YP_001089976.1', 'cdf:CD630_34560;pdc:CDIF630_03767', '272563.120.peg.3653']\n",
      "['Q18CW2_CLOD6', '6.3.3.1', 'WP_009895312.1;YP_001086689.1', 'cdf:CD630_02210;pdc:CDIF630_00343', '272563.120.peg.237']\n",
      "['MURD_CLOD6', '6.3.2.9', 'WP_004454967.1;YP_001089164.1', 'cdf:CD630_26530;pdc:CDIF630_02907', '272563.120.peg.2796']\n",
      "['MURC_CLOD6', '6.3.2.8', 'WP_003425936.1;YP_001090039.1', 'cdf:CD630_35180;pdc:CDIF630_03832', '272563.120.peg.3718']\n",
      "['Q188T0_CLOD6', '6.3.2.6', 'WP_003426355.1;YP_001086963.1', 'cdf:CD630_04890;pdc:CDIF630_00618', '272563.120.peg.514']\n",
      "['Q18CV5_CLOD6', '6.3.2.6', 'WP_003420966.1;YP_001086687.1', 'cdf:CD630_02190;pdc:CDIF630_00341', '272563.120.peg.235']\n",
      "['Q186I2_CLOD6', '6.3.2.4', 'WP_011861276.1;YP_001088127.1', 'cdf:CD630_16260;pdc:CDIF630_01803', '272563.120.peg.1704']\n",
      "['Q18BS9_CLOD6', '6.3.2.4', 'WP_003429344.1;YP_001087908.1', 'cdf:CD630_14080;pdc:CDIF630_01569', '272563.120.peg.1475']\n",
      "['Q17ZZ4_CLOD6', '6.3.2.17', 'WP_009903674.1;YP_001089769.1', 'cdf:CD630_32530;pdc:CDIF630_03550', '272563.120.peg.3437']\n",
      "['MURE_CLOD6', '6.3.2.13', 'WP_011861629.1;YP_001089175.1', 'cdf:CD630_26640;pdc:CDIF630_02918', '272563.120.peg.2807']\n",
      "['Q182Y9_CLOD6', '6.3.2.10', 'WP_011861626.1;YP_001089166.1', 'cdf:CD630_26550;pdc:CDIF630_02909', '272563.120.peg.2798']\n",
      "['PANC_CLOD6', '6.3.2.1', 'WP_009902543.1;YP_001088013.1', 'cdf:CD630_15120;pdc:CDIF630_01678', '272563.120.peg.1583']\n",
      "['Q189Z0_CLOD6', '6.3.1.5', 'WP_003437535.1;YP_001087271.1', 'cdf:CD630_07940;pdc:CDIF630_00913', '272563.120.peg.817']\n",
      "['Q186L0_CLOD6', '6.3.1.20', 'WP_011861285.1;YP_001088155.1', 'cdf:CD630_16540;pdc:CDIF630_01832', '272563.120.peg.1732']\n",
      "['Q18BL0_CLOD6', '6.3.1.2', 'WP_003428202.1;YP_001087839.1', 'cdf:CD630_13430;pdc:CDIF630_01499', '272563.120.peg.1404']\n",
      "['Q183C9_CLOD6', '6.3.1.1', 'WP_003427105.1;YP_001089206.2', 'cdf:CD630_26950;pdc:CDIF630_02950', '272563.120.peg.2839']\n",
      "['Q183T6_CLOD6', '6.2.1.54', 'WP_011861716.1;YP_001089366.1', 'cdf:CD630_28530;pdc:CDIF630_03120', '272563.120.peg.3005']\n",
      "['Q180A1_CLOD6', '6.1.1.9', 'WP_011861890.1;YP_001089772.1', 'cdf:CD630_32560;pdc:CDIF630_03553', '272563.120.peg.3440']\n",
      "['SYA_CLOD6', '6.1.1.7', 'WP_009889064.1;YP_001087776.1', 'cdf:CD630_12820;pdc:CDIF630_01436', '272563.120.peg.1340']\n",
      "['Q181F2_CLOD6', '6.1.1.6', 'WP_011861998.1;YP_001090075.1', 'cdf:CD630_35520;pdc:CDIF630_03872', '272563.120.peg.3755']\n",
      "['Q182V2_CLOD6', '6.1.1.5', 'WP_004454939.1;YP_001089129.1', 'cdf:CD630_26180', '272563.8.peg.2741']\n",
      "['SYL_CLOD6', '6.1.1.4', 'WP_003416419.1;YP_001089035.1', 'cdf:CD630_25210;pdc:CDIF630_02772', '272563.120.peg.2661']\n",
      "['SYT_CLOD6', '6.1.1.3', 'WP_009888359.1;YP_001087049.1', 'cdf:CD630_05740;pdc:CDIF630_00687', '272563.120.peg.584']\n",
      "['Q185B5_CLOD6', '6.1.1.22', 'WP_003430444.1;YP_001088759.1', 'cdf:CD630_22450;pdc:CDIF630_02478', '272563.120.peg.2372']\n",
      "['SYH_CLOD6', '6.1.1.21', 'WP_003422928.1;YP_001089252.1', 'cdf:CD630_27400;pdc:CDIF630_03003', '272563.120.peg.2887']\n",
      "['Q18C68_CLOD6', '6.1.1.21', 'WP_011861248.1;YP_001088048.1', 'cdf:CD630_15470;pdc:CDIF630_01716', '272563.120.peg.1621']\n",
      "['Q189P6_CLOD6', '6.1.1.20', 'WP_011860925.1;YP_001087176.1', 'cdf:CD630_07000;pdc:CDIF630_00815', '272563.120.peg.720']\n",
      "['Q189P4_CLOD6', '6.1.1.20', 'WP_003436795.1;YP_001087175.2', 'cdf:CD630_06990;pdc:CDIF630_00814', '272563.120.peg.719']\n",
      "['Q182U4_CLOD6', '6.1.1.2', 'WP_004454934.1;YP_001089121.1', 'cdf:CD630_26100;pdc:CDIF630_02862', '272563.120.peg.2753']\n",
      "['SYR_CLOD6', '6.1.1.19', 'WP_003436885.1;YP_001087187.1', 'cdf:CD630_07110;pdc:CDIF630_00826', '272563.120.peg.731']\n",
      "['SYQ_CLOD6', '6.1.1.18', 'WP_003428989.1;YP_001088573.1', 'cdf:CD630_20590;pdc:CDIF630_02282', '272563.120.peg.2173']\n",
      "['SYE_CLOD6', '6.1.1.17', 'WP_003435581.1;YP_001086514.1', 'cdf:CD630_00510;pdc:CDIF630_00114', '272563.120.peg.55']\n",
      "['SYC_CLOD6', '6.1.1.16', 'WP_009895184.1;YP_001086515.2', 'cdf:CD630_00520;pdc:CDIF630_00116', '272563.120.peg.57']\n",
      "['SYP2_CLOD6', '6.1.1.15', 'WP_009895182.1;YP_001086513.1', 'cdf:CD630_00500;pdc:CDIF630_00113', '272563.120.peg.54']\n",
      "['SYP1_CLOD6', '6.1.1.15', 'WP_009895180.1;YP_001086512.1', 'cdf:CD630_00490;pdc:CDIF630_00112', '272563.120.peg.53']\n",
      "['SYGA_CLOD6', '6.1.1.14', 'WP_003416656.1;YP_001088946.1', 'cdf:CD630_24330;pdc:CDIF630_02676', '272563.120.peg.2568']\n",
      "['SYGB_CLOD6', '6.1.1.14', 'WP_011861551.1;YP_001088945.1', 'cdf:CD630_24320;pdc:CDIF630_02675', '272563.120.peg.2567']\n",
      "['Q183H4_CLOD6', '6.1.1.12', 'WP_004454140.1;YP_001089251.1', 'cdf:CD630_27390;pdc:CDIF630_03002', '272563.120.peg.2886']\n",
      "['Q183S4_CLOD6', '6.1.1.11', 'WP_011861710.1;YP_001089353.1', 'cdf:CD630_28400;pdc:CDIF630_03107', '272563.120.peg.2992']\n",
      "['Q18C97_CLOD6', '6.1.1.11', 'WP_003435562.1;YP_001086477.1', 'cdf:CD630_00140;pdc:CDIF630_00024', '272563.120.peg.14']\n",
      "['Q181D9_CLOD6', '6.1.1.10', 'WP_011861995.1;YP_001090061.1', 'cdf:CD630_35400;pdc:CDIF630_03854', '272563.120.peg.3741']\n",
      "['Q18C40_CLOD6', '6.1.1.1', 'WP_009889364.1;YP_001088022.1', 'cdf:CD630_15210;pdc:CDIF630_01689', '272563.120.peg.1594']\n",
      "['Q188K6_CLOD6', '5.99.1.2', 'WP_011860791.1;YP_001086889.1', 'cdf:CD630_04210;pdc:CDIF630_00549', '272563.120.peg.441']\n",
      "['Q181Z2_CLOD6', '5.99.1.2', 'WP_011861544.1;YP_001088916.1', 'cdf:CD630_24040;pdc:CDIF630_02646', '272563.120.peg.2538']\n",
      "['Q18AW1_CLOD6', '5.99.1.2', 'WP_011861108.1;YP_001087593.1', 'cdf:CD630_11060;pdc:CDIF630_01251', '272563.120.peg.1152']\n",
      "['Q18C90_CLOD6', '5.6.2.2', 'WP_003429305.1;YP_001086469.1', 'cdf:CD630_00060;pdc:CDIF630_00006', '272563.120.peg.6']\n",
      "['Q18C89_CLOD6', '5.6.2.2', 'WP_003434229.1;YP_001086468.1', 'cdf:CD630_00050;pdc:CDIF630_00005', '272563.120.peg.5']\n",
      "['Q185E5_CLOD6', '5.6.2.1', 'WP_003440272.1;YP_001088787.1', 'cdf:CD630_22750;pdc:CDIF630_02509', '272563.120.peg.2402']\n",
      "['Q18BD6_CLOD6', '5.6.2.1', 'WP_009892959.1;YP_001087768.1', 'cdf:CD630_12740;pdc:CDIF630_01427', '272563.120.peg.1332']\n",
      "['TRUB_CLOD6', '5.4.99.25', 'WP_003428232.1;YP_001087809.1', 'cdf:CD630_13140;pdc:CDIF630_01470', '272563.120.peg.1374']\n",
      "['Q18CV6_CLOD6', '5.4.99.18', 'WP_003425290.1;YP_001086686.1', 'cdf:CD630_02180;pdc:CDIF630_00340', '272563.120.peg.234']\n",
      "['Q186Z4_CLOD6', '5.4.99.12', 'WP_003430279.1;YP_001088290.1', 'cdf:CD630_17840;pdc:CDIF630_01982', '272563.120.peg.1877']\n",
      "['Q18CJ2_CLOD6', '5.4.99.12', 'WP_003435654.1;YP_001086571.1', 'cdf:CD630_01030;pdc:CDIF630_00173', '272563.120.peg.113']\n",
      "['Q188I9_CLOD6', '5.4.99.-', 'WP_011860778.1;YP_001086871.1', 'cdf:CD630_04050;pdc:CDIF630_00533', '272563.120.peg.425']\n",
      "['Q181I2_CLOD6', '5.4.99.-', 'WP_009892161.1;YP_001090106.1', 'cdf:CD630_35820;pdc:CDIF630_03903', '272563.120.peg.3786']\n",
      "['Q182T6_CLOD6', '5.4.99.-', 'WP_011861614.1;YP_001089113.1', 'cdf:CD630_25960;pdc:CDIF630_02850', '272563.120.peg.2738']\n",
      "['Q188B7_CLOD6', '5.4.99.-', 'WP_003435147.1;YP_001088564.1', 'cdf:CD630_20500;pdc:CDIF630_02272', '272563.120.peg.2163']\n",
      "['Q18AQ9_CLOD6', '5.4.99.-', 'WP_011861083.1;YP_001087541.1', 'cdf:CD630_10600;pdc:CDIF630_01200', '272563.120.peg.1100']\n",
      "['EAM_CLOD6', '5.4.3.9', 'WP_003430434.1;YP_001088767.1', 'cdf:CD630_22520;pdc:CDIF630_02487', '272563.120.peg.2380']\n",
      "['Q188N7_CLOD6', '5.4.3.5', 'WP_009895465.1;YP_001086917.1', 'cdf:CD630_04460;pdc:CDIF630_00574', '272563.120.peg.471']\n",
      "['Q188N8_CLOD6', '5.4.3.5', 'WP_003434665.1;YP_001086916.1', 'cdf:CD630_04450;pdc:CDIF630_00573', '272563.120.peg.470']\n",
      "['DEOB_CLOD6', '5.4.2.7', 'WP_011861145.1;YP_001087716.1', 'cdf:CD630_12230;pdc:CDIF630_01372', '272563.120.peg.1274']\n",
      "['Q184W6_CLOD6', '5.4.2.6', 'WP_011861837.1;YP_001089643.1', 'cdf:CD630_31280;pdc:CDIF630_03413', '272563.120.peg.3296']\n",
      "['Q185X7_CLOD6', '5.4.2.6', 'WP_003433900.1;YP_001088702.1', 'cdf:CD630_21890;pdc:CDIF630_02421', '272563.120.peg.2311']\n",
      "['Q183L5_CLOD6', '5.4.2.2', 'WP_004454107.1;YP_001089293.1', 'cdf:CD630_27800;pdc:CDIF630_03043', '272563.120.peg.2928']\n",
      "['Q186D0_CLOD6', '5.4.2.2', 'WP_009896733.1;YP_001088077.1', 'cdf:CD630_15770;pdc:CDIF630_01746', '272563.120.peg.1650']\n",
      "['GPMI_CLOD6', '5.4.2.12', 'WP_011861866.1;YP_001089688.1', 'cdf:CD630_31710', '272563.8.peg.3322']\n",
      "['GLMM_CLOD6', '5.4.2.10', 'WP_011860638.1;YP_001086588.2', 'cdf:CD630_01190;pdc:CDIF630_00234', '272563.120.peg.131']\n",
      "['Q185J2_CLOD6', '5.4.2.-', 'WP_004454621.1;YP_001088833.1', 'cdf:CD630_23180;pdc:CDIF630_02556', '272563.120.peg.2449']\n",
      "['Q180S7_CLOD6', '5.4.1.2', 'WP_003437747.1;YP_001089950.1', 'cdf:CD630_34300;pdc:CDIF630_03738', '272563.120.peg.3624']\n",
      "['G6PI_CLOD6', '5.3.1.9', 'WP_003435635.1;YP_001089801.1', 'cdf:CD630_32850;pdc:CDIF630_03585', '272563.120.peg.3470']\n",
      "['Q182H9_CLOD6', '5.3.1.8', 'WP_004454782.1;YP_001089005.1', 'cdf:CD630_24910;pdc:CDIF630_02738', '272563.120.peg.2630']\n",
      "['Q180X7_CLOD6', '5.3.1.6', 'WP_003437841.1;YP_001090000.1', 'cdf:CD630_34800;pdc:CDIF630_03791', '272563.120.peg.3678']\n",
      "['Q185J3_CLOD6', '5.3.1.6', 'WP_003430700.1;YP_001088835.1', 'cdf:CD630_23200;pdc:CDIF630_02558', '272563.120.peg.2451']\n",
      "['Q184Q2_CLOD6', '5.3.1.5', 'WP_009898127.1;YP_001089578.1', 'cdf:CD630_30640;pdc:CDIF630_03347', '272563.120.peg.3230']\n",
      "['HIS4_CLOD6', '5.3.1.16', 'WP_003436675.1;YP_001088053.1', 'cdf:CD630_15520;pdc:CDIF630_01721', '272563.120.peg.1626']\n",
      "['Q181T6_CLOD6', '5.3.1.1', 'WP_011861867.1;YP_001089689.1', 'cdf:CD630_31720;pdc:CDIF630_03464', '272563.120.peg.3347']\n",
      "['Q180Z8_CLOD6', '5.2.1.8', 'WP_009892066.1;YP_001090021.1', 'cdf:CD630_35000;pdc:CDIF630_03813', '272563.120.peg.3700']\n",
      "['Q18D70_CLOD6', '5.2.1.8', 'WP_003434191.1;YP_001086800.1', 'cdf:CD630_03310;pdc:CDIF630_00459', '272563.120.peg.350']\n",
      "['TIG_CLOD6', '5.2.1.8', 'WP_003417090.1;YP_001089822.1', 'cdf:CD630_33060;pdc:CDIF630_03607', '272563.120.peg.3491']\n",
      "['Q185D5_CLOD6', '5.2.1.8', 'WP_009897434.1;YP_001088777.1', 'cdf:CD630_22630;pdc:CDIF630_02497', '272563.120.peg.2390']\n",
      "['Q18C77_CLOD6', '5.2.1.8', 'WP_009902572.1;YP_001088058.1', 'cdf:CD630_15570;pdc:CDIF630_01726', '272563.120.peg.1631']\n",
      "['Q18BM5_CLOD6', '5.2.1.8', 'WP_009889131.1;YP_001087853.1', 'cdf:CD630_13570;pdc:CDIF630_01513', '272563.120.peg.1418']\n",
      "['NANE_CLOD6', '5.1.3.9', 'WP_003430452.1;YP_001088755.1', 'cdf:CD630_22410;pdc:CDIF630_02474', '272563.120.peg.2368']\n",
      "['Q185F0_CLOD6', '5.1.3.4', 'WP_009897453.1;YP_001088790.1', 'cdf:CD630_22780;pdc:CDIF630_02513', '272563.120.peg.2405']\n",
      "['Q183E8_CLOD6', '5.1.3.2', 'WP_011861656.1;YP_001089225.1', 'cdf:CD630_27140;pdc:CDIF630_02971', '272563.120.peg.2858']\n",
      "['Q18AN3_CLOD6', '5.1.3.14', 'WP_003437003.1;YP_001087514.1', 'cdf:CD630_10330;pdc:CDIF630_01169', '272563.120.peg.1073']\n",
      "['Q184U7_CLOD6', '5.1.3.1', 'WP_009891451.1;YP_001089623.1', 'cdf:CD630_31090;pdc:CDIF630_03393', '272563.120.peg.3275']\n",
      "['Q182R3_CLOD6', '5.1.3.1', 'WP_003431138.1;YP_001089091.1', 'cdf:CD630_25750;pdc:CDIF630_02828', '272563.120.peg.2716']\n",
      "['Q185J4_CLOD6', '5.1.3.1', 'WP_003430699.1;YP_001088834.1', 'cdf:CD630_23190;pdc:CDIF630_02557', '272563.120.peg.2450']\n",
      "['Q182T1_CLOD6', '5.1.1.7', 'WP_009897787.1;YP_001089107.1', 'cdf:CD630_25900;pdc:CDIF630_02844', '272563.120.peg.2732']\n",
      "['PRAC_CLOD6', '5.1.1.4', 'WP_003422090.1;YP_001089754.1', 'cdf:CD630_32370;pdc:CDIF630_03533', '272563.120.peg.3419']\n",
      "['Q181G1_CLOD6', '5.1.1.3', 'WP_003425992.1;YP_001090085.1', 'cdf:CD630_35620;pdc:CDIF630_03882', '272563.120.peg.3765']\n",
      "['Q187Z6_CLOD6', '5.1.1.13', 'WP_011861432.1;YP_001088542.1', 'cdf:CD630_20280;pdc:CDIF630_02249', '272563.120.peg.2139']\n",
      "['Q188N5_CLOD6', '5.1.1.12', 'WP_011860813.1;YP_001086919.1', 'cdf:CD630_04480;pdc:CDIF630_00576', '272563.120.peg.473']\n",
      "['Q180W0_CLOD6', '5.1.1.1', 'WP_003437830.1;YP_001089983.1', 'cdf:CD630_34630;pdc:CDIF630_03774', '272563.120.peg.3660']\n",
      "['Q186I3_CLOD6', '5.1.1.1', 'WP_009893196.1;YP_001088129.1', 'cdf:CD630_16280;pdc:CDIF630_01805', '272563.120.peg.1706']\n",
      "['Q18BQ6_CLOD6', '5.1.1.-', 'WP_003438474.1;YP_001087888.1', 'cdf:CD630_13890;pdc:CDIF630_01549', '272563.120.peg.1455']\n",
      "['Q180U7_CLOD6', '5.-.-.-', 'WP_009898687.1;YP_001089969.2', 'cdf:CD630_34490;pdc:CDIF630_03760', '272563.120.peg.3645']\n",
      "['Q180S2_CLOD6', '4.99.1.3', 'WP_003437727.1;YP_001089942.1', 'cdf:CD630_34220;pdc:CDIF630_03730', '272563.120.peg.3616']\n",
      "['Q186S3_CLOD6', '4.6.1.17', 'WP_004454438.1;YP_001088218.1', 'cdf:CD630_17140;pdc:CDIF630_01902', '272563.120.peg.1798']\n",
      "['ISPF_CLOD6', '4.6.1.12', 'WP_003425513.1;YP_001086511.1', 'cdf:CD630_00480;pdc:CDIF630_00110', '272563.120.peg.52']\n",
      "['Q184L6_CLOD6', '4.4.1.8', 'WP_011861787.1;YP_001089544.1', 'cdf:CD630_30290;pdc:CDIF630_03313', '272563.120.peg.3195']\n",
      "['Q18AZ1_CLOD6', '4.4.1.5', 'WP_003419037.1;YP_001087625.1', 'cdf:CD630_11340;pdc:CDIF630_01281', '272563.120.peg.1182']\n",
      "['LUXS_CLOD6', '4.4.1.21', 'WP_003422312.1;YP_001090122.1', 'cdf:CD630_35980;pdc:CDIF630_03920', '272563.120.peg.3804']\n",
      "['Q181I0_CLOD6', '4.4.1.11', 'WP_011862002.1;YP_001090101.1', 'cdf:CD630_35770;pdc:CDIF630_03898', '272563.120.peg.3781']\n",
      "['Q181M9_CLOD6', '4.3.3.7', 'WP_003435760.1;YP_001090150.1', 'cdf:CD630_36270', '272563.8.peg.3800']\n",
      "['Q17ZX0_CLOD6', '4.3.3.7', 'WP_003422063.1;YP_001089742.1', 'cdf:CD630_32250;pdc:CDIF630_03521', '272563.120.peg.3406']\n",
      "['Q17ZW8_CLOD6', '4.3.3.7', 'WP_003438799.1;YP_001089740.1', 'cdf:CD630_32230;pdc:CDIF630_03518', '272563.120.peg.3404']\n",
      "['Q18BJ9_CLOD6', '4.3.2.2', 'WP_003438353.1;YP_001087830.1', 'cdf:CD630_13340;pdc:CDIF630_01490', '272563.120.peg.1394']\n",
      "['Q18C72_CLOD6', '4.3.2.10;_3.5.1.2', 'WP_003436678.1;YP_001088052.1', 'cdf:CD630_15510;pdc:CDIF630_01720', '272563.120.peg.1625']\n",
      "['HIS6_CLOD6', '4.3.2.10', 'WP_003436672.1;YP_001088054.1', 'cdf:CD630_15530;pdc:CDIF630_01722', '272563.120.peg.1627']\n",
      "['ARLY_CLOD6', '4.3.2.1', 'WP_009890714.1;YP_001089013.2', 'cdf:CD630_25000;pdc:CDIF630_02747', '272563.120.peg.2639']\n",
      "['EUTC_CLOD6', '4.3.1.7', 'WP_009897052.1;YP_001088421.1', 'cdf:CD630_19140;pdc:CDIF630_02118', '272563.120.peg.2010']\n",
      "['Q187M3_CLOD6', '4.3.1.7', 'WP_011861389.1;YP_001088420.1', 'cdf:CD630_19130;pdc:CDIF630_02117', '272563.120.peg.2009']\n",
      "['Q182J9_CLOD6', '4.3.1.19', 'WP_009890721.1;YP_001089027.1', 'cdf:CD630_25140;pdc:CDIF630_02763', '272563.120.peg.2653']\n",
      "['Q17ZW7_CLOD6', '4.3.1.17', 'WP_003422058.1;YP_001089739.1', 'cdf:CD630_32220;pdc:CDIF630_03517', '272563.120.peg.3403']\n",
      "['Q181U9_CLOD6', '4.3.1.15', 'WP_011861869.1;YP_001089701.1', 'cdf:CD630_31840;pdc:CDIF630_03476', '272563.120.peg.3359']\n",
      "['Q188F2_CLOD6', '4.3.1.15', 'WP_009890217.1;YP_001088599.1', 'cdf:CD630_20850;pdc:CDIF630_02308', '272563.120.peg.2199']\n",
      "['Q188Y6_CLOD6', '4.3.1.12', 'WP_003439110.1;YP_001087020.1', 'cdf:CD630_05440;pdc:CDIF630_00657', '272563.120.peg.553']\n",
      "['Q189B2_CLOD6', '4.2.99.18', 'WP_011860857.1;YP_001087041.1', 'cdf:CD630_05650', '272563.8.peg.592']\n",
      "['Q187E5_CLOD6', '4.2.3.5', 'WP_011861346.1;YP_001088341.1', 'cdf:CD630_18350;pdc:CDIF630_02037', '272563.120.peg.1930']\n",
      "['Q187E4_CLOD6', '4.2.3.4', 'WP_009893310.1;YP_001088339.1', 'cdf:CD630_18330;pdc:CDIF630_02035', '272563.120.peg.1928']\n",
      "['MGSA_CLOD6', '4.2.3.3', 'WP_003438060.1;YP_001087645.1', 'cdf:CD630_11530;pdc:CDIF630_01300', '272563.120.peg.1202']\n",
      "['Q185Q9_CLOD6', '4.2.3.1', 'WP_003434031.1;YP_001088632.1', 'cdf:CD630_21180;pdc:CDIF630_02347', '272563.120.peg.2236']\n",
      "['Q18B90_CLOD6', '4.2.2.-', 'WP_003438164.1;YP_001087719.1', 'cdf:CD630_12260;pdc:CDIF630_01375', '272563.120.peg.1277']\n",
      "['Q187Y2_CLOD6', '4.2.1.9', 'WP_011861428.1;YP_001088528.1', 'cdf:CD630_20140;pdc:CDIF630_02233', '272563.120.peg.2123']\n",
      "['Q184J4_CLOD6', '4.2.1.7', 'WP_003439842.1;YP_001089520.1', 'cdf:CD630_30020;pdc:CDIF630_03287', '272563.120.peg.3169']\n",
      "['Q183W0_CLOD6', '4.2.1.7', 'WP_009898047.1;YP_001089385.1', 'cdf:CD630_28720;pdc:CDIF630_03140', '272563.120.peg.3024']\n",
      "['FABZ_CLOD6', '4.2.1.59', 'WP_003420731.1;YP_001086597.1', 'cdf:CD630_01280;pdc:CDIF630_00244', '272563.120.peg.140']\n",
      "['Q189Z6_CLOD6', '4.2.1.55', 'WP_011860959.1;YP_001087277.1', 'cdf:CD630_08000', '272563.8.peg.837']\n",
      "['Q18AQ8_CLOD6', '4.2.1.55', 'WP_011861082.1;YP_001087538.1', 'cdf:CD630_10570;pdc:CDIF630_01197', '272563.120.peg.1097']\n",
      "['Q184J1_CLOD6', '4.2.1.52', 'WP_003439840.1;YP_001089518.1', 'cdf:CD630_30000;pdc:CDIF630_03285', '272563.120.peg.3167']\n",
      "['Q187E6_CLOD6', '4.2.1.51;_5.4.99.5', 'WP_011861347.1;YP_001088342.1', 'cdf:CD630_18360;pdc:CDIF630_02038', '272563.120.peg.1931']\n",
      "['LEUD_CLOD6', '4.2.1.33', 'WP_003437092.1;YP_001087472.1', 'cdf:CD630_09910;pdc:CDIF630_01127', '272563.120.peg.1030']\n",
      "['LEUC_CLOD6', '4.2.1.33', 'WP_009895941.1;YP_001087471.1', 'cdf:CD630_09900;pdc:CDIF630_01126', '272563.120.peg.1029']\n",
      "['Q18A28_CLOD6', '4.2.1.3', 'WP_004454235.1;YP_001087310.1', 'cdf:CD630_08330;pdc:CDIF630_00952', '272563.120.peg.856']\n",
      "['Q180R7_CLOD6', '4.2.1.24', 'WP_003437722.1;YP_001089939.2', 'cdf:CD630_34190;pdc:CDIF630_03727', '272563.120.peg.3613']\n",
      "['Q18AK1_CLOD6', '4.2.1.2', 'WP_003437060.1;YP_001087485.1', 'cdf:CD630_10040;pdc:CDIF630_01140', '272563.120.peg.1043']\n",
      "['Q18AK2_CLOD6', '4.2.1.2', 'WP_003437062.1;YP_001087484.1', 'cdf:CD630_10030;pdc:CDIF630_01139', '272563.120.peg.1042']\n",
      "['HIS7_CLOD6', '4.2.1.19', 'WP_003436680.1;YP_001088051.1', 'cdf:CD630_15500;pdc:CDIF630_01719', '272563.120.peg.1624']\n",
      "['Q18A74_CLOD6', '4.2.1.136;_5.1.99.6', 'WP_009892822.1;YP_001087353.1', 'cdf:CD630_08790;pdc:CDIF630_00999', '272563.120.peg.903']\n",
      "['MURQ_CLOD6', '4.2.1.126', 'WP_003431390.1;YP_001089561.1', 'cdf:CD630_30460;pdc:CDIF630_03330', '272563.120.peg.3212']\n",
      "['ENO_CLOD6', '4.2.1.11', 'WP_009898301.1;YP_001089687.1', 'cdf:CD630_31700', '272563.8.peg.3321']\n",
      "['AROD_CLOD6', '4.2.1.10', 'WP_009897376.1;YP_001088731.2', 'cdf:CD630_22170;pdc:CDIF630_02450', '272563.120.peg.2341']\n",
      "['Q186A4_CLOD6', '4.2.1.1', 'WP_003423380.1;YP_001088727.2', 'cdf:CD630_22130;pdc:CDIF630_02446', '272563.120.peg.2337']\n",
      "['Q185L4_CLOD6', '4.2.1.-;_5.3.3.3', 'WP_003430738.1;YP_001088856.1', 'cdf:CD630_23410;pdc:CDIF630_02580', '272563.120.peg.2472']\n",
      "['Q186V7_CLOD6', '4.2.1.-', 'WP_004454390.1;YP_001088253.1', 'cdf:CD630_17490;pdc:CDIF630_01943', '272563.120.peg.1838']\n",
      "['Q18AK3_CLOD6', '4.2.-.-', 'WP_003437064.1;YP_001087483.1', 'cdf:CD630_10020;pdc:CDIF630_01138', '272563.120.peg.1041']\n",
      "['THIC_CLOD6', '4.1.99.17', 'WP_009889633.1;YP_001088205.1', 'cdf:CD630_17020', '272563.8.peg.1780']\n",
      "['Q186Q4_CLOD6', '4.1.99.12;_3.5.4.25', 'WP_009889591.1;YP_001088201.1', 'cdf:CD630_16980;pdc:CDIF630_01883', '272563.120.peg.1781']\n",
      "['Q18D17_CLOD6', '4.1.99.-', 'WP_003435960.1;YP_001086744.1', 'cdf:CD630_02750;pdc:CDIF630_00399', '272563.120.peg.293']\n",
      "['Q18BW9_CLOD6', '4.1.3.38', 'WP_003438634.1;YP_001087948.1', 'cdf:CD630_14470;pdc:CDIF630_01611', '272563.120.peg.1516']\n",
      "['Q185B3_CLOD6', '4.1.3.3', 'WP_009890428.1;YP_001088754.1', 'cdf:CD630_22400;pdc:CDIF630_02473', '272563.120.peg.2367']\n",
      "['Q181M7_CLOD6', '4.1.3.16', 'WP_011862015.1;YP_001090149.1', 'cdf:CD630_36260;pdc:CDIF630_03949', '272563.120.peg.3833']\n",
      "['Q182T0_CLOD6', '4.1.2.5', 'WP_009903270.1;YP_001089108.1', 'cdf:CD630_25910;pdc:CDIF630_02845', '272563.120.peg.2733']\n",
      "['DEOC_CLOD6', '4.1.2.4', 'WP_003439345.1;YP_001088002.1', 'cdf:CD630_15020;pdc:CDIF630_01667', '272563.120.peg.1571']\n",
      "['Q18BX1_CLOD6', '4.1.2.25', 'WP_011861219.1;YP_001087952.1', 'cdf:CD630_14510;pdc:CDIF630_01615', '272563.120.peg.1520']\n",
      "['Q184R0_CLOD6', '4.1.2.2', 'WP_011861804.1;YP_001089588.1', 'cdf:CD630_30740;pdc:CDIF630_03357', '272563.120.peg.3240']\n",
      "['Q185E8_CLOD6', '4.1.2.17', 'WP_011861507.1;YP_001088789.1', 'cdf:CD630_22770;pdc:CDIF630_02512', '272563.120.peg.2404']\n",
      "['Q188J1_CLOD6', '4.1.2.13', 'WP_011860777.1;YP_001086869.1', 'cdf:CD630_04030;pdc:CDIF630_00531', '272563.120.peg.423']\n",
      "['Q184X4_CLOD6', '4.1.2.13', 'WP_003431312.1;YP_001089650.1', 'cdf:CD630_31350;pdc:CDIF630_03420', '272563.120.peg.3303']\n",
      "['Q180U5_CLOD6', '4.1.2.-', 'WP_009898685.1;YP_001089968.1', 'cdf:CD630_34480;pdc:CDIF630_03759', '272563.120.peg.3644']\n",
      "['HPDS_CLOD6', '4.1.1.83', 'WP_003425410.1;YP_001086623.1', 'cdf:CD630_01540;pdc:CDIF630_00273', '272563.120.peg.167']\n",
      "['HPDL_CLOD6', '4.1.1.83', 'WP_009901609.1;YP_001086622.1', 'cdf:CD630_01530;pdc:CDIF630_00272', '272563.120.peg.166']\n",
      "['Q180S9_CLOD6', '4.1.1.81', 'WP_011861966.1;YP_001089952.1', 'cdf:CD630_34320;pdc:CDIF630_03740', '272563.120.peg.3626']\n",
      "['Q183C6_CLOD6', '4.1.1.81', 'WP_011861645.1;YP_001089203.1', 'cdf:CD630_26920;pdc:CDIF630_02947', '272563.120.peg.2836']\n",
      "['Q18A81_CLOD6', '4.1.1.50', 'WP_003418655.1;YP_001087363.1', 'cdf:CD630_08890;pdc:CDIF630_01009', '272563.120.peg.913']\n",
      "['Q182S7_CLOD6', '4.1.1.36;_6.3.2.5', 'WP_011861611.1;YP_001089103.1', 'cdf:CD630_25870;pdc:CDIF630_02840', '272563.120.peg.2728']\n",
      "['Q181J5_CLOD6', '4.1.1.23', 'WP_003435818.1;YP_001090116.1', 'cdf:CD630_35920;pdc:CDIF630_03914', '272563.120.peg.3797']\n",
      "['Q188C1_CLOD6', '4.1.1.20', 'WP_009902969.1;YP_001088567.1', 'cdf:CD630_20530;pdc:CDIF630_02275', '272563.120.peg.2167']\n",
      "['Q187B7_CLOD6', '4.1.1.20', 'WP_011861341.1;YP_001088314.1', 'cdf:CD630_18080;pdc:CDIF630_02007', '272563.120.peg.1901']\n",
      "['Q18A82_CLOD6', '4.1.1.19', 'WP_011860979.1;YP_001087362.1', 'cdf:CD630_08880;pdc:CDIF630_01008', '272563.120.peg.912']\n",
      "['Q181F1_CLOD6', '4.1.1.19', 'WP_011861997.1;YP_001090072.1', 'cdf:CD630_35510;pdc:CDIF630_03866', '272563.120.peg.3752']\n",
      "['LEPA_CLOD6', '3.6.5.n1', 'WP_003416556.1;YP_001088981.1', 'cdf:CD630_24670;pdc:CDIF630_02713', '272563.120.peg.2606']\n",
      "['OBG_CLOD6', '3.6.5.-', 'WP_009888921.1;YP_001087656.1', 'cdf:CD630_11640;pdc:CDIF630_01311', '272563.120.peg.1214']\n",
      "['Q189U3_CLOD6', '3.6.4.12', 'WP_011860946.1;YP_001087226.1', 'cdf:CD630_07490', '272563.8.peg.786']\n",
      "['Q188P5_CLOD6', '3.6.4.12', 'WP_011860818.1;YP_001086928.1', 'cdf:CD630_04550;pdc:CDIF630_00583', '272563.120.peg.480']\n",
      "['Q181Q8_CLOD6', '3.6.4.12', 'WP_003420541.1;YP_001090181.1', 'cdf:CD630_36570;pdc:CDIF630_03985', '272563.120.peg.3867']\n",
      "['Q18D66_CLOD6', '3.6.4.12', 'WP_009888186.1;YP_001086797.1', 'cdf:CD630_03280;pdc:CDIF630_00456', '272563.120.peg.347']\n",
      "['Q183P4_CLOD6', '3.6.4.12', 'WP_003426582.1;YP_001089319.2', 'cdf:CD630_28060;pdc:CDIF630_03071', '272563.120.peg.2956']\n",
      "['RUVB_CLOD6', '3.6.4.12', 'WP_003426579.1;YP_001089318.1', 'cdf:CD630_28050;pdc:CDIF630_03070', '272563.120.peg.2955']\n",
      "['Q182P8_CLOD6', '3.6.4.12', 'WP_011861601.1;YP_001089075.2', 'cdf:CD630_25600;pdc:CDIF630_02811', '272563.120.peg.2699']\n",
      "['Q187I0_CLOD6', '3.6.4.12', 'WP_009889821.1;YP_001088374.1', 'cdf:CD630_18660;pdc:CDIF630_02069', '272563.120.peg.1961']\n",
      "['Q18CM1_CLOD6', '3.6.4.12', 'WP_003434402.1;YP_001086601.1', 'cdf:CD630_01320;pdc:CDIF630_00249', '272563.120.peg.144']\n",
      "['Q18BJ7_CLOD6', '3.6.4.12', 'WP_011861177.1;YP_001087826.1', 'cdf:CD630_13300;pdc:CDIF630_01486', '272563.120.peg.1390']\n",
      "['Q18AN6_CLOD6', '3.6.4.12', 'WP_003437942.1;YP_001087519.1', 'cdf:CD630_10380;pdc:CDIF630_01178', '272563.120.peg.1078']\n",
      "['Q181A0_CLOD6', '3.6.4.-', 'WP_003425912.1;YP_001090022.1', 'cdf:CD630_35010;pdc:CDIF630_03814', '272563.120.peg.3701']\n",
      "['Q180Y6_CLOD6', '3.6.4.-', 'WP_009898714.1;YP_001090008.2', 'cdf:CD630_34870;pdc:CDIF630_03799', '272563.120.peg.3686']\n",
      "['Q182S5_CLOD6', '3.6.4.-', 'WP_011861610.1;YP_001089102.1', 'cdf:CD630_25860;pdc:CDIF630_02839', '272563.120.peg.2727']\n",
      "['Q183R9_CLOD6', '3.6.3.8', 'WP_009898004.1;YP_001089346.1', 'cdf:CD630_28330;pdc:CDIF630_03100', '272563.120.peg.2985']\n",
      "['Q182I9_CLOD6', '3.6.3.8', 'WP_011861573.1;YP_001089016.1', 'cdf:CD630_25030;pdc:CDIF630_02751', '272563.120.peg.2642']\n",
      "['Q187D1_CLOD6', '3.6.3.8', 'WP_009902789.1;YP_001088329.1', 'cdf:CD630_18240;pdc:CDIF630_02025', '272563.120.peg.1919']\n",
      "['Q186J2_CLOD6', '3.6.3.8', 'WP_009893200.1;YP_001088138.1', 'cdf:CD630_16370;pdc:CDIF630_01814', '272563.120.peg.1715']\n",
      "['Q18D50_CLOD6', '3.6.3.5', 'WP_011860727.1;YP_001086782.1', 'cdf:CD630_03130;pdc:CDIF630_00438', '272563.120.peg.332']\n",
      "['Q185Q4_CLOD6', '3.6.3.4', 'WP_011861457.1;YP_001088629.1', 'cdf:CD630_21150;pdc:CDIF630_02340', '272563.120.peg.2230']\n",
      "['Q18D31_CLOD6', '3.6.3.30', 'WP_011860718.1;YP_001086762.2', 'cdf:CD630_02930', '272563.8.peg.314']\n",
      "['Q180M4_CLOD6', '3.6.3.2', 'WP_011861944.1;YP_001089897.1', 'cdf:CD630_33770;pdc:CDIF630_03681', '272563.120.peg.3570']\n",
      "['Q180M0_CLOD6', '3.6.3.2', 'WP_011861942.1;YP_001089892.1', 'cdf:CD630_33730;pdc:CDIF630_03677', '272563.120.peg.3566']\n",
      "['Q180X3_CLOD6', '3.6.3.14', 'WP_003437838.1;YP_001089995.1', 'cdf:CD630_34750;pdc:CDIF630_03786', '272563.120.peg.3672']\n",
      "['Q180W6_CLOD6', '3.6.3.14', 'WP_003425849.1;YP_001089987.1', 'cdf:CD630_34670;pdc:CDIF630_03778', '272563.120.peg.3664']\n",
      "['Q184E9_CLOD6', '3.6.3.14', 'WP_003422657.1;YP_001089476.1', 'cdf:CD630_29590;pdc:CDIF630_03243', '272563.120.peg.3126']\n",
      "['Q184E5_CLOD6', '3.6.3.14', 'WP_003439726.1;YP_001089474.1', 'cdf:CD630_29570;pdc:CDIF630_03241', '272563.120.peg.3124']\n",
      "['Q184E6_CLOD6', '3.6.3.14', 'WP_009893746.1;YP_001089473.2', 'cdf:CD630_29561;pdc:CDIF630_03240', '272563.120.peg.3123']\n",
      "['Q18CY8_CLOD6', '3.6.3.14', 'WP_011860700.1;YP_001086719.1', 'cdf:CD630_02510;pdc:CDIF630_00374', '272563.120.peg.267']\n",
      "['Q18A20_CLOD6', '3.6.3.-', 'WP_004454245.1;YP_001087299.1', 'cdf:CD630_08220;pdc:CDIF630_00940', '272563.8.peg.860']\n",
      "['Q187Z3_CLOD6', '3.6.3.-', 'WP_003435188.1;YP_001088538.1', 'cdf:CD630_20240', '272563.8.peg.2129']\n",
      "['Q187T8_CLOD6', '3.6.3.-', 'WP_011861410.1;YP_001088484.1', 'cdf:CD630_19780', '272563.8.peg.2074']\n",
      "['Q18C60_CLOD6', '3.6.3.-', 'WP_011861245.1;YP_001088040.1', 'cdf:CD630_15390', '272563.8.peg.1615']\n",
      "['Q18BS1_CLOD6', '3.6.3.-', 'WP_011861202.1;YP_001087905.1', 'cdf:CD630_14050', '272563.8.peg.1477']\n",
      "['Q18A90_CLOD6', '3.6.3.-', 'WP_003437132.1;YP_001087374.1', 'cdf:CD630_09000;pdc:CDIF630_01020', '272563.120.peg.924']\n",
      "['Q18A70_CLOD6', '3.6.3.-', 'WP_009895911.1;YP_001087351.1', 'cdf:CD630_08770;pdc:CDIF630_00997', '272563.120.peg.901']\n",
      "['Q18A68_CLOD6', '3.6.3.-', 'WP_003437183.1;YP_001087348.1', 'cdf:CD630_08740;pdc:CDIF630_00994', '272563.120.peg.898']\n",
      "['Q18A62_CLOD6', '3.6.3.-', 'WP_009888694.1;YP_001087345.1', 'cdf:CD630_08710;pdc:CDIF630_00991', '272563.120.peg.895']\n",
      "['Q189Y0_CLOD6', '3.6.3.-', 'WP_009902156.1;YP_001087262.1', 'cdf:CD630_07850;pdc:CDIF630_00904', '272563.120.peg.808']\n",
      "['Q189V0_CLOD6', '3.6.3.-', 'WP_011860947.1;YP_001087229.1', 'cdf:CD630_07520;pdc:CDIF630_00870', '272563.120.peg.774']\n",
      "['Q189K8_CLOD6', '3.6.3.-', 'WP_011860908.1;YP_001087143.2', 'cdf:CD630_06670;pdc:CDIF630_00782', '272563.120.peg.686']\n",
      "['Q189J8_CLOD6', '3.6.3.-', 'WP_011860900.1;YP_001087128.1', 'cdf:CD630_06530;pdc:CDIF630_00765', '272563.120.peg.669']\n",
      "['Q189F5_CLOD6', '3.6.3.-', 'WP_009895631.1;YP_001087088.2', 'cdf:CD630_06130;pdc:CDIF630_00725', '272563.120.peg.625']\n",
      "['Q188S6_CLOD6', '3.6.3.-', 'WP_009895525.1;YP_001086958.1', 'cdf:CD630_04840;pdc:CDIF630_00612', '272563.120.peg.509']\n",
      "['Q188Q2_CLOD6', '3.6.3.-', 'WP_009888282.1;YP_001086932.1', 'cdf:CD630_04590;pdc:CDIF630_00587', '272563.120.peg.484']\n",
      "['Q188Q0_CLOD6', '3.6.3.-', 'WP_011860819.1;YP_001086929.1', 'cdf:CD630_04560;pdc:CDIF630_00584', '272563.120.peg.481']\n",
      "['Q188L7_CLOD6', '3.6.3.-', 'WP_011860798.1;YP_001086899.1', 'cdf:CD630_04310;pdc:CDIF630_00559', '272563.120.peg.451']\n",
      "['Q18DA4_CLOD6', '3.6.3.-', 'WP_011860754.1;YP_001086833.1', 'cdf:CD630_03660;pdc:CDIF630_00493', '272563.120.peg.386']\n",
      "['Q181M4_CLOD6', '3.6.3.-', 'WP_003435768.1;YP_001090146.1', 'cdf:CD630_36230;pdc:CDIF630_03946', '272563.120.peg.3830']\n",
      "['Q181L0_CLOD6', '3.6.3.-', 'WP_003429190.1;YP_001090133.1', 'cdf:CD630_36080;pdc:CDIF630_03931', '272563.120.peg.3815']\n",
      "['Q181I7_CLOD6', '3.6.3.-', 'WP_003426034.1;YP_001090109.1', 'cdf:CD630_35850;pdc:CDIF630_03906', '272563.120.peg.3789']\n",
      "['Q181D3_CLOD6', '3.6.3.-', 'WP_003435860.1;YP_001090055.1', 'cdf:CD630_35340;pdc:CDIF630_03848', '272563.120.peg.3735']\n",
      "['Q181D2_CLOD6', '3.6.3.-', 'WP_009892101.1;YP_001090054.1', 'cdf:CD630_35330;pdc:CDIF630_03847', '272563.120.peg.3734']\n",
      "['Q181C9_CLOD6', '3.6.3.-', 'WP_003435855.1;YP_001090051.1', 'cdf:CD630_35300;pdc:CDIF630_03844', '272563.120.peg.3731']\n",
      "['Q181C6_CLOD6', '3.6.3.-', 'WP_011861991.1;YP_001090048.1', 'cdf:CD630_35270;pdc:CDIF630_03841', '272563.120.peg.3728']\n",
      "['Q180R5_CLOD6', '3.6.3.-', 'WP_003436292.1;YP_001089937.1', 'cdf:CD630_34170;pdc:CDIF630_03722', '272563.120.peg.3611']\n",
      "['Q180K9_CLOD6', '3.6.3.-', 'WP_011861934.1;YP_001089881.1', 'cdf:CD630_33640;pdc:CDIF630_03664', '272563.120.peg.3553']\n",
      "['Q18D75_CLOD6', '3.6.3.-', 'WP_011860736.1;YP_001086805.1', 'cdf:CD630_03360;pdc:CDIF630_00464', '272563.120.peg.355']\n",
      "['Q180K4_CLOD6', '3.6.3.-', 'WP_011861929.1;YP_001089876.1', 'cdf:CD630_33590;pdc:CDIF630_03659', '272563.120.peg.3548']\n",
      "['Q17ZW1_CLOD6', '3.6.3.-', 'WP_011861883.1;YP_001089733.1', 'cdf:CD630_32170;pdc:CDIF630_03511', '272563.120.peg.3397']\n",
      "['Q17ZV9_CLOD6', '3.6.3.-', 'WP_003438780.1;YP_001089731.1', 'cdf:CD630_32150;pdc:CDIF630_03509', '272563.120.peg.3394']\n",
      "['Q17ZU4_CLOD6', '3.6.3.-', 'WP_011861875.1;YP_001089717.1', 'cdf:CD630_32010;pdc:CDIF630_03493', '272563.120.peg.3378']\n",
      "['Q185A0_CLOD6', '3.6.3.-', 'WP_003433732.1;YP_001089678.1', 'cdf:CD630_31620;pdc:CDIF630_03453', '272563.120.peg.3335']\n",
      "['Q184P3_CLOD6', '3.6.3.-', 'WP_003431374.1;YP_001089570.1', 'cdf:CD630_30560;pdc:CDIF630_03339', '272563.120.peg.3221']\n",
      "['Q18D42_CLOD6', '3.6.3.-', 'WP_011860721.1;YP_001086770.1', 'cdf:CD630_03010;pdc:CDIF630_00426', '272563.120.peg.320']\n",
      "['Q184I7_CLOD6', '3.6.3.-', 'WP_003439823.1;YP_001089515.1', 'cdf:CD630_29970;pdc:CDIF630_03281', '272563.120.peg.3164']\n",
      "['Q184I0_CLOD6', '3.6.3.-', 'WP_011861776.1;YP_001089508.1', 'cdf:CD630_29900;pdc:CDIF630_03274', '272563.120.peg.3157']\n",
      "['Q184H5_CLOD6', '3.6.3.-', 'WP_003439795.1;YP_001089503.1', 'cdf:CD630_29850;pdc:CDIF630_03269', '272563.120.peg.3152']\n",
      "['Q183W2_CLOD6', '3.6.3.-', 'WP_011861723.1;YP_001089388.1', 'cdf:CD630_28750;pdc:CDIF630_03143', '272563.120.peg.3027']\n",
      "['Q183Q4_CLOD6', '3.6.3.-', 'WP_009897981.1;YP_001089331.2', 'cdf:CD630_28180;pdc:CDIF630_03083', '272563.120.peg.2968']\n",
      "['Q183Q0_CLOD6', '3.6.3.-', 'WP_003426617.1;YP_001089330.1', 'cdf:CD630_28170;pdc:CDIF630_03082', '272563.120.peg.2967']\n",
      "['Q182T3_CLOD6', '3.6.3.-', 'WP_011861613.1;YP_001089110.1', 'cdf:CD630_25930;pdc:CDIF630_02847', '272563.120.peg.2735']\n",
      "['Q182M1_CLOD6', '3.6.3.-', 'WP_004454838.1;YP_001089048.1', 'cdf:CD630_25340;pdc:CDIF630_02786', '272563.120.peg.2674']\n",
      "['Q182E3_CLOD6', '3.6.3.-', 'WP_011861561.1;YP_001088970.1', 'cdf:CD630_24560;pdc:CDIF630_02702', '272563.120.peg.2594']\n",
      "['Q185N2_CLOD6', '3.6.3.-', 'WP_003430772.1;YP_001088874.2', 'cdf:CD630_23610;pdc:CDIF630_02601', '272563.120.peg.2493']\n",
      "['Q185I7_CLOD6', '3.6.3.-', 'WP_011861522.1;YP_001088828.1', 'cdf:CD630_23130;pdc:CDIF630_02551', '272563.120.peg.2444']\n",
      "['Q186A1_CLOD6', '3.6.3.-', 'WP_003433859.1;YP_001088725.1', 'cdf:CD630_22110;pdc:CDIF630_02444', '272563.120.peg.2334']\n",
      "['Q185Z8_CLOD6', '3.6.3.-', 'WP_003433862.1;YP_001088724.1', 'cdf:CD630_22100;pdc:CDIF630_02443', '272563.120.peg.2333']\n",
      "['Q185W1_CLOD6', '3.6.3.-', 'WP_003433939.1;YP_001088685.1', 'cdf:CD630_21720;pdc:CDIF630_02404', '272563.120.peg.2294']\n",
      "['Q188H3_CLOD6', '3.6.3.-', 'WP_003434049.1;YP_001088619.1', 'cdf:CD630_21050;pdc:CDIF630_02329', '272563.120.peg.2220']\n",
      "['Q188D7_CLOD6', '3.6.3.-', 'WP_003429004.1;YP_001088582.1', 'cdf:CD630_20680;pdc:CDIF630_02291', '272563.120.peg.2182']\n",
      "['Q187S0_CLOD6', '3.6.3.-', 'WP_011861403.1;YP_001088467.1', 'cdf:CD630_19610;pdc:CDIF630_02166', '272563.120.peg.2058']\n",
      "['Q187R6_CLOD6', '3.6.3.-', 'WP_003435313.1;YP_001088461.1', 'cdf:CD630_19550;pdc:CDIF630_02160', '272563.120.peg.2052']\n",
      "['Q187Q7_CLOD6', '3.6.3.-', 'WP_009897077.1;YP_001088453.2', 'cdf:CD630_19470;pdc:CDIF630_02152', '272563.120.peg.2044']\n",
      "['Q187K0_CLOD6', '3.6.3.-', 'WP_011861379.1;YP_001088397.1', 'cdf:CD630_18890;pdc:CDIF630_02094', '272563.120.peg.1986']\n",
      "['Q187I8_CLOD6', '3.6.3.-', 'WP_003426165.1;YP_001088383.1', 'cdf:CD630_18740;pdc:CDIF630_02077', '272563.120.peg.1970']\n",
      "['Q186Y5_CLOD6', '3.6.3.-', 'WP_003423753.1;YP_001088282.1', 'cdf:CD630_17760;pdc:CDIF630_01974', '272563.120.peg.1869']\n",
      "['Q186W1_CLOD6', '3.6.3.-', 'WP_004454385.1;YP_001088257.1', 'cdf:CD630_17530;pdc:CDIF630_01948', '272563.120.peg.1843']\n",
      "['Q186M6_CLOD6', '3.6.3.-', 'WP_011861296.1;YP_001088171.1', 'cdf:CD630_16690;pdc:CDIF630_01854', '272563.120.peg.1752']\n",
      "['Q186K5_CLOD6', '3.6.3.-', 'WP_009889549.1;YP_001088150.1', 'cdf:CD630_16490;pdc:CDIF630_01826', '272563.120.peg.1727']\n",
      "['Q186H3_CLOD6', '3.6.3.-', 'WP_003436372.1;YP_001088119.1', 'cdf:CD630_16180;pdc:CDIF630_01794', '272563.120.peg.1695']\n",
      "['Q186G9_CLOD6', '3.6.3.-', 'WP_009896767.1;YP_001088115.1', 'cdf:CD630_16140;pdc:CDIF630_01790', '272563.120.peg.1691']\n",
      "['Q186G2_CLOD6', '3.6.3.-', 'WP_009893181.1;YP_001088108.1', 'cdf:CD630_16070;pdc:CDIF630_01783', '272563.120.peg.1684']\n",
      "['Q186F9_CLOD6', '3.6.3.-', 'WP_009902652.1;YP_001088105.1', 'cdf:CD630_16040;pdc:CDIF630_01779', '272563.120.peg.1680']\n",
      "['Q186E2_CLOD6', '3.6.3.-', 'WP_011861266.1;YP_001088088.1', 'cdf:CD630_15870;pdc:CDIF630_01759', '272563.120.peg.1661']\n",
      "['Q18C53_CLOD6', '3.6.3.-', 'WP_003436725.1;YP_001088033.1', 'cdf:CD630_15320;pdc:CDIF630_01700', '272563.120.peg.1605']\n",
      "['Q18C48_CLOD6', '3.6.3.-', 'WP_003436744.1;YP_001088029.1', 'cdf:CD630_15280;pdc:CDIF630_01696', '272563.120.peg.1601']\n",
      "['Q18C24_CLOD6', '3.6.3.-', 'WP_003439347.1;YP_001088004.1', 'cdf:CD630_15040;pdc:CDIF630_01669', '272563.120.peg.1573']\n",
      "['Q18C02_CLOD6', '3.6.3.-', 'WP_009893083.1;YP_001087984.1', 'cdf:CD630_14830;pdc:CDIF630_01647', '272563.120.peg.1552']\n",
      "['Q18BZ3_CLOD6', '3.6.3.-', 'WP_011861230.1;YP_001087974.1', 'cdf:CD630_14730;pdc:CDIF630_01637', '272563.120.peg.1542']\n",
      "['Q18BZ2_CLOD6', '3.6.3.-', 'WP_011861229.1;YP_001087973.1', 'cdf:CD630_14720;pdc:CDIF630_01636', '272563.120.peg.1541']\n",
      "['Q18BY7_CLOD6', '3.6.3.-', 'WP_003438655.1;YP_001087967.1', 'cdf:CD630_14660;pdc:CDIF630_01630', '272563.120.peg.1535']\n",
      "['Q18BL2_CLOD6', '3.6.3.-', 'WP_009896245.1;YP_001087845.1', 'cdf:CD630_13490;pdc:CDIF630_01505', '272563.120.peg.1410']\n",
      "['Q18BD2_CLOD6', '3.6.3.-', 'WP_009902391.1;YP_001087762.1', 'cdf:CD630_12680;pdc:CDIF630_01421', '272563.120.peg.1326']\n",
      "['Q18AU9_CLOD6', '3.6.3.-', 'WP_011861095.1;YP_001087580.1', 'cdf:CD630_10970;pdc:CDIF630_01240', '272563.120.peg.1139']\n",
      "['Q18AT8_CLOD6', '3.6.3.-', 'WP_003438014.1;YP_001087568.1', 'cdf:CD630_10840;pdc:CDIF630_01228', '272563.120.peg.1127']\n",
      "['ECFA2_CLOD6', '3.6.3.-', 'WP_003435657.1;YP_001086569.1', 'cdf:CD630_01010;pdc:CDIF630_00171', '272563.120.peg.111']\n",
      "['Q18AP8_CLOD6', '3.6.3.-', 'WP_009896018.1;YP_001087531.1', 'cdf:CD630_10500;pdc:CDIF630_01190', '272563.120.peg.1090']\n",
      "['Q18AL8_CLOD6', '3.6.3.-', 'WP_011861068.1;YP_001087499.1', 'cdf:CD630_10180;pdc:CDIF630_01154', '272563.120.peg.1058']\n",
      "['Q18AL5_CLOD6', '3.6.3.-', 'WP_011861067.1;YP_001087498.1', 'cdf:CD630_10170;pdc:CDIF630_01153', '272563.120.peg.1057']\n",
      "['Q18AK4_CLOD6', '3.6.3.-', 'WP_003437066.1;YP_001087482.1', 'cdf:CD630_10010;pdc:CDIF630_01137', '272563.120.peg.1040']\n",
      "['Q180R2_CLOD6', '3.6.1.9', 'WP_011861962.1;YP_001089933.1', 'cdf:CD630_34130;pdc:CDIF630_03718', '272563.120.peg.3607']\n",
      "['NTPPA_CLOD6', '3.6.1.9', 'WP_003428485.1;YP_001087635.1', 'cdf:CD630_11430;pdc:CDIF630_01290', '272563.120.peg.1192']\n",
      "['Q180F2_CLOD6', '3.6.1.66;_2.7.7.56', 'WP_003435607.1;YP_001089824.1', 'cdf:CD630_33080;pdc:CDIF630_03609', '272563.120.peg.3493']\n",
      "['SP4A_CLOD6', '3.6.1.3', 'WP_004454943.1;YP_001089140.2', 'cdf:CD630_26290;pdc:CDIF630_02883', '272563.120.peg.2772']\n",
      "['UPPP2_CLOD6', '3.6.1.27', 'WP_011861775.1;YP_001089504.1', 'cdf:CD630_29860', '272563.8.peg.3130']\n",
      "['UPPP1_CLOD6', '3.6.1.27', 'WP_009905410.1;YP_001087828.1', 'cdf:CD630_13320;pdc:CDIF630_01488', '272563.120.peg.1392']\n",
      "['DUT_CLOD6', '3.6.1.23', 'WP_004454700.1;YP_001088915.1', 'cdf:CD630_24030;pdc:CDIF630_02645', '272563.120.peg.2537']\n",
      "['Q18B82_CLOD6', '3.6.1.13', 'WP_003428379.1;YP_001087713.1', 'cdf:CD630_12200;pdc:CDIF630_01369', '272563.120.peg.1271']\n",
      "['Q18D73_CLOD6', '3.6.1.1', 'WP_003434813.1;YP_001086802.1', 'cdf:CD630_03330;pdc:CDIF630_00461', '272563.120.peg.352']\n",
      "['Q180L2_CLOD6', '3.6.1.-', 'WP_011861937.1;YP_001089885.1', 'cdf:CD630_33680', '272563.8.peg.3526']\n",
      "['Q180Z0_CLOD6', '3.6.1.-', 'WP_009894028.1;YP_001090012.1', 'cdf:CD630_34910;pdc:CDIF630_03804', '272563.120.peg.3691']\n",
      "['Q182R5_CLOD6', '3.6.1.-', 'WP_009893669.1;YP_001089092.1', 'cdf:CD630_25760;pdc:CDIF630_02829', '272563.120.peg.2717']\n",
      "['Q186Q8_CLOD6', '3.6.1.-', 'WP_011861310.1;YP_001088204.1', 'cdf:CD630_17010;pdc:CDIF630_01887', '272563.120.peg.1784']\n",
      "['MNME_CLOD6', '3.6.-.-', 'WP_009898860.1;YP_001090200.1', 'cdf:CD630_36760;pdc:CDIF630_04006', '272563.120.peg.3888']\n",
      "['NAGB_CLOD6', '3.5.99.6', 'WP_003437039.1;YP_001087492.1', 'cdf:CD630_10110;pdc:CDIF630_01147', '272563.120.peg.1050']\n",
      "['Q183S8_CLOD6', '3.5.5.1', 'WP_003436788.1;YP_001089355.1', 'cdf:CD630_28420;pdc:CDIF630_03109', '272563.120.peg.2994']\n",
      "['Q189R9_CLOD6', '3.5.4.9', 'WP_003432265.1;YP_001087196.1', 'cdf:CD630_07190;pdc:CDIF630_00835', '272563.120.peg.740']\n",
      "['Q182C5_CLOD6', '3.5.4.5', 'WP_009890647.1;YP_001088951.1', 'cdf:CD630_24380;pdc:CDIF630_02682', '272563.120.peg.2574']\n",
      "['Q186V5_CLOD6', '3.5.4.4', 'WP_004454392.1;YP_001088251.1', 'cdf:CD630_17470;pdc:CDIF630_01941', '272563.120.peg.1836']\n",
      "['Q18CA0_CLOD6', '3.5.4.33', 'WP_003421624.1;YP_001086478.1', 'cdf:CD630_00150;pdc:CDIF630_00025', '272563.120.peg.15']\n",
      "['Q186L8_CLOD6', '3.5.4.3', 'WP_009896806.1;YP_001088165.1', 'cdf:CD630_16630;pdc:CDIF630_01844', '272563.120.peg.1741']\n",
      "['Q188F8_CLOD6', '3.5.4.28', 'WP_009897237.1;YP_001088603.2', 'cdf:CD630_20890;pdc:CDIF630_02312', '272563.120.peg.2203']\n",
      "['Q188E5_CLOD6', '3.5.4.28', 'WP_009897229.1;YP_001088592.1', 'cdf:CD630_20780;pdc:CDIF630_02301', '272563.120.peg.2192']\n",
      "['Q186Q7_CLOD6', '3.5.4.26;_1.1.1.193', 'WP_009896839.1;YP_001088203.1', 'cdf:CD630_17000;pdc:CDIF630_01885', '272563.120.peg.1783']\n",
      "['Q18C73_CLOD6', '3.5.4.19;_3.6.1.31', 'WP_011861249.1;YP_001088055.1', 'cdf:CD630_15540;pdc:CDIF630_01723', '272563.120.peg.1628']\n",
      "['GCH1_CLOD6', '3.5.4.16', 'WP_003429416.1;YP_001087950.1', 'cdf:CD630_14490;pdc:CDIF630_01613', '272563.120.peg.1518']\n",
      "['Q180X5_CLOD6', '3.5.4.12', 'WP_003421374.1;YP_001089997.1', 'cdf:CD630_34770;pdc:CDIF630_03788', '272563.120.peg.3674']\n",
      "['Q187L0_CLOD6', '3.5.4.12', 'WP_009889849.1;YP_001088406.1', 'cdf:CD630_18990;pdc:CDIF630_02104', '272563.120.peg.1996']\n",
      "['Q18A84_CLOD6', '3.5.3.11', 'WP_009888724.1;YP_001087365.1', 'cdf:CD630_08910;pdc:CDIF630_01011', '272563.120.peg.915']\n",
      "['PXPA_CLOD6', '3.5.2.9', 'WP_009896287.1;YP_001087884.1', 'cdf:CD630_13840;pdc:CDIF630_01544', '272563.120.peg.1449']\n",
      "['Q188Q3_CLOD6', '3.5.2.6', 'WP_011860820.1;YP_001086931.1', 'cdf:CD630_04580', '272563.8.peg.482']\n",
      "['Q188R3_CLOD6', '3.5.2.6', 'WP_011860827.1;YP_001086943.1', 'cdf:CD630_04700;pdc:CDIF630_00598', '272563.120.peg.495']\n",
      "['Q181P4_CLOD6', '3.5.2.3', 'WP_009903845.1;YP_001090167.1', 'cdf:CD630_36430;pdc:CDIF630_03967', '272563.120.peg.3853']\n",
      "['PYRC_CLOD6', '3.5.2.3', 'WP_011861243.1;YP_001088027.1', 'cdf:CD630_15260;pdc:CDIF630_01694', '272563.120.peg.1599']\n",
      "['Q181U3_CLOD6', '3.5.2.2', 'WP_009891590.1;YP_001089695.1', 'cdf:CD630_31780;pdc:CDIF630_03470', '272563.120.peg.3353']\n",
      "['Q188F0_CLOD6', '3.5.2.2', 'WP_011861449.1;YP_001088597.1', 'cdf:CD630_20830;pdc:CDIF630_02306', '272563.120.peg.2197']\n",
      "['Q184T1_CLOD6', '3.5.1.94', 'WP_011861815.1;YP_001089607.2', 'cdf:CD630_30930;pdc:CDIF630_03376', '272563.120.peg.3259']\n",
      "['Q182S4_CLOD6', '3.5.1.88', 'WP_003431159.1;YP_001089101.1', 'cdf:CD630_25850;pdc:CDIF630_02838', '272563.120.peg.2726']\n",
      "['Q186W3_CLOD6', '3.5.1.88', 'WP_003423696.1;YP_001088261.1', 'cdf:CD630_17560;pdc:CDIF630_01952', '272563.120.peg.1847']\n",
      "['Q187Z4_CLOD6', '3.5.1.87', 'WP_011861431.1;YP_001088541.1', 'cdf:CD630_20270;pdc:CDIF630_02248', '272563.120.peg.2138']\n",
      "['Q181U6_CLOD6', '3.5.1.81', 'WP_009893877.1;YP_001089699.1', 'cdf:CD630_31820;pdc:CDIF630_03474', '272563.120.peg.3357']\n",
      "['Q183U7_CLOD6', '3.5.1.81', 'WP_011861717.1;YP_001089372.1', 'cdf:CD630_28590;pdc:CDIF630_03127', '272563.120.peg.3011']\n",
      "['Q188X7_CLOD6', '3.5.1.44', 'WP_003425085.1;YP_001087011.1', 'cdf:CD630_05350;pdc:CDIF630_00648', '272563.120.peg.544']\n",
      "['Q183L9_CLOD6', '3.5.1.28', 'WP_011861677.1;YP_001089297.1', 'cdf:CD630_27840', '272563.8.peg.2911']\n",
      "['Q183J9_CLOD6', '3.5.1.28', 'WP_011861670.1;YP_001089274.1', 'cdf:CD630_27610', '272563.8.peg.2888']\n",
      "['Q18CJ4_CLOD6', '3.5.1.28', 'WP_011860636.1;YP_001086574.1', 'cdf:CD630_01060', '272563.8.peg.121']\n",
      "['Q18AN8_CLOD6', '3.5.1.28', 'WP_011861076.1;YP_001087517.2', 'cdf:CD630_10360;pdc:CDIF630_01176', '272563.120.peg.1076']\n",
      "['Q18AN5_CLOD6', '3.5.1.28', 'WP_011861075.1;YP_001087516.2', 'cdf:CD630_10350;pdc:CDIF630_01175', '272563.120.peg.1075']\n",
      "['Q18CM5_CLOD6', '3.5.1.26', 'WP_009895213.1;YP_001086608.1', 'cdf:CD630_01390;pdc:CDIF630_00256', '272563.120.peg.151']\n",
      "['Q180V0_CLOD6', '3.5.1.25', 'WP_009903776.1;YP_001089973.1', 'cdf:CD630_34530;pdc:CDIF630_03764', '272563.120.peg.3649']\n",
      "['Q18AL1_CLOD6', '3.5.1.25', 'WP_011861065.1;YP_001087491.2', 'cdf:CD630_10100;pdc:CDIF630_01146', '272563.120.peg.1049']\n",
      "['GLSA_CLOD6', '3.5.1.2', 'WP_003417962.1;YP_001087034.1', 'cdf:CD630_05580;pdc:CDIF630_00671', '272563.120.peg.568']\n",
      "['Q18D47_CLOD6', '3.5.1.16', 'WP_011860723.1;YP_001086772.1', 'cdf:CD630_03030;pdc:CDIF630_00428', '272563.120.peg.322']\n",
      "['Q182K2_CLOD6', '3.5.1.1', 'WP_003431031.1;YP_001089029.1', 'cdf:CD630_25160;pdc:CDIF630_02765', '272563.120.peg.2655']\n",
      "['Q189Y1_CLOD6', '3.5.1.-', 'WP_011860957.1;YP_001087261.1', 'cdf:CD630_07840;pdc:CDIF630_00902', '272563.8.peg.821']\n",
      "['Q18BR9_CLOD6', '3.5.1.-', 'WP_011861197.1;YP_001087897.1', 'cdf:CD630_13980', '272563.8.peg.1469']\n",
      "['Q18A44_CLOD6', '3.5.1.-', 'WP_004453606.1;YP_001087326.1', 'cdf:CD630_08490;pdc:CDIF630_00968', '272563.120.peg.872']\n",
      "['Q180A0_CLOD6', '3.5.1.-', 'WP_003432345.1;YP_001089773.1', 'cdf:CD630_32570;pdc:CDIF630_03555', '272563.120.peg.3441']\n",
      "['Q17ZW3_CLOD6', '3.5.1.-', 'WP_009891676.1;YP_001089738.1', 'cdf:CD630_32210;pdc:CDIF630_03516', '272563.120.peg.3402']\n",
      "['Q184U9_CLOD6', '3.5.1.-', 'WP_011861827.1;YP_001089626.1', 'cdf:CD630_31120;pdc:CDIF630_03396', '272563.120.peg.3279']\n",
      "['Q18D45_CLOD6', '3.5.1.-', 'WP_011860725.1;YP_001086774.1', 'cdf:CD630_03050;pdc:CDIF630_00430', '272563.120.peg.324']\n",
      "['Q18D33_CLOD6', '3.5.1.-', 'WP_003435936.1;YP_001086760.1', 'cdf:CD630_02910;pdc:CDIF630_00415', '272563.120.peg.309']\n",
      "['Q183S1_CLOD6', '3.5.1.-', 'WP_003436121.1;YP_001089349.1', 'cdf:CD630_28360;pdc:CDIF630_03103', '272563.120.peg.2988']\n",
      "['Q183D2_CLOD6', '3.5.1.-', 'WP_011861648.1;YP_001089208.1', 'cdf:CD630_26970;pdc:CDIF630_02953', '272563.120.peg.2841']\n",
      "['Q185W0_CLOD6', '3.5.1.-', 'WP_009897326.1;YP_001088686.2', 'cdf:CD630_21730;pdc:CDIF630_02405', '272563.120.peg.2295']\n",
      "['Q18CQ8_CLOD6', '3.5.1.-', 'WP_011860656.1;YP_001086635.1', 'cdf:CD630_01660;pdc:CDIF630_00287', '272563.120.peg.181']\n",
      "['Q18BV2_CLOD6', '3.5.1.-', 'WP_003438596.1;YP_001087931.1', 'cdf:CD630_14300;pdc:CDIF630_01594', '272563.120.peg.1499']\n",
      "['Q18AU1_CLOD6', '3.5.1.-', 'WP_011861090.1;YP_001087570.2', 'cdf:CD630_10860;pdc:CDIF630_01230', '272563.120.peg.1129']\n",
      "['Q183D9_CLOD6', '3.5.-.-', 'WP_004455014.1;YP_001089216.1', 'cdf:CD630_27050;pdc:CDIF630_02962', '272563.120.peg.2849']\n",
      "['PPEP1_CLOD6', '3.4.24.89', 'WP_011861706.1;YP_001089343.1', 'cdf:CD630_28300;pdc:CDIF630_03094', '272563.120.peg.2981']\n",
      "['Q182F6_CLOD6', '3.4.24.78', 'WP_003430954.1;YP_001088984.1', 'cdf:CD630_24700;pdc:CDIF630_02716', '272563.120.peg.2609']\n",
      "['Q181G0_CLOD6', '3.4.24.-', 'WP_003422216.1;YP_001090082.1', 'cdf:CD630_35590', '272563.8.peg.3729']\n",
      "['Q189M8_CLOD6', '3.4.24.-', 'WP_003439135.1;YP_001087160.1', 'cdf:CD630_06840;pdc:CDIF630_00797', '272563.120.peg.702']\n",
      "['Q185R9_CLOD6', '3.4.24.-', 'WP_003430605.1;YP_001088643.1', 'cdf:CD630_21290;pdc:CDIF630_02360', '272563.120.peg.2249']\n",
      "['Q187D7_CLOD6', '3.4.24.-', 'WP_003428619.1;YP_001088333.1', 'cdf:CD630_18280;pdc:CDIF630_02030', '272563.120.peg.1923']\n",
      "['LSPA_CLOD6', '3.4.23.36', 'WP_004454916.1;YP_001089114.1', 'cdf:CD630_25970;pdc:CDIF630_02851', '272563.120.peg.2739']\n",
      "['Q182X8_CLOD6', '3.4.23.-', 'WP_011861622.1;YP_001089155.1', 'cdf:CD630_26440;pdc:CDIF630_02898', '272563.120.peg.2787']\n",
      "['Q183M1_CLOD6', '3.4.22.15', 'WP_011861680.1;YP_001089300.1', 'cdf:CD630_27870;pdc:CDIF630_03050', '272563.120.peg.2935']\n",
      "['Q180J6_CLOD6', '3.4.21.92', 'WP_003436212.1;YP_001089868.1', 'cdf:CD630_33510;pdc:CDIF630_03651', '272563.120.peg.3539']\n",
      "['Q180F0_CLOD6', '3.4.21.92', 'WP_003417102.1;YP_001089821.1', 'cdf:CD630_33050;pdc:CDIF630_03606', '272563.120.peg.3490']\n",
      "['Q188Z8_CLOD6', '3.4.21.89', 'WP_003425129.1;YP_001087031.1', 'cdf:CD630_05550;pdc:CDIF630_00668', '272563.120.peg.565']\n",
      "['Q188Z9_CLOD6', '3.4.21.89', 'WP_003425127.1;YP_001087030.1', 'cdf:CD630_05540;pdc:CDIF630_00667', '272563.120.peg.564']\n",
      "['Q18BJ6_CLOD6', '3.4.21.89', 'WP_011861178.1;YP_001087827.1', 'cdf:CD630_13310;pdc:CDIF630_01487', '272563.120.peg.1391']\n",
      "['LEXA_CLOD6', '3.4.21.88', 'WP_009897067.1;YP_001088437.2', 'cdf:CD630_19310;pdc:CDIF630_02135', '272563.120.peg.2027']\n",
      "['Q189A7_CLOD6', '3.4.21.53', 'WP_003438993.1;YP_001087040.1', 'cdf:CD630_05640;pdc:CDIF630_00677', '272563.120.peg.574']\n",
      "['LON_CLOD6', '3.4.21.53', 'WP_003435613.1;YP_001089817.2', 'cdf:CD630_33010;pdc:CDIF630_03602', '272563.120.peg.3486']\n",
      "['Q18B75_CLOD6', '3.4.21.116', 'WP_003438145.1;YP_001087706.1', 'cdf:CD630_12130;pdc:CDIF630_01362', '272563.120.peg.1264']\n",
      "['Q180V3_CLOD6', '3.4.21.102', 'WP_003437815.1;YP_001089975.1', 'cdf:CD630_34550', '272563.8.peg.3620']\n",
      "['Q187W2_CLOD6', '3.4.21.-', 'WP_011861421.1;YP_001088508.1', 'cdf:CD630_20000', '272563.8.peg.2105']\n",
      "['Q180C8_CLOD6', '3.4.21.-', 'WP_003435637.1;YP_001089800.1', 'cdf:CD630_32840;pdc:CDIF630_03584', '272563.120.peg.3469']\n",
      "['PCP_CLOD6', '3.4.19.3', 'WP_011861300.1;YP_001088178.1', 'cdf:CD630_16760;pdc:CDIF630_01861', '272563.120.peg.1759']\n",
      "['Q18BR8_CLOD6', '3.4.17.13', 'WP_009896305.1;YP_001087898.1', 'cdf:CD630_13990;pdc:CDIF630_01559', '272563.120.peg.1465']\n",
      "['Q183Q5_CLOD6', '3.4.17.11', 'WP_011861703.1;YP_001089335.1', 'cdf:CD630_28220;pdc:CDIF630_03087', '272563.120.peg.2973']\n",
      "['Q185T0_CLOD6', '3.4.16.4', 'WP_011861467.1;YP_001088655.1', 'cdf:CD630_21410', '272563.8.peg.2254']\n",
      "['Q188W3_CLOD6', '3.4.16.4', 'WP_009895546.1;YP_001086993.2', 'cdf:CD630_05150;pdc:CDIF630_00627', '272563.120.peg.523']\n",
      "['Q181K3_CLOD6', '3.4.16.4', 'WP_003435804.1;YP_001090125.1', 'cdf:CD630_36010;pdc:CDIF630_03923', '272563.120.peg.3807']\n",
      "['Q182J0_CLOD6', '3.4.16.4', 'WP_004454801.1;YP_001089017.1', 'cdf:CD630_25040;pdc:CDIF630_02753', '272563.120.peg.2643']\n",
      "['Q182I4_CLOD6', '3.4.16.4', 'WP_004454794.1;YP_001089011.1', 'cdf:CD630_24980;pdc:CDIF630_02745', '272563.120.peg.2636']\n",
      "['Q186I1_CLOD6', '3.4.16.4', 'WP_003436406.1;YP_001088128.1', 'cdf:CD630_16270;pdc:CDIF630_01804', '272563.120.peg.1705']\n",
      "['Q18BF4_CLOD6', '3.4.16.4', 'WP_003428267.1;YP_001087785.1', 'cdf:CD630_12910;pdc:CDIF630_01445', '272563.120.peg.1349']\n",
      "['Q182H3_CLOD6', '3.4.13.9', 'WP_009897674.1;YP_001088999.1', 'cdf:CD630_24850;pdc:CDIF630_02732', '272563.120.peg.2624']\n",
      "['Q185M2_CLOD6', '3.4.13.9', 'WP_004454650.1;YP_001088862.1', 'cdf:CD630_23470;pdc:CDIF630_02586', '272563.120.peg.2478']\n",
      "['Q189Q4_CLOD6', '3.4.13.3', 'WP_003436882.1;YP_001087184.1', 'cdf:CD630_07080;pdc:CDIF630_00823', '272563.120.peg.728']\n",
      "['Q181H2_CLOD6', '3.4.13.19', 'WP_003434938.1;YP_001090093.1', 'cdf:CD630_35700;pdc:CDIF630_03890', '272563.120.peg.3773']\n",
      "['Q186K2_CLOD6', '3.4.13.19', 'WP_009889546.1;YP_001088147.1', 'cdf:CD630_16460;pdc:CDIF630_01823', '272563.120.peg.1724']\n",
      "['Q182U6_CLOD6', '3.4.11.9', 'WP_003426912.1;YP_001089124.1', 'cdf:CD630_26130;pdc:CDIF630_02866', '272563.120.peg.2756']\n",
      "['Q185U0_CLOD6', '3.4.11.7', 'WP_003433978.1;YP_001088665.1', 'cdf:CD630_21520;pdc:CDIF630_02383', '272563.120.peg.2272']\n",
      "['Q187V8_CLOD6', '3.4.11.5', 'WP_011861419.1;YP_001088505.1', 'cdf:CD630_19970', '272563.8.peg.2100']\n",
      "['PIP_CLOD6', '3.4.11.5', 'WP_003439900.1;YP_001089556.1', 'cdf:CD630_30410;pdc:CDIF630_03325', '272563.120.peg.3207']\n",
      "['Q18AP6_CLOD6', '3.4.11.4', 'WP_003437954.1;YP_001087527.1', 'cdf:CD630_10460;pdc:CDIF630_01186', '272563.120.peg.1086']\n",
      "['Q186F1_CLOD6', '3.4.11.18', 'WP_003430012.1;YP_001088097.1', 'cdf:CD630_15960;pdc:CDIF630_01770', '272563.120.peg.1672']\n",
      "['Q18CI4_CLOD6', '3.4.11.18', 'WP_003421130.1;YP_001086559.1', 'cdf:CD630_00920;pdc:CDIF630_00161', '272563.120.peg.101']\n",
      "['Q18BF9_CLOD6', '3.4.11.1;_3.4.11.10', 'WP_011861171.1;YP_001087794.1', 'cdf:CD630_13000;pdc:CDIF630_01454', '272563.120.peg.1358']\n",
      "['Q188X0_CLOD6', '3.4.11.-', 'WP_011860849.1;YP_001087002.1', 'cdf:CD630_05240;pdc:CDIF630_00637', '272563.120.peg.533']\n",
      "['Q181J7_CLOD6', '3.4.11.-', 'WP_003429165.1;YP_001090119.1', 'cdf:CD630_35950;pdc:CDIF630_03917', '272563.120.peg.3801']\n",
      "['Q18D72_CLOD6', '3.4.11.-', 'WP_003434194.1;YP_001086798.1', 'cdf:CD630_03290;pdc:CDIF630_00457', '272563.120.peg.348']\n",
      "['Q18AS5_CLOD6', '3.4.11.-', 'WP_011861085.1;YP_001087556.1', 'cdf:CD630_10720;pdc:CDIF630_01216', '272563.120.peg.1115']\n",
      "['PRSW_CLOD6', '3.4.-.-', 'WP_009901989.1;YP_001087028.1', 'cdf:CD630_05520;pdc:CDIF630_00665', '272563.120.peg.562']\n",
      "['Q180U0_CLOD6', '3.4.-.-', 'WP_011861971.1;YP_001089962.1', 'cdf:CD630_34420;pdc:CDIF630_03753', '272563.120.peg.3638']\n",
      "['Q181Y8_CLOD6', '3.4.-.-', 'WP_009903185.1;YP_001088914.1', 'cdf:CD630_24020;pdc:CDIF630_02644', '272563.120.peg.2536']\n",
      "['Q181N3_CLOD6', '3.3.2.1', 'WP_003429214.1;YP_001090155.1', 'cdf:CD630_36320;pdc:CDIF630_03955', '272563.120.peg.3840']\n",
      "['Q182U5_CLOD6', '3.2.2.9', 'WP_003416187.1;YP_001089122.1', 'cdf:CD630_26110;pdc:CDIF630_02863', '272563.120.peg.2754']\n",
      "['UNG_CLOD6', '3.2.2.27', 'WP_003416528.1;YP_001088995.1', 'cdf:CD630_24810;pdc:CDIF630_02728', '272563.120.peg.2620']\n",
      "['Q186N9_CLOD6', '3.2.2.1', 'WP_004454479.1;YP_001088185.1', 'cdf:CD630_16820;pdc:CDIF630_01867', '272563.120.peg.1765']\n",
      "['3MGH_CLOD6', '3.2.2.-', 'WP_003427435.1;YP_001087994.1', 'cdf:CD630_14930;pdc:CDIF630_01658', '272563.120.peg.1562']\n",
      "['Q5S4P9_CLOD6', '3.2.1.96', 'WP_011861172.1;YP_001087799.1', 'cdf:CD630_13040;pdc:CDIF630_01459', '272563.120.peg.1363']\n",
      "['Q184S7_CLOD6', '3.2.1.93', 'WP_003439981.1;YP_001089605.1', 'cdf:CD630_30910;pdc:CDIF630_03374', '272563.120.peg.3257']\n",
      "['Q184W2_CLOD6', '3.2.1.86', 'WP_011861833.1;YP_001089639.1', 'cdf:CD630_31240', '272563.8.peg.3269']\n",
      "['Q184R5_CLOD6', '3.2.1.86', 'WP_011861807.1;YP_001089593.1', 'cdf:CD630_30790', '272563.8.peg.3223']\n",
      "['Q18BK1_CLOD6', '3.2.1.86', 'WP_011861180.1;YP_001087834.1', 'cdf:CD630_13380', '272563.8.peg.1405']\n",
      "['Q18A15_CLOD6', '3.2.1.86', 'WP_011860965.1;YP_001087295.1', 'cdf:CD630_08180;pdc:CDIF630_00936', '272563.120.peg.840']\n",
      "['Q18DC7_CLOD6', '3.2.1.86', 'WP_003431416.1;YP_001086855.1', 'cdf:CD630_03890;pdc:CDIF630_00516', '272563.120.peg.408']\n",
      "['Q184X3_CLOD6', '3.2.1.86', 'WP_011861842.1;YP_001089651.1', 'cdf:CD630_31360;pdc:CDIF630_03421', '272563.120.peg.3304']\n",
      "['Q184W9_CLOD6', '3.2.1.86', 'WP_011861839.1;YP_001089645.1', 'cdf:CD630_31300;pdc:CDIF630_03415', '272563.120.peg.3298']\n",
      "['Q184V1_CLOD6', '3.2.1.86', 'WP_003440025.1;YP_001089629.1', 'cdf:CD630_31150;pdc:CDIF630_03399', '272563.120.peg.3282']\n",
      "['Q184T3_CLOD6', '3.2.1.86', 'WP_011861818.1;YP_001089610.1', 'cdf:CD630_30960;pdc:CDIF630_03379', '272563.120.peg.3262']\n",
      "['Q184T2_CLOD6', '3.2.1.86', 'WP_011861817.1;YP_001089609.1', 'cdf:CD630_30950;pdc:CDIF630_03378', '272563.120.peg.3261']\n",
      "['Q183W7_CLOD6', '3.2.1.86', 'WP_011861728.1;YP_001089395.1', 'cdf:CD630_28820;pdc:CDIF630_03150', '272563.120.peg.3034']\n",
      "['Q18CC0_CLOD6', '3.2.1.58', 'WP_009901552.1;YP_001086495.1', 'cdf:CD630_00320;pdc:CDIF630_00094', '272563.120.peg.36']\n",
      "['Q185X6_CLOD6', '3.2.1.54', 'WP_009897348.1;YP_001088703.1', 'cdf:CD630_21900;pdc:CDIF630_02422', '272563.120.peg.2312']\n",
      "['Q182N3_CLOD6', '3.2.1.4', 'WP_011861595.1;YP_001089060.1', 'cdf:CD630_25460;pdc:CDIF630_02798', '272563.120.peg.2686']\n",
      "['Q187B5_CLOD6', '3.2.1.26', 'WP_009896953.1;YP_001088311.1', 'cdf:CD630_18050;pdc:CDIF630_02004', '272563.120.peg.1898']\n",
      "['Q184S2_CLOD6', '3.2.1.24', 'WP_011861811.1;YP_001089599.1', 'cdf:CD630_30850;pdc:CDIF630_03368', '272563.120.peg.3251']\n",
      "['Q184K1_CLOD6', '3.2.1.24', 'WP_011861780.1;YP_001089529.1', 'cdf:CD630_30120;pdc:CDIF630_03297', '272563.120.peg.3179']\n",
      "['Q182Q9_CLOD6', '3.2.1.24', 'WP_011861603.1;YP_001089085.1', 'cdf:CD630_25690;pdc:CDIF630_02822', '272563.120.peg.2710']\n",
      "['Q18A53_CLOD6', '3.2.1.122', 'WP_004454201.1;YP_001087338.1', 'cdf:CD630_08640;pdc:CDIF630_00984', '272563.120.peg.888']\n",
      "['Q184P6_CLOD6', '3.2.1.122', 'WP_011861799.1;YP_001089574.1', 'cdf:CD630_30600;pdc:CDIF630_03343', '272563.120.peg.3226']\n",
      "['Q188R1_CLOD6', '3.2.1.1', 'WP_011860825.1;YP_001086941.1', 'cdf:CD630_04680;pdc:CDIF630_00596', '272563.120.peg.493']\n",
      "['Q18A77_CLOD6', '3.2.1.-', 'WP_003437164.1;YP_001087360.1', 'cdf:CD630_08860;pdc:CDIF630_01006', '272563.120.peg.910']\n",
      "['Q18CB9_CLOD6', '3.2.1.-', 'WP_004453968.1;YP_001086496.1', 'cdf:CD630_00330;pdc:CDIF630_00095', '272563.120.peg.37']\n",
      "['Q184Q9_CLOD6', '3.2.1.-', 'WP_003439934.1;YP_001089585.1', 'cdf:CD630_30710;pdc:CDIF630_03354', '272563.120.peg.3237']\n",
      "['Q182J3_CLOD6', '3.2.1.-', 'WP_011861576.1;YP_001089022.1', 'cdf:CD630_25090;pdc:CDIF630_02758', '272563.120.peg.2648']\n",
      "['Q18AY9_CLOD6', '3.2.1.-', 'WP_003438043.1;YP_001087621.1', 'cdf:CD630_11300;pdc:CDIF630_01277', '272563.120.peg.1178']\n",
      "['PHNXW_CLOD6', '3.11.1.1;_2.6.1.37', 'WP_011861713.1;YP_001089362.1', 'cdf:CD630_28490;pdc:CDIF630_03116', '272563.120.peg.3001']\n",
      "['Q186M2_CLOD6', '3.1.4.46', 'WP_011861293.1;YP_001088168.1', 'cdf:CD630_16660;pdc:CDIF630_01850', '272563.120.peg.1748']\n",
      "['Q18BR5_CLOD6', '3.1.4.46', 'WP_011861199.1;YP_001087901.1', 'cdf:CD630_14020;pdc:CDIF630_01562', '272563.120.peg.1468']\n",
      "['Q181R1_CLOD6', '3.1.4.-', 'WP_003429268.1;YP_001090183.1', 'cdf:CD630_36590;pdc:CDIF630_03987', '272563.120.peg.3869']\n",
      "['Q180F1_CLOD6', '3.1.4.-', 'WP_003435609.1;YP_001089823.1', 'cdf:CD630_33070;pdc:CDIF630_03608', '272563.120.peg.3492']\n",
      "['Q18B21_CLOD6', '3.1.4.-', 'WP_003438072.1;YP_001087652.1', 'cdf:CD630_11600;pdc:CDIF630_01307', '272563.120.peg.1210']\n",
      "['Q18AQ0_CLOD6', '3.1.4.-', 'WP_003428581.1;YP_001087529.1', 'cdf:CD630_10480;pdc:CDIF630_01188', '272563.120.peg.1088']\n",
      "['Q182M3_CLOD6', '3.1.3.5', 'WP_009903241.1;YP_001089051.1', 'cdf:CD630_25370;pdc:CDIF630_02789', '272563.120.peg.2677']\n",
      "['Q180Y0_CLOD6', '3.1.3.48', 'WP_003437844.1;YP_001090001.1', 'cdf:CD630_34810;pdc:CDIF630_03792', '272563.120.peg.3679']\n",
      "['Q18CY3_CLOD6', '3.1.3.3', 'WP_011860692.1;YP_001086709.1', 'cdf:CD630_02410', '272563.8.peg.259']\n",
      "['Q180Y8_CLOD6', '3.1.3.16', 'WP_009898719.1;YP_001090011.1', 'cdf:CD630_34900;pdc:CDIF630_03803', '272563.120.peg.3690']\n",
      "['Q189T7_CLOD6', '3.1.3.15', 'WP_009895786.1;YP_001087214.1', 'cdf:CD630_07370;pdc:CDIF630_00853', '272563.120.peg.758']\n",
      "['Q185Y0_CLOD6', '3.1.3.15', 'WP_009897350.1;YP_001088704.1', 'cdf:CD630_21910;pdc:CDIF630_02423', '272563.120.peg.2313']\n",
      "['Q18AS0_CLOD6', '3.1.3.15', 'WP_009896027.1;YP_001087553.1', 'cdf:CD630_10690;pdc:CDIF630_01213', '272563.120.peg.1112']\n",
      "['F16PC_CLOD6', '3.1.3.11', 'WP_003438115.1;YP_001087684.1', 'cdf:CD630_11910;pdc:CDIF630_01340', '272563.120.peg.1242']\n",
      "['Q188R2_CLOD6', '3.1.3.-', 'WP_009895496.1;YP_001086940.2', 'cdf:CD630_04670;pdc:CDIF630_00595', '272563.120.peg.492']\n",
      "['Q181K6_CLOD6', '3.1.3.-', 'WP_003435799.1;YP_001090129.1', 'cdf:CD630_36050;pdc:CDIF630_03927', '272563.120.peg.3811']\n",
      "['Q18D90_CLOD6', '3.1.3.-', 'WP_009888208.1;YP_001086817.1', 'cdf:CD630_03500;pdc:CDIF630_00478', '272563.120.peg.369']\n",
      "['Q180T3_CLOD6', '3.1.3.-', 'WP_011861969.1;YP_001089956.1', 'cdf:CD630_34360;pdc:CDIF630_03745', '272563.120.peg.3630']\n",
      "['Q18D48_CLOD6', '3.1.3.-', 'WP_003421079.1;YP_001086778.1', 'cdf:CD630_03090;pdc:CDIF630_00434', '272563.120.peg.328']\n",
      "['Q184T9_CLOD6', '3.1.3.-', 'WP_011861824.1;YP_001089618.2', 'cdf:CD630_31040;pdc:CDIF630_03388', '272563.120.peg.3270']\n",
      "['Q183U3_CLOD6', '3.1.3.-', 'WP_009898027.1;YP_001089370.1', 'cdf:CD630_28570;pdc:CDIF630_03125', '272563.120.peg.3009']\n",
      "['Q183K1_CLOD6', '3.1.3.-', 'WP_011861671.1;YP_001089277.1', 'cdf:CD630_27640;pdc:CDIF630_03027', '272563.120.peg.2911']\n",
      "['Q182R1_CLOD6', '3.1.3.-', 'WP_009903260.1;YP_001089089.1', 'cdf:CD630_25730;pdc:CDIF630_02826', '272563.120.peg.2714']\n",
      "['Q182I7_CLOD6', '3.1.3.-', 'WP_004454798.1;YP_001089014.1', 'cdf:CD630_25010;pdc:CDIF630_02749', '272563.120.peg.2640']\n",
      "['Q185N0_CLOD6', '3.1.3.-', 'WP_011861537.1;YP_001088872.1', 'cdf:CD630_23590;pdc:CDIF630_02599', '272563.120.peg.2490']\n",
      "['Q18CU7_CLOD6', '3.1.3.-', 'WP_011860672.1;YP_001086678.1', 'cdf:CD630_02100;pdc:CDIF630_00332', '272563.120.peg.226']\n",
      "['Q18BS7_CLOD6', '3.1.3.-', 'WP_003438534.1;YP_001087910.1', 'cdf:CD630_14100;pdc:CDIF630_01571', '272563.120.peg.1477']\n",
      "['Q18AI1_CLOD6', '3.1.3.-', 'WP_011861058.1;YP_001087464.1', 'cdf:CD630_09830;pdc:CDIF630_01118', '272563.120.peg.1022']\n",
      "['Q181C4_CLOD6', '3.1.26.8', 'WP_003437901.1;YP_001090045.1', 'cdf:CD630_35240;pdc:CDIF630_03838', '272563.120.peg.3724']\n",
      "['RNPA_CLOD6', '3.1.26.5', 'WP_003429302.1;YP_001090204.1', 'cdf:CD630_36790;pdc:CDIF630_04010', '272563.120.peg.3892']\n",
      "['RNH2_CLOD6', '3.1.26.4', 'WP_003438239.1;YP_001087756.1', 'cdf:CD630_12620;pdc:CDIF630_01415', '272563.120.peg.1320']\n",
      "['Q18BA7_CLOD6', '3.1.26.3', 'WP_003428341.1;YP_001087741.1', 'cdf:CD630_12480;pdc:CDIF630_01400', '272563.120.peg.1305']\n",
      "['RNZ_CLOD6', '3.1.26.11', 'WP_004454848.1;YP_001089053.1', 'cdf:CD630_25390;pdc:CDIF630_02791', '272563.120.peg.2679']\n",
      "['Q18CD4_CLOD6', '3.1.26.-', 'WP_011860630.1;YP_001086516.1', 'cdf:CD630_00530;pdc:CDIF630_00117', '272563.120.peg.58']\n",
      "['RUVC_CLOD6', '3.1.22.4', 'WP_011861695.1;YP_001089320.1', 'cdf:CD630_28070;pdc:CDIF630_03072', '272563.120.peg.2957']\n",
      "['END4_CLOD6', '3.1.21.2', 'WP_003438990.1;YP_001087036.1', 'cdf:CD630_05600;pdc:CDIF630_00673', '272563.120.peg.570']\n",
      "['Q185A2_CLOD6', '3.1.13.1', 'WP_003434984.1;YP_001089680.1', 'cdf:CD630_31640;pdc:CDIF630_03455', '272563.120.peg.3337']\n",
      "['Q184G9_CLOD6', '3.1.12.1', 'WP_009903365.1;YP_001089495.1', 'cdf:CD630_29770', '272563.8.peg.3121']\n",
      "['Q18B65_CLOD6', '3.1.11.6', 'WP_003428410.1;YP_001087697.1', 'cdf:CD630_12040;pdc:CDIF630_01353', '272563.120.peg.1255']\n",
      "['EX7L_CLOD6', '3.1.11.6', 'WP_003438130.1;YP_001087696.1', 'cdf:CD630_12030;pdc:CDIF630_01352', '272563.120.peg.1254']\n",
      "['Q18BK4_CLOD6', '3.1.11.2', 'WP_003438367.1;YP_001087837.1', 'cdf:CD630_13410;pdc:CDIF630_01497', '272563.120.peg.1401']\n",
      "['DTD_CLOD6', '3.1.1.96', 'WP_003426448.1;YP_001089255.1', 'cdf:CD630_27430;pdc:CDIF630_03006', '272563.120.peg.2890']\n",
      "['Q188Y9_CLOD6', '3.1.1.61', 'WP_011860852.1;YP_001087018.1', 'cdf:CD630_05420;pdc:CDIF630_00655', '272563.120.peg.551']\n",
      "['PTH_CLOD6', '3.1.1.29', 'WP_009892068.1;YP_001090023.1', 'cdf:CD630_35020;pdc:CDIF630_03815', '272563.120.peg.3702']\n",
      "['ADDA_CLOD6', '3.1.-.-;_3.6.4.12', 'WP_011861079.1;YP_001087522.1', 'cdf:CD630_10410;pdc:CDIF630_01181', '272563.120.peg.1081']\n",
      "['ADDB_CLOD6', '3.1.-.-;_3.6.4.12', 'WP_011861078.1;YP_001087521.1', 'cdf:CD630_10400;pdc:CDIF630_01180', '272563.120.peg.1080']\n",
      "['RNY_CLOD6', '3.1.-.-', 'WP_003428216.1;YP_001087825.2', 'cdf:CD630_13290', '272563.8.peg.1396']\n",
      "['MUTS2_CLOD6', '3.1.-.-', 'WP_011860930.1;YP_001087185.1', 'cdf:CD630_07090;pdc:CDIF630_00824', '272563.120.peg.729']\n",
      "['Q180V9_CLOD6', '3.1.-.-', 'WP_003425840.1;YP_001089981.1', 'cdf:CD630_34610;pdc:CDIF630_03772', '272563.120.peg.3658']\n",
      "['Q184G6_CLOD6', '3.1.-.-', 'WP_011861768.1;YP_001089494.1', 'cdf:CD630_29760;pdc:CDIF630_03260', '272563.120.peg.3143']\n",
      "['Q184G7_CLOD6', '3.1.-.-', 'WP_003439780.1;YP_001089493.2', 'cdf:CD630_29750;pdc:CDIF630_03259', '272563.120.peg.3142']\n",
      "['Q183V0_CLOD6', '3.1.-.-', 'WP_009898037.1;YP_001089377.1', 'cdf:CD630_28640;pdc:CDIF630_03132', '272563.120.peg.3016']\n",
      "['Q183I1_CLOD6', '3.1.-.-', 'WP_011861665.1;YP_001089258.1', 'cdf:CD630_27460;pdc:CDIF630_03009', '272563.120.peg.2893']\n",
      "['YBEY_CLOD6', '3.1.-.-', 'WP_003430894.1;YP_001088953.1', 'cdf:CD630_24400;pdc:CDIF630_02684', '272563.120.peg.2576']\n",
      "['Q185T3_CLOD6', '3.1.-.-', 'WP_011861470.1;YP_001088659.1', 'cdf:CD630_21450;pdc:CDIF630_02376', '272563.120.peg.2265']\n",
      "['Q186D2_CLOD6', '3.1.-.-', 'WP_011861260.1;YP_001088078.1', 'cdf:CD630_15780;pdc:CDIF630_01747', '272563.120.peg.1651']\n",
      "['Q18BF6_CLOD6', '3.1.-.-', 'WP_011861169.1;YP_001087783.1', 'cdf:CD630_12890;pdc:CDIF630_01443', '272563.120.peg.1347']\n",
      "['YQGF_CLOD6', '3.1.-.-', 'WP_003428279.1;YP_001087779.1', 'cdf:CD630_12850;pdc:CDIF630_01439', '272563.120.peg.1343']\n",
      "['Q188X8_CLOD6', '3.-.-.-', 'WP_011860850.1;YP_001087010.1', 'cdf:CD630_05340', '272563.8.peg.560']\n",
      "['Q181Q3_CLOD6', '3.-.-.-', 'WP_003435698.1;YP_001090175.1', 'cdf:CD630_36510;pdc:CDIF630_03975', '272563.120.peg.3861']\n",
      "['Q180U3_CLOD6', '3.-.-.-', 'WP_009891992.1;YP_001089966.1', 'cdf:CD630_34460;pdc:CDIF630_03757', '272563.120.peg.3642']\n",
      "['Q18D12_CLOD6', '3.-.-.-', 'WP_003425190.1;YP_001086740.1', 'cdf:CD630_02710;pdc:CDIF630_00395', '272563.120.peg.289']\n",
      "['Q18BR6_CLOD6', '3.-.-.-', 'WP_011861198.1;YP_001087900.1', 'cdf:CD630_14010;pdc:CDIF630_01561', '272563.120.peg.1467']\n",
      "['Q18AY0_CLOD6', '3.-.-.-', 'WP_009888897.1;YP_001087612.1', 'cdf:CD630_11230;pdc:CDIF630_01267', '272563.120.peg.1169']\n",
      "['Q182I2_CLOD6', '2.9.1.1', 'WP_009903217.1;YP_001089008.2', 'cdf:CD630_24950', '272563.8.peg.2616']\n",
      "['Q181P5_CLOD6', '2.9.1.1', 'WP_009898827.1;YP_001090166.1', 'cdf:CD630_36420;pdc:CDIF630_03966', '272563.120.peg.3852']\n",
      "['RIMO_CLOD6', '2.8.4.4', 'WP_011861176.1;YP_001087822.1', 'cdf:CD630_13260;pdc:CDIF630_01482', '272563.120.peg.1386']\n",
      "['MIAB_CLOD6', '2.8.4.3', 'WP_003435252.1;YP_001088491.1', 'cdf:CD630_19860;pdc:CDIF630_02195', '272563.120.peg.2085']\n",
      "['Q183B1_CLOD6', '2.8.3.5', 'WP_009897858.1;YP_001089189.1', 'cdf:CD630_26780;pdc:CDIF630_02932', '272563.120.peg.2821']\n",
      "['Q188I3_CLOD6', '2.8.3.-', 'WP_003427766.1;YP_001086861.1', 'cdf:CD630_03950;pdc:CDIF630_00523', '272563.120.peg.415']\n",
      "['Q183B2_CLOD6', '2.8.3.-', 'WP_011861637.1;YP_001089188.1', 'cdf:CD630_26770;pdc:CDIF630_02931', '272563.120.peg.2820']\n",
      "['Q185L6_CLOD6', '2.8.3.-', 'WP_011861532.1;YP_001088858.1', 'cdf:CD630_23430;pdc:CDIF630_02582', '272563.120.peg.2474']\n",
      "['Q189U9_CLOD6', '2.8.1.7', 'WP_009895806.1;YP_001087230.1', 'cdf:CD630_07530;pdc:CDIF630_00871', '272563.120.peg.775']\n",
      "['Q181S2_CLOD6', '2.8.1.7', 'WP_003434244.1;YP_001090195.1', 'cdf:CD630_36700;pdc:CDIF630_03999', '272563.120.peg.3881']\n",
      "['Q184M0_CLOD6', '2.8.1.7', 'WP_011861788.1;YP_001089547.1', 'cdf:CD630_30320;pdc:CDIF630_03316', '272563.120.peg.3198']\n",
      "['Q18BE4_CLOD6', '2.8.1.7', 'WP_011861166.1;YP_001087773.1', 'cdf:CD630_12790;pdc:CDIF630_01432', '272563.120.peg.1337']\n",
      "['BIOB_CLOD6', '2.8.1.6', 'WP_009895370.1;YP_001086766.1', 'cdf:CD630_02970;pdc:CDIF630_00422', '272563.120.peg.316']\n",
      "['Q185U5_CLOD6', '2.8.1.6', 'WP_003433972.1;YP_001088668.2', 'cdf:CD630_21550;pdc:CDIF630_02386', '272563.120.peg.2275']\n",
      "['THII_CLOD6', '2.8.1.4', 'WP_011860948.1;YP_001087231.1', 'cdf:CD630_07540', '272563.8.peg.791']\n",
      "['Q18C56_CLOD6', '2.8.1.2', 'WP_003436717.1;YP_001088036.1', 'cdf:CD630_15350;pdc:CDIF630_01703', '272563.120.peg.1608']\n",
      "['MNMA_CLOD6', '2.8.1.13', 'WP_011861167.1;YP_001087775.1', 'cdf:CD630_12810', '272563.8.peg.1346']\n",
      "['THIG_CLOD6', '2.8.1.10', 'WP_009889636.1;YP_001088208.1', 'cdf:CD630_17040;pdc:CDIF630_01892', '272563.120.peg.1788']\n",
      "['Q18CL3_CLOD6', '2.8.1.1', 'WP_011860639.1;YP_001086590.1', 'cdf:CD630_01210;pdc:CDIF630_00237', '272563.120.peg.133']\n",
      "['SELD_CLOD6', '2.7.9.3', 'YP_001089009.2', 'cdf:CD630_24960', '272563.8.peg.2617']\n",
      "['Q181Z4_CLOD6', '2.7.9.1', 'WP_011861547.1;YP_001088922.2', 'cdf:CD630_24100;pdc:CDIF630_02652', '272563.120.peg.2544']\n",
      "['Q180W4_CLOD6', '2.7.8.7', 'WP_011861979.1;YP_001089986.1', 'cdf:CD630_34660;pdc:CDIF630_03777', '272563.120.peg.3663']\n",
      "['Q18BJ1_CLOD6', '2.7.8.5', 'WP_009889106.1;YP_001087823.1', 'cdf:CD630_13270;pdc:CDIF630_01483', '272563.120.peg.1387']\n",
      "['COBS_CLOD6', '2.7.8.26', 'WP_003437766.1;YP_001089957.1', 'cdf:CD630_34370;pdc:CDIF630_03746', '272563.120.peg.3631']\n",
      "['MRAY_CLOD6', '2.7.8.13', 'WP_003427000.1;YP_001089165.1', 'cdf:CD630_26540', '272563.8.peg.2778']\n",
      "['Q180Q2_CLOD6', '2.7.8.-', 'WP_003429110.1;YP_001089924.1', 'cdf:CD630_34040;pdc:CDIF630_03709', '272563.120.peg.3598']\n",
      "['Q18CT1_CLOD6', '2.7.8.-', 'WP_003435010.1;YP_001086662.1', 'cdf:CD630_01920;pdc:CDIF630_00313', '272563.120.peg.207']\n",
      "['Q180Y5_CLOD6', '2.7.7.9', 'WP_009892053.1;YP_001090009.1', 'cdf:CD630_34880;pdc:CDIF630_03801', '272563.120.peg.3688']\n",
      "['Q183E9_CLOD6', '2.7.7.9', 'WP_009890959.1;YP_001089226.1', 'cdf:CD630_27150;pdc:CDIF630_02972', '272563.120.peg.2859']\n",
      "['Q180X9_CLOD6', '2.7.7.87', 'WP_009892045.1;YP_001090002.2', 'cdf:CD630_34820;pdc:CDIF630_03793', '272563.120.peg.3680']\n",
      "['DISA_CLOD6', '2.7.7.85', 'WP_004453975.1;YP_001086491.1', 'cdf:CD630_00280;pdc:CDIF630_00090', '272563.120.peg.32']\n",
      "['Q18CK0_CLOD6', '2.7.7.85', 'WP_003420697.1;YP_001086578.1', 'cdf:CD630_01100;pdc:CDIF630_00225', '272563.120.peg.122']\n",
      "['PNP_CLOD6', '2.7.7.8', 'WP_003438316.1;YP_001087814.1', 'cdf:CD630_13180;pdc:CDIF630_01474', '272563.120.peg.1378']\n",
      "['Q186R9_CLOD6', '2.7.7.77', 'WP_011861311.1;YP_001088213.1', 'cdf:CD630_17090', '272563.8.peg.1788']\n",
      "['DPO4_CLOD6', '2.7.7.7', 'WP_011860981.1;YP_001087373.1', 'cdf:CD630_08990;pdc:CDIF630_01019', '272563.120.peg.923']\n",
      "['Q181E8_CLOD6', '2.7.7.7', 'WP_003435878.1;YP_001090070.1', 'cdf:CD630_35490;pdc:CDIF630_03864', '272563.120.peg.3750']\n",
      "['Q180P3_CLOD6', '2.7.7.7', 'WP_003436251.1;YP_001089916.1', 'cdf:CD630_33960;pdc:CDIF630_03701', '272563.120.peg.3590']\n",
      "['Q181X9_CLOD6', '2.7.7.7', 'WP_004454691.1;YP_001088906.1', 'cdf:CD630_23940;pdc:CDIF630_02636', '272563.120.peg.2528']\n",
      "['Q18C99_CLOD6', '2.7.7.7', 'WP_003421622.1;YP_001086479.1', 'cdf:CD630_00160;pdc:CDIF630_00030', '272563.120.peg.16']\n",
      "['Q18BG8_CLOD6', '2.7.7.7', 'WP_003438301.1;YP_001087800.1', 'cdf:CD630_13050;pdc:CDIF630_01460', '272563.120.peg.1364']\n",
      "['Q18AY7_CLOD6', '2.7.7.7', 'WP_009896068.1;YP_001087619.1', 'cdf:CD630_11280;pdc:CDIF630_01275', '272563.120.peg.1176']\n",
      "['Q183X4_CLOD6', '2.7.7.65', 'WP_003439713.1;YP_001089400.1', 'cdf:CD630_28870', '272563.8.peg.3016']\n",
      "['Q189V1_CLOD6', '2.7.7.65', 'WP_003437448.1;YP_001087234.1', 'cdf:CD630_07570;pdc:CDIF630_00875', '272563.120.peg.779']\n",
      "['Q183J0_CLOD6', '2.7.7.65', 'WP_011861669.1;YP_001089267.1', 'cdf:CD630_27540;pdc:CDIF630_03017', '272563.120.peg.2901']\n",
      "['Q18C34_CLOD6', '2.7.7.65', 'WP_009902545.1;YP_001088016.1', 'cdf:CD630_15150;pdc:CDIF630_01681', '272563.120.peg.1586']\n",
      "['Q18BU3_CLOD6', '2.7.7.65', 'WP_011861208.1;YP_001087923.1', 'cdf:CD630_14210;pdc:CDIF630_01583', '272563.120.peg.1489']\n",
      "['Q18BU4_CLOD6', '2.7.7.65', 'WP_009889222.1;YP_001087922.1', 'cdf:CD630_14200;pdc:CDIF630_01582', '272563.120.peg.1488']\n",
      "['Q18BU0_CLOD6', '2.7.7.65', 'WP_009902497.1;YP_001087921.1', 'cdf:CD630_14190;pdc:CDIF630_01581', '272563.120.peg.1487']\n",
      "['Q18CC7_CLOD6', '2.7.7.63', 'WP_004453953.1;YP_001086509.1', 'cdf:CD630_00460;pdc:CDIF630_00108', '272563.120.peg.50']\n",
      "['ISPD_CLOD6', '2.7.7.60', 'WP_011860629.1;YP_001086510.1', 'cdf:CD630_00470;pdc:CDIF630_00109', '272563.120.peg.51']\n",
      "['RPOB_CLOD6', '2.7.7.6', 'WP_003436174.1;YP_001086530.1', 'cdf:CD630_00660', '272563.8.peg.77']\n",
      "['Q18CD8_CLOD6', '2.7.7.6', 'WP_003421193.1;YP_001086520.1', 'cdf:CD630_00570', '272563.8.peg.67']\n",
      "['RPOC_CLOD6', '2.7.7.6', 'WP_003429485.1;YP_001086531.1', 'cdf:CD630_00670;pdc:CDIF630_00133', '272563.120.peg.73']\n",
      "['RPOZ_CLOD6', '2.7.7.6', 'WP_003431167.1;YP_001089104.1', 'cdf:CD630_25871;pdc:CDIF630_02841', '272563.120.peg.2729']\n",
      "['RPOA_CLOD6', '2.7.7.6', 'WP_003427733.1;YP_001086566.1', 'cdf:CD630_00980;pdc:CDIF630_00168', '272563.120.peg.108']\n",
      "['Q188V0_CLOD6', '2.7.7.49', 'WP_002335621.1;YP_001086980.1', 'cdf:CD630_05060', '272563.8.peg.530']\n",
      "['Q185S2_CLOD6', '2.7.7.41', 'WP_003434000.1;YP_001088648.1', 'cdf:CD630_21350;pdc:CDIF630_02365', '272563.120.peg.2254']\n",
      "['Q182P5_CLOD6', '2.7.7.3', 'WP_003416314.1;YP_001089073.1', 'cdf:CD630_25580;pdc:CDIF630_02809', '272563.120.peg.2697']\n",
      "['GLGC_CLOD6', '2.7.7.27', 'WP_003418638.1;YP_001087356.1', 'cdf:CD630_08820;pdc:CDIF630_01002', '272563.120.peg.906']\n",
      "['GLMU_CLOD6', '2.7.7.23;_2.3.1.157', 'WP_009898727.1;YP_001090036.1', 'cdf:CD630_35150;pdc:CDIF630_03829', '272563.120.peg.3715']\n",
      "['Q183L6_CLOD6', '2.7.7.22', 'WP_003426511.1;YP_001089292.1', 'cdf:CD630_27790;pdc:CDIF630_03042', '272563.120.peg.2927']\n",
      "['Q182L2_CLOD6', '2.7.7.18', 'WP_003416411.1;YP_001089038.1', 'cdf:CD630_25240;pdc:CDIF630_02776', '272563.120.peg.2664']\n",
      "['Q18BK5_CLOD6', '2.7.7.12', 'WP_003438360.1;YP_001087836.1', 'cdf:CD630_13400;pdc:CDIF630_01496', '272563.120.peg.1400']\n",
      "['Q18BX2_CLOD6', '2.7.7.-', 'WP_009896376.1;YP_001087955.2', 'cdf:CD630_14540;pdc:CDIF630_01618', '272563.120.peg.1523']\n",
      "['Q18D82_CLOD6', '2.7.6.5', 'WP_003417310.1;YP_001086814.1', 'cdf:CD630_03450;pdc:CDIF630_00473', '272563.120.peg.364']\n",
      "['Q183H8_CLOD6', '2.7.6.5', 'WP_003422921.1;YP_001089256.1', 'cdf:CD630_27440;pdc:CDIF630_03007', '272563.120.peg.2891']\n",
      "['Q18BX4_CLOD6', '2.7.6.3', 'WP_003438641.1;YP_001087953.1', 'cdf:CD630_14520;pdc:CDIF630_01616', '272563.120.peg.1521']\n",
      "['Q182R4_CLOD6', '2.7.6.2', 'WP_004454892.1;YP_001089090.1', 'cdf:CD630_25740;pdc:CDIF630_02827', '272563.120.peg.2715']\n",
      "['Q181B3_CLOD6', '2.7.6.1', 'WP_003421448.1;YP_001090035.1', 'cdf:CD630_35140;pdc:CDIF630_03828', '272563.120.peg.3714']\n",
      "['Q181E7_CLOD6', '2.7.4.9', 'WP_009892113.1;YP_001090071.1', 'cdf:CD630_35500;pdc:CDIF630_03865', '272563.120.peg.3751']\n",
      "['KGUA_CLOD6', '2.7.4.8', 'WP_003431170.1;YP_001089105.1', 'cdf:CD630_25880;pdc:CDIF630_02842', '272563.120.peg.2730']\n",
      "['Q186F5_CLOD6', '2.7.4.7', 'WP_003436492.1;YP_001088100.1', 'cdf:CD630_15990;pdc:CDIF630_01774', '272563.120.peg.1675']\n",
      "['KAD_CLOD6', '2.7.4.3', 'WP_011860634.1;YP_001086558.1', 'cdf:CD630_00910;pdc:CDIF630_00160', '272563.120.peg.100']\n",
      "['KCY_CLOD6', '2.7.4.25', 'WP_003439454.1;YP_001088321.1', 'cdf:CD630_18160;pdc:CDIF630_02016', '272563.120.peg.1910']\n",
      "['PYRH_CLOD6', '2.7.4.22', 'WP_003430597.1;YP_001088652.1', 'cdf:CD630_21380;pdc:CDIF630_02369', '272563.120.peg.2258']\n",
      "['Q183J1_CLOD6', '2.7.3.9', 'WP_003431795.1;YP_001089268.1', 'cdf:CD630_27550;pdc:CDIF630_03018', '272563.120.peg.2902']\n",
      "['Q188S3_CLOD6', '2.7.3.-', 'WP_003434751.1;YP_001086956.1', 'cdf:CD630_04820;pdc:CDIF630_00610', '272563.120.peg.507']\n",
      "['Q187D8_CLOD6', '2.7.3.-', 'WP_011861344.1;YP_001088334.1', 'cdf:CD630_18290;pdc:CDIF630_02031', '272563.120.peg.1924']\n",
      "['ARGB_CLOD6', '2.7.2.8', 'WP_003435171.1;YP_001088546.1', 'cdf:CD630_20320;pdc:CDIF630_02253', '272563.120.peg.2143']\n",
      "['Q182B0_CLOD6', '2.7.2.7', 'WP_009890632.1;YP_001088938.1', 'cdf:CD630_24260;pdc:CDIF630_02668', '272563.120.peg.2560']\n",
      "['Q181W4_CLOD6', '2.7.2.7', 'WP_004454677.1;YP_001088891.1', 'cdf:CD630_23790;pdc:CDIF630_02619', '272563.120.peg.2512']\n",
      "['Q18CK3_CLOD6', '2.7.2.7', 'WP_003420702.1;YP_001086582.1', 'cdf:CD630_01130;pdc:CDIF630_00228', '272563.120.peg.125']\n",
      "['Q188C0_CLOD6', '2.7.2.4', 'WP_003424287.1;YP_001088568.1', 'cdf:CD630_20540;pdc:CDIF630_02276', '272563.120.peg.2168']\n",
      "['Q18BI5_CLOD6', '2.7.2.4', 'WP_003438324.1;YP_001087818.1', 'cdf:CD630_13220;pdc:CDIF630_01478', '272563.120.peg.1382']\n",
      "['PGK_CLOD6', '2.7.2.3', 'WP_003429565.1;YP_001089690.1', 'cdf:CD630_31730;pdc:CDIF630_03465', '272563.120.peg.3348']\n",
      "['ACKA_CLOD6', '2.7.2.1', 'WP_003438096.1;YP_001087667.1', 'cdf:CD630_11750;pdc:CDIF630_01323', '272563.120.peg.1225']\n",
      "['Q18CB0_CLOD6', '2.7.14.1', 'WP_004453979.1;YP_001086488.1', 'cdf:CD630_00250;pdc:CDIF630_00087', '272563.120.peg.29']\n",
      "['Q18A16_CLOD6', '2.7.13.3', 'WP_011860966.1;YP_001087298.1', 'cdf:CD630_08210', '272563.8.peg.859']\n",
      "['Q188Y5_CLOD6', '2.7.13.3', 'WP_011860851.1;YP_001087015.1', 'cdf:CD630_05390', '272563.8.peg.565']\n",
      "['Q188S9_CLOD6', '2.7.13.3', 'WP_011860833.1;YP_001086959.1', 'cdf:CD630_04850', '272563.8.peg.509']\n",
      "['Q181K0_CLOD6', '2.7.13.3', 'WP_011862007.1;YP_001090123.1', 'cdf:CD630_35990', '272563.8.peg.3771']\n",
      "['Q183F6_CLOD6', '2.7.13.3', 'WP_004454160.1;YP_001089233.1', 'cdf:CD630_27210', '272563.8.peg.2847']\n",
      "['Q182Q8_CLOD6', '2.7.13.3', 'WP_011861604.1;YP_001089086.2', 'cdf:CD630_25700', '272563.8.peg.2694']\n",
      "['Q182H8_CLOD6', '2.7.13.3', 'WP_004454784.1;YP_001089006.1', 'cdf:CD630_24920;pdc:CDIF630_02739', '272563.8.peg.2614']\n",
      "['Q185J1_CLOD6', '2.7.13.3', 'WP_011861523.1;YP_001088832.1', 'cdf:CD630_23170', '272563.8.peg.2434']\n",
      "['Q18C12_CLOD6', '2.7.13.3', 'WP_011861234.1;YP_001087993.1', 'cdf:CD630_14920', '272563.8.peg.1564']\n",
      "['Q181I8_CLOD6', '2.7.13.3', 'WP_003434897.1;YP_001090110.1', 'cdf:CD630_35860;pdc:CDIF630_03907', '272563.120.peg.3790']\n",
      "['Q187S1_CLOD6', '2.7.13.3', 'WP_011861402.1;YP_001088466.1', 'cdf:CD630_19600;pdc:CDIF630_02165', '272563.120.peg.2057']\n",
      "['Q187R5_CLOD6', '2.7.13.3', 'WP_009897088.1;YP_001088462.1', 'cdf:CD630_19560;pdc:CDIF630_02161', '272563.120.peg.2053']\n",
      "['Q187J0_CLOD6', '2.7.13.3', 'WP_009889841.1;YP_001088386.1', 'cdf:CD630_18770;pdc:CDIF630_02080', '272563.120.peg.1973']\n",
      "['Q186U6_CLOD6', '2.7.13.3', 'WP_011861320.1;YP_001088242.1', 'cdf:CD630_17380;pdc:CDIF630_01929', '272563.120.peg.1825']\n",
      "['Q186P7_CLOD6', '2.7.13.3', 'WP_004454470.1;YP_001088192.1', 'cdf:CD630_16890;pdc:CDIF630_01874', '272563.120.peg.1772']\n",
      "['Q186I0_CLOD6', '2.7.13.3', 'WP_011861275.1;YP_001088126.1', 'cdf:CD630_16250;pdc:CDIF630_01802', '272563.120.peg.1703']\n",
      "['Q186D3_CLOD6', '2.7.13.3', 'WP_003436553.1;YP_001088079.1', 'cdf:CD630_15790;pdc:CDIF630_01748', '272563.120.peg.1652']\n",
      "['PDRP_CLOD6', '2.7.11.32;_2.7.4.27', 'WP_004454708.1;YP_001088923.1', 'cdf:CD630_24110', '272563.8.peg.2529']\n",
      "['Q189W4_CLOD6', '2.7.11.1', 'WP_003418415.1;YP_001087248.1', 'cdf:CD630_07710;pdc:CDIF630_00889', '272563.120.peg.793']\n",
      "['Q180Q7_CLOD6', '2.7.11.-;_2.7.4.-', 'WP_003416935.1;YP_001089929.1', 'cdf:CD630_34090;pdc:CDIF630_03714', '272563.120.peg.3603']\n",
      "['Q184U6_CLOD6', '2.7.1.90', 'WP_009891452.1;YP_001089624.1', 'cdf:CD630_31100;pdc:CDIF630_03394', '272563.120.peg.3276']\n",
      "['Q187E9_CLOD6', '2.7.1.71', 'WP_003439413.1;YP_001088344.1', 'cdf:CD630_18380;pdc:CDIF630_02040', '272563.120.peg.1933']\n",
      "['Q189W2_CLOD6', '2.7.1.69', 'WP_003418410.1', 'pdc:CDIF630_00884', '272563.120.peg.788']\n",
      "['Q189W3_CLOD6', '2.7.1.69', 'WP_003437467.1', 'pdc:CDIF630_00883', '272563.120.peg.787']\n",
      "['Q188T4_CLOD6', '2.7.1.69', 'WP_003426364.1;YP_001086966.1', 'cdf:CD630_04920;pdc:CDIF630_00621', '272563.120.peg.517']\n",
      "['Q188R0_CLOD6', '2.7.1.69', 'WP_011860826.1;YP_001086942.1', 'cdf:CD630_04690;pdc:CDIF630_00597', '272563.120.peg.494']\n",
      "['Q18DC5_CLOD6', '2.7.1.69', 'WP_011860773.1;YP_001086854.1', 'cdf:CD630_03880;pdc:CDIF630_00515', '272563.120.peg.407']\n",
      "['Q181P9_CLOD6', '2.7.1.69', 'WP_003429250.1;YP_001090172.1', 'cdf:CD630_36480;pdc:CDIF630_03972', '272563.120.peg.3858']\n",
      "['Q180C2_CLOD6', '2.7.1.69', 'WP_009898441.1;YP_001089795.1', 'cdf:CD630_32790;pdc:CDIF630_03579', '272563.120.peg.3464']\n",
      "['Q184X5_CLOD6', '2.7.1.69', 'WP_011861843.1;YP_001089652.1', 'cdf:CD630_31370;pdc:CDIF630_03422', '272563.120.peg.3305']\n",
      "['Q184X2_CLOD6', '2.7.1.69', 'WP_011861841.1;YP_001089649.1', 'cdf:CD630_31340;pdc:CDIF630_03419', '272563.120.peg.3302']\n",
      "['Q184W5_CLOD6', '2.7.1.69', 'WP_011861836.1;YP_001089642.1', 'cdf:CD630_31270;pdc:CDIF630_03412', '272563.120.peg.3295']\n",
      "['Q184W3_CLOD6', '2.7.1.69', 'WP_011861834.1;YP_001089640.1', 'cdf:CD630_31250;pdc:CDIF630_03410', '272563.120.peg.3293']\n",
      "['Q184V3_CLOD6', '2.7.1.69', 'WP_011861829.1;YP_001089630.1', 'cdf:CD630_31160;pdc:CDIF630_03400', '272563.120.peg.3283']\n",
      "['Q184T8_CLOD6', '2.7.1.69', 'WP_011861822.1;YP_001089615.1', 'cdf:CD630_31010;pdc:CDIF630_03385', '272563.120.peg.3267']\n",
      "['Q184T4_CLOD6', '2.7.1.69', 'WP_011861819.1;YP_001089611.1', 'cdf:CD630_30970;pdc:CDIF630_03380', '272563.120.peg.3263']\n",
      "['Q184S5_CLOD6', '2.7.1.69', 'WP_009903457.1;YP_001089603.1', 'cdf:CD630_30890;pdc:CDIF630_03372', '272563.120.peg.3255']\n",
      "['Q184S3_CLOD6', '2.7.1.69', 'WP_011861812.1;YP_001089600.1', 'cdf:CD630_30860;pdc:CDIF630_03369', '272563.120.peg.3252']\n",
      "['Q184R2_CLOD6', '2.7.1.69', 'WP_011861805.1;YP_001089589.1', 'cdf:CD630_30750;pdc:CDIF630_03358', '272563.120.peg.3241']\n",
      "['Q184Q6_CLOD6', '2.7.1.69', 'WP_009891383.1;YP_001089582.1', 'cdf:CD630_30680;pdc:CDIF630_03351', '272563.120.peg.3234']\n",
      "['Q184P8_CLOD6', '2.7.1.69', 'WP_011861800.1;YP_001089575.1', 'cdf:CD630_30610;pdc:CDIF630_03344', '272563.120.peg.3227']\n",
      "['Q184L8_CLOD6', '2.7.1.69', 'WP_009893767.1;YP_001089545.1', 'cdf:CD630_30300;pdc:CDIF630_03314', '272563.120.peg.3196']\n",
      "['Q184K5_CLOD6', '2.7.1.69', 'WP_003422575.1;YP_001089531.1', 'cdf:CD630_30140;pdc:CDIF630_03299', '272563.120.peg.3181']\n",
      "['Q18D28_CLOD6', '2.7.1.69', 'WP_003425168.1;YP_001086756.1', 'cdf:CD630_02870;pdc:CDIF630_00411', '272563.120.peg.305']\n",
      "['Q183X2_CLOD6', '2.7.1.69', 'WP_003422688.1;YP_001089397.1', 'cdf:CD630_28840;pdc:CDIF630_03152', '272563.120.peg.3036']\n",
      "['Q183W9_CLOD6', '2.7.1.69', 'WP_003422696.1;YP_001089393.1', 'cdf:CD630_28800;pdc:CDIF630_03148', '272563.120.peg.3032']\n",
      "['Q18D23_CLOD6', '2.7.1.69', 'WP_003435943.1;YP_001086754.1', 'cdf:CD630_02850;pdc:CDIF630_00409', '272563.120.peg.303']\n",
      "['Q183A0_CLOD6', '2.7.1.69', 'WP_011861632.1;YP_001089178.1', 'cdf:CD630_26670;pdc:CDIF630_02921', '272563.120.peg.2810']\n",
      "['Q183A1_CLOD6', '2.7.1.69', 'WP_011861631.1;YP_001089177.1', 'cdf:CD630_26660;pdc:CDIF630_02920', '272563.120.peg.2809']\n",
      "['Q182Q5_CLOD6', '2.7.1.69', 'WP_004454881.1;YP_001089083.1', 'cdf:CD630_25670;pdc:CDIF630_02820', '272563.120.peg.2708']\n",
      "['Q182P2_CLOD6', '2.7.1.69', 'WP_004454867.1;YP_001089070.1', 'cdf:CD630_25550;pdc:CDIF630_02806', '272563.120.peg.2694']\n",
      "['Q182J6_CLOD6', '2.7.1.69', 'WP_009897702.1;YP_001089025.1', 'cdf:CD630_25120;pdc:CDIF630_02761', '272563.120.peg.2651']\n",
      "['Q182J8_CLOD6', '2.7.1.69', 'WP_009890719.1;YP_001089023.1', 'cdf:CD630_25100;pdc:CDIF630_02759', '272563.120.peg.2649']\n",
      "['Q182H4_CLOD6', '2.7.1.69', 'WP_003416512.1;YP_001089001.1', 'cdf:CD630_24870;pdc:CDIF630_02734', '272563.120.peg.2626']\n",
      "['Q181Z8_CLOD6', '2.7.1.69', 'WP_004454715.1;YP_001088929.1', 'cdf:CD630_24170;pdc:CDIF630_02659', '272563.120.peg.2551']\n",
      "['Q185K8_CLOD6', '2.7.1.69', 'WP_011861530.1;YP_001088849.1', 'cdf:CD630_23340;pdc:CDIF630_02573', '272563.120.peg.2465']\n",
      "['Q185K6_CLOD6', '2.7.1.69', 'WP_004454636.1;YP_001088847.1', 'cdf:CD630_23320;pdc:CDIF630_02571', '272563.120.peg.2463']\n",
      "['Q185K0_CLOD6', '2.7.1.69', 'WP_011861526.1;YP_001088842.1', 'cdf:CD630_23270;pdc:CDIF630_02565', '272563.120.peg.2458']\n",
      "['Q185K1_CLOD6', '2.7.1.69', 'WP_003430711.1;YP_001088841.1', 'cdf:CD630_23260;pdc:CDIF630_02564', '272563.120.peg.2457']\n",
      "['Q185E0_CLOD6', '2.7.1.69', 'WP_009897443.1;YP_001088781.1', 'cdf:CD630_22690;pdc:CDIF630_02503', '272563.120.peg.2396']\n",
      "['Q18CU9_CLOD6', '2.7.1.69', 'WP_003435043.1;YP_001086676.1', 'cdf:CD630_02080;pdc:CDIF630_00330', '272563.120.peg.224']\n",
      "['Q18BK3_CLOD6', '2.7.1.69', 'WP_011861179.1;YP_001087832.1', 'cdf:CD630_13360;pdc:CDIF630_01492', '272563.120.peg.1396']\n",
      "['Q186D9_CLOD6', '2.7.1.59', 'WP_011861264.1;YP_001088086.1', 'cdf:CD630_15850;pdc:CDIF630_01757', '272563.120.peg.1659']\n",
      "['THIM_CLOD6', '2.7.1.50', 'WP_011861269.1;YP_001088101.1', 'cdf:CD630_16000;pdc:CDIF630_01775', '272563.120.peg.1676']\n",
      "['Q189A1_CLOD6', '2.7.1.48', 'WP_003436982.1;YP_001087033.1', 'cdf:CD630_05570;pdc:CDIF630_00670', '272563.120.peg.567']\n",
      "['Q180P2_CLOD6', '2.7.1.40', 'WP_003436250.1;YP_001089914.1', 'cdf:CD630_33940;pdc:CDIF630_03699', '272563.120.peg.3588']\n",
      "['Q187B6_CLOD6', '2.7.1.4', 'WP_003439483.1;YP_001088312.1', 'cdf:CD630_18060;pdc:CDIF630_02005', '272563.120.peg.1899']\n",
      "['Q185Q8_CLOD6', '2.7.1.39', 'WP_009897262.1;YP_001088633.2', 'cdf:CD630_21190;pdc:CDIF630_02348', '272563.120.peg.2237']\n",
      "['Q18BM1_CLOD6', '2.7.1.35', 'WP_003428191.1;YP_001087849.1', 'cdf:CD630_13530;pdc:CDIF630_01509', '272563.120.peg.1414']\n",
      "['COAX_CLOD6', '2.7.1.33', 'WP_011861999.1;YP_001090078.1', 'cdf:CD630_35550;pdc:CDIF630_03875', '272563.120.peg.3758']\n",
      "['Q184S0_CLOD6', '2.7.1.31', 'WP_011861810.1;YP_001089598.1', 'cdf:CD630_30840;pdc:CDIF630_03367', '272563.120.peg.3250']\n",
      "['Q189T3_CLOD6', '2.7.1.30', 'WP_009902132.1;YP_001087218.1', 'cdf:CD630_07410;pdc:CDIF630_00857', '272563.120.peg.762']\n",
      "['Q18BQ0_CLOD6', '2.7.1.30', 'WP_011861192.1;YP_001087882.1', 'cdf:CD630_13820;pdc:CDIF630_01542', '272563.120.peg.1447']\n",
      "['Q18BI2_CLOD6', '2.7.1.26;_2.7.7.2', 'WP_003438310.1;YP_001087810.1', 'cdf:CD630_13150;pdc:CDIF630_01471', '272563.120.peg.1375']\n",
      "['Q18AZ0_CLOD6', '2.7.1.24', 'WP_003438041.1;YP_001087620.1', 'cdf:CD630_11290;pdc:CDIF630_01276', '272563.120.peg.1177']\n",
      "['Q18AP9_CLOD6', '2.7.1.23', 'WP_003418878.1;YP_001087530.1', 'cdf:CD630_10490;pdc:CDIF630_01189', '272563.120.peg.1089']\n",
      "['Q182H0_CLOD6', '2.7.1.180', 'WP_009897671.1;YP_001088997.1', 'cdf:CD630_24830;pdc:CDIF630_02730', '272563.120.peg.2622']\n",
      "['ANMK_CLOD6', '2.7.1.170', 'WP_011861795.1;YP_001089566.2', 'cdf:CD630_30510;pdc:CDIF630_03335', '272563.120.peg.3217']\n",
      "['Q184Q1_CLOD6', '2.7.1.17', 'WP_009898129.1;YP_001089579.1', 'cdf:CD630_30650;pdc:CDIF630_03348', '272563.120.peg.3231']\n",
      "['Q180T7_CLOD6', '2.7.1.156;_2.7.7.62', 'WP_009891982.1;YP_001089958.1', 'cdf:CD630_34380;pdc:CDIF630_03747', '272563.120.peg.3632']\n",
      "['Q18D39_CLOD6', '2.7.1.15', 'WP_011860720.1;YP_001086768.1', 'cdf:CD630_02990;pdc:CDIF630_00424', '272563.120.peg.318']\n",
      "['ISPE_CLOD6', '2.7.1.148', 'WP_009894070.1;YP_001090089.1', 'cdf:CD630_35660;pdc:CDIF630_03886', '272563.120.peg.3769']\n",
      "['Q180U9_CLOD6', '2.7.1.144', 'WP_011861975.1;YP_001089971.1', 'cdf:CD630_34510;pdc:CDIF630_03762', '272563.120.peg.3647']\n",
      "['Q184R4_CLOD6', '2.7.1.144', 'WP_009891391.1;YP_001089590.1', 'cdf:CD630_30760;pdc:CDIF630_03359', '272563.120.peg.3242']\n",
      "['Q185E1_CLOD6', '2.7.1.144', 'WP_011861506.1;YP_001088782.1', 'cdf:CD630_22700;pdc:CDIF630_02504', '272563.120.peg.2397']\n",
      "['PFKA_CLOD6', '2.7.1.11', 'WP_003429132.1;YP_001089915.1', 'cdf:CD630_33950;pdc:CDIF630_03700', '272563.120.peg.3589']\n",
      "['Q189W1_CLOD6', '2.7.1.-', 'WP_003437468.1;YP_001087244.1', 'cdf:CD630_07670;pdc:CDIF630_00885', '272563.120.peg.789']\n",
      "['Q182E6_CLOD6', '2.7.1.-', 'WP_003416574.1;YP_001088973.1', 'cdf:CD630_24590;pdc:CDIF630_02705', '272563.120.peg.2597']\n",
      "['Q182C4_CLOD6', '2.7.1.-', 'WP_003416641.1;YP_001088952.1', 'cdf:CD630_24390;pdc:CDIF630_02683', '272563.120.peg.2575']\n",
      "['Q182A1_CLOD6', '2.7.1.-', 'WP_003430855.1;YP_001088926.1', 'cdf:CD630_24140;pdc:CDIF630_02656', '272563.120.peg.2548']\n",
      "['Q185B0_CLOD6', '2.7.1.-', 'WP_003433828.1;YP_001088750.1', 'cdf:CD630_22360;pdc:CDIF630_02469', '272563.120.peg.2363']\n",
      "['Q189C8_CLOD6', '2.7.-.-', 'WP_011860863.1;YP_001087057.2', 'cdf:CD630_05820;pdc:CDIF630_00695', '272563.120.peg.592']\n",
      "['Q181P2_CLOD6', '2.7.-.-', 'WP_011862023.1;YP_001090164.1', 'cdf:CD630_36400;pdc:CDIF630_03964', '272563.120.peg.3850']\n",
      "['Q185Y8_CLOD6', '2.6.1.9', 'WP_009897361.1;YP_001088714.1', 'cdf:CD630_22000;pdc:CDIF630_02433', '272563.120.peg.2323']\n",
      "['Q18C70_CLOD6', '2.6.1.9', 'WP_009889400.1;YP_001088050.1', 'cdf:CD630_15490;pdc:CDIF630_01718', '272563.120.peg.1623']\n",
      "['Q18BW6_CLOD6', '2.6.1.85;_4.1.3.27', 'WP_011861216.1;YP_001087946.1', 'cdf:CD630_14450;pdc:CDIF630_01609', '272563.120.peg.1514']\n",
      "['Q18BW5_CLOD6', '2.6.1.85', 'WP_011861217.1;YP_001087947.1', 'cdf:CD630_14460;pdc:CDIF630_01610', '272563.120.peg.1515']\n",
      "['Q180J8_CLOD6', '2.6.1.39', 'WP_009903720.1;YP_001089870.1', 'cdf:CD630_33530;pdc:CDIF630_03653', '272563.120.peg.3541']\n",
      "['Q18CK9_CLOD6', '2.6.1.16', 'WP_003432446.1;YP_001086589.1', 'cdf:CD630_01200;pdc:CDIF630_00236', '272563.120.peg.132']\n",
      "['Q187Z9_CLOD6', '2.6.1.11', 'WP_011861434.1;YP_001088545.1', 'cdf:CD630_20310;pdc:CDIF630_02252', '272563.120.peg.2142']\n",
      "['Q183R4_CLOD6', '2.6.1.-', 'WP_011861705.1;YP_001089341.1', 'cdf:CD630_28280', '272563.8.peg.2955']\n",
      "['Q181R8_CLOD6', '2.6.1.-', 'WP_003429271.1;YP_001090189.1', 'cdf:CD630_36640;pdc:CDIF630_03993', '272563.120.peg.3875']\n",
      "['Q180T8_CLOD6', '2.6.1.-', 'WP_003437779.1;YP_001089961.1', 'cdf:CD630_34410;pdc:CDIF630_03752', '272563.120.peg.3637']\n",
      "['Q182L9_CLOD6', '2.6.1.-', 'WP_011861586.1;YP_001089046.1', 'cdf:CD630_25320;pdc:CDIF630_02784', '272563.120.peg.2672']\n",
      "['Q182K1_CLOD6', '2.6.1.-', 'WP_004454813.1;YP_001089028.1', 'cdf:CD630_25150;pdc:CDIF630_02764', '272563.120.peg.2654']\n",
      "['Q182A9_CLOD6', '2.6.1.-', 'WP_004454722.1;YP_001088936.1', 'cdf:CD630_24240;pdc:CDIF630_02666', '272563.120.peg.2558']\n",
      "['Q185U6_CLOD6', '2.6.1.-', 'WP_003424572.1;YP_001088671.1', 'cdf:CD630_21580;pdc:CDIF630_02389', '272563.120.peg.2278']\n",
      "['Q18BK6_CLOD6', '2.6.1.-', 'WP_003438357.1;YP_001087835.1', 'cdf:CD630_13390;pdc:CDIF630_01495', '272563.120.peg.1399']\n",
      "['Q18CJ7_CLOD6', '2.6.1.-', 'WP_009892490.1;YP_001086575.1', 'cdf:CD630_01070;pdc:CDIF630_00203', '272563.120.peg.118']\n",
      "['Q18AJ4_CLOD6', '2.6.1.-', 'WP_009888750.1;YP_001087475.1', 'cdf:CD630_09940;pdc:CDIF630_01130', '272563.120.peg.1033']\n",
      "['Q186Q6_CLOD6', '2.5.1.9', 'WP_009889593.1;YP_001088202.1', 'cdf:CD630_16990;pdc:CDIF630_01884', '272563.120.peg.1782']\n",
      "['Q186Q5_CLOD6', '2.5.1.78', 'WP_004454467.1;YP_001088200.1', 'cdf:CD630_16970;pdc:CDIF630_01882', '272563.120.peg.1780']\n",
      "['MIAA_CLOD6', '2.5.1.75', 'WP_011861408.1;YP_001088481.2', 'cdf:CD630_19750;pdc:CDIF630_02180', '272563.120.peg.2072']\n",
      "['NADA_CLOD6', '2.5.1.72', 'WP_004454668.1;YP_001088884.1', 'cdf:CD630_23720;pdc:CDIF630_02612', '272563.120.peg.2505']\n",
      "['Q18CL1_CLOD6', '2.5.1.7', 'WP_011860640.1;YP_001086592.1', 'cdf:CD630_01230', '272563.8.peg.139']\n",
      "['HEM3_CLOD6', '2.5.1.61', 'WP_011861963.1;YP_001089941.1', 'cdf:CD630_34210;pdc:CDIF630_03729', '272563.120.peg.3615']\n",
      "['METK_CLOD6', '2.5.1.6', 'WP_003432462.1;YP_001086599.1', 'cdf:CD630_01300;pdc:CDIF630_00247', '272563.120.peg.142']\n",
      "['Q187E1_CLOD6', '2.5.1.54', 'WP_003439427.1;YP_001088338.1', 'cdf:CD630_18320;pdc:CDIF630_02034', '272563.120.peg.1927']\n",
      "['Q18BX3_CLOD6', '2.5.1.54', 'WP_003419887.1;YP_001087954.1', 'cdf:CD630_14530;pdc:CDIF630_01617', '272563.120.peg.1522']\n",
      "['Q187D4_CLOD6', '2.5.1.49', 'WP_003439438.1;YP_001088330.1', 'cdf:CD630_18250;pdc:CDIF630_02027', '272563.120.peg.1920']\n",
      "['Q186M0_CLOD6', '2.5.1.47', 'WP_004454499.1;YP_001088167.1', 'cdf:CD630_16650;pdc:CDIF630_01849', '272563.120.peg.1747']\n",
      "['Q186F0_CLOD6', '2.5.1.47', 'WP_003436514.1;YP_001088095.1', 'cdf:CD630_15940;pdc:CDIF630_01767', '272563.120.peg.1669']\n",
      "['Q183J8_CLOD6', '2.5.1.31', 'WP_003422884.1;YP_001089275.1', 'cdf:CD630_27620;pdc:CDIF630_03025', '272563.120.peg.2909']\n",
      "['Q186R4_CLOD6', '2.5.1.3', 'WP_009889638.1;YP_001088210.1', 'cdf:CD630_17060;pdc:CDIF630_01894', '272563.120.peg.1790']\n",
      "['THIE_CLOD6', '2.5.1.3', 'WP_003436488.1;YP_001088102.1', 'cdf:CD630_16010;pdc:CDIF630_01776', '272563.120.peg.1677']\n",
      "['AROA_CLOD6', '2.5.1.19', 'WP_011861345.1;YP_001088340.1', 'cdf:CD630_18340;pdc:CDIF630_02036', '272563.120.peg.1929']\n",
      "['Q187M9_CLOD6', '2.5.1.17', 'WP_011861391.1;YP_001088426.1', 'cdf:CD630_19190;pdc:CDIF630_02123', '272563.120.peg.2015']\n",
      "['SPEE_CLOD6', '2.5.1.16', 'WP_009888722.1;YP_001087364.1', 'cdf:CD630_08900;pdc:CDIF630_01010', '272563.120.peg.914']\n",
      "['Q18BX0_CLOD6', '2.5.1.15', 'WP_011861218.1;YP_001087951.1', 'cdf:CD630_14500', '272563.8.peg.1522']\n",
      "['LGT_CLOD6', '2.5.1.145', 'WP_009897834.1;YP_001089170.1', 'cdf:CD630_26590;pdc:CDIF630_02913', '272563.120.peg.2802']\n",
      "['Q18B64_CLOD6', '2.5.1.10', 'WP_009888951.1;YP_001087698.1', 'cdf:CD630_12050;pdc:CDIF630_01354', '272563.120.peg.1256']\n",
      "['Q185S4_CLOD6', '2.5.1.-', 'WP_003424529.1;YP_001088649.1', 'cdf:CD630_21360;pdc:CDIF630_02366', '272563.120.peg.2255']\n",
      "['Q18AI2_CLOD6', '2.5.1.-', 'WP_011861057.1;YP_001087463.1', 'cdf:CD630_09820;pdc:CDIF630_01117', '272563.120.peg.1021']\n",
      "['QUEA_CLOD6', '2.4.99.17', 'WP_011861694.1;YP_001089317.1', 'cdf:CD630_28040;pdc:CDIF630_03069', '272563.120.peg.2953']\n",
      "['UPP_CLOD6', '2.4.2.9', 'WP_003437840.1;YP_001089999.1', 'cdf:CD630_34790;pdc:CDIF630_03790', '272563.120.peg.3677']\n",
      "['PYRR_CLOD6', '2.4.2.9', 'WP_003431186.1;YP_001089112.1', 'cdf:CD630_25950;pdc:CDIF630_02849', '272563.120.peg.2737']\n",
      "['Q17ZX6_CLOD6', '2.4.2.8', 'WP_004454045.1;YP_001089748.1', 'cdf:CD630_32310;pdc:CDIF630_03527', '272563.120.peg.3412']\n",
      "['Q183C4_CLOD6', '2.4.2.8', 'WP_003416006.1;YP_001089202.1', 'cdf:CD630_26910;pdc:CDIF630_02946', '272563.120.peg.2835']\n",
      "['Q181B6_CLOD6', '2.4.2.7', 'WP_003425934.1;YP_001090038.1', 'cdf:CD630_35170;pdc:CDIF630_03831', '272563.120.peg.3717']\n",
      "['APT_CLOD6', '2.4.2.7', 'WP_003426450.1;YP_001089257.1', 'cdf:CD630_27450;pdc:CDIF630_03008', '272563.120.peg.2892']\n",
      "['Q183P1_CLOD6', '2.4.2.29', 'WP_009897970.1;YP_001089315.2', 'cdf:CD630_28020;pdc:CDIF630_03067', '272563.120.peg.2951']\n",
      "['XPT_CLOD6', '2.4.2.22', 'WP_004454633.1;YP_001088845.1', 'cdf:CD630_23300;pdc:CDIF630_02568', '272563.120.peg.2461']\n",
      "['COBT_CLOD6', '2.4.2.21', 'WP_003437772.1;YP_001089959.1', 'cdf:CD630_34390;pdc:CDIF630_03748', '272563.120.peg.3633']\n",
      "['PPNP_CLOD6', '2.4.2.2;_2.4.2.15;_2.4.2.4;_2.4.2.3', 'WP_004454504.1;YP_001088161.1', 'cdf:CD630_16600;pdc:CDIF630_01841', '272563.120.peg.1738']\n",
      "['Q18B84_CLOD6', '2.4.2.2', 'WP_003438162.1;YP_001087718.1', 'cdf:CD630_12250', '272563.8.peg.1284']\n",
      "['Q185P1_CLOD6', '2.4.2.19', 'WP_011861538.1;YP_001088882.1', 'cdf:CD630_23700', '272563.8.peg.2487']\n",
      "['HIS1_CLOD6', '2.4.2.17', 'WP_003436683.1;YP_001088049.1', 'cdf:CD630_15480;pdc:CDIF630_01717', '272563.120.peg.1622']\n",
      "['Q18CV9_CLOD6', '2.4.2.14', 'WP_003425288.1;YP_001086688.1', 'cdf:CD630_02200;pdc:CDIF630_00342', '272563.120.peg.236']\n",
      "['PYRE_CLOD6', '2.4.2.10', 'WP_003420885.1;YP_001086657.1', 'cdf:CD630_01870;pdc:CDIF630_00308', '272563.120.peg.202']\n",
      "['Q18B85_CLOD6', '2.4.2.1', 'WP_011861146.1;YP_001087717.1', 'cdf:CD630_12240;pdc:CDIF630_01373', '272563.120.peg.1275']\n",
      "['Q185X8_CLOD6', '2.4.1.8', 'WP_011861487.1;YP_001088701.1', 'cdf:CD630_21880;pdc:CDIF630_02420', '272563.120.peg.2310']\n",
      "['MURG_CLOD6', '2.4.1.227', 'WP_011861625.1;YP_001089162.1', 'cdf:CD630_26510', '272563.8.peg.2775']\n",
      "['GLGA_CLOD6', '2.4.1.21', 'WP_011860977.1;YP_001087358.1', 'cdf:CD630_08840;pdc:CDIF630_01004', '272563.120.peg.908']\n",
      "['Q182L3_CLOD6', '2.4.1.18', 'WP_011861581.1;YP_001089040.1', 'cdf:CD630_25260;pdc:CDIF630_02778', '272563.120.peg.2666']\n",
      "['Q18B92_CLOD6', '2.4.1.129', 'WP_003438168.1;YP_001087722.1', 'cdf:CD630_12290;pdc:CDIF630_01378', '272563.120.peg.1280']\n",
      "['Q18B08_CLOD6', '2.4.1.129', 'WP_011861128.1;YP_001087640.1', 'cdf:CD630_11480;pdc:CDIF630_01295', '272563.120.peg.1197']\n",
      "['Q182N2_CLOD6', '2.4.1.12', 'WP_011861594.1;YP_001089059.1', 'cdf:CD630_25450;pdc:CDIF630_02797', '272563.120.peg.2685']\n",
      "['Q18B77_CLOD6', '2.4.1.117', 'WP_003438150.1;YP_001087710.1', 'cdf:CD630_12170;pdc:CDIF630_01366', '272563.120.peg.1268']\n",
      "['Q18A78_CLOD6', '2.4.1.1', 'WP_003437168.1;YP_001087359.1', 'cdf:CD630_08850', '272563.8.peg.924']\n",
      "['Q189K5_CLOD6', '2.4.1.-', 'WP_011860904.1;YP_001087137.1', 'cdf:CD630_06630;pdc:CDIF630_00776', '272563.120.peg.680']\n",
      "['Q189K3_CLOD6', '2.4.1.-', 'WP_009902069.1;YP_001087135.1', 'cdf:CD630_06600;pdc:CDIF630_00773', '272563.120.peg.677']\n",
      "['Q184W7_CLOD6', '2.4.1.-', 'WP_011861838.1;YP_001089644.1', 'cdf:CD630_31290;pdc:CDIF630_03414', '272563.120.peg.3297']\n",
      "['Q183K7_CLOD6', '2.4.1.-', 'WP_003422862.1;YP_001089287.1', 'cdf:CD630_27740;pdc:CDIF630_03037', '272563.120.peg.2922']\n",
      "['Q183K9_CLOD6', '2.4.1.-', 'WP_003426498.1;YP_001089285.1', 'cdf:CD630_27720;pdc:CDIF630_03035', '272563.120.peg.2920']\n",
      "['Q18CX6_CLOD6', '2.4.1.-', 'WP_011860691.1;YP_001086708.1', 'cdf:CD630_02400;pdc:CDIF630_00362', '272563.120.peg.256']\n",
      "['Q18CA1_CLOD6', '2.4.1.-', 'WP_003426064.1;YP_001086482.1', 'cdf:CD630_00190;pdc:CDIF630_00034', '272563.120.peg.20']\n",
      "['Q18AN1_CLOD6', '2.4.1.-', 'WP_003437008.1;YP_001087511.1', 'cdf:CD630_10300;pdc:CDIF630_01166', '272563.120.peg.1070']\n",
      "['Q18AI7_CLOD6', '2.3.3.13', 'WP_003437098.1;YP_001087470.1', 'cdf:CD630_09890;pdc:CDIF630_01125', '272563.120.peg.1028']\n",
      "['Q180G3_CLOD6', '2.3.2.3', 'WP_003429718.1;YP_001089835.1', 'cdf:CD630_33190;pdc:CDIF630_03621', '272563.120.peg.3505']\n",
      "['Q183F7_CLOD6', '2.3.2.3', 'WP_009890975.1;YP_001089235.1', 'cdf:CD630_27230;pdc:CDIF630_02985', '272563.120.peg.2869']\n",
      "['Q188H2_CLOD6', '2.3.2.3', 'WP_003434053.1;YP_001088617.1', 'cdf:CD630_21030;pdc:CDIF630_02327', '272563.120.peg.2218']\n",
      "['Q18C54_CLOD6', '2.3.2.2', 'WP_003436720.1;YP_001088035.1', 'cdf:CD630_15340;pdc:CDIF630_01702', '272563.120.peg.1607']\n",
      "['Q183B0_CLOD6', '2.3.1.9', 'WP_011861636.1;YP_001089187.1', 'cdf:CD630_26760;pdc:CDIF630_02930', '272563.120.peg.2819']\n",
      "['THLA_CLOD6', '2.3.1.9', 'WP_009888868.1;YP_001087540.1', 'cdf:CD630_10590;pdc:CDIF630_01199', '272563.120.peg.1099']\n",
      "['DAPH_CLOD6', '2.3.1.89', 'WP_003428057.1;YP_001089744.1', 'cdf:CD630_32270;pdc:CDIF630_03523', '272563.120.peg.3408']\n",
      "['Q18A66_CLOD6', '2.3.1.79', 'WP_009888695.1;YP_001087346.1', 'cdf:CD630_08720;pdc:CDIF630_00992', '272563.120.peg.896']\n",
      "['Q186M4_CLOD6', '2.3.1.79', 'WP_011861294.1;YP_001088169.1', 'cdf:CD630_16670;pdc:CDIF630_01851', '272563.120.peg.1749']\n",
      "['Q18B76_CLOD6', '2.3.1.57', 'WP_011861143.1;YP_001087705.1', 'cdf:CD630_12120;pdc:CDIF630_01361', '272563.120.peg.1263']\n",
      "['Q189V5_CLOD6', '2.3.1.54', 'WP_003437454.1;YP_001087236.1', 'cdf:CD630_07590;pdc:CDIF630_00877', '272563.120.peg.781']\n",
      "['Q180C6_CLOD6', '2.3.1.54', 'WP_009903701.1;YP_001089798.1', 'cdf:CD630_32820;pdc:CDIF630_03582', '272563.120.peg.3467']\n",
      "['Q18AX8_CLOD6', '2.3.1.54', 'WP_003438022.1;YP_001087609.1', 'cdf:CD630_11200;pdc:CDIF630_01264', '272563.120.peg.1166']\n",
      "['Q187C7_CLOD6', '2.3.1.51', 'WP_003439452.1;YP_001088322.1', 'cdf:CD630_18170;pdc:CDIF630_02017', '272563.120.peg.1911']\n",
      "['Q186S9_CLOD6', '2.3.1.51', 'WP_004454428.1;YP_001088226.1', 'cdf:CD630_17230;pdc:CDIF630_01912', '272563.120.peg.1807']\n",
      "['Q18B46_CLOD6', '2.3.1.39', 'WP_011861134.1;YP_001087674.1', 'cdf:CD630_11810;pdc:CDIF630_01330', '272563.120.peg.1232']\n",
      "['Q188A1_CLOD6', '2.3.1.35;_2.3.1.1', 'WP_003435168.1;YP_001088547.1', 'cdf:CD630_20330;pdc:CDIF630_02254', '272563.120.peg.2144']\n",
      "['Q187D6_CLOD6', '2.3.1.31', 'WP_003439436.1;YP_001088331.1', 'cdf:CD630_18260;pdc:CDIF630_02028', '272563.120.peg.1921']\n",
      "['Q186E9_CLOD6', '2.3.1.30', 'WP_009889486.1;YP_001088096.2', 'cdf:CD630_15950;pdc:CDIF630_01768', '272563.120.peg.1670']\n",
      "['Q18BG4_CLOD6', '2.3.1.286', 'WP_003428245.1;YP_001087798.1', 'cdf:CD630_13030;pdc:CDIF630_01458', '272563.120.peg.1362']\n",
      "['PLSY_CLOD6', '2.3.1.275', 'WP_003426957.1;YP_001089142.1', 'cdf:CD630_26310;pdc:CDIF630_02885', '272563.120.peg.2774']\n",
      "['PLSX_CLOD6', '2.3.1.274', 'WP_003428446.1;YP_001087671.1', 'cdf:CD630_11780;pdc:CDIF630_01327', '272563.120.peg.1229']\n",
      "['TSAD_CLOD6', '2.3.1.234', 'WP_011860650.1;YP_001086621.1', 'cdf:CD630_01520;pdc:CDIF630_00271', '272563.120.peg.165']\n",
      "['Q18B67_CLOD6', '2.3.1.234', 'WP_011861140.1;YP_001087695.1', 'cdf:CD630_12020;pdc:CDIF630_01351', '272563.120.peg.1253']\n",
      "['Q183B8_CLOD6', '2.3.1.222', 'WP_011861641.1;YP_001089194.1', 'cdf:CD630_26830;pdc:CDIF630_02937', '272563.120.peg.2826']\n",
      "['PDUL_CLOD6', '2.3.1.222', 'WP_009889877.1;YP_001088427.1', 'cdf:CD630_19200;pdc:CDIF630_02124', '272563.120.peg.2016']\n",
      "['Q189Q9_CLOD6', '2.3.1.19', 'WP_003436909.1;YP_001087192.1', 'cdf:CD630_07150;pdc:CDIF630_00831', '272563.120.peg.736']\n",
      "['Q182B1_CLOD6', '2.3.1.19', 'WP_004454723.1;YP_001088937.1', 'cdf:CD630_24250;pdc:CDIF630_02667', '272563.120.peg.2559']\n",
      "['Q18CK4_CLOD6', '2.3.1.19', 'WP_003420701.1;YP_001086581.1', 'cdf:CD630_01120;pdc:CDIF630_00227', '272563.120.peg.124']\n",
      "['LIPM_CLOD6', '2.3.1.181', 'WP_003436933.1;YP_001087210.1', 'cdf:CD630_07330;pdc:CDIF630_00849', '272563.120.peg.754']\n",
      "['FABH_CLOD6', '2.3.1.180', 'WP_003438099.1;YP_001087672.1', 'cdf:CD630_11790;pdc:CDIF630_01328', '272563.120.peg.1230']\n",
      "['Q18B43_CLOD6', '2.3.1.179', 'WP_003438103.1;YP_001087677.1', 'cdf:CD630_11840;pdc:CDIF630_01333', '272563.120.peg.1235']\n",
      "['Q184F9_CLOD6', '2.3.1.16', 'WP_011861765.1;YP_001089488.1', 'cdf:CD630_29700;pdc:CDIF630_03255', '272563.120.peg.3138']\n",
      "['Q18CP1_CLOD6', '2.3.1.128', 'WP_003434429.1;YP_001086620.1', 'cdf:CD630_01510;pdc:CDIF630_00270', '272563.120.peg.164']\n",
      "['Q18CC2_CLOD6', '2.3.1.12', 'WP_004453963.1;YP_001086501.1', 'cdf:CD630_00380;pdc:CDIF630_00100', '272563.120.peg.42']\n",
      "['ATSE_CLOD6', '2.3.1.-', 'WP_009902308.1;YP_001087704.1', 'cdf:CD630_12110', '272563.8.peg.1270']\n",
      "['Q182M9_CLOD6', '2.3.1.-', 'WP_011861592.1;YP_001089057.1', 'cdf:CD630_25430;pdc:CDIF630_02795', '272563.120.peg.2683']\n",
      "['Q186B4_CLOD6', '2.3.1.-', 'WP_003433837.1;YP_001088740.1', 'cdf:CD630_22260;pdc:CDIF630_02459', '272563.120.peg.2353']\n",
      "['Q18B56_CLOD6', '2.3.1.-', 'WP_011861137.1;YP_001087683.1', 'cdf:CD630_11900;pdc:CDIF630_01339', '272563.120.peg.1241']\n",
      "['Q18A29_CLOD6', '2.3.-.-', 'WP_003418528.1;YP_001087309.1', 'cdf:CD630_08320;pdc:CDIF630_00951', '272563.120.peg.855']\n",
      "['DXS_CLOD6', '2.2.1.7', 'WP_011861141.1;YP_001087700.1', 'cdf:CD630_12070;pdc:CDIF630_01356', '272563.120.peg.1258']\n",
      "['Q186C0_CLOD6', '2.2.1.6', 'WP_011861253.1;YP_001088066.1', 'cdf:CD630_15660', '272563.8.peg.1641']\n",
      "['Q184U2_CLOD6', '2.2.1.2', 'WP_003440006.1;YP_001089620.1', 'cdf:CD630_31060;pdc:CDIF630_03390', '272563.120.peg.3272']\n",
      "['TAL_CLOD6', '2.2.1.2', 'WP_003423115.1;YP_001088844.1', 'cdf:CD630_23290;pdc:CDIF630_02567', '272563.120.peg.2460']\n",
      "['Q185P9_CLOD6', '2.2.1.2', 'WP_003430626.1;YP_001088624.1', 'cdf:CD630_21100;pdc:CDIF630_02335', '272563.120.peg.2225']\n",
      "['Q180V7_CLOD6', '2.2.1.1', 'WP_003421354.1;YP_001089980.1', 'cdf:CD630_34600;pdc:CDIF630_03771', '272563.120.peg.3657']\n",
      "['Q180V8_CLOD6', '2.2.1.1', 'WP_009903779.1;YP_001089979.1', 'cdf:CD630_34590;pdc:CDIF630_03770', '272563.120.peg.3656']\n",
      "['Q185J6_CLOD6', '2.2.1.1', 'WP_003430704.1;YP_001088837.1', 'cdf:CD630_23220;pdc:CDIF630_02560', '272563.120.peg.2453']\n",
      "['Q185J5_CLOD6', '2.2.1.1', 'WP_003430702.1;YP_001088836.1', 'cdf:CD630_23210;pdc:CDIF630_02559', '272563.120.peg.2452']\n",
      "['Q188D6_CLOD6', '2.10.1.1', 'WP_003435117.1;YP_001088583.1', 'cdf:CD630_20690;pdc:CDIF630_02292', '272563.120.peg.2183']\n",
      "['Q18AT6_CLOD6', '2.10.1.1', 'WP_009902263.1;YP_001087565.1', 'cdf:CD630_10810;pdc:CDIF630_01225', '272563.120.peg.1124']\n",
      "['Q187Z7_CLOD6', '2.1.3.3', 'WP_011861433.1;YP_001088544.1', 'cdf:CD630_20300;pdc:CDIF630_02251', '272563.120.peg.2141']\n",
      "['PYRB_CLOD6', '2.1.3.2', 'WP_003434504.1;YP_001086654.1', 'cdf:CD630_01840;pdc:CDIF630_00305', '272563.120.peg.199']\n",
      "['ACCD_CLOD6', '2.1.3.15', 'WP_003424064.1;YP_001088443.1', 'cdf:CD630_19370;pdc:CDIF630_02141', '272563.120.peg.2033']\n",
      "['Q187P7_CLOD6', '2.1.3.15', 'WP_009889903.1;YP_001088442.1', 'cdf:CD630_19360;pdc:CDIF630_02140', '272563.120.peg.2032']\n",
      "['FMT_CLOD6', '2.1.2.9', 'WP_011861609.1;YP_001089100.1', 'cdf:CD630_25840;pdc:CDIF630_02837', '272563.120.peg.2725']\n",
      "['PUR9_CLOD6', '2.1.2.3;_3.5.4.10', 'WP_003436063.1;YP_001086691.1', 'cdf:CD630_02230;pdc:CDIF630_00345', '272563.120.peg.239']\n",
      "['Q18A34_CLOD6', '2.1.2.3', 'WP_011860970.1;YP_001087318.1', 'cdf:CD630_08410;pdc:CDIF630_00960', '272563.120.peg.864']\n",
      "['Q18CW1_CLOD6', '2.1.2.2', 'WP_003436064.1;YP_001086690.1', 'cdf:CD630_02220;pdc:CDIF630_00344', '272563.120.peg.238']\n",
      "['PANB_CLOD6', '2.1.2.11', 'WP_003439367.1;YP_001088014.1', 'cdf:CD630_15130;pdc:CDIF630_01679', '272563.120.peg.1584']\n",
      "['Q183S3_CLOD6', '2.1.2.1', 'WP_003436123.1;YP_001089347.1', 'cdf:CD630_28340;pdc:CDIF630_03101', '272563.120.peg.2986']\n",
      "['GLYA_CLOD6', '2.1.2.1', 'WP_011861660.1;YP_001089238.1', 'cdf:CD630_27260;pdc:CDIF630_02988', '272563.120.peg.2872']\n",
      "['Q188Z0_CLOD6', '2.1.1.80', 'WP_003425096.1;YP_001087017.1', 'cdf:CD630_05410;pdc:CDIF630_00654', '272563.120.peg.550']\n",
      "['Q18CS1_CLOD6', '2.1.1.79', 'WP_003434490.1;YP_001086647.1', 'cdf:CD630_01770;pdc:CDIF630_00299', '272563.120.peg.193']\n",
      "['Q183J3_CLOD6', '2.1.1.72', 'WP_003431790.1;YP_001089271.1', 'cdf:CD630_27580;pdc:CDIF630_03021', '272563.120.peg.2905']\n",
      "['Q189E1_CLOD6', '2.1.1.63', 'WP_009902015.1;YP_001087076.1', 'cdf:CD630_06010;pdc:CDIF630_00714', '272563.120.peg.612']\n",
      "['Q186Y0_CLOD6', '2.1.1.63', 'WP_003439552.1;YP_001088276.1', 'cdf:CD630_17700;pdc:CDIF630_01967', '272563.120.peg.1863']\n",
      "['Q187X3_CLOD6', '2.1.1.48', 'WP_011861426.1;YP_001088518.1;YP_001088523.1', 'cdf:CD630_20070;cdf:CD630_20100;pdc:CDIF630_02225;pdc:CDIF630_02229', '272563.120.peg.2115']\n",
      "['Q188J6_CLOD6', '2.1.1.35', 'WP_011860779.1;YP_001086872.2', 'cdf:CD630_04060;pdc:CDIF630_00534', '272563.120.peg.426']\n",
      "['TRMB_CLOD6', '2.1.1.33', 'WP_009889910.1;YP_001088448.2', 'cdf:CD630_19420;pdc:CDIF630_02147', '272563.120.peg.2039']\n",
      "['Q180Y4_CLOD6', '2.1.1.297', 'WP_003437849.1;YP_001090005.1', 'cdf:CD630_34850;pdc:CDIF630_03796', '272563.120.peg.3683']\n",
      "['TRMD_CLOD6', '2.1.1.228', 'WP_003438222.1;YP_001087750.1', 'cdf:CD630_12560;pdc:CDIF630_01409', '272563.120.peg.1314']\n",
      "['Q189B1_CLOD6', '2.1.1.207', 'WP_003429657.1;YP_001087042.1', 'cdf:CD630_05660;pdc:CDIF630_00679', '272563.120.peg.576']\n",
      "['RSMH_CLOD6', '2.1.1.199', 'WP_011861627.1;YP_001089169.1', 'cdf:CD630_26580;pdc:CDIF630_02912', '272563.120.peg.2801']\n",
      "['Q181E6_CLOD6', '2.1.1.198', 'WP_003435873.1;YP_001090067.1', 'cdf:CD630_35460;pdc:CDIF630_03861', '272563.120.peg.3747']\n",
      "['CBID_CLOD6', '2.1.1.195', 'WP_011861964.1;YP_001089949.1', 'cdf:CD630_34290;pdc:CDIF630_03737', '272563.120.peg.3623']\n",
      "['Q182D6_CLOD6', '2.1.1.193', 'WP_003430913.1;YP_001088963.1', 'cdf:CD630_24490;pdc:CDIF630_02694', '272563.120.peg.2586']\n",
      "['RLMN_CLOD6', '2.1.1.192', 'WP_003431147.1;YP_001089096.1', 'cdf:CD630_25800;pdc:CDIF630_02833', '272563.120.peg.2721']\n",
      "['RSMA_CLOD6', '2.1.1.182', 'WP_009892086.1;YP_001090044.1', 'cdf:CD630_35230;pdc:CDIF630_03837', '272563.120.peg.3723']\n",
      "['RLMH_CLOD6', '2.1.1.177', 'WP_003435703.1;YP_001090173.1', 'cdf:CD630_36490;pdc:CDIF630_03973', '272563.120.peg.3859']\n",
      "['Q18CE1_CLOD6', '2.1.1.148', 'WP_003421197.1;YP_001086517.1', 'cdf:CD630_00540;pdc:CDIF630_00118', '272563.120.peg.59']\n",
      "['Q181J6_CLOD6', '2.1.1.13', 'WP_009903830.1;YP_001090120.1', 'cdf:CD630_35960;pdc:CDIF630_03918', '272563.120.peg.3802']\n",
      "['Q180R8_CLOD6', '2.1.1.107;_4.2.1.75', 'WP_009898657.1;YP_001089940.2', 'cdf:CD630_34200;pdc:CDIF630_03728', '272563.120.peg.3614']\n",
      "['Q181A3_CLOD6', '2.1.1.-;_3.4.23.43', 'WP_011861982.1;YP_001090025.1', 'cdf:CD630_35040;pdc:CDIF630_03817', '272563.120.peg.3704']\n",
      "['Q18CE0_CLOD6', '2.1.1.-', 'WP_003425498.1;YP_001086518.1', 'cdf:CD630_00550', '272563.8.peg.65']\n",
      "['Q180S5_CLOD6', '2.1.1.-', 'WP_003437741.1;YP_001089948.1', 'cdf:CD630_34280;pdc:CDIF630_03736', '272563.120.peg.3622']\n",
      "['Q180S6_CLOD6', '2.1.1.-', 'WP_009898664.1;YP_001089947.1', 'cdf:CD630_34270;pdc:CDIF630_03735', '272563.120.peg.3621']\n",
      "['Q180S3_CLOD6', '2.1.1.-', 'WP_009891957.1;YP_001089946.1', 'cdf:CD630_34260;pdc:CDIF630_03734', '272563.120.peg.3620']\n",
      "['Q180S0_CLOD6', '2.1.1.-', 'WP_009903755.1;YP_001089944.1', 'cdf:CD630_34240;pdc:CDIF630_03732', '272563.120.peg.3618']\n",
      "['Q180P1_CLOD6', '2.1.1.-', 'WP_011861959.1;YP_001089913.1', 'cdf:CD630_33930;pdc:CDIF630_03698', '272563.120.peg.3587']\n",
      "['Q182R9_CLOD6', '2.1.1.-', 'WP_011861607.1;YP_001089097.1', 'cdf:CD630_25810;pdc:CDIF630_02834', '272563.120.peg.2722']\n",
      "['Q182D8_CLOD6', '2.1.1.-', 'WP_011861555.1;YP_001088964.1', 'cdf:CD630_24500;pdc:CDIF630_02695', '272563.120.peg.2587']\n",
      "['Q187S5_CLOD6', '2.1.1.-', 'WP_011861405.1;YP_001088469.1', 'cdf:CD630_19630;pdc:CDIF630_02168', '272563.120.peg.2060']\n",
      "['Q18B89_CLOD6', '2.1.1.-', 'WP_003438166.1;YP_001087720.1', 'cdf:CD630_12270;pdc:CDIF630_01376', '272563.120.peg.1278']\n",
      "['Q18AW5_CLOD6', '2.1.1.-', 'WP_011861113.1;YP_001087598.1', 'cdf:CD630_11090;pdc:CDIF630_01255', '272563.120.peg.1156']\n",
      "['Q189V6_CLOD6', '1.97.1.4', 'WP_011860950.1;YP_001087235.1', 'cdf:CD630_07580;pdc:CDIF630_00876', '272563.120.peg.780']\n",
      "['Q180C7_CLOD6', '1.97.1.4', 'WP_003432388.1;YP_001089799.1', 'cdf:CD630_32830;pdc:CDIF630_03583', '272563.120.peg.3468']\n",
      "['Q18AX7_CLOD6', '1.97.1.4', 'WP_009896060.1;YP_001087610.1', 'cdf:CD630_11210;pdc:CDIF630_01265', '272563.120.peg.1167']\n",
      "['HPDA_CLOD6', '1.97.1.-', 'WP_011860651.1;YP_001086624.1', 'cdf:CD630_01550;pdc:CDIF630_00274', '272563.120.peg.168']\n",
      "['Q18CJ8_CLOD6', '1.97.1.-', 'WP_003432598.1;YP_001086577.1', 'cdf:CD630_01090;pdc:CDIF630_00205', '272563.120.peg.120']\n",
      "['Q185V3_CLOD6', '1.8.4.11;_1.8.4.12', 'WP_003433953.1;YP_001088679.1', 'cdf:CD630_21660', '272563.8.peg.2280']\n",
      "['Q186P9_CLOD6', '1.8.1.9', 'WP_011861308.1;YP_001088194.1', 'cdf:CD630_16910', '272563.8.peg.1769']\n",
      "['Q185M7_CLOD6', '1.8.1.9', 'WP_003416869.1;YP_001088869.1', 'cdf:CD630_23560;pdc:CDIF630_02596', '272563.120.peg.2487']\n",
      "['Q185Q5_CLOD6', '1.8.1.9', 'WP_003434033.1;YP_001088631.1', 'cdf:CD630_21170;pdc:CDIF630_02344', '272563.120.peg.2234']\n",
      "['Q189R5_CLOD6', '1.8.1.4', 'WP_003436922.1;YP_001087200.1', 'cdf:CD630_07230;pdc:CDIF630_00839', '272563.120.peg.744']\n",
      "['Q18CC1_CLOD6', '1.8.1.4', 'WP_011860626.1;YP_001086502.1', 'cdf:CD630_00390;pdc:CDIF630_00101', '272563.120.peg.43']\n",
      "['Q182B6_CLOD6', '1.8.1.2', 'WP_009890637.1;YP_001088944.1', 'cdf:CD630_24310;pdc:CDIF630_02674', '272563.120.peg.2566']\n",
      "['Q182M6_CLOD6', '1.8.1.14', 'WP_011861590.1;YP_001089054.1', 'cdf:CD630_25400;pdc:CDIF630_02792', '272563.120.peg.2680']\n",
      "['Q187A7_CLOD6', '1.8.1.14', 'WP_003439500.1;YP_001088303.1', 'cdf:CD630_17970;pdc:CDIF630_01996', '272563.120.peg.1890']\n",
      "['Q185A4_CLOD6', '1.8.1.-', 'WP_003433832.1;YP_001088745.2', 'cdf:CD630_22310;pdc:CDIF630_02464', '272563.120.peg.2358']\n",
      "['Q185W8_CLOD6', '1.7.99.4', 'WP_011861481.1;YP_001088692.1', 'cdf:CD630_21790', '272563.8.peg.2294']\n",
      "['Q185V5_CLOD6', '1.7.99.1', 'WP_003433949.1;YP_001088681.1', 'cdf:CD630_21680;pdc:CDIF630_02400', '272563.120.peg.2289']\n",
      "['Q18C63_CLOD6', '1.7.1.17', 'WP_009902564.1;YP_001088044.1', 'cdf:CD630_15430;pdc:CDIF630_01711', '272563.120.peg.1616']\n",
      "['Q180Q5_CLOD6', '1.6.99.5', 'WP_003429109.1;YP_001089927.1', 'cdf:CD630_34070;pdc:CDIF630_03712', '272563.120.peg.3601']\n",
      "['Q180Q4_CLOD6', '1.6.99.5', 'WP_003416938.1;YP_001089926.1', 'cdf:CD630_34060;pdc:CDIF630_03711', '272563.120.peg.3600']\n",
      "['Q180Q3_CLOD6', '1.6.99.5', 'WP_003416946.1;YP_001089925.1', 'cdf:CD630_34050;pdc:CDIF630_03710', '272563.120.peg.3599']\n",
      "['Q183S7_CLOD6', '1.6.5.5', 'WP_003436791.1;YP_001089356.1', 'cdf:CD630_28430;pdc:CDIF630_03110', '272563.120.peg.2995']\n",
      "['FOLD_CLOD6', '1.5.1.5;_3.5.4.9', 'WP_011860936.1;YP_001087197.2', 'cdf:CD630_07200', '272563.8.peg.757']\n",
      "['Q189R6_CLOD6', '1.5.1.20', 'WP_003436920.1;YP_001087199.1', 'cdf:CD630_07220;pdc:CDIF630_00838', '272563.120.peg.743']\n",
      "['Q180C5_CLOD6', '1.5.1.2', 'WP_009898446.1;YP_001089797.1', 'cdf:CD630_32810;pdc:CDIF630_03581', '272563.120.peg.3466']\n",
      "['Q18C14_CLOD6', '1.5.1.2', 'WP_003434098.1;YP_001087996.1', 'cdf:CD630_14950;pdc:CDIF630_01660', '272563.120.peg.1564']\n",
      "['Q186L1_CLOD6', '1.4.4.2;_2.1.2.10', 'WP_011861287.1;YP_001088158.1', 'cdf:CD630_16570;pdc:CDIF630_01838', '272563.120.peg.1735']\n",
      "['Q186L4_CLOD6', '1.4.4.2', 'WP_011861288.1;YP_001088159.1', 'cdf:CD630_16580;pdc:CDIF630_01839', '272563.120.peg.1736']\n",
      "['Q185P0_CLOD6', '1.4.3.16', 'WP_003430790.1;YP_001088883.1', 'cdf:CD630_23710;pdc:CDIF630_02611', '272563.120.peg.2504']\n",
      "['Q18C58_CLOD6', '1.4.1.13', 'WP_003420072.1;YP_001088038.1', 'cdf:CD630_15370;pdc:CDIF630_01705', '272563.120.peg.1610']\n",
      "['Q188N4_CLOD6', '1.4.1.12', 'WP_009895461.1;YP_001086913.1', 'cdf:CD630_04420;pdc:CDIF630_00570', '272563.120.peg.467']\n",
      "['Q183H7_CLOD6', '1.3.99.22', 'WP_011861664.1;YP_001089253.1', 'cdf:CD630_27410', '272563.8.peg.2868']\n",
      "['Q189Z8_CLOD6', '1.3.99.-', 'WP_003437554.1;YP_001087280.1', 'cdf:CD630_08030;pdc:CDIF630_00922', '272563.120.peg.826']\n",
      "['Q188I5_CLOD6', '1.3.99.-', 'WP_003427762.1;YP_001086865.1', 'cdf:CD630_03990;pdc:CDIF630_00527', '272563.120.peg.419']\n",
      "['Q18AQ1_CLOD6', '1.3.99.-', 'WP_003418888.1;YP_001087535.1', 'cdf:CD630_10540;pdc:CDIF630_01194', '272563.120.peg.1094']\n",
      "['Q181U4_CLOD6', '1.3.3.1', 'WP_009891591.1;YP_001089696.1', 'cdf:CD630_31790;pdc:CDIF630_03471', '272563.120.peg.3354']\n",
      "['Q188E6_CLOD6', '1.3.3.1', 'WP_009897227.1;YP_001088591.1', 'cdf:CD630_20770;pdc:CDIF630_02300', '272563.120.peg.2191']\n",
      "['Q18CS7_CLOD6', '1.3.3.1', 'WP_003434506.1;YP_001086655.1', 'cdf:CD630_01850;pdc:CDIF630_00306', '272563.120.peg.200']\n",
      "['MURB_CLOD6', '1.3.1.98', 'WP_009894000.1;YP_001089922.1', 'cdf:CD630_34020;pdc:CDIF630_03707', '272563.120.peg.3596']\n",
      "['Q18B39_CLOD6', '1.3.1.9', 'WP_003419125.1;YP_001087673.1', 'cdf:CD630_11800;pdc:CDIF630_01329', '272563.120.peg.1231']\n",
      "['Q180R6_CLOD6', '1.3.1.76', 'WP_009898654.1;YP_001089938.1', 'cdf:CD630_34180;pdc:CDIF630_03726', '272563.120.peg.3612']\n",
      "['PYRDB_CLOD6', '1.3.1.14', 'WP_011860662.1;YP_001086656.1', 'cdf:CD630_01860;pdc:CDIF630_00307', '272563.120.peg.201']\n",
      "['Q187E8_CLOD6', '1.3.1.12', 'WP_003428643.1;YP_001088345.1', 'cdf:CD630_18390;pdc:CDIF630_02041', '272563.120.peg.1934']\n",
      "['Q181F5_CLOD6', '1.3.1.-', 'WP_003434962.1;YP_001090077.1', 'cdf:CD630_35540;pdc:CDIF630_03874', '272563.120.peg.3757']\n",
      "['Q180S1_CLOD6', '1.3.1.-', 'WP_003437730.1;YP_001089943.1', 'cdf:CD630_34230;pdc:CDIF630_03731', '272563.120.peg.3617']\n",
      "['Q185M6_CLOD6', '1.21.4.2;_1.21.4.3;_1.21.4.4', 'WP_011861535.1;YP_001088866.2', 'cdf:CD630_23520;pdc:CDIF630_02592', '272563.8.peg.2468']\n",
      "['Q185M1_CLOD6', '1.21.4.2;_1.21.4.3;_1.21.4.4', 'WP_003430750.1;YP_001088863.1', 'cdf:CD630_23480;pdc:CDIF630_02587', '272563.120.peg.2479']\n",
      "['Q185M3_CLOD6', '1.21.4.2', 'WP_011861534.1;YP_001088865.2', 'cdf:CD630_23510;pdc:CDIF630_02589', '272563.8.peg.2467']\n",
      "['Q185M5_CLOD6', '1.21.4.2', 'WP_003416871.1;YP_001088867.1', 'cdf:CD630_23540;pdc:CDIF630_02594', '272563.120.peg.2485']\n",
      "['Q185M4_CLOD6', '1.21.4.2', 'WP_009897537.1;YP_001088864.1', 'cdf:CD630_23490;pdc:CDIF630_02588', '272563.120.peg.2480']\n",
      "['Q17ZY6_CLOD6', '1.21.4.1', 'WP_011861888.1;YP_001089758.2', 'cdf:CD630_32410;pdc:CDIF630_03537', '272563.8.peg.3394']\n",
      "['Q17ZY9_CLOD6', '1.21.4.1', 'WP_003422104.1;YP_001089760.1', 'cdf:CD630_32440;pdc:CDIF630_03540', '272563.120.peg.3426']\n",
      "['Q17ZY7_CLOD6', '1.21.4.1', 'WP_003428092.1;YP_001089757.2', 'cdf:CD630_32400;pdc:CDIF630_03536', '272563.120.peg.3422']\n",
      "['Q17ZY2_CLOD6', '1.21.4.1', 'WP_003422094.1;YP_001089756.1', 'cdf:CD630_32390;pdc:CDIF630_03535', '272563.120.peg.3421']\n",
      "['Q186U8_CLOD6', '1.21.4.-', 'WP_004454406.1;YP_001088244.1', 'cdf:CD630_17400;pdc:CDIF630_01931', '272563.120.peg.1827']\n",
      "['Q181W5_CLOD6', '1.2.7.8', 'WP_003430808.1;YP_001088893.1', 'cdf:CD630_23810;pdc:CDIF630_02621', '272563.120.peg.2514']\n",
      "['Q181W6_CLOD6', '1.2.7.8', 'WP_003430806.1;YP_001088892.1', 'cdf:CD630_23800;pdc:CDIF630_02620', '272563.120.peg.2513']\n",
      "['Q189R4_CLOD6', '1.2.7.4', 'WP_011860935.1;YP_001087193.1', 'cdf:CD630_07160;pdc:CDIF630_00832', '272563.120.peg.737']\n",
      "['Q18CR1_CLOD6', '1.2.7.4', 'WP_009888019.1;YP_001086644.1', 'cdf:CD630_01740;pdc:CDIF630_00296', '272563.120.peg.190']\n",
      "['Q183B6_CLOD6', '1.2.7.-', 'WP_003427073.1;YP_001089193.1', 'cdf:CD630_26820;pdc:CDIF630_02936', '272563.120.peg.2825']\n",
      "['Q182B5_CLOD6', '1.2.7.-', 'WP_009890635.1;YP_001088941.1', 'cdf:CD630_24290;pdc:CDIF630_02671', '272563.120.peg.2563']\n",
      "['Q182B2_CLOD6', '1.2.7.-', 'WP_004454726.1;YP_001088940.1', 'cdf:CD630_24280;pdc:CDIF630_02670', '272563.120.peg.2562']\n",
      "['Q182B3_CLOD6', '1.2.7.-', 'WP_003430880.1;YP_001088939.1', 'cdf:CD630_24270;pdc:CDIF630_02669', '272563.120.peg.2561']\n",
      "['Q185Y6_CLOD6', '1.2.7.-', 'WP_003433885.1;YP_001088712.1', 'cdf:CD630_21990;pdc:CDIF630_02431', '272563.120.peg.2321']\n",
      "['Q185Y7_CLOD6', '1.2.7.-', 'WP_003433887.1;YP_001088711.1', 'cdf:CD630_21980;pdc:CDIF630_02430', '272563.120.peg.2320']\n",
      "['Q185Y4_CLOD6', '1.2.7.-', 'WP_003433888.1;YP_001088710.1', 'cdf:CD630_21970;pdc:CDIF630_02429', '272563.120.peg.2319']\n",
      "['Q18CK6_CLOD6', '1.2.7.-', 'WP_003420711.1;YP_001086587.1', 'cdf:CD630_01180;pdc:CDIF630_00233', '272563.120.peg.130']\n",
      "['Q18CK7_CLOD6', '1.2.7.-', 'WP_003420710.1;YP_001086586.1', 'cdf:CD630_01170;pdc:CDIF630_00232', '272563.120.peg.129']\n",
      "['Q18CK8_CLOD6', '1.2.7.-', 'WP_003432442.1;YP_001086585.1', 'cdf:CD630_01160;pdc:CDIF630_00231', '272563.120.peg.128']\n",
      "['Q18CB6_CLOD6', '1.2.4.1', 'WP_004453965.1;YP_001086499.1', 'cdf:CD630_00360', '272563.8.peg.45']\n",
      "['Q18CC3_CLOD6', '1.2.4.1', 'WP_004453964.1;YP_001086500.1', 'cdf:CD630_00370;pdc:CDIF630_00099', '272563.120.peg.41']\n",
      "['Q189C4_CLOD6', '1.2.1.9', 'WP_009895591.1;YP_001087055.1', 'cdf:CD630_05800;pdc:CDIF630_00693', '272563.120.peg.590']\n",
      "['Q188A0_CLOD6', '1.2.1.38', 'WP_011861435.1;YP_001088548.1', 'cdf:CD630_20340;pdc:CDIF630_02255', '272563.120.peg.2145']\n",
      "['Q180G2_CLOD6', '1.2.1.2', 'WP_011861905.1;YP_001089834.2', 'cdf:CD630_33170;pdc:CDIF630_03619', '272563.8.peg.3472']\n",
      "['Q185L7_CLOD6', '1.2.1.16', 'WP_004454646.1;YP_001088857.1', 'cdf:CD630_23420;pdc:CDIF630_02581', '272563.120.peg.2473']\n",
      "['Q17ZW9_CLOD6', '1.2.1.11', 'WP_003422061.1;YP_001089741.1', 'cdf:CD630_32240;pdc:CDIF630_03520', '272563.120.peg.3405']\n",
      "['Q187M7_CLOD6', '1.2.1.10', 'WP_011861390.1;YP_001088424.1', 'cdf:CD630_19170;pdc:CDIF630_02121', '272563.120.peg.2013']\n",
      "['Q186X6_CLOD6', '1.2.1.-', 'WP_011861329.1;YP_001088273.1', 'cdf:CD630_17670', '272563.8.peg.1855']\n",
      "['Q181T9_CLOD6', '1.2.1.-', 'WP_003421962.1;YP_001089691.1', 'cdf:CD630_31740;pdc:CDIF630_03466', '272563.120.peg.3349']\n",
      "['Q18C55_CLOD6', '1.18.1.2', 'WP_009896496.1;YP_001088037.1', 'cdf:CD630_15360;pdc:CDIF630_01704', '272563.120.peg.1609']\n",
      "['ISPH_CLOD6', '1.17.7.4', 'WP_003439450.1;YP_001088323.1', 'cdf:CD630_18180;pdc:CDIF630_02018', '272563.120.peg.1912']\n",
      "['Q185R7_CLOD6', '1.17.7.3', 'WP_011861462.1;YP_001088642.1', 'cdf:CD630_21280;pdc:CDIF630_02359', '272563.120.peg.2248']\n",
      "['Q18CJ6_CLOD6', '1.17.4.2', 'WP_003436320.1;YP_001086576.1', 'cdf:CD630_01080;pdc:CDIF630_00204', '272563.120.peg.119']\n",
      "['Q184I5_CLOD6', '1.17.4.1', 'WP_011861778.1;YP_001089513.1', 'cdf:CD630_29950;pdc:CDIF630_03279', '272563.120.peg.3162']\n",
      "['Q184I6_CLOD6', '1.17.4.1', 'WP_003439817.1;YP_001089512.1', 'cdf:CD630_29940;pdc:CDIF630_03278', '272563.120.peg.3161']\n",
      "['Q18BC3_CLOD6', '1.17.4.1', 'WP_003438236.1;YP_001087755.1', 'cdf:CD630_12610;pdc:CDIF630_01414', '272563.120.peg.1319']\n",
      "['Q17ZX4_CLOD6', '1.17.1.8', 'WP_011861884.1;YP_001089746.1', 'cdf:CD630_32290;pdc:CDIF630_03525', '272563.120.peg.3410']\n",
      "['Q17ZX1_CLOD6', '1.17.1.8', 'WP_003438808.1;YP_001089743.1', 'cdf:CD630_32260;pdc:CDIF630_03522', '272563.120.peg.3407']\n",
      "['Q181U2_CLOD6', '1.17.1.4', 'WP_009891588.1;YP_001089694.1', 'cdf:CD630_31770;pdc:CDIF630_03469', '272563.120.peg.3352']\n",
      "['Q188G9_CLOD6', '1.17.1.4', 'WP_003434060.1;YP_001088614.1', 'cdf:CD630_21000;pdc:CDIF630_02323', '272563.120.peg.2214']\n",
      "['Q188G6_CLOD6', '1.17.1.4', 'WP_009897245.1;YP_001088613.1', 'cdf:CD630_20990;pdc:CDIF630_02322', '272563.120.peg.2213']\n",
      "['Q188F6_CLOD6', '1.17.1.4', 'WP_003434079.1;YP_001088602.1', 'cdf:CD630_20880;pdc:CDIF630_02311', '272563.120.peg.2202']\n",
      "['Q188F4_CLOD6', '1.17.1.4', 'WP_011861450.1;YP_001088601.1', 'cdf:CD630_20870;pdc:CDIF630_02310', '272563.120.peg.2201']\n",
      "['Q188E9_CLOD6', '1.17.1.4', 'WP_009902988.1;YP_001088595.2', 'cdf:CD630_20810;pdc:CDIF630_02304', '272563.120.peg.2195']\n",
      "['Q188E7_CLOD6', '1.17.1.4', 'WP_011861448.1;YP_001088594.1', 'cdf:CD630_20800;pdc:CDIF630_02303', '272563.120.peg.2194']\n",
      "['Q188E8_CLOD6', '1.17.1.4', 'WP_011861447.1;YP_001088593.1', 'cdf:CD630_20790;pdc:CDIF630_02302', '272563.120.peg.2193']\n",
      "['Q188E2_CLOD6', '1.17.1.4', 'WP_011861444.1;YP_001088587.1', 'cdf:CD630_20730;pdc:CDIF630_02296', '272563.120.peg.2187']\n",
      "['Q185Y2_CLOD6', '1.16.3.2', 'WP_003433890.1;YP_001088708.1', 'cdf:CD630_21950;pdc:CDIF630_02427', '272563.120.peg.2317']\n",
      "['Q18A22_CLOD6', '1.15.1.1', 'WP_004453642.1;YP_001087304.1', 'cdf:CD630_08270;pdc:CDIF630_00946', '272563.120.peg.850']\n",
      "['Q186I6_CLOD6', '1.15.1.1', 'WP_009889536.1;YP_001088132.1', 'cdf:CD630_16310;pdc:CDIF630_01808', '272563.120.peg.1709']\n",
      "['Q180F8_CLOD6', '1.12.7.2', 'WP_011861902.1;YP_001089830.1', 'cdf:CD630_33140', '272563.8.peg.3468']\n",
      "['Q18A86_CLOD6', '1.12.7.2', 'WP_009895925.1;YP_001087368.1', 'cdf:CD630_08940;pdc:CDIF630_01014', '272563.120.peg.918']\n",
      "['Q18A87_CLOD6', '1.12.7.2', 'WP_009888725.1;YP_001087367.2', 'cdf:CD630_08930;pdc:CDIF630_01013', '272563.120.peg.917']\n",
      "['Q185A6_CLOD6', '1.12.2.1', 'WP_003433830.1;YP_001088747.2', 'cdf:CD630_22330;pdc:CDIF630_02466', '272563.120.peg.2360']\n",
      "['Q186C1_CLOD6', '1.11.1.6', 'WP_009896673.1;YP_001088067.1', 'cdf:CD630_15670;pdc:CDIF630_01736', '272563.120.peg.1640']\n",
      "['Q18BV5_CLOD6', '1.11.1.15', 'WP_011861211.1;YP_001087934.1', 'cdf:CD630_14330', '272563.8.peg.1505']\n",
      "['Q187D3_CLOD6', '1.11.1.15', 'WP_003439443.1;YP_001088327.1', 'cdf:CD630_18220;pdc:CDIF630_02023', '272563.120.peg.1917']\n",
      "['Q18AJ3_CLOD6', '1.1.1.95', 'WP_003437082.1;YP_001087476.1', 'cdf:CD630_09950;pdc:CDIF630_01131', '272563.120.peg.1034']\n",
      "['Q182W5_CLOD6', '1.1.1.94', 'WP_011861619.1;YP_001089141.1', 'cdf:CD630_26300', '272563.8.peg.2754']\n",
      "['ILVC_CLOD6', '1.1.1.86', 'WP_011861252.1;YP_001088065.1', 'cdf:CD630_15650;pdc:CDIF630_01734', '272563.120.peg.1638']\n",
      "['Q18AJ0_CLOD6', '1.1.1.85', 'WP_011861060.1;YP_001087473.1', 'cdf:CD630_09920;pdc:CDIF630_01128', '272563.120.peg.1031']\n",
      "['Q185L3_CLOD6', '1.1.1.61', 'WP_011861531.1;YP_001088853.1', 'cdf:CD630_23380;pdc:CDIF630_02577', '272563.120.peg.2469']\n",
      "['Q183P5_CLOD6', '1.1.1.60', 'WP_011861698.1;YP_001089326.1', 'cdf:CD630_28130;pdc:CDIF630_03078', '272563.120.peg.2963']\n",
      "['Q18A33_CLOD6', '1.1.1.41', 'WP_004454234.1;YP_001087311.1', 'cdf:CD630_08340;pdc:CDIF630_00953', '272563.120.peg.857']\n",
      "['Q18AK8_CLOD6', '1.1.1.38', 'WP_011861063.1;YP_001087486.1', 'cdf:CD630_10050', '272563.8.peg.1056']\n",
      "['Q183B5_CLOD6', '1.1.1.30', 'WP_009890906.1;YP_001089190.1', 'cdf:CD630_26790;pdc:CDIF630_02933', '272563.120.peg.2822']\n",
      "['Q186D4_CLOD6', '1.1.1.3', 'WP_003429993.1;YP_001088080.1', 'cdf:CD630_15800;pdc:CDIF630_01751', '272563.120.peg.1653']\n",
      "['Q18B25_CLOD6', '1.1.1.3', 'WP_011861130.1;YP_001087658.1', 'cdf:CD630_11660;pdc:CDIF630_01314', '272563.120.peg.1216']\n",
      "['Q188H7_CLOD6', '1.1.1.28', 'WP_003431407.1;YP_001086860.1', 'cdf:CD630_03940;pdc:CDIF630_00522', '272563.120.peg.414']\n",
      "['Q189A3_CLOD6', '1.1.1.274', 'WP_009895580.1;YP_001087037.1', 'cdf:CD630_05610;pdc:CDIF630_00674', '272563.120.peg.571']\n",
      "['Q185V1_CLOD6', '1.1.1.27', 'WP_011861477.1;YP_001088677.1', 'cdf:CD630_21640;pdc:CDIF630_02396', '272563.120.peg.2285']\n",
      "['DXR_CLOD6', '1.1.1.267', 'WP_011861463.1;YP_001088644.1', 'cdf:CD630_21300;pdc:CDIF630_02361', '272563.120.peg.2250']\n",
      "['Q183E3_CLOD6', '1.1.1.25', 'WP_011861653.1;YP_001089219.1', 'cdf:CD630_27080;pdc:CDIF630_02965', '272563.120.peg.2852']\n",
      "['Q187E7_CLOD6', '1.1.1.25', 'WP_003439416.1;YP_001088343.1', 'cdf:CD630_18370;pdc:CDIF630_02039', '272563.120.peg.1932']\n",
      "['Q186D5_CLOD6', '1.1.1.23', 'WP_011861261.1;YP_001088082.1', 'cdf:CD630_15820;pdc:CDIF630_01753', '272563.120.peg.1655']\n",
      "['Q183L0_CLOD6', '1.1.1.22', 'WP_011861675.1;YP_001089284.1', 'cdf:CD630_27710;pdc:CDIF630_03034', '272563.120.peg.2919']\n",
      "['Q185K9_CLOD6', '1.1.1.205', 'WP_003423106.1;YP_001088850.1', 'cdf:CD630_23350;pdc:CDIF630_02574', '272563.120.peg.2466']\n",
      "['Q185K5_CLOD6', '1.1.1.17', 'WP_011861528.1;YP_001088846.1', 'cdf:CD630_23310;pdc:CDIF630_02570', '272563.120.peg.2462']\n",
      "['Q18CT0_CLOD6', '1.1.1.169', 'WP_011860663.1;YP_001086658.1', 'cdf:CD630_01880;pdc:CDIF630_00309', '272563.120.peg.203']\n",
      "['Q18AQ7_CLOD6', '1.1.1.157', 'WP_003418897.1;YP_001087539.1', 'cdf:CD630_10580;pdc:CDIF630_01198', '272563.120.peg.1098']\n",
      "['Q189W7_CLOD6', '1.1.1.140', 'WP_009895815.1;YP_001087245.1', 'cdf:CD630_07680;pdc:CDIF630_00886', '272563.120.peg.790']\n",
      "['Q187A2_CLOD6', '1.1.1.103', 'WP_003433612.1;YP_001088298.1', 'cdf:CD630_17920;pdc:CDIF630_01991', '272563.120.peg.1885']\n",
      "['Q182R6_CLOD6', '1.1.1.100', 'WP_009897780.1;YP_001089093.1', 'cdf:CD630_25770;pdc:CDIF630_02830', '272563.120.peg.2718']\n",
      "['Q18B45_CLOD6', '1.1.1.100', 'WP_003428442.1;YP_001087675.1', 'cdf:CD630_11820;pdc:CDIF630_01331', '272563.120.peg.1233']\n",
      "['Q184J7_CLOD6', '1.1.1.1', 'WP_003439845.1;YP_001089524.1', 'cdf:CD630_30060;pdc:CDIF630_03291', '272563.120.peg.3173']\n",
      "['Q187L6_CLOD6', '1.1.1.1', 'WP_011861386.1;YP_001088414.1', 'cdf:CD630_19070;pdc:CDIF630_02111', '272563.120.peg.2003']\n",
      "['Q18CF0_CLOD6', '1.1.1.-', 'WP_011860631.1;YP_001086529.1', 'cdf:CD630_00650;pdc:CDIF630_00131', '272563.120.peg.71']\n",
      "['Q188T6_CLOD6', '1.1.1.-', 'WP_011860836.1;YP_001086964.1', 'cdf:CD630_04900;pdc:CDIF630_00619', '272563.120.peg.515']\n",
      "['Q180F5_CLOD6', '1.1.1.-', 'WP_009891785.1;YP_001089826.1', 'cdf:CD630_33100;pdc:CDIF630_03611', '272563.120.peg.3495']\n",
      "['Q184U3_CLOD6', '1.1.1.-', 'WP_011861825.1;YP_001089619.1', 'cdf:CD630_31050;pdc:CDIF630_03389', '272563.120.peg.3271']\n",
      "['Q18D18_CLOD6', '1.1.1.-', 'WP_003435962.1;YP_001086743.1', 'cdf:CD630_02740;pdc:CDIF630_00398', '272563.120.peg.292']\n",
      "['Q185J9_CLOD6', '1.1.1.-', 'WP_004454628.1;YP_001088839.1', 'cdf:CD630_23240;pdc:CDIF630_02562', '272563.120.peg.2455']\n",
      "['Q185J7_CLOD6', '1.1.1.-', 'WP_011861524.1;YP_001088838.1', 'cdf:CD630_23230;pdc:CDIF630_02561', '272563.120.peg.2454']\n",
      "['Q185E9_CLOD6', '1.1.1.-', 'WP_011861508.1;YP_001088791.1', 'cdf:CD630_22790;pdc:CDIF630_02514', '272563.120.peg.2406']\n",
      "['Q18B31_CLOD6', '1.1.-.-', 'WP_003438093.1;YP_001087665.1', 'cdf:CD630_11730;pdc:CDIF630_01321', '272563.120.peg.1223']\n",
      "['Q18AE6_CLOD6', 'none', 'WP_003433538.1;YP_001087428.1', 'cdf:CD630_09500', '272563.8.peg.997']\n",
      "['Q18AD5_CLOD6', 'none', 'WP_011861024.1;YP_001087416.1', 'cdf:CD630_09400', '272563.8.peg.985']\n",
      "['Q18AD6_CLOD6', 'none', 'WP_011861023.1;YP_001087415.1', 'cdf:CD630_09390', '272563.8.peg.984']\n",
      "['Q18AC7_CLOD6', 'none', 'WP_011861018.1;YP_001087410.1', 'cdf:CD630_09341', '272563.8.peg.979']\n",
      "['Q18AC8_CLOD6', 'none', 'WP_011861017.1;YP_001087409.1', 'cdf:CD630_09340', '272563.8.peg.978']\n",
      "['Q18AC1_CLOD6', 'none', 'WP_011861013.1;YP_001087405.1', 'cdf:CD630_09301;pdc:CDIF630_01056', '272563.8.peg.974']\n",
      "['Q18AB5_CLOD6', 'none', 'WP_011861003.1;YP_001087397.1', 'cdf:CD630_09230', '272563.8.peg.964']\n",
      "['Q18AB9_CLOD6', 'none', 'WP_011860999.1;YP_001087393.1', 'cdf:CD630_09190;pdc:CDIF630_01042', '272563.8.peg.960']\n",
      "['Q18AB0_CLOD6', 'none', 'WP_011860994.1;YP_001087388.1', 'cdf:CD630_09140', '272563.8.peg.955']\n",
      "['Q18AA3_CLOD6', 'none', 'WP_011860990.1;YP_001087384.1', 'cdf:CD630_09100', '272563.8.peg.951']\n",
      "['RL24_CLOD6', 'none', 'WP_003427723.1;YP_001086548.2', 'cdf:CD630_00830', '272563.8.peg.95']\n",
      "['Q18A95_CLOD6', 'none', 'WP_011860982.1;YP_001087375.1', 'cdf:CD630_09010', '272563.8.peg.940']\n",
      "['Q18A63_CLOD6', 'none', 'WP_009888685.1;YP_001087344.1', 'cdf:CD630_08700', '272563.8.peg.909']\n",
      "['Q18A55_CLOD6', 'none', 'WP_004454202.1;YP_001087336.1', 'cdf:CD630_08620', '272563.8.peg.901']\n",
      "['Q18A52_CLOD6', 'none', 'WP_004454207.1;YP_001087333.1', 'cdf:CD630_08560', '272563.8.peg.894']\n",
      "['Q18A36_CLOD6', 'none', 'WP_009895864.1;YP_001087316.1', 'cdf:CD630_08390', '272563.8.peg.877']\n",
      "['Q18A23_CLOD6', 'none', 'WP_011860967.1;YP_001087303.1', 'cdf:CD630_08260', '272563.8.peg.864']\n",
      "['Q18A13_CLOD6', 'none', 'WP_011860964.1;YP_001087294.1', 'cdf:CD630_08170', '272563.8.peg.855']\n",
      "['Q18CF9_CLOD6', 'none', 'WP_009887887.1;YP_001086538.2', 'cdf:CD630_00740', '272563.8.peg.85']\n",
      "['Q18A01_CLOD6', 'none', 'WP_003437556.1;YP_001087282.1', 'cdf:CD630_08050', '272563.8.peg.842']\n",
      "['Q189Z9_CLOD6', 'none', 'WP_011860960.1;YP_001087279.1', 'cdf:CD630_08020', '272563.8.peg.839']\n",
      "['Q189X5_CLOD6', 'none', 'WP_011860954.1;YP_001087255.1', 'cdf:CD630_07780', '272563.8.peg.815']\n",
      "['Q17ZT3_CLOD6', 'none', 'WP_011572951.1;YP_659586.1', 'cdf:CD630_p100', '272563.8.peg.8']\n",
      "['Q189U4_CLOD6', 'none', 'WP_003436957.1;YP_001087225.1', 'cdf:CD630_07480', '272563.8.peg.785']\n",
      "['Q189U1_CLOD6', 'none', 'WP_011860943.1;YP_001087222.1', 'cdf:CD630_07450', '272563.8.peg.782']\n",
      "['Q189P1_CLOD6', 'none', 'WP_011860924.1;YP_001087174.1', 'cdf:CD630_06980', '272563.8.peg.733']\n",
      "['Q189N3_CLOD6', 'none', 'WP_011860917.1;YP_001087164.1', 'cdf:CD630_06880', '272563.8.peg.723']\n",
      "['Q189N1_CLOD6', 'none', 'WP_011860916.1;YP_001087161.1', 'cdf:CD630_06850', '272563.8.peg.720']\n",
      "['Q189M3_CLOD6', 'none', 'WP_011860915.1;YP_001087158.1', 'cdf:CD630_06820', '272563.8.peg.717']\n",
      "['Q189L6_CLOD6', 'none', 'WP_011860910.1;YP_001087148.1', 'cdf:CD630_06720', '272563.8.peg.708']\n",
      "['Q189L4_CLOD6', 'none', 'WP_003439182.1;YP_001087144.1', 'cdf:CD630_06680', '272563.8.peg.704']\n",
      "['Q17ZT4_CLOD6', 'none', 'WP_011572950.1;YP_659585.1', 'cdf:CD630_p090', '272563.8.peg.7']\n",
      "['Q189K1_CLOD6', 'none', 'WP_011860902.1;YP_001087131.1', 'cdf:CD630_06560', '272563.8.peg.690']\n",
      "['Q189J3_CLOD6', 'none', 'WP_011860895.1;YP_001087124.1', 'cdf:CD630_06490', '272563.8.peg.681']\n",
      "['Q189J4_CLOD6', 'none', 'WP_003440167.1;YP_001087123.1', 'cdf:CD630_06480', '272563.8.peg.680']\n",
      "['Q189I2_CLOD6', 'none', 'WP_011860891.1;YP_001087117.1', 'cdf:CD630_06420', '272563.8.peg.672']\n",
      "['Q189H9_CLOD6', 'none', 'WP_011860889.1;YP_001087110.1', 'cdf:CD630_06350', '272563.8.peg.665']\n",
      "['Q189H3_CLOD6', 'none', 'WP_011860886.1;YP_001087107.1', 'cdf:CD630_06320', '272563.8.peg.662']\n",
      "['Q189H5_CLOD6', 'none', 'WP_011860884.1;YP_001087105.1', 'cdf:CD630_06300', '272563.8.peg.660']\n",
      "['Q189F0_CLOD6', 'none', 'WP_011860873.1;YP_001087077.1', 'cdf:CD630_06020', '272563.8.peg.632']\n",
      "['Q189C5_CLOD6', 'none', 'WP_011860861.1;YP_001087054.1', 'cdf:CD630_05790', '272563.8.peg.607']\n",
      "['Q17ZT5_CLOD6', 'none', 'WP_011572949.1;YP_659584.1', 'cdf:CD630_p080', '272563.8.peg.6']\n",
      "['Q189B4_CLOD6', 'none', 'WP_011860859.1;YP_001087045.1', 'cdf:CD630_05700', '272563.8.peg.597']\n",
      "['Q188W0_CLOD6', 'none', 'WP_011860844.1;YP_001086992.1', 'cdf:CD630_05140', '272563.8.peg.539']\n",
      "['Q188V9_CLOD6', 'none', 'WP_002324544.1;YP_001086990.1', 'cdf:CD630_05110', '272563.8.peg.537']\n",
      "['Q188V4_CLOD6', 'none', 'WP_002334012.1;YP_001086988.1', 'cdf:CD630_05101', '272563.8.peg.535']\n",
      "['Q188V5_CLOD6', 'none', 'WP_002324547.1;YP_001086987.1', 'cdf:CD630_05100', '272563.8.peg.534']\n",
      "['Q188V7_CLOD6', 'none', 'WP_002324548.1;YP_001086985.1', 'cdf:CD630_05090', '272563.8.peg.533']\n",
      "['Q188V1_CLOD6', 'none', 'WP_011860841.1;YP_001086983.1', 'cdf:CD630_05080', '272563.8.peg.532']\n",
      "['Q188U9_CLOD6', 'none', 'WP_002324550.1;YP_001086981.1', 'cdf:CD630_05070', '272563.8.peg.531']\n",
      "['Q188U3_CLOD6', 'none', 'WP_011860840.1;YP_001086978.1', 'cdf:CD630_05030', '272563.8.peg.528']\n",
      "['Q188U4_CLOD6', 'none', 'WP_002324553.1;YP_001086977.1', 'cdf:CD630_05020', '272563.8.peg.527']\n",
      "['Q188U5_CLOD6', 'none', 'WP_002324554.1;YP_001086976.1', 'cdf:CD630_05010', '272563.8.peg.526']\n",
      "['Q188U6_CLOD6', 'none', 'WP_002324555.1;YP_001086975.1', 'cdf:CD630_05000', '272563.8.peg.525']\n",
      "['Q188U7_CLOD6', 'none', 'WP_001009056.1;YP_001086974.1', 'cdf:CD630_04991', '272563.8.peg.524']\n",
      "['Q188U0_CLOD6', 'none', 'WP_002364938.1;YP_001086973.1', 'cdf:CD630_04990', '272563.8.peg.523']\n",
      "['Q188U2_CLOD6', 'none', 'WP_002324557.1;YP_001086971.1', 'cdf:CD630_04980', '272563.8.peg.522']\n",
      "['Q188T7_CLOD6', 'none', 'WP_002324558.1;YP_001086970.1', 'cdf:CD630_04970', '272563.8.peg.521']\n",
      "['Q188T8_CLOD6', 'none', 'WP_002324559.1;YP_001086969.1', 'cdf:CD630_04960', '272563.8.peg.520']\n",
      "['Q18CC5_CLOD6', 'none', 'WP_004453958.1;YP_001086505.1', 'cdf:CD630_00420', '272563.8.peg.51']\n",
      "['Q188S0_CLOD6', 'none', 'WP_003434737.1;YP_001086952.1', 'cdf:CD630_04780', '272563.8.peg.502']\n",
      "['Q17ZT6_CLOD6', 'none', 'WP_011572948.1;YP_659583.1', 'cdf:CD630_p070', '272563.8.peg.5']\n",
      "['Q188R6_CLOD6', 'none', 'WP_011860828.1;YP_001086946.1', 'cdf:CD630_04730', '272563.8.peg.497']\n",
      "['Q188N9_CLOD6', 'none', 'WP_011860812.1;YP_001086915.1', 'cdf:CD630_04440', '272563.8.peg.468']\n",
      "['Q188N1_CLOD6', 'none', 'WP_011860810.1;YP_001086912.1', 'cdf:CD630_04410', '272563.8.peg.465']\n",
      "['Q188M6_CLOD6', 'none', 'WP_011860803.1;YP_001086903.1', 'cdf:CD630_04350', '272563.8.peg.457']\n",
      "['Q188L8_CLOD6', 'none', 'WP_011860797.1;YP_001086898.1', 'cdf:CD630_04300', '272563.8.peg.451']\n",
      "['Q188K3_CLOD6', 'none', 'WP_011860787.1;YP_001086884.1', 'cdf:CD630_04170;pdc:CDIF630_00544', '272563.8.peg.437']\n",
      "['Q188K5_CLOD6', 'none', 'WP_000466908.1;YP_001086882.1', 'cdf:CD630_04150;pdc:CDIF630_00542', '272563.8.peg.435']\n",
      "['Q188J4_CLOD6', 'none', 'WP_011860780.1;YP_001086874.1', 'cdf:CD630_04081', '272563.8.peg.428']\n",
      "['Q188J5_CLOD6', 'none', 'WP_003042651.1;YP_001086873.1', 'cdf:CD630_04080', '272563.8.peg.427']\n",
      "['Q18DB9_CLOD6', 'none', 'WP_011860769.1;YP_009109548.1', 'cdf:CD630_03840', '272563.8.peg.404']\n",
      "['Q17ZT1_CLOD6', 'none', 'WP_011572947.1;YP_659582.1', 'cdf:CD630_p060', '272563.8.peg.4']\n",
      "['Q18DB5_CLOD6', 'none', 'WP_011860763.1;YP_001086844.1', 'cdf:CD630_03770', '272563.8.peg.398']\n",
      "['Q181T1_CLOD6', 'none', 'WP_011862027.1;YP_001090203.1', 'cdf:CD630_36781', '272563.8.peg.3856']\n",
      "['Q181Q6_CLOD6', 'none', 'WP_011862025.1;YP_001090176.1', 'cdf:CD630_36520', '272563.8.peg.3827']\n",
      "['Q181N8_CLOD6', 'none', 'WP_011862021.1;YP_001090161.1', 'cdf:CD630_36370', '272563.8.peg.3812']\n",
      "['Q18D98_CLOD6', 'none', 'WP_011860749.1;YP_001086828.2', 'cdf:CD630_03610', '272563.8.peg.381']\n",
      "['Q181N7_CLOD6', 'none', 'WP_011862019.1;YP_001090158.1', 'cdf:CD630_36350', '272563.8.peg.3809']\n",
      "['Q181M8_CLOD6', 'none', 'WP_011862016.1;YP_001090151.1', 'cdf:CD630_36280', '272563.8.peg.3801']\n",
      "['Q181M3_CLOD6', 'none', 'WP_011862011.1;YP_001090143.1', 'cdf:CD630_36200', '272563.8.peg.3793']\n",
      "['Q181M0_CLOD6', 'none', 'WP_011862009.1;YP_001090141.1', 'cdf:CD630_36180', '272563.8.peg.3791']\n",
      "['Q181L7_CLOD6', 'none', 'WP_011862008.1;YP_001090140.1', 'cdf:CD630_36170', '272563.8.peg.3790']\n",
      "['Q18DA0_CLOD6', 'none', 'WP_009264383.1;YP_001086826.1', 'cdf:CD630_03590', '272563.8.peg.379']\n",
      "['Q181J4_CLOD6', 'none', 'WP_011862005.1;YP_001090117.1', 'cdf:CD630_35930', '272563.8.peg.3764']\n",
      "['Q18D95_CLOD6', 'none', 'WP_011860746.1;YP_001086822.1', 'cdf:CD630_03550', '272563.8.peg.375']\n",
      "['Q18D96_CLOD6', 'none', 'WP_011860745.1;YP_001086821.1', 'cdf:CD630_03541', '272563.8.peg.374']\n",
      "['Q181G5_CLOD6', 'none', 'WP_011862000.1;YP_001090088.1', 'cdf:CD630_35650', '272563.8.peg.3735']\n",
      "['Q181F3_CLOD6', 'none', 'WP_003425974.1;YP_001090074.1', 'cdf:CD630_35512', '272563.8.peg.3721']\n",
      "['Q181E2_CLOD6', 'none', 'WP_003427659.1;YP_001090063.1', 'cdf:CD630_35420', '272563.8.peg.3710']\n",
      "['Q181A7_CLOD6', 'none', 'WP_011861985.1;YP_001090029.1', 'cdf:CD630_35080', '272563.8.peg.3676']\n",
      "['Q181A1_CLOD6', 'none', 'WP_011861981.1;YP_001090024.1', 'cdf:CD630_35030', '272563.8.peg.3671']\n",
      "['Q18D91_CLOD6', 'none', 'WP_011860743.1;YP_001086816.1', 'cdf:CD630_03470', '272563.8.peg.367']\n",
      "['Q180Y9_CLOD6', 'none', 'WP_011861980.1;YP_001090010.1', 'cdf:CD630_34890', '272563.8.peg.3657']\n",
      "['Q18D86_CLOD6', 'none', 'WP_011860740.1;YP_001086810.1', 'cdf:CD630_03410', '272563.8.peg.361']\n",
      "['Q180P6_CLOD6', 'none', 'WP_003429125.1;YP_001089919.1', 'cdf:CD630_33990', '272563.8.peg.3562']\n",
      "['Q180N4_CLOD6', 'none', 'WP_011861952.1;YP_001089905.1', 'cdf:CD630_33870', '272563.8.peg.3548']\n",
      "['Q180M7_CLOD6', 'none', 'WP_009898612.1;YP_001089898.1', 'cdf:CD630_33780', '272563.8.peg.3538']\n",
      "['Q180K8_CLOD6', 'none', 'WP_011861933.1;YP_001089880.1', 'cdf:CD630_33630', '272563.8.peg.3521']\n",
      "['Q180K6_CLOD6', 'none', 'WP_011861932.1;YP_001089879.1', 'cdf:CD630_33620', '272563.8.peg.3520']\n",
      "['Q180J4_CLOD6', 'none', 'WP_011861926.1;YP_001089866.1', 'cdf:CD630_33490', '272563.8.peg.3506']\n",
      "['Q180J1_CLOD6', 'none', 'WP_002578441.1;YP_001089863.1', 'cdf:CD630_33460', '272563.8.peg.3503']\n",
      "['Q180I6_CLOD6', 'none', 'WP_002570387.1;YP_001089859.1', 'cdf:CD630_33421', '272563.8.peg.3498']\n",
      "['Q180I5_CLOD6', 'none', 'WP_005427021.1;YP_001089856.1', 'cdf:CD630_33390', '272563.8.peg.3493']\n",
      "['Q180H7_CLOD6', 'none', 'WP_011861912.1;YP_001089849.1', 'cdf:CD630_33330', '272563.8.peg.3487']\n",
      "['Q180H1_CLOD6', 'none', 'WP_009009557.1;YP_001089845.1', 'cdf:CD630_33290', '272563.8.peg.3483']\n",
      "['Q180G0_CLOD6', 'none', 'WP_011861904.1;YP_001089833.1', 'cdf:CD630_33160', '272563.8.peg.3471']\n",
      "['Q180G1_CLOD6', 'none', 'WP_003437324.1;YP_001089832.1', 'cdf:CD630_33151', '272563.8.peg.3470']\n",
      "['Q180D4_CLOD6', 'none', 'WP_003435630.1;YP_001089806.1', 'cdf:CD630_32900', '272563.8.peg.3443']\n",
      "['Q180B9_CLOD6', 'none', 'WP_011861894.1;YP_001089792.1', 'cdf:CD630_32760', '272563.8.peg.3429']\n",
      "['Q180B5_CLOD6', 'none', 'WP_011861893.1;YP_001089787.1', 'cdf:CD630_32710', '272563.8.peg.3424']\n",
      "['Q18D65_CLOD6', 'none', 'WP_011860732.1;YP_001086791.1', 'cdf:CD630_03220', '272563.8.peg.342']\n",
      "['Q180A9_CLOD6', 'none', 'WP_011861892.1;YP_001089782.1', 'cdf:CD630_32660', '272563.8.peg.3419']\n",
      "['Q17ZY3_CLOD6', 'none', 'WP_003428088.1;YP_001089755.1', 'cdf:CD630_32380', '272563.8.peg.3391']\n",
      "['Q17ZX8_CLOD6', 'none', 'WP_011861887.1;YP_001089750.1', 'cdf:CD630_32330', '272563.8.peg.3386']\n",
      "['Y3232_CLOD6', 'none', 'WP_011861886.1;YP_001089749.1', 'cdf:CD630_32320', '272563.8.peg.3385']\n",
      "['Q17ZW6_CLOD6', 'none', 'WP_003438792.1;YP_001089737.1', 'cdf:CD630_32201', '272563.8.peg.3373']\n",
      "['Q17ZV1_CLOD6', 'none', 'WP_011861878.1;YP_001089723.1', 'cdf:CD630_32070', '272563.8.peg.3357']\n",
      "['Q17ZU0_CLOD6', 'none', 'WP_011861873.1;YP_001089712.1', 'cdf:CD630_31960', '272563.8.peg.3347']\n",
      "['Q184Z8_CLOD6', 'none', 'WP_011861861.1;YP_001089676.1', 'cdf:CD630_31600', '272563.8.peg.3309']\n",
      "['SSRP_CLOD6', 'none', 'WP_011861858.1;YP_001089673.1', 'cdf:CD630_31570', '272563.8.peg.3306']\n",
      "['Q184Y9_CLOD6', 'none', 'WP_011861853.1;YP_001089665.1', 'cdf:CD630_31510', '272563.8.peg.3295']\n",
      "['Q18D49_CLOD6', 'none', 'WP_003435902.1;YP_001086777.1', 'cdf:CD630_03080', '272563.8.peg.329']\n",
      "['Q184X7_CLOD6', 'none', 'WP_011861845.1;YP_001089654.1', 'cdf:CD630_31390', '272563.8.peg.3284']\n",
      "['Q184W4_CLOD6', 'none', 'WP_011861835.1;YP_001089641.1', 'cdf:CD630_31260', '272563.8.peg.3271']\n",
      "['Q184V7_CLOD6', 'none', 'WP_011861831.1;YP_001089635.1', 'cdf:CD630_31201', '272563.8.peg.3265']\n",
      "['Q184T7_CLOD6', 'none', 'WP_011861820.1;YP_001089614.1', 'cdf:CD630_31000', '272563.8.peg.3244']\n",
      "['Q184Q3_CLOD6', 'none', 'WP_009898137.1;YP_001089581.1', 'cdf:CD630_30670', '272563.8.peg.3211']\n",
      "['Q184Q0_CLOD6', 'none', 'WP_011861801.1;YP_001089577.1', 'cdf:CD630_30630', '272563.8.peg.3207']\n",
      "['Q184P4_CLOD6', 'none', 'WP_011861798.1;YP_001089572.1', 'cdf:CD630_30580', '272563.8.peg.3202']\n",
      "['Q18CA8_CLOD6', 'none', 'WP_003425587.1;YP_001086486.1', 'cdf:CD630_00230', '272563.8.peg.32']\n",
      "['Q184N5_CLOD6', 'none', 'WP_011861794.1;YP_001089565.1', 'cdf:CD630_30500', '272563.8.peg.3194']\n",
      "['Q184M6_CLOD6', 'none', 'WP_009898107.1;YP_001089552.1', 'cdf:CD630_30370', '272563.8.peg.3181']\n",
      "['Q184K6_CLOD6', 'none', 'WP_011861781.1;YP_001089533.1', 'cdf:CD630_30160', '272563.8.peg.3160']\n",
      "['Q184K3_CLOD6', 'none', 'WP_003431989.1;YP_001089530.1', 'cdf:CD630_30130', '272563.8.peg.3157']\n",
      "['Q184J0_CLOD6', 'none', 'WP_003439828.1;YP_001089517.1', 'cdf:CD630_29990', '272563.8.peg.3143']\n",
      "['Q184H3_CLOD6', 'none', 'WP_011861773.1;YP_001089501.1', 'cdf:CD630_29830', '272563.8.peg.3127']\n",
      "['Q184F6_CLOD6', 'none', 'WP_011861762.1;YP_001089483.1', 'cdf:CD630_29660', '272563.8.peg.3111']\n",
      "['Q184D6_CLOD6', 'none', 'WP_011861755.1;YP_001089463.1', 'cdf:CD630_29471;pdc:CDIF630_03229', '272563.8.peg.3090']\n",
      "['Q184D7_CLOD6', 'none', 'WP_011861754.1;YP_001089462.2', 'cdf:CD630_29470', '272563.8.peg.3089']\n",
      "['Q184C2_CLOD6', 'none', 'WP_011861742.1;YP_001089450.1', 'cdf:CD630_29350;pdc:CDIF630_03212', '272563.8.peg.3074']\n",
      "['Q184A6_CLOD6', 'none', 'WP_011861026.1;YP_001087421.1;YP_001089437.2', 'cdf:CD630_09440;cdf:CD630_29220', '272563.8.peg.3057']\n",
      "['Q184A9_CLOD6', 'none', 'WP_003429839.1;YP_001087424.1;YP_001089434.2', 'cdf:CD630_09461;cdf:CD630_29190', '272563.8.peg.3054']\n",
      "['Q184A1_CLOD6', 'none', 'WP_011861733.1;YP_001089427.1', 'cdf:CD630_29110', '272563.8.peg.3047']\n",
      "['Q17ZS7_CLOD6', 'none', 'WP_011572945.1;YP_659580.1', 'cdf:CD630_p040', '272563.8.peg.3']\n",
      "['Q183S0_CLOD6', 'none', 'WP_011861708.1;YP_001089350.1', 'cdf:CD630_28370', '272563.8.peg.2965']\n",
      "['Q183S2_CLOD6', 'none', 'WP_009891132.1;YP_001089348.1', 'cdf:CD630_28350', '272563.8.peg.2963']\n",
      "['Q183Q2_CLOD6', 'none', 'WP_011861700.1;YP_001089328.1', 'cdf:CD630_28150', '272563.8.peg.2942']\n",
      "['Q183P6_CLOD6', 'none', 'WP_011861697.1;YP_001089325.1', 'cdf:CD630_28120', '272563.8.peg.2939']\n",
      "['Q183N7_CLOD6', 'none', 'WP_011861693.1;YP_001089313.1', 'cdf:CD630_28000', '272563.8.peg.2928']\n",
      "['Q183N0_CLOD6', 'none', 'WP_011861688.1;YP_001089308.1', 'cdf:CD630_27950', '272563.8.peg.2922']\n",
      "['Q183M3_CLOD6', 'none', 'WP_011861678.1;YP_001089298.1', 'cdf:CD630_27850', '272563.8.peg.2912']\n",
      "['Q183L4_CLOD6', 'none', 'WP_004454110.1;YP_001089288.1', 'cdf:CD630_27750', '272563.8.peg.2902']\n",
      "['Q183K6_CLOD6', 'none', 'WP_011861674.1;YP_001089281.1', 'cdf:CD630_27680', '272563.8.peg.2895']\n",
      "['Q18D06_CLOD6', 'none', 'WP_011860710.1;YP_001086736.1', 'cdf:CD630_02670', '272563.8.peg.286']\n",
      "['Q183F0_CLOD6', 'none', 'WP_011861658.1;YP_001089229.1', 'cdf:CD630_27170', '272563.8.peg.2843']\n",
      "['Q183E5_CLOD6', 'none', 'WP_011861654.1;YP_001089223.2', 'cdf:CD630_27120', '272563.8.peg.2837']\n",
      "['Q183E1_CLOD6', 'none', 'WP_011861652.1;YP_001089217.1', 'cdf:CD630_27060', '272563.8.peg.2831']\n",
      "['Q183C0_CLOD6', 'none', 'WP_004455001.1;YP_001089197.2', 'cdf:CD630_26860', '272563.8.peg.2810']\n",
      "['Q183C1_CLOD6', 'none', 'WP_011861643.1;YP_001089196.1', 'cdf:CD630_26850', '272563.8.peg.2809']\n",
      "['Q183B7_CLOD6', 'none', 'WP_011861642.1;YP_001089195.1', 'cdf:CD630_26840', '272563.8.peg.2808']\n",
      "['Q183A3_CLOD6', 'none', 'WP_011861633.1;YP_001089179.1', 'cdf:CD630_26680', '272563.8.peg.2792']\n",
      "['Q182Y4_CLOD6', 'none', 'WP_011861624.1;YP_001089161.1', 'cdf:CD630_26500', '272563.8.peg.2774']\n",
      "['Q182X4_CLOD6', 'none', 'WP_011861621.1;YP_001089150.1', 'cdf:CD630_26390', '272563.8.peg.2763']\n",
      "['Q182W7_CLOD6', 'none', 'WP_011861620.1;YP_001089144.1', 'cdf:CD630_26330', '272563.8.peg.2757']\n",
      "['Q182V8_CLOD6', 'none', 'WP_003426933.1;YP_001089134.1', 'cdf:CD630_26230', '272563.8.peg.2746']\n",
      "['Q182U7_CLOD6', 'none', 'WP_003426910.1;YP_001089123.1', 'cdf:CD630_26120', '272563.8.peg.2735']\n",
      "['Q18CZ7_CLOD6', 'none', 'WP_011860701.1;YP_001086720.2', 'cdf:CD630_02520', '272563.8.peg.270']\n",
      "['Q18CY9_CLOD6', 'none', 'WP_011860699.1;YP_001086718.1', 'cdf:CD630_02500', '272563.8.peg.268']\n",
      "['Q182P1_CLOD6', 'none', 'WP_011861600.1;YP_001089068.1', 'cdf:CD630_25531', '272563.8.peg.2676']\n",
      "['Q182N1_CLOD6', 'none', 'WP_011861593.1;YP_001089058.1', 'cdf:CD630_25440', '272563.8.peg.2667']\n",
      "['Q182M2_CLOD6', 'none', 'WP_011861588.1;YP_001089049.2', 'cdf:CD630_25350', '272563.8.peg.2658']\n",
      "['Q182L4_CLOD6', 'none', 'WP_011861583.1;YP_001089042.1', 'cdf:CD630_25280', '272563.8.peg.2651']\n",
      "['Q182J4_CLOD6', 'none', 'WP_009897697.1;YP_001089021.2', 'cdf:CD630_25080', '272563.8.peg.2630']\n",
      "['Q18CY0_CLOD6', 'none', 'WP_011860695.1;YP_001086712.1', 'cdf:CD630_02440', '272563.8.peg.262']\n",
      "['Q182H7_CLOD6', 'none', 'WP_011861571.1;YP_001089004.1', 'cdf:CD630_24900', '272563.8.peg.2612']\n",
      "['Q182G0_CLOD6', 'none', 'WP_011861567.1;YP_001088988.1', 'cdf:CD630_24740', '272563.8.peg.2596']\n",
      "['Q182F3_CLOD6', 'none', 'WP_011861564.1;YP_001088980.1', 'cdf:CD630_24660', '272563.8.peg.2588']\n",
      "['Q182D0_CLOD6', 'none', 'WP_003416626.1;YP_001088956.1', 'cdf:CD630_24430', '272563.8.peg.2563']\n",
      "['Q182C7_CLOD6', 'none', 'WP_011861553.1;YP_001088954.1', 'cdf:CD630_24411', '272563.8.peg.2561']\n",
      "['Q182C2_CLOD6', 'none', 'WP_003430887.1;YP_001088949.1', 'cdf:CD630_24360', '272563.8.peg.2556']\n",
      "['Q181Z5_CLOD6', 'none', 'WP_004454705.1;YP_001088921.1', 'cdf:CD630_24090', '272563.8.peg.2527']\n",
      "['Q181Y3_CLOD6', 'none', 'WP_003416732.1;YP_001088911.1', 'cdf:CD630_23990;pdc:CDIF630_02641', '272563.8.peg.2517']\n",
      "['Q181X5_CLOD6', 'none', 'WP_011861542.1;YP_001088903.1', 'cdf:CD630_23910', '272563.8.peg.2509']\n",
      "['Q181X3_CLOD6', 'none', 'WP_011861541.1;YP_001088901.1', 'cdf:CD630_23890', '272563.8.peg.2507']\n",
      "['Q181X0_CLOD6', 'none', 'WP_011861540.1;YP_001088896.1', 'cdf:CD630_23840', '272563.8.peg.2501']\n",
      "['Q181W2_CLOD6', 'none', 'WP_011861539.1;YP_001088890.1', 'cdf:CD630_23780', '272563.8.peg.2495']\n",
      "['Q185P3_CLOD6', 'none', 'WP_009890565.1;YP_001088885.1', 'cdf:CD630_23730', '272563.8.peg.2490']\n",
      "['Q185N9_CLOD6', 'none', 'WP_003430783.1;YP_001088880.1', 'cdf:CD630_23680', '272563.8.peg.2485']\n",
      "['Q185M9_CLOD6', 'none', 'WP_009897539.1;YP_001088870.1', 'cdf:CD630_23570', '272563.8.peg.2472']\n",
      "['Q185L8_CLOD6', 'none', 'WP_011861533.1;YP_001088859.1', 'cdf:CD630_23440', '272563.8.peg.2461']\n",
      "['Q185K2_CLOD6', 'none', 'WP_011861527.1;YP_001088843.1', 'cdf:CD630_23280', '272563.8.peg.2445']\n",
      "['Q185F9_CLOD6', 'none', 'WP_011861512.1;YP_001088800.1', 'cdf:CD630_22880', '272563.8.peg.2403']\n",
      "['Q185E4_CLOD6', 'none', 'WP_009890462.1;YP_001088784.1', 'cdf:CD630_22720', '272563.8.peg.2386']\n",
      "['Q185D7_CLOD6', 'none', 'WP_011861505.1;YP_001088779.1', 'cdf:CD630_22650', '272563.8.peg.2380']\n",
      "['Q186B8_CLOD6', 'none', 'WP_011861500.1;YP_001088743.1', 'cdf:CD630_22290', '272563.8.peg.2344']\n",
      "['Q186B9_CLOD6', 'none', 'WP_011861499.1;YP_001088742.1', 'cdf:CD630_22280', '272563.8.peg.2343']\n",
      "['Q186B3_CLOD6', 'none', 'WP_011861498.1;YP_001088741.1', 'cdf:CD630_22270', '272563.8.peg.2342']\n",
      "['Q185Z9_CLOD6', 'none', 'WP_011861493.1;YP_001088723.1', 'cdf:CD630_22090', '272563.8.peg.2324']\n",
      "['Q185Z0_CLOD6', 'none', 'WP_003423398.1;YP_001088717.1', 'cdf:CD630_22030', '272563.8.peg.2318']\n",
      "['Q185Y3_CLOD6', 'none', 'WP_011861489.1;YP_001088707.1', 'cdf:CD630_21940', '272563.8.peg.2308']\n",
      "['Q185X3_CLOD6', 'none', 'WP_011861485.1;YP_001088698.1', 'cdf:CD630_21850', '272563.8.peg.2300']\n",
      "['Q185W7_CLOD6', 'none', 'WP_011861482.1;YP_001088693.1', 'cdf:CD630_21800', '272563.8.peg.2295']\n",
      "['Q185V7_CLOD6', 'none', 'WP_011861479.1;YP_001088683.2', 'cdf:CD630_21700', '272563.8.peg.2284']\n",
      "['Q185V6_CLOD6', 'none', 'WP_011861478.1;YP_001088680.1', 'cdf:CD630_21670', '272563.8.peg.2281']\n",
      "['Q185T8_CLOD6', 'none', 'WP_003430586.1;YP_001088664.2', 'cdf:CD630_21510', '272563.8.peg.2264']\n",
      "['Q185Q7_CLOD6', 'none', 'WP_011861458.1;YP_001088634.1', 'cdf:CD630_21200', '272563.8.peg.2231']\n",
      "['Q185Q3_CLOD6', 'none', 'WP_011861456.1;YP_001088626.1', 'cdf:CD630_21120', '272563.8.peg.2220']\n",
      "['Q188E4_CLOD6', 'none', 'WP_011861446.1;YP_001088590.1', 'cdf:CD630_20760', '272563.8.peg.2183']\n",
      "['Q188E0_CLOD6', 'none', 'WP_011861442.1;YP_001088584.1', 'cdf:CD630_20700', '272563.8.peg.2177']\n",
      "['Q188D1_CLOD6', 'none', 'WP_003435125.1;YP_001088578.1', 'cdf:CD630_20640', '272563.8.peg.2171']\n",
      "['Q18CT7_CLOD6', 'none', 'WP_011860665.1;YP_001086667.1', 'cdf:CD630_01990', '272563.8.peg.217']\n",
      "['Q188C5_CLOD6', 'none', 'WP_003428987.1;YP_001088571.1', 'cdf:CD630_20570', '272563.8.peg.2164']\n",
      "['Q188C3_CLOD6', 'none', 'WP_003435139.1;YP_001088569.1', 'cdf:CD630_20550', '272563.8.peg.2162']\n",
      "['Q188B4_CLOD6', 'none', 'WP_003428956.1;YP_001088560.1', 'cdf:CD630_20460', '272563.8.peg.2152']\n",
      "['Q187W5_CLOD6', 'none', 'WP_011861422.1;YP_001088509.1', 'cdf:CD630_20010', '272563.8.peg.2106']\n",
      "['Q187S8_CLOD6', 'none', 'WP_003428769.1;YP_001088475.1', 'cdf:CD630_19690', '272563.8.peg.2065']\n",
      "['Q187R7_CLOD6', 'none', 'WP_003435305.1;YP_001088464.1', 'cdf:CD630_19580', '272563.8.peg.2054']\n",
      "['Q187Q2_CLOD6', 'none', 'WP_011861396.1;YP_001088447.1', 'cdf:CD630_19410', '272563.8.peg.2037']\n",
      "['Q187P9_CLOD6', 'none', 'WP_011861395.1;YP_001088446.1', 'cdf:CD630_19400', '272563.8.peg.2036']\n",
      "['Q187L7_CLOD6', 'none', 'WP_011861385.1;YP_001088413.1', 'cdf:CD630_19060', '272563.8.peg.2002']\n",
      "['Q18C96_CLOD6', 'none', 'WP_011860621.1;YP_001086475.1', 'cdf:CD630_00120', '272563.8.peg.20']\n",
      "['Q17ZS8_CLOD6', 'none', 'WP_011572944.1;YP_659579.1', 'cdf:CD630_p030', '272563.8.peg.2']\n",
      "['Q187L2_CLOD6', 'none', 'WP_009902829.1;YP_001088410.1', 'cdf:CD630_19030', '272563.8.peg.1998']\n",
      "['Q187J5_CLOD6', 'none', 'WP_011861375.1;YP_001088390.1', 'cdf:CD630_18820', '272563.8.peg.1977']\n",
      "['Q187G9_CLOD6', 'none', 'WP_011861364.1;YP_001088365.1', 'cdf:CD630_18580', '272563.8.peg.1950']\n",
      "['Q187G3_CLOD6', 'none', 'WP_011861360.1;YP_001088361.1', 'cdf:CD630_18550', '272563.8.peg.1946']\n",
      "['Q187F7_CLOD6', 'none', 'WP_011861355.1;YP_001088355.1', 'cdf:CD630_18490', '272563.8.peg.1940']\n",
      "['Q18CR7_CLOD6', 'none', 'WP_011860659.1;YP_001086645.1', 'cdf:CD630_01750', '272563.8.peg.194']\n",
      "['Q187F3_CLOD6', 'none', 'WP_011861350.1;YP_001088349.1', 'cdf:CD630_18430;pdc:CDIF630_02045', '272563.8.peg.1934']\n",
      "['Q187F0_CLOD6', 'none', 'WP_011861348.1;YP_001088346.1', 'cdf:CD630_18400', '272563.8.peg.1931']\n",
      "['Q187D0_CLOD6', 'none', 'WP_011861342.1;YP_001088326.1', 'cdf:CD630_18210', '272563.8.peg.1911']\n",
      "['Q187B9_CLOD6', 'none', 'WP_009902781.1;YP_001088318.1', 'cdf:CD630_18130', '272563.8.peg.1903']\n",
      "['Q18CR4_CLOD6', 'none', 'WP_011860658.1;YP_001086641.1', 'cdf:CD630_01720', '272563.8.peg.190']\n",
      "['Q187B2_CLOD6', 'none', 'WP_011861337.1;YP_001088307.1', 'cdf:CD630_18010', '272563.8.peg.1890']\n",
      "['Q187A6_CLOD6', 'none', 'WP_003430297.1;YP_001088300.1', 'cdf:CD630_17940;pdc:CDIF630_01993', '272563.8.peg.1883']\n",
      "['Q186Z9_CLOD6', 'none', 'WP_011861336.1;YP_001088296.1', 'cdf:CD630_17900', '272563.8.peg.1879']\n",
      "['Q186Z2_CLOD6', 'none', 'WP_003430276.1;YP_001088288.1', 'cdf:CD630_17820', '272563.8.peg.1870']\n",
      "['Q18CQ6_CLOD6', 'none', 'WP_011860657.1;YP_001086637.1', 'cdf:CD630_01680', '272563.8.peg.186']\n",
      "['Q186X8_CLOD6', 'none', 'WP_011861331.1;YP_001088274.2', 'cdf:CD630_17680', '272563.8.peg.1856']\n",
      "['Q18CQ3_CLOD6', 'none', 'WP_009888010.1;YP_001086632.1', 'cdf:CD630_01632;pdc:CDIF630_00284', '272563.8.peg.181']\n",
      "['Q186T3_CLOD6', 'none', 'WP_003430200.1;YP_001088230.1', 'cdf:CD630_17270', '272563.8.peg.1807']\n",
      "['X1W3M5_CLOD6', 'none', 'WP_033562367.1;YP_009109553.1', 'cdf:CD630_17180', '272563.8.peg.1800']\n",
      "['Q186S7_CLOD6', 'none', 'WP_011861313.1;YP_001088223.1', 'cdf:CD630_17190', '272563.8.peg.1799']\n",
      "['Q186P2_CLOD6', 'none', 'WP_011861307.1;YP_001088190.1', 'cdf:CD630_16870', '272563.8.peg.1765']\n",
      "['Q186L7_CLOD6', 'none', 'WP_011861290.1;YP_001088163.1', 'cdf:CD630_16610', '272563.8.peg.1738']\n",
      "['Q186J9_CLOD6', 'none', 'WP_011861282.1;YP_001088145.1', 'cdf:CD630_16440', '272563.8.peg.1721']\n",
      "['Q186J7_CLOD6', 'none', 'WP_011861279.1;YP_001088141.1', 'cdf:CD630_16400', '272563.8.peg.1717']\n",
      "['Q18C94_CLOD6', 'none', 'WP_011860620.1;YP_001086472.1', 'cdf:CD630_00090', '272563.8.peg.17']\n",
      "['Q186G3_CLOD6', 'none', 'WP_011861270.1;YP_001088111.1', 'cdf:CD630_16100', '272563.8.peg.1687']\n",
      "['Q186F6_CLOD6', 'none', 'WP_003436484.1;YP_001088104.1', 'cdf:CD630_16030', '272563.8.peg.1680']\n",
      "['Q186F3_CLOD6', 'none', 'WP_011861268.1;YP_001088098.1', 'cdf:CD630_15970', '272563.8.peg.1674']\n",
      "['Q186E7_CLOD6', 'none', 'WP_011861267.1;YP_001088092.1', 'cdf:CD630_15910', '272563.8.peg.1667']\n",
      "['Q186C5_CLOD6', 'none', 'WP_011861256.1;YP_001088072.1', 'cdf:CD630_15720', '272563.8.peg.1646']\n",
      "['Q18CN8_CLOD6', 'none', 'WP_011860648.1;YP_001086616.1', 'cdf:CD630_01470', '272563.8.peg.164']\n",
      "['Q18CN5_CLOD6', 'none', 'WP_011860646.1;YP_001086613.1', 'cdf:CD630_01440', '272563.8.peg.161']\n",
      "['Q18C28_CLOD6', 'none', 'WP_009889347.1;YP_001088008.1', 'cdf:CD630_15071', '272563.8.peg.1580']\n",
      "['Q18C19_CLOD6', 'none', 'WP_011861238.1;YP_001088001.1', 'cdf:CD630_15000', '272563.8.peg.1572']\n",
      "['Q18C16_CLOD6', 'none', 'WP_011861237.1;YP_001087999.1', 'cdf:CD630_14980', '272563.8.peg.1570']\n",
      "['Q18C01_CLOD6', 'none', 'WP_003434129.1;YP_001087981.1', 'cdf:CD630_14800', '272563.8.peg.1552']\n",
      "['Q18BZ6_CLOD6', 'none', 'WP_011861231.1;YP_001087976.2', 'cdf:CD630_14750', '272563.8.peg.1547']\n",
      "['Q18BY8_CLOD6', 'none', 'WP_003438661.1;YP_001087969.1', 'cdf:CD630_14680', '272563.8.peg.1540']\n",
      "['Q18BY5_CLOD6', 'none', 'WP_011861225.1;YP_001087966.1', 'cdf:CD630_14650', '272563.8.peg.1537']\n",
      "['Q18BX8_CLOD6', 'none', 'WP_011861221.1;YP_001087959.1', 'cdf:CD630_14580', '272563.8.peg.1530']\n",
      "['Q18BX5_CLOD6', 'none', 'WP_011861220.1;YP_001087956.1', 'cdf:CD630_14550', '272563.8.peg.1527']\n",
      "['Q18BV6_CLOD6', 'none', 'WP_011861214.1;YP_001087939.1', 'cdf:CD630_14380', '272563.8.peg.1510']\n",
      "['Q18BV4_CLOD6', 'none', 'WP_011861212.1;YP_001087935.1', 'cdf:CD630_14340', '272563.8.peg.1506']\n",
      "['Q18CM3_CLOD6', 'none', 'WP_011860642.1;YP_001086603.1', 'cdf:CD630_01340', '272563.8.peg.150']\n",
      "['Q18BT6_CLOD6', 'none', 'WP_003429358.1;YP_001087917.1', 'cdf:CD630_14170', '272563.8.peg.1489']\n",
      "['Q18BT3_CLOD6', 'none', 'WP_011861205.1;YP_001087914.1', 'cdf:CD630_14140', '272563.8.peg.1486']\n",
      "['Q18BS6_CLOD6', 'none', 'WP_011861203.1;YP_001087906.1', 'cdf:CD630_14060', '272563.8.peg.1478']\n",
      "['Q18BR3_CLOD6', 'none', 'WP_003438483.1;YP_001087894.1', 'cdf:CD630_13950', '272563.8.peg.1466']\n",
      "['Q18BR1_CLOD6', 'none', 'WP_011861194.1;YP_001087890.1', 'cdf:CD630_13910', '272563.8.peg.1462']\n",
      "['Q18CL8_CLOD6', 'none', 'WP_003421842.1;YP_001086598.1', 'cdf:CD630_01290', '272563.8.peg.145']\n",
      "['Q18BP6_CLOD6', 'none', 'WP_003428145.1;YP_001087878.1', 'cdf:CD630_13782', '272563.8.peg.1449']\n",
      "['Q18BH8_CLOD6', 'none', 'WP_011861173.1;YP_001087808.1', 'cdf:CD630_13120', '272563.8.peg.1379']\n",
      "['Q18BH1_CLOD6', 'none', 'WP_003419469.1;YP_001087803.1', 'cdf:CD630_13071', '272563.8.peg.1374']\n",
      "['Q18BG1_CLOD6', 'none', 'WP_003428254.1;YP_001087792.1', 'cdf:CD630_12980', '272563.8.peg.1363']\n",
      "['Q18BF5_CLOD6', 'none', 'WP_003428269.1;YP_001087784.1', 'cdf:CD630_12900;pdc:CDIF630_01444', '272563.8.peg.1355']\n",
      "['Q18BD7_CLOD6', 'none', 'WP_011861164.1;YP_001087767.1', 'cdf:CD630_12730', '272563.8.peg.1338']\n",
      "['Q18BC0_CLOD6', 'none', 'WP_011861161.1;YP_001087752.1', 'cdf:CD630_12580', '272563.8.peg.1323']\n",
      "['Q18BA6_CLOD6', 'none', 'WP_011861155.1;YP_001087735.1', 'cdf:CD630_12420', '272563.8.peg.1306']\n",
      "['Q18B74_CLOD6', 'none', 'WP_003424711.1;YP_001087707.1', 'cdf:CD630_12140', '272563.8.peg.1273']\n",
      "['NUSB_CLOD6', 'none', 'WP_011861139.1;YP_001087694.1', 'cdf:CD630_12010', '272563.8.peg.1260']\n",
      "['Q18B59_CLOD6', 'none', 'WP_011861138.1;YP_001087689.1', 'cdf:CD630_11960', '272563.8.peg.1255']\n",
      "['Q18B49_CLOD6', 'none', 'WP_011861136.1;YP_001087682.1', 'cdf:CD630_11890', '272563.8.peg.1248']\n",
      "['Q18B51_CLOD6', 'none', 'WP_003438109.1;YP_001087680.1', 'cdf:CD630_11870', '272563.8.peg.1246']\n",
      "['Q18B33_CLOD6', 'none', 'WP_011861132.1;YP_001087663.1', 'cdf:CD630_11710', '272563.8.peg.1229']\n",
      "['Q18B30_CLOD6', 'none', 'WP_011861131.1;YP_001087659.1', 'cdf:CD630_11670', '272563.8.peg.1225']\n",
      "['Q18B04_CLOD6', 'none', 'WP_011861126.1;YP_001087638.1', 'cdf:CD630_11460', '272563.8.peg.1204']\n",
      "['Q18AZ7_CLOD6', 'none', 'WP_011861123.1;YP_001087626.1', 'cdf:CD630_11350', '272563.8.peg.1192']\n",
      "['Q18AY5_CLOD6', 'none', 'WP_003428522.1;YP_001087614.1', 'cdf:CD630_11241;pdc:CDIF630_01269', '272563.8.peg.1180']\n",
      "['Q18AX2_CLOD6', 'none', 'WP_011861119.1;YP_001087607.1', 'cdf:CD630_11180', '272563.8.peg.1173']\n",
      "['Q18AW8_CLOD6', 'none', 'WP_002610321.1;YP_001087602.1', 'cdf:CD630_11130', '272563.8.peg.1170']\n",
      "['Q18AW9_CLOD6', 'none', 'WP_002610323.1;YP_001087601.1', 'cdf:CD630_11120', '272563.8.peg.1169']\n",
      "['Q18AV9_CLOD6', 'none', 'WP_011861110.1;YP_001087595.1', 'cdf:CD630_11070', '272563.8.peg.1161']\n",
      "['Q18AV2_CLOD6', 'none', 'WP_011861098.1;YP_001087583.1', 'cdf:CD630_10991', '272563.8.peg.1152']\n",
      "['Q18AT9_CLOD6', 'none', 'WP_003438018.1;YP_001087572.1', 'cdf:CD630_10880', '272563.8.peg.1142']\n",
      "['Q18AS7_CLOD6', 'none', 'WP_003438000.1;YP_001087561.1', 'cdf:CD630_10770', '272563.8.peg.1131']\n",
      "['Q18AS4_CLOD6', 'none', 'WP_011861086.1;YP_001087557.1', 'cdf:CD630_10730', '272563.8.peg.1127']\n",
      "['Q18AR9_CLOD6', 'none', 'WP_003437991.1;YP_001087554.1', 'cdf:CD630_10700', '272563.8.peg.1124']\n",
      "['Q18AS1_CLOD6', 'none', 'WP_003437987.1;YP_001087552.1', 'cdf:CD630_10680', '272563.8.peg.1122']\n",
      "['Q18AR7_CLOD6', 'none', 'WP_003428554.1;YP_001087549.1', 'cdf:CD630_10650', '272563.8.peg.1119']\n",
      "['Q18AR5_CLOD6', 'none', 'WP_011861084.1;YP_001087542.1', 'cdf:CD630_10610', '272563.8.peg.1113']\n",
      "['Q18AN4_CLOD6', 'none', 'WP_009895992.1;YP_001087513.1', 'cdf:CD630_10320', '272563.8.peg.1084']\n",
      "['Q18AM1_CLOD6', 'none', 'WP_011861070.1;YP_001087501.1', 'cdf:CD630_10200', '272563.8.peg.1072']\n",
      "['Q18AL3_CLOD6', 'none', 'WP_011861066.1;YP_001087495.1', 'cdf:CD630_10140', '272563.8.peg.1065']\n",
      "['Q18AJ5_CLOD6', 'none', 'WP_003437086.1;YP_001087474.1', 'cdf:CD630_09930', '272563.8.peg.1044']\n",
      "['Q18AI3_CLOD6', 'none', 'WP_011861056.1;YP_001087462.1', 'cdf:CD630_09810', '272563.8.peg.1032']\n",
      "['Q18AH7_CLOD6', 'none', 'WP_011861053.1;YP_001087458.1', 'cdf:CD630_09770', '272563.8.peg.1027']\n",
      "['Q183Y9_CLOD6', 'none', 'WP_011861043.1;YP_001087448.1;YP_001089411.1', 'cdf:CD630_09670;cdf:CD630_28970', '272563.8.peg.1018']\n",
      "['Q18AF3_CLOD6', 'none', 'WP_011861033.1;YP_001087436.1', 'cdf:CD630_09561', '272563.8.peg.1005']\n",
      "['Q184A2_CLOD6', 'none', 'WP_011861031.1;YP_001087432.1;YP_001089426.2', 'cdf:CD630_09531;cdf:CD630_29101', '272563.8.peg.1001']\n",
      "['Q17ZS9_CLOD6', 'none', 'WP_011572943.1;YP_659578.1', 'cdf:CD630_p020', '272563.8.peg.1']\n",
      "['Q18CH6_CLOD6', 'none', 'WP_003427725.1;YP_001086557.1', 'cdf:CD630_00900;pdc:CDIF630_00159', '272563.120.peg.99']\n",
      "['RL15_CLOD6', 'none', 'WP_003421135.1;YP_001086556.1', 'cdf:CD630_00890;pdc:CDIF630_00158', '272563.120.peg.98']\n",
      "['RL30_CLOD6', 'none', 'WP_003421137.1;YP_001086555.1', 'cdf:CD630_00881;pdc:CDIF630_00157', '272563.120.peg.97']\n",
      "['Q18AD8_CLOD6', 'none', 'WP_011861021.1;YP_001087413.1', 'cdf:CD630_09370;pdc:CDIF630_01063', '272563.120.peg.967']\n",
      "['Q18AD1_CLOD6', 'none', 'WP_011861020.1;YP_001087412.1', 'cdf:CD630_09360;pdc:CDIF630_01062', '272563.120.peg.966']\n",
      "['Q18AC6_CLOD6', 'none', 'WP_011861019.1;YP_001087411.1', 'cdf:CD630_09350;pdc:CDIF630_01061', '272563.120.peg.965']\n",
      "['Q18AC9_CLOD6', 'none', 'WP_011861016.1;YP_001087408.1', 'cdf:CD630_09330;pdc:CDIF630_01059', '272563.120.peg.962']\n",
      "['Q18AD0_CLOD6', 'none', 'WP_011861015.1;YP_001087407.1', 'cdf:CD630_09320;pdc:CDIF630_01058', '272563.120.peg.961']\n",
      "['Q18AC5_CLOD6', 'none', 'WP_011861014.1;YP_001087406.1', 'cdf:CD630_09310;pdc:CDIF630_01057', '272563.120.peg.960']\n",
      "['RS5_CLOD6', 'none', 'WP_003421138.1;YP_001086554.1', 'cdf:CD630_00880;pdc:CDIF630_00156', '272563.120.peg.96']\n",
      "['Q18AC2_CLOD6', 'none', 'WP_011861012.1;YP_001087404.1', 'cdf:CD630_09300;pdc:CDIF630_01055', '272563.120.peg.958']\n",
      "['Q18AC3_CLOD6', 'none', 'WP_011861011.1;YP_001087403.1', 'cdf:CD630_09290;pdc:CDIF630_01054', '272563.120.peg.957']\n",
      "['Q18AC4_CLOD6', 'none', 'WP_011861010.1;YP_001087402.1', 'cdf:CD630_09280;pdc:CDIF630_01053', '272563.120.peg.956']\n",
      "['Q18AC0_CLOD6', 'none', 'WP_011861009.1;YP_001087401.1', 'cdf:CD630_09270;pdc:CDIF630_01052', '272563.120.peg.955']\n",
      "['Q18AB2_CLOD6', 'none', 'WP_011861008.1;YP_001087400.1', 'cdf:CD630_09260;pdc:CDIF630_01051', '272563.120.peg.954']\n",
      "['Q18AB3_CLOD6', 'none', 'WP_011861007.1;YP_001087399.1', 'cdf:CD630_09250;pdc:CDIF630_01050', '272563.120.peg.953']\n",
      "['F3Y628_CLOD6', 'none', 'WP_011861006.1;YP_006589360.1', 'cdf:CD630_09241;pdc:CDIF630_01049', '272563.120.peg.952']\n",
      "['Q18AB4_CLOD6', 'none', 'WP_011861005.1;YP_001087398.1', 'cdf:CD630_09240;pdc:CDIF630_01048', '272563.120.peg.951']\n",
      "['F3Y627_CLOD6', 'none', 'WP_011861004.1;YP_006589359.1', 'cdf:CD630_09231;pdc:CDIF630_01047', '272563.120.peg.950']\n",
      "['RL18_CLOD6', 'none', 'WP_003421141.1;YP_001086553.1', 'cdf:CD630_00870;pdc:CDIF630_00155', '272563.120.peg.95']\n",
      "['Q18AB6_CLOD6', 'none', 'WP_011861002.1;YP_001087396.1', 'cdf:CD630_09220;pdc:CDIF630_01045', '272563.120.peg.949']\n",
      "['Q18AB7_CLOD6', 'none', 'WP_011861001.1;YP_001087395.1', 'cdf:CD630_09210;pdc:CDIF630_01044', '272563.120.peg.948']\n",
      "['Q18AB8_CLOD6', 'none', 'WP_011861000.1;YP_001087394.1', 'cdf:CD630_09200;pdc:CDIF630_01043', '272563.120.peg.947']\n",
      "['Q18AA6_CLOD6', 'none', 'WP_011860998.1;YP_001087392.1', 'cdf:CD630_09180;pdc:CDIF630_01041', '272563.120.peg.945']\n",
      "['Q18AA7_CLOD6', 'none', 'WP_011860997.1;YP_001087391.1', 'cdf:CD630_09170;pdc:CDIF630_01040', '272563.120.peg.944']\n",
      "['Q18AA8_CLOD6', 'none', 'WP_011860996.1;YP_001087390.1', 'cdf:CD630_09160;pdc:CDIF630_01039', '272563.120.peg.943']\n",
      "['Q18AA9_CLOD6', 'none', 'WP_011860995.1;YP_001087389.1', 'cdf:CD630_09150;pdc:CDIF630_01038', '272563.120.peg.942']\n",
      "['Q18AB1_CLOD6', 'none', 'WP_011860993.1;YP_001087387.1', 'cdf:CD630_09130;pdc:CDIF630_01036', '272563.120.peg.940']\n",
      "['RL6_CLOD6', 'none', 'WP_003435668.1;YP_001086552.1', 'cdf:CD630_00860;pdc:CDIF630_00154', '272563.120.peg.94']\n",
      "['Q18AA1_CLOD6', 'none', 'WP_011860992.1;YP_001087386.1', 'cdf:CD630_09120;pdc:CDIF630_01035', '272563.120.peg.939']\n",
      "['Q18AA2_CLOD6', 'none', 'WP_011860991.1;YP_001087385.1', 'cdf:CD630_09110;pdc:CDIF630_01034', '272563.120.peg.938']\n",
      "['Q18AA4_CLOD6', 'none', 'WP_011860989.1;YP_001087383.1', 'cdf:CD630_09090;pdc:CDIF630_01032', '272563.120.peg.936']\n",
      "['Q18AA5_CLOD6', 'none', 'WP_011860988.1;YP_001087382.1', 'cdf:CD630_09080;pdc:CDIF630_01031', '272563.120.peg.935']\n",
      "['Q18AA0_CLOD6', 'none', 'WP_011860987.1;YP_001087381.1', 'cdf:CD630_09071;pdc:CDIF630_01030', '272563.120.peg.934']\n",
      "['Q18A97_CLOD6', 'none', 'WP_011860986.1;YP_001087380.1', 'cdf:CD630_09070;pdc:CDIF630_01029', '272563.120.peg.933']\n",
      "['Q18A98_CLOD6', 'none', 'WP_011860985.1;YP_001087379.1', 'cdf:CD630_09060;pdc:CDIF630_01027', '272563.120.peg.931']\n",
      "['Q18A99_CLOD6', 'none', 'WP_011860984.1;YP_001087378.1', 'cdf:CD630_09050;pdc:CDIF630_01026', '272563.120.peg.930']\n",
      "['RS8_CLOD6', 'none', 'WP_003421148.1;YP_001086551.1', 'cdf:CD630_00850;pdc:CDIF630_00153', '272563.120.peg.93']\n",
      "['Q18A96_CLOD6', 'none', 'WP_011860983.1;YP_001087377.1', 'cdf:CD630_09040;pdc:CDIF630_01024', '272563.120.peg.928']\n",
      "['Q18A94_CLOD6', 'none', 'WP_003437125.1;YP_001087376.1', 'cdf:CD630_09020;pdc:CDIF630_01022', '272563.120.peg.926']\n",
      "['Y898_CLOD6', 'none', 'WP_003437136.1;YP_001087372.1', 'cdf:CD630_08980;pdc:CDIF630_01018', '272563.120.peg.922']\n",
      "['Q18A93_CLOD6', 'none', 'WP_003437137.1;YP_001087371.1', 'cdf:CD630_08970;pdc:CDIF630_01017', '272563.120.peg.921']\n",
      "['Q18A88_CLOD6', 'none', 'WP_003418667.1;YP_001087370.1', 'cdf:CD630_08960;pdc:CDIF630_01016', '272563.120.peg.920']\n",
      "['RS14Z_CLOD6', 'none', 'WP_003421149.1;YP_001086550.1', 'cdf:CD630_00841;pdc:CDIF630_00152', '272563.120.peg.92']\n",
      "['Q18A89_CLOD6', 'none', 'WP_011860980.1;YP_001087369.1', 'cdf:CD630_08950;pdc:CDIF630_01015', '272563.120.peg.919']\n",
      "['Q18A83_CLOD6', 'none', 'WP_003418660.1;YP_001087366.1', 'cdf:CD630_08920;pdc:CDIF630_01012', '272563.120.peg.916']\n",
      "['Q18A76_CLOD6', 'none', 'WP_011860978.1;YP_001087361.1', 'cdf:CD630_08870;pdc:CDIF630_01007', '272563.120.peg.911']\n",
      "['Q18CH2_CLOD6', 'none', 'WP_003435673.1;YP_001086549.1', 'cdf:CD630_00840;pdc:CDIF630_00151', '272563.120.peg.91']\n",
      "['Q18A80_CLOD6', 'none', 'WP_003418641.1;YP_001087357.1', 'cdf:CD630_08830;pdc:CDIF630_01003', '272563.120.peg.907']\n",
      "['Q18A72_CLOD6', 'none', 'WP_003437173.1;YP_001087355.1', 'cdf:CD630_08810;pdc:CDIF630_01001', '272563.120.peg.905']\n",
      "['Q18A73_CLOD6', 'none', 'WP_003437174.1;YP_001087354.1', 'cdf:CD630_08800;pdc:CDIF630_01000', '272563.120.peg.904']\n",
      "['Q18A71_CLOD6', 'none', 'WP_003439336.1;YP_001087350.1', 'cdf:CD630_08760;pdc:CDIF630_00996', '272563.120.peg.900']\n",
      "['Q18A67_CLOD6', 'none', 'WP_009902204.1;YP_001087349.1;YP_001087352.1', 'cdf:CD630_08750;cdf:CD630_08780;pdc:CDIF630_00995;pdc:CDIF630_00998', '272563.120.peg.899']\n",
      "['Q18A65_CLOD6', 'none', 'WP_011860976.1;YP_001087347.1', 'cdf:CD630_08730;pdc:CDIF630_00993', '272563.120.peg.897']\n",
      "['Q18A64_CLOD6', 'none', 'WP_011860975.1;YP_001087343.1', 'cdf:CD630_08690;pdc:CDIF630_00989', '272563.120.peg.893']\n",
      "['Q18A58_CLOD6', 'none', 'WP_004454197.1;YP_001087342.1', 'cdf:CD630_08680;pdc:CDIF630_00988', '272563.120.peg.892']\n",
      "['Q18A59_CLOD6', 'none', 'WP_011860974.1;YP_001087341.1', 'cdf:CD630_08670;pdc:CDIF630_00987', '272563.120.peg.891']\n",
      "['Q18A60_CLOD6', 'none', 'WP_004454199.1;YP_001087340.1', 'cdf:CD630_08660;pdc:CDIF630_00986', '272563.120.peg.890']\n",
      "['RL14_CLOD6', 'none', 'WP_003421155.1;YP_001086547.1', 'cdf:CD630_00820;pdc:CDIF630_00149', '272563.120.peg.89']\n",
      "['Q18A61_CLOD6', 'none', 'WP_011860973.1;YP_001087339.1', 'cdf:CD630_08650;pdc:CDIF630_00985', '272563.120.peg.889']\n",
      "['Q18A54_CLOD6', 'none', 'WP_009895893.1;YP_001087337.1', 'cdf:CD630_08630;pdc:CDIF630_00983', '272563.120.peg.887']\n",
      "['Q18A56_CLOD6', 'none', 'WP_004453591.1;YP_001087335.1', 'cdf:CD630_08610;pdc:CDIF630_00981', '272563.120.peg.885']\n",
      "['Q18A57_CLOD6', 'none', 'WP_004453592.1;YP_001087334.1', 'cdf:CD630_08600;pdc:CDIF630_00980', '272563.120.peg.884']\n",
      "['RS17_CLOD6', 'none', 'WP_003421156.1;YP_001086546.1', 'cdf:CD630_00810;pdc:CDIF630_00148', '272563.120.peg.88']\n",
      "['Q18A51_CLOD6', 'none', 'WP_004454209.1;YP_001087332.1', 'cdf:CD630_08550;pdc:CDIF630_00974', '272563.120.peg.878']\n",
      "['Q18A48_CLOD6', 'none', 'WP_009902199.1;YP_001087331.1', 'cdf:CD630_08540;pdc:CDIF630_00973', '272563.120.peg.877']\n",
      "['Q18A49_CLOD6', 'none', 'WP_009895878.1;YP_001087330.1', 'cdf:CD630_08530;pdc:CDIF630_00972', '272563.120.peg.876']\n",
      "['Q18A50_CLOD6', 'none', 'WP_009895876.1;YP_001087329.2', 'cdf:CD630_08520;pdc:CDIF630_00971', '272563.120.peg.875']\n",
      "['Q18A42_CLOD6', 'none', 'WP_004453603.1;YP_001087328.1', 'cdf:CD630_08510;pdc:CDIF630_00970', '272563.120.peg.874']\n",
      "['Q18A43_CLOD6', 'none', 'WP_009888660.1;YP_001087327.1', 'cdf:CD630_08500;pdc:CDIF630_00969', '272563.120.peg.873']\n",
      "['Q18A45_CLOD6', 'none', 'WP_009902197.1;YP_001087325.1', 'cdf:CD630_08480;pdc:CDIF630_00967', '272563.120.peg.871']\n",
      "['Q18A46_CLOD6', 'none', 'WP_009902196.1;YP_001087324.1', 'cdf:CD630_08470;pdc:CDIF630_00966', '272563.120.peg.870']\n",
      "['RL29_CLOD6', 'none', 'WP_003421158.1;YP_001086545.1', 'cdf:CD630_00801;pdc:CDIF630_00147', '272563.120.peg.87']\n",
      "['Q18A47_CLOD6', 'none', 'WP_011860972.1;YP_001087323.1', 'cdf:CD630_08460;pdc:CDIF630_00965', '272563.120.peg.869']\n",
      "['Q18A39_CLOD6', 'none', 'WP_011860971.1;YP_001087322.1', 'cdf:CD630_08450;pdc:CDIF630_00964', '272563.120.peg.868']\n",
      "['Q18A40_CLOD6', 'none', 'WP_004454223.1;YP_001087321.1', 'cdf:CD630_08440;pdc:CDIF630_00963', '272563.120.peg.867']\n",
      "['Q18A41_CLOD6', 'none', 'WP_009895871.1;YP_001087320.1', 'cdf:CD630_08430;pdc:CDIF630_00962', '272563.120.peg.866']\n",
      "['Q18A38_CLOD6', 'none', 'WP_004454225.1;YP_001087319.1', 'cdf:CD630_08420;pdc:CDIF630_00961', '272563.120.peg.865']\n",
      "['Q18A35_CLOD6', 'none', 'WP_011860969.1;YP_001087317.1', 'cdf:CD630_08400;pdc:CDIF630_00959', '272563.120.peg.863']\n",
      "['Q18A37_CLOD6', 'none', 'WP_004453624.1;YP_001087315.1', 'cdf:CD630_08380;pdc:CDIF630_00957', '272563.120.peg.861']\n",
      "['Q18A30_CLOD6', 'none', 'WP_009895862.1;YP_001087314.1', 'cdf:CD630_08370;pdc:CDIF630_00956', '272563.120.peg.860']\n",
      "['RL16_CLOD6', 'none', 'WP_003421160.1;YP_001086544.1', 'cdf:CD630_00800;pdc:CDIF630_00146', '272563.120.peg.86']\n",
      "['Q18A31_CLOD6', 'none', 'WP_004454230.1;YP_001087313.1', 'cdf:CD630_08360;pdc:CDIF630_00955', '272563.120.peg.859']\n",
      "['Q18A32_CLOD6', 'none', 'WP_009902189.1;YP_001087312.1', 'cdf:CD630_08350;pdc:CDIF630_00954', '272563.120.peg.858']\n",
      "['Q18A25_CLOD6', 'none', 'WP_009888643.1;YP_001087308.1', 'cdf:CD630_08310;pdc:CDIF630_00950', '272563.120.peg.854']\n",
      "['Q18A26_CLOD6', 'none', 'WP_004454238.1;YP_001087307.1', 'cdf:CD630_08300;pdc:CDIF630_00949', '272563.120.peg.853']\n",
      "['Q18A27_CLOD6', 'none', 'WP_011860968.1;YP_001087306.1', 'cdf:CD630_08290;pdc:CDIF630_00948', '272563.120.peg.852']\n",
      "['Q18A21_CLOD6', 'none', 'WP_009895854.1;YP_001087305.1', 'cdf:CD630_08280;pdc:CDIF630_00947', '272563.120.peg.851']\n",
      "['RS3_CLOD6', 'none', 'WP_003421163.1;YP_001086543.1', 'cdf:CD630_00790;pdc:CDIF630_00145', '272563.120.peg.85']\n",
      "['Q18A24_CLOD6', 'none', 'WP_003418513.1;YP_001087302.1', 'cdf:CD630_08250;pdc:CDIF630_00944', '272563.120.peg.848']\n",
      "['Q18A18_CLOD6', 'none', 'WP_004454242.1;YP_001087301.1', 'cdf:CD630_08240;pdc:CDIF630_00943', '272563.120.peg.847']\n",
      "['Q18A19_CLOD6', 'none', 'WP_009902187.1;YP_001087300.1', 'cdf:CD630_08230;pdc:CDIF630_00942', '272563.120.peg.846']\n",
      "['Q18A17_CLOD6', 'none', 'WP_009888625.1;YP_001087297.1', 'cdf:CD630_08200;pdc:CDIF630_00938', '272563.120.peg.842']\n",
      "['Q18A14_CLOD6', 'none', 'WP_004454252.1;YP_001087296.1', 'cdf:CD630_08190;pdc:CDIF630_00937', '272563.120.peg.841']\n",
      "['RL22_CLOD6', 'none', 'WP_009887891.1;YP_001086542.1', 'cdf:CD630_00780;pdc:CDIF630_00144', '272563.120.peg.84']\n",
      "['Q18A12_CLOD6', 'none', 'WP_011860963.1;YP_001087292.1', 'cdf:CD630_08150;pdc:CDIF630_00933', '272563.120.peg.837']\n",
      "['Q18A07_CLOD6', 'none', 'WP_003437572.1;YP_001087291.1', 'cdf:CD630_08140;pdc:CDIF630_00932', '272563.120.peg.836']\n",
      "['Q18A08_CLOD6', 'none', 'WP_003437569.1;YP_001087290.1', 'cdf:CD630_08130;pdc:CDIF630_00931', '272563.120.peg.835']\n",
      "['Q18A09_CLOD6', 'none', 'WP_003418487.1;YP_001087289.1', 'cdf:CD630_08120;pdc:CDIF630_00930', '272563.120.peg.834']\n",
      "['Q18A10_CLOD6', 'none', 'WP_011860962.1;YP_001087288.1', 'cdf:CD630_08110;pdc:CDIF630_00929', '272563.120.peg.833']\n",
      "['Q18A03_CLOD6', 'none', 'WP_009902172.1;YP_001087287.1', 'cdf:CD630_08100;pdc:CDIF630_00928', '272563.120.peg.832']\n",
      "['Q18A04_CLOD6', 'none', 'WP_003437564.1;YP_001087286.1', 'cdf:CD630_08090;pdc:CDIF630_00927', '272563.120.peg.831']\n",
      "['Q18A06_CLOD6', 'none', 'WP_003437561.1;YP_001087284.1', 'cdf:CD630_08070;pdc:CDIF630_00926', '272563.120.peg.830']\n",
      "['RS19_CLOD6', 'none', 'WP_003421166.1;YP_001086541.1', 'cdf:CD630_00770;pdc:CDIF630_00143', '272563.120.peg.83']\n",
      "['Q18A00_CLOD6', 'none', 'WP_003437559.1;YP_001087283.1', 'cdf:CD630_08060;pdc:CDIF630_00925', '272563.120.peg.829']\n",
      "['Q18A02_CLOD6', 'none', 'WP_003437555.1;YP_001087281.1', 'cdf:CD630_08040;pdc:CDIF630_00923', '272563.120.peg.827']\n",
      "['Q189Z5_CLOD6', 'none', 'WP_003437552.1;YP_001087278.1', 'cdf:CD630_08010;pdc:CDIF630_00920', '272563.120.peg.824']\n",
      "['Q189Z7_CLOD6', 'none', 'WP_009895828.1;YP_001087276.1', 'cdf:CD630_07990;pdc:CDIF630_00918', '272563.120.peg.822']\n",
      "['Q189Z2_CLOD6', 'none', 'WP_009888596.1;YP_001087275.1', 'cdf:CD630_07980;pdc:CDIF630_00917', '272563.120.peg.821']\n",
      "['Q189Z3_CLOD6', 'none', 'WP_011860958.1;YP_001087274.1', 'cdf:CD630_07970;pdc:CDIF630_00916', '272563.120.peg.820']\n",
      "['RL2_CLOD6', 'none', 'WP_003421168.1;YP_001086540.1', 'cdf:CD630_00760;pdc:CDIF630_00142', '272563.120.peg.82']\n",
      "['Q189Z4_CLOD6', 'none', 'WP_003437541.1;YP_001087273.1', 'cdf:CD630_07960;pdc:CDIF630_00915', '272563.120.peg.819']\n",
      "['Y795_CLOD6', 'none', 'WP_003437537.1;YP_001087272.1', 'cdf:CD630_07950;pdc:CDIF630_00914', '272563.120.peg.818']\n",
      "['Q189Z1_CLOD6', 'none', 'WP_003437532.1;YP_001087270.1', 'cdf:CD630_07930;pdc:CDIF630_00912', '272563.120.peg.816']\n",
      "['Q189Y4_CLOD6', 'none', 'WP_003437529.1;YP_001087269.1', 'cdf:CD630_07920;pdc:CDIF630_00911', '272563.120.peg.815']\n",
      "['Q189Y5_CLOD6', 'none', 'WP_009895825.1;YP_001087268.1', 'cdf:CD630_07910;pdc:CDIF630_00910', '272563.120.peg.814']\n",
      "['Q189Y6_CLOD6', 'none', 'WP_003437525.1;YP_001087267.1', 'cdf:CD630_07900;pdc:CDIF630_00909', '272563.120.peg.813']\n",
      "['Q189Y7_CLOD6', 'none', 'WP_003437523.1;YP_001087266.2', 'cdf:CD630_07890;pdc:CDIF630_00908', '272563.120.peg.812']\n",
      "['Q189Y8_CLOD6', 'none', 'WP_009895824.1;YP_001087265.1', 'cdf:CD630_07880;pdc:CDIF630_00907', '272563.120.peg.811']\n",
      "['Q189Y2_CLOD6', 'none', 'WP_003437519.1;YP_001087264.1', 'cdf:CD630_07870;pdc:CDIF630_00906', '272563.120.peg.810']\n",
      "['RL23_CLOD6', 'none', 'WP_003435681.1;YP_001086539.1', 'cdf:CD630_00750;pdc:CDIF630_00141', '272563.120.peg.81']\n",
      "['Q189Y3_CLOD6', 'none', 'WP_003437518.1;YP_001087263.1', 'cdf:CD630_07860;pdc:CDIF630_00905', '272563.120.peg.809']\n",
      "['Q189X8_CLOD6', 'none', 'WP_003437515.1;YP_001087260.1', 'cdf:CD630_07830;pdc:CDIF630_00901', '272563.120.peg.805']\n",
      "['Q189X9_CLOD6', 'none', 'WP_003418439.1;YP_001087259.1', 'cdf:CD630_07820;pdc:CDIF630_00900', '272563.120.peg.804']\n",
      "['Q189X7_CLOD6', 'none', 'WP_011860956.1;YP_001087258.1', 'cdf:CD630_07810;pdc:CDIF630_00899', '272563.120.peg.803']\n",
      "['Q189X3_CLOD6', 'none', 'WP_003437502.1;YP_001087257.1', 'cdf:CD630_07800;pdc:CDIF630_00898', '272563.120.peg.802']\n",
      "['Q189X4_CLOD6', 'none', 'WP_011860955.1;YP_001087256.1', 'cdf:CD630_07790;pdc:CDIF630_00897', '272563.120.peg.801']\n",
      "['Q18C95_CLOD6', 'none', 'WP_003420469.1;YP_001086471.1', 'cdf:CD630_00080;pdc:CDIF630_00008', '272563.120.peg.8']\n",
      "['Q189X6_CLOD6', 'none', 'WP_003437481.1;YP_001087254.1', 'cdf:CD630_07770;pdc:CDIF630_00895', '272563.120.peg.799']\n",
      "['Q189W9_CLOD6', 'none', 'WP_003437479.1;YP_001087253.1', 'cdf:CD630_07760;pdc:CDIF630_00894', '272563.120.peg.798']\n",
      "['Q189X0_CLOD6', 'none', 'WP_003418423.1;YP_001087252.1', 'cdf:CD630_07750;pdc:CDIF630_00893', '272563.120.peg.797']\n",
      "['Q189X1_CLOD6', 'none', 'WP_003437476.1;YP_001087251.1', 'cdf:CD630_07740;pdc:CDIF630_00892', '272563.120.peg.796']\n",
      "['Q189X2_CLOD6', 'none', 'WP_003418420.1;YP_001087250.1', 'cdf:CD630_07730;pdc:CDIF630_00891', '272563.120.peg.795']\n",
      "['Q189W8_CLOD6', 'none', 'WP_011860953.1;YP_001087249.1', 'cdf:CD630_07720;pdc:CDIF630_00890', '272563.120.peg.794']\n",
      "['Q189W5_CLOD6', 'none', 'WP_003418414.1;YP_001087247.1', 'cdf:CD630_07700;pdc:CDIF630_00888', '272563.120.peg.792']\n",
      "['Q189W6_CLOD6', 'none', 'WP_011860952.1;YP_001087246.1', 'cdf:CD630_07690;pdc:CDIF630_00887', '272563.120.peg.791']\n",
      "['RL3_CLOD6', 'none', 'WP_003421173.1;YP_001086537.1', 'cdf:CD630_00730;pdc:CDIF630_00139', '272563.120.peg.79']\n",
      "['Q189V9_CLOD6', 'none', 'WP_003418408.1;YP_001087241.1', 'cdf:CD630_07640;pdc:CDIF630_00882', '272563.120.peg.786']\n",
      "['Q189W0_CLOD6', 'none', 'WP_003437466.1;YP_001087240.1', 'cdf:CD630_07630;pdc:CDIF630_00881', '272563.120.peg.785']\n",
      "['Q189V8_CLOD6', 'none', 'WP_011860951.1;YP_001087239.1', 'cdf:CD630_07620;pdc:CDIF630_00880', '272563.120.peg.784']\n",
      "['Q189V7_CLOD6', 'none', 'WP_003437464.1;YP_001087238.1', 'cdf:CD630_07610;pdc:CDIF630_00879', '272563.120.peg.783']\n",
      "['Q189V4_CLOD6', 'none', 'WP_009902145.1;YP_001087237.1', 'cdf:CD630_07600;pdc:CDIF630_00878', '272563.120.peg.782']\n",
      "['RS10_CLOD6', 'none', 'WP_011860633.1;YP_001086536.1', 'cdf:CD630_00720;pdc:CDIF630_00138', '272563.120.peg.78']\n",
      "['Q189V2_CLOD6', 'none', 'WP_003437445.1;YP_001087233.1', 'cdf:CD630_07560;pdc:CDIF630_00874', '272563.120.peg.778']\n",
      "['Q189V3_CLOD6', 'none', 'WP_011860949.1;YP_001087232.1', 'cdf:CD630_07550;pdc:CDIF630_00873', '272563.120.peg.777']\n",
      "['Q189U6_CLOD6', 'none', 'WP_009888558.1;YP_001087228.1', 'cdf:CD630_07510;pdc:CDIF630_00869', '272563.120.peg.773']\n",
      "['Q189U7_CLOD6', 'none', 'WP_003432207.1;YP_001087227.1', 'cdf:CD630_07500;pdc:CDIF630_00868', '272563.120.peg.772']\n",
      "['Q189U5_CLOD6', 'none', 'WP_011860945.1;YP_001087224.1', 'cdf:CD630_07470;pdc:CDIF630_00864', '272563.120.peg.768']\n",
      "['Q189U0_CLOD6', 'none', 'WP_011860944.1;YP_001087223.1', 'cdf:CD630_07460;pdc:CDIF630_00863', '272563.120.peg.767']\n",
      "['Q189U2_CLOD6', 'none', 'WP_011860942.1;YP_001087221.1', 'cdf:CD630_07440;pdc:CDIF630_00861', '272563.120.peg.765']\n",
      "['Q189T8_CLOD6', 'none', 'WP_003418385.1;YP_001087220.1', 'cdf:CD630_07430;pdc:CDIF630_00859', '272563.120.peg.764']\n",
      "['Q189T9_CLOD6', 'none', 'WP_011860941.1;YP_001087219.1', 'cdf:CD630_07420;pdc:CDIF630_00858', '272563.120.peg.763']\n",
      "['Q189T4_CLOD6', 'none', 'WP_011860940.1;YP_001087217.1', 'cdf:CD630_07400;pdc:CDIF630_00856', '272563.120.peg.761']\n",
      "['Q189T5_CLOD6', 'none', 'WP_011860939.1;YP_001087216.1', 'cdf:CD630_07390;pdc:CDIF630_00855', '272563.120.peg.760']\n",
      "['EFG_CLOD6', 'none', 'WP_003436178.1;YP_001086534.1', 'cdf:CD630_00700;pdc:CDIF630_00136', '272563.120.peg.76']\n",
      "['Q189T6_CLOD6', 'none', 'WP_011860938.1;YP_001087215.1', 'cdf:CD630_07380;pdc:CDIF630_00854', '272563.120.peg.759']\n",
      "['Q189S8_CLOD6', 'none', 'WP_003432237.1;YP_001087213.1', 'cdf:CD630_07360;pdc:CDIF630_00852', '272563.120.peg.757']\n",
      "['Q189S9_CLOD6', 'none', 'WP_003418377.1;YP_001087212.1', 'cdf:CD630_07350;pdc:CDIF630_00851', '272563.120.peg.756']\n",
      "['Q189T0_CLOD6', 'none', 'WP_003418375.1;YP_001087211.1', 'cdf:CD630_07340;pdc:CDIF630_00850', '272563.120.peg.755']\n",
      "['Q189T2_CLOD6', 'none', 'WP_011860937.1;YP_001087209.1', 'cdf:CD630_07320;pdc:CDIF630_00848', '272563.120.peg.753']\n",
      "['Q189S7_CLOD6', 'none', 'WP_003436928.1;YP_001087208.1', 'cdf:CD630_07310;pdc:CDIF630_00847', '272563.120.peg.752']\n",
      "['Q189S3_CLOD6', 'none', 'WP_003436926.1;YP_001087207.1', 'cdf:CD630_07300;pdc:CDIF630_00846', '272563.120.peg.751']\n",
      "['GCSH_CLOD6', 'none', 'WP_003418363.1;YP_001087206.1', 'cdf:CD630_07290;pdc:CDIF630_00845', '272563.120.peg.750']\n",
      "['RS7_CLOD6', 'none', 'WP_003421180.1;YP_001086533.1', 'cdf:CD630_00690;pdc:CDIF630_00135', '272563.120.peg.75']\n",
      "['Q189S5_CLOD6', 'none', 'WP_003418359.1;YP_001087205.1', 'cdf:CD630_07280;pdc:CDIF630_00844', '272563.120.peg.749']\n",
      "['Q189S6_CLOD6', 'none', 'WP_003418356.1;YP_001087204.1', 'cdf:CD630_07270;pdc:CDIF630_00843', '272563.120.peg.748']\n",
      "['Q189S0_CLOD6', 'none', 'WP_003436924.1;YP_001087203.1', 'cdf:CD630_07260;pdc:CDIF630_00842', '272563.120.peg.747']\n",
      "['Q189S1_CLOD6', 'none', 'WP_003432254.1;YP_001087202.1', 'cdf:CD630_07250;pdc:CDIF630_00841', '272563.120.peg.746']\n",
      "['Q189S2_CLOD6', 'none', 'WP_003418348.1;YP_001087201.1', 'cdf:CD630_07240;pdc:CDIF630_00840', '272563.120.peg.745']\n",
      "['Q189R7_CLOD6', 'none', 'WP_003436912.1;YP_001087198.1', 'cdf:CD630_07210;pdc:CDIF630_00837', '272563.120.peg.742']\n",
      "['RS12_CLOD6', 'none', 'WP_011860632.1;YP_001086532.2', 'cdf:CD630_00680;pdc:CDIF630_00134', '272563.120.peg.74']\n",
      "['Q189R3_CLOD6', 'none', 'WP_009888532.1;YP_001087194.1', 'cdf:CD630_07170;pdc:CDIF630_00833', '272563.120.peg.738']\n",
      "['Q189R1_CLOD6', 'none', 'WP_011860934.1;YP_001087190.1', 'cdf:CD630_07140;pdc:CDIF630_00830', '272563.120.peg.735']\n",
      "['Q189Q5_CLOD6', 'none', 'WP_011860933.1;YP_001087189.1', 'cdf:CD630_07130;pdc:CDIF630_00829', '272563.120.peg.734']\n",
      "['F3Y5Y4_CLOD6', 'none', 'WP_003418305.1;YP_006589358.1', 'cdf:CD630_07121;pdc:CDIF630_00828', '272563.120.peg.733']\n",
      "['Q189Q6_CLOD6', 'none', 'WP_011860932.1;YP_001087188.1', 'cdf:CD630_07120;pdc:CDIF630_00827', '272563.120.peg.732']\n",
      "['Q189Q8_CLOD6', 'none', 'WP_011860931.1;YP_001087186.1', 'cdf:CD630_07100;pdc:CDIF630_00825', '272563.120.peg.730']\n",
      "['Q189Q1_CLOD6', 'none', 'WP_011860929.1;YP_001087183.2', 'cdf:CD630_07070;pdc:CDIF630_00822', '272563.120.peg.727']\n",
      "['Q189Q2_CLOD6', 'none', 'WP_011860928.1;YP_001087182.1', 'cdf:CD630_07060;pdc:CDIF630_00821', '272563.120.peg.726']\n",
      "['Q189Q0_CLOD6', 'none', 'WP_003436879.1;YP_001087181.1', 'cdf:CD630_07050;pdc:CDIF630_00820', '272563.120.peg.725']\n",
      "['Q189P8_CLOD6', 'none', 'WP_011860927.1;YP_001087180.1', 'cdf:CD630_07040;pdc:CDIF630_00819', '272563.120.peg.724']\n",
      "['Q189P9_CLOD6', 'none', 'WP_011860926.1;YP_001087179.1', 'cdf:CD630_07030;pdc:CDIF630_00818', '272563.120.peg.723']\n",
      "['Q189P7_CLOD6', 'none', 'WP_003436876.1;YP_001087178.1', 'cdf:CD630_07020;pdc:CDIF630_00817', '272563.120.peg.722']\n",
      "['Q189P5_CLOD6', 'none', 'WP_009895750.1;YP_001087177.1', 'cdf:CD630_07010;pdc:CDIF630_00816', '272563.120.peg.721']\n",
      "['Q189P2_CLOD6', 'none', 'WP_003418257.1;YP_001087173.1', 'cdf:CD630_06970;pdc:CDIF630_00811', '272563.120.peg.717']\n",
      "['Q189P3_CLOD6', 'none', 'WP_009888508.1;YP_001087172.1', 'cdf:CD630_06960;pdc:CDIF630_00810', '272563.120.peg.716']\n",
      "['Q189N8_CLOD6', 'none', 'WP_011860923.1;YP_001087171.1', 'cdf:CD630_06950;pdc:CDIF630_00809', '272563.120.peg.715']\n",
      "['F3Y5Y3_CLOD6', 'none', 'WP_011860922.1;YP_006589357.1', 'cdf:CD630_06941;pdc:CDIF630_00808', '272563.120.peg.714']\n",
      "['Q189N9_CLOD6', 'none', 'WP_011860921.1;YP_001087170.1', 'cdf:CD630_06940;pdc:CDIF630_00807', '272563.120.peg.713']\n",
      "['Q189P0_CLOD6', 'none', 'WP_011860920.1;YP_001087169.1', 'cdf:CD630_06930;pdc:CDIF630_00806', '272563.120.peg.712']\n",
      "['Q189N7_CLOD6', 'none', 'WP_011860919.1;YP_001087168.1', 'cdf:CD630_06920;pdc:CDIF630_00805', '272563.120.peg.711']\n",
      "['Q189N5_CLOD6', 'none', 'WP_003436810.1;YP_001087167.1', 'cdf:CD630_06910;pdc:CDIF630_00804', '272563.120.peg.710']\n",
      "['Q189N6_CLOD6', 'none', 'WP_011860918.1;YP_001087166.1', 'cdf:CD630_06900;pdc:CDIF630_00803', '272563.120.peg.709']\n",
      "['Q189N2_CLOD6', 'none', 'WP_009902098.1;YP_001087165.1', 'cdf:CD630_06890;pdc:CDIF630_00802', '272563.120.peg.708']\n",
      "['RL20_CLOD6', 'none', 'WP_003429730.1;YP_001087163.1', 'cdf:CD630_06870;pdc:CDIF630_00800', '272563.120.peg.706']\n",
      "['RL35_CLOD6', 'none', 'WP_003429729.1;YP_001087162.1', 'cdf:CD630_06860;pdc:CDIF630_00799', '272563.120.peg.705']\n",
      "['Q189M9_CLOD6', 'none', 'WP_003429728.1;YP_001087159.1', 'cdf:CD630_06830;pdc:CDIF630_00796', '272563.120.peg.701']\n",
      "['RL7_CLOD6', 'none', 'WP_003421186.1;YP_001086528.1', 'cdf:CD630_00640;pdc:CDIF630_00130', '272563.120.peg.70']\n",
      "['Q18C91_CLOD6', 'none', 'WP_003420471.1;YP_001086470.1', 'cdf:CD630_00070;pdc:CDIF630_00007', '272563.120.peg.7']\n",
      "['Q189M4_CLOD6', 'none', 'WP_009895721.1;YP_001087157.1', 'cdf:CD630_06810;pdc:CDIF630_00794', '272563.120.peg.699']\n",
      "['Q189M5_CLOD6', 'none', 'WP_011860914.1;YP_001087156.1', 'cdf:CD630_06800;pdc:CDIF630_00793', '272563.120.peg.698']\n",
      "['Q189M6_CLOD6', 'none', 'WP_009902083.1;YP_001087155.1', 'cdf:CD630_06790;pdc:CDIF630_00792', '272563.120.peg.697']\n",
      "['Q189L9_CLOD6', 'none', 'WP_003439159.1;YP_001087153.1', 'cdf:CD630_06770;pdc:CDIF630_00791', '272563.120.peg.695']\n",
      "['Q189M0_CLOD6', 'none', 'WP_011860913.1;YP_001087152.1', 'cdf:CD630_06760;pdc:CDIF630_00790', '272563.120.peg.694']\n",
      "['Q189L5_CLOD6', 'none', 'WP_003439170.1;YP_001087149.1', 'cdf:CD630_06730;pdc:CDIF630_00788', '272563.120.peg.692']\n",
      "['Q189L7_CLOD6', 'none', 'WP_009888463.1;YP_001087147.1', 'cdf:CD630_06710;pdc:CDIF630_00786', '272563.120.peg.690']\n",
      "['RL10_CLOD6', 'none', 'WP_003421187.1;YP_001086527.1', 'cdf:CD630_00630;pdc:CDIF630_00129', '272563.120.peg.69']\n",
      "['Q189L8_CLOD6', 'none', 'WP_003426867.1;YP_001087146.1', 'cdf:CD630_06700;pdc:CDIF630_00785', '272563.120.peg.689']\n",
      "['Q189L3_CLOD6', 'none', 'WP_011860909.1;YP_001087145.1', 'cdf:CD630_06690;pdc:CDIF630_00784', '272563.120.peg.688']\n",
      "['Q189K9_CLOD6', 'none', 'WP_011860907.1;YP_001087142.1', 'cdf:CD630_06660;pdc:CDIF630_00781', '272563.120.peg.685']\n",
      "['Q189L0_CLOD6', 'none', 'WP_011860906.1;YP_001087141.1', 'cdf:CD630_06650;pdc:CDIF630_00780', '272563.120.peg.684']\n",
      "['Q189L1_CLOD6', 'none', 'WP_003426848.1;YP_001087140.1', 'cdf:CD630_06642;pdc:CDIF630_00779', '272563.120.peg.683']\n",
      "['Q189L2_CLOD6', 'none', 'WP_009888450.1;YP_001087139.1', 'cdf:CD630_06641;pdc:CDIF630_00778', '272563.120.peg.682']\n",
      "['Q189K7_CLOD6', 'none', 'WP_011860905.1;YP_001087138.1', 'cdf:CD630_06640;pdc:CDIF630_00777', '272563.120.peg.681']\n",
      "['RL1_CLOD6', 'none', 'WP_003421188.1;YP_001086526.1', 'cdf:CD630_00620;pdc:CDIF630_00127', '272563.120.peg.68']\n",
      "['Q189K6_CLOD6', 'none', 'WP_003434540.1;YP_001087136.1', 'cdf:CD630_06610;pdc:CDIF630_00774', '272563.120.peg.678']\n",
      "['Q189K4_CLOD6', 'none', 'WP_003434528.1;YP_001087134.1', 'cdf:CD630_06590;pdc:CDIF630_00772', '272563.120.peg.676']\n",
      "['Q189K2_CLOD6', 'none', 'WP_003434522.1;YP_001087133.1', 'cdf:CD630_06580;pdc:CDIF630_00771', '272563.120.peg.675']\n",
      "['Q189K0_CLOD6', 'none', 'WP_011860903.1;YP_001087132.1', 'cdf:CD630_06570;pdc:CDIF630_00770', '272563.120.peg.674']\n",
      "['Q189J6_CLOD6', 'none', 'WP_011860901.1;YP_001087130.1', 'cdf:CD630_06550;pdc:CDIF630_00768', '272563.120.peg.672']\n",
      "['Q189J7_CLOD6', 'none', 'WP_003440185.1;YP_001087129.1', 'cdf:CD630_06540;pdc:CDIF630_00766', '272563.120.peg.670']\n",
      "['RL11_CLOD6', 'none', 'WP_003421189.1;YP_001086525.1', 'cdf:CD630_00610;pdc:CDIF630_00126', '272563.120.peg.67']\n",
      "['Q189J9_CLOD6', 'none', 'WP_003440181.1;YP_001087127.1', 'cdf:CD630_06520;pdc:CDIF630_00764', '272563.120.peg.668']\n",
      "['Q189J1_CLOD6', 'none', 'WP_011860897.1;YP_001087126.1', 'cdf:CD630_06510;pdc:CDIF630_00763', '272563.120.peg.666']\n",
      "['Q189J2_CLOD6', 'none', 'WP_011860896.1;YP_001087125.1', 'cdf:CD630_06500;pdc:CDIF630_00762', '272563.120.peg.665']\n",
      "['Q189I9_CLOD6', 'none', 'WP_011860894.1;YP_001087121.1', 'cdf:CD630_06460;pdc:CDIF630_00759', '272563.120.peg.661']\n",
      "['Q189J0_CLOD6', 'none', 'WP_011860893.1;YP_001087120.1', 'cdf:CD630_06450;pdc:CDIF630_00758', '272563.120.peg.660']\n",
      "['Q18CE6_CLOD6', 'none', 'WP_003421190.1;YP_001086524.1', 'cdf:CD630_00600;pdc:CDIF630_00125', '272563.120.peg.66']\n",
      "['Q189I7_CLOD6', 'none', 'WP_009895668.1;YP_001087119.1', 'cdf:CD630_06440;pdc:CDIF630_00757', '272563.120.peg.659']\n",
      "['Q189I8_CLOD6', 'none', 'WP_009895666.1;YP_001087118.1', 'cdf:CD630_06430;pdc:CDIF630_00756', '272563.120.peg.658']\n",
      "['F3Y5X9_CLOD6', 'none', 'pdc:CDIF630_00755', '272563.120.peg.656']\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-153-38696f6b008e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     10\u001b[0m         \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m         \u001b[0muniprot_gene_ids\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m         \u001b[0mec_gene_ids\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mrefseq_gene_ids\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "uniprot_gene_ids = {}\n",
    "ec_gene_ids = {}\n",
    "refseq_gene_ids = {}\n",
    "kegg_gene_ids = {}\n",
    "patric_genes = set()\n",
    "\n",
    "with open('/home/mjenior/Desktop/repos/uniprot_Cd630.tab', 'r') as gene_annotation:\n",
    "    header = gene_annotation.readline()\n",
    "    for line in gene_annotation:\n",
    "        line = line.split()\n",
    "        print(line)\n",
    "        uniprot_gene_ids[line[4]] = line[0]\n",
    "        ec_gene_ids[line[4]] = line[1]\n",
    "        refseq_gene_ids[line[4]] = line[2]\n",
    "        kegg_gene_ids[line[4]] = line[3]\n",
    "        patric_genes |= set([line[4]])\n",
    "        \n",
    "for gene in iCdG707.genes:\n",
    "    if not gene.id in patric_genes: continue\n",
    "        \n",
    "    gene.annotation['uniprot'] = uniprot_gene_ids[gene.id]\n",
    "    gene.annotation['kegg.genes'] = kegg_gene_ids[gene.id]\n",
    "    \n",
    "    if refseq_gene_ids[gene.id] != 'none':\n",
    "        gene.annotation['refseq'] = refseq_gene_ids[gene.id]\n",
    "    \n",
    "    if ec_gene_ids[gene.id] != 'none':\n",
    "        gene.annotation['ec_number'] = ec_gene_ids[gene.id]\n",
    "    \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save model\n",
    "cobra.io.write_sbml_model(iCdG707, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG707.sbml')\n",
    "cobra.io.save_json_model(iCdG707, '/home/mjenior/Desktop/repos/Jenior_Cdifficile_2019/data/reconstructions/iCdG707.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'272563.8.peg.3384'"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iCdG707.genes[2].id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
